diff --git a/internal/e2e/otlphttp_test.go b/internal/e2e/otlphttp_test.go
index d9de99807..cfcba3158 100644
--- a/internal/e2e/otlphttp_test.go
+++ b/internal/e2e/otlphttp_test.go
@@ -117,7 +117,7 @@ func TestTraceRoundTrip(t *testing.T) {
 			}, 1*time.Second, 10*time.Millisecond)
 			allTraces := sink.AllTraces()
 			require.Len(t, allTraces, 1)
-			assert.EqualValues(t, td, allTraces[0])
+			assert.Equal(t, td, allTraces[0])
 		})
 	}
 }
@@ -170,7 +170,7 @@ func TestMetricsRoundTrip(t *testing.T) {
 			}, 1*time.Second, 10*time.Millisecond)
 			allMetrics := sink.AllMetrics()
 			require.Len(t, allMetrics, 1)
-			assert.EqualValues(t, md, allMetrics[0])
+			assert.Equal(t, md, allMetrics[0])
 		})
 	}
 }
@@ -223,7 +223,7 @@ func TestLogsRoundTrip(t *testing.T) {
 			}, 1*time.Second, 10*time.Millisecond)
 			allLogs := sink.AllLogs()
 			require.Len(t, allLogs, 1)
-			assert.EqualValues(t, md, allLogs[0])
+			assert.Equal(t, md, allLogs[0])
 		})
 	}
 }
