diff --git a/receiver/go.mod b/receiver/go.mod
index 82689654e..dbe8afbc5 100644
--- a/receiver/go.mod
+++ b/receiver/go.mod
@@ -10,6 +10,7 @@ require (
 	go.opentelemetry.io/collector/consumer v0.114.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.114.0
 	go.opentelemetry.io/collector/pdata v1.20.0
+	go.opentelemetry.io/collector/pdata/testdata v0.114.0
 	go.opentelemetry.io/collector/pipeline v0.114.0
 	go.opentelemetry.io/collector/receiver/receivertest v0.114.0
 	go.opentelemetry.io/collector/scraper v0.114.0
