diff --git a/exporter/exporterhelper/resource_to_label.go b/exporter/exporterhelper/resource_to_label.go
index c7e1f02fa..6ce056ad2 100644
--- a/exporter/exporterhelper/resource_to_label.go
+++ b/exporter/exporterhelper/resource_to_label.go
@@ -60,9 +60,10 @@ func extractLabelsFromResource(resource *pdata.Resource) pdata.StringMap {
 	labelMap := pdata.NewStringMap()
 
 	attrMap := resource.Attributes()
-	attrMap.ForEach(func(k string, av pdata.AttributeValue) {
+	attrMap.Range(func(k string, av pdata.AttributeValue) bool {
 		stringLabel := tracetranslator.AttributeValueToString(av, false)
 		labelMap.Upsert(k, stringLabel)
+		return true
 	})
 	return labelMap
 }
@@ -110,7 +111,8 @@ func addLabelsToDoubleHistogramDataPoints(ps pdata.HistogramDataPointSlice, newL
 }
 
 func joinStringMaps(from, to pdata.StringMap) {
-	from.ForEach(func(k, v string) {
+	from.Range(func(k, v string) bool {
 		to.Upsert(k, v)
+		return true
 	})
 }
