diff --git a/component/componenttest/nop_exporter_test.go b/component/componenttest/nop_exporter_test.go
index 8a1863c99..2d404184d 100644
--- a/component/componenttest/nop_exporter_test.go
+++ b/component/componenttest/nop_exporter_test.go
@@ -33,19 +33,19 @@ func TestNewNopExporterFactory(t *testing.T) {
 	cfg := factory.CreateDefaultConfig()
 	assert.Equal(t, &nopExporterConfig{ExporterSettings: config.NewExporterSettings(config.NewID("nop"))}, cfg)
 
-	traces, err := factory.CreateTracesExporter(context.Background(), component.ExporterCreateParams{}, cfg)
+	traces, err := factory.CreateTracesExporter(context.Background(), component.ExporterCreateSettings{}, cfg)
 	require.NoError(t, err)
 	assert.NoError(t, traces.Start(context.Background(), NewNopHost()))
 	assert.NoError(t, traces.ConsumeTraces(context.Background(), pdata.NewTraces()))
 	assert.NoError(t, traces.Shutdown(context.Background()))
 
-	metrics, err := factory.CreateMetricsExporter(context.Background(), component.ExporterCreateParams{}, cfg)
+	metrics, err := factory.CreateMetricsExporter(context.Background(), component.ExporterCreateSettings{}, cfg)
 	require.NoError(t, err)
 	assert.NoError(t, metrics.Start(context.Background(), NewNopHost()))
 	assert.NoError(t, metrics.ConsumeMetrics(context.Background(), pdata.NewMetrics()))
 	assert.NoError(t, metrics.Shutdown(context.Background()))
 
-	logs, err := factory.CreateLogsExporter(context.Background(), component.ExporterCreateParams{}, cfg)
+	logs, err := factory.CreateLogsExporter(context.Background(), component.ExporterCreateSettings{}, cfg)
 	require.NoError(t, err)
 	assert.NoError(t, logs.Start(context.Background(), NewNopHost()))
 	assert.NoError(t, logs.ConsumeLogs(context.Background(), pdata.NewLogs()))
