diff --git a/exporter/exporterhelper/traces.go b/exporter/exporterhelper/traces.go
index 7d7bedbd2..ee05a547d 100644
--- a/exporter/exporterhelper/traces.go
+++ b/exporter/exporterhelper/traces.go
@@ -90,9 +90,6 @@ func NewTraces(
 	return NewTracesRequest(ctx, set, requestFromTraces(pusher), append(tracesOpts, options...)...)
 }
 
-// Deprecated: [v0.112.0] use NewTraces.
-var NewTracesExporter = NewTraces
-
 // RequestFromTracesFunc converts ptrace.Traces into a user-defined Request.
 // Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
@@ -148,9 +145,6 @@ func NewTracesRequest(
 	}, err
 }
 
-// Deprecated: [v0.112.0] use NewTracesRequest.
-var NewTracesRequestExporter = NewTracesRequest
-
 type tracesWithObservability struct {
 	internal.BaseRequestSender
 	obsrep *internal.ObsReport
