diff --git a/pdata/pprofile/pprofileotlp/generated_exportresponse.go b/pdata/pprofile/pprofileotlp/generated_exportresponse.go
index 721b78feb..6236b2ece 100644
--- a/pdata/pprofile/pprofileotlp/generated_exportresponse.go
+++ b/pdata/pprofile/pprofileotlp/generated_exportresponse.go
@@ -32,7 +32,7 @@ func newExportResponse(orig *otlpcollectorprofiles.ExportProfilesServiceResponse
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewExportResponse() ExportResponse {
-	return newExportResponse(internal.NewOrigPtrExportProfilesServiceResponse(), internal.NewState())
+	return newExportResponse(internal.NewOrigExportProfilesServiceResponse(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
