diff --git a/service/internal/graph/graph.go b/service/internal/graph/graph.go
index 754d6259e..b9f1f2b1d 100644
--- a/service/internal/graph/graph.go
+++ b/service/internal/graph/graph.go
@@ -28,7 +28,6 @@ import (
 	"go.opentelemetry.io/collector/component/componentstatus"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/consumer"
-	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/internal/fanoutconsumer"
 	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/service/internal/builders"
@@ -44,7 +43,7 @@ type Settings struct {
 
 	ReceiverBuilder  builders.Receiver
 	ProcessorBuilder *processor.Builder
-	ExporterBuilder  *exporter.Builder
+	ExporterBuilder  builders.Exporter
 	ConnectorBuilder builders.Connector
 
 	// PipelineConfigs is a map of component.ID to PipelineConfig.
