diff --git a/processor/processorhelper/traces_test.go b/processor/processorhelper/traces_test.go
index ff5e3bd8d..54adcfcab 100644
--- a/processor/processorhelper/traces_test.go
+++ b/processor/processorhelper/traces_test.go
@@ -33,7 +33,7 @@ import (
 var testTracesCfg = config.NewProcessorSettings(config.NewComponentID("test"))
 
 func TestNewTracesProcessor(t *testing.T) {
-	tp, err := NewTracesProcessor(&testTracesCfg, consumertest.NewNop(), newTestTProcessor(nil))
+	tp, err := NewTracesProcessorWithCreateSettings(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, consumertest.NewNop(), newTestTProcessor(nil))
 	require.NoError(t, err)
 
 	assert.True(t, tp.Capabilities().MutatesData)
@@ -44,7 +44,7 @@ func TestNewTracesProcessor(t *testing.T) {
 
 func TestNewTracesProcessor_WithOptions(t *testing.T) {
 	want := errors.New("my_error")
-	tp, err := NewTracesProcessor(&testTracesCfg, consumertest.NewNop(), newTestTProcessor(nil),
+	tp, err := NewTracesProcessorWithCreateSettings(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, consumertest.NewNop(), newTestTProcessor(nil),
 		WithStart(func(context.Context, component.Host) error { return want }),
 		WithShutdown(func(context.Context) error { return want }),
 		WithCapabilities(consumer.Capabilities{MutatesData: false}))
@@ -56,22 +56,22 @@ func TestNewTracesProcessor_WithOptions(t *testing.T) {
 }
 
 func TestNewTracesProcessor_NilRequiredFields(t *testing.T) {
-	_, err := NewTracesProcessor(&testTracesCfg, consumertest.NewNop(), nil)
+	_, err := NewTracesProcessorWithCreateSettings(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, consumertest.NewNop(), nil)
 	assert.Error(t, err)
 
-	_, err = NewTracesProcessor(&testTracesCfg, nil, newTestTProcessor(nil))
+	_, err = NewTracesProcessorWithCreateSettings(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, nil, newTestTProcessor(nil))
 	assert.Equal(t, component.ErrNilNextConsumer, err)
 }
 
 func TestNewTracesProcessor_ProcessTraceError(t *testing.T) {
 	want := errors.New("my_error")
-	tp, err := NewTracesProcessor(&testTracesCfg, consumertest.NewNop(), newTestTProcessor(want))
+	tp, err := NewTracesProcessorWithCreateSettings(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, consumertest.NewNop(), newTestTProcessor(want))
 	require.NoError(t, err)
 	assert.Equal(t, want, tp.ConsumeTraces(context.Background(), ptrace.NewTraces()))
 }
 
 func TestNewTracesProcessor_ProcessTracesErrSkipProcessingData(t *testing.T) {
-	tp, err := NewTracesProcessor(&testTracesCfg, consumertest.NewNop(), newTestTProcessor(ErrSkipProcessingData))
+	tp, err := NewTracesProcessorWithCreateSettings(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, consumertest.NewNop(), newTestTProcessor(ErrSkipProcessingData))
 	require.NoError(t, err)
 	assert.Equal(t, nil, tp.ConsumeTraces(context.Background(), ptrace.NewTraces()))
 }
