diff --git a/service/pipelines/config.go b/service/pipelines/config.go
index d22e71fd6..ff898aafa 100644
--- a/service/pipelines/config.go
+++ b/service/pipelines/config.go
@@ -46,34 +46,6 @@ func (cfg Config) Validate() error {
 	return nil
 }
 
-// Deprecated: [v0.111.0] Use Config instead
-type ConfigWithPipelineID map[pipeline.ID]*PipelineConfig
-
-func (cfg ConfigWithPipelineID) Validate() error {
-	// Must have at least one pipeline.
-	if len(cfg) == 0 {
-		return errMissingServicePipelines
-	}
-
-	// Check that all pipelines have at least one receiver and one exporter, and they reference
-	// only configured components.
-	for pipelineID, p := range cfg {
-		switch pipelineID.Signal() {
-		case pipeline.SignalTraces, pipeline.SignalMetrics, pipeline.SignalLogs, componentprofiles.SignalProfiles:
-			// Continue
-		default:
-			return fmt.Errorf("pipeline %q: unknown signal %q", pipelineID.String(), pipelineID.Signal())
-		}
-
-		// Validate pipeline has at least one receiver.
-		if err := p.Validate(); err != nil {
-			return fmt.Errorf("pipeline %q: %w", pipelineID.String(), err)
-		}
-	}
-
-	return nil
-}
-
 // PipelineConfig defines the configuration of a Pipeline.
 type PipelineConfig struct {
 	Receivers  []component.ID `mapstructure:"receivers"`
