diff --git a/confmap/provider.go b/confmap/provider.go
index 47e95768d..c81036e6b 100644
--- a/confmap/provider.go
+++ b/confmap/provider.go
@@ -41,9 +41,13 @@ type Provider interface {
 	//
 	// `uri` must follow the "<scheme>:<opaque_data>" format. This format is compatible
 	// with the URI definition (see https://datatracker.ietf.org/doc/html/rfc3986). The "<scheme>"
-	// must be always included in the `uri`. The scheme supported by any provider MUST be at
-	// least 2 characters long to avoid conflicting with a driver-letter identifier as specified
-	// in https://tools.ietf.org/id/draft-kerwin-file-scheme-07.html#syntax.
+	// must be always included in the `uri`. The "<scheme>" supported by any provider:
+	//   - MUST consist of a sequence of characters beginning with a letter and followed by any
+	//     combination of letters, digits, plus ("+"), period ("."), or hyphen ("-").
+	//     See https://datatracker.ietf.org/doc/html/rfc3986#section-3.1.
+	//   - MUST be at least 2 characters long to avoid conflicting with a driver-letter identifier as specified
+	//     in https://tools.ietf.org/id/draft-kerwin-file-scheme-07.html#syntax.
+	//   - For testing, all implementation MUST check that confmaptest.ValidateProviderScheme returns no error.
 	//
 	// `watcher` callback is called when the config changes. watcher may be called from
 	// a different go routine. After watcher is called Retrieved.Get should be called
