diff --git a/internal/data/testdata/resource.go b/internal/data/testdata/resource.go
index 8eb2eb7dd..956bf5681 100644
--- a/internal/data/testdata/resource.go
+++ b/internal/data/testdata/resource.go
@@ -1,4 +1,4 @@
-// Copyright 2020 OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
