diff --git a/pdata/pcommon/alias.go b/pdata/pcommon/alias.go
index c3d9a84e0..23861ddda 100644
--- a/pdata/pcommon/alias.go
+++ b/pdata/pcommon/alias.go
@@ -75,12 +75,6 @@ var (
 
 	// NewValueBytes creates a new Value with the given ImmutableByteSlice value.
 	NewValueBytes = internal.NewValueBytes
-
-	// NewValueMBytes creates a new Value with the given []byte value.
-	// The caller must ensure the []byte passed in is not modified after the call is made, sharing the data
-	// across multiple attributes is forbidden.
-	// Deprecated: [0.54.0] Use NewValueBytes instead.
-	NewValueMBytes = internal.NewValueMBytes
 )
 
 // Map stores a map of string keys to elements of Value type.
