diff --git a/service/parserprovider/file.go b/service/parserprovider/file.go
index 9c0ea40bd..3aabdfcc7 100644
--- a/service/parserprovider/file.go
+++ b/service/parserprovider/file.go
@@ -29,7 +29,7 @@ func NewFile() ParserProvider {
 	return &fileProvider{}
 }
 
-func (fl *fileProvider) Get() (*configparser.Parser, error) {
+func (fl *fileProvider) Get() (*configparser.ConfigMap, error) {
 	fileName := getConfigFlag()
 	if fileName == "" {
 		return nil, errors.New("config file not specified")
