diff --git a/service/parserprovider/inmemory.go b/service/parserprovider/inmemory.go
index 80dd4120b..d73c28bdb 100644
--- a/service/parserprovider/inmemory.go
+++ b/service/parserprovider/inmemory.go
@@ -29,6 +29,6 @@ func NewInMemory(buf io.Reader) ParserProvider {
 	return &inMemoryProvider{buf: buf}
 }
 
-func (inp *inMemoryProvider) Get() (*configparser.Parser, error) {
+func (inp *inMemoryProvider) Get() (*configparser.ConfigMap, error) {
 	return configparser.NewParserFromBuffer(inp.buf)
 }
