diff --git a/service/parserprovider/setflag_test.go b/service/parserprovider/setflag_test.go
index be9e07490..0f2923749 100644
--- a/service/parserprovider/setflag_test.go
+++ b/service/parserprovider/setflag_test.go
@@ -59,6 +59,6 @@ func TestSetFlags_empty(t *testing.T) {
 
 type emptyProvider struct{}
 
-func (el *emptyProvider) Get() (*configparser.Parser, error) {
+func (el *emptyProvider) Get() (*configparser.ConfigMap, error) {
 	return configparser.NewParser(), nil
 }
