diff --git a/cmd/builder/internal/builder/config_test.go b/cmd/builder/internal/builder/config_test.go
index 14ca9bbe5..116ae9f10 100644
--- a/cmd/builder/internal/builder/config_test.go
+++ b/cmd/builder/internal/builder/config_test.go
@@ -70,6 +70,9 @@ func TestAliases(t *testing.T) {
 				GoMod: "github.com/another3/module v0.1.2",
 			},
 		},
+		Telemetry: Module{
+			GoMod: "github.com/another3/module v0.1.2",
+		},
 	}
 
 	// test
@@ -128,6 +131,10 @@ func TestAliases(t *testing.T) {
 	assert.Equal(t, "github.com/another3/module v0.1.2", cfg.Connectors[2].GoMod)
 	assert.Equal(t, "github.com/another3/module", cfg.Connectors[2].Import)
 	assert.Equal(t, "module5", cfg.Connectors[2].Name)
+
+	assert.Equal(t, "github.com/another3/module v0.1.2", cfg.Telemetry.GoMod)
+	assert.Equal(t, "github.com/another3/module", cfg.Telemetry.Import)
+	assert.Equal(t, "module6", cfg.Telemetry.Name)
 }
 
 func TestParseModules(t *testing.T) {
@@ -275,6 +282,15 @@ func TestMissingModule(t *testing.T) {
 			},
 			err: errMissingGoMod,
 		},
+		{
+			cfg: Config{
+				Logger: zap.NewNop(),
+				Telemetry: Module{
+					Import: "go.opentelemetry.io/collector/service/telemetry/otelconftelemetry",
+				},
+			},
+			err: errMissingGoMod,
+		},
 	}
 
 	for _, test := range configurations {
@@ -285,10 +301,11 @@ func TestMissingModule(t *testing.T) {
 func TestNewDefaultConfig(t *testing.T) {
 	cfg, err := NewDefaultConfig()
 	require.NoError(t, err)
-	require.NoError(t, cfg.ParseModules())
-	assert.NoError(t, cfg.Validate())
-	assert.NoError(t, cfg.SetGoPath())
+	assert.Empty(t, cfg.Telemetry.GoMod)
 	require.NoError(t, cfg.Validate())
+	require.NoError(t, cfg.SetGoPath())
+	require.NoError(t, cfg.ParseModules())
+	assert.NotEmpty(t, cfg.Telemetry.GoMod)
 	assert.False(t, cfg.Distribution.DebugCompilation)
 	assert.Empty(t, cfg.Distribution.BuildTags)
 	assert.False(t, cfg.LDSet)
@@ -305,9 +322,9 @@ func TestNewBuiltinConfig(t *testing.T) {
 	cfg := Config{Logger: zaptest.NewLogger(t)}
 
 	require.NoError(t, k.UnmarshalWithConf("", &cfg, koanf.UnmarshalConf{Tag: "mapstructure"}))
-	assert.NoError(t, cfg.ParseModules())
-	assert.NoError(t, cfg.Validate())
-	assert.NoError(t, cfg.SetGoPath())
+	require.NoError(t, cfg.Validate())
+	require.NoError(t, cfg.SetGoPath())
+	require.NoError(t, cfg.ParseModules())
 
 	// Unlike the config initialized in NewDefaultConfig(), we expect
 	// the builtin default to be practically useful, so there must be
