diff --git a/pdata/internal/generated_wrapper_scopelogs.go b/pdata/internal/generated_wrapper_scopelogs.go
index 1b2b763a1..c98a86112 100644
--- a/pdata/internal/generated_wrapper_scopelogs.go
+++ b/pdata/internal/generated_wrapper_scopelogs.go
@@ -15,11 +15,13 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
-var protoPoolScopeLogs = sync.Pool{
-	New: func() any {
-		return &otlplogs.ScopeLogs{}
-	},
-}
+var (
+	protoPoolScopeLogs = sync.Pool{
+		New: func() any {
+			return &otlplogs.ScopeLogs{}
+		},
+	}
+)
 
 func NewOrigScopeLogs() *otlplogs.ScopeLogs {
 	if !UseProtoPooling.IsEnabled() {
