diff --git a/pdata/internal/generated_wrapper_logrecord.go b/pdata/internal/generated_wrapper_logrecord.go
index f9c546bd5..a43fdb38b 100644
--- a/pdata/internal/generated_wrapper_logrecord.go
+++ b/pdata/internal/generated_wrapper_logrecord.go
@@ -18,11 +18,13 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
-var protoPoolLogRecord = sync.Pool{
-	New: func() any {
-		return &otlplogs.LogRecord{}
-	},
-}
+var (
+	protoPoolLogRecord = sync.Pool{
+		New: func() any {
+			return &otlplogs.LogRecord{}
+		},
+	}
+)
 
 func NewOrigLogRecord() *otlplogs.LogRecord {
 	if !UseProtoPooling.IsEnabled() {
