diff --git a/extension/memorylimiterextension/go.mod b/extension/memorylimiterextension/go.mod
index df7cf0f39..09baf6c12 100644
--- a/extension/memorylimiterextension/go.mod
+++ b/extension/memorylimiterextension/go.mod
@@ -81,3 +81,7 @@ replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/con
 replace go.opentelemetry.io/collector/pdata/testdata => ../../pdata/testdata
 
 replace go.opentelemetry.io/collector/pdata/pprofile => ../../pdata/pprofile
+
+replace go.opentelemetry.io/collector/consumer/consumerprofiles => ../../consumer/consumerprofiles
+
+replace go.opentelemetry.io/collector/consumer/consumertest => ../../consumer/consumertest
