diff --git a/pdata/pprofile/generated_mapping_test.go b/pdata/pprofile/generated_mapping_test.go
index 1ad90e623..601f55eba 100644
--- a/pdata/pprofile/generated_mapping_test.go
+++ b/pdata/pprofile/generated_mapping_test.go
@@ -66,37 +66,37 @@ func TestMapping_MarshalAndUnmarshalJSON(t *testing.T) {
 func TestMapping_MemoryStart(t *testing.T) {
 	ms := NewMapping()
 	assert.Equal(t, uint64(0), ms.MemoryStart())
-	ms.SetMemoryStart(uint64(1))
-	assert.Equal(t, uint64(1), ms.MemoryStart())
+	ms.SetMemoryStart(uint64(13))
+	assert.Equal(t, uint64(13), ms.MemoryStart())
 	sharedState := internal.StateReadOnly
-	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetMemoryStart(uint64(1)) })
+	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetMemoryStart(uint64(13)) })
 }
 
 func TestMapping_MemoryLimit(t *testing.T) {
 	ms := NewMapping()
 	assert.Equal(t, uint64(0), ms.MemoryLimit())
-	ms.SetMemoryLimit(uint64(1))
-	assert.Equal(t, uint64(1), ms.MemoryLimit())
+	ms.SetMemoryLimit(uint64(13))
+	assert.Equal(t, uint64(13), ms.MemoryLimit())
 	sharedState := internal.StateReadOnly
-	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetMemoryLimit(uint64(1)) })
+	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetMemoryLimit(uint64(13)) })
 }
 
 func TestMapping_FileOffset(t *testing.T) {
 	ms := NewMapping()
 	assert.Equal(t, uint64(0), ms.FileOffset())
-	ms.SetFileOffset(uint64(1))
-	assert.Equal(t, uint64(1), ms.FileOffset())
+	ms.SetFileOffset(uint64(13))
+	assert.Equal(t, uint64(13), ms.FileOffset())
 	sharedState := internal.StateReadOnly
-	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetFileOffset(uint64(1)) })
+	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetFileOffset(uint64(13)) })
 }
 
 func TestMapping_FilenameStrindex(t *testing.T) {
 	ms := NewMapping()
 	assert.Equal(t, int32(0), ms.FilenameStrindex())
-	ms.SetFilenameStrindex(int32(1))
-	assert.Equal(t, int32(1), ms.FilenameStrindex())
+	ms.SetFilenameStrindex(int32(13))
+	assert.Equal(t, int32(13), ms.FilenameStrindex())
 	sharedState := internal.StateReadOnly
-	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetFilenameStrindex(int32(1)) })
+	assert.Panics(t, func() { newMapping(&otlpprofiles.Mapping{}, &sharedState).SetFilenameStrindex(int32(13)) })
 }
 
 func TestMapping_AttributeIndices(t *testing.T) {
@@ -149,10 +149,10 @@ func generateTestMapping() Mapping {
 }
 
 func fillTestMapping(tv Mapping) {
-	tv.orig.MemoryStart = uint64(1)
-	tv.orig.MemoryLimit = uint64(1)
-	tv.orig.FileOffset = uint64(1)
-	tv.orig.FilenameStrindex = int32(1)
+	tv.orig.MemoryStart = uint64(13)
+	tv.orig.MemoryLimit = uint64(13)
+	tv.orig.FileOffset = uint64(13)
+	tv.orig.FilenameStrindex = int32(13)
 	internal.FillTestInt32Slice(internal.NewInt32Slice(&tv.orig.AttributeIndices, tv.state))
 	tv.orig.HasFunctions = true
 	tv.orig.HasFilenames = true
