diff --git a/processor/processorhelper/go.mod b/processor/processorhelper/go.mod
index 88a291794..e5c042cc4 100644
--- a/processor/processorhelper/go.mod
+++ b/processor/processorhelper/go.mod
@@ -41,7 +41,7 @@ require (
 	go.opentelemetry.io/otel/sdk v1.38.0 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
 	go.uber.org/zap v1.27.0 // indirect
-	golang.org/x/sys v0.35.0 // indirect
+	golang.org/x/sys v0.37.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
