diff --git a/otelcol/collector_test.go b/otelcol/collector_test.go
index 3b100a346..3c5b4b00c 100644
--- a/otelcol/collector_test.go
+++ b/otelcol/collector_test.go
@@ -436,7 +436,7 @@ func TestPassConfmapToServiceFailure(t *testing.T) {
 		ResolverSettings: confmap.ResolverSettings{
 			URIs:       []string{filepath.Join("testdata", "otelcol-invalid.yaml")},
 			Providers:  makeMapProvidersMap(newFailureProvider()),
-			Converters: []confmap.Converter{expandconverter.New()},
+			Converters: []confmap.Converter{expandconverter.New(confmap.ConverterSettings{})},
 		},
 	})
 	require.NoError(t, err)
