diff --git a/processor/resourceprocessor/config.go b/processor/resourceprocessor/config.go
index 4c5e020c1..5e61758b5 100644
--- a/processor/resourceprocessor/config.go
+++ b/processor/resourceprocessor/config.go
@@ -16,7 +16,7 @@ package resourceprocessor
 
 import (
 	"go.opentelemetry.io/collector/config/configmodels"
-	"go.opentelemetry.io/collector/internal/processor/attraction"
+	"go.opentelemetry.io/collector/processor/processorhelper"
 )
 
 // Config defines configuration for Resource processor.
@@ -25,7 +25,7 @@ type Config struct {
 
 	// AttributesActions specifies the list of actions to be applied on resource attributes.
 	// The set of actions are {INSERT, UPDATE, UPSERT, DELETE, HASH, EXTRACT}.
-	AttributesActions []attraction.ActionKeyValue `mapstructure:"attributes"`
+	AttributesActions []processorhelper.ActionKeyValue `mapstructure:"attributes"`
 
 	// ResourceType field is deprecated. Set "opencensus.type" key in "attributes.upsert" map instead.
 	ResourceType string `mapstructure:"type"`
