diff --git a/model/pdata/generated_metrics_test.go b/model/pdata/generated_metrics_test.go
index ab35cfe4c..7fadd97df 100644
--- a/model/pdata/generated_metrics_test.go
+++ b/model/pdata/generated_metrics_test.go
@@ -645,6 +645,14 @@ func TestNumberDataPoint_LabelsMap(t *testing.T) {
 	assert.EqualValues(t, testValLabelsMap, ms.LabelsMap())
 }
 
+func TestNumberDataPoint_Attributes(t *testing.T) {
+	ms := NewNumberDataPoint()
+	assert.EqualValues(t, NewAttributeMap(), ms.Attributes())
+	fillTestAttributeMap(ms.Attributes())
+	testValAttributes := generateTestAttributeMap()
+	assert.EqualValues(t, testValAttributes, ms.Attributes())
+}
+
 func TestNumberDataPoint_StartTimestamp(t *testing.T) {
 	ms := NewNumberDataPoint()
 	assert.EqualValues(t, Timestamp(0), ms.StartTimestamp())
@@ -808,6 +816,14 @@ func TestHistogramDataPoint_LabelsMap(t *testing.T) {
 	assert.EqualValues(t, testValLabelsMap, ms.LabelsMap())
 }
 
+func TestHistogramDataPoint_Attributes(t *testing.T) {
+	ms := NewHistogramDataPoint()
+	assert.EqualValues(t, NewAttributeMap(), ms.Attributes())
+	fillTestAttributeMap(ms.Attributes())
+	testValAttributes := generateTestAttributeMap()
+	assert.EqualValues(t, testValAttributes, ms.Attributes())
+}
+
 func TestHistogramDataPoint_StartTimestamp(t *testing.T) {
 	ms := NewHistogramDataPoint()
 	assert.EqualValues(t, Timestamp(0), ms.StartTimestamp())
@@ -988,6 +1004,14 @@ func TestSummaryDataPoint_LabelsMap(t *testing.T) {
 	assert.EqualValues(t, testValLabelsMap, ms.LabelsMap())
 }
 
+func TestSummaryDataPoint_Attributes(t *testing.T) {
+	ms := NewSummaryDataPoint()
+	assert.EqualValues(t, NewAttributeMap(), ms.Attributes())
+	fillTestAttributeMap(ms.Attributes())
+	testValAttributes := generateTestAttributeMap()
+	assert.EqualValues(t, testValAttributes, ms.Attributes())
+}
+
 func TestSummaryDataPoint_StartTimestamp(t *testing.T) {
 	ms := NewSummaryDataPoint()
 	assert.EqualValues(t, Timestamp(0), ms.StartTimestamp())
@@ -1443,6 +1467,7 @@ func generateTestNumberDataPoint() NumberDataPoint {
 
 func fillTestNumberDataPoint(tv NumberDataPoint) {
 	fillTestStringMap(tv.LabelsMap())
+	fillTestAttributeMap(tv.Attributes())
 	tv.SetStartTimestamp(Timestamp(1234567890))
 	tv.SetTimestamp(Timestamp(1234567890))
 	tv.SetDoubleVal(float64(17.13))
@@ -1472,6 +1497,7 @@ func generateTestHistogramDataPoint() HistogramDataPoint {
 
 func fillTestHistogramDataPoint(tv HistogramDataPoint) {
 	fillTestStringMap(tv.LabelsMap())
+	fillTestAttributeMap(tv.Attributes())
 	tv.SetStartTimestamp(Timestamp(1234567890))
 	tv.SetTimestamp(Timestamp(1234567890))
 	tv.SetCount(uint64(17))
@@ -1503,6 +1529,7 @@ func generateTestSummaryDataPoint() SummaryDataPoint {
 
 func fillTestSummaryDataPoint(tv SummaryDataPoint) {
 	fillTestStringMap(tv.LabelsMap())
+	fillTestAttributeMap(tv.Attributes())
 	tv.SetStartTimestamp(Timestamp(1234567890))
 	tv.SetTimestamp(Timestamp(1234567890))
 	tv.SetCount(uint64(17))
