diff --git a/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go b/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go
index 7d16af7e4..d9d844e14 100644
--- a/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go
+++ b/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go
@@ -3,7 +3,6 @@
 package metadatatest
 
 import (
-	"context"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -16,21 +15,6 @@ import (
 	"go.opentelemetry.io/collector/processor/processortest"
 )
 
-// Deprecated: [v0.119.0] Use componenttest.Telemetry
-type Telemetry struct {
-	*componenttest.Telemetry
-}
-
-// Deprecated: [v0.119.0] Use componenttest.NewTelemetry
-func SetupTelemetry(opts ...componenttest.TelemetryOption) Telemetry {
-	return Telemetry{Telemetry: componenttest.NewTelemetry(opts...)}
-}
-
-// Deprecated: [v0.119.0] Use metadatatest.NewSettings
-func (tt *Telemetry) NewSettings() processor.Settings {
-	return NewSettings(tt.Telemetry)
-}
-
 func NewSettings(tt *componenttest.Telemetry) processor.Settings {
 	set := processortest.NewNopSettings()
 	set.ID = component.NewID(component.MustNewType("processorhelper"))
@@ -38,20 +22,6 @@ func NewSettings(tt *componenttest.Telemetry) processor.Settings {
 	return set
 }
 
-// Deprecated: [v0.119.0] Use metadatatest.AssertEqual*
-func (tt *Telemetry) AssertMetrics(t *testing.T, expected []metricdata.Metrics, opts ...metricdatatest.Option) {
-	var md metricdata.ResourceMetrics
-	require.NoError(t, tt.Reader.Collect(context.Background(), &md))
-	// ensure all required metrics are present
-	for _, want := range expected {
-		got := getMetricFromResource(want.Name, md)
-		metricdatatest.AssertEqual(t, want, got, opts...)
-	}
-
-	// ensure no additional metrics are emitted
-	require.Equal(t, len(expected), lenMetrics(md))
-}
-
 func AssertEqualProcessorIncomingItems(t *testing.T, tt *componenttest.Telemetry, dps []metricdata.DataPoint[int64], opts ...metricdatatest.Option) {
 	want := metricdata.Metrics{
 		Name:        "otelcol_processor_incoming_items",
