diff --git a/processor/processorhelper/traces_test.go b/processor/processorhelper/traces_test.go
index df5808c13..8aaf81adc 100644
--- a/processor/processorhelper/traces_test.go
+++ b/processor/processorhelper/traces_test.go
@@ -24,13 +24,12 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
-	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
-var testTracesCfg = config.NewProcessorSettings(config.NewComponentID("test"))
+var testTracesCfg = component.NewProcessorConfigSettings(component.NewID("test"))
 
 func TestNewTracesProcessor(t *testing.T) {
 	tp, err := NewTracesProcessor(context.Background(), componenttest.NewNopProcessorCreateSettings(), &testTracesCfg, consumertest.NewNop(), newTestTProcessor(nil))
