diff --git a/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_metrics_test.go b/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_metrics_test.go
index 9c1895ccf..f20f2db07 100644
--- a/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_metrics_test.go
+++ b/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_metrics_test.go
@@ -450,7 +450,7 @@ func TestMetricsBuilder(t *testing.T) {
 						validatedMetrics["reaggregate.metric"] = true
 						assert.Equal(t, pmetric.MetricTypeGauge, ms.At(i).Type())
 						assert.Equal(t, 1, ms.At(i).Gauge().DataPoints().Len())
-						assert.Equal(t, "Metric for testing spacial reaggregation", ms.At(i).Description())
+						assert.Equal(t, "Metric for testing spatial reaggregation", ms.At(i).Description())
 						assert.Equal(t, "1", ms.At(i).Unit())
 						dp := ms.At(i).Gauge().DataPoints().At(0)
 						assert.Equal(t, start, dp.StartTimestamp())
@@ -468,7 +468,7 @@ func TestMetricsBuilder(t *testing.T) {
 						validatedMetrics["reaggregate.metric"] = true
 						assert.Equal(t, pmetric.MetricTypeGauge, ms.At(i).Type())
 						assert.Equal(t, 1, ms.At(i).Gauge().DataPoints().Len())
-						assert.Equal(t, "Metric for testing spacial reaggregation", ms.At(i).Description())
+						assert.Equal(t, "Metric for testing spatial reaggregation", ms.At(i).Description())
 						assert.Equal(t, "1", ms.At(i).Unit())
 						dp := ms.At(i).Gauge().DataPoints().At(0)
 						assert.Equal(t, start, dp.StartTimestamp())
