diff --git a/consumer/simple/metrics_test.go b/consumer/simple/metrics_test.go
index 802d2c276..ab8a76378 100644
--- a/consumer/simple/metrics_test.go
+++ b/consumer/simple/metrics_test.go
@@ -467,7 +467,7 @@ func BenchmarkSimpleMetrics(b *testing.B) {
 
 func BenchmarkPdataMetrics(b *testing.B) {
 	for n := 0; n < b.N; n++ {
-		tsNano := pdata.TimestampUnixNano(time.Now().UnixNano())
+		tsNano := pdata.TimestampFromTime(time.Now())
 
 		m := pdata.NewMetrics()
 
