diff --git a/receiver/prometheusreceiver/internal/ocastore.go b/receiver/prometheusreceiver/internal/ocastore.go
index 1260ed49b..85b1931e7 100644
--- a/receiver/prometheusreceiver/internal/ocastore.go
+++ b/receiver/prometheusreceiver/internal/ocastore.go
@@ -37,6 +37,7 @@ const (
 
 var idSeq int64
 var errAlreadyStop = errors.New("ocaStore already stopped")
+var noop = &noopAppender{}
 
 // OcaStore is an interface combines io.Closer and prometheus' scrape.Appendable
 type OcaStore interface {
@@ -53,16 +54,18 @@ type ocaStore struct {
 	mc      *mService
 	once    *sync.Once
 	ctx     context.Context
+	jobsMap *JobsMap
 }
 
 // NewOcaStore returns an ocaStore instance, which can be acted as prometheus' scrape.Appendable
-func NewOcaStore(ctx context.Context, sink consumer.MetricsConsumer, logger *zap.SugaredLogger) OcaStore {
+func NewOcaStore(ctx context.Context, sink consumer.MetricsConsumer, logger *zap.SugaredLogger, jobsMap *JobsMap) OcaStore {
 	return &ocaStore{
 		running: runningStateInit,
 		ctx:     ctx,
 		sink:    sink,
 		logger:  logger,
 		once:    &sync.Once{},
+		jobsMap: jobsMap,
 	}
 }
 
@@ -70,19 +73,19 @@ func NewOcaStore(ctx context.Context, sink consumer.MetricsConsumer, logger *zap
 // cannot accept any Appender() request
 func (o *ocaStore) SetScrapeManager(scrapeManager *scrape.Manager) {
 	if scrapeManager != nil && atomic.CompareAndSwapInt32(&o.running, runningStateInit, runningStateReady) {
-		o.mc = &mService{scrapeManager}
+		o.mc = &mService{sm: scrapeManager}
 	}
 }
 
 func (o *ocaStore) Appender() (storage.Appender, error) {
 	state := atomic.LoadInt32(&o.running)
 	if state == runningStateReady {
-		return newTransaction(o.ctx, o.mc, o.sink, o.logger), nil
+		return newTransaction(o.ctx, o.jobsMap, o.mc, o.sink, o.logger), nil
 	} else if state == runningStateInit {
 		return nil, errors.New("ScrapeManager is not set")
 	}
 	// instead of returning an error, return a dummy appender instead, otherwise it can trigger panic
-	return noopAppender(true), nil
+	return noop, nil
 }
 
 func (o *ocaStore) Close() error {
@@ -91,22 +94,20 @@ func (o *ocaStore) Close() error {
 }
 
 // noopAppender, always return error on any operations
-type noopAppender bool
+type noopAppender struct{}
 
-func (noopAppender) Add(l labels.Labels, t int64, v float64) (uint64, error) {
+func (*noopAppender) Add(l labels.Labels, t int64, v float64) (uint64, error) {
 	return 0, errAlreadyStop
 }
 
-func (noopAppender) AddFast(l labels.Labels, ref uint64, t int64, v float64) error {
+func (*noopAppender) AddFast(l labels.Labels, ref uint64, t int64, v float64) error {
 	return errAlreadyStop
 }
 
-func (noopAppender) Commit() error {
+func (*noopAppender) Commit() error {
 	return errAlreadyStop
 }
 
-func (noopAppender) Rollback() error {
-	return errAlreadyStop
+func (*noopAppender) Rollback() error {
+	return nil
 }
-
-var _ storage.Appender = noopAppender(true)
