diff --git a/receiver/prometheusreceiver/internal/transaction_test.go b/receiver/prometheusreceiver/internal/transaction_test.go
index 4cdd0911e..36a4c2ae6 100644
--- a/receiver/prometheusreceiver/internal/transaction_test.go
+++ b/receiver/prometheusreceiver/internal/transaction_test.go
@@ -16,6 +16,7 @@ package internal
 
 import (
 	"context"
+	"math"
 	"reflect"
 	"testing"
 	"time"
@@ -33,7 +34,7 @@ func Test_transaction(t *testing.T) {
 
 	t.Run("Commit Without Adding", func(t *testing.T) {
 		mcon := newMockConsumer()
-		tr := newTransaction(context.Background(), ms, mcon, testLogger)
+		tr := newTransaction(context.Background(), nil, ms, mcon, testLogger)
 		if got := tr.Commit(); got != nil {
 			t.Errorf("expecting nil from Commit() but got err %v", got)
 		}
@@ -41,7 +42,7 @@ func Test_transaction(t *testing.T) {
 
 	t.Run("Rollback dose nothing", func(t *testing.T) {
 		mcon := newMockConsumer()
-		tr := newTransaction(context.Background(), ms, mcon, testLogger)
+		tr := newTransaction(context.Background(), nil, ms, mcon, testLogger)
 		if got := tr.Rollback(); got != nil {
 			t.Errorf("expecting nil from Rollback() but got err %v", got)
 		}
@@ -50,7 +51,7 @@ func Test_transaction(t *testing.T) {
 	badLabels := labels.Labels([]labels.Label{{Name: "foo", Value: "bar"}})
 	t.Run("Add One No Target", func(t *testing.T) {
 		mcon := newMockConsumer()
-		tr := newTransaction(context.Background(), ms, mcon, testLogger)
+		tr := newTransaction(context.Background(), nil, ms, mcon, testLogger)
 		if _, got := tr.Add(badLabels, time.Now().Unix()*1000, 1.0); got == nil {
 			t.Errorf("expecting error from Add() but got nil")
 		}
@@ -62,7 +63,7 @@ func Test_transaction(t *testing.T) {
 		{Name: "foo", Value: "bar"}})
 	t.Run("Add One Job not found", func(t *testing.T) {
 		mcon := newMockConsumer()
-		tr := newTransaction(context.Background(), ms, mcon, testLogger)
+		tr := newTransaction(context.Background(), nil, ms, mcon, testLogger)
 		if _, got := tr.Add(jobNotFoundLb, time.Now().Unix()*1000, 1.0); got == nil {
 			t.Errorf("expecting error from Add() but got nil")
 		}
@@ -73,7 +74,7 @@ func Test_transaction(t *testing.T) {
 		{Name: "__name__", Value: "foo"}})
 	t.Run("Add One Good", func(t *testing.T) {
 		mcon := newMockConsumer()
-		tr := newTransaction(context.Background(), ms, mcon, testLogger)
+		tr := newTransaction(context.Background(), nil, ms, mcon, testLogger)
 		if _, got := tr.Add(goodLabels, time.Now().Unix()*1000, 1.0); got != nil {
 			t.Errorf("expecting error == nil from Add() but got: %v\n", got)
 		}
@@ -82,7 +83,7 @@ func Test_transaction(t *testing.T) {
 		}
 
 		expected := createNode("test", "localhost:8080", "http")
-		md := <-mcon.Metrics
+		md := mcon.md
 		if !reflect.DeepEqual(md.Node, expected) {
 			t.Errorf("generated node %v and expected node %v is different\n", md.Node, expected)
 		}
@@ -92,4 +93,19 @@ func Test_transaction(t *testing.T) {
 		}
 	})
 
+	t.Run("Drop NaN value", func(t *testing.T) {
+		mcon := newMockConsumer()
+		tr := newTransaction(context.Background(), nil, ms, mcon, testLogger)
+		if _, got := tr.Add(goodLabels, time.Now().Unix()*1000, math.NaN()); got != nil {
+			t.Errorf("expecting error == nil from Add() but got: %v\n", got)
+		}
+		if got := tr.Commit(); got != nil {
+			t.Errorf("expecting nil from Commit() but got err %v", got)
+		}
+
+		if mcon.md != nil {
+			t.Errorf("wanted nil, got %v\n", mcon.md)
+		}
+	})
+
 }
