diff --git a/obsreport/obsreport_exporter.go b/obsreport/obsreport_exporter.go
index f80e00354..3e6bb11fe 100644
--- a/obsreport/obsreport_exporter.go
+++ b/obsreport/obsreport_exporter.go
@@ -38,7 +38,7 @@ const (
 	exporterScope = scopeName + nameSep + exporterName
 )
 
-// Exporter is a helper to add observability to a component.Exporter.
+// Exporter is a helper to add observability to an exporter.
 type Exporter struct {
 	level          configtelemetry.Level
 	spanNamePrefix string
