diff --git a/consumer/go.mod b/consumer/go.mod
index fb5505af1..0acb7760d 100644
--- a/consumer/go.mod
+++ b/consumer/go.mod
@@ -4,8 +4,8 @@ go 1.20
 
 require (
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector v0.91.0
-	go.opentelemetry.io/collector/pdata v1.0.0
+	go.opentelemetry.io/collector v0.92.0
+	go.opentelemetry.io/collector/pdata v1.0.1
 	go.uber.org/goleak v1.3.0
 )
 
