diff --git a/service/internal/builder/pipelines_builder.go b/service/internal/builder/pipelines_builder.go
index 320f0d5d4..3175b241e 100644
--- a/service/internal/builder/pipelines_builder.go
+++ b/service/internal/builder/pipelines_builder.go
@@ -24,7 +24,7 @@ import (
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumererror"
-	"go.opentelemetry.io/collector/consumer/fanoutconsumer"
+	"go.opentelemetry.io/collector/service/internal/fanoutconsumer"
 )
 
 // builtPipeline is a pipeline that is built based on a config.
