diff --git a/receiver/scraperhelper/scrapercontroller_test.go b/receiver/scraperhelper/scrapercontroller_test.go
index c1556b908..0adff240e 100644
--- a/receiver/scraperhelper/scrapercontroller_test.go
+++ b/receiver/scraperhelper/scrapercontroller_test.go
@@ -71,7 +71,7 @@ func (ts *testScrapeMetrics) scrape(_ context.Context) (pdata.Metrics, error) {
 	}
 
 	md := pdata.NewMetrics()
-	metric := md.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty().Metrics().AppendEmpty()
+	metric := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
 	metric.SetDataType(pdata.MetricDataTypeGauge)
 	metric.Gauge().DataPoints().AppendEmpty()
 	return md, nil
