diff --git a/service/service_test.go b/service/service_test.go
index b1875c790..f9643e830 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -104,7 +104,7 @@ func createExampleService(t *testing.T) *service {
 	require.NoError(t, err)
 
 	srv, err := newService(&svcSettings{
-		BuildInfo: component.DefaultBuildInfo(),
+		BuildInfo: component.NewDefaultBuildInfo(),
 		Factories: factories,
 		Telemetry: componenttest.NewNopTelemetrySettings(),
 		Config:    cfg,
