diff --git a/exporter/otlpexporter/go.mod b/exporter/otlpexporter/go.mod
index 85a07dd89..e9675b632 100644
--- a/exporter/otlpexporter/go.mod
+++ b/exporter/otlpexporter/go.mod
@@ -16,7 +16,7 @@ require (
 	go.opentelemetry.io/collector/exporter v0.90.0
 	go.opentelemetry.io/collector/pdata v1.0.0
 	go.opentelemetry.io/collector/receiver/otlpreceiver v0.90.0
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20231120223509-83a465c0220f
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20231127180814-3a041ad873d4
 	google.golang.org/grpc v1.59.0
 	google.golang.org/protobuf v1.31.0
 )
