diff --git a/exporter/loggingexporter/logging_exporter.go b/exporter/loggingexporter/logging_exporter.go
index 242790888..37d6035df 100644
--- a/exporter/loggingexporter/logging_exporter.go
+++ b/exporter/loggingexporter/logging_exporter.go
@@ -325,9 +325,7 @@ func (s *loggingExporter) pushTraceData(
 			buf.logEntry("* Nil ResourceSpans")
 			continue
 		}
-		if !rs.Resource().IsNil() {
-			buf.logAttributeMap("Resource labels", rs.Resource().Attributes())
-		}
+		buf.logAttributeMap("Resource labels", rs.Resource().Attributes())
 		ilss := rs.InstrumentationLibrarySpans()
 		for j := 0; j < ilss.Len(); j++ {
 			buf.logEntry("InstrumentationLibrarySpans #%d", j)
@@ -391,9 +389,7 @@ func (s *loggingExporter) pushMetricsData(
 			buf.logEntry("* Nil ResourceMetrics")
 			continue
 		}
-		if !rm.Resource().IsNil() {
-			buf.logAttributeMap("Resource labels", rm.Resource().Attributes())
-		}
+		buf.logAttributeMap("Resource labels", rm.Resource().Attributes())
 		ilms := rm.InstrumentationLibraryMetrics()
 		for j := 0; j < ilms.Len(); j++ {
 			buf.logEntry("InstrumentationLibraryMetrics #%d", j)
@@ -504,10 +500,7 @@ func (s *loggingExporter) pushLogData(
 			buf.logEntry("* Nil ResourceLog")
 			continue
 		}
-		if !rl.Resource().IsNil() {
-			buf.logAttributeMap("Resource labels", rl.Resource().Attributes())
-		}
-
+		buf.logAttributeMap("Resource labels", rl.Resource().Attributes())
 		ills := rl.InstrumentationLibraryLogs()
 		for j := 0; j < ills.Len(); j++ {
 			buf.logEntry("InstrumentationLibraryLogs #%d", j)
