diff --git a/pdata/ptrace/generated_status.go b/pdata/ptrace/generated_status.go
index c0850997b..2b3e66a92 100644
--- a/pdata/ptrace/generated_status.go
+++ b/pdata/ptrace/generated_status.go
@@ -7,6 +7,7 @@
 package ptrace
 
 import (
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 )
 
@@ -19,11 +20,12 @@ import (
 // Must use NewStatus function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type Status struct {
-	orig *otlptrace.Status
+	orig  *otlptrace.Status
+	state *internal.State
 }
 
-func newStatus(orig *otlptrace.Status) Status {
-	return Status{orig}
+func newStatus(orig *otlptrace.Status, state *internal.State) Status {
+	return Status{orig: orig, state: state}
 }
 
 // NewStatus creates a new empty Status.
@@ -31,12 +33,15 @@ func newStatus(orig *otlptrace.Status) Status {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewStatus() Status {
-	return newStatus(&otlptrace.Status{})
+	state := internal.StateMutable
+	return newStatus(&otlptrace.Status{}, &state)
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms Status) MoveTo(dest Status) {
+	ms.state.AssertMutable()
+	dest.state.AssertMutable()
 	*dest.orig = *ms.orig
 	*ms.orig = otlptrace.Status{}
 }
@@ -48,6 +53,7 @@ func (ms Status) Code() StatusCode {
 
 // SetCode replaces the code associated with this Status.
 func (ms Status) SetCode(v StatusCode) {
+	ms.state.AssertMutable()
 	ms.orig.Code = otlptrace.Status_StatusCode(v)
 }
 
@@ -58,11 +64,13 @@ func (ms Status) Message() string {
 
 // SetMessage replaces the message associated with this Status.
 func (ms Status) SetMessage(v string) {
+	ms.state.AssertMutable()
 	ms.orig.Message = v
 }
 
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Status) CopyTo(dest Status) {
+	dest.state.AssertMutable()
 	dest.SetCode(ms.Code())
 	dest.SetMessage(ms.Message())
 }
