diff --git a/exporter/exportertest/sink_exporter.go b/exporter/exportertest/sink_exporter.go
index 6c4eb1056..ef837d1c3 100644
--- a/exporter/exportertest/sink_exporter.go
+++ b/exporter/exportertest/sink_exporter.go
@@ -21,6 +21,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/consumerdata"
 	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
 	"go.opentelemetry.io/collector/internal/data"
 )
 
@@ -34,12 +35,12 @@ type SinkTraceExporterOld struct {
 // Start tells the exporter to start. The exporter may prepare for exporting
 // by connecting to the endpoint. Host parameter can be used for communicating
 // with the host after Start() has already returned.
-func (ste *SinkTraceExporterOld) Start(ctx context.Context, host component.Host) error {
+func (ste *SinkTraceExporterOld) Start(context.Context, component.Host) error {
 	return nil
 }
 
 // ConsumeTraceData stores traces for tests.
-func (ste *SinkTraceExporterOld) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
+func (ste *SinkTraceExporterOld) ConsumeTraceData(_ context.Context, td consumerdata.TraceData) error {
 	if ste.consumeTraceError != nil {
 		return ste.consumeTraceError
 	}
@@ -72,28 +73,30 @@ func (ste *SinkTraceExporterOld) Shutdown(context.Context) error {
 
 // SinkTraceExporter acts as a trace receiver for use in tests.
 type SinkTraceExporter struct {
-	consumeTraceError error // to be returned by ConsumeTraces, if set
 	mu                sync.Mutex
+	consumeTraceError error // to be returned by ConsumeTraces, if set
 	traces            []pdata.Traces
+	spansCount        int
 }
 
 // Start tells the exporter to start. The exporter may prepare for exporting
 // by connecting to the endpoint. Host parameter can be used for communicating
 // with the host after Start() has already returned.
-func (ste *SinkTraceExporter) Start(ctx context.Context, host component.Host) error {
+func (ste *SinkTraceExporter) Start(context.Context, component.Host) error {
 	return nil
 }
 
 // ConsumeTraceData stores traces for tests.
-func (ste *SinkTraceExporter) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+func (ste *SinkTraceExporter) ConsumeTraces(_ context.Context, td pdata.Traces) error {
+	ste.mu.Lock()
+	defer ste.mu.Unlock()
+
 	if ste.consumeTraceError != nil {
 		return ste.consumeTraceError
 	}
 
-	ste.mu.Lock()
-	defer ste.mu.Unlock()
-
 	ste.traces = append(ste.traces, td)
+	ste.spansCount += td.SpanCount()
 
 	return nil
 }
@@ -103,11 +106,22 @@ func (ste *SinkTraceExporter) AllTraces() []pdata.Traces {
 	ste.mu.Lock()
 	defer ste.mu.Unlock()
 
-	return ste.traces
+	copyTraces := make([]pdata.Traces, len(ste.traces))
+	copy(copyTraces, ste.traces)
+	return copyTraces
+}
+
+// SpansCount return the number of spans sent to the test sing.
+func (ste *SinkTraceExporter) SpansCount() int {
+	ste.mu.Lock()
+	defer ste.mu.Unlock()
+	return ste.spansCount
 }
 
 // SetConsumeTraceError sets an error that will be returned by ConsumeTraces
 func (ste *SinkTraceExporter) SetConsumeTraceError(err error) {
+	ste.mu.Lock()
+	defer ste.mu.Unlock()
 	ste.consumeTraceError = err
 }
 
@@ -118,20 +132,20 @@ func (ste *SinkTraceExporter) Shutdown(context.Context) error {
 
 // SinkMetricsExporterOld acts as a metrics receiver for use in tests.
 type SinkMetricsExporterOld struct {
-	consumeMetricsError error // to be returned by ConsumeMetricsData, if set
 	mu                  sync.Mutex
+	consumeMetricsError error // to be returned by ConsumeMetricsData, if set
 	metrics             []consumerdata.MetricsData
 }
 
 // Start tells the exporter to start. The exporter may prepare for exporting
 // by connecting to the endpoint. Host parameter can be used for communicating
 // with the host after Start() has already returned.
-func (sme *SinkMetricsExporterOld) Start(ctx context.Context, host component.Host) error {
+func (sme *SinkMetricsExporterOld) Start(context.Context, component.Host) error {
 	return nil
 }
 
 // ConsumeMetricsData stores traces for tests.
-func (sme *SinkMetricsExporterOld) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
+func (sme *SinkMetricsExporterOld) ConsumeMetricsData(_ context.Context, md consumerdata.MetricsData) error {
 	if sme.consumeMetricsError != nil {
 		return sme.consumeMetricsError
 	}
@@ -164,33 +178,36 @@ func (sme *SinkMetricsExporterOld) Shutdown(context.Context) error {
 
 // SinkMetricsExporter acts as a metrics receiver for use in tests.
 type SinkMetricsExporter struct {
-	consumeMetricsError error // to be returned by ConsumeMetrics, if set
 	mu                  sync.Mutex
+	consumeMetricsError error // to be returned by ConsumeMetrics, if set
 	metrics             []pdata.Metrics
+	metricsCount        int
 }
 
 // SetConsumeMetricsError sets an error that will be returned by ConsumeMetrics
 func (sme *SinkMetricsExporter) SetConsumeMetricsError(err error) {
+	sme.mu.Lock()
+	defer sme.mu.Unlock()
 	sme.consumeMetricsError = err
 }
 
 // Start tells the exporter to start. The exporter may prepare for exporting
 // by connecting to the endpoint. Host parameter can be used for communicating
 // with the host after Start() has already returned.
-func (sme *SinkMetricsExporter) Start(ctx context.Context, host component.Host) error {
+func (sme *SinkMetricsExporter) Start(context.Context, component.Host) error {
 	return nil
 }
 
 // ConsumeMetricsData stores traces for tests.
 func (sme *SinkMetricsExporter) ConsumeMetrics(_ context.Context, md pdata.Metrics) error {
+	sme.mu.Lock()
+	defer sme.mu.Unlock()
 	if sme.consumeMetricsError != nil {
 		return sme.consumeMetricsError
 	}
 
-	sme.mu.Lock()
-	defer sme.mu.Unlock()
-
 	sme.metrics = append(sme.metrics, md)
+	sme.metricsCount += pdatautil.MetricsToInternalMetrics(md).MetricCount()
 
 	return nil
 }
@@ -200,7 +217,16 @@ func (sme *SinkMetricsExporter) AllMetrics() []pdata.Metrics {
 	sme.mu.Lock()
 	defer sme.mu.Unlock()
 
-	return sme.metrics
+	copyMetrics := make([]pdata.Metrics, len(sme.metrics))
+	copy(copyMetrics, sme.metrics)
+	return copyMetrics
+}
+
+// MetricsCount return the number of metrics sent to the test sing.
+func (sme *SinkMetricsExporter) MetricsCount() int {
+	sme.mu.Lock()
+	defer sme.mu.Unlock()
+	return sme.metricsCount
 }
 
 // Shutdown stops the exporter and is invoked during shutdown.
@@ -212,44 +238,56 @@ func (sme *SinkMetricsExporter) Shutdown(context.Context) error {
 type SinkLogExporter struct {
 	consumeLogError error // to be returned by ConsumeLog, if set
 	mu              sync.Mutex
-	metrics         []data.Logs
+	logs            []data.Logs
+	logRecordsCount int
 }
 
 // SetConsumeLogError sets an error that will be returned by ConsumeLog
-func (sme *SinkLogExporter) SetConsumeLogError(err error) {
-	sme.consumeLogError = err
+func (sle *SinkLogExporter) SetConsumeLogError(err error) {
+	sle.mu.Lock()
+	defer sle.mu.Unlock()
+	sle.consumeLogError = err
 }
 
 // Start tells the exporter to start. The exporter may prepare for exporting
 // by connecting to the endpoint. Host parameter can be used for communicating
 // with the host after Start() has already returned.
-func (sme *SinkLogExporter) Start(ctx context.Context, host component.Host) error {
+func (sle *SinkLogExporter) Start(context.Context, component.Host) error {
 	return nil
 }
 
 // ConsumeLogData stores traces for tests.
-func (sme *SinkLogExporter) ConsumeLogs(_ context.Context, md data.Logs) error {
-	if sme.consumeLogError != nil {
-		return sme.consumeLogError
+func (sle *SinkLogExporter) ConsumeLogs(_ context.Context, ld data.Logs) error {
+	sle.mu.Lock()
+	defer sle.mu.Unlock()
+	if sle.consumeLogError != nil {
+		return sle.consumeLogError
 	}
 
-	sme.mu.Lock()
-	defer sme.mu.Unlock()
-
-	sme.metrics = append(sme.metrics, md)
+	sle.logs = append(sle.logs, ld)
+	sle.logRecordsCount += ld.LogRecordCount()
 
 	return nil
 }
 
 // AllLog returns the metrics sent to the test sink.
-func (sme *SinkLogExporter) AllLogs() []data.Logs {
-	sme.mu.Lock()
-	defer sme.mu.Unlock()
+func (sle *SinkLogExporter) AllLogs() []data.Logs {
+	sle.mu.Lock()
+	defer sle.mu.Unlock()
 
-	return sme.metrics
+	copyLogs := make([]data.Logs, len(sle.logs))
+	copy(copyLogs, sle.logs)
+	return copyLogs
+}
+
+// LogRecordsCount return the number of log records sent to the test sing.
+func (sle *SinkLogExporter) LogRecordsCount() int {
+	sle.mu.Lock()
+	defer sle.mu.Unlock()
+	return sle.logRecordsCount
 }
 
 // Shutdown stops the exporter and is invoked during shutdown.
-func (sme *SinkLogExporter) Shutdown(context.Context) error {
+func (sle *SinkLogExporter) Shutdown(context.Context) error {
 	return nil
 }
