diff --git a/service/internal/builders/processor.go b/service/internal/builders/processor.go
index 4200a9972..35b327c3e 100644
--- a/service/internal/builders/processor.go
+++ b/service/internal/builders/processor.go
@@ -9,7 +9,9 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerprofiles"
 	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorprofiles"
 	"go.opentelemetry.io/collector/processor/processortest"
 )
 
@@ -91,6 +93,25 @@ func (b *ProcessorBuilder) CreateLogs(ctx context.Context, set processor.Setting
 	return f.CreateLogsProcessor(ctx, set, cfg, next)
 }
 
+// CreateProfiles creates a Profiles processor based on the settings and config.
+func (b *ProcessorBuilder) CreateProfiles(ctx context.Context, set processor.Settings, next consumerprofiles.Profiles) (processorprofiles.Profiles, error) {
+	if next == nil {
+		return nil, errNilNextConsumer
+	}
+	cfg, existsCfg := b.cfgs[set.ID]
+	if !existsCfg {
+		return nil, fmt.Errorf("processor %q is not configured", set.ID)
+	}
+
+	f, existsFactory := b.factories[set.ID.Type()]
+	if !existsFactory {
+		return nil, fmt.Errorf("processor factory not available for: %q", set.ID)
+	}
+
+	logStabilityLevel(set.Logger, f.ProfilesProcessorStability())
+	return f.CreateProfilesProcessor(ctx, set, cfg, next)
+}
+
 func (b *ProcessorBuilder) Factory(componentType component.Type) component.Factory {
 	return b.factories[componentType]
 }
