diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 6b9d1074a..aa890b586 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -47,7 +47,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/prometheus/client_golang v1.20.2 // indirect
 	github.com/prometheus/client_model v0.6.1 // indirect
-	github.com/prometheus/common v0.56.0 // indirect
+	github.com/prometheus/common v0.57.0 // indirect
 	github.com/prometheus/procfs v0.15.1 // indirect
 	go.opentelemetry.io/collector/confmap v1.14.1 // indirect
 	go.opentelemetry.io/collector/extension v0.108.1 // indirect
