diff --git a/config/configtls/configtls_test.go b/config/configtls/configtls_test.go
index db1658f1d..ca7054348 100644
--- a/config/configtls/configtls_test.go
+++ b/config/configtls/configtls_test.go
@@ -627,3 +627,52 @@ func TestMinMaxTLSVersions(t *testing.T) {
 		})
 	}
 }
+
+func TestCipherSuites(t *testing.T) {
+	tests := []struct {
+		name       string
+		tlsSetting TLSSetting
+		wantErr    string
+		result     []uint16
+	}{
+		{
+			name:       "no suites set",
+			tlsSetting: TLSSetting{},
+			result:     nil,
+		},
+		{
+			name: "one cipher suite set",
+			tlsSetting: TLSSetting{
+				CipherSuites: []string{"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"},
+			},
+			result: []uint16{tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA},
+		},
+		{
+			name: "invalid cipher suite set",
+			tlsSetting: TLSSetting{
+				CipherSuites: []string{"FOO"},
+			},
+			wantErr: `invalid TLS cipher suite: "FOO"`,
+		},
+		{
+			name: "multiple invalid cipher suites set",
+			tlsSetting: TLSSetting{
+				CipherSuites: []string{"FOO", "BAR"},
+			},
+			wantErr: `invalid TLS cipher suite: "FOO"
+invalid TLS cipher suite: "BAR"`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			config, err := test.tlsSetting.loadTLSConfig()
+			if test.wantErr != "" {
+				assert.EqualError(t, err, test.wantErr)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, test.result, config.CipherSuites)
+			}
+		})
+	}
+}
