diff --git a/pdata/pmetric/generated_scopemetricsslice.go b/pdata/pmetric/generated_scopemetricsslice.go
index 5cdfac9d7..2a1fc16c6 100644
--- a/pdata/pmetric/generated_scopemetricsslice.go
+++ b/pdata/pmetric/generated_scopemetricsslice.go
@@ -145,22 +145,7 @@ func (es ScopeMetricsSlice) RemoveIf(f func(ScopeMetrics) bool) {
 // CopyTo copies all elements from the current slice overriding the destination.
 func (es ScopeMetricsSlice) CopyTo(dest ScopeMetricsSlice) {
 	dest.state.AssertMutable()
-	srcLen := es.Len()
-	destCap := cap(*dest.orig)
-	if srcLen <= destCap {
-		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
-		for i := range *es.orig {
-			newScopeMetrics((*es.orig)[i], es.state).CopyTo(newScopeMetrics((*dest.orig)[i], dest.state))
-		}
-		return
-	}
-	origs := make([]otlpmetrics.ScopeMetrics, srcLen)
-	wrappers := make([]*otlpmetrics.ScopeMetrics, srcLen)
-	for i := range *es.orig {
-		wrappers[i] = &origs[i]
-		newScopeMetrics((*es.orig)[i], es.state).CopyTo(newScopeMetrics(wrappers[i], dest.state))
-	}
-	*dest.orig = wrappers
+	*dest.orig = copyOrigScopeMetricsSlice(*dest.orig, *es.orig)
 }
 
 // Sort sorts the ScopeMetrics elements within ScopeMetricsSlice given the
@@ -170,3 +155,18 @@ func (es ScopeMetricsSlice) Sort(less func(a, b ScopeMetrics) bool) {
 	es.state.AssertMutable()
 	sort.SliceStable(*es.orig, func(i, j int) bool { return less(es.At(i), es.At(j)) })
 }
+
+func copyOrigScopeMetricsSlice(dest, src []*otlpmetrics.ScopeMetrics) []*otlpmetrics.ScopeMetrics {
+	if cap(dest) < len(src) {
+		dest = make([]*otlpmetrics.ScopeMetrics, len(src))
+		data := make([]otlpmetrics.ScopeMetrics, len(src))
+		for i := range src {
+			dest[i] = &data[i]
+		}
+	}
+	dest = dest[:len(src)]
+	for i := range src {
+		copyOrigScopeMetrics(dest[i], src[i])
+	}
+	return dest
+}
