diff --git a/pdata/pmetric/generated_scopemetrics.go b/pdata/pmetric/generated_scopemetrics.go
index 667bb8406..d13626f06 100644
--- a/pdata/pmetric/generated_scopemetrics.go
+++ b/pdata/pmetric/generated_scopemetrics.go
@@ -74,7 +74,11 @@ func (ms ScopeMetrics) Metrics() MetricSlice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms ScopeMetrics) CopyTo(dest ScopeMetrics) {
 	dest.state.AssertMutable()
-	ms.Scope().CopyTo(dest.Scope())
-	dest.SetSchemaUrl(ms.SchemaUrl())
-	ms.Metrics().CopyTo(dest.Metrics())
+	copyOrigScopeMetrics(dest.orig, ms.orig)
+}
+
+func copyOrigScopeMetrics(dest, src *otlpmetrics.ScopeMetrics) {
+	internal.CopyOrigInstrumentationScope(&dest.Scope, &src.Scope)
+	dest.SchemaUrl = src.SchemaUrl
+	dest.Metrics = copyOrigMetricSlice(dest.Metrics, src.Metrics)
 }
