diff --git a/exporter/exportertest/go.mod b/exporter/exportertest/go.mod
index 67439128e..7aeb13ba7 100644
--- a/exporter/exportertest/go.mod
+++ b/exporter/exportertest/go.mod
@@ -19,7 +19,7 @@ require (
 	go.opentelemetry.io/collector/pipeline v1.53.0
 	go.opentelemetry.io/collector/receiver v1.53.0
 	go.opentelemetry.io/collector/receiver/receivertest v0.147.0
-	google.golang.org/grpc v1.79.1
+	google.golang.org/grpc v1.79.2
 )
 
 require (
