diff --git a/confmap/provider/httpprovider/provider.go b/confmap/provider/httpprovider/provider.go
index 4c762adc2..6ed2b6d10 100644
--- a/confmap/provider/httpprovider/provider.go
+++ b/confmap/provider/httpprovider/provider.go
@@ -13,6 +13,17 @@ import (
 // This Provider supports "http" scheme.
 //
 // One example for HTTP URI is: http://localhost:3333/getConfig
+//
+// Deprecated: [v0.99.0] Use NewFactory instead.
 func NewWithSettings(set confmap.ProviderSettings) confmap.Provider {
 	return configurablehttpprovider.New(configurablehttpprovider.HTTPScheme, set)
 }
+
+// NewFactory returns a factory for a confmap.Provider that reads the configuration from a http server.
+//
+// This Provider supports "http" scheme.
+//
+// One example for HTTP URI is: http://localhost:3333/getConfig
+func NewFactory() confmap.ProviderFactory {
+	return confmap.NewProviderFactory(NewWithSettings)
+}
