diff --git a/service/internal/builders/receiver.go b/service/internal/builders/receiver.go
index 5831993d4..ceb78b8ef 100644
--- a/service/internal/builders/receiver.go
+++ b/service/internal/builders/receiver.go
@@ -44,8 +44,8 @@ func (b *ReceiverBuilder) CreateTraces(ctx context.Context, set receiver.Setting
 		return nil, fmt.Errorf("receiver factory not available for: %q", set.ID)
 	}
 
-	logStabilityLevel(set.Logger, f.TracesReceiverStability())
-	return f.CreateTracesReceiver(ctx, set, cfg, next)
+	logStabilityLevel(set.Logger, f.TracesStability())
+	return f.CreateTraces(ctx, set, cfg, next)
 }
 
 // CreateMetrics creates a Metrics receiver based on the settings and config.
@@ -63,8 +63,8 @@ func (b *ReceiverBuilder) CreateMetrics(ctx context.Context, set receiver.Settin
 		return nil, fmt.Errorf("receiver factory not available for: %q", set.ID)
 	}
 
-	logStabilityLevel(set.Logger, f.MetricsReceiverStability())
-	return f.CreateMetricsReceiver(ctx, set, cfg, next)
+	logStabilityLevel(set.Logger, f.MetricsStability())
+	return f.CreateMetrics(ctx, set, cfg, next)
 }
 
 // CreateLogs creates a Logs receiver based on the settings and config.
@@ -82,8 +82,8 @@ func (b *ReceiverBuilder) CreateLogs(ctx context.Context, set receiver.Settings,
 		return nil, fmt.Errorf("receiver factory not available for: %q", set.ID)
 	}
 
-	logStabilityLevel(set.Logger, f.LogsReceiverStability())
-	return f.CreateLogsReceiver(ctx, set, cfg, next)
+	logStabilityLevel(set.Logger, f.LogsStability())
+	return f.CreateLogs(ctx, set, cfg, next)
 }
 
 // CreateProfiles creates a Profiles receiver based on the settings and config.
@@ -106,8 +106,8 @@ func (b *ReceiverBuilder) CreateProfiles(ctx context.Context, set receiver.Setti
 		return nil, pipeline.ErrSignalNotSupported
 	}
 
-	logStabilityLevel(set.Logger, f.ProfilesReceiverStability())
-	return f.CreateProfilesReceiver(ctx, set, cfg, next)
+	logStabilityLevel(set.Logger, f.ProfilesStability())
+	return f.CreateProfiles(ctx, set, cfg, next)
 }
 
 func (b *ReceiverBuilder) Factory(componentType component.Type) component.Factory {
