diff --git a/config/configauth/go.mod b/config/configauth/go.mod
index 935359329..6e3b20e3a 100644
--- a/config/configauth/go.mod
+++ b/config/configauth/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v0.120.0
 	go.opentelemetry.io/collector/extension v0.120.0
-	go.opentelemetry.io/collector/extension/extensionauth v0.120.0
+	go.opentelemetry.io/collector/extension/extensionauth v0.0.0-20250227134758-6a98ee8b14a2
 	go.uber.org/goleak v1.3.0
 )
 
