diff --git a/exporter/exporterhelper/logs_batch.go b/exporter/exporterhelper/logs_batch.go
index 0043e323e..7da063aa8 100644
--- a/exporter/exporterhelper/logs_batch.go
+++ b/exporter/exporterhelper/logs_batch.go
@@ -7,19 +7,18 @@ import (
 	"context"
 	"errors"
 
-	"go.opentelemetry.io/collector/exporter/exporterbatcher"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/sizer"
 	"go.opentelemetry.io/collector/pdata/plog"
 )
 
 // MergeSplit splits and/or merges the provided logs request and the current request into one or more requests
 // conforming with the MaxSizeConfig.
-func (req *logsRequest) MergeSplit(_ context.Context, maxSize int, szt exporterbatcher.SizerType, r2 Request) ([]Request, error) {
+func (req *logsRequest) MergeSplit(_ context.Context, maxSize int, szt RequestSizerType, r2 Request) ([]Request, error) {
 	var sz sizer.LogsSizer
 	switch szt {
-	case exporterbatcher.SizerTypeItems:
+	case RequestSizerTypeItems:
 		sz = &sizer.LogsCountSizer{}
-	case exporterbatcher.SizerTypeBytes:
+	case RequestSizerTypeBytes:
 		sz = &sizer.LogsBytesSizer{}
 	default:
 		return nil, errors.New("unknown sizer type")
