diff --git a/pdata/pprofile/generated_profilesdictionary.go b/pdata/pprofile/generated_profilesdictionary.go
index fe1f650f6..f3d4d102e 100644
--- a/pdata/pprofile/generated_profilesdictionary.go
+++ b/pdata/pprofile/generated_profilesdictionary.go
@@ -33,7 +33,7 @@ func newProfilesDictionary(orig *otlpprofiles.ProfilesDictionary, state *interna
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewProfilesDictionary() ProfilesDictionary {
-	return newProfilesDictionary(internal.NewOrigProfilesDictionary(), internal.NewState())
+	return newProfilesDictionary(internal.NewProfilesDictionary(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
@@ -45,7 +45,7 @@ func (ms ProfilesDictionary) MoveTo(dest ProfilesDictionary) {
 	if ms.orig == dest.orig {
 		return
 	}
-	internal.DeleteOrigProfilesDictionary(dest.orig, false)
+	internal.DeleteProfilesDictionary(dest.orig, false)
 	*dest.orig, *ms.orig = *ms.orig, *dest.orig
 }
 
@@ -71,7 +71,7 @@ func (ms ProfilesDictionary) LinkTable() LinkSlice {
 
 // StringTable returns the StringTable associated with this ProfilesDictionary.
 func (ms ProfilesDictionary) StringTable() pcommon.StringSlice {
-	return pcommon.StringSlice(internal.NewStringSlice(&ms.orig.StringTable, ms.state))
+	return pcommon.StringSlice(internal.NewStringSliceWrapper(&ms.orig.StringTable, ms.state))
 }
 
 // AttributeTable returns the AttributeTable associated with this ProfilesDictionary.
@@ -87,5 +87,5 @@ func (ms ProfilesDictionary) StackTable() StackSlice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms ProfilesDictionary) CopyTo(dest ProfilesDictionary) {
 	dest.state.AssertMutable()
-	internal.CopyOrigProfilesDictionary(dest.orig, ms.orig)
+	internal.CopyProfilesDictionary(dest.orig, ms.orig)
 }
