diff --git a/otelcol/command_components.go b/otelcol/command_components.go
index 233882801..f33beb27d 100644
--- a/otelcol/command_components.go
+++ b/otelcol/command_components.go
@@ -21,9 +21,9 @@ type componentsOutput struct {
 	Extensions []component.Type
 }
 
-// newBuildSubCommand constructs a new cobra.Command sub command using the given CollectorSettings.
-func newBuildSubCommand(set CollectorSettings) *cobra.Command {
-	buildCmd := &cobra.Command{
+// newComponentsCommand constructs a new components command using the given CollectorSettings.
+func newComponentsCommand(set CollectorSettings) *cobra.Command {
+	return &cobra.Command{
 		Use:   "components",
 		Short: "Outputs available components in this collector distribution",
 		Args:  cobra.ExactArgs(0),
@@ -54,5 +54,4 @@ func newBuildSubCommand(set CollectorSettings) *cobra.Command {
 			return nil
 		},
 	}
-	return buildCmd
 }
