diff --git a/otelcol/command.go b/otelcol/command.go
index 9350b8b68..a9baff305 100644
--- a/otelcol/command.go
+++ b/otelcol/command.go
@@ -27,7 +27,7 @@ func NewCommand(set CollectorSettings) *cobra.Command {
 			return col.Run(cmd.Context())
 		},
 	}
-	rootCmd.AddCommand(newBuildSubCommand(set))
+	rootCmd.AddCommand(newComponentsCommand(set))
 	rootCmd.AddCommand(newValidateSubCommand(set, flagSet))
 	rootCmd.Flags().AddGoFlagSet(flagSet)
 	return rootCmd
