diff --git a/exporter/loggingexporter/internal/otlptext/metrics.go b/exporter/loggingexporter/internal/otlptext/metrics.go
index 6c7a41337..639474663 100644
--- a/exporter/loggingexporter/internal/otlptext/metrics.go
+++ b/exporter/loggingexporter/internal/otlptext/metrics.go
@@ -31,7 +31,7 @@ func (textMetricsMarshaler) MarshalMetrics(md pmetric.Metrics) ([]byte, error) {
 		buf.logEntry("ResourceMetrics #%d", i)
 		rm := rms.At(i)
 		buf.logEntry("Resource SchemaURL: %s", rm.SchemaUrl())
-		buf.logAttributes("Resource labels", rm.Resource().Attributes())
+		buf.logAttributes("Resource attributes", rm.Resource().Attributes())
 		ilms := rm.ScopeMetrics()
 		for j := 0; j < ilms.Len(); j++ {
 			buf.logEntry("ScopeMetrics #%d", j)
