diff --git a/component/componenttest/nop_host_test.go b/component/componenttest/nop_host_test.go
index aeb3a02c5..0856c6918 100644
--- a/component/componenttest/nop_host_test.go
+++ b/component/componenttest/nop_host_test.go
@@ -18,12 +18,19 @@ import (
 	"errors"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/component"
 )
 
 func TestNewNopHost(t *testing.T) {
-	got := NewNopHost()
-	require.IsType(t, &NopHost{}, got)
+	nh := NewNopHost()
+	require.NotNil(t, nh)
+	require.IsType(t, &NopHost{}, nh)
 
-	got.ReportFatalError(errors.New("TestError"))
+	nh.ReportFatalError(errors.New("TestError"))
+	assert.Nil(t, nh.GetExporters())
+	assert.Nil(t, nh.GetExtensions())
+	assert.Nil(t, nh.GetFactory(component.KindReceiver, "test"))
 }
