diff --git a/exporter/zipkinexporter/zipkin_test.go b/exporter/zipkinexporter/zipkin_test.go
index db8b27d2e..9f8c2f774 100644
--- a/exporter/zipkinexporter/zipkin_test.go
+++ b/exporter/zipkinexporter/zipkin_test.go
@@ -70,7 +70,7 @@ func TestZipkinExporter_roundtripJSON(t *testing.T) {
 	mzr := newMockZipkinReporter(cst.URL)
 
 	// Run the Zipkin receiver to "receive spans upload from a client application"
-	zexp := processor.NewTraceFanOutConnector([]consumer.TraceConsumerOld{tes})
+	zexp := processor.NewTraceFanOutConnectorOld([]consumer.TraceConsumerOld{tes})
 	addr := testutils.GetAvailableLocalAddress(t)
 	zi, err := zipkinreceiver.New("zipkin_receiver", addr, zexp)
 	assert.NoError(t, err)
@@ -300,7 +300,7 @@ func TestZipkinExporter_roundtripProto(t *testing.T) {
 	mzr.serializer = zipkinproto.SpanSerializer{}
 
 	// Run the Zipkin receiver to "receive spans upload from a client application"
-	zexp := processor.NewTraceFanOutConnector([]consumer.TraceConsumerOld{tes})
+	zexp := processor.NewTraceFanOutConnectorOld([]consumer.TraceConsumerOld{tes})
 	port := testutils.GetAvailablePort(t)
 	zi, err := zipkinreceiver.New(
 		"zipkin_receiver", fmt.Sprintf(":%d", port), zexp)
