diff --git a/pdata/pmetric/pmetricotlp/generated_exportresponse.go b/pdata/pmetric/pmetricotlp/generated_exportresponse.go
index 5592ce857..491e5259e 100644
--- a/pdata/pmetric/pmetricotlp/generated_exportresponse.go
+++ b/pdata/pmetric/pmetricotlp/generated_exportresponse.go
@@ -32,7 +32,7 @@ func newExportResponse(orig *otlpcollectormetrics.ExportMetricsServiceResponse,
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewExportResponse() ExportResponse {
-	return newExportResponse(internal.NewOrigExportMetricsServiceResponse(), internal.NewState())
+	return newExportResponse(internal.NewExportMetricsServiceResponse(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
@@ -44,7 +44,7 @@ func (ms ExportResponse) MoveTo(dest ExportResponse) {
 	if ms.orig == dest.orig {
 		return
 	}
-	internal.DeleteOrigExportMetricsServiceResponse(dest.orig, false)
+	internal.DeleteExportMetricsServiceResponse(dest.orig, false)
 	*dest.orig, *ms.orig = *ms.orig, *dest.orig
 }
 
@@ -56,5 +56,5 @@ func (ms ExportResponse) PartialSuccess() ExportPartialSuccess {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms ExportResponse) CopyTo(dest ExportResponse) {
 	dest.state.AssertMutable()
-	internal.CopyOrigExportMetricsServiceResponse(dest.orig, ms.orig)
+	internal.CopyExportMetricsServiceResponse(dest.orig, ms.orig)
 }
