diff --git a/translator/trace/jaeger/constants.go b/translator/trace/jaeger/constants.go
index adb17a6fa..65b8ddb6d 100644
--- a/translator/trace/jaeger/constants.go
+++ b/translator/trace/jaeger/constants.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
