diff --git a/internal/config/config.go b/internal/config/config.go
index f1e36e6a2..b71969ad5 100644
--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -26,6 +26,7 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 
+	"github.com/census-instrumentation/opencensus-service/consumer"
 	"github.com/census-instrumentation/opencensus-service/exporter/awsexporter"
 	"github.com/census-instrumentation/opencensus-service/exporter/datadogexporter"
 	"github.com/census-instrumentation/opencensus-service/exporter/honeycombexporter"
@@ -35,7 +36,6 @@ import (
 	"github.com/census-instrumentation/opencensus-service/exporter/prometheusexporter"
 	"github.com/census-instrumentation/opencensus-service/exporter/stackdriverexporter"
 	"github.com/census-instrumentation/opencensus-service/exporter/zipkinexporter"
-	"github.com/census-instrumentation/opencensus-service/processor"
 	"github.com/census-instrumentation/opencensus-service/receiver/opencensusreceiver"
 	"github.com/census-instrumentation/opencensus-service/receiver/prometheusreceiver"
 )
@@ -443,10 +443,10 @@ func eqLocalHost(host string) bool {
 //  + prometheus
 //  + aws-xray
 //  + honeycomb
-func ExportersFromViperConfig(logger *zap.Logger, v *viper.Viper) ([]processor.TraceDataProcessor, []processor.MetricsDataProcessor, []func() error, error) {
+func ExportersFromViperConfig(logger *zap.Logger, v *viper.Viper) ([]consumer.TraceConsumer, []consumer.MetricsConsumer, []func() error, error) {
 	parseFns := []struct {
 		name string
-		fn   func(*viper.Viper) ([]processor.TraceDataProcessor, []processor.MetricsDataProcessor, []func() error, error)
+		fn   func(*viper.Viper) ([]consumer.TraceConsumer, []consumer.MetricsConsumer, []func() error, error)
 	}{
 		{name: "datadog", fn: datadogexporter.DatadogTraceExportersFromViper},
 		{name: "stackdriver", fn: stackdriverexporter.StackdriverTraceExportersFromViper},
@@ -459,8 +459,8 @@ func ExportersFromViperConfig(logger *zap.Logger, v *viper.Viper) ([]processor.T
 		{name: "honeycomb", fn: honeycombexporter.HoneycombTraceExportersFromViper},
 	}
 
-	var traceExporters []processor.TraceDataProcessor
-	var metricsExporters []processor.MetricsDataProcessor
+	var traceExporters []consumer.TraceConsumer
+	var metricsExporters []consumer.MetricsConsumer
 	var doneFns []func() error
 	exportersViper := v.Sub("exporters")
 	if exportersViper == nil {
