diff --git a/cmd/mdatagen/internal/templates/config.go.tmpl b/cmd/mdatagen/internal/templates/config.go.tmpl
index 461061d83..a3f5bc4d6 100644
--- a/cmd/mdatagen/internal/templates/config.go.tmpl
+++ b/cmd/mdatagen/internal/templates/config.go.tmpl
@@ -33,36 +33,36 @@ func (ms *MetricConfig) Unmarshal(parser *confmap.Conf) error {
 
 {{- range $name, $metric := .Metrics }}
 {{- if hasAggregatableAttributes $metric.Attributes }}
-// {{ $name.Render }}AttributeKey specifies the key of an attribute for the {{ $name }} metric.
-type {{ $name.Render }}AttributeKey string
+// {{ $name.Render }}MetricAttributeKey specifies the key of an attribute for the {{ $name }} metric.
+type {{ $name.Render }}MetricAttributeKey string
 
 const (
 	{{- range $metric.Attributes }}
-	{{ $name.Render }}AttributeKey{{ .Render }} {{ $name.Render }}AttributeKey = "{{ (attributeInfo .).Name }}"
+	{{ $name.Render }}MetricAttributeKey{{ .Render }} {{ $name.Render }}MetricAttributeKey = "{{ (attributeInfo .).Name }}"
 	{{- end }}
 )
 {{- end }}
 
-// {{ $name.Render }}Config provides config for the {{ $name }} metric.
-type {{ $name.Render }}Config struct {
+// {{ $name.Render }}MetricConfig provides config for the {{ $name }} metric.
+type {{ $name.Render }}MetricConfig struct {
 	Enabled          bool `mapstructure:"enabled"`
 	enabledSetByUser bool
 	{{- if hasAggregatableAttributes $metric.Attributes }}
 
-	AggregationStrategy string                           `mapstructure:"aggregation_strategy"`
-	EnabledAttributes   []{{ $name.Render }}AttributeKey `mapstructure:"attributes"`
+	AggregationStrategy string                                 `mapstructure:"aggregation_strategy"`
+	EnabledAttributes   []{{ $name.Render }}MetricAttributeKey `mapstructure:"attributes"`
 	{{- end }}
 }
 
-func (ms *{{ $name.Render }}Config) Unmarshal(parser *confmap.Conf) error {
+func (ms *{{ $name.Render }}MetricConfig) Unmarshal(parser *confmap.Conf) error {
 	{{- template "metricUnmarshal" "ms" }}
 }
 
 {{ if hasAggregatableAttributes $metric.Attributes -}}
-func (ms *{{ $name.Render }}Config) Validate() error {
+func (ms *{{ $name.Render }}MetricConfig) Validate() error {
 	for _, val := range ms.EnabledAttributes {
 		switch val {
-		case {{ range $index, $element := $metric.Attributes -}}{{ if $index }}, {{ end }}{{ $name.Render }}AttributeKey{{ $element.Render }}{{ end }}:
+		case {{ range $index, $element := $metric.Attributes -}}{{ if $index }}, {{ end }}{{ $name.Render }}MetricAttributeKey{{ $element.Render }}{{ end }}:
 		default:
 			return fmt.Errorf("metric {{ $name }} doesn't have an attribute %v, valid attributes: [{{ range $index, $element := $metric.Attributes -}}{{ if $index }}, {{ end }}{{ (attributeInfo $element).Name }}{{ end }}]", val)
 		}
@@ -70,7 +70,7 @@ func (ms *{{ $name.Render }}Config) Validate() error {
 
 	{{- range $element := $metric.Attributes }}
 	{{- if (attributeInfo $element).IsRequired }}
-	if !slices.Contains(ms.EnabledAttributes, {{ $name.Render }}AttributeKey{{ $element.Render }}) {
+	if !slices.Contains(ms.EnabledAttributes, {{ $name.Render }}MetricAttributeKey{{ $element.Render }}) {
 		return fmt.Errorf("{{ (attributeInfo $element).Name }} is a required attribute for {{ $name }} metric and must be included")
 	}
 	{{- end }}
@@ -91,7 +91,7 @@ func (ms *{{ $name.Render }}Config) Validate() error {
 // MetricsConfig provides config for {{ .Type }} metrics.
 type MetricsConfig struct {
 	{{- range $name, $metric := .Metrics }}
-	{{ $name.Render }} {{ if $reag }}{{ $name.Render }}Config{{ else }}MetricConfig{{ end }} `mapstructure:"{{ $name }}"`
+	{{ $name.Render }} {{ if $reag }}{{ $name.Render }}{{ end }}MetricConfig `mapstructure:"{{ $name }}"`
 	{{- end }}
 }
 
@@ -99,11 +99,11 @@ func DefaultMetricsConfig() MetricsConfig {
 	return MetricsConfig{
 		{{- range $name, $metric := .Metrics }}
 		{{- $metricReag := and $reag (hasAggregatableAttributes $metric.Attributes) }}
-		{{ $name.Render }}: {{ if $reag }}{{ $name.Render }}{{ else }}Metric{{ end }}Config{
+		{{ $name.Render }}: {{ if $reag }}{{ $name.Render }}{{ end }}MetricConfig{
 			Enabled: {{ $metric.Enabled }},
 			{{- if $metricReag }}
 			AggregationStrategy: AggregationStrategy{{ if eq $metric.Data.Type "Sum" }}Sum{{ else }}Avg{{ end }},
-			EnabledAttributes:   []{{ $name.Render }}AttributeKey{ {{- range $element := $metric.Attributes -}}{{- if (attributeInfo $element).IsNotOptIn }}{{ $name.Render }}AttributeKey{{ $element.Render }}, {{ end }}{{- end -}} },
+			EnabledAttributes:   []{{ $name.Render }}MetricAttributeKey{ {{- range $element := $metric.Attributes -}}{{- if (attributeInfo $element).IsNotOptIn }}{{ $name.Render }}MetricAttributeKey{{ $element.Render }}, {{ end }}{{- end -}} },
 			{{- end }}
 		},
 		{{- end }}
