diff --git a/service/service.go b/service/service.go
index 32f158b41..1cde310a5 100644
--- a/service/service.go
+++ b/service/service.go
@@ -164,9 +164,10 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 
 	telFactory := telemetry.NewFactory()
 	telset := telemetry.Settings{
-		BuildInfo:  set.BuildInfo,
-		ZapOptions: set.LoggingOptions,
-		SDK:        &sdk,
+		AsyncErrorChannel: set.AsyncErrorChannel,
+		BuildInfo:         set.BuildInfo,
+		ZapOptions:        set.LoggingOptions,
+		SDK:               &sdk,
 	}
 
 	logger, lp, err := telFactory.CreateLogger(ctx, telset, &cfg.Telemetry)
