diff --git a/model/otlpgrpc/metrics.go b/model/otlpgrpc/metrics.go
index 84097a6f5..7ecc8904f 100644
--- a/model/otlpgrpc/metrics.go
+++ b/model/otlpgrpc/metrics.go
@@ -19,8 +19,8 @@ import (
 
 	"google.golang.org/grpc"
 
-	otlpcollectormetrics "go.opentelemetry.io/collector/internal/data/protogen/collector/metrics/v1"
 	"go.opentelemetry.io/collector/model/internal"
+	otlpcollectormetrics "go.opentelemetry.io/collector/model/internal/data/protogen/collector/metrics/v1"
 	"go.opentelemetry.io/collector/model/pdata"
 )
 
