diff --git a/confmap/provider/internal/provider.go b/confmap/provider/internal/provider.go
index a1f62c5a0..b8ff514be 100644
--- a/confmap/provider/internal/provider.go
+++ b/confmap/provider/internal/provider.go
@@ -23,10 +23,10 @@ import (
 // NewRetrievedFromYAML returns a new Retrieved instance that contains the deserialized data from the yaml bytes.
 // * yamlBytes the yaml bytes that will be deserialized.
 // * opts specifies options associated with this Retrieved value, such as CloseFunc.
-func NewRetrievedFromYAML(yamlBytes []byte, opts ...confmap.RetrievedOption) (confmap.Retrieved, error) {
+func NewRetrievedFromYAML(yamlBytes []byte, opts ...confmap.RetrievedOption) (*confmap.Retrieved, error) {
 	var rawConf interface{}
 	if err := yaml.Unmarshal(yamlBytes, &rawConf); err != nil {
-		return confmap.Retrieved{}, err
+		return nil, err
 	}
 	return confmap.NewRetrieved(rawConf, opts...)
 }
