diff --git a/confmap/confmap.go b/confmap/confmap.go
index 39bbc53cc..f24d12de1 100644
--- a/confmap/confmap.go
+++ b/confmap/confmap.go
@@ -192,6 +192,7 @@ func decodeConfig(m *Conf, result any, errorUnused bool, skipTopLevelUnmarshaler
 func encoderConfig(rawVal any) *encoder.EncoderConfig {
 	return &encoder.EncoderConfig{
 		EncodeHook: mapstructure.ComposeDecodeHookFunc(
+			encoder.YamlMarshalerHookFunc(),
 			encoder.TextMarshalerHookFunc(),
 			marshalerHookFunc(rawVal),
 		),
