diff --git a/service/service_test.go b/service/service_test.go
index 4e74ebfd5..8d0000db7 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -16,7 +16,6 @@ import (
 	"time"
 
 	"github.com/prometheus/common/expfmt"
-	"github.com/prometheus/common/model"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	config "go.opentelemetry.io/contrib/otelconf/v0.3.0"
@@ -700,7 +699,7 @@ func assertMetrics(t *testing.T, metricsAddr string, expectedLabels map[string]l
 	})
 	reader := bufio.NewReader(resp.Body)
 
-	parser := expfmt.NewTextParser(model.UTF8Validation)
+	var parser expfmt.TextParser
 	parsed, err := parser.TextToMetricFamilies(reader)
 	require.NoError(t, err)
 
