diff --git a/testbed/tests/scenarios.go b/testbed/tests/scenarios.go
index 9e6366ec0..37dbda92b 100644
--- a/testbed/tests/scenarios.go
+++ b/testbed/tests/scenarios.go
@@ -32,6 +32,7 @@ import (
 // createConfigFile creates a collector config file that corresponds to the
 // sender and receiver used in the test and returns the config file name.
 func createConfigFile(
+	t *testing.T,
 	sender testbed.DataSender, // Sender to send test data.
 	receiver testbed.DataReceiver, // Receiver to receive test data.
 	resultDir string, // Directory to write config file to.
@@ -128,14 +129,20 @@ service:
 	// Write the config string to a temporary file.
 	file, err := ioutil.TempFile("", "agent*.yaml")
 	if err != nil {
-		fmt.Print(err)
+		t.Error(err)
 		return ""
 	}
-	defer file.Close()
+
+	defer func() {
+		err := file.Close()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
 
 	_, err = file.WriteString(config)
 	if err != nil {
-		fmt.Print(err)
+		t.Error(err)
 		return ""
 	}
 
@@ -156,7 +163,7 @@ func Scenario10kItemsPerSecond(
 		t.Fatal(err)
 	}
 
-	configFile := createConfigFile(sender, receiver, resultDir, nil)
+	configFile := createConfigFile(t, sender, receiver, resultDir, nil)
 	defer os.Remove(configFile)
 
 	if configFile == "" {
