diff --git a/service/internal/configunmarshaler/defaultunmarshaler.go b/service/internal/configunmarshaler/defaultunmarshaler.go
index 1fe5ccc46..c29f6e7cb 100644
--- a/service/internal/configunmarshaler/defaultunmarshaler.go
+++ b/service/internal/configunmarshaler/defaultunmarshaler.go
@@ -199,7 +199,7 @@ func unmarshalService(srvRaw map[string]interface{}) (config.Service, error) {
 
 func defaultServiceTelemetryMetricsSettings() telemetry.MetricsConfig {
 	return telemetry.MetricsConfig{
-		Level:   configtelemetry.LevelBasic, //nolint:staticcheck
+		Level:   configtelemetry.LevelBasic,
 		Address: ":8888",
 	}
 }
