diff --git a/pdata/go.mod b/pdata/go.mod
index de7058b47..4d6e1c5b7 100644
--- a/pdata/go.mod
+++ b/pdata/go.mod
@@ -7,8 +7,8 @@ require (
 	github.com/json-iterator/go v1.1.12
 	github.com/stretchr/testify v1.8.4
 	go.uber.org/multierr v1.11.0
-	google.golang.org/grpc v1.56.0
-	google.golang.org/protobuf v1.30.0
+	google.golang.org/grpc v1.56.1
+	google.golang.org/protobuf v1.31.0
 )
 
 require (
