diff --git a/processor/probabilisticsamplerprocessor/probabilisticsampler.go b/processor/probabilisticsamplerprocessor/probabilisticsampler.go
index 08a81cb5b..69cc366b9 100644
--- a/processor/probabilisticsamplerprocessor/probabilisticsampler.go
+++ b/processor/probabilisticsamplerprocessor/probabilisticsampler.go
@@ -16,11 +16,8 @@ package probabilisticsamplerprocessor
 
 import (
 	"context"
-	"errors"
-	"fmt"
 
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
-	"github.com/spf13/viper"
 
 	"github.com/open-telemetry/opentelemetry-service/consumer"
 	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
@@ -29,30 +26,12 @@ import (
 )
 
 const (
-	// The constants below are tags used to read the configuration via viper.
-	samplingPercentageCfgTag = "sampling-percentage"
-	hashSeedCfgTag           = "hash-seed"
-
 	// The constants help translate user friendly percentages to numbers direct used in sampling.
 	numHashBuckets        = 0x4000 // Using a power of 2 to avoid division.
 	bitMaskHashBuckets    = numHashBuckets - 1
 	percentageScaleFactor = numHashBuckets / 100.0
 )
 
-// InitFromViper updates TraceSampler config according to the viper configuration.
-func (tsc *Config) InitFromViper(v *viper.Viper) (*Config, error) {
-	if v == nil {
-		return nil, errors.New("v is nil")
-	}
-	if err := v.UnmarshalKey(samplingPercentageCfgTag, &tsc.SamplingPercentage); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal %q: %v", samplingPercentageCfgTag, err)
-	}
-	if err := v.UnmarshalKey(hashSeedCfgTag, &tsc.HashSeed); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal %q: %v", hashSeedCfgTag, err)
-	}
-	return tsc, nil
-}
-
 type tracesamplerprocessor struct {
 	nextConsumer       consumer.TraceConsumer
 	scaledSamplingRate uint32
