diff --git a/exporter/loggingexporter/factory_test.go b/exporter/loggingexporter/factory_test.go
index 78ed3381a..3ddaa4dc2 100644
--- a/exporter/loggingexporter/factory_test.go
+++ b/exporter/loggingexporter/factory_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
