diff --git a/exporter/nopexporter/go.mod b/exporter/nopexporter/go.mod
index 28e5db2c6..fafa01dda 100644
--- a/exporter/nopexporter/go.mod
+++ b/exporter/nopexporter/go.mod
@@ -10,7 +10,9 @@ require (
 	go.opentelemetry.io/collector/consumer/consumertest v0.142.0
 	go.opentelemetry.io/collector/exporter v1.48.0
 	go.opentelemetry.io/collector/exporter/exportertest v0.142.0
+	go.opentelemetry.io/collector/exporter/xexporter v0.142.0
 	go.opentelemetry.io/collector/pdata v1.48.0
+	go.opentelemetry.io/collector/pdata/pprofile v0.142.0
 	go.uber.org/goleak v1.3.0
 )
 
@@ -36,9 +38,7 @@ require (
 	go.opentelemetry.io/collector/consumer v1.48.0 // indirect
 	go.opentelemetry.io/collector/consumer/consumererror v0.142.0 // indirect
 	go.opentelemetry.io/collector/consumer/xconsumer v0.142.0 // indirect
-	go.opentelemetry.io/collector/exporter/xexporter v0.142.0 // indirect
 	go.opentelemetry.io/collector/featuregate v1.48.0 // indirect
-	go.opentelemetry.io/collector/pdata/pprofile v0.142.0 // indirect
 	go.opentelemetry.io/collector/pipeline v1.48.0 // indirect
 	go.opentelemetry.io/collector/receiver v1.48.0 // indirect
 	go.opentelemetry.io/collector/receiver/receivertest v0.142.0 // indirect
