diff --git a/service/graph.go b/service/graph.go
index a0014cd1c..5c84efc66 100644
--- a/service/graph.go
+++ b/service/graph.go
@@ -16,6 +16,7 @@ package service // import "go.opentelemetry.io/collector/service"
 
 import (
 	"context"
+	"errors"
 	"net/http"
 
 	"go.uber.org/multierr"
@@ -32,6 +33,11 @@ type pipelinesGraph struct {
 	componentGraph *simple.DirectedGraph
 }
 
+func buildPipelinesGraph(_ context.Context, _ pipelinesSettings) (pipelines, error) {
+	err := errors.New("not yet implemented")
+	return &pipelinesGraph{componentGraph: simple.NewDirectedGraph()}, err
+}
+
 func (g *pipelinesGraph) StartAll(ctx context.Context, host component.Host) error {
 	nodes, err := topo.Sort(g.componentGraph)
 	if err != nil {
