diff --git a/pdata/ptrace/ptraceotlp/grpc.go b/pdata/ptrace/ptraceotlp/grpc.go
index 5479976ab..1314a2e15 100644
--- a/pdata/ptrace/ptraceotlp/grpc.go
+++ b/pdata/ptrace/ptraceotlp/grpc.go
@@ -11,8 +11,7 @@ import (
 	"google.golang.org/grpc/status"
 
 	"go.opentelemetry.io/collector/pdata/internal"
-	otlpcollectortrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1"
-	_ "go.opentelemetry.io/collector/pdata/internal/grpcencoding" // enforces custom gRPC encoding to be loaded.
+	"go.opentelemetry.io/collector/pdata/internal/otelgrpc"
 	"go.opentelemetry.io/collector/pdata/internal/otlp"
 )
 
@@ -32,11 +31,11 @@ type GRPCClient interface {
 
 // NewGRPCClient returns a new GRPCClient connected using the given connection.
 func NewGRPCClient(cc *grpc.ClientConn) GRPCClient {
-	return &grpcClient{rawClient: otlpcollectortrace.NewTraceServiceClient(cc)}
+	return &grpcClient{rawClient: otelgrpc.NewTraceServiceClient(cc)}
 }
 
 type grpcClient struct {
-	rawClient otlpcollectortrace.TraceServiceClient
+	rawClient otelgrpc.TraceServiceClient
 }
 
 // Export implements the Client interface.
@@ -76,14 +75,14 @@ func (*UnimplementedGRPCServer) unexported() {}
 
 // RegisterGRPCServer registers the GRPCServer to the grpc.Server.
 func RegisterGRPCServer(s *grpc.Server, srv GRPCServer) {
-	otlpcollectortrace.RegisterTraceServiceServer(s, &rawTracesServer{srv: srv})
+	otelgrpc.RegisterTraceServiceServer(s, &rawTracesServer{srv: srv})
 }
 
 type rawTracesServer struct {
 	srv GRPCServer
 }
 
-func (s rawTracesServer) Export(ctx context.Context, request *otlpcollectortrace.ExportTraceServiceRequest) (*otlpcollectortrace.ExportTraceServiceResponse, error) {
+func (s rawTracesServer) Export(ctx context.Context, request *internal.ExportTraceServiceRequest) (*internal.ExportTraceServiceResponse, error) {
 	otlp.MigrateTraces(request.ResourceSpans)
 	rsp, err := s.srv.Export(ctx, ExportRequest{orig: request, state: internal.NewState()})
 	return rsp.orig, err
