diff --git a/exporter/exporterhelper/batch_sender_test.go b/exporter/exporterhelper/batch_sender_test.go
index 5a5a175bd..4651e9c61 100644
--- a/exporter/exporterhelper/batch_sender_test.go
+++ b/exporter/exporterhelper/batch_sender_test.go
@@ -136,7 +136,7 @@ func TestBatchSender_BatchExportError(t *testing.T) {
 			assert.Eventually(t, func() bool {
 				return sink.requestsCount.Load() == tt.expectedRequests &&
 					sink.itemsCount.Load() == tt.expectedItems &&
-					be.batchSender.(*batchSender).activeRequests.Load() == uint64(0) &&
+					be.batchSender.(*batchSender).activeRequests.Load() == 0 &&
 					be.queueSender.(*queueSender).queue.Size() == 0
 			}, 100*time.Millisecond, 10*time.Millisecond)
 		})
@@ -270,27 +270,101 @@ func TestBatchSender_PostShutdown(t *testing.T) {
 }
 
 func TestBatchSender_ConcurrencyLimitReached(t *testing.T) {
-	qCfg := exporterqueue.NewDefaultConfig()
-	qCfg.NumConsumers = 2
-	be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
-		WithBatcher(exporterbatcher.NewDefaultConfig(), WithRequestBatchFuncs(fakeBatchMergeFunc, fakeBatchMergeSplitFunc)),
-		WithRequestQueue(qCfg, exporterqueue.NewMemoryQueueFactory[Request]()))
-	require.NotNil(t, be)
-	require.NoError(t, err)
-	assert.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))
-	t.Cleanup(func() {
-		assert.NoError(t, be.Shutdown(context.Background()))
-	})
 
-	sink := newFakeRequestSink()
-	assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 8, sink: sink}))
+	tests := []struct {
+		name             string
+		batcherCfg       exporterbatcher.Config
+		expectedRequests uint64
+		expectedItems    uint64
+	}{
+		{
+			name: "merge_only",
+			batcherCfg: func() exporterbatcher.Config {
+				cfg := exporterbatcher.NewDefaultConfig()
+				cfg.FlushTimeout = 20 * time.Millisecond
+				return cfg
+			}(),
+			expectedRequests: 6,
+			expectedItems:    51,
+		},
+		{
+			name: "merge_without_split_triggered",
+			batcherCfg: func() exporterbatcher.Config {
+				cfg := exporterbatcher.NewDefaultConfig()
+				cfg.FlushTimeout = 20 * time.Millisecond
+				cfg.MaxSizeItems = 200
+				return cfg
+			}(),
+			expectedRequests: 6,
+			expectedItems:    51,
+		},
+		{
+			name: "merge_with_split_triggered",
+			batcherCfg: func() exporterbatcher.Config {
+				cfg := exporterbatcher.NewDefaultConfig()
+				cfg.FlushTimeout = 50 * time.Millisecond
+				cfg.MaxSizeItems = 10
+				return cfg
+			}(),
+			expectedRequests: 8,
+			expectedItems:    51,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			qCfg := exporterqueue.NewDefaultConfig()
+			qCfg.NumConsumers = 2
+			be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
+				WithBatcher(tt.batcherCfg, WithRequestBatchFuncs(fakeBatchMergeFunc, fakeBatchMergeSplitFunc)),
+				WithRequestQueue(qCfg, exporterqueue.NewMemoryQueueFactory[Request]()))
+			require.NotNil(t, be)
+			require.NoError(t, err)
+			assert.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))
+			t.Cleanup(func() {
+				assert.NoError(t, be.Shutdown(context.Background()))
+			})
+
+			sink := newFakeRequestSink()
+			// the 1st and 2nd request should be flushed in the same batched request by max concurrency limit.
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 2, sink: sink}))
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 2, sink: sink}))
 
-	// the second request should be sent by reaching max concurrency limit.
-	assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 8, sink: sink}))
+			assert.Eventually(t, func() bool {
+				return sink.requestsCount.Load() == 1 && sink.itemsCount.Load() == 4
+			}, 100*time.Millisecond, 10*time.Millisecond)
 
-	assert.Eventually(t, func() bool {
-		return sink.requestsCount.Load() == 1 && sink.itemsCount.Load() == 16
-	}, 100*time.Millisecond, 10*time.Millisecond)
+			// the 3rd request should be flushed by itself due to flush interval
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 2, sink: sink}))
+			assert.Eventually(t, func() bool {
+				return sink.requestsCount.Load() == 2 && sink.itemsCount.Load() == 6
+			}, 100*time.Millisecond, 10*time.Millisecond)
+
+			// the 4th and 5th request should be flushed in the same batched request by max concurrency limit.
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 2, sink: sink}))
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 2, sink: sink}))
+			assert.Eventually(t, func() bool {
+				return sink.requestsCount.Load() == 3 && sink.itemsCount.Load() == 10
+			}, 100*time.Millisecond, 10*time.Millisecond)
+
+			// do it a few more times to ensure it produces the correct batch size regardless of goroutine scheduling.
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 5, sink: sink}))
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 6, sink: sink}))
+			if tt.batcherCfg.MaxSizeItems == 10 {
+				// in case of MaxSizeItems=10, wait for the leftover request to send
+				assert.Eventually(t, func() bool {
+					return sink.requestsCount.Load() == 5 && sink.itemsCount.Load() == 21
+				}, 50*time.Millisecond, 10*time.Millisecond)
+			}
+
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 6, sink: sink}))
+			assert.NoError(t, be.send(context.Background(), &fakeRequest{items: 20, sink: sink}))
+			assert.Eventually(t, func() bool {
+				return sink.requestsCount.Load() == tt.expectedRequests && sink.itemsCount.Load() == tt.expectedItems
+			}, 100*time.Millisecond, 10*time.Millisecond)
+		})
+	}
 }
 
 func TestBatchSender_BatchBlocking(t *testing.T) {
