diff --git a/service/go.mod b/service/go.mod
index e340b4c59..33c8b23ef 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/prometheus/client_model v0.6.1
 	github.com/prometheus/common v0.63.0
-	github.com/shirou/gopsutil/v4 v4.25.2
+	github.com/shirou/gopsutil/v4 v4.25.1
 	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector v0.122.1
 	go.opentelemetry.io/collector/component v1.28.1
