diff --git a/receiver/otlpreceiver/config_test.go b/receiver/otlpreceiver/config_test.go
index c9efcfd90..410ae09c1 100644
--- a/receiver/otlpreceiver/config_test.go
+++ b/receiver/otlpreceiver/config_test.go
@@ -44,24 +44,21 @@ func TestLoadConfig(t *testing.T) {
 
 	assert.Equal(t, len(cfg.Receivers), 10)
 
-	assert.Equal(t, cfg.Receivers["otlp"], factory.CreateDefaultConfig())
+	assert.Equal(t, cfg.Receivers[config.NewID(typeStr)], factory.CreateDefaultConfig())
 
 	defaultOnlyGRPC := factory.CreateDefaultConfig().(*Config)
-	defaultOnlyGRPC.SetName("otlp/only_grpc")
+	defaultOnlyGRPC.SetIDName("only_grpc")
 	defaultOnlyGRPC.HTTP = nil
-	assert.Equal(t, cfg.Receivers["otlp/only_grpc"], defaultOnlyGRPC)
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "only_grpc")], defaultOnlyGRPC)
 
 	defaultOnlyHTTP := factory.CreateDefaultConfig().(*Config)
-	defaultOnlyHTTP.SetName("otlp/only_http")
+	defaultOnlyHTTP.SetIDName("only_http")
 	defaultOnlyHTTP.GRPC = nil
-	assert.Equal(t, cfg.Receivers["otlp/only_http"], defaultOnlyHTTP)
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "only_http")], defaultOnlyHTTP)
 
-	assert.Equal(t, cfg.Receivers["otlp/customname"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "customname")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/customname",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "customname")),
 			Protocols: Protocols{
 				GRPC: &configgrpc.GRPCServerSettings{
 					NetAddr: confignet.NetAddr{
@@ -73,12 +70,9 @@ func TestLoadConfig(t *testing.T) {
 			},
 		})
 
-	assert.Equal(t, cfg.Receivers["otlp/keepalive"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "keepalive")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/keepalive",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "keepalive")),
 			Protocols: Protocols{
 				GRPC: &configgrpc.GRPCServerSettings{
 					NetAddr: confignet.NetAddr{
@@ -103,12 +97,9 @@ func TestLoadConfig(t *testing.T) {
 			},
 		})
 
-	assert.Equal(t, cfg.Receivers["otlp/msg-size-conc-connect-max-idle"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "msg-size-conc-connect-max-idle")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/msg-size-conc-connect-max-idle",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "msg-size-conc-connect-max-idle")),
 			Protocols: Protocols{
 				GRPC: &configgrpc.GRPCServerSettings{
 					NetAddr: confignet.NetAddr{
@@ -130,12 +121,9 @@ func TestLoadConfig(t *testing.T) {
 
 	// NOTE: Once the config loader checks for the files existence, this test may fail and require
 	// 	use of fake cert/key for test purposes.
-	assert.Equal(t, cfg.Receivers["otlp/tlscredentials"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "tlscredentials")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/tlscredentials",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "tlscredentials")),
 			Protocols: Protocols{
 				GRPC: &configgrpc.GRPCServerSettings{
 					NetAddr: confignet.NetAddr{
@@ -162,12 +150,9 @@ func TestLoadConfig(t *testing.T) {
 			},
 		})
 
-	assert.Equal(t, cfg.Receivers["otlp/cors"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "cors")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/cors",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "cors")),
 			Protocols: Protocols{
 				HTTP: &confighttp.HTTPServerSettings{
 					Endpoint:    "0.0.0.0:55681",
@@ -176,12 +161,9 @@ func TestLoadConfig(t *testing.T) {
 			},
 		})
 
-	assert.Equal(t, cfg.Receivers["otlp/corsheader"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "corsheader")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/corsheader",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "corsheader")),
 			Protocols: Protocols{
 				HTTP: &confighttp.HTTPServerSettings{
 					Endpoint:    "0.0.0.0:55681",
@@ -191,12 +173,9 @@ func TestLoadConfig(t *testing.T) {
 			},
 		})
 
-	assert.Equal(t, cfg.Receivers["otlp/uds"],
+	assert.Equal(t, cfg.Receivers[config.NewIDWithName(typeStr, "uds")],
 		&Config{
-			ReceiverSettings: config.ReceiverSettings{
-				TypeVal: typeStr,
-				NameVal: "otlp/uds",
-			},
+			ReceiverSettings: config.NewReceiverSettings(config.NewIDWithName(typeStr, "uds")),
 			Protocols: Protocols{
 				GRPC: &configgrpc.GRPCServerSettings{
 					NetAddr: confignet.NetAddr{
