diff --git a/exporter/exporterprofiles/go.mod b/exporter/exporterprofiles/go.mod
index 86817aac2..e659701ee 100644
--- a/exporter/exporterprofiles/go.mod
+++ b/exporter/exporterprofiles/go.mod
@@ -41,8 +41,6 @@ replace go.opentelemetry.io/collector/extension => ../../extension
 
 replace go.opentelemetry.io/collector/extension/experimental/storage => ../../extension/experimental/storage
 
-replace go.opentelemetry.io/collector => ../..
-
 replace go.opentelemetry.io/collector/pdata/pprofile => ../../pdata/pprofile
 
 replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/configtelemetry
@@ -51,10 +49,6 @@ replace go.opentelemetry.io/collector/config/configretry => ../../config/configr
 
 replace go.opentelemetry.io/collector/pdata/testdata => ../../pdata/testdata
 
-replace go.opentelemetry.io/collector/featuregate => ../../featuregate
-
-replace go.opentelemetry.io/collector/confmap => ../../confmap
-
 replace go.opentelemetry.io/collector/component => ../../component
 
 replace go.opentelemetry.io/collector/pdata => ../../pdata
@@ -67,6 +61,4 @@ replace go.opentelemetry.io/collector/consumer => ../../consumer
 
 replace go.opentelemetry.io/collector/exporter => ../
 
-replace go.opentelemetry.io/collector/component/componentstatus => ../../component/componentstatus
-
 replace go.opentelemetry.io/collector/receiver/receiverprofiles => ../../receiver/receiverprofiles
