diff --git a/processor/attributesprocessor/attributes.go b/processor/attributesprocessor/attributes.go
index 9d1cd6784..f81d10e3b 100644
--- a/processor/attributesprocessor/attributes.go
+++ b/processor/attributesprocessor/attributes.go
@@ -19,10 +19,10 @@ import (
 
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 
-	"github.com/open-telemetry/opentelemetry-service/consumer"
-	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
-	"github.com/open-telemetry/opentelemetry-service/oterr"
-	"github.com/open-telemetry/opentelemetry-service/processor"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+	"github.com/open-telemetry/opentelemetry-collector/processor"
 )
 
 type attributesProcessor struct {
@@ -42,7 +42,7 @@ type attributesConfig struct {
 type attributeAction struct {
 	Key           string
 	FromAttribute string
-	// TODO https://github.com/open-telemetry/opentelemetry-service/issues/296
+	// TODO https://github.com/open-telemetry/opentelemetry-collector/issues/296
 	// Do benchmark testing between having action be of type string vs integer.
 	// The reason is attributes processor will most likely be commonly used
 	// and could impact performance.
@@ -101,7 +101,7 @@ func (a *attributesProcessor) ConsumeTraceData(ctx context.Context, td consumerd
 
 		for _, action := range a.config.actions {
 
-			// TODO https://github.com/open-telemetry/opentelemetry-service/issues/296
+			// TODO https://github.com/open-telemetry/opentelemetry-collector/issues/296
 			// Do benchmark testing between having action be of type string vs integer.
 			// The reason is attributes processor will most likely be commonly used
 			// and could impact performance.
