diff --git a/connector/traces_router_test.go b/connector/traces_router_test.go
index 125d159ca..2034b306d 100644
--- a/connector/traces_router_test.go
+++ b/connector/traces_router_test.go
@@ -49,7 +49,7 @@ func fuzzTraces(numIDs, numCons, numTraces int) func(*testing.T) {
 
 		// If any consumer is mutating, the router must report mutating
 		for i := 0; i < numCons; i++ {
-			allIDs = append(allIDs, pipeline.MustNewIDWithName("sink", strconv.Itoa(numCons)))
+			allIDs = append(allIDs, pipeline.NewIDWithName(pipeline.SignalTraces, "sink_"+strconv.Itoa(numCons)))
 			// Random chance for each consumer to be mutating
 			if (numCons+numTraces+i)%4 == 0 {
 				allCons = append(allCons, &mutatingTracesSink{TracesSink: new(consumertest.TracesSink)})
@@ -109,8 +109,8 @@ func TestTracesRouterConsumer(t *testing.T) {
 	ctx := context.Background()
 	td := testdata.GenerateTraces(1)
 
-	fooID := pipeline.MustNewID("foo")
-	barID := pipeline.MustNewID("bar")
+	fooID := pipeline.NewIDWithName(pipeline.SignalTraces, "foo")
+	barID := pipeline.NewIDWithName(pipeline.SignalTraces, "bar")
 
 	foo := new(consumertest.TracesSink)
 	bar := new(consumertest.TracesSink)
@@ -151,7 +151,7 @@ func TestTracesRouterConsumer(t *testing.T) {
 	assert.Nil(t, none)
 	require.Error(t, err)
 
-	fake, err := r.Consumer(pipeline.MustNewID("fake"))
+	fake, err := r.Consumer(pipeline.NewIDWithName(pipeline.SignalTraces, "fake"))
 	assert.Nil(t, fake)
 	assert.Error(t, err)
 }
