diff --git a/pdata/internal/generated_wrapper_histogram_test.go b/pdata/internal/generated_wrapper_histogram_test.go
index e1f06b667..191e02ee7 100644
--- a/pdata/internal/generated_wrapper_histogram_test.go
+++ b/pdata/internal/generated_wrapper_histogram_test.go
@@ -23,7 +23,7 @@ import (
 func TestCopyOrigHistogram(t *testing.T) {
 	for name, src := range genTestEncodingValuesHistogram() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -52,7 +52,7 @@ func TestMarshalAndUnmarshalJSONOrigHistogramUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalJSONOrigHistogram(t *testing.T) {
 	for name, src := range genTestEncodingValuesHistogram() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -96,7 +96,7 @@ func TestMarshalAndUnmarshalProtoOrigHistogramUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalProtoOrigHistogram(t *testing.T) {
 	for name, src := range genTestEncodingValuesHistogram() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
