diff --git a/pdata/internal/generated_wrapper_link_test.go b/pdata/internal/generated_wrapper_link_test.go
index bb166eef1..c4db1ba88 100644
--- a/pdata/internal/generated_wrapper_link_test.go
+++ b/pdata/internal/generated_wrapper_link_test.go
@@ -23,7 +23,7 @@ import (
 func TestCopyOrigLink(t *testing.T) {
 	for name, src := range genTestEncodingValuesLink() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -52,7 +52,7 @@ func TestMarshalAndUnmarshalJSONOrigLinkUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalJSONOrigLink(t *testing.T) {
 	for name, src := range genTestEncodingValuesLink() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -96,7 +96,7 @@ func TestMarshalAndUnmarshalProtoOrigLinkUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalProtoOrigLink(t *testing.T) {
 	for name, src := range genTestEncodingValuesLink() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
