diff --git a/pdata/internal/generated_wrapper_exporttracepartialsuccess_test.go b/pdata/internal/generated_wrapper_exporttracepartialsuccess_test.go
index 5ebeb6fce..6aa9b873c 100644
--- a/pdata/internal/generated_wrapper_exporttracepartialsuccess_test.go
+++ b/pdata/internal/generated_wrapper_exporttracepartialsuccess_test.go
@@ -23,7 +23,7 @@ import (
 func TestCopyOrigExportTracePartialSuccess(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportTracePartialSuccess() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -52,7 +52,7 @@ func TestMarshalAndUnmarshalJSONOrigExportTracePartialSuccessUnknown(t *testing.
 func TestMarshalAndUnmarshalJSONOrigExportTracePartialSuccess(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportTracePartialSuccess() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -96,7 +96,7 @@ func TestMarshalAndUnmarshalProtoOrigExportTracePartialSuccessUnknown(t *testing
 func TestMarshalAndUnmarshalProtoOrigExportTracePartialSuccess(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportTracePartialSuccess() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
