diff --git a/pdata/internal/generated_wrapper_entityref_test.go b/pdata/internal/generated_wrapper_entityref_test.go
index 19e9fd998..017393122 100644
--- a/pdata/internal/generated_wrapper_entityref_test.go
+++ b/pdata/internal/generated_wrapper_entityref_test.go
@@ -23,7 +23,7 @@ import (
 func TestCopyOrigEntityRef(t *testing.T) {
 	for name, src := range genTestEncodingValuesEntityRef() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -52,7 +52,7 @@ func TestMarshalAndUnmarshalJSONOrigEntityRefUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalJSONOrigEntityRef(t *testing.T) {
 	for name, src := range genTestEncodingValuesEntityRef() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -96,7 +96,7 @@ func TestMarshalAndUnmarshalProtoOrigEntityRefUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalProtoOrigEntityRef(t *testing.T) {
 	for name, src := range genTestEncodingValuesEntityRef() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
