diff --git a/pdata/internal/generated_wrapper_scopelogs_test.go b/pdata/internal/generated_wrapper_scopelogs_test.go
index be51af8d4..77610d6fb 100644
--- a/pdata/internal/generated_wrapper_scopelogs_test.go
+++ b/pdata/internal/generated_wrapper_scopelogs_test.go
@@ -23,7 +23,7 @@ import (
 func TestCopyOrigScopeLogs(t *testing.T) {
 	for name, src := range genTestEncodingValuesScopeLogs() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -52,7 +52,7 @@ func TestMarshalAndUnmarshalJSONOrigScopeLogsUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalJSONOrigScopeLogs(t *testing.T) {
 	for name, src := range genTestEncodingValuesScopeLogs() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -96,7 +96,7 @@ func TestMarshalAndUnmarshalProtoOrigScopeLogsUnknown(t *testing.T) {
 func TestMarshalAndUnmarshalProtoOrigScopeLogs(t *testing.T) {
 	for name, src := range genTestEncodingValuesScopeLogs() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+			t.Run(name+"/Pooling="+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
