diff --git a/scraper/scrapertest/go.mod b/scraper/scrapertest/go.mod
index cb90008b4..fa6b0ad34 100644
--- a/scraper/scrapertest/go.mod
+++ b/scraper/scrapertest/go.mod
@@ -41,3 +41,5 @@ replace go.opentelemetry.io/collector/pdata => ../../pdata
 replace go.opentelemetry.io/collector/scraper => ../
 
 replace go.opentelemetry.io/collector/featuregate => ../../featuregate
+
+replace go.opentelemetry.io/collector/internal/testutil => ../../internal/testutil
