diff --git a/processor/batchprocessor/factory_test.go b/processor/batchprocessor/factory_test.go
index 4a30adedf..28d36d15d 100644
--- a/processor/batchprocessor/factory_test.go
+++ b/processor/batchprocessor/factory_test.go
@@ -38,7 +38,7 @@ func TestCreateProcessor(t *testing.T) {
 
 	cfg := factory.CreateDefaultConfig()
 	creationParams := component.ProcessorCreateParams{Logger: zap.NewNop()}
-	tp, err := factory.CreateTraceProcessor(context.Background(), creationParams, cfg, nil)
+	tp, err := factory.CreateTracesProcessor(context.Background(), creationParams, cfg, nil)
 	assert.NotNil(t, tp)
 	assert.NoError(t, err, "cannot create trace processor")
 
