diff --git a/service/service_test.go b/service/service_test.go
index 0d3403a5d..c082622ae 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -49,6 +49,7 @@ func TestApplication_Start(t *testing.T) {
 		"--config=testdata/otelcol-config.yaml",
 		"--metrics-port=" + strconv.FormatUint(uint64(metricsPort), 10),
 		"--metrics-prefix=" + testPrefix,
+		"--add-instance-id=true",
 	})
 
 	appDone := make(chan struct{})
