diff --git a/processor/addattributesprocessor/factory.go b/processor/addattributesprocessor/factory.go
index 8a846b680..f2e89f05c 100644
--- a/processor/addattributesprocessor/factory.go
+++ b/processor/addattributesprocessor/factory.go
@@ -23,24 +23,24 @@ import (
 	"github.com/open-telemetry/opentelemetry-service/processor"
 )
 
-var _ = processor.RegisterProcessorFactory(&processorFactory{})
+var _ = processor.RegisterFactory(&factory{})
 
 const (
 	// The value of "type" key in configuration.
 	typeStr = "attributes"
 )
 
-// processorFactory is the factory for Attributes processor.
-type processorFactory struct {
+// factory is the factory for Attributes processor.
+type factory struct {
 }
 
 // Type gets the type of the config created by this factory.
-func (f *processorFactory) Type() string {
+func (f *factory) Type() string {
 	return typeStr
 }
 
 // CreateDefaultConfig creates the default configuration for processor.
-func (f *processorFactory) CreateDefaultConfig() models.Processor {
+func (f *factory) CreateDefaultConfig() models.Processor {
 	return &ConfigV2{
 		ProcessorSettings: models.ProcessorSettings{
 			TypeVal: typeStr,
@@ -51,7 +51,7 @@ func (f *processorFactory) CreateDefaultConfig() models.Processor {
 }
 
 // CreateTraceProcessor creates a trace processor based on this config.
-func (f *processorFactory) CreateTraceProcessor(
+func (f *factory) CreateTraceProcessor(
 	logger *zap.Logger,
 	nextConsumer consumer.TraceConsumer,
 	cfg models.Processor,
@@ -61,7 +61,7 @@ func (f *processorFactory) CreateTraceProcessor(
 }
 
 // CreateMetricsProcessor creates a metrics processor based on this config.
-func (f *processorFactory) CreateMetricsProcessor(
+func (f *factory) CreateMetricsProcessor(
 	logger *zap.Logger,
 	nextConsumer consumer.MetricsConsumer,
 	cfg models.Processor,
