diff --git a/internal/collector/processor/exporter_processor.go b/internal/collector/processor/exporter_processor.go
index b6aca5a41..b7a27341d 100644
--- a/internal/collector/processor/exporter_processor.go
+++ b/internal/collector/processor/exporter_processor.go
@@ -19,22 +19,22 @@ import (
 
 	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
 	"github.com/census-instrumentation/opencensus-service/data"
-	"github.com/census-instrumentation/opencensus-service/exporter"
+	"github.com/census-instrumentation/opencensus-service/processor"
 )
 
 type exporterSpanProcessor struct {
-	tes exporter.TraceExporterSink
+	tdp processor.TraceDataProcessor
 }
 
 var _ SpanProcessor = (*exporterSpanProcessor)(nil)
 
 // NewTraceExporterProcessor creates processor that feeds SpanData to the given trace exporters.
-func NewTraceExporterProcessor(traceExporters ...exporter.TraceExporter) SpanProcessor {
-	return &exporterSpanProcessor{tes: exporter.MultiTraceExporters(traceExporters...)}
+func NewTraceExporterProcessor(traceExporters ...processor.TraceDataProcessor) SpanProcessor {
+	return &exporterSpanProcessor{tdp: processor.NewMultiTraceDataProcessor(traceExporters)}
 }
 
 func (sp *exporterSpanProcessor) ProcessSpans(batch *agenttracepb.ExportTraceServiceRequest, spanFormat string) (uint64, error) {
-	err := sp.tes.ProcessTraceData(context.Background(), data.TraceData{Node: batch.Node, Resource: batch.Resource, Spans: batch.Spans})
+	err := sp.tdp.ProcessTraceData(context.Background(), data.TraceData{Node: batch.Node, Resource: batch.Resource, Spans: batch.Spans})
 	if err != nil {
 		// TODO: determine if the number of dropped spans is needed because it was wrong anyway.
 		return 0, err
