diff --git a/cmd/otelcorecol/main.go b/cmd/otelcorecol/main.go
index 358cb8644..45b25c6bb 100644
--- a/cmd/otelcorecol/main.go
+++ b/cmd/otelcorecol/main.go
@@ -21,7 +21,7 @@ func main() {
 	info := component.BuildInfo{
 		Command:     "otelcorecol",
 		Description: "Local OpenTelemetry Collector binary, testing only.",
-		Version:     "0.103.0-dev",
+		Version:     "0.104.0-dev",
 	}
 
 	set := otelcol.CollectorSettings{
