diff --git a/exporter/debugexporter/internal/otlptext/databuffer.go b/exporter/debugexporter/internal/otlptext/databuffer.go
index 4e73794da..37ca74449 100644
--- a/exporter/debugexporter/internal/otlptext/databuffer.go
+++ b/exporter/debugexporter/internal/otlptext/databuffer.go
@@ -350,7 +350,6 @@ func (b *dataBuffer) logProfileMappings(ms pprofile.MappingSlice) {
 		b.logEntry("        Has filenames: %t", mapping.HasFilenames())
 		b.logEntry("        Has line numbers: %t", mapping.HasLineNumbers())
 		b.logEntry("        Has inline frames: %t", mapping.HasInlineFrames())
-
 	}
 }
 
