diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 37d4e255d..4fa75ef3c 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -7,15 +7,15 @@ require (
 	github.com/klauspost/compress v1.17.5
 	github.com/rs/cors v1.10.1
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector v0.94.0
-	go.opentelemetry.io/collector/component v0.94.0
-	go.opentelemetry.io/collector/config/configauth v0.94.0
-	go.opentelemetry.io/collector/config/configcompression v0.94.0
-	go.opentelemetry.io/collector/config/configopaque v0.94.0
-	go.opentelemetry.io/collector/config/configtelemetry v0.94.0
-	go.opentelemetry.io/collector/config/configtls v0.94.0
-	go.opentelemetry.io/collector/config/internal v0.94.0
-	go.opentelemetry.io/collector/extension/auth v0.94.0
+	go.opentelemetry.io/collector v0.94.1
+	go.opentelemetry.io/collector/component v0.94.1
+	go.opentelemetry.io/collector/config/configauth v0.94.1
+	go.opentelemetry.io/collector/config/configcompression v0.94.1
+	go.opentelemetry.io/collector/config/configopaque v0.94.1
+	go.opentelemetry.io/collector/config/configtelemetry v0.94.1
+	go.opentelemetry.io/collector/config/configtls v0.94.1
+	go.opentelemetry.io/collector/config/internal v0.94.1
+	go.opentelemetry.io/collector/extension/auth v0.94.1
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.47.0
 	go.opentelemetry.io/otel v1.23.0
 	go.uber.org/goleak v1.3.0
@@ -46,8 +46,8 @@ require (
 	github.com/prometheus/client_model v0.5.0 // indirect
 	github.com/prometheus/common v0.46.0 // indirect
 	github.com/prometheus/procfs v0.12.0 // indirect
-	go.opentelemetry.io/collector/confmap v0.94.0 // indirect
-	go.opentelemetry.io/collector/extension v0.94.0 // indirect
+	go.opentelemetry.io/collector/confmap v0.94.1 // indirect
+	go.opentelemetry.io/collector/extension v0.94.1 // indirect
 	go.opentelemetry.io/collector/featuregate v1.1.0 // indirect
 	go.opentelemetry.io/collector/pdata v1.1.0 // indirect
 	go.opentelemetry.io/otel/exporters/prometheus v0.45.1 // indirect
