diff --git a/internal/e2e/configauth_test.go b/internal/e2e/configauth_test.go
index 859b45c93..137a7cad2 100644
--- a/internal/e2e/configauth_test.go
+++ b/internal/e2e/configauth_test.go
@@ -15,6 +15,6 @@ import (
 
 func TestConfmapMarshalConfigAuth(t *testing.T) {
 	conf := confmap.New()
-	require.NoError(t, conf.Marshal(configauth.Authentication{}))
+	require.NoError(t, conf.Marshal(configauth.Config{}))
 	assert.Equal(t, map[string]any{}, conf.ToStringMap())
 }
