diff --git a/extension/healthcheckextension/config.go b/extension/healthcheckextension/config.go
index e3558ae95..1137eb874 100644
--- a/extension/healthcheckextension/config.go
+++ b/extension/healthcheckextension/config.go
@@ -34,3 +34,10 @@ type Config struct {
 	// The default endpoint is "0.0.0.0:13133".
 	TCPAddr confignet.TCPAddr `mapstructure:",squash"`
 }
+
+var _ config.Extension = (*Config)(nil)
+
+// Validate checks if the extension configuration is valid
+func (cfg *Config) Validate() error {
+	return nil
+}
