diff --git a/pdata/pcommon/map.go b/pdata/pcommon/map.go
index e7f48b079..04af92110 100644
--- a/pdata/pcommon/map.go
+++ b/pdata/pcommon/map.go
@@ -125,7 +125,7 @@ func (m Map) PutEmpty(k string) Value {
 // PutStr performs the Insert or Update action. The Value is
 // inserted to the map that did not originally have the key. The key/value is
 // updated to the map where the key already existed.
-func (m Map) PutStr(k string, v string) {
+func (m Map) PutStr(k, v string) {
 	m.getState().AssertMutable()
 	if av, existing := m.Get(k); existing {
 		av.SetStr(v)
