diff --git a/component/experimental/component/factory.go b/component/experimental/component/factory.go
index 03c0d49d9..3532b3bb3 100644
--- a/component/experimental/component/factory.go
+++ b/component/experimental/component/factory.go
@@ -44,7 +44,7 @@ type ConfigSourceFactory interface {
 	// configuration and should not cause side-effects that prevent the creation
 	// of multiple instances of the Source.
 	// The object returned by this method needs to pass the checks implemented by
-	// 'configcheck.ValidateConfig'. It is recommended to have such check in the
+	// 'configtest.CheckConfigStruct'. It is recommended to have such check in the
 	// tests of any implementation of the ConfigSourceFactory interface.
 	CreateDefaultConfig() config.Source
 
