diff --git a/receiver/end_to_end_test.go b/receiver/end_to_end_test.go
index a05dcc2ed..94f1d8e04 100644
--- a/receiver/end_to_end_test.go
+++ b/receiver/end_to_end_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
