diff --git a/pdata/ptrace/generated_status_test.go b/pdata/ptrace/generated_status_test.go
index 97d158ef4..0ca2ba296 100644
--- a/pdata/ptrace/generated_status_test.go
+++ b/pdata/ptrace/generated_status_test.go
@@ -48,7 +48,7 @@ func TestStatus_Code(t *testing.T) {
 
 func TestStatus_Message(t *testing.T) {
 	ms := NewStatus()
-	assert.Equal(t, "", ms.Message())
+	assert.Empty(t, ms.Message())
 	ms.SetMessage("cancelled")
 	assert.Equal(t, "cancelled", ms.Message())
 	sharedState := internal.StateReadOnly
