diff --git a/connector/connectortest/go.mod b/connector/connectortest/go.mod
index 6e9bdb774..523699cfc 100644
--- a/connector/connectortest/go.mod
+++ b/connector/connectortest/go.mod
@@ -46,7 +46,7 @@ require (
 	golang.org/x/sys v0.32.0 // indirect
 	golang.org/x/text v0.24.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20250218202821-56aae31c358a // indirect
-	google.golang.org/grpc v1.72.1 // indirect
+	google.golang.org/grpc v1.72.2 // indirect
 	google.golang.org/protobuf v1.36.6 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
