diff --git a/internal/cmd/pdatagen/internal/primitive_slice_field.go b/internal/cmd/pdatagen/internal/primitive_slice_field.go
index 5578aec43..6c67f695c 100644
--- a/internal/cmd/pdatagen/internal/primitive_slice_field.go
+++ b/internal/cmd/pdatagen/internal/primitive_slice_field.go
@@ -26,6 +26,14 @@ const primitiveSliceCopyOrigTemplate = `dest.{{ .originFieldName }} =
 {{- if .isCommon }}{{ if not .isBaseStructCommon }}internal.{{ end }}CopyOrig{{ else }}copyOrig{{ end }}
 {{- .returnType }}(dest.{{ .originFieldName }}, src.{{ .originFieldName }})`
 
+const primitiveSliceMarshalJSONTemplate = `dest.WriteObjectField("{{ lowerFirst .originFieldName }}")
+	{{- if .isCommon }}
+	{{ if not .isBaseStructCommon }}internal.{{ end }}MarshalJSONStream{{ .returnType }}(
+	{{- if not .isBaseStructCommon }}internal.{{ end }}New{{ .returnType }}(&ms.orig.{{ .originFieldName }}, ms.state), dest)
+	{{- else }}
+	ms.{{ .fieldName }}().marshalJSONStream(dest)
+	{{- end }}`
+
 // PrimitiveSliceField is used to generate fields for slice of primitive types
 type PrimitiveSliceField struct {
 	fieldName         string
@@ -56,6 +64,11 @@ func (psf *PrimitiveSliceField) GenerateCopyOrig(ms *messageStruct) string {
 	return executeTemplate(t, psf.templateFields(ms))
 }
 
+func (psf *PrimitiveSliceField) GenerateMarshalJSON(ms *messageStruct) string {
+	t := template.Must(templateNew("primitiveSliceMarshalJSONTemplate").Parse(primitiveSliceMarshalJSONTemplate))
+	return executeTemplate(t, psf.templateFields(ms))
+}
+
 func (psf *PrimitiveSliceField) templateFields(ms *messageStruct) map[string]any {
 	return map[string]any{
 		"structName": ms.getName(),
