diff --git a/receiver/otlpreceiver/otlp_test.go b/receiver/otlpreceiver/otlp_test.go
index ba8a341dd..9a58c8d54 100644
--- a/receiver/otlpreceiver/otlp_test.go
+++ b/receiver/otlpreceiver/otlp_test.go
@@ -703,7 +703,7 @@ func TestGRPCInvalidTLSCredentials(t *testing.T) {
 
 	r, err := NewFactory().CreateTraces(
 		context.Background(),
-		receivertest.NewNopSettingsWithType(receivertest.NopType),
+		receivertest.NewNopSettings(receivertest.NopType),
 		cfg,
 		consumertest.NewNop())
 	require.NoError(t, err)
@@ -772,7 +772,7 @@ func TestHTTPInvalidTLSCredentials(t *testing.T) {
 	// TLS is resolved during Start for HTTP.
 	r, err := NewFactory().CreateTraces(
 		context.Background(),
-		receivertest.NewNopSettingsWithType(receivertest.NopType),
+		receivertest.NewNopSettings(receivertest.NopType),
 		cfg,
 		consumertest.NewNop())
 	require.NoError(t, err)
@@ -838,7 +838,7 @@ func newHTTPReceiver(t *testing.T, settings component.TelemetrySettings, endpoin
 }
 
 func newReceiver(t *testing.T, settings component.TelemetrySettings, cfg *Config, id component.ID, c consumertest.Consumer) component.Component {
-	set := receivertest.NewNopSettingsWithType(receivertest.NopType)
+	set := receivertest.NewNopSettings(receivertest.NopType)
 	set.TelemetrySettings = settings
 	set.ID = id
 	r, err := newOtlpReceiver(cfg, &set)
@@ -1015,7 +1015,7 @@ func TestShutdown(t *testing.T) {
 	cfg := factory.CreateDefaultConfig().(*Config)
 	cfg.GRPC.NetAddr.Endpoint = endpointGrpc
 	cfg.HTTP.Endpoint = endpointHTTP
-	set := receivertest.NewNopSettingsWithType(receivertest.NopType)
+	set := receivertest.NewNopSettings(receivertest.NopType)
 	set.ID = otlpReceiverID
 	r, err := NewFactory().CreateTraces(
 		context.Background(),
