diff --git a/exporter/prometheusremotewriteexporter/testutil_test.go b/exporter/prometheusremotewriteexporter/testutil_test.go
index 1a603f3c2..fd5f13aee 100644
--- a/exporter/prometheusremotewriteexporter/testutil_test.go
+++ b/exporter/prometheusremotewriteexporter/testutil_test.go
@@ -15,6 +15,7 @@
 package prometheusremotewriteexporter
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/prometheus/prometheus/prompb"
@@ -25,9 +26,11 @@ import (
 
 var (
 	time1   = uint64(time.Now().UnixNano())
-	time2   = uint64(time.Date(1970, 1, 0, 0, 0, 0, 0, time.UTC).UnixNano())
+	time2   = uint64(time.Now().UnixNano() - 5)
+	time3   = uint64(time.Date(1970, 1, 0, 0, 0, 0, 0, time.UTC).UnixNano())
 	msTime1 = int64(time1 / uint64(int64(time.Millisecond)/int64(time.Nanosecond)))
 	msTime2 = int64(time2 / uint64(int64(time.Millisecond)/int64(time.Nanosecond)))
+	msTime3 = int64(time3 / uint64(int64(time.Millisecond)/int64(time.Nanosecond)))
 
 	label11 = "test_label11"
 	value11 = "test_value11"
@@ -50,6 +53,7 @@ var (
 	intVal2   int64 = 2
 	floatVal1       = 1.0
 	floatVal2       = 2.0
+	floatVal3       = 3.0
 
 	lbs1      = getLabels(label11, value11, label12, value12)
 	lbs2      = getLabels(label21, value21, label22, value22)
@@ -564,3 +568,13 @@ func getQuantiles(bounds []float64, values []float64) []*otlp.DoubleSummaryDataP
 	}
 	return quantiles
 }
+
+func getTimeseriesMap(timeseries []*prompb.TimeSeries) map[string]*prompb.TimeSeries {
+	tsMap := make(map[string]*prompb.TimeSeries)
+
+	for i, v := range timeseries {
+		tsMap[fmt.Sprintf("%s%d", "timeseries_name", i)] = v
+	}
+
+	return tsMap
+}
