diff --git a/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go b/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go
index 577a53121..e954f9aaf 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go
@@ -12,8 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// +build linux windows
-
 package processscraper
 
 import (
@@ -36,7 +34,15 @@ import (
 	"go.opentelemetry.io/collector/translator/conventions"
 )
 
+func skipTestOnUnsupportedOS(t *testing.T) {
+	if runtime.GOOS != "linux" && runtime.GOOS != "windows" {
+		t.Skipf("skipping test on %v", runtime.GOOS)
+	}
+}
+
 func TestScrapeMetrics(t *testing.T) {
+	skipTestOnUnsupportedOS(t)
+
 	const bootTime = 100
 	const expectedStartTime = 100 * 1e9
 
@@ -131,6 +137,8 @@ func getMetricSlice(t *testing.T, rm pdata.ResourceMetrics) pdata.MetricSlice {
 }
 
 func TestScrapeMetrics_NewError(t *testing.T) {
+	skipTestOnUnsupportedOS(t)
+
 	_, err := newProcessScraper(&Config{Include: MatchConfig{Names: []string{"test"}}})
 	require.Error(t, err)
 	require.Regexp(t, "^error creating process include filters:", err.Error())
@@ -141,6 +149,8 @@ func TestScrapeMetrics_NewError(t *testing.T) {
 }
 
 func TestScrapeMetrics_GetProcessesError(t *testing.T) {
+	skipTestOnUnsupportedOS(t)
+
 	scraper, err := newProcessScraper(&Config{})
 	require.NoError(t, err, "Failed to create process scraper: %v", err)
 
@@ -227,6 +237,8 @@ func newDefaultHandleMock() *processHandleMock {
 }
 
 func TestScrapeMetrics_Filtered(t *testing.T) {
+	skipTestOnUnsupportedOS(t)
+
 	type testCase struct {
 		name          string
 		names         []string
@@ -318,6 +330,8 @@ func TestScrapeMetrics_Filtered(t *testing.T) {
 }
 
 func TestScrapeMetrics_ProcessErrors(t *testing.T) {
+	skipTestOnUnsupportedOS(t)
+
 	type testCase struct {
 		name            string
 		osFilter        string
