diff --git a/consumer/pdata/trace.go b/consumer/pdata/trace.go
index 45cf0e259..0a52fdee7 100644
--- a/consumer/pdata/trace.go
+++ b/consumer/pdata/trace.go
@@ -112,6 +112,10 @@ type SpanKind otlptrace.Span_SpanKind
 
 func (sk SpanKind) String() string { return otlptrace.Span_SpanKind(sk).String() }
 
+const (
+	TraceStateEmpty TraceState = ""
+)
+
 const (
 	SpanKindUNSPECIFIED = SpanKind(0)
 	SpanKindINTERNAL    = SpanKind(otlptrace.Span_INTERNAL)
