diff --git a/processor/processorhelper/metrics_test.go b/processor/processorhelper/metrics_test.go
index cb841abfa..537c7c564 100644
--- a/processor/processorhelper/metrics_test.go
+++ b/processor/processorhelper/metrics_test.go
@@ -13,12 +13,14 @@ import (
 	"github.com/stretchr/testify/require"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/processor/processorhelper/internal/metadatatest"
 	"go.opentelemetry.io/collector/processor/processortest"
 )
 
@@ -118,7 +120,7 @@ func TestMetrics_RecordInOut(t *testing.T) {
 	dps.AppendEmpty()
 	dps.AppendEmpty()
 
-	testTelemetry := setupTestTelemetry()
+	testTelemetry := metadatatest.SetupTelemetry()
 	mp, err := NewMetrics(context.Background(), testTelemetry.NewSettings(), &testMetricsCfg, consumertest.NewNop(), mockAggregate)
 	require.NoError(t, err)
 
@@ -126,7 +128,7 @@ func TestMetrics_RecordInOut(t *testing.T) {
 	assert.NoError(t, mp.ConsumeMetrics(context.Background(), incomingMetrics))
 	assert.NoError(t, mp.Shutdown(context.Background()))
 
-	testTelemetry.assertMetrics(t, []metricdata.Metrics{
+	testTelemetry.AssertMetrics(t, []metricdata.Metrics{
 		{
 			Name:        "otelcol_processor_incoming_items",
 			Description: "Number of items passed to the processor. [alpha]",
@@ -157,7 +159,7 @@ func TestMetrics_RecordInOut(t *testing.T) {
 				},
 			},
 		},
-	})
+	}, metricdatatest.IgnoreTimestamp())
 }
 
 func TestMetrics_RecordIn_ErrorOut(t *testing.T) {
@@ -173,7 +175,7 @@ func TestMetrics_RecordIn_ErrorOut(t *testing.T) {
 	dps.AppendEmpty()
 	dps.AppendEmpty()
 
-	testTelemetry := setupTestTelemetry()
+	testTelemetry := metadatatest.SetupTelemetry()
 	mp, err := NewMetrics(context.Background(), testTelemetry.NewSettings(), &testMetricsCfg, consumertest.NewNop(), mockErr)
 	require.NoError(t, err)
 
@@ -181,7 +183,7 @@ func TestMetrics_RecordIn_ErrorOut(t *testing.T) {
 	require.Error(t, mp.ConsumeMetrics(context.Background(), incomingMetrics))
 	require.NoError(t, mp.Shutdown(context.Background()))
 
-	testTelemetry.assertMetrics(t, []metricdata.Metrics{
+	testTelemetry.AssertMetrics(t, []metricdata.Metrics{
 		{
 			Name:        "otelcol_processor_incoming_items",
 			Description: "Number of items passed to the processor. [alpha]",
@@ -212,5 +214,5 @@ func TestMetrics_RecordIn_ErrorOut(t *testing.T) {
 				},
 			},
 		},
-	})
+	}, metricdatatest.IgnoreTimestamp())
 }
