diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index ecace88f5..17c156cf6 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -71,6 +71,7 @@ require (
 	github.com/tklauser/numcpus v0.6.0 // indirect
 	github.com/yusufpapurcu/wmi v1.2.3 // indirect
 	go.opencensus.io v0.24.0 // indirect
+	go.opentelemetry.io/collector/config/configauth v0.0.0-00010101000000-000000000000 // indirect
 	go.opentelemetry.io/collector/config/configcompression v0.0.0-00010101000000-000000000000 // indirect
 	go.opentelemetry.io/collector/config/confignet v0.0.0-00010101000000-000000000000 // indirect
 	go.opentelemetry.io/collector/config/configopaque v0.0.0-00010101000000-000000000000 // indirect
@@ -110,6 +111,8 @@ replace go.opentelemetry.io/collector => ../../
 
 replace go.opentelemetry.io/collector/component => ../../component
 
+replace go.opentelemetry.io/collector/config/configauth => ../../config/configauth
+
 replace go.opentelemetry.io/collector/config/configcompression => ../../config/configcompression
 
 replace go.opentelemetry.io/collector/config/confignet => ../../config/confignet
