diff --git a/receiver/receiver.go b/receiver/receiver.go
index 795ed5ede..b102b5070 100644
--- a/receiver/receiver.go
+++ b/receiver/receiver.go
@@ -5,7 +5,6 @@ package receiver // import "go.opentelemetry.io/collector/receiver"
 
 import (
 	"context"
-	"fmt"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
@@ -103,39 +102,12 @@ func (f factoryOptionFunc) applyOption(o *factory) {
 // CreateTracesFunc is the equivalent of Factory.CreateTraces.
 type CreateTracesFunc func(context.Context, Settings, component.Config, consumer.Traces) (Traces, error)
 
-// CreateTraces implements Factory.CreateTraces().
-// Deprecated: [v0.120.0] No longer used, will be removed.
-func (f CreateTracesFunc) CreateTraces(ctx context.Context, set Settings, cfg component.Config, next consumer.Traces) (Traces, error) {
-	if f == nil {
-		return nil, pipeline.ErrSignalNotSupported
-	}
-	return f(ctx, set, cfg, next)
-}
-
 // CreateMetricsFunc is the equivalent of Factory.CreateMetrics.
 type CreateMetricsFunc func(context.Context, Settings, component.Config, consumer.Metrics) (Metrics, error)
 
-// CreateMetrics implements Factory.CreateMetrics.
-// Deprecated: [v0.120.0] No longer used, will be removed.
-func (f CreateMetricsFunc) CreateMetrics(ctx context.Context, set Settings, cfg component.Config, next consumer.Metrics) (Metrics, error) {
-	if f == nil {
-		return nil, pipeline.ErrSignalNotSupported
-	}
-	return f(ctx, set, cfg, next)
-}
-
 // CreateLogsFunc is the equivalent of Factory.CreateLogs.
 type CreateLogsFunc func(context.Context, Settings, component.Config, consumer.Logs) (Logs, error)
 
-// CreateLogs implements Factory.CreateLogs.
-// Deprecated: [v0.120.0] No longer used, will be removed.
-func (f CreateLogsFunc) CreateLogs(ctx context.Context, set Settings, cfg component.Config, next consumer.Logs) (Logs, error) {
-	if f == nil {
-		return nil, pipeline.ErrSignalNotSupported
-	}
-	return f(ctx, set, cfg, next)
-}
-
 type factory struct {
 	cfgType component.Type
 	component.CreateDefaultConfigFunc
@@ -221,18 +193,3 @@ func NewFactory(cfgType component.Type, createDefaultConfig component.CreateDefa
 	}
 	return f
 }
-
-// MakeFactoryMap takes a list of receiver factories and returns a map with factory type as keys.
-// It returns a non-nil error when there are factories with duplicate type.
-//
-// Deprecated: [v0.120.0] Use otelcol.MakeFactoryMap[receiver.Factory] instead
-func MakeFactoryMap(factories ...Factory) (map[component.Type]Factory, error) {
-	fMap := map[component.Type]Factory{}
-	for _, f := range factories {
-		if _, ok := fMap[f.Type()]; ok {
-			return fMap, fmt.Errorf("duplicate receiver factory %q", f.Type())
-		}
-		fMap[f.Type()] = f
-	}
-	return fMap, nil
-}
