diff --git a/receiver/otlpreceiver/otlphttp.go b/receiver/otlpreceiver/otlphttp.go
index e719cd58e..2b0f7b106 100644
--- a/receiver/otlpreceiver/otlphttp.go
+++ b/receiver/otlpreceiver/otlphttp.go
@@ -195,7 +195,11 @@ func writeError(w http.ResponseWriter, encoder encoder, err error, statusCode in
 // by the OTLP protocol.
 func errorHandler(w http.ResponseWriter, r *http.Request, errMsg string, statusCode int) {
 	s := statusutil.NewStatusFromMsgAndHTTPCode(errMsg, statusCode)
-	switch getMimeTypeFromContentType(r.Header.Get("Content-Type")) {
+	contentType := r.Header.Get("Content-Type")
+	if contentType == "" {
+		contentType = fallbackContentType
+	}
+	switch getMimeTypeFromContentType(contentType) {
 	case pbContentType:
 		writeStatusResponse(w, pbEncoder, statusCode, s)
 		return
