diff --git a/otelcol/command.go b/otelcol/command.go
index 9db850bcc..648b3a5ff 100644
--- a/otelcol/command.go
+++ b/otelcol/command.go
@@ -43,22 +43,20 @@ func NewCommand(set CollectorSettings) *cobra.Command {
 
 // Puts command line flags from flags into the CollectorSettings, to be used during config resolution.
 func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet) error {
-	if set.ConfigProvider == nil {
-		resolverSet := &set.ConfigProviderSettings.ResolverSettings
-		configFlags := getConfigFlag(flags)
+	resolverSet := &set.ConfigProviderSettings.ResolverSettings
+	configFlags := getConfigFlag(flags)
 
-		if len(configFlags) > 0 {
-			resolverSet.URIs = configFlags
-		}
-		if len(resolverSet.URIs) == 0 {
-			return errors.New("at least one config flag must be provided")
-		}
-		// Provide a default set of providers and converters if none have been specified.
-		// TODO: Remove this after CollectorSettings.ConfigProvider is removed and instead
-		// do it in the builder.
-		if len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
-			set.ConfigProviderSettings = newDefaultConfigProviderSettings(resolverSet.URIs)
-		}
+	if len(configFlags) > 0 {
+		resolverSet.URIs = configFlags
+	}
+	if len(resolverSet.URIs) == 0 {
+		return errors.New("at least one config flag must be provided")
+	}
+	// Provide a default set of providers and converters if none have been specified.
+	// TODO: Remove this after CollectorSettings.ConfigProvider is removed and instead
+	// do it in the builder.
+	if len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
+		set.ConfigProviderSettings = newDefaultConfigProviderSettings(resolverSet.URIs)
 	}
 	return nil
 }
