diff --git a/processor/spanprocessor/span.go b/processor/spanprocessor/span.go
index 27c16d195..92d3a8885 100644
--- a/processor/spanprocessor/span.go
+++ b/processor/spanprocessor/span.go
@@ -24,10 +24,10 @@ import (
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 
 	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/component/componenterr"
 	"github.com/open-telemetry/opentelemetry-collector/consumer"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
 	"github.com/open-telemetry/opentelemetry-collector/internal/processor/span"
-	"github.com/open-telemetry/opentelemetry-collector/oterr"
 	"github.com/open-telemetry/opentelemetry-collector/processor"
 )
 
@@ -51,7 +51,7 @@ type toAttributeRule struct {
 // newSpanProcessor returns the span processor.
 func newSpanProcessor(nextConsumer consumer.TraceConsumerOld, config Config) (*spanProcessor, error) {
 	if nextConsumer == nil {
-		return nil, oterr.ErrNilNextConsumer
+		return nil, componenterr.ErrNilNextConsumer
 	}
 
 	include, err := span.NewMatcher(config.Include)
