diff --git a/cmd/configschema/docsgen/docsgen/render.go b/cmd/configschema/docsgen/docsgen/render.go
index 45e7ade17..ef40d027b 100644
--- a/cmd/configschema/docsgen/docsgen/render.go
+++ b/cmd/configschema/docsgen/docsgen/render.go
@@ -20,7 +20,7 @@ import (
 	"strings"
 	"text/template"
 
-	"go.opentelemetry.io/collector/cmd/configschema/configschema"
+	"go.opentelemetry.io/collector/cmd/configschema"
 )
 
 func renderHeader(typ, group, doc string) []byte {
