diff --git a/pdata/ptrace/generated_traces.go b/pdata/ptrace/generated_traces.go
index 3b5ac63b9..e43c70ba9 100644
--- a/pdata/ptrace/generated_traces.go
+++ b/pdata/ptrace/generated_traces.go
@@ -69,7 +69,7 @@ func (es ResourceSpansSlice) At(ix int) ResourceSpans {
 	return newResourceSpans((*es.getOrig())[ix])
 }
 
-// CopyTo copies all elements from the current slice to the dest.
+// CopyTo copies all elements from the current slice overriding the destination.
 func (es ResourceSpansSlice) CopyTo(dest ResourceSpansSlice) {
 	srcLen := es.Len()
 	destCap := cap(*dest.getOrig())
@@ -192,7 +192,7 @@ func NewResourceSpans() ResourceSpans {
 	return newResourceSpans(&otlptrace.ResourceSpans{})
 }
 
-// MoveTo moves all properties from the current struct to dest
+// MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms ResourceSpans) MoveTo(dest ResourceSpans) {
 	*dest.getOrig() = *ms.getOrig()
@@ -219,7 +219,7 @@ func (ms ResourceSpans) ScopeSpans() ScopeSpansSlice {
 	return ScopeSpansSlice(internal.NewScopeSpansSlice(&ms.getOrig().ScopeSpans))
 }
 
-// CopyTo copies all properties from the current struct to the dest.
+// CopyTo copies all properties from the current struct overriding the destination.
 func (ms ResourceSpans) CopyTo(dest ResourceSpans) {
 	ms.Resource().CopyTo(dest.Resource())
 	dest.SetSchemaUrl(ms.SchemaUrl())
@@ -269,7 +269,7 @@ func (es ScopeSpansSlice) At(ix int) ScopeSpans {
 	return newScopeSpans((*es.getOrig())[ix])
 }
 
-// CopyTo copies all elements from the current slice to the dest.
+// CopyTo copies all elements from the current slice overriding the destination.
 func (es ScopeSpansSlice) CopyTo(dest ScopeSpansSlice) {
 	srcLen := es.Len()
 	destCap := cap(*dest.getOrig())
@@ -392,7 +392,7 @@ func NewScopeSpans() ScopeSpans {
 	return newScopeSpans(&otlptrace.ScopeSpans{})
 }
 
-// MoveTo moves all properties from the current struct to dest
+// MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms ScopeSpans) MoveTo(dest ScopeSpans) {
 	*dest.getOrig() = *ms.getOrig()
@@ -419,7 +419,7 @@ func (ms ScopeSpans) Spans() SpanSlice {
 	return SpanSlice(internal.NewSpanSlice(&ms.getOrig().Spans))
 }
 
-// CopyTo copies all properties from the current struct to the dest.
+// CopyTo copies all properties from the current struct overriding the destination.
 func (ms ScopeSpans) CopyTo(dest ScopeSpans) {
 	ms.Scope().CopyTo(dest.Scope())
 	dest.SetSchemaUrl(ms.SchemaUrl())
@@ -469,7 +469,7 @@ func (es SpanSlice) At(ix int) Span {
 	return newSpan((*es.getOrig())[ix])
 }
 
-// CopyTo copies all elements from the current slice to the dest.
+// CopyTo copies all elements from the current slice overriding the destination.
 func (es SpanSlice) CopyTo(dest SpanSlice) {
 	srcLen := es.Len()
 	destCap := cap(*dest.getOrig())
@@ -593,7 +593,7 @@ func NewSpan() Span {
 	return newSpan(&otlptrace.Span{})
 }
 
-// MoveTo moves all properties from the current struct to dest
+// MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms Span) MoveTo(dest Span) {
 	*dest.getOrig() = *ms.getOrig()
@@ -725,7 +725,7 @@ func (ms Span) Status() Status {
 	return Status(internal.NewStatus(&ms.getOrig().Status))
 }
 
-// CopyTo copies all properties from the current struct to the dest.
+// CopyTo copies all properties from the current struct overriding the destination.
 func (ms Span) CopyTo(dest Span) {
 	dest.SetTraceID(ms.TraceID())
 	dest.SetSpanID(ms.SpanID())
@@ -787,7 +787,7 @@ func (es SpanEventSlice) At(ix int) SpanEvent {
 	return newSpanEvent((*es.getOrig())[ix])
 }
 
-// CopyTo copies all elements from the current slice to the dest.
+// CopyTo copies all elements from the current slice overriding the destination.
 func (es SpanEventSlice) CopyTo(dest SpanEventSlice) {
 	srcLen := es.Len()
 	destCap := cap(*dest.getOrig())
@@ -911,7 +911,7 @@ func NewSpanEvent() SpanEvent {
 	return newSpanEvent(&otlptrace.Span_Event{})
 }
 
-// MoveTo moves all properties from the current struct to dest
+// MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms SpanEvent) MoveTo(dest SpanEvent) {
 	*dest.getOrig() = *ms.getOrig()
@@ -953,7 +953,7 @@ func (ms SpanEvent) SetDroppedAttributesCount(v uint32) {
 	ms.getOrig().DroppedAttributesCount = v
 }
 
-// CopyTo copies all properties from the current struct to the dest.
+// CopyTo copies all properties from the current struct overriding the destination.
 func (ms SpanEvent) CopyTo(dest SpanEvent) {
 	dest.SetTimestamp(ms.Timestamp())
 	dest.SetName(ms.Name())
@@ -1004,7 +1004,7 @@ func (es SpanLinkSlice) At(ix int) SpanLink {
 	return newSpanLink((*es.getOrig())[ix])
 }
 
-// CopyTo copies all elements from the current slice to the dest.
+// CopyTo copies all elements from the current slice overriding the destination.
 func (es SpanLinkSlice) CopyTo(dest SpanLinkSlice) {
 	srcLen := es.Len()
 	destCap := cap(*dest.getOrig())
@@ -1129,7 +1129,7 @@ func NewSpanLink() SpanLink {
 	return newSpanLink(&otlptrace.Span_Link{})
 }
 
-// MoveTo moves all properties from the current struct to dest
+// MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms SpanLink) MoveTo(dest SpanLink) {
 	*dest.getOrig() = *ms.getOrig()
@@ -1176,7 +1176,7 @@ func (ms SpanLink) SetDroppedAttributesCount(v uint32) {
 	ms.getOrig().DroppedAttributesCount = v
 }
 
-// CopyTo copies all properties from the current struct to the dest.
+// CopyTo copies all properties from the current struct overriding the destination.
 func (ms SpanLink) CopyTo(dest SpanLink) {
 	dest.SetTraceID(ms.TraceID())
 	dest.SetSpanID(ms.SpanID())
@@ -1212,7 +1212,7 @@ func NewStatus() Status {
 	return newStatus(&otlptrace.Status{})
 }
 
-// MoveTo moves all properties from the current struct to dest
+// MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms Status) MoveTo(dest Status) {
 	*dest.getOrig() = *ms.getOrig()
@@ -1239,7 +1239,7 @@ func (ms Status) SetMessage(v string) {
 	ms.getOrig().Message = v
 }
 
-// CopyTo copies all properties from the current struct to the dest.
+// CopyTo copies all properties from the current struct overriding the destination.
 func (ms Status) CopyTo(dest Status) {
 	dest.SetCode(ms.Code())
 	dest.SetMessage(ms.Message())
