diff --git a/service/telemetry/otelconftelemetry/config.go b/service/telemetry/otelconftelemetry/config.go
index 0229e7c74..5b131538a 100644
--- a/service/telemetry/otelconftelemetry/config.go
+++ b/service/telemetry/otelconftelemetry/config.go
@@ -7,7 +7,7 @@ import (
 	"errors"
 
 	"go.opentelemetry.io/collector/config/configtelemetry"
-	"go.opentelemetry.io/collector/service/telemetry/internal/migration"
+	"go.opentelemetry.io/collector/service/telemetry/otelconftelemetry/internal/migration"
 )
 
 // Config defines the configurable settings for service telemetry.
