diff --git a/translator/trace/zipkin/zipkinv1_to_traces.go b/translator/trace/zipkin/zipkinv1_to_traces.go
index a32312376..2f894cfbb 100644
--- a/translator/trace/zipkin/zipkinv1_to_traces.go
+++ b/translator/trace/zipkin/zipkinv1_to_traces.go
@@ -28,7 +28,7 @@ func V1JSONBatchToInternalTraces(blob []byte, parseStringTags bool) (pdata.Trace
 	}
 
 	for _, td := range ocTraces {
-		tmp := internaldata.OCToTraceData(td)
+		tmp := internaldata.OCToTraces(td.Node, td.Resource, td.Spans)
 		tmp.ResourceSpans().MoveAndAppendTo(traces.ResourceSpans())
 	}
 	return traces, nil
