diff --git a/otelcol/signals_others.go b/otelcol/signals_others.go
new file mode 100644
index 000000000..64747d319
--- /dev/null
+++ b/otelcol/signals_others.go
@@ -0,0 +1,15 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !(js && wasm)
+
+package otelcol // import "go.opentelemetry.io/collector/otelcol"
+
+import (
+	"syscall"
+)
+
+const (
+	SIGHUP  = syscall.SIGHUP
+	SIGTERM = syscall.SIGTERM
+)
