diff --git a/receiver/go.mod b/receiver/go.mod
index 4c1ac9091..1056d7848 100644
--- a/receiver/go.mod
+++ b/receiver/go.mod
@@ -49,3 +49,5 @@ retract v0.76.0 // Depends on retracted pdata v1.0.0-rc10 module
 replace go.opentelemetry.io/collector/pipeline => ../pipeline
 
 replace go.opentelemetry.io/collector/featuregate => ../featuregate
+
+replace go.opentelemetry.io/collector/internal/testutil => ../internal/testutil
