diff --git a/consumer/go.mod b/consumer/go.mod
index 2b3d7a98b..aa8f4c458 100644
--- a/consumer/go.mod
+++ b/consumer/go.mod
@@ -22,7 +22,7 @@ require (
 	golang.org/x/text v0.21.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20241015192408-796eee8c2d53 // indirect
 	google.golang.org/grpc v1.69.2 // indirect
-	google.golang.org/protobuf v1.36.1 // indirect
+	google.golang.org/protobuf v1.36.2 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
