diff --git a/internal/obsreportconfig/obsmetrics/obs_receiver.go b/internal/obsreportconfig/obsmetrics/obs_receiver.go
index 0f380b4aa..d70148c1f 100644
--- a/internal/obsreportconfig/obsmetrics/obs_receiver.go
+++ b/internal/obsreportconfig/obsmetrics/obs_receiver.go
@@ -3,11 +3,6 @@
 
 package obsmetrics // import "go.opentelemetry.io/collector/internal/obsreportconfig/obsmetrics"
 
-import (
-	"go.opencensus.io/stats"
-	"go.opencensus.io/tag"
-)
-
 const (
 	// ReceiverKey used to identify receivers in metrics and traces.
 	ReceiverKey = "receiver"
@@ -35,41 +30,8 @@ const (
 )
 
 var (
-	TagKeyReceiver, _  = tag.NewKey(ReceiverKey)
-	TagKeyTransport, _ = tag.NewKey(TransportKey)
-
 	ReceiverPrefix                  = ReceiverKey + NameSep
 	ReceiveTraceDataOperationSuffix = NameSep + "TraceDataReceived"
 	ReceiverMetricsOperationSuffix  = NameSep + "MetricsReceived"
 	ReceiverLogsOperationSuffix     = NameSep + "LogsReceived"
-
-	// Receiver metrics. Any count of data items below is in the original format
-	// that they were received, reasoning: reconciliation is easier if measurement
-	// on clients and receiver are expected to be the same. Translation issues
-	// that result in a different number of elements should be reported in a
-	// separate way.
-	ReceiverAcceptedSpans = stats.Int64(
-		ReceiverPrefix+AcceptedSpansKey,
-		"Number of spans successfully pushed into the pipeline.",
-		stats.UnitDimensionless)
-	ReceiverRefusedSpans = stats.Int64(
-		ReceiverPrefix+RefusedSpansKey,
-		"Number of spans that could not be pushed into the pipeline.",
-		stats.UnitDimensionless)
-	ReceiverAcceptedMetricPoints = stats.Int64(
-		ReceiverPrefix+AcceptedMetricPointsKey,
-		"Number of metric points successfully pushed into the pipeline.",
-		stats.UnitDimensionless)
-	ReceiverRefusedMetricPoints = stats.Int64(
-		ReceiverPrefix+RefusedMetricPointsKey,
-		"Number of metric points that could not be pushed into the pipeline.",
-		stats.UnitDimensionless)
-	ReceiverAcceptedLogRecords = stats.Int64(
-		ReceiverPrefix+AcceptedLogRecordsKey,
-		"Number of log records successfully pushed into the pipeline.",
-		stats.UnitDimensionless)
-	ReceiverRefusedLogRecords = stats.Int64(
-		ReceiverPrefix+RefusedLogRecordsKey,
-		"Number of log records that could not be pushed into the pipeline.",
-		stats.UnitDimensionless)
 )
