diff --git a/internal/data_generator/internal/base_structs.go b/internal/data_generator/internal/base_structs.go
index 3e4ef7907..3d393a766 100644
--- a/internal/data_generator/internal/base_structs.go
+++ b/internal/data_generator/internal/base_structs.go
@@ -83,6 +83,57 @@ func (es ${structName}) Resize(from, to int) {
 	*es.orig = (*es.orig)[from:to]
 }`
 
+const sliceTestTemplate = `func Test${structName}(t *testing.T) {
+	es := New${structName}(0)
+	assert.EqualValues(t, 0, es.Len())
+	es = new${structName}(&[]*${originName}{})
+	assert.EqualValues(t, 0, es.Len())
+	es = New${structName}(13)
+	defaultVal := New${elementName}()
+	testVal := generateTest${elementName}()
+	assert.EqualValues(t, 13, es.Len())
+	for i := 0; i < es.Len(); i++ {
+		assert.EqualValues(t, defaultVal, es.Get(i))
+		fillTest${elementName}(es.Get(i))
+		assert.EqualValues(t, testVal, es.Get(i))
+	}
+
+	// Test resize.
+	const resizeLo = 2
+	const resizeHi = 10
+	expectedEs := make(map[*${originName}]bool, resizeHi-resizeLo)
+	for i := resizeLo; i < resizeHi; i++ {
+		expectedEs[es.Get(i).orig] = true
+	}
+	assert.EqualValues(t, resizeHi-resizeLo, len(expectedEs))
+	es.Resize(resizeLo, resizeHi)
+	assert.EqualValues(t, resizeHi-resizeLo, es.Len())
+	foundEs := make(map[*${originName}]bool, resizeHi-resizeLo)
+	for i := 0; i < es.Len(); i++ {
+		foundEs[es.Get(i).orig] = true
+	}
+	assert.EqualValues(t, expectedEs, foundEs)
+
+	// Test remove.
+	const removePos = 2
+	delete(expectedEs, es.Get(removePos).orig)
+	es.Remove(removePos)
+	assert.EqualValues(t, resizeHi-resizeLo-1, es.Len())
+	foundEs = make(map[*${originName}]bool, resizeHi-resizeLo)
+	for i := 0; i < es.Len(); i++ {
+		foundEs[es.Get(i).orig] = true
+	}
+	assert.EqualValues(t, expectedEs, foundEs)
+}`
+
+const sliceGenerateTest = `func generateTest${structName}() ${structName} {
+	tv := New${structName}(13)
+	for i := 0; i < tv.Len(); i++ {
+		fillTest${elementName}(tv.Get(i))
+	}
+	return tv
+}`
+
 const messageTemplate = `${description}
 //
 // This is a reference type, if passsed by value and callee modifies it the
@@ -104,10 +155,29 @@ func new${structName}(orig *${originName}) ${structName} {
 	return ${structName}{orig}
 }`
 
+const messageTestHeaderTemplate = `func Test${structName}(t *testing.T) {
+	ms := New${structName}()
+	assert.EqualValues(t, new${structName}(&${originName}{}), ms)`
+
+const messageTestFooterTemplate = `	assert.EqualValues(t, generateTest${structName}(), ms)
+}`
+
+const messageGenerateTestHeaderTemplate = `func generateTest${structName}() ${structName} {
+	tv := New${structName}()`
+const messageGenerateTestFooterTemplate = `	return tv
+}`
+
+const messageFillTestHeaderTemplate = `func fillTest${structName}(tv ${structName}) {`
+const messageFillTestFooterTemplate = `}`
+
 const newLine = "\n"
 
 type baseStruct interface {
 	generateStruct(sb *strings.Builder)
+
+	generateTests(sb *strings.Builder)
+
+	generateTestValueHelpers(sb *strings.Builder)
 }
 
 // Will generate code only for the slice struct.
@@ -131,13 +201,43 @@ func (ss *sliceStruct) generateStruct(sb *strings.Builder) {
 	}))
 }
 
+func (ss *sliceStruct) generateTests(sb *strings.Builder) {
+	sb.WriteString(os.Expand(sliceTestTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ss.structName
+		case "elementName":
+			return ss.element.structName
+		case "originName":
+			return ss.element.originFullName
+		default:
+			panic(name)
+		}
+	}))
+	ss.element.generateFillTest = true
+}
+
+func (ss *sliceStruct) generateTestValueHelpers(sb *strings.Builder) {
+	sb.WriteString(os.Expand(sliceGenerateTest, func(name string) string {
+		switch name {
+		case "structName":
+			return ss.structName
+		case "elementName":
+			return ss.element.structName
+		default:
+			panic(name)
+		}
+	}))
+}
+
 var _ baseStruct = (*sliceStruct)(nil)
 
 type messageStruct struct {
-	structName     string
-	description    string
-	originFullName string
-	fields         []baseField
+	structName       string
+	description      string
+	originFullName   string
+	fields           []baseField
+	generateFillTest bool
 }
 
 func (ms *messageStruct) generateStruct(sb *strings.Builder) {
@@ -160,4 +260,78 @@ func (ms *messageStruct) generateStruct(sb *strings.Builder) {
 	}
 }
 
+func (ms *messageStruct) generateTests(sb *strings.Builder) {
+	sb.WriteString(os.Expand(messageTestHeaderTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.structName
+		case "originName":
+			return ms.originFullName
+		default:
+			panic(name)
+		}
+	}))
+	// Write accessors tests for the struct
+	for _, f := range ms.fields {
+		sb.WriteString(newLine + newLine)
+		f.generateAccessorsTests(ms, sb)
+	}
+	sb.WriteString(newLine + newLine)
+	sb.WriteString(os.Expand(messageTestFooterTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.structName
+		default:
+			panic(name)
+		}
+	}))
+}
+
+func (ms *messageStruct) generateTestValueHelpers(sb *strings.Builder) {
+	sb.WriteString(os.Expand(messageGenerateTestHeaderTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.structName
+		default:
+			panic(name)
+		}
+	}))
+	// Write accessors test value for the struct
+	for _, f := range ms.fields {
+		sb.WriteString(newLine)
+		f.generateSetWithTestValue(sb)
+	}
+	sb.WriteString(newLine)
+	sb.WriteString(os.Expand(messageGenerateTestFooterTemplate, func(name string) string {
+		switch name {
+		default:
+			panic(name)
+		}
+	}))
+
+	if ms.generateFillTest {
+		sb.WriteString(newLine + newLine)
+		sb.WriteString(os.Expand(messageFillTestHeaderTemplate, func(name string) string {
+			switch name {
+			case "structName":
+				return ms.structName
+			default:
+				panic(name)
+			}
+		}))
+		// Write accessors test value for the struct
+		for _, f := range ms.fields {
+			sb.WriteString(newLine)
+			f.generateSetWithTestValue(sb)
+		}
+		sb.WriteString(newLine)
+		sb.WriteString(os.Expand(messageFillTestFooterTemplate, func(name string) string {
+			switch name {
+			default:
+				panic(name)
+			}
+		}))
+	}
+}
+
 var _ baseStruct = (*messageStruct)(nil)
