diff --git a/exporter/exporterhelper/common_test.go b/exporter/exporterhelper/common_test.go
index 92d959251..6b6428459 100644
--- a/exporter/exporterhelper/common_test.go
+++ b/exporter/exporterhelper/common_test.go
@@ -24,12 +24,12 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
-	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config"
 )
 
 var okStatus = trace.Status{Code: trace.StatusCodeOK}
 
-var defaultExporterCfg = &configmodels.ExporterSettings{
+var defaultExporterCfg = &config.ExporterSettings{
 	TypeVal: "test",
 	NameVal: "test",
 }
