diff --git a/exporter/exporterhelper/factory.go b/exporter/exporterhelper/factory.go
index 0855d2ecc..044632939 100644
--- a/exporter/exporterhelper/factory.go
+++ b/exporter/exporterhelper/factory.go
@@ -20,27 +20,27 @@ import (
 	"github.com/spf13/viper"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configerror"
-	"go.opentelemetry.io/collector/config/configmodels"
 )
 
 // FactoryOption apply changes to ExporterOptions.
 type FactoryOption func(o *factory)
 
 // CreateDefaultConfig is the equivalent of component.ExporterFactory.CreateDefaultConfig()
-type CreateDefaultConfig func() configmodels.Exporter
+type CreateDefaultConfig func() config.Exporter
 
 // CreateTracesExporter is the equivalent of component.ExporterFactory.CreateTracesExporter()
-type CreateTracesExporter func(context.Context, component.ExporterCreateParams, configmodels.Exporter) (component.TracesExporter, error)
+type CreateTracesExporter func(context.Context, component.ExporterCreateParams, config.Exporter) (component.TracesExporter, error)
 
 // CreateMetricsExporter is the equivalent of component.ExporterFactory.CreateMetricsExporter()
-type CreateMetricsExporter func(context.Context, component.ExporterCreateParams, configmodels.Exporter) (component.MetricsExporter, error)
+type CreateMetricsExporter func(context.Context, component.ExporterCreateParams, config.Exporter) (component.MetricsExporter, error)
 
 // CreateMetricsExporter is the equivalent of component.ExporterFactory.CreateLogsExporter()
-type CreateLogsExporter func(context.Context, component.ExporterCreateParams, configmodels.Exporter) (component.LogsExporter, error)
+type CreateLogsExporter func(context.Context, component.ExporterCreateParams, config.Exporter) (component.LogsExporter, error)
 
 type factory struct {
-	cfgType               configmodels.Type
+	cfgType               config.Type
 	customUnmarshaler     component.CustomUnmarshaler
 	createDefaultConfig   CreateDefaultConfig
 	createTracesExporter  CreateTracesExporter
@@ -78,7 +78,7 @@ func WithCustomUnmarshaler(customUnmarshaler component.CustomUnmarshaler) Factor
 
 // NewFactory returns a component.ExporterFactory.
 func NewFactory(
-	cfgType configmodels.Type,
+	cfgType config.Type,
 	createDefaultConfig CreateDefaultConfig,
 	options ...FactoryOption) component.ExporterFactory {
 	f := &factory{
@@ -98,12 +98,12 @@ func NewFactory(
 }
 
 // Type gets the type of the Exporter config created by this factory.
-func (f *factory) Type() configmodels.Type {
+func (f *factory) Type() config.Type {
 	return f.cfgType
 }
 
 // CreateDefaultConfig creates the default configuration for processor.
-func (f *factory) CreateDefaultConfig() configmodels.Exporter {
+func (f *factory) CreateDefaultConfig() config.Exporter {
 	return f.createDefaultConfig()
 }
 
@@ -111,7 +111,7 @@ func (f *factory) CreateDefaultConfig() configmodels.Exporter {
 func (f *factory) CreateTracesExporter(
 	ctx context.Context,
 	params component.ExporterCreateParams,
-	cfg configmodels.Exporter) (component.TracesExporter, error) {
+	cfg config.Exporter) (component.TracesExporter, error) {
 	if f.createTracesExporter != nil {
 		return f.createTracesExporter(ctx, params, cfg)
 	}
@@ -122,7 +122,7 @@ func (f *factory) CreateTracesExporter(
 func (f *factory) CreateMetricsExporter(
 	ctx context.Context,
 	params component.ExporterCreateParams,
-	cfg configmodels.Exporter) (component.MetricsExporter, error) {
+	cfg config.Exporter) (component.MetricsExporter, error) {
 	if f.createMetricsExporter != nil {
 		return f.createMetricsExporter(ctx, params, cfg)
 	}
@@ -133,7 +133,7 @@ func (f *factory) CreateMetricsExporter(
 func (f *factory) CreateLogsExporter(
 	ctx context.Context,
 	params component.ExporterCreateParams,
-	cfg configmodels.Exporter,
+	cfg config.Exporter,
 ) (component.LogsExporter, error) {
 	if f.createLogsExporter != nil {
 		return f.createLogsExporter(ctx, params, cfg)
