diff --git a/config/configtest/configtest_test.go b/config/configtest/configtest_test.go
index 8e47551ea..6b2a0a0d2 100644
--- a/config/configtest/configtest_test.go
+++ b/config/configtest/configtest_test.go
@@ -16,7 +16,6 @@ package configtest
 
 import (
 	"io"
-	"path/filepath"
 	"strings"
 	"testing"
 
@@ -24,17 +23,6 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestLoadConfigMap_FileNotFound(t *testing.T) {
-	_, err := LoadConfigMap("file/not/found")
-	assert.Error(t, err)
-}
-
-func TestLoadConfigMap(t *testing.T) {
-	cfg, err := LoadConfigMap(filepath.Join("testdata", "simple.yaml"))
-	require.NoError(t, err)
-	assert.Equal(t, map[string]interface{}{"floating": 3.14}, cfg.ToStringMap())
-}
-
 func TestCheckConfigStructPointerAndValue(t *testing.T) {
 	config := struct {
 		SomeFiled string `mapstructure:"test"`
