diff --git a/exporter/exporterhelper/internal/queue_sender.go b/exporter/exporterhelper/internal/queue_sender.go
index d6b081957..dbc6b8e05 100644
--- a/exporter/exporterhelper/internal/queue_sender.go
+++ b/exporter/exporterhelper/internal/queue_sender.go
@@ -203,9 +203,9 @@ type SizeConfig struct {
 	Sizer request.SizerType `mapstructure:"sizer"`
 
 	// MinSize defines the configuration for the minimum size of a batch.
-	MinSize int `mapstructure:"min_size"`
+	MinSize int64 `mapstructure:"min_size"`
 	// MaxSize defines the configuration for the maximum size of a batch.
-	MaxSize int `mapstructure:"max_size"`
+	MaxSize int64 `mapstructure:"max_size"`
 }
 
 func (c *BatcherConfig) Validate() error {
