diff --git a/pdata/internal/generated_wrapper_scopespans_test.go b/pdata/internal/generated_wrapper_scopespans_test.go
index dd1ae77c4..8dfded0d4 100644
--- a/pdata/internal/generated_wrapper_scopespans_test.go
+++ b/pdata/internal/generated_wrapper_scopespans_test.go
@@ -11,6 +11,8 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	gootlptrace "go.opentelemetry.io/proto/slim/otlp/trace/v1"
+	"google.golang.org/protobuf/proto"
 
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
@@ -54,6 +56,13 @@ func TestMarshalAndUnmarshalJSONOrigScopeSpans(t *testing.T) {
 	}
 }
 
+func TestMarshalAndUnmarshalProtoOrigScopeSpansUnknown(t *testing.T) {
+	dest := &otlptrace.ScopeSpans{}
+	// message Test { required int64 field = 1313; } encoding { "field": "1234" }
+	require.NoError(t, UnmarshalProtoOrigScopeSpans(dest, []byte{0x88, 0x52, 0xD2, 0x09}))
+	assert.Equal(t, &otlptrace.ScopeSpans{}, dest)
+}
+
 func TestMarshalAndUnmarshalProtoOrigScopeSpans(t *testing.T) {
 	for name, src := range getEncodingTestValuesScopeSpans() {
 		t.Run(name, func(t *testing.T) {
@@ -68,6 +77,26 @@ func TestMarshalAndUnmarshalProtoOrigScopeSpans(t *testing.T) {
 	}
 }
 
+func TestMarshalAndUnmarshalProtoViaProtobufScopeSpans(t *testing.T) {
+	for name, src := range getEncodingTestValuesScopeSpans() {
+		t.Run(name, func(t *testing.T) {
+			buf := make([]byte, SizeProtoOrigScopeSpans(src))
+			gotSize := MarshalProtoOrigScopeSpans(src, buf)
+			assert.Equal(t, len(buf), gotSize)
+
+			goDest := &gootlptrace.ScopeSpans{}
+			require.NoError(t, proto.Unmarshal(buf, goDest))
+
+			goBuf, err := proto.Marshal(goDest)
+			require.NoError(t, err)
+
+			dest := &otlptrace.ScopeSpans{}
+			require.NoError(t, UnmarshalProtoOrigScopeSpans(dest, goBuf))
+			assert.Equal(t, src, dest)
+		})
+	}
+}
+
 func getEncodingTestValuesScopeSpans() map[string]*otlptrace.ScopeSpans {
 	return map[string]*otlptrace.ScopeSpans{
 		"empty": {},
