diff --git a/pdata/internal/generated_wrapper_profile_test.go b/pdata/internal/generated_wrapper_profile_test.go
index c6fa494ce..754dcde20 100644
--- a/pdata/internal/generated_wrapper_profile_test.go
+++ b/pdata/internal/generated_wrapper_profile_test.go
@@ -11,6 +11,8 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	gootlpprofiles "go.opentelemetry.io/proto/slim/otlp/profiles/v1development"
+	"google.golang.org/protobuf/proto"
 
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
 	"go.opentelemetry.io/collector/pdata/internal/json"
@@ -54,6 +56,13 @@ func TestMarshalAndUnmarshalJSONOrigProfile(t *testing.T) {
 	}
 }
 
+func TestMarshalAndUnmarshalProtoOrigProfileUnknown(t *testing.T) {
+	dest := &otlpprofiles.Profile{}
+	// message Test { required int64 field = 1313; } encoding { "field": "1234" }
+	require.NoError(t, UnmarshalProtoOrigProfile(dest, []byte{0x88, 0x52, 0xD2, 0x09}))
+	assert.Equal(t, &otlpprofiles.Profile{}, dest)
+}
+
 func TestMarshalAndUnmarshalProtoOrigProfile(t *testing.T) {
 	for name, src := range getEncodingTestValuesProfile() {
 		t.Run(name, func(t *testing.T) {
@@ -68,6 +77,26 @@ func TestMarshalAndUnmarshalProtoOrigProfile(t *testing.T) {
 	}
 }
 
+func TestMarshalAndUnmarshalProtoViaProtobufProfile(t *testing.T) {
+	for name, src := range getEncodingTestValuesProfile() {
+		t.Run(name, func(t *testing.T) {
+			buf := make([]byte, SizeProtoOrigProfile(src))
+			gotSize := MarshalProtoOrigProfile(src, buf)
+			assert.Equal(t, len(buf), gotSize)
+
+			goDest := &gootlpprofiles.Profile{}
+			require.NoError(t, proto.Unmarshal(buf, goDest))
+
+			goBuf, err := proto.Marshal(goDest)
+			require.NoError(t, err)
+
+			dest := &otlpprofiles.Profile{}
+			require.NoError(t, UnmarshalProtoOrigProfile(dest, goBuf))
+			assert.Equal(t, src, dest)
+		})
+	}
+}
+
 func getEncodingTestValuesProfile() map[string]*otlpprofiles.Profile {
 	return map[string]*otlpprofiles.Profile{
 		"empty": {},
