diff --git a/pdata/internal/generated_wrapper_status.go b/pdata/internal/generated_wrapper_status.go
index 8ac3de771..62dcd121e 100644
--- a/pdata/internal/generated_wrapper_status.go
+++ b/pdata/internal/generated_wrapper_status.go
@@ -7,11 +7,21 @@
 package internal
 
 import (
+	"fmt"
+
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
+func NewOrigStatus() otlptrace.Status {
+	return otlptrace.Status{}
+}
+
+func NewOrigPtrStatus() *otlptrace.Status {
+	return &otlptrace.Status{}
+}
+
 func CopyOrigStatus(dest, src *otlptrace.Status) {
 	dest.Message = src.Message
 	dest.Code = src.Code
@@ -87,5 +97,49 @@ func MarshalProtoOrigStatus(orig *otlptrace.Status, buf []byte) int {
 }
 
 func UnmarshalProtoOrigStatus(orig *otlptrace.Status, buf []byte) error {
-	return orig.Unmarshal(buf)
+	var err error
+	var fieldNum int32
+	var wireType proto.WireType
+
+	l := len(buf)
+	pos := 0
+	for pos < l {
+		// If in a group parsing, move to the next tag.
+		fieldNum, wireType, pos, err = proto.ConsumeTag(buf, pos)
+		if err != nil {
+			return err
+		}
+		switch fieldNum {
+
+		case 2:
+			if wireType != proto.WireTypeLen {
+				return fmt.Errorf("proto: wrong wireType = %d for field Message", wireType)
+			}
+			var length int
+			length, pos, err = proto.ConsumeLen(buf, pos)
+			if err != nil {
+				return err
+			}
+			startPos := pos - length
+			orig.Message = string(buf[startPos:pos])
+
+		case 3:
+			if wireType != proto.WireTypeVarint {
+				return fmt.Errorf("proto: wrong wireType = %d for field Code", wireType)
+			}
+			var num uint64
+			num, pos, err = proto.ConsumeVarint(buf, pos)
+			if err != nil {
+				return err
+			}
+
+			orig.Code = otlptrace.Status_StatusCode(num)
+		default:
+			pos, err = proto.ConsumeUnknown(buf, pos, wireType)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
 }
