diff --git a/pdata/internal/generated_wrapper_scopelogs.go b/pdata/internal/generated_wrapper_scopelogs.go
index 28bb03792..09fb4ae35 100644
--- a/pdata/internal/generated_wrapper_scopelogs.go
+++ b/pdata/internal/generated_wrapper_scopelogs.go
@@ -7,11 +7,21 @@
 package internal
 
 import (
+	"fmt"
+
 	otlplogs "go.opentelemetry.io/collector/pdata/internal/data/protogen/logs/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
+func NewOrigScopeLogs() otlplogs.ScopeLogs {
+	return otlplogs.ScopeLogs{}
+}
+
+func NewOrigPtrScopeLogs() *otlplogs.ScopeLogs {
+	return &otlplogs.ScopeLogs{}
+}
+
 func CopyOrigScopeLogs(dest, src *otlplogs.ScopeLogs) {
 	CopyOrigInstrumentationScope(&dest.Scope, &src.Scope)
 	dest.LogRecords = CopyOrigLogRecordSlice(dest.LogRecords, src.LogRecords)
@@ -110,5 +120,69 @@ func MarshalProtoOrigScopeLogs(orig *otlplogs.ScopeLogs, buf []byte) int {
 }
 
 func UnmarshalProtoOrigScopeLogs(orig *otlplogs.ScopeLogs, buf []byte) error {
-	return orig.Unmarshal(buf)
+	var err error
+	var fieldNum int32
+	var wireType proto.WireType
+
+	l := len(buf)
+	pos := 0
+	for pos < l {
+		// If in a group parsing, move to the next tag.
+		fieldNum, wireType, pos, err = proto.ConsumeTag(buf, pos)
+		if err != nil {
+			return err
+		}
+		switch fieldNum {
+
+		case 1:
+			if wireType != proto.WireTypeLen {
+				return fmt.Errorf("proto: wrong wireType = %d for field Scope", wireType)
+			}
+			var length int
+			length, pos, err = proto.ConsumeLen(buf, pos)
+			if err != nil {
+				return err
+			}
+			startPos := pos - length
+
+			err = UnmarshalProtoOrigInstrumentationScope(&orig.Scope, buf[startPos:pos])
+			if err != nil {
+				return err
+			}
+
+		case 2:
+			if wireType != proto.WireTypeLen {
+				return fmt.Errorf("proto: wrong wireType = %d for field LogRecords", wireType)
+			}
+			var length int
+			length, pos, err = proto.ConsumeLen(buf, pos)
+			if err != nil {
+				return err
+			}
+			startPos := pos - length
+			orig.LogRecords = append(orig.LogRecords, NewOrigPtrLogRecord())
+			err = UnmarshalProtoOrigLogRecord(orig.LogRecords[len(orig.LogRecords)-1], buf[startPos:pos])
+			if err != nil {
+				return err
+			}
+
+		case 3:
+			if wireType != proto.WireTypeLen {
+				return fmt.Errorf("proto: wrong wireType = %d for field SchemaUrl", wireType)
+			}
+			var length int
+			length, pos, err = proto.ConsumeLen(buf, pos)
+			if err != nil {
+				return err
+			}
+			startPos := pos - length
+			orig.SchemaUrl = string(buf[startPos:pos])
+		default:
+			pos, err = proto.ConsumeUnknown(buf, pos, wireType)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
 }
