diff --git a/pdata/internal/generated_wrapper_link.go b/pdata/internal/generated_wrapper_link.go
index 789f43654..82b2e90c9 100644
--- a/pdata/internal/generated_wrapper_link.go
+++ b/pdata/internal/generated_wrapper_link.go
@@ -7,12 +7,22 @@
 package internal
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/pdata/internal/data"
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
+func NewOrigLink() otlpprofiles.Link {
+	return otlpprofiles.Link{}
+}
+
+func NewOrigPtrLink() *otlpprofiles.Link {
+	return &otlpprofiles.Link{}
+}
+
 func CopyOrigLink(dest, src *otlpprofiles.Link) {
 	dest.TraceId = src.TraceId
 	dest.SpanId = src.SpanId
@@ -84,5 +94,57 @@ func MarshalProtoOrigLink(orig *otlpprofiles.Link, buf []byte) int {
 }
 
 func UnmarshalProtoOrigLink(orig *otlpprofiles.Link, buf []byte) error {
-	return orig.Unmarshal(buf)
+	var err error
+	var fieldNum int32
+	var wireType proto.WireType
+
+	l := len(buf)
+	pos := 0
+	for pos < l {
+		// If in a group parsing, move to the next tag.
+		fieldNum, wireType, pos, err = proto.ConsumeTag(buf, pos)
+		if err != nil {
+			return err
+		}
+		switch fieldNum {
+
+		case 1:
+			if wireType != proto.WireTypeLen {
+				return fmt.Errorf("proto: wrong wireType = %d for field TraceId", wireType)
+			}
+			var length int
+			length, pos, err = proto.ConsumeLen(buf, pos)
+			if err != nil {
+				return err
+			}
+			startPos := pos - length
+
+			err = UnmarshalProtoOrigTraceID(&orig.TraceId, buf[startPos:pos])
+			if err != nil {
+				return err
+			}
+
+		case 2:
+			if wireType != proto.WireTypeLen {
+				return fmt.Errorf("proto: wrong wireType = %d for field SpanId", wireType)
+			}
+			var length int
+			length, pos, err = proto.ConsumeLen(buf, pos)
+			if err != nil {
+				return err
+			}
+			startPos := pos - length
+
+			err = UnmarshalProtoOrigSpanID(&orig.SpanId, buf[startPos:pos])
+			if err != nil {
+				return err
+			}
+		default:
+			pos, err = proto.ConsumeUnknown(buf, pos, wireType)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
 }
