diff --git a/translator/trace/protospan_translation.go b/translator/trace/protospan_translation.go
index 07ea690ff..116bab83f 100644
--- a/translator/trace/protospan_translation.go
+++ b/translator/trace/protospan_translation.go
@@ -14,7 +14,7 @@
 
 package tracetranslator
 
-// Some of the keys used to represent OC proto constructs as tags or annotations in other formats.
+// Some of the keys used to represent OTLP constructs as tags or annotations in other formats.
 const (
 	AnnotationDescriptionKey = "description"
 
@@ -23,6 +23,8 @@ const (
 	MessageEventCompressedSizeKey   = "message.compressed_size"
 	MessageEventUncompressedSizeKey = "message.uncompressed_size"
 
+	TagMessage = "message"
+
 	TagSpanKind = "span.kind"
 
 	TagStatusCode          = "status.code"
@@ -33,14 +35,21 @@ const (
 	TagZipkinCensusCode    = "census.status_code"
 	TagZipkinCensusMsg     = "census.status_description"
 	TagZipkinOpenCensusMsg = "opencensus.status_description"
+
+	TagW3CTraceState = "w3c.tracestate"
+)
+
+// Constants used for signifying batch-level attribute values where not supplied by OTLP data but required
+// by other protocols.
+const (
+	ResourceNotSet  = "OTLPResourceNotSet"
+	ResourceNoAttrs = "OTLPResourceNoAttributes"
 )
 
 // OpenTracingSpanKind are possible values for TagSpanKind and match the OpenTracing
 // conventions: https://github.com/opentracing/specification/blob/master/semantic_conventions.md
-// These values are also used for representing internally span kinds that have no
+// These values are used for representing span kinds that have no
 // equivalents in OpenCensus format. They are stored as values of TagSpanKind
-// Note: this internal usage needs to be eliminated when we move to OTLP for internal
-// in-memory representation since OTLP has the equivalents.
 type OpenTracingSpanKind string
 
 const (
