diff --git a/consumer/pdata/metric_test.go b/consumer/pdata/metric_test.go
index 6414b0dd6..54729466b 100644
--- a/consumer/pdata/metric_test.go
+++ b/consumer/pdata/metric_test.go
@@ -168,11 +168,9 @@ func TestMetricCount(t *testing.T) {
 
 	rms := md.ResourceMetrics()
 	rms.Resize(3)
-	rms.At(0).InstrumentationLibraryMetrics().Resize(1)
-	rms.At(0).InstrumentationLibraryMetrics().At(0).Metrics().Resize(1)
-	rms.At(1).InstrumentationLibraryMetrics().Resize(1)
-	rms.At(2).InstrumentationLibraryMetrics().Resize(1)
-	rms.At(2).InstrumentationLibraryMetrics().At(0).Metrics().Resize(5)
+	rms.At(1).InstrumentationLibraryMetrics().AppendEmpty()
+	rms.At(2).InstrumentationLibraryMetrics().AppendEmpty().Metrics().Resize(5)
+	// 5 + 1 (from rms.At(0) initialized first)
 	assert.EqualValues(t, 6, md.MetricCount())
 }
 
@@ -180,15 +178,12 @@ func TestMetricSize(t *testing.T) {
 	md := NewMetrics()
 	assert.Equal(t, 0, md.OtlpProtoSize())
 	rms := md.ResourceMetrics()
-	rms.Resize(1)
-	rms.At(0).InstrumentationLibraryMetrics().Resize(1)
-	rms.At(0).InstrumentationLibraryMetrics().At(0).Metrics().Resize(1)
-	metric := rms.At(0).InstrumentationLibraryMetrics().At(0).Metrics().At(0)
+	metric := rms.AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty().Metrics().AppendEmpty()
 	metric.SetDataType(MetricDataTypeHistogram)
 	doubleHistogram := metric.Histogram()
-	doubleHistogram.DataPoints().Resize(1)
-	doubleHistogram.DataPoints().At(0).SetCount(123)
-	doubleHistogram.DataPoints().At(0).SetSum(123)
+	pt := doubleHistogram.DataPoints().AppendEmpty()
+	pt.SetCount(123)
+	pt.SetSum(123)
 	otlp := internal.MetricsToOtlp(md.InternalRep())
 	size := otlp.Size()
 	bytes, err := otlp.Marshal()
@@ -280,10 +275,7 @@ func TestMetricAndDataPointCountWithEmpty(t *testing.T) {
 
 func TestMetricAndDataPointCountWithNilDataPoints(t *testing.T) {
 	metrics := NewMetrics()
-	metrics.ResourceMetrics().Resize(1)
-	rm := metrics.ResourceMetrics().At(0)
-	rm.InstrumentationLibraryMetrics().Resize(1)
-	ilm := rm.InstrumentationLibraryMetrics().At(0)
+	ilm := metrics.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
 	intGauge := ilm.Metrics().AppendEmpty()
 	intGauge.SetDataType(MetricDataTypeIntGauge)
 	doubleGauge := ilm.Metrics().AppendEmpty()
