diff --git a/receiver/jaegerreceiver/config_test.go b/receiver/jaegerreceiver/config_test.go
index eb6060918..5159461e4 100644
--- a/receiver/jaegerreceiver/config_test.go
+++ b/receiver/jaegerreceiver/config_test.go
@@ -76,6 +76,9 @@ func TestLoadConfig(t *testing.T) {
 					},
 				},
 			},
+			RemoteSampling: &RemoteSamplingConfig{
+				FetchEndpoint: "jaeger-collector:1234",
+			},
 		})
 
 	tlsConfig := cfg.Receivers["jaeger/tls"].(*Config)
