diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index cb3c1243e..0d3278455 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -18,7 +18,7 @@ require (
 	go.opentelemetry.io/collector/config/configtls v1.24.0
 	go.opentelemetry.io/collector/extension/auth v0.118.0
 	go.opentelemetry.io/collector/extension/auth/authtest v0.118.0
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.56.0
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.59.0
 	go.opentelemetry.io/otel v1.34.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
