diff --git a/internal/memorylimiter/go.mod b/internal/memorylimiter/go.mod
index 90beb22b1..4103f3701 100644
--- a/internal/memorylimiter/go.mod
+++ b/internal/memorylimiter/go.mod
@@ -4,7 +4,7 @@ go 1.22.0
 
 require (
 	github.com/shirou/gopsutil/v4 v4.24.10
-	github.com/stretchr/testify v1.9.0
+	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v0.114.0
 	go.opentelemetry.io/collector/confmap v1.20.0
 	go.uber.org/goleak v1.3.0
