diff --git a/internal/data/testdata/trace.go b/internal/data/testdata/trace.go
index 1c833185a..0e8d178bd 100644
--- a/internal/data/testdata/trace.go
+++ b/internal/data/testdata/trace.go
@@ -22,14 +22,14 @@ import (
 )
 
 var (
-	TestStartTime      = time.Date(2020, 2, 11, 20, 26, 12, 321, time.UTC)
-	TestStartTimestamp = data.TimestampUnixNano(TestStartTime.UnixNano())
+	TestSpanStartTime      = time.Date(2020, 2, 11, 20, 26, 12, 321, time.UTC)
+	TestSpanStartTimestamp = data.TimestampUnixNano(TestSpanStartTime.UnixNano())
 
-	TestEventTime      = time.Date(2020, 2, 11, 20, 26, 13, 123, time.UTC)
-	TestEventTimestamp = data.TimestampUnixNano(TestEventTime.UnixNano())
+	TestSpanEventTime      = time.Date(2020, 2, 11, 20, 26, 13, 123, time.UTC)
+	TestSpanEventTimestamp = data.TimestampUnixNano(TestSpanEventTime.UnixNano())
 
-	TestEndTime      = time.Date(2020, 2, 11, 20, 26, 13, 789, time.UTC)
-	TestEndTimestamp = data.TimestampUnixNano(TestEndTime.UnixNano())
+	TestSpanEndTime      = time.Date(2020, 2, 11, 20, 26, 13, 789, time.UTC)
+	TestSpanEndTimestamp = data.TimestampUnixNano(TestSpanEndTime.UnixNano())
 )
 
 // TraceTestCase represents one test case with the coresponding TraceData and OTLP representation.
@@ -47,37 +47,37 @@ var (
 	}
 	OneEmptyResourceSpansTraceTestCase = TraceTestCase{
 		Name:      "one-empty-resource-spans",
-		TraceData: generateTraceDataOneEmptyResourceSpans(),
+		TraceData: GenerateTraceDataOneEmptyResourceSpans(),
 		OtlpData:  generateTraceOtlpOneEmptyResourceSpans(),
 	}
 	NoLibrariesTraceTestCase = TraceTestCase{
 		Name:      "no-libraries",
-		TraceData: generateTraceDataNoLibraries(),
+		TraceData: GenerateTraceDataNoLibraries(),
 		OtlpData:  generateTraceOtlpNoLibraries(),
 	}
 	NoSpansTraceTestCase = TraceTestCase{
 		Name:      "no-spans",
-		TraceData: generateTraceDataNoSpans(),
+		TraceData: GenerateTraceDataNoSpans(),
 		OtlpData:  generateTraceOtlpNoSpans(),
 	}
 	OneSpanNoResourceTraceTestCase = TraceTestCase{
 		Name:      "one-span-no-resource",
-		TraceData: generateTraceDataOneSpanNoResource(),
+		TraceData: GenerateTraceDataOneSpanNoResource(),
 		OtlpData:  generateTraceOtlpOneSpanNoResource(),
 	}
 	OneSpanTraceTestCase = TraceTestCase{
 		Name:      "one-span",
-		TraceData: generateTraceDataOneSpan(),
+		TraceData: GenerateTraceDataOneSpan(),
 		OtlpData:  generateTraceOtlpOneSpan(),
 	}
 	TwoSpansSameResourceTraceTestCase = TraceTestCase{
 		Name:      "two-spans-same-resource",
-		TraceData: generateTraceDataSameResourceTwoSpans(),
+		TraceData: GenerateTraceDataSameResourceTwoSpans(),
 		OtlpData:  generateTraceOtlpSameResourceTwoSpans(),
 	}
 	TwoSpansSameResourceOneDifferentTraceTestCase = TraceTestCase{
 		Name:      "two-spans-same-resource-one-different",
-		TraceData: generateTraceDataTwoSpansSameResourceOneDifferent(),
+		TraceData: GenerateTraceDataTwoSpansSameResourceOneDifferent(),
 		OtlpData:  generateTraceOtlpTwoSpansSameResourceOneDifferent(),
 	}
 
@@ -94,27 +94,27 @@ var (
 	}
 )
 
-func generateTraceDataOneEmptyResourceSpans() data.TraceData {
+func GenerateTraceDataOneEmptyResourceSpans() data.TraceData {
 	td := data.NewTraceData()
 	td.SetResourceSpans(data.NewResourceSpansSlice(1))
 	return td
 }
 
-// generateTraceOtlpOneEmptyResourceSpans returns the OTLP representation of the generateTraceDataOneEmptyResourceSpans
+// generateTraceOtlpOneEmptyResourceSpans returns the OTLP representation of the GenerateTraceDataOneEmptyResourceSpans
 func generateTraceOtlpOneEmptyResourceSpans() []*otlptrace.ResourceSpans {
 	return []*otlptrace.ResourceSpans{
 		{},
 	}
 }
 
-func generateTraceDataNoLibraries() data.TraceData {
-	td := generateTraceDataOneEmptyResourceSpans()
+func GenerateTraceDataNoLibraries() data.TraceData {
+	td := GenerateTraceDataOneEmptyResourceSpans()
 	rs0 := td.ResourceSpans().Get(0)
 	fillResource1(rs0.Resource())
 	return td
 }
 
-// generateTraceOtlpDataNoLibraries returns the OTLP representation of the generateTraceDataNoLibraries.
+// generateTraceOtlpDataNoLibraries returns the OTLP representation of the GenerateTraceDataNoLibraries.
 func generateTraceOtlpNoLibraries() []*otlptrace.ResourceSpans {
 	return []*otlptrace.ResourceSpans{
 		{
@@ -123,14 +123,14 @@ func generateTraceOtlpNoLibraries() []*otlptrace.ResourceSpans {
 	}
 }
 
-func generateTraceDataNoSpans() data.TraceData {
-	td := generateTraceDataNoLibraries()
+func GenerateTraceDataNoSpans() data.TraceData {
+	td := GenerateTraceDataNoLibraries()
 	rs0 := td.ResourceSpans().Get(0)
 	rs0.SetInstrumentationLibrarySpans(data.NewInstrumentationLibrarySpansSlice(1))
 	return td
 }
 
-// generateTraceOtlpNoSpans returns the OTLP representation of the generateTraceDataNoSpans.
+// generateTraceOtlpNoSpans returns the OTLP representation of the GenerateTraceDataNoSpans.
 func generateTraceOtlpNoSpans() []*otlptrace.ResourceSpans {
 	return []*otlptrace.ResourceSpans{
 		{
@@ -142,8 +142,8 @@ func generateTraceOtlpNoSpans() []*otlptrace.ResourceSpans {
 	}
 }
 
-func generateTraceDataOneSpanNoResource() data.TraceData {
-	td := generateTraceDataOneEmptyResourceSpans()
+func GenerateTraceDataOneSpanNoResource() data.TraceData {
+	td := GenerateTraceDataOneEmptyResourceSpans()
 	rs0 := td.ResourceSpans().Get(0)
 	rs0.SetInstrumentationLibrarySpans(data.NewInstrumentationLibrarySpansSlice(1))
 	rs0ils0 := rs0.InstrumentationLibrarySpans().Get(0)
@@ -152,7 +152,7 @@ func generateTraceDataOneSpanNoResource() data.TraceData {
 	return td
 }
 
-// generateTraceOtlpOneSpanNoResource returns the OTLP representation of the generateTraceDataOneSpanNoResource.
+// generateTraceOtlpOneSpanNoResource returns the OTLP representation of the GenerateTraceDataOneSpanNoResource.
 func generateTraceOtlpOneSpanNoResource() []*otlptrace.ResourceSpans {
 	return []*otlptrace.ResourceSpans{
 		{
@@ -167,15 +167,15 @@ func generateTraceOtlpOneSpanNoResource() []*otlptrace.ResourceSpans {
 	}
 }
 
-func generateTraceDataOneSpan() data.TraceData {
-	td := generateTraceDataNoSpans()
+func GenerateTraceDataOneSpan() data.TraceData {
+	td := GenerateTraceDataNoSpans()
 	rs0ils0 := td.ResourceSpans().Get(0).InstrumentationLibrarySpans().Get(0)
 	rs0ils0.SetSpans(data.NewSpanSlice(1))
 	fillSpanOne(rs0ils0.Spans().Get(0))
 	return td
 }
 
-// generateTraceOtlpOneSpan returns the OTLP representation of the generateTraceDataOneSpan.
+// generateTraceOtlpOneSpan returns the OTLP representation of the GenerateTraceDataOneSpan.
 func generateTraceOtlpOneSpan() []*otlptrace.ResourceSpans {
 	return []*otlptrace.ResourceSpans{
 		{
@@ -191,8 +191,8 @@ func generateTraceOtlpOneSpan() []*otlptrace.ResourceSpans {
 	}
 }
 
-func generateTraceDataSameResourceTwoSpans() data.TraceData {
-	td := generateTraceDataNoSpans()
+func GenerateTraceDataSameResourceTwoSpans() data.TraceData {
+	td := GenerateTraceDataNoSpans()
 	rs0ils0 := td.ResourceSpans().Get(0).InstrumentationLibrarySpans().Get(0)
 	rs0ils0.SetSpans(data.NewSpanSlice(2))
 	fillSpanOne(rs0ils0.Spans().Get(0))
@@ -217,11 +217,11 @@ func generateTraceOtlpSameResourceTwoSpans() []*otlptrace.ResourceSpans {
 	}
 }
 
-func generateTraceDataTwoSpansSameResourceOneDifferent() data.TraceData {
+func GenerateTraceDataTwoSpansSameResourceOneDifferent() data.TraceData {
 	td := data.NewTraceData()
 	td.SetResourceSpans(data.NewResourceSpansSlice(2))
 	rs0 := td.ResourceSpans().Get(0)
-	rs0.Resource().SetAttributes(generateResourceAttributes1())
+	fillResource1(rs0.Resource())
 	rs0.SetInstrumentationLibrarySpans(data.NewInstrumentationLibrarySpansSlice(1))
 	rs0ils0 := rs0.InstrumentationLibrarySpans().Get(0)
 	rs0ils0.SetSpans(data.NewSpanSlice(2))
@@ -236,7 +236,7 @@ func generateTraceDataTwoSpansSameResourceOneDifferent() data.TraceData {
 	return td
 }
 
-// generateTraceOtlpTwoSpansSameResourceOneDifferent returns the OTLP representation of the generateTraceDataTwoSpansSameResourceOneDifferent.
+// generateTraceOtlpTwoSpansSameResourceOneDifferent returns the OTLP representation of the GenerateTraceDataTwoSpansSameResourceOneDifferent.
 func generateTraceOtlpTwoSpansSameResourceOneDifferent() []*otlptrace.ResourceSpans {
 	return []*otlptrace.ResourceSpans{
 		{
@@ -265,17 +265,17 @@ func generateTraceOtlpTwoSpansSameResourceOneDifferent() []*otlptrace.ResourceSp
 
 func fillSpanOne(span data.Span) {
 	span.SetName("operationA")
-	span.SetStartTime(TestStartTimestamp)
-	span.SetEndTime(TestEndTimestamp)
+	span.SetStartTime(TestSpanStartTimestamp)
+	span.SetEndTime(TestSpanEndTimestamp)
 	span.SetDroppedAttributesCount(1)
 	span.SetEvents(data.NewSpanEventSlice(2))
 	ev0 := span.Events().Get(0)
-	ev0.SetTimestamp(TestEventTimestamp)
+	ev0.SetTimestamp(TestSpanEventTimestamp)
 	ev0.SetName("event-with-attr")
 	ev0.SetAttributes(generateSpanEventAttributes())
 	ev0.SetDroppedAttributesCount(2)
 	ev1 := span.Events().Get(1)
-	ev1.SetTimestamp(TestEventTimestamp)
+	ev1.SetTimestamp(TestSpanEventTimestamp)
 	ev1.SetName("event")
 	ev1.SetDroppedAttributesCount(2)
 	span.SetDroppedEventsCount(1)
@@ -286,19 +286,19 @@ func fillSpanOne(span data.Span) {
 func generateOtlpSpanOne() *otlptrace.Span {
 	return &otlptrace.Span{
 		Name:                   "operationA",
-		StartTimeUnixNano:      uint64(TestStartTimestamp),
-		EndTimeUnixNano:        uint64(TestEndTimestamp),
+		StartTimeUnixNano:      uint64(TestSpanStartTimestamp),
+		EndTimeUnixNano:        uint64(TestSpanEndTimestamp),
 		DroppedAttributesCount: 1,
 		Events: []*otlptrace.Span_Event{
 			{
 				Name:                   "event-with-attr",
-				TimeUnixNano:           uint64(TestEventTimestamp),
+				TimeUnixNano:           uint64(TestSpanEventTimestamp),
 				Attributes:             generateOtlpSpanEventAttributes(),
 				DroppedAttributesCount: 2,
 			},
 			{
 				Name:                   "event",
-				TimeUnixNano:           uint64(TestEventTimestamp),
+				TimeUnixNano:           uint64(TestSpanEventTimestamp),
 				DroppedAttributesCount: 2,
 			},
 		},
@@ -312,8 +312,8 @@ func generateOtlpSpanOne() *otlptrace.Span {
 
 func fillSpanTwo(span data.Span) {
 	span.SetName("operationB")
-	span.SetStartTime(TestStartTimestamp)
-	span.SetEndTime(TestEndTimestamp)
+	span.SetStartTime(TestSpanStartTimestamp)
+	span.SetEndTime(TestSpanEndTimestamp)
 	span.SetLinks(data.NewSpanLinkSlice(2))
 	span.Links().Get(0).SetAttributes(generateSpanLinkAttributes())
 	span.Links().Get(0).SetDroppedAttributesCount(4)
@@ -324,8 +324,8 @@ func fillSpanTwo(span data.Span) {
 func generateOtlpSpanTwo() *otlptrace.Span {
 	return &otlptrace.Span{
 		Name:              "operationB",
-		StartTimeUnixNano: uint64(TestStartTimestamp),
-		EndTimeUnixNano:   uint64(TestEndTimestamp),
+		StartTimeUnixNano: uint64(TestSpanStartTimestamp),
+		EndTimeUnixNano:   uint64(TestSpanEndTimestamp),
 		Links: []*otlptrace.Span_Link{
 			{
 				Attributes:             generateOtlpSpanLinkAttributes(),
@@ -341,17 +341,17 @@ func generateOtlpSpanTwo() *otlptrace.Span {
 
 func fillSpanThree(span data.Span) {
 	span.SetName("operationC")
-	span.SetStartTime(TestStartTimestamp)
-	span.SetEndTime(TestEndTimestamp)
-	span.SetAttributes(generateSpanAttributes())
+	span.SetStartTime(TestSpanStartTimestamp)
+	span.SetEndTime(TestSpanEndTimestamp)
+	span.SetAttributes(data.NewAttributeMap(spanAttributes))
 	span.SetDroppedAttributesCount(5)
 }
 
 func generateOtlpSpanThree() *otlptrace.Span {
 	return &otlptrace.Span{
 		Name:                   "operationC",
-		StartTimeUnixNano:      uint64(TestStartTimestamp),
-		EndTimeUnixNano:        uint64(TestEndTimestamp),
+		StartTimeUnixNano:      uint64(TestSpanStartTimestamp),
+		EndTimeUnixNano:        uint64(TestSpanEndTimestamp),
 		Attributes:             generateOtlpSpanAttributes(),
 		DroppedAttributesCount: 5,
 	}
