diff --git a/processor/spanprocessor/span_test.go b/processor/spanprocessor/span_test.go
index c8b68a505..d5b0c9cdf 100644
--- a/processor/spanprocessor/span_test.go
+++ b/processor/spanprocessor/span_test.go
@@ -27,6 +27,7 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/exporter/exportertest"
 	"go.opentelemetry.io/collector/internal/data/testdata"
+	"go.opentelemetry.io/collector/internal/processor/filterset"
 	"go.opentelemetry.io/collector/internal/processor/filterspan"
 	"go.opentelemetry.io/collector/translator/conventions"
 )
@@ -149,8 +150,8 @@ func TestSpanProcessor_NilEmptyData(t *testing.T) {
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Include = &filterspan.MatchProperties{
-		MatchType: "strict",
-		Services:  []string{"service"},
+		Config:   *createMatchConfig(filterset.Strict),
+		Services: []string{"service"},
 	}
 	oCfg.Rename.FromAttributes = []string{"key"}
 
@@ -607,12 +608,12 @@ func TestSpanProcessor_skipSpan(t *testing.T) {
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Include = &filterspan.MatchProperties{
-		MatchType: filterspan.MatchTypeRegexp,
+		Config:    *createMatchConfig(filterset.Regexp),
 		Services:  []string{`^banks$`},
 		SpanNames: []string{"/"},
 	}
 	oCfg.Exclude = &filterspan.MatchProperties{
-		MatchType: filterspan.MatchTypeStrict,
+		Config:    *createMatchConfig(filterset.Strict),
 		SpanNames: []string{`donot/change`},
 	}
 	oCfg.Rename.ToAttributes = &ToAttributes{
