diff --git a/component/receiver_test.go b/component/receiver_test.go
index 2bd811b56..bbec820e4 100644
--- a/component/receiver_test.go
+++ b/component/receiver_test.go
@@ -15,17 +15,15 @@
 package component
 
 import (
-	"context"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
-	"go.opentelemetry.io/collector/component/componenterror"
 	"go.opentelemetry.io/collector/config"
-	"go.opentelemetry.io/collector/consumer"
 )
 
 type TestReceiverFactory struct {
+	ReceiverFactory
 	name config.Type
 }
 
@@ -34,26 +32,6 @@ func (f *TestReceiverFactory) Type() config.Type {
 	return f.name
 }
 
-// CreateDefaultConfig creates the default configuration for the Receiver.
-func (f *TestReceiverFactory) CreateDefaultConfig() config.Receiver {
-	return nil
-}
-
-// CreateTracesReceiver creates a trace receiver based on this config.
-func (f *TestReceiverFactory) CreateTracesReceiver(context.Context, ReceiverCreateSettings, config.Receiver, consumer.Traces) (TracesReceiver, error) {
-	return nil, componenterror.ErrDataTypeIsNotSupported
-}
-
-// CreateMetricsReceiver creates a metrics receiver based on this config.
-func (f *TestReceiverFactory) CreateMetricsReceiver(context.Context, ReceiverCreateSettings, config.Receiver, consumer.Metrics) (MetricsReceiver, error) {
-	return nil, componenterror.ErrDataTypeIsNotSupported
-}
-
-// CreateMetricsReceiver creates a metrics receiver based on this config.
-func (f *TestReceiverFactory) CreateLogsReceiver(context.Context, ReceiverCreateSettings, config.Receiver, consumer.Logs) (LogsReceiver, error) {
-	return nil, componenterror.ErrDataTypeIsNotSupported
-}
-
 func TestBuildReceivers(t *testing.T) {
 	type testCase struct {
 		in  []ReceiverFactory
@@ -63,18 +41,18 @@ func TestBuildReceivers(t *testing.T) {
 	testCases := []testCase{
 		{
 			in: []ReceiverFactory{
-				&TestReceiverFactory{"e1"},
-				&TestReceiverFactory{"e2"},
+				&TestReceiverFactory{name: "e1"},
+				&TestReceiverFactory{name: "e2"},
 			},
 			out: map[config.Type]ReceiverFactory{
-				"e1": &TestReceiverFactory{"e1"},
-				"e2": &TestReceiverFactory{"e2"},
+				"e1": &TestReceiverFactory{name: "e1"},
+				"e2": &TestReceiverFactory{name: "e2"},
 			},
 		},
 		{
 			in: []ReceiverFactory{
-				&TestReceiverFactory{"e1"},
-				&TestReceiverFactory{"e1"},
+				&TestReceiverFactory{name: "e1"},
+				&TestReceiverFactory{name: "e1"},
 			},
 		},
 	}
