diff --git a/component/receiver.go b/component/receiver.go
index 3f5511c4d..421c0268a 100644
--- a/component/receiver.go
+++ b/component/receiver.go
@@ -99,6 +99,9 @@ type ReceiverCreateSettings struct {
 
 // ReceiverFactory can create TracesReceiver, MetricsReceiver and
 // and LogsReceiver. This is the new preferred factory type to create receivers.
+//
+// This interface cannot be directly implemented. Implementations must
+// use the receiverhelper.NewFactory to implement it.
 type ReceiverFactory interface {
 	Factory
 
