diff --git a/service/internal/builders/exporter.go b/service/internal/builders/exporter.go
index bb2367d22..9520eb4ed 100644
--- a/service/internal/builders/exporter.go
+++ b/service/internal/builders/exporter.go
@@ -37,6 +37,7 @@ func (b *ExporterBuilder) CreateTraces(ctx context.Context, set exporter.Setting
 		return nil, fmt.Errorf("exporter factory not available for: %q", set.ID)
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.TracesStability())
 	return f.CreateTraces(ctx, set, cfg)
 }
@@ -53,6 +54,7 @@ func (b *ExporterBuilder) CreateMetrics(ctx context.Context, set exporter.Settin
 		return nil, fmt.Errorf("exporter factory not available for: %q", set.ID)
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.MetricsStability())
 	return f.CreateMetrics(ctx, set, cfg)
 }
@@ -69,6 +71,7 @@ func (b *ExporterBuilder) CreateLogs(ctx context.Context, set exporter.Settings)
 		return nil, fmt.Errorf("exporter factory not available for: %q", set.ID)
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.LogsStability())
 	return f.CreateLogs(ctx, set, cfg)
 }
@@ -90,6 +93,7 @@ func (b *ExporterBuilder) CreateProfiles(ctx context.Context, set exporter.Setti
 		return nil, pipeline.ErrSignalNotSupported
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.ProfilesStability())
 	return f.CreateProfiles(ctx, set, cfg)
 }
