diff --git a/service/internal/builders/receiver.go b/service/internal/builders/receiver.go
index 256839262..2277b1b4c 100644
--- a/service/internal/builders/receiver.go
+++ b/service/internal/builders/receiver.go
@@ -44,6 +44,7 @@ func (b *ReceiverBuilder) CreateTraces(ctx context.Context, set receiver.Setting
 		return nil, fmt.Errorf("receiver factory not available for: %q", set.ID)
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.TracesStability())
 	return f.CreateTraces(ctx, set, cfg, next)
 }
@@ -63,6 +64,7 @@ func (b *ReceiverBuilder) CreateMetrics(ctx context.Context, set receiver.Settin
 		return nil, fmt.Errorf("receiver factory not available for: %q", set.ID)
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.MetricsStability())
 	return f.CreateMetrics(ctx, set, cfg, next)
 }
@@ -82,6 +84,7 @@ func (b *ReceiverBuilder) CreateLogs(ctx context.Context, set receiver.Settings,
 		return nil, fmt.Errorf("receiver factory not available for: %q", set.ID)
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.LogsStability())
 	return f.CreateLogs(ctx, set, cfg, next)
 }
@@ -106,6 +109,7 @@ func (b *ReceiverBuilder) CreateProfiles(ctx context.Context, set receiver.Setti
 		return nil, pipeline.ErrSignalNotSupported
 	}
 
+	logDeprecatedTypeAlias(set.Logger, f, set.ID.Type())
 	logStabilityLevel(set.Logger, f.ProfilesStability())
 	return f.CreateProfiles(ctx, set, cfg, next)
 }
