diff --git a/confmap/provider/envprovider/go.mod b/confmap/provider/envprovider/go.mod
index 6149ddd9f..8a63caa16 100644
--- a/confmap/provider/envprovider/go.mod
+++ b/confmap/provider/envprovider/go.mod
@@ -4,7 +4,7 @@ go 1.21.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/confmap v0.102.0
+	go.opentelemetry.io/collector/confmap v0.102.1
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
 )
