diff --git a/Makefile.Common b/Makefile.Common
index daf1e14dc..662c55190 100644
--- a/Makefile.Common
+++ b/Makefile.Common
@@ -18,10 +18,6 @@ GH := $(shell which gh)
 test:
 	$(GOTEST) $(GOTEST_OPT) ./...
 
-.PHONY: test-unstable
-test-unstable:
-	$(GOTEST) $(GOTEST_OPT) -tags enable_unstable ./...
-
 .PHONY: test-with-cover
 test-with-cover:
 	$(GO_ACC) --output=coverage.out ./...
@@ -44,10 +40,6 @@ tidy:
 lint:
 	$(LINT) run
 
-.PHONY: lint-unstable
-lint-unstable:
-	$(LINT) run --build-tags enable_unstable
-
 .PHONY: generate
 generate:
 	$(GOCMD) generate ./...
