diff --git a/receiver/prometheusreceiver/internal/ocastore.go b/receiver/prometheusreceiver/internal/ocastore.go
index 1d404e860..765949690 100644
--- a/receiver/prometheusreceiver/internal/ocastore.go
+++ b/receiver/prometheusreceiver/internal/ocastore.go
@@ -99,11 +99,11 @@ func (o *ocaStore) Close() error {
 // noopAppender, always return error on any operations
 type noopAppender struct{}
 
-func (*noopAppender) Add(l labels.Labels, t int64, v float64) (uint64, error) {
+func (*noopAppender) Add(labels.Labels, int64, float64) (uint64, error) {
 	return 0, componenterror.ErrAlreadyStopped
 }
 
-func (*noopAppender) AddFast(ref uint64, t int64, v float64) error {
+func (*noopAppender) AddFast(uint64, int64, float64) error {
 	return componenterror.ErrAlreadyStopped
 }
 
