diff --git a/service/service.go b/service/service.go
index 526cbc16e..986e6daa8 100644
--- a/service/service.go
+++ b/service/service.go
@@ -18,7 +18,6 @@ import (
 	"context"
 	"fmt"
 
-	"go.opentelemetry.io/contrib/zpages"
 	"go.uber.org/multierr"
 
 	"go.opentelemetry.io/collector/component"
@@ -29,31 +28,26 @@ import (
 
 // service represents the implementation of a component.Host.
 type service struct {
-	factories           component.Factories
-	buildInfo           component.BuildInfo
-	config              *config.Config
-	telemetry           component.TelemetrySettings
-	zPagesSpanProcessor *zpages.SpanProcessor
-	asyncErrorChannel   chan error
-
-	builtExporters  builder.Exporters
-	builtReceivers  builder.Receivers
-	builtPipelines  builder.BuiltPipelines
-	builtExtensions extensions.Extensions
+	buildInfo component.BuildInfo
+	config    *config.Config
+	telemetry component.TelemetrySettings
+	host      *serviceHost
 }
 
 func newService(set *svcSettings) (*service, error) {
 	srv := &service{
-		factories:           set.Factories,
-		buildInfo:           set.BuildInfo,
-		config:              set.Config,
-		telemetry:           set.Telemetry,
-		zPagesSpanProcessor: set.ZPagesSpanProcessor,
-		asyncErrorChannel:   set.AsyncErrorChannel,
+		buildInfo: set.BuildInfo,
+		config:    set.Config,
+		telemetry: set.Telemetry,
+		host: &serviceHost{
+			factories:           set.Factories,
+			zPagesSpanProcessor: set.ZPagesSpanProcessor,
+			asyncErrorChannel:   set.AsyncErrorChannel,
+		},
 	}
 
 	var err error
-	if srv.builtExtensions, err = extensions.Build(srv.telemetry, srv.buildInfo, srv.config, srv.factories.Extensions); err != nil {
+	if srv.host.builtExtensions, err = extensions.Build(srv.telemetry, srv.buildInfo, srv.config, srv.host.factories.Extensions); err != nil {
 		return nil, fmt.Errorf("cannot build extensions: %w", err)
 	}
 
@@ -61,17 +55,17 @@ func newService(set *svcSettings) (*service, error) {
 	// which are referenced before objects which reference them.
 
 	// First create exporters.
-	if srv.builtExporters, err = builder.BuildExporters(srv.telemetry, srv.buildInfo, srv.config, srv.factories.Exporters); err != nil {
+	if srv.host.builtExporters, err = builder.BuildExporters(srv.telemetry, srv.buildInfo, srv.config, srv.host.factories.Exporters); err != nil {
 		return nil, fmt.Errorf("cannot build exporters: %w", err)
 	}
 
 	// Create pipelines and their processors and plug exporters to the end of the pipelines.
-	if srv.builtPipelines, err = builder.BuildPipelines(srv.telemetry, srv.buildInfo, srv.config, srv.builtExporters, srv.factories.Processors); err != nil {
+	if srv.host.builtPipelines, err = builder.BuildPipelines(srv.telemetry, srv.buildInfo, srv.config, srv.host.builtExporters, srv.host.factories.Processors); err != nil {
 		return nil, fmt.Errorf("cannot build pipelines: %w", err)
 	}
 
 	// Create receivers and plug them into the start of the pipelines.
-	if srv.builtReceivers, err = builder.BuildReceivers(srv.telemetry, srv.buildInfo, srv.config, srv.builtPipelines, srv.factories.Receivers); err != nil {
+	if srv.host.builtReceivers, err = builder.BuildReceivers(srv.telemetry, srv.buildInfo, srv.config, srv.host.builtPipelines, srv.host.factories.Receivers); err != nil {
 		return nil, fmt.Errorf("cannot build receivers: %w", err)
 	}
 
@@ -80,33 +74,33 @@ func newService(set *svcSettings) (*service, error) {
 
 func (srv *service) Start(ctx context.Context) error {
 	srv.telemetry.Logger.Info("Starting extensions...")
-	if err := srv.builtExtensions.StartAll(ctx, srv); err != nil {
+	if err := srv.host.builtExtensions.StartAll(ctx, srv.host); err != nil {
 		return fmt.Errorf("failed to start extensions: %w", err)
 	}
 
 	srv.telemetry.Logger.Info("Starting exporters...")
-	if err := srv.builtExporters.StartAll(ctx, srv); err != nil {
+	if err := srv.host.builtExporters.StartAll(ctx, srv.host); err != nil {
 		return fmt.Errorf("cannot start exporters: %w", err)
 	}
 
 	srv.telemetry.Logger.Info("Starting processors...")
-	if err := srv.builtPipelines.StartProcessors(ctx, srv); err != nil {
+	if err := srv.host.builtPipelines.StartProcessors(ctx, srv.host); err != nil {
 		return fmt.Errorf("cannot start processors: %w", err)
 	}
 
 	srv.telemetry.Logger.Info("Starting receivers...")
-	if err := srv.builtReceivers.StartAll(ctx, srv); err != nil {
+	if err := srv.host.builtReceivers.StartAll(ctx, srv.host); err != nil {
 		return fmt.Errorf("cannot start receivers: %w", err)
 	}
 
-	return srv.builtExtensions.NotifyPipelineReady()
+	return srv.host.builtExtensions.NotifyPipelineReady()
 }
 
 func (srv *service) Shutdown(ctx context.Context) error {
 	// Accumulate errors and proceed with shutting down remaining components.
 	var errs error
 
-	if err := srv.builtExtensions.NotifyPipelineNotReady(); err != nil {
+	if err := srv.host.builtExtensions.NotifyPipelineNotReady(); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to notify that pipeline is not ready: %w", err))
 	}
 
@@ -115,53 +109,24 @@ func (srv *service) Shutdown(ctx context.Context) error {
 	// time should be part of configuration.
 
 	srv.telemetry.Logger.Info("Stopping receivers...")
-	if err := srv.builtReceivers.ShutdownAll(ctx); err != nil {
+	if err := srv.host.builtReceivers.ShutdownAll(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown receivers: %w", err))
 	}
 
 	srv.telemetry.Logger.Info("Stopping processors...")
-	if err := srv.builtPipelines.ShutdownProcessors(ctx); err != nil {
+	if err := srv.host.builtPipelines.ShutdownProcessors(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown processors: %w", err))
 	}
 
 	srv.telemetry.Logger.Info("Stopping exporters...")
-	if err := srv.builtExporters.ShutdownAll(ctx); err != nil {
+	if err := srv.host.builtExporters.ShutdownAll(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown exporters: %w", err))
 	}
 
 	srv.telemetry.Logger.Info("Stopping extensions...")
-	if err := srv.builtExtensions.ShutdownAll(ctx); err != nil {
+	if err := srv.host.builtExtensions.ShutdownAll(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown extensions: %w", err))
 	}
 
 	return errs
 }
-
-// ReportFatalError is used to report to the host that the receiver encountered
-// a fatal error (i.e.: an error that the instance can't recover from) after
-// its start function has already returned.
-func (srv *service) ReportFatalError(err error) {
-	srv.asyncErrorChannel <- err
-}
-
-func (srv *service) GetFactory(kind component.Kind, componentType config.Type) component.Factory {
-	switch kind {
-	case component.KindReceiver:
-		return srv.factories.Receivers[componentType]
-	case component.KindProcessor:
-		return srv.factories.Processors[componentType]
-	case component.KindExporter:
-		return srv.factories.Exporters[componentType]
-	case component.KindExtension:
-		return srv.factories.Extensions[componentType]
-	}
-	return nil
-}
-
-func (srv *service) GetExtensions() map[config.ComponentID]component.Extension {
-	return srv.builtExtensions.ToMap()
-}
-
-func (srv *service) GetExporters() map[config.DataType]map[config.ComponentID]component.Exporter {
-	return srv.builtExporters.ToMapByDataType()
-}
