diff --git a/processor/spanprocessor/span_test.go b/processor/spanprocessor/span_test.go
index d5b0c9cdf..816308c74 100644
--- a/processor/spanprocessor/span_test.go
+++ b/processor/spanprocessor/span_test.go
@@ -33,7 +33,7 @@ import (
 )
 
 func TestNewTraceProcessor(t *testing.T) {
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	tp, err := newSpanProcessor(nil, *oCfg)
@@ -146,7 +146,7 @@ func TestSpanProcessor_NilEmptyData(t *testing.T) {
 			output: testdata.GenerateTraceDataOneEmptyOneNilInstrumentationLibrary(),
 		},
 	}
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Include = &filterspan.MatchProperties{
@@ -255,7 +255,7 @@ func TestSpanProcessor_Values(t *testing.T) {
 		},*/
 	}
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.FromAttributes = []string{"key1"}
@@ -331,7 +331,7 @@ func TestSpanProcessor_MissingKeys(t *testing.T) {
 			},
 		},
 	}
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.FromAttributes = []string{"key1", "key2", "key3", "key4"}
@@ -350,7 +350,7 @@ func TestSpanProcessor_MissingKeys(t *testing.T) {
 // the single key.
 func TestSpanProcessor_Separator(t *testing.T) {
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.FromAttributes = []string{"key1"}
@@ -380,7 +380,7 @@ func TestSpanProcessor_Separator(t *testing.T) {
 // TestSpanProcessor_NoSeparatorMultipleKeys tests naming a span using multiple keys and no separator.
 func TestSpanProcessor_NoSeparatorMultipleKeys(t *testing.T) {
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.FromAttributes = []string{"key1", "key2"}
@@ -411,7 +411,7 @@ func TestSpanProcessor_NoSeparatorMultipleKeys(t *testing.T) {
 // TestSpanProcessor_SeparatorMultipleKeys tests naming a span with multiple keys and a separator.
 func TestSpanProcessor_SeparatorMultipleKeys(t *testing.T) {
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.FromAttributes = []string{"key1", "key2", "key3", "key4"}
@@ -447,7 +447,7 @@ func TestSpanProcessor_SeparatorMultipleKeys(t *testing.T) {
 // TestSpanProcessor_NilName tests naming a span when the input span had no name.
 func TestSpanProcessor_NilName(t *testing.T) {
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.FromAttributes = []string{"key1"}
@@ -543,7 +543,7 @@ func TestSpanProcessor_ToAttributes(t *testing.T) {
 		},
 	}
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Rename.ToAttributes = &ToAttributes{}
@@ -604,7 +604,7 @@ func TestSpanProcessor_skipSpan(t *testing.T) {
 		},
 	}
 
-	factory := Factory{}
+	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
 	oCfg.Include = &filterspan.MatchProperties{
