diff --git a/go.mod b/go.mod
index e7d9b6b1d..bbc490873 100644
--- a/go.mod
+++ b/go.mod
@@ -22,7 +22,7 @@ require (
 	go.opentelemetry.io/collector/model v0.39.0
 	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.26.1
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.26.1
-	go.opentelemetry.io/contrib/zpages v0.26.1
+	go.opentelemetry.io/contrib/zpages v0.27.0
 	go.opentelemetry.io/otel v1.2.0
 	go.opentelemetry.io/otel/exporters/prometheus v0.25.0
 	go.opentelemetry.io/otel/metric v0.25.0
