diff --git a/internal/pdatagrpc/traces.go b/internal/pdatagrpc/traces.go
index 513fe6f7d..c581073aa 100644
--- a/internal/pdatagrpc/traces.go
+++ b/internal/pdatagrpc/traces.go
@@ -19,9 +19,9 @@ import (
 
 	"google.golang.org/grpc"
 
-	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/internal"
 	otlpcollectortraces "go.opentelemetry.io/collector/internal/data/protogen/collector/trace/v1"
+	"go.opentelemetry.io/collector/model/pdata"
 )
 
 // TODO: Consider to add `TracesRequest`. If we add non pdata properties we can add them to the request.
