diff --git a/exporter/opencensusexporter/config.go b/exporter/opencensusexporter/config.go
index 83200f7ab..8b1b2a63e 100644
--- a/exporter/opencensusexporter/config.go
+++ b/exporter/opencensusexporter/config.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -17,7 +17,7 @@ package opencensusexporter
 import (
 	"time"
 
-	"github.com/census-instrumentation/opencensus-service/internal/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/internal/configmodels"
 )
 
 // ConfigV2 defines configuration for OpenCensus exporter.
