diff --git a/exporter/fileexporter/config.go b/exporter/fileexporter/config.go
index c70450857..6e6213e42 100644
--- a/exporter/fileexporter/config.go
+++ b/exporter/fileexporter/config.go
@@ -20,7 +20,7 @@ import (
 
 // Config defines configuration for file exporter.
 type Config struct {
-	*config.ExporterSettings `mapstructure:"-"`
+	config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
 	// Path of the file to write to. Path is relative to current directory.
 	Path string `mapstructure:"path"`
