diff --git a/processor/filterprocessor/factory.go b/processor/filterprocessor/factory.go
index 11ef8ec85..0a1d07b4a 100644
--- a/processor/filterprocessor/factory.go
+++ b/processor/filterprocessor/factory.go
@@ -28,7 +28,7 @@ const (
 	typeStr = "filter"
 )
 
-var processorCapabilities = component.ProcessorCapabilities{MutatesConsumedData: true}
+var processorCapabilities = consumer.Capabilities{MutatesData: true}
 
 // NewFactory returns a new factory for the Filter processor.
 func NewFactory() component.ProcessorFactory {
