diff --git a/internal/data/testdata/trace.go b/internal/data/testdata/trace.go
index fa6d9df6d..2b88fd8c4 100644
--- a/internal/data/testdata/trace.go
+++ b/internal/data/testdata/trace.go
@@ -286,7 +286,7 @@ func fillSpanOne(span pdata.Span) {
 	span.SetDroppedEventsCount(1)
 	status := span.Status()
 	status.InitEmpty()
-	status.SetCode(pdata.StatusCode(1))
+	status.SetCode(pdata.StatusCodeCancelled)
 	status.SetMessage("status-cancelled")
 }
 
