diff --git a/config/processor.go b/config/processor.go
index ec416e92f..c511fe7c7 100644
--- a/config/processor.go
+++ b/config/processor.go
@@ -25,11 +25,17 @@ type Processors map[string]Processor
 
 // ProcessorSettings defines common settings for a processor configuration.
 // Specific processors can embed this struct and extend it with more fields if needed.
+// When embedded in the processor config it must be with `mapstructure:"-"` tag.
 type ProcessorSettings struct {
 	TypeVal Type   `mapstructure:"-"`
 	NameVal string `mapstructure:"-"`
 }
 
+// NewProcessorSettings return a new ProcessorSettings with the given type.
+func NewProcessorSettings(typeVal Type) *ProcessorSettings {
+	return &ProcessorSettings{TypeVal: typeVal, NameVal: string(typeVal)}
+}
+
 var _ Processor = (*ProcessorSettings)(nil)
 
 // Name gets the processor name.
