diff --git a/exporter/zipkinexporter/config_test.go b/exporter/zipkinexporter/config_test.go
index c1f40f62c..797673930 100644
--- a/exporter/zipkinexporter/config_test.go
+++ b/exporter/zipkinexporter/config_test.go
@@ -48,6 +48,7 @@ func TestLoadConfig(t *testing.T) {
 	assert.Equal(t, "zipkin/2", e1.(*Config).Name())
 	assert.Equal(t, "https://somedest:1234/api/v2/spans", e1.(*Config).URL)
 	assert.Equal(t, "proto", e1.(*Config).Format)
+	assert.Equal(t, false, *e1.(*Config).ExportResourceLabels)
 	_, err = factory.CreateTraceExporter(zap.NewNop(), e1)
 	require.NoError(t, err)
 }
