diff --git a/pdata/plog/json.go b/pdata/plog/json.go
index e7ac2b8f5..71a20849b 100644
--- a/pdata/plog/json.go
+++ b/pdata/plog/json.go
@@ -12,7 +12,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal/otlp"
 )
 
-// JSONMarshaler marshals pdata.Logs to JSON bytes using the OTLP/JSON format.
+// JSONMarshaler marshals Logs to JSON bytes using the OTLP/JSON format.
 type JSONMarshaler struct{}
 
 // MarshalLogs to the OTLP/JSON format.
@@ -25,10 +25,10 @@ func (*JSONMarshaler) MarshalLogs(ld Logs) ([]byte, error) {
 
 var _ Unmarshaler = (*JSONUnmarshaler)(nil)
 
-// JSONUnmarshaler unmarshals OTLP/JSON formatted-bytes to pdata.Logs.
+// JSONUnmarshaler unmarshals OTLP/JSON formatted-bytes to Logs.
 type JSONUnmarshaler struct{}
 
-// UnmarshalLogs from OTLP/JSON format into pdata.Logs.
+// UnmarshalLogs from OTLP/JSON format into Logs.
 func (*JSONUnmarshaler) UnmarshalLogs(buf []byte) (Logs, error) {
 	iter := json.BorrowIterator(buf)
 	defer json.ReturnIterator(iter)
