diff --git a/receiver/opencensusreceiver/opencensus_test.go b/receiver/opencensusreceiver/opencensus_test.go
index 190b9f8fb..04e5360b0 100644
--- a/receiver/opencensusreceiver/opencensus_test.go
+++ b/receiver/opencensusreceiver/opencensus_test.go
@@ -36,8 +36,6 @@ import (
 )
 
 func TestGrpcGateway_endToEnd(t *testing.T) {
-	t.Skip("gRPC Gateway version changed, fix this test.")
-
 	addr := ":35993"
 
 	// Set the buffer count to 1 to make it flush the test span immediately.
@@ -136,6 +134,7 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 					},
 				},
 			},
+			SourceFormat: "oc_trace",
 		},
 	}
 
