diff --git a/featuregate/registry.go b/featuregate/registry.go
index e7d532621..c1d4c26c7 100644
--- a/featuregate/registry.go
+++ b/featuregate/registry.go
@@ -19,48 +19,64 @@ import (
 	"sync"
 )
 
-// RegistryOption allows for configuration additional information about a Gate that can be exposed throughout the application.
-type RegistryOption func(g *Gate)
+var reg = NewRegistry()
+
+// GetRegistry returns the global Registry.
+func GetRegistry() *Registry {
+	return reg
+}
+
+type Registry struct {
+	mu    sync.RWMutex
+	gates map[string]Gate
+}
+
+// NewRegistry returns a new empty Registry.
+func NewRegistry() *Registry {
+	return &Registry{gates: make(map[string]Gate)}
+}
+
+// RegistryOption allows to configure additional information about a Gate during registration.
+type RegistryOption interface {
+	apply(g *Gate)
+}
+
+type registerOption struct {
+	applyFunc func(g *Gate)
+}
+
+func (ro registerOption) apply(g *Gate) {
+	ro.applyFunc(g)
+}
 
 // WithRegisterDescription adds description for the Gate.
 func WithRegisterDescription(description string) RegistryOption {
-	return func(g *Gate) {
-		g.description = description
+	return registerOption{
+		applyFunc: func(g *Gate) {
+			g.description = description
+		},
 	}
 }
 
 // WithRegisterReferenceURL adds an URL that has all the contextual information about the Gate.
 func WithRegisterReferenceURL(url string) RegistryOption {
-	return func(g *Gate) {
-		g.referenceURL = url
+	return registerOption{
+		applyFunc: func(g *Gate) {
+			g.referenceURL = url
+		},
 	}
 }
 
 // WithRegisterRemovalVersion is used when the Gate is considered StageStable,
 // to inform users that referencing the gate is no longer needed.
 func WithRegisterRemovalVersion(version string) RegistryOption {
-	return func(g *Gate) {
-		g.removalVersion = version
+	return registerOption{
+		applyFunc: func(g *Gate) {
+			g.removalVersion = version
+		},
 	}
 }
 
-var reg = NewRegistry()
-
-// GetRegistry returns the global Registry.
-func GetRegistry() *Registry {
-	return reg
-}
-
-// NewRegistry returns a new empty Registry.
-func NewRegistry() *Registry {
-	return &Registry{gates: make(map[string]Gate)}
-}
-
-type Registry struct {
-	mu    sync.RWMutex
-	gates map[string]Gate
-}
-
 // Apply a configuration in the form of a map of Gate identifiers to boolean values.
 // Sets only those values provided in the map, other gate values are not changed.
 func (r *Registry) Apply(cfg map[string]bool) error {
@@ -106,7 +122,7 @@ func (r *Registry) RegisterID(id string, stage Stage, opts ...RegistryOption) er
 		stage: stage,
 	}
 	for _, opt := range opts {
-		opt(&g)
+		opt.apply(&g)
 	}
 	switch g.stage {
 	case StageAlpha:
