diff --git a/exporter/debugexporter/factory.go b/exporter/debugexporter/factory.go
index 72a2f45e3..15f6407e9 100644
--- a/exporter/debugexporter/factory.go
+++ b/exporter/debugexporter/factory.go
@@ -15,8 +15,8 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/debugexporter/internal/metadata"
+	"go.opentelemetry.io/collector/exporter/debugexporter/internal/otlptext"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
-	"go.opentelemetry.io/collector/exporter/internal/otlptext"
 )
 
 // The value of "type" key in configuration.
