diff --git a/pdata/internal/data/profileid_test.go b/pdata/internal/data/profileid_test.go
index 4cf2eb5ee..e1e27c2db 100644
--- a/pdata/internal/data/profileid_test.go
+++ b/pdata/internal/data/profileid_test.go
@@ -63,13 +63,13 @@ func TestProfileIDMarshalAndUnmarshalJSON(t *testing.T) {
 	defer json.ReturnStream(stream)
 	src := ProfileID([16]byte{0x12, 0x34, 0x56, 0x78, 0x12, 0x34, 0x56, 0x78, 0x12, 0x34, 0x56, 0x78, 0x12, 0x34, 0x56, 0x78})
 	src.MarshalJSONStream(stream)
-	require.NoError(t, stream.Error)
+	require.NoError(t, stream.Error())
 
 	iter := json.BorrowIterator(stream.Buffer())
 	defer json.ReturnIterator(iter)
 	dest := ProfileID{}
 	dest.UnmarshalJSONIter(iter)
-	require.NoError(t, iter.Error)
+	require.NoError(t, iter.Error())
 
 	assert.Equal(t, src, dest)
 }
