diff --git a/model/pdata/metrics_alias.go b/model/pdata/metrics_alias.go
index 685db4bde..e178d6b98 100644
--- a/model/pdata/metrics_alias.go
+++ b/model/pdata/metrics_alias.go
@@ -17,7 +17,6 @@ package pdata // import "go.opentelemetry.io/collector/model/pdata"
 // This file contains aliases for metric data structures.
 
 import (
-	"go.opentelemetry.io/collector/model/internal"
 	"go.opentelemetry.io/collector/model/internal/pdata"
 )
 
@@ -36,11 +35,6 @@ type Metrics = pdata.Metrics
 // NewMetrics is an alias for a function to create new Metrics.
 var NewMetrics = pdata.NewMetrics
 
-// Deprecated: [v0.47.0] will be removed soon, only used internally.
-func MetricsFromInternalRep(internal.MetricsWrapper) Metrics {
-	return Metrics{}
-}
-
 // MetricDataType is an alias for pdata.MetricDataType type.
 type MetricDataType = pdata.MetricDataType
 
