diff --git a/internal/testdata/metric_test.go b/internal/testdata/metric_test.go
index b686cb9f9..8d7ebfcc2 100644
--- a/internal/testdata/metric_test.go
+++ b/internal/testdata/metric_test.go
@@ -20,7 +20,7 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
-	otlpmetrics "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/metrics/v1"
+	otlpmetrics "go.opentelemetry.io/collector/internal/data/protogen/metrics/v1"
 )
 
 type traceMetricsCase struct {
