diff --git a/service/go.mod b/service/go.mod
index 4a7d49f9b..aa6bc9fad 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -27,6 +27,7 @@ require (
 	go.opentelemetry.io/collector/exporter/exportertest v0.113.0
 	go.opentelemetry.io/collector/extension v0.113.0
 	go.opentelemetry.io/collector/extension/extensioncapabilities v0.113.0
+	go.opentelemetry.io/collector/extension/extensiontest v0.113.0
 	go.opentelemetry.io/collector/extension/zpagesextension v0.113.0
 	go.opentelemetry.io/collector/featuregate v1.19.0
 	go.opentelemetry.io/collector/internal/fanoutconsumer v0.113.0
@@ -209,3 +210,5 @@ replace go.opentelemetry.io/collector/consumer/consumererror => ../consumer/cons
 replace go.opentelemetry.io/collector/connector/connectorprofiles => ../connector/connectorprofiles
 
 replace go.opentelemetry.io/collector/internal/fanoutconsumer => ../internal/fanoutconsumer
+
+replace go.opentelemetry.io/collector/extension/extensiontest => ../extension/extensiontest
