diff --git a/component/telemetry.go b/component/telemetry.go
index 6f8f0ec84..febb6d6cb 100644
--- a/component/telemetry.go
+++ b/component/telemetry.go
@@ -13,10 +13,6 @@ import (
 )
 
 // TelemetrySettings provides components with APIs to report telemetry.
-//
-// Note: there is a service version of this struct, servicetelemetry.TelemetrySettings, that mirrors
-// this struct except ReportStatus. When adding or removing anything from
-// this struct consider whether the same should be done for the service version.
 type TelemetrySettings struct {
 	// Logger that the factory can use during creation and can pass to the created
 	// component to be used later as well.
