diff --git a/service/defaultcomponents/defaults_test.go b/service/defaultcomponents/defaults_test.go
index 4faacbc34..d0c793723 100644
--- a/service/defaultcomponents/defaults_test.go
+++ b/service/defaultcomponents/defaults_test.go
@@ -40,6 +40,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/processor/samplingprocessor/probabilisticsamplerprocessor"
 	"github.com/open-telemetry/opentelemetry-collector/processor/samplingprocessor/tailsamplingprocessor"
 	"github.com/open-telemetry/opentelemetry-collector/processor/spanprocessor"
+	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver"
 	"github.com/open-telemetry/opentelemetry-collector/receiver/jaegerreceiver"
 	"github.com/open-telemetry/opentelemetry-collector/receiver/opencensusreceiver"
 	"github.com/open-telemetry/opentelemetry-collector/receiver/otlpreceiver"
@@ -55,12 +56,13 @@ func TestDefaultComponents(t *testing.T) {
 		"zpages":       &zpagesextension.Factory{},
 	}
 	expectedReceivers := map[string]component.ReceiverFactoryBase{
-		"jaeger":     &jaegerreceiver.Factory{},
-		"zipkin":     &zipkinreceiver.Factory{},
-		"prometheus": &prometheusreceiver.Factory{},
-		"opencensus": &opencensusreceiver.Factory{},
-		"otlp":       &otlpreceiver.Factory{},
-		"vmmetrics":  &vmmetricsreceiver.Factory{},
+		"jaeger":      &jaegerreceiver.Factory{},
+		"zipkin":      &zipkinreceiver.Factory{},
+		"prometheus":  &prometheusreceiver.Factory{},
+		"opencensus":  &opencensusreceiver.Factory{},
+		"otlp":        &otlpreceiver.Factory{},
+		"vmmetrics":   &vmmetricsreceiver.Factory{},
+		"hostmetrics": hostmetricsreceiver.NewFactory(),
 	}
 	expectedProcessors := map[string]component.ProcessorFactoryBase{
 		"attributes":            &attributesprocessor.Factory{},
