diff --git a/internal/e2e/go.mod b/internal/e2e/go.mod
index 8ab98fe15..81679c74b 100644
--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -13,7 +13,7 @@ require (
 	go.opentelemetry.io/collector/consumer v0.95.0
 	go.opentelemetry.io/collector/exporter v0.95.0
 	go.opentelemetry.io/collector/exporter/otlpexporter v0.94.1
-	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.94.1
+	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.95.0
 	go.opentelemetry.io/collector/pdata v1.2.0
 	go.opentelemetry.io/collector/receiver v0.95.0
 	go.opentelemetry.io/collector/receiver/otlpreceiver v0.94.1
