diff --git a/connector/connectortest/go.mod b/connector/connectortest/go.mod
index 2fb012bd5..6e23739c7 100644
--- a/connector/connectortest/go.mod
+++ b/connector/connectortest/go.mod
@@ -6,14 +6,14 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/stretchr/testify v1.11.1
 	go.opentelemetry.io/collector/component v1.51.0
-	go.opentelemetry.io/collector/component/componenttest v0.145.0
-	go.opentelemetry.io/collector/connector v0.145.0
-	go.opentelemetry.io/collector/connector/xconnector v0.145.0
+	go.opentelemetry.io/collector/component/componenttest v0.146.0
+	go.opentelemetry.io/collector/connector v0.146.0
+	go.opentelemetry.io/collector/connector/xconnector v0.146.0
 	go.opentelemetry.io/collector/consumer v1.51.0
-	go.opentelemetry.io/collector/consumer/consumertest v0.145.0
-	go.opentelemetry.io/collector/consumer/xconsumer v0.145.0
+	go.opentelemetry.io/collector/consumer/consumertest v0.146.0
+	go.opentelemetry.io/collector/consumer/xconsumer v0.146.0
 	go.opentelemetry.io/collector/pdata v1.51.0
-	go.opentelemetry.io/collector/pdata/pprofile v0.145.0
+	go.opentelemetry.io/collector/pdata/pprofile v0.146.0
 	go.uber.org/goleak v1.3.0
 )
 
@@ -29,10 +29,10 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	go.opentelemetry.io/auto/sdk v1.2.1 // indirect
 	go.opentelemetry.io/collector/featuregate v1.51.0 // indirect
-	go.opentelemetry.io/collector/internal/componentalias v0.145.0 // indirect
-	go.opentelemetry.io/collector/internal/fanoutconsumer v0.145.0 // indirect
+	go.opentelemetry.io/collector/internal/componentalias v0.146.0 // indirect
+	go.opentelemetry.io/collector/internal/fanoutconsumer v0.146.0 // indirect
 	go.opentelemetry.io/collector/pipeline v1.51.0 // indirect
-	go.opentelemetry.io/collector/pipeline/xpipeline v0.145.0 // indirect
+	go.opentelemetry.io/collector/pipeline/xpipeline v0.146.0 // indirect
 	go.opentelemetry.io/otel v1.40.0 // indirect
 	go.opentelemetry.io/otel/metric v1.40.0 // indirect
 	go.opentelemetry.io/otel/sdk v1.40.0 // indirect
