diff --git a/exporter/otlpexporter/otlp.go b/exporter/otlpexporter/otlp.go
index cfa3bc805..0d8e6b680 100644
--- a/exporter/otlpexporter/otlp.go
+++ b/exporter/otlpexporter/otlp.go
@@ -31,10 +31,10 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/consumer/pdatautil"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
-	"go.opentelemetry.io/collector/internal/data"
 	otlplogs "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/collector/logs/v1"
 	otlpmetrics "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/collector/metrics/v1"
 	otlptrace "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/collector/trace/v1"
+	"go.opentelemetry.io/collector/internal/dataold"
 )
 
 type exporterImp struct {
@@ -92,7 +92,7 @@ func (e *exporterImp) pushTraceData(ctx context.Context, td pdata.Traces) (int,
 func (e *exporterImp) pushMetricsData(ctx context.Context, md pdata.Metrics) (int, error) {
 	imd := pdatautil.MetricsToInternalMetrics(md)
 	request := &otlpmetrics.ExportMetricsServiceRequest{
-		ResourceMetrics: data.MetricDataToOtlp(imd),
+		ResourceMetrics: dataold.MetricDataToOtlp(imd),
 	}
 	err := e.w.exportMetrics(ctx, request)
 
