diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index 5f06db763..fc8d597de 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -21,6 +21,7 @@
 cmd/builder/                                 @open-telemetry/collector-approvers @ArthurSens @dmathieu
 cmd/mdatagen/                                @open-telemetry/collector-approvers @dmitryax
 cmd/mdatagen/internal/sampleconnector/       @open-telemetry/collector-approvers
+cmd/mdatagen/internal/sampleentityreceiver/  @open-telemetry/collector-approvers @dmitryax
 cmd/mdatagen/internal/samplefactoryreceiver/ @open-telemetry/collector-approvers @dmitryax
 cmd/mdatagen/internal/sampleprocessor/       @open-telemetry/collector-approvers
 cmd/mdatagen/internal/samplereceiver/        @open-telemetry/collector-approvers @dmitryax
