diff --git a/processor/batchprocessor/factory_test.go b/processor/batchprocessor/factory_test.go
index c73021e3a..a9b3aaa16 100644
--- a/processor/batchprocessor/factory_test.go
+++ b/processor/batchprocessor/factory_test.go
@@ -43,6 +43,6 @@ func TestCreateProcessor(t *testing.T) {
 	assert.NoError(t, err, "cannot create trace processor")
 
 	mp, err := factory.CreateMetricsProcessor(context.Background(), creationParams, nil, cfg)
-	assert.Nil(t, mp)
-	assert.Error(t, err, "should not be able to create metric processor")
+	assert.NotNil(t, mp)
+	assert.NoError(t, err, "cannot create metric processor")
 }
