diff --git a/receiver/opencensusreceiver/opencensus_test.go b/receiver/opencensusreceiver/opencensus_test.go
index d01a97cff..4786b2173 100644
--- a/receiver/opencensusreceiver/opencensus_test.go
+++ b/receiver/opencensusreceiver/opencensus_test.go
@@ -27,11 +27,12 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 
-	"github.com/census-instrumentation/opencensus-service/receiver/testhelper"
+	"github.com/census-instrumentation/opencensus-service/processor/processortest"
 
 	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
 	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/census-instrumentation/opencensus-service/data"
 	"github.com/census-instrumentation/opencensus-service/internal"
 )
 
@@ -45,7 +46,7 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 	}
 	defer ocr.StopTraceReception(context.Background())
 
-	sink := new(testhelper.ConcurrentSpanSink)
+	sink := new(processortest.ConcurrentTraceDataSink)
 
 	go func() {
 		if err := ocr.StartTraceReception(context.Background(), sink); err != nil {
@@ -113,7 +114,7 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 
 	got := sink.AllTraces()
 
-	want := []*agenttracepb.ExportTraceServiceRequest{
+	want := []data.TraceData{
 		{
 			Node: &commonpb.Node{
 				Identifier: &commonpb.ProcessIdentifier{HostName: "testHost"},
@@ -139,7 +140,7 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 	}
 
 	if !reflect.DeepEqual(got, want) {
-		gj, wj := testhelper.ToJSON(got), testhelper.ToJSON(want)
+		gj, wj := processortest.ToJSON(got), processortest.ToJSON(want)
 		if !bytes.Equal(gj, wj) {
 			t.Errorf("Mismatched responses\nGot:\n\t%v\n\t%s\nWant:\n\t%v\n\t%s", got, gj, want, wj)
 		}
@@ -156,7 +157,7 @@ func TestGrpcGatewayCors_endToEnd(t *testing.T) {
 	}
 	defer ocr.Stop()
 
-	sink := new(testhelper.ConcurrentSpanSink)
+	sink := new(processortest.ConcurrentTraceDataSink)
 	go func() {
 		if err := ocr.StartTraceReception(context.Background(), sink); err != nil {
 			t.Fatalf("Failed to start trace receiver: %v", err)
@@ -187,7 +188,7 @@ func TestAcceptAllGRPCProtoAffiliatedContentTypes(t *testing.T) {
 		t.Fatalf("Failed to create trace receiver: %v", err)
 	}
 
-	cbts := new(testhelper.ConcurrentSpanSink)
+	cbts := new(processortest.ConcurrentTraceDataSink)
 	if err := ocr.StartTraceReception(context.Background(), cbts); err != nil {
 		t.Fatalf("Failed to start the trace receiver: %v", err)
 	}
