diff --git a/receiver/zipkinreceiver/factory_test.go b/receiver/zipkinreceiver/factory_test.go
index a85986f80..4fe150d73 100644
--- a/receiver/zipkinreceiver/factory_test.go
+++ b/receiver/zipkinreceiver/factory_test.go
@@ -35,7 +35,7 @@ func TestCreateDefaultConfig(t *testing.T) {
 func TestCreateReceiver(t *testing.T) {
 	cfg := createDefaultConfig()
 
-	tReceiver, err := createTraceReceiver(
+	tReceiver, err := createTracesReceiver(
 		context.Background(),
 		component.ReceiverCreateParams{Logger: zap.NewNop()},
 		cfg,
@@ -43,7 +43,7 @@ func TestCreateReceiver(t *testing.T) {
 	assert.NoError(t, err, "receiver creation failed")
 	assert.NotNil(t, tReceiver, "receiver creation failed")
 
-	tReceiver, err = createTraceReceiver(
+	tReceiver, err = createTracesReceiver(
 		context.Background(),
 		component.ReceiverCreateParams{Logger: zap.NewNop()},
 		cfg,
