diff --git a/exporter/prometheusexporter/factory.go b/exporter/prometheusexporter/factory.go
index d85050715..95442358a 100644
--- a/exporter/prometheusexporter/factory.go
+++ b/exporter/prometheusexporter/factory.go
@@ -28,24 +28,22 @@ import (
 	"github.com/orijtech/prometheus-go-metrics-exporter"
 )
 
-var _ = exporter.RegisterFactory(&factory{})
-
 const (
 	// The value of "type" key in configuration.
 	typeStr = "prometheus"
 )
 
-// factory is the factory for Prometheus exporter.
-type factory struct {
+// Factory is the factory for Prometheus exporter.
+type Factory struct {
 }
 
 // Type gets the type of the Exporter config created by this factory.
-func (f *factory) Type() string {
+func (f *Factory) Type() string {
 	return typeStr
 }
 
 // CreateDefaultConfig creates the default configuration for exporter.
-func (f *factory) CreateDefaultConfig() configmodels.Exporter {
+func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
 	return &Config{
 		ExporterSettings: configmodels.ExporterSettings{
 			TypeVal: typeStr,
@@ -56,12 +54,12 @@ func (f *factory) CreateDefaultConfig() configmodels.Exporter {
 }
 
 // CreateTraceExporter creates a trace exporter based on this config.
-func (f *factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+func (f *Factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
 	return nil, nil, configerror.ErrDataTypeIsNotSupported
 }
 
 // CreateMetricsExporter creates a metrics exporter based on this config.
-func (f *factory) CreateMetricsExporter(logger *zap.Logger, cfg configmodels.Exporter) (consumer.MetricsConsumer, exporter.StopFunc, error) {
+func (f *Factory) CreateMetricsExporter(logger *zap.Logger, cfg configmodels.Exporter) (consumer.MetricsConsumer, exporter.StopFunc, error) {
 	pcfg := cfg.(*Config)
 
 	addr := strings.TrimSpace(pcfg.Endpoint)
