diff --git a/exporter/exporterhelper/queue_sender.go b/exporter/exporterhelper/queue_sender.go
index d71663355..060edab81 100644
--- a/exporter/exporterhelper/queue_sender.go
+++ b/exporter/exporterhelper/queue_sender.go
@@ -73,12 +73,12 @@ type queueSender struct {
 	traceAttribute attribute.KeyValue
 	consumers      *queue.Consumers[Request]
 
-	obsrep     *ObsReport
+	obsrep     *obsReport
 	exporterID component.ID
 }
 
 func newQueueSender(q exporterqueue.Queue[Request], set exporter.Settings, numConsumers int,
-	exportFailureMessage string, obsrep *ObsReport) *queueSender {
+	exportFailureMessage string, obsrep *obsReport) *queueSender {
 	qs := &queueSender{
 		queue:          q,
 		numConsumers:   numConsumers,
