diff --git a/processor/memorylimiterprocessor/go.mod b/processor/memorylimiterprocessor/go.mod
index 6040afece..62b7249a5 100644
--- a/processor/memorylimiterprocessor/go.mod
+++ b/processor/memorylimiterprocessor/go.mod
@@ -4,13 +4,13 @@ go 1.20
 
 require (
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector v0.90.1
-	go.opentelemetry.io/collector/component v0.90.1
-	go.opentelemetry.io/collector/config/configtelemetry v0.90.1
-	go.opentelemetry.io/collector/confmap v0.90.1
-	go.opentelemetry.io/collector/consumer v0.90.1
+	go.opentelemetry.io/collector v0.91.0
+	go.opentelemetry.io/collector/component v0.91.0
+	go.opentelemetry.io/collector/config/configtelemetry v0.91.0
+	go.opentelemetry.io/collector/confmap v0.91.0
+	go.opentelemetry.io/collector/consumer v0.91.0
 	go.opentelemetry.io/collector/pdata v1.0.0
-	go.opentelemetry.io/collector/processor v0.90.1
+	go.opentelemetry.io/collector/processor v0.91.0
 	go.uber.org/zap v1.26.0
 )
 
