diff --git a/pdata/pmetric/generated_metric.go b/pdata/pmetric/generated_metric.go
index aba0d8e04..839628520 100644
--- a/pdata/pmetric/generated_metric.go
+++ b/pdata/pmetric/generated_metric.go
@@ -7,6 +7,7 @@
 package pmetric
 
 import (
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
 )
 
@@ -19,11 +20,12 @@ import (
 // Must use NewMetric function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type Metric struct {
-	orig *otlpmetrics.Metric
+	orig  *otlpmetrics.Metric
+	state *internal.State
 }
 
-func newMetric(orig *otlpmetrics.Metric) Metric {
-	return Metric{orig}
+func newMetric(orig *otlpmetrics.Metric, state *internal.State) Metric {
+	return Metric{orig: orig, state: state}
 }
 
 // NewMetric creates a new empty Metric.
@@ -31,12 +33,15 @@ func newMetric(orig *otlpmetrics.Metric) Metric {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewMetric() Metric {
-	return newMetric(&otlpmetrics.Metric{})
+	state := internal.StateMutable
+	return newMetric(&otlpmetrics.Metric{}, &state)
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
 // resetting the current instance to its zero value
 func (ms Metric) MoveTo(dest Metric) {
+	ms.state.AssertMutable()
+	dest.state.AssertMutable()
 	*dest.orig = *ms.orig
 	*ms.orig = otlpmetrics.Metric{}
 }
@@ -48,6 +53,7 @@ func (ms Metric) Name() string {
 
 // SetName replaces the name associated with this Metric.
 func (ms Metric) SetName(v string) {
+	ms.state.AssertMutable()
 	ms.orig.Name = v
 }
 
@@ -58,6 +64,7 @@ func (ms Metric) Description() string {
 
 // SetDescription replaces the description associated with this Metric.
 func (ms Metric) SetDescription(v string) {
+	ms.state.AssertMutable()
 	ms.orig.Description = v
 }
 
@@ -68,6 +75,7 @@ func (ms Metric) Unit() string {
 
 // SetUnit replaces the unit associated with this Metric.
 func (ms Metric) SetUnit(v string) {
+	ms.state.AssertMutable()
 	ms.orig.Unit = v
 }
 
@@ -100,7 +108,7 @@ func (ms Metric) Gauge() Gauge {
 	if !ok {
 		return Gauge{}
 	}
-	return newGauge(v.Gauge)
+	return newGauge(v.Gauge, ms.state)
 }
 
 // SetEmptyGauge sets an empty gauge to this Metric.
@@ -109,9 +117,10 @@ func (ms Metric) Gauge() Gauge {
 //
 // Calling this function on zero-initialized Metric will cause a panic.
 func (ms Metric) SetEmptyGauge() Gauge {
+	ms.state.AssertMutable()
 	val := &otlpmetrics.Gauge{}
 	ms.orig.Data = &otlpmetrics.Metric_Gauge{Gauge: val}
-	return newGauge(val)
+	return newGauge(val, ms.state)
 }
 
 // Sum returns the sum associated with this Metric.
@@ -125,7 +134,7 @@ func (ms Metric) Sum() Sum {
 	if !ok {
 		return Sum{}
 	}
-	return newSum(v.Sum)
+	return newSum(v.Sum, ms.state)
 }
 
 // SetEmptySum sets an empty sum to this Metric.
@@ -134,9 +143,10 @@ func (ms Metric) Sum() Sum {
 //
 // Calling this function on zero-initialized Metric will cause a panic.
 func (ms Metric) SetEmptySum() Sum {
+	ms.state.AssertMutable()
 	val := &otlpmetrics.Sum{}
 	ms.orig.Data = &otlpmetrics.Metric_Sum{Sum: val}
-	return newSum(val)
+	return newSum(val, ms.state)
 }
 
 // Histogram returns the histogram associated with this Metric.
@@ -150,7 +160,7 @@ func (ms Metric) Histogram() Histogram {
 	if !ok {
 		return Histogram{}
 	}
-	return newHistogram(v.Histogram)
+	return newHistogram(v.Histogram, ms.state)
 }
 
 // SetEmptyHistogram sets an empty histogram to this Metric.
@@ -159,9 +169,10 @@ func (ms Metric) Histogram() Histogram {
 //
 // Calling this function on zero-initialized Metric will cause a panic.
 func (ms Metric) SetEmptyHistogram() Histogram {
+	ms.state.AssertMutable()
 	val := &otlpmetrics.Histogram{}
 	ms.orig.Data = &otlpmetrics.Metric_Histogram{Histogram: val}
-	return newHistogram(val)
+	return newHistogram(val, ms.state)
 }
 
 // ExponentialHistogram returns the exponentialhistogram associated with this Metric.
@@ -175,7 +186,7 @@ func (ms Metric) ExponentialHistogram() ExponentialHistogram {
 	if !ok {
 		return ExponentialHistogram{}
 	}
-	return newExponentialHistogram(v.ExponentialHistogram)
+	return newExponentialHistogram(v.ExponentialHistogram, ms.state)
 }
 
 // SetEmptyExponentialHistogram sets an empty exponentialhistogram to this Metric.
@@ -184,9 +195,10 @@ func (ms Metric) ExponentialHistogram() ExponentialHistogram {
 //
 // Calling this function on zero-initialized Metric will cause a panic.
 func (ms Metric) SetEmptyExponentialHistogram() ExponentialHistogram {
+	ms.state.AssertMutable()
 	val := &otlpmetrics.ExponentialHistogram{}
 	ms.orig.Data = &otlpmetrics.Metric_ExponentialHistogram{ExponentialHistogram: val}
-	return newExponentialHistogram(val)
+	return newExponentialHistogram(val, ms.state)
 }
 
 // Summary returns the summary associated with this Metric.
@@ -200,7 +212,7 @@ func (ms Metric) Summary() Summary {
 	if !ok {
 		return Summary{}
 	}
-	return newSummary(v.Summary)
+	return newSummary(v.Summary, ms.state)
 }
 
 // SetEmptySummary sets an empty summary to this Metric.
@@ -209,13 +221,15 @@ func (ms Metric) Summary() Summary {
 //
 // Calling this function on zero-initialized Metric will cause a panic.
 func (ms Metric) SetEmptySummary() Summary {
+	ms.state.AssertMutable()
 	val := &otlpmetrics.Summary{}
 	ms.orig.Data = &otlpmetrics.Metric_Summary{Summary: val}
-	return newSummary(val)
+	return newSummary(val, ms.state)
 }
 
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Metric) CopyTo(dest Metric) {
+	dest.state.AssertMutable()
 	dest.SetName(ms.Name())
 	dest.SetDescription(ms.Description())
 	dest.SetUnit(ms.Unit())
