diff --git a/pdata/internal/cmd/pdatagen/internal/primitive_slice_structs.go b/pdata/internal/cmd/pdatagen/internal/primitive_slice_structs.go
index 15827d283..ca814a046 100644
--- a/pdata/internal/cmd/pdatagen/internal/primitive_slice_structs.go
+++ b/pdata/internal/cmd/pdatagen/internal/primitive_slice_structs.go
@@ -223,6 +223,7 @@ func (iss *primitiveSliceStruct) generateInternal(sb *bytes.Buffer) {
 }
 
 var primitiveSliceFile = &File{
+	Path:        "pdata",
 	Name:        "primitive_slice",
 	PackageName: "pcommon",
 	testImports: []string{
