diff --git a/receiver/otlpreceiver/factory.go b/receiver/otlpreceiver/factory.go
index b07ccd76c..be4755563 100644
--- a/receiver/otlpreceiver/factory.go
+++ b/receiver/otlpreceiver/factory.go
@@ -42,7 +42,7 @@ func NewFactory() component.ReceiverFactory {
 	return receiverhelper.NewFactory(
 		typeStr,
 		createDefaultConfig,
-		receiverhelper.WithTraces(createTraceReceiver),
+		receiverhelper.WithTraces(createTracesReceiver),
 		receiverhelper.WithMetrics(createMetricsReceiver),
 		receiverhelper.WithLogs(createLogReceiver))
 }
@@ -71,7 +71,7 @@ func createDefaultConfig() config.Receiver {
 }
 
 // CreateTracesReceiver creates a  trace receiver based on provided config.
-func createTraceReceiver(
+func createTracesReceiver(
 	ctx context.Context,
 	params component.ReceiverCreateParams,
 	cfg config.Receiver,
