diff --git a/exporter/prometheusremotewriteexporter/exporter.go b/exporter/prometheusremotewriteexporter/exporter.go
index 2e5eed447..61d30f1d7 100644
--- a/exporter/prometheusremotewriteexporter/exporter.go
+++ b/exporter/prometheusremotewriteexporter/exporter.go
@@ -32,7 +32,6 @@ import (
 
 	"go.opentelemetry.io/collector/component/componenterror"
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/collector/consumer/pdatautil"
 	otlp "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/metrics/v1"
 )
 
@@ -83,7 +82,7 @@ func (prwe *PrwExporter) PushMetrics(ctx context.Context, md pdata.Metrics) (int
 	defer prwe.wg.Done()
 	select {
 	case <-prwe.closeChan:
-		return pdatautil.MetricCount(md), errors.New("shutdown has been called")
+		return md.MetricCount(), errors.New("shutdown has been called")
 	default:
 		tsMap := map[string]*prompb.TimeSeries{}
 		dropped := 0
@@ -132,7 +131,7 @@ func (prwe *PrwExporter) PushMetrics(ctx context.Context, md pdata.Metrics) (int
 		}
 
 		if err := prwe.export(ctx, tsMap); err != nil {
-			dropped = pdatautil.MetricCount(md)
+			dropped = md.MetricCount()
 			errs = append(errs, err)
 		}
 
