diff --git a/service/telemetry/telemetrytest/go.mod b/service/telemetry/telemetrytest/go.mod
index 2e6250d98..c8497a24b 100644
--- a/service/telemetry/telemetrytest/go.mod
+++ b/service/telemetry/telemetrytest/go.mod
@@ -62,7 +62,7 @@ require (
 	golang.org/x/text v0.34.0 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20251222181119-0a764e51fe1b // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20251222181119-0a764e51fe1b // indirect
-	google.golang.org/grpc v1.78.0 // indirect
+	google.golang.org/grpc v1.79.1 // indirect
 	google.golang.org/protobuf v1.36.11 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
