diff --git a/internal/goldendataset/metric_gen.go b/internal/goldendataset/metric_gen.go
index 76e376647..68ee094e1 100644
--- a/internal/goldendataset/metric_gen.go
+++ b/internal/goldendataset/metric_gen.go
@@ -164,7 +164,7 @@ func populateIntPoints(cfg MetricCfg, pts pdata.IntDataPointSlice) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
 		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
 		pt.SetValue(int64(cfg.PtVal + i))
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -175,7 +175,7 @@ func populateDoublePoints(cfg MetricCfg, pts pdata.DoubleDataPointSlice) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
 		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
 		pt.SetValue(float64(cfg.PtVal + i))
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -187,7 +187,7 @@ func populateDoubleHistogram(cfg MetricCfg, dh pdata.DoubleHistogram) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
 		ts := getTimestamp(cfg.StartTime, cfg.StepSize, i)
 		pt.SetTimestamp(ts)
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -224,7 +224,7 @@ func populateIntHistogram(cfg MetricCfg, dh pdata.IntHistogram) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
 		ts := getTimestamp(cfg.StartTime, cfg.StepSize, i)
 		pt.SetTimestamp(ts)
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -264,6 +264,6 @@ func populatePtLabels(cfg MetricCfg, lm pdata.StringMap) {
 	}
 }
 
-func getTimestamp(startTime uint64, stepSize uint64, i int) pdata.TimestampUnixNano {
-	return pdata.TimestampUnixNano(startTime + (stepSize * uint64(i+1)))
+func getTimestamp(startTime uint64, stepSize uint64, i int) pdata.Timestamp {
+	return pdata.Timestamp(startTime + (stepSize * uint64(i+1)))
 }
