diff --git a/internal/collector/processor/nodebatcher/metrics.go b/internal/collector/processor/nodebatcher/metrics.go
index f02b63010..4147b7486 100644
--- a/internal/collector/processor/nodebatcher/metrics.go
+++ b/internal/collector/processor/nodebatcher/metrics.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -19,8 +19,8 @@ import (
 	"go.opencensus.io/stats/view"
 	"go.opencensus.io/tag"
 
-	"github.com/census-instrumentation/opencensus-service/internal/collector/processor"
-	"github.com/census-instrumentation/opencensus-service/internal/collector/telemetry"
+	"github.com/open-telemetry/opentelemetry-service/internal/collector/processor"
+	"github.com/open-telemetry/opentelemetry-service/internal/collector/telemetry"
 )
 
 var (
