diff --git a/internal/tools/go.mod b/internal/tools/go.mod
index 83b62a512..7dbb779ee 100644
--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -281,7 +281,7 @@ require (
 	golang.org/x/text v0.32.0 // indirect
 	golang.org/x/tools v0.40.0 // indirect
 	golang.org/x/vuln v1.1.4 // indirect
-	google.golang.org/protobuf v1.36.8 // indirect
+	google.golang.org/protobuf v1.36.11 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	gotest.tools/gotestsum v1.13.0 // indirect
