diff --git a/exporter/internal/queue/consumers.go b/exporter/internal/queue/consumers.go
index 4db4b830e..dc7418c4e 100644
--- a/exporter/internal/queue/consumers.go
+++ b/exporter/internal/queue/consumers.go
@@ -36,12 +36,11 @@ func (qc *Consumers[T]) Start(_ context.Context, _ component.Host) error {
 			startWG.Done()
 			defer qc.stopWG.Done()
 			for {
-				index, ctx, req, ok := qc.queue.Read(context.Background())
+				ctx, req, done, ok := qc.queue.Read(context.Background())
 				if !ok {
 					return
 				}
-				consumeErr := qc.consumeFunc(ctx, req)
-				qc.queue.OnProcessingFinished(index, consumeErr)
+				done(qc.consumeFunc(ctx, req))
 			}
 		}()
 	}
