diff --git a/config/configtls/configtls_test.go b/config/configtls/configtls_test.go
index 534177ef1..9824c20db 100644
--- a/config/configtls/configtls_test.go
+++ b/config/configtls/configtls_test.go
@@ -635,30 +635,25 @@ func TestMinMaxTLSVersions(t *testing.T) {
 
 func TestTLSSettingValidate(t *testing.T) {
 	tests := []struct {
-		name       string
-		minVersion string
-		maxVersion string
-		errorTxt   string
+		name      string
+		tlsConfig TLSSetting
+		errorTxt  string
 	}{
-		{name: `TLS Config ["", ""] to be valid`, minVersion: "", maxVersion: ""},
-		{name: `TLS Config ["", "1.3"] to be valid`, minVersion: "", maxVersion: "1.3"},
-		{name: `TLS Config ["1.2", ""] to be valid`, minVersion: "1.2", maxVersion: ""},
-		{name: `TLS Config ["1.3", "1.3"] to be valid`, minVersion: "1.3", maxVersion: "1.3"},
-		{name: `TLS Config ["1.0", "1.1"] to be valid`, minVersion: "1.0", maxVersion: "1.1"},
-		{name: `TLS Config ["asd", ""] to give [Error]`, minVersion: "asd", maxVersion: "", errorTxt: `invalid TLS min_version: unsupported TLS version: "asd"`},
-		{name: `TLS Config ["", "asd"] to give [Error]`, minVersion: "", maxVersion: "asd", errorTxt: `invalid TLS max_version: unsupported TLS version: "asd"`},
-		{name: `TLS Config ["0.4", ""] to give [Error]`, minVersion: "0.4", maxVersion: "", errorTxt: `invalid TLS min_version: unsupported TLS version: "0.4"`},
-		{name: `TLS Config ["1.2", "1.1"] to give [Error]`, minVersion: "1.2", maxVersion: "1.1", errorTxt: `invalid TLS configuration: min_version cannot be greater than max_version`},
+		{name: `TLS Config ["", ""] to be valid`, tlsConfig: Config{MinVersion: "", MaxVersion: ""}},
+		{name: `TLS Config ["", "1.3"] to be valid`, tlsConfig: Config{MinVersion: "", MaxVersion: "1.3"}},
+		{name: `TLS Config ["1.2", ""] to be valid`, tlsConfig: Config{MinVersion: "1.2", MaxVersion: ""}},
+		{name: `TLS Config ["1.3", "1.3"] to be valid`, tlsConfig: Config{MinVersion: "1.3", MaxVersion: "1.3"}},
+		{name: `TLS Config ["1.0", "1.1"] to be valid`, tlsConfig: Config{MinVersion: "1.0", MaxVersion: "1.1"}},
+		{name: `TLS Config ["asd", ""] to give [Error]`, tlsConfig: Config{MinVersion: "asd", MaxVersion: ""}, errorTxt: `invalid TLS min_version: unsupported TLS version: "asd"`},
+		{name: `TLS Config ["", "asd"] to give [Error]`, tlsConfig: Config{MinVersion: "", MaxVersion: "asd"}, errorTxt: `invalid TLS max_version: unsupported TLS version: "asd"`},
+		{name: `TLS Config ["0.4", ""] to give [Error]`, tlsConfig: Config{MinVersion: "0.4", MaxVersion: ""}, errorTxt: `invalid TLS min_version: unsupported TLS version: "0.4"`},
+		{name: `TLS Config ["1.2", "1.1"] to give [Error]`, tlsConfig: Config{MinVersion: "1.2", MaxVersion: "1.1"}, errorTxt: `invalid TLS configuration: min_version cannot be greater than max_version`},
+		{name: `TLS Config with both CA File and PEM`, tlsConfig: Config{CAFile: "test", CAPem: "test"}, errorTxt: `provide either a CA file or the PEM-encoded string, but not both`},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			setting := TLSSetting{
-				MinVersion: test.minVersion,
-				MaxVersion: test.maxVersion,
-			}
-
-			err := setting.Validate()
+			err := test.tlsConfig.Validate()
 
 			if test.errorTxt == "" {
 				assert.Nil(t, err)
@@ -719,6 +714,86 @@ invalid TLS cipher suite: "BAR"`,
 }
 
 func TestSystemCertPool(t *testing.T) {
+	anError := errors.New("my error")
+	tests := []struct {
+		name         string
+		tlsSetting   TLSSetting
+		wantErr      error
+		systemCertFn func() (*x509.CertPool, error)
+	}{
+		{
+			name: "not using system cert pool",
+			tlsSetting: TLSSetting{
+				IncludeSystemCACertsPool: false,
+				CAFile:                   filepath.Join("testdata", "ca-1.crt"),
+			},
+			wantErr:      nil,
+			systemCertFn: x509.SystemCertPool,
+		},
+		{
+			name: "using system cert pool",
+			tlsSetting: TLSSetting{
+				IncludeSystemCACertsPool: true,
+				CAFile:                   filepath.Join("testdata", "ca-1.crt"),
+			},
+			wantErr:      nil,
+			systemCertFn: x509.SystemCertPool,
+		},
+		{
+			name: "error loading system cert pool",
+			tlsSetting: TLSSetting{
+				IncludeSystemCACertsPool: true,
+				CAFile:                   filepath.Join("testdata", "ca-1.crt"),
+			},
+			wantErr: anError,
+			systemCertFn: func() (*x509.CertPool, error) {
+				return nil, anError
+			},
+		},
+		{
+			name: "nil system cert pool",
+			tlsSetting: TLSSetting{
+				IncludeSystemCACertsPool: true,
+				CAFile:                   filepath.Join("testdata", "ca-1.crt"),
+			},
+			wantErr: nil,
+			systemCertFn: func() (*x509.CertPool, error) {
+				return nil, nil
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			oldSystemCertPool := systemCertPool
+			systemCertPool = test.systemCertFn
+			defer func() {
+				systemCertPool = oldSystemCertPool
+			}()
+
+			serverConfig := ServerConfig{
+				TLSSetting: test.tlsSetting,
+			}
+			c, err := serverConfig.LoadTLSConfig()
+			if test.wantErr != nil {
+				assert.ErrorContains(t, err, test.wantErr.Error())
+			} else {
+				assert.NotNil(t, c.RootCAs)
+			}
+
+			clientConfig := ClientConfig{
+				TLSSetting: test.tlsSetting,
+			}
+			c, err = clientConfig.LoadTLSConfig()
+			if test.wantErr != nil {
+				assert.ErrorContains(t, err, test.wantErr.Error())
+			} else {
+				assert.NotNil(t, c.RootCAs)
+			}
+		})
+	}
+}
+
+func TestSystemCertPool_loadCert(t *testing.T) {
 	anError := errors.New("my error")
 	tests := []struct {
 		name         string
