diff --git a/service/builder/pipelines_builder_test.go b/service/builder/pipelines_builder_test.go
index ee9a21b3e..c92dedfe4 100644
--- a/service/builder/pipelines_builder_test.go
+++ b/service/builder/pipelines_builder_test.go
@@ -27,6 +27,7 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/component/componenttest"
 	"github.com/open-telemetry/opentelemetry-collector/config"
 	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
 	"github.com/open-telemetry/opentelemetry-collector/consumer"
@@ -96,8 +97,7 @@ func testPipeline(t *testing.T, pipelineName string, exporterNames []string) {
 	assert.NoError(t, err)
 	require.NotNil(t, pipelineProcessors)
 
-	mh := component.NewMockHost()
-	err = pipelineProcessors.StartProcessors(zap.NewNop(), mh)
+	err = pipelineProcessors.StartProcessors(zap.NewNop(), componenttest.NewNopHost())
 	assert.NoError(t, err)
 
 	processor := pipelineProcessors[cfg.Service.Pipelines[pipelineName]]
