diff --git a/cmd/mdatagen/main.go b/cmd/mdatagen/main.go
index 11411f449..9f099af85 100644
--- a/cmd/mdatagen/main.go
+++ b/cmd/mdatagen/main.go
@@ -24,6 +24,7 @@ import (
 	"log"
 	"os"
 	"path"
+	"path/filepath"
 	"runtime"
 	"strings"
 	"text/template"
@@ -42,7 +43,7 @@ func run(ymlPath string) error {
 		return errors.New("argument must be metadata.yaml file")
 	}
 
-	ymlData, err := ioutil.ReadFile(ymlPath)
+	ymlData, err := ioutil.ReadFile(filepath.Clean(ymlPath))
 	if err != nil {
 		return fmt.Errorf("unable to read file %v: %v", ymlPath, err)
 	}
