diff --git a/service/service.go b/service/service.go
index 181302ba2..c93e8ea8c 100644
--- a/service/service.go
+++ b/service/service.go
@@ -520,7 +520,7 @@ func (app *Application) getPipelinesSummaryTableData() internal.SummaryPipelines
 	for c, p := range app.builtPipelines {
 		row := internal.SummaryPipelinesTableRowData{
 			FullName:            c.Name,
-			InputType:           c.InputType.GetString(),
+			InputType:           string(c.InputType),
 			MutatesConsumedData: p.MutatesConsumedData,
 			Receivers:           c.Receivers,
 			Processors:          c.Processors,
