diff --git a/service/internal/proctelemetry/process_telemetry.go b/service/internal/proctelemetry/process_telemetry.go
index e7a0cc145..0fe58d5d6 100644
--- a/service/internal/proctelemetry/process_telemetry.go
+++ b/service/internal/proctelemetry/process_telemetry.go
@@ -15,7 +15,6 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/service/internal/metadata"
-	"go.opentelemetry.io/collector/service/internal/servicetelemetry"
 )
 
 // processMetrics is a struct that contains views related to process metrics (cpu, mem, etc)
@@ -53,7 +52,7 @@ func WithHostProc(hostProc string) RegisterOption {
 
 // RegisterProcessMetrics creates a new set of processMetrics (mem, cpu) that can be used to measure
 // basic information about this process.
-func RegisterProcessMetrics(cfg servicetelemetry.TelemetrySettings, opts ...RegisterOption) error {
+func RegisterProcessMetrics(cfg component.TelemetrySettings, opts ...RegisterOption) error {
 	set := registerOption{}
 	for _, opt := range opts {
 		opt.apply(&set)
@@ -74,7 +73,7 @@ func RegisterProcessMetrics(cfg servicetelemetry.TelemetrySettings, opts ...Regi
 		return err
 	}
 
-	_, err = metadata.NewTelemetryBuilder(cfg.ToComponentTelemetrySettings(&component.InstanceID{}),
+	_, err = metadata.NewTelemetryBuilder(cfg,
 		metadata.WithProcessUptimeCallback(pm.updateProcessUptime),
 		metadata.WithProcessRuntimeHeapAllocBytesCallback(pm.updateAllocMem),
 		metadata.WithProcessRuntimeTotalAllocBytesCallback(pm.updateTotalAllocMem),
