diff --git a/pdata/pprofile/go.mod b/pdata/pprofile/go.mod
index 61b29996a..5886e447a 100644
--- a/pdata/pprofile/go.mod
+++ b/pdata/pprofile/go.mod
@@ -5,7 +5,7 @@ go 1.22.0
 require (
 	github.com/json-iterator/go v1.1.12
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/pdata v1.22.0
+	go.opentelemetry.io/collector/pdata v1.23.0
 	go.uber.org/goleak v1.3.0
 	google.golang.org/grpc v1.69.2
 )
