diff --git a/extension/ballastextension/memory_ballast.go b/extension/ballastextension/memory_ballast.go
index 05c3cff79..ac307a1c5 100644
--- a/extension/ballastextension/memory_ballast.go
+++ b/extension/ballastextension/memory_ballast.go
@@ -24,9 +24,6 @@ import (
 
 const megaBytes = 1024 * 1024
 
-// Deprecated: [v0.64.0] will be made private.
-type MemoryBallast = memoryBallast
-
 type memoryBallast struct {
 	cfg              *Config
 	logger           *zap.Logger
