diff --git a/internal/testutils/testutils.go b/internal/testutils/testutils.go
index 5e8c11379..42991442a 100644
--- a/internal/testutils/testutils.go
+++ b/internal/testutils/testutils.go
@@ -1,4 +1,4 @@
-// Copyright 2018, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
