diff --git a/receiver/otlpreceiver/internal/metrics/otlp.go b/receiver/otlpreceiver/internal/metrics/otlp.go
index 8449d9b81..44f81446d 100644
--- a/receiver/otlpreceiver/internal/metrics/otlp.go
+++ b/receiver/otlpreceiver/internal/metrics/otlp.go
@@ -35,15 +35,20 @@ type Receiver struct {
 }
 
 // New creates a new Receiver reference.
-func New(id component.ID, nextConsumer consumer.Metrics, set component.ReceiverCreateSettings) *Receiver {
+func New(id component.ID, nextConsumer consumer.Metrics, set component.ReceiverCreateSettings) (*Receiver, error) {
+	obsrecv, err := obsreport.NewReceiver(obsreport.ReceiverSettings{
+		ReceiverID:             id,
+		Transport:              receiverTransport,
+		ReceiverCreateSettings: set,
+	})
+	if err != nil {
+		return nil, err
+	}
+
 	return &Receiver{
 		nextConsumer: nextConsumer,
-		obsrecv: obsreport.MustNewReceiver(obsreport.ReceiverSettings{
-			ReceiverID:             id,
-			Transport:              receiverTransport,
-			ReceiverCreateSettings: set,
-		}),
-	}
+		obsrecv:      obsrecv,
+	}, nil
 }
 
 // Export implements the service Export metrics func.
