diff --git a/internal/tools/go.mod b/internal/tools/go.mod
index 9dde6b6ed..835432ad1 100644
--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -21,7 +21,7 @@ require (
 	golang.org/x/tools v0.36.0
 	golang.org/x/tools/gopls v0.20.0
 	golang.org/x/vuln v1.1.4
-	gotest.tools/gotestsum v1.12.3
+	gotest.tools/gotestsum v1.13.0
 	mvdan.cc/gofumpt v0.9.1
 )
 
@@ -244,10 +244,10 @@ require (
 	golang.org/x/exp/typeparams v0.0.0-20250620022241-b7579e27df2b // indirect
 	golang.org/x/mod v0.27.0 // indirect
 	golang.org/x/net v0.43.0 // indirect
-	golang.org/x/sync v0.16.0 // indirect
-	golang.org/x/sys v0.35.0 // indirect
+	golang.org/x/sync v0.17.0 // indirect
+	golang.org/x/sys v0.36.0 // indirect
 	golang.org/x/telemetry v0.0.0-20250807160809-1a19826ec488 // indirect
-	golang.org/x/term v0.34.0 // indirect
+	golang.org/x/term v0.35.0 // indirect
 	golang.org/x/text v0.28.0 // indirect
 	google.golang.org/protobuf v1.36.8 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
