diff --git a/receiver/otlpreceiver/internal/metadata/generated_logs.go b/receiver/otlpreceiver/internal/metadata/generated_logs.go
index 0c12252bd..37d557a42 100644
--- a/receiver/otlpreceiver/internal/metadata/generated_logs.go
+++ b/receiver/otlpreceiver/internal/metadata/generated_logs.go
@@ -62,7 +62,7 @@ func (lb *LogsBuilder) AppendLogRecord(lr plog.LogRecord) {
 // just `Emit` function can be called instead.
 // Resource attributes should be provided as ResourceLogsOption arguments.
 func (lb *LogsBuilder) EmitForResource(options ...ResourceLogsOption) {
-	rl := lb.logsBuffer.ResourceLogs().AppendEmpty()
+	rl := plog.NewResourceLogs()
 	ils := rl.ScopeLogs().AppendEmpty()
 	ils.Scope().SetName(ScopeName)
 	ils.Scope().SetVersion(lb.buildInfo.Version)
@@ -75,6 +75,10 @@ func (lb *LogsBuilder) EmitForResource(options ...ResourceLogsOption) {
 		lb.logRecordsBuffer.MoveAndAppendTo(ils.LogRecords())
 		lb.logRecordsBuffer = plog.NewLogRecordSlice()
 	}
+
+	if ils.LogRecords().Len() > 0 {
+		rl.MoveTo(lb.logsBuffer.ResourceLogs().AppendEmpty())
+	}
 }
 
 // Emit returns all the logs accumulated by the logs builder and updates the internal state to be ready for
