diff --git a/exporter/loggingexporter/internal/otlptext/metrics_test.go b/exporter/loggingexporter/internal/otlptext/metrics_test.go
index cb6a7cbbc..bb0107645 100644
--- a/exporter/loggingexporter/internal/otlptext/metrics_test.go
+++ b/exporter/loggingexporter/internal/otlptext/metrics_test.go
@@ -15,36 +15,53 @@
 package otlptext
 
 import (
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/internal/testdata"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 )
 
 func TestMetricsText(t *testing.T) {
-	type args struct {
-		md pmetric.Metrics
-	}
 	tests := []struct {
-		name  string
-		args  args
-		empty bool
+		name string
+		in   pmetric.Metrics
+		out  string
 	}{
-		{"empty metrics", args{pmetric.NewMetrics()}, true},
-		{"metrics with all types and datapoints", args{testdata.GenerateMetricsAllTypes()}, false},
-		{"metrics with all types without datapoints", args{testdata.GenerateMetricsAllTypesEmpty()}, false},
-		{"metrics with invalid metric types", args{testdata.GenerateMetricsMetricTypeInvalid()}, false},
-		{"metrics with lots of metrics", args{testdata.GenerateMetrics(10)}, false},
+		{
+			name: "empty_metrics",
+			in:   pmetric.NewMetrics(),
+			out:  "empty.out",
+		},
+		{
+			name: "metrics_with_all_types",
+			in:   testdata.GenerateMetricsAllTypes(),
+			out:  "metrics_with_all_types.out",
+		},
+		{
+			name: "two_metrics",
+			in:   testdata.GenerateMetrics(2),
+			out:  "two_metrics.out",
+		},
+		{
+			name: "invalid_metric_type",
+			in:   testdata.GenerateMetricsMetricTypeInvalid(),
+			out:  "invalid_metric_type.out",
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			metrics, err := NewTextMetricsMarshaler().MarshalMetrics(tt.args.md)
+			got, err := NewTextMetricsMarshaler().MarshalMetrics(tt.in)
 			assert.NoError(t, err)
-			if !tt.empty {
-				assert.NotEmpty(t, metrics)
-			}
+			out, err := os.ReadFile(filepath.Join("testdata", "metrics", tt.out))
+			require.NoError(t, err)
+			expected := strings.ReplaceAll(string(out), "\r", "")
+			assert.Equal(t, expected, string(got))
 		})
 	}
 }
