diff --git a/translator/trace/zipkin/zipkinv2_to_traces.go b/translator/trace/zipkin/zipkinv2_to_traces.go
index dd257c626..6e7ecf7b0 100644
--- a/translator/trace/zipkin/zipkinv2_to_traces.go
+++ b/translator/trace/zipkin/zipkinv2_to_traces.go
@@ -176,21 +176,21 @@ func populateSpanStatus(tags map[string]string, status pdata.SpanStatus) {
 func zipkinKindToSpanKind(kind zipkinmodel.Kind, tags map[string]string) pdata.SpanKind {
 	switch kind {
 	case zipkinmodel.Client:
-		return pdata.SpanKindCLIENT
+		return pdata.SpanKindClient
 	case zipkinmodel.Server:
-		return pdata.SpanKindSERVER
+		return pdata.SpanKindServer
 	case zipkinmodel.Producer:
-		return pdata.SpanKindPRODUCER
+		return pdata.SpanKindProducer
 	case zipkinmodel.Consumer:
-		return pdata.SpanKindCONSUMER
+		return pdata.SpanKindConsumer
 	default:
 		if value, ok := tags[tracetranslator.TagSpanKind]; ok {
 			delete(tags, tracetranslator.TagSpanKind)
 			if value == "internal" {
-				return pdata.SpanKindINTERNAL
+				return pdata.SpanKindInternal
 			}
 		}
-		return pdata.SpanKindUNSPECIFIED
+		return pdata.SpanKindUnspecified
 	}
 }
 
@@ -348,13 +348,13 @@ func tagsToAttributeMap(tags map[string]string, dest pdata.AttributeMap, parseSt
 
 		if parseStringTags {
 			switch determineValueType(val) {
-			case pdata.AttributeValueINT:
+			case pdata.AttributeValueTypeInt:
 				iValue, _ := strconv.ParseInt(val, 10, 64)
 				dest.UpsertInt(key, iValue)
-			case pdata.AttributeValueDOUBLE:
+			case pdata.AttributeValueTypeDouble:
 				fValue, _ := strconv.ParseFloat(val, 64)
 				dest.UpsertDouble(key, fValue)
-			case pdata.AttributeValueBOOL:
+			case pdata.AttributeValueTypeBool:
 				bValue, _ := strconv.ParseBool(val)
 				dest.UpsertBool(key, bValue)
 			default:
