diff --git a/service/telemetry/internal/migration/v0.3.0.go b/service/telemetry/internal/migration/v0.3.0.go
index ceda52ce0..d2634b5f6 100644
--- a/service/telemetry/internal/migration/v0.3.0.go
+++ b/service/telemetry/internal/migration/v0.3.0.go
@@ -70,9 +70,7 @@ type MetricsConfigV030 struct {
 	// Deprecated: [v0.111.0] use readers configuration.
 	Address string `mapstructure:"address,omitempty"`
 
-	// Readers allow configuration of metric readers to emit metrics to
-	// any number of supported backends.
-	Readers []config.MetricReader `mapstructure:"readers"`
+	config.MeterProvider `mapstructure:",squash"`
 }
 
 func (c *MetricsConfigV030) Unmarshal(conf *confmap.Conf) error {
