diff --git a/translator/trace/big_endian_converter.go b/translator/trace/big_endian_converter.go
index 84c3e1ee0..25daea54e 100644
--- a/translator/trace/big_endian_converter.go
+++ b/translator/trace/big_endian_converter.go
@@ -16,26 +16,14 @@ package tracetranslator
 
 import (
 	"encoding/binary"
-	"errors"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
 )
 
-var (
-	// ErrNilTraceID error returned when the TraceID is nil
-	ErrNilTraceID = errors.New("TraceID is nil")
-	// ErrWrongLenTraceID error returned when the TraceID does not have 16 bytes.
-	ErrWrongLenTraceID = errors.New("TraceID does not have 16 bytes")
-	// ErrNilSpanID error returned when the SpanID is nil
-	ErrNilSpanID = errors.New("SpanID is nil")
-	// ErrWrongLenSpanID error returned when the SpanID does not have 8 bytes.
-	ErrWrongLenSpanID = errors.New("SpanID does not have 8 bytes")
-)
-
 // UInt64ToByteTraceID takes a two uint64 representation of a TraceID and
 // converts it to a []byte representation.
 func UInt64ToTraceID(high, low uint64) pdata.TraceID {
-	traceID := make([]byte, 16)
+	traceID := [16]byte{}
 	binary.BigEndian.PutUint64(traceID[:8], high)
 	binary.BigEndian.PutUint64(traceID[8:], low)
 	return pdata.NewTraceID(traceID)
@@ -43,8 +31,8 @@ func UInt64ToTraceID(high, low uint64) pdata.TraceID {
 
 // UInt64ToByteTraceID takes a two uint64 representation of a TraceID and
 // converts it to a []byte representation.
-func UInt64ToByteTraceID(high, low uint64) []byte {
-	traceID := make([]byte, 16)
+func UInt64ToByteTraceID(high, low uint64) [16]byte {
+	traceID := [16]byte{}
 	binary.BigEndian.PutUint64(traceID[:8], high)
 	binary.BigEndian.PutUint64(traceID[8:], low)
 	return traceID
@@ -58,39 +46,33 @@ func Int64ToTraceID(high, low int64) pdata.TraceID {
 
 // Int64ToByteTraceID takes a two int64 representation of a TraceID and
 // converts it to a []byte representation.
-func Int64ToByteTraceID(high, low int64) []byte {
+func Int64ToByteTraceID(high, low int64) [16]byte {
 	return UInt64ToByteTraceID(uint64(high), uint64(low))
 }
 
 // BytesToUInt64TraceID takes a []byte representation of a TraceID and
 // converts it to a two uint64 representation.
-func BytesToUInt64TraceID(traceID []byte) (uint64, uint64, error) {
-	if traceID == nil {
-		return 0, 0, ErrNilTraceID
-	}
-	if len(traceID) != 16 {
-		return 0, 0, ErrWrongLenTraceID
-	}
-	return binary.BigEndian.Uint64(traceID[:8]), binary.BigEndian.Uint64(traceID[8:]), nil
+func BytesToUInt64TraceID(traceID [16]byte) (uint64, uint64) {
+	return binary.BigEndian.Uint64(traceID[:8]), binary.BigEndian.Uint64(traceID[8:])
 }
 
 // BytesToInt64TraceID takes a []byte representation of a TraceID and
 // converts it to a two int64 representation.
-func BytesToInt64TraceID(traceID []byte) (int64, int64, error) {
-	traceIDHigh, traceIDLow, err := BytesToUInt64TraceID(traceID)
-	return int64(traceIDHigh), int64(traceIDLow), err
+func BytesToInt64TraceID(traceID [16]byte) (int64, int64) {
+	traceIDHigh, traceIDLow := BytesToUInt64TraceID(traceID)
+	return int64(traceIDHigh), int64(traceIDLow)
 }
 
 // TraceIDToUInt64Pair takes a pdata.TraceID and converts it to a pair of uint64 representation.
-func TraceIDToUInt64Pair(traceID pdata.TraceID) (uint64, uint64, error) {
+func TraceIDToUInt64Pair(traceID pdata.TraceID) (uint64, uint64) {
 	return BytesToUInt64TraceID(traceID.Bytes())
 }
 
 // UInt64ToByteSpanID takes a uint64 representation of a SpanID and
 // converts it to a []byte representation.
-func UInt64ToByteSpanID(id uint64) []byte {
-	spanID := make([]byte, 8)
-	binary.BigEndian.PutUint64(spanID, id)
+func UInt64ToByteSpanID(id uint64) [8]byte {
+	spanID := [8]byte{}
+	binary.BigEndian.PutUint64(spanID[:], id)
 	return spanID
 }
 
@@ -102,7 +84,7 @@ func UInt64ToSpanID(id uint64) pdata.SpanID {
 
 // Int64ToByteSpanID takes a int64 representation of a SpanID and
 // converts it to a []byte representation.
-func Int64ToByteSpanID(id int64) []byte {
+func Int64ToByteSpanID(id int64) [8]byte {
 	return UInt64ToByteSpanID(uint64(id))
 }
 
@@ -114,19 +96,12 @@ func Int64ToSpanID(id int64) pdata.SpanID {
 
 // BytesToUInt64SpanID takes a []byte representation of a SpanID and
 // converts it to a uint64 representation.
-func BytesToUInt64SpanID(b []byte) (uint64, error) {
-	if b == nil {
-		return 0, ErrNilSpanID
-	}
-	if len(b) != 8 {
-		return 0, ErrWrongLenSpanID
-	}
-	return binary.BigEndian.Uint64(b), nil
+func BytesToUInt64SpanID(b [8]byte) uint64 {
+	return binary.BigEndian.Uint64(b[:])
 }
 
 // BytesToInt64SpanID takes a []byte representation of a SpanID and
 // converts it to a int64 representation.
-func BytesToInt64SpanID(b []byte) (int64, error) {
-	id, err := BytesToUInt64SpanID(b)
-	return int64(id), err
+func BytesToInt64SpanID(b [8]byte) int64 {
+	return int64(BytesToUInt64SpanID(b))
 }
