diff --git a/internal/fanoutconsumer/metrics.go b/internal/fanoutconsumer/metrics.go
index 32d951456..57cf10668 100644
--- a/internal/fanoutconsumer/metrics.go
+++ b/internal/fanoutconsumer/metrics.go
@@ -13,7 +13,7 @@ import (
 )
 
 // NewMetrics wraps multiple metrics consumers in a single one.
-// It fanouts the incoming data to all the consumers, and does smart routing:
+// It fans out the incoming data to all the consumers, and does smart routing:
 //   - Clones only to the consumer that needs to mutate the data.
 //   - If all consumers needs to mutate the data one will get the original mutable data.
 func NewMetrics(mcs []consumer.Metrics) consumer.Metrics {
