diff --git a/confmap/provider/httpprovider/provider_test.go b/confmap/provider/httpprovider/provider_test.go
index 86e744fd4..705e476d3 100644
--- a/confmap/provider/httpprovider/provider_test.go
+++ b/confmap/provider/httpprovider/provider_test.go
@@ -16,101 +16,14 @@ package httpprovider
 
 import (
 	"context"
-	"fmt"
-	"net/http"
-	"net/http/httptest"
-	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
-	"go.opentelemetry.io/collector/confmap/confmaptest"
 )
 
-func TestFunctionalityDownloadFileHTTP(t *testing.T) {
-	fp := New()
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		f, err := os.ReadFile("./testdata/otel-config.yaml")
-		if err != nil {
-			w.WriteHeader(404)
-			_, innerErr := w.Write([]byte("Cannot find the config file"))
-			if innerErr != nil {
-				fmt.Println("Write failed: ", innerErr)
-			}
-			return
-		}
-		w.WriteHeader(200)
-		_, err = w.Write(f)
-		if err != nil {
-			fmt.Println("Write failed: ", err)
-		}
-	}))
-	defer ts.Close()
-	_, err := fp.Retrieve(context.Background(), ts.URL, nil)
-	assert.NoError(t, err)
-	assert.NoError(t, fp.Shutdown(context.Background()))
-}
-
-func TestUnsupportedScheme(t *testing.T) {
-	fp := New()
-	_, err := fp.Retrieve(context.Background(), "https://...", nil)
-	assert.Error(t, err)
-	assert.NoError(t, fp.Shutdown(context.Background()))
-}
-
-func TestEmptyURI(t *testing.T) {
-	fp := New()
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(400)
-	}))
-	defer ts.Close()
-	_, err := fp.Retrieve(context.Background(), ts.URL, nil)
-	require.Error(t, err)
-	require.NoError(t, fp.Shutdown(context.Background()))
-}
-
-func TestRetrieveFromShutdownServer(t *testing.T) {
-	fp := New()
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
-	ts.Close()
-	_, err := fp.Retrieve(context.Background(), ts.URL, nil)
-	assert.Error(t, err)
-	require.NoError(t, fp.Shutdown(context.Background()))
-}
-
-func TestNonExistent(t *testing.T) {
-	fp := New()
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(404)
-	}))
-	defer ts.Close()
-	_, err := fp.Retrieve(context.Background(), ts.URL, nil)
-	assert.Error(t, err)
-	require.NoError(t, fp.Shutdown(context.Background()))
-}
-
-func TestInvalidYAML(t *testing.T) {
-	fp := New()
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(200)
-		_, err := w.Write([]byte("wrong : ["))
-		if err != nil {
-			fmt.Println("Write failed: ", err)
-		}
-	}))
-	defer ts.Close()
-	_, err := fp.Retrieve(context.Background(), ts.URL, nil)
-	assert.Error(t, err)
-	require.NoError(t, fp.Shutdown(context.Background()))
-}
-
-func TestScheme(t *testing.T) {
+func TestSupportedScheme(t *testing.T) {
 	fp := New()
 	assert.Equal(t, "http", fp.Scheme())
 	require.NoError(t, fp.Shutdown(context.Background()))
 }
-
-func TestValidateProviderScheme(t *testing.T) {
-	assert.NoError(t, confmaptest.ValidateProviderScheme(New()))
-}
