diff --git a/receiver/otlpreceiver/metrics/otlp_test.go b/receiver/otlpreceiver/metrics/otlp_test.go
index 965d8edb6..896f4a526 100644
--- a/receiver/otlpreceiver/metrics/otlp_test.go
+++ b/receiver/otlpreceiver/metrics/otlp_test.go
@@ -27,15 +27,12 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/collector/internal"
 	collectormetrics "go.opentelemetry.io/collector/internal/data/protogen/collector/metrics/v1"
-	otlpcommon "go.opentelemetry.io/collector/internal/data/protogen/common/v1"
-	otlpmetrics "go.opentelemetry.io/collector/internal/data/protogen/metrics/v1"
+	"go.opentelemetry.io/collector/internal/pdatagrpc"
+	"go.opentelemetry.io/collector/internal/testdata"
 	"go.opentelemetry.io/collector/obsreport"
 )
 
-var _ collectormetrics.MetricsServiceServer = (*Receiver)(nil)
-
 func TestExport(t *testing.T) {
 	// given
 
@@ -50,60 +47,11 @@ func TestExport(t *testing.T) {
 
 	// when
 
-	unixnanos1 := uint64(12578940000000012345)
-	unixnanos2 := uint64(12578940000000054321)
-
-	req := &collectormetrics.ExportMetricsServiceRequest{
-		ResourceMetrics: []*otlpmetrics.ResourceMetrics{
-			{
-				InstrumentationLibraryMetrics: []*otlpmetrics.InstrumentationLibraryMetrics{
-					{
-						Metrics: []*otlpmetrics.Metric{
-							{
-								Name:        "mymetric",
-								Description: "My metric",
-								Unit:        "ms",
-								Data: &otlpmetrics.Metric_IntSum{
-									IntSum: &otlpmetrics.IntSum{
-										IsMonotonic:            true,
-										AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
-										DataPoints: []*otlpmetrics.IntDataPoint{
-											{
-												Labels: []otlpcommon.StringKeyValue{
-													{
-														Key:   "key1",
-														Value: "value1",
-													},
-												},
-												StartTimeUnixNano: unixnanos1,
-												TimeUnixNano:      unixnanos2,
-												Value:             123,
-											},
-											{
-												Labels: []otlpcommon.StringKeyValue{
-													{
-														Key:   "key2",
-														Value: "value2",
-													},
-												},
-												StartTimeUnixNano: unixnanos1,
-												TimeUnixNano:      unixnanos2,
-												Value:             456,
-											},
-										},
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-		},
-	}
+	req := testdata.GenerateMetricsOneMetric()
 
 	// Keep metric data to compare the test result against it
 	// Clone needed because OTLP proto XXX_ fields are altered in the GRPC downstream
-	metricData := pdata.MetricsFromInternalRep(internal.MetricsFromOtlp(req)).Clone()
+	metricData := req.Clone()
 
 	resp, err := metricsClient.Export(context.Background(), req)
 	require.NoError(t, err, "Failed to export metrics: %v", err)
@@ -111,10 +59,9 @@ func TestExport(t *testing.T) {
 
 	// assert
 
-	require.Equal(t, 1, len(metricSink.AllMetrics()),
-		"unexpected length: %v", len(metricSink.AllMetrics()))
-
-	assert.EqualValues(t, metricData, metricSink.AllMetrics()[0])
+	mds := metricSink.AllMetrics()
+	require.Len(t, mds, 1)
+	assert.EqualValues(t, metricData, mds[0])
 }
 
 func TestExport_EmptyRequest(t *testing.T) {
@@ -129,7 +76,7 @@ func TestExport_EmptyRequest(t *testing.T) {
 	require.NoError(t, err, "Failed to create the MetricsServiceClient: %v", err)
 	defer metricsClientDoneFn()
 
-	resp, err := metricsClient.Export(context.Background(), &collectormetrics.ExportMetricsServiceRequest{})
+	resp, err := metricsClient.Export(context.Background(), pdata.NewMetrics())
 	require.NoError(t, err)
 	require.NotNil(t, resp)
 }
@@ -144,47 +91,20 @@ func TestExport_ErrorConsumer(t *testing.T) {
 	require.NoError(t, err, "Failed to create the MetricsServiceClient: %v", err)
 	defer metricsClientDoneFn()
 
-	req := &collectormetrics.ExportMetricsServiceRequest{ResourceMetrics: []*otlpmetrics.ResourceMetrics{
-		{
-			InstrumentationLibraryMetrics: []*otlpmetrics.InstrumentationLibraryMetrics{
-				{
-					Metrics: []*otlpmetrics.Metric{
-						{
-							Name:        "mymetric",
-							Description: "My metric",
-							Unit:        "ms",
-							Data: &otlpmetrics.Metric_IntSum{
-								IntSum: &otlpmetrics.IntSum{
-									IsMonotonic:            true,
-									AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
-									DataPoints: []*otlpmetrics.IntDataPoint{
-										{
-											Value: 123,
-										},
-										{
-											Value: 456,
-										},
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-		},
-	}}
+	req := testdata.GenerateMetricsOneMetric()
+
 	resp, err := metricsClient.Export(context.Background(), req)
 	assert.EqualError(t, err, "rpc error: code = Unknown desc = my error")
 	assert.Nil(t, resp)
 }
 
-func makeMetricsServiceClient(addr net.Addr) (collectormetrics.MetricsServiceClient, func(), error) {
+func makeMetricsServiceClient(addr net.Addr) (pdatagrpc.MetricsClient, func(), error) {
 	cc, err := grpc.Dial(addr.String(), grpc.WithInsecure(), grpc.WithBlock())
 	if err != nil {
 		return nil, nil, err
 	}
 
-	metricsClient := collectormetrics.NewMetricsServiceClient(cc)
+	metricsClient := pdatagrpc.NewMetricsClient(cc)
 
 	doneFn := func() { _ = cc.Close() }
 	return metricsClient, doneFn, nil
