diff --git a/config/configtls/configtls_test.go b/config/configtls/configtls_test.go
index 823c35e7f..57b0f920d 100644
--- a/config/configtls/configtls_test.go
+++ b/config/configtls/configtls_test.go
@@ -456,7 +456,7 @@ func overwriteClientCA(t *testing.T, targetFilePath string, testdataFileName str
 }
 
 func createTempClientCaFile(t *testing.T) string {
-	tmpCa, err := os.CreateTemp("", "ca-tmp.crt")
+	tmpCa, err := os.CreateTemp(t.TempDir(), "ca-tmp.crt")
 	require.NoError(t, err)
 	tmpCaPath, err := filepath.Abs(tmpCa.Name())
 	assert.NoError(t, err)
@@ -533,18 +533,20 @@ func TestCertificateReload(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			// Copy certs into a temp dir so we can safely modify them
-			certFile, err := os.CreateTemp("", "cert")
+			tempDir := t.TempDir()
+			certFile, err := os.CreateTemp(tempDir, "cert")
 			require.NoError(t, err)
-			defer os.Remove(certFile.Name())
+			defer certFile.Close()
 
-			keyFile, err := os.CreateTemp("", "key")
+			keyFile, err := os.CreateTemp(tempDir, "key")
 			require.NoError(t, err)
-			defer os.Remove(keyFile.Name())
+			defer keyFile.Close()
 
 			fdc, err := os.Open(filepath.Join("testdata", "client-1.crt"))
 			require.NoError(t, err)
 			_, err = io.Copy(certFile, fdc)
 			require.NoError(t, err)
+			require.NoError(t, fdc.Close())
 
 			fdk, err := os.Open(filepath.Join("testdata", "client-1.key"))
 			require.NoError(t, err)
