diff --git a/exporter/exportertest/nop_exporter.go b/exporter/exportertest/nop_exporter.go
index c1c9fc1e3..f504a547b 100644
--- a/exporter/exportertest/nop_exporter.go
+++ b/exporter/exportertest/nop_exporter.go
@@ -20,6 +20,8 @@ import (
 	"go.opentelemetry.io/collector/exporter"
 )
 
+const typeStr = "nop"
+
 // NewNopCreateSettings returns a new nop settings for Create*Exporter functions.
 func NewNopCreateSettings() exporter.CreateSettings {
 	return exporter.CreateSettings{
@@ -30,3 +32,11 @@ func NewNopCreateSettings() exporter.CreateSettings {
 
 // NewNopFactory returns an exporter.Factory that constructs nop exporters.
 var NewNopFactory = componenttest.NewNopExporterFactory //nolint:staticcheck
+
+// NewNopBuilder returns a exporter.Builder that constructs nop receivers.
+func NewNopBuilder() *exporter.Builder {
+	nopFactory := NewNopFactory()
+	return exporter.NewBuilder(
+		map[component.ID]component.Config{component.NewID(typeStr): nopFactory.CreateDefaultConfig()},
+		map[component.Type]exporter.Factory{typeStr: nopFactory})
+}
