diff --git a/exporter/exporterhelper/internal/queue/memory_queue.go b/exporter/exporterhelper/internal/queue/memory_queue.go
index d724259aa..24529e6ac 100644
--- a/exporter/exporterhelper/internal/queue/memory_queue.go
+++ b/exporter/exporterhelper/internal/queue/memory_queue.go
@@ -44,7 +44,7 @@ type memoryQueue[T any] struct {
 
 // newMemoryQueue creates a sized elements channel. Each element is assigned a size by the provided sizer.
 // capacity is the capacity of the queue.
-func newMemoryQueue[T any](set Settings[T]) readableQueue[T] {
+func newMemoryQueue[T request.Request](set Settings[T]) readableQueue[T] {
 	sq := &memoryQueue[T]{
 		refCounter:      set.ReferenceCounter,
 		sizer:           set.activeSizer(),
