diff --git a/config/configgrpc/gzip.go b/config/configgrpc/gzip.go
index 6a9a36f01..db96de2a4 100644
--- a/config/configgrpc/gzip.go
+++ b/config/configgrpc/gzip.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
