diff --git a/exporter/exporterhelper/xexporterhelper/profiles_batch.go b/exporter/exporterhelper/xexporterhelper/profiles_batch.go
index 8fd337878..2aa47c950 100644
--- a/exporter/exporterhelper/xexporterhelper/profiles_batch.go
+++ b/exporter/exporterhelper/xexporterhelper/profiles_batch.go
@@ -37,13 +37,15 @@ func (req *profilesRequest) MergeSplit(_ context.Context, cfg exporterbatcher.Ma
 		if srcReq == nil {
 			continue
 		}
-		if srcReq.pd.SampleCount() <= capacityLeft {
+
+		srcCount := srcReq.pd.SampleCount()
+		if srcCount <= capacityLeft {
 			if destReq == nil {
 				destReq = srcReq
 			} else {
 				srcReq.pd.ResourceProfiles().MoveAndAppendTo(destReq.pd.ResourceProfiles())
 			}
-			capacityLeft -= destReq.pd.SampleCount()
+			capacityLeft -= srcCount
 			continue
 		}
 
