diff --git a/translator/internaldata/oc_to_traces.go b/translator/internaldata/oc_to_traces.go
index 28c4fa5a5..760fc6ab9 100644
--- a/translator/internaldata/oc_to_traces.go
+++ b/translator/internaldata/oc_to_traces.go
@@ -245,10 +245,10 @@ func initAttributeMapFromOC(ocAttrs *octrace.Span_Attributes, dest pdata.Attribu
 func ocSpanKindToInternal(ocKind octrace.Span_SpanKind, ocAttrs *octrace.Span_Attributes) pdata.SpanKind {
 	switch ocKind {
 	case octrace.Span_SERVER:
-		return pdata.SpanKindSERVER
+		return pdata.SpanKindServer
 
 	case octrace.Span_CLIENT:
-		return pdata.SpanKindCLIENT
+		return pdata.SpanKindClient
 
 	case octrace.Span_SPAN_KIND_UNSPECIFIED:
 		// Span kind field is unspecified, check if TagSpanKind attribute is set.
@@ -262,23 +262,23 @@ func ocSpanKindToInternal(ocKind octrace.Span_SpanKind, ocAttrs *octrace.Span_At
 					var otlpKind pdata.SpanKind
 					switch tracetranslator.OpenTracingSpanKind(strVal.StringValue.GetValue()) {
 					case tracetranslator.OpenTracingSpanKindConsumer:
-						otlpKind = pdata.SpanKindCONSUMER
+						otlpKind = pdata.SpanKindConsumer
 					case tracetranslator.OpenTracingSpanKindProducer:
-						otlpKind = pdata.SpanKindPRODUCER
+						otlpKind = pdata.SpanKindProducer
 					case tracetranslator.OpenTracingSpanKindInternal:
-						otlpKind = pdata.SpanKindINTERNAL
+						otlpKind = pdata.SpanKindInternal
 					default:
-						return pdata.SpanKindUNSPECIFIED
+						return pdata.SpanKindUnspecified
 					}
 					delete(ocAttrs.AttributeMap, tracetranslator.TagSpanKind)
 					return otlpKind
 				}
 			}
 		}
-		return pdata.SpanKindUNSPECIFIED
+		return pdata.SpanKindUnspecified
 
 	default:
-		return pdata.SpanKindUNSPECIFIED
+		return pdata.SpanKindUnspecified
 	}
 }
 
