diff --git a/config/configmapprovider/merge_test.go b/config/configmapprovider/merge_test.go
index a5a06e0a7..385144a04 100644
--- a/config/configmapprovider/merge_test.go
+++ b/config/configmapprovider/merge_test.go
@@ -21,12 +21,10 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
-	"go.opentelemetry.io/collector/config"
 )
 
 func TestMerge_GetError(t *testing.T) {
-	pl := NewMerge(&errProvider{err: nil}, &errProvider{err: errors.New("my error")})
+	pl := NewMerge(&mockProvider{}, &mockProvider{retrieved: &mockRetrieved{getErr: errors.New("my error")}})
 	require.NotNil(t, pl)
 	cp, err := pl.Retrieve(context.Background(), nil)
 	assert.Error(t, err)
@@ -34,7 +32,7 @@ func TestMerge_GetError(t *testing.T) {
 }
 
 func TestMerge_CloseError(t *testing.T) {
-	pl := NewMerge(&errProvider{err: nil}, &errProvider{closeErr: errors.New("my error")})
+	pl := NewMerge(&mockProvider{}, &mockProvider{retrieved: &mockRetrieved{closeErr: errors.New("my error")}})
 	require.NotNil(t, pl)
 	cp, err := pl.Retrieve(context.Background(), nil)
 	assert.NoError(t, err)
@@ -42,23 +40,7 @@ func TestMerge_CloseError(t *testing.T) {
 }
 
 func TestMerge_ShutdownError(t *testing.T) {
-	pl := NewMerge(&errProvider{err: nil}, &errProvider{err: errors.New("my error")})
+	pl := NewMerge(&mockProvider{}, &mockProvider{shutdownErr: errors.New("my error")})
 	require.NotNil(t, pl)
 	assert.Error(t, pl.Shutdown(context.Background()))
 }
-
-type errProvider struct {
-	err      error
-	closeErr error
-}
-
-func (epl *errProvider) Retrieve(context.Context, func(*ChangeEvent)) (Retrieved, error) {
-	if epl.err == nil {
-		return &simpleRetrieved{confMap: config.NewMap(), closeFunc: func(context.Context) error { return epl.closeErr }}, nil
-	}
-	return nil, epl.err
-}
-
-func (epl *errProvider) Shutdown(context.Context) error {
-	return epl.err
-}
