diff --git a/exporter/exporterhelper/internal/queuebatch/config.go b/exporter/exporterhelper/internal/queuebatch/config.go
index 645969c13..77f44c56f 100644
--- a/exporter/exporterhelper/internal/queuebatch/config.go
+++ b/exporter/exporterhelper/internal/queuebatch/config.go
@@ -44,7 +44,7 @@ type Config struct {
 
 	// BatchConfig it configures how the requests are consumed from the queue and batch together during consumption.
 	// TODO: This will be changed to Optional when available.
-	BatchConfig *BatchConfig `mapstructure:"batch"`
+	Batch *BatchConfig `mapstructure:"batch"`
 }
 
 // Validate checks if the Config is valid
