diff --git a/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_metadata.go b/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_metadata.go
index cb7e9f0b8..5437e1ff9 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_metadata.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_metadata.go
@@ -54,7 +54,7 @@ var swapUsageDescriptor = func() pdata.Metric {
 	metric.InitEmpty()
 	metric.SetName("system.swap.usage")
 	metric.SetDescription("Swap (unix) or pagefile (windows) usage.")
-	metric.SetUnit("pages")
+	metric.SetUnit("bytes")
 	metric.SetDataType(pdata.MetricDataTypeIntSum)
 	sum := metric.IntSum()
 	sum.InitEmpty()
