diff --git a/pdata/internal/cmd/pdatagen/internal/pmetricotlp_structs.go b/pdata/internal/cmd/pdatagen/internal/pmetricotlp_structs.go
index 9b8dcc186..8a00220e8 100644
--- a/pdata/internal/cmd/pdatagen/internal/pmetricotlp_structs.go
+++ b/pdata/internal/cmd/pdatagen/internal/pmetricotlp_structs.go
@@ -42,10 +42,9 @@ var metricsOtlpFile = &File{
 }
 
 var exportMetricsPartialSuccess = &messageValueStruct{
-	structName:         "ExportPartialSuccess",
-	internalStructName: "MetricsExportPartialSuccess",
-	description:        "// ExportPartialSuccess represents the details of a partially successful export request.",
-	originFullName:     "otlpcollectormetrics.ExportMetricsPartialSuccess",
+	structName:     "ExportPartialSuccess",
+	description:    "// ExportPartialSuccess represents the details of a partially successful export request.",
+	originFullName: "otlpcollectormetrics.ExportMetricsPartialSuccess",
 	fields: []baseField{
 		&primitiveField{
 			fieldName:  "RejectedDataPoints",
