diff --git a/extension/healthcheckextension/healthcheckextension_test.go b/extension/healthcheckextension/healthcheckextension_test.go
index 603756e1f..f59019275 100644
--- a/extension/healthcheckextension/healthcheckextension_test.go
+++ b/extension/healthcheckextension/healthcheckextension_test.go
@@ -21,11 +21,12 @@ import (
 	"runtime"
 	"strconv"
 	"testing"
-	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/testutil"
 )
@@ -86,14 +87,8 @@ func TestHealthCheckExtensionPortAlreadyInUse(t *testing.T) {
 	hcExt := newServer(config, zap.NewNop())
 	require.NotNil(t, hcExt)
 
-	// Health check will report port already in use in a goroutine, use the error waiting
-	// host to get it.
-	mh := componenttest.NewErrorWaitingHost()
-	require.NoError(t, hcExt.Start(context.Background(), mh))
-
-	receivedError, receivedErr := mh.WaitForFatalError(500 * time.Millisecond)
-	require.True(t, receivedError)
-	require.Error(t, receivedErr)
+	mh := newAssertNoErrorHost(t)
+	require.Error(t, hcExt.Start(context.Background(), mh))
 }
 
 func TestHealthCheckMultipleStarts(t *testing.T) {
@@ -104,17 +99,11 @@ func TestHealthCheckMultipleStarts(t *testing.T) {
 	hcExt := newServer(config, zap.NewNop())
 	require.NotNil(t, hcExt)
 
-	mh := componenttest.NewErrorWaitingHost()
+	mh := newAssertNoErrorHost(t)
 	require.NoError(t, hcExt.Start(context.Background(), mh))
 	defer hcExt.Shutdown(context.Background())
 
-	// Health check will report already in use in a goroutine, use the error waiting
-	// host to get it.
-	require.NoError(t, hcExt.Start(context.Background(), mh))
-
-	receivedError, receivedErr := mh.WaitForFatalError(500 * time.Millisecond)
-	require.True(t, receivedError)
-	require.Error(t, receivedErr)
+	require.Error(t, hcExt.Start(context.Background(), mh))
 }
 
 func TestHealthCheckMultipleShutdowns(t *testing.T) {
@@ -140,3 +129,23 @@ func TestHealthCheckShutdownWithoutStart(t *testing.T) {
 
 	require.NoError(t, hcExt.Shutdown(context.Background()))
 }
+
+// assertNoErrorHost implements a component.Host that asserts that there were no errors.
+type assertNoErrorHost struct {
+	component.Host
+	*testing.T
+}
+
+var _ component.Host = (*assertNoErrorHost)(nil)
+
+// newAssertNoErrorHost returns a new instance of assertNoErrorHost.
+func newAssertNoErrorHost(t *testing.T) component.Host {
+	return &assertNoErrorHost{
+		componenttest.NewNopHost(),
+		t,
+	}
+}
+
+func (aneh *assertNoErrorHost) ReportFatalError(err error) {
+	assert.NoError(aneh, err)
+}
