diff --git a/exporter/exporterhelper/queue_batch.go b/exporter/exporterhelper/queue_batch.go
index f4e3a4390..0e90b6bcf 100644
--- a/exporter/exporterhelper/queue_batch.go
+++ b/exporter/exporterhelper/queue_batch.go
@@ -11,14 +11,6 @@ import (
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/queuebatch"
 )
 
-// Deprecated: [v0.123.0] use QueueBatchConfig.
-type QueueConfig = QueueBatchConfig
-
-// Deprecated: [v0.123.0] use WithQueueBatch.
-func WithRequestQueue(cfg QueueBatchConfig, encoding QueueBatchEncoding[Request]) Option {
-	return WithQueueBatch(cfg, QueueBatchSettings{Encoding: encoding})
-}
-
 // WithQueue overrides the default QueueBatchConfig for an exporter.
 // The default QueueBatchConfig is to disable queueing.
 // This option cannot be used with the new exporter helpers New[Traces|Metrics|Logs]RequestExporter.
@@ -26,11 +18,6 @@ func WithQueue(config QueueBatchConfig) Option {
 	return internal.WithQueue(config)
 }
 
-// Deprecated: [v0.123.0] use WithQueueBatch.
-func WithBatcher(cfg BatcherConfig) Option {
-	return internal.WithBatcher(cfg)
-}
-
 // QueueBatchConfig defines configuration for queueing and batching for the exporter.
 type QueueBatchConfig = queuebatch.Config
 
@@ -64,12 +51,3 @@ func WithQueueBatch(cfg QueueBatchConfig, set QueueBatchSettings) Option {
 // NewDefaultQueueConfig returns the default config for QueueBatchConfig.
 // By default, the queue stores 1000 requests of telemetry and is non-blocking when full.
 var NewDefaultQueueConfig = internal.NewDefaultQueueConfig
-
-// Deprecated: [v0.123.0] use WithQueueBatch.
-type BatcherConfig = internal.BatcherConfig
-
-// Deprecated: [v0.123.0] use WithQueueBatch.
-type SizeConfig = internal.SizeConfig
-
-// Deprecated: [v0.123.0] use WithQueueBatch.
-var NewDefaultBatcherConfig = internal.NewDefaultBatcherConfig
