diff --git a/semconv/template.j2 b/semconv/template.j2
index 904d1e15a..dfe143f8e 100644
--- a/semconv/template.j2
+++ b/semconv/template.j2
@@ -16,12 +16,14 @@ Type: {{ attr.attr_type }}
 {%- else %}
 Type: Enum
 {%- endif %}
-{%- if attr.required == Required.ALWAYS %}
-Required: Always
-{%- elif attr.required == Required.CONDITIONAL %}
-Required: {{ attr.required_msg }}
+{%- if attr.requirement_level == RequirementLevel.REQUIRED %}
+Requirement Level: Required
+{%- elif attr.requirement_level == RequirementLevel.CONDITIONALLY_REQUIRED %}
+Requirement Level: Conditionally Required{% if attr.requirement_level_msg != "" %} - {{ attr.requirement_level_msg }} {% endif %}
+{%- elif attr.requirement_level == RequirementLevel.RECOMMENDED %}
+Requirement Level: Recommended{% if attr.requirement_level_msg != "" %} - {{ attr.requirement_level_msg }} {% endif %}
 {%- else %}
-Required: No
+Requirement Level: Optional
 {%- endif %}
 {{ attr.stability |  replace("Level.", ": ") | capitalize }}
 {%- if attr.deprecated != None %}
