diff --git a/exporter/exportertest/sink_exporter_test.go b/exporter/exportertest/sink_exporter_test.go
index ac76c9a77..fb50f8f0b 100644
--- a/exporter/exportertest/sink_exporter_test.go
+++ b/exporter/exportertest/sink_exporter_test.go
@@ -39,7 +39,7 @@ func TestSinkTraceExporter(t *testing.T) {
 	if !reflect.DeepEqual(got, want) {
 		t.Errorf("Mismatches responses\nGot:\n\t%v\nWant:\n\t%v\n", got, want)
 	}
-	if "sink_trace" != sink.TraceExportFormat() {
+	if sink.TraceExportFormat() != "sink_trace" {
 		t.Errorf("Wanted sink_trace got %s", sink.TraceExportFormat())
 	}
 }
@@ -60,7 +60,7 @@ func TestSinkMetricsExporter(t *testing.T) {
 	if !reflect.DeepEqual(got, want) {
 		t.Errorf("Mismatches responses\nGot:\n\t%v\nWant:\n\t%v\n", got, want)
 	}
-	if "sink_metrics" != sink.MetricsExportFormat() {
+	if sink.MetricsExportFormat() != "sink_metrics" {
 		t.Errorf("Wanted sink_metrics got %s", sink.MetricsExportFormat())
 	}
 }
