diff --git a/service/internal/graph/util_test.go b/service/internal/graph/util_test.go
index 43f1e64bb..0a9165622 100644
--- a/service/internal/graph/util_test.go
+++ b/service/internal/graph/util_test.go
@@ -22,6 +22,7 @@ import (
 	"go.opentelemetry.io/collector/processor/processorprofiles"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/receiver/receiverprofiles"
+	"go.opentelemetry.io/collector/service/internal/testcomponents"
 	"go.opentelemetry.io/collector/service/pipelines"
 )
 
@@ -146,6 +147,29 @@ func expectedInstances(m pipelines.Config, pID pipeline.ID) (int, int) {
 	return r, e
 }
 
+// connector needs to be unwrapped to access component as ExampleConnector
+func unwrapExampleConnector(c *connectorNode) *testcomponents.ExampleConnector {
+	switch ct := c.Component.(type) {
+	case componentTraces: // consumes traces, emits traces
+		return ct.Component.(*testcomponents.ExampleConnector)
+	case connector.Traces: // consumes traces, emits something else
+		return ct.(*testcomponents.ExampleConnector)
+	case componentMetrics: // consumes metrics, emits metrics
+		return ct.Component.(*testcomponents.ExampleConnector)
+	case connector.Metrics: // consumes metrics, emits something else
+		return ct.(*testcomponents.ExampleConnector)
+	case componentLogs: // consumes logs, emits logs
+		return ct.Component.(*testcomponents.ExampleConnector)
+	case connector.Logs: // consumes logs, emits something else
+		return ct.(*testcomponents.ExampleConnector)
+	case componentProfiles: // consumes profiles, emits profiles
+		return ct.Component.(*testcomponents.ExampleConnector)
+	case connectorprofiles.Profiles: // consumes profiles, emits something else
+		return ct.(*testcomponents.ExampleConnector)
+	}
+	return nil
+}
+
 func newBadReceiverFactory() receiver.Factory {
 	return receiver.NewFactory(component.MustNewType("bf"), func() component.Config {
 		return &struct{}{}
