diff --git a/pdata/pmetric/metrics.go b/pdata/pmetric/metrics.go
index 998ef8e41..caf681a06 100644
--- a/pdata/pmetric/metrics.go
+++ b/pdata/pmetric/metrics.go
@@ -212,3 +212,9 @@ func (nt ExemplarValueType) String() string {
 	}
 	return ""
 }
+
+// Deprecated: [0.62.0] Use ExponentialHistogramDataPointBuckets instead.
+type Buckets = ExponentialHistogramDataPointBuckets
+
+// Deprecated: [0.62.0] Use NewExponentialHistogramDataPointBuckets instead.
+var NewBuckets = NewExponentialHistogramDataPointBuckets
