diff --git a/internal/collector/processor/queued/queued_processor_test.go b/internal/collector/processor/queued/queued_processor_test.go
index 3d8e6bd3b..5a62c2808 100644
--- a/internal/collector/processor/queued/queued_processor_test.go
+++ b/internal/collector/processor/queued/queued_processor_test.go
@@ -20,16 +20,17 @@ import (
 	"sync/atomic"
 	"testing"
 
+	"github.com/census-instrumentation/opencensus-service/consumer"
+
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 	"github.com/census-instrumentation/opencensus-service/data"
-	"github.com/census-instrumentation/opencensus-service/internal/collector/processor"
 )
 
 func TestQueueProcessorHappyPath(t *testing.T) {
 	mockProc := newMockConcurrentSpanProcessor()
 	qp := NewQueuedSpanProcessor(mockProc)
 	goFn := func(td data.TraceData) {
-		qp.ProcessSpans(context.Background(), td)
+		qp.ConsumeTraceData(context.Background(), td)
 	}
 
 	spans := []*tracepb.Span{{}}
@@ -63,9 +64,9 @@ type mockConcurrentSpanProcessor struct {
 	spanCount  int32
 }
 
-var _ processor.SpanProcessor = (*mockConcurrentSpanProcessor)(nil)
+var _ consumer.TraceConsumer = (*mockConcurrentSpanProcessor)(nil)
 
-func (p *mockConcurrentSpanProcessor) ProcessSpans(ctx context.Context, td data.TraceData) error {
+func (p *mockConcurrentSpanProcessor) ConsumeTraceData(ctx context.Context, td data.TraceData) error {
 	atomic.AddInt32(&p.batchCount, 1)
 	atomic.AddInt32(&p.spanCount, int32(len(td.Spans)))
 	p.waitGroup.Done()
