diff --git a/service/telemetry_test.go b/service/telemetry_test.go
index 4d9420594..afd3766fe 100644
--- a/service/telemetry_test.go
+++ b/service/telemetry_test.go
@@ -102,29 +102,13 @@ func TestTelemetryInit(t *testing.T) {
 
 	for _, tc := range []struct {
 		name            string
-		useOtel         bool
 		disableHighCard bool
 		expectedMetrics map[string]metricValue
 		extendedConfig  bool
 		cfg             *telemetry.Config
 	}{
 		{
-			name:    "UseOpenCensusForInternalMetrics",
-			useOtel: false,
-			expectedMetrics: map[string]metricValue{
-				metricPrefix + ocPrefix + counterName: {
-					value: 13,
-					labels: map[string]string{
-						"service_name":        "otelcol",
-						"service_version":     "latest",
-						"service_instance_id": testInstanceID,
-					},
-				},
-			},
-		},
-		{
-			name:    "UseOpenTelemetryForInternalMetrics",
-			useOtel: true,
+			name: "UseOpenTelemetryForInternalMetrics",
 			expectedMetrics: map[string]metricValue{
 				metricPrefix + ocPrefix + counterName: {
 					value: 13,
@@ -175,7 +159,6 @@ func TestTelemetryInit(t *testing.T) {
 		},
 		{
 			name:            "DisableHighCardinalityWithOtel",
-			useOtel:         true,
 			disableHighCard: true,
 			expectedMetrics: map[string]metricValue{
 				metricPrefix + ocPrefix + counterName: {
@@ -292,7 +275,7 @@ func TestTelemetryInit(t *testing.T) {
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			tel := newColTelemetry(tc.useOtel, tc.disableHighCard, tc.extendedConfig)
+			tel := newColTelemetry(tc.disableHighCard, tc.extendedConfig)
 			buildInfo := component.NewDefaultBuildInfo()
 			if tc.extendedConfig {
 				tc.cfg.Metrics.Readers = []config.MetricReader{
