diff --git a/service/internal/telemetrylogs/logger.go b/service/internal/telemetrylogs/logger.go
index ee1c4efa8..a903e3e0b 100644
--- a/service/internal/telemetrylogs/logger.go
+++ b/service/internal/telemetrylogs/logger.go
@@ -30,10 +30,13 @@ func NewLogger(cfg config.ServiceTelemetryLogs, options []zap.Option) (*zap.Logg
 			Initial:    100,
 			Thereafter: 100,
 		},
-		Encoding:         cfg.Encoding,
-		EncoderConfig:    zap.NewProductionEncoderConfig(),
-		OutputPaths:      []string{"stderr"},
-		ErrorOutputPaths: []string{"stderr"},
+		Encoding:          cfg.Encoding,
+		EncoderConfig:     zap.NewProductionEncoderConfig(),
+		OutputPaths:       cfg.OutputPaths,
+		ErrorOutputPaths:  cfg.ErrorOutputPaths,
+		DisableCaller:     cfg.DisableCaller,
+		DisableStacktrace: cfg.DisableStacktrace,
+		InitialFields:     cfg.InitialFields,
 	}
 
 	if zapCfg.Encoding == "console" {
