diff --git a/config/configtls/configtls.go b/config/configtls/configtls.go
index 93444cb0b..22c05b00a 100644
--- a/config/configtls/configtls.go
+++ b/config/configtls/configtls.go
@@ -169,8 +169,8 @@ func (c TLSSetting) loadTLSConfig() (*tls.Config, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to load TLS cert and key: %w", err)
 		}
-		getCertificate = func(_ *tls.ClientHelloInfo) (*tls.Certificate, error) { return certReloader.GetCertificate() }
-		getClientCertificate = func(_ *tls.CertificateRequestInfo) (*tls.Certificate, error) { return certReloader.GetCertificate() }
+		getCertificate = func(*tls.ClientHelloInfo) (*tls.Certificate, error) { return certReloader.GetCertificate() }
+		getClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) { return certReloader.GetCertificate() }
 	}
 
 	minTLS, err := convertVersion(c.MinVersion, defaultMinTLSVersion)
@@ -342,7 +342,7 @@ func (c TLSServerSetting) LoadTLSConfig() (*tls.Config, error) {
 			if err != nil {
 				return nil, err
 			}
-			tlsCfg.GetConfigForClient = func(_ *tls.ClientHelloInfo) (*tls.Config, error) { return reloader.getClientConfig(tlsCfg) }
+			tlsCfg.GetConfigForClient = func(*tls.ClientHelloInfo) (*tls.Config, error) { return reloader.getClientConfig(tlsCfg) }
 		}
 		tlsCfg.ClientCAs = reloader.certPool
 		tlsCfg.ClientAuth = tls.RequireAndVerifyClientCert
