diff --git a/pdata/testdata/profile.go b/pdata/testdata/profile.go
index a065a3db0..76f5429c4 100644
--- a/pdata/testdata/profile.go
+++ b/pdata/testdata/profile.go
@@ -45,7 +45,7 @@ func GenerateProfiles(profilesCount int) pprofile.Profiles {
 func fillProfileOne(dic pprofile.ProfilesDictionary, profile pprofile.Profile) {
 	profile.SetProfileID([16]byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10})
 	profile.SetTime(profileStartTimestamp)
-	profile.SetDurationNano(uint64(time.Second.Nanoseconds())) //nolint:gosec // G115 overflow checked
+	profile.SetDurationNano(uint64(time.Second.Nanoseconds()))
 	profile.SetDroppedAttributesCount(1)
 
 	loc := pprofile.NewLocation()
@@ -63,7 +63,7 @@ func fillProfileOne(dic pprofile.ProfilesDictionary, profile pprofile.Profile) {
 func fillProfileTwo(dic pprofile.ProfilesDictionary, profile pprofile.Profile) {
 	profile.SetProfileID([16]byte{0x02, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10})
 	profile.SetTime(profileStartTimestamp)
-	profile.SetDurationNano(uint64(time.Second.Nanoseconds())) //nolint:gosec // G115 overflow checked
+	profile.SetDurationNano(uint64(time.Second.Nanoseconds()))
 
 	loc := pprofile.NewLocation()
 	loc.SetAddress(2)
