diff --git a/processor/processorhelper/metrics_test.go b/processor/processorhelper/metrics_test.go
index 6d2a0fbb8..10353a28e 100644
--- a/processor/processorhelper/metrics_test.go
+++ b/processor/processorhelper/metrics_test.go
@@ -77,14 +77,8 @@ func TestNewMetricsProcessor_ProcessMetricsErrSkipProcessingData(t *testing.T) {
 	assert.Equal(t, nil, mp.ConsumeMetrics(context.Background(), pdata.NewMetrics()))
 }
 
-type testMProcessor struct {
-	retError error
-}
-
-func newTestMProcessor(retError error) MProcessor {
-	return &testMProcessor{retError: retError}
-}
-
-func (tmp *testMProcessor) ProcessMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
-	return md, tmp.retError
+func newTestMProcessor(retError error) ProcessMetricsFunc {
+	return func(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+		return md, retError
+	}
 }
