diff --git a/config/configoptional/optional.go b/config/configoptional/optional.go
index b6eabafe9..13deba6c2 100644
--- a/config/configoptional/optional.go
+++ b/config/configoptional/optional.go
@@ -10,6 +10,7 @@ import (
 	"strings"
 
 	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/xconfmap"
 )
 
 type flavor int
@@ -189,3 +190,26 @@ func (o Optional[T]) Marshal(conf *confmap.Conf) error {
 
 	return nil
 }
+
+var _ xconfmap.Validator = (*Optional[any])(nil)
+
+// Validate implements [xconfmap.Validator]. This is required because the
+// private fields in [xconfmap.Validator] can't be seen by the reflection used
+// by [xconfmap.Validate], and therefore we have to continue the validation
+// chain manually. This method isn't meant to be called directly, and should
+// generally only be called by [xconfmap.Validate].
+func (o *Optional[T]) Validate() error {
+	// When the flavor is None, the user has not passed this value,
+	// and therefore we should not validate it. The parent struct holding
+	// the Optional type can determine whether a None value is valid for
+	// a given config.
+	//
+	// If the flavor is still Default, then the user has not passed this
+	// value and we should also not validate it.
+	if o.flavor == noneFlavor || o.flavor == defaultFlavor {
+		return nil
+	}
+
+	// For the some flavor, validate the actual value.
+	return xconfmap.Validate(o.value)
+}
