diff --git a/pdata/xpdata/go.mod b/pdata/xpdata/go.mod
index c2d82e6d1..8a20530a9 100644
--- a/pdata/xpdata/go.mod
+++ b/pdata/xpdata/go.mod
@@ -5,10 +5,10 @@ go 1.23.0
 require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/client v1.36.0
-	go.opentelemetry.io/collector/pdata v1.36.1
-	go.opentelemetry.io/collector/pdata/pprofile v0.130.1
-	go.opentelemetry.io/collector/pdata/testdata v0.130.0
+	go.opentelemetry.io/collector/client v1.37.0
+	go.opentelemetry.io/collector/pdata v1.37.0
+	go.opentelemetry.io/collector/pdata/pprofile v0.131.0
+	go.opentelemetry.io/collector/pdata/testdata v0.131.0
 	go.opentelemetry.io/otel/trace v1.36.0
 )
 
