diff --git a/service/internal/graph/graph.go b/service/internal/graph/graph.go
index 3e0c1e5d0..26754e8fc 100644
--- a/service/internal/graph/graph.go
+++ b/service/internal/graph/graph.go
@@ -25,7 +25,6 @@ import (
 	"gonum.org/v1/gonum/graph/topo"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/component/componentprofiles"
 	"go.opentelemetry.io/collector/component/componentstatus"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/connector/connectorprofiles"
@@ -33,6 +32,7 @@ import (
 	"go.opentelemetry.io/collector/consumer/consumerprofiles"
 	"go.opentelemetry.io/collector/internal/fanoutconsumer"
 	"go.opentelemetry.io/collector/pipeline"
+	"go.opentelemetry.io/collector/pipeline/pipelineprofiles"
 	"go.opentelemetry.io/collector/service/internal/builders"
 	"go.opentelemetry.io/collector/service/internal/capabilityconsumer"
 	"go.opentelemetry.io/collector/service/internal/status"
@@ -318,7 +318,7 @@ func (g *Graph) buildComponents(ctx context.Context, set Settings) error {
 				cc := capabilityconsumer.NewLogs(next.(consumer.Logs), capability)
 				n.baseConsumer = cc
 				n.ConsumeLogsFunc = cc.ConsumeLogs
-			case componentprofiles.SignalProfiles:
+			case pipelineprofiles.SignalProfiles:
 				cc := capabilityconsumer.NewProfiles(next.(consumerprofiles.Profiles), capability)
 				n.baseConsumer = cc
 				n.ConsumeProfilesFunc = cc.ConsumeProfiles
@@ -344,7 +344,7 @@ func (g *Graph) buildComponents(ctx context.Context, set Settings) error {
 					consumers = append(consumers, next.(consumer.Logs))
 				}
 				n.baseConsumer = fanoutconsumer.NewLogs(consumers)
-			case componentprofiles.SignalProfiles:
+			case pipelineprofiles.SignalProfiles:
 				consumers := make([]consumerprofiles.Profiles, 0, len(nexts))
 				for _, next := range nexts {
 					consumers = append(consumers, next.(consumerprofiles.Profiles))
@@ -484,7 +484,7 @@ func (g *Graph) GetExporters() map[pipeline.Signal]map[component.ID]component.Co
 	exportersMap[pipeline.SignalTraces] = make(map[component.ID]component.Component)
 	exportersMap[pipeline.SignalMetrics] = make(map[component.ID]component.Component)
 	exportersMap[pipeline.SignalLogs] = make(map[component.ID]component.Component)
-	exportersMap[componentprofiles.SignalProfiles] = make(map[component.ID]component.Component)
+	exportersMap[pipelineprofiles.SignalProfiles] = make(map[component.ID]component.Component)
 
 	for _, pg := range g.pipelines {
 		for _, expNode := range pg.exporters {
@@ -547,7 +547,7 @@ func connectorStability(f connector.Factory, expType, recType pipeline.Signal) c
 			return f.TracesToMetricsStability()
 		case pipeline.SignalLogs:
 			return f.TracesToLogsStability()
-		case componentprofiles.SignalProfiles:
+		case pipelineprofiles.SignalProfiles:
 			fprof, ok := f.(connectorprofiles.Factory)
 			if !ok {
 				return component.StabilityLevelUndefined
@@ -562,7 +562,7 @@ func connectorStability(f connector.Factory, expType, recType pipeline.Signal) c
 			return f.MetricsToMetricsStability()
 		case pipeline.SignalLogs:
 			return f.MetricsToLogsStability()
-		case componentprofiles.SignalProfiles:
+		case pipelineprofiles.SignalProfiles:
 			fprof, ok := f.(connectorprofiles.Factory)
 			if !ok {
 				return component.StabilityLevelUndefined
@@ -577,14 +577,14 @@ func connectorStability(f connector.Factory, expType, recType pipeline.Signal) c
 			return f.LogsToMetricsStability()
 		case pipeline.SignalLogs:
 			return f.LogsToLogsStability()
-		case componentprofiles.SignalProfiles:
+		case pipelineprofiles.SignalProfiles:
 			fprof, ok := f.(connectorprofiles.Factory)
 			if !ok {
 				return component.StabilityLevelUndefined
 			}
 			return fprof.LogsToProfilesStability()
 		}
-	case componentprofiles.SignalProfiles:
+	case pipelineprofiles.SignalProfiles:
 		fprof, ok := f.(connectorprofiles.Factory)
 		if !ok {
 			return component.StabilityLevelUndefined
@@ -596,7 +596,7 @@ func connectorStability(f connector.Factory, expType, recType pipeline.Signal) c
 			return fprof.ProfilesToMetricsStability()
 		case pipeline.SignalLogs:
 			return fprof.ProfilesToLogsStability()
-		case componentprofiles.SignalProfiles:
+		case pipelineprofiles.SignalProfiles:
 			return fprof.ProfilesToProfilesStability()
 		}
 	}
