diff --git a/service/internal/graph/receiver.go b/service/internal/graph/receiver.go
index e462671bf..d5de95ca3 100644
--- a/service/internal/graph/receiver.go
+++ b/service/internal/graph/receiver.go
@@ -8,11 +8,11 @@ import (
 	"fmt"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/component/componentprofiles"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumerprofiles"
 	"go.opentelemetry.io/collector/internal/fanoutconsumer"
 	"go.opentelemetry.io/collector/pipeline"
+	"go.opentelemetry.io/collector/pipeline/pipelineprofiles"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/service/internal/builders"
 	"go.opentelemetry.io/collector/service/internal/components"
@@ -65,7 +65,7 @@ func (n *receiverNode) buildComponent(ctx context.Context,
 			consumers = append(consumers, next.(consumer.Logs))
 		}
 		n.Component, err = builder.CreateLogs(ctx, set, fanoutconsumer.NewLogs(consumers))
-	case componentprofiles.SignalProfiles:
+	case pipelineprofiles.SignalProfiles:
 		var consumers []consumerprofiles.Profiles
 		for _, next := range nexts {
 			consumers = append(consumers, next.(consumerprofiles.Profiles))
