diff --git a/config/confighttp/confighttp.go b/config/confighttp/confighttp.go
index 90b05c1ac..84c7f6425 100644
--- a/config/confighttp/confighttp.go
+++ b/config/confighttp/confighttp.go
@@ -20,6 +20,8 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/rs/cors"
+
 	"go.opentelemetry.io/collector/config/configtls"
 )
 
@@ -55,6 +57,12 @@ type HTTPServerSettings struct {
 
 	// TLSSetting struct exposes TLS client configuration.
 	TLSSetting *configtls.TLSServerSetting `mapstructure:"tls_settings, omitempty"`
+
+	// CorsOrigins are the allowed CORS origins for HTTP/JSON requests to grpc-gateway adapter
+	// for the OTLP receiver. See github.com/rs/cors
+	// An empty list means that CORS is not enabled at all. A wildcard (*) can be
+	// used to match any origin or one or more characters of an origin.
+	CorsOrigins []string `mapstructure:"cors_allowed_origins"`
 }
 
 func (hss *HTTPServerSettings) ToListener() (net.Listener, error) {
@@ -75,6 +83,10 @@ func (hss *HTTPServerSettings) ToListener() (net.Listener, error) {
 }
 
 func (hss *HTTPServerSettings) ToServer(handler http.Handler) *http.Server {
+	if len(hss.CorsOrigins) > 0 {
+		co := cors.Options{AllowedOrigins: hss.CorsOrigins}
+		handler = cors.New(co).Handler(handler)
+	}
 	return &http.Server{
 		Handler: handler,
 	}
