diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index bac81994c..4086b4fb4 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -29,7 +29,6 @@ extension/experimental/storage  @open-telemetry/collector-approvers @swiatekm
 
 # Profiling-related modules
 pdata/pprofile                          @open-telemetry/collector-approvers @mx-psi @dmathieu
-consumer/consumerprofiles               @open-telemetry/collector-approvers @mx-psi @dmathieu
 consumer/xconsumer                      @open-telemetry/collector-approvers @mx-psi @dmathieu
 connector/xconnector                    @open-telemetry/collector-approvers @mx-psi @dmathieu
 exporter/exporterhelper/xexporterhelper @open-telemetry/collector-approvers @mx-psi @dmathieu
