diff --git a/model/pdata/generated_log.go b/model/pdata/generated_log.go
index 110c9f88d..ee46494ec 100644
--- a/model/pdata/generated_log.go
+++ b/model/pdata/generated_log.go
@@ -183,6 +183,13 @@ func NewResourceLogs() ResourceLogs {
 	return newResourceLogs(&otlplogs.ResourceLogs{})
 }
 
+// MoveTo moves all properties from the current struct to dest
+// reseting the current instance to its zero value
+func (ms ResourceLogs) MoveTo(dest ResourceLogs) {
+	*dest.orig = *ms.orig
+	*ms.orig = otlplogs.ResourceLogs{}
+}
+
 // Resource returns the resource associated with this ResourceLogs.
 func (ms ResourceLogs) Resource() Resource {
 	return newResource(&(*ms.orig).Resource)
@@ -370,6 +377,13 @@ func NewInstrumentationLibraryLogs() InstrumentationLibraryLogs {
 	return newInstrumentationLibraryLogs(&otlplogs.InstrumentationLibraryLogs{})
 }
 
+// MoveTo moves all properties from the current struct to dest
+// reseting the current instance to its zero value
+func (ms InstrumentationLibraryLogs) MoveTo(dest InstrumentationLibraryLogs) {
+	*dest.orig = *ms.orig
+	*ms.orig = otlplogs.InstrumentationLibraryLogs{}
+}
+
 // InstrumentationLibrary returns the instrumentationlibrary associated with this InstrumentationLibraryLogs.
 func (ms InstrumentationLibraryLogs) InstrumentationLibrary() InstrumentationLibrary {
 	return newInstrumentationLibrary(&(*ms.orig).InstrumentationLibrary)
@@ -558,6 +572,13 @@ func NewLogRecord() LogRecord {
 	return newLogRecord(&otlplogs.LogRecord{})
 }
 
+// MoveTo moves all properties from the current struct to dest
+// reseting the current instance to its zero value
+func (ms LogRecord) MoveTo(dest LogRecord) {
+	*dest.orig = *ms.orig
+	*ms.orig = otlplogs.LogRecord{}
+}
+
 // Timestamp returns the timestamp associated with this LogRecord.
 func (ms LogRecord) Timestamp() Timestamp {
 	return Timestamp((*ms.orig).TimeUnixNano)
