diff --git a/service/service_test.go b/service/service_test.go
index a332dbf6c..f705d1593 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -23,7 +23,6 @@ import (
 	"go.uber.org/zap/zaptest/observer"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/component/componentstatus"
 	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/config/confignet"
 	"go.opentelemetry.io/collector/config/configtelemetry"
@@ -46,81 +45,7 @@ const (
 	otelCommand = "otelcoltest"
 )
 
-var (
-	nopType   = component.MustNewType("nop")
-	wrongType = component.MustNewType("wrong")
-)
-
-func TestServiceGetFactory(t *testing.T) {
-	set := newNopSettings()
-	srv, err := New(context.Background(), set, newNopConfig())
-	require.NoError(t, err)
-
-	assert.NoError(t, srv.Start(context.Background()))
-	t.Cleanup(func() {
-		assert.NoError(t, srv.Shutdown(context.Background()))
-	})
-
-	assert.Nil(t, srv.host.GetFactory(component.KindReceiver, wrongType))
-	assert.Equal(t, srv.host.Receivers.Factory(nopType), srv.host.GetFactory(component.KindReceiver, nopType))
-
-	assert.Nil(t, srv.host.GetFactory(component.KindProcessor, wrongType))
-	assert.Equal(t, srv.host.Processors.Factory(nopType), srv.host.GetFactory(component.KindProcessor, nopType))
-
-	assert.Nil(t, srv.host.GetFactory(component.KindExporter, wrongType))
-	assert.Equal(t, srv.host.Exporters.Factory(nopType), srv.host.GetFactory(component.KindExporter, nopType))
-
-	assert.Nil(t, srv.host.GetFactory(component.KindConnector, wrongType))
-	assert.Equal(t, srv.host.Connectors.Factory(nopType), srv.host.GetFactory(component.KindConnector, nopType))
-
-	assert.Nil(t, srv.host.GetFactory(component.KindExtension, wrongType))
-	assert.Equal(t, srv.host.Extensions.Factory(nopType), srv.host.GetFactory(component.KindExtension, nopType))
-
-	// Try retrieve non existing component.Kind.
-	assert.Nil(t, srv.host.GetFactory(component.Kind{}, nopType))
-}
-
-func TestServiceGetExtensions(t *testing.T) {
-	srv, err := New(context.Background(), newNopSettings(), newNopConfig())
-	require.NoError(t, err)
-
-	assert.NoError(t, srv.Start(context.Background()))
-	t.Cleanup(func() {
-		assert.NoError(t, srv.Shutdown(context.Background()))
-	})
-
-	extMap := srv.host.GetExtensions()
-
-	assert.Len(t, extMap, 1)
-	assert.Contains(t, extMap, component.NewID(nopType))
-}
-
-func TestServiceGetExporters(t *testing.T) {
-	srv, err := New(context.Background(), newNopSettings(), newNopConfig())
-	require.NoError(t, err)
-
-	assert.NoError(t, srv.Start(context.Background()))
-	t.Cleanup(func() {
-		assert.NoError(t, srv.Shutdown(context.Background()))
-	})
-
-	//nolint:staticcheck
-	expMap := srv.host.GetExporters()
-
-	v, ok := expMap[pipeline.SignalTraces]
-	assert.True(t, ok)
-	assert.NotNil(t, v)
-
-	assert.Len(t, expMap, 4)
-	assert.Len(t, expMap[pipeline.SignalTraces], 1)
-	assert.Contains(t, expMap[pipeline.SignalTraces], component.NewID(nopType))
-	assert.Len(t, expMap[pipeline.SignalMetrics], 1)
-	assert.Contains(t, expMap[pipeline.SignalMetrics], component.NewID(nopType))
-	assert.Len(t, expMap[pipeline.SignalLogs], 1)
-	assert.Contains(t, expMap[pipeline.SignalLogs], component.NewID(nopType))
-	assert.Len(t, expMap[xpipeline.SignalProfiles], 1)
-	assert.Contains(t, expMap[xpipeline.SignalProfiles], component.NewID(nopType))
-}
+var nopType = component.MustNewType("nop")
 
 // TestServiceTelemetryCleanupOnError tests that if newService errors due to an invalid config telemetry is cleaned up
 // and another service with a valid config can be started right after.
@@ -494,28 +419,6 @@ func TestServiceTelemetryLogger(t *testing.T) {
 	assert.NotNil(t, srv.telemetrySettings.Logger)
 }
 
-func TestServiceFatalError(t *testing.T) {
-	set := newNopSettings()
-	set.AsyncErrorChannel = make(chan error)
-
-	srv, err := New(context.Background(), set, newNopConfig())
-	require.NoError(t, err)
-
-	assert.NoError(t, srv.Start(context.Background()))
-	t.Cleanup(func() {
-		assert.NoError(t, srv.Shutdown(context.Background()))
-	})
-
-	go func() {
-		ev := componentstatus.NewFatalErrorEvent(assert.AnError)
-		srv.host.NotifyComponentStatusChange(&componentstatus.InstanceID{}, ev)
-	}()
-
-	err = <-srv.host.AsyncErrorChannel
-
-	require.ErrorIs(t, err, assert.AnError)
-}
-
 func TestServiceTelemetryCreateProvidersError(t *testing.T) {
 	loggerOpt := telemetry.WithCreateLogger(
 		func(context.Context, telemetry.LoggerSettings, component.Config) (*zap.Logger, component.ShutdownFunc, error) {
