diff --git a/receiver/opencensusreceiver/opencensus_test.go b/receiver/opencensusreceiver/opencensus_test.go
index aba656f7a..2fd4fd85f 100644
--- a/receiver/opencensusreceiver/opencensus_test.go
+++ b/receiver/opencensusreceiver/opencensus_test.go
@@ -63,7 +63,7 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 
 	err = ocr.Start(context.Background(), componenttest.NewNopHost())
 	require.NoError(t, err, "Failed to start trace receiver: %v", err)
-	defer ocr.Shutdown(context.Background())
+	t.Cleanup(func() { require.NoError(t, ocr.Shutdown(context.Background())) })
 
 	// TODO(songy23): make starting server deterministic
 	// Wait for the servers to start
@@ -157,7 +157,7 @@ func TestTraceGrpcGatewayCors_endToEnd(t *testing.T) {
 
 	ocr, err := newOpenCensusReceiver(ocReceiverName, "tcp", addr, consumertest.NewNop(), nil, withCorsOrigins(corsOrigins))
 	require.NoError(t, err, "Failed to create trace receiver: %v", err)
-	defer ocr.Shutdown(context.Background())
+	t.Cleanup(func() { require.NoError(t, ocr.Shutdown(context.Background())) })
 
 	err = ocr.Start(context.Background(), componenttest.NewNopHost())
 	require.NoError(t, err, "Failed to start trace receiver: %v", err)
@@ -181,7 +181,7 @@ func TestMetricsGrpcGatewayCors_endToEnd(t *testing.T) {
 
 	ocr, err := newOpenCensusReceiver(ocReceiverName, "tcp", addr, nil, consumertest.NewNop(), withCorsOrigins(corsOrigins))
 	require.NoError(t, err, "Failed to create metrics receiver: %v", err)
-	defer ocr.Shutdown(context.Background())
+	t.Cleanup(func() { require.NoError(t, ocr.Shutdown(context.Background())) })
 
 	err = ocr.Start(context.Background(), componenttest.NewNopHost())
 	require.NoError(t, err, "Failed to start metrics receiver: %v", err)
@@ -241,7 +241,7 @@ func TestStopWithoutStartNeverCrashes(t *testing.T) {
 	ocr, err := newOpenCensusReceiver(ocReceiverName, "tcp", addr, nil, nil)
 	require.NoError(t, err, "Failed to create an OpenCensus receiver: %v", err)
 	// Stop it before ever invoking Start*.
-	ocr.stop()
+	require.NoError(t, ocr.stop())
 }
 
 func TestNewPortAlreadyUsed(t *testing.T) {
@@ -290,7 +290,7 @@ func TestReceiveOnUnixDomainSocket_endToEnd(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, r)
 	require.NoError(t, r.Start(context.Background(), componenttest.NewNopHost()))
-	defer r.Shutdown(context.Background())
+	t.Cleanup(func() { require.NoError(t, r.Shutdown(context.Background())) })
 
 	// Wait for the servers to start
 	<-time.After(10 * time.Millisecond)
@@ -393,7 +393,7 @@ func TestOCReceiverTrace_HandleNextConsumerResponse(t *testing.T) {
 		require.NotNil(t, stream)
 
 		err = stream.Send(msg)
-		stream.CloseSend()
+		require.NoError(t, stream.CloseSend())
 		if err == nil {
 			for {
 				if _, err = stream.Recv(); err != nil {
@@ -436,7 +436,7 @@ func TestOCReceiverTrace_HandleNextConsumerResponse(t *testing.T) {
 
 				ocr.traceConsumer = sink
 				require.NoError(t, ocr.Start(context.Background(), componenttest.NewNopHost()))
-				defer ocr.Shutdown(context.Background())
+				t.Cleanup(func() { require.NoError(t, ocr.Shutdown(context.Background())) })
 
 				cc, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock())
 				if err != nil {
@@ -542,7 +542,7 @@ func TestOCReceiverMetrics_HandleNextConsumerResponse(t *testing.T) {
 		require.NotNil(t, stream)
 
 		err = stream.Send(msg)
-		stream.CloseSend()
+		require.NoError(t, stream.CloseSend())
 		if err == nil {
 			for {
 				if _, err = stream.Recv(); err != nil {
@@ -585,7 +585,7 @@ func TestOCReceiverMetrics_HandleNextConsumerResponse(t *testing.T) {
 
 				ocr.metricsConsumer = sink
 				require.Nil(t, ocr.Start(context.Background(), componenttest.NewNopHost()))
-				defer ocr.Shutdown(context.Background())
+				t.Cleanup(func() { require.NoError(t, ocr.Shutdown(context.Background())) })
 
 				cc, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock())
 				if err != nil {
