diff --git a/otelcol/command_components_test.go b/otelcol/command_components_test.go
index 8609f3ae0..5b429dd3f 100644
--- a/otelcol/command_components_test.go
+++ b/otelcol/command_components_test.go
@@ -21,6 +21,9 @@ func TestNewBuildSubCommand(t *testing.T) {
 		BuildInfo:              component.NewDefaultBuildInfo(),
 		Factories:              nopFactories,
 		ConfigProviderSettings: newDefaultConfigProviderSettings(t, []string{filepath.Join("testdata", "otelcol-nop.yaml")}),
+		ProviderModules: map[string]string{
+			"nop": "go.opentelemetry.io/collector/confmap/provider/testprovider v1.2.3",
+		},
 	}
 	cmd := NewCommand(set)
 	cmd.SetArgs([]string{"components"})
