diff --git a/exporter/opencensusexporter/config.go b/exporter/opencensusexporter/config.go
index eb64b08eb..a7f63098d 100644
--- a/exporter/opencensusexporter/config.go
+++ b/exporter/opencensusexporter/config.go
@@ -17,18 +17,18 @@ package opencensusexporter
 import (
 	"time"
 
-	"github.com/open-telemetry/opentelemetry-service/models"
+	"github.com/open-telemetry/opentelemetry-service/configv2/configmodels"
 )
 
 // ConfigV2 defines configuration for OpenCensus exporter.
 type ConfigV2 struct {
-	models.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
-	Endpoint                string                   `mapstructure:"endpoint"`
-	Compression             string                   `mapstructure:"compression"`
-	Headers                 map[string]string        `mapstructure:"headers"`
-	NumWorkers              int                      `mapstructure:"num-workers"`
-	CertPemFile             string                   `mapstructure:"cert-pem-file"`
-	UseSecure               bool                     `mapstructure:"secure,omitempty"`
-	ReconnectionDelay       time.Duration            `mapstructure:"reconnection-delay,omitempty"`
-	KeepaliveParameters     *keepaliveConfig         `mapstructure:"keepalive,omitempty"`
+	configmodels.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	Endpoint                      string                   `mapstructure:"endpoint"`
+	Compression                   string                   `mapstructure:"compression"`
+	Headers                       map[string]string        `mapstructure:"headers"`
+	NumWorkers                    int                      `mapstructure:"num-workers"`
+	CertPemFile                   string                   `mapstructure:"cert-pem-file"`
+	UseSecure                     bool                     `mapstructure:"secure,omitempty"`
+	ReconnectionDelay             time.Duration            `mapstructure:"reconnection-delay,omitempty"`
+	KeepaliveParameters           *keepaliveConfig         `mapstructure:"keepalive,omitempty"`
 }
