diff --git a/consumer/pdata/metric_test.go b/consumer/pdata/metric_test.go
index c2223f486..5d414ca1b 100644
--- a/consumer/pdata/metric_test.go
+++ b/consumer/pdata/metric_test.go
@@ -25,6 +25,72 @@ import (
 	otlpmetrics "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/metrics/v1"
 )
 
+func TestCopyData(t *testing.T) {
+	tests := []struct {
+		name string
+		src  *otlpmetrics.Metric
+	}{
+		{
+			name: "IntGauge",
+			src: &otlpmetrics.Metric{
+				Data: &otlpmetrics.Metric_IntGauge{
+					IntGauge: &otlpmetrics.IntGauge{},
+				},
+			},
+		},
+		{
+			name: "DoubleGauge",
+			src: &otlpmetrics.Metric{
+				Data: &otlpmetrics.Metric_DoubleGauge{
+					DoubleGauge: &otlpmetrics.DoubleGauge{},
+				},
+			},
+		},
+		{
+			name: "IntSum",
+			src: &otlpmetrics.Metric{
+				Data: &otlpmetrics.Metric_IntSum{
+					IntSum: &otlpmetrics.IntSum{},
+				},
+			},
+		},
+		{
+			name: "DoubleSum",
+			src: &otlpmetrics.Metric{
+				Data: &otlpmetrics.Metric_DoubleSum{
+					DoubleSum: &otlpmetrics.DoubleSum{},
+				},
+			},
+		},
+		{
+			name: "IntHistogram",
+			src: &otlpmetrics.Metric{
+				Data: &otlpmetrics.Metric_IntHistogram{
+					IntHistogram: &otlpmetrics.IntHistogram{},
+				},
+			},
+		},
+		{
+			name: "DoubleHistogram",
+			src: &otlpmetrics.Metric{
+				Data: &otlpmetrics.Metric_DoubleHistogram{
+					DoubleHistogram: &otlpmetrics.DoubleHistogram{},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dest := &otlpmetrics.Metric{}
+			assert.Nil(t, dest.Data)
+			assert.NotNil(t, test.src.Data)
+			copyData(test.src, dest)
+			assert.EqualValues(t, test.src, dest)
+		})
+	}
+}
+
 func TestResourceMetricsWireCompatibility(t *testing.T) {
 	// This test verifies that OTLP ProtoBufs generated using goproto lib in
 	// opentelemetry-proto repository OTLP ProtoBufs generated using gogoproto lib in
