diff --git a/receiver/scrapererror/doc.go b/receiver/scrapererror/doc.go
index a66d4a30b..75970a9ce 100644
--- a/receiver/scrapererror/doc.go
+++ b/receiver/scrapererror/doc.go
@@ -12,5 +12,5 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package scraperror provides custom error types for scrapers.
+// Package scrapererror provides custom error types for scrapers.
 package scrapererror // import "go.opentelemetry.io/collector/receiver/scrapererror"
