diff --git a/pdata/pcommon/slice_test.go b/pdata/pcommon/slice_test.go
index 209d4b77c..dd3540958 100644
--- a/pdata/pcommon/slice_test.go
+++ b/pdata/pcommon/slice_test.go
@@ -48,7 +48,7 @@ func TestSliceReadOnly(t *testing.T) {
 	assert.Panics(t, func() { es.MoveAndAppendTo(es2) })
 	assert.Panics(t, func() { es2.MoveAndAppendTo(es) })
 
-	assert.Panics(t, func() { es.RemoveIf(func(el Value) bool { return false }) })
+	assert.Panics(t, func() { es.RemoveIf(func(_ Value) bool { return false }) })
 
 	assert.Equal(t, []any{int64(3)}, es.AsRaw())
 	assert.Panics(t, func() { _ = es.FromRaw([]any{3}) })
@@ -122,7 +122,7 @@ func TestSlice_MoveAndAppendTo(t *testing.T) {
 func TestSlice_RemoveIf(t *testing.T) {
 	// Test RemoveIf on empty slice
 	emptySlice := NewSlice()
-	emptySlice.RemoveIf(func(el Value) bool {
+	emptySlice.RemoveIf(func(_ Value) bool {
 		t.Fail()
 		return false
 	})
@@ -130,7 +130,7 @@ func TestSlice_RemoveIf(t *testing.T) {
 	// Test RemoveIf
 	filtered := Slice(internal.GenerateTestSlice())
 	pos := 0
-	filtered.RemoveIf(func(el Value) bool {
+	filtered.RemoveIf(func(_ Value) bool {
 		pos++
 		return pos%3 == 0
 	})
