diff --git a/pdata/xpdata/entity/generated_entityrefslice.go b/pdata/xpdata/entity/generated_entityrefslice.go
index ea5b09aec..705dadb05 100644
--- a/pdata/xpdata/entity/generated_entityrefslice.go
+++ b/pdata/xpdata/entity/generated_entityrefslice.go
@@ -96,7 +96,7 @@ func (es EntityRefSlice) EnsureCapacity(newCap int) {
 // It returns the newly added EntityRef.
 func (es EntityRefSlice) AppendEmpty() EntityRef {
 	es.getState().AssertMutable()
-	*es.getOrig() = append(*es.getOrig(), &otlpcommon.EntityRef{})
+	*es.getOrig() = append(*es.getOrig(), internal.NewOrigPtrEntityRef())
 	return es.At(es.Len() - 1)
 }
 
