diff --git a/pdata/pmetric/pmetricotlp/fuzz_test.go b/pdata/pmetric/pmetricotlp/fuzz_test.go
index 137b98b61..1e5e9381d 100644
--- a/pdata/pmetric/pmetricotlp/fuzz_test.go
+++ b/pdata/pmetric/pmetricotlp/fuzz_test.go
@@ -10,31 +10,27 @@ import (
 func FuzzRequestUnmarshalJSON(f *testing.F) {
 	f.Fuzz(func(_ *testing.T, data []byte) {
 		er := NewExportRequest()
-		//nolint: errcheck
-		er.UnmarshalJSON(data)
+		_ = er.UnmarshalJSON(data)
 	})
 }
 
 func FuzzResponseUnmarshalJSON(f *testing.F) {
 	f.Fuzz(func(_ *testing.T, data []byte) {
 		er := NewExportResponse()
-		//nolint: errcheck
-		er.UnmarshalJSON(data)
+		_ = er.UnmarshalJSON(data)
 	})
 }
 
 func FuzzRequestUnmarshalProto(f *testing.F) {
 	f.Fuzz(func(_ *testing.T, data []byte) {
 		er := NewExportRequest()
-		//nolint: errcheck
-		er.UnmarshalProto(data)
+		_ = er.UnmarshalJSON(data)
 	})
 }
 
 func FuzzResponseUnmarshalProto(f *testing.F) {
 	f.Fuzz(func(_ *testing.T, data []byte) {
 		er := NewExportResponse()
-		//nolint: errcheck
-		er.UnmarshalProto(data)
+		_ = er.UnmarshalJSON(data)
 	})
 }
