diff --git a/component/componenttest/nop_receiver.go b/component/componenttest/nop_receiver.go
index 828a5a6dc..a44452940 100644
--- a/component/componenttest/nop_receiver.go
+++ b/component/componenttest/nop_receiver.go
@@ -52,7 +52,7 @@ func (f *nopReceiverFactory) CreateDefaultConfig() config.Receiver {
 // CreateTracesReceiver implements component.ReceiverFactory interface.
 func (f *nopReceiverFactory) CreateTracesReceiver(
 	_ context.Context,
-	_ component.ReceiverCreateParams,
+	_ component.ReceiverCreateSettings,
 	_ config.Receiver,
 	_ consumer.Traces,
 ) (component.TracesReceiver, error) {
@@ -62,7 +62,7 @@ func (f *nopReceiverFactory) CreateTracesReceiver(
 // CreateMetricsReceiver implements component.ReceiverFactory interface.
 func (f *nopReceiverFactory) CreateMetricsReceiver(
 	_ context.Context,
-	_ component.ReceiverCreateParams,
+	_ component.ReceiverCreateSettings,
 	_ config.Receiver,
 	_ consumer.Metrics,
 ) (component.MetricsReceiver, error) {
@@ -72,7 +72,7 @@ func (f *nopReceiverFactory) CreateMetricsReceiver(
 // CreateLogsReceiver implements component.ReceiverFactory interface.
 func (f *nopReceiverFactory) CreateLogsReceiver(
 	_ context.Context,
-	_ component.ReceiverCreateParams,
+	_ component.ReceiverCreateSettings,
 	_ config.Receiver,
 	_ consumer.Logs,
 ) (component.LogsReceiver, error) {
