diff --git a/exporter/otlpexporter/go.mod b/exporter/otlpexporter/go.mod
index 21ce184f4..3194e0340 100644
--- a/exporter/otlpexporter/go.mod
+++ b/exporter/otlpexporter/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/stretchr/testify v1.8.1
 	go.opentelemetry.io/collector v0.66.0
 	go.opentelemetry.io/collector/component v0.66.0
-	go.opentelemetry.io/collector/confmap v0.66.0
+	go.opentelemetry.io/collector/confmap v0.0.0-20221201172708-2bdff61fa52a
 	go.opentelemetry.io/collector/consumer v0.66.0
 	go.opentelemetry.io/collector/pdata v0.66.0
 	go.uber.org/atomic v1.10.0
