diff --git a/receiver/scrapererror/doc.go b/receiver/scrapererror/doc.go
deleted file mode 100644
index c893f0f8e..000000000
--- a/receiver/scrapererror/doc.go
+++ /dev/null
@@ -1,5 +0,0 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
-
-// Package scrapererror provides custom error types for scrapers.
-package scrapererror // import "go.opentelemetry.io/collector/receiver/scrapererror"
