diff --git a/receiver/receivertest/go.mod b/receiver/receivertest/go.mod
index b5b2b9f36..35da8cdff 100644
--- a/receiver/receivertest/go.mod
+++ b/receiver/receivertest/go.mod
@@ -4,7 +4,7 @@ go 1.22.0
 
 require (
 	github.com/google/uuid v1.6.0
-	github.com/stretchr/testify v1.9.0
+	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v0.114.0
 	go.opentelemetry.io/collector/component/componenttest v0.114.0
 	go.opentelemetry.io/collector/consumer v0.114.0
