diff --git a/cmd/builder/internal/builder/main.go b/cmd/builder/internal/builder/main.go
index fbbb3a094..3c62bd3ff 100644
--- a/cmd/builder/internal/builder/main.go
+++ b/cmd/builder/internal/builder/main.go
@@ -139,7 +139,7 @@ func GetModules(cfg Config) error {
 		return nil
 	}
 
-	if _, err := runGoCommand(cfg, "mod", "tidy", "-compat=1.21"); err != nil {
+	if _, err := runGoCommand(cfg, "mod", "tidy", "-compat=1.22"); err != nil {
 		return fmt.Errorf("failed to update go.mod: %w", err)
 	}
 
