diff --git a/exporter/debugexporter/doc.go b/exporter/debugexporter/doc.go
new file mode 100644
index 000000000..9edb9ae09
--- /dev/null
+++ b/exporter/debugexporter/doc.go
@@ -0,0 +1,5 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package debugexporter exports data to console as logs.
+package debugexporter // import "go.opentelemetry.io/collector/exporter/debugexporter"
