diff --git a/service/defaultcomponents/defaults.go b/service/defaultcomponents/defaults.go
index bb503c05e..f542b9d23 100644
--- a/service/defaultcomponents/defaults.go
+++ b/service/defaultcomponents/defaults.go
@@ -29,8 +29,7 @@ import (
 	"go.opentelemetry.io/collector/receiver/otlpreceiver"
 )
 
-// Components returns the default set of components used by the
-// OpenTelemetry collector.
+// Deprecated: Will be removed soon.
 func Components() (
 	component.Factories,
 	error,
