diff --git a/receiver/kafkareceiver/config_test.go b/receiver/kafkareceiver/config_test.go
index ca9ef1ac5..4d09dca56 100644
--- a/receiver/kafkareceiver/config_test.go
+++ b/receiver/kafkareceiver/config_test.go
@@ -23,7 +23,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/component/componenttest"
-	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configtest"
 	"go.opentelemetry.io/collector/config/configtls"
 	"go.opentelemetry.io/collector/exporter/kafkaexporter"
@@ -41,7 +41,7 @@ func TestLoadConfig(t *testing.T) {
 
 	r := cfg.Receivers[typeStr].(*Config)
 	assert.Equal(t, &Config{
-		ReceiverSettings: configmodels.ReceiverSettings{
+		ReceiverSettings: config.ReceiverSettings{
 			NameVal: typeStr,
 			TypeVal: typeStr,
 		},
