diff --git a/exporter/exporterhelper/metrics.go b/exporter/exporterhelper/metrics.go
index 4a9811e94..3dd7de63b 100644
--- a/exporter/exporterhelper/metrics.go
+++ b/exporter/exporterhelper/metrics.go
@@ -126,7 +126,7 @@ func NewMetricsRequest(
 		return nil, errNilMetricsConverter
 	}
 
-	be, err := internal.NewBaseExporter(set, pipeline.SignalMetrics, newMetricsSenderWithObservability, options...)
+	be, err := internal.NewBaseExporter(set, pipeline.SignalMetrics, internal.NewObsReportSender, options...)
 	if err != nil {
 		return nil, err
 	}
@@ -141,7 +141,7 @@ func NewMetricsRequest(
 		}
 		sErr := be.Send(ctx, req)
 		if errors.Is(sErr, exporterqueue.ErrQueueIsFull) {
-			be.Obsrep.RecordEnqueueFailure(ctx, pipeline.SignalMetrics, int64(req.ItemsCount()))
+			be.Obsrep.RecordEnqueueFailure(ctx, int64(req.ItemsCount()))
 		}
 		return sErr
 	}, be.ConsumerOptions...)
@@ -151,20 +151,3 @@ func NewMetricsRequest(
 		Metrics:      mc,
 	}, err
 }
-
-type metricsSenderWithObservability struct {
-	internal.BaseSender[Request]
-	obsrep *internal.ObsReport
-}
-
-func newMetricsSenderWithObservability(obsrep *internal.ObsReport) internal.Sender[Request] {
-	return &metricsSenderWithObservability{obsrep: obsrep}
-}
-
-func (mewo *metricsSenderWithObservability) Send(ctx context.Context, req Request) error {
-	c := mewo.obsrep.StartMetricsOp(ctx)
-	numMetricDataPoints := req.ItemsCount()
-	err := mewo.NextSender.Send(c, req)
-	mewo.obsrep.EndMetricsOp(c, numMetricDataPoints, err)
-	return err
-}
