diff --git a/receiver/otlpreceiver/internal/logs/otlp.go b/receiver/otlpreceiver/internal/logs/otlp.go
index 8bdca9ec0..898a6ae9f 100644
--- a/receiver/otlpreceiver/internal/logs/otlp.go
+++ b/receiver/otlpreceiver/internal/logs/otlp.go
@@ -20,8 +20,8 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer"
-	"go.opentelemetry.io/collector/model/otlpgrpc"
 	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
 )
 
 const (
@@ -48,16 +48,16 @@ func New(id config.ComponentID, nextConsumer consumer.Logs, set component.Receiv
 }
 
 // Export implements the service Export logs func.
-func (r *Receiver) Export(ctx context.Context, req otlpgrpc.LogsRequest) (otlpgrpc.LogsResponse, error) {
+func (r *Receiver) Export(ctx context.Context, req plogotlp.Request) (plogotlp.Response, error) {
 	ld := req.Logs()
 	numSpans := ld.LogRecordCount()
 	if numSpans == 0 {
-		return otlpgrpc.NewLogsResponse(), nil
+		return plogotlp.NewResponse(), nil
 	}
 
 	ctx = r.obsrecv.StartLogsOp(ctx)
 	err := r.nextConsumer.ConsumeLogs(ctx, ld)
 	r.obsrecv.EndLogsOp(ctx, dataFormatProtobuf, numSpans, err)
 
-	return otlpgrpc.NewLogsResponse(), err
+	return plogotlp.NewResponse(), err
 }
