diff --git a/internal/telemetry/componentattribute/meter_provider.go b/internal/telemetry/componentattribute/meter_provider.go
index 1126cc3cc..d17732dde 100644
--- a/internal/telemetry/componentattribute/meter_provider.go
+++ b/internal/telemetry/componentattribute/meter_provider.go
@@ -15,7 +15,7 @@ type meterProviderWithAttributes struct {
 	attrs []attribute.KeyValue
 }
 
-// Creates a MeterProvider with a new set of injected instrumentation scope attributes.
+// MeterProviderWithAttributes creates a MeterProvider with a new set of injected instrumentation scope attributes.
 func MeterProviderWithAttributes(mp metric.MeterProvider, attrs attribute.Set) metric.MeterProvider {
 	if mpwa, ok := mp.(meterProviderWithAttributes); ok {
 		mp = mpwa.MeterProvider
