diff --git a/pdata/plog/plogotlp/response.go b/pdata/plog/plogotlp/response.go
index 3aa9a3856..293d2d2cf 100644
--- a/pdata/plog/plogotlp/response.go
+++ b/pdata/plog/plogotlp/response.go
@@ -73,7 +73,7 @@ func (ms ExportResponse) unmarshalJsoniter(iter *jsoniter.Iterator) {
 }
 
 func (ms ExportPartialSuccess) unmarshalJsoniter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iterator *jsoniter.Iterator, f string) bool {
+	iter.ReadObjectCB(func(_ *jsoniter.Iterator, f string) bool {
 		switch f {
 		case "rejected_log_records", "rejectedLogRecords":
 			ms.orig.RejectedLogRecords = json.ReadInt64(iter)
