diff --git a/receiver/opencensusreceiver/config_test.go b/receiver/opencensusreceiver/config_test.go
index be893cb1f..a6829dfee 100644
--- a/receiver/opencensusreceiver/config_test.go
+++ b/receiver/opencensusreceiver/config_test.go
@@ -52,7 +52,8 @@ func TestLoadConfig(t *testing.T) {
 				NameVal: "opencensus/customname",
 			},
 			GRPCServerSettings: configgrpc.GRPCServerSettings{
-				Endpoint: "0.0.0.0:9090",
+				Endpoint:       "0.0.0.0:9090",
+				ReadBufferSize: 512 * 1024,
 			},
 			Transport: "tcp",
 		})
@@ -65,7 +66,8 @@ func TestLoadConfig(t *testing.T) {
 				NameVal: "opencensus/keepalive",
 			},
 			GRPCServerSettings: configgrpc.GRPCServerSettings{
-				Endpoint: "0.0.0.0:55678",
+				Endpoint:       "0.0.0.0:55678",
+				ReadBufferSize: 512 * 1024,
 				Keepalive: &configgrpc.KeepaliveServerConfig{
 					ServerParameters: &configgrpc.KeepaliveServerParameters{
 						MaxConnectionIdle:     11 * time.Second,
@@ -94,6 +96,8 @@ func TestLoadConfig(t *testing.T) {
 				Endpoint:             "0.0.0.0:55678",
 				MaxRecvMsgSizeMiB:    32,
 				MaxConcurrentStreams: 16,
+				ReadBufferSize:       1024,
+				WriteBufferSize:      1024,
 				Keepalive: &configgrpc.KeepaliveServerConfig{
 					ServerParameters: &configgrpc.KeepaliveServerParameters{
 						MaxConnectionIdle: 10 * time.Second,
@@ -113,7 +117,8 @@ func TestLoadConfig(t *testing.T) {
 				NameVal: "opencensus/tlscredentials",
 			},
 			GRPCServerSettings: configgrpc.GRPCServerSettings{
-				Endpoint: "0.0.0.0:55678",
+				Endpoint:       "0.0.0.0:55678",
+				ReadBufferSize: 512 * 1024,
 				TLSCredentials: &configtls.TLSServerSetting{
 					TLSSetting: configtls.TLSSetting{
 						CertFile: "test.crt",
@@ -132,7 +137,8 @@ func TestLoadConfig(t *testing.T) {
 				NameVal: "opencensus/cors",
 			},
 			GRPCServerSettings: configgrpc.GRPCServerSettings{
-				Endpoint: "0.0.0.0:55678",
+				Endpoint:       "0.0.0.0:55678",
+				ReadBufferSize: 512 * 1024,
 			},
 			Transport:   "tcp",
 			CorsOrigins: []string{"https://*.test.com", "https://test.com"},
@@ -146,7 +152,8 @@ func TestLoadConfig(t *testing.T) {
 				NameVal: "opencensus/uds",
 			},
 			GRPCServerSettings: configgrpc.GRPCServerSettings{
-				Endpoint: "/tmp/opencensus.sock",
+				Endpoint:       "/tmp/opencensus.sock",
+				ReadBufferSize: 512 * 1024,
 			},
 			Transport: "unix",
 		})
