diff --git a/cmd/ocagent/config.go b/cmd/ocagent/config.go
index 17e3c6462..41d5b7a9b 100644
--- a/cmd/ocagent/config.go
+++ b/cmd/ocagent/config.go
@@ -113,6 +113,7 @@ func exportersFromYAMLConfig(config []byte) (traceExporters []exporter.TraceExpo
 		{name: "datadog", fn: exporterparser.DatadogTraceExportersFromYAML},
 		{name: "stackdriver", fn: exporterparser.StackdriverTraceExportersFromYAML},
 		{name: "zipkin", fn: exporterparser.ZipkinExportersFromYAML},
+		{name: "jaeger", fn: exporterparser.JaegerExportersFromYAML},
 	}
 
 	for _, cfg := range parseFns {
