diff --git a/receiver/jaegerreceiver/trace_receiver.go b/receiver/jaegerreceiver/trace_receiver.go
index 5b87366c8..f10e8a708 100644
--- a/receiver/jaegerreceiver/trace_receiver.go
+++ b/receiver/jaegerreceiver/trace_receiver.go
@@ -120,13 +120,13 @@ func newJaegerReceiver(
 	config *configuration,
 	nextConsumer consumer.TracesConsumer,
 	params component.ReceiverCreateParams,
-) (*jReceiver, error) {
+) *jReceiver {
 	return &jReceiver{
 		config:       config,
 		nextConsumer: nextConsumer,
 		instanceName: instanceName,
 		logger:       params.Logger,
-	}, nil
+	}
 }
 
 func (jr *jReceiver) agentCompactThriftAddr() string {
