diff --git a/pdata/pcommon/trace_state.go b/pdata/pcommon/trace_state.go
index 440b4b1ad..dd163629c 100644
--- a/pdata/pcommon/trace_state.go
+++ b/pdata/pcommon/trace_state.go
@@ -11,13 +11,18 @@ import (
 type TraceState internal.TraceState
 
 func NewTraceState() TraceState {
-	return TraceState(internal.NewTraceState(new(string)))
+	state := internal.StateMutable
+	return TraceState(internal.NewTraceState(new(string), &state))
 }
 
 func (ms TraceState) getOrig() *string {
 	return internal.GetOrigTraceState(internal.TraceState(ms))
 }
 
+func (ms TraceState) getState() *internal.State {
+	return internal.GetTraceStateState(internal.TraceState(ms))
+}
+
 // AsRaw returns the string representation of the tracestate in w3c-trace-context format: https://www.w3.org/TR/trace-context/#tracestate-header
 func (ms TraceState) AsRaw() string {
 	return *ms.getOrig()
@@ -25,17 +30,21 @@ func (ms TraceState) AsRaw() string {
 
 // FromRaw copies the string representation in w3c-trace-context format of the tracestate into this TraceState.
 func (ms TraceState) FromRaw(v string) {
+	ms.getState().AssertMutable()
 	*ms.getOrig() = v
 }
 
 // MoveTo moves the TraceState instance overriding the destination
 // and resetting the current instance to its zero value.
 func (ms TraceState) MoveTo(dest TraceState) {
+	ms.getState().AssertMutable()
+	dest.getState().AssertMutable()
 	*dest.getOrig() = *ms.getOrig()
 	*ms.getOrig() = ""
 }
 
 // CopyTo copies the TraceState instance overriding the destination.
 func (ms TraceState) CopyTo(dest TraceState) {
+	dest.getState().AssertMutable()
 	*dest.getOrig() = *ms.getOrig()
 }
