diff --git a/pdata/pcommon/value_test.go b/pdata/pcommon/value_test.go
index 1a4ed8fea..9d2e5f439 100644
--- a/pdata/pcommon/value_test.go
+++ b/pdata/pcommon/value_test.go
@@ -11,6 +11,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 )
 
@@ -44,6 +45,36 @@ func TestValue(t *testing.T) {
 	assert.EqualValues(t, ValueTypeSlice, v.Type())
 }
 
+func TestValueReadOnly(t *testing.T) {
+	state := internal.StateReadOnly
+	v := newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "v"}}, &state)
+
+	assert.EqualValues(t, ValueTypeStr, v.Type())
+	assert.EqualValues(t, "v", v.Str())
+	assert.EqualValues(t, 0, v.Int())
+	assert.EqualValues(t, 0, v.Double())
+	assert.False(t, v.Bool())
+	assert.EqualValues(t, ByteSlice{}, v.Bytes())
+	assert.EqualValues(t, Map{}, v.Map())
+	assert.EqualValues(t, Slice{}, v.Slice())
+
+	assert.EqualValues(t, "v", v.AsString())
+
+	assert.Panics(t, func() { v.SetStr("abc") })
+	assert.Panics(t, func() { v.SetInt(123) })
+	assert.Panics(t, func() { v.SetDouble(3.4) })
+	assert.Panics(t, func() { v.SetBool(true) })
+	assert.Panics(t, func() { v.SetEmptyBytes() })
+	assert.Panics(t, func() { v.SetEmptyMap() })
+	assert.Panics(t, func() { v.SetEmptySlice() })
+
+	v2 := NewValueEmpty()
+	v.CopyTo(v2)
+	assert.Equal(t, v.AsRaw(), v2.AsRaw())
+	assert.Panics(t, func() { v2.CopyTo(v) })
+
+}
+
 func TestValueType(t *testing.T) {
 	assert.EqualValues(t, "Empty", ValueTypeEmpty.String())
 	assert.EqualValues(t, "Str", ValueTypeStr.String())
@@ -128,7 +159,8 @@ func TestValueMap(t *testing.T) {
 
 	// Test nil KvlistValue case for Map() func.
 	orig := &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_KvlistValue{KvlistValue: nil}}
-	m1 = newValue(orig)
+	state := internal.StateMutable
+	m1 = newValue(orig, &state)
 	assert.EqualValues(t, Map{}, m1.Map())
 }
 
@@ -165,8 +197,9 @@ func TestValueSlice(t *testing.T) {
 	assert.EqualValues(t, "somestr", v.Str())
 
 	// Test nil values case for Slice() func.
-	a1 = newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_ArrayValue{ArrayValue: nil}})
-	assert.EqualValues(t, newSlice(nil), a1.Slice())
+	state := internal.StateMutable
+	a1 = newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_ArrayValue{ArrayValue: nil}}, &state)
+	assert.EqualValues(t, newSlice(nil, nil), a1.Slice())
 }
 
 func TestNilOrigSetValue(t *testing.T) {
@@ -200,26 +233,28 @@ func TestNilOrigSetValue(t *testing.T) {
 }
 
 func TestValue_CopyTo(t *testing.T) {
+	state := internal.StateMutable
+
 	// Test nil KvlistValue case for Map() func.
 	dest := NewValueEmpty()
 	orig := &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_KvlistValue{KvlistValue: nil}}
-	newValue(orig).CopyTo(dest)
+	newValue(orig, &state).CopyTo(dest)
 	assert.Nil(t, dest.getOrig().Value.(*otlpcommon.AnyValue_KvlistValue).KvlistValue)
 
 	// Test nil ArrayValue case for Slice() func.
 	dest = NewValueEmpty()
 	orig = &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_ArrayValue{ArrayValue: nil}}
-	newValue(orig).CopyTo(dest)
+	newValue(orig, &state).CopyTo(dest)
 	assert.Nil(t, dest.getOrig().Value.(*otlpcommon.AnyValue_ArrayValue).ArrayValue)
 
 	// Test copy empty value.
 	orig = &otlpcommon.AnyValue{}
-	newValue(orig).CopyTo(dest)
+	newValue(orig, &state).CopyTo(dest)
 	assert.Nil(t, dest.getOrig().Value)
 
 	av := NewValueEmpty()
 	destVal := otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_IntValue{}}
-	av.CopyTo(newValue(&destVal))
+	av.CopyTo(newValue(&destVal, &state))
 	assert.EqualValues(t, nil, destVal.Value)
 }
 
@@ -228,7 +263,8 @@ func TestSliceWithNilValues(t *testing.T) {
 		{},
 		{Value: &otlpcommon.AnyValue_StringValue{StringValue: "test_value"}},
 	}
-	sm := newSlice(&origWithNil)
+	state := internal.StateMutable
+	sm := newSlice(&origWithNil, &state)
 
 	val := sm.At(0)
 	assert.EqualValues(t, ValueTypeEmpty, val.Type())
