diff --git a/cmd/otelcorecol/main.go b/cmd/otelcorecol/main.go
index b0b3972fd..bf2e84be9 100644
--- a/cmd/otelcorecol/main.go
+++ b/cmd/otelcorecol/main.go
@@ -20,7 +20,7 @@ func main() {
 	info := component.BuildInfo{
 		Command:     "otelcorecol",
 		Description: "Local OpenTelemetry Collector binary, testing only.",
-		Version:     "0.111.0-dev",
+		Version:     "0.112.0-dev",
 	}
 
 	set := otelcol.CollectorSettings{
