diff --git a/pdata/pprofile/generated_location_test.go b/pdata/pprofile/generated_location_test.go
index 30bd9cd69..8ff5cf456 100644
--- a/pdata/pprofile/generated_location_test.go
+++ b/pdata/pprofile/generated_location_test.go
@@ -47,14 +47,10 @@ func TestLocation_MappingIndex(t *testing.T) {
 	ms := NewLocation()
 	assert.Equal(t, int32(0), ms.MappingIndex())
 	ms.SetMappingIndex(int32(13))
-	assert.True(t, ms.HasMappingIndex())
 	assert.Equal(t, int32(13), ms.MappingIndex())
-	ms.RemoveMappingIndex()
-	assert.False(t, ms.HasMappingIndex())
-	dest := NewLocation()
-	dest.SetMappingIndex(int32(13))
-	ms.CopyTo(dest)
-	assert.False(t, dest.HasMappingIndex())
+	sharedState := internal.NewState()
+	sharedState.MarkReadOnly()
+	assert.Panics(t, func() { newLocation(&otlpprofiles.Location{}, sharedState).SetMappingIndex(int32(13)) })
 }
 
 func TestLocation_Address(t *testing.T) {
@@ -74,16 +70,6 @@ func TestLocation_Line(t *testing.T) {
 	assert.Equal(t, generateTestLineSlice(), ms.Line())
 }
 
-func TestLocation_IsFolded(t *testing.T) {
-	ms := NewLocation()
-	assert.False(t, ms.IsFolded())
-	ms.SetIsFolded(true)
-	assert.True(t, ms.IsFolded())
-	sharedState := internal.NewState()
-	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { newLocation(&otlpprofiles.Location{}, sharedState).SetIsFolded(true) })
-}
-
 func TestLocation_AttributeIndices(t *testing.T) {
 	ms := NewLocation()
 	assert.Equal(t, pcommon.NewInt32Slice(), ms.AttributeIndices())
