diff --git a/service/defaultcomponents/default_receivers_test.go b/service/defaultcomponents/default_receivers_test.go
index ac7e40940..6be710108 100644
--- a/service/defaultcomponents/default_receivers_test.go
+++ b/service/defaultcomponents/default_receivers_test.go
@@ -82,7 +82,7 @@ func TestDefaultReceivers(t *testing.T) {
 			factory, ok := rcvrFactories[tt.receiver]
 			require.True(t, ok)
 			assert.Equal(t, tt.receiver, factory.Type())
-			assert.Equal(t, string(tt.receiver), factory.CreateDefaultConfig().ID().String())
+			assert.Equal(t, config.NewID(tt.receiver), factory.CreateDefaultConfig().ID())
 
 			if tt.skipLifecyle {
 				t.Log("Skipping lifecycle test", tt.receiver)
