diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index 49873e2c7..aef879086 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -5,7 +5,7 @@ go 1.23.0
 require (
 	github.com/fsnotify/fsnotify v1.9.0
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/config/configopaque v1.29.0
+	go.opentelemetry.io/collector/config/configopaque v1.30.0
 )
 
 require (
