diff --git a/service/service.go b/service/service.go
index 22c8f20ba..22eff4b75 100644
--- a/service/service.go
+++ b/service/service.go
@@ -40,6 +40,7 @@ import (
 
 // Application represents a collector application
 type Application struct {
+	info           ApplicationStartInfo
 	rootCmd        *cobra.Command
 	v              *viper.Viper
 	logger         *zap.Logger
@@ -60,6 +61,22 @@ type Application struct {
 	asyncErrorChannel chan error
 }
 
+// ApplicationStartInfo is the information that is logged at the application start.
+// This information can be overridden in custom builds.
+type ApplicationStartInfo struct {
+	// Executable file name, e.g. "otelcol".
+	ExeName string
+
+	// Long name, used e.g. in the logs.
+	LongName string
+
+	// Version string.
+	Version string
+
+	// Git hash of the source code.
+	GitHash string
+}
+
 var _ receiver.Host = (*Application)(nil)
 
 // Context returns a context provided by the host to be used on the receiver
@@ -72,6 +89,7 @@ func (app *Application) Context() context.Context {
 // New creates and returns a new instance of Application.
 func New(
 	factories config.Factories,
+	appInfo ApplicationStartInfo,
 ) (*Application, error) {
 
 	if err := configcheck.ValidateConfigFromFactories(factories); err != nil {
@@ -79,14 +97,15 @@ func New(
 	}
 
 	app := &Application{
+		info:      appInfo,
 		v:         viper.New(),
 		readyChan: make(chan struct{}),
 		factories: factories,
 	}
 
 	rootCmd := &cobra.Command{
-		Use:  "otelcol",
-		Long: "OpenTelemetry Collector",
+		Use:  appInfo.ExeName,
+		Long: appInfo.LongName,
 		Run: func(cmd *cobra.Command, args []string) {
 			app.init()
 			app.execute()
@@ -312,7 +331,11 @@ func (app *Application) shutdownExtensions() {
 }
 
 func (app *Application) execute() {
-	app.logger.Info("Starting...", zap.Int("NumCPU", runtime.NumCPU()))
+	app.logger.Info("Starting "+app.info.LongName+"...",
+		zap.String("Version", app.info.Version),
+		zap.String("GitHash", app.info.GitHash),
+		zap.Int("NumCPU", runtime.NumCPU()),
+	)
 
 	// Set memory ballast
 	ballast, ballastSizeBytes := app.createMemoryBallast()
