diff --git a/examples/demo/app/main.go b/examples/demo/app/main.go
index 4764a0590..77773d57e 100644
--- a/examples/demo/app/main.go
+++ b/examples/demo/app/main.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Sample contains a program that exports to the OpenCensus service.
+// Sample contains a program that exports to the OpenTelemetry service.
 package main
 
 import (
