diff --git a/component/componenttest/example_factories_test.go b/component/componenttest/example_factories_test.go
index 2c5cc4c42..935da61f8 100644
--- a/component/componenttest/example_factories_test.go
+++ b/component/componenttest/example_factories_test.go
@@ -4,7 +4,7 @@
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//      http://www.apache.org/licenses/LICENSE-2.0
+//       http://www.apache.org/licenses/LICENSE-2.0
 //
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
