diff --git a/processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go b/processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go
index db9fa0bf0..aaf8a87e4 100644
--- a/processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go
+++ b/processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go
@@ -18,14 +18,10 @@ import (
 	"context"
 	"strconv"
 
-	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
-
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenterror"
 	"go.opentelemetry.io/collector/consumer"
-	"go.opentelemetry.io/collector/consumer/consumerdata"
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/collector/translator/internaldata"
 )
 
 // samplingPriority has the semantic result of parsing the "sampling.priority"
@@ -74,49 +70,55 @@ func newTraceProcessor(nextConsumer consumer.TraceConsumer, cfg Config) (compone
 }
 
 func (tsp *tracesamplerprocessor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
-	octds := internaldata.TraceDataToOC(td)
-	var errs []error
-	for _, octd := range octds {
-		if err := tsp.processTraces(ctx, octd); err != nil {
-			errs = append(errs, err)
+	rspans := td.ResourceSpans()
+	sampledTraceData := pdata.NewTraces()
+	for i := 0; i < rspans.Len(); i++ {
+		rspan := rspans.At(i)
+		if rspan.IsNil() {
+			continue
 		}
+		tsp.processTraces(ctx, rspan, sampledTraceData)
 	}
-	return componenterror.CombineErrors(errs)
+	return tsp.nextConsumer.ConsumeTraces(ctx, sampledTraceData)
 }
 
-func (tsp *tracesamplerprocessor) processTraces(ctx context.Context, td consumerdata.TraceData) error {
+func (tsp *tracesamplerprocessor) processTraces(ctx context.Context, resourceSpans pdata.ResourceSpans, sampledTraceData pdata.Traces) {
 	scaledSamplingRate := tsp.scaledSamplingRate
 
-	sampledTraceData := consumerdata.TraceData{
-		Node:         td.Node,
-		Resource:     td.Resource,
-		SourceFormat: td.SourceFormat,
-	}
-
-	sampledSpans := make([]*tracepb.Span, 0, len(td.Spans))
-	for _, span := range td.Spans {
-		sp := parseSpanSamplingPriority(span)
-		if sp == doNotSampleSpan {
-			// The OpenTelemetry mentions this as a "hint" we take a stronger
-			// approach and do not sample the span since some may use it to
-			// remove specific spans from traces.
+	sampledTraceData.ResourceSpans().Resize(sampledTraceData.ResourceSpans().Len() + 1)
+	rs := sampledTraceData.ResourceSpans().At(sampledTraceData.ResourceSpans().Len() - 1)
+	rs.Resource().InitEmpty()
+	resourceSpans.Resource().CopyTo(rs.Resource())
+	rs.InstrumentationLibrarySpans().Resize(1)
+	spns := rs.InstrumentationLibrarySpans().At(0).Spans()
+
+	ilss := resourceSpans.InstrumentationLibrarySpans()
+	for j := 0; j < ilss.Len(); j++ {
+		ils := ilss.At(j)
+		if ils.IsNil() {
 			continue
 		}
+		for k := 0; k < ils.Spans().Len(); k++ {
+			span := ils.Spans().At(k)
+			sp := parseSpanSamplingPriority(span)
+			if sp == doNotSampleSpan {
+				// The OpenTelemetry mentions this as a "hint" we take a stronger
+				// approach and do not sample the span since some may use it to
+				// remove specific spans from traces.
+				continue
+			}
 
-		// If one assumes random trace ids hashing may seems avoidable, however, traces can be coming from sources
-		// with various different criteria to generate trace id and perhaps were already sampled without hashing.
-		// Hashing here prevents bias due to such systems.
-		sampled := sp == mustSampleSpan ||
-			hash(span.TraceId, tsp.hashSeed)&bitMaskHashBuckets < scaledSamplingRate
+			// If one assumes random trace ids hashing may seems avoidable, however, traces can be coming from sources
+			// with various different criteria to generate trace id and perhaps were already sampled without hashing.
+			// Hashing here prevents bias due to such systems.
+			sampled := sp == mustSampleSpan ||
+				hash(span.TraceID(), tsp.hashSeed)&bitMaskHashBuckets < scaledSamplingRate
 
-		if sampled {
-			sampledSpans = append(sampledSpans, span)
+			if sampled {
+				spns.Append(&span)
+			}
 		}
 	}
-
-	sampledTraceData.Spans = sampledSpans
-
-	return tsp.nextConsumer.ConsumeTraces(ctx, internaldata.OCToTraceData(sampledTraceData))
 }
 
 func (tsp *tracesamplerprocessor) GetCapabilities() component.ProcessorCapabilities {
@@ -137,14 +139,17 @@ func (tsp *tracesamplerprocessor) Shutdown(context.Context) error {
 // decide if the span should be sampled or not. The usage of the tag follows the
 // OpenTracing semantic tags:
 // https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table
-func parseSpanSamplingPriority(span *tracepb.Span) samplingPriority {
-	attribMap := span.GetAttributes().GetAttributeMap()
-	if attribMap == nil {
+func parseSpanSamplingPriority(span pdata.Span) samplingPriority {
+	if span.IsNil() {
+		return deferDecision
+	}
+	attribMap := span.Attributes()
+	if attribMap.Len() <= 0 {
 		return deferDecision
 	}
 
-	samplingPriorityAttrib := attribMap["sampling.priority"]
-	if samplingPriorityAttrib == nil {
+	samplingPriorityAttrib, ok := attribMap.Get("sampling.priority")
+	if !ok {
 		return deferDecision
 	}
 
@@ -155,23 +160,23 @@ func parseSpanSamplingPriority(span *tracepb.Span) samplingPriority {
 	// using different conventions regarding "sampling.priority". Besides the
 	// client libraries it is also possible that the type was lost in translation
 	// between different formats.
-	switch samplingPriorityAttrib.Value.(type) {
-	case *tracepb.AttributeValue_IntValue:
-		value := samplingPriorityAttrib.GetIntValue()
+	switch samplingPriorityAttrib.Type() {
+	case pdata.AttributeValueINT:
+		value := samplingPriorityAttrib.IntVal()
 		if value == 0 {
 			decision = doNotSampleSpan
 		} else if value > 0 {
 			decision = mustSampleSpan
 		}
-	case *tracepb.AttributeValue_DoubleValue:
-		value := samplingPriorityAttrib.GetDoubleValue()
+	case pdata.AttributeValueDOUBLE:
+		value := samplingPriorityAttrib.DoubleVal()
 		if value == 0.0 {
 			decision = doNotSampleSpan
 		} else if value > 0.0 {
 			decision = mustSampleSpan
 		}
-	case *tracepb.AttributeValue_StringValue:
-		attribVal := samplingPriorityAttrib.GetStringValue().GetValue()
+	case pdata.AttributeValueSTRING:
+		attribVal := samplingPriorityAttrib.StringVal()
 		if value, err := strconv.ParseFloat(attribVal, 64); err == nil {
 			if value == 0.0 {
 				decision = doNotSampleSpan
