diff --git a/exporter/prometheusremotewriteexporter/helper.go b/exporter/prometheusremotewriteexporter/helper.go
index 51a58bd3c..8c42d3e5b 100644
--- a/exporter/prometheusremotewriteexporter/helper.go
+++ b/exporter/prometheusremotewriteexporter/helper.go
@@ -272,7 +272,7 @@ func addSingleNumberDataPoint(pt pdata.NumberDataPoint, resource pdata.Resource,
 	name := getPromMetricName(metric, namespace)
 	labels := createLabelSet(resource, pt.LabelsMap(), externalLabels, nameStr, name)
 	sample := &prompb.Sample{
-		Value: pt.Value(),
+		Value: pt.DoubleVal(),
 		// convert ns to ms
 		Timestamp: convertTimeStamp(pt.Timestamp()),
 	}
