diff --git a/exporter/exporterhelper/exporterhelperprofiles/go.mod b/exporter/exporterhelper/exporterhelperprofiles/go.mod
index 1057dfd35..ffd56a1cd 100644
--- a/exporter/exporterhelper/exporterhelperprofiles/go.mod
+++ b/exporter/exporterhelper/exporterhelperprofiles/go.mod
@@ -13,6 +13,7 @@ require (
 	go.opentelemetry.io/collector/consumer/consumertest v0.111.0
 	go.opentelemetry.io/collector/exporter v0.111.0
 	go.opentelemetry.io/collector/exporter/exporterprofiles v0.111.0
+	go.opentelemetry.io/collector/exporter/exportertest v0.111.0
 	go.opentelemetry.io/collector/pdata v1.17.0
 	go.opentelemetry.io/collector/pdata/pprofile v0.111.0
 	go.opentelemetry.io/collector/pdata/testdata v0.111.0
@@ -88,4 +89,6 @@ replace go.opentelemetry.io/collector/extension/experimental/storage => ../../..
 
 replace go.opentelemetry.io/collector/pipeline => ../../../pipeline
 
+replace go.opentelemetry.io/collector/exporter/exportertest => ../../exportertest
+
 replace go.opentelemetry.io/collector/consumer/consumererror => ../../../consumer/consumererror
