diff --git a/extension/auth/authtest/go.mod b/extension/auth/authtest/go.mod
index dc2bd2709..d7613ce45 100644
--- a/extension/auth/authtest/go.mod
+++ b/extension/auth/authtest/go.mod
@@ -7,7 +7,7 @@ require (
 	go.opentelemetry.io/collector/component v0.116.0
 	go.opentelemetry.io/collector/extension/auth v0.116.0
 	go.uber.org/goleak v1.3.0
-	google.golang.org/grpc v1.69.0
+	google.golang.org/grpc v1.69.2
 )
 
 require (
