diff --git a/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go b/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go
index 887332319..f95d89544 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go
@@ -15,7 +15,7 @@
 package processscraper
 
 import (
-	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/dataold"
 )
 
 // labels
@@ -42,42 +42,42 @@ const (
 
 // descriptors
 
-var cpuTimeDescriptor = func() pdata.MetricDescriptor {
-	descriptor := pdata.NewMetricDescriptor()
+var cpuTimeDescriptor = func() dataold.MetricDescriptor {
+	descriptor := dataold.NewMetricDescriptor()
 	descriptor.InitEmpty()
 	descriptor.SetName("process.cpu.time")
 	descriptor.SetDescription("Total CPU seconds broken down by different states.")
 	descriptor.SetUnit("s")
-	descriptor.SetType(pdata.MetricTypeMonotonicDouble)
+	descriptor.SetType(dataold.MetricTypeMonotonicDouble)
 	return descriptor
 }()
 
-var physicalMemoryUsageDescriptor = func() pdata.MetricDescriptor {
-	descriptor := pdata.NewMetricDescriptor()
+var physicalMemoryUsageDescriptor = func() dataold.MetricDescriptor {
+	descriptor := dataold.NewMetricDescriptor()
 	descriptor.InitEmpty()
 	descriptor.SetName("process.memory.physical_usage")
 	descriptor.SetDescription("The amount of physical memory in use.")
 	descriptor.SetUnit("bytes")
-	descriptor.SetType(pdata.MetricTypeInt64)
+	descriptor.SetType(dataold.MetricTypeInt64)
 	return descriptor
 }()
 
-var virtualMemoryUsageDescriptor = func() pdata.MetricDescriptor {
-	descriptor := pdata.NewMetricDescriptor()
+var virtualMemoryUsageDescriptor = func() dataold.MetricDescriptor {
+	descriptor := dataold.NewMetricDescriptor()
 	descriptor.InitEmpty()
 	descriptor.SetName("process.memory.virtual_usage")
 	descriptor.SetDescription("Virtual memory size.")
 	descriptor.SetUnit("bytes")
-	descriptor.SetType(pdata.MetricTypeInt64)
+	descriptor.SetType(dataold.MetricTypeInt64)
 	return descriptor
 }()
 
-var diskIODescriptor = func() pdata.MetricDescriptor {
-	descriptor := pdata.NewMetricDescriptor()
+var diskIODescriptor = func() dataold.MetricDescriptor {
+	descriptor := dataold.NewMetricDescriptor()
 	descriptor.InitEmpty()
 	descriptor.SetName("process.disk.io")
 	descriptor.SetDescription("Disk bytes transferred.")
 	descriptor.SetUnit("bytes")
-	descriptor.SetType(pdata.MetricTypeMonotonicInt64)
+	descriptor.SetType(dataold.MetricTypeMonotonicInt64)
 	return descriptor
 }()
