diff --git a/consumer/consumertest/sink_test.go b/consumer/consumertest/sink_test.go
index bd29ebc6a..3c813cf2f 100644
--- a/consumer/consumertest/sink_test.go
+++ b/consumer/consumertest/sink_test.go
@@ -12,6 +12,7 @@ import (
 
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pprofile"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.opentelemetry.io/collector/pdata/testdata"
 )
@@ -60,3 +61,16 @@ func TestLogsSink(t *testing.T) {
 	assert.Equal(t, 0, len(sink.AllLogs()))
 	assert.Equal(t, 0, sink.LogRecordCount())
 }
+
+func TestProfilesSink(t *testing.T) {
+	sink := new(ProfilesSink)
+	td := testdata.GenerateProfiles(1)
+	want := make([]pprofile.Profiles, 0, 7)
+	for i := 0; i < 7; i++ {
+		require.NoError(t, sink.ConsumeProfiles(context.Background(), td))
+		want = append(want, td)
+	}
+	assert.Equal(t, want, sink.AllProfiles())
+	sink.Reset()
+	assert.Equal(t, 0, len(sink.AllProfiles()))
+}
