diff --git a/component/componenttest/nop_telemetry.go b/component/componenttest/nop_telemetry.go
index d872b45cc..6b0fd94ee 100644
--- a/component/componenttest/nop_telemetry.go
+++ b/component/componenttest/nop_telemetry.go
@@ -15,7 +15,7 @@
 package componenttest // import "go.opentelemetry.io/collector/component/componenttest"
 
 import (
-	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/nonrecording"
 	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 
@@ -28,7 +28,7 @@ func NewNopTelemetrySettings() component.TelemetrySettings {
 	return component.TelemetrySettings{
 		Logger:         zap.NewNop(),
 		TracerProvider: trace.NewNoopTracerProvider(),
-		MeterProvider:  metric.NewNoopMeterProvider(),
+		MeterProvider:  nonrecording.NewNoopMeterProvider(),
 		MetricsLevel:   configtelemetry.LevelNone,
 	}
 }
