diff --git a/service/internal/builders/exporter_test.go b/service/internal/builders/exporter_test.go
index e4a3cdd19..802a42f2a 100644
--- a/service/internal/builders/exporter_test.go
+++ b/service/internal/builders/exporter_test.go
@@ -65,37 +65,37 @@ func TestExporterBuilder(t *testing.T) {
 
 			te, err := b.CreateTraces(context.Background(), createExporterSettings(tt.id))
 			if tt.err != "" {
-				assert.EqualError(t, err, tt.err)
+				require.EqualError(t, err, tt.err)
 				assert.Nil(t, te)
 			} else {
-				assert.NoError(t, err)
+				require.NoError(t, err)
 				assert.Equal(t, nopExporterInstance, te)
 			}
 
 			me, err := b.CreateMetrics(context.Background(), createExporterSettings(tt.id))
 			if tt.err != "" {
-				assert.EqualError(t, err, tt.err)
+				require.EqualError(t, err, tt.err)
 				assert.Nil(t, me)
 			} else {
-				assert.NoError(t, err)
+				require.NoError(t, err)
 				assert.Equal(t, nopExporterInstance, me)
 			}
 
 			le, err := b.CreateLogs(context.Background(), createExporterSettings(tt.id))
 			if tt.err != "" {
-				assert.EqualError(t, err, tt.err)
+				require.EqualError(t, err, tt.err)
 				assert.Nil(t, le)
 			} else {
-				assert.NoError(t, err)
+				require.NoError(t, err)
 				assert.Equal(t, nopExporterInstance, le)
 			}
 
 			pe, err := b.CreateProfiles(context.Background(), createExporterSettings(tt.id))
 			if tt.err != "" {
-				assert.EqualError(t, err, tt.err)
+				require.EqualError(t, err, tt.err)
 				assert.Nil(t, pe)
 			} else {
-				assert.NoError(t, err)
+				require.NoError(t, err)
 				assert.Equal(t, nopExporterInstance, pe)
 			}
 		})
@@ -121,19 +121,19 @@ func TestExporterBuilderMissingConfig(t *testing.T) {
 	missingID := component.MustNewIDWithName("all", "missing")
 
 	te, err := bErr.CreateTraces(context.Background(), createExporterSettings(missingID))
-	assert.EqualError(t, err, "exporter \"all/missing\" is not configured")
+	require.EqualError(t, err, "exporter \"all/missing\" is not configured")
 	assert.Nil(t, te)
 
 	me, err := bErr.CreateMetrics(context.Background(), createExporterSettings(missingID))
-	assert.EqualError(t, err, "exporter \"all/missing\" is not configured")
+	require.EqualError(t, err, "exporter \"all/missing\" is not configured")
 	assert.Nil(t, me)
 
 	le, err := bErr.CreateLogs(context.Background(), createExporterSettings(missingID))
-	assert.EqualError(t, err, "exporter \"all/missing\" is not configured")
+	require.EqualError(t, err, "exporter \"all/missing\" is not configured")
 	assert.Nil(t, le)
 
 	pe, err := bErr.CreateProfiles(context.Background(), createExporterSettings(missingID))
-	assert.EqualError(t, err, "exporter \"all/missing\" is not configured")
+	require.EqualError(t, err, "exporter \"all/missing\" is not configured")
 	assert.Nil(t, pe)
 }
 
