diff --git a/exporter/otlphttpexporter/config.go b/exporter/otlphttpexporter/config.go
index 892dff4e8..eec59c144 100644
--- a/exporter/otlphttpexporter/config.go
+++ b/exporter/otlphttpexporter/config.go
@@ -45,9 +45,9 @@ func (e *EncodingType) UnmarshalText(text []byte) error {
 
 // Config defines configuration for OTLP/HTTP exporter.
 type Config struct {
-	ClientConfig confighttp.ClientConfig    `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
-	QueueConfig  exporterhelper.QueueConfig `mapstructure:"sending_queue"`
-	RetryConfig  configretry.BackOffConfig  `mapstructure:"retry_on_failure"`
+	ClientConfig confighttp.ClientConfig         `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	QueueConfig  exporterhelper.QueueBatchConfig `mapstructure:"sending_queue"`
+	RetryConfig  configretry.BackOffConfig       `mapstructure:"retry_on_failure"`
 
 	// The URL to send traces to. If omitted the Endpoint + "/v1/traces" will be used.
 	TracesEndpoint string `mapstructure:"traces_endpoint"`
