diff --git a/pdata/pmetric/generated_exponentialhistogramdatapointbuckets_test.go b/pdata/pmetric/generated_exponentialhistogramdatapointbuckets_test.go
index ee516c433..d5da345ea 100644
--- a/pdata/pmetric/generated_exponentialhistogramdatapointbuckets_test.go
+++ b/pdata/pmetric/generated_exponentialhistogramdatapointbuckets_test.go
@@ -72,11 +72,11 @@ func TestExponentialHistogramDataPointBuckets_MarshalAndUnmarshalJSON(t *testing
 func TestExponentialHistogramDataPointBuckets_Offset(t *testing.T) {
 	ms := NewExponentialHistogramDataPointBuckets()
 	assert.Equal(t, int32(0), ms.Offset())
-	ms.SetOffset(int32(909))
-	assert.Equal(t, int32(909), ms.Offset())
+	ms.SetOffset(int32(13))
+	assert.Equal(t, int32(13), ms.Offset())
 	sharedState := internal.StateReadOnly
 	assert.Panics(t, func() {
-		newExponentialHistogramDataPointBuckets(&otlpmetrics.ExponentialHistogramDataPoint_Buckets{}, &sharedState).SetOffset(int32(909))
+		newExponentialHistogramDataPointBuckets(&otlpmetrics.ExponentialHistogramDataPoint_Buckets{}, &sharedState).SetOffset(int32(13))
 	})
 }
 
@@ -94,6 +94,6 @@ func generateTestExponentialHistogramDataPointBuckets() ExponentialHistogramData
 }
 
 func fillTestExponentialHistogramDataPointBuckets(tv ExponentialHistogramDataPointBuckets) {
-	tv.orig.Offset = int32(909)
+	tv.orig.Offset = int32(13)
 	internal.FillTestUInt64Slice(internal.NewUInt64Slice(&tv.orig.BucketCounts, tv.state))
 }
