diff --git a/internal/cmd/pdatagen/internal/pprofile_package.go b/internal/cmd/pdatagen/internal/pprofile_package.go
index 82565d746..c0cf42ed8 100644
--- a/internal/cmd/pdatagen/internal/pprofile_package.go
+++ b/internal/cmd/pdatagen/internal/pprofile_package.go
@@ -14,6 +14,7 @@ var pprofile = &Package{
 			`"go.opentelemetry.io/collector/pdata/internal"`,
 			`"go.opentelemetry.io/collector/pdata/internal/data"`,
 			`"go.opentelemetry.io/collector/pdata/internal/json"`,
+			`"go.opentelemetry.io/collector/pdata/internal/proto"`,
 			`otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"`,
 			`"go.opentelemetry.io/collector/pdata/pcommon"`,
 		},
@@ -69,12 +70,22 @@ var resourceProfiles = &messageStruct{
 	description:    "// ResourceProfiles is a collection of profiles from a Resource.",
 	originFullName: "otlpprofiles.ResourceProfiles",
 	fields: []Field{
-		resourceField,
-		schemaURLField,
+		&MessageField{
+			fieldName:     "Resource",
+			protoID:       1,
+			returnMessage: resource,
+		},
 		&SliceField{
 			fieldName:   "ScopeProfiles",
+			protoID:     2,
+			protoType:   ProtoTypeMessage,
 			returnSlice: scopeProfilesSlice,
 		},
+		&PrimitiveField{
+			fieldName: "SchemaUrl",
+			protoID:   3,
+			protoType: ProtoTypeString,
+		},
 	},
 }
 
@@ -85,30 +96,44 @@ var profilesDictionary = &messageStruct{
 	fields: []Field{
 		&SliceField{
 			fieldName:   "MappingTable",
+			protoID:     1,
+			protoType:   ProtoTypeMessage,
 			returnSlice: mappingSlice,
 		},
 		&SliceField{
 			fieldName:   "LocationTable",
+			protoID:     2,
+			protoType:   ProtoTypeMessage,
 			returnSlice: locationSlice,
 		},
 		&SliceField{
 			fieldName:   "FunctionTable",
+			protoID:     3,
+			protoType:   ProtoTypeMessage,
 			returnSlice: functionSlice,
 		},
 		&SliceField{
 			fieldName:   "LinkTable",
+			protoID:     4,
+			protoType:   ProtoTypeMessage,
 			returnSlice: linkSlice,
 		},
 		&SliceField{
 			fieldName:   "StringTable",
+			protoID:     5,
+			protoType:   ProtoTypeString,
 			returnSlice: stringSlice,
 		},
 		&SliceField{
 			fieldName:   "AttributeTable",
+			protoID:     6,
+			protoType:   ProtoTypeMessage,
 			returnSlice: attributeTableSlice,
 		},
 		&SliceField{
 			fieldName:   "AttributeUnits",
+			protoID:     7,
+			protoType:   ProtoTypeMessage,
 			returnSlice: attributeUnitSlice,
 		},
 	},
@@ -124,12 +149,22 @@ var scopeProfiles = &messageStruct{
 	description:    "// ScopeProfiles is a collection of profiles from a LibraryInstrumentation.",
 	originFullName: "otlpprofiles.ScopeProfiles",
 	fields: []Field{
-		scopeField,
-		schemaURLField,
+		&MessageField{
+			fieldName:     "Scope",
+			protoID:       1,
+			returnMessage: scope,
+		},
 		&SliceField{
 			fieldName:   "Profiles",
+			protoID:     2,
+			protoType:   ProtoTypeMessage,
 			returnSlice: profilesSlice,
 		},
+		&PrimitiveField{
+			fieldName: "SchemaUrl",
+			protoID:   3,
+			protoType: ProtoTypeString,
+		},
 	},
 }
 
@@ -145,23 +180,30 @@ var profile = &messageStruct{
 	fields: []Field{
 		&SliceField{
 			fieldName:   "SampleType",
+			protoID:     1,
+			protoType:   ProtoTypeMessage,
 			returnSlice: valueTypeSlice,
 		},
 		&SliceField{
 			fieldName:   "Sample",
+			protoID:     2,
+			protoType:   ProtoTypeMessage,
 			returnSlice: sampleSlice,
 		},
 		&SliceField{
 			fieldName:   "LocationIndices",
+			protoID:     3,
+			protoType:   ProtoTypeInt32,
 			returnSlice: int32Slice,
 		},
 		&TypedField{
 			fieldName:       "Time",
 			originFieldName: "TimeNanos",
+			protoID:         4,
 			returnType: &TypedType{
 				structName:  "Timestamp",
 				packageName: "pcommon",
-				rawType:     "int64",
+				protoType:   ProtoTypeInt64,
 				defaultVal:  "0",
 				testVal:     "1234567890",
 			},
@@ -169,55 +211,68 @@ var profile = &messageStruct{
 		&TypedField{
 			fieldName:       "Duration",
 			originFieldName: "DurationNanos",
+			protoID:         5,
 			returnType: &TypedType{
 				structName:  "Timestamp",
 				packageName: "pcommon",
-				rawType:     "int64",
+				protoType:   ProtoTypeInt64,
 				defaultVal:  "0",
 				testVal:     "1234567890",
 			},
 		},
 		&MessageField{
 			fieldName:     "PeriodType",
+			protoID:       6,
 			returnMessage: valueType,
 		},
 		&PrimitiveField{
 			fieldName: "Period",
+			protoID:   7,
 			protoType: ProtoTypeInt64,
 		},
 		&SliceField{
 			fieldName:   "CommentStrindices",
+			protoID:     8,
+			protoType:   ProtoTypeInt32,
 			returnSlice: int32Slice,
 		},
 		&PrimitiveField{
 			fieldName: "DefaultSampleTypeIndex",
+			protoID:   9,
 			protoType: ProtoTypeInt32,
 		},
 		&TypedField{
 			fieldName:       "ProfileID",
 			originFieldName: "ProfileId",
+			protoID:         10,
 			returnType: &TypedType{
-				structName: "ProfileID",
-				rawType:    "data.ProfileID",
-				isType:     true,
-				defaultVal: "data.ProfileID([16]byte{})",
-				testVal:    "data.ProfileID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1})",
+				structName:  "ProfileID",
+				protoType:   ProtoTypeMessage,
+				messageName: "data.ProfileID",
+				defaultVal:  "data.ProfileID([16]byte{})",
+				testVal:     "data.ProfileID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1})",
 			},
 		},
 		&PrimitiveField{
 			fieldName: "DroppedAttributesCount",
+			protoID:   11,
 			protoType: ProtoTypeUint32,
 		},
 		&PrimitiveField{
 			fieldName: "OriginalPayloadFormat",
+			protoID:   12,
 			protoType: ProtoTypeString,
 		},
 		&SliceField{
 			fieldName:   "OriginalPayload",
+			protoID:     13,
+			protoType:   ProtoTypeBytes,
 			returnSlice: byteSlice,
 		},
 		&SliceField{
 			fieldName:   "AttributeIndices",
+			protoID:     14,
+			protoType:   ProtoTypeInt32,
 			returnSlice: int32Slice,
 		},
 	},
@@ -235,10 +290,12 @@ var attributeUnit = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "AttributeKeyStrindex",
+			protoID:   1,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "UnitStrindex",
+			protoID:   2,
 			protoType: ProtoTypeInt32,
 		},
 	},
@@ -254,8 +311,18 @@ var link = &messageStruct{
 	description:    "// Link represents a pointer from a profile Sample to a trace Span.",
 	originFullName: "otlpprofiles.Link",
 	fields: []Field{
-		traceIDField,
-		spanIDField,
+		&TypedField{
+			fieldName:       "TraceID",
+			originFieldName: "TraceId",
+			protoID:         1,
+			returnType:      traceIDType,
+		},
+		&TypedField{
+			fieldName:       "SpanID",
+			originFieldName: "SpanId",
+			protoID:         2,
+			returnType:      spanIDType,
+		},
 	},
 }
 
@@ -271,20 +338,23 @@ var valueType = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "TypeStrindex",
+			protoID:   1,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "UnitStrindex",
+			protoID:   2,
 			protoType: ProtoTypeInt32,
 		},
 		&TypedField{
 			fieldName: "AggregationTemporality",
+			protoID:   3,
 			returnType: &TypedType{
-				structName: "AggregationTemporality",
-				rawType:    "otlpprofiles.AggregationTemporality",
-				isEnum:     true,
-				defaultVal: "otlpprofiles.AggregationTemporality(0)",
-				testVal:    "otlpprofiles.AggregationTemporality(1)",
+				structName:  "AggregationTemporality",
+				protoType:   ProtoTypeEnum,
+				messageName: "otlpprofiles.AggregationTemporality",
+				defaultVal:  "otlpprofiles.AggregationTemporality(0)",
+				testVal:     "otlpprofiles.AggregationTemporality(1)",
 			},
 		},
 	},
@@ -302,26 +372,35 @@ var sample = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "LocationsStartIndex",
+			protoID:   1,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "LocationsLength",
+			protoID:   2,
 			protoType: ProtoTypeInt32,
 		},
 		&SliceField{
 			fieldName:   "Value",
+			protoID:     3,
+			protoType:   ProtoTypeInt64,
 			returnSlice: int64Slice,
 		},
 		&SliceField{
 			fieldName:   "AttributeIndices",
+			protoID:     4,
+			protoType:   ProtoTypeInt32,
 			returnSlice: int32Slice,
 		},
 		&OptionalPrimitiveField{
 			fieldName: "LinkIndex",
+			protoID:   5,
 			protoType: ProtoTypeInt32,
 		},
 		&SliceField{
 			fieldName:   "TimestampsUnixNano",
+			protoID:     6,
+			protoType:   ProtoTypeUint64,
 			returnSlice: uInt64Slice,
 		},
 	},
@@ -339,38 +418,48 @@ var mapping = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "MemoryStart",
+			protoID:   1,
 			protoType: ProtoTypeUint64,
 		},
 		&PrimitiveField{
 			fieldName: "MemoryLimit",
+			protoID:   2,
 			protoType: ProtoTypeUint64,
 		},
 		&PrimitiveField{
 			fieldName: "FileOffset",
+			protoID:   3,
 			protoType: ProtoTypeUint64,
 		},
 		&PrimitiveField{
 			fieldName: "FilenameStrindex",
+			protoID:   4,
 			protoType: ProtoTypeInt32,
 		},
 		&SliceField{
 			fieldName:   "AttributeIndices",
+			protoID:     15,
+			protoType:   ProtoTypeInt32,
 			returnSlice: int32Slice,
 		},
 		&PrimitiveField{
 			fieldName: "HasFunctions",
+			protoID:   6,
 			protoType: ProtoTypeBool,
 		},
 		&PrimitiveField{
 			fieldName: "HasFilenames",
+			protoID:   7,
 			protoType: ProtoTypeBool,
 		},
 		&PrimitiveField{
 			fieldName: "HasLineNumbers",
+			protoID:   8,
 			protoType: ProtoTypeBool,
 		},
 		&PrimitiveField{
 			fieldName: "HasInlineFrames",
+			protoID:   9,
 			protoType: ProtoTypeBool,
 		},
 	},
@@ -388,22 +477,29 @@ var location = &messageStruct{
 	fields: []Field{
 		&OptionalPrimitiveField{
 			fieldName: "MappingIndex",
+			protoID:   1,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "Address",
+			protoID:   2,
 			protoType: ProtoTypeUint64,
 		},
 		&SliceField{
 			fieldName:   "Line",
+			protoID:     3,
+			protoType:   ProtoTypeMessage,
 			returnSlice: lineSlice,
 		},
 		&PrimitiveField{
 			fieldName: "IsFolded",
+			protoID:   4,
 			protoType: ProtoTypeBool,
 		},
 		&SliceField{
 			fieldName:   "AttributeIndices",
+			protoID:     5,
+			protoType:   ProtoTypeInt32,
 			returnSlice: int32Slice,
 		},
 	},
@@ -421,14 +517,17 @@ var line = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "FunctionIndex",
+			protoID:   1,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "Line",
+			protoID:   2,
 			protoType: ProtoTypeInt64,
 		},
 		&PrimitiveField{
 			fieldName: "Column",
+			protoID:   3,
 			protoType: ProtoTypeInt64,
 		},
 	},
@@ -446,18 +545,22 @@ var function = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "NameStrindex",
+			protoID:   1,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "SystemNameStrindex",
+			protoID:   2,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "FilenameStrindex",
+			protoID:   3,
 			protoType: ProtoTypeInt32,
 		},
 		&PrimitiveField{
 			fieldName: "StartLine",
+			protoID:   4,
 			protoType: ProtoTypeInt64,
 		},
 	},
@@ -475,10 +578,12 @@ var attribute = &messageStruct{
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "Key",
+			protoID:   1,
 			protoType: ProtoTypeString,
 		},
 		&MessageField{
 			fieldName:     "Value",
+			protoID:       2,
 			returnMessage: anyValue,
 		},
 	},
