diff --git a/confmap/internal/e2e/go.mod b/confmap/internal/e2e/go.mod
index 37622cd80..705f96795 100644
--- a/confmap/internal/e2e/go.mod
+++ b/confmap/internal/e2e/go.mod
@@ -5,8 +5,8 @@ go 1.21.0
 require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/confmap v0.105.0
-	go.opentelemetry.io/collector/confmap/provider/envprovider v0.104.0
-	go.opentelemetry.io/collector/confmap/provider/fileprovider v0.104.0
+	go.opentelemetry.io/collector/confmap/provider/envprovider v0.105.0
+	go.opentelemetry.io/collector/confmap/provider/fileprovider v0.105.0
 	go.opentelemetry.io/collector/featuregate v1.12.0
 	go.opentelemetry.io/collector/internal/globalgates v0.105.0
 )
