diff --git a/cmd/mdatagen/internal/samplereceiver/internal/metadata/generated_logs.go b/cmd/mdatagen/internal/samplereceiver/internal/metadata/generated_logs.go
index edd2bdc9e..ea414e8c6 100644
--- a/cmd/mdatagen/internal/samplereceiver/internal/metadata/generated_logs.go
+++ b/cmd/mdatagen/internal/samplereceiver/internal/metadata/generated_logs.go
@@ -15,29 +15,54 @@ import (
 	"go.opentelemetry.io/collector/receiver"
 )
 
+type EventAttributeOption interface {
+	apply(plog.LogRecord)
+}
+
+type eventAttributeOptionFunc func(plog.LogRecord)
+
+func (eaof eventAttributeOptionFunc) apply(lr plog.LogRecord) {
+	eaof(lr)
+}
+
+func WithOptionalIntAttrEventAttribute(optionalIntAttrAttributeValue int64) EventAttributeOption {
+	return eventAttributeOptionFunc(func(dp plog.LogRecord) {
+		dp.Attributes().PutInt("optional_int_attr", optionalIntAttrAttributeValue)
+	})
+}
+
+func WithOptionalStringAttrEventAttribute(optionalStringAttrAttributeValue string) EventAttributeOption {
+	return eventAttributeOptionFunc(func(dp plog.LogRecord) {
+		dp.Attributes().PutStr("optional_string_attr", optionalStringAttrAttributeValue)
+	})
+}
+
 type eventDefaultEvent struct {
 	data   plog.LogRecordSlice // data buffer for generated log records.
 	config EventConfig         // event config provided by user.
 }
 
-func (e *eventDefaultEvent) recordEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, overriddenIntAttrAttributeValue int64, enumAttrAttributeValue string, sliceAttrAttributeValue []any, mapAttrAttributeValue map[string]any) {
+func (e *eventDefaultEvent) recordEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, overriddenIntAttrAttributeValue int64, enumAttrAttributeValue string, sliceAttrAttributeValue []any, mapAttrAttributeValue map[string]any, options ...EventAttributeOption) {
 	if !e.config.Enabled {
 		return
 	}
-	lr := e.data.AppendEmpty()
-	lr.SetEventName("default.event")
-	lr.SetTimestamp(timestamp)
+	dp := e.data.AppendEmpty()
+	dp.SetEventName("default.event")
+	dp.SetTimestamp(timestamp)
 
 	if span := trace.SpanContextFromContext(ctx); span.IsValid() {
-		lr.SetTraceID(pcommon.TraceID(span.TraceID()))
-		lr.SetSpanID(pcommon.SpanID(span.SpanID()))
+		dp.SetTraceID(pcommon.TraceID(span.TraceID()))
+		dp.SetSpanID(pcommon.SpanID(span.SpanID()))
 	}
+	dp.Attributes().PutStr("string_attr", stringAttrAttributeValue)
+	dp.Attributes().PutInt("state", overriddenIntAttrAttributeValue)
+	dp.Attributes().PutStr("enum_attr", enumAttrAttributeValue)
+	dp.Attributes().PutEmptySlice("slice_attr").FromRaw(sliceAttrAttributeValue)
+	dp.Attributes().PutEmptyMap("map_attr").FromRaw(mapAttrAttributeValue)
 
-	lr.Attributes().PutStr("string_attr", stringAttrAttributeValue)
-	lr.Attributes().PutInt("state", overriddenIntAttrAttributeValue)
-	lr.Attributes().PutStr("enum_attr", enumAttrAttributeValue)
-	lr.Attributes().PutEmptySlice("slice_attr").FromRaw(sliceAttrAttributeValue)
-	lr.Attributes().PutEmptyMap("map_attr").FromRaw(mapAttrAttributeValue)
+	for _, op := range options {
+		op.apply(dp)
+	}
 }
 
 // emit appends recorded event data to a events slice and prepares it for recording another set of log records.
@@ -64,20 +89,20 @@ func (e *eventDefaultEventToBeRemoved) recordEvent(ctx context.Context, timestam
 	if !e.config.Enabled {
 		return
 	}
-	lr := e.data.AppendEmpty()
-	lr.SetEventName("default.event.to_be_removed")
-	lr.SetTimestamp(timestamp)
+	dp := e.data.AppendEmpty()
+	dp.SetEventName("default.event.to_be_removed")
+	dp.SetTimestamp(timestamp)
 
 	if span := trace.SpanContextFromContext(ctx); span.IsValid() {
-		lr.SetTraceID(pcommon.TraceID(span.TraceID()))
-		lr.SetSpanID(pcommon.SpanID(span.SpanID()))
+		dp.SetTraceID(pcommon.TraceID(span.TraceID()))
+		dp.SetSpanID(pcommon.SpanID(span.SpanID()))
 	}
+	dp.Attributes().PutStr("string_attr", stringAttrAttributeValue)
+	dp.Attributes().PutInt("state", overriddenIntAttrAttributeValue)
+	dp.Attributes().PutStr("enum_attr", enumAttrAttributeValue)
+	dp.Attributes().PutEmptySlice("slice_attr").FromRaw(sliceAttrAttributeValue)
+	dp.Attributes().PutEmptyMap("map_attr").FromRaw(mapAttrAttributeValue)
 
-	lr.Attributes().PutStr("string_attr", stringAttrAttributeValue)
-	lr.Attributes().PutInt("state", overriddenIntAttrAttributeValue)
-	lr.Attributes().PutStr("enum_attr", enumAttrAttributeValue)
-	lr.Attributes().PutEmptySlice("slice_attr").FromRaw(sliceAttrAttributeValue)
-	lr.Attributes().PutEmptyMap("map_attr").FromRaw(mapAttrAttributeValue)
 }
 
 // emit appends recorded event data to a events slice and prepares it for recording another set of log records.
@@ -100,22 +125,25 @@ type eventDefaultEventToBeRenamed struct {
 	config EventConfig         // event config provided by user.
 }
 
-func (e *eventDefaultEventToBeRenamed) recordEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, booleanAttrAttributeValue bool, booleanAttr2AttributeValue bool) {
+func (e *eventDefaultEventToBeRenamed) recordEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, booleanAttrAttributeValue bool, booleanAttr2AttributeValue bool, options ...EventAttributeOption) {
 	if !e.config.Enabled {
 		return
 	}
-	lr := e.data.AppendEmpty()
-	lr.SetEventName("default.event.to_be_renamed")
-	lr.SetTimestamp(timestamp)
+	dp := e.data.AppendEmpty()
+	dp.SetEventName("default.event.to_be_renamed")
+	dp.SetTimestamp(timestamp)
 
 	if span := trace.SpanContextFromContext(ctx); span.IsValid() {
-		lr.SetTraceID(pcommon.TraceID(span.TraceID()))
-		lr.SetSpanID(pcommon.SpanID(span.SpanID()))
+		dp.SetTraceID(pcommon.TraceID(span.TraceID()))
+		dp.SetSpanID(pcommon.SpanID(span.SpanID()))
 	}
+	dp.Attributes().PutStr("string_attr", stringAttrAttributeValue)
+	dp.Attributes().PutBool("boolean_attr", booleanAttrAttributeValue)
+	dp.Attributes().PutBool("boolean_attr2", booleanAttr2AttributeValue)
 
-	lr.Attributes().PutStr("string_attr", stringAttrAttributeValue)
-	lr.Attributes().PutBool("boolean_attr", booleanAttrAttributeValue)
-	lr.Attributes().PutBool("boolean_attr2", booleanAttr2AttributeValue)
+	for _, op := range options {
+		op.apply(dp)
+	}
 }
 
 // emit appends recorded event data to a events slice and prepares it for recording another set of log records.
@@ -314,8 +342,8 @@ func (lb *LogsBuilder) Emit(options ...ResourceLogsOption) plog.Logs {
 }
 
 // RecordDefaultEventEvent adds a log record of default.event event.
-func (lb *LogsBuilder) RecordDefaultEventEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, overriddenIntAttrAttributeValue int64, enumAttrAttributeValue AttributeEnumAttr, sliceAttrAttributeValue []any, mapAttrAttributeValue map[string]any) {
-	lb.eventDefaultEvent.recordEvent(ctx, timestamp, stringAttrAttributeValue, overriddenIntAttrAttributeValue, enumAttrAttributeValue.String(), sliceAttrAttributeValue, mapAttrAttributeValue)
+func (lb *LogsBuilder) RecordDefaultEventEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, overriddenIntAttrAttributeValue int64, enumAttrAttributeValue AttributeEnumAttr, sliceAttrAttributeValue []any, mapAttrAttributeValue map[string]any, options ...EventAttributeOption) {
+	lb.eventDefaultEvent.recordEvent(ctx, timestamp, stringAttrAttributeValue, overriddenIntAttrAttributeValue, enumAttrAttributeValue.String(), sliceAttrAttributeValue, mapAttrAttributeValue, options...)
 }
 
 // RecordDefaultEventToBeRemovedEvent adds a log record of default.event.to_be_removed event.
@@ -324,6 +352,6 @@ func (lb *LogsBuilder) RecordDefaultEventToBeRemovedEvent(ctx context.Context, t
 }
 
 // RecordDefaultEventToBeRenamedEvent adds a log record of default.event.to_be_renamed event.
-func (lb *LogsBuilder) RecordDefaultEventToBeRenamedEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, booleanAttrAttributeValue bool, booleanAttr2AttributeValue bool) {
-	lb.eventDefaultEventToBeRenamed.recordEvent(ctx, timestamp, stringAttrAttributeValue, booleanAttrAttributeValue, booleanAttr2AttributeValue)
+func (lb *LogsBuilder) RecordDefaultEventToBeRenamedEvent(ctx context.Context, timestamp pcommon.Timestamp, stringAttrAttributeValue string, booleanAttrAttributeValue bool, booleanAttr2AttributeValue bool, options ...EventAttributeOption) {
+	lb.eventDefaultEventToBeRenamed.recordEvent(ctx, timestamp, stringAttrAttributeValue, booleanAttrAttributeValue, booleanAttr2AttributeValue, options...)
 }
