diff --git a/testbed/tests/trace_test.go b/testbed/tests/trace_test.go
index 52c25db9c..9fb65b978 100644
--- a/testbed/tests/trace_test.go
+++ b/testbed/tests/trace_test.go
@@ -61,7 +61,7 @@ func TestTrace10kSPS(t *testing.T) {
 			testbed.NewOCDataReceiver(testbed.GetAvailablePort(t)),
 			testbed.ResourceSpec{
 				ExpectedMaxCPU: 39,
-				ExpectedMaxRAM: 70,
+				ExpectedMaxRAM: 82,
 			},
 		},
 		{
