diff --git a/service/builder/exporters_builder.go b/service/builder/exporters_builder.go
index 38afb7a0e..7e9d0cce7 100644
--- a/service/builder/exporters_builder.go
+++ b/service/builder/exporters_builder.go
@@ -232,11 +232,7 @@ func (eb *ExportersBuilder) buildExporter(
 
 	inputDataTypes := exportersInputDataTypes[config]
 	if inputDataTypes == nil {
-		// TODO  https://go.opentelemetry.io/collector/issues/294
-		// Move this validation to config/config.go:validateConfig
-		// No data types where requested for this exporter. This can only happen
-		// if there are no pipelines associated with the exporter.
-		logger.Warn("Exporter is not associated with any pipeline and will not export data.")
+		logger.Info("Ignoring exporter as it is not used by any pipeline")
 		return exporter, nil
 	}
 
