diff --git a/pdata/pprofile/pprofileotlp/request_test.go b/pdata/pprofile/pprofileotlp/request_test.go
index 799543170..ac3a933c9 100644
--- a/pdata/pprofile/pprofileotlp/request_test.go
+++ b/pdata/pprofile/pprofileotlp/request_test.go
@@ -70,10 +70,10 @@ func TestProfilesProtoWireCompatibility(t *testing.T) {
 	// this repository are wire compatible.
 
 	// Generate Profiles as pdata struct.
-	td := NewExportRequestFromProfiles(pprofile.Profiles(internal.GenerateTestProfiles()))
+	pd := NewExportRequestFromProfiles(pprofile.Profiles(internal.NewProfiles(internal.GenTestOrigExportProfilesServiceRequest(), internal.NewState())))
 
 	// Marshal its underlying ProtoBuf to wire.
-	wire1, err := td.MarshalProto()
+	wire1, err := pd.MarshalProto()
 	require.NoError(t, err)
 	assert.NotNil(t, wire1)
 
@@ -94,5 +94,5 @@ func TestProfilesProtoWireCompatibility(t *testing.T) {
 
 	// Now compare that the original and final ProtoBuf messages are the same.
 	// This proves that goproto and gogoproto marshaling/unmarshaling are wire compatible.
-	assert.Equal(t, td, td2)
+	assert.Equal(t, pd, td2)
 }
