diff --git a/testbed/testbed/test_case.go b/testbed/testbed/test_case.go
index 058aa73bc..cba059e77 100644
--- a/testbed/testbed/test_case.go
+++ b/testbed/testbed/test_case.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -197,9 +197,9 @@ func (tc *TestCase) Stop() {
 
 	var result string
 	if tc.t.Failed() {
-		result = "Failed"
+		result = "FAIL"
 	} else {
-		result = "Passed"
+		result = "PASS"
 	}
 
 	results.Add(tc.t.Name(), &TestResult{
