diff --git a/exporter/exporterhelper/internal/bounded_queue.go b/exporter/exporterhelper/internal/bounded_queue.go
index 6bb87cdfe..ff1598f29 100644
--- a/exporter/exporterhelper/internal/bounded_queue.go
+++ b/exporter/exporterhelper/internal/bounded_queue.go
@@ -19,10 +19,8 @@ package internal
 import (
 	"sync"
 	"sync/atomic"
-	"time"
 	"unsafe"
 
-	"github.com/uber/jaeger-lib/metrics"
 	uatomic "go.uber.org/atomic"
 )
 
@@ -162,24 +160,6 @@ func (q *BoundedQueue) Capacity() int {
 	return int(q.capacity.Load())
 }
 
-// StartLengthReporting starts a timer-based goroutine that periodically reports
-// current queue length to a given metrics gauge.
-func (q *BoundedQueue) StartLengthReporting(reportPeriod time.Duration, gauge metrics.Gauge) {
-	ticker := time.NewTicker(reportPeriod)
-	go func() {
-		defer ticker.Stop()
-		for {
-			select {
-			case <-ticker.C:
-				size := q.Size()
-				gauge.Update(int64(size))
-			case <-q.stopCh:
-				return
-			}
-		}
-	}()
-}
-
 // Resize changes the capacity of the queue, returning whether the action was successful
 func (q *BoundedQueue) Resize(capacity int) bool {
 	if capacity == q.Capacity() {
