diff --git a/exporter/exporterhelper/internal/base_exporter.go b/exporter/exporterhelper/internal/base_exporter.go
index 0155cb532..d7f931364 100644
--- a/exporter/exporterhelper/internal/base_exporter.go
+++ b/exporter/exporterhelper/internal/base_exporter.go
@@ -340,7 +340,7 @@ func WithBatchFuncs(mf exporterbatcher.BatchMergeFunc[internal.Request], msf exp
 func CheckStatus(t *testing.T, sd sdktrace.ReadOnlySpan, err error) {
 	if err != nil {
 		require.Equal(t, codes.Error, sd.Status().Code, "SpanData %v", sd)
-		require.Equal(t, err.Error(), sd.Status().Description, "SpanData %v", sd)
+		require.EqualError(t, err, sd.Status().Description, "SpanData %v", sd)
 	} else {
 		require.Equal(t, codes.Unset, sd.Status().Code, "SpanData %v", sd)
 	}
