diff --git a/internal/cmd/pdatagen/internal/pprofile_package.go b/internal/cmd/pdatagen/internal/pprofile_package.go
index b170b5ea9..4ef4f7781 100644
--- a/internal/cmd/pdatagen/internal/pprofile_package.go
+++ b/internal/cmd/pdatagen/internal/pprofile_package.go
@@ -9,6 +9,7 @@ var pprofile = &Package{
 		path: "pprofile",
 		imports: []string{
 			`"encoding/binary"`,
+			`"fmt"`,
 			`"iter"`,
 			`"math"`,
 			`"sort"`,
@@ -17,7 +18,8 @@ var pprofile = &Package{
 			`"go.opentelemetry.io/collector/pdata/internal/data"`,
 			`"go.opentelemetry.io/collector/pdata/internal/json"`,
 			`"go.opentelemetry.io/collector/pdata/internal/proto"`,
-			`otlpcollectorprofile "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/profiles/v1development"`,
+			`otlpcollectorprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/profiles/v1development"`,
+			`otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"`,
 			`otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"`,
 			`"go.opentelemetry.io/collector/pdata/pcommon"`,
 		},
@@ -27,10 +29,14 @@ var pprofile = &Package{
 			``,
 			`"github.com/stretchr/testify/assert"`,
 			`"github.com/stretchr/testify/require"`,
+			`"google.golang.org/protobuf/proto"`,
+			`gootlpcollectorprofiles "go.opentelemetry.io/proto/slim/otlp/collector/profiles/v1development"`,
+			`gootlpcommon "go.opentelemetry.io/proto/slim/otlp/common/v1"`,
+			`gootlpprofiles "go.opentelemetry.io/proto/slim/otlp/profiles/v1development"`,
 			``,
 			`"go.opentelemetry.io/collector/pdata/internal"`,
 			`"go.opentelemetry.io/collector/pdata/internal/json"`,
-			`otlpcollectorprofile "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/profiles/v1development"`,
+			`otlpcollectorprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/profiles/v1development"`,
 			`otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"`,
 			`"go.opentelemetry.io/collector/pdata/pcommon"`,
 		},
@@ -68,7 +74,7 @@ var pprofile = &Package{
 var profiles = &messageStruct{
 	structName:     "Profiles",
 	description:    "// Profiles is the top-level struct that is propagated through the profiles pipeline.\n// Use NewProfiles to create new instance, zero-initialized instance is not valid for use.",
-	originFullName: "otlpcollectorprofile.ExportProfilesServiceRequest",
+	originFullName: "otlpcollectorprofiles.ExportProfilesServiceRequest",
 	fields: []Field{
 		&SliceField{
 			fieldName:   "ResourceProfiles",
@@ -600,7 +606,7 @@ var attributeTableSlice = &sliceOfValues{
 var attribute = &messageStruct{
 	structName:     "Attribute",
 	description:    "// Attribute describes an attribute stored in a profile's attribute table.",
-	originFullName: "v1.KeyValue",
+	originFullName: "otlpcommon.KeyValue",
 	fields: []Field{
 		&PrimitiveField{
 			fieldName: "Key",
