diff --git a/internal/memorylimiter/iruntime/total_memory_linux_test.go b/internal/memorylimiter/iruntime/total_memory_linux_test.go
new file mode 100644
index 000000000..3e05aee21
--- /dev/null
+++ b/internal/memorylimiter/iruntime/total_memory_linux_test.go
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build linux
+
+package iruntime
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTotalMemory(t *testing.T) {
+	totalMemory, err := TotalMemory()
+	require.NoError(t, err)
+	assert.Positive(t, totalMemory)
+}
