diff --git a/config/confighttp/confighttp.go b/config/confighttp/confighttp.go
index 4fed24e08..b4a7acb60 100644
--- a/config/confighttp/confighttp.go
+++ b/config/confighttp/confighttp.go
@@ -40,6 +40,10 @@ type HTTPClientSettings struct {
 
 	// Timeout parameter configures `http.Client.Timeout`.
 	Timeout time.Duration `mapstructure:"timeout,omitempty"`
+
+	// Additional headers attached to each HTTP request sent by the client.
+	// Existing header values are overwritten if collision happens.
+	Headers map[string]string `mapstructure:"headers,omitempty"`
 }
 
 func (hcs *HTTPClientSettings) ToClient() (*http.Client, error) {
@@ -57,12 +61,40 @@ func (hcs *HTTPClientSettings) ToClient() (*http.Client, error) {
 	if hcs.WriteBufferSize > 0 {
 		transport.WriteBufferSize = hcs.WriteBufferSize
 	}
+	var clientTransport http.RoundTripper
+
+	if hcs.Headers != nil && len(hcs.Headers) > 0 {
+		clientTransport = &clientInterceptorRoundTripper{
+			transport: transport,
+			headers:   hcs.Headers,
+		}
+	} else {
+		clientTransport = transport
+	}
+
 	return &http.Client{
-		Transport: transport,
+		Transport: clientTransport,
 		Timeout:   hcs.Timeout,
 	}, nil
 }
 
+// Custom RoundTripper that add headers
+type clientInterceptorRoundTripper struct {
+	transport http.RoundTripper
+	headers   map[string]string
+}
+
+// Custom RoundTrip that add headers
+func (interceptor *clientInterceptorRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	for k, v := range interceptor.headers {
+		req.Header.Set(k, v)
+	}
+	// Send the request to Cortex
+	response, err := interceptor.transport.RoundTrip(req)
+
+	return response, err
+}
+
 type HTTPServerSettings struct {
 	// Endpoint configures the listening address for the server.
 	Endpoint string `mapstructure:"endpoint"`
