diff --git a/observability/observability.go b/observability/observability.go
index ad5ec734b..9bc79d53e 100644
--- a/observability/observability.go
+++ b/observability/observability.go
@@ -31,18 +31,18 @@ import (
 )
 
 var (
-	mReceiverReceivedSpans = stats.Int64("oc.io/receiver/received_spans", "Counts the number of spans received by the receiver", "1")
-	mReceiverDroppedSpans  = stats.Int64("oc.io/receiver/dropped_spans", "Counts the number of spans dropped by the receiver", "1")
+	mReceiverReceivedSpans = stats.Int64("otelsvc/receiver/received_spans", "Counts the number of spans received by the receiver", "1")
+	mReceiverDroppedSpans  = stats.Int64("otelsvc/receiver/dropped_spans", "Counts the number of spans dropped by the receiver", "1")
 
-	mExporterReceivedSpans = stats.Int64("oc.io/exporter/received_spans", "Counts the number of spans received by the exporter", "1")
-	mExporterDroppedSpans  = stats.Int64("oc.io/exporter/dropped_spans", "Counts the number of spans received by the exporter", "1")
+	mExporterReceivedSpans = stats.Int64("otelsvc/exporter/received_spans", "Counts the number of spans received by the exporter", "1")
+	mExporterDroppedSpans  = stats.Int64("otelsvc/exporter/dropped_spans", "Counts the number of spans received by the exporter", "1")
 )
 
 // TagKeyReceiver defines tag key for Receiver.
-var TagKeyReceiver, _ = tag.NewKey("oc_receiver")
+var TagKeyReceiver, _ = tag.NewKey("otelsvc_receiver")
 
 // TagKeyExporter defines tag key for Exporter.
-var TagKeyExporter, _ = tag.NewKey("oc_exporter")
+var TagKeyExporter, _ = tag.NewKey("otelsvc_exporter")
 
 // ViewReceiverReceivedSpans defines the view for the receiver received spans metric.
 var ViewReceiverReceivedSpans = &view.View{
@@ -88,31 +88,31 @@ var AllViews = []*view.View{
 	ViewExporterDroppedSpans,
 }
 
-// ContextWithReceiverName adds the tag "oc_receiver" and the name of the receiver as the value,
+// ContextWithReceiverName adds the tag "otelsvc_receiver" and the name of the receiver as the value,
 // and returns the newly created context. For receivers that can receive multiple signals it is
 // recommended to encode the signal as suffix (e.g. "oc_trace" and "oc_metrics").
 func ContextWithReceiverName(ctx context.Context, receiverName string) context.Context {
-	ctx, _ = tag.New(ctx, tag.Upsert(TagKeyReceiver, receiverName))
+	ctx, _ = tag.New(ctx, tag.Upsert(TagKeyReceiver, receiverName, tag.WithTTL(tag.TTLNoPropagation)))
 	return ctx
 }
 
-// RecordTraceReceiverMetrics records the number of the spans received and dropped by the receiver.
+// RecordMetricsForTraceReceiver records the number of the spans received and dropped by the receiver.
 // Use it with a context.Context generated using ContextWithReceiverName().
-func RecordTraceReceiverMetrics(ctxWithTraceReceiverName context.Context, receivedSpans int, droppedSpans int) {
+func RecordMetricsForTraceReceiver(ctxWithTraceReceiverName context.Context, receivedSpans int, droppedSpans int) {
 	stats.Record(ctxWithTraceReceiverName, mReceiverReceivedSpans.M(int64(receivedSpans)), mReceiverDroppedSpans.M(int64(droppedSpans)))
 }
 
-// ContextWithExporterName adds the tag "oc_exporter" and the name of the exporter as the value,
+// ContextWithExporterName adds the tag "otelsvc_exporter" and the name of the exporter as the value,
 // and returns the newly created context. For exporters that can export multiple signals it is
 // recommended to encode the signal as suffix (e.g. "oc_trace" and "oc_metrics").
 func ContextWithExporterName(ctx context.Context, exporterName string) context.Context {
-	ctx, _ = tag.New(ctx, tag.Upsert(TagKeyExporter, exporterName))
+	ctx, _ = tag.New(ctx, tag.Upsert(TagKeyExporter, exporterName, tag.WithTTL(tag.TTLNoPropagation)))
 	return ctx
 }
 
-// RecordTraceExporterMetrics records the number of the spans received and dropped by the exporter.
+// RecordMetricsForTraceExporter records the number of the spans received and dropped by the exporter.
 // Use it with a context.Context generated using ContextWithExporterName().
-func RecordTraceExporterMetrics(ctx context.Context, receivedSpans int, droppedSpans int) {
+func RecordMetricsForTraceExporter(ctx context.Context, receivedSpans int, droppedSpans int) {
 	stats.Record(ctx, mExporterReceivedSpans.M(int64(receivedSpans)), mExporterDroppedSpans.M(int64(droppedSpans)))
 }
 
