diff --git a/receiver/opencensusreceiver/factory_test.go b/receiver/opencensusreceiver/factory_test.go
index e08c525e1..5d17d6c1f 100644
--- a/receiver/opencensusreceiver/factory_test.go
+++ b/receiver/opencensusreceiver/factory_test.go
@@ -115,9 +115,9 @@ func TestCreateTraceReceiver(t *testing.T) {
 			}
 			if tr != nil {
 				mh := component.NewMockHost()
-				err := tr.Start(mh)
+				err := tr.Start(context.Background(), mh)
 				require.NoError(t, err, "Start() error = %v", err)
-				tr.Shutdown()
+				tr.Shutdown(context.Background())
 			}
 		})
 	}
@@ -189,9 +189,9 @@ func TestCreateMetricReceiver(t *testing.T) {
 			}
 			if tc != nil {
 				mh := component.NewMockHost()
-				err := tc.Start(mh)
+				err := tc.Start(context.Background(), mh)
 				require.NoError(t, err, "Start() error = %v", err)
-				tc.Shutdown()
+				tc.Shutdown(context.Background())
 			}
 		})
 	}
