diff --git a/receiver/internal/factory.go b/receiver/internal/factory.go
index 0b6cb17c7..24d05785f 100644
--- a/receiver/internal/factory.go
+++ b/receiver/internal/factory.go
@@ -18,24 +18,27 @@ type Factory interface {
 	component.Factory
 
 	// CreateTracesReceiver creates a TracesReceiver based on this config.
-	// If the receiver type does not support tracing or if the config is not valid
-	// an error will be returned instead. `nextConsumer` is never nil.
+	// If the receiver type does not support traces,
+	// this function returns the error [component.ErrDataTypeIsNotSupported].
+	// Implementers can assume `nextConsumer` is never nil.
 	CreateTracesReceiver(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Traces) (Traces, error)
 
 	// TracesReceiverStability gets the stability level of the TracesReceiver.
 	TracesReceiverStability() component.StabilityLevel
 
 	// CreateMetricsReceiver creates a MetricsReceiver based on this config.
-	// If the receiver type does not support metrics or if the config is not valid
-	// an error will be returned instead. `nextConsumer` is never nil.
+	// If the receiver type does not support metrics,
+	// this function returns the error [component.ErrDataTypeIsNotSupported].
+	// Implementers can assume `nextConsumer` is never nil.
 	CreateMetricsReceiver(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Metrics) (Metrics, error)
 
 	// MetricsReceiverStability gets the stability level of the MetricsReceiver.
 	MetricsReceiverStability() component.StabilityLevel
 
 	// CreateLogsReceiver creates a LogsReceiver based on this config.
-	// If the receiver type does not support the data type or if the config is not valid
-	// an error will be returned instead. `nextConsumer` is never nil.
+	// If the receiver type does not support logs,
+	// this function returns the error [component.ErrDataTypeIsNotSupported].
+	// Implementers can assume `nextConsumer` is never nil.
 	CreateLogsReceiver(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Logs) (Logs, error)
 
 	// LogsReceiverStability gets the stability level of the LogsReceiver.
