diff --git a/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_others_test.go b/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_others_test.go
index 516e233e6..c53dbbc9d 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_others_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_others_test.go
@@ -26,7 +26,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/component/componenttest"
-	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
 )
 
 func TestScrape_Others(t *testing.T) {
@@ -59,10 +59,10 @@ func TestScrape_Others(t *testing.T) {
 			_, err = scraper.scrape(context.Background())
 			assert.EqualError(t, err, test.expectedErr)
 
-			isPartial := consumererror.IsPartialScrapeError(err)
+			isPartial := scrapererror.IsPartialScrapeError(err)
 			assert.True(t, isPartial)
 			if isPartial {
-				assert.Equal(t, metricsLen, err.(consumererror.PartialScrapeError).Failed)
+				assert.Equal(t, metricsLen, err.(scrapererror.PartialScrapeError).Failed)
 			}
 		})
 	}
