diff --git a/service/service.go b/service/service.go
index 5874b72f8..bf17b4183 100644
--- a/service/service.go
+++ b/service/service.go
@@ -26,6 +26,7 @@ import (
 	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/internal/globalgates"
 	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pipeline"
 	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/service/extensions"
@@ -34,6 +35,7 @@ import (
 	"go.opentelemetry.io/collector/service/internal/proctelemetry"
 	"go.opentelemetry.io/collector/service/internal/resource"
 	"go.opentelemetry.io/collector/service/internal/status"
+	"go.opentelemetry.io/collector/service/pipelines"
 	"go.opentelemetry.io/collector/service/telemetry"
 )
 
@@ -303,8 +305,19 @@ func (srv *Service) initExtensions(ctx context.Context, cfg extensions.Config) e
 	return nil
 }
 
+func convertFromComponentIDToPipelineID(id component.ID) pipeline.ID {
+	return pipeline.MustNewIDWithName(id.Type().String(), id.Name())
+}
+
 // Creates the pipeline graph.
 func (srv *Service) initGraph(ctx context.Context, cfg Config) error {
+	if len(cfg.Pipelines) > 0 {
+		cfg.PipelinesWithPipelineID = make(pipelines.ConfigWithPipelineID, len(cfg.Pipelines))
+		for k, v := range cfg.Pipelines {
+			cfg.PipelinesWithPipelineID[convertFromComponentIDToPipelineID(k)] = v
+		}
+	}
+
 	var err error
 	if srv.host.Pipelines, err = graph.Build(ctx, graph.Settings{
 		Telemetry:        srv.telemetrySettings,
@@ -313,7 +326,7 @@ func (srv *Service) initGraph(ctx context.Context, cfg Config) error {
 		ProcessorBuilder: srv.host.Processors,
 		ExporterBuilder:  srv.host.Exporters,
 		ConnectorBuilder: srv.host.Connectors,
-		PipelineConfigs:  cfg.Pipelines,
+		PipelineConfigs:  cfg.PipelinesWithPipelineID,
 		ReportStatus:     srv.host.Reporter.ReportStatus,
 	}); err != nil {
 		return fmt.Errorf("failed to build pipelines: %w", err)
