diff --git a/confmap/go.mod b/confmap/go.mod
index bc210f4c2..fb8d56623 100644
--- a/confmap/go.mod
+++ b/confmap/go.mod
@@ -3,7 +3,7 @@ module go.opentelemetry.io/collector/confmap
 go 1.21.0
 
 require (
-	github.com/go-viper/mapstructure/v2 v2.0.0-alpha.1
+	github.com/go-viper/mapstructure/v2 v2.0.0
 	github.com/knadh/koanf/maps v0.1.1
 	github.com/knadh/koanf/providers/confmap v0.1.0
 	github.com/knadh/koanf/v2 v2.1.1
