diff --git a/service/internal/graph/graph_test.go b/service/internal/graph/graph_test.go
index 8b98d020f..3593d4533 100644
--- a/service/internal/graph/graph_test.go
+++ b/service/internal/graph/graph_test.go
@@ -2260,12 +2260,12 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 	eSdErr := &testNode{id: component.MustNewIDWithName("e_sd_err", "1"), shutdownErr: assert.AnError}
 
 	instanceIDs := map[*testNode]*componentstatus.InstanceID{
-		rNoErr: {ID: rNoErr.id},
-		rStErr: {ID: rStErr.id},
-		rSdErr: {ID: rSdErr.id},
-		eNoErr: {ID: eNoErr.id},
-		eStErr: {ID: eStErr.id},
-		eSdErr: {ID: eSdErr.id},
+		rNoErr: componentstatus.NewInstanceID(rNoErr.id, component.KindReceiver),
+		rStErr: componentstatus.NewInstanceID(rStErr.id, component.KindReceiver),
+		rSdErr: componentstatus.NewInstanceID(rSdErr.id, component.KindReceiver),
+		eNoErr: componentstatus.NewInstanceID(eNoErr.id, component.KindExporter),
+		eStErr: componentstatus.NewInstanceID(eStErr.id, component.KindExporter),
+		eSdErr: componentstatus.NewInstanceID(eSdErr.id, component.KindExporter),
 	}
 
 	// compare two maps of status events ignoring timestamp
