diff --git a/processor/memorylimiter/internal/cgroups/subsys.go b/processor/memorylimiter/internal/cgroups/subsys.go
index b92119c50..97b9f11f6 100644
--- a/processor/memorylimiter/internal/cgroups/subsys.go
+++ b/processor/memorylimiter/internal/cgroups/subsys.go
@@ -41,6 +41,7 @@ package cgroups
 import (
 	"bufio"
 	"os"
+	"path/filepath"
 	"strconv"
 	"strings"
 )
@@ -91,7 +92,7 @@ func NewCGroupSubsysFromLine(line string) (*CGroupSubsys, error) {
 // parseCGroupSubsystems parses procPathCGroup (usually at `/proc/$PID/cgroup`)
 // and returns a new map[string]*CGroupSubsys.
 func parseCGroupSubsystems(procPathCGroup string) (map[string]*CGroupSubsys, error) {
-	cgroupFile, err := os.Open(procPathCGroup)
+	cgroupFile, err := os.Open(filepath.Clean(procPathCGroup))
 	if err != nil {
 		return nil, err
 	}
