diff --git a/cmd/mdatagen/templates/component_test.go.tmpl b/cmd/mdatagen/templates/component_test.go.tmpl
index 57090bb5c..8ec280966 100644
--- a/cmd/mdatagen/templates/component_test.go.tmpl
+++ b/cmd/mdatagen/templates/component_test.go.tmpl
@@ -108,20 +108,26 @@ func TestComponentLifecycle(t *testing.T) {
 			err = c.Start(context.Background(), host)
 			require.NoError(t, err)
 			require.NotPanics(t, func() {
-				switch e := c.(type) {
-				case exporter.Logs:
+				switch test.name {
+				case "logs":
+					e, ok := c.(exporter.Logs)
+					require.True(t, ok)				
 					logs := generateLifecycleTestLogs()
 					if !e.Capabilities().MutatesData {
 						logs.MarkReadOnly()
 					}
 					err = e.ConsumeLogs(context.Background(), logs)
-				case exporter.Metrics:
+				case "metrics":
+					e, ok := c.(exporter.Metrics)
+					require.True(t, ok)
 					metrics := generateLifecycleTestMetrics()
 					if !e.Capabilities().MutatesData {
 						metrics.MarkReadOnly()
 					}
 					err = e.ConsumeMetrics(context.Background(), metrics)
-				case exporter.Traces:
+				case "traces":
+					e, ok := c.(exporter.Traces)
+					require.True(t, ok)
 					traces := generateLifecycleTestTraces()
 					if !e.Capabilities().MutatesData {
 						traces.MarkReadOnly()
@@ -199,20 +205,26 @@ func TestComponentLifecycle(t *testing.T) {
 			err = c.Start(context.Background(), host)
 			require.NoError(t, err)
 			require.NotPanics(t, func() {
-				switch e := c.(type) {
-				case processor.Logs:
+				switch test.name {
+				case "logs":
+					e, ok := c.(processor.Logs)
+					require.True(t, ok)
 					logs := generateLifecycleTestLogs()
 					if !e.Capabilities().MutatesData {
 						logs.MarkReadOnly()
 					}
 					err = e.ConsumeLogs(context.Background(), logs)
-				case processor.Metrics:
+				case "metrics":
+					e, ok := c.(processor.Metrics)
+					require.True(t, ok)
 					metrics := generateLifecycleTestMetrics()
 					if !e.Capabilities().MutatesData {
 						metrics.MarkReadOnly()
 					}
 					err = e.ConsumeMetrics(context.Background(), metrics)
-				case processor.Traces:
+				case "traces":
+					e, ok := c.(processor.Traces)
+					require.True(t, ok)
 					traces := generateLifecycleTestTraces()
 					if !e.Capabilities().MutatesData {
 						traces.MarkReadOnly()
