diff --git a/exporter/factory_test.go b/exporter/factory_test.go
index d317bb16b..399aa8bd6 100644
--- a/exporter/factory_test.go
+++ b/exporter/factory_test.go
@@ -23,37 +23,37 @@ import (
 	"github.com/open-telemetry/opentelemetry-service/models"
 )
 
-type ExampleExporterFactory struct {
+type TestFactory struct {
 }
 
 // Type gets the type of the Exporter config created by this factory.
-func (f *ExampleExporterFactory) Type() string {
+func (f *TestFactory) Type() string {
 	return "exampleexporter"
 }
 
 // CreateDefaultConfig creates the default configuration for the Exporter.
-func (f *ExampleExporterFactory) CreateDefaultConfig() models.Exporter {
+func (f *TestFactory) CreateDefaultConfig() models.Exporter {
 	return nil
 }
 
 // CreateTraceExporter creates a trace exporter based on this config.
-func (f *ExampleExporterFactory) CreateTraceExporter(logger *zap.Logger, cfg models.Exporter) (consumer.TraceConsumer, StopFunc, error) {
+func (f *TestFactory) CreateTraceExporter(logger *zap.Logger, cfg models.Exporter) (consumer.TraceConsumer, StopFunc, error) {
 	return nil, nil, nil
 }
 
 // CreateMetricsExporter creates a metrics exporter based on this config.
-func (f *ExampleExporterFactory) CreateMetricsExporter(logger *zap.Logger, cfg models.Exporter) (consumer.MetricsConsumer, StopFunc, error) {
+func (f *TestFactory) CreateMetricsExporter(logger *zap.Logger, cfg models.Exporter) (consumer.MetricsConsumer, StopFunc, error) {
 	return nil, nil, nil
 }
 
-func TestRegisterExporterFactory(t *testing.T) {
-	f := ExampleExporterFactory{}
-	err := RegisterExporterFactory(&f)
+func TestRegisterFactory(t *testing.T) {
+	f := TestFactory{}
+	err := RegisterFactory(&f)
 	if err != nil {
 		t.Fatalf("cannot register factory")
 	}
 
-	if &f != GetExporterFactory(f.Type()) {
+	if &f != GetFactory(f.Type()) {
 		t.Fatalf("cannot find factory")
 	}
 
@@ -66,7 +66,7 @@ func TestRegisterExporterFactory(t *testing.T) {
 			}
 		}()
 
-		err = RegisterExporterFactory(&f)
+		err = RegisterFactory(&f)
 	}()
 
 	if !panicked {
