diff --git a/connector/connectorprofiles/profiles_router_test.go b/connector/connectorprofiles/profiles_router_test.go
index fcec379df..08dc2490c 100644
--- a/connector/connectorprofiles/profiles_router_test.go
+++ b/connector/connectorprofiles/profiles_router_test.go
@@ -120,8 +120,8 @@ func TestProfilessRouterConsumer(t *testing.T) {
 	assert.Len(t, rcs, 2)
 	assert.ElementsMatch(t, []component.ID{fooID, barID}, rcs)
 
-	assert.Len(t, foo.AllProfiles(), 0)
-	assert.Len(t, bar.AllProfiles(), 0)
+	assert.Empty(t, foo.AllProfiles())
+	assert.Empty(t, bar.AllProfiles())
 
 	both, err := r.Consumer(fooID, barID)
 	assert.NotNil(t, both)
