diff --git a/service/collector_test.go b/service/collector_test.go
index 600333d6d..8a137347c 100644
--- a/service/collector_test.go
+++ b/service/collector_test.go
@@ -54,7 +54,7 @@ func TestCollector_StartAsGoRoutine(t *testing.T) {
 	set := CollectorSettings{
 		BuildInfo:      component.NewDefaultBuildInfo(),
 		Factories:      factories,
-		ConfigProvider: NewDefaultConfigProvider([]string{path.Join("testdata", "otelcol-config.yaml")}, nil),
+		ConfigProvider: MustNewDefaultConfigProvider([]string{path.Join("testdata", "otelcol-config.yaml")}, nil),
 	}
 	col, err := New(set)
 	require.NoError(t, err)
@@ -96,7 +96,7 @@ func TestCollector_Start(t *testing.T) {
 	col, err := New(CollectorSettings{
 		BuildInfo: component.NewDefaultBuildInfo(),
 		Factories: factories,
-		ConfigProvider: NewDefaultConfigProvider(
+		ConfigProvider: MustNewDefaultConfigProvider(
 			[]string{path.Join("testdata", "otelcol-config.yaml")},
 			[]string{"service.telemetry.metrics.address=localhost:" + strconv.FormatUint(uint64(metricsPort), 10)}),
 		LoggingOptions: []zap.Option{zap.Hooks(hook)},
@@ -157,7 +157,7 @@ func TestCollector_ReportError(t *testing.T) {
 	col, err := New(CollectorSettings{
 		BuildInfo:      component.NewDefaultBuildInfo(),
 		Factories:      factories,
-		ConfigProvider: NewDefaultConfigProvider([]string{path.Join("testdata", "otelcol-config.yaml")}, nil),
+		ConfigProvider: MustNewDefaultConfigProvider([]string{path.Join("testdata", "otelcol-config.yaml")}, nil),
 	})
 	require.NoError(t, err)
 
