diff --git a/internal/data/testdata/resource.go b/internal/data/testdata/resource.go
index 956bf5681..67371501f 100644
--- a/internal/data/testdata/resource.go
+++ b/internal/data/testdata/resource.go
@@ -15,7 +15,7 @@
 package testdata
 
 import (
-	otlpresource "github.com/open-telemetry/opentelemetry-proto/gen/go/resource/v1"
+	otlpresource "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/resource/v1"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
 )
