diff --git a/config/configgrpc/client_middleware_test.go b/config/configgrpc/client_middleware_test.go
index c5676df34..4bd3dcf8c 100644
--- a/config/configgrpc/client_middleware_test.go
+++ b/config/configgrpc/client_middleware_test.go
@@ -114,7 +114,7 @@ func TestClientMiddlewareOrdering(t *testing.T) {
 	// Create client config with middleware extensions
 	clientConfig := ClientConfig{
 		Endpoint: addr,
-		TLSSetting: configtls.ClientConfig{
+		TLS: configtls.ClientConfig{
 			Insecure: true,
 		},
 		Middlewares: []configmiddleware.Config{
@@ -158,7 +158,7 @@ func TestClientMiddlewareToClientErrors(t *testing.T) {
 			},
 			config: ClientConfig{
 				Endpoint: "localhost:1234",
-				TLSSetting: configtls.ClientConfig{
+				TLS: configtls.ClientConfig{
 					Insecure: true,
 				},
 				Middlewares: []configmiddleware.Config{
@@ -178,7 +178,7 @@ func TestClientMiddlewareToClientErrors(t *testing.T) {
 			},
 			config: ClientConfig{
 				Endpoint: "localhost:1234",
-				TLSSetting: configtls.ClientConfig{
+				TLS: configtls.ClientConfig{
 					Insecure: true,
 				},
 				Middlewares: []configmiddleware.Config{
