diff --git a/pdata/pprofile/generated_line_test.go b/pdata/pprofile/generated_line_test.go
index 010799265..b8b0484a9 100644
--- a/pdata/pprofile/generated_line_test.go
+++ b/pdata/pprofile/generated_line_test.go
@@ -13,6 +13,7 @@ import (
 
 	"go.opentelemetry.io/collector/pdata/internal"
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestLine_MoveTo(t *testing.T) {
@@ -40,6 +41,20 @@ func TestLine_CopyTo(t *testing.T) {
 	assert.Panics(t, func() { ms.CopyTo(newLine(&otlpprofiles.Line{}, &sharedState)) })
 }
 
+func TestLine_MarshalAndUnmarshal(t *testing.T) {
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	src := generateTestLine()
+	src.marshalJSONStream(stream)
+
+	iter := json.BorrowIterator(stream.Buffer())
+	defer json.ReturnIterator(iter)
+	dest := NewLine()
+	dest.unmarshalJsoniter(iter)
+
+	assert.Equal(t, src, dest)
+}
+
 func TestLine_FunctionIndex(t *testing.T) {
 	ms := NewLine()
 	assert.Equal(t, int32(0), ms.FunctionIndex())
