diff --git a/pdata/pprofile/generated_profile_test.go b/pdata/pprofile/generated_profile_test.go
index 07f1602b1..a8bb55bb5 100644
--- a/pdata/pprofile/generated_profile_test.go
+++ b/pdata/pprofile/generated_profile_test.go
@@ -14,6 +14,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal"
 	"go.opentelemetry.io/collector/pdata/internal/data"
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
@@ -42,6 +43,20 @@ func TestProfile_CopyTo(t *testing.T) {
 	assert.Panics(t, func() { ms.CopyTo(newProfile(&otlpprofiles.Profile{}, &sharedState)) })
 }
 
+func TestProfile_MarshalAndUnmarshal(t *testing.T) {
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	src := generateTestProfile()
+	src.marshalJSONStream(stream)
+
+	iter := json.BorrowIterator(stream.Buffer())
+	defer json.ReturnIterator(iter)
+	dest := NewProfile()
+	dest.unmarshalJsoniter(iter)
+
+	assert.Equal(t, src, dest)
+}
+
 func TestProfile_SampleType(t *testing.T) {
 	ms := NewProfile()
 	assert.Equal(t, NewValueTypeSlice(), ms.SampleType())
