diff --git a/extension/pprofextension/factory_test.go b/extension/pprofextension/factory_test.go
index 0ada40d1d..8244b081b 100644
--- a/extension/pprofextension/factory_test.go
+++ b/extension/pprofextension/factory_test.go
@@ -32,11 +32,8 @@ import (
 func TestFactory_CreateDefaultConfig(t *testing.T) {
 	cfg := createDefaultConfig()
 	assert.Equal(t, &Config{
-		ExtensionSettings: config.ExtensionSettings{
-			NameVal: typeStr,
-			TypeVal: typeStr,
-		},
-		TCPAddr: confignet.TCPAddr{Endpoint: defaultEndpoint},
+		ExtensionSettings: config.NewExtensionSettings(typeStr),
+		TCPAddr:           confignet.TCPAddr{Endpoint: defaultEndpoint},
 	},
 		cfg)
 
