diff --git a/service/go.mod b/service/go.mod
index 879f7478a..16db8d101 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -21,7 +21,7 @@ require (
 	go.opentelemetry.io/collector/extension v0.104.0
 	go.opentelemetry.io/collector/extension/zpagesextension v0.104.0
 	go.opentelemetry.io/collector/featuregate v1.11.0
-	go.opentelemetry.io/collector/internal/featuregates v0.0.0-20240705161705-b127da089038
+	go.opentelemetry.io/collector/internal/globalgates v0.0.0-20240705161705-b127da089038
 	go.opentelemetry.io/collector/pdata v1.11.0
 	go.opentelemetry.io/collector/pdata/testdata v0.104.0
 	go.opentelemetry.io/collector/processor v0.104.0
@@ -155,4 +155,4 @@ replace go.opentelemetry.io/collector/config/configcompression => ../config/conf
 
 replace go.opentelemetry.io/collector/pdata/pprofile => ../pdata/pprofile
 
-replace go.opentelemetry.io/collector/internal/featuregates => ../internal/featuregates
+replace go.opentelemetry.io/collector/internal/globalgates => ../internal/globalgates
