diff --git a/receiver/kafkareceiver/kafka_receiver.go b/receiver/kafkareceiver/kafka_receiver.go
index 99a66a951..964bfb5f3 100644
--- a/receiver/kafkareceiver/kafka_receiver.go
+++ b/receiver/kafkareceiver/kafka_receiver.go
@@ -40,7 +40,7 @@ var errUnrecognizedEncoding = fmt.Errorf("unrecognized encoding")
 type kafkaConsumer struct {
 	name              string
 	consumerGroup     sarama.ConsumerGroup
-	nextConsumer      consumer.TraceConsumer
+	nextConsumer      consumer.TracesConsumer
 	topics            []string
 	cancelConsumeLoop context.CancelFunc
 	unmarshaller      Unmarshaller
@@ -50,7 +50,7 @@ type kafkaConsumer struct {
 
 var _ component.Receiver = (*kafkaConsumer)(nil)
 
-func newReceiver(config Config, params component.ReceiverCreateParams, unmarshalers map[string]Unmarshaller, nextConsumer consumer.TraceConsumer) (*kafkaConsumer, error) {
+func newReceiver(config Config, params component.ReceiverCreateParams, unmarshalers map[string]Unmarshaller, nextConsumer consumer.TracesConsumer) (*kafkaConsumer, error) {
 	unmarshaller := unmarshalers[config.Encoding]
 	if unmarshaller == nil {
 		return nil, errUnrecognizedEncoding
@@ -124,7 +124,7 @@ func (c *kafkaConsumer) Shutdown(context.Context) error {
 type consumerGroupHandler struct {
 	name         string
 	unmarshaller Unmarshaller
-	nextConsumer consumer.TraceConsumer
+	nextConsumer consumer.TracesConsumer
 	ready        chan bool
 	readyCloser  sync.Once
 
