diff --git a/pdata/pprofile/generated_resourceprofilesslice_test.go b/pdata/pprofile/generated_resourceprofilesslice_test.go
index 16bb16242..043d8692e 100644
--- a/pdata/pprofile/generated_resourceprofilesslice_test.go
+++ b/pdata/pprofile/generated_resourceprofilesslice_test.go
@@ -28,9 +28,9 @@ func TestResourceProfilesSlice(t *testing.T) {
 	emptyVal := NewResourceProfiles()
 	testVal := generateTestResourceProfiles()
 	for i := 0; i < 7; i++ {
-		el := es.AppendEmpty()
+		es.AppendEmpty()
 		assert.Equal(t, emptyVal, es.At(i))
-		fillTestResourceProfiles(el)
+		internal.FillOrigTestResourceProfiles((*es.orig)[i])
 		assert.Equal(t, testVal, es.At(i))
 	}
 	assert.Equal(t, 7, es.Len())
@@ -51,26 +51,9 @@ func TestResourceProfilesSliceReadOnly(t *testing.T) {
 
 func TestResourceProfilesSlice_CopyTo(t *testing.T) {
 	dest := NewResourceProfilesSlice()
-	// Test CopyTo empty
-	NewResourceProfilesSlice().CopyTo(dest)
-	assert.Equal(t, NewResourceProfilesSlice(), dest)
-
-	// Test CopyTo larger slice
 	src := generateTestResourceProfilesSlice()
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestResourceProfilesSlice(), dest)
-
-	// Test CopyTo same size slice
-	src.CopyTo(dest)
-	assert.Equal(t, generateTestResourceProfilesSlice(), dest)
-
-	// Test CopyTo smaller size slice
-	NewResourceProfilesSlice().CopyTo(dest)
-	assert.Equal(t, 0, dest.Len())
-
-	// Test CopyTo larger slice with enough capacity
-	src.CopyTo(dest)
-	assert.Equal(t, generateTestResourceProfilesSlice(), dest)
 }
 
 func TestResourceProfilesSlice_EnsureCapacity(t *testing.T) {
@@ -194,15 +177,7 @@ func TestResourceProfilesSlice_Sort(t *testing.T) {
 }
 
 func generateTestResourceProfilesSlice() ResourceProfilesSlice {
-	es := NewResourceProfilesSlice()
-	fillTestResourceProfilesSlice(es)
-	return es
-}
-
-func fillTestResourceProfilesSlice(es ResourceProfilesSlice) {
-	*es.orig = make([]*otlpprofiles.ResourceProfiles, 7)
-	for i := 0; i < 7; i++ {
-		(*es.orig)[i] = &otlpprofiles.ResourceProfiles{}
-		fillTestResourceProfiles(newResourceProfiles((*es.orig)[i], es.state))
-	}
+	ms := NewResourceProfilesSlice()
+	*ms.orig = internal.GenerateOrigTestResourceProfilesSlice()
+	return ms
 }
