diff --git a/pdata/pprofile/generated_sample_test.go b/pdata/pprofile/generated_sample_test.go
index 375b4b59c..0d27d1e37 100644
--- a/pdata/pprofile/generated_sample_test.go
+++ b/pdata/pprofile/generated_sample_test.go
@@ -84,14 +84,14 @@ func TestSample_LocationsLength(t *testing.T) {
 func TestSample_Value(t *testing.T) {
 	ms := NewSample()
 	assert.Equal(t, pcommon.NewInt64Slice(), ms.Value())
-	internal.FillTestInt64Slice(internal.Int64Slice(ms.Value()))
+	ms.orig.Value = internal.GenerateOrigTestInt64Slice()
 	assert.Equal(t, pcommon.Int64Slice(internal.GenerateTestInt64Slice()), ms.Value())
 }
 
 func TestSample_AttributeIndices(t *testing.T) {
 	ms := NewSample()
 	assert.Equal(t, pcommon.NewInt32Slice(), ms.AttributeIndices())
-	internal.FillTestInt32Slice(internal.Int32Slice(ms.AttributeIndices()))
+	ms.orig.AttributeIndices = internal.GenerateOrigTestInt32Slice()
 	assert.Equal(t, pcommon.Int32Slice(internal.GenerateTestInt32Slice()), ms.AttributeIndices())
 }
 
@@ -112,21 +112,12 @@ func TestSample_LinkIndex(t *testing.T) {
 func TestSample_TimestampsUnixNano(t *testing.T) {
 	ms := NewSample()
 	assert.Equal(t, pcommon.NewUInt64Slice(), ms.TimestampsUnixNano())
-	internal.FillTestUInt64Slice(internal.UInt64Slice(ms.TimestampsUnixNano()))
+	ms.orig.TimestampsUnixNano = internal.GenerateOrigTestUint64Slice()
 	assert.Equal(t, pcommon.UInt64Slice(internal.GenerateTestUInt64Slice()), ms.TimestampsUnixNano())
 }
 
 func generateTestSample() Sample {
-	tv := NewSample()
-	fillTestSample(tv)
-	return tv
-}
-
-func fillTestSample(tv Sample) {
-	tv.orig.LocationsStartIndex = int32(13)
-	tv.orig.LocationsLength = int32(13)
-	internal.FillTestInt64Slice(internal.NewInt64Slice(&tv.orig.Value, tv.state))
-	internal.FillTestInt32Slice(internal.NewInt32Slice(&tv.orig.AttributeIndices, tv.state))
-	tv.orig.LinkIndex_ = &otlpprofiles.Sample_LinkIndex{LinkIndex: int32(13)}
-	internal.FillTestUInt64Slice(internal.NewUInt64Slice(&tv.orig.TimestampsUnixNano, tv.state))
+	ms := NewSample()
+	internal.FillOrigTestSample(ms.orig)
+	return ms
 }
