diff --git a/service/telemetry.go b/service/telemetry.go
index 35eeee600..cb48395cb 100644
--- a/service/telemetry.go
+++ b/service/telemetry.go
@@ -91,7 +91,7 @@ func (tel *colTelemetry) init(col *Collector) error {
 }
 
 func (tel *colTelemetry) initOnce(col *Collector) error {
-	logger := col.logger
+	logger := col.telemetry.Logger
 	cfg := col.service.config.Telemetry
 
 	level := cfg.Metrics.Level
@@ -210,7 +210,7 @@ func (tel *colTelemetry) initOpenTelemetry(col *Collector) (http.Handler, error)
 		return nil, err
 	}
 
-	col.meterProvider = pe.MeterProvider()
+	col.telemetry.MeterProvider = pe.MeterProvider()
 	return pe, err
 }
 
