diff --git a/exporter/exporterhelper/internal/persistent_queue.go b/exporter/exporterhelper/internal/persistent_queue.go
index 3ceb1d0f6..ae7803a67 100644
--- a/exporter/exporterhelper/internal/persistent_queue.go
+++ b/exporter/exporterhelper/internal/persistent_queue.go
@@ -28,7 +28,6 @@ var (
 // persistentQueue holds the queue backed by file storage
 type persistentQueue struct {
 	stopWG       sync.WaitGroup
-	stopChan     chan struct{}
 	set          exporter.CreateSettings
 	storageID    component.ID
 	storage      *persistentContiguousStorage
@@ -48,7 +47,6 @@ func NewPersistentQueue(capacity int, numConsumers int, storageID component.ID,
 		storageID:    storageID,
 		marshaler:    marshaler,
 		unmarshaler:  unmarshaler,
-		stopChan:     make(chan struct{}),
 	}
 }
 
@@ -64,12 +62,11 @@ func (pq *persistentQueue) Start(ctx context.Context, host component.Host, set Q
 		go func() {
 			defer pq.stopWG.Done()
 			for {
-				select {
-				case req := <-pq.storage.get():
-					set.Callback(req)
-				case <-pq.stopChan:
+				req, found := pq.storage.get()
+				if !found {
 					return
 				}
+				set.Callback(req)
 			}
 		}()
 	}
@@ -85,7 +82,9 @@ func (pq *persistentQueue) Produce(_ context.Context, item any) bool {
 
 // Shutdown stops accepting items, shuts down the queue and closes the persistent queue
 func (pq *persistentQueue) Shutdown(ctx context.Context) error {
-	close(pq.stopChan)
+	if pq.storage != nil {
+		close(pq.storage.stopChan)
+	}
 	pq.stopWG.Wait()
 	return stopStorage(pq.storage, ctx)
 }
