diff --git a/internal/cmd/pdatagen/internal/one_of_field.go b/internal/cmd/pdatagen/internal/one_of_field.go
index 11269f754..e4b7ff8ca 100644
--- a/internal/cmd/pdatagen/internal/one_of_field.go
+++ b/internal/cmd/pdatagen/internal/one_of_field.go
@@ -45,6 +45,11 @@ const oneOfMarshalJSONTemplate = `switch ov := ms.{{ .origAccessor }}.{{ .origin
 		{{- end }}
 	}`
 
+const oneOfUnmarshalJSONTemplate = `
+	{{- range .values }}
+	{{ .GenerateUnmarshalJSON $.baseStruct $.OneOfField }}
+	{{- end }}`
+
 type OneOfField struct {
 	originFieldName            string
 	typeName                   string
@@ -85,6 +90,11 @@ func (of *OneOfField) GenerateMarshalJSON(ms *messageStruct) string {
 	return executeTemplate(t, of.templateFields(ms))
 }
 
+func (of *OneOfField) GenerateUnmarshalJSON(ms *messageStruct) string {
+	t := template.Must(templateNew("oneOfUnmarshalJSONTemplate").Parse(oneOfUnmarshalJSONTemplate))
+	return executeTemplate(t, of.templateFields(ms))
+}
+
 func (of *OneOfField) templateFields(ms *messageStruct) map[string]any {
 	return map[string]any{
 		"baseStruct":           ms,
@@ -111,4 +121,5 @@ type oneOfValue interface {
 	GenerateCopyOrig(ms *messageStruct, of *OneOfField) string
 	GenerateType(ms *messageStruct, of *OneOfField) string
 	GenerateMarshalJSON(ms *messageStruct, of *OneOfField) string
+	GenerateUnmarshalJSON(ms *messageStruct, of *OneOfField) string
 }
