diff --git a/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_darwin.go b/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_darwin.go
index 2499ae809..9be55ba4a 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_darwin.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_darwin.go
@@ -24,6 +24,6 @@ import (
 
 const unixSystemSpecificMetricsLen = 0
 
-func appendUnixSystemSpecificProcessesMetrics(metrics pdata.MetricSlice, startTime pdata.Timestamp, startIndex int, now pdata.Timestamp, misc *load.MiscStat) error {
+func appendUnixSystemSpecificProcessesMetrics(metrics pdata.MetricSlice, startTime pdata.Timestamp, now pdata.Timestamp, misc *load.MiscStat) error {
 	return nil
 }
