diff --git a/pdata/pmetric/generated_scopemetricsslice_test.go b/pdata/pmetric/generated_scopemetricsslice_test.go
index c583e1611..6f509fec4 100644
--- a/pdata/pmetric/generated_scopemetricsslice_test.go
+++ b/pdata/pmetric/generated_scopemetricsslice_test.go
@@ -141,6 +141,14 @@ func TestScopeMetricsSlice_RemoveIf(t *testing.T) {
 	assert.Equal(t, 5, filtered.Len())
 }
 
+func TestScopeMetricsSlice_RemoveIfAll(t *testing.T) {
+	got := generateTestScopeMetricsSlice()
+	got.RemoveIf(func(el ScopeMetrics) bool {
+		return true
+	})
+	assert.Equal(t, 0, got.Len())
+}
+
 func TestScopeMetricsSliceAll(t *testing.T) {
 	ms := generateTestScopeMetricsSlice()
 	assert.NotEmpty(t, ms.Len())
