diff --git a/exporter/exporterhelper/resource_to_label.go b/exporter/exporterhelper/resource_to_label.go
index 6ce056ad2..0c4b6bd8e 100644
--- a/exporter/exporterhelper/resource_to_label.go
+++ b/exporter/exporterhelper/resource_to_label.go
@@ -61,7 +61,7 @@ func extractLabelsFromResource(resource *pdata.Resource) pdata.StringMap {
 
 	attrMap := resource.Attributes()
 	attrMap.Range(func(k string, av pdata.AttributeValue) bool {
-		stringLabel := tracetranslator.AttributeValueToString(av, false)
+		stringLabel := tracetranslator.AttributeValueToString(av)
 		labelMap.Upsert(k, stringLabel)
 		return true
 	})
