diff --git a/exporter/loggingexporter/logging_exporter.go b/exporter/loggingexporter/logging_exporter.go
index fdac6f172..3bff58f42 100644
--- a/exporter/loggingexporter/logging_exporter.go
+++ b/exporter/loggingexporter/logging_exporter.go
@@ -45,7 +45,7 @@ func (b *logDataBuffer) logAttr(label string, value string) {
 }
 
 func (b *logDataBuffer) logAttributeMap(label string, am pdata.AttributeMap) {
-	if am.Cap() == 0 {
+	if am.Len() == 0 {
 		return
 	}
 
