diff --git a/component/componenttest/nop_exporter.go b/component/componenttest/nop_exporter.go
index c1f540170..ac038d413 100644
--- a/component/componenttest/nop_exporter.go
+++ b/component/componenttest/nop_exporter.go
@@ -20,11 +20,12 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/exporter"
 )
 
 // NewNopExporterCreateSettings returns a new nop settings for Create*Exporter functions.
-func NewNopExporterCreateSettings() component.ExporterCreateSettings {
-	return component.ExporterCreateSettings{
+func NewNopExporterCreateSettings() exporter.CreateSettings {
+	return exporter.CreateSettings{
 		TelemetrySettings: NewNopTelemetrySettings(),
 		BuildInfo:         component.NewDefaultBuildInfo(),
 	}
@@ -34,30 +35,30 @@ type nopExporterConfig struct {
 	config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 }
 
-// NewNopExporterFactory returns a component.ExporterFactory that constructs nop exporters.
-func NewNopExporterFactory() component.ExporterFactory {
-	return component.NewExporterFactory(
+// NewNopExporterFactory returns an exporter.Factory that constructs nop exporters.
+func NewNopExporterFactory() exporter.Factory {
+	return exporter.NewFactory(
 		"nop",
 		func() component.Config {
 			return &nopExporterConfig{
 				ExporterSettings: config.NewExporterSettings(component.NewID("nop")),
 			}
 		},
-		component.WithTracesExporter(createTracesExporter, component.StabilityLevelStable),
-		component.WithMetricsExporter(createMetricsExporter, component.StabilityLevelStable),
-		component.WithLogsExporter(createLogsExporter, component.StabilityLevelStable),
+		exporter.WithTraces(createTracesExporter, component.StabilityLevelStable),
+		exporter.WithMetrics(createMetricsExporter, component.StabilityLevelStable),
+		exporter.WithLogs(createLogsExporter, component.StabilityLevelStable),
 	)
 }
 
-func createTracesExporter(context.Context, component.ExporterCreateSettings, component.Config) (component.TracesExporter, error) {
+func createTracesExporter(context.Context, exporter.CreateSettings, component.Config) (exporter.Traces, error) {
 	return nopExporterInstance, nil
 }
 
-func createMetricsExporter(context.Context, component.ExporterCreateSettings, component.Config) (component.MetricsExporter, error) {
+func createMetricsExporter(context.Context, exporter.CreateSettings, component.Config) (exporter.Metrics, error) {
 	return nopExporterInstance, nil
 }
 
-func createLogsExporter(context.Context, component.ExporterCreateSettings, component.Config) (component.LogsExporter, error) {
+func createLogsExporter(context.Context, exporter.CreateSettings, component.Config) (exporter.Logs, error) {
 	return nopExporterInstance, nil
 }
 
