diff --git a/pdata/pmetric/generated_sum.go b/pdata/pmetric/generated_sum.go
index 382ea7a7e..e8c3155f5 100644
--- a/pdata/pmetric/generated_sum.go
+++ b/pdata/pmetric/generated_sum.go
@@ -9,6 +9,7 @@ package pmetric
 import (
 	"go.opentelemetry.io/collector/pdata/internal"
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 // Sum represents the type of a numeric metric that is calculated as a sum of all reported measurements over a time interval.
@@ -82,6 +83,24 @@ func (ms Sum) CopyTo(dest Sum) {
 	copyOrigSum(dest.orig, ms.orig)
 }
 
+// marshalJSONStream marshals all properties from the current struct to the destination stream.
+func (ms Sum) marshalJSONStream(dest *json.Stream) {
+	dest.WriteObjectStart()
+	if ms.orig.AggregationTemporality != otlpmetrics.AggregationTemporality(0) {
+		dest.WriteObjectField("aggregationTemporality")
+		ms.AggregationTemporality().marshalJSONStream(dest)
+	}
+	if ms.orig.IsMonotonic != false {
+		dest.WriteObjectField("isMonotonic")
+		dest.WriteBool(ms.orig.IsMonotonic)
+	}
+	if len(ms.orig.DataPoints) > 0 {
+		dest.WriteObjectField("dataPoints")
+		ms.DataPoints().marshalJSONStream(dest)
+	}
+	dest.WriteObjectEnd()
+}
+
 func copyOrigSum(dest, src *otlpmetrics.Sum) {
 	dest.AggregationTemporality = src.AggregationTemporality
 	dest.IsMonotonic = src.IsMonotonic
