diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index ac77d2119..3a4a21290 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -6,16 +6,16 @@ go 1.18
 
 require (
 	github.com/stretchr/testify v1.8.1
-	go.opentelemetry.io/collector v0.68.0
+	go.opentelemetry.io/collector v0.69.0
 	go.opentelemetry.io/collector/component v0.68.0
-	go.opentelemetry.io/collector/exporter/loggingexporter v0.68.0
-	go.opentelemetry.io/collector/exporter/otlpexporter v0.68.0
-	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.68.0
-	go.opentelemetry.io/collector/extension/ballastextension v0.68.0
-	go.opentelemetry.io/collector/extension/zpagesextension v0.68.0
-	go.opentelemetry.io/collector/processor/batchprocessor v0.68.0
-	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.68.0
-	go.opentelemetry.io/collector/receiver/otlpreceiver v0.68.0
+	go.opentelemetry.io/collector/exporter/loggingexporter v0.69.0
+	go.opentelemetry.io/collector/exporter/otlpexporter v0.69.0
+	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.69.0
+	go.opentelemetry.io/collector/extension/ballastextension v0.69.0
+	go.opentelemetry.io/collector/extension/zpagesextension v0.69.0
+	go.opentelemetry.io/collector/processor/batchprocessor v0.69.0
+	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.69.0
+	go.opentelemetry.io/collector/receiver/otlpreceiver v0.69.0
 	golang.org/x/sys v0.3.0
 )
 
