diff --git a/receiver/go.mod b/receiver/go.mod
index 9a2a9410d..e58f27522 100644
--- a/receiver/go.mod
+++ b/receiver/go.mod
@@ -44,9 +44,9 @@ require (
 	go.opentelemetry.io/collector/confmap v0.72.0 // indirect
 	go.opentelemetry.io/collector/exporter v0.0.0-20230303211526-ec5d71fec2da // indirect
 	go.opentelemetry.io/collector/featuregate v0.72.0 // indirect
-	go.opentelemetry.io/otel/exporters/prometheus v0.36.0 // indirect
-	go.opentelemetry.io/otel/metric v0.36.0 // indirect
-	go.opentelemetry.io/otel/sdk/metric v0.36.0 // indirect
+	go.opentelemetry.io/otel/exporters/prometheus v0.37.0 // indirect
+	go.opentelemetry.io/otel/metric v0.37.0 // indirect
+	go.opentelemetry.io/otel/sdk/metric v0.37.0 // indirect
 	go.opentelemetry.io/otel/trace v1.14.0 // indirect
 	go.uber.org/atomic v1.10.0 // indirect
 	golang.org/x/net v0.7.0 // indirect
