diff --git a/otelcol/go.mod b/otelcol/go.mod
index 25a03b1d8..5870df34e 100644
--- a/otelcol/go.mod
+++ b/otelcol/go.mod
@@ -114,7 +114,7 @@ require (
 	gonum.org/v1/gonum v0.15.1 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20250115164207-1a7da9e5054f // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20250115164207-1a7da9e5054f // indirect
-	google.golang.org/protobuf v1.36.4 // indirect
+	google.golang.org/protobuf v1.36.5 // indirect
 )
 
 replace go.opentelemetry.io/collector => ../
