diff --git a/exporter/internal/queue/queue.go b/exporter/internal/queue/queue.go
index ee7940836..bec2c3048 100644
--- a/exporter/internal/queue/queue.go
+++ b/exporter/internal/queue/queue.go
@@ -38,22 +38,11 @@ type Queue[T any] interface {
 	OnProcessingFinished(index uint64, consumeErr error)
 }
 
-type itemsCounter interface {
-	ItemsCount() int
-}
-
 // Sizer is an interface that returns the size of the given element.
 type Sizer[T any] interface {
 	Sizeof(T) int64
 }
 
-// ItemsSizer is a Sizer implementation that returns the size of a queue element as the number of items it contains.
-type ItemsSizer[T itemsCounter] struct{}
-
-func (is *ItemsSizer[T]) Sizeof(el T) int64 {
-	return int64(el.ItemsCount())
-}
-
 // RequestSizer is a Sizer implementation that returns the size of a queue element as one request.
 type RequestSizer[T any] struct{}
 
