diff --git a/otelcol/go.mod b/otelcol/go.mod
index 867ad1eb4..73819429c 100644
--- a/otelcol/go.mod
+++ b/otelcol/go.mod
@@ -110,7 +110,7 @@ require (
 	gonum.org/v1/gonum v0.15.1 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20241104194629-dd2ea8efbc28 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20241104194629-dd2ea8efbc28 // indirect
-	google.golang.org/protobuf v1.36.0 // indirect
+	google.golang.org/protobuf v1.36.1 // indirect
 )
 
 replace go.opentelemetry.io/collector => ../
