diff --git a/config/configgrpc/configgrpc.go b/config/configgrpc/configgrpc.go
index 59ea2bccb..be9c9f359 100644
--- a/config/configgrpc/configgrpc.go
+++ b/config/configgrpc/configgrpc.go
@@ -192,11 +192,6 @@ func (gcs *GRPCClientSettings) ToClientConn(ctx context.Context, host component.
 	return grpc.DialContext(ctx, gcs.SanitizedEndpoint(), opts...)
 }
 
-// Deprecated: [v0.64.0] use ToClientConn.
-func (gcs *GRPCClientSettings) ToDialOptions(host component.Host, settings component.TelemetrySettings) ([]grpc.DialOption, error) {
-	return gcs.toDialOptions(host, settings)
-}
-
 func (gcs *GRPCClientSettings) toDialOptions(host component.Host, settings component.TelemetrySettings) ([]grpc.DialOption, error) {
 	var opts []grpc.DialOption
 	if configcompression.IsCompressed(gcs.Compression) {
@@ -298,11 +293,6 @@ func (gss *GRPCServerSettings) ToServer(host component.Host, settings component.
 	return grpc.NewServer(opts...), nil
 }
 
-// Deprecated: [v0.64.0] use ToServer.
-func (gss *GRPCServerSettings) ToServerOption(host component.Host, settings component.TelemetrySettings) ([]grpc.ServerOption, error) {
-	return gss.toServerOption(host, settings)
-}
-
 func (gss *GRPCServerSettings) toServerOption(host component.Host, settings component.TelemetrySettings) ([]grpc.ServerOption, error) {
 	switch gss.NetAddr.Transport {
 	case "tcp", "tcp4", "tcp6", "udp", "udp4", "udp6":
