diff --git a/otelcol/go.mod b/otelcol/go.mod
index 439d26e2a..378ed3808 100644
--- a/otelcol/go.mod
+++ b/otelcol/go.mod
@@ -82,6 +82,7 @@ require (
 	go.opentelemetry.io/collector/processor/processorprofiles v0.112.0 // indirect
 	go.opentelemetry.io/collector/receiver/receiverprofiles v0.112.0 // indirect
 	go.opentelemetry.io/collector/semconv v0.112.0 // indirect
+	go.opentelemetry.io/contrib/bridges/otelzap v0.6.0 // indirect
 	go.opentelemetry.io/contrib/propagators/b3 v1.31.0 // indirect
 	go.opentelemetry.io/otel v1.31.0 // indirect
 	go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp v0.7.0 // indirect
