diff --git a/model/pdata/metrics.go b/model/pdata/metrics.go
index ca4d7954f..efae3d48b 100644
--- a/model/pdata/metrics.go
+++ b/model/pdata/metrics.go
@@ -310,18 +310,6 @@ func (ms NumberDataPoint) Type() MetricValueType {
 	return MetricValueTypeNone
 }
 
-// Value returns the value associated with this NumberDataPoint.
-// Deprecated: Use DoubleVal instead.
-func (ms NumberDataPoint) Value() float64 {
-	return ms.DoubleVal()
-}
-
-// SetValue replaces the value associated with this NumberDataPoint.
-// Deprecated: Use SetDoubleVal instead.
-func (ms NumberDataPoint) SetValue(v float64) {
-	ms.SetDoubleVal(v)
-}
-
 // Type returns the type of the value for this Exemplar.
 // Calling this function on zero-initialized Exemplar will cause a panic.
 func (ms Exemplar) Type() MetricValueType {
@@ -336,15 +324,3 @@ func (ms Exemplar) Type() MetricValueType {
 	}
 	return MetricValueTypeNone
 }
-
-// Value returns the value associated with this Exemplar.
-// Deprecated: Use DoubleVal instead.
-func (ms Exemplar) Value() float64 {
-	return ms.DoubleVal()
-}
-
-// SetValue replaces the value associated with this Exemplar.
-// Deprecated: Use SetDoubleVal instead.
-func (ms Exemplar) SetValue(v float64) {
-	ms.SetDoubleVal(v)
-}
