diff --git a/service/command.go b/service/command.go
index 593fc46be..7a287773f 100644
--- a/service/command.go
+++ b/service/command.go
@@ -15,6 +15,8 @@
 package service // import "go.opentelemetry.io/collector/service"
 
 import (
+	"errors"
+
 	"github.com/spf13/cobra"
 
 	"go.opentelemetry.io/collector/confmap"
@@ -35,7 +37,13 @@ func NewCommand(set CollectorSettings) *cobra.Command {
 			}
 			if set.ConfigProvider == nil {
 				var err error
-				cfgSet := newDefaultConfigProviderSettings(getConfigFlag(flagSet))
+
+				configFlags := getConfigFlag(flagSet)
+				if len(configFlags) == 0 {
+					return errors.New("at least one config flag must be provided")
+				}
+
+				cfgSet := newDefaultConfigProviderSettings(configFlags)
 				// Append the "overwrite properties converter" as the first converter.
 				cfgSet.ResolverSettings.Converters = append(
 					[]confmap.Converter{overwritepropertiesconverter.New(getSetFlag(flagSet))},
