diff --git a/confmap/expand_test.go b/confmap/expand_test.go
index 02db3ea81..b7bfc5948 100644
--- a/confmap/expand_test.go
+++ b/confmap/expand_test.go
@@ -99,9 +99,10 @@ func TestResolverExpandMapAndSliceValues(t *testing.T) {
 
 func TestResolverExpandStringValues(t *testing.T) {
 	tests := []struct {
-		name   string
-		input  string
-		output any
+		name            string
+		input           string
+		output          any
+		defaultProvider bool
 	}{
 		// Embedded.
 		{
@@ -109,11 +110,23 @@ func TestResolverExpandStringValues(t *testing.T) {
 			input:  "${HOST}:${PORT}",
 			output: "${HOST}:${PORT}",
 		},
+		{
+			name:            "NoMatchOldStyleDefaultProvider",
+			input:           "${HOST}:${PORT}",
+			output:          "localhost:3044",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchOldStyleNoBrackets",
 			input:  "${HOST}:$PORT",
 			output: "${HOST}:$PORT",
 		},
+		{
+			name:            "NoMatchOldStyleNoBracketsDefaultProvider",
+			input:           "${HOST}:$PORT",
+			output:          "localhost:$PORT",
+			defaultProvider: true,
+		},
 		{
 			name:   "ComplexValue",
 			input:  "${env:COMPLEX_VALUE}",
@@ -139,6 +152,12 @@ func TestResolverExpandStringValues(t *testing.T) {
 			input:  "${env:HOST}:${PORT}",
 			output: "localhost:${PORT}",
 		},
+		{
+			name:            "EmbeddedNewAndOldStyleDefaultProvider",
+			input:           "${env:HOST}:${PORT}",
+			output:          "localhost:3044",
+			defaultProvider: true,
+		},
 		{
 			name:   "Int",
 			input:  "test_${env:INT}",
@@ -181,11 +200,23 @@ func TestResolverExpandStringValues(t *testing.T) {
 			input:  "${test:localhost:${env:PORT}}",
 			output: "localhost:3044",
 		},
+		{
+			name:            "NestedDefaultProvider",
+			input:           "${test:localhost:${PORT}}",
+			output:          "localhost:3044",
+			defaultProvider: true,
+		},
 		{
 			name:   "EmbeddedInNested",
 			input:  "${test:${env:HOST}:${env:PORT}}",
 			output: "localhost:3044",
 		},
+		{
+			name:            "EmbeddedInNestedDefaultProvider",
+			input:           "${test:${HOST}:${PORT}}",
+			output:          "localhost:3044",
+			defaultProvider: true,
+		},
 		{
 			name:   "EmbeddedAndNested",
 			input:  "${test:localhost:${env:PORT}}?os=${env:OS}",
@@ -202,31 +233,67 @@ func TestResolverExpandStringValues(t *testing.T) {
 			input:  "env:HOST}",
 			output: "env:HOST}",
 		},
+		{
+			name:            "NoMatchMissingOpeningBracketDefaultProvider",
+			input:           "env:HOST}",
+			output:          "env:HOST}",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchMissingClosingBracket",
 			input:  "${HOST",
 			output: "${HOST",
 		},
+		{
+			name:            "NoMatchMissingClosingBracketDefaultProvider",
+			input:           "${HOST",
+			output:          "${HOST",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchBracketsWithout$",
 			input:  "HO{ST}",
 			output: "HO{ST}",
 		},
+		{
+			name:            "NoMatchBracketsWithout$DefaultProvider",
+			input:           "HO{ST}",
+			output:          "HO{ST}",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchOnlyMissingClosingBracket",
 			input:  "${env:HOST${env:PORT?os=${env:OS",
 			output: "${env:HOST${env:PORT?os=${env:OS",
 		},
+		{
+			name:            "NoMatchOnlyMissingClosingBracketDefaultProvider",
+			input:           "${env:HOST${env:PORT?os=${env:OS",
+			output:          "${env:HOST${env:PORT?os=${env:OS",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchOnlyMissingOpeningBracket",
 			input:  "env:HOST}env:PORT}?os=env:OS}",
 			output: "env:HOST}env:PORT}?os=env:OS}",
 		},
+		{
+			name:            "NoMatchOnlyMissingOpeningBracketDefaultProvider",
+			input:           "env:HOST}env:PORT}?os=env:OS}",
+			output:          "env:HOST}env:PORT}?os=env:OS}",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchCloseBeforeOpen",
 			input:  "env:HOST}${env:PORT",
 			output: "env:HOST}${env:PORT",
 		},
+		{
+			name:            "NoMatchCloseBeforeOpenDefaultProvider",
+			input:           "env:HOST}${env:PORT",
+			output:          "env:HOST}${env:PORT",
+			defaultProvider: true,
+		},
 		{
 			name:   "NoMatchOldStyleNested",
 			input:  "${test:localhost:${PORT}}",
@@ -238,6 +305,12 @@ func TestResolverExpandStringValues(t *testing.T) {
 			input:  "env:HOST}${env:PORT}",
 			output: "env:HOST}3044",
 		},
+		{
+			name:            "PartialMatchMissingOpeningBracketFirstDefaultProvider",
+			input:           "env:HOST}${PORT}",
+			output:          "env:HOST}3044",
+			defaultProvider: true,
+		},
 		{
 			name:   "PartialMatchMissingOpeningBracketLast",
 			input:  "${env:HOST}env:PORT}",
@@ -288,11 +361,23 @@ func TestResolverExpandStringValues(t *testing.T) {
 			input:  "${HOST}${env:PORT}",
 			output: "${HOST}3044",
 		},
+		{
+			name:            "SchemeAfterNoSchemeIsExpandedDefaultProvider",
+			input:           "${HOST}${env:PORT}",
+			output:          "localhost3044",
+			defaultProvider: true,
+		},
 		{
 			name:   "SchemeBeforeNoSchemeIsExpanded",
 			input:  "${env:HOST}${PORT}",
 			output: "localhost${PORT}",
 		},
+		{
+			name:            "SchemeBeforeNoSchemeIsExpandedDefaultProvider",
+			input:           "${env:HOST}${PORT}",
+			output:          "localhost3044",
+			defaultProvider: true,
+		},
 	}
 
 	for _, tt := range tests {
@@ -305,7 +390,12 @@ func TestResolverExpandStringValues(t *testing.T) {
 				return NewRetrieved(uri[5:])
 			})
 
-			resolver, err := NewResolver(ResolverSettings{URIs: []string{"input:"}, ProviderFactories: []ProviderFactory{provider, newEnvProvider(), testProvider}, ConverterFactories: nil})
+			envProvider := newEnvProvider()
+			set := ResolverSettings{URIs: []string{"input:"}, ProviderFactories: []ProviderFactory{provider, envProvider, testProvider}, ConverterFactories: nil}
+			if tt.defaultProvider {
+				set.DefaultScheme = "env"
+			}
+			resolver, err := NewResolver(set)
 			require.NoError(t, err)
 
 			cfgMap, err := resolver.Resolve(context.Background())
@@ -317,6 +407,11 @@ func TestResolverExpandStringValues(t *testing.T) {
 
 func newEnvProvider() ProviderFactory {
 	return newFakeProvider("env", func(_ context.Context, uri string, _ WatcherFunc) (*Retrieved, error) {
+		// When using `env` as the default scheme for tests, the uri will not include `env:`.
+		// Instead of duplicating the switch cases, the scheme is added instead.
+		if uri[0:4] != "env:" {
+			uri = "env:" + uri
+		}
 		switch uri {
 		case "env:COMPLEX_VALUE":
 			return NewRetrieved([]any{"localhost:3042"})
@@ -469,3 +564,16 @@ func TestResolverExpandStringValueInvalidReturnValue(t *testing.T) {
 	_, err = resolver.Resolve(context.Background())
 	assert.EqualError(t, err, `expanding ${test:PORT}: expected convertable to string value type, got ['ӛ']([]interface {})`)
 }
+
+func TestResolverDefaultProviderExpand(t *testing.T) {
+	provider := newFakeProvider("input", func(context.Context, string, WatcherFunc) (*Retrieved, error) {
+		return NewRetrieved(map[string]any{"foo": "${HOST}"})
+	})
+
+	resolver, err := NewResolver(ResolverSettings{URIs: []string{"input:"}, ProviderFactories: []ProviderFactory{provider, newEnvProvider()}, DefaultScheme: "env", ConverterFactories: nil})
+	require.NoError(t, err)
+
+	cfgMap, err := resolver.Resolve(context.Background())
+	require.NoError(t, err)
+	assert.Equal(t, map[string]any{"foo": "localhost"}, cfgMap.ToStringMap())
+}
