diff --git a/pdata/plog/generated_logs.go b/pdata/plog/generated_logs.go
index 7ff82a7c1..0c70eba36 100644
--- a/pdata/plog/generated_logs.go
+++ b/pdata/plog/generated_logs.go
@@ -30,7 +30,7 @@ func newLogs(orig *otlpcollectorlogs.ExportLogsServiceRequest, state *internal.S
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewLogs() Logs {
-	return newLogs(&otlpcollectorlogs.ExportLogsServiceRequest{}, internal.NewState())
+	return newLogs(internal.NewOrigPtrExportLogsServiceRequest(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
