diff --git a/component/go.mod b/component/go.mod
index 014a7ce03..7ee259e83 100644
--- a/component/go.mod
+++ b/component/go.mod
@@ -31,6 +31,8 @@ replace go.opentelemetry.io/collector/confmap => ../confmap
 
 replace go.opentelemetry.io/collector/consumer => ../consumer
 
+replace go.opentelemetry.io/collector/exporter => ../exporter
+
 replace go.opentelemetry.io/collector/featuregate => ../featuregate
 
 replace go.opentelemetry.io/collector/pdata => ../pdata
