diff --git a/exporter/exporterhelper/internal/base_exporter.go b/exporter/exporterhelper/internal/base_exporter.go
index ea68353ab..a2b299136 100644
--- a/exporter/exporterhelper/internal/base_exporter.go
+++ b/exporter/exporterhelper/internal/base_exporter.go
@@ -32,7 +32,7 @@ var usePullingBasedExporterQueueBatcher = featuregate.GlobalRegistry().MustRegis
 	featuregate.WithRegisterDescription("if set to true, turns on the pulling-based exporter queue bathcer"),
 )
 
-type ObsrepSenderFactory = func(obsrep *ObsReport) Sender[internal.Request]
+type ObsrepSenderFactory = func(obsrep *ObsReport, next Sender[internal.Request]) Sender[internal.Request]
 
 // Option apply changes to BaseExporter.
 type Option func(*BaseExporter) error
@@ -52,17 +52,20 @@ type BaseExporter struct {
 	// Chain of senders that the exporter helper applies before passing the data to the actual exporter.
 	// The data is handled by each sender in the respective order starting from the queueSender.
 	// Most of the senders are optional, and initialized with a no-op path-through sender.
-	BatchSender   Sender[internal.Request]
-	QueueSender   Sender[internal.Request]
-	ObsrepSender  Sender[internal.Request]
-	RetrySender   Sender[internal.Request]
-	TimeoutSender *TimeoutSender // TimeoutSender is always initialized.
+	BatchSender  Sender[internal.Request]
+	QueueSender  Sender[internal.Request]
+	ObsrepSender Sender[internal.Request]
+	RetrySender  Sender[internal.Request]
+
+	firstSender Sender[internal.Request]
 
 	ConsumerOptions []consumer.Option
 
-	queueCfg     exporterqueue.Config
+	timeoutCfg   TimeoutConfig
+	retryCfg     configretry.BackOffConfig
 	queueFactory exporterqueue.Factory[internal.Request]
-	BatcherCfg   exporterbatcher.Config
+	queueCfg     exporterqueue.Config
+	batcherCfg   exporterbatcher.Config
 }
 
 func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, osf ObsrepSenderFactory, options ...Option) (*BaseExporter, error) {
@@ -72,50 +75,51 @@ func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, osf ObsrepSe
 	}
 
 	be := &BaseExporter{
-		BatchSender:   &BaseSender[internal.Request]{},
-		QueueSender:   &BaseSender[internal.Request]{},
-		ObsrepSender:  osf(obsReport),
-		RetrySender:   &BaseSender[internal.Request]{},
-		TimeoutSender: &TimeoutSender{cfg: NewDefaultTimeoutConfig()},
-
-		Set: set,
+		timeoutCfg: NewDefaultTimeoutConfig(),
+		Set:        set,
 	}
 
 	for _, op := range options {
-		err = multierr.Append(err, op(be))
+		if err = op(be); err != nil {
+			return nil, err
+		}
 	}
-	if err != nil {
-		return nil, err
+
+	// TimeoutSender is always initialized.
+	be.firstSender = &TimeoutSender{cfg: be.timeoutCfg}
+	if be.retryCfg.Enabled {
+		be.RetrySender = newRetrySender(be.retryCfg, set, be.firstSender)
+		be.firstSender = be.RetrySender
+	}
+
+	be.ObsrepSender = osf(obsReport, be.firstSender)
+	be.firstSender = be.ObsrepSender
+
+	if be.batcherCfg.Enabled {
+		// Batcher mutates the data.
+		be.ConsumerOptions = append(be.ConsumerOptions, consumer.WithCapabilities(consumer.Capabilities{MutatesData: true}))
+	}
+
+	if !usePullingBasedExporterQueueBatcher.IsEnabled() && be.batcherCfg.Enabled ||
+		usePullingBasedExporterQueueBatcher.IsEnabled() && be.batcherCfg.Enabled && !be.queueCfg.Enabled {
+		concurrencyLimit := int64(0)
+		if be.queueCfg.Enabled {
+			concurrencyLimit = int64(be.queueCfg.NumConsumers)
+		}
+		be.BatchSender = NewBatchSender(be.batcherCfg, set, concurrencyLimit, be.firstSender)
+		be.firstSender = be.BatchSender
 	}
 
 	if be.queueCfg.Enabled {
 		qSet := exporterqueue.Settings{
 			Signal:           signal,
-			ExporterSettings: be.Set,
+			ExporterSettings: set,
 		}
-		q := be.queueFactory(context.Background(), qSet, be.queueCfg)
-		q, err = newObsQueue(qSet, q)
+		be.QueueSender, err = NewQueueSender(be.queueFactory, qSet, be.queueCfg, be.batcherCfg, be.ExportFailureMessage, be.firstSender)
 		if err != nil {
 			return nil, err
 		}
-		be.QueueSender = NewQueueSender(q, be.Set, be.queueCfg.NumConsumers, be.ExportFailureMessage, be.BatcherCfg)
-	}
-
-	if !usePullingBasedExporterQueueBatcher.IsEnabled() && be.BatcherCfg.Enabled ||
-		usePullingBasedExporterQueueBatcher.IsEnabled() && be.BatcherCfg.Enabled && !be.queueCfg.Enabled {
-		bs := NewBatchSender(be.BatcherCfg, be.Set)
-		be.BatchSender = bs
-	}
-
-	be.connectSenders()
-
-	if bs, ok := be.BatchSender.(*BatchSender); ok {
-		// If queue sender is enabled assign to the batch sender the same number of workers.
-		if qs, ok := be.QueueSender.(*QueueSender); ok {
-			bs.concurrencyLimit = int64(qs.numConsumers)
-		}
-		// Batcher sender mutates the data.
-		be.ConsumerOptions = append(be.ConsumerOptions, consumer.WithCapabilities(consumer.Capabilities{MutatesData: true}))
+		be.firstSender = be.QueueSender
 	}
 
 	return be, nil
@@ -123,7 +127,7 @@ func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, osf ObsrepSe
 
 // Send sends the request using the first sender in the chain.
 func (be *BaseExporter) Send(ctx context.Context, req internal.Request) error {
-	err := be.QueueSender.Send(ctx, req)
+	err := be.firstSender.Send(ctx, req)
 	if err != nil {
 		be.Set.Logger.Error("Exporting failed. Rejecting data."+be.ExportFailureMessage,
 			zap.Error(err), zap.Int("rejected_items", req.ItemsCount()))
@@ -131,39 +135,47 @@ func (be *BaseExporter) Send(ctx context.Context, req internal.Request) error {
 	return err
 }
 
-// connectSenders connects the senders in the predefined order.
-func (be *BaseExporter) connectSenders() {
-	be.QueueSender.SetNextSender(be.BatchSender)
-	be.BatchSender.SetNextSender(be.ObsrepSender)
-	be.ObsrepSender.SetNextSender(be.RetrySender)
-	be.RetrySender.SetNextSender(be.TimeoutSender)
-}
-
 func (be *BaseExporter) Start(ctx context.Context, host component.Host) error {
 	// First start the wrapped exporter.
 	if err := be.StartFunc.Start(ctx, host); err != nil {
 		return err
 	}
 
-	// If no error then start the BatchSender.
-	if err := be.BatchSender.Start(ctx, host); err != nil {
-		return err
+	if be.BatchSender != nil {
+		// If no error then start the BatchSender.
+		if err := be.BatchSender.Start(ctx, host); err != nil {
+			return err
+		}
 	}
 
 	// Last start the queueSender.
-	return be.QueueSender.Start(ctx, host)
+	if be.QueueSender != nil {
+		return be.QueueSender.Start(ctx, host)
+	}
+
+	return nil
 }
 
 func (be *BaseExporter) Shutdown(ctx context.Context) error {
-	return multierr.Combine(
-		// First shutdown the retry sender, so the queue sender can flush the queue without retries.
-		be.RetrySender.Shutdown(ctx),
-		// Then shutdown the batch sender
-		be.BatchSender.Shutdown(ctx),
-		// Then shutdown the queue sender.
-		be.QueueSender.Shutdown(ctx),
-		// Last shutdown the wrapped exporter itself.
-		be.ShutdownFunc.Shutdown(ctx))
+	var err error
+
+	// First shutdown the retry sender, so the queue sender can flush the queue without retries.
+	if be.RetrySender != nil {
+		err = multierr.Append(err, be.RetrySender.Shutdown(ctx))
+	}
+
+	// Then shutdown the batch sender
+	if be.BatchSender != nil {
+		err = multierr.Append(err, be.BatchSender.Shutdown(ctx))
+	}
+
+	// Then shutdown the queue sender.
+	if be.QueueSender != nil {
+		err = multierr.Append(err, be.QueueSender.Shutdown(ctx))
+	}
+
+	// Last shutdown the wrapped exporter itself.
+	return multierr.Append(err, be.ShutdownFunc.Shutdown(ctx))
 }
 
 // WithStart overrides the default Start function for an exporter.
@@ -188,7 +200,7 @@ func WithShutdown(shutdown component.ShutdownFunc) Option {
 // The default TimeoutConfig is 5 seconds.
 func WithTimeout(timeoutConfig TimeoutConfig) Option {
 	return func(o *BaseExporter) error {
-		o.TimeoutSender.cfg = timeoutConfig
+		o.timeoutCfg = timeoutConfig
 		return nil
 	}
 }
@@ -201,7 +213,7 @@ func WithRetry(config configretry.BackOffConfig) Option {
 			o.ExportFailureMessage += " Try enabling retry_on_failure config option to retry on retryable errors."
 			return nil
 		}
-		o.RetrySender = newRetrySender(config, o.Set)
+		o.retryCfg = config
 		return nil
 	}
 }
@@ -268,7 +280,7 @@ func WithCapabilities(capabilities consumer.Capabilities) Option {
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
 func WithBatcher(cfg exporterbatcher.Config) Option {
 	return func(o *BaseExporter) error {
-		o.BatcherCfg = cfg
+		o.batcherCfg = cfg
 		return nil
 	}
 }
