diff --git a/pipeline/xpipeline/go.mod b/pipeline/xpipeline/go.mod
index b35925af4..0263e11af 100644
--- a/pipeline/xpipeline/go.mod
+++ b/pipeline/xpipeline/go.mod
@@ -2,6 +2,6 @@ module go.opentelemetry.io/collector/pipeline/xpipeline
 
 go 1.24.0
 
-require go.opentelemetry.io/collector/pipeline v1.48.0
+require go.opentelemetry.io/collector/pipeline v1.49.0
 
 replace go.opentelemetry.io/collector/pipeline => ../
