diff --git a/internal/processor/filterexpr/matcher.go b/internal/processor/filterexpr/matcher.go
index d3ae381ac..220c6d9ca 100644
--- a/internal/processor/filterexpr/matcher.go
+++ b/internal/processor/filterexpr/matcher.go
@@ -58,7 +58,7 @@ func (m *Matcher) MatchMetric(metric pdata.Metric) (bool, error) {
 func (m *Matcher) matchGauge(metricName string, gauge pdata.Gauge) (bool, error) {
 	pts := gauge.DataPoints()
 	for i := 0; i < pts.Len(); i++ {
-		matched, err := m.matchEnv(metricName, pts.At(i).LabelsMap())
+		matched, err := m.matchEnv(metricName, pts.At(i).Attributes())
 		if err != nil {
 			return false, err
 		}
@@ -72,7 +72,7 @@ func (m *Matcher) matchGauge(metricName string, gauge pdata.Gauge) (bool, error)
 func (m *Matcher) matchSum(metricName string, sum pdata.Sum) (bool, error) {
 	pts := sum.DataPoints()
 	for i := 0; i < pts.Len(); i++ {
-		matched, err := m.matchEnv(metricName, pts.At(i).LabelsMap())
+		matched, err := m.matchEnv(metricName, pts.At(i).Attributes())
 		if err != nil {
 			return false, err
 		}
@@ -86,7 +86,7 @@ func (m *Matcher) matchSum(metricName string, sum pdata.Sum) (bool, error) {
 func (m *Matcher) matchDoubleHistogram(metricName string, histogram pdata.Histogram) (bool, error) {
 	pts := histogram.DataPoints()
 	for i := 0; i < pts.Len(); i++ {
-		matched, err := m.matchEnv(metricName, pts.At(i).LabelsMap())
+		matched, err := m.matchEnv(metricName, pts.At(i).Attributes())
 		if err != nil {
 			return false, err
 		}
@@ -97,20 +97,20 @@ func (m *Matcher) matchDoubleHistogram(metricName string, histogram pdata.Histog
 	return false, nil
 }
 
-func (m *Matcher) matchEnv(metricName string, labelsMap pdata.StringMap) (bool, error) {
-	return m.match(createEnv(metricName, labelsMap))
+func (m *Matcher) matchEnv(metricName string, attributes pdata.AttributeMap) (bool, error) {
+	return m.match(createEnv(metricName, attributes))
 }
 
-func createEnv(metricName string, labelsMap pdata.StringMap) env {
+func createEnv(metricName string, attributes pdata.AttributeMap) env {
 	return env{
 		MetricName: metricName,
 		HasLabel: func(key string) bool {
-			_, ok := labelsMap.Get(key)
+			_, ok := attributes.Get(key)
 			return ok
 		},
 		Label: func(key string) string {
-			v, _ := labelsMap.Get(key)
-			return v
+			v, _ := attributes.Get(key)
+			return v.StringVal()
 		},
 	}
 }
