diff --git a/service/service.go b/service/service.go
index 9ed279160..a30adf6fd 100644
--- a/service/service.go
+++ b/service/service.go
@@ -28,7 +28,6 @@ import (
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/extension"
-	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/internal/obsreportconfig"
 	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/receiver"
@@ -37,13 +36,6 @@ import (
 	"go.opentelemetry.io/collector/service/telemetry"
 )
 
-var _ = featuregate.GlobalRegistry().MustRegister(
-	"service.graph",
-	featuregate.StageStable,
-	featuregate.WithRegisterRemovalVersion("v0.74.0"),
-	featuregate.WithRegisterDescription("Enables the new graph based implementations for pipelines."),
-	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector/issues/2336"))
-
 // Settings holds configuration for building a new service.
 type Settings struct {
 	// BuildInfo provides collector start information.
