diff --git a/service/internal/builder/extensions_builder_test.go b/service/internal/builder/extensions_builder_test.go
index 9e053b1b3..f7254fe63 100644
--- a/service/internal/builder/extensions_builder_test.go
+++ b/service/internal/builder/extensions_builder_test.go
@@ -31,7 +31,7 @@ func TestService_setupExtensions(t *testing.T) {
 	errExtensionFactory := extensionhelper.NewFactory(
 		"err",
 		func() config.Extension {
-			cfg := config.NewExtensionSettings(config.NewID("err"))
+			cfg := config.NewExtensionSettings(config.NewComponentID("err"))
 			return &cfg
 		},
 		func(ctx context.Context, set component.ExtensionCreateSettings, extension config.Extension) (component.Extension, error) {
@@ -53,7 +53,7 @@ func TestService_setupExtensions(t *testing.T) {
 			config: &config.Config{
 				Service: config.Service{
 					Extensions: []config.ComponentID{
-						config.NewID("myextension"),
+						config.NewComponentID("myextension"),
 					},
 				},
 			},
@@ -63,11 +63,11 @@ func TestService_setupExtensions(t *testing.T) {
 			name: "missing_extension_factory",
 			config: &config.Config{
 				Extensions: map[config.ComponentID]config.Extension{
-					config.NewID(errExtensionFactory.Type()): errExtensionConfig,
+					config.NewComponentID(errExtensionFactory.Type()): errExtensionConfig,
 				},
 				Service: config.Service{
 					Extensions: []config.ComponentID{
-						config.NewID(errExtensionFactory.Type()),
+						config.NewComponentID(errExtensionFactory.Type()),
 					},
 				},
 			},
@@ -82,11 +82,11 @@ func TestService_setupExtensions(t *testing.T) {
 			},
 			config: &config.Config{
 				Extensions: map[config.ComponentID]config.Extension{
-					config.NewID(errExtensionFactory.Type()): errExtensionConfig,
+					config.NewComponentID(errExtensionFactory.Type()): errExtensionConfig,
 				},
 				Service: config.Service{
 					Extensions: []config.ComponentID{
-						config.NewID(errExtensionFactory.Type()),
+						config.NewComponentID(errExtensionFactory.Type()),
 					},
 				},
 			},
@@ -101,11 +101,11 @@ func TestService_setupExtensions(t *testing.T) {
 			},
 			config: &config.Config{
 				Extensions: map[config.ComponentID]config.Extension{
-					config.NewID(badExtensionFactory.Type()): badExtensionCfg,
+					config.NewComponentID(badExtensionFactory.Type()): badExtensionCfg,
 				},
 				Service: config.Service{
 					Extensions: []config.ComponentID{
-						config.NewID(badExtensionFactory.Type()),
+						config.NewComponentID(badExtensionFactory.Type()),
 					},
 				},
 			},
