diff --git a/service/go.mod b/service/go.mod
index e64e29caa..56a066dad 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -7,7 +7,7 @@ require (
 	github.com/prometheus/client_golang v1.20.5
 	github.com/prometheus/client_model v0.6.1
 	github.com/prometheus/common v0.61.0
-	github.com/shirou/gopsutil/v4 v4.24.11
+	github.com/shirou/gopsutil/v4 v4.24.12
 	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector v0.116.0
 	go.opentelemetry.io/collector/component v0.116.0
