diff --git a/service/parserprovider/file.go b/service/parserprovider/file.go
index 71553593a..04286923d 100644
--- a/service/parserprovider/file.go
+++ b/service/parserprovider/file.go
@@ -22,28 +22,30 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-type fileMapProvider struct{}
+type fileMapProvider struct {
+	fileName string
+}
 
-// NewFileMapProvider returns a new MapProvider that reads the configuration from a file configured
-// via the --config command line flag.
-func NewFileMapProvider() MapProvider {
-	return &fileMapProvider{}
+// NewFileMapProvider returns a new MapProvider that reads the configuration from the given file.
+func NewFileMapProvider(fileName string) MapProvider {
+	return &fileMapProvider{
+		fileName: fileName,
+	}
 }
 
-func (fl *fileMapProvider) Get(context.Context) (*config.Map, error) {
-	fileName := getConfigFlag()
-	if fileName == "" {
+func (fmp *fileMapProvider) Get(context.Context) (*config.Map, error) {
+	if fmp.fileName == "" {
 		return nil, errors.New("config file not specified")
 	}
 
-	cp, err := config.NewMapFromFile(fileName)
+	cp, err := config.NewMapFromFile(fmp.fileName)
 	if err != nil {
-		return nil, fmt.Errorf("error loading config file %q: %w", fileName, err)
+		return nil, fmt.Errorf("error loading config file %q: %w", fmp.fileName, err)
 	}
 
 	return cp, nil
 }
 
-func (fl *fileMapProvider) Close(context.Context) error {
+func (*fileMapProvider) Close(context.Context) error {
 	return nil
 }
