diff --git a/pdata/pmetric/generated_numberdatapointslice.go b/pdata/pmetric/generated_numberdatapointslice.go
index 9bf396e16..b12dd84c5 100644
--- a/pdata/pmetric/generated_numberdatapointslice.go
+++ b/pdata/pmetric/generated_numberdatapointslice.go
@@ -60,28 +60,8 @@ func (es NumberDataPointSlice) Len() int {
 //	    e := es.At(i)
 //	    ... // Do something with the element
 //	}
-func (es NumberDataPointSlice) At(ix int) NumberDataPoint {
-	return newNumberDataPoint((*es.orig)[ix])
-}
-
-// CopyTo copies all elements from the current slice overriding the destination.
-func (es NumberDataPointSlice) CopyTo(dest NumberDataPointSlice) {
-	srcLen := es.Len()
-	destCap := cap(*dest.orig)
-	if srcLen <= destCap {
-		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
-		for i := range *es.orig {
-			newNumberDataPoint((*es.orig)[i]).CopyTo(newNumberDataPoint((*dest.orig)[i]))
-		}
-		return
-	}
-	origs := make([]otlpmetrics.NumberDataPoint, srcLen)
-	wrappers := make([]*otlpmetrics.NumberDataPoint, srcLen)
-	for i := range *es.orig {
-		wrappers[i] = &origs[i]
-		newNumberDataPoint((*es.orig)[i]).CopyTo(newNumberDataPoint(wrappers[i]))
-	}
-	*dest.orig = wrappers
+func (es NumberDataPointSlice) At(i int) NumberDataPoint {
+	return newNumberDataPoint((*es.orig)[i])
 }
 
 // EnsureCapacity is an operation that ensures the slice has at least the specified capacity.
@@ -114,13 +94,6 @@ func (es NumberDataPointSlice) AppendEmpty() NumberDataPoint {
 	return es.At(es.Len() - 1)
 }
 
-// Sort sorts the NumberDataPoint elements within NumberDataPointSlice given the
-// provided less function so that two instances of NumberDataPointSlice
-// can be compared.
-func (es NumberDataPointSlice) Sort(less func(a, b NumberDataPoint) bool) {
-	sort.SliceStable(*es.orig, func(i, j int) bool { return less(es.At(i), es.At(j)) })
-}
-
 // MoveAndAppendTo moves all elements from the current slice and appends them to the dest.
 // The current slice will be cleared.
 func (es NumberDataPointSlice) MoveAndAppendTo(dest NumberDataPointSlice) {
@@ -152,3 +125,30 @@ func (es NumberDataPointSlice) RemoveIf(f func(NumberDataPoint) bool) {
 	// TODO: Prevent memory leak by erasing truncated values.
 	*es.orig = (*es.orig)[:newLen]
 }
+
+// CopyTo copies all elements from the current slice overriding the destination.
+func (es NumberDataPointSlice) CopyTo(dest NumberDataPointSlice) {
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newNumberDataPoint((*es.orig)[i]).CopyTo(newNumberDataPoint((*dest.orig)[i]))
+		}
+		return
+	}
+	origs := make([]otlpmetrics.NumberDataPoint, srcLen)
+	wrappers := make([]*otlpmetrics.NumberDataPoint, srcLen)
+	for i := range *es.orig {
+		wrappers[i] = &origs[i]
+		newNumberDataPoint((*es.orig)[i]).CopyTo(newNumberDataPoint(wrappers[i]))
+	}
+	*dest.orig = wrappers
+}
+
+// Sort sorts the NumberDataPoint elements within NumberDataPointSlice given the
+// provided less function so that two instances of NumberDataPointSlice
+// can be compared.
+func (es NumberDataPointSlice) Sort(less func(a, b NumberDataPoint) bool) {
+	sort.SliceStable(*es.orig, func(i, j int) bool { return less(es.At(i), es.At(j)) })
+}
