diff --git a/pdata/pmetric/generated_metricslice_test.go b/pdata/pmetric/generated_metricslice_test.go
index 0b4c44c56..4a6c05ad9 100644
--- a/pdata/pmetric/generated_metricslice_test.go
+++ b/pdata/pmetric/generated_metricslice_test.go
@@ -119,15 +119,15 @@ func TestMetricSlice_RemoveIf(t *testing.T) {
 }
 
 func generateTestMetricSlice() MetricSlice {
-	tv := NewMetricSlice()
-	fillTestMetricSlice(tv)
-	return tv
+	es := NewMetricSlice()
+	fillTestMetricSlice(es)
+	return es
 }
 
-func fillTestMetricSlice(tv MetricSlice) {
-	*tv.orig = make([]*otlpmetrics.Metric, 7)
+func fillTestMetricSlice(es MetricSlice) {
+	*es.orig = make([]*otlpmetrics.Metric, 7)
 	for i := 0; i < 7; i++ {
-		(*tv.orig)[i] = &otlpmetrics.Metric{}
-		fillTestMetric(newMetric((*tv.orig)[i]))
+		(*es.orig)[i] = &otlpmetrics.Metric{}
+		fillTestMetric(newMetric((*es.orig)[i]))
 	}
 }
