diff --git a/processor/processortest/nop_processor.go b/processor/processortest/nop_processor.go
index 6e16dbc78..408130d7b 100644
--- a/processor/processortest/nop_processor.go
+++ b/processor/processortest/nop_processor.go
@@ -11,8 +11,10 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerprofiles"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorprofiles"
 )
 
 var nopType = component.MustNewType("nop")
@@ -34,6 +36,7 @@ func NewNopFactory() processor.Factory {
 		processor.WithTraces(createTracesProcessor, component.StabilityLevelStable),
 		processor.WithMetrics(createMetricsProcessor, component.StabilityLevelStable),
 		processor.WithLogs(createLogsProcessor, component.StabilityLevelStable),
+		processorprofiles.WithProfiles(createProfilesProcessor, component.StabilityLevelAlpha),
 	)
 }
 
@@ -49,6 +52,10 @@ func createLogsProcessor(context.Context, processor.Settings, component.Config,
 	return nopInstance, nil
 }
 
+func createProfilesProcessor(context.Context, processor.Settings, component.Config, consumerprofiles.Profiles) (processorprofiles.Profiles, error) {
+	return nopInstance, nil
+}
+
 type nopConfig struct{}
 
 var nopInstance = &nopProcessor{
