diff --git a/service/builder/exporters_builder.go b/service/builder/exporters_builder.go
index 1fce88674..3b6324a21 100644
--- a/service/builder/exporters_builder.go
+++ b/service/builder/exporters_builder.go
@@ -15,6 +15,7 @@
 package builder
 
 import (
+	"context"
 	"fmt"
 
 	"go.uber.org/zap"
@@ -28,8 +29,8 @@ import (
 // builtExporter is an exporter that is built based on a config. It can have
 // a trace and/or a metrics consumer and have a shutdown function.
 type builtExporter struct {
-	te component.TraceExporterOld
-	me component.MetricsExporterOld
+	te component.TraceExporterBase
+	me component.MetricsExporterBase
 }
 
 // Start the exporter.
@@ -192,11 +193,6 @@ func (eb *ExportersBuilder) buildExporter(
 		return nil, fmt.Errorf("exporter factory not found for type: %s", config.Type())
 	}
 
-	factoryV1, ok := factory.(component.ExporterFactoryOld)
-	if !ok {
-		return nil, fmt.Errorf("exporter factory must implement ExporterFactoryOld: %s", config.Type())
-	}
-
 	exporter := &builtExporter{}
 
 	inputDataTypes := exportersInputDataTypes[config]
@@ -212,7 +208,7 @@ func (eb *ExportersBuilder) buildExporter(
 
 	if requirement, ok := inputDataTypes[configmodels.TracesDataType]; ok {
 		// Traces data type is required. Create a trace exporter based on config.
-		te, err := factoryV1.CreateTraceExporter(eb.logger, config)
+		te, err := createTraceExporter(factory, eb.logger, config)
 		if err != nil {
 			if err == configerror.ErrDataTypeIsNotSupported {
 				// Could not create because this exporter does not support this data type.
@@ -231,7 +227,7 @@ func (eb *ExportersBuilder) buildExporter(
 
 	if requirement, ok := inputDataTypes[configmodels.MetricsDataType]; ok {
 		// Metrics data type is required. Create a trace exporter based on config.
-		me, err := factoryV1.CreateMetricsExporter(eb.logger, config)
+		me, err := createMetricsExporter(factory, eb.logger, config)
 		if err != nil {
 			if err == configerror.ErrDataTypeIsNotSupported {
 				// Could not create because this exporter does not support this data type.
@@ -264,3 +260,42 @@ func typeMismatchErr(
 		config.Name(), dataType.GetString(),
 	)
 }
+
+// createTraceProcessor creates a trace exporter based on provided factory type.
+func createTraceExporter(
+	factoryBase component.ExporterFactoryBase,
+	logger *zap.Logger,
+	cfg configmodels.Exporter,
+) (component.TraceExporterBase, error) {
+	if factory, ok := factoryBase.(component.ExporterFactory); ok {
+		creationParams := component.ExporterCreateParams{Logger: logger}
+		ctx := context.Background()
+
+		// If exporter is of the new type (can manipulate on internal data structure),
+		// use ExporterFactory.CreateTraceExporter.
+		return factory.CreateTraceExporter(ctx, creationParams, cfg)
+	}
+
+	// If exporter is of the old type (can manipulate on OC traces only),
+	// use ExporterFactoryOld.CreateTraceExporter.
+	return factoryBase.(component.ExporterFactoryOld).CreateTraceExporter(logger, cfg)
+}
+
+// createMetricsExporter creates a metrics exporter based on provided factory type.
+func createMetricsExporter(factoryBase component.ExporterFactoryBase,
+	logger *zap.Logger,
+	cfg configmodels.Exporter,
+) (component.MetricsExporterBase, error) {
+	if factory, ok := factoryBase.(component.ExporterFactory); ok {
+		creationParams := component.ExporterCreateParams{Logger: logger}
+		ctx := context.Background()
+
+		// If exporter is of the new type (can manipulate on internal data structure),
+		// use ExporterFactory.CreateMetricsExporter.
+		return factory.CreateMetricsExporter(ctx, creationParams, cfg)
+	}
+
+	// If exporter is of the old type (can manipulate on OC metrics only),
+	// use ExporterFactoryOld.CreateMetricsExporter.
+	return factoryBase.(component.ExporterFactoryOld).CreateMetricsExporter(logger, cfg)
+}
