diff --git a/service/service.go b/service/service.go
index 642f73936..a4d186904 100644
--- a/service/service.go
+++ b/service/service.go
@@ -71,8 +71,6 @@ type Service struct {
 	telemetrySettings component.TelemetrySettings
 	host              *graph.Host
 	collectorConf     *confmap.Conf
-
-	reporter status.Reporter
 }
 
 // New creates a new Service, its telemetry, and Components.
@@ -136,7 +134,7 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 		// Construct telemetry attributes from build info and config's resource attributes.
 		Resource: pcommonRes,
 	}
-	srv.reporter = status.NewReporter(srv.host.NotifyComponentStatusChange, func(err error) {
+	srv.host.Reporter = status.NewReporter(srv.host.NotifyComponentStatusChange, func(err error) {
 		if errors.Is(err, status.ErrStatusNotReady) {
 			logger.Warn("Invalid transition", zap.Error(err))
 		}
@@ -198,7 +196,7 @@ func (srv *Service) Start(ctx context.Context) error {
 	)
 
 	// enable status reporting
-	srv.reporter.Ready()
+	srv.host.Reporter.Ready()
 
 	if err := srv.host.ServiceExtensions.Start(ctx, srv.host); err != nil {
 		return fmt.Errorf("failed to start extensions: %w", err)
@@ -210,7 +208,7 @@ func (srv *Service) Start(ctx context.Context) error {
 		}
 	}
 
-	if err := srv.host.Pipelines.StartAll(ctx, srv.host, srv.reporter); err != nil {
+	if err := srv.host.Pipelines.StartAll(ctx, srv.host); err != nil {
 		return fmt.Errorf("cannot start pipelines: %w", err)
 	}
 
@@ -261,7 +259,7 @@ func (srv *Service) Shutdown(ctx context.Context) error {
 		errs = multierr.Append(errs, fmt.Errorf("failed to notify that pipeline is not ready: %w", err))
 	}
 
-	if err := srv.host.Pipelines.ShutdownAll(ctx, srv.reporter); err != nil {
+	if err := srv.host.Pipelines.ShutdownAll(ctx, srv.host.Reporter); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown pipelines: %w", err))
 	}
 
@@ -284,7 +282,7 @@ func (srv *Service) initExtensions(ctx context.Context, cfg extensions.Config) e
 		BuildInfo:  srv.buildInfo,
 		Extensions: srv.host.Extensions,
 	}
-	if srv.host.ServiceExtensions, err = extensions.New(ctx, extensionsSettings, cfg, extensions.WithReporter(srv.reporter)); err != nil {
+	if srv.host.ServiceExtensions, err = extensions.New(ctx, extensionsSettings, cfg, extensions.WithReporter(srv.host.Reporter)); err != nil {
 		return fmt.Errorf("failed to build extensions: %w", err)
 	}
 	return nil
@@ -301,7 +299,7 @@ func (srv *Service) initGraph(ctx context.Context, set Settings, cfg Config) err
 		ExporterBuilder:  set.Exporters,
 		ConnectorBuilder: set.Connectors,
 		PipelineConfigs:  cfg.Pipelines,
-		ReportStatus:     srv.reporter.ReportStatus,
+		ReportStatus:     srv.host.Reporter.ReportStatus,
 	}); err != nil {
 		return fmt.Errorf("failed to build pipelines: %w", err)
 	}
