diff --git a/pdata/pmetric/generated_metrics_test.go b/pdata/pmetric/generated_metrics_test.go
index c7e468ad1..68d349110 100644
--- a/pdata/pmetric/generated_metrics_test.go
+++ b/pdata/pmetric/generated_metrics_test.go
@@ -1063,18 +1063,16 @@ func TestHistogramDataPoint_Sum(t *testing.T) {
 
 func TestHistogramDataPoint_BucketCounts(t *testing.T) {
 	ms := NewHistogramDataPoint()
-	assert.Equal(t, pcommon.NewImmutableUInt64Slice([]uint64(nil)), ms.BucketCounts())
-	testValBucketCounts := pcommon.NewImmutableUInt64Slice([]uint64{1, 2, 3})
-	ms.SetBucketCounts(testValBucketCounts)
-	assert.Equal(t, testValBucketCounts, ms.BucketCounts())
+	assert.Equal(t, []uint64(nil), ms.BucketCounts().AsRaw())
+	ms.BucketCounts().FromRaw([]uint64{1, 2, 3})
+	assert.Equal(t, []uint64{1, 2, 3}, ms.BucketCounts().AsRaw())
 }
 
 func TestHistogramDataPoint_ExplicitBounds(t *testing.T) {
 	ms := NewHistogramDataPoint()
-	assert.Equal(t, pcommon.NewImmutableFloat64Slice([]float64(nil)), ms.ExplicitBounds())
-	testValExplicitBounds := pcommon.NewImmutableFloat64Slice([]float64{1, 2, 3})
-	ms.SetExplicitBounds(testValExplicitBounds)
-	assert.Equal(t, testValExplicitBounds, ms.ExplicitBounds())
+	assert.Equal(t, []float64(nil), ms.ExplicitBounds().AsRaw())
+	ms.ExplicitBounds().FromRaw([]float64{1, 2, 3})
+	assert.Equal(t, []float64{1, 2, 3}, ms.ExplicitBounds().AsRaw())
 }
 
 func TestHistogramDataPoint_Exemplars(t *testing.T) {
@@ -1353,10 +1351,9 @@ func TestBuckets_Offset(t *testing.T) {
 
 func TestBuckets_BucketCounts(t *testing.T) {
 	ms := NewBuckets()
-	assert.Equal(t, pcommon.NewImmutableUInt64Slice([]uint64(nil)), ms.BucketCounts())
-	testValBucketCounts := pcommon.NewImmutableUInt64Slice([]uint64{1, 2, 3})
-	ms.SetBucketCounts(testValBucketCounts)
-	assert.Equal(t, testValBucketCounts, ms.BucketCounts())
+	assert.Equal(t, []uint64(nil), ms.BucketCounts().AsRaw())
+	ms.BucketCounts().FromRaw([]uint64{1, 2, 3})
+	assert.Equal(t, []uint64{1, 2, 3}, ms.BucketCounts().AsRaw())
 }
 
 func TestSummaryDataPointSlice(t *testing.T) {
