diff --git a/processor/batchprocessor/go.mod b/processor/batchprocessor/go.mod
index b488e9255..17fffdc8a 100644
--- a/processor/batchprocessor/go.mod
+++ b/processor/batchprocessor/go.mod
@@ -4,14 +4,14 @@ go 1.21.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector v0.102.0
-	go.opentelemetry.io/collector/component v0.102.0
-	go.opentelemetry.io/collector/config/configtelemetry v0.102.0
-	go.opentelemetry.io/collector/confmap v0.102.0
-	go.opentelemetry.io/collector/consumer v0.102.0
+	go.opentelemetry.io/collector v0.102.1
+	go.opentelemetry.io/collector/component v0.102.1
+	go.opentelemetry.io/collector/config/configtelemetry v0.102.1
+	go.opentelemetry.io/collector/confmap v0.102.1
+	go.opentelemetry.io/collector/consumer v0.102.1
 	go.opentelemetry.io/collector/pdata v1.9.0
-	go.opentelemetry.io/collector/pdata/testdata v0.102.0
-	go.opentelemetry.io/collector/processor v0.102.0
+	go.opentelemetry.io/collector/pdata/testdata v0.102.1
+	go.opentelemetry.io/collector/processor v0.102.1
 	go.opentelemetry.io/otel v1.27.0
 	go.opentelemetry.io/otel/metric v1.27.0
 	go.opentelemetry.io/otel/sdk/metric v1.27.0
