diff --git a/exporter/prometheusremotewriteexporter/exporter_test.go b/exporter/prometheusremotewriteexporter/exporter_test.go
index 6a15745dd..c5533cfd7 100644
--- a/exporter/prometheusremotewriteexporter/exporter_test.go
+++ b/exporter/prometheusremotewriteexporter/exporter_test.go
@@ -16,11 +16,11 @@ package prometheusremotewriteexporter
 
 import (
 	"context"
-	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"strconv"
 	"sync"
 	"testing"
 
@@ -142,7 +142,114 @@ func Test_handleScalarMetric(t *testing.T) {
 	}
 }
 
-// Test_newPrwExporter checks that a new exporter instance with non-nil fields is initialized
+// Test_handleHistogramMetric checks whether data points(sum, count, buckets) within a single Histogram metric can be
+// added to a map of TimeSeries correctly.
+// Test cases are a histogram data point with two buckets and nil data points case.
+func Test_handleHistogramMetric(t *testing.T) {
+	sum := "sum"
+	count := "count"
+	bucket1 := "bucket1"
+	bucket2 := "bucket2"
+	bucketInf := "bucketInf"
+	histPoint := getHistogramDataPoint(
+		lbs1,
+		time1,
+		floatVal2,
+		uint64(intVal2), []float64{floatVal1, floatVal2},
+		[]uint64{uint64(intVal1), uint64(intVal1)})
+
+	// string signature of the data point is the key of the map
+	sigs := map[string]string{
+		sum:   typeHistogram + "-" + nameStr + "-" + name1 + "_sum" + lb1Sig,
+		count: typeHistogram + "-" + nameStr + "-" + name1 + "_count" + lb1Sig,
+		bucket1: typeHistogram + "-" + nameStr + "-" + name1 + "_bucket" + "-" + "le-" +
+			strconv.FormatFloat(floatVal1, 'f', -1, 64) + lb1Sig,
+		bucket2: typeHistogram + "-" + nameStr + "-" + name1 + "_bucket" + "-" + "le-" +
+			strconv.FormatFloat(floatVal2, 'f', -1, 64) + lb1Sig,
+		bucketInf: typeHistogram + "-" + nameStr + "-" + name1 + "_bucket" + "-" + "le-" +
+			"+Inf" + lb1Sig,
+	}
+	labels := map[string][]prompb.Label{
+		sum:   append(promLbs1, getPromLabels(nameStr, name1+"_sum")...),
+		count: append(promLbs1, getPromLabels(nameStr, name1+"_count")...),
+		bucket1: append(promLbs1, getPromLabels(nameStr, name1+"_bucket", "le",
+			strconv.FormatFloat(floatVal1, 'f', -1, 64))...),
+		bucket2: append(promLbs1, getPromLabels(nameStr, name1+"_bucket", "le",
+			strconv.FormatFloat(floatVal2, 'f', -1, 64))...),
+		bucketInf: append(promLbs1, getPromLabels(nameStr, name1+"_bucket", "le",
+			"+Inf")...),
+	}
+	tests := []struct {
+		name        string
+		m           otlp.Metric
+		returnError bool
+		want        map[string]*prompb.TimeSeries
+	}{
+		{
+			"invalid_type_array",
+			otlp.Metric{
+				MetricDescriptor:    getDescriptor("invalid_type_array", histogramComb, validCombinations),
+				Int64DataPoints:     nil,
+				DoubleDataPoints:    nil,
+				HistogramDataPoints: nil,
+				SummaryDataPoints:   nil,
+			},
+			true,
+			map[string]*prompb.TimeSeries{},
+		},
+		{
+			"hist_nil_pt",
+			otlp.Metric{
+				MetricDescriptor:    getDescriptor("hist_nil_pt", histogramComb, validCombinations),
+				Int64DataPoints:     nil,
+				DoubleDataPoints:    nil,
+				HistogramDataPoints: []*otlp.HistogramDataPoint{nil},
+				SummaryDataPoints:   nil,
+			},
+			false,
+			map[string]*prompb.TimeSeries{},
+		},
+		{
+			"single_histogram_point",
+			otlp.Metric{
+				MetricDescriptor:    getDescriptor(name1+"", histogramComb, validCombinations),
+				Int64DataPoints:     nil,
+				DoubleDataPoints:    nil,
+				HistogramDataPoints: []*otlp.HistogramDataPoint{histPoint},
+				SummaryDataPoints:   nil,
+			},
+			false,
+			map[string]*prompb.TimeSeries{
+				sigs[sum]:       getTimeSeries(labels[sum], getSample(floatVal2, msTime1)),
+				sigs[count]:     getTimeSeries(labels[count], getSample(float64(intVal2), msTime1)),
+				sigs[bucket1]:   getTimeSeries(labels[bucket1], getSample(float64(intVal1), msTime1)),
+				sigs[bucket2]:   getTimeSeries(labels[bucket2], getSample(float64(intVal1), msTime1)),
+				sigs[bucketInf]: getTimeSeries(labels[bucketInf], getSample(float64(intVal2), msTime1)),
+			},
+		},
+	}
+
+	// run tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tsMap := map[string]*prompb.TimeSeries{}
+			prw := &prwExporter{}
+			ok := prw.handleHistogramMetric(tsMap, &tt.m)
+			if tt.returnError {
+				assert.Error(t, ok)
+				return
+			}
+			assert.Exactly(t, len(tt.want), len(tsMap))
+			for k, v := range tsMap {
+				require.NotNil(t, tt.want[k], k)
+				assert.ElementsMatch(t, tt.want[k].Labels, v.Labels)
+				assert.ElementsMatch(t, tt.want[k].Samples, v.Samples)
+			}
+		})
+	}
+}
+
+// Test_ newPrwExporter checks that a new exporter instance with non-nil fields is initialized
 func Test_newPrwExporter(t *testing.T) {
 	config := &Config{
 		ExporterSettings:   configmodels.ExporterSettings{},
@@ -263,7 +370,6 @@ func Test_export(t *testing.T) {
 		require.NotNil(t, writeReq.GetTimeseries())
 		assert.Equal(t, *ts1, writeReq.GetTimeseries()[0])
 		w.WriteHeader(code)
-		fmt.Fprintf(w, "error message")
 	}
 
 	// Create in test table format to check if different HTTP response codes or server errors
@@ -299,7 +405,9 @@ func Test_export(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-				handleFunc(w, r, tt.httpResponseCode)
+				if handleFunc != nil {
+					handleFunc(w, r, tt.httpResponseCode)
+				}
 			}))
 			defer server.Close()
 			serverURL, uErr := url.Parse(server.URL)
@@ -335,38 +443,65 @@ func runExportPipeline(t *testing.T, ts *prompb.TimeSeries, endpoint *url.URL) e
 // Test_pushMetrics checks the number of TimeSeries received by server and the number of metrics dropped is the same as
 // expected
 func Test_pushMetrics(t *testing.T) {
-	// fail cases
-	noTempBatch := pdatautil.MetricsFromOldInternalMetrics(testdataold.GenerateMetricDataManyMetricsSameResource(10))
-	invalidTypeBatch := pdatautil.MetricsFromOldInternalMetrics(testdataold.GenerateMetricDataMetricTypeInvalid())
 
-	invalidTemp := testdataold.GenerateMetricDataManyMetricsSameResource(10)
-	setTemporality(&invalidTemp, otlp.MetricDescriptor_INVALID_TEMPORALITY)
-	invalidTempBatch := pdatautil.MetricsFromOldInternalMetrics(invalidTemp)
+	noTempBatch := pdatautil.MetricsFromOldInternalMetrics((testdataold.GenerateMetricDataManyMetricsSameResource(10)))
+	invalidTypeBatch := pdatautil.MetricsFromOldInternalMetrics((testdataold.GenerateMetricDataMetricTypeInvalid()))
+	nilDescBatch := pdatautil.MetricsFromOldInternalMetrics((testdataold.GenerateMetricDataNilMetricDescriptor()))
+
+	// 10 counter metrics, 2 points in each. Two TimeSeries in total
+	batch := testdataold.GenerateMetricDataManyMetricsSameResource(10)
+	setCumulative(&batch)
+	scalarBatch := pdatautil.MetricsFromOldInternalMetrics((batch))
 
-	nilDescBatch := pdatautil.MetricsFromOldInternalMetrics(testdataold.GenerateMetricDataNilMetricDescriptor())
 	nilBatch1 := testdataold.GenerateMetricDataManyMetricsSameResource(10)
 	nilBatch2 := testdataold.GenerateMetricDataManyMetricsSameResource(10)
+	nilBatch3 := testdataold.GenerateMetricDataManyMetricsSameResource(10)
+	nilBatch4 := testdataold.GenerateMetricDataManyMetricsSameResource(10)
+	nilBatch5 := testdataold.GenerateMetricDataOneEmptyResourceMetrics()
+	nilBatch6 := testdataold.GenerateMetricDataOneEmptyInstrumentationLibrary()
+	nilBatch7 := testdataold.GenerateMetricDataOneMetric()
+
+	nilResource := dataold.MetricDataToOtlp(nilBatch5)
+	nilResource[0] = nil
+	nilResourceBatch := pdatautil.MetricsFromOldInternalMetrics(dataold.MetricDataFromOtlp(nilResource))
+
+	nilInstrumentation := dataold.MetricDataToOtlp(nilBatch6)
+	nilInstrumentation[0].InstrumentationLibraryMetrics[0] = nil
+	nilInstrumentationBatch := pdatautil.MetricsFromOldInternalMetrics(dataold.MetricDataFromOtlp(nilInstrumentation))
+
+	nilMetric := dataold.MetricDataToOtlp(nilBatch7)
+	nilMetric[0].InstrumentationLibraryMetrics[0].Metrics[0] = nil
+	nilMetricBatch := pdatautil.MetricsFromOldInternalMetrics(dataold.MetricDataFromOtlp(nilMetric))
+
+	setCumulative(&nilBatch1)
+	setCumulative(&nilBatch2)
+	setCumulative(&nilBatch3)
+	setCumulative(&nilBatch4)
 
-	setTemporality(&nilBatch1, otlp.MetricDescriptor_CUMULATIVE)
-	setTemporality(&nilBatch2, otlp.MetricDescriptor_CUMULATIVE)
 	setDataPointToNil(&nilBatch1, typeMonotonicInt64)
 	setType(&nilBatch2, typeMonotonicDouble)
+	setType(&nilBatch3, typeHistogram)
+	setType(&nilBatch4, typeSummary)
 
-	nilIntDataPointsBatch := pdatautil.MetricsFromOldInternalMetrics(nilBatch1)
-	nilDoubleDataPointsBatch := pdatautil.MetricsFromOldInternalMetrics(nilBatch2)
-
-	// Success cases: 10 counter metrics, 2 points in each. Two TimeSeries in total
-	batch1 := testdataold.GenerateMetricDataManyMetricsSameResource(10)
-	setTemporality(&batch1, otlp.MetricDescriptor_CUMULATIVE)
-	scalarBatch := pdatautil.MetricsFromOldInternalMetrics(batch1)
+	nilIntDataPointsBatch := pdatautil.MetricsFromOldInternalMetrics((nilBatch1))
+	nilDoubleDataPointsBatch := pdatautil.MetricsFromOldInternalMetrics((nilBatch2))
+	nilHistogramDataPointsBatch := pdatautil.MetricsFromOldInternalMetrics((nilBatch3))
 
-	// Partial Success cases
-	batch2 := testdataold.GenerateMetricDataManyMetricsSameResource(10)
-	setTemporality(&batch2, otlp.MetricDescriptor_CUMULATIVE)
-	failDesc := dataold.MetricDataToOtlp(batch2)[0].InstrumentationLibraryMetrics[0].Metrics[0].GetMetricDescriptor()
-	failDesc.Temporality = otlp.MetricDescriptor_INVALID_TEMPORALITY
-	partialBatch := pdatautil.MetricsFromOldInternalMetrics(batch2)
+	hist := dataold.MetricDataToOtlp(testdataold.GenerateMetricDataOneMetric())
+	hist[0].InstrumentationLibraryMetrics[0].Metrics[0] = &otlp.Metric{
+		MetricDescriptor: getDescriptor("hist_test", histogramComb, validCombinations),
+		HistogramDataPoints: []*otlp.HistogramDataPoint{getHistogramDataPoint(
+			lbs1,
+			time1,
+			floatVal1,
+			uint64(intVal1),
+			[]float64{floatVal1},
+			[]uint64{uint64(intVal1)},
+		),
+		},
+	}
 
+	histBatch := pdatautil.MetricsFromOldInternalMetrics((dataold.MetricDataFromOtlp(hist)))
 	checkFunc := func(t *testing.T, r *http.Request, expected int) {
 		body, err := ioutil.ReadAll(r.Body)
 		if err != nil {
@@ -385,6 +520,13 @@ func Test_pushMetrics(t *testing.T) {
 		assert.EqualValues(t, expected, len(wr.Timeseries))
 	}
 
+	summary := dataold.MetricDataToOtlp(testdataold.GenerateMetricDataOneMetric())
+	summary[0].InstrumentationLibraryMetrics[0].Metrics[0] = &otlp.Metric{
+		MetricDescriptor:  getDescriptor("summary_test", summaryComb, validCombinations),
+		SummaryDataPoints: []*otlp.SummaryDataPoint{},
+	}
+	summaryBatch := pdatautil.MetricsFromOldInternalMetrics(dataold.MetricDataFromOtlp(summary))
+
 	tests := []struct {
 		name                 string
 		md                   *pdata.Metrics
@@ -404,21 +546,39 @@ func Test_pushMetrics(t *testing.T) {
 			true,
 		},
 		{
-			"invalid_temporality_case",
-			&invalidTempBatch,
+			"nil_desc_case",
+			&nilDescBatch,
 			nil,
 			0,
 			http.StatusAccepted,
-			pdatautil.MetricCount(invalidTempBatch),
+			pdatautil.MetricCount(nilDescBatch),
 			true,
 		},
 		{
-			"nil_desc_case",
-			&nilDescBatch,
+			"nil_resourece_case",
+			&nilResourceBatch,
 			nil,
 			0,
 			http.StatusAccepted,
-			pdatautil.MetricCount(nilDescBatch),
+			pdatautil.MetricCount(nilResourceBatch),
+			false,
+		},
+		{
+			"nil_instrumentation_case",
+			&nilInstrumentationBatch,
+			nil,
+			0,
+			http.StatusAccepted,
+			pdatautil.MetricCount(nilInstrumentationBatch),
+			false,
+		},
+		{
+			"nil_metric_case",
+			&nilMetricBatch,
+			nil,
+			0,
+			http.StatusAccepted,
+			pdatautil.MetricCount(nilMetricBatch),
 			true,
 		},
 		{
@@ -439,6 +599,24 @@ func Test_pushMetrics(t *testing.T) {
 			pdatautil.MetricCount(nilDoubleDataPointsBatch),
 			true,
 		},
+		{
+			"nil_histogram_point_case",
+			&nilHistogramDataPointsBatch,
+			nil,
+			0,
+			http.StatusAccepted,
+			pdatautil.MetricCount(nilHistogramDataPointsBatch),
+			true,
+		},
+		{
+			"nil_histogram_point_case",
+			&nilHistogramDataPointsBatch,
+			nil,
+			0,
+			http.StatusAccepted,
+			pdatautil.MetricCount(nilHistogramDataPointsBatch),
+			true,
+		},
 		{
 			"no_temp_case",
 			&noTempBatch,
@@ -466,13 +644,20 @@ func Test_pushMetrics(t *testing.T) {
 			0,
 			false,
 		},
-		{
-			"partial_success_case",
-			&partialBatch,
+		{"histogram_case",
+			&histBatch,
 			checkFunc,
-			2,
+			4,
+			http.StatusAccepted,
+			0,
+			false,
+		},
+		{"summary_case",
+			&summaryBatch,
+			checkFunc,
+			0,
 			http.StatusAccepted,
-			1,
+			pdatautil.MetricCount(summaryBatch),
 			true,
 		},
 	}
