diff --git a/exporter/exportertest/sink_exporter.go b/exporter/exportertest/sink_exporter.go
index c25250f69..784eeb4cb 100644
--- a/exporter/exportertest/sink_exporter.go
+++ b/exporter/exportertest/sink_exporter.go
@@ -40,11 +40,14 @@ func (ste *SinkTraceExporter) ProcessTraceData(ctx context.Context, td data.Trac
 	return nil
 }
 
-const sinkExportFormat = "SinkExporter"
+const (
+	sinkTraceExportFormat   = "sink_trace"
+	sinkMetricsExportFormat = "sink_metrics"
+)
 
-// ExportFormat retruns the name of this TraceExporter
-func (ste *SinkTraceExporter) ExportFormat() string {
-	return sinkExportFormat
+// TraceExportFormat retruns the name of this TraceExporter
+func (ste *SinkTraceExporter) TraceExportFormat() string {
+	return sinkTraceExportFormat
 }
 
 // AllTraces returns the traces sent to the test sink.
@@ -73,9 +76,9 @@ func (sme *SinkMetricsExporter) ProcessMetricsData(ctx context.Context, md data.
 	return nil
 }
 
-// ExportFormat retruns the name of this TraceExporter
-func (sme *SinkMetricsExporter) ExportFormat() string {
-	return sinkExportFormat
+// MetricsExportFormat retruns the name of this MetricsExporter
+func (sme *SinkMetricsExporter) MetricsExportFormat() string {
+	return sinkMetricsExportFormat
 }
 
 // AllMetrics returns the metrics sent to the test sink.
