diff --git a/pdata/plog/generated_logrecordslice.go b/pdata/plog/generated_logrecordslice.go
index 757f991d8..b4419ba90 100644
--- a/pdata/plog/generated_logrecordslice.go
+++ b/pdata/plog/generated_logrecordslice.go
@@ -148,7 +148,7 @@ func (es LogRecordSlice) RemoveIf(f func(LogRecord) bool) {
 // CopyTo copies all elements from the current slice overriding the destination.
 func (es LogRecordSlice) CopyTo(dest LogRecordSlice) {
 	dest.state.AssertMutable()
-	*dest.orig = copyOrigLogRecordSlice(*dest.orig, *es.orig)
+	*dest.orig = internal.CopyOrigLogRecordSlice(*dest.orig, *es.orig)
 }
 
 // Sort sorts the LogRecord elements within LogRecordSlice given the
@@ -180,32 +180,3 @@ func (ms LogRecordSlice) unmarshalJSONIter(iter *json.Iterator) {
 		return true
 	})
 }
-
-func copyOrigLogRecordSlice(dest, src []*otlplogs.LogRecord) []*otlplogs.LogRecord {
-	var newDest []*otlplogs.LogRecord
-	if cap(dest) < len(src) {
-		newDest = make([]*otlplogs.LogRecord, len(src))
-		// Copy old pointers to re-use.
-		copy(newDest, dest)
-		// Add new pointers for missing elements from len(dest) to len(srt).
-		for i := len(dest); i < len(src); i++ {
-			newDest[i] = &otlplogs.LogRecord{}
-		}
-	} else {
-		newDest = dest[:len(src)]
-		// Cleanup the rest of the elements so GC can free the memory.
-		// This can happen when len(src) < len(dest) < cap(dest).
-		for i := len(src); i < len(dest); i++ {
-			dest[i] = nil
-		}
-		// Add new pointers for missing elements.
-		// This can happen when len(dest) < len(src) < cap(dest).
-		for i := len(dest); i < len(src); i++ {
-			newDest[i] = &otlplogs.LogRecord{}
-		}
-	}
-	for i := range src {
-		copyOrigLogRecord(newDest[i], src[i])
-	}
-	return newDest
-}
