diff --git a/extension/fluentbitextension/factory.go b/extension/fluentbitextension/factory.go
index bcc007e1d..396048834 100644
--- a/extension/fluentbitextension/factory.go
+++ b/extension/fluentbitextension/factory.go
@@ -19,6 +19,7 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/extension/extensionhelper"
 )
 
 const (
@@ -26,17 +27,15 @@ const (
 	typeStr = "fluentbit"
 )
 
-// ExtensionFactory is the factory for the extension.
-type Factory struct {
+// NewFactory creates a factory for FluentBit extension.
+func NewFactory() component.ExtensionFactory {
+	return extensionhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		createExtension)
 }
 
-// Type gets the type of the config created by this factory.
-func (f *Factory) Type() configmodels.Type {
-	return typeStr
-}
-
-// CreateDefaultConfig creates the default configuration for the extension.
-func (f *Factory) CreateDefaultConfig() configmodels.Extension {
+func createDefaultConfig() configmodels.Extension {
 	return &Config{
 		ExtensionSettings: configmodels.ExtensionSettings{
 			TypeVal: typeStr,
@@ -45,9 +44,7 @@ func (f *Factory) CreateDefaultConfig() configmodels.Extension {
 	}
 }
 
-// CreateExtension creates the extension based on this config.
-func (f *Factory) CreateExtension(_ context.Context, params component.ExtensionCreateParams, cfg configmodels.Extension) (component.ServiceExtension, error) {
+func createExtension(_ context.Context, params component.ExtensionCreateParams, cfg configmodels.Extension) (component.ServiceExtension, error) {
 	config := cfg.(*Config)
-
 	return newProcessManager(config, params.Logger), nil
 }
