diff --git a/pipeline/xpipeline/go.mod b/pipeline/xpipeline/go.mod
index df1c6f9c5..18af37d7f 100644
--- a/pipeline/xpipeline/go.mod
+++ b/pipeline/xpipeline/go.mod
@@ -2,6 +2,6 @@ module go.opentelemetry.io/collector/pipeline/xpipeline
 
 go 1.23.0
 
-require go.opentelemetry.io/collector/pipeline v0.130.0
+require go.opentelemetry.io/collector/pipeline v0.130.1
 
 replace go.opentelemetry.io/collector/pipeline => ../
