diff --git a/receiver/otlpreceiver/otlp_test.go b/receiver/otlpreceiver/otlp_test.go
index 23e2115a9..237ccc854 100644
--- a/receiver/otlpreceiver/otlp_test.go
+++ b/receiver/otlpreceiver/otlp_test.go
@@ -1204,30 +1204,30 @@ func (esc *errOrSinkConsumer) Reset() {
 }
 
 // Reset deletes any stored in the sinks, resets error to nil.
-func (esc *errOrSinkConsumer) checkData(t *testing.T, data any, len int) {
+func (esc *errOrSinkConsumer) checkData(t *testing.T, data any, dataLen int) {
 	switch data.(type) {
 	case ptrace.Traces:
 		allTraces := esc.TracesSink.AllTraces()
-		require.Len(t, allTraces, len)
-		if len > 0 {
+		require.Len(t, allTraces, dataLen)
+		if dataLen > 0 {
 			require.Equal(t, allTraces[0], data)
 		}
 	case pmetric.Metrics:
 		allMetrics := esc.MetricsSink.AllMetrics()
-		require.Len(t, allMetrics, len)
-		if len > 0 {
+		require.Len(t, allMetrics, dataLen)
+		if dataLen > 0 {
 			require.Equal(t, allMetrics[0], data)
 		}
 	case plog.Logs:
 		allLogs := esc.LogsSink.AllLogs()
-		require.Len(t, allLogs, len)
-		if len > 0 {
+		require.Len(t, allLogs, dataLen)
+		if dataLen > 0 {
 			require.Equal(t, allLogs[0], data)
 		}
 	case pprofile.Profiles:
 		allProfiles := esc.ProfilesSink.AllProfiles()
-		require.Len(t, allProfiles, len)
-		if len > 0 {
+		require.Len(t, allProfiles, dataLen)
+		if dataLen > 0 {
 			require.Equal(t, allProfiles[0], data)
 		}
 	}
