diff --git a/service/internal/telemetrylogs/logger.go b/service/internal/telemetrylogs/logger.go
index a903e3e0b..456385d26 100644
--- a/service/internal/telemetrylogs/logger.go
+++ b/service/internal/telemetrylogs/logger.go
@@ -15,8 +15,12 @@
 package telemetrylogs // import "go.opentelemetry.io/collector/service/internal/telemetrylogs"
 
 import (
+	"sync"
+
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zapgrpc"
+	"google.golang.org/grpc/grpclog"
 
 	"go.opentelemetry.io/collector/config"
 )
@@ -48,5 +52,44 @@ func NewLogger(cfg config.ServiceTelemetryLogs, options []zap.Option) (*zap.Logg
 	if err != nil {
 		return nil, err
 	}
+
 	return logger, nil
 }
+
+// SettableGRPCLoggerV2 sets grpc framework's logger with internal logger.
+type SettableGRPCLoggerV2 interface {
+	SetGRPCLogger()
+}
+
+type colGRPCLogger struct {
+	setOnce  sync.Once
+	loggerV2 grpclog.LoggerV2
+}
+
+// NewColGRPCLogger constructs a grpclog.LoggerV2 instance cloned from baseLogger with exact configuration.
+// The minimum level of gRPC logs is set to WARN should the loglevel of the collector is set to INFO to avoid
+// copious logging from grpc framework.
+func NewColGRPCLogger(baseLogger *zap.Logger, loglevel zapcore.Level) SettableGRPCLoggerV2 {
+	logger := baseLogger.WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+		var c zapcore.Core
+		if loglevel == zap.InfoLevel {
+			// NewIncreaseLevelCore errors only if the new log level is less than the initial core level.
+			// In this case it never happens as WARN is always greater than INFO, therefore ignoring it.
+			c, _ = zapcore.NewIncreaseLevelCore(core, zap.WarnLevel)
+		} else {
+			c = core
+		}
+		return c.With([]zapcore.Field{zap.Bool("grpc_log", true)})
+	}))
+	return &colGRPCLogger{
+		loggerV2: zapgrpc.NewLogger(logger),
+	}
+}
+
+// SetGRPCLogger needs to be run before any grpc calls and this implementation requires it to be run
+// only once.
+func (gl *colGRPCLogger) SetGRPCLogger() {
+	gl.setOnce.Do(func() {
+		grpclog.SetLoggerV2(gl.loggerV2)
+	})
+}
