diff --git a/exporter/exporterhelper/internal/base_exporter.go b/exporter/exporterhelper/internal/base_exporter.go
index 1a85eb32d..35b748b00 100644
--- a/exporter/exporterhelper/internal/base_exporter.go
+++ b/exporter/exporterhelper/internal/base_exporter.go
@@ -48,7 +48,6 @@ type BaseExporter struct {
 
 	queueBatchSettings QueueBatchSettings[request.Request]
 	queueCfg           queuebatch.Config
-	batcherCfg         BatcherConfig
 }
 
 func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, pusher sender.SendFunc[request.Request], options ...Option) (*BaseExporter, error) {
@@ -83,12 +82,12 @@ func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, pusher sende
 		return nil, err
 	}
 
-	if be.batcherCfg.Enabled || be.queueCfg.Batch != nil {
+	if be.queueCfg.Batch != nil {
 		// Batcher mutates the data.
 		be.ConsumerOptions = append(be.ConsumerOptions, consumer.WithCapabilities(consumer.Capabilities{MutatesData: true}))
 	}
 
-	if be.queueCfg.Enabled || be.batcherCfg.Enabled {
+	if be.queueCfg.Enabled {
 		qSet := queuebatch.Settings[request.Request]{
 			Signal:      signal,
 			ID:          set.ID,
@@ -98,7 +97,7 @@ func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, pusher sende
 			BytesSizer:  be.queueBatchSettings.BytesSizer,
 			Partitioner: be.queueBatchSettings.Partitioner,
 		}
-		be.QueueSender, err = NewQueueSender(qSet, be.queueCfg, be.batcherCfg, be.ExportFailureMessage, be.firstSender)
+		be.QueueSender, err = NewQueueSender(qSet, be.queueCfg, be.ExportFailureMessage, be.firstSender)
 		if err != nil {
 			return nil, err
 		}
@@ -233,18 +232,6 @@ func WithCapabilities(capabilities consumer.Capabilities) Option {
 	}
 }
 
-// WithBatcher enables batching for an exporter based on custom request types.
-// For now, it can be used only with the New[Traces|Metrics|Logs|Profiles]Request exporter helpers and
-// WithRequestBatchFuncs provided.
-// This API is at the early stage of development and may change without backward compatibility
-// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
-func WithBatcher(cfg BatcherConfig) Option {
-	return func(o *BaseExporter) error {
-		o.batcherCfg = cfg
-		return nil
-	}
-}
-
 // WithQueueBatchSettings is used to set the QueueBatchSettings for the new request based exporter helper.
 // It must be provided as the first option when creating a new exporter helper.
 func WithQueueBatchSettings(set QueueBatchSettings[request.Request]) Option {
