diff --git a/pdata/internal/generated_wrapper_valuetype.go b/pdata/internal/generated_wrapper_valuetype.go
index b9f61b2b9..a2e1b0e2c 100644
--- a/pdata/internal/generated_wrapper_valuetype.go
+++ b/pdata/internal/generated_wrapper_valuetype.go
@@ -35,7 +35,8 @@ func MarshalJSONOrigValueType(orig *otlpprofiles.ValueType, dest *json.Stream) {
 		dest.WriteObjectField("unitStrindex")
 		dest.WriteInt32(orig.UnitStrindex)
 	}
-	if orig.AggregationTemporality != otlpprofiles.AggregationTemporality(0) {
+
+	if int32(orig.AggregationTemporality) != 0 {
 		dest.WriteObjectField("aggregationTemporality")
 		dest.WriteInt32(int32(orig.AggregationTemporality))
 	}
