diff --git a/pdata/internal/generated_wrapper_summarydatapointslice.go b/pdata/internal/generated_wrapper_summarydatapointslice.go
index 7d5669c6a..6696c9836 100644
--- a/pdata/internal/generated_wrapper_summarydatapointslice.go
+++ b/pdata/internal/generated_wrapper_summarydatapointslice.go
@@ -49,19 +49,6 @@ func GenerateOrigTestSummaryDataPointSlice() []*otlpmetrics.SummaryDataPoint {
 	return orig
 }
 
-// MarshalJSONOrigSummaryDataPointSlice marshals all properties from the current struct to the destination stream.
-func MarshalJSONOrigSummaryDataPointSlice(orig []*otlpmetrics.SummaryDataPoint, dest *json.Stream) {
-	dest.WriteArrayStart()
-	if len(orig) > 0 {
-		MarshalJSONOrigSummaryDataPoint(orig[0], dest)
-	}
-	for i := 1; i < len(orig); i++ {
-		dest.WriteMore()
-		MarshalJSONOrigSummaryDataPoint(orig[i], dest)
-	}
-	dest.WriteArrayEnd()
-}
-
 // UnmarshalJSONOrigSummaryDataPointSlice unmarshals all properties from the current struct from the source iterator.
 func UnmarshalJSONOrigSummaryDataPointSlice(iter *json.Iterator) []*otlpmetrics.SummaryDataPoint {
 	var orig []*otlpmetrics.SummaryDataPoint
