diff --git a/pdata/internal/generated_wrapper_scopeprofiles.go b/pdata/internal/generated_wrapper_scopeprofiles.go
index 8ed6d925a..5af7bd070 100644
--- a/pdata/internal/generated_wrapper_scopeprofiles.go
+++ b/pdata/internal/generated_wrapper_scopeprofiles.go
@@ -31,7 +31,13 @@ func MarshalJSONOrigScopeProfiles(orig *otlpprofiles.ScopeProfiles, dest *json.S
 	MarshalJSONOrigInstrumentationScope(&orig.Scope, dest)
 	if len(orig.Profiles) > 0 {
 		dest.WriteObjectField("profiles")
-		MarshalJSONOrigProfileSlice(orig.Profiles, dest)
+		dest.WriteArrayStart()
+		MarshalJSONOrigProfile(orig.Profiles[0], dest)
+		for i := 1; i < len(orig.Profiles); i++ {
+			dest.WriteMore()
+			MarshalJSONOrigProfile(orig.Profiles[i], dest)
+		}
+		dest.WriteArrayEnd()
 	}
 	if orig.SchemaUrl != "" {
 		dest.WriteObjectField("schemaUrl")
