diff --git a/pdata/internal/generated_wrapper_numberdatapointslice.go b/pdata/internal/generated_wrapper_numberdatapointslice.go
index 39c59ec41..526fe675d 100644
--- a/pdata/internal/generated_wrapper_numberdatapointslice.go
+++ b/pdata/internal/generated_wrapper_numberdatapointslice.go
@@ -49,19 +49,6 @@ func GenerateOrigTestNumberDataPointSlice() []*otlpmetrics.NumberDataPoint {
 	return orig
 }
 
-// MarshalJSONOrigNumberDataPointSlice marshals all properties from the current struct to the destination stream.
-func MarshalJSONOrigNumberDataPointSlice(orig []*otlpmetrics.NumberDataPoint, dest *json.Stream) {
-	dest.WriteArrayStart()
-	if len(orig) > 0 {
-		MarshalJSONOrigNumberDataPoint(orig[0], dest)
-	}
-	for i := 1; i < len(orig); i++ {
-		dest.WriteMore()
-		MarshalJSONOrigNumberDataPoint(orig[i], dest)
-	}
-	dest.WriteArrayEnd()
-}
-
 // UnmarshalJSONOrigNumberDataPointSlice unmarshals all properties from the current struct from the source iterator.
 func UnmarshalJSONOrigNumberDataPointSlice(iter *json.Iterator) []*otlpmetrics.NumberDataPoint {
 	var orig []*otlpmetrics.NumberDataPoint
