diff --git a/pdata/internal/generated_wrapper_resourcelogsslice_test.go b/pdata/internal/generated_wrapper_resourcelogsslice_test.go
index 278595dcb..f05862e82 100644
--- a/pdata/internal/generated_wrapper_resourcelogsslice_test.go
+++ b/pdata/internal/generated_wrapper_resourcelogsslice_test.go
@@ -10,10 +10,8 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 
 	otlplogs "go.opentelemetry.io/collector/pdata/internal/data/protogen/logs/v1"
-	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestCopyOrigResourceLogsSlice(t *testing.T) {
@@ -40,19 +38,3 @@ func TestCopyOrigResourceLogsSlice(t *testing.T) {
 	dest = CopyOrigResourceLogsSlice(dest, src)
 	assert.Equal(t, GenerateOrigTestResourceLogsSlice(), dest)
 }
-
-func TestMarshalAndUnmarshalJSONOrigResourceLogsSlice(t *testing.T) {
-	src := GenerateOrigTestResourceLogsSlice()
-
-	stream := json.BorrowStream(nil)
-	defer json.ReturnStream(stream)
-	MarshalJSONOrigResourceLogsSlice(src, stream)
-	require.NoError(t, stream.Error())
-
-	iter := json.BorrowIterator(stream.Buffer())
-	defer json.ReturnIterator(iter)
-	dest := UnmarshalJSONOrigResourceLogsSlice(iter)
-	require.NoError(t, iter.Error())
-
-	assert.Equal(t, src, dest)
-}
