diff --git a/internal/processor/filterset/strict/strictfilterset_test.go b/internal/processor/filterset/strict/strictfilterset_test.go
index f2afa5e97..668b80f70 100644
--- a/internal/processor/filterset/strict/strictfilterset_test.go
+++ b/internal/processor/filterset/strict/strictfilterset_test.go
@@ -1,4 +1,4 @@
-// Copyright 2020 OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
