diff --git a/pdata/internal/generated_proto_histogramdatapoint.go b/pdata/internal/generated_proto_histogramdatapoint.go
index 7306f55eb..6d1bbde52 100644
--- a/pdata/internal/generated_proto_histogramdatapoint.go
+++ b/pdata/internal/generated_proto_histogramdatapoint.go
@@ -56,10 +56,10 @@ func DeleteHistogramDataPoint(orig *HistogramDataPoint, nullable bool) {
 		orig.Reset()
 		return
 	}
-
 	for i := range orig.Attributes {
 		DeleteKeyValue(&orig.Attributes[i], false)
 	}
+
 	for i := range orig.Exemplars {
 		DeleteExemplar(&orig.Exemplars[i], false)
 	}
