diff --git a/pdata/internal/generated_proto_requestcontext.go b/pdata/internal/generated_proto_requestcontext.go
index 4921ea97f..0dd5d7e8c 100644
--- a/pdata/internal/generated_proto_requestcontext.go
+++ b/pdata/internal/generated_proto_requestcontext.go
@@ -119,7 +119,6 @@ func DeleteRequestContext(orig *RequestContext, nullable bool) {
 		orig.Reset()
 		return
 	}
-
 	DeleteSpanContext(orig.SpanContext, true)
 	for i := range orig.ClientMetadata {
 		DeleteKeyValue(&orig.ClientMetadata[i], false)
@@ -141,9 +140,7 @@ func DeleteRequestContext(orig *RequestContext, nullable bool) {
 		DeleteUnixAddr(ov.Unix, true)
 		ov.Unix = nil
 		ProtoPoolRequestContext_Unix.Put(ov)
-
 	}
-
 	orig.Reset()
 	if nullable {
 		protoPoolRequestContext.Put(orig)
@@ -338,7 +335,6 @@ func (orig *RequestContext) UnmarshalJSON(iter *json.Iterator) {
 				ov.IP.UnmarshalJSON(iter)
 				orig.ClientAddress = ov
 			}
-
 		case "tCP":
 			{
 				var ov *RequestContext_TCP
@@ -351,7 +347,6 @@ func (orig *RequestContext) UnmarshalJSON(iter *json.Iterator) {
 				ov.TCP.UnmarshalJSON(iter)
 				orig.ClientAddress = ov
 			}
-
 		case "uDP":
 			{
 				var ov *RequestContext_UDP
@@ -364,7 +359,6 @@ func (orig *RequestContext) UnmarshalJSON(iter *json.Iterator) {
 				ov.UDP.UnmarshalJSON(iter)
 				orig.ClientAddress = ov
 			}
-
 		case "unix":
 			{
 				var ov *RequestContext_Unix
@@ -618,6 +612,7 @@ func (orig *RequestContext) UnmarshalProto(buf []byte) error {
 				return err
 			}
 			orig.ClientAddress = ov
+
 		default:
 			pos, err = proto.ConsumeUnknown(buf, pos, wireType)
 			if err != nil {
