diff --git a/consumer/pdata/trace.go b/consumer/pdata/trace.go
index 0a52fdee7..e0c5e536f 100644
--- a/consumer/pdata/trace.go
+++ b/consumer/pdata/trace.go
@@ -79,6 +79,18 @@ func (td Traces) SpanCount() int {
 	return spanCount
 }
 
+// Size returns size in bytes.
+func (td Traces) Size() int {
+	size := 0
+	for i := 0; i < len(*td.orig); i++ {
+		if (*td.orig)[i] == nil {
+			continue
+		}
+		size += (*td.orig)[i].Size()
+	}
+	return size
+}
+
 func (td Traces) ResourceSpans() ResourceSpansSlice {
 	return newResourceSpansSlice(td.orig)
 }
