diff --git a/exporter/exporterqueue/persistent_queue.go b/exporter/exporterqueue/persistent_queue.go
index 5e0195ece..f7665357a 100644
--- a/exporter/exporterqueue/persistent_queue.go
+++ b/exporter/exporterqueue/persistent_queue.go
@@ -273,14 +273,14 @@ func (pq *persistentQueue[T]) putInternal(ctx context.Context, req T) error {
 	return nil
 }
 
-func (pq *persistentQueue[T]) Read(ctx context.Context) (uint64, context.Context, T, bool) {
+func (pq *persistentQueue[T]) Read(ctx context.Context) (context.Context, T, DoneCallback, bool) {
 	pq.mu.Lock()
 	defer pq.mu.Unlock()
 
 	for {
 		if pq.stopped {
 			var req T
-			return 0, context.Background(), req, false
+			return context.Background(), req, nil, false
 		}
 
 		// Read until either a successful retrieved element or no more elements in the storage.
@@ -301,7 +301,7 @@ func (pq *persistentQueue[T]) Read(ctx context.Context) (uint64, context.Context
 				}
 				pq.hasMoreSpace.Signal()
 
-				return index, context.Background(), req, true
+				return context.Background(), req, func(processErr error) { pq.onDone(index, processErr) }, true
 			}
 		}
 
@@ -312,7 +312,7 @@ func (pq *persistentQueue[T]) Read(ctx context.Context) (uint64, context.Context
 }
 
 // getNextItem pulls the next available item from the persistent storage along with its index. Once processing is
-// finished, the index should be called with OnProcessingFinished to clean up the storage. If no new item is available,
+// finished, the index should be called with onDone to clean up the storage. If no new item is available,
 // returns false.
 func (pq *persistentQueue[T]) getNextItem(ctx context.Context) (uint64, T, bool) {
 	index := pq.readIndex
@@ -347,8 +347,8 @@ func (pq *persistentQueue[T]) getNextItem(ctx context.Context) (uint64, T, bool)
 	return index, request, true
 }
 
-// OnProcessingFinished should be called to remove the item of the given index from the queue once processing is finished.
-func (pq *persistentQueue[T]) OnProcessingFinished(index uint64, consumeErr error) {
+// onDone should be called to remove the item of the given index from the queue once processing is finished.
+func (pq *persistentQueue[T]) onDone(index uint64, consumeErr error) {
 	// Delete the item from the persistent storage after it was processed.
 	pq.mu.Lock()
 	// Always unref client even if the consumer is shutdown because we always ref it for every valid request.
