diff --git a/processor/batchprocessor/node_batcher.go b/processor/batchprocessor/node_batcher.go
index 1d1e53ee5..ef7cf1753 100644
--- a/processor/batchprocessor/node_batcher.go
+++ b/processor/batchprocessor/node_batcher.go
@@ -104,12 +104,12 @@ func (b *batchProcessor) GetCapabilities() component.ProcessorCapabilities {
 }
 
 // Start is invoked during service startup.
-func (b *batchProcessor) Start(host component.Host) error {
+func (b *batchProcessor) Start(ctx context.Context, host component.Host) error {
 	return nil
 }
 
 // Shutdown is invoked during service shutdown.
-func (b *batchProcessor) Shutdown() error {
+func (b *batchProcessor) Shutdown(context.Context) error {
 	// TODO: flush accumulated data.
 	return nil
 }
