diff --git a/processor/batchprocessor/metrics.go b/processor/batchprocessor/metrics.go
index 01f2d89fb..d10727813 100644
--- a/processor/batchprocessor/metrics.go
+++ b/processor/batchprocessor/metrics.go
@@ -6,9 +6,6 @@ package batchprocessor // import "go.opentelemetry.io/collector/processor/batchp
 import (
 	"context"
 
-	"go.opencensus.io/stats"
-	"go.opencensus.io/stats/view"
-	"go.opencensus.io/tag"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/metric"
 	"go.uber.org/multierr"
@@ -23,14 +20,6 @@ const (
 	scopeName = "go.opentelemetry.io/collector/processor/batchprocessor"
 )
 
-var (
-	processorTagKey          = tag.MustNewKey(obsmetrics.ProcessorKey)
-	statBatchSizeTriggerSend = stats.Int64("batch_size_trigger_send", "Number of times the batch was sent due to a size trigger", stats.UnitDimensionless)
-	statTimeoutTriggerSend   = stats.Int64("timeout_trigger_send", "Number of times the batch was sent due to a timeout trigger", stats.UnitDimensionless)
-	statBatchSendSize        = stats.Int64("batch_send_size", "Number of units in the batch", stats.UnitDimensionless)
-	statBatchSendSizeBytes   = stats.Int64("batch_send_size_bytes", "Number of bytes in batch that was sent", stats.UnitBytes)
-)
-
 type trigger int
 
 const (
@@ -39,57 +28,6 @@ const (
 	triggerBatchSize
 )
 
-func init() {
-	// TODO: Find a way to handle the error.
-	_ = view.Register(metricViews()...)
-}
-
-// MetricViews returns the metrics views related to batching
-func metricViews() []*view.View {
-	processorTagKeys := []tag.Key{processorTagKey}
-
-	countBatchSizeTriggerSendView := &view.View{
-		Name:        processorhelper.BuildCustomMetricName(typeStr, statBatchSizeTriggerSend.Name()),
-		Measure:     statBatchSizeTriggerSend,
-		Description: statBatchSizeTriggerSend.Description(),
-		TagKeys:     processorTagKeys,
-		Aggregation: view.Sum(),
-	}
-
-	countTimeoutTriggerSendView := &view.View{
-		Name:        processorhelper.BuildCustomMetricName(typeStr, statTimeoutTriggerSend.Name()),
-		Measure:     statTimeoutTriggerSend,
-		Description: statTimeoutTriggerSend.Description(),
-		TagKeys:     processorTagKeys,
-		Aggregation: view.Sum(),
-	}
-
-	distributionBatchSendSizeView := &view.View{
-		Name:        processorhelper.BuildCustomMetricName(typeStr, statBatchSendSize.Name()),
-		Measure:     statBatchSendSize,
-		Description: statBatchSendSize.Description(),
-		TagKeys:     processorTagKeys,
-		Aggregation: view.Distribution(10, 25, 50, 75, 100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000, 100000),
-	}
-
-	distributionBatchSendSizeBytesView := &view.View{
-		Name:        processorhelper.BuildCustomMetricName(typeStr, statBatchSendSizeBytes.Name()),
-		Measure:     statBatchSendSizeBytes,
-		Description: statBatchSendSizeBytes.Description(),
-		TagKeys:     processorTagKeys,
-		Aggregation: view.Distribution(10, 25, 50, 75, 100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000,
-			100_000, 200_000, 300_000, 400_000, 500_000, 600_000, 700_000, 800_000, 900_000,
-			1000_000, 2000_000, 3000_000, 4000_000, 5000_000, 6000_000, 7000_000, 8000_000, 9000_000),
-	}
-
-	return []*view.View{
-		countBatchSizeTriggerSendView,
-		countTimeoutTriggerSendView,
-		distributionBatchSendSizeView,
-		distributionBatchSendSizeBytesView,
-	}
-}
-
 type batchProcessorTelemetry struct {
 	level    configtelemetry.Level
 	detailed bool
@@ -105,19 +43,14 @@ type batchProcessorTelemetry struct {
 }
 
 func newBatchProcessorTelemetry(set processor.CreateSettings, currentMetadataCardinality func() int) (*batchProcessorTelemetry, error) {
-	exportCtx, err := tag.New(context.Background(), tag.Insert(processorTagKey, set.ID.String()))
-	if err != nil {
-		return nil, err
-	}
-
 	bpt := &batchProcessorTelemetry{
 		processorAttr: []attribute.KeyValue{attribute.String(obsmetrics.ProcessorKey, set.ID.String())},
-		exportCtx:     exportCtx,
+		exportCtx:     context.Background(),
 		level:         set.MetricsLevel,
 		detailed:      set.MetricsLevel == configtelemetry.LevelDetailed,
 	}
 
-	if err = bpt.createOtelMetrics(set.MeterProvider, currentMetadataCardinality); err != nil {
+	if err := bpt.createOtelMetrics(set.MeterProvider, currentMetadataCardinality); err != nil {
 		return nil, err
 	}
 
