diff --git a/exporter/exporterhelper/internal/queuebatch/batcher.go b/exporter/exporterhelper/internal/queuebatch/batcher.go
index 111b2cfd6..98c2e5a1f 100644
--- a/exporter/exporterhelper/internal/queuebatch/batcher.go
+++ b/exporter/exporterhelper/internal/queuebatch/batcher.go
@@ -7,6 +7,7 @@ import (
 	"context"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configoptional"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/queue"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/request"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/sender"
@@ -26,14 +27,14 @@ type batcherSettings[T any] struct {
 	maxWorkers  int
 }
 
-func NewBatcher(cfg *BatchConfig, set batcherSettings[request.Request]) Batcher[request.Request] {
-	if cfg == nil {
+func NewBatcher(cfg configoptional.Optional[BatchConfig], set batcherSettings[request.Request]) Batcher[request.Request] {
+	if !cfg.HasValue() {
 		return newDisabledBatcher[request.Request](set.next)
 	}
 
 	if set.partitioner == nil {
-		return newPartitionBatcher(*cfg, set.sizerType, set.sizer, newWorkerPool(set.maxWorkers), set.next)
+		return newPartitionBatcher(*cfg.Get(), set.sizerType, set.sizer, newWorkerPool(set.maxWorkers), set.next)
 	}
 
-	return newMultiBatcher(*cfg, set.sizerType, set.sizer, newWorkerPool(set.maxWorkers), set.partitioner, set.next)
+	return newMultiBatcher(*cfg.Get(), set.sizerType, set.sizer, newWorkerPool(set.maxWorkers), set.partitioner, set.next)
 }
