diff --git a/exporter/exporterhelper/internal/queuebatch/batcher.go b/exporter/exporterhelper/internal/queuebatch/batcher.go
index 6d4fcfb7e..7a13bbcca 100644
--- a/exporter/exporterhelper/internal/queuebatch/batcher.go
+++ b/exporter/exporterhelper/internal/queuebatch/batcher.go
@@ -44,5 +44,9 @@ func NewBatcher(cfg configoptional.Optional[BatchConfig], set batcherSettings[re
 		return newPartitionBatcher(*cfg.Get(), sizer, set.mergeCtx, newWorkerPool(set.maxWorkers), set.next, set.logger), nil
 	}
 
-	return newMultiBatcher(*cfg.Get(), sizer, newWorkerPool(set.maxWorkers), set.partitioner, set.mergeCtx, set.next, set.logger), nil
+	mb, err := newMultiBatcher(*cfg.Get(), sizer, newWorkerPool(set.maxWorkers), set.partitioner, set.mergeCtx, set.next, set.logger)
+	if err != nil {
+		return nil, fmt.Errorf("error during creating multi batcher: %w", err)
+	}
+	return mb, nil
 }
