diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 1ac50bf02..7e9d40a08 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -5,7 +5,7 @@ go 1.25.0
 require (
 	github.com/golang/snappy v1.0.0
 	github.com/klauspost/compress v1.18.4
-	github.com/pierrec/lz4/v4 v4.1.25
+	github.com/pierrec/lz4/v4 v4.1.26
 	github.com/rs/cors v1.11.1
 	github.com/stretchr/testify v1.11.1
 	go.opentelemetry.io/collector/client v1.53.0
