diff --git a/consumer/pdata/generated_metrics.go b/consumer/pdata/generated_metrics.go
index 2fed5d755..bca1c4c78 100644
--- a/consumer/pdata/generated_metrics.go
+++ b/consumer/pdata/generated_metrics.go
@@ -77,32 +77,27 @@ func (es ResourceMetricsSlice) MoveAndAppendTo(dest ResourceMetricsSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es ResourceMetricsSlice) CopyTo(dest ResourceMetricsSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.ResourceMetrics(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newResourceMetrics(&el).CopyTo(newResourceMetrics(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newResourceMetrics(&(*es.orig)[i]).CopyTo(newResourceMetrics(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.ResourceMetrics, newLen)
-	wrappers := make([]*otlpmetrics.ResourceMetrics, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.ResourceMetrics, srcLen)
+	wrappers := make([]*otlpmetrics.ResourceMetrics, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newResourceMetrics(&el).CopyTo(newResourceMetrics(&wrappers[i]))
+		newResourceMetrics(&(*es.orig)[i]).CopyTo(newResourceMetrics(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new ResourceMetricsSlice can be initialized:
 // es := NewResourceMetricsSlice()
@@ -112,22 +107,24 @@ func (es ResourceMetricsSlice) CopyTo(dest ResourceMetricsSlice) {
 //     // Here should set all the values for e.
 // }
 func (es ResourceMetricsSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.ResourceMetrics(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.ResourceMetrics, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.ResourceMetrics, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the ResourceMetricsSlice by one and set the
@@ -259,32 +256,27 @@ func (es InstrumentationLibraryMetricsSlice) MoveAndAppendTo(dest Instrumentatio
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es InstrumentationLibraryMetricsSlice) CopyTo(dest InstrumentationLibraryMetricsSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.InstrumentationLibraryMetrics(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newInstrumentationLibraryMetrics(&el).CopyTo(newInstrumentationLibraryMetrics(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newInstrumentationLibraryMetrics(&(*es.orig)[i]).CopyTo(newInstrumentationLibraryMetrics(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.InstrumentationLibraryMetrics, newLen)
-	wrappers := make([]*otlpmetrics.InstrumentationLibraryMetrics, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.InstrumentationLibraryMetrics, srcLen)
+	wrappers := make([]*otlpmetrics.InstrumentationLibraryMetrics, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newInstrumentationLibraryMetrics(&el).CopyTo(newInstrumentationLibraryMetrics(&wrappers[i]))
+		newInstrumentationLibraryMetrics(&(*es.orig)[i]).CopyTo(newInstrumentationLibraryMetrics(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new InstrumentationLibraryMetricsSlice can be initialized:
 // es := NewInstrumentationLibraryMetricsSlice()
@@ -294,22 +286,24 @@ func (es InstrumentationLibraryMetricsSlice) CopyTo(dest InstrumentationLibraryM
 //     // Here should set all the values for e.
 // }
 func (es InstrumentationLibraryMetricsSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.InstrumentationLibraryMetrics(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.InstrumentationLibraryMetrics, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.InstrumentationLibraryMetrics, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the InstrumentationLibraryMetricsSlice by one and set the
@@ -444,32 +438,27 @@ func (es MetricSlice) MoveAndAppendTo(dest MetricSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es MetricSlice) CopyTo(dest MetricSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.Metric(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newMetric(&el).CopyTo(newMetric(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newMetric(&(*es.orig)[i]).CopyTo(newMetric(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.Metric, newLen)
-	wrappers := make([]*otlpmetrics.Metric, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.Metric, srcLen)
+	wrappers := make([]*otlpmetrics.Metric, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newMetric(&el).CopyTo(newMetric(&wrappers[i]))
+		newMetric(&(*es.orig)[i]).CopyTo(newMetric(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new MetricSlice can be initialized:
 // es := NewMetricSlice()
@@ -479,22 +468,24 @@ func (es MetricSlice) CopyTo(dest MetricSlice) {
 //     // Here should set all the values for e.
 // }
 func (es MetricSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.Metric(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.Metric, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.Metric, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the MetricSlice by one and set the
@@ -1146,32 +1137,27 @@ func (es IntDataPointSlice) MoveAndAppendTo(dest IntDataPointSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es IntDataPointSlice) CopyTo(dest IntDataPointSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.IntDataPoint(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newIntDataPoint(&el).CopyTo(newIntDataPoint(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newIntDataPoint(&(*es.orig)[i]).CopyTo(newIntDataPoint(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.IntDataPoint, newLen)
-	wrappers := make([]*otlpmetrics.IntDataPoint, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.IntDataPoint, srcLen)
+	wrappers := make([]*otlpmetrics.IntDataPoint, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newIntDataPoint(&el).CopyTo(newIntDataPoint(&wrappers[i]))
+		newIntDataPoint(&(*es.orig)[i]).CopyTo(newIntDataPoint(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new IntDataPointSlice can be initialized:
 // es := NewIntDataPointSlice()
@@ -1181,22 +1167,24 @@ func (es IntDataPointSlice) CopyTo(dest IntDataPointSlice) {
 //     // Here should set all the values for e.
 // }
 func (es IntDataPointSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.IntDataPoint(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.IntDataPoint, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.IntDataPoint, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the IntDataPointSlice by one and set the
@@ -1373,32 +1361,27 @@ func (es DoubleDataPointSlice) MoveAndAppendTo(dest DoubleDataPointSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es DoubleDataPointSlice) CopyTo(dest DoubleDataPointSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.DoubleDataPoint(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newDoubleDataPoint(&el).CopyTo(newDoubleDataPoint(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newDoubleDataPoint(&(*es.orig)[i]).CopyTo(newDoubleDataPoint(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.DoubleDataPoint, newLen)
-	wrappers := make([]*otlpmetrics.DoubleDataPoint, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.DoubleDataPoint, srcLen)
+	wrappers := make([]*otlpmetrics.DoubleDataPoint, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newDoubleDataPoint(&el).CopyTo(newDoubleDataPoint(&wrappers[i]))
+		newDoubleDataPoint(&(*es.orig)[i]).CopyTo(newDoubleDataPoint(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new DoubleDataPointSlice can be initialized:
 // es := NewDoubleDataPointSlice()
@@ -1408,22 +1391,24 @@ func (es DoubleDataPointSlice) CopyTo(dest DoubleDataPointSlice) {
 //     // Here should set all the values for e.
 // }
 func (es DoubleDataPointSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.DoubleDataPoint(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.DoubleDataPoint, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.DoubleDataPoint, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the DoubleDataPointSlice by one and set the
@@ -1600,32 +1585,27 @@ func (es IntHistogramDataPointSlice) MoveAndAppendTo(dest IntHistogramDataPointS
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es IntHistogramDataPointSlice) CopyTo(dest IntHistogramDataPointSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.IntHistogramDataPoint(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newIntHistogramDataPoint(&el).CopyTo(newIntHistogramDataPoint(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newIntHistogramDataPoint(&(*es.orig)[i]).CopyTo(newIntHistogramDataPoint(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.IntHistogramDataPoint, newLen)
-	wrappers := make([]*otlpmetrics.IntHistogramDataPoint, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.IntHistogramDataPoint, srcLen)
+	wrappers := make([]*otlpmetrics.IntHistogramDataPoint, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newIntHistogramDataPoint(&el).CopyTo(newIntHistogramDataPoint(&wrappers[i]))
+		newIntHistogramDataPoint(&(*es.orig)[i]).CopyTo(newIntHistogramDataPoint(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new IntHistogramDataPointSlice can be initialized:
 // es := NewIntHistogramDataPointSlice()
@@ -1635,22 +1615,24 @@ func (es IntHistogramDataPointSlice) CopyTo(dest IntHistogramDataPointSlice) {
 //     // Here should set all the values for e.
 // }
 func (es IntHistogramDataPointSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.IntHistogramDataPoint(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.IntHistogramDataPoint, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.IntHistogramDataPoint, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the IntHistogramDataPointSlice by one and set the
@@ -1872,32 +1854,27 @@ func (es DoubleHistogramDataPointSlice) MoveAndAppendTo(dest DoubleHistogramData
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es DoubleHistogramDataPointSlice) CopyTo(dest DoubleHistogramDataPointSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.DoubleHistogramDataPoint(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newDoubleHistogramDataPoint(&el).CopyTo(newDoubleHistogramDataPoint(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newDoubleHistogramDataPoint(&(*es.orig)[i]).CopyTo(newDoubleHistogramDataPoint(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.DoubleHistogramDataPoint, newLen)
-	wrappers := make([]*otlpmetrics.DoubleHistogramDataPoint, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.DoubleHistogramDataPoint, srcLen)
+	wrappers := make([]*otlpmetrics.DoubleHistogramDataPoint, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newDoubleHistogramDataPoint(&el).CopyTo(newDoubleHistogramDataPoint(&wrappers[i]))
+		newDoubleHistogramDataPoint(&(*es.orig)[i]).CopyTo(newDoubleHistogramDataPoint(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new DoubleHistogramDataPointSlice can be initialized:
 // es := NewDoubleHistogramDataPointSlice()
@@ -1907,22 +1884,24 @@ func (es DoubleHistogramDataPointSlice) CopyTo(dest DoubleHistogramDataPointSlic
 //     // Here should set all the values for e.
 // }
 func (es DoubleHistogramDataPointSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.DoubleHistogramDataPoint(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.DoubleHistogramDataPoint, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.DoubleHistogramDataPoint, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the DoubleHistogramDataPointSlice by one and set the
@@ -2144,32 +2123,27 @@ func (es DoubleSummaryDataPointSlice) MoveAndAppendTo(dest DoubleSummaryDataPoin
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es DoubleSummaryDataPointSlice) CopyTo(dest DoubleSummaryDataPointSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.DoubleSummaryDataPoint(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newDoubleSummaryDataPoint(&el).CopyTo(newDoubleSummaryDataPoint(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newDoubleSummaryDataPoint(&(*es.orig)[i]).CopyTo(newDoubleSummaryDataPoint(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.DoubleSummaryDataPoint, newLen)
-	wrappers := make([]*otlpmetrics.DoubleSummaryDataPoint, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.DoubleSummaryDataPoint, srcLen)
+	wrappers := make([]*otlpmetrics.DoubleSummaryDataPoint, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newDoubleSummaryDataPoint(&el).CopyTo(newDoubleSummaryDataPoint(&wrappers[i]))
+		newDoubleSummaryDataPoint(&(*es.orig)[i]).CopyTo(newDoubleSummaryDataPoint(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new DoubleSummaryDataPointSlice can be initialized:
 // es := NewDoubleSummaryDataPointSlice()
@@ -2179,22 +2153,24 @@ func (es DoubleSummaryDataPointSlice) CopyTo(dest DoubleSummaryDataPointSlice) {
 //     // Here should set all the values for e.
 // }
 func (es DoubleSummaryDataPointSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.DoubleSummaryDataPoint(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.DoubleSummaryDataPoint, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.DoubleSummaryDataPoint, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the DoubleSummaryDataPointSlice by one and set the
@@ -2386,32 +2362,27 @@ func (es ValueAtQuantileSlice) MoveAndAppendTo(dest ValueAtQuantileSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es ValueAtQuantileSlice) CopyTo(dest ValueAtQuantileSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newValueAtQuantile(&el).CopyTo(newValueAtQuantile(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newValueAtQuantile(&(*es.orig)[i]).CopyTo(newValueAtQuantile(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile, newLen)
-	wrappers := make([]*otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile, srcLen)
+	wrappers := make([]*otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newValueAtQuantile(&el).CopyTo(newValueAtQuantile(&wrappers[i]))
+		newValueAtQuantile(&(*es.orig)[i]).CopyTo(newValueAtQuantile(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new ValueAtQuantileSlice can be initialized:
 // es := NewValueAtQuantileSlice()
@@ -2421,22 +2392,24 @@ func (es ValueAtQuantileSlice) CopyTo(dest ValueAtQuantileSlice) {
 //     // Here should set all the values for e.
 // }
 func (es ValueAtQuantileSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.DoubleSummaryDataPoint_ValueAtQuantile, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the ValueAtQuantileSlice by one and set the
@@ -2582,32 +2555,27 @@ func (es IntExemplarSlice) MoveAndAppendTo(dest IntExemplarSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es IntExemplarSlice) CopyTo(dest IntExemplarSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.IntExemplar(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newIntExemplar(&el).CopyTo(newIntExemplar(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newIntExemplar(&(*es.orig)[i]).CopyTo(newIntExemplar(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.IntExemplar, newLen)
-	wrappers := make([]*otlpmetrics.IntExemplar, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.IntExemplar, srcLen)
+	wrappers := make([]*otlpmetrics.IntExemplar, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newIntExemplar(&el).CopyTo(newIntExemplar(&wrappers[i]))
+		newIntExemplar(&(*es.orig)[i]).CopyTo(newIntExemplar(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new IntExemplarSlice can be initialized:
 // es := NewIntExemplarSlice()
@@ -2617,22 +2585,24 @@ func (es IntExemplarSlice) CopyTo(dest IntExemplarSlice) {
 //     // Here should set all the values for e.
 // }
 func (es IntExemplarSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.IntExemplar(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.IntExemplar, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.IntExemplar, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the IntExemplarSlice by one and set the
@@ -2789,32 +2759,27 @@ func (es DoubleExemplarSlice) MoveAndAppendTo(dest DoubleExemplarSlice) {
 
 // CopyTo copies all elements from the current slice to the dest.
 func (es DoubleExemplarSlice) CopyTo(dest DoubleExemplarSlice) {
-	newLen := es.Len()
-	if newLen == 0 {
-		*dest.orig = []*otlpmetrics.DoubleExemplar(nil)
-		return
-	}
-	oldLen := dest.Len()
-	if newLen <= oldLen {
-		(*dest.orig) = (*dest.orig)[:newLen]
-		for i, el := range *es.orig {
-			newDoubleExemplar(&el).CopyTo(newDoubleExemplar(&(*dest.orig)[i]))
+	srcLen := es.Len()
+	destCap := cap(*dest.orig)
+	if srcLen <= destCap {
+		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
+		for i := range *es.orig {
+			newDoubleExemplar(&(*es.orig)[i]).CopyTo(newDoubleExemplar(&(*dest.orig)[i]))
 		}
 		return
 	}
-	origs := make([]otlpmetrics.DoubleExemplar, newLen)
-	wrappers := make([]*otlpmetrics.DoubleExemplar, newLen)
-	for i, el := range *es.orig {
+	origs := make([]otlpmetrics.DoubleExemplar, srcLen)
+	wrappers := make([]*otlpmetrics.DoubleExemplar, srcLen)
+	for i := range *es.orig {
 		wrappers[i] = &origs[i]
-		newDoubleExemplar(&el).CopyTo(newDoubleExemplar(&wrappers[i]))
+		newDoubleExemplar(&(*es.orig)[i]).CopyTo(newDoubleExemplar(&wrappers[i]))
 	}
 	*dest.orig = wrappers
 }
 
 // Resize is an operation that resizes the slice:
-// 1. If newLen is 0 then the slice is replaced with a nil slice.
-// 2. If the newLen <= len then equivalent with slice[0:newLen].
-// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+// 1. If the newLen <= len then equivalent with slice[0:newLen:cap].
+// 2. If the newLen > len then (newLen - cap) empty elements will be appended to the slice.
 //
 // Here is how a new DoubleExemplarSlice can be initialized:
 // es := NewDoubleExemplarSlice()
@@ -2824,22 +2789,24 @@ func (es DoubleExemplarSlice) CopyTo(dest DoubleExemplarSlice) {
 //     // Here should set all the values for e.
 // }
 func (es DoubleExemplarSlice) Resize(newLen int) {
-	if newLen == 0 {
-		(*es.orig) = []*otlpmetrics.DoubleExemplar(nil)
-		return
-	}
 	oldLen := len(*es.orig)
+	oldCap := cap(*es.orig)
 	if newLen <= oldLen {
-		(*es.orig) = (*es.orig)[:newLen]
+		*es.orig = (*es.orig)[:newLen:oldCap]
 		return
 	}
-	// TODO: Benchmark and optimize this logic.
+
+	if newLen > oldCap {
+		newOrig := make([]*otlpmetrics.DoubleExemplar, oldLen, newLen)
+		copy(newOrig, *es.orig)
+		*es.orig = newOrig
+	}
+
+	// Add extra empty elements to the array.
 	extraOrigs := make([]otlpmetrics.DoubleExemplar, newLen-oldLen)
-	oldOrig := (*es.orig)
 	for i := range extraOrigs {
-		oldOrig = append(oldOrig, &extraOrigs[i])
+		*es.orig = append(*es.orig, &extraOrigs[i])
 	}
-	(*es.orig) = oldOrig
 }
 
 // Append will increase the length of the DoubleExemplarSlice by one and set the
