diff --git a/extension/ballastextension/go.mod b/extension/ballastextension/go.mod
index cfe0167d4..246fa7c6d 100644
--- a/extension/ballastextension/go.mod
+++ b/extension/ballastextension/go.mod
@@ -54,8 +54,6 @@ replace go.opentelemetry.io/collector/exporter => ../../exporter
 
 replace go.opentelemetry.io/collector/extension => ../
 
-replace go.opentelemetry.io/collector/extension/auth => ../auth
-
 replace go.opentelemetry.io/collector/featuregate => ../../featuregate
 
 replace go.opentelemetry.io/collector/pdata => ../../pdata
@@ -77,14 +75,6 @@ replace go.opentelemetry.io/collector/processor => ../../processor
 
 replace go.opentelemetry.io/collector/connector => ../../connector
 
-replace go.opentelemetry.io/collector/config/configcompression => ../../config/configcompression
-
 replace go.opentelemetry.io/collector/config/confignet => ../../config/confignet
 
-replace go.opentelemetry.io/collector/config/configopaque => ../../config/configopaque
-
-replace go.opentelemetry.io/collector/config/configtls => ../../config/configtls
-
 replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/configtelemetry
-
-replace go.opentelemetry.io/collector/config/configauth => ../../config/configauth
