diff --git a/config/internal/configsource/manager_test.go b/config/internal/configsource/manager_test.go
index 709139400..7eaf149c8 100644
--- a/config/internal/configsource/manager_test.go
+++ b/config/internal/configsource/manager_test.go
@@ -18,6 +18,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"os"
 	"path"
 	"testing"
 
@@ -81,15 +82,6 @@ func TestConfigSourceManager_ResolveErrors(t *testing.T) {
 		config          map[string]interface{}
 		configSourceMap map[string]ConfigSource
 	}{
-		{
-			name: "not_found_config_source",
-			config: map[string]interface{}{
-				"cfgsrc": "$unknown:test_selector",
-			},
-			configSourceMap: map[string]ConfigSource{
-				"tstcfgsrc": &testConfigSource{},
-			},
-		},
 		{
 			name: "incorrect_cfgsrc_ref",
 			config: map[string]interface{}{
@@ -335,38 +327,155 @@ func TestConfigSourceManager_MultipleWatchForUpdate(t *testing.T) {
 	assert.NoError(t, manager.Close(ctx))
 }
 
-func TestConfigSourceManager_expandConfigSources(t *testing.T) {
+func TestConfigSourceManager_EnvVarHandling(t *testing.T) {
+	require.NoError(t, os.Setenv("envvar", "envvar_value"))
+	defer func() {
+		assert.NoError(t, os.Unsetenv("envvar"))
+	}()
+
+	ctx := context.Background()
+	tstCfgSrc := testConfigSource{
+		ValueMap: map[string]valueEntry{
+			"int_key": {Value: 42},
+		},
+	}
+
+	// Intercept "params_key" and create an entry with the params themselves.
+	tstCfgSrc.OnRetrieve = func(ctx context.Context, selector string, params interface{}) error {
+		if selector == "params_key" {
+			tstCfgSrc.ValueMap[selector] = valueEntry{Value: params}
+		}
+		return nil
+	}
+
+	manager, err := NewManager(nil)
+	require.NoError(t, err)
+	manager.configSources = map[string]ConfigSource{
+		"tstcfgsrc": &tstCfgSrc,
+	}
+
+	file := path.Join("testdata", "envvar_cfgsrc_mix.yaml")
+	cp, err := config.NewParserFromFile(file)
+	require.NoError(t, err)
+
+	expectedFile := path.Join("testdata", "envvar_cfgsrc_mix_expected.yaml")
+	expectedParser, err := config.NewParserFromFile(expectedFile)
+	require.NoError(t, err)
+	expectedCfg := expectedParser.Viper().AllSettings()
+
+	res, err := manager.Resolve(ctx, cp)
+	require.NoError(t, err)
+	actualCfg := res.Viper().AllSettings()
+	assert.Equal(t, expectedCfg, actualCfg)
+	assert.NoError(t, manager.Close(ctx))
+}
+
+func TestManager_expandString(t *testing.T) {
 	ctx := context.Background()
 	csp, err := NewManager(nil)
 	require.NoError(t, err)
 	csp.configSources = map[string]ConfigSource{
 		"tstcfgsrc": &testConfigSource{
 			ValueMap: map[string]valueEntry{
-				"test_selector": {Value: "test_value"},
+				"str_key": {Value: "test_value"},
+				"int_key": {Value: 1},
 			},
 		},
 	}
 
-	v, err := csp.expandConfigSources(ctx, "")
-	assert.NoError(t, err)
-	assert.Equal(t, "", v)
-
-	v, err = csp.expandConfigSources(ctx, "no_cfgsrc")
-	assert.NoError(t, err)
-	assert.Equal(t, "no_cfgsrc", v)
-
-	// Not found config source.
-	v, err = csp.expandConfigSources(ctx, "$cfgsrc:selector")
-	assert.Error(t, err)
-	assert.Nil(t, v)
-
-	v, err = csp.expandConfigSources(ctx, "$tstcfgsrc:test_selector")
-	assert.NoError(t, err)
-	assert.Equal(t, "test_value", v)
+	require.NoError(t, os.Setenv("envvar", "envvar_value"))
+	defer func() {
+		assert.NoError(t, os.Unsetenv("envvar"))
+	}()
+	require.NoError(t, os.Setenv("envvar_str_key", "str_key"))
+	defer func() {
+		assert.NoError(t, os.Unsetenv("envvar_str_key"))
+	}()
 
-	v, err = csp.expandConfigSources(ctx, "$tstcfgsrc:invalid_selector")
-	assert.Error(t, err)
-	assert.Nil(t, v)
+	tests := []struct {
+		name    string
+		input   string
+		want    interface{}
+		wantErr error
+	}{
+		{
+			name:  "literal_string",
+			input: "literal_string",
+			want:  "literal_string",
+		},
+		{
+			name:  "escaped_$",
+			input: "$$tstcfgsrc:int_key$$envvar",
+			want:  "$tstcfgsrc:int_key$envvar",
+		},
+		{
+			name:  "cfgsrc_int",
+			input: "$tstcfgsrc:int_key",
+			want:  1,
+		},
+		{
+			name:  "concatenate_cfgsrc_string",
+			input: "prefix-$tstcfgsrc:str_key",
+			want:  "prefix-test_value",
+		},
+		{
+			name:  "concatenate_cfgsrc_non_string",
+			input: "prefix-$tstcfgsrc:int_key",
+			want:  "prefix-1",
+		},
+		{
+			name:  "envvar",
+			input: "$envvar",
+			want:  "envvar_value",
+		},
+		{
+			name:  "prefixed_envvar",
+			input: "prefix-$envvar",
+			want:  "prefix-envvar_value",
+		},
+		{
+			name:    "envvar_treated_as_cfgsrc",
+			input:   "$envvar:suffix",
+			wantErr: &errUnknownConfigSource{},
+		},
+		{
+			name:  "cfgsrc_using_envvar",
+			input: "$tstcfgsrc:$envvar_str_key",
+			want:  "test_value",
+		},
+		{
+			name:  "envvar_cfgsrc_using_envvar",
+			input: "$envvar/$tstcfgsrc:$envvar_str_key",
+			want:  "envvar_value/test_value",
+		},
+		{
+			name:  "delimited_cfgsrc",
+			input: "${tstcfgsrc:int_key}",
+			want:  1,
+		},
+		{
+			name:    "unknown_delimited_cfgsrc",
+			input:   "${cfgsrc:int_key}",
+			wantErr: &errUnknownConfigSource{},
+		},
+		{
+			name:  "delimited_cfgsrc_with_spaces",
+			input: "${ tstcfgsrc: int_key }",
+			want:  1,
+		},
+		{
+			name:  "interpolated_and_delimited_cfgsrc",
+			input: "0/${ tstcfgsrc: $envvar_str_key }/2/${tstcfgsrc:int_key}",
+			want:  "0/test_value/2/1",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := csp.expandString(ctx, tt.input)
+			require.IsType(t, tt.wantErr, err)
+			require.Equal(t, tt.want, got)
+		})
+	}
 }
 
 func Test_parseCfgSrc(t *testing.T) {
@@ -400,6 +509,17 @@ func Test_parseCfgSrc(t *testing.T) {
 				"p2": true,
 			},
 		},
+		{
+			name:       "query_pass_nil",
+			str:        "cfgsrc:selector?p0&p1&p2",
+			cfgSrcName: "cfgsrc",
+			selector:   "selector",
+			params: map[string]interface{}{
+				"p0": nil,
+				"p1": nil,
+				"p2": nil,
+			},
+		},
 		{
 			name:       "array_in_params",
 			str:        "cfgsrc:selector?p0=0&p0=1&p0=2&p1=done",
