diff --git a/internal/builder/config.go b/internal/builder/config.go
index b6d0ecdd9..fa253e882 100644
--- a/internal/builder/config.go
+++ b/internal/builder/config.go
@@ -27,7 +27,7 @@ import (
 	"go.uber.org/zap"
 )
 
-const defaultOtelColVersion = "0.35.0"
+const defaultOtelColVersion = "0.36.0"
 
 // ErrInvalidGoMod indicates an invalid gomod
 var ErrInvalidGoMod = errors.New("invalid gomod specification for module")
