diff --git a/processor/factory.go b/processor/factory.go
index f40ff74c6..d575910bd 100644
--- a/processor/factory.go
+++ b/processor/factory.go
@@ -19,8 +19,8 @@ import (
 
 	"go.uber.org/zap"
 
-	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
-	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
 )
 
 // Factory is factory interface for processors.
