diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index 150d2b24a..1d8979f5e 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -5,7 +5,7 @@ go 1.22.0
 require (
 	github.com/fsnotify/fsnotify v1.8.0
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/config/configopaque v1.24.0
+	go.opentelemetry.io/collector/config/configopaque v1.25.0
 )
 
 require (
