diff --git a/exporter/exportertest/go.mod b/exporter/exportertest/go.mod
index 0390a377a..d86c78194 100644
--- a/exporter/exportertest/go.mod
+++ b/exporter/exportertest/go.mod
@@ -4,7 +4,7 @@ go 1.24
 
 require (
 	github.com/google/uuid v1.6.0
-	github.com/stretchr/testify v1.11.0
+	github.com/stretchr/testify v1.11.1
 	go.opentelemetry.io/collector/component v1.39.0
 	go.opentelemetry.io/collector/component/componenttest v0.133.0
 	go.opentelemetry.io/collector/config/configretry v1.39.0
@@ -12,6 +12,7 @@ require (
 	go.opentelemetry.io/collector/consumer/consumererror v0.133.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.133.0
 	go.opentelemetry.io/collector/exporter v0.133.0
+	go.opentelemetry.io/collector/exporter/exporterhelper v0.0.0-00010101000000-000000000000
 	go.opentelemetry.io/collector/exporter/xexporter v0.133.0
 	go.opentelemetry.io/collector/pdata v1.39.0
 	go.opentelemetry.io/collector/pdata/pprofile v0.133.0
@@ -120,3 +121,5 @@ replace go.opentelemetry.io/collector/config/configoptional => ../../config/conf
 replace go.opentelemetry.io/collector/confmap => ../../confmap
 
 replace go.opentelemetry.io/collector/confmap/xconfmap => ../../confmap/xconfmap
+
+replace go.opentelemetry.io/collector/exporter/exporterhelper => ../exporterhelper
