diff --git a/exporter/exporterhelper/logs.go b/exporter/exporterhelper/logs.go
index f2219f4db..4f5b977b2 100644
--- a/exporter/exporterhelper/logs.go
+++ b/exporter/exporterhelper/logs.go
@@ -70,8 +70,8 @@ type logsExporter struct {
 	consumer.Logs
 }
 
-// NewLogsExporter creates an exporter.Logs that records observability metrics and wraps every request with a Span.
-func NewLogsExporter(
+// NewLogs creates an exporter.Logs that records observability metrics and wraps every request with a Span.
+func NewLogs(
 	ctx context.Context,
 	set exporter.Settings,
 	cfg component.Config,
@@ -88,9 +88,12 @@ func NewLogsExporter(
 		internal.WithMarshaler(logsRequestMarshaler), internal.WithUnmarshaler(newLogsRequestUnmarshalerFunc(pusher)),
 		internal.WithBatchFuncs(mergeLogs, mergeSplitLogs),
 	}
-	return NewLogsRequestExporter(ctx, set, requestFromLogs(pusher), append(logsOpts, options...)...)
+	return NewLogsRequest(ctx, set, requestFromLogs(pusher), append(logsOpts, options...)...)
 }
 
+// Deprecated: [v0.112.0] use NewLogs.
+var NewLogsExporter = NewLogs
+
 // RequestFromLogsFunc converts plog.Logs data into a user-defined request.
 // Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
@@ -103,10 +106,10 @@ func requestFromLogs(pusher consumer.ConsumeLogsFunc) RequestFromLogsFunc {
 	}
 }
 
-// NewLogsRequestExporter creates new logs exporter based on custom LogsConverter and RequestSender.
+// NewLogsRequest creates new logs exporter based on custom LogsConverter and RequestSender.
 // Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
-func NewLogsRequestExporter(
+func NewLogsRequest(
 	_ context.Context,
 	set exporter.Settings,
 	converter RequestFromLogsFunc,
@@ -120,7 +123,7 @@ func NewLogsRequestExporter(
 		return nil, errNilLogsConverter
 	}
 
-	be, err := internal.NewBaseExporter(set, pipeline.SignalLogs, newLogsExporterWithObservability, options...)
+	be, err := internal.NewBaseExporter(set, pipeline.SignalLogs, newLogsWithObservability, options...)
 	if err != nil {
 		return nil, err
 	}
@@ -146,12 +149,15 @@ func NewLogsRequestExporter(
 	}, err
 }
 
+// Deprecated: [v0.112.0] use NewLogsRequest.
+var NewLogsRequestExporter = NewLogsRequest
+
 type logsExporterWithObservability struct {
 	internal.BaseRequestSender
 	obsrep *internal.ObsReport
 }
 
-func newLogsExporterWithObservability(obsrep *internal.ObsReport) internal.RequestSender {
+func newLogsWithObservability(obsrep *internal.ObsReport) internal.RequestSender {
 	return &logsExporterWithObservability{obsrep: obsrep}
 }
 
