diff --git a/translator/trace/protospan_translation.go b/translator/trace/protospan_translation.go
index 5dba6dc1f..9623e26af 100644
--- a/translator/trace/protospan_translation.go
+++ b/translator/trace/protospan_translation.go
@@ -161,6 +161,7 @@ func AttributeMapToMap(attrMap pdata.AttributeMap) map[string]interface{} {
 	return rawMap
 }
 
+// AttributeArrayToSlice creates a slice out of a pdata.AnyValueArray.
 func AttributeArrayToSlice(attrArray pdata.AnyValueArray) []interface{} {
 	rawSlice := make([]interface{}, 0, attrArray.Len())
 	for i := 0; i < attrArray.Len(); i++ {
