diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index eaec52e10..0538b67c9 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -63,7 +63,7 @@ require (
 	github.com/tklauser/numcpus v0.6.0 // indirect
 	github.com/yusufpapurcu/wmi v1.2.2 // indirect
 	go.opencensus.io v0.24.0 // indirect
-	go.opentelemetry.io/collector/confmap v0.66.0 // indirect
+	go.opentelemetry.io/collector/confmap v0.0.0-20221201172708-2bdff61fa52a // indirect
 	go.opentelemetry.io/collector/consumer v0.66.0 // indirect
 	go.opentelemetry.io/collector/featuregate v0.66.0 // indirect
 	go.opentelemetry.io/collector/pdata v0.66.0 // indirect
