diff --git a/receiver/receivertest/nop_receiver.go b/receiver/receivertest/nop_receiver.go
index a144b7828..2e7e8e721 100644
--- a/receiver/receivertest/nop_receiver.go
+++ b/receiver/receivertest/nop_receiver.go
@@ -83,14 +83,3 @@ type nopReceiver struct {
 	component.StartFunc
 	component.ShutdownFunc
 }
-
-// NewNopBuilder returns a receiver.Builder that constructs nop receivers.
-//
-// Deprecated: [v0.108.0] this builder is being internalized within the service module,
-// and will be removed soon.
-func NewNopBuilder() *receiver.Builder {
-	nopFactory := NewNopFactory()
-	return receiver.NewBuilder(
-		map[component.ID]component.Config{component.NewID(defaultComponentType): nopFactory.CreateDefaultConfig()},
-		map[component.Type]receiver.Factory{defaultComponentType: nopFactory})
-}
