diff --git a/config/configgrpc/configgrpc.go b/config/configgrpc/configgrpc.go
index 1b1b15963..5def5bbb0 100644
--- a/config/configgrpc/configgrpc.go
+++ b/config/configgrpc/configgrpc.go
@@ -16,6 +16,8 @@ import (
 	"github.com/mostynb/go-grpc-compression/nonclobbering/zstd"
 	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/noop"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/balancer"
 	"google.golang.org/grpc/codes"
@@ -337,7 +339,7 @@ func (gcs *ClientConfig) getGrpcDialOptions(
 	otelOpts := []otelgrpc.Option{
 		otelgrpc.WithTracerProvider(settings.TracerProvider),
 		otelgrpc.WithPropagators(otel.GetTextMapPropagator()),
-		otelgrpc.WithMeterProvider(settings.LeveledMeterProvider(configtelemetry.LevelDetailed)),
+		otelgrpc.WithMeterProvider(getLeveledMeterProvider(settings)),
 	}
 
 	// Enable OpenTelemetry observability plugin.
@@ -481,7 +483,7 @@ func (gss *ServerConfig) getGrpcServerOptions(
 	otelOpts := []otelgrpc.Option{
 		otelgrpc.WithTracerProvider(settings.TracerProvider),
 		otelgrpc.WithPropagators(otel.GetTextMapPropagator()),
-		otelgrpc.WithMeterProvider(settings.LeveledMeterProvider(configtelemetry.LevelDetailed)),
+		otelgrpc.WithMeterProvider(getLeveledMeterProvider(settings)),
 	}
 
 	// Enable OpenTelemetry observability plugin.
@@ -575,3 +577,10 @@ func authStreamServerInterceptor(srv any, stream grpc.ServerStream, _ *grpc.Stre
 
 	return handler(srv, wrapServerStream(ctx, stream))
 }
+
+func getLeveledMeterProvider(settings component.TelemetrySettings) metric.MeterProvider {
+	if configtelemetry.LevelDetailed <= settings.MetricsLevel {
+		return settings.MeterProvider
+	}
+	return noop.MeterProvider{}
+}
