diff --git a/internal/collector/processor/nodebatcher/factory_test.go b/internal/collector/processor/nodebatcher/factory_test.go
index 6ecffefbb..1bd5712f8 100644
--- a/internal/collector/processor/nodebatcher/factory_test.go
+++ b/internal/collector/processor/nodebatcher/factory_test.go
@@ -22,7 +22,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/open-telemetry/opentelemetry-service/internal/factories"
+	"github.com/open-telemetry/opentelemetry-service/pkg/factories"
 )
 
 func TestCreateDefaultConfig(t *testing.T) {
