diff --git a/confmap/go.mod b/confmap/go.mod
index e340d653c..21d9c3f1c 100644
--- a/confmap/go.mod
+++ b/confmap/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/knadh/koanf v1.4.4
 	github.com/mitchellh/mapstructure v1.5.0
 	github.com/stretchr/testify v1.8.1
-	go.opentelemetry.io/collector/featuregate v0.65.0
+	go.opentelemetry.io/collector/featuregate v0.67.0
 	go.uber.org/multierr v1.8.0
 	gopkg.in/yaml.v3 v3.0.1
 )
