diff --git a/pdata/pprofile/function_test.go b/pdata/pprofile/function_test.go
index 1de4f9d46..f5ef9b625 100644
--- a/pdata/pprofile/function_test.go
+++ b/pdata/pprofile/function_test.go
@@ -9,6 +9,8 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/internal/testutil"
 )
 
 func TestFunctionEqual(t *testing.T) {
@@ -252,6 +254,28 @@ func TestFunctionSwitchDictionary(t *testing.T) {
 	}
 }
 
+func BenchmarkFunctionSwitchDictionary(b *testing.B) {
+	testutil.SkipMemoryBench(b)
+
+	fn := NewFunction()
+	fn.SetNameStrindex(1)
+	fn.SetSystemNameStrindex(2)
+
+	src := NewProfilesDictionary()
+	src.StringTable().Append("", "test", "foo")
+
+	b.ReportAllocs()
+
+	for b.Loop() {
+		b.StopTimer()
+		dst := NewProfilesDictionary()
+		dst.StringTable().Append("", "foo")
+		b.StartTimer()
+
+		_ = fn.switchDictionary(src, dst)
+	}
+}
+
 func buildFunction(name, sName, fileName int32, startLine int64) Function {
 	f := NewFunction()
 	f.SetNameStrindex(name)
