diff --git a/pdata/pprofile/scopeprofiles_test.go b/pdata/pprofile/scopeprofiles_test.go
index 784146eb2..d080567b6 100644
--- a/pdata/pprofile/scopeprofiles_test.go
+++ b/pdata/pprofile/scopeprofiles_test.go
@@ -9,6 +9,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"go.opentelemetry.io/collector/internal/testutil"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
@@ -89,3 +90,23 @@ func TestScopeProfilesSwitchDictionary(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkScopeProfilesSwitchDictionary(b *testing.B) {
+	testutil.SkipMemoryBench(b)
+
+	s := NewScopeProfiles()
+	profile := s.Profiles().AppendEmpty()
+	profile.Samples().AppendEmpty().SetLinkIndex(1)
+
+	src := NewProfilesDictionary()
+	src.LinkTable().AppendEmpty()
+	src.LinkTable().AppendEmpty().SetSpanID(pcommon.SpanID([8]byte{1, 2, 3, 4, 5, 6, 7, 8}))
+
+	dst := NewProfilesDictionary()
+
+	b.ReportAllocs()
+
+	for b.Loop() {
+		_ = s.switchDictionary(src, dst)
+	}
+}
