diff --git a/internal/zpagesserver/zpagesserver.go b/internal/zpagesserver/zpagesserver.go
index 6436d28a2..0ff5f1770 100644
--- a/internal/zpagesserver/zpagesserver.go
+++ b/internal/zpagesserver/zpagesserver.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
