diff --git a/extension/ballastextension/go.mod b/extension/ballastextension/go.mod
index 9bc8c5c5f..8f13cbf2d 100644
--- a/extension/ballastextension/go.mod
+++ b/extension/ballastextension/go.mod
@@ -58,6 +58,4 @@ replace go.opentelemetry.io/collector/semconv => ../../semconv
 
 replace go.opentelemetry.io/collector/extension/zpagesextension => ../zpagesextension
 
-replace go.opentelemetry.io/collector/processor/batchprocessor => ../../processor/batchprocessor
-
 replace go.opentelemetry.io/collector/consumer => ../../consumer
