diff --git a/internal/fanoutconsumer/profiles_test.go b/internal/fanoutconsumer/profiles_test.go
index 58d01119f..eab0d51bb 100644
--- a/internal/fanoutconsumer/profiles_test.go
+++ b/internal/fanoutconsumer/profiles_test.go
@@ -82,13 +82,13 @@ func TestProfilesMultiplexingMutating(t *testing.T) {
 		}
 	}
 
-	assert.NotSame(t, td, p1.AllProfiles()[0])
-	assert.NotSame(t, td, p1.AllProfiles()[1])
+	assert.NotSame(t, &td, &p1.AllProfiles()[0])
+	assert.NotSame(t, &td, &p1.AllProfiles()[1])
 	assert.EqualValues(t, td, p1.AllProfiles()[0])
 	assert.EqualValues(t, td, p1.AllProfiles()[1])
 
-	assert.NotSame(t, td, p2.AllProfiles()[0])
-	assert.NotSame(t, td, p2.AllProfiles()[1])
+	assert.NotSame(t, &td, &p2.AllProfiles()[0])
+	assert.NotSame(t, &td, &p2.AllProfiles()[1])
 	assert.EqualValues(t, td, p2.AllProfiles()[0])
 	assert.EqualValues(t, td, p2.AllProfiles()[1])
 
@@ -157,8 +157,8 @@ func TestProfilesMultiplexingMixLastMutating(t *testing.T) {
 		}
 	}
 
-	assert.NotSame(t, td, p1.AllProfiles()[0])
-	assert.NotSame(t, td, p1.AllProfiles()[1])
+	assert.NotSame(t, &td, &p1.AllProfiles()[0])
+	assert.NotSame(t, &td, &p1.AllProfiles()[1])
 	assert.EqualValues(t, td, p1.AllProfiles()[0])
 	assert.EqualValues(t, td, p1.AllProfiles()[1])
 
@@ -169,8 +169,8 @@ func TestProfilesMultiplexingMixLastMutating(t *testing.T) {
 	assert.EqualValues(t, td, p2.AllProfiles()[1])
 
 	// For this consumer, will clone the initial data.
-	assert.NotSame(t, td, p3.AllProfiles()[0])
-	assert.NotSame(t, td, p3.AllProfiles()[1])
+	assert.NotSame(t, &td, &p3.AllProfiles()[0])
+	assert.NotSame(t, &td, &p3.AllProfiles()[1])
 	assert.EqualValues(t, td, p3.AllProfiles()[0])
 	assert.EqualValues(t, td, p3.AllProfiles()[1])
 
@@ -195,13 +195,13 @@ func TestProfilesMultiplexingMixLastNonMutating(t *testing.T) {
 		}
 	}
 
-	assert.NotSame(t, td, p1.AllProfiles()[0])
-	assert.NotSame(t, td, p1.AllProfiles()[1])
+	assert.NotSame(t, &td, &p1.AllProfiles()[0])
+	assert.NotSame(t, &td, &p1.AllProfiles()[1])
 	assert.EqualValues(t, td, p1.AllProfiles()[0])
 	assert.EqualValues(t, td, p1.AllProfiles()[1])
 
-	assert.NotSame(t, td, p2.AllProfiles()[0])
-	assert.NotSame(t, td, p2.AllProfiles()[1])
+	assert.NotSame(t, &td, &p2.AllProfiles()[0])
+	assert.NotSame(t, &td, &p2.AllProfiles()[1])
 	assert.EqualValues(t, td, p2.AllProfiles()[0])
 	assert.EqualValues(t, td, p2.AllProfiles()[1])
 
