diff --git a/pdata/pmetric/generated_resourcemetricsslice_test.go b/pdata/pmetric/generated_resourcemetricsslice_test.go
index 6b561907c..a090587b0 100644
--- a/pdata/pmetric/generated_resourcemetricsslice_test.go
+++ b/pdata/pmetric/generated_resourcemetricsslice_test.go
@@ -52,6 +52,8 @@ func TestResourceMetricsSlice_CopyTo(t *testing.T) {
 	src := generateTestResourceMetricsSlice()
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestResourceMetricsSlice(), dest)
+	dest.CopyTo(dest)
+	assert.Equal(t, generateTestResourceMetricsSlice(), dest)
 }
 
 func TestResourceMetricsSlice_EnsureCapacity(t *testing.T) {
