diff --git a/receiver/jaegerreceiver/factory.go b/receiver/jaegerreceiver/factory.go
index c8aa207fa..15c217d4f 100644
--- a/receiver/jaegerreceiver/factory.go
+++ b/receiver/jaegerreceiver/factory.go
@@ -111,9 +111,9 @@ func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
 // CreateTraceReceiver creates a trace receiver based on provided config.
 func (f *Factory) CreateTraceReceiver(
 	ctx context.Context,
-	logger *zap.Logger,
+	params component.ReceiverCreateParams,
 	cfg configmodels.Receiver,
-	nextConsumer consumer.TraceConsumerOld,
+	nextConsumer consumer.TraceConsumer,
 ) (component.TraceReceiver, error) {
 
 	// Convert settings in the source config to Configuration struct
@@ -130,6 +130,7 @@ func (f *Factory) CreateTraceReceiver(
 
 	config := Configuration{}
 	var grpcServerOptions []grpc.ServerOption
+	logger := params.Logger
 
 	// Set ports
 	if protoGRPC != nil && protoGRPC.IsEnabled() {
@@ -217,14 +218,15 @@ func (f *Factory) CreateTraceReceiver(
 	}
 
 	// Create the receiver.
-	return New(rCfg.Name(), &config, nextConsumer, logger)
+	return New(rCfg.Name(), &config, nextConsumer, params)
 }
 
 // CreateMetricsReceiver creates a metrics receiver based on provided config.
 func (f *Factory) CreateMetricsReceiver(
-	logger *zap.Logger,
-	cfg configmodels.Receiver,
-	consumer consumer.MetricsConsumerOld,
+	_ context.Context,
+	_ component.ReceiverCreateParams,
+	_ configmodels.Receiver,
+	_ consumer.MetricsConsumer,
 ) (component.MetricsReceiver, error) {
 	return nil, configerror.ErrDataTypeIsNotSupported
 }
