diff --git a/service/telemetry/telemetry_test.go b/service/telemetry/telemetry_test.go
index 3a46c3cd8..25c42d018 100644
--- a/service/telemetry/telemetry_test.go
+++ b/service/telemetry/telemetry_test.go
@@ -9,9 +9,6 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	"go.opentelemetry.io/otel/trace"
-	"go.opentelemetry.io/otel/trace/noop"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 
@@ -122,26 +119,3 @@ func TestSampledLogger(t *testing.T) {
 		})
 	}
 }
-
-func TestTelemetryShutdown(t *testing.T) {
-	tests := []struct {
-		name     string
-		provider trace.TracerProvider
-		wantErr  error
-	}{
-		{
-			name: "No provider",
-		},
-		{
-			name:     "Non-SDK provider",
-			provider: noop.NewTracerProvider(),
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			telemetry := Telemetry{tracerProvider: tt.provider}
-			err := telemetry.Shutdown(context.Background())
-			require.Equal(t, tt.wantErr, err)
-		})
-	}
-}
