diff --git a/component/component.go b/component/component.go
index 90397a89a..c5574eb67 100644
--- a/component/component.go
+++ b/component/component.go
@@ -108,7 +108,7 @@ func (k Kind) String() string {
 // StabilityLevel represents the stability level of the component created by the factory.
 // The stability level is used to determine if the component should be used in production
 // or not. For more details see:
-// https://github.com/open-telemetry/opentelemetry-collector#stability-levels
+// https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-stability.md#stability-levels
 type StabilityLevel int
 
 const (
