diff --git a/internal/testdata/metric_test.go b/internal/testdata/metric_test.go
index ea4f17f09..3e0b6a6b7 100644
--- a/internal/testdata/metric_test.go
+++ b/internal/testdata/metric_test.go
@@ -26,60 +26,55 @@ import (
 
 type traceMetricsCase struct {
 	name string
-	td   pdata.Metrics
+	md   pdata.Metrics
 	otlp *otlpcollectormetrics.ExportMetricsServiceRequest
 }
 
 func generateAllMetricsTestCases() []traceMetricsCase {
 	return []traceMetricsCase{
-		{
-			name: "empty",
-			td:   GenerateMetricsEmpty(),
-			otlp: generateMetricsOtlpEmpty(),
-		},
 		{
 			name: "one-empty-resource-metrics",
-			td:   GenerateMetricsOneEmptyResourceMetrics(),
+			md:   GenerateMetricsOneEmptyResourceMetrics(),
 			otlp: generateMetricsOtlpOneEmptyResourceMetrics(),
 		},
 		{
 			name: "no-libraries",
-			td:   GenerateMetricsNoLibraries(),
+			md:   GenerateMetricsNoLibraries(),
 			otlp: generateMetricsOtlpNoLibraries(),
 		},
 		{
 			name: "one-empty-instrumentation-library",
-			td:   GenerateMetricsOneEmptyInstrumentationLibrary(),
+			md:   GenerateMetricsOneEmptyInstrumentationLibrary(),
 			otlp: generateMetricsOtlpOneEmptyInstrumentationLibrary(),
 		},
 		{
 			name: "one-metric-no-resource",
-			td:   GenerateMetricsOneMetricNoResource(),
+			md:   GenerateMetricsOneMetricNoResource(),
 			otlp: generateMetricsOtlpOneMetricNoResource(),
 		},
 		{
 			name: "one-metric",
-			td:   GenerateMetricsOneMetric(),
+			md:   GenerateMetricsOneMetric(),
 			otlp: generateMetricsOtlpOneMetric(),
 		},
 		{
 			name: "two-metrics",
-			td:   GenerateMetricsTwoMetrics(),
+			md:   GenerateMetricsTwoMetrics(),
 			otlp: generateMetricsOtlpTwoMetrics(),
 		},
 		{
 			name: "one-metric-no-labels",
-			td:   GenerateMetricsOneMetricNoLabels(),
+			md:   GenerateMetricsOneMetricNoLabels(),
 			otlp: generateMetricsOtlpOneMetricNoLabels(),
 		},
 		{
 			name: "all-types-no-data-points",
-			td:   GenerateMetricsAllTypesNoDataPoints(),
+			md:   GenerateMetricsAllTypesNoDataPoints(),
 			otlp: generateMetricsOtlpAllTypesNoDataPoints(),
 		},
 		{
 			name: "all-metric-types",
-			td:   GeneratMetricsAllTypesWithSampleDatapoints(),
+			md:   GeneratMetricsAllTypesWithSampleDatapoints(),
 			otlp: generateMetricsOtlpAllTypesWithSampleDatapoints(),
 		},
 	}
@@ -92,7 +87,7 @@ func TestToFromOtlpMetrics(t *testing.T) {
 		test := allTestCases[i]
 		t.Run(test.name, func(t *testing.T) {
 			td := pdata.MetricsFromInternalRep(internal.MetricsFromOtlp(test.otlp))
-			assert.EqualValues(t, test.td, td)
+			assert.EqualValues(t, test.md, td)
 			otlp := internal.MetricsToOtlp(td.InternalRep())
 			assert.EqualValues(t, test.otlp, otlp)
 		})
