diff --git a/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_test.go b/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_test.go
index 12d2aee33..7e92ca825 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_test.go
@@ -28,6 +28,7 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/internal/processor/filterset"
 	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
+	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal/metadata"
 )
 
 func TestScrape(t *testing.T) {
@@ -216,7 +217,7 @@ func TestScrape(t *testing.T) {
 			assertFileSystemUsageMetricValid(
 				t,
 				metrics.At(0),
-				fileSystemUsageDescriptor,
+				metadata.Metrics.SystemFilesystemUsage.New(),
 				test.expectedDeviceDataPoints*fileSystemStatesLen,
 				test.expectedDeviceLabelValues,
 			)
@@ -226,7 +227,7 @@ func TestScrape(t *testing.T) {
 				assertFileSystemUsageMetricValid(
 					t,
 					metrics.At(1),
-					fileSystemINodesUsageDescriptor,
+					metadata.Metrics.SystemFilesystemInodesUsage.New(),
 					test.expectedDeviceDataPoints*2,
 					test.expectedDeviceLabelValues,
 				)
@@ -245,7 +246,7 @@ func assertFileSystemUsageMetricValid(
 	expectedDeviceLabelValues []map[string]string) {
 	internal.AssertDescriptorEqual(t, descriptor, metric)
 	for i := 0; i < metric.IntSum().DataPoints().Len(); i++ {
-		for _, label := range []string{deviceLabelName, typeLabelName, mountModeLabelName, mountPointLabelName} {
+		for _, label := range []string{"device", "type", "mode", "mountpoint"} {
 			internal.AssertIntSumMetricLabelExists(t, metric, i, label)
 		}
 	}
@@ -269,12 +270,12 @@ func assertFileSystemUsageMetricValid(
 	} else {
 		assert.GreaterOrEqual(t, metric.IntSum().DataPoints().Len(), fileSystemStatesLen)
 	}
-	internal.AssertIntSumMetricLabelHasValue(t, metric, 0, stateLabelName, usedLabelValue)
-	internal.AssertIntSumMetricLabelHasValue(t, metric, 1, stateLabelName, freeLabelValue)
+	internal.AssertIntSumMetricLabelHasValue(t, metric, 0, "state", "used")
+	internal.AssertIntSumMetricLabelHasValue(t, metric, 1, "state", "free")
 }
 
 func assertFileSystemUsageMetricHasUnixSpecificStateLabels(t *testing.T, metric pdata.Metric) {
-	internal.AssertIntSumMetricLabelHasValue(t, metric, 2, stateLabelName, reservedLabelValue)
+	internal.AssertIntSumMetricLabelHasValue(t, metric, 2, "state", "reserved")
 }
 
 func isUnix() bool {
