diff --git a/translator/trace/zipkinv1_to_protospan_test.go b/translator/trace/zipkinv1_to_protospan_test.go
index 9615c2af6..96997550b 100644
--- a/translator/trace/zipkinv1_to_protospan_test.go
+++ b/translator/trace/zipkinv1_to_protospan_test.go
@@ -409,4 +409,21 @@ var ocBatchesFromZipkinV1 = []*agenttracepb.ExportTraceServiceRequest{
 			},
 		},
 	},
+	{
+		Node: &commonpb.Node{
+			ServiceInfo: &commonpb.ServiceInfo{Name: "unknown-service"},
+		},
+		Spans: []*tracepb.Span{
+			{
+				TraceId:      []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0e, 0xd2, 0xe6, 0x3c, 0xbe, 0x71, 0xf5, 0xa8},
+				SpanId:       []byte{0xfe, 0x35, 0x1a, 0x05, 0x3f, 0xbc, 0xac, 0x1f},
+				ParentSpanId: []byte{0x0e, 0xd2, 0xe6, 0x3c, 0xbe, 0x71, 0xf5, 0xa8},
+				Name:         &tracepb.TruncatableString{Value: "checkStock"},
+				Kind:         tracepb.Span_SPAN_KIND_UNSPECIFIED,
+				StartTime:    &timestamp.Timestamp{Seconds: 1544805927, Nanos: 453923000},
+				EndTime:      &timestamp.Timestamp{Seconds: 1544805927, Nanos: 457663000},
+				Attributes:   nil,
+			},
+		},
+	},
 }
