diff --git a/pdata/internal/cmd/pdatagen/internal/pprofile_package.go b/pdata/internal/cmd/pdatagen/internal/pprofile_package.go
index 5e7a7b3df..88c316ab7 100644
--- a/pdata/internal/cmd/pdatagen/internal/pprofile_package.go
+++ b/pdata/internal/cmd/pdatagen/internal/pprofile_package.go
@@ -171,17 +171,6 @@ var profile = &messageValueStruct{
 				testVal:     "1234567890",
 			},
 		},
-		&primitiveTypedField{
-			fieldName:       "StartTime",
-			originFieldName: "TimeNanos",
-			returnType: &primitiveType{
-				structName:  "Timestamp",
-				packageName: "pcommon",
-				rawType:     "int64",
-				defaultVal:  "0",
-				testVal:     "1234567890",
-			},
-		},
 		&messageValueField{
 			fieldName:     "PeriodType",
 			returnMessage: valueType,
@@ -212,10 +201,6 @@ var profile = &messageValueStruct{
 				testVal:    "data.ProfileID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1})",
 			},
 		},
-		&sliceField{
-			fieldName:   "AttributeIndices",
-			returnSlice: int32Slice,
-		},
 		droppedAttributesCount,
 		&primitiveField{
 			fieldName:  "OriginalPayloadFormat",
@@ -227,6 +212,10 @@ var profile = &messageValueStruct{
 			fieldName:   "OriginalPayload",
 			returnSlice: byteSlice,
 		},
+		&sliceField{
+			fieldName:   "AttributeIndices",
+			returnSlice: int32Slice,
+		},
 	},
 }
 
