diff --git a/cmd/mdatagen/templates/config.go.tmpl b/cmd/mdatagen/templates/config.go.tmpl
index 32025863d..e4a9f52f8 100644
--- a/cmd/mdatagen/templates/config.go.tmpl
+++ b/cmd/mdatagen/templates/config.go.tmpl
@@ -2,12 +2,12 @@
 
 package {{ .Package }}
 
-{{ if or .Metrics .ResourceAttributes -}}
-import "go.opentelemetry.io/collector/confmap"
-{{- end }}
-{{ if .ResourceAttributes -}}
-import "go.opentelemetry.io/collector/filter"
-{{- end }}
+import (
+    "go.opentelemetry.io/collector/confmap"
+    {{ if and .Metrics .ResourceAttributes -}}
+    "go.opentelemetry.io/collector/filter"
+    {{- end }}
+)
 {{ if .Metrics -}}
 
 // MetricConfig provides common config for a particular metric.
@@ -51,8 +51,10 @@ func DefaultMetricsConfig() MetricsConfig {
 // ResourceAttributeConfig provides common config for a particular resource attribute.
 type ResourceAttributeConfig struct {
 	Enabled bool `mapstructure:"enabled"`
+	{{- if .Metrics }}
 	Include []filter.Config `mapstructure:"include"`
 	Exclude []filter.Config `mapstructure:"exclude"`
+	{{- end }}
 
 	enabledSetByUser bool
 }
