diff --git a/confmap/internal/mapstructure/encoder.go b/confmap/internal/mapstructure/encoder.go
index d0e222e08..994ad038f 100644
--- a/confmap/internal/mapstructure/encoder.go
+++ b/confmap/internal/mapstructure/encoder.go
@@ -11,6 +11,7 @@ import (
 	"strings"
 
 	"github.com/go-viper/mapstructure/v2"
+	"gopkg.in/yaml.v3"
 )
 
 const (
@@ -228,3 +229,35 @@ func TextMarshalerHookFunc() mapstructure.DecodeHookFuncValue {
 		return string(out), nil
 	}
 }
+
+// YamlMarshalerHookFunc returns a DecodeHookFuncValue that checks for structs
+// that have yaml tags but no mapstructure tags. If found, it will convert the struct
+// to map[string]any using the yaml package, which respects the yaml tags. Ultimately,
+// this allows mapstructure to later marshal the map[string]any in a generic way.
+func YamlMarshalerHookFunc() mapstructure.DecodeHookFuncValue {
+	return func(from reflect.Value, _ reflect.Value) (any, error) {
+		if from.Kind() == reflect.Struct {
+			for i := 0; i < from.NumField(); i++ {
+				if _, ok := from.Type().Field(i).Tag.Lookup("mapstructure"); ok {
+					// The struct has at least one mapstructure tag so don't do anything.
+					return from.Interface(), nil
+				}
+
+				if _, ok := from.Type().Field(i).Tag.Lookup("yaml"); ok {
+					// The struct has at least one yaml tag, so convert it to map[string]any using yaml.
+					yamlBytes, err := yaml.Marshal(from.Interface())
+					if err != nil {
+						return nil, err
+					}
+					var m map[string]any
+					err = yaml.Unmarshal(yamlBytes, &m)
+					if err != nil {
+						return nil, err
+					}
+					return m, nil
+				}
+			}
+		}
+		return from.Interface(), nil
+	}
+}
