diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index 493c16ade..0e74635fe 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -5,7 +5,7 @@ go 1.23.0
 require (
 	github.com/fsnotify/fsnotify v1.8.0
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/config/configopaque v1.27.0
+	go.opentelemetry.io/collector/config/configopaque v1.28.0
 )
 
 require (
