diff --git a/config/configgrpc/configgrpc.go b/config/configgrpc/configgrpc.go
index 4ed941f8c..c3b6afee9 100644
--- a/config/configgrpc/configgrpc.go
+++ b/config/configgrpc/configgrpc.go
@@ -17,6 +17,7 @@ package configgrpc
 
 import (
 	"fmt"
+	"net"
 	"strings"
 	"time"
 
@@ -179,6 +180,15 @@ func (gcs *GRPCClientSettings) ToDialOptions() ([]grpc.DialOption, error) {
 	return opts, nil
 }
 
+func (gss *GRPCServerSettings) ToListener() (net.Listener, error) {
+	listener, err := net.Listen("tcp", gss.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return listener, nil
+}
+
 // ToServerOption maps configgrpc.GRPCServerSettings to a slice of server options for gRPC
 func (gss *GRPCServerSettings) ToServerOption() ([]grpc.ServerOption, error) {
 	var opts []grpc.ServerOption
