diff --git a/consumer/consumererror/permanent.go b/consumer/consumererror/permanent.go
index c4b2c4fc0..dbf331358 100644
--- a/consumer/consumererror/permanent.go
+++ b/consumer/consumererror/permanent.go
@@ -22,9 +22,9 @@ type permanent struct {
 	err error
 }
 
-// Permanent wraps an error to indicate that it is a permanent error, i.e. an
+// NewPermanent wraps an error to indicate that it is a permanent error, i.e. an
 // error that will be always returned if its source receives the same inputs.
-func Permanent(err error) error {
+func NewPermanent(err error) error {
 	return permanent{err: err}
 }
 
@@ -37,7 +37,7 @@ func (p permanent) Unwrap() error {
 	return p.err
 }
 
-// IsPermanent checks if an error was wrapped with the Permanent function, which
+// IsPermanent checks if an error was wrapped with the NewPermanent function, which
 // is used to indicate that a given error will always be returned in the case
 // that its sources receives the same input.
 func IsPermanent(err error) bool {
