diff --git a/cmd/builder/internal/builder/config.go b/cmd/builder/internal/builder/config.go
index 6ef3a4efe..a3a66962f 100644
--- a/cmd/builder/internal/builder/config.go
+++ b/cmd/builder/internal/builder/config.go
@@ -18,8 +18,8 @@ import (
 )
 
 const (
-	defaultBetaOtelColVersion   = "v0.122.0"
-	defaultStableOtelColVersion = "v1.28.0"
+	defaultBetaOtelColVersion   = "v0.122.1"
+	defaultStableOtelColVersion = "v1.28.1"
 )
 
 // errMissingGoMod indicates an empty gomod field
