diff --git a/pdata/xpdata/go.mod b/pdata/xpdata/go.mod
index b7510645e..5551019fc 100644
--- a/pdata/xpdata/go.mod
+++ b/pdata/xpdata/go.mod
@@ -5,10 +5,10 @@ go 1.23.0
 require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/client v1.34.0
-	go.opentelemetry.io/collector/pdata v1.34.0
-	go.opentelemetry.io/collector/pdata/pprofile v0.128.0
-	go.opentelemetry.io/collector/pdata/testdata v0.128.0
+	go.opentelemetry.io/collector/client v1.35.0
+	go.opentelemetry.io/collector/pdata v1.35.0
+	go.opentelemetry.io/collector/pdata/pprofile v0.129.0
+	go.opentelemetry.io/collector/pdata/testdata v0.129.0
 	go.opentelemetry.io/otel/trace v1.36.0
 )
 
