diff --git a/service/internal/configunmarshaler/extensions.go b/service/internal/configunmarshaler/extensions.go
index cb3478a77..b44024c13 100644
--- a/service/internal/configunmarshaler/extensions.go
+++ b/service/internal/configunmarshaler/extensions.go
@@ -57,7 +57,7 @@ func (e *Extensions) Unmarshal(conf *confmap.Conf) error {
 
 		// Now that the default config struct is created we can Unmarshal into it,
 		// and it will apply user-defined config on top of the default.
-		if err := component.UnmarshalExtensionConfig(confmap.NewFromStringMap(value), extensionCfg); err != nil {
+		if err := component.UnmarshalConfig(confmap.NewFromStringMap(value), extensionCfg); err != nil {
 			return errorUnmarshalError(extensionsKeyName, id, err)
 		}
 
