diff --git a/client/client_test.go b/client/client_test.go
index 37f300de4..c3d33c9f4 100644
--- a/client/client_test.go
+++ b/client/client_test.go
@@ -15,11 +15,11 @@ import (
 
 func TestNewContext(t *testing.T) {
 	testCases := []struct {
-		desc string
+		name string
 		cl   Info
 	}{
 		{
-			desc: "valid client",
+			name: "valid client",
 			cl: Info{
 				Addr: &net.IPAddr{
 					IP: net.IPv4(1, 2, 3, 4),
@@ -27,26 +27,26 @@ func TestNewContext(t *testing.T) {
 			},
 		},
 		{
-			desc: "nil client",
+			name: "nil client",
 			cl:   Info{},
 		},
 	}
-	for _, tC := range testCases {
-		t.Run(tC.desc, func(t *testing.T) {
-			ctx := NewContext(context.Background(), tC.cl)
-			assert.Equal(t, ctx.Value(ctxKey{}), tC.cl)
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := NewContext(context.Background(), tt.cl)
+			assert.Equal(t, ctx.Value(ctxKey{}), tt.cl)
 		})
 	}
 }
 
 func TestFromContext(t *testing.T) {
 	testCases := []struct {
-		desc     string
+		name     string
 		input    context.Context
 		expected Info
 	}{
 		{
-			desc: "context with client",
+			name: "context with client",
 			input: context.WithValue(context.Background(), ctxKey{}, Info{
 				Addr: &net.IPAddr{
 					IP: net.IPv4(1, 2, 3, 4),
@@ -59,19 +59,19 @@ func TestFromContext(t *testing.T) {
 			},
 		},
 		{
-			desc:     "context without client",
+			name:     "context without client",
 			input:    context.Background(),
 			expected: Info{},
 		},
 		{
-			desc:     "context with something else in the key",
+			name:     "context with something else in the key",
 			input:    context.WithValue(context.Background(), ctxKey{}, "unexpected!"),
 			expected: Info{},
 		},
 	}
-	for _, tC := range testCases {
-		t.Run(tC.desc, func(t *testing.T) {
-			assert.Equal(t, tC.expected, FromContext(tC.input))
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.expected, FromContext(tt.input))
 		})
 	}
 }
