diff --git a/exporter/factory.go b/exporter/factory.go
index ded017441..d0cfdb4b0 100644
--- a/exporter/factory.go
+++ b/exporter/factory.go
@@ -42,20 +42,16 @@ type Factory interface {
 	CreateMetricsExporter(logger *zap.Logger, cfg configmodels.Exporter) (consumer.MetricsConsumer, StopFunc, error)
 }
 
-// List of registered exporter factories.
-var exporterFactories = make(map[string]Factory)
-
-// RegisterFactory registers a exporter factory.
-func RegisterFactory(factory Factory) error {
-	if exporterFactories[factory.Type()] != nil {
-		panic(fmt.Sprintf("duplicate exporter factory %q", factory.Type()))
+// Build takes a list of exporter factories and returns a map of type map[string]Factory
+// with factory type as keys. It returns a non-nil error when more than one factories
+// have the same type.
+func Build(factories ...Factory) (map[string]Factory, error) {
+	fMap := map[string]Factory{}
+	for _, f := range factories {
+		if _, ok := fMap[f.Type()]; ok {
+			return fMap, fmt.Errorf("duplicate exporter factory %q", f.Type())
+		}
+		fMap[f.Type()] = f
 	}
-
-	exporterFactories[factory.Type()] = factory
-	return nil
-}
-
-// GetFactory gets a exporter factory by type string.
-func GetFactory(typeStr string) Factory {
-	return exporterFactories[typeStr]
+	return fMap, nil
 }
