diff --git a/connector/connector_test.go b/connector/connector_test.go
index ebf727a05..7d8db010e 100644
--- a/connector/connector_test.go
+++ b/connector/connector_test.go
@@ -433,13 +433,16 @@ func TestBuilderMissingConfig(t *testing.T) {
 	assert.Nil(t, l2l)
 }
 
-func TestBuilderFactory(t *testing.T) {
+func TestBuilderGetters(t *testing.T) {
 	factories, err := MakeFactoryMap([]Factory{NewFactory("foo", nil)}...)
 	require.NoError(t, err)
 
 	cfgs := map[component.ID]component.Config{component.NewID("foo"): struct{}{}}
 	b := NewBuilder(cfgs, factories)
 
+	assert.True(t, b.IsConfigured(component.NewID("foo")))
+	assert.False(t, b.IsConfigured(component.NewID("bar")))
+
 	assert.NotNil(t, b.Factory(component.NewID("foo").Type()))
 	assert.Nil(t, b.Factory(component.NewID("bar").Type()))
 }
