diff --git a/config/confighttp/clientinfohandler.go b/config/confighttp/clientinfohandler.go
index 025bdb851..67d09c6f8 100644
--- a/config/confighttp/clientinfohandler.go
+++ b/config/confighttp/clientinfohandler.go
@@ -22,7 +22,7 @@ type clientInfoHandler struct {
 // ServeHTTP intercepts incoming HTTP requests, replacing the request's context with one that contains
 // a client.Info containing the client's IP address.
 func (h *clientInfoHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	req = req.WithContext(contextWithClient(req, h.includeMetadata))
+	req = req.WithContext(contextWithClient(req, h.includeMetadata)) //nolint:contextcheck //context already handled through contextWithClient
 	h.next.ServeHTTP(w, req)
 }
 
