diff --git a/pdata/pprofile/generated_sample.go b/pdata/pprofile/generated_sample.go
index 7e7d0723d..04007774b 100644
--- a/pdata/pprofile/generated_sample.go
+++ b/pdata/pprofile/generated_sample.go
@@ -114,7 +114,7 @@ func (ms Sample) TimestampsUnixNano() pcommon.UInt64Slice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Sample) CopyTo(dest Sample) {
 	dest.state.AssertMutable()
-	copyOrigSample(dest.orig, ms.orig)
+	internal.CopyOrigSample(dest.orig, ms.orig)
 }
 
 // marshalJSONStream marshals all properties from the current struct to the destination stream.
@@ -169,21 +169,3 @@ func (ms Sample) unmarshalJSONIter(iter *json.Iterator) {
 		return true
 	})
 }
-
-func copyOrigSample(dest, src *otlpprofiles.Sample) {
-	dest.LocationsStartIndex = src.LocationsStartIndex
-	dest.LocationsLength = src.LocationsLength
-	dest.Value = internal.CopyOrigInt64Slice(dest.Value, src.Value)
-	dest.AttributeIndices = internal.CopyOrigInt32Slice(dest.AttributeIndices, src.AttributeIndices)
-	if srcLinkIndex, ok := src.LinkIndex_.(*otlpprofiles.Sample_LinkIndex); ok {
-		destLinkIndex, ok := dest.LinkIndex_.(*otlpprofiles.Sample_LinkIndex)
-		if !ok {
-			destLinkIndex = &otlpprofiles.Sample_LinkIndex{}
-			dest.LinkIndex_ = destLinkIndex
-		}
-		destLinkIndex.LinkIndex = srcLinkIndex.LinkIndex
-	} else {
-		dest.LinkIndex_ = nil
-	}
-	dest.TimestampsUnixNano = internal.CopyOrigUInt64Slice(dest.TimestampsUnixNano, src.TimestampsUnixNano)
-}
