diff --git a/pdata/pprofile/generated_lineslice.go b/pdata/pprofile/generated_lineslice.go
index 7f0498b8e..736aaf250 100644
--- a/pdata/pprofile/generated_lineslice.go
+++ b/pdata/pprofile/generated_lineslice.go
@@ -148,7 +148,7 @@ func (es LineSlice) RemoveIf(f func(Line) bool) {
 // CopyTo copies all elements from the current slice overriding the destination.
 func (es LineSlice) CopyTo(dest LineSlice) {
 	dest.state.AssertMutable()
-	*dest.orig = copyOrigLineSlice(*dest.orig, *es.orig)
+	*dest.orig = internal.CopyOrigLineSlice(*dest.orig, *es.orig)
 }
 
 // Sort sorts the Line elements within LineSlice given the
@@ -180,32 +180,3 @@ func (ms LineSlice) unmarshalJSONIter(iter *json.Iterator) {
 		return true
 	})
 }
-
-func copyOrigLineSlice(dest, src []*otlpprofiles.Line) []*otlpprofiles.Line {
-	var newDest []*otlpprofiles.Line
-	if cap(dest) < len(src) {
-		newDest = make([]*otlpprofiles.Line, len(src))
-		// Copy old pointers to re-use.
-		copy(newDest, dest)
-		// Add new pointers for missing elements from len(dest) to len(srt).
-		for i := len(dest); i < len(src); i++ {
-			newDest[i] = &otlpprofiles.Line{}
-		}
-	} else {
-		newDest = dest[:len(src)]
-		// Cleanup the rest of the elements so GC can free the memory.
-		// This can happen when len(src) < len(dest) < cap(dest).
-		for i := len(src); i < len(dest); i++ {
-			dest[i] = nil
-		}
-		// Add new pointers for missing elements.
-		// This can happen when len(dest) < len(src) < cap(dest).
-		for i := len(dest); i < len(src); i++ {
-			newDest[i] = &otlpprofiles.Line{}
-		}
-	}
-	for i := range src {
-		copyOrigLine(newDest[i], src[i])
-	}
-	return newDest
-}
