diff --git a/config/configauth/mock_serverauth.go b/config/configauth/mock_serverauth.go
index 1f5ead659..cd11c34f7 100644
--- a/config/configauth/mock_serverauth.go
+++ b/config/configauth/mock_serverauth.go
@@ -42,13 +42,13 @@ func (m *MockAuthenticator) Authenticate(ctx context.Context, headers map[string
 	return m.AuthenticateFunc(ctx, headers)
 }
 
-// GrpcUnaryServerInterceptor isn't currently implemented and always returns nil.
-func (m *MockAuthenticator) GrpcUnaryServerInterceptor(context.Context, interface{}, *grpc.UnaryServerInfo, grpc.UnaryHandler) (interface{}, error) {
+// GRPCUnaryServerInterceptor isn't currently implemented and always returns nil.
+func (m *MockAuthenticator) GRPCUnaryServerInterceptor(context.Context, interface{}, *grpc.UnaryServerInfo, grpc.UnaryHandler) (interface{}, error) {
 	return nil, nil
 }
 
-// GrpcStreamServerInterceptor isn't currently implemented and always returns nil.
-func (m *MockAuthenticator) GrpcStreamServerInterceptor(interface{}, grpc.ServerStream, *grpc.StreamServerInfo, grpc.StreamHandler) error {
+// GRPCStreamServerInterceptor isn't currently implemented and always returns nil.
+func (m *MockAuthenticator) GRPCStreamServerInterceptor(interface{}, grpc.ServerStream, *grpc.StreamServerInfo, grpc.StreamHandler) error {
 	return nil
 }
 
