diff --git a/exporter/debugexporter/exporter_test.go b/exporter/debugexporter/exporter_test.go
index 42cd116ca..c35420661 100644
--- a/exporter/debugexporter/exporter_test.go
+++ b/exporter/debugexporter/exporter_test.go
@@ -13,6 +13,7 @@ import (
 	"go.uber.org/zap/zaptest"
 
 	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/exporter/debugexporter/internal/metadata"
 	"go.opentelemetry.io/collector/exporter/exportertest"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
@@ -24,7 +25,7 @@ import (
 func TestTracesNoErrors(t *testing.T) {
 	for _, tc := range createTestCases() {
 		t.Run(tc.name, func(t *testing.T) {
-			lte, err := createTraces(context.Background(), exportertest.NewNopSettings(), tc.config)
+			lte, err := createTraces(context.Background(), exportertest.NewNopSettingsWithType(metadata.Type), tc.config)
 			require.NotNil(t, lte)
 			assert.NoError(t, err)
 
@@ -39,7 +40,7 @@ func TestTracesNoErrors(t *testing.T) {
 func TestMetricsNoErrors(t *testing.T) {
 	for _, tc := range createTestCases() {
 		t.Run(tc.name, func(t *testing.T) {
-			lme, err := createMetrics(context.Background(), exportertest.NewNopSettings(), tc.config)
+			lme, err := createMetrics(context.Background(), exportertest.NewNopSettingsWithType(metadata.Type), tc.config)
 			require.NotNil(t, lme)
 			assert.NoError(t, err)
 
@@ -57,7 +58,7 @@ func TestMetricsNoErrors(t *testing.T) {
 func TestLogsNoErrors(t *testing.T) {
 	for _, tc := range createTestCases() {
 		t.Run(tc.name, func(t *testing.T) {
-			lle, err := createLogs(context.Background(), exportertest.NewNopSettings(), createDefaultConfig())
+			lle, err := createLogs(context.Background(), exportertest.NewNopSettingsWithType(metadata.Type), createDefaultConfig())
 			require.NotNil(t, lle)
 			assert.NoError(t, err)
 
@@ -72,7 +73,7 @@ func TestLogsNoErrors(t *testing.T) {
 func TestProfilesNoErrors(t *testing.T) {
 	for _, tc := range createTestCases() {
 		t.Run(tc.name, func(t *testing.T) {
-			lle, err := createProfiles(context.Background(), exportertest.NewNopSettings(), createDefaultConfig())
+			lle, err := createProfiles(context.Background(), exportertest.NewNopSettingsWithType(metadata.Type), createDefaultConfig())
 			require.NotNil(t, lle)
 			assert.NoError(t, err)
 
