diff --git a/internal/iruntime/mem_info_test.go b/internal/iruntime/mem_info_test.go
index 5f4e11105..ec6d7f597 100644
--- a/internal/iruntime/mem_info_test.go
+++ b/internal/iruntime/mem_info_test.go
@@ -1,4 +1,4 @@
-// Copyright  The OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
