diff --git a/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_unix.go b/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_unix.go
index 8b431fed1..4f1013b93 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_unix.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_unix.go
@@ -20,11 +20,10 @@ import (
 	"time"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/collector/internal/dataold"
 	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
 )
 
-func appendSystemSpecificProcessesMetrics(metrics dataold.MetricSlice, startIndex int, miscFunc getMiscStats) error {
+func appendSystemSpecificProcessesMetrics(metrics pdata.MetricSlice, startIndex int, miscFunc getMiscStats) error {
 	now := internal.TimeToUnixNano(time.Now())
 	misc, err := miscFunc()
 	if err != nil {
@@ -37,10 +36,10 @@ func appendSystemSpecificProcessesMetrics(metrics dataold.MetricSlice, startInde
 	return nil
 }
 
-func initializeProcessesMetric(metric dataold.Metric, descriptor dataold.MetricDescriptor, now pdata.TimestampUnixNano, value int64) {
-	descriptor.CopyTo(metric.MetricDescriptor())
+func initializeProcessesMetric(metric pdata.Metric, descriptor pdata.Metric, now pdata.TimestampUnixNano, value int64) {
+	descriptor.CopyTo(metric)
 
-	ddps := metric.Int64DataPoints()
+	ddps := metric.IntSum().DataPoints()
 	ddps.Resize(1)
 	ddps.At(0).SetTimestamp(now)
 	ddps.At(0).SetValue(value)
