diff --git a/config/confighttp/confighttp.go b/config/confighttp/confighttp.go
index 600b58302..bc957ee02 100644
--- a/config/confighttp/confighttp.go
+++ b/config/confighttp/confighttp.go
@@ -391,7 +391,7 @@ func (hss *ServerConfig) ToServer(host component.Host, settings component.Teleme
 		otelhttp.WithTracerProvider(settings.TracerProvider),
 		otelhttp.WithMeterProvider(settings.MeterProvider),
 		otelhttp.WithPropagators(otel.GetTextMapPropagator()),
-		otelhttp.WithSpanNameFormatter(func(operation string, r *http.Request) string {
+		otelhttp.WithSpanNameFormatter(func(_ string, r *http.Request) string {
 			return r.URL.Path
 		}),
 	)
