diff --git a/exporter/exporterhelper/internal/persistent_queue.go b/exporter/exporterhelper/internal/persistent_queue.go
index 24f886a04..e2e0b3c6b 100644
--- a/exporter/exporterhelper/internal/persistent_queue.go
+++ b/exporter/exporterhelper/internal/persistent_queue.go
@@ -25,6 +25,13 @@ import (
 	"go.opentelemetry.io/collector/extension/experimental/storage"
 )
 
+// Monkey patching for unit test
+var (
+	stopStorage = func(queue *persistentQueue) {
+		queue.storage.stop()
+	}
+)
+
 // persistentQueue holds the queue backed by file storage
 type persistentQueue struct {
 	logger     *zap.Logger
@@ -80,9 +87,10 @@ func (pq *persistentQueue) Produce(item Request) bool {
 // Stop stops accepting items, shuts down the queue and closes the persistent queue
 func (pq *persistentQueue) Stop() {
 	pq.stopOnce.Do(func() {
-		pq.storage.stop()
+		// stop the consumers before the storage or the successful processing result will fail to write to persistent storage
 		close(pq.stopChan)
 		pq.stopWG.Wait()
+		stopStorage(pq)
 	})
 }
 
