diff --git a/exporter/exporterhelper/logs_test.go b/exporter/exporterhelper/logs_test.go
index 2d464f080..0a99672ff 100644
--- a/exporter/exporterhelper/logs_test.go
+++ b/exporter/exporterhelper/logs_test.go
@@ -30,6 +30,7 @@ import (
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/metadatatest"
 	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/exporter/internal/requesttest"
 	"go.opentelemetry.io/collector/exporter/internal/storagetest"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/testdata"
@@ -68,7 +69,7 @@ func TestLogs_NilLogger(t *testing.T) {
 }
 
 func TestLogsRequest_NilLogger(t *testing.T) {
-	le, err := NewLogsRequest(context.Background(), exporter.Settings{}, internal.RequestFromLogsFunc(nil))
+	le, err := NewLogsRequest(context.Background(), exporter.Settings{}, requesttest.RequestFromLogsFunc(nil))
 	require.Nil(t, le)
 	require.Equal(t, errNilLogger, err)
 }
@@ -100,7 +101,7 @@ func TestLogs_Default(t *testing.T) {
 func TestLogsRequest_Default(t *testing.T) {
 	ld := plog.NewLogs()
 	le, err := NewLogsRequest(context.Background(), exportertest.NewNopSettings(),
-		internal.RequestFromLogsFunc(nil))
+		requesttest.RequestFromLogsFunc(nil))
 	assert.NotNil(t, le)
 	require.NoError(t, err)
 
@@ -122,7 +123,7 @@ func TestLogs_WithCapabilities(t *testing.T) {
 func TestLogsRequest_WithCapabilities(t *testing.T) {
 	capabilities := consumer.Capabilities{MutatesData: true}
 	le, err := NewLogsRequest(context.Background(), exportertest.NewNopSettings(),
-		internal.RequestFromLogsFunc(nil), WithCapabilities(capabilities))
+		requesttest.RequestFromLogsFunc(nil), WithCapabilities(capabilities))
 	require.NoError(t, err)
 	require.NotNil(t, le)
 
@@ -154,7 +155,7 @@ func TestLogsRequest_Default_ExportError(t *testing.T) {
 	ld := plog.NewLogs()
 	want := errors.New("export_error")
 	le, err := NewLogsRequest(context.Background(), exportertest.NewNopSettings(),
-		internal.RequestFromLogsFunc(want))
+		requesttest.RequestFromLogsFunc(want))
 	require.NoError(t, err)
 	require.NotNil(t, le)
 	require.Equal(t, want, le.ConsumeLogs(context.Background(), ld))
@@ -226,7 +227,7 @@ func TestLogsRequest_WithRecordMetrics(t *testing.T) {
 
 	le, err := NewLogsRequest(context.Background(),
 		exporter.Settings{ID: fakeLogsName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
-		internal.RequestFromLogsFunc(nil))
+		requesttest.RequestFromLogsFunc(nil))
 	require.NoError(t, err)
 	require.NotNil(t, le)
 
@@ -253,7 +254,7 @@ func TestLogsRequest_WithRecordMetrics_ExportError(t *testing.T) {
 	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
 
 	le, err := NewLogsRequest(context.Background(), exporter.Settings{ID: fakeLogsName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
-		internal.RequestFromLogsFunc(want))
+		requesttest.RequestFromLogsFunc(want))
 	require.NoError(t, err)
 	require.NotNil(t, le)
 
@@ -312,7 +313,7 @@ func TestLogsRequest_WithSpan(t *testing.T) {
 	otel.SetTracerProvider(set.TracerProvider)
 	defer otel.SetTracerProvider(nooptrace.NewTracerProvider())
 
-	le, err := NewLogsRequest(context.Background(), set, internal.RequestFromLogsFunc(nil))
+	le, err := NewLogsRequest(context.Background(), set, requesttest.RequestFromLogsFunc(nil))
 	require.NoError(t, err)
 	require.NotNil(t, le)
 	checkWrapSpanForLogs(t, sr, set.TracerProvider.Tracer("test"), le, nil)
@@ -340,7 +341,7 @@ func TestLogsRequest_WithSpan_ReturnError(t *testing.T) {
 	defer otel.SetTracerProvider(nooptrace.NewTracerProvider())
 
 	want := errors.New("my_error")
-	le, err := NewLogsRequest(context.Background(), set, internal.RequestFromLogsFunc(want))
+	le, err := NewLogsRequest(context.Background(), set, requesttest.RequestFromLogsFunc(want))
 	require.NoError(t, err)
 	require.NotNil(t, le)
 	checkWrapSpanForLogs(t, sr, set.TracerProvider.Tracer("test"), le, want)
@@ -363,7 +364,7 @@ func TestLogsRequest_WithShutdown(t *testing.T) {
 	shutdown := func(context.Context) error { shutdownCalled = true; return nil }
 
 	le, err := NewLogsRequest(context.Background(), exportertest.NewNopSettings(),
-		internal.RequestFromLogsFunc(nil), WithShutdown(shutdown))
+		requesttest.RequestFromLogsFunc(nil), WithShutdown(shutdown))
 	assert.NotNil(t, le)
 	assert.NoError(t, err)
 
@@ -387,7 +388,7 @@ func TestLogsRequest_WithShutdown_ReturnError(t *testing.T) {
 	shutdownErr := func(context.Context) error { return want }
 
 	le, err := NewLogsRequest(context.Background(), exportertest.NewNopSettings(),
-		internal.RequestFromLogsFunc(nil), WithShutdown(shutdownErr))
+		requesttest.RequestFromLogsFunc(nil), WithShutdown(shutdownErr))
 	assert.NotNil(t, le)
 	require.NoError(t, err)
 
