diff --git a/Makefile.Common b/Makefile.Common
index ba93e825b..f4b9b56da 100644
--- a/Makefile.Common
+++ b/Makefile.Common
@@ -5,16 +5,21 @@ ALL_PKGS := $(sort $(shell go list ./...))
 # wide resources, e.g. ports.
 GOTEST_OPT?= -v -p 1 -race -timeout 60s
 GOTEST=go test
+GO_ACC=go-acc
 LINT=golangci-lint
 IMPI=impi
 
 .PHONY: test
 test:
-	@echo $(ALL_PKGS) | xargs -n 10 $(GOTEST) $(GOTEST_OPT)
+	$(GOTEST) $(GOTEST_OPT) ./...
 
 .PHONY: test-unstable
 test-unstable:
-	@echo $(ALL_PKGS) | xargs -n 10 $(GOTEST) $(GOTEST_OPT) -tags enable_unstable
+	$(GOTEST) $(GOTEST_OPT) -tags enable_unstable ./...
+
+.PHONY: test-with-cover
+test-with-cover:
+	$(GO_ACC) --output=coverage.out ./...
 
 .PHONY: benchmark
 benchmark:
