diff --git a/config/configgrpc/go.mod b/config/configgrpc/go.mod
index ddb210db9..3fc8ec046 100644
--- a/config/configgrpc/go.mod
+++ b/config/configgrpc/go.mod
@@ -64,7 +64,7 @@ require (
 	golang.org/x/net v0.26.0 // indirect
 	golang.org/x/sys v0.21.0 // indirect
 	golang.org/x/text v0.16.0 // indirect
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20240528184218-531527333157 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240701130421-f6361c86f094 // indirect
 	google.golang.org/protobuf v1.34.2 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
