diff --git a/pdata/pmetric/generated_summary.go b/pdata/pmetric/generated_summary.go
index 5359de47f..2a1f6a7d3 100644
--- a/pdata/pmetric/generated_summary.go
+++ b/pdata/pmetric/generated_summary.go
@@ -32,8 +32,7 @@ func newSummary(orig *otlpmetrics.Summary, state *internal.State) Summary {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewSummary() Summary {
-	state := internal.StateMutable
-	return newSummary(&otlpmetrics.Summary{}, &state)
+	return newSummary(&otlpmetrics.Summary{}, internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
