diff --git a/cmd/mdatagen/internal/templates/metrics.go.tmpl b/cmd/mdatagen/internal/templates/metrics.go.tmpl
index 5e7700c23..4e990321c 100644
--- a/cmd/mdatagen/internal/templates/metrics.go.tmpl
+++ b/cmd/mdatagen/internal/templates/metrics.go.tmpl
@@ -3,6 +3,8 @@
 package {{ .Package }}
 
 {{ $reag := .ReaggregationEnabled }}
+{{- $hasReagMetrics := false }}
+{{- range $name, $metric := .Metrics }}{{- if and $reag (hasAggregatableAttributes $metric.Attributes) }}{{- $hasReagMetrics = true }}{{- end }}{{- end }}
 
 import (
 	{{- if .Metrics | parseImportsRequired }}
@@ -10,7 +12,7 @@ import (
 	"fmt"
 	{{- end }}
 	"time"
-	{{- if $reag }}
+	{{- if $hasReagMetrics }}
 	"slices"
 	{{- end }}
 
@@ -38,7 +40,7 @@ const (
 {{- end }}
 
 {{ range $name, $info := .Attributes }}
-{{- if $info.Enum -}}
+{{- if $info.Enum }}
 // Attribute{{ $name.Render }} specifies the value {{ $name }} attribute.
 type Attribute{{ $name.Render }} int
 
@@ -66,8 +68,7 @@ var MapAttribute{{ $name.Render }} = map[string]Attribute{{ $name.Render }}{
 	"{{ . }}": Attribute{{ $name.Render }}{{ . | publicVar }},
 	{{- end }}
 }
-
-{{ end }}
+{{- end }}
 {{- end }}
 
 var MetricsInfo = metricsInfo{
@@ -108,12 +109,12 @@ func With{{ .Render }}MetricAttribute({{ .RenderUnexported }}AttributeValue {{ (
 {{ end }}
 {{ end }}
 
-{{ range $name, $metric := .Metrics -}}
+{{ range $name, $metric := .Metrics }}
 {{- $metricReag := and $reag (hasAggregatableAttributes $metric.Attributes) -}}
 type metric{{ $name.Render }} struct {
 	data                     pmetric.Metric // data buffer for generated metric.
-	config                   MetricConfig   // metric config provided by user.
-	capacity                 int            // max observed number of data points added to the metric.
+	config                   {{ if $reag }}{{ $name.Render }}Config{{ else }}MetricConfig{{ end }} // metric config provided by user.
+	capacity                 int // max observed number of data points added to the metric.
 	{{- if $metricReag }}
 	aggDataPoints            []{{ $metric.Data.MetricValueType.BasicType }} // slice containing number of aggregated datapoints at each index
 	{{- end }}
@@ -152,7 +153,9 @@ func (m *metric{{ $name.Render }}) recordDataPoint(start pcommon.Timestamp, ts p
 	dp.SetTimestamp(ts)
 	{{- range $metric.Attributes }}
 	{{- if not (attributeInfo .).IsConditional }}
-	{{ template "putMetricAttribute" . }}
+	if slices.Contains(m.config.EnabledAttributes, {{ $name.Render }}AttributeKey{{ .Render }}) {
+		{{- template "putAttribute" . }}
+	}
 	{{- end }}
 	{{- end }}
 
@@ -239,7 +242,7 @@ func (m *metric{{ $name.Render }}) emit(metrics pmetric.MetricSlice) {
 	}
 }
 
-func newMetric{{ $name.Render }}(cfg MetricConfig) metric{{ $name.Render }} {
+func newMetric{{ $name.Render }}(cfg {{ if $reag }}{{ $name.Render }}Config{{ else }}MetricConfig{{ end }}) metric{{ $name.Render }} {
 	m := metric{{ $name.Render }}{config: cfg}
 
 	if cfg.Enabled {
