diff --git a/processor/processorhelper/logs_test.go b/processor/processorhelper/logs_test.go
index 611c242e7..d1e2fddab 100644
--- a/processor/processorhelper/logs_test.go
+++ b/processor/processorhelper/logs_test.go
@@ -27,7 +27,7 @@ import (
 var testLogsCfg = struct{}{}
 
 func TestNewLogs(t *testing.T) {
-	lp, err := NewLogs(context.Background(), processortest.NewNopSettings(), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(nil))
+	lp, err := NewLogs(context.Background(), processortest.NewNopSettingsWithType(processortest.NopType), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(nil))
 	require.NoError(t, err)
 
 	assert.True(t, lp.Capabilities().MutatesData)
@@ -38,7 +38,7 @@ func TestNewLogs(t *testing.T) {
 
 func TestNewLogs_WithOptions(t *testing.T) {
 	want := errors.New("my_error")
-	lp, err := NewLogs(context.Background(), processortest.NewNopSettings(), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(nil),
+	lp, err := NewLogs(context.Background(), processortest.NewNopSettingsWithType(processortest.NopType), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(nil),
 		WithStart(func(context.Context, component.Host) error { return want }),
 		WithShutdown(func(context.Context) error { return want }),
 		WithCapabilities(consumer.Capabilities{MutatesData: false}))
@@ -50,19 +50,19 @@ func TestNewLogs_WithOptions(t *testing.T) {
 }
 
 func TestNewLogs_NilRequiredFields(t *testing.T) {
-	_, err := NewLogs(context.Background(), processortest.NewNopSettings(), &testLogsCfg, consumertest.NewNop(), nil)
+	_, err := NewLogs(context.Background(), processortest.NewNopSettingsWithType(processortest.NopType), &testLogsCfg, consumertest.NewNop(), nil)
 	assert.Error(t, err)
 }
 
 func TestNewLogs_ProcessLogError(t *testing.T) {
 	want := errors.New("my_error")
-	lp, err := NewLogs(context.Background(), processortest.NewNopSettings(), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(want))
+	lp, err := NewLogs(context.Background(), processortest.NewNopSettingsWithType(processortest.NopType), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(want))
 	require.NoError(t, err)
 	assert.Equal(t, want, lp.ConsumeLogs(context.Background(), plog.NewLogs()))
 }
 
 func TestNewLogs_ProcessLogsErrSkipProcessingData(t *testing.T) {
-	lp, err := NewLogs(context.Background(), processortest.NewNopSettings(), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(ErrSkipProcessingData))
+	lp, err := NewLogs(context.Background(), processortest.NewNopSettingsWithType(processortest.NopType), &testLogsCfg, consumertest.NewNop(), newTestLProcessor(ErrSkipProcessingData))
 	require.NoError(t, err)
 	assert.NoError(t, lp.ConsumeLogs(context.Background(), plog.NewLogs()))
 }
@@ -86,7 +86,7 @@ func TestLogsConcurrency(t *testing.T) {
 	incomingLogRecords.AppendEmpty()
 	incomingLogRecords.AppendEmpty()
 
-	lp, err := NewLogs(context.Background(), processortest.NewNopSettings(), &testLogsCfg, consumertest.NewNop(), logsFunc)
+	lp, err := NewLogs(context.Background(), processortest.NewNopSettingsWithType(processortest.NopType), &testLogsCfg, consumertest.NewNop(), logsFunc)
 	require.NoError(t, err)
 	assert.NoError(t, lp.Start(context.Background(), componenttest.NewNopHost()))
 
