diff --git a/receiver/zipkinreceiver/trace_receiver_test.go b/receiver/zipkinreceiver/trace_receiver_test.go
index c07467f3b..8fad5f183 100644
--- a/receiver/zipkinreceiver/trace_receiver_test.go
+++ b/receiver/zipkinreceiver/trace_receiver_test.go
@@ -48,6 +48,12 @@ import (
 	"go.opentelemetry.io/collector/translator/conventions"
 )
 
+const (
+	zipkinV2Single      = "../../translator/trace/zipkinv2/testdata/zipkin_v2_single.json"
+	zipkinV2NoTimestamp = "../../translator/trace/zipkinv2/testdata/zipkin_v2_notimestamp.json"
+	zipkinV1SingleBatch = "../../translator/trace/zipkin/testdata/zipkin_v1_single_batch.json"
+)
+
 var zipkinReceiverID = config.NewIDWithName(typeStr, "receiver_test")
 
 func TestNew(t *testing.T) {
@@ -113,8 +119,7 @@ func TestConvertSpansToTraceSpans_json(t *testing.T) {
 	// Using Adrian Cole's sample at https://gist.github.com/adriancole/e8823c19dfed64e2eb71
 	blob, err := ioutil.ReadFile("./testdata/sample1.json")
 	require.NoError(t, err, "Failed to read sample JSON file: %v", err)
-	zi := new(ZipkinReceiver)
-	zi.config = createDefaultConfig().(*Config)
+	zi := &ZipkinReceiver{translator: translator, config: createDefaultConfig().(*Config)}
 	reqs, err := zi.v2ToTraceSpans(blob, nil)
 	require.NoError(t, err, "Failed to parse convert Zipkin spans in JSON to Trace spans: %v", err)
 
@@ -202,7 +207,7 @@ func TestConversionRoundtrip(t *testing.T) {
   }
 }]`)
 
-	zi := &ZipkinReceiver{nextConsumer: consumertest.NewNop()}
+	zi := &ZipkinReceiver{translator: translator, nextConsumer: consumertest.NewNop()}
 	zi.config = &Config{}
 	ereqs, err := zi.v2ToTraceSpans(receiverInputJSON, nil)
 	require.NoError(t, err)
@@ -314,7 +319,7 @@ func TestReceiverContentTypes(t *testing.T) {
 			content:  "application/json",
 			encoding: "gzip",
 			bodyFn: func() ([]byte, error) {
-				return ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v1_single_batch.json")
+				return ioutil.ReadFile(zipkinV1SingleBatch)
 			},
 		},
 
@@ -332,7 +337,7 @@ func TestReceiverContentTypes(t *testing.T) {
 			content:  "application/json",
 			encoding: "gzip",
 			bodyFn: func() ([]byte, error) {
-				return ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v2_single.json")
+				return ioutil.ReadFile(zipkinV2Single)
 			},
 		},
 
@@ -341,7 +346,7 @@ func TestReceiverContentTypes(t *testing.T) {
 			content:  "application/json",
 			encoding: "zlib",
 			bodyFn: func() ([]byte, error) {
-				return ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v2_single.json")
+				return ioutil.ReadFile(zipkinV2Single)
 			},
 		},
 
@@ -350,7 +355,7 @@ func TestReceiverContentTypes(t *testing.T) {
 			content:  "application/json",
 			encoding: "",
 			bodyFn: func() ([]byte, error) {
-				return ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v2_single.json")
+				return ioutil.ReadFile(zipkinV2Single)
 			},
 		},
 	}
@@ -422,7 +427,7 @@ func TestReceiverInvalidContentType(t *testing.T) {
 }
 
 func TestReceiverConsumerError(t *testing.T) {
-	body, err := ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v2_single.json")
+	body, err := ioutil.ReadFile(zipkinV2Single)
 	require.NoError(t, err)
 
 	r := httptest.NewRequest("POST", "/api/v2/spans", bytes.NewBuffer(body))
@@ -499,8 +504,7 @@ func compressZlib(body []byte) (*bytes.Buffer, error) {
 func TestConvertSpansToTraceSpans_JSONWithoutSerivceName(t *testing.T) {
 	blob, err := ioutil.ReadFile("./testdata/sample2.json")
 	require.NoError(t, err, "Failed to read sample JSON file: %v", err)
-	zi := new(ZipkinReceiver)
-	zi.config = createDefaultConfig().(*Config)
+	zi := &ZipkinReceiver{translator: translator, config: createDefaultConfig().(*Config)}
 	reqs, err := zi.v2ToTraceSpans(blob, nil)
 	require.NoError(t, err, "Failed to parse convert Zipkin spans in JSON to Trace spans: %v", err)
 
@@ -511,7 +515,7 @@ func TestConvertSpansToTraceSpans_JSONWithoutSerivceName(t *testing.T) {
 }
 
 func TestReceiverConvertsStringsToTypes(t *testing.T) {
-	body, err := ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v2_single.json")
+	body, err := ioutil.ReadFile(zipkinV2Single)
 	require.NoError(t, err, "Failed to read sample JSON file: %v", err)
 
 	r := httptest.NewRequest("POST", "/api/v2/spans", bytes.NewBuffer(body))
@@ -559,7 +563,7 @@ func TestReceiverConvertsStringsToTypes(t *testing.T) {
 }
 
 func TestFromBytesWithNoTimestamp(t *testing.T) {
-	noTimestampBytes, err := ioutil.ReadFile("../../translator/trace/zipkin/testdata/zipkin_v2_notimestamp.json")
+	noTimestampBytes, err := ioutil.ReadFile(zipkinV2NoTimestamp)
 	require.NoError(t, err, "Failed to read sample JSON file: %v", err)
 
 	cfg := &Config{
