diff --git a/cmd/builder/internal/builder/main.go b/cmd/builder/internal/builder/main.go
index 781ce4495..37085ec5a 100644
--- a/cmd/builder/internal/builder/main.go
+++ b/cmd/builder/internal/builder/main.go
@@ -148,7 +148,7 @@ func GetModules(cfg *Config) error {
 		return nil
 	}
 
-	if _, err := runGoCommand(cfg, "mod", "tidy", "-compat=1.23"); err != nil {
+	if _, err := runGoCommand(cfg, "mod", "tidy", "-compat=1.24"); err != nil {
 		return fmt.Errorf("failed to update go.mod: %w", err)
 	}
 
