diff --git a/otelcol/factories.go b/otelcol/factories.go
index 7528fa006..c080dbde5 100644
--- a/otelcol/factories.go
+++ b/otelcol/factories.go
@@ -4,6 +4,8 @@
 package otelcol // import "go.opentelemetry.io/collector/otelcol"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/exporter"
@@ -45,3 +47,16 @@ type Factories struct {
 	// ConnectorModules maps connector types to their respective go modules.
 	ConnectorModules map[component.Type]string
 }
+
+// MakeFactoryMap takes a list of factories and returns a map with Factory type as keys.
+// It returns a non-nil error when there are factories with duplicate type.
+func MakeFactoryMap[T component.Factory](factories ...T) (map[component.Type]T, error) {
+	fMap := map[component.Type]T{}
+	for _, f := range factories {
+		if _, ok := fMap[f.Type()]; ok {
+			return fMap, fmt.Errorf("duplicate component factory %q", f.Type())
+		}
+		fMap[f.Type()] = f
+	}
+	return fMap, nil
+}
