diff --git a/pdata/testdata/go.mod b/pdata/testdata/go.mod
index 88492c965..520e8c9f3 100644
--- a/pdata/testdata/go.mod
+++ b/pdata/testdata/go.mod
@@ -18,7 +18,7 @@ require (
 	golang.org/x/text v0.21.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20241015192408-796eee8c2d53 // indirect
 	google.golang.org/grpc v1.69.2 // indirect
-	google.golang.org/protobuf v1.36.0 // indirect
+	google.golang.org/protobuf v1.36.1 // indirect
 )
 
 replace go.opentelemetry.io/collector/pdata => ../
