diff --git a/internal/e2e/go.mod b/internal/e2e/go.mod
index 94e4887f7..d9f3732b8 100644
--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -12,7 +12,7 @@ require (
 	go.opentelemetry.io/collector/config/configtls v0.99.0
 	go.opentelemetry.io/collector/consumer v0.99.0
 	go.opentelemetry.io/collector/exporter v0.99.0
-	go.opentelemetry.io/collector/exporter/otlpexporter v0.98.0
+	go.opentelemetry.io/collector/exporter/otlpexporter v0.99.0
 	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.98.0
 	go.opentelemetry.io/collector/pdata v1.6.0
 	go.opentelemetry.io/collector/pdata/testdata v0.99.0
