diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 97c61c66c..f502b0f2d 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -16,7 +16,7 @@ require (
 	go.opentelemetry.io/collector/config/configtls v0.98.0
 	go.opentelemetry.io/collector/config/internal v0.98.0
 	go.opentelemetry.io/collector/extension/auth v0.98.0
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.49.0
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.50.0
 	go.opentelemetry.io/otel v1.25.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
