diff --git a/translator/trace/protospan_translation_test.go b/translator/trace/protospan_translation_test.go
index a3d04ffa8..b70e455fa 100644
--- a/translator/trace/protospan_translation_test.go
+++ b/translator/trace/protospan_translation_test.go
@@ -134,16 +134,11 @@ func TestAttributeMapToStringAndBack(t *testing.T) {
 func TestAttributeArrayToStringAndBack(t *testing.T) {
 	expected := pdata.NewAttributeValueArray()
 	attrArr := expected.ArrayVal()
-	strAV := pdata.NewAttributeValueString("strVal")
-	attrArr.Append(strAV)
-	intAV := pdata.NewAttributeValueInt(7)
-	attrArr.Append(intAV)
-	floatAV := pdata.NewAttributeValueDouble(18.6)
-	attrArr.Append(floatAV)
-	boolAV := pdata.NewAttributeValueBool(false)
-	attrArr.Append(boolAV)
-	nullAV := pdata.NewAttributeValueNull()
-	attrArr.Append(nullAV)
+	attrArr.Append(pdata.NewAttributeValueString("strVal"))
+	attrArr.Append(pdata.NewAttributeValueInt(7))
+	attrArr.Append(pdata.NewAttributeValueDouble(18.6))
+	attrArr.Append(pdata.NewAttributeValueBool(false))
+	attrArr.Append(pdata.NewAttributeValueNull())
 	strVal := AttributeValueToString(expected, false)
 	dest := pdata.NewAttributeMap()
 	UpsertStringToAttributeMap("parent", strVal, dest, false)
