diff --git a/connector/metrics_router_test.go b/connector/metrics_router_test.go
index f82f770c2..77cebe318 100644
--- a/connector/metrics_router_test.go
+++ b/connector/metrics_router_test.go
@@ -48,7 +48,7 @@ func fuzzMetrics(numIDs, numCons, numMetrics int) func(*testing.T) {
 		allConsMap := make(map[pipeline.ID]consumer.Metrics)
 
 		// If any consumer is mutating, the router must report mutating
-		for i := 0; i < numCons; i++ {
+		for i := range numCons {
 			allIDs = append(allIDs, pipeline.NewIDWithName(pipeline.SignalMetrics, "sink_"+strconv.Itoa(numCons)))
 			// Random chance for each consumer to be mutating
 			if (numCons+numMetrics+i)%4 == 0 {
@@ -66,10 +66,10 @@ func fuzzMetrics(numIDs, numCons, numMetrics int) func(*testing.T) {
 		// This will be validated after every call to RouteMetrics.
 		expected := make(map[pipeline.ID]int, numCons)
 
-		for i := 0; i < numMetrics; i++ {
+		for i := range numMetrics {
 			// Build a random set of ids (no duplicates)
 			randCons := make(map[pipeline.ID]bool, numIDs)
-			for j := 0; j < numIDs; j++ {
+			for j := range numIDs {
 				// This number should be pretty random and less than numCons
 				conNum := (numCons + numIDs + i + j) % numCons
 				randCons[allIDs[conNum]] = true
