diff --git a/cmd/mdatagen/internal/samplescraper/generated_component_test.go b/cmd/mdatagen/internal/samplescraper/generated_component_test.go
index 2cf588df5..c43f41b49 100644
--- a/cmd/mdatagen/internal/samplescraper/generated_component_test.go
+++ b/cmd/mdatagen/internal/samplescraper/generated_component_test.go
@@ -16,8 +16,10 @@ import (
 	"go.opentelemetry.io/collector/scraper/scrapertest"
 )
 
+var typ = component.MustNewType("sample")
+
 func TestComponentFactoryType(t *testing.T) {
-	require.Equal(t, "sample", NewFactory().Type().String())
+	require.Equal(t, typ, NewFactory().Type())
 }
 
 func TestComponentConfigStruct(t *testing.T) {
