diff --git a/pdata/pcommon/generated_int32slice.go b/pdata/pcommon/generated_int32slice.go
index 427090b99..d2bb746df 100644
--- a/pdata/pcommon/generated_int32slice.go
+++ b/pdata/pcommon/generated_int32slice.go
@@ -36,13 +36,13 @@ func NewInt32Slice() Int32Slice {
 
 // AsRaw returns a copy of the []int32 slice.
 func (ms Int32Slice) AsRaw() []int32 {
-	return internal.CopyInt32Slice(nil, *ms.getOrig())
+	return copyInt32Slice(nil, *ms.getOrig())
 }
 
 // FromRaw copies raw []int32 into the slice Int32Slice.
 func (ms Int32Slice) FromRaw(val []int32) {
 	ms.getState().AssertMutable()
-	*ms.getOrig() = internal.CopyInt32Slice(*ms.getOrig(), val)
+	*ms.getOrig() = copyInt32Slice(*ms.getOrig(), val)
 }
 
 // Len returns length of the []int32 slice value.
@@ -155,10 +155,14 @@ func (ms Int32Slice) CopyTo(dest Int32Slice) {
 	if ms.getOrig() == dest.getOrig() {
 		return
 	}
-	*dest.getOrig() = internal.CopyInt32Slice(*dest.getOrig(), *ms.getOrig())
+	*dest.getOrig() = copyInt32Slice(*dest.getOrig(), *ms.getOrig())
 }
 
 // Equal checks equality with another Int32Slice
 func (ms Int32Slice) Equal(val Int32Slice) bool {
 	return slices.Equal(*ms.getOrig(), *val.getOrig())
 }
+
+func copyInt32Slice(dst, src []int32) []int32 {
+	return append(dst[:0], src...)
+}
