diff --git a/extension/memorylimiterextension/go.mod b/extension/memorylimiterextension/go.mod
index 78ef9d907..a2e28b1c7 100644
--- a/extension/memorylimiterextension/go.mod
+++ b/extension/memorylimiterextension/go.mod
@@ -9,7 +9,7 @@ require (
 	go.opentelemetry.io/collector/extension v0.95.0
 	go.opentelemetry.io/otel/metric v1.24.0
 	go.opentelemetry.io/otel/trace v1.24.0
-	go.uber.org/zap v1.26.0
+	go.uber.org/zap v1.27.0
 )
 
 require (
