diff --git a/go.mod b/go.mod
index 3b48f945e..dcae4fdaf 100644
--- a/go.mod
+++ b/go.mod
@@ -23,7 +23,7 @@ require (
 	go.opentelemetry.io/collector/component v0.0.0-20221115212826-19db2f15c85c
 	go.opentelemetry.io/collector/consumer v0.0.0-20221116212002-2454cfb9ab7f
 	go.opentelemetry.io/collector/extension/zpagesextension v0.64.1
-	go.opentelemetry.io/collector/featuregate v0.0.0-00010101000000-000000000000
+	go.opentelemetry.io/collector/featuregate v0.0.0-20221117214536-6a117bfc3737
 	go.opentelemetry.io/collector/pdata v0.64.1
 	go.opentelemetry.io/collector/processor/batchprocessor v0.64.1
 	go.opentelemetry.io/collector/semconv v0.64.1
