diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index bad5e7c38..3ff44a688 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -5,7 +5,7 @@ go 1.21
 require (
 	github.com/fsnotify/fsnotify v1.7.0
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/config/configopaque v1.7.0
+	go.opentelemetry.io/collector/config/configopaque v1.8.0
 )
 
 require (
