diff --git a/config/configoptional/go.mod b/config/configoptional/go.mod
index 97fb4a3c6..cd5b26886 100644
--- a/config/configoptional/go.mod
+++ b/config/configoptional/go.mod
@@ -4,9 +4,9 @@ go 1.24.0
 
 require (
 	github.com/stretchr/testify v1.11.1
-	go.opentelemetry.io/collector/confmap v1.47.0
-	go.opentelemetry.io/collector/confmap/xconfmap v0.141.0
-	go.opentelemetry.io/collector/featuregate v1.47.0
+	go.opentelemetry.io/collector/confmap v1.48.0
+	go.opentelemetry.io/collector/confmap/xconfmap v0.142.0
+	go.opentelemetry.io/collector/featuregate v1.48.0
 	go.uber.org/goleak v1.3.0
 )
 
