diff --git a/cmd/mdatagen/internal/templates/component_test.go.tmpl b/cmd/mdatagen/internal/templates/component_test.go.tmpl
index f224dec8a..bb847f662 100644
--- a/cmd/mdatagen/internal/templates/component_test.go.tmpl
+++ b/cmd/mdatagen/internal/templates/component_test.go.tmpl
@@ -36,6 +36,11 @@ import (
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/receiver/receivertest"
 	{{- end }}
+	{{- if isScraper }}
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/scraper"
+	"go.opentelemetry.io/collector/scraper/scrapertest"
+	{{- end }}
 	{{- if isExtension }}
 	"go.opentelemetry.io/collector/extension/extensiontest"
 	{{- end }}
@@ -325,6 +330,75 @@ func TestComponentLifecycle(t *testing.T) {
 }
 {{ end }}
 
+{{ if isScraper }}
+func TestComponentLifecycle(t *testing.T) {
+	factory := NewFactory()
+
+	tests := []struct{
+		name string
+		createFn func(ctx context.Context, set scraper.Settings, cfg component.Config) (component.Component, error)
+	}{
+{{ if supportsLogs }}
+		{
+			name: "logs",
+			createFn: func(ctx context.Context, set scraper.Settings, cfg component.Config) (component.Component, error) {
+				return factory.CreateLogs(ctx, set, cfg)
+			},
+		},
+{{ end }}
+{{ if supportsMetrics }}
+		{
+			name: "metrics",
+			createFn: func(ctx context.Context, set scraper.Settings, cfg component.Config) (component.Component, error) {
+				return factory.CreateMetrics(ctx, set, cfg)
+			},
+		},
+{{ end }}
+{{ if supportsTraces }}
+		{
+			name: "traces",
+			createFn: func(ctx context.Context, set scraper.Settings, cfg component.Config) (component.Component, error) {
+				return factory.CreateTraces(ctx, set, cfg)
+			},
+		},
+{{ end }}
+	}
+
+	cm, err := confmaptest.LoadConf("metadata.yaml")
+	require.NoError(t, err)
+	cfg := factory.CreateDefaultConfig()
+	sub, err := cm.Sub("tests::config")
+	require.NoError(t, err)
+	require.NoError(t, sub.Unmarshal(&cfg))
+
+	for _, tt := range tests {
+		{{- if not .Tests.SkipShutdown }}
+		t.Run(tt.name + "-shutdown", func(t *testing.T) {
+			c, err := tt.createFn(context.Background(), scrapertest.NewNopSettings(), cfg)
+			require.NoError(t, err)
+			err = c.Shutdown(context.Background())
+			require.NoError(t, err)
+		})
+		{{- end }}
+
+		{{- if not .Tests.SkipLifecycle }}
+		t.Run(tt.name + "-lifecycle", func(t *testing.T) {
+			firstRcvr, err := tt.createFn(context.Background(), scrapertest.NewNopSettings(), cfg)
+			require.NoError(t, err)
+			host := {{ .Tests.Host }}
+			require.NoError(t, err)
+			require.NoError(t, firstRcvr.Start(context.Background(), host))
+			require.NoError(t, firstRcvr.Shutdown(context.Background()))
+			secondRcvr, err := tt.createFn(context.Background(), scrapertest.NewNopSettings(), cfg)
+			require.NoError(t, err)
+			require.NoError(t, secondRcvr.Start(context.Background(), host))
+			require.NoError(t, secondRcvr.Shutdown(context.Background()))
+		})
+		{{- end }}
+	}
+}
+{{ end }}
+
 {{ if isExtension }}
 func TestComponentLifecycle(t *testing.T) {
 	factory := NewFactory()
