diff --git a/internal/e2e/go.mod b/internal/e2e/go.mod
index 63e30b39a..fb2df57ef 100644
--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -99,7 +99,7 @@ require (
 	go.opentelemetry.io/collector/processor v0.115.0 // indirect
 	go.opentelemetry.io/collector/processor/processorprofiles v0.115.0 // indirect
 	go.opentelemetry.io/collector/processor/processortest v0.115.0 // indirect
-	go.opentelemetry.io/collector/receiver/receiverprofiles v0.115.0 // indirect
+	go.opentelemetry.io/collector/receiver/xreceiver v0.115.0 // indirect
 	go.opentelemetry.io/collector/semconv v0.115.0 // indirect
 	go.opentelemetry.io/contrib/bridges/otelzap v0.6.0 // indirect
 	go.opentelemetry.io/contrib/config v0.10.0 // indirect
@@ -212,7 +212,7 @@ replace go.opentelemetry.io/collector/service => ../../service
 
 replace go.opentelemetry.io/collector/extension/extensioncapabilities => ../../extension/extensioncapabilities
 
-replace go.opentelemetry.io/collector/receiver/receiverprofiles => ../../receiver/receiverprofiles
+replace go.opentelemetry.io/collector/receiver/xreceiver => ../../receiver/xreceiver
 
 replace go.opentelemetry.io/collector/receiver/receivertest => ../../receiver/receivertest
 
