diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index 78e1328d1..7b81b2287 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -3,7 +3,7 @@ module go.opentelemetry.io/collector/config/configtls
 go 1.24.0
 
 require (
-	github.com/foxboron/go-tpm-keyfiles v0.0.0-20250323135004-b31fac66206e
+	github.com/foxboron/go-tpm-keyfiles v0.0.0-20250903184740-5d135037bd4d
 	github.com/fsnotify/fsnotify v1.9.0
 	github.com/google/go-tpm v0.9.6
 	github.com/stretchr/testify v1.11.1
