diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 0d3278455..e554d2450 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -14,7 +14,6 @@ require (
 	go.opentelemetry.io/collector/config/configauth v0.118.0
 	go.opentelemetry.io/collector/config/configcompression v1.24.0
 	go.opentelemetry.io/collector/config/configopaque v1.24.0
-	go.opentelemetry.io/collector/config/configtelemetry v0.118.0
 	go.opentelemetry.io/collector/config/configtls v1.24.0
 	go.opentelemetry.io/collector/extension/auth v0.118.0
 	go.opentelemetry.io/collector/extension/auth/authtest v0.118.0
@@ -35,6 +34,7 @@ require (
 	github.com/google/uuid v1.6.0 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	go.opentelemetry.io/auto/sdk v1.1.0 // indirect
+	go.opentelemetry.io/collector/config/configtelemetry v0.118.0 // indirect
 	go.opentelemetry.io/collector/extension v0.118.0 // indirect
 	go.opentelemetry.io/collector/pdata v1.24.0 // indirect
 	go.opentelemetry.io/otel/metric v1.34.0 // indirect
