diff --git a/extension/ballastextension/factory.go b/extension/ballastextension/factory.go
index 13bb28c22..d0446d6a6 100644
--- a/extension/ballastextension/factory.go
+++ b/extension/ballastextension/factory.go
@@ -18,7 +18,6 @@ import (
 	"context"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/extension"
 	"go.opentelemetry.io/collector/internal/iruntime"
 )
@@ -37,9 +36,7 @@ func NewFactory() extension.Factory {
 }
 
 func createDefaultConfig() component.Config {
-	return &Config{
-		ExtensionSettings: config.NewExtensionSettings(component.NewID(typeStr)),
-	}
+	return &Config{}
 }
 
 func createExtension(_ context.Context, set extension.CreateSettings, cfg component.Config) (extension.Extension, error) {
