diff --git a/service/extensions/config.go b/service/extensions/config.go
index 44c75b239..da7b851ee 100644
--- a/service/extensions/config.go
+++ b/service/extensions/config.go
@@ -14,9 +14,7 @@
 
 package extensions // import "go.opentelemetry.io/collector/service/extensions"
 
-import (
-	"go.opentelemetry.io/collector/config"
-)
+import "go.opentelemetry.io/collector/component"
 
 // Config represents the ordered list of extensions configured for the service.
-type Config []config.ComponentID
+type Config []component.ID
