diff --git a/pdata/xpdata/entity/generated_entityrefslice.go b/pdata/xpdata/entity/generated_entityrefslice.go
index a5c4cc1ff..268b35a67 100644
--- a/pdata/xpdata/entity/generated_entityrefslice.go
+++ b/pdata/xpdata/entity/generated_entityrefslice.go
@@ -21,13 +21,13 @@ import (
 //
 // Must use NewEntityRefSlice function to create new instances.
 // Important: zero-initialized instance is not valid for use.
-type EntityRefSlice internal.EntityRefSlice
+type EntityRefSlice internal.EntityRefSliceWrapper
 
 func newEntityRefSlice(orig *[]*otlpcommon.EntityRef, state *internal.State) EntityRefSlice {
-	return EntityRefSlice(internal.NewEntityRefSlice(orig, state))
+	return EntityRefSlice(internal.NewEntityRefSliceWrapper(orig, state))
 }
 
-// NewEntityRefSlice creates a EntityRefSlice with 0 elements.
+// NewEntityRefSlice creates a EntityRefSliceWrapper with 0 elements.
 // Can use "EnsureCapacity" to initialize with a given capacity.
 func NewEntityRefSlice() EntityRefSlice {
 	orig := []*otlpcommon.EntityRef(nil)
@@ -96,7 +96,7 @@ func (es EntityRefSlice) EnsureCapacity(newCap int) {
 // It returns the newly added EntityRef.
 func (es EntityRefSlice) AppendEmpty() EntityRef {
 	es.getState().AssertMutable()
-	*es.getOrig() = append(*es.getOrig(), internal.NewOrigEntityRef())
+	*es.getOrig() = append(*es.getOrig(), internal.NewEntityRef())
 	return es.At(es.Len() - 1)
 }
 
@@ -125,7 +125,7 @@ func (es EntityRefSlice) RemoveIf(f func(EntityRef) bool) {
 	newLen := 0
 	for i := 0; i < len(*es.getOrig()); i++ {
 		if f(es.At(i)) {
-			internal.DeleteOrigEntityRef((*es.getOrig())[i], true)
+			internal.DeleteEntityRef((*es.getOrig())[i], true)
 			(*es.getOrig())[i] = nil
 
 			continue
@@ -149,7 +149,7 @@ func (es EntityRefSlice) CopyTo(dest EntityRefSlice) {
 	if es.getOrig() == dest.getOrig() {
 		return
 	}
-	*dest.getOrig() = internal.CopyOrigEntityRefSlice(*dest.getOrig(), *es.getOrig())
+	*dest.getOrig() = internal.CopyEntityRefSlice(*dest.getOrig(), *es.getOrig())
 }
 
 // Sort sorts the EntityRef elements within EntityRefSlice given the
@@ -161,9 +161,9 @@ func (es EntityRefSlice) Sort(less func(a, b EntityRef) bool) {
 }
 
 func (ms EntityRefSlice) getOrig() *[]*otlpcommon.EntityRef {
-	return internal.GetOrigEntityRefSlice(internal.EntityRefSlice(ms))
+	return internal.GetEntityRefSliceOrig(internal.EntityRefSliceWrapper(ms))
 }
 
 func (ms EntityRefSlice) getState() *internal.State {
-	return internal.GetEntityRefSliceState(internal.EntityRefSlice(ms))
+	return internal.GetEntityRefSliceState(internal.EntityRefSliceWrapper(ms))
 }
