diff --git a/pdata/xpdata/entity/generated_entityref.go b/pdata/xpdata/entity/generated_entityref.go
index a56e7d93d..100c381c3 100644
--- a/pdata/xpdata/entity/generated_entityref.go
+++ b/pdata/xpdata/entity/generated_entityref.go
@@ -17,10 +17,10 @@ import (
 //
 // Must use NewEntityRef function to create new instances.
 // Important: zero-initialized instance is not valid for use.
-type EntityRef internal.EntityRef
+type EntityRef internal.EntityRefWrapper
 
 func newEntityRef(orig *otlpcommon.EntityRef, state *internal.State) EntityRef {
-	return EntityRef(internal.NewEntityRef(orig, state))
+	return EntityRef(internal.NewEntityRefWrapper(orig, state))
 }
 
 // NewEntityRef creates a new empty EntityRef.
@@ -28,7 +28,7 @@ func newEntityRef(orig *otlpcommon.EntityRef, state *internal.State) EntityRef {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewEntityRef() EntityRef {
-	return newEntityRef(internal.NewOrigEntityRef(), internal.NewState())
+	return newEntityRef(internal.NewEntityRef(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
@@ -40,7 +40,7 @@ func (ms EntityRef) MoveTo(dest EntityRef) {
 	if ms.getOrig() == dest.getOrig() {
 		return
 	}
-	internal.DeleteOrigEntityRef(dest.getOrig(), false)
+	internal.DeleteEntityRef(dest.getOrig(), false)
 	*dest.getOrig(), *ms.getOrig() = *ms.getOrig(), *dest.getOrig()
 }
 
@@ -68,24 +68,24 @@ func (ms EntityRef) SetType(v string) {
 
 // IdKeys returns the IdKeys associated with this EntityRef.
 func (ms EntityRef) IdKeys() pcommon.StringSlice {
-	return pcommon.StringSlice(internal.NewStringSlice(&ms.getOrig().IdKeys, ms.getState()))
+	return pcommon.StringSlice(internal.NewStringSliceWrapper(&ms.getOrig().IdKeys, ms.getState()))
 }
 
 // DescriptionKeys returns the DescriptionKeys associated with this EntityRef.
 func (ms EntityRef) DescriptionKeys() pcommon.StringSlice {
-	return pcommon.StringSlice(internal.NewStringSlice(&ms.getOrig().DescriptionKeys, ms.getState()))
+	return pcommon.StringSlice(internal.NewStringSliceWrapper(&ms.getOrig().DescriptionKeys, ms.getState()))
 }
 
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms EntityRef) CopyTo(dest EntityRef) {
 	dest.getState().AssertMutable()
-	internal.CopyOrigEntityRef(dest.getOrig(), ms.getOrig())
+	internal.CopyEntityRef(dest.getOrig(), ms.getOrig())
 }
 
 func (ms EntityRef) getOrig() *otlpcommon.EntityRef {
-	return internal.GetOrigEntityRef(internal.EntityRef(ms))
+	return internal.GetEntityRefOrig(internal.EntityRefWrapper(ms))
 }
 
 func (ms EntityRef) getState() *internal.State {
-	return internal.GetEntityRefState(internal.EntityRef(ms))
+	return internal.GetEntityRefState(internal.EntityRefWrapper(ms))
 }
