diff --git a/exporter/zipkinexporter/config_test.go b/exporter/zipkinexporter/config_test.go
index a8fc34d76..2c14dba5f 100644
--- a/exporter/zipkinexporter/config_test.go
+++ b/exporter/zipkinexporter/config_test.go
@@ -15,12 +15,15 @@
 package zipkinexporter
 
 import (
+	"context"
 	"path"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config/configtest"
 )
@@ -48,4 +51,7 @@ func TestLoadConfig(t *testing.T) {
 	assert.Equal(t, "zipkin/2", e1.(*Config).Name())
 	assert.Equal(t, "https://somedest:1234/api/v2/spans", e1.(*Config).Endpoint)
 	assert.Equal(t, "proto", e1.(*Config).Format)
+	params := component.ExporterCreateParams{Logger: zap.NewNop()}
+	_, err = factory.CreateTraceExporter(context.Background(), params, e1)
+	require.NoError(t, err)
 }
