diff --git a/component/receiver.go b/component/receiver.go
index 28a8164b9..fb2c88542 100644
--- a/component/receiver.go
+++ b/component/receiver.go
@@ -17,6 +17,7 @@ package component
 import (
 	"context"
 
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/config"
@@ -61,6 +62,9 @@ type ReceiverCreateSettings struct {
 	// component to be used later as well.
 	Logger *zap.Logger
 
+	// TracerProvider that the factory can pass to other instrumented third-party libraries.
+	TracerProvider trace.TracerProvider
+
 	// BuildInfo can be used by components for informational purposes.
 	BuildInfo BuildInfo
 }
