diff --git a/internal/pprofserver/pprofserver.go b/internal/pprofserver/pprofserver.go
index 50346afda..fd3505b1f 100644
--- a/internal/pprofserver/pprofserver.go
+++ b/internal/pprofserver/pprofserver.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
