diff --git a/processor/internal/processor.go b/processor/internal/processor.go
deleted file mode 100644
index 4e7a07a18..000000000
--- a/processor/internal/processor.go
+++ /dev/null
@@ -1,17 +0,0 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
-
-package internal // import "go.opentelemetry.io/collector/processor/internal"
-
-import "go.opentelemetry.io/collector/component"
-
-// Settings is passed to Create* functions in Factory.
-type Settings struct {
-	// ID returns the ID of the component that will be created.
-	ID component.ID
-
-	component.TelemetrySettings
-
-	// BuildInfo can be used by components for informational purposes
-	BuildInfo component.BuildInfo
-}
