diff --git a/cmd/builder/internal/builder/main_test.go b/cmd/builder/internal/builder/main_test.go
index 400740610..aa90c99ca 100644
--- a/cmd/builder/internal/builder/main_test.go
+++ b/cmd/builder/internal/builder/main_test.go
@@ -128,12 +128,12 @@ func TestGenerateInvalidOutputPath(t *testing.T) {
 func TestVersioning(t *testing.T) {
 	replaces := generateReplaces()
 	tests := []struct {
-		description string
+		name        string
 		cfgBuilder  func() Config
 		expectedErr error
 	}{
 		{
-			description: "defaults",
+			name: "defaults",
 			cfgBuilder: func() Config {
 				cfg := newTestConfig()
 				cfg.Distribution.Go = "go"
@@ -143,7 +143,7 @@ func TestVersioning(t *testing.T) {
 			expectedErr: nil,
 		},
 		{
-			description: "require otelcol",
+			name: "require otelcol",
 			cfgBuilder: func() Config {
 				cfg := newTestConfig()
 				cfg.Distribution.Go = "go"
@@ -154,7 +154,7 @@ func TestVersioning(t *testing.T) {
 			expectedErr: nil,
 		},
 		{
-			description: "only gomod file, skip generate",
+			name: "only gomod file, skip generate",
 			cfgBuilder: func() Config {
 				cfg := newTestConfig()
 				tempDir := t.TempDir()
@@ -168,7 +168,7 @@ func TestVersioning(t *testing.T) {
 			expectedErr: ErrDepNotFound,
 		},
 		{
-			description: "old otel version",
+			name: "old otel version",
 			cfgBuilder: func() Config {
 				cfg := newTestConfig()
 				cfg.Verbose = true
@@ -200,7 +200,7 @@ func TestVersioning(t *testing.T) {
 			expectedErr: nil,
 		},
 		{
-			description: "old component version",
+			name: "old component version",
 			cfgBuilder: func() Config {
 				cfg := newTestConfig()
 				cfg.Distribution.Go = "go"
@@ -216,7 +216,7 @@ func TestVersioning(t *testing.T) {
 			expectedErr: nil,
 		},
 		{
-			description: "old component version without strict mode",
+			name: "old component version without strict mode",
 			cfgBuilder: func() Config {
 				cfg := newTestConfig()
 				cfg.Distribution.Go = "go"
@@ -233,14 +233,14 @@ func TestVersioning(t *testing.T) {
 			expectedErr: nil,
 		},
 	}
-	for _, tc := range tests {
-		t.Run(tc.description, func(t *testing.T) {
-			cfg := tc.cfgBuilder()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := tt.cfgBuilder()
 			require.NoError(t, cfg.SetBackwardsCompatibility())
 			require.NoError(t, cfg.Validate())
 			require.NoError(t, cfg.ParseModules())
 			err := GenerateAndCompile(cfg)
-			require.ErrorIs(t, err, tc.expectedErr)
+			require.ErrorIs(t, err, tt.expectedErr)
 		})
 	}
 }
@@ -263,11 +263,11 @@ func TestSkipGenerate(t *testing.T) {
 func TestGenerateAndCompile(t *testing.T) {
 	replaces := generateReplaces()
 	testCases := []struct {
-		testCase   string
+		name       string
 		cfgBuilder func(t *testing.T) Config
 	}{
 		{
-			testCase: "Default Configuration Compilation",
+			name: "Default Configuration Compilation",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -278,7 +278,7 @@ func TestGenerateAndCompile(t *testing.T) {
 			},
 		},
 		{
-			testCase: "LDFlags Compilation",
+			name: "LDFlags Compilation",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -290,7 +290,7 @@ func TestGenerateAndCompile(t *testing.T) {
 			},
 		},
 		{
-			testCase: "Debug Compilation",
+			name: "Debug Compilation",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -303,7 +303,7 @@ func TestGenerateAndCompile(t *testing.T) {
 			},
 		},
 		{
-			testCase: "No providers",
+			name: "No providers",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -315,7 +315,7 @@ func TestGenerateAndCompile(t *testing.T) {
 			},
 		},
 		{
-			testCase: "Pre-confmap factories",
+			name: "Pre-confmap factories",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -328,7 +328,7 @@ func TestGenerateAndCompile(t *testing.T) {
 			},
 		},
 		{
-			testCase: "With confmap factories",
+			name: "With confmap factories",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -341,7 +341,7 @@ func TestGenerateAndCompile(t *testing.T) {
 			},
 		},
 		{
-			testCase: "ConfResolverDefaultURIScheme set",
+			name: "ConfResolverDefaultURIScheme set",
 			cfgBuilder: func(t *testing.T) Config {
 				cfg := newTestConfig()
 				err := cfg.SetBackwardsCompatibility()
@@ -357,7 +357,7 @@ func TestGenerateAndCompile(t *testing.T) {
 	}
 
 	for _, tt := range testCases {
-		t.Run(tt.testCase, func(t *testing.T) {
+		t.Run(tt.name, func(t *testing.T) {
 			cfg := tt.cfgBuilder(t)
 			assert.NoError(t, cfg.Validate())
 			assert.NoError(t, cfg.SetGoPath())
