diff --git a/pdata/internal/generated_wrapper_mapping.go b/pdata/internal/generated_wrapper_mapping.go
index 357ceeb18..b9220b056 100644
--- a/pdata/internal/generated_wrapper_mapping.go
+++ b/pdata/internal/generated_wrapper_mapping.go
@@ -56,10 +56,6 @@ func CopyOrigMapping(dest, src *otlpprofiles.Mapping) {
 	dest.FileOffset = src.FileOffset
 	dest.FilenameStrindex = src.FilenameStrindex
 	dest.AttributeIndices = CopyOrigInt32Slice(dest.AttributeIndices, src.AttributeIndices)
-	dest.HasFunctions = src.HasFunctions
-	dest.HasFilenames = src.HasFilenames
-	dest.HasLineNumbers = src.HasLineNumbers
-	dest.HasInlineFrames = src.HasInlineFrames
 }
 
 func GenTestOrigMapping() *otlpprofiles.Mapping {
@@ -69,10 +65,6 @@ func GenTestOrigMapping() *otlpprofiles.Mapping {
 	orig.FileOffset = uint64(13)
 	orig.FilenameStrindex = int32(13)
 	orig.AttributeIndices = GenerateOrigTestInt32Slice()
-	orig.HasFunctions = true
-	orig.HasFilenames = true
-	orig.HasLineNumbers = true
-	orig.HasInlineFrames = true
 	return orig
 }
 
@@ -105,22 +97,6 @@ func MarshalJSONOrigMapping(orig *otlpprofiles.Mapping, dest *json.Stream) {
 		}
 		dest.WriteArrayEnd()
 	}
-	if orig.HasFunctions != false {
-		dest.WriteObjectField("hasFunctions")
-		dest.WriteBool(orig.HasFunctions)
-	}
-	if orig.HasFilenames != false {
-		dest.WriteObjectField("hasFilenames")
-		dest.WriteBool(orig.HasFilenames)
-	}
-	if orig.HasLineNumbers != false {
-		dest.WriteObjectField("hasLineNumbers")
-		dest.WriteBool(orig.HasLineNumbers)
-	}
-	if orig.HasInlineFrames != false {
-		dest.WriteObjectField("hasInlineFrames")
-		dest.WriteBool(orig.HasInlineFrames)
-	}
 	dest.WriteObjectEnd()
 }
 
@@ -141,14 +117,6 @@ func UnmarshalJSONOrigMapping(orig *otlpprofiles.Mapping, iter *json.Iterator) {
 				orig.AttributeIndices = append(orig.AttributeIndices, iter.ReadInt32())
 			}
 
-		case "hasFunctions", "has_functions":
-			orig.HasFunctions = iter.ReadBool()
-		case "hasFilenames", "has_filenames":
-			orig.HasFilenames = iter.ReadBool()
-		case "hasLineNumbers", "has_line_numbers":
-			orig.HasLineNumbers = iter.ReadBool()
-		case "hasInlineFrames", "has_inline_frames":
-			orig.HasInlineFrames = iter.ReadBool()
 		default:
 			iter.Skip()
 		}
@@ -178,18 +146,6 @@ func SizeProtoOrigMapping(orig *otlpprofiles.Mapping) int {
 		}
 		n += 1 + proto.Sov(uint64(l)) + l
 	}
-	if orig.HasFunctions {
-		n += 2
-	}
-	if orig.HasFilenames {
-		n += 2
-	}
-	if orig.HasLineNumbers {
-		n += 2
-	}
-	if orig.HasInlineFrames {
-		n += 2
-	}
 	return n
 }
 
@@ -227,46 +183,6 @@ func MarshalProtoOrigMapping(orig *otlpprofiles.Mapping, buf []byte) int {
 		pos--
 		buf[pos] = 0x2a
 	}
-	if orig.HasFunctions {
-		pos--
-		if orig.HasFunctions {
-			buf[pos] = 1
-		} else {
-			buf[pos] = 0
-		}
-		pos--
-		buf[pos] = 0x30
-	}
-	if orig.HasFilenames {
-		pos--
-		if orig.HasFilenames {
-			buf[pos] = 1
-		} else {
-			buf[pos] = 0
-		}
-		pos--
-		buf[pos] = 0x38
-	}
-	if orig.HasLineNumbers {
-		pos--
-		if orig.HasLineNumbers {
-			buf[pos] = 1
-		} else {
-			buf[pos] = 0
-		}
-		pos--
-		buf[pos] = 0x40
-	}
-	if orig.HasInlineFrames {
-		pos--
-		if orig.HasInlineFrames {
-			buf[pos] = 1
-		} else {
-			buf[pos] = 0
-		}
-		pos--
-		buf[pos] = 0x48
-	}
 	return len(buf) - pos
 }
 
@@ -362,54 +278,6 @@ func UnmarshalProtoOrigMapping(orig *otlpprofiles.Mapping, buf []byte) error {
 			default:
 				return fmt.Errorf("proto: wrong wireType = %d for field AttributeIndices", wireType)
 			}
-
-		case 6:
-			if wireType != proto.WireTypeVarint {
-				return fmt.Errorf("proto: wrong wireType = %d for field HasFunctions", wireType)
-			}
-			var num uint64
-			num, pos, err = proto.ConsumeVarint(buf, pos)
-			if err != nil {
-				return err
-			}
-
-			orig.HasFunctions = num != 0
-
-		case 7:
-			if wireType != proto.WireTypeVarint {
-				return fmt.Errorf("proto: wrong wireType = %d for field HasFilenames", wireType)
-			}
-			var num uint64
-			num, pos, err = proto.ConsumeVarint(buf, pos)
-			if err != nil {
-				return err
-			}
-
-			orig.HasFilenames = num != 0
-
-		case 8:
-			if wireType != proto.WireTypeVarint {
-				return fmt.Errorf("proto: wrong wireType = %d for field HasLineNumbers", wireType)
-			}
-			var num uint64
-			num, pos, err = proto.ConsumeVarint(buf, pos)
-			if err != nil {
-				return err
-			}
-
-			orig.HasLineNumbers = num != 0
-
-		case 9:
-			if wireType != proto.WireTypeVarint {
-				return fmt.Errorf("proto: wrong wireType = %d for field HasInlineFrames", wireType)
-			}
-			var num uint64
-			num, pos, err = proto.ConsumeVarint(buf, pos)
-			if err != nil {
-				return err
-			}
-
-			orig.HasInlineFrames = num != 0
 		default:
 			pos, err = proto.ConsumeUnknown(buf, pos, wireType)
 			if err != nil {
