diff --git a/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl b/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl
index d8b0c6678..4818fd4ca 100644
--- a/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl
+++ b/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl
@@ -19,7 +19,11 @@ func Test{{ .structName }}(t *testing.T) {
 	es = new{{ .structName }}(&[]{{ .originElementType }}{}, &state)
 	assert.Equal(t, 0, es.Len())
 
+	{{ if eq .elementName "Value" -}}
+	emptyVal := New{{ .elementName }}Empty()
+	{{- else }}
 	emptyVal := New{{ .elementName }}()
+	{{- end }}
 	{{- if .isCommon }}
 	testVal := {{ .elementName }}(internal.GenerateTest{{ .elementName }}())
 	{{- else }}
