diff --git a/exporter/exporterhelper/exporterhelperprofiles/profiles.go b/exporter/exporterhelper/exporterhelperprofiles/profiles.go
index bb6aa00de..ca33c94c1 100644
--- a/exporter/exporterhelper/exporterhelperprofiles/profiles.go
+++ b/exporter/exporterhelper/exporterhelperprofiles/profiles.go
@@ -12,7 +12,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/consumer/consumererror/consumererrorprofiles"
-	"go.opentelemetry.io/collector/consumer/consumerprofiles"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
 	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
@@ -29,17 +29,17 @@ var (
 
 type profilesRequest struct {
 	pd     pprofile.Profiles
-	pusher consumerprofiles.ConsumeProfilesFunc
+	pusher xconsumer.ConsumeProfilesFunc
 }
 
-func newProfilesRequest(pd pprofile.Profiles, pusher consumerprofiles.ConsumeProfilesFunc) exporterhelper.Request {
+func newProfilesRequest(pd pprofile.Profiles, pusher xconsumer.ConsumeProfilesFunc) exporterhelper.Request {
 	return &profilesRequest{
 		pd:     pd,
 		pusher: pusher,
 	}
 }
 
-func newProfileRequestUnmarshalerFunc(pusher consumerprofiles.ConsumeProfilesFunc) exporterqueue.Unmarshaler[exporterhelper.Request] {
+func newProfileRequestUnmarshalerFunc(pusher xconsumer.ConsumeProfilesFunc) exporterqueue.Unmarshaler[exporterhelper.Request] {
 	return func(bytes []byte) (exporterhelper.Request, error) {
 		profiles, err := profilesUnmarshaler.UnmarshalProfiles(bytes)
 		if err != nil {
@@ -71,7 +71,7 @@ func (req *profilesRequest) ItemsCount() int {
 
 type profileExporter struct {
 	*internal.BaseExporter
-	consumerprofiles.Profiles
+	xconsumer.Profiles
 }
 
 // NewProfilesExporter creates an exporterprofiles.Profiles that records observability metrics and wraps every request with a Span.
@@ -79,7 +79,7 @@ func NewProfilesExporter(
 	ctx context.Context,
 	set exporter.Settings,
 	cfg component.Config,
-	pusher consumerprofiles.ConsumeProfilesFunc,
+	pusher xconsumer.ConsumeProfilesFunc,
 	options ...exporterhelper.Option,
 ) (exporterprofiles.Profiles, error) {
 	if cfg == nil {
@@ -100,7 +100,7 @@ func NewProfilesExporter(
 type RequestFromProfilesFunc func(context.Context, pprofile.Profiles) (exporterhelper.Request, error)
 
 // requestFromProfiles returns a RequestFromProfilesFunc that converts pprofile.Profiles into a Request.
-func requestFromProfiles(pusher consumerprofiles.ConsumeProfilesFunc) RequestFromProfilesFunc {
+func requestFromProfiles(pusher xconsumer.ConsumeProfilesFunc) RequestFromProfilesFunc {
 	return func(_ context.Context, profiles pprofile.Profiles) (exporterhelper.Request, error) {
 		return newProfilesRequest(profiles, pusher), nil
 	}
@@ -128,7 +128,7 @@ func NewProfilesRequestExporter(
 		return nil, err
 	}
 
-	tc, err := consumerprofiles.NewProfiles(func(ctx context.Context, pd pprofile.Profiles) error {
+	tc, err := xconsumer.NewProfiles(func(ctx context.Context, pd pprofile.Profiles) error {
 		req, cErr := converter(ctx, pd)
 		if cErr != nil {
 			set.Logger.Error("Failed to convert profiles. Dropping data.",
