diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index d58a71fb0..fe7fd2685 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -25,7 +25,7 @@ require (
 	go.opentelemetry.io/collector/extension/memorylimiterextension v0.111.0
 	go.opentelemetry.io/collector/extension/zpagesextension v0.111.0
 	go.opentelemetry.io/collector/otelcol v0.111.0
-	go.opentelemetry.io/collector/processor v0.111.1-0.20241021181817-007f06b7c4a8
+	go.opentelemetry.io/collector/processor v0.111.1-0.20241021235809-403c782d50c6
 	go.opentelemetry.io/collector/processor/batchprocessor v0.111.0
 	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.111.0
 	go.opentelemetry.io/collector/receiver v0.111.0
@@ -93,13 +93,13 @@ require (
 	go.opentelemetry.io/collector/config/configtls v1.17.0 // indirect
 	go.opentelemetry.io/collector/config/internal v0.111.0 // indirect
 	go.opentelemetry.io/collector/connector/connectorprofiles v0.111.0 // indirect
-	go.opentelemetry.io/collector/connector/connectortest v0.0.0-20241021181817-007f06b7c4a8 // indirect
-	go.opentelemetry.io/collector/consumer v0.111.1-0.20241021181817-007f06b7c4a8 // indirect
-	go.opentelemetry.io/collector/consumer/consumererror v0.0.0-20241021181817-007f06b7c4a8 // indirect
-	go.opentelemetry.io/collector/consumer/consumererror/consumererrorprofiles v0.0.0-20241021181817-007f06b7c4a8 // indirect
+	go.opentelemetry.io/collector/connector/connectortest v0.0.0-20241021235809-403c782d50c6 // indirect
+	go.opentelemetry.io/collector/consumer v0.111.1-0.20241021235809-403c782d50c6 // indirect
+	go.opentelemetry.io/collector/consumer/consumererror v0.0.0-20241021235809-403c782d50c6 // indirect
+	go.opentelemetry.io/collector/consumer/consumererror/consumererrorprofiles v0.0.0-20241021235809-403c782d50c6 // indirect
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.111.0 // indirect
 	go.opentelemetry.io/collector/consumer/consumertest v0.111.0 // indirect
-	go.opentelemetry.io/collector/exporter/exporterhelper/exporterhelperprofiles v0.0.0-20241021181817-007f06b7c4a8 // indirect
+	go.opentelemetry.io/collector/exporter/exporterhelper/exporterhelperprofiles v0.0.0-20241021235809-403c782d50c6 // indirect
 	go.opentelemetry.io/collector/exporter/exporterprofiles v0.111.0 // indirect
 	go.opentelemetry.io/collector/exporter/exportertest v0.111.0 // indirect
 	go.opentelemetry.io/collector/extension/auth v0.111.0 // indirect
@@ -111,9 +111,9 @@ require (
 	go.opentelemetry.io/collector/pdata/pprofile v0.111.0 // indirect
 	go.opentelemetry.io/collector/pdata/testdata v0.111.0 // indirect
 	go.opentelemetry.io/collector/pipeline v0.111.0 // indirect
-	go.opentelemetry.io/collector/pipeline/pipelineprofiles v0.0.0-20241021181817-007f06b7c4a8 // indirect
+	go.opentelemetry.io/collector/pipeline/pipelineprofiles v0.0.0-20241021235809-403c782d50c6 // indirect
 	go.opentelemetry.io/collector/processor/processorprofiles v0.111.0 // indirect
-	go.opentelemetry.io/collector/processor/processortest v0.0.0-20241021181817-007f06b7c4a8 // indirect
+	go.opentelemetry.io/collector/processor/processortest v0.0.0-20241021235809-403c782d50c6 // indirect
 	go.opentelemetry.io/collector/receiver/receiverprofiles v0.111.0 // indirect
 	go.opentelemetry.io/collector/semconv v0.111.0 // indirect
 	go.opentelemetry.io/collector/service v0.111.0 // indirect
