diff --git a/processor/memorylimiter/factory.go b/processor/memorylimiter/factory.go
index ca5d1fcd7..d3d06332f 100644
--- a/processor/memorylimiter/factory.go
+++ b/processor/memorylimiter/factory.go
@@ -17,8 +17,6 @@ package memorylimiter
 import (
 	"context"
 
-	"go.uber.org/zap"
-
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer"
@@ -30,6 +28,8 @@ const (
 	typeStr = "memory_limiter"
 )
 
+var processorCapabilities = component.ProcessorCapabilities{MutatesConsumedData: false}
+
 // NewFactory returns a new factory for the Memory Limiter processor.
 func NewFactory() component.ProcessorFactory {
 	return processorhelper.NewFactory(
@@ -57,7 +57,16 @@ func createTraceProcessor(
 	cfg configmodels.Processor,
 	nextConsumer consumer.TraceConsumer,
 ) (component.TraceProcessor, error) {
-	return createProcessor(params.Logger, nextConsumer, nil, nil, cfg)
+	ml, err := newMemoryLimiter(params.Logger, cfg.(*Config))
+	if err != nil {
+		return nil, err
+	}
+	return processorhelper.NewTraceProcessor(
+		cfg,
+		nextConsumer,
+		ml,
+		processorhelper.WithCapabilities(processorCapabilities),
+		processorhelper.WithShutdown(ml.shutdown))
 }
 
 func createMetricsProcessor(
@@ -66,7 +75,16 @@ func createMetricsProcessor(
 	cfg configmodels.Processor,
 	nextConsumer consumer.MetricsConsumer,
 ) (component.MetricsProcessor, error) {
-	return createProcessor(params.Logger, nil, nextConsumer, nil, cfg)
+	ml, err := newMemoryLimiter(params.Logger, cfg.(*Config))
+	if err != nil {
+		return nil, err
+	}
+	return processorhelper.NewMetricsProcessor(
+		cfg,
+		nextConsumer,
+		ml,
+		processorhelper.WithCapabilities(processorCapabilities),
+		processorhelper.WithShutdown(ml.shutdown))
 }
 
 func createLogProcessor(
@@ -75,29 +93,14 @@ func createLogProcessor(
 	cfg configmodels.Processor,
 	nextConsumer consumer.LogConsumer,
 ) (component.LogProcessor, error) {
-	return createProcessor(params.Logger, nil, nil, nextConsumer, cfg)
-}
-
-type TripleTypeProcessor interface {
-	consumer.TraceConsumer
-	consumer.MetricsConsumer
-	consumer.LogConsumer
-	component.Processor
-}
-
-func createProcessor(
-	logger *zap.Logger,
-	traceConsumer consumer.TraceConsumer,
-	metricConsumer consumer.MetricsConsumer,
-	logConsumer consumer.LogConsumer,
-	cfg configmodels.Processor,
-) (TripleTypeProcessor, error) {
-	pCfg := cfg.(*Config)
-	return newMemoryLimiter(
-		logger,
-		traceConsumer,
-		metricConsumer,
-		logConsumer,
-		pCfg,
-	)
+	ml, err := newMemoryLimiter(params.Logger, cfg.(*Config))
+	if err != nil {
+		return nil, err
+	}
+	return processorhelper.NewLogProcessor(
+		cfg,
+		nextConsumer,
+		ml,
+		processorhelper.WithCapabilities(processorCapabilities),
+		processorhelper.WithShutdown(ml.shutdown))
 }
