diff --git a/processor/filterprocessor/factory.go b/processor/filterprocessor/factory.go
index 81e6fa4c7..c5af4fdcf 100644
--- a/processor/filterprocessor/factory.go
+++ b/processor/filterprocessor/factory.go
@@ -15,8 +15,6 @@
 package filterprocessor
 
 import (
-	"context"
-
 	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/component"
@@ -50,12 +48,11 @@ func (f Factory) CreateDefaultConfig() configmodels.Processor {
 }
 
 // CreateTraceProcessor creates a trace processor based on this config.
-func (f *Factory) CreateTraceProcessor(
-	ctx context.Context,
-	params component.ProcessorCreateParams,
-	nextConsumer consumer.TraceConsumer,
-	cfg configmodels.Processor,
-) (component.TraceProcessor, error) {
+func (f Factory) CreateTraceProcessor(
+	logger *zap.Logger,
+	nextConsumer consumer.TraceConsumerOld,
+	c configmodels.Processor,
+) (component.TraceProcessorOld, error) {
 	return nil, configerror.ErrDataTypeIsNotSupported
 }
 
