diff --git a/processor/processorhelper/processor.go b/processor/processorhelper/processor.go
index 85c1a40c6..286fd25ba 100644
--- a/processor/processorhelper/processor.go
+++ b/processor/processorhelper/processor.go
@@ -20,6 +20,7 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/component/componenthelper"
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/pdata"
@@ -30,12 +31,6 @@ import (
 // to stop further processing without propagating an error back up the pipeline to logs.
 var ErrSkipProcessingData = errors.New("sentinel error to skip processing data from the remainder of the pipeline")
 
-// Start specifies the function invoked when the processor is being started.
-type Start func(context.Context, component.Host) error
-
-// Shutdown specifies the function invoked when the processor is being shutdown.
-type Shutdown func(context.Context) error
-
 // TProcessor is a helper interface that allows avoiding implementing all functions in TracesProcessor by using NewTraceProcessor.
 type TProcessor interface {
 	// ProcessTraces is a helper function that processes the incoming data and returns the data to be sent to the next component.
@@ -58,74 +53,75 @@ type LProcessor interface {
 }
 
 // Option apply changes to internalOptions.
-type Option func(*baseProcessor)
+type Option func(*baseSettings)
 
 // WithStart overrides the default Start function for an processor.
 // The default shutdown function does nothing and always returns nil.
-func WithStart(start Start) Option {
-	return func(o *baseProcessor) {
-		o.start = start
+func WithStart(start componenthelper.Start) Option {
+	return func(o *baseSettings) {
+		o.Start = start
 	}
 }
 
 // WithShutdown overrides the default Shutdown function for an processor.
 // The default shutdown function does nothing and always returns nil.
-func WithShutdown(shutdown Shutdown) Option {
-	return func(o *baseProcessor) {
-		o.shutdown = shutdown
+func WithShutdown(shutdown componenthelper.Shutdown) Option {
+	return func(o *baseSettings) {
+		o.Shutdown = shutdown
 	}
 }
 
 // WithShutdown overrides the default GetCapabilities function for an processor.
 // The default GetCapabilities function returns mutable capabilities.
 func WithCapabilities(capabilities component.ProcessorCapabilities) Option {
-	return func(o *baseProcessor) {
+	return func(o *baseSettings) {
 		o.capabilities = capabilities
 	}
 }
 
+type baseSettings struct {
+	*componenthelper.ComponentSettings
+	capabilities component.ProcessorCapabilities
+}
+
+// fromOptions returns the internal settings starting from the default and applying all options.
+func fromOptions(options []Option) *baseSettings {
+	// Start from the default options:
+	opts := &baseSettings{
+		ComponentSettings: componenthelper.DefaultComponentSettings(),
+		capabilities:      component.ProcessorCapabilities{MutatesConsumedData: true},
+	}
+
+	for _, op := range options {
+		op(opts)
+	}
+
+	return opts
+}
+
 // internalOptions contains internalOptions concerning how an Processor is configured.
 type baseProcessor struct {
+	component.Component
 	fullName     string
-	start        Start
-	shutdown     Shutdown
 	capabilities component.ProcessorCapabilities
 }
 
 // Construct the internalOptions from multiple Option.
 func newBaseProcessor(fullName string, options ...Option) baseProcessor {
+	bs := fromOptions(options)
 	be := baseProcessor{
+		Component:    componenthelper.NewComponent(bs.ComponentSettings),
 		fullName:     fullName,
-		capabilities: component.ProcessorCapabilities{MutatesConsumedData: true},
-	}
-
-	for _, op := range options {
-		op(&be)
+		capabilities: bs.capabilities,
 	}
 
 	return be
 }
 
-// Start the processor, invoked during service start.
-func (bp *baseProcessor) Start(ctx context.Context, host component.Host) error {
-	if bp.start != nil {
-		return bp.start(ctx, host)
-	}
-	return nil
-}
-
 func (bp *baseProcessor) GetCapabilities() component.ProcessorCapabilities {
 	return bp.capabilities
 }
 
-// Shutdown the processor, invoked during service shutdown.
-func (bp *baseProcessor) Shutdown(ctx context.Context) error {
-	if bp.shutdown != nil {
-		return bp.shutdown(ctx)
-	}
-	return nil
-}
-
 type tracesProcessor struct {
 	baseProcessor
 	processor    TProcessor
