diff --git a/go.mod b/go.mod
index bf2a6731c..d37817f42 100644
--- a/go.mod
+++ b/go.mod
@@ -14,7 +14,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/component v0.111.0
 	go.opentelemetry.io/collector/component/componentstatus v0.111.0
-	go.opentelemetry.io/collector/consumer v0.111.1-0.20241021235809-403c782d50c6
+	go.opentelemetry.io/collector/consumer v0.111.1-0.20241022132505-11767fed9dfe
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.111.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.111.0
 	go.opentelemetry.io/collector/pdata v1.17.0
