diff --git a/service/pipelines_test.go b/service/pipelines_test.go
index a467487a8..ebc9befca 100644
--- a/service/pipelines_test.go
+++ b/service/pipelines_test.go
@@ -182,7 +182,7 @@ func TestBuildPipelines(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			// Build the pipeline
-			pips, err := buildPipelines(context.Background(), pipelinesSettings{
+			pipelines, err := buildPipelines(context.Background(), pipelinesSettings{
 				Telemetry: componenttest.NewNopTelemetrySettings(),
 				BuildInfo: component.NewDefaultBuildInfo(),
 				ReceiverBuilder: receiver.NewBuilder(
@@ -212,9 +212,7 @@ func TestBuildPipelines(t *testing.T) {
 				PipelineConfigs: test.pipelineConfigs,
 			})
 			assert.NoError(t, err)
-			assert.IsType(t, &builtPipelines{}, pips)
 
-			pipelines := pips.(*builtPipelines)
 			assert.NoError(t, pipelines.StartAll(context.Background(), componenttest.NewNopHost()))
 
 			for dt, pipeline := range test.pipelineConfigs {
