diff --git a/exporter/exporterhelper/internal/queuebatch/config.go b/exporter/exporterhelper/internal/queuebatch/config.go
index 022845788..b30277695 100644
--- a/exporter/exporterhelper/internal/queuebatch/config.go
+++ b/exporter/exporterhelper/internal/queuebatch/config.go
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/request"
 )
 
@@ -31,6 +32,9 @@ type Config struct {
 	// If true, the component will wait for space; otherwise, operations will immediately return a retryable error.
 	BlockOnOverflow bool `mapstructure:"block_on_overflow"`
 
+	// Deprecated: [v0.123.0] use `block_on_overflow`.
+	Blocking bool `mapstructure:"blocking"`
+
 	// StorageID if not empty, enables the persistent storage and uses the component specified
 	// as a storage extension for the persistent queue.
 	// TODO: This will be changed to Optional when available.
@@ -45,6 +49,23 @@ type Config struct {
 	// BatchConfig it configures how the requests are consumed from the queue and batch together during consumption.
 	// TODO: This will be changed to Optional when available.
 	Batch *BatchConfig `mapstructure:"batch"`
+
+	// TODO: Remove when deprecated "blocking" is removed.
+	hasBlocking bool
+}
+
+func (cfg *Config) Unmarshal(conf *confmap.Conf) error {
+	if err := conf.Unmarshal(cfg); err != nil {
+		return err
+	}
+
+	// If user still uses the old blocking, override and will log error during initialization.
+	if conf.IsSet("blocking") {
+		cfg.hasBlocking = true
+		cfg.BlockOnOverflow = cfg.Blocking
+	}
+
+	return nil
 }
 
 // Validate checks if the Config is valid
@@ -61,6 +82,7 @@ func (cfg *Config) Validate() error {
 		return errors.New("`queue_size` must be positive")
 	}
 
+	// Only support request sizer for persistent queue at this moment.
 	if cfg.StorageID != nil && cfg.WaitForResult {
 		return errors.New("`wait_for_result` is not supported with a persistent queue configured with `storage`")
 	}
@@ -70,6 +92,11 @@ func (cfg *Config) Validate() error {
 		return errors.New("persistent queue configured with `storage` only supports `requests` sizer")
 	}
 
+	// Only support items sizer for batch at this moment.
+	if cfg.Batch != nil && cfg.Sizer != request.SizerTypeItems {
+		return errors.New("`batch` supports only `items` sizer")
+	}
+
 	return nil
 }
 
