diff --git a/service/internal/testcomponents/example_exporter.go b/service/internal/testcomponents/example_exporter.go
index fe91a5e2f..df3539910 100644
--- a/service/internal/testcomponents/example_exporter.go
+++ b/service/internal/testcomponents/example_exporter.go
@@ -20,6 +20,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
@@ -36,12 +37,12 @@ type ExampleExporterConfig struct {
 }
 
 // ExampleExporterFactory is factory for ExampleExporter.
-var ExampleExporterFactory = component.NewExporterFactory(
+var ExampleExporterFactory = exporter.NewFactory(
 	typeStr,
 	createExporterDefaultConfig,
-	component.WithTracesExporter(createTracesExporter, stability),
-	component.WithMetricsExporter(createMetricsExporter, stability),
-	component.WithLogsExporter(createLogsExporter, stability),
+	exporter.WithTraces(createTracesExporter, stability),
+	exporter.WithMetrics(createMetricsExporter, stability),
+	exporter.WithLogs(createLogsExporter, stability),
 )
 
 func createExporterDefaultConfig() component.Config {
@@ -50,15 +51,15 @@ func createExporterDefaultConfig() component.Config {
 	}
 }
 
-func createTracesExporter(context.Context, component.ExporterCreateSettings, component.Config) (component.TracesExporter, error) {
+func createTracesExporter(context.Context, exporter.CreateSettings, component.Config) (exporter.Traces, error) {
 	return &ExampleExporter{}, nil
 }
 
-func createMetricsExporter(context.Context, component.ExporterCreateSettings, component.Config) (component.MetricsExporter, error) {
+func createMetricsExporter(context.Context, exporter.CreateSettings, component.Config) (exporter.Metrics, error) {
 	return &ExampleExporter{}, nil
 }
 
-func createLogsExporter(context.Context, component.ExporterCreateSettings, component.Config) (component.LogsExporter, error) {
+func createLogsExporter(context.Context, exporter.CreateSettings, component.Config) (exporter.Logs, error) {
 	return &ExampleExporter{}, nil
 }
 
