diff --git a/cmd/mdatagen/main.go b/cmd/mdatagen/main.go
index 20e625d77..11411f449 100644
--- a/cmd/mdatagen/main.go
+++ b/cmd/mdatagen/main.go
@@ -96,12 +96,6 @@ func run(ymlPath string) error {
 	if err := ioutil.WriteFile(outputFile, formatted, 0600); err != nil {
 		return fmt.Errorf("failed writing %q: %v", outputFile, err)
 	}
-	// Exclude generated code from coverage and testing requirements.
-	nocoverFile := path.Join(outputDir, ".nocover")
-	if err := ioutil.WriteFile(nocoverFile, []byte("We test the generated code in the package it is used."),
-		0600); err != nil {
-		return fmt.Errorf("failed writing %v: %v", nocoverFile, err)
-	}
 
 	return nil
 }
