diff --git a/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl b/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl
index 6c778d5ee..ec47fc5a8 100644
--- a/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl
+++ b/internal/cmd/pdatagen/internal/templates/slice_test.go.tmpl
@@ -15,7 +15,7 @@ import (
 func Test{{ .structName }}(t *testing.T) {
 	es := New{{ .structName }}()
 	assert.Equal(t, 0, es.Len())
-	es = new{{ .structName }}(&[]{{ .originElementType }}{}, internal.NewState())
+	es = new{{ .structName }}(&[]{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }}{}, internal.NewState())
 	assert.Equal(t, 0, es.Len())
 
 	{{ if eq .elementName "Value" -}}
@@ -31,7 +31,7 @@ func Test{{ .structName }}(t *testing.T) {
 	for i := 0; i < 7; i++ {
 		es.AppendEmpty()
 		assert.Equal(t, emptyVal, es.At(i))
-		internal.FillOrigTest{{ .elementOriginName }}({{ .originElementPtr }}(*es.{{ .origAccessor }})[i])
+		internal.FillOrigTest{{ .elementOriginName }}({{ if not .elementNullable }}&{{ end }}(*es.{{ .origAccessor }})[i])
 		assert.Equal(t, testVal, es.At(i))
 	}
 	assert.Equal(t, 7, es.Len())
@@ -40,7 +40,7 @@ func Test{{ .structName }}(t *testing.T) {
 func Test{{ .structName }}ReadOnly(t *testing.T) {
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	es := new{{ .structName }}(&[]{{ .originElementType }}{}, sharedState)
+	es := new{{ .structName }}(&[]{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }}{}, sharedState)
 	assert.Equal(t, 0, es.Len())
 	assert.Panics(t, func() { es.AppendEmpty() })
 	assert.Panics(t, func() { es.EnsureCapacity(2) })
@@ -146,7 +146,7 @@ func Test{{ .structName }}All(t *testing.T) {
 	assert.Equal(t, ms.Len(), c, "All elements should have been visited")
 }
 
-{{ if eq .type "sliceOfPtrs" -}}
+{{ if .elementNullable -}}
 func Test{{ .structName }}_Sort(t *testing.T) {
 	es := generateTest{{ .structName }}()
 	es.Sort(func(a, b {{ .elementName }}) bool {
