diff --git a/translator/trace/zipkin/zipkinv1_thrift_to_protospan.go b/translator/trace/zipkin/zipkinv1_thrift_to_protospan.go
index fabad007b..bfbfc75c3 100644
--- a/translator/trace/zipkin/zipkinv1_thrift_to_protospan.go
+++ b/translator/trace/zipkin/zipkinv1_thrift_to_protospan.go
@@ -35,11 +35,7 @@ import (
 func V1ThriftBatchToOCProto(zSpans []*zipkincore.Span) ([]consumerdata.TraceData, error) {
 	ocSpansAndParsedAnnotations := make([]ocSpanAndParsedAnnotations, 0, len(zSpans))
 	for _, zSpan := range zSpans {
-		ocSpan, parsedAnnotations, err := zipkinV1ThriftToOCSpan(zSpan)
-		if err != nil {
-			// error from internal package function, it already wraps the error to give better context.
-			return nil, err
-		}
+		ocSpan, parsedAnnotations := zipkinV1ThriftToOCSpan(zSpan)
 		ocSpansAndParsedAnnotations = append(ocSpansAndParsedAnnotations, ocSpanAndParsedAnnotations{
 			ocSpan:            ocSpan,
 			parsedAnnotations: parsedAnnotations,
@@ -49,7 +45,7 @@ func V1ThriftBatchToOCProto(zSpans []*zipkincore.Span) ([]consumerdata.TraceData
 	return zipkinToOCProtoBatch(ocSpansAndParsedAnnotations)
 }
 
-func zipkinV1ThriftToOCSpan(zSpan *zipkincore.Span) (*tracepb.Span, *annotationParseResult, error) {
+func zipkinV1ThriftToOCSpan(zSpan *zipkincore.Span) (*tracepb.Span, *annotationParseResult) {
 	traceIDHigh := int64(0)
 	if zSpan.TraceIDHigh != nil {
 		traceIDHigh = *zSpan.TraceIDHigh
@@ -104,7 +100,7 @@ func zipkinV1ThriftToOCSpan(zSpan *zipkincore.Span) (*tracepb.Span, *annotationP
 
 	setSpanKind(ocSpan, parsedAnnotations.Kind, parsedAnnotations.ExtendedKind)
 
-	return ocSpan, parsedAnnotations, nil
+	return ocSpan, parsedAnnotations
 }
 
 func parseZipkinV1ThriftAnnotations(ztAnnotations []*zipkincore.Annotation) *annotationParseResult {
