diff --git a/receiver/jaegerreceiver/config.go b/receiver/jaegerreceiver/config.go
index 30de0fe0b..c5efcfe70 100644
--- a/receiver/jaegerreceiver/config.go
+++ b/receiver/jaegerreceiver/config.go
@@ -23,6 +23,7 @@ const protocolsFieldName = "protocols"
 
 // RemoteSamplingConfig defines config key for remote sampling fetch endpoint
 type RemoteSamplingConfig struct {
+	HostEndpoint  string `mapstructure:"host_endpoint"`
 	FetchEndpoint string `mapstructure:"fetch_endpoint"`
 }
 
