diff --git a/receiver/jaegerreceiver/factory_test.go b/receiver/jaegerreceiver/factory_test.go
index 58bbf58b5..dc4958783 100644
--- a/receiver/jaegerreceiver/factory_test.go
+++ b/receiver/jaegerreceiver/factory_test.go
@@ -16,6 +16,7 @@ package jaegerreceiver
 
 import (
 	"context"
+	"fmt"
 	"testing"
 
 	"github.com/spf13/viper"
@@ -117,6 +118,23 @@ func TestCreateInvalidThriftCompactEndpoint(t *testing.T) {
 	assert.Equal(t, 6831, r.(*jReceiver).config.AgentCompactThriftPort, "thrift port should be default")
 }
 
+func TestDefaultAgentRemoteSamplingHTTPPort(t *testing.T) {
+	factory := Factory{}
+	cfg := factory.CreateDefaultConfig()
+	rCfg := cfg.(*Config)
+
+	endpoint := "localhost:1234"
+	rCfg.Protocols[protoThriftCompact], _ = defaultsForProtocol(protoThriftCompact)
+	rCfg.RemoteSampling = &RemoteSamplingConfig{
+		FetchEndpoint: endpoint,
+	}
+	r, err := factory.CreateTraceReceiver(context.Background(), zap.NewNop(), cfg, nil)
+
+	assert.NoError(t, err, "create trace receiver should not error")
+	assert.Equal(t, endpoint, r.(*jReceiver).config.RemoteSamplingEndpoint)
+	assert.Equal(t, defaultAgentRemoteSamplingHTTPPort, r.(*jReceiver).config.AgentHTTPPort, "agent http port should be default")
+}
+
 func TestCreateNoPort(t *testing.T) {
 	factory := Factory{}
 	cfg := factory.CreateDefaultConfig()
@@ -205,15 +223,18 @@ func TestRemoteSamplingConfigPropagation(t *testing.T) {
 	cfg := factory.CreateDefaultConfig()
 	rCfg := cfg.(*Config)
 
+	hostPort := 5778
 	endpoint := "localhost:1234"
 	rCfg.Protocols[protoThriftCompact], _ = defaultsForProtocol(protoThriftCompact)
 	rCfg.RemoteSampling = &RemoteSamplingConfig{
 		FetchEndpoint: endpoint,
+		HostEndpoint:  fmt.Sprintf("localhost:%d", hostPort),
 	}
 	r, err := factory.CreateTraceReceiver(context.Background(), zap.NewNop(), cfg, nil)
 
 	assert.NoError(t, err, "create trace receiver should not error")
 	assert.Equal(t, endpoint, r.(*jReceiver).config.RemoteSamplingEndpoint)
+	assert.Equal(t, hostPort, r.(*jReceiver).config.AgentHTTPPort, "agent http port should be configured value")
 }
 
 func TestCustomUnmarshalErrors(t *testing.T) {
