diff --git a/cmd/mdatagen/go.mod b/cmd/mdatagen/go.mod
index cdfce449e..46d806204 100644
--- a/cmd/mdatagen/go.mod
+++ b/cmd/mdatagen/go.mod
@@ -5,7 +5,7 @@ go 1.22.0
 require (
 	github.com/google/go-cmp v0.6.0
 	github.com/spf13/cobra v1.8.1
-	github.com/stretchr/testify v1.9.0
+	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v0.114.0
 	go.opentelemetry.io/collector/component/componenttest v0.114.0
 	go.opentelemetry.io/collector/config/configtelemetry v0.114.0
