diff --git a/Makefile.Common b/Makefile.Common
index de10f4d8a..70a630be9 100644
--- a/Makefile.Common
+++ b/Makefile.Common
@@ -38,7 +38,7 @@ SEMCONVGEN   := $(TOOLS_BIN_DIR)/semconvgen
 SEMCONVKIT   := $(TOOLS_BIN_DIR)/semconvkit
 TESTIFYLINT  := $(TOOLS_BIN_DIR)/testifylint
 
-TESTIFYLINT_OPT?= --enable-all --disable=error-is-as,expected-actual,float-compare,formatter,go-require,require-error
+TESTIFYLINT_OPT?= --enable-all --disable=error-is-as,float-compare,formatter,go-require,require-error
 
 .PHONY: install-tools
 install-tools: $(TOOLS_BIN_NAMES)
