diff --git a/processor/processortest/unhealthy_processor.go b/processor/processortest/unhealthy_processor.go
index 196787de9..c537ab534 100644
--- a/processor/processortest/unhealthy_processor.go
+++ b/processor/processortest/unhealthy_processor.go
@@ -14,8 +14,8 @@ import (
 )
 
 // NewUnhealthyProcessorCreateSettings returns a new nop settings for Create*Processor functions.
-func NewUnhealthyProcessorCreateSettings() processor.CreateSettings {
-	return processor.CreateSettings{
+func NewUnhealthyProcessorCreateSettings() processor.Settings {
+	return processor.Settings{
 		TelemetrySettings: componenttest.NewNopTelemetrySettings(),
 		BuildInfo:         component.NewDefaultBuildInfo(),
 	}
@@ -34,21 +34,21 @@ func NewUnhealthyProcessorFactory() processor.Factory {
 	)
 }
 
-func createUnhealthyTracesProcessor(_ context.Context, set processor.CreateSettings, _ component.Config, _ consumer.Traces) (processor.Traces, error) {
+func createUnhealthyTracesProcessor(_ context.Context, set processor.Settings, _ component.Config, _ consumer.Traces) (processor.Traces, error) {
 	return &unhealthyProcessor{
 		Consumer:  consumertest.NewNop(),
 		telemetry: set.TelemetrySettings,
 	}, nil
 }
 
-func createUnhealthyMetricsProcessor(_ context.Context, set processor.CreateSettings, _ component.Config, _ consumer.Metrics) (processor.Metrics, error) {
+func createUnhealthyMetricsProcessor(_ context.Context, set processor.Settings, _ component.Config, _ consumer.Metrics) (processor.Metrics, error) {
 	return &unhealthyProcessor{
 		Consumer:  consumertest.NewNop(),
 		telemetry: set.TelemetrySettings,
 	}, nil
 }
 
-func createUnhealthyLogsProcessor(_ context.Context, set processor.CreateSettings, _ component.Config, _ consumer.Logs) (processor.Logs, error) {
+func createUnhealthyLogsProcessor(_ context.Context, set processor.Settings, _ component.Config, _ consumer.Logs) (processor.Logs, error) {
 	return &unhealthyProcessor{
 		Consumer:  consumertest.NewNop(),
 		telemetry: set.TelemetrySettings,
