diff --git a/exporter/exporterhelper/queue_sender_test.go b/exporter/exporterhelper/queue_sender_test.go
index 322282a69..928fbe1e6 100644
--- a/exporter/exporterhelper/queue_sender_test.go
+++ b/exporter/exporterhelper/queue_sender_test.go
@@ -173,6 +173,11 @@ func TestQueueSettings_Validate(t *testing.T) {
 	qCfg.QueueSize = 0
 	assert.EqualError(t, qCfg.Validate(), "queue size must be positive")
 
+	qCfg = NewDefaultQueueSettings()
+	qCfg.NumConsumers = 0
+
+	assert.EqualError(t, qCfg.Validate(), "number of queue consumers must be positive")
+
 	// Confirm Validate doesn't return error with invalid config when feature is disabled
 	qCfg.Enabled = false
 	assert.NoError(t, qCfg.Validate())
