diff --git a/translator/trace/zipkin/zipkinv2_to_traces_test.go b/translator/trace/zipkin/zipkinv2_to_traces_test.go
index 1df2e2abb..e1efe0c21 100644
--- a/translator/trace/zipkin/zipkinv2_to_traces_test.go
+++ b/translator/trace/zipkin/zipkinv2_to_traces_test.go
@@ -22,7 +22,6 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/collector/internal/testdata"
 	"go.opentelemetry.io/collector/translator/conventions"
 )
 
@@ -36,13 +35,7 @@ func TestZipkinSpansToInternalTraces(t *testing.T) {
 		{
 			name: "empty",
 			zs:   make([]*zipkinmodel.SpanModel, 0),
-			td:   testdata.GenerateTraceDataEmpty(),
-			err:  nil,
-		},
-		{
-			name: "nilSpan",
-			zs:   generateNilSpan(),
-			td:   testdata.GenerateTraceDataEmpty(),
+			td:   pdata.NewTraces(),
 			err:  nil,
 		},
 		{
@@ -76,10 +69,6 @@ func TestZipkinSpansToInternalTraces(t *testing.T) {
 	}
 }
 
-func generateNilSpan() []*zipkinmodel.SpanModel {
-	return make([]*zipkinmodel.SpanModel, 1)
-}
-
 func generateSpanNoEndpoints() []*zipkinmodel.SpanModel {
 	spans := make([]*zipkinmodel.SpanModel, 1)
 	spans[0] = &zipkinmodel.SpanModel{
