diff --git a/exporter/otlphttpexporter/otlp_test.go b/exporter/otlphttpexporter/otlp_test.go
index a19298c77..95ca2ce06 100644
--- a/exporter/otlphttpexporter/otlp_test.go
+++ b/exporter/otlphttpexporter/otlp_test.go
@@ -251,7 +251,7 @@ func TestUserAgent(t *testing.T) {
 
 				cfg := &Config{
 					TracesEndpoint: fmt.Sprintf("%s/v1/traces", srv.URL),
-					HTTPClientConfig: confighttp.HTTPClientConfig{
+					ClientConfig: confighttp.ClientConfig{
 						Headers: test.headers,
 					},
 				}
@@ -284,7 +284,7 @@ func TestUserAgent(t *testing.T) {
 
 				cfg := &Config{
 					MetricsEndpoint: fmt.Sprintf("%s/v1/metrics", srv.URL),
-					HTTPClientConfig: confighttp.HTTPClientConfig{
+					ClientConfig: confighttp.ClientConfig{
 						Headers: test.headers,
 					},
 				}
@@ -317,7 +317,7 @@ func TestUserAgent(t *testing.T) {
 
 				cfg := &Config{
 					LogsEndpoint: fmt.Sprintf("%s/v1/logs", srv.URL),
-					HTTPClientConfig: confighttp.HTTPClientConfig{
+					ClientConfig: confighttp.ClientConfig{
 						Headers: test.headers,
 					},
 				}
@@ -423,8 +423,8 @@ func TestPartialSuccess_logs(t *testing.T) {
 	defer srv.Close()
 
 	cfg := &Config{
-		LogsEndpoint:     fmt.Sprintf("%s/v1/logs", srv.URL),
-		HTTPClientConfig: confighttp.HTTPClientConfig{},
+		LogsEndpoint: fmt.Sprintf("%s/v1/logs", srv.URL),
+		ClientConfig: confighttp.ClientConfig{},
 	}
 	exp, err := createLogsExporter(context.Background(), exportertest.NewNopCreateSettings(), cfg)
 	require.NoError(t, err)
@@ -547,8 +547,8 @@ func TestPartialSuccess_traces(t *testing.T) {
 	defer srv.Close()
 
 	cfg := &Config{
-		TracesEndpoint:   fmt.Sprintf("%s/v1/traces", srv.URL),
-		HTTPClientConfig: confighttp.HTTPClientConfig{},
+		TracesEndpoint: fmt.Sprintf("%s/v1/traces", srv.URL),
+		ClientConfig:   confighttp.ClientConfig{},
 	}
 	exp, err := createTracesExporter(context.Background(), exportertest.NewNopCreateSettings(), cfg)
 	require.NoError(t, err)
@@ -581,8 +581,8 @@ func TestPartialSuccess_metrics(t *testing.T) {
 	defer srv.Close()
 
 	cfg := &Config{
-		MetricsEndpoint:  fmt.Sprintf("%s/v1/metrics", srv.URL),
-		HTTPClientConfig: confighttp.HTTPClientConfig{},
+		MetricsEndpoint: fmt.Sprintf("%s/v1/metrics", srv.URL),
+		ClientConfig:    confighttp.ClientConfig{},
 	}
 	exp, err := createMetricsExporter(context.Background(), exportertest.NewNopCreateSettings(), cfg)
 	require.NoError(t, err)
