diff --git a/pdata/pcommon/value.go b/pdata/pcommon/value.go
index 64ac3f258..ab82d7fde 100644
--- a/pdata/pcommon/value.go
+++ b/pdata/pcommon/value.go
@@ -76,37 +76,64 @@ func NewValueEmpty() Value {
 
 // NewValueStr creates a new Value with the given string value.
 func NewValueStr(v string) Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: v}}, internal.NewState())
+	ov := internal.NewOrigAnyValueStringValue()
+	ov.StringValue = v
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 // NewValueInt creates a new Value with the given int64 value.
 func NewValueInt(v int64) Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_IntValue{IntValue: v}}, internal.NewState())
+	ov := internal.NewOrigAnyValueIntValue()
+	ov.IntValue = v
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 // NewValueDouble creates a new Value with the given float64 value.
 func NewValueDouble(v float64) Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_DoubleValue{DoubleValue: v}}, internal.NewState())
+	ov := internal.NewOrigAnyValueDoubleValue()
+	ov.DoubleValue = v
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 // NewValueBool creates a new Value with the given bool value.
 func NewValueBool(v bool) Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_BoolValue{BoolValue: v}}, internal.NewState())
+	ov := internal.NewOrigAnyValueBoolValue()
+	ov.BoolValue = v
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 // NewValueMap creates a new Value of map type.
 func NewValueMap() Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_KvlistValue{KvlistValue: &otlpcommon.KeyValueList{}}}, internal.NewState())
+	ov := internal.NewOrigAnyValueKeyValueList()
+	ov.KvlistValue = internal.NewOrigKeyValueList()
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 // NewValueSlice creates a new Value of array type.
 func NewValueSlice() Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_ArrayValue{ArrayValue: &otlpcommon.ArrayValue{}}}, internal.NewState())
+	ov := internal.NewOrigAnyValueArrayValue()
+	ov.ArrayValue = internal.NewOrigArrayValue()
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 // NewValueBytes creates a new empty Value of byte type.
 func NewValueBytes() Value {
-	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_BytesValue{BytesValue: nil}}, internal.NewState())
+	ov := internal.NewOrigAnyValueBytesValue()
+	orig := internal.NewOrigAnyValue()
+	orig.Value = ov
+	return newValue(orig, internal.NewState())
 }
 
 func newValue(orig *otlpcommon.AnyValue, state *internal.State) Value {
@@ -256,7 +283,11 @@ func (v Value) Bytes() ByteSlice {
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetStr(sv string) {
 	v.getState().AssertMutable()
-	v.getOrig().Value = &otlpcommon.AnyValue_StringValue{StringValue: sv}
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	ov := internal.NewOrigAnyValueStringValue()
+	ov.StringValue = sv
+	v.getOrig().Value = ov
 }
 
 // SetInt replaces the int64 value associated with this Value,
@@ -264,7 +295,11 @@ func (v Value) SetStr(sv string) {
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetInt(iv int64) {
 	v.getState().AssertMutable()
-	v.getOrig().Value = &otlpcommon.AnyValue_IntValue{IntValue: iv}
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	ov := internal.NewOrigAnyValueIntValue()
+	ov.IntValue = iv
+	v.getOrig().Value = ov
 }
 
 // SetDouble replaces the float64 value associated with this Value,
@@ -272,7 +307,11 @@ func (v Value) SetInt(iv int64) {
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetDouble(dv float64) {
 	v.getState().AssertMutable()
-	v.getOrig().Value = &otlpcommon.AnyValue_DoubleValue{DoubleValue: dv}
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	ov := internal.NewOrigAnyValueDoubleValue()
+	ov.DoubleValue = dv
+	v.getOrig().Value = ov
 }
 
 // SetBool replaces the bool value associated with this Value,
@@ -280,15 +319,21 @@ func (v Value) SetDouble(dv float64) {
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetBool(bv bool) {
 	v.getState().AssertMutable()
-	v.getOrig().Value = &otlpcommon.AnyValue_BoolValue{BoolValue: bv}
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	ov := internal.NewOrigAnyValueBoolValue()
+	ov.BoolValue = bv
+	v.getOrig().Value = ov
 }
 
 // SetEmptyBytes sets value to an empty byte slice and returns it.
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetEmptyBytes() ByteSlice {
 	v.getState().AssertMutable()
-	bv := otlpcommon.AnyValue_BytesValue{BytesValue: nil}
-	v.getOrig().Value = &bv
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	bv := internal.NewOrigAnyValueBytesValue()
+	v.getOrig().Value = bv
 	return ByteSlice(internal.NewByteSlice(&bv.BytesValue, v.getState()))
 }
 
@@ -296,18 +341,24 @@ func (v Value) SetEmptyBytes() ByteSlice {
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetEmptyMap() Map {
 	v.getState().AssertMutable()
-	kv := &otlpcommon.AnyValue_KvlistValue{KvlistValue: &otlpcommon.KeyValueList{}}
-	v.getOrig().Value = kv
-	return newMap(&kv.KvlistValue.Values, v.getState())
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	ov := internal.NewOrigAnyValueKeyValueList()
+	ov.KvlistValue = internal.NewOrigKeyValueList()
+	v.getOrig().Value = ov
+	return newMap(&ov.KvlistValue.Values, v.getState())
 }
 
 // SetEmptySlice sets value to an empty slice and returns it.
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) SetEmptySlice() Slice {
 	v.getState().AssertMutable()
-	av := &otlpcommon.AnyValue_ArrayValue{ArrayValue: &otlpcommon.ArrayValue{}}
-	v.getOrig().Value = av
-	return newSlice(&av.ArrayValue.Values, v.getState())
+	// Delete everything but the AnyValue object itself.
+	internal.DeleteOrigAnyValue(v.getOrig(), false)
+	ov := internal.NewOrigAnyValueArrayValue()
+	ov.ArrayValue = internal.NewOrigArrayValue()
+	v.getOrig().Value = ov
+	return newSlice(&ov.ArrayValue.Values, v.getState())
 }
 
 // MoveTo moves the Value from current overriding the destination and
