diff --git a/pdata/pprofile/generated_profilesslice_test.go b/pdata/pprofile/generated_profilesslice_test.go
index b188c6181..456ad98ae 100644
--- a/pdata/pprofile/generated_profilesslice_test.go
+++ b/pdata/pprofile/generated_profilesslice_test.go
@@ -123,6 +123,18 @@ func TestProfilesSlice_RemoveIf(t *testing.T) {
 	assert.Equal(t, 5, filtered.Len())
 }
 
+func TestProfilesSliceAll(t *testing.T) {
+	ms := generateTestProfilesSlice()
+	assert.NotEmpty(t, ms.Len())
+
+	var c int
+	for i, v := range ms.All() {
+		assert.Equal(t, ms.At(i), v, "element should match")
+		c++
+	}
+	assert.Equal(t, ms.Len(), c, "All elements should have been visited")
+}
+
 func TestProfilesSlice_Sort(t *testing.T) {
 	es := generateTestProfilesSlice()
 	es.Sort(func(a, b Profile) bool {
