diff --git a/pdata/pprofile/generated_valuetypeslice_test.go b/pdata/pprofile/generated_valuetypeslice_test.go
index f41809cc7..821d33843 100644
--- a/pdata/pprofile/generated_valuetypeslice_test.go
+++ b/pdata/pprofile/generated_valuetypeslice_test.go
@@ -123,6 +123,18 @@ func TestValueTypeSlice_RemoveIf(t *testing.T) {
 	assert.Equal(t, 5, filtered.Len())
 }
 
+func TestValueTypeSliceAll(t *testing.T) {
+	ms := generateTestValueTypeSlice()
+	assert.NotEmpty(t, ms.Len())
+
+	var c int
+	for i, v := range ms.All() {
+		assert.Equal(t, ms.At(i), v, "element should match")
+		c++
+	}
+	assert.Equal(t, ms.Len(), c, "All elements should have been visited")
+}
+
 func TestValueTypeSlice_Sort(t *testing.T) {
 	es := generateTestValueTypeSlice()
 	es.Sort(func(a, b ValueType) bool {
