diff --git a/pdata/pprofile/generated_locationslice_test.go b/pdata/pprofile/generated_locationslice_test.go
index cdf2ebeba..e13dba14e 100644
--- a/pdata/pprofile/generated_locationslice_test.go
+++ b/pdata/pprofile/generated_locationslice_test.go
@@ -123,6 +123,18 @@ func TestLocationSlice_RemoveIf(t *testing.T) {
 	assert.Equal(t, 5, filtered.Len())
 }
 
+func TestLocationSliceAll(t *testing.T) {
+	ms := generateTestLocationSlice()
+	assert.NotEmpty(t, ms.Len())
+
+	var c int
+	for i, v := range ms.All() {
+		assert.Equal(t, ms.At(i), v, "element should match")
+		c++
+	}
+	assert.Equal(t, ms.Len(), c, "All elements should have been visited")
+}
+
 func TestLocationSlice_Sort(t *testing.T) {
 	es := generateTestLocationSlice()
 	es.Sort(func(a, b Location) bool {
