diff --git a/receiver/receiver.go b/receiver/receiver.go
index 368768989..785ef38c1 100644
--- a/receiver/receiver.go
+++ b/receiver/receiver.go
@@ -18,7 +18,6 @@ import (
 	"context"
 
 	"github.com/census-instrumentation/opencensus-service/data"
-	"github.com/census-instrumentation/opencensus-service/internal"
 	_ "github.com/census-instrumentation/opencensus-service/internal/compression/grpc" // load in supported grpc compression encodings
 )
 
@@ -74,65 +73,3 @@ type MetricsReceiverAcknowledgement struct {
 	SavedMetrics   uint64
 	DroppedMetrics uint64
 }
-
-// MultiTraceReceiver wraps multiple trace receivers in a single one.
-func MultiTraceReceiver(trs ...TraceReceiver) TraceReceiver {
-	return traceReceivers(trs)
-}
-
-type traceReceivers []TraceReceiver
-
-func (trs traceReceivers) StartTraceReception(ctx context.Context, destination TraceReceiverSink) error {
-	var errs []error
-	for _, tr := range trs {
-		err := tr.StartTraceReception(ctx, destination)
-		if err != nil {
-			errs = append(errs, err)
-		}
-	}
-
-	return internal.CombineErrors(errs)
-}
-
-func (trs traceReceivers) StopTraceReception(ctx context.Context) error {
-	var errs []error
-	for _, tr := range trs {
-		err := tr.StopTraceReception(ctx)
-		if err != nil {
-			errs = append(errs, err)
-		}
-	}
-
-	return internal.CombineErrors(errs)
-}
-
-// MultiMetricsReceiver wraps multiple metrics receivers in a single one.
-func MultiMetricsReceiver(mrs ...MetricsReceiver) MetricsReceiver {
-	return metricsReceivers(mrs)
-}
-
-type metricsReceivers []MetricsReceiver
-
-func (mrs metricsReceivers) StartMetricsReception(ctx context.Context, destination MetricsReceiverSink) error {
-	var errs []error
-	for _, mr := range mrs {
-		err := mr.StartMetricsReception(ctx, destination)
-		if err != nil {
-			errs = append(errs, err)
-		}
-	}
-
-	return internal.CombineErrors(errs)
-}
-
-func (mrs metricsReceivers) StopMetricsReception(ctx context.Context) error {
-	var errs []error
-	for _, mr := range mrs {
-		err := mr.StopMetricsReception(ctx)
-		if err != nil {
-			errs = append(errs, err)
-		}
-	}
-
-	return internal.CombineErrors(errs)
-}
