diff --git a/pdata/pcommon/timestamp_test.go b/pdata/pcommon/timestamp_test.go
index edd94596e..3fa5ffbc5 100644
--- a/pdata/pcommon/timestamp_test.go
+++ b/pdata/pcommon/timestamp_test.go
@@ -11,7 +11,7 @@ import (
 )
 
 func TestUnixNanosConverters(t *testing.T) {
-	t1 := time.Date(2020, 03, 24, 1, 13, 23, 789, time.UTC)
+	t1 := time.Date(2020, 3, 24, 1, 13, 23, 789, time.UTC)
 	// nolint:gosec
 	tun := Timestamp(t1.UnixNano())
 
