diff --git a/interceptor/opencensus/options.go b/interceptor/opencensus/options.go
index f7ce54d60..15bf930ce 100644
--- a/interceptor/opencensus/options.go
+++ b/interceptor/opencensus/options.go
@@ -16,7 +16,7 @@ package ocinterceptor
 
 import "time"
 
-type OCOption interface {
+type Option interface {
 	WithInterceptor(*Interceptor)
 }
 
@@ -24,7 +24,7 @@ type spanBufferPeriod struct {
 	period time.Duration
 }
 
-var _ OCOption = (*spanBufferPeriod)(nil)
+var _ Option = (*spanBufferPeriod)(nil)
 
 func (sfd *spanBufferPeriod) WithInterceptor(oci *Interceptor) {
 	oci.spanBufferPeriod = sfd.period
@@ -33,13 +33,13 @@ func (sfd *spanBufferPeriod) WithInterceptor(oci *Interceptor) {
 // WithSpanBufferPeriod is an option that allows one to configure
 // the period that spans are buffered for before the Interceptor
 // sends them to its SpanReceiver.
-func WithSpanBufferPeriod(period time.Duration) OCOption {
+func WithSpanBufferPeriod(period time.Duration) Option {
 	return &spanBufferPeriod{period: period}
 }
 
 type spanBufferCount int
 
-var _ OCOption = (*spanBufferCount)(nil)
+var _ Option = (*spanBufferCount)(nil)
 
 func (spc spanBufferCount) WithInterceptor(oci *Interceptor) {
 	oci.spanBufferCount = int(spc)
@@ -48,6 +48,6 @@ func (spc spanBufferCount) WithInterceptor(oci *Interceptor) {
 // WithSpanBufferCount is an option that allows one to configure
 // the number of spans that are buffered before the Interceptor
 // send them to its SpanReceiver.
-func WithSpanBufferCount(count int) OCOption {
+func WithSpanBufferCount(count int) Option {
 	return spanBufferCount(count)
 }
