diff --git a/translator/trace/protospan_translation.go b/translator/trace/protospan_translation.go
index 83f0196f9..5d11a60a8 100644
--- a/translator/trace/protospan_translation.go
+++ b/translator/trace/protospan_translation.go
@@ -29,13 +29,10 @@ const (
 
 	TagSpanKind = "span.kind"
 
-	TagStatusCode          = "status.code"
-	TagStatusMsg           = "status.message"
-	TagError               = "error"
-	TagHTTPStatusMsg       = "http.status_message"
-	TagZipkinCensusCode    = "census.status_code"
-	TagZipkinCensusMsg     = "census.status_description"
-	TagZipkinOpenCensusMsg = "opencensus.status_description"
+	TagStatusCode    = "status.code"
+	TagStatusMsg     = "status.message"
+	TagError         = "error"
+	TagHTTPStatusMsg = "http.status_message"
 
 	TagW3CTraceState     = "w3c.tracestate"
 	TagServiceNameSource = "otlp.service.name.source"
@@ -62,11 +59,6 @@ const (
 	OpenTracingSpanKindInternal    OpenTracingSpanKind = "internal"
 )
 
-const (
-	SpanLinkDataFormat  = "%s|%s|%s|%s|%d"
-	SpanEventDataFormat = "%s|%s|%d"
-)
-
 type attrValDescript struct {
 	regex    *regexp.Regexp
 	attrType pdata.AttributeValueType
