diff --git a/config/exporter.go b/config/exporter.go
index f56d55e42..1b3176fb6 100644
--- a/config/exporter.go
+++ b/config/exporter.go
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 package config // import "go.opentelemetry.io/collector/config"
+
 import (
 	"go.opentelemetry.io/collector/component"
 )
@@ -24,23 +25,12 @@ import (
 //
 // When embedded in the exporter config, it must be with `mapstructure:",squash"` tag.
 type ExporterSettings struct {
-	id component.ID `mapstructure:"-"`
-	component.ExporterConfig
+	settings
 }
 
 // NewExporterSettings return a new ExporterSettings with the given ComponentID.
 func NewExporterSettings(id component.ID) ExporterSettings {
-	return ExporterSettings{id: id}
+	return ExporterSettings{settings: newSettings(id)}
 }
 
 var _ component.ExporterConfig = (*ExporterSettings)(nil)
-
-// ID returns the receiver component.ID.
-func (es *ExporterSettings) ID() component.ID {
-	return es.id
-}
-
-// SetIDName sets the receiver name.
-func (es *ExporterSettings) SetIDName(idName string) {
-	es.id = component.NewIDWithName(es.id.Type(), idName)
-}
