diff --git a/config/mapprovider/filemapprovider/mapprovider.go b/config/mapprovider/filemapprovider/mapprovider.go
index 0c22e6eea..51c03f4aa 100644
--- a/config/mapprovider/filemapprovider/mapprovider.go
+++ b/config/mapprovider/filemapprovider/mapprovider.go
@@ -63,7 +63,7 @@ func (fmp *mapProvider) Retrieve(_ context.Context, uri string, _ config.Watcher
 		return config.Retrieved{}, fmt.Errorf("unable to parse yaml: %w", err)
 	}
 
-	return config.Retrieved{Map: config.NewMapFromStringMap(data)}, nil
+	return config.NewRetrievedFromMap(config.NewMapFromStringMap(data)), nil
 }
 
 func (*mapProvider) Scheme() string {
