diff --git a/exporter/exporterhelper/internal/queuebatch/config.go b/exporter/exporterhelper/internal/queuebatch/config.go
index 2c4837f31..7caa75e08 100644
--- a/exporter/exporterhelper/internal/queuebatch/config.go
+++ b/exporter/exporterhelper/internal/queuebatch/config.go
@@ -9,6 +9,7 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configoptional"
+	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/request"
 )
 
@@ -46,6 +47,18 @@ type Config struct {
 	Batch configoptional.Optional[BatchConfig] `mapstructure:"batch"`
 }
 
+func (cfg *Config) Unmarshal(conf *confmap.Conf) error {
+	if err := conf.Unmarshal(cfg); err != nil {
+		return err
+	}
+
+	// If the batch sizer is not set, use the same value as the queue sizer.
+	if !conf.IsSet("batch::sizer") && cfg.Batch.HasValue() {
+		cfg.Batch.Get().Sizer = cfg.Sizer
+	}
+	return nil
+}
+
 // Validate checks if the Config is valid
 func (cfg *Config) Validate() error {
 	if !cfg.Enabled {
@@ -65,12 +78,7 @@ func (cfg *Config) Validate() error {
 		return errors.New("`wait_for_result` is not supported with a persistent queue configured with `storage`")
 	}
 
-	if cfg.Batch.HasValue() {
-		// Only support items or bytes sizer for batch at this moment.
-		if cfg.Sizer != request.SizerTypeItems && cfg.Sizer != request.SizerTypeBytes {
-			return errors.New("`batch` supports only `items` or `bytes` sizer")
-		}
-
+	if cfg.Batch.HasValue() && cfg.Batch.Get().Sizer == cfg.Sizer {
 		// Avoid situations where the queue is not able to hold any data.
 		if cfg.Batch.Get().MinSize > cfg.QueueSize {
 			return errors.New("`min_size` must be less than or equal to `queue_size`")
@@ -85,6 +93,11 @@ type BatchConfig struct {
 	// FlushTimeout sets the time after which a batch will be sent regardless of its size.
 	FlushTimeout time.Duration `mapstructure:"flush_timeout"`
 
+	// Sizer determines the type of size measurement used by the batch.
+	// If not configured, use the same configuration as the queue.
+	// It accepts "requests", "items", or "bytes".
+	Sizer request.SizerType `mapstructure:"sizer"`
+
 	// MinSize defines the configuration for the minimum size of a batch.
 	MinSize int64 `mapstructure:"min_size"`
 
@@ -97,6 +110,11 @@ func (cfg *BatchConfig) Validate() error {
 		return nil
 	}
 
+	// Only support items or bytes sizer for batch at this moment.
+	if cfg.Sizer != request.SizerTypeItems && cfg.Sizer != request.SizerTypeBytes {
+		return errors.New("`batch` supports only `items` or `bytes` sizer")
+	}
+
 	if cfg.FlushTimeout <= 0 {
 		return errors.New("`flush_timeout` must be positive")
 	}
