diff --git a/exporter/exportertest/nop_exporter_test.go b/exporter/exportertest/nop_exporter_test.go
index 456d7c89c..93f5ea6f9 100644
--- a/exporter/exportertest/nop_exporter_test.go
+++ b/exporter/exportertest/nop_exporter_test.go
@@ -49,31 +49,3 @@ func TestNewNopFactory(t *testing.T) {
 	assert.NoError(t, profiles.ConsumeProfiles(context.Background(), pprofile.NewProfiles()))
 	assert.NoError(t, profiles.Shutdown(context.Background()))
 }
-
-func TestNewNopBuilder(t *testing.T) {
-	builder := NewNopBuilder()
-	require.NotNil(t, builder)
-
-	factory := NewNopFactory()
-	cfg := factory.CreateDefaultConfig()
-	set := NewNopSettings()
-	set.ID = component.NewID(nopType)
-
-	traces, err := factory.CreateTracesExporter(context.Background(), set, cfg)
-	require.NoError(t, err)
-	bTraces, err := builder.CreateTraces(context.Background(), set)
-	require.NoError(t, err)
-	assert.IsType(t, traces, bTraces)
-
-	metrics, err := factory.CreateMetricsExporter(context.Background(), set, cfg)
-	require.NoError(t, err)
-	bMetrics, err := builder.CreateMetrics(context.Background(), set)
-	require.NoError(t, err)
-	assert.IsType(t, metrics, bMetrics)
-
-	logs, err := factory.CreateLogsExporter(context.Background(), set, cfg)
-	require.NoError(t, err)
-	bLogs, err := builder.CreateLogs(context.Background(), set)
-	require.NoError(t, err)
-	assert.IsType(t, logs, bLogs)
-}
