diff --git a/service/builder/receivers_builder_test.go b/service/builder/receivers_builder_test.go
index 2ab5a29c0..70998080b 100644
--- a/service/builder/receivers_builder_test.go
+++ b/service/builder/receivers_builder_test.go
@@ -29,7 +29,6 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/consumer/pdatautil"
 	"go.opentelemetry.io/collector/internal/data/testdata"
-	"go.opentelemetry.io/collector/internal/dataold/testdataold"
 	"go.opentelemetry.io/collector/processor/attributesprocessor"
 	"go.opentelemetry.io/collector/receiver/receiverhelper"
 )
@@ -140,7 +139,7 @@ func testReceivers(
 		traceProducer.TraceConsumer.ConsumeTraces(context.Background(), testdata.GenerateTraceDataOneSpan())
 	}
 
-	metrics := pdatautil.MetricsFromOldInternalMetrics(testdataold.GenerateMetricDataOneMetric())
+	metrics := pdatautil.MetricsFromInternalMetrics(testdata.GenerateMetricsOneMetric())
 	if test.hasMetrics {
 		metricsProducer := receiver.receiver.(*componenttest.ExampleReceiverProducer)
 		metricsProducer.MetricsConsumer.ConsumeMetrics(context.Background(), metrics)
