diff --git a/exporter/exporterhelper/traces_test.go b/exporter/exporterhelper/traces_test.go
index 2d3d1cfdf..d6feaedc1 100644
--- a/exporter/exporterhelper/traces_test.go
+++ b/exporter/exporterhelper/traces_test.go
@@ -61,7 +61,7 @@ func TestTracesExporter_NilLogger(t *testing.T) {
 }
 
 func TestTracesRequestExporter_NilLogger(t *testing.T) {
-	te, err := NewTracesRequestExporter(context.Background(), exporter.Settings{}, (&fakeRequestConverter{}).requestFromTracesFunc)
+	te, err := NewTracesRequestExporter(context.Background(), exporter.Settings{}, (&internal.FakeRequestConverter{}).RequestFromTracesFunc)
 	require.Nil(t, te)
 	require.Equal(t, errNilLogger, err)
 }
@@ -93,7 +93,7 @@ func TestTracesExporter_Default(t *testing.T) {
 func TestTracesRequestExporter_Default(t *testing.T) {
 	td := ptrace.NewTraces()
 	te, err := NewTracesRequestExporter(context.Background(), exportertest.NewNopSettings(),
-		(&fakeRequestConverter{}).requestFromTracesFunc)
+		(&internal.FakeRequestConverter{}).RequestFromTracesFunc)
 	assert.NotNil(t, te)
 	require.NoError(t, err)
 
@@ -115,7 +115,7 @@ func TestTracesExporter_WithCapabilities(t *testing.T) {
 func TestTracesRequestExporter_WithCapabilities(t *testing.T) {
 	capabilities := consumer.Capabilities{MutatesData: true}
 	te, err := NewTracesRequestExporter(context.Background(), exportertest.NewNopSettings(),
-		(&fakeRequestConverter{}).requestFromTracesFunc, WithCapabilities(capabilities))
+		(&internal.FakeRequestConverter{}).RequestFromTracesFunc, WithCapabilities(capabilities))
 	assert.NotNil(t, te)
 	require.NoError(t, err)
 
@@ -137,7 +137,7 @@ func TestTracesRequestExporter_Default_ConvertError(t *testing.T) {
 	td := ptrace.NewTraces()
 	want := errors.New("convert_error")
 	te, err := NewTracesRequestExporter(context.Background(), exportertest.NewNopSettings(),
-		(&fakeRequestConverter{tracesError: want}).requestFromTracesFunc)
+		(&internal.FakeRequestConverter{TracesError: want}).RequestFromTracesFunc)
 	require.NoError(t, err)
 	require.NotNil(t, te)
 	require.Equal(t, consumererror.NewPermanent(want), te.ConsumeTraces(context.Background(), td))
@@ -147,7 +147,7 @@ func TestTracesRequestExporter_Default_ExportError(t *testing.T) {
 	td := ptrace.NewTraces()
 	want := errors.New("export_error")
 	te, err := NewTracesRequestExporter(context.Background(), exportertest.NewNopSettings(),
-		(&fakeRequestConverter{requestError: want}).requestFromTracesFunc)
+		(&internal.FakeRequestConverter{RequestError: want}).RequestFromTracesFunc)
 	require.NoError(t, err)
 	require.NotNil(t, te)
 	require.Equal(t, want, te.ConsumeTraces(context.Background(), td))
@@ -164,7 +164,7 @@ func TestTracesExporter_WithPersistentQueue(t *testing.T) {
 	te, err := NewTracesExporter(context.Background(), set, &fakeTracesExporterConfig, ts.ConsumeTraces, WithRetry(rCfg), WithQueue(qCfg))
 	require.NoError(t, err)
 
-	host := &mockHost{ext: map[component.ID]component.Component{
+	host := &internal.MockHost{Ext: map[component.ID]component.Component{
 		storageID: queue.NewMockStorageExtension(nil),
 	}}
 	require.NoError(t, te.Start(context.Background(), host))
@@ -211,7 +211,7 @@ func TestTracesRequestExporter_WithRecordMetrics(t *testing.T) {
 
 	te, err := NewTracesRequestExporter(context.Background(),
 		exporter.Settings{ID: fakeTracesExporterName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
-		(&fakeRequestConverter{}).requestFromTracesFunc)
+		(&internal.FakeRequestConverter{}).RequestFromTracesFunc)
 	require.NoError(t, err)
 	require.NotNil(t, te)
 
@@ -239,7 +239,7 @@ func TestTracesRequestExporter_WithRecordMetrics_RequestSenderError(t *testing.T
 
 	te, err := NewTracesRequestExporter(context.Background(),
 		exporter.Settings{ID: fakeTracesExporterName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
-		(&fakeRequestConverter{requestError: want}).requestFromTracesFunc)
+		(&internal.FakeRequestConverter{RequestError: want}).RequestFromTracesFunc)
 	require.NoError(t, err)
 	require.NotNil(t, te)
 
@@ -292,7 +292,7 @@ func TestTracesRequestExporter_WithSpan(t *testing.T) {
 	otel.SetTracerProvider(set.TracerProvider)
 	defer otel.SetTracerProvider(nooptrace.NewTracerProvider())
 
-	te, err := NewTracesRequestExporter(context.Background(), set, (&fakeRequestConverter{}).requestFromTracesFunc)
+	te, err := NewTracesRequestExporter(context.Background(), set, (&internal.FakeRequestConverter{}).RequestFromTracesFunc)
 	require.NoError(t, err)
 	require.NotNil(t, te)
 
@@ -322,7 +322,7 @@ func TestTracesRequestExporter_WithSpan_ExportError(t *testing.T) {
 	defer otel.SetTracerProvider(nooptrace.NewTracerProvider())
 
 	want := errors.New("export_error")
-	te, err := NewTracesRequestExporter(context.Background(), set, (&fakeRequestConverter{requestError: want}).requestFromTracesFunc)
+	te, err := NewTracesRequestExporter(context.Background(), set, (&internal.FakeRequestConverter{RequestError: want}).RequestFromTracesFunc)
 	require.NoError(t, err)
 	require.NotNil(t, te)
 
@@ -347,7 +347,7 @@ func TestTracesRequestExporter_WithShutdown(t *testing.T) {
 	shutdown := func(context.Context) error { shutdownCalled = true; return nil }
 
 	te, err := NewTracesRequestExporter(context.Background(), exportertest.NewNopSettings(),
-		(&fakeRequestConverter{}).requestFromTracesFunc, WithShutdown(shutdown))
+		(&internal.FakeRequestConverter{}).RequestFromTracesFunc, WithShutdown(shutdown))
 	assert.NotNil(t, te)
 	assert.NoError(t, err)
 
@@ -373,7 +373,7 @@ func TestTracesRequestExporter_WithShutdown_ReturnError(t *testing.T) {
 	shutdownErr := func(context.Context) error { return want }
 
 	te, err := NewTracesRequestExporter(context.Background(), exportertest.NewNopSettings(),
-		(&fakeRequestConverter{}).requestFromTracesFunc, WithShutdown(shutdownErr))
+		(&internal.FakeRequestConverter{}).RequestFromTracesFunc, WithShutdown(shutdownErr))
 	assert.NotNil(t, te)
 	assert.NoError(t, err)
 
@@ -433,7 +433,7 @@ func checkWrapSpanForTracesExporter(t *testing.T, sr *tracetest.SpanRecorder, tr
 
 	for _, sd := range gotSpanData[:numRequests] {
 		require.Equalf(t, parentSpan.SpanContext(), sd.Parent(), "Exporter span not a child\nSpanData %v", sd)
-		checkStatus(t, sd, wantError)
+		internal.CheckStatus(t, sd, wantError)
 
 		sentSpans := numSpans
 		var failedToSendSpans int64
