diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index 78a0bb2ac..26ed0a890 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -9,8 +9,8 @@ toolchain go1.22.7
 require (
 	go.opentelemetry.io/collector/component v0.109.0
 	go.opentelemetry.io/collector/confmap v1.15.0
-	go.opentelemetry.io/collector/confmap/provider/envprovider v0.109.0
-	go.opentelemetry.io/collector/confmap/provider/fileprovider v0.109.0
+	go.opentelemetry.io/collector/confmap/provider/envprovider v1.15.0
+	go.opentelemetry.io/collector/confmap/provider/fileprovider v1.15.0
 	go.opentelemetry.io/collector/confmap/provider/httpprovider v0.109.0
 	go.opentelemetry.io/collector/confmap/provider/httpsprovider v0.109.0
 	go.opentelemetry.io/collector/confmap/provider/yamlprovider v0.109.0
