diff --git a/processor/tailsampling/processor_test.go b/processor/tailsampling/processor_test.go
index 542df7e80..dda08d87d 100644
--- a/processor/tailsampling/processor_test.go
+++ b/processor/tailsampling/processor_test.go
@@ -25,9 +25,9 @@ import (
 
 	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
 	"github.com/open-telemetry/opentelemetry-service/exporter/exportertest"
-	"github.com/open-telemetry/opentelemetry-service/internal/collector/processor/idbatcher"
-	"github.com/open-telemetry/opentelemetry-service/internal/collector/sampling"
 	"github.com/open-telemetry/opentelemetry-service/processor"
+	"github.com/open-telemetry/opentelemetry-service/processor/tailsampling/idbatcher"
+	"github.com/open-telemetry/opentelemetry-service/processor/tailsampling/sampling"
 	tracetranslator "github.com/open-telemetry/opentelemetry-service/translator/trace"
 )
 
