diff --git a/internal/e2e/go.mod b/internal/e2e/go.mod
index 074fa0e1e..05f5d0537 100644
--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -94,8 +94,6 @@ replace go.opentelemetry.io/collector/config/configtls => ../../config/configtls
 
 replace go.opentelemetry.io/collector/extension/auth => ../../extension/auth
 
-replace go.opentelemetry.io/collector/semconv => ../../semconv
-
 replace go.opentelemetry.io/collector/exporter/otlpexporter => ../../exporter/otlpexporter
 
 replace go.opentelemetry.io/collector/config/configcompression => ../../config/configcompression
