diff --git a/exporter/exporter.go b/exporter/exporter.go
index 2a427933d..98329fdcf 100644
--- a/exporter/exporter.go
+++ b/exporter/exporter.go
@@ -48,28 +48,46 @@ type Settings struct {
 type Factory interface {
 	component.Factory
 
-	// CreateTracesExporter creates a TracesExporter based on this config.
+	// CreateTraces creates a Traces exporter based on this config.
 	// If the exporter type does not support tracing,
 	// this function returns the error [pipeline.ErrSignalNotSupported].
+	CreateTraces(ctx context.Context, set Settings, cfg component.Config) (Traces, error)
+
+	// TracesStability gets the stability level of the Traces exporter.
+	TracesStability() component.StabilityLevel
+
+	// Deprecated: [v0.112.0] use CreateTraces.
 	CreateTracesExporter(ctx context.Context, set Settings, cfg component.Config) (Traces, error)
 
-	// TracesExporterStability gets the stability level of the TracesExporter.
+	// Deprecated: [v0.112.0] use TracesStability.
 	TracesExporterStability() component.StabilityLevel
 
-	// CreateMetricsExporter creates a MetricsExporter based on this config.
+	// CreateMetrics creates a Metrics exporter based on this config.
 	// If the exporter type does not support metrics,
 	// this function returns the error [pipeline.ErrSignalNotSupported].
+	CreateMetrics(ctx context.Context, set Settings, cfg component.Config) (Metrics, error)
+
+	// MetricsStability gets the stability level of the Metrics exporter.
+	MetricsStability() component.StabilityLevel
+
+	// Deprecated: [v0.112.0] use CreateMetrics.
 	CreateMetricsExporter(ctx context.Context, set Settings, cfg component.Config) (Metrics, error)
 
-	// MetricsExporterStability gets the stability level of the MetricsExporter.
+	// Deprecated: [v0.112.0] use MetricsStability.
 	MetricsExporterStability() component.StabilityLevel
 
-	// CreateLogsExporter creates a LogsExporter based on the config.
+	// CreateLogs creates a Logs exporter based on the config.
 	// If the exporter type does not support logs,
 	// this function returns the error [pipeline.ErrSignalNotSupported].
+	CreateLogs(ctx context.Context, set Settings, cfg component.Config) (Logs, error)
+
+	// LogsStability gets the stability level of the Logs exporter.
+	LogsStability() component.StabilityLevel
+
+	// Deprecated: [v0.112.0] use CreateLogs.
 	CreateLogsExporter(ctx context.Context, set Settings, cfg component.Config) (Logs, error)
 
-	// LogsExporterStability gets the stability level of the LogsExporter.
+	// Deprecated: [v0.112.0] use LogsStability.
 	LogsExporterStability() component.StabilityLevel
 
 	unexportedFactoryFunc()
@@ -77,52 +95,67 @@ type Factory interface {
 
 // FactoryOption apply changes to Factory.
 type FactoryOption interface {
-	// applyExporterFactoryOption applies the option.
-	applyExporterFactoryOption(o *factory)
+	// applyOption applies the option.
+	applyOption(o *factory)
 }
 
 var _ FactoryOption = (*factoryOptionFunc)(nil)
 
-// factoryOptionFunc is an ExporterFactoryOption created through a function.
+// factoryOptionFunc is an FactoryOption created through a function.
 type factoryOptionFunc func(*factory)
 
-func (f factoryOptionFunc) applyExporterFactoryOption(o *factory) {
+func (f factoryOptionFunc) applyOption(o *factory) {
 	f(o)
 }
 
 // CreateTracesFunc is the equivalent of Factory.CreateTraces.
 type CreateTracesFunc func(context.Context, Settings, component.Config) (Traces, error)
 
-// CreateTracesExporter implements Factory.CreateTraces.
-func (f CreateTracesFunc) CreateTracesExporter(ctx context.Context, set Settings, cfg component.Config) (Traces, error) {
+// CreateTraces implements Factory.CreateTraces.
+func (f CreateTracesFunc) CreateTraces(ctx context.Context, set Settings, cfg component.Config) (Traces, error) {
 	if f == nil {
 		return nil, pipeline.ErrSignalNotSupported
 	}
 	return f(ctx, set, cfg)
 }
 
+// Deprecated: [v0.112.0] use CreateTraces.
+func (f CreateTracesFunc) CreateTracesExporter(ctx context.Context, set Settings, cfg component.Config) (Traces, error) {
+	return f.CreateTraces(ctx, set, cfg)
+}
+
 // CreateMetricsFunc is the equivalent of Factory.CreateMetrics.
 type CreateMetricsFunc func(context.Context, Settings, component.Config) (Metrics, error)
 
-// CreateMetricsExporter implements Factory.CreateMetrics.
-func (f CreateMetricsFunc) CreateMetricsExporter(ctx context.Context, set Settings, cfg component.Config) (Metrics, error) {
+// CreateMetrics implements Factory.CreateMetrics.
+func (f CreateMetricsFunc) CreateMetrics(ctx context.Context, set Settings, cfg component.Config) (Metrics, error) {
 	if f == nil {
 		return nil, pipeline.ErrSignalNotSupported
 	}
 	return f(ctx, set, cfg)
 }
 
+// Deprecated: [v0.112.0] use CreateMetrics.
+func (f CreateMetricsFunc) CreateMetricsExporter(ctx context.Context, set Settings, cfg component.Config) (Metrics, error) {
+	return f.CreateMetrics(ctx, set, cfg)
+}
+
 // CreateLogsFunc is the equivalent of Factory.CreateLogs.
 type CreateLogsFunc func(context.Context, Settings, component.Config) (Logs, error)
 
-// CreateLogsExporter implements Factory.CreateLogs.
-func (f CreateLogsFunc) CreateLogsExporter(ctx context.Context, set Settings, cfg component.Config) (Logs, error) {
+// CreateLogs implements Factory.CreateLogs.
+func (f CreateLogsFunc) CreateLogs(ctx context.Context, set Settings, cfg component.Config) (Logs, error) {
 	if f == nil {
 		return nil, pipeline.ErrSignalNotSupported
 	}
 	return f(ctx, set, cfg)
 }
 
+// Deprecated: [v0.112.0] use CreateLogs.
+func (f CreateLogsFunc) CreateLogsExporter(ctx context.Context, set Settings, cfg component.Config) (Logs, error) {
+	return f.CreateLogs(ctx, set, cfg)
+}
+
 type factory struct {
 	cfgType component.Type
 	component.CreateDefaultConfigFunc
@@ -140,19 +173,34 @@ func (f *factory) Type() component.Type {
 
 func (f *factory) unexportedFactoryFunc() {}
 
+func (f *factory) TracesStability() component.StabilityLevel {
+	return f.tracesStabilityLevel
+}
+
+func (f *factory) MetricsStability() component.StabilityLevel {
+	return f.metricsStabilityLevel
+}
+
+func (f *factory) LogsStability() component.StabilityLevel {
+	return f.logsStabilityLevel
+}
+
+// Deprecated: [v0.112.0] use TracesStability.
 func (f *factory) TracesExporterStability() component.StabilityLevel {
 	return f.tracesStabilityLevel
 }
 
+// Deprecated: [v0.112.0] use MetricsStability.
 func (f *factory) MetricsExporterStability() component.StabilityLevel {
 	return f.metricsStabilityLevel
 }
 
+// Deprecated: [v0.112.0] use LogsStability.
 func (f *factory) LogsExporterStability() component.StabilityLevel {
 	return f.logsStabilityLevel
 }
 
-// WithTraces overrides the default "error not supported" implementation for CreateTracesExporter and the default "undefined" stability level.
+// WithTraces overrides the default "error not supported" implementation for Factory.CreateTraces and the default "undefined" stability level.
 func WithTraces(createTraces CreateTracesFunc, sl component.StabilityLevel) FactoryOption {
 	return factoryOptionFunc(func(o *factory) {
 		o.tracesStabilityLevel = sl
@@ -160,7 +208,7 @@ func WithTraces(createTraces CreateTracesFunc, sl component.StabilityLevel) Fact
 	})
 }
 
-// WithMetrics overrides the default "error not supported" implementation for CreateMetricsExporter and the default "undefined" stability level.
+// WithMetrics overrides the default "error not supported" implementation for Factory.CreateMetrics and the default "undefined" stability level.
 func WithMetrics(createMetrics CreateMetricsFunc, sl component.StabilityLevel) FactoryOption {
 	return factoryOptionFunc(func(o *factory) {
 		o.metricsStabilityLevel = sl
@@ -168,7 +216,7 @@ func WithMetrics(createMetrics CreateMetricsFunc, sl component.StabilityLevel) F
 	})
 }
 
-// WithLogs overrides the default "error not supported" implementation for CreateLogsExporter and the default "undefined" stability level.
+// WithLogs overrides the default "error not supported" implementation for Factory.CreateLogs and the default "undefined" stability level.
 func WithLogs(createLogs CreateLogsFunc, sl component.StabilityLevel) FactoryOption {
 	return factoryOptionFunc(func(o *factory) {
 		o.logsStabilityLevel = sl
@@ -183,7 +231,7 @@ func NewFactory(cfgType component.Type, createDefaultConfig component.CreateDefa
 		CreateDefaultConfigFunc: createDefaultConfig,
 	}
 	for _, opt := range options {
-		opt.applyExporterFactoryOption(f)
+		opt.applyOption(f)
 	}
 	return f
 }
