diff --git a/receiver/prometheusreceiver/factory_test.go b/receiver/prometheusreceiver/factory_test.go
index 0e8857bf4..db840a1d5 100644
--- a/receiver/prometheusreceiver/factory_test.go
+++ b/receiver/prometheusreceiver/factory_test.go
@@ -18,6 +18,8 @@ import (
 	"context"
 	"testing"
 
+	"go.uber.org/zap"
+
 	"github.com/stretchr/testify/assert"
 
 	"github.com/open-telemetry/opentelemetry-service/factories"
@@ -33,13 +35,13 @@ func TestCreateReceiver(t *testing.T) {
 	factory := factories.GetReceiverFactory(typeStr)
 	cfg := factory.CreateDefaultConfig()
 
-	tReceiver, err := factory.CreateTraceReceiver(context.Background(), cfg, nil)
+	tReceiver, err := factory.CreateTraceReceiver(context.Background(), zap.NewNop(), cfg, nil)
 	assert.Equal(t, err, factories.ErrDataTypeIsNotSupported)
 	assert.Nil(t, tReceiver)
 
 	// The default config does not provide scrape_config so we expect that metrics receiver
 	// creation must also fail.
-	mReceiver, err := factory.CreateMetricsReceiver(cfg, nil)
+	mReceiver, err := factory.CreateMetricsReceiver(zap.NewNop(), cfg, nil)
 	assert.Equal(t, err, errNilScrapeConfig)
 	assert.Nil(t, mReceiver)
 }
