diff --git a/translator/internaldata/resource_to_oc_test.go b/translator/internaldata/resource_to_oc_test.go
index 35002c31c..28620cfa0 100644
--- a/translator/internaldata/resource_to_oc_test.go
+++ b/translator/internaldata/resource_to_oc_test.go
@@ -31,6 +31,7 @@ import (
 	otlptrace "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/trace/v1"
 	"go.opentelemetry.io/collector/internal/goldendataset"
 	"go.opentelemetry.io/collector/translator/conventions"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
 )
 
 func TestResourceToOC(t *testing.T) {
@@ -121,19 +122,19 @@ func TestContainerResourceToOC(t *testing.T) {
 }
 
 func TestAttributeValueToString(t *testing.T) {
-	assert.EqualValues(t, "", attributeValueToString(pdata.NewAttributeValueNull(), false))
-	assert.EqualValues(t, "abc", attributeValueToString(pdata.NewAttributeValueString("abc"), false))
-	assert.EqualValues(t, `"abc"`, attributeValueToString(pdata.NewAttributeValueString("abc"), true))
-	assert.EqualValues(t, "123", attributeValueToString(pdata.NewAttributeValueInt(123), false))
-	assert.EqualValues(t, "1.23", attributeValueToString(pdata.NewAttributeValueDouble(1.23), false))
-	assert.EqualValues(t, "true", attributeValueToString(pdata.NewAttributeValueBool(true), false))
+	assert.EqualValues(t, "", tracetranslator.AttributeValueToString(pdata.NewAttributeValueNull(), false))
+	assert.EqualValues(t, "abc", tracetranslator.AttributeValueToString(pdata.NewAttributeValueString("abc"), false))
+	assert.EqualValues(t, `"abc"`, tracetranslator.AttributeValueToString(pdata.NewAttributeValueString("abc"), true))
+	assert.EqualValues(t, "123", tracetranslator.AttributeValueToString(pdata.NewAttributeValueInt(123), false))
+	assert.EqualValues(t, "1.23", tracetranslator.AttributeValueToString(pdata.NewAttributeValueDouble(1.23), false))
+	assert.EqualValues(t, "true", tracetranslator.AttributeValueToString(pdata.NewAttributeValueBool(true), false))
 
 	v := pdata.NewAttributeValueMap()
 	v.MapVal().InsertString(`a"\`, `b"\`)
 	v.MapVal().InsertInt("c", 123)
 	v.MapVal().Insert("d", pdata.NewAttributeValueNull())
 	v.MapVal().Insert("e", v)
-	assert.EqualValues(t, `{"a\"\\":"b\"\\","c":123,"d":null,"e":{"a\"\\":"b\"\\","c":123,"d":null}}`, attributeValueToString(v, false))
+	assert.EqualValues(t, `{"a\"\\":"b\"\\","c":123,"d":null,"e":{"a\"\\":"b\"\\","c":123,"d":null}}`, tracetranslator.AttributeValueToString(v, false))
 
 	v = pdata.NewAttributeValueArray()
 	av := pdata.NewAttributeValueString(`b"\`)
@@ -144,7 +145,7 @@ func TestAttributeValueToString(t *testing.T) {
 	v.ArrayVal().Append(av)
 	av = pdata.NewAttributeValueArray()
 	v.ArrayVal().Append(av)
-	assert.EqualValues(t, `["b\"\\",123,null,"\u003cInvalid array value\u003e"]`, attributeValueToString(v, false))
+	assert.EqualValues(t, `["b\"\\",123,null,"\u003cInvalid array value\u003e"]`, tracetranslator.AttributeValueToString(v, false))
 }
 
 func TestInferResourceType(t *testing.T) {
