diff --git a/internal/data/trace_test.go b/internal/data/trace_test.go
index 064e9530c..ac61d87bf 100644
--- a/internal/data/trace_test.go
+++ b/internal/data/trace_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019 OpenTelemetry Authors
+// Copyright 2020 OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
