diff --git a/internal/processor/filterhelper/filterhelper_test.go b/internal/processor/filterhelper/filterhelper_test.go
index e0eaa13b8..c5b3cd0a3 100644
--- a/internal/processor/filterhelper/filterhelper_test.go
+++ b/internal/processor/filterhelper/filterhelper_test.go
@@ -19,40 +19,40 @@ import (
 
 	"github.com/stretchr/testify/assert"
 
-	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
 )
 
 func TestHelper_AttributeValue(t *testing.T) {
 	val, err := NewAttributeValueRaw(uint8(123))
-	assert.Equal(t, data.NewAttributeValueInt(123), val)
+	assert.Equal(t, pdata.NewAttributeValueInt(123), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(uint16(123))
-	assert.Equal(t, data.NewAttributeValueInt(123), val)
+	assert.Equal(t, pdata.NewAttributeValueInt(123), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(int8(123))
-	assert.Equal(t, data.NewAttributeValueInt(123), val)
+	assert.Equal(t, pdata.NewAttributeValueInt(123), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(int16(123))
-	assert.Equal(t, data.NewAttributeValueInt(123), val)
+	assert.Equal(t, pdata.NewAttributeValueInt(123), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(float32(234.129312))
-	assert.Equal(t, data.NewAttributeValueDouble(float64(float32(234.129312))), val)
+	assert.Equal(t, pdata.NewAttributeValueDouble(float64(float32(234.129312))), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(234.129312)
-	assert.Equal(t, data.NewAttributeValueDouble(234.129312), val)
+	assert.Equal(t, pdata.NewAttributeValueDouble(234.129312), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(true)
-	assert.Equal(t, data.NewAttributeValueBool(true), val)
+	assert.Equal(t, pdata.NewAttributeValueBool(true), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw("bob the builder")
-	assert.Equal(t, data.NewAttributeValueString("bob the builder"), val)
+	assert.Equal(t, pdata.NewAttributeValueString("bob the builder"), val)
 	assert.NoError(t, err)
 
 	val, err = NewAttributeValueRaw(nil)
