diff --git a/component/componenttest/nop_processor.go b/component/componenttest/nop_processor.go
index 6fa0a4052..1ef661f00 100644
--- a/component/componenttest/nop_processor.go
+++ b/component/componenttest/nop_processor.go
@@ -39,7 +39,7 @@ type nopProcessorConfig struct {
 func NewNopProcessorFactory() component.ProcessorFactory {
 	return component.NewProcessorFactory(
 		"nop",
-		func() component.ProcessorConfig {
+		func() component.Config {
 			return &nopProcessorConfig{
 				ProcessorSettings: config.NewProcessorSettings(component.NewID("nop")),
 			}
@@ -50,15 +50,15 @@ func NewNopProcessorFactory() component.ProcessorFactory {
 	)
 }
 
-func createTracesProcessor(context.Context, component.ProcessorCreateSettings, component.ProcessorConfig, consumer.Traces) (component.TracesProcessor, error) {
+func createTracesProcessor(context.Context, component.ProcessorCreateSettings, component.Config, consumer.Traces) (component.TracesProcessor, error) {
 	return nopProcessorInstance, nil
 }
 
-func createMetricsProcessor(context.Context, component.ProcessorCreateSettings, component.ProcessorConfig, consumer.Metrics) (component.MetricsProcessor, error) {
+func createMetricsProcessor(context.Context, component.ProcessorCreateSettings, component.Config, consumer.Metrics) (component.MetricsProcessor, error) {
 	return nopProcessorInstance, nil
 }
 
-func createLogsProcessor(context.Context, component.ProcessorCreateSettings, component.ProcessorConfig, consumer.Logs) (component.LogsProcessor, error) {
+func createLogsProcessor(context.Context, component.ProcessorCreateSettings, component.Config, consumer.Logs) (component.LogsProcessor, error) {
 	return nopProcessorInstance, nil
 }
 
