diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index 49590ba5e..84cebcba8 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -6,23 +6,23 @@ go 1.20
 
 require (
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector/component v0.86.0
-	go.opentelemetry.io/collector/connector v0.86.0
-	go.opentelemetry.io/collector/connector/forwardconnector v0.86.0
-	go.opentelemetry.io/collector/exporter v0.86.0
-	go.opentelemetry.io/collector/exporter/debugexporter v0.86.0
-	go.opentelemetry.io/collector/exporter/loggingexporter v0.86.0
-	go.opentelemetry.io/collector/exporter/otlpexporter v0.86.0
-	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.86.0
-	go.opentelemetry.io/collector/extension v0.86.0
-	go.opentelemetry.io/collector/extension/ballastextension v0.86.0
-	go.opentelemetry.io/collector/extension/zpagesextension v0.86.0
+	go.opentelemetry.io/collector/component v0.87.0
+	go.opentelemetry.io/collector/connector v0.87.0
+	go.opentelemetry.io/collector/connector/forwardconnector v0.87.0
+	go.opentelemetry.io/collector/exporter v0.87.0
+	go.opentelemetry.io/collector/exporter/debugexporter v0.87.0
+	go.opentelemetry.io/collector/exporter/loggingexporter v0.87.0
+	go.opentelemetry.io/collector/exporter/otlpexporter v0.87.0
+	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.87.0
+	go.opentelemetry.io/collector/extension v0.87.0
+	go.opentelemetry.io/collector/extension/ballastextension v0.87.0
+	go.opentelemetry.io/collector/extension/zpagesextension v0.87.0
 	go.opentelemetry.io/collector/otelcol v0.0.0-00010101000000-000000000000
-	go.opentelemetry.io/collector/processor v0.86.0
-	go.opentelemetry.io/collector/processor/batchprocessor v0.86.0
-	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.86.0
-	go.opentelemetry.io/collector/receiver v0.86.0
-	go.opentelemetry.io/collector/receiver/otlpreceiver v0.86.0
+	go.opentelemetry.io/collector/processor v0.87.0
+	go.opentelemetry.io/collector/processor/batchprocessor v0.87.0
+	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.87.0
+	go.opentelemetry.io/collector/receiver v0.87.0
+	go.opentelemetry.io/collector/receiver/otlpreceiver v0.87.0
 	golang.org/x/sys v0.12.0
 )
 
@@ -75,23 +75,23 @@ require (
 	github.com/tklauser/numcpus v0.6.1 // indirect
 	github.com/yusufpapurcu/wmi v1.2.3 // indirect
 	go.opencensus.io v0.24.0 // indirect
-	go.opentelemetry.io/collector v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/configauth v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/configcompression v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/configgrpc v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/confighttp v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/confignet v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/configopaque v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/configtelemetry v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/configtls v0.86.0 // indirect
-	go.opentelemetry.io/collector/config/internal v0.86.0 // indirect
-	go.opentelemetry.io/collector/confmap v0.86.0 // indirect
-	go.opentelemetry.io/collector/consumer v0.86.0 // indirect
-	go.opentelemetry.io/collector/extension/auth v0.86.0 // indirect
-	go.opentelemetry.io/collector/featuregate v1.0.0-rcv0015 // indirect
-	go.opentelemetry.io/collector/pdata v1.0.0-rcv0015 // indirect
-	go.opentelemetry.io/collector/semconv v0.86.0 // indirect
-	go.opentelemetry.io/collector/service v0.86.0 // indirect
+	go.opentelemetry.io/collector v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/configauth v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/configcompression v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/configgrpc v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/confighttp v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/confignet v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/configopaque v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/configtelemetry v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/configtls v0.87.0 // indirect
+	go.opentelemetry.io/collector/config/internal v0.87.0 // indirect
+	go.opentelemetry.io/collector/confmap v0.87.0 // indirect
+	go.opentelemetry.io/collector/consumer v0.87.0 // indirect
+	go.opentelemetry.io/collector/extension/auth v0.87.0 // indirect
+	go.opentelemetry.io/collector/featuregate v1.0.0-rcv0016 // indirect
+	go.opentelemetry.io/collector/pdata v1.0.0-rcv0016 // indirect
+	go.opentelemetry.io/collector/semconv v0.87.0 // indirect
+	go.opentelemetry.io/collector/service v0.87.0 // indirect
 	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.45.0 // indirect
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.45.0 // indirect
 	go.opentelemetry.io/contrib/propagators/b3 v1.19.0 // indirect
