diff --git a/pdata/pcommon/common.go b/pdata/pcommon/common.go
index 35451dcac..3661e7401 100644
--- a/pdata/pcommon/common.go
+++ b/pdata/pcommon/common.go
@@ -717,6 +717,9 @@ func (m Map) PutEmptySlice(k string) Slice {
 // Returns the same instance to allow nicer code like:
 //
 //	assert.EqualValues(t, expected.Sort(), actual.Sort())
+//
+// IMPORTANT: Sort mutates the data, if you call this function in a consumer,
+// the consumer must be configured that it mutates data.
 func (m Map) Sort() Map {
 	// Intention is to move the nil values at the end.
 	sort.SliceStable(*m.getOrig(), func(i, j int) bool {
