diff --git a/pdata/internal/generated_wrapper_resource.go b/pdata/internal/generated_wrapper_resource.go
index aa9bb057c..41cc9a190 100644
--- a/pdata/internal/generated_wrapper_resource.go
+++ b/pdata/internal/generated_wrapper_resource.go
@@ -30,16 +30,9 @@ func NewResource(orig *otlpresource.Resource, state *State) Resource {
 
 func GenerateTestResource() Resource {
 	orig := otlpresource.Resource{}
+	FillOrigTestResource(&orig)
 	state := StateMutable
-	tv := NewResource(&orig, &state)
-	FillTestResource(tv)
-	return tv
-}
-
-func FillTestResource(tv Resource) {
-	FillTestMap(NewMap(&tv.orig.Attributes, tv.state))
-	tv.orig.DroppedAttributesCount = uint32(13)
-	FillTestEntityRefSlice(NewEntityRefSlice(&tv.orig.EntityRefs, tv.state))
+	return NewResource(&orig, &state)
 }
 
 // MarshalJSONStream marshals all properties from the current struct to the destination stream.
@@ -82,3 +75,9 @@ func CopyOrigResource(dest, src *otlpresource.Resource) {
 	dest.DroppedAttributesCount = src.DroppedAttributesCount
 	dest.EntityRefs = CopyOrigEntityRefSlice(dest.EntityRefs, src.EntityRefs)
 }
+
+func FillOrigTestResource(orig *otlpresource.Resource) {
+	orig.Attributes = GenerateOrigTestKeyValueSlice()
+	orig.DroppedAttributesCount = uint32(13)
+	orig.EntityRefs = GenerateOrigTestEntityRefSlice()
+}
