diff --git a/pdata/internal/generated_wrapper_uint64slice.go b/pdata/internal/generated_wrapper_uint64slice.go
index ff68aed95..1f678b7dc 100644
--- a/pdata/internal/generated_wrapper_uint64slice.go
+++ b/pdata/internal/generated_wrapper_uint64slice.go
@@ -27,16 +27,10 @@ func NewUInt64Slice(orig *[]uint64, state *State) UInt64Slice {
 	return UInt64Slice{orig: orig, state: state}
 }
 
-func FillTestUInt64Slice(ms UInt64Slice) {
-	*ms.orig = []uint64{1, 2, 3}
-}
-
 func GenerateTestUInt64Slice() UInt64Slice {
-	orig := []uint64(nil)
+	orig := GenerateOrigTestUint64Slice()
 	state := StateMutable
-	ms := NewUInt64Slice(&orig, &state)
-	FillTestUInt64Slice(ms)
-	return ms
+	return NewUInt64Slice(&orig, &state)
 }
 
 // MarshalJSONStreamUInt64Slice marshals all properties from the current struct to the destination stream.
@@ -63,3 +57,7 @@ func UnmarshalJSONIterUInt64Slice(ms UInt64Slice, iter *json.Iterator) {
 func CopyOrigUint64Slice(dst, src []uint64) []uint64 {
 	return append(dst[:0], src...)
 }
+
+func GenerateOrigTestUint64Slice() []uint64 {
+	return []uint64{1, 2, 3}
+}
