diff --git a/pdata/internal/generated_wrapper_span_link.go b/pdata/internal/generated_wrapper_span_link.go
index 7fec6db0b..529c07126 100644
--- a/pdata/internal/generated_wrapper_span_link.go
+++ b/pdata/internal/generated_wrapper_span_link.go
@@ -7,6 +7,7 @@
 package internal
 
 import (
+	"go.opentelemetry.io/collector/pdata/internal/data"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 )
 
@@ -18,3 +19,12 @@ func CopyOrigSpan_Link(dest, src *otlptrace.Span_Link) {
 	dest.Attributes = CopyOrigKeyValueSlice(dest.Attributes, src.Attributes)
 	dest.DroppedAttributesCount = src.DroppedAttributesCount
 }
+
+func FillOrigTestSpan_Link(orig *otlptrace.Span_Link) {
+	orig.TraceId = data.TraceID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1})
+	orig.SpanId = data.SpanID([8]byte{8, 7, 6, 5, 4, 3, 2, 1})
+	FillOrigTestTraceState(&orig.TraceState)
+	orig.Flags = uint32(13)
+	orig.Attributes = GenerateOrigTestKeyValueSlice()
+	orig.DroppedAttributesCount = uint32(13)
+}
