diff --git a/pdata/internal/generated_wrapper_exporttracepartialsuccess.go b/pdata/internal/generated_wrapper_exporttracepartialsuccess.go
index 559986f5b..2b25e2e5c 100644
--- a/pdata/internal/generated_wrapper_exporttracepartialsuccess.go
+++ b/pdata/internal/generated_wrapper_exporttracepartialsuccess.go
@@ -14,3 +14,8 @@ func CopyOrigExportTracePartialSuccess(dest, src *otlpcollectortrace.ExportTrace
 	dest.RejectedSpans = src.RejectedSpans
 	dest.ErrorMessage = src.ErrorMessage
 }
+
+func FillOrigTestExportTracePartialSuccess(orig *otlpcollectortrace.ExportTracePartialSuccess) {
+	orig.RejectedSpans = int64(13)
+	orig.ErrorMessage = "test_errormessage"
+}
