diff --git a/.github/workflows/scripts/get-components.sh b/.github/workflows/scripts/get-components.sh
new file mode 100755
index 000000000..7730ce5fe
--- /dev/null
+++ b/.github/workflows/scripts/get-components.sh
@@ -0,0 +1,11 @@
+#!/usr/bin/env sh
+#
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Get a list of components within the repository that have some form of ownership
+# ascribed to them.
+
+grep -E '^[A-Za-z0-9/]' .github/CODEOWNERS | \
+    awk '{ print $1 }' | \
+    sed -E 's%(.+)/$%\1%'
