diff --git a/internal/goldendataset/resource_generator_test.go b/internal/goldendataset/resource_generator_test.go
index f11b9ca0b..d13a2dab9 100644
--- a/internal/goldendataset/resource_generator_test.go
+++ b/internal/goldendataset/resource_generator_test.go
@@ -19,7 +19,7 @@ import (
 
 	"github.com/stretchr/testify/assert"
 
-	otlpresource "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/resource/v1"
+	otlpresource "go.opentelemetry.io/collector/internal/data/protogen/resource/v1"
 )
 
 func TestGenerateResource(t *testing.T) {
