diff --git a/exporter/go.mod b/exporter/go.mod
index 3805cf5e7..0f21a263f 100644
--- a/exporter/go.mod
+++ b/exporter/go.mod
@@ -16,9 +16,11 @@ require (
 	go.opentelemetry.io/collector/exporter/exportertest v0.128.0
 	go.opentelemetry.io/collector/extension/extensiontest v0.128.0
 	go.opentelemetry.io/collector/extension/xextension v0.128.0
+	go.opentelemetry.io/collector/featuregate v1.34.0
 	go.opentelemetry.io/collector/pdata v1.34.0
 	go.opentelemetry.io/collector/pdata/pprofile v0.128.0
 	go.opentelemetry.io/collector/pdata/testdata v0.128.0
+	go.opentelemetry.io/collector/pdata/xpdata v0.0.0-00010101000000-000000000000
 	go.opentelemetry.io/collector/pipeline v0.128.0
 	go.opentelemetry.io/otel v1.36.0
 	go.opentelemetry.io/otel/metric v1.36.0
@@ -53,7 +55,6 @@ require (
 	go.opentelemetry.io/collector/consumer/xconsumer v0.128.0 // indirect
 	go.opentelemetry.io/collector/exporter/xexporter v0.128.0 // indirect
 	go.opentelemetry.io/collector/extension v1.34.0 // indirect
-	go.opentelemetry.io/collector/featuregate v1.34.0 // indirect
 	go.opentelemetry.io/collector/internal/telemetry v0.128.0 // indirect
 	go.opentelemetry.io/collector/receiver v1.34.0 // indirect
 	go.opentelemetry.io/collector/receiver/receivertest v0.128.0 // indirect
@@ -116,3 +117,5 @@ replace go.opentelemetry.io/collector/confmap => ../confmap
 replace go.opentelemetry.io/collector/internal/telemetry => ../internal/telemetry
 
 replace go.opentelemetry.io/collector/client => ../client
+
+replace go.opentelemetry.io/collector/pdata/xpdata => ../pdata/xpdata
