diff --git a/vendor/github.com/census-instrumentation/opencensus-proto/build.gradle b/vendor/github.com/census-instrumentation/opencensus-proto/build.gradle
index d05ba4878..a2c8522d1 100644
--- a/vendor/github.com/census-instrumentation/opencensus-proto/build.gradle
+++ b/vendor/github.com/census-instrumentation/opencensus-proto/build.gradle
@@ -25,7 +25,7 @@ sourceSets {
         proto {
             // In addition to the default 'src/main/proto'
             srcDir 'opencensus/proto/stats'
-            srcDir 'opencensus/proto/stats/metrics'
+            srcDir 'opencensus/proto/metrics'
             srcDir 'opencensus/proto/trace'
         }
     }
