diff --git a/component/componenttest/doc.go b/component/componenttest/doc.go
index 176244611..b26d40711 100644
--- a/component/componenttest/doc.go
+++ b/component/componenttest/doc.go
@@ -15,3 +15,6 @@
 // Package componenttest define types and functions used to help test packages
 // implementing the component package interfaces.
 package componenttest
+
+// TODO: This file will be deleted after the checkdoc tool
+// is imported in contrib repo
