diff --git a/exporter/exporterhelper/queued_retry_experimental.go b/exporter/exporterhelper/queued_retry_experimental.go
index 9663c833c..4c03410e6 100644
--- a/exporter/exporterhelper/queued_retry_experimental.go
+++ b/exporter/exporterhelper/queued_retry_experimental.go
@@ -48,8 +48,11 @@ type QueueSettings struct {
 	PersistentStorageEnabled bool `mapstructure:"persistent_storage_enabled"`
 }
 
-// DefaultQueueSettings returns the default settings for QueueSettings.
-func DefaultQueueSettings() QueueSettings {
+// Deprecated: [v0.46.0] use NewDefaultQueueSettings instead.
+var DefaultQueueSettings = NewDefaultQueueSettings
+
+// NewDefaultQueueSettings returns the default settings for QueueSettings.
+func NewDefaultQueueSettings() QueueSettings {
 	return QueueSettings{
 		Enabled:      true,
 		NumConsumers: 10,
