diff --git a/translator/internaldata/resource_to_oc_test.go b/translator/internaldata/resource_to_oc_test.go
index 28620cfa0..d2e5ccd6b 100644
--- a/translator/internaldata/resource_to_oc_test.go
+++ b/translator/internaldata/resource_to_oc_test.go
@@ -36,7 +36,6 @@ import (
 
 func TestResourceToOC(t *testing.T) {
 	emptyResource := pdata.NewResource()
-	emptyResource.InitEmpty()
 
 	ocNode := generateOcNode()
 	ocResource := generateOcResource()
@@ -61,8 +60,8 @@ func TestResourceToOC(t *testing.T) {
 		{
 			name:       "empty",
 			resource:   emptyResource,
-			ocNode:     &occommon.Node{},
-			ocResource: &ocresource.Resource{},
+			ocNode:     nil,
+			ocResource: nil,
 		},
 
 		{
@@ -84,7 +83,6 @@ func TestResourceToOC(t *testing.T) {
 
 func TestContainerResourceToOC(t *testing.T) {
 	resource := pdata.NewResource()
-	resource.InitEmpty()
 	resource.Attributes().InitFromMap(map[string]pdata.AttributeValue{
 		conventions.AttributeK8sCluster:    pdata.NewAttributeValueString("cluster1"),
 		conventions.AttributeK8sPod:        pdata.NewAttributeValueString("pod1"),
@@ -243,22 +241,21 @@ func TestResourceToOCAndBack(t *testing.T) {
 			ocNode, ocResource := internalResourceToOC(expected)
 			actual := pdata.NewResource()
 			ocNodeResourceToInternal(ocNode, ocResource, actual)
-			if expected.IsNil() {
-				assert.True(t, actual.IsNil())
-			} else {
-				expected.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
-					a, ok := actual.Attributes().Get(k)
-					assert.True(t, ok)
-					switch v.Type() {
-					case pdata.AttributeValueINT:
-						assert.Equal(t, strconv.FormatInt(v.IntVal(), 10), a.StringVal())
-					case pdata.AttributeValueMAP, pdata.AttributeValueARRAY:
-						assert.Equal(t, a, a)
-					default:
-						assert.Equal(t, v, a)
-					}
-				})
-			}
+			// Remove opencensus resource type from actual. This will be added during translation.
+			actual.Attributes().Delete(conventions.OCAttributeResourceType)
+			assert.Equal(t, expected.Attributes().Len(), actual.Attributes().Len())
+			expected.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+				a, ok := actual.Attributes().Get(k)
+				assert.True(t, ok)
+				switch v.Type() {
+				case pdata.AttributeValueINT:
+					assert.Equal(t, strconv.FormatInt(v.IntVal(), 10), a.StringVal())
+				case pdata.AttributeValueMAP, pdata.AttributeValueARRAY:
+					assert.Equal(t, a, a)
+				default:
+					assert.Equal(t, v, a)
+				}
+			})
 		})
 	}
 }
