diff --git a/model/pdata/generated_metrics.go b/model/pdata/generated_metrics.go
index 0d15e16f9..047d1614f 100644
--- a/model/pdata/generated_metrics.go
+++ b/model/pdata/generated_metrics.go
@@ -975,6 +975,16 @@ func (ms NumberDataPoint) Exemplars() ExemplarSlice {
 	return newExemplarSlice(&(*ms.orig).Exemplars)
 }
 
+// Flags returns the flags associated with this NumberDataPoint.
+func (ms NumberDataPoint) Flags() MetricDataPointFlags {
+	return MetricDataPointFlags((*ms.orig).Flags)
+}
+
+// SetFlags replaces the flags associated with this NumberDataPoint.
+func (ms NumberDataPoint) SetFlags(v MetricDataPointFlags) {
+	(*ms.orig).Flags = uint32(v)
+}
+
 // CopyTo copies all properties from the current struct to the dest.
 func (ms NumberDataPoint) CopyTo(dest NumberDataPoint) {
 	ms.Attributes().CopyTo(dest.Attributes())
@@ -988,6 +998,7 @@ func (ms NumberDataPoint) CopyTo(dest NumberDataPoint) {
 	}
 
 	ms.Exemplars().CopyTo(dest.Exemplars())
+	dest.SetFlags(ms.Flags())
 }
 
 // HistogramDataPointSlice logically represents a slice of HistogramDataPoint.
@@ -1220,6 +1231,16 @@ func (ms HistogramDataPoint) Exemplars() ExemplarSlice {
 	return newExemplarSlice(&(*ms.orig).Exemplars)
 }
 
+// Flags returns the flags associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) Flags() MetricDataPointFlags {
+	return MetricDataPointFlags((*ms.orig).Flags)
+}
+
+// SetFlags replaces the flags associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) SetFlags(v MetricDataPointFlags) {
+	(*ms.orig).Flags = uint32(v)
+}
+
 // CopyTo copies all properties from the current struct to the dest.
 func (ms HistogramDataPoint) CopyTo(dest HistogramDataPoint) {
 	ms.Attributes().CopyTo(dest.Attributes())
@@ -1230,6 +1251,7 @@ func (ms HistogramDataPoint) CopyTo(dest HistogramDataPoint) {
 	dest.SetBucketCounts(ms.BucketCounts())
 	dest.SetExplicitBounds(ms.ExplicitBounds())
 	ms.Exemplars().CopyTo(dest.Exemplars())
+	dest.SetFlags(ms.Flags())
 }
 
 // SummaryDataPointSlice logically represents a slice of SummaryDataPoint.
@@ -1442,6 +1464,16 @@ func (ms SummaryDataPoint) QuantileValues() ValueAtQuantileSlice {
 	return newValueAtQuantileSlice(&(*ms.orig).QuantileValues)
 }
 
+// Flags returns the flags associated with this SummaryDataPoint.
+func (ms SummaryDataPoint) Flags() MetricDataPointFlags {
+	return MetricDataPointFlags((*ms.orig).Flags)
+}
+
+// SetFlags replaces the flags associated with this SummaryDataPoint.
+func (ms SummaryDataPoint) SetFlags(v MetricDataPointFlags) {
+	(*ms.orig).Flags = uint32(v)
+}
+
 // CopyTo copies all properties from the current struct to the dest.
 func (ms SummaryDataPoint) CopyTo(dest SummaryDataPoint) {
 	ms.Attributes().CopyTo(dest.Attributes())
@@ -1450,6 +1482,7 @@ func (ms SummaryDataPoint) CopyTo(dest SummaryDataPoint) {
 	dest.SetCount(ms.Count())
 	dest.SetSum(ms.Sum())
 	ms.QuantileValues().CopyTo(dest.QuantileValues())
+	dest.SetFlags(ms.Flags())
 }
 
 // ValueAtQuantileSlice logically represents a slice of ValueAtQuantile.
