diff --git a/extension/auth/client.go b/extension/auth/client.go
index e786c65bd..7e031eeaf 100644
--- a/extension/auth/client.go
+++ b/extension/auth/client.go
@@ -57,5 +57,9 @@ func WithClientPerRPCCredentials(perRPCCredentialsFunc ClientPerRPCCredentialsFu
 // NewClient returns a Client configured with the provided options.
 // Deprecated: [v0.121.0] Use extensionauth.NewClient instead.
 func NewClient(options ...ClientOption) Client {
-	return extensionauth.NewClient(options...)
+	client, err := extensionauth.NewClient(options...)
+	if err != nil {
+		panic("unexpected error when calling extensionauth.NewClient: " + err.Error())
+	}
+	return client
 }
