diff --git a/internal/telemetry/go.mod b/internal/telemetry/go.mod
index 057416cbe..7aea60f60 100644
--- a/internal/telemetry/go.mod
+++ b/internal/telemetry/go.mod
@@ -31,7 +31,7 @@ require (
 	golang.org/x/sys v0.32.0 // indirect
 	golang.org/x/text v0.24.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20250218202821-56aae31c358a // indirect
-	google.golang.org/grpc v1.72.0 // indirect
+	google.golang.org/grpc v1.72.1 // indirect
 	google.golang.org/protobuf v1.36.6 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
