diff --git a/pdata/pmetric/alias.go b/pdata/pmetric/alias.go
index 3594ce139..1f3e1cba1 100644
--- a/pdata/pmetric/alias.go
+++ b/pdata/pmetric/alias.go
@@ -57,34 +57,6 @@ type MetricDataPointFlagsStruct = internal.MetricDataPointFlagsStruct
 // NewMetricDataPointFlagsStruct returns a new empty MetricDataPointFlagsStruct.
 var NewMetricDataPointFlagsStruct = internal.NewMetricDataPointFlagsStruct
 
-// MetricDataPointFlags defines how a metric aggregator reports aggregated values.
-// It describes how those values relate to the time interval over which they are aggregated.
-// Deprecated: [v0.57.0] Use MetricDataPointFlagsStruct instead.
-type MetricDataPointFlags = internal.MetricDataPointFlags
-
-const (
-	// MetricDataPointFlagsNone is the default MetricDataPointFlags.
-	// Deprecated: [v0.57.0] Use MetricDataPointFlagsStruct.NoRecordedValue instead.
-	MetricDataPointFlagsNone = internal.MetricDataPointFlagsNone
-)
-
-// NewMetricDataPointFlags returns a new MetricDataPointFlags combining the flags passed
-// in as parameters.
-// Deprecated: [v0.57.0] Use NewMetricDataPointFlagsStruct and MetricDataPointFlagsStruct.SetNoRecordedValue instead.
-var NewMetricDataPointFlags = internal.NewMetricDataPointFlags
-
-// MetricDataPointFlag allow users to configure DataPointFlags. This is achieved via NewMetricDataPointFlags.
-// The separation between MetricDataPointFlags and MetricDataPointFlag exists to prevent users accidentally
-// comparing the value of individual flags with MetricDataPointFlags. Instead, users must use the HasFlag method.
-// Deprecated: [v0.57.0] Use MetricDataPointFlagsStruct instead.
-type MetricDataPointFlag = internal.MetricDataPointFlag
-
-const (
-	// MetricDataPointFlagNoRecordedValue is flag for a metric aggregator which reports changes since last report time.
-	// Deprecated: [v0.57.0] Use MetricDataPointFlagsStruct.NoRecordedValue instead.
-	MetricDataPointFlagNoRecordedValue = internal.MetricDataPointFlagNoRecordedValue
-)
-
 // NumberDataPointValueType specifies the type of NumberDataPoint value.
 type NumberDataPointValueType = internal.NumberDataPointValueType
 
