diff --git a/receiver/factory.go b/receiver/factory.go
index adde007b4..6c6231a41 100644
--- a/receiver/factory.go
+++ b/receiver/factory.go
@@ -21,8 +21,8 @@ import (
 	"github.com/spf13/viper"
 	"go.uber.org/zap"
 
-	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
-	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
 )
 
 // Factory is factory interface for receivers.
