diff --git a/pdata/pmetric/pmetricotlp/metrics.go b/pdata/pmetric/pmetricotlp/metrics.go
index e911ba841..3c461d78c 100644
--- a/pdata/pmetric/pmetricotlp/metrics.go
+++ b/pdata/pmetric/pmetricotlp/metrics.go
@@ -151,11 +151,14 @@ type GRPCServer interface {
 // Deprecated: [0.61.0] Use GRPCServer instead
 type Server = GRPCServer
 
-// RegisterServer registers the GRPCServer to the grpc.Server.
-func RegisterServer(s *grpc.Server, srv GRPCServer) {
+// RegisterGRPCServer registers the GRPCServer to the grpc.Server.
+func RegisterGRPCServer(s *grpc.Server, srv GRPCServer) {
 	otlpcollectormetrics.RegisterMetricsServiceServer(s, &rawMetricsServer{srv: srv})
 }
 
+// Deprecated: [0.62.0] Use RegisterGRPCServer instead
+var RegisterServer = RegisterGRPCServer
+
 type rawMetricsServer struct {
 	srv GRPCServer
 }
