diff --git a/service/featuregate/gates.go b/service/featuregate/gates.go
index 47d14f934..68869aac8 100644
--- a/service/featuregate/gates.go
+++ b/service/featuregate/gates.go
@@ -27,40 +27,40 @@ type Gate struct {
 	Enabled     bool
 }
 
-var reg = &registry{gates: make(map[string]Gate)}
+var reg = NewRegistry()
 
-// IsEnabled returns true if a registered feature gate is enabled and false otherwise.
-func IsEnabled(id string) bool {
-	return reg.isEnabled(id)
+// GetRegistry returns the global Registry.
+func GetRegistry() *Registry {
+	return reg
 }
 
-// List returns a slice of copies of all registered Gates.
-func List() []Gate {
-	return reg.list()
-}
+// Deprecated: [v0.50.0] Use GetRegistry().Apply.
+var Apply = GetRegistry().Apply
 
-// Register a Gate. May only be called in an init() function.
-// Will panic() if a Gate with the same ID is already registered.
-func Register(g Gate) {
-	if err := reg.add(g); err != nil {
-		panic(err)
-	}
-}
+// Deprecated: [v0.50.0] Use GetRegistry().IsEnabled.
+var IsEnabled = GetRegistry().IsEnabled
 
-// Apply a configuration in the form of a map of Gate identifiers to boolean values.
-// Sets only those values provided in the map, other gate values are not changed.
-func Apply(cfg map[string]bool) {
-	reg.apply(cfg)
+// Deprecated: [v0.50.0] Use GetRegistry().List.
+var List = GetRegistry().List
+
+// Deprecated: [v0.50.0] Use GetRegistry().MustRegister.
+var Register = GetRegistry().MustRegister
+
+// NewRegistry returns a new empty Registry.
+func NewRegistry() *Registry {
+	return &Registry{gates: make(map[string]Gate)}
 }
 
-type registry struct {
-	sync.RWMutex
+type Registry struct {
+	mu    sync.RWMutex
 	gates map[string]Gate
 }
 
-func (r *registry) apply(cfg map[string]bool) {
-	r.Lock()
-	defer r.Unlock()
+// Apply a configuration in the form of a map of Gate identifiers to boolean values.
+// Sets only those values provided in the map, other gate values are not changed.
+func (r *Registry) Apply(cfg map[string]bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
 	for id, val := range cfg {
 		if g, ok := r.gates[id]; ok {
 			g.Enabled = val
@@ -69,27 +69,36 @@ func (r *registry) apply(cfg map[string]bool) {
 	}
 }
 
-func (r *registry) add(g Gate) error {
-	r.Lock()
-	defer r.Unlock()
+// IsEnabled returns true if a registered feature gate is enabled and false otherwise.
+func (r *Registry) IsEnabled(id string) bool {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	g, ok := r.gates[id]
+	return ok && g.Enabled
+}
+
+// MustRegister like Register but panics if a Gate with the same ID is already registered.
+func (r *Registry) MustRegister(g Gate) {
+	if err := r.Register(g); err != nil {
+		panic(err)
+	}
+}
+
+// Register registers a Gate. May only be called in an init() function.
+func (r *Registry) Register(g Gate) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
 	if _, ok := r.gates[g.ID]; ok {
 		return fmt.Errorf("attempted to add pre-existing gate %q", g.ID)
 	}
-
 	r.gates[g.ID] = g
 	return nil
 }
 
-func (r *registry) isEnabled(id string) bool {
-	r.RLock()
-	defer r.RUnlock()
-	g, ok := r.gates[id]
-	return ok && g.Enabled
-}
-
-func (r *registry) list() []Gate {
-	r.RLock()
-	defer r.RUnlock()
+// List returns a slice of copies of all registered Gates.
+func (r *Registry) List() []Gate {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
 	ret := make([]Gate, len(r.gates))
 	i := 0
 	for _, gate := range r.gates {
