diff --git a/exporter/loggingexporter/go.mod b/exporter/loggingexporter/go.mod
index 2aa4bf9b2..efac4afb6 100644
--- a/exporter/loggingexporter/go.mod
+++ b/exporter/loggingexporter/go.mod
@@ -59,8 +59,6 @@ replace go.opentelemetry.io/collector/exporter => ../
 
 replace go.opentelemetry.io/collector/extension => ../../extension
 
-replace go.opentelemetry.io/collector/extension/auth => ../../extension/auth
-
 replace go.opentelemetry.io/collector/featuregate => ../../featuregate
 
 replace go.opentelemetry.io/collector/pdata => ../../pdata
@@ -80,14 +78,6 @@ retract (
 
 replace go.opentelemetry.io/collector/connector => ../../connector
 
-replace go.opentelemetry.io/collector/config/configcompression => ../../config/configcompression
-
 replace go.opentelemetry.io/collector/config/confignet => ../../config/confignet
 
-replace go.opentelemetry.io/collector/config/configopaque => ../../config/configopaque
-
-replace go.opentelemetry.io/collector/config/configtls => ../../config/configtls
-
 replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/configtelemetry
-
-replace go.opentelemetry.io/collector/config/configauth => ../../config/configauth
