diff --git a/internal/goldendataset/span_generator.go b/internal/goldendataset/span_generator.go
index dc96657d3..0da00200d 100644
--- a/internal/goldendataset/span_generator.go
+++ b/internal/goldendataset/span_generator.go
@@ -85,7 +85,7 @@ func GenerateSpans(count int, startPos int, pictFile string, random io.Reader) (
 	var inputs []string
 	var spanInputs *PICTSpanInputs
 	var traceID otlpcommon.TraceID
-	var parentID []byte
+	var parentID otlpcommon.SpanID
 	for i := 0; i < count; i++ {
 		if index >= pairsTotal {
 			index = 1
@@ -103,13 +103,13 @@ func GenerateSpans(count int, startPos int, pictFile string, random io.Reader) (
 		switch spanInputs.Parent {
 		case SpanParentRoot:
 			traceID = generateTraceID(random)
-			parentID = nil
+			parentID = otlpcommon.NewSpanID(nil)
 		case SpanParentChild:
 			// use existing if available
 			if traceID.Bytes() == nil {
 				traceID = generateTraceID(random)
 			}
-			if parentID == nil {
+			if parentID.Bytes() == nil {
 				parentID = generateSpanID(random)
 			}
 		}
@@ -134,7 +134,7 @@ func generateSpanName(spanInputs *PICTSpanInputs) string {
 //   random - the random number generator to use in generating ID values
 //
 // The generated span is returned.
-func GenerateSpan(traceID otlpcommon.TraceID, parentID []byte, spanName string, spanInputs *PICTSpanInputs,
+func GenerateSpan(traceID otlpcommon.TraceID, parentID otlpcommon.SpanID, spanName string, spanInputs *PICTSpanInputs,
 	random io.Reader) *otlptrace.Span {
 	endTime := time.Now().Add(-50 * time.Microsecond)
 	return &otlptrace.Span{
