diff --git a/go.mod b/go.mod
index 205d476d4..83aafd4f0 100644
--- a/go.mod
+++ b/go.mod
@@ -32,6 +32,7 @@ require (
 	github.com/jaegertracing/jaeger v1.19.2
 	github.com/joshdk/go-junit v0.0.0-20200702055522-6efcf4050909
 	github.com/jstemmer/go-junit-report v0.9.1
+	github.com/mitchellh/mapstructure v1.3.2
 	github.com/mjibson/esc v0.2.0
 	github.com/openzipkin/zipkin-go v0.2.4-0.20200818204336-dc18516bbb4c
 	github.com/orijtech/prometheus-go-metrics-exporter v0.0.5
