diff --git a/receiver/otlpreceiver/trace/otlp.go b/receiver/otlpreceiver/trace/otlp.go
index 851a8275b..6dd2d3cf8 100644
--- a/receiver/otlpreceiver/trace/otlp.go
+++ b/receiver/otlpreceiver/trace/otlp.go
@@ -55,26 +55,22 @@ func (r *Receiver) Export(ctx context.Context, req *collectortrace.ExportTraceSe
 	ctxWithReceiverName := obsreport.ReceiverContext(ctx, r.instanceName, receiverTransport)
 
 	// Perform backward compatibility conversion of Span Status code according to
-	// OTLP specification.
-	// See https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L231
-	//
-	//   If code==STATUS_CODE_UNSET then the value of `deprecated_code` is the
-	//   carrier of the overall status according to these rules:
-	//
-	//     if deprecated_code==DEPRECATED_STATUS_CODE_OK then the receiver MUST interpret
-	//     the overall status to be STATUS_CODE_UNSET.
-	//
-	//     if deprecated_code!=DEPRECATED_STATUS_CODE_OK then the receiver MUST interpret
-	//     the overall status to be STATUS_CODE_ERROR.
-	//
-	//   If code!=STATUS_CODE_UNSET then the value of `deprecated_code` MUST be
-	//   ignored, the `code` field is the sole carrier of the status.
+	// OTLP specification as we are a new receiver and sender (we are pushing data to the pipelines):
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L239
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L253
 	for _, rss := range req.ResourceSpans {
 		for _, ils := range rss.InstrumentationLibrarySpans {
 			for _, span := range ils.Spans {
-				if span.Status.Code == otlptrace.Status_STATUS_CODE_UNSET &&
-					span.Status.DeprecatedCode != otlptrace.Status_DEPRECATED_STATUS_CODE_OK {
-					span.Status.Code = otlptrace.Status_STATUS_CODE_ERROR
+				switch span.Status.Code {
+				case otlptrace.Status_STATUS_CODE_UNSET:
+					if span.Status.DeprecatedCode != otlptrace.Status_DEPRECATED_STATUS_CODE_OK {
+						span.Status.Code = otlptrace.Status_STATUS_CODE_ERROR
+					}
+				case otlptrace.Status_STATUS_CODE_OK:
+					// If status code is set then overwrites deprecated.
+					span.Status.DeprecatedCode = otlptrace.Status_DEPRECATED_STATUS_CODE_OK
+				case otlptrace.Status_STATUS_CODE_ERROR:
+					span.Status.DeprecatedCode = otlptrace.Status_DEPRECATED_STATUS_CODE_UNKNOWN_ERROR
 				}
 			}
 		}
