diff --git a/internal/fanoutconsumer/package_test.go b/internal/fanoutconsumer/package_test.go
new file mode 100644
index 000000000..b2da7cedc
--- /dev/null
+++ b/internal/fanoutconsumer/package_test.go
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package fanoutconsumer
+
+import (
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
