diff --git a/translator/trace/jaeger/jaegerproto_to_traces.go b/translator/trace/jaeger/jaegerproto_to_traces.go
index 1cf2d8068..c6945d8bf 100644
--- a/translator/trace/jaeger/jaegerproto_to_traces.go
+++ b/translator/trace/jaeger/jaegerproto_to_traces.go
@@ -146,7 +146,6 @@ func jSpansToInternal(spans []*model.Span) map[instrumentationLibrary]pdata.Inst
 		ils, found := spansByLibrary[library]
 		if !found {
 			ils = pdata.NewInstrumentationLibrarySpans()
-			ils.InitEmpty()
 			spansByLibrary[library] = ils
 
 			if library.name != "" {
@@ -165,7 +164,6 @@ type instrumentationLibrary struct {
 
 func jSpanToInternal(span *model.Span) (pdata.Span, instrumentationLibrary) {
 	dest := pdata.NewSpan()
-	dest.InitEmpty()
 	dest.SetTraceID(tracetranslator.UInt64ToTraceID(span.TraceID.High, span.TraceID.Low))
 	dest.SetSpanID(tracetranslator.UInt64ToSpanID(uint64(span.SpanID)))
 	dest.SetName(span.OperationName)
