diff --git a/config/configmodels/configmodels.go b/config/configmodels/configmodels.go
index 1451fe466..3084eab5b 100644
--- a/config/configmodels/configmodels.go
+++ b/config/configmodels/configmodels.go
@@ -46,7 +46,6 @@ type Type string
 // NamedEntity is a configuration entity that has a type and a name.
 type NamedEntity interface {
 	Type() Type
-	SetType(typeStr Type)
 	Name() string
 	SetName(name string)
 }
@@ -151,11 +150,6 @@ func (rs *ReceiverSettings) Type() Type {
 	return rs.TypeVal
 }
 
-// SetType sets the receiver type.
-func (rs *ReceiverSettings) SetType(typeStr Type) {
-	rs.TypeVal = typeStr
-}
-
 // ExporterSettings defines common settings for an exporter configuration.
 // Specific exporters can embed this struct and extend it with more fields if needed.
 type ExporterSettings struct {
@@ -180,11 +174,6 @@ func (es *ExporterSettings) Type() Type {
 	return es.TypeVal
 }
 
-// SetType sets the exporter type.
-func (es *ExporterSettings) SetType(typeStr Type) {
-	es.TypeVal = typeStr
-}
-
 // ProcessorSettings defines common settings for a processor configuration.
 // Specific processors can embed this struct and extend it with more fields if needed.
 type ProcessorSettings struct {
@@ -207,11 +196,6 @@ func (proc *ProcessorSettings) Type() Type {
 	return proc.TypeVal
 }
 
-// SetType sets the processor type.
-func (proc *ProcessorSettings) SetType(typeStr Type) {
-	proc.TypeVal = typeStr
-}
-
 var _ Processor = (*ProcessorSettings)(nil)
 
 // ExtensionSettings defines common settings for a service extension configuration.
@@ -236,9 +220,4 @@ func (ext *ExtensionSettings) Type() Type {
 	return ext.TypeVal
 }
 
-// SetType sets the extension type.
-func (ext *ExtensionSettings) SetType(typeStr Type) {
-	ext.TypeVal = typeStr
-}
-
 var _ Extension = (*ExtensionSettings)(nil)
