diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 1fffb04e5..c3c4dee4d 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -10,7 +10,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/client v1.19.0
 	go.opentelemetry.io/collector/component v0.113.0
-	go.opentelemetry.io/collector/component/componenttest v0.113.0
+	go.opentelemetry.io/collector/component/componenttest v0.0.0-20241112045949-9c4c007a12b4
 	go.opentelemetry.io/collector/config/configauth v0.113.0
 	go.opentelemetry.io/collector/config/configcompression v1.19.0
 	go.opentelemetry.io/collector/config/configopaque v1.19.0
