diff --git a/exporter/loggingexporter/config.go b/exporter/loggingexporter/config.go
index 530551e2d..5ef5772b3 100644
--- a/exporter/loggingexporter/config.go
+++ b/exporter/loggingexporter/config.go
@@ -20,7 +20,6 @@ import (
 	"go.uber.org/zap/zapcore"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configtelemetry"
 	"go.opentelemetry.io/collector/confmap"
 )
@@ -37,8 +36,6 @@ var (
 
 // Config defines configuration for logging exporter.
 type Config struct {
-	// Deprecated: [v0.68.0] will be removed soon.
-	config.ExporterSettings `mapstructure:",squash"`
 	// LogLevel defines log level of the logging exporter; options are debug, info, warn, error.
 	// Deprecated: Use `Verbosity` instead.
 	LogLevel zapcore.Level `mapstructure:"loglevel"`
