diff --git a/receiver/zipkinreceiver/config.go b/receiver/zipkinreceiver/config.go
index 45367f5b3..54e354dfd 100644
--- a/receiver/zipkinreceiver/config.go
+++ b/receiver/zipkinreceiver/config.go
@@ -14,9 +14,9 @@
 
 package zipkinreceiver
 
-import "github.com/open-telemetry/opentelemetry-service/pkg/configmodels"
+import "github.com/open-telemetry/opentelemetry-service/models"
 
 // ConfigV2 defines configuration for Zipkin receiver.
 type ConfigV2 struct {
-	configmodels.ReceiverSettings `mapstructure:",squash"`
+	models.ReceiverSettings `mapstructure:",squash"`
 }
