diff --git a/receiver/prometheusreceiver/config_test.go b/receiver/prometheusreceiver/config_test.go
index 4a8d73c4d..0affb5a19 100644
--- a/receiver/prometheusreceiver/config_test.go
+++ b/receiver/prometheusreceiver/config_test.go
@@ -26,7 +26,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/component/componenttest"
-	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configtest"
 )
 
@@ -47,7 +47,7 @@ func TestLoadConfig(t *testing.T) {
 
 	r1 := cfg.Receivers["prometheus/customname"].(*Config)
 	assert.Equal(t, r1.ReceiverSettings,
-		configmodels.ReceiverSettings{
+		config.ReceiverSettings{
 			TypeVal: typeStr,
 			NameVal: "prometheus/customname",
 		})
@@ -73,7 +73,7 @@ func TestLoadConfigWithEnvVar(t *testing.T) {
 
 	r := cfg.Receivers["prometheus"].(*Config)
 	assert.Equal(t, r.ReceiverSettings,
-		configmodels.ReceiverSettings{
+		config.ReceiverSettings{
 			TypeVal: typeStr,
 			NameVal: "prometheus",
 		})
@@ -98,7 +98,7 @@ func TestLoadConfigK8s(t *testing.T) {
 
 	r := cfg.Receivers["prometheus"].(*Config)
 	assert.Equal(t, r.ReceiverSettings,
-		configmodels.ReceiverSettings{
+		config.ReceiverSettings{
 			TypeVal: typeStr,
 			NameVal: "prometheus",
 		})
