diff --git a/exporter/loggingexporter/doc.go b/exporter/loggingexporter/doc.go
index d6652178b..4465a6e90 100644
--- a/exporter/loggingexporter/doc.go
+++ b/exporter/loggingexporter/doc.go
@@ -13,4 +13,4 @@
 // limitations under the License.
 
 // Package loggingexporter exports data to console as logs.
-package loggingexporter
+package loggingexporter // import "go.opentelemetry.io/collector/exporter/loggingexporter"
