diff --git a/service/telemetry/logger_test.go b/service/telemetry/logger_test.go
index 7b162d1fe..1cf4f1441 100644
--- a/service/telemetry/logger_test.go
+++ b/service/telemetry/logger_test.go
@@ -104,16 +104,18 @@ func TestNewLogger(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			sdk, _ := config.NewSDK(config.WithOpenTelemetryConfiguration(config.OpenTelemetryConfiguration{LoggerProvider: &config.LoggerProvider{
-				Processors: tt.cfg.Logs.Processors,
-			}}))
+			buildInfo := component.BuildInfo{}
+			sdk, err := NewSDK(context.Background(), &tt.cfg, resource.New(buildInfo, nil))
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, sdk.Shutdown(context.Background()))
+			}()
 
-			_, _, err := newLogger(Settings{SDK: &sdk}, tt.cfg)
+			_, _, err = newLogger(Settings{SDK: sdk}, tt.cfg)
 			if tt.wantErr != nil {
 				require.ErrorContains(t, err, tt.wantErr.Error())
 			} else {
 				require.NoError(t, err)
-				assert.NoError(t, sdk.Shutdown(context.Background()))
 			}
 		})
 	}
@@ -313,29 +315,19 @@ func newOTLPLoggerProvider(t *testing.T, level zapcore.Level, handler http.Handl
 		},
 	}
 
-	sdk, err := config.NewSDK(
-		config.WithOpenTelemetryConfiguration(
-			config.OpenTelemetryConfiguration{
-				LoggerProvider: &config.LoggerProvider{
-					Processors: processors,
-				},
-				Resource: &config.Resource{
-					SchemaUrl: ptr(""),
-					Attributes: []config.AttributeNameValue{
-						{Name: string(semconv.ServiceNameKey), Value: service},
-						{Name: string(semconv.ServiceVersionKey), Value: version},
-						{Name: testAttribute, Value: testValue},
-					},
-				},
-			},
-		),
-	)
+	buildInfo := component.BuildInfo{}
+	res := resource.New(buildInfo, map[string]*string{
+		string(semconv.ServiceNameKey):    ptr(service),
+		string(semconv.ServiceVersionKey): ptr(version),
+		testAttribute:                     ptr(testValue),
+	})
+	sdk, err := NewSDK(context.Background(), &cfg, res)
 	require.NoError(t, err)
 	t.Cleanup(func() {
 		assert.NoError(t, sdk.Shutdown(context.Background()))
 	})
 
-	_, lp, err := newLogger(Settings{SDK: &sdk}, cfg)
+	_, lp, err := newLogger(Settings{SDK: sdk}, cfg)
 	require.NoError(t, err)
 	require.NotNil(t, lp)
 	return lp
