diff --git a/exporter/internal/queue/batcher.go b/exporter/internal/queue/batcher.go
index 9149dda66..2f728b073 100644
--- a/exporter/internal/queue/batcher.go
+++ b/exporter/internal/queue/batcher.go
@@ -28,14 +28,11 @@ type BaseBatcher struct {
 	batchCfg   exporterbatcher.Config
 	queue      Queue[internal.Request]
 	maxWorkers int
+	workerPool chan bool
 	stopWG     sync.WaitGroup
 }
 
 func NewBatcher(batchCfg exporterbatcher.Config, queue Queue[internal.Request], maxWorkers int) (Batcher, error) {
-	if maxWorkers != 0 {
-		return nil, errors.ErrUnsupported
-	}
-
 	if batchCfg.Enabled {
 		return nil, errors.ErrUnsupported
 	}
@@ -50,6 +47,16 @@ func NewBatcher(batchCfg exporterbatcher.Config, queue Queue[internal.Request],
 	}, nil
 }
 
+func (qb *BaseBatcher) startWorkerPool() {
+	if qb.maxWorkers == 0 {
+		return
+	}
+	qb.workerPool = make(chan bool, qb.maxWorkers)
+	for i := 0; i < qb.maxWorkers; i++ {
+		qb.workerPool <- true
+	}
+}
+
 // flush exports the incoming batch synchronously.
 func (qb *BaseBatcher) flush(batchToFlush batch) {
 	err := batchToFlush.req.Export(batchToFlush.ctx)
@@ -61,9 +68,18 @@ func (qb *BaseBatcher) flush(batchToFlush batch) {
 // flushAsync starts a goroutine that calls flushIfNecessary. It blocks until a worker is available.
 func (qb *BaseBatcher) flushAsync(batchToFlush batch) {
 	qb.stopWG.Add(1)
+	if qb.maxWorkers == 0 {
+		go func() {
+			defer qb.stopWG.Done()
+			qb.flush(batchToFlush)
+		}()
+		return
+	}
+	<-qb.workerPool
 	go func() {
 		defer qb.stopWG.Done()
 		qb.flush(batchToFlush)
+		qb.workerPool <- true
 	}()
 }
 
