diff --git a/pdata/testdata/go.mod b/pdata/testdata/go.mod
index 4d7e6ec45..a8f417aa1 100644
--- a/pdata/testdata/go.mod
+++ b/pdata/testdata/go.mod
@@ -3,8 +3,8 @@ module go.opentelemetry.io/collector/pdata/testdata
 go 1.23.0
 
 require (
-	go.opentelemetry.io/collector/pdata v1.25.0
-	go.opentelemetry.io/collector/pdata/pprofile v0.119.0
+	go.opentelemetry.io/collector/pdata v1.26.0
+	go.opentelemetry.io/collector/pdata/pprofile v0.120.0
 )
 
 require (
