diff --git a/receiver/otlpreceiver/otlp.go b/receiver/otlpreceiver/otlp.go
index 9b5c17502..0f66905e7 100644
--- a/receiver/otlpreceiver/otlp.go
+++ b/receiver/otlpreceiver/otlp.go
@@ -250,18 +250,18 @@ func (r *Receiver) startServer(host component.Host) error {
 
 		httpL := m.Match(cmux.Any())
 		go func() {
-			if err := r.serverGRPC.Serve(grpcL); err != nil {
-				host.ReportFatalError(err)
+			if errGrpc := r.serverGRPC.Serve(grpcL); errGrpc != nil {
+				host.ReportFatalError(errGrpc)
 			}
 		}()
 		go func() {
-			if err := r.httpServer().Serve(httpL); err != nil {
-				host.ReportFatalError(err)
+			if errHTTP := r.httpServer().Serve(httpL); errHTTP != nil {
+				host.ReportFatalError(errHTTP)
 			}
 		}()
 		go func() {
-			if err := m.Serve(); err != nil {
-				host.ReportFatalError(err)
+			if errServe := m.Serve(); errServe != nil {
+				host.ReportFatalError(errServe)
 			}
 		}()
 	})
