diff --git a/connector/forwardconnector/forward.go b/connector/forwardconnector/forward.go
index 83914a05f..1a9bc2344 100644
--- a/connector/forwardconnector/forward.go
+++ b/connector/forwardconnector/forward.go
@@ -9,17 +9,20 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/connector/forwardconnector/internal/metadata"
+	"go.opentelemetry.io/collector/connector/xconnector"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
 )
 
 // NewFactory returns a connector.Factory.
-func NewFactory() connector.Factory {
-	return connector.NewFactory(
+func NewFactory() xconnector.Factory {
+	return xconnector.NewFactory(
 		metadata.Type,
 		createDefaultConfig,
-		connector.WithTracesToTraces(createTracesToTraces, metadata.TracesToTracesStability),
-		connector.WithMetricsToMetrics(createMetricsToMetrics, metadata.MetricsToMetricsStability),
-		connector.WithLogsToLogs(createLogsToLogs, metadata.LogsToLogsStability),
+		xconnector.WithTracesToTraces(createTracesToTraces, metadata.TracesToTracesStability),
+		xconnector.WithMetricsToMetrics(createMetricsToMetrics, metadata.MetricsToMetricsStability),
+		xconnector.WithLogsToLogs(createLogsToLogs, metadata.LogsToLogsStability),
+		xconnector.WithProfilesToProfiles(createProfilesToProfiles, metadata.ProfilesToProfilesStability),
 	)
 }
 
@@ -60,6 +63,16 @@ func createLogsToLogs(
 	return &forward{Logs: nextConsumer}, nil
 }
 
+// createProfilesToProfiles creates a profile receiver based on provided config.
+func createProfilesToProfiles(
+	_ context.Context,
+	_ connector.Settings,
+	_ component.Config,
+	nextConsumer xconsumer.Profiles,
+) (xconnector.Profiles, error) {
+	return &forward{Profiles: nextConsumer}, nil
+}
+
 // forward is used to pass signals directly from one pipeline to another.
 // This is useful when there is a need to replicate data and process it in more
 // than one way. It can also be used to join pipelines together.
@@ -67,6 +80,7 @@ type forward struct {
 	consumer.Traces
 	consumer.Metrics
 	consumer.Logs
+	xconsumer.Profiles
 	component.StartFunc
 	component.ShutdownFunc
 }
