diff --git a/exporter/prometheusexporter/accumulator.go b/exporter/prometheusexporter/accumulator.go
index f4c356adf..d2af0dd63 100644
--- a/exporter/prometheusexporter/accumulator.go
+++ b/exporter/prometheusexporter/accumulator.go
@@ -123,7 +123,7 @@ func (a *lastValueAccumulator) accumulateSummary(metric pdata.Metric, il pdata.I
 		}
 
 		mm := createMetric(metric)
-		mm.Summary().DataPoints().Append(ip)
+		ip.CopyTo(mm.Summary().DataPoints().AppendEmpty())
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: mm, instrumentationLibrary: il, updated: now})
 		n++
 	}
@@ -141,7 +141,7 @@ func (a *lastValueAccumulator) accumulateIntGauge(metric pdata.Metric, il pdata.
 		v, ok := a.registeredMetrics.Load(signature)
 		if !ok {
 			m := createMetric(metric)
-			m.IntGauge().DataPoints().Append(ip)
+			ip.CopyTo(m.IntGauge().DataPoints().AppendEmpty())
 			a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 			n++
 			continue
@@ -154,7 +154,7 @@ func (a *lastValueAccumulator) accumulateIntGauge(metric pdata.Metric, il pdata.
 		}
 
 		m := createMetric(metric)
-		m.IntGauge().DataPoints().Append(ip)
+		ip.CopyTo(m.IntGauge().DataPoints().AppendEmpty())
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 		n++
 	}
@@ -171,7 +171,7 @@ func (a *lastValueAccumulator) accumulateDoubleGauge(metric pdata.Metric, il pda
 		v, ok := a.registeredMetrics.Load(signature)
 		if !ok {
 			m := createMetric(metric)
-			m.DoubleGauge().DataPoints().Append(ip)
+			ip.CopyTo(m.DoubleGauge().DataPoints().AppendEmpty())
 			a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 			n++
 			continue
@@ -184,7 +184,7 @@ func (a *lastValueAccumulator) accumulateDoubleGauge(metric pdata.Metric, il pda
 		}
 
 		m := createMetric(metric)
-		m.DoubleGauge().DataPoints().Append(ip)
+		ip.CopyTo(m.DoubleGauge().DataPoints().AppendEmpty())
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 		n++
 	}
@@ -210,7 +210,7 @@ func (a *lastValueAccumulator) accumulateIntSum(metric pdata.Metric, il pdata.In
 			m := createMetric(metric)
 			m.IntSum().SetIsMonotonic(metric.IntSum().IsMonotonic())
 			m.IntSum().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
-			m.IntSum().DataPoints().Append(ip)
+			ip.CopyTo(m.IntSum().DataPoints().AppendEmpty())
 			a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 			n++
 			continue
@@ -225,7 +225,7 @@ func (a *lastValueAccumulator) accumulateIntSum(metric pdata.Metric, il pdata.In
 		m := createMetric(metric)
 		m.IntSum().SetIsMonotonic(metric.IntSum().IsMonotonic())
 		m.IntSum().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
-		m.IntSum().DataPoints().Append(ip)
+		ip.CopyTo(m.IntSum().DataPoints().AppendEmpty())
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 		n++
 	}
@@ -251,7 +251,7 @@ func (a *lastValueAccumulator) accumulateDoubleSum(metric pdata.Metric, il pdata
 			m := createMetric(metric)
 			m.DoubleSum().SetIsMonotonic(metric.DoubleSum().IsMonotonic())
 			m.DoubleSum().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
-			m.DoubleSum().DataPoints().Append(ip)
+			ip.CopyTo(m.DoubleSum().DataPoints().AppendEmpty())
 			a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 			n++
 			continue
@@ -266,7 +266,7 @@ func (a *lastValueAccumulator) accumulateDoubleSum(metric pdata.Metric, il pdata
 		m := createMetric(metric)
 		m.DoubleSum().SetIsMonotonic(metric.DoubleSum().IsMonotonic())
 		m.DoubleSum().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
-		m.DoubleSum().DataPoints().Append(ip)
+		ip.CopyTo(m.DoubleSum().DataPoints().AppendEmpty())
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 		n++
 	}
@@ -290,7 +290,7 @@ func (a *lastValueAccumulator) accumulateIntHistogram(metric pdata.Metric, il pd
 		v, ok := a.registeredMetrics.Load(signature)
 		if !ok {
 			m := createMetric(metric)
-			m.IntHistogram().DataPoints().Append(ip)
+			ip.CopyTo(m.IntHistogram().DataPoints().AppendEmpty())
 			a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 			n++
 			continue
@@ -303,7 +303,7 @@ func (a *lastValueAccumulator) accumulateIntHistogram(metric pdata.Metric, il pd
 		}
 
 		m := createMetric(metric)
-		m.IntHistogram().DataPoints().Append(ip)
+		ip.CopyTo(m.IntHistogram().DataPoints().AppendEmpty())
 		m.IntHistogram().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 		n++
@@ -328,7 +328,7 @@ func (a *lastValueAccumulator) accumulateDoubleHistogram(metric pdata.Metric, il
 		v, ok := a.registeredMetrics.Load(signature)
 		if !ok {
 			m := createMetric(metric)
-			m.Histogram().DataPoints().Append(ip)
+			ip.CopyTo(m.Histogram().DataPoints().AppendEmpty())
 			a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 			n++
 			continue
@@ -341,7 +341,7 @@ func (a *lastValueAccumulator) accumulateDoubleHistogram(metric pdata.Metric, il
 		}
 
 		m := createMetric(metric)
-		m.Histogram().DataPoints().Append(ip)
+		ip.CopyTo(m.Histogram().DataPoints().AppendEmpty())
 		m.Histogram().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
 		a.registeredMetrics.Store(signature, &accumulatedValue{value: m, instrumentationLibrary: il, updated: now})
 		n++
