diff --git a/service/internal/graph/graph.go b/service/internal/graph/graph.go
index ee39954a3..97fa45af6 100644
--- a/service/internal/graph/graph.go
+++ b/service/internal/graph/graph.go
@@ -310,6 +310,12 @@ func (g *Graph) ShutdownAll(ctx context.Context) error {
 	return errs
 }
 
+// Deprecated: [0.79.0] This function will be removed in the future.
+// Several components in the contrib repository use this function so it cannot be removed
+// before those cases are removed. In most cases, use of this function can be replaced by a
+// connector. See https://github.com/open-telemetry/opentelemetry-collector/issues/7370 and
+// https://github.com/open-telemetry/opentelemetry-collector/pull/7390#issuecomment-1483710184
+// for additional information.
 func (g *Graph) GetExporters() map[component.DataType]map[component.ID]component.Component {
 	exportersMap := make(map[component.DataType]map[component.ID]component.Component)
 	exportersMap[component.DataTypeTraces] = make(map[component.ID]component.Component)
