diff --git a/factories/factories.go b/factories/factories.go
index 41998c97e..91aaca1bb 100644
--- a/factories/factories.go
+++ b/factories/factories.go
@@ -47,13 +47,13 @@ type ReceiverFactory interface {
 	// CreateTraceReceiver creates a trace receiver based on this config.
 	// If the receiver type does not support tracing or if the config is not valid
 	// error will be returned instead.
-	CreateTraceReceiver(ctx context.Context, cfg models.Receiver,
+	CreateTraceReceiver(ctx context.Context, logger *zap.Logger, cfg models.Receiver,
 		nextConsumer consumer.TraceConsumer) (receiver.TraceReceiver, error)
 
 	// CreateMetricsReceiver creates a metrics receiver based on this config.
 	// If the receiver type does not support metrics or if the config is not valid
 	// error will be returned instead.
-	CreateMetricsReceiver(cfg models.Receiver,
+	CreateMetricsReceiver(logger *zap.Logger, cfg models.Receiver,
 		consumer consumer.MetricsConsumer) (receiver.MetricsReceiver, error)
 }
 
