diff --git a/service/service.go b/service/service.go
index 3398397d9..3983e7989 100644
--- a/service/service.go
+++ b/service/service.go
@@ -169,11 +169,10 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 
 	telFactory := telemetry.NewFactory()
 	telset := telemetry.Settings{
-		AsyncErrorChannel: set.AsyncErrorChannel,
-		BuildInfo:         set.BuildInfo,
-		ZapOptions:        set.LoggingOptions,
-		SDK:               &sdk,
-		Resource:          res,
+		BuildInfo:  set.BuildInfo,
+		ZapOptions: set.LoggingOptions,
+		SDK:        &sdk,
+		Resource:   res,
 	}
 
 	logger, loggerProvider, err := telFactory.CreateLogger(ctx, telset, &cfg.Telemetry)
