diff --git a/pdata/internal/generated_proto_profile_test.go b/pdata/internal/generated_proto_profile_test.go
index 872f0823c..76b7d69c6 100644
--- a/pdata/internal/generated_proto_profile_test.go
+++ b/pdata/internal/generated_proto_profile_test.go
@@ -191,8 +191,8 @@ func genTestFailingUnmarshalProtoValuesProfile() map[string][]byte {
 		"invalid_field":                          {0x02},
 		"SampleType/wrong_wire_type":             {0xc},
 		"SampleType/missing_value":               {0xa},
-		"Sample/wrong_wire_type":                 {0x14},
-		"Sample/missing_value":                   {0x12},
+		"Samples/wrong_wire_type":                {0x14},
+		"Samples/missing_value":                  {0x12},
 		"TimeUnixNano/wrong_wire_type":           {0x1c},
 		"TimeUnixNano/missing_value":             {0x19},
 		"DurationNano/wrong_wire_type":           {0x24},
@@ -201,18 +201,16 @@ func genTestFailingUnmarshalProtoValuesProfile() map[string][]byte {
 		"PeriodType/missing_value":               {0x2a},
 		"Period/wrong_wire_type":                 {0x34},
 		"Period/missing_value":                   {0x30},
-		"CommentStrindices/wrong_wire_type":      {0x3c},
-		"CommentStrindices/missing_value":        {0x3a},
-		"ProfileId/wrong_wire_type":              {0x44},
-		"ProfileId/missing_value":                {0x42},
-		"DroppedAttributesCount/wrong_wire_type": {0x4c},
-		"DroppedAttributesCount/missing_value":   {0x48},
-		"OriginalPayloadFormat/wrong_wire_type":  {0x54},
-		"OriginalPayloadFormat/missing_value":    {0x52},
-		"OriginalPayload/wrong_wire_type":        {0x5c},
-		"OriginalPayload/missing_value":          {0x5a},
-		"AttributeIndices/wrong_wire_type":       {0x64},
-		"AttributeIndices/missing_value":         {0x62},
+		"ProfileId/wrong_wire_type":              {0x3c},
+		"ProfileId/missing_value":                {0x3a},
+		"DroppedAttributesCount/wrong_wire_type": {0x44},
+		"DroppedAttributesCount/missing_value":   {0x40},
+		"OriginalPayloadFormat/wrong_wire_type":  {0x4c},
+		"OriginalPayloadFormat/missing_value":    {0x4a},
+		"OriginalPayload/wrong_wire_type":        {0x54},
+		"OriginalPayload/missing_value":          {0x52},
+		"AttributeIndices/wrong_wire_type":       {0x5c},
+		"AttributeIndices/missing_value":         {0x5a},
 	}
 }
 
@@ -220,12 +218,11 @@ func genTestEncodingValuesProfile() map[string]*Profile {
 	return map[string]*Profile{
 		"empty":                       NewProfile(),
 		"SampleType/test":             {SampleType: *GenTestValueType()},
-		"Sample/test":                 {Sample: []*Sample{{}, GenTestSample()}},
+		"Samples/test":                {Samples: []*Sample{{}, GenTestSample()}},
 		"TimeUnixNano/test":           {TimeUnixNano: uint64(13)},
 		"DurationNano/test":           {DurationNano: uint64(13)},
 		"PeriodType/test":             {PeriodType: *GenTestValueType()},
 		"Period/test":                 {Period: int64(13)},
-		"CommentStrindices/test":      {CommentStrindices: []int32{int32(0), int32(13)}},
 		"ProfileId/test":              {ProfileId: *GenTestProfileID()},
 		"DroppedAttributesCount/test": {DroppedAttributesCount: uint32(13)},
 		"OriginalPayloadFormat/test":  {OriginalPayloadFormat: "test_originalpayloadformat"},
