diff --git a/exporter/opencensusexporter/opencensus_test.go b/exporter/opencensusexporter/opencensus_test.go
index 9fa17831d..75135e0f0 100644
--- a/exporter/opencensusexporter/opencensus_test.go
+++ b/exporter/opencensusexporter/opencensus_test.go
@@ -167,18 +167,18 @@ func TestSendMetrics(t *testing.T) {
 	})
 
 	md := testdataold.GenerateMetricDataOneMetric()
-	assert.NoError(t, exp.ConsumeMetrics(context.Background(), pdatautil.MetricsFromInternalMetrics(md)))
+	assert.NoError(t, exp.ConsumeMetrics(context.Background(), pdatautil.MetricsFromOldInternalMetrics(md)))
 	testutil.WaitFor(t, func() bool {
 		return len(sink.AllMetrics()) == 1
 	})
 	metrics := sink.AllMetrics()
 	require.Len(t, metrics, 1)
-	assert.Equal(t, md, pdatautil.MetricsToInternalMetrics(metrics[0]))
+	assert.Equal(t, md, pdatautil.MetricsToOldInternalMetrics(metrics[0]))
 
 	// Sending data no node.
 	sink.Reset()
 	pdata.NewResource().CopyTo(md.ResourceMetrics().At(0).Resource())
-	assert.NoError(t, exp.ConsumeMetrics(context.Background(), pdatautil.MetricsFromInternalMetrics(md)))
+	assert.NoError(t, exp.ConsumeMetrics(context.Background(), pdatautil.MetricsFromOldInternalMetrics(md)))
 	testutil.WaitFor(t, func() bool {
 		return len(sink.AllMetrics()) == 1
 	})
@@ -186,7 +186,7 @@ func TestSendMetrics(t *testing.T) {
 	require.Len(t, metrics, 1)
 	// The conversion will initialize the Resource
 	md.ResourceMetrics().At(0).Resource().InitEmpty()
-	assert.Equal(t, md, pdatautil.MetricsToInternalMetrics(metrics[0]))
+	assert.Equal(t, md, pdatautil.MetricsToOldInternalMetrics(metrics[0]))
 }
 
 func TestSendMetrics_NoBackend(t *testing.T) {
@@ -207,7 +207,7 @@ func TestSendMetrics_NoBackend(t *testing.T) {
 		assert.NoError(t, exp.Shutdown(context.Background()))
 	})
 
-	md := pdatautil.MetricsFromInternalMetrics(testdataold.GenerateMetricDataOneMetric())
+	md := pdatautil.MetricsFromOldInternalMetrics(testdataold.GenerateMetricDataOneMetric())
 	for i := 0; i < 10000; i++ {
 		assert.Error(t, exp.ConsumeMetrics(context.Background(), md))
 	}
@@ -229,6 +229,6 @@ func TestSendMetrics_AfterStop(t *testing.T) {
 	require.NoError(t, exp.Start(context.Background(), host))
 	assert.NoError(t, exp.Shutdown(context.Background()))
 
-	md := pdatautil.MetricsFromInternalMetrics(testdataold.GenerateMetricDataOneMetric())
+	md := pdatautil.MetricsFromOldInternalMetrics(testdataold.GenerateMetricDataOneMetric())
 	assert.Error(t, exp.ConsumeMetrics(context.Background(), md))
 }
