diff --git a/config/confignet/confignet_test.go b/config/confignet/confignet_test.go
index 22eef3031..9375561dc 100644
--- a/config/confignet/confignet_test.go
+++ b/config/confignet/confignet_test.go
@@ -14,6 +14,24 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestNewDefaultDialerConfig(t *testing.T) {
+	expectedDialerConfig := DialerConfig{}
+	dialerConfig := NewDefaultDialerConfig()
+	require.Equal(t, expectedDialerConfig, dialerConfig)
+}
+
+func TestNewDefaultAddrConfig(t *testing.T) {
+	expectedAddrConfig := AddrConfig{}
+	addrConfig := NewDefaultAddrConfig()
+	require.Equal(t, expectedAddrConfig, addrConfig)
+}
+
+func TestNewDefaultTCPAddrConfig(t *testing.T) {
+	expectedTCPAddrConfig := TCPAddrConfig{}
+	tcpAddrconfig := NewDefaultTCPAddrConfig()
+	require.Equal(t, expectedTCPAddrConfig, tcpAddrconfig)
+}
+
 func TestAddrConfigTimeout(t *testing.T) {
 	nac := &AddrConfig{
 		Endpoint:  "localhost:0",
