diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index ac1fbdb28..5037b7f19 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -8,8 +8,8 @@ require (
 	github.com/google/go-tpm v0.9.8
 	github.com/stretchr/testify v1.11.1
 	go.opentelemetry.io/collector/config/configopaque v1.51.0
-	go.opentelemetry.io/collector/confmap/xconfmap v0.145.0
-	go.opentelemetry.io/collector/internal/testutil v0.145.0
+	go.opentelemetry.io/collector/confmap/xconfmap v0.146.0
+	go.opentelemetry.io/collector/internal/testutil v0.146.0
 )
 
 require (
