diff --git a/exporter/kafkaexporter/factory_test.go b/exporter/kafkaexporter/factory_test.go
index bd4d923e2..76e1834a4 100644
--- a/exporter/kafkaexporter/factory_test.go
+++ b/exporter/kafkaexporter/factory_test.go
@@ -59,6 +59,18 @@ func TestCreateMetricsExport(t *testing.T) {
 	assert.NotNil(t, mr)
 }
 
+func TestCreateLogsExport(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	cfg.Brokers = []string{"invalid:9092"}
+	cfg.ProtocolVersion = "2.0.0"
+	// this disables contacting the broker so we can successfully create the exporter
+	cfg.Metadata.Full = false
+	mf := kafkaExporterFactory{logsMarshallers: logsMarshallers()}
+	mr, err := mf.createLogsExporter(context.Background(), component.ExporterCreateParams{Logger: zap.NewNop()}, cfg)
+	require.NoError(t, err)
+	assert.NotNil(t, mr)
+}
+
 func TestCreateTracesExporter_err(t *testing.T) {
 	cfg := createDefaultConfig().(*Config)
 	cfg.Brokers = []string{"invalid:9092"}
@@ -80,6 +92,16 @@ func TestCreateMetricsExporter_err(t *testing.T) {
 	assert.Nil(t, mr)
 }
 
+func TestCreateLogsExporter_err(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	cfg.Brokers = []string{"invalid:9092"}
+	cfg.ProtocolVersion = "2.0.0"
+	mf := kafkaExporterFactory{logsMarshallers: logsMarshallers()}
+	mr, err := mf.createLogsExporter(context.Background(), component.ExporterCreateParams{Logger: zap.NewNop()}, cfg)
+	require.Error(t, err)
+	assert.Nil(t, mr)
+}
+
 func TestWithMarshallers(t *testing.T) {
 	cm := &customMarshaller{}
 	f := NewFactory(WithAddTracesMarshallers(map[string]TracesMarshaller{cm.Encoding(): cm}))
