diff --git a/exporter/kafkaexporter/marshaller.go b/exporter/kafkaexporter/marshaller.go
index 7e1bdba21..9785a0086 100644
--- a/exporter/kafkaexporter/marshaller.go
+++ b/exporter/kafkaexporter/marshaller.go
@@ -36,6 +36,15 @@ type MetricsMarshaller interface {
 	Encoding() string
 }
 
+// LogsMarshaller marshals logs into Message array
+type LogsMarshaller interface {
+	// Marshal serializes logs into Messages
+	Marshal(logs pdata.Logs) ([]Message, error)
+
+	// Encoding returns encoding name
+	Encoding() string
+}
+
 // Message encapsulates Kafka's message payload.
 type Message struct {
 	Value []byte
@@ -61,3 +70,11 @@ func metricsMarshallers() map[string]MetricsMarshaller {
 		otlppb.Encoding(): otlppb,
 	}
 }
+
+// logsMarshallers returns map of supported encodings and LogsMarshaller
+func logsMarshallers() map[string]LogsMarshaller {
+	otlppb := &otlpLogsPbMarshaller{}
+	return map[string]LogsMarshaller{
+		otlppb.Encoding(): otlppb,
+	}
+}
