diff --git a/receiver/receivertest/metric_appender.go b/receiver/receivertest/metric_appender.go
new file mode 100644
index 000000000..706a629d5
--- /dev/null
+++ b/receiver/receivertest/metric_appender.go
@@ -0,0 +1,29 @@
+package receivertest
+
+import (
+	"context"
+	"sync"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+)
+
+type MetricAppender struct {
+	sync.RWMutex
+	MetricsDataList []consumerdata.MetricsData
+}
+
+func NewMetricAppender() *MetricAppender {
+	return &MetricAppender{}
+}
+
+var _ consumer.MetricsConsumer = (*MetricAppender)(nil)
+
+func (ma *MetricAppender) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
+	ma.Lock()
+	defer ma.Unlock()
+
+	ma.MetricsDataList = append(ma.MetricsDataList, md)
+
+	return nil
+}
