diff --git a/extension/extensionauth/server.go b/extension/extensionauth/server.go
index 511d43a22..56cf59335 100644
--- a/extension/extensionauth/server.go
+++ b/extension/extensionauth/server.go
@@ -7,9 +7,9 @@ import (
 	"context"
 )
 
-// Server is an optional Extension interface that can be used as an authenticator for the configauth.Authentication option.
+// Server is an optional Extension interface that can be used as an authenticator for the configauth.Config option.
 // Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
-// names from the Authentication configuration. Each Server is free to define its own behavior and configuration options,
+// names from the [configauth.Config] configuration. Each Server is free to define its own behavior and configuration options,
 // but note that the expectations that come as part of Extensions exist here as well. For instance, multiple instances of the same
 // authenticator should be possible to exist under different names.
 type Server interface {
