diff --git a/cmd/builder/go.mod b/cmd/builder/go.mod
index 33061f897..b6b63cb3d 100644
--- a/cmd/builder/go.mod
+++ b/cmd/builder/go.mod
@@ -11,7 +11,7 @@ require (
 	github.com/knadh/koanf/providers/env v0.1.0
 	github.com/knadh/koanf/providers/file v0.1.0
 	github.com/knadh/koanf/providers/fs v0.1.0
-	github.com/knadh/koanf/v2 v2.1.0
+	github.com/knadh/koanf/v2 v2.1.1
 	github.com/spf13/cobra v1.8.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.9.0
