diff --git a/consumer/consumerhelper/doc.go b/consumer/consumerhelper/doc.go
index 7dd8fa6fb..977dc8fae 100644
--- a/consumer/consumerhelper/doc.go
+++ b/consumer/consumerhelper/doc.go
@@ -14,4 +14,4 @@
 
 // Package consumerhelper defines types and functions used to create consumer
 // Logs, Metrics, and Traces.
-package consumerhelper
+package consumerhelper // import "go.opentelemetry.io/collector/consumer/consumerhelper"
