diff --git a/service/service_test.go b/service/service_test.go
index c5dcfb46c..2074a526a 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -92,6 +92,53 @@ func TestServiceGetExporters(t *testing.T) {
 	assert.Contains(t, expMap[config.LogsDataType], config.NewComponentID("nop"))
 }
 
+// TestServiceTelemetryCleanupOnError tests that if newService errors due to an invalid config telemetry is cleaned up
+// and another service with a valid config can be started right after.
+func TestServiceTelemetryCleanupOnError(t *testing.T) {
+	factories, err := componenttest.NopFactories()
+	require.NoError(t, err)
+
+	// Read invalid yaml config from file
+	invalidConf, err := confmaptest.LoadConf(filepath.Join("testdata", "otelcol-invalid.yaml"))
+	require.NoError(t, err)
+	invalidCfg, err := configunmarshaler.New().Unmarshal(invalidConf, factories)
+	require.NoError(t, err)
+
+	// Read valid yaml config from file
+	validConf, err := confmaptest.LoadConf(filepath.Join("testdata", "otelcol-nop.yaml"))
+	require.NoError(t, err)
+	validCfg, err := configunmarshaler.New().Unmarshal(validConf, factories)
+	require.NoError(t, err)
+
+	// Create a service with an invalid config and expect an error
+	telemetryOne := newColTelemetry(featuregate.NewRegistry())
+	_, err = newService(&settings{
+		BuildInfo: component.NewDefaultBuildInfo(),
+		Factories: factories,
+		Config:    invalidCfg,
+		telemetry: telemetryOne,
+	})
+	require.Error(t, err)
+
+	// Create a service with a valid config and expect no error
+	telemetryTwo := newColTelemetry(featuregate.NewRegistry())
+	srv, err := newService(&settings{
+		BuildInfo: component.NewDefaultBuildInfo(),
+		Factories: factories,
+		Config:    validCfg,
+		telemetry: telemetryTwo,
+	})
+	require.NoError(t, err)
+
+	// For safety ensure everything is cleaned up
+	t.Cleanup(func() {
+		assert.NoError(t, telemetryOne.shutdown())
+		assert.NoError(t, telemetryTwo.shutdown())
+		assert.NoError(t, srv.Shutdown(context.Background()))
+	})
+
+}
+
 func createExampleService(t *testing.T, factories component.Factories) *service {
 	// Read yaml config from file
 	conf, err := confmaptest.LoadConf(filepath.Join("testdata", "otelcol-nop.yaml"))
