diff --git a/cmd/builder/internal/builder/config.go b/cmd/builder/internal/builder/config.go
index c0ed0050c..3fdc33da3 100644
--- a/cmd/builder/internal/builder/config.go
+++ b/cmd/builder/internal/builder/config.go
@@ -18,8 +18,8 @@ import (
 )
 
 const (
-	defaultBetaOtelColVersion   = "v0.136.0"
-	defaultStableOtelColVersion = "v1.42.0"
+	defaultBetaOtelColVersion   = "v0.137.0"
+	defaultStableOtelColVersion = "v1.43.0"
 )
 
 // errMissingGoMod indicates an empty gomod field
