diff --git a/pdata/plog/logs_test.go b/pdata/plog/logs_test.go
index 66a317b7e..cb5366e9b 100644
--- a/pdata/plog/logs_test.go
+++ b/pdata/plog/logs_test.go
@@ -132,14 +132,10 @@ func TestLogsClone(t *testing.T) {
 	assert.EqualValues(t, logs, logs.Clone())
 }
 
-func BenchmarkLogsClone(b *testing.B) {
+func TestLogsCopyTo(t *testing.T) {
 	logs := NewLogs()
 	internal.FillTestResourceLogsSlice(internal.ResourceLogsSlice(logs.ResourceLogs()))
-	b.ResetTimer()
-	for n := 0; n < b.N; n++ {
-		clone := logs.Clone()
-		if clone.ResourceLogs().Len() != logs.ResourceLogs().Len() {
-			b.Fail()
-		}
-	}
+	logsCopy := NewLogs()
+	logs.CopyTo(logsCopy)
+	assert.EqualValues(t, logs, logsCopy)
 }
