diff --git a/pipeline/xpipeline/go.mod b/pipeline/xpipeline/go.mod
index 99248b854..8fd27d675 100644
--- a/pipeline/xpipeline/go.mod
+++ b/pipeline/xpipeline/go.mod
@@ -2,6 +2,6 @@ module go.opentelemetry.io/collector/pipeline/xpipeline
 
 go 1.25.0
 
-require go.opentelemetry.io/collector/pipeline v1.53.0
+require go.opentelemetry.io/collector/pipeline v1.54.0
 
 replace go.opentelemetry.io/collector/pipeline => ../
