diff --git a/translator/internaldata/resource_to_oc.go b/translator/internaldata/resource_to_oc.go
index 418510596..91d1f9394 100644
--- a/translator/internaldata/resource_to_oc.go
+++ b/translator/internaldata/resource_to_oc.go
@@ -89,7 +89,7 @@ func internalResourceToOC(resource pdata.Resource) (*occommon.Node, *ocresource.
 	ocResource := &ocresource.Resource{}
 	labels := make(map[string]string, attrs.Len())
 	attrs.Range(func(k string, v pdata.AttributeValue) bool {
-		val := tracetranslator.AttributeValueToString(v, false)
+		val := tracetranslator.AttributeValueToString(v)
 
 		switch k {
 		case conventions.AttributeCloudAvailabilityZone:
