diff --git a/pdata/ptrace/generated_spanlink.go b/pdata/ptrace/generated_spanlink.go
index 47635b4c6..66040556d 100644
--- a/pdata/ptrace/generated_spanlink.go
+++ b/pdata/ptrace/generated_spanlink.go
@@ -144,6 +144,29 @@ func (ms SpanLink) marshalJSONStream(dest *json.Stream) {
 	dest.WriteObjectEnd()
 }
 
+// unmarshalJSONIter unmarshals all properties from the current struct from the source iterator.
+func (ms SpanLink) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
+		switch f {
+		case "traceId", "trace_id":
+			ms.orig.TraceId.UnmarshalJSONIter(iter)
+		case "spanId", "span_id":
+			ms.orig.SpanId.UnmarshalJSONIter(iter)
+		case "traceState", "trace_state":
+			internal.UnmarshalJSONIterTraceState(internal.NewTraceState(&ms.orig.TraceState, ms.state), iter)
+		case "flags":
+			ms.orig.Flags = iter.ReadUint32()
+		case "attributes":
+			internal.UnmarshalJSONIterMap(internal.NewMap(&ms.orig.Attributes, ms.state), iter)
+		case "droppedAttributesCount", "dropped_attributes_count":
+			ms.orig.DroppedAttributesCount = iter.ReadUint32()
+		default:
+			iter.Skip()
+		}
+		return true
+	})
+}
+
 func copyOrigSpanLink(dest, src *otlptrace.Span_Link) {
 	dest.TraceId = src.TraceId
 	dest.SpanId = src.SpanId
