diff --git a/receiver/nopreceiver/go.mod b/receiver/nopreceiver/go.mod
index 1f3396647..b212a4557 100644
--- a/receiver/nopreceiver/go.mod
+++ b/receiver/nopreceiver/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/component v0.111.0
 	go.opentelemetry.io/collector/confmap v1.17.0
-	go.opentelemetry.io/collector/consumer v0.111.1-0.20241022132505-11767fed9dfe
+	go.opentelemetry.io/collector/consumer v0.111.1-0.20241022144230-02ceda1496be
 	go.opentelemetry.io/collector/consumer/consumertest v0.111.0
 	go.opentelemetry.io/collector/receiver v0.111.0
 	go.uber.org/goleak v1.3.0
