diff --git a/internal/collector/telemetry/telemetry.go b/internal/collector/telemetry/telemetry.go
index ab2c7b790..d9c496e52 100644
--- a/internal/collector/telemetry/telemetry.go
+++ b/internal/collector/telemetry/telemetry.go
@@ -17,8 +17,6 @@ package telemetry
 
 import (
 	"flag"
-	"fmt"
-	"strings"
 
 	"go.opentelemetry.io/collector/config/configtelemetry"
 	"go.opentelemetry.io/collector/internal/version"
@@ -80,27 +78,11 @@ func GetAddInstanceID() bool {
 // GetLevel returns the Level represented by the string. The parsing is case-insensitive
 // and it returns error if the string value is unknown.
 func GetLevel() (configtelemetry.Level, error) {
-	var level configtelemetry.Level
-	var str string
-
-	if metricsLevelPtr != nil {
-		str = strings.ToLower(*metricsLevelPtr)
-	}
-
-	switch str {
-	case "none":
-		level = configtelemetry.LevelNone
-	case "", "basic":
-		level = configtelemetry.LevelBasic
-	case "normal":
-		level = configtelemetry.LevelNormal
-	case "detailed":
-		level = configtelemetry.LevelDetailed
-	default:
-		return level, fmt.Errorf("unknown metrics level %q", str)
+	if metricsLevelPtr != nil && *metricsLevelPtr != "" {
+		return configtelemetry.ParseLevel(*metricsLevelPtr)
 	}
 
-	return level, nil
+	return configtelemetry.LevelBasic, nil
 }
 
 func GetMetricsAddr() string {
