diff --git a/service/service_test.go b/service/service_test.go
index cdb20024d..d8f0b2855 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -25,7 +25,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config"
-	"go.opentelemetry.io/collector/config/configtest"
+	"go.opentelemetry.io/collector/confmap/confmaptest"
 	"go.opentelemetry.io/collector/service/internal/configunmarshaler"
 )
 
@@ -93,9 +93,9 @@ func TestService_GetExporters(t *testing.T) {
 
 func createExampleService(t *testing.T, factories component.Factories) *service {
 	// Read yaml config from file
-	cfgMap, err := configtest.LoadConfigMap(filepath.Join("testdata", "otelcol-nop.yaml"))
+	conf, err := confmaptest.LoadConf(filepath.Join("testdata", "otelcol-nop.yaml"))
 	require.NoError(t, err)
-	cfg, err := configunmarshaler.New().Unmarshal(cfgMap, factories)
+	cfg, err := configunmarshaler.New().Unmarshal(conf, factories)
 	require.NoError(t, err)
 
 	srv, err := newService(&svcSettings{
