diff --git a/component/componenttest/example_factories.go b/component/componenttest/example_factories.go
index 3420d0c3f..fbd463744 100644
--- a/component/componenttest/example_factories.go
+++ b/component/componenttest/example_factories.go
@@ -15,97 +15,9 @@
 package componenttest
 
 import (
-	"context"
-
-	"github.com/spf13/viper"
-
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/config/configmodels"
-	"go.opentelemetry.io/collector/consumer"
 )
 
-// MultiProtoReceiver is for testing purposes. We are defining an example multi protocol
-// config and factory for "multireceiver" receiver type.
-type MultiProtoReceiver struct {
-	configmodels.ReceiverSettings `mapstructure:",squash"`            // squash ensures fields are correctly decoded in embedded struct
-	Protocols                     map[string]MultiProtoReceiverOneCfg `mapstructure:"protocols"`
-}
-
-// MultiProtoReceiverOneCfg is multi proto receiver config.
-type MultiProtoReceiverOneCfg struct {
-	Endpoint     string `mapstructure:"endpoint"`
-	ExtraSetting string `mapstructure:"extra"`
-}
-
-// MultiProtoReceiverFactory is factory for MultiProtoReceiver.
-type MultiProtoReceiverFactory struct {
-}
-
-var _ component.ReceiverFactory = (*MultiProtoReceiverFactory)(nil)
-
-// Type gets the type of the Receiver config created by this factory.
-func (f *MultiProtoReceiverFactory) Type() configmodels.Type {
-	return "multireceiver"
-}
-
-// Unmarshal implements the ConfigUnmarshaler interface.
-func (f *MultiProtoReceiverFactory) Unmarshal(componentViperSection *viper.Viper, intoCfg interface{}) error {
-	return componentViperSection.UnmarshalExact(intoCfg)
-}
-
-// CreateDefaultConfig creates the default configuration for the Receiver.
-func (f *MultiProtoReceiverFactory) CreateDefaultConfig() configmodels.Receiver {
-	return &MultiProtoReceiver{
-		ReceiverSettings: configmodels.ReceiverSettings{
-			TypeVal: f.Type(),
-			NameVal: string(f.Type()),
-		},
-		Protocols: map[string]MultiProtoReceiverOneCfg{
-			"http": {
-				Endpoint:     "example.com:8888",
-				ExtraSetting: "extra string 1",
-			},
-			"tcp": {
-				Endpoint:     "omnition.com:9999",
-				ExtraSetting: "extra string 2",
-			},
-		},
-	}
-}
-
-// CreateTraceReceiver creates a trace receiver based on this config.
-func (f *MultiProtoReceiverFactory) CreateTracesReceiver(
-	_ context.Context,
-	_ component.ReceiverCreateParams,
-	_ configmodels.Receiver,
-	_ consumer.TracesConsumer,
-) (component.TracesReceiver, error) {
-	// Not used for this test, just return nil
-	return nil, nil
-}
-
-// CreateMetricsReceiver creates a metrics receiver based on this config.
-func (f *MultiProtoReceiverFactory) CreateMetricsReceiver(
-	_ context.Context,
-	_ component.ReceiverCreateParams,
-	_ configmodels.Receiver,
-	_ consumer.MetricsConsumer,
-) (component.MetricsReceiver, error) {
-	// Not used for this test, just return nil
-	return nil, nil
-}
-
-// CreateMetricsReceiver creates a metrics receiver based on this config.
-func (f *MultiProtoReceiverFactory) CreateLogsReceiver(
-	_ context.Context,
-	_ component.ReceiverCreateParams,
-	_ configmodels.Receiver,
-	_ consumer.LogsConsumer,
-) (component.LogsReceiver, error) {
-	// Not used for this test, just return nil
-	return nil, nil
-}
-
 // ExampleComponents registers example factories. This is only used by tests.
 func ExampleComponents() (
 	factories component.Factories,
@@ -117,7 +29,6 @@ func ExampleComponents() (
 
 	factories.Receivers, err = component.MakeReceiverFactoryMap(
 		&ExampleReceiverFactory{},
-		&MultiProtoReceiverFactory{},
 	)
 	if err != nil {
 		return
