diff --git a/translator/internaldata/resource_to_oc.go b/translator/internaldata/resource_to_oc.go
index 5f156bbf6..58330c8b8 100644
--- a/translator/internaldata/resource_to_oc.go
+++ b/translator/internaldata/resource_to_oc.go
@@ -91,6 +91,8 @@ func internalResourceToOC(resource pdata.Resource) (*occommon.Node, *ocresource.
 		val := tracetranslator.AttributeValueToString(v, false)
 
 		switch k {
+		case conventions.AttributeCloudAvailabilityZone:
+			labels[resourcekeys.CloudKeyZone] = val
 		case conventions.OCAttributeResourceType:
 			ocResource.Type = val
 		case conventions.AttributeServiceName:
