diff --git a/extension/extensiontest/nop_extension_test.go b/extension/extensiontest/nop_extension_test.go
index 1e01f9076..b76df9253 100644
--- a/extension/extensiontest/nop_extension_test.go
+++ b/extension/extensiontest/nop_extension_test.go
@@ -21,7 +21,7 @@ func TestNewNopFactory(t *testing.T) {
 	cfg := factory.CreateDefaultConfig()
 	assert.Equal(t, &nopConfig{}, cfg)
 
-	traces, err := factory.CreateExtension(context.Background(), NewNopSettings(), cfg)
+	traces, err := factory.Create(context.Background(), NewNopSettings(), cfg)
 	require.NoError(t, err)
 	assert.NoError(t, traces.Start(context.Background(), componenttest.NewNopHost()))
 	assert.NoError(t, traces.Shutdown(context.Background()))
