diff --git a/processor/batchprocessor/generated_component_telemetry_test.go b/processor/batchprocessor/generated_component_telemetry_test.go
index 4747507bc..cad0f0fff 100644
--- a/processor/batchprocessor/generated_component_telemetry_test.go
+++ b/processor/batchprocessor/generated_component_telemetry_test.go
@@ -13,6 +13,7 @@ import (
 	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config/configtelemetry"
 	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/processor/processortest"
@@ -24,14 +25,19 @@ type componentTestTelemetry struct {
 }
 
 func (tt *componentTestTelemetry) NewSettings() processor.Settings {
-	settings := processortest.NewNopSettings()
-	settings.MeterProvider = tt.meterProvider
-	settings.LeveledMeterProvider = func(_ configtelemetry.Level) metric.MeterProvider {
+	set := processortest.NewNopSettings()
+	set.TelemetrySettings = tt.newTelemetrySettings()
+	set.ID = component.NewID(component.MustNewType("batch"))
+	return set
+}
+
+func (tt *componentTestTelemetry) newTelemetrySettings() component.TelemetrySettings {
+	set := componenttest.NewNopTelemetrySettings()
+	set.MeterProvider = tt.meterProvider
+	set.LeveledMeterProvider = func(_ configtelemetry.Level) metric.MeterProvider {
 		return tt.meterProvider
 	}
-	settings.ID = component.NewID(component.MustNewType("batch"))
-
-	return settings
+	return set
 }
 
 func setupTestTelemetry() componentTestTelemetry {
