diff --git a/cmd/occollector/app/builder/pipelines_builder_test.go b/cmd/occollector/app/builder/pipelines_builder_test.go
index 646cb380d..ee3c3c519 100644
--- a/cmd/occollector/app/builder/pipelines_builder_test.go
+++ b/cmd/occollector/app/builder/pipelines_builder_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -25,10 +25,10 @@ import (
 
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 
-	"github.com/census-instrumentation/opencensus-service/data"
-	"github.com/census-instrumentation/opencensus-service/internal/configmodels"
-	"github.com/census-instrumentation/opencensus-service/internal/configv2"
-	"github.com/census-instrumentation/opencensus-service/processor/addattributesprocessor"
+	"github.com/open-telemetry/opentelemetry-service/data"
+	"github.com/open-telemetry/opentelemetry-service/internal/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/internal/configv2"
+	"github.com/open-telemetry/opentelemetry-service/processor/addattributesprocessor"
 )
 
 // Ensure attributes processor is registered.
