diff --git a/config/configtls/configtls.go b/config/configtls/configtls.go
index e2a1734a5..4f97346e6 100644
--- a/config/configtls/configtls.go
+++ b/config/configtls/configtls.go
@@ -26,10 +26,6 @@ const defaultMaxTLSVersion = 0
 
 var systemCertPool = x509.SystemCertPool
 
-// TLSSetting exposes the common client and server TLS configurations.
-// Deprecated: [v0.96.0] Use Config instead.
-type TLSSetting = Config
-
 // Config exposes the common client and server TLS configurations.
 // Note: Since there isn't anything specific to a server connection. Components
 // with server connections should use Config.
@@ -76,17 +72,12 @@ type Config struct {
 	ReloadInterval time.Duration `mapstructure:"reload_interval"`
 }
 
-// TSLClientSetting contains TLS configurations that are specific to client
-// connections in addition to the common configurations.
-// Deprecated: [v0.96.0] Use ClientConfig instead.
-type TLSClientSetting = ClientConfig
-
 // ClientConfig contains TLS configurations that are specific to client
 // connections in addition to the common configurations. This should be used by
 // components configuring TLS client connections.
 type ClientConfig struct {
 	// squash ensures fields are correctly decoded in embedded struct.
-	TLSSetting `mapstructure:",squash"`
+	Config `mapstructure:",squash"`
 
 	// These are config options specific to client connections.
 
@@ -105,17 +96,12 @@ type ClientConfig struct {
 	ServerName string `mapstructure:"server_name_override"`
 }
 
-// TLSServerSetting contains TLS configurations that are specific to server
-// connections in addition to the common configurations.
-// Deprecated: [v0.96.0] Use ServerConfig instead.
-type TLSServerSetting = ServerConfig
-
 // ServerConfig contains TLS configurations that are specific to server
 // connections in addition to the common configurations. This should be used by
 // components configuring TLS server connections.
 type ServerConfig struct {
 	// squash ensures fields are correctly decoded in embedded struct.
-	TLSSetting `mapstructure:",squash"`
+	Config `mapstructure:",squash"`
 
 	// These are config options specific to server connections.
 
@@ -174,7 +160,7 @@ func (r *certReloader) GetCertificate() (*tls.Certificate, error) {
 	return r.cert, nil
 }
 
-func (c TLSSetting) Validate() error {
+func (c Config) Validate() error {
 	if c.hasCAFile() && c.hasCAPem() {
 		return fmt.Errorf("provide either a CA file or the PEM-encoded string, but not both")
 	}
@@ -377,7 +363,7 @@ func (c ClientConfig) LoadTLSConfigContext(_ context.Context) (*tls.Config, erro
 		return nil, nil
 	}
 
-	tlsCfg, err := c.TLSSetting.loadTLSConfig()
+	tlsCfg, err := c.loadTLSConfig()
 	if err != nil {
 		return nil, fmt.Errorf("failed to load TLS config: %w", err)
 	}
