diff --git a/internal/cmd/pdatagen/internal/proto/field.go b/internal/cmd/pdatagen/internal/proto/field.go
index e91386915..e347537bd 100644
--- a/internal/cmd/pdatagen/internal/proto/field.go
+++ b/internal/cmd/pdatagen/internal/proto/field.go
@@ -128,6 +128,7 @@ func (pf *Field) getTemplateFields() map[string]any {
 		"origName":             pf.messageName(),
 		"origFullName":         pf.MessageFullName,
 		"oneOfGroup":           pf.OneOfGroup,
+		"oneOfMessageName":     ExtractNameFromFull(pf.OneOfMessageFullName),
 		"oneOfMessageFullName": pf.OneOfMessageFullName,
 		"repeated":             pf.Repeated,
 		"nullable":             pf.Nullable,
