diff --git a/config/configtls/tpm_test.go b/config/configtls/tpm_test.go
index c9dd8d63f..72f49877b 100644
--- a/config/configtls/tpm_test.go
+++ b/config/configtls/tpm_test.go
@@ -28,9 +28,12 @@ import (
 	"github.com/google/go-tpm/tpm2/transport/simulator"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/internal/testutil"
 )
 
 func TestTPM_loadCertificate(t *testing.T) {
+	testutil.SkipIfFIPSOnly(t, "use of CFB is not allowed in FIPS 140-only mode")
 	tpm, err := simulator.OpenSimulator()
 	require.NoError(t, err)
 	defer tpm.Close()
