diff --git a/service/service.go b/service/service.go
index 80983067c..22c8f20ba 100644
--- a/service/service.go
+++ b/service/service.go
@@ -31,6 +31,7 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector/config"
+	"github.com/open-telemetry/opentelemetry-collector/config/configcheck"
 	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
 	"github.com/open-telemetry/opentelemetry-collector/extension"
 	"github.com/open-telemetry/opentelemetry-collector/receiver"
@@ -68,17 +69,15 @@ func (app *Application) Context() context.Context {
 	return context.Background()
 }
 
-// ReportFatalError is used to report to the host that the receiver encountered
-// a fatal error (i.e.: an error that the instance can't recover from) after
-// its start function has already returned.
-func (app *Application) ReportFatalError(err error) {
-	app.asyncErrorChannel <- err
-}
-
 // New creates and returns a new instance of Application.
 func New(
 	factories config.Factories,
-) *Application {
+) (*Application, error) {
+
+	if err := configcheck.ValidateConfigFromFactories(factories); err != nil {
+		return nil, err
+	}
+
 	app := &Application{
 		v:         viper.New(),
 		readyChan: make(chan struct{}),
@@ -108,7 +107,14 @@ func New(
 
 	app.rootCmd = rootCmd
 
-	return app
+	return app, nil
+}
+
+// ReportFatalError is used to report to the host that the receiver encountered
+// a fatal error (i.e.: an error that the instance can't recover from) after
+// its start function has already returned.
+func (app *Application) ReportFatalError(err error) {
+	app.asyncErrorChannel <- err
 }
 
 func (app *Application) init() {
