diff --git a/config/configtls/configtls.go b/config/configtls/configtls.go
index 2090f775d..569ae6061 100644
--- a/config/configtls/configtls.go
+++ b/config/configtls/configtls.go
@@ -376,13 +376,6 @@ func (c Config) loadCert(caPath string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
-// LoadTLSConfigContext loads the TLS configuration.
-//
-// Deprecated: [v0.99.0] Use LoadTLSConfig instead.
-func (c ClientConfig) LoadTLSConfigContext(ctx context.Context) (*tls.Config, error) {
-	return c.LoadTLSConfig(ctx)
-}
-
 // LoadTLSConfig loads the TLS configuration.
 func (c ClientConfig) LoadTLSConfig(_ context.Context) (*tls.Config, error) {
 	if c.Insecure && !c.hasCA() {
@@ -398,13 +391,6 @@ func (c ClientConfig) LoadTLSConfig(_ context.Context) (*tls.Config, error) {
 	return tlsCfg, nil
 }
 
-// LoadTLSConfigContext loads the TLS configuration.
-//
-// Deprecated: [v0.99.0] Use LoadTLSConfig instead.
-func (c ServerConfig) LoadTLSConfigContext(ctx context.Context) (*tls.Config, error) {
-	return c.LoadTLSConfig(ctx)
-}
-
 // LoadTLSConfig loads the TLS configuration.
 func (c ServerConfig) LoadTLSConfig(_ context.Context) (*tls.Config, error) {
 	tlsCfg, err := c.loadTLSConfig()
