diff --git a/confmap/confmap.go b/confmap/confmap.go
index 3040778dc..6e3bacb30 100644
--- a/confmap/confmap.go
+++ b/confmap/confmap.go
@@ -235,6 +235,7 @@ func decodeConfig(m *Conf, result any, errorUnused bool, skipTopLevelUnmarshaler
 		TagName:          MapstructureTag,
 		WeaklyTypedInput: false,
 		MatchName:        caseSensitiveMatchName,
+		DecodeNil:        true,
 		DecodeHook: composehook.ComposeDecodeHookFunc(
 			useExpandValue(),
 			expandNilStructPointersHookFunc(),
@@ -582,10 +583,7 @@ type Marshaler interface {
 func zeroSliceHookFunc() mapstructure.DecodeHookFuncValue {
 	return safeWrapDecodeHookFunc(func(from reflect.Value, to reflect.Value) (any, error) {
 		if to.CanSet() && to.Kind() == reflect.Slice && from.Kind() == reflect.Slice {
-			if from.IsNil() {
-				// input slice is nil, set output slice to nil.
-				to.Set(reflect.Zero(to.Type()))
-			} else {
+			if !from.IsNil() {
 				// input slice is not nil, set the output slice to a new slice of the same type.
 				to.Set(reflect.MakeSlice(to.Type(), from.Len(), from.Cap()))
 			}
