diff --git a/receiver/opencensusreceiver/factory.go b/receiver/opencensusreceiver/factory.go
index 2617b39ae..d5a5753af 100644
--- a/receiver/opencensusreceiver/factory.go
+++ b/receiver/opencensusreceiver/factory.go
@@ -22,6 +22,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configgrpc"
 	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/confignet"
 	"go.opentelemetry.io/collector/consumer"
 )
 
@@ -52,11 +53,13 @@ func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
 			NameVal: typeStr,
 		},
 		GRPCServerSettings: configgrpc.GRPCServerSettings{
-			Endpoint: "0.0.0.0:55678",
+			NetAddr: confignet.NetAddr{
+				Endpoint:  "0.0.0.0:55678",
+				Transport: "tcp",
+			},
 			// We almost write 0 bytes, so no need to tune WriteBufferSize.
 			ReadBufferSize: 512 * 1024,
 		},
-		Transport: "tcp",
 	}
 }
 
@@ -107,7 +110,7 @@ func (f *Factory) createReceiver(cfg configmodels.Receiver) (*Receiver, error) {
 
 		// We don't have a receiver, so create one.
 		receiver, err = New(
-			rCfg.Name(), rCfg.Transport, rCfg.Endpoint, nil, nil, opts...)
+			rCfg.Name(), rCfg.NetAddr.Transport, rCfg.NetAddr.Endpoint, nil, nil, opts...)
 		if err != nil {
 			return nil, err
 		}
