diff --git a/config/confighttp/compression.go b/config/confighttp/compression.go
index 62f1ff6d7..3e357ad16 100644
--- a/config/confighttp/compression.go
+++ b/config/confighttp/compression.go
@@ -22,19 +22,11 @@ import (
 	"github.com/pierrec/lz4/v4"
 
 	"go.opentelemetry.io/collector/config/configcompression"
-	"go.opentelemetry.io/collector/featuregate"
-)
-
-var enableFramedSnappy = featuregate.GlobalRegistry().MustRegister(
-	"confighttp.framedSnappy",
-	featuregate.StageBeta,
-	featuregate.WithRegisterFromVersion("v0.125.0"),
-	featuregate.WithRegisterDescription("Content encoding 'snappy' will compress/decompress block snappy format while 'x-snappy-framed' will compress/decompress framed snappy format."),
-	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector/issues/10584"),
+	"go.opentelemetry.io/collector/config/confighttp/internal/metadata"
 )
 
 func defaultCompressionAlgorithms() []string {
-	if enableFramedSnappy.IsEnabled() {
+	if metadata.ConfighttpFramedSnappyFeatureGate.IsEnabled() {
 		return []string{"", "gzip", "zstd", "zlib", "snappy", "deflate", "lz4", "x-snappy-framed"}
 	}
 	return []string{"", "gzip", "zstd", "zlib", "snappy", "deflate", "lz4"}
@@ -236,7 +228,7 @@ func httpContentDecompressor(h http.Handler, maxRequestBodySize int64, eh func(w
 
 	enabled := map[string]func(body io.ReadCloser) (io.ReadCloser, error){}
 	for _, dec := range enableDecoders {
-		if dec == "x-frame-snappy" && !enableFramedSnappy.IsEnabled() {
+		if dec == "x-frame-snappy" && !metadata.ConfighttpFramedSnappyFeatureGate.IsEnabled() {
 			continue
 		}
 		enabled[dec] = availableDecoders[dec]
