diff --git a/internal/version/version.go b/internal/version/version.go
index 4969c15d2..c81f50eb3 100644
--- a/internal/version/version.go
+++ b/internal/version/version.go
@@ -20,8 +20,8 @@ import (
 	"runtime"
 )
 
-// version variable will be replaced at link time after `make` has been run.
-var version = "latest"
+// Version variable will be replaced at link time after `make` has been run.
+var Version = "latest"
 
 // GitHash variable will be replaced at link time after `make` has been run.
 var GitHash = "<NOT PROPERLY GENERATED>"
@@ -31,7 +31,7 @@ var GitHash = "<NOT PROPERLY GENERATED>"
 func Info() string {
 	buf := new(bytes.Buffer)
 	rows := [][2]string{
-		{"Version", version},
+		{"Version", Version},
 		{"GitHash", GitHash},
 		{"Goversion", runtime.Version()},
 		{"OS", runtime.GOOS},
