diff --git a/exporter/loggingexporter/factory.go b/exporter/loggingexporter/factory.go
index 6a830fbfb..d407b3c41 100644
--- a/exporter/loggingexporter/factory.go
+++ b/exporter/loggingexporter/factory.go
@@ -22,24 +22,22 @@ import (
 	"github.com/open-telemetry/opentelemetry-service/exporter"
 )
 
-var _ = exporter.RegisterFactory(&factory{})
-
 const (
 	// The value of "type" key in configuration.
 	typeStr = "logging"
 )
 
-// factory is the factory for logging exporter.
-type factory struct {
+// Factory is the factory for logging exporter.
+type Factory struct {
 }
 
 // Type gets the type of the Exporter config created by this factory.
-func (f *factory) Type() string {
+func (f *Factory) Type() string {
 	return typeStr
 }
 
 // CreateDefaultConfig creates the default configuration for exporter.
-func (f *factory) CreateDefaultConfig() configmodels.Exporter {
+func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
 	return &Config{
 		ExporterSettings: configmodels.ExporterSettings{
 			TypeVal: typeStr,
@@ -53,7 +51,7 @@ func noopStopFunc() error {
 }
 
 // CreateTraceExporter creates a trace exporter based on this config.
-func (f *factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+func (f *Factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
 
 	lexp, err := NewTraceExporter(logger)
 	if err != nil {
@@ -63,7 +61,7 @@ func (f *factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Ex
 }
 
 // CreateMetricsExporter creates a metrics exporter based on this config.
-func (f *factory) CreateMetricsExporter(logger *zap.Logger, cfg configmodels.Exporter) (consumer.MetricsConsumer, exporter.StopFunc, error) {
+func (f *Factory) CreateMetricsExporter(logger *zap.Logger, cfg configmodels.Exporter) (consumer.MetricsConsumer, exporter.StopFunc, error) {
 	lexp, err := NewMetricsExporter(logger)
 	if err != nil {
 		return nil, nil, err
