diff --git a/internal/cmd/pdatagen/internal/templates/message.go.tmpl b/internal/cmd/pdatagen/internal/templates/message.go.tmpl
index aa6700028..b78e0dc45 100644
--- a/internal/cmd/pdatagen/internal/templates/message.go.tmpl
+++ b/internal/cmd/pdatagen/internal/templates/message.go.tmpl
@@ -19,7 +19,7 @@ import (
 //
 // Must use New{{ .structName }} function to create new instances.
 // Important: zero-initialized instance is not valid for use.
-{{- if .isCommon }}
+{{- if .hasWrapper }}
 type {{ .structName }} internal.{{ .structName }}
 {{- else }}
 type {{ .structName }} struct {
@@ -29,7 +29,7 @@ type {{ .structName }} struct {
 {{- end }}
 
 func new{{ .structName }}(orig *{{ .originFullName }}, state *internal.State) {{ .structName }} {
-	{{- if .isCommon }}
+	{{- if .hasWrapper }}
 	return {{ .structName }}(internal.New{{ .structName }}(orig, state))
 	{{- else }}
 	return {{ .structName }}{orig: orig, state: state}
@@ -68,7 +68,7 @@ func (ms {{ .structName }}) CopyTo(dest {{ .structName }}) {
     internal.CopyOrig{{ .originName }}(dest.{{ .origAccessor }}, ms.{{ .origAccessor }})
 }
 
-{{ if .isCommon -}}
+{{ if .hasWrapper -}}
 func (ms {{ .structName }}) getOrig() *{{ .originFullName }} {
 	return internal.GetOrig{{ .structName }}(internal.{{ .structName }}(ms))
 }
