diff --git a/exporter/otlpexporter/config_test.go b/exporter/otlpexporter/config_test.go
index 4a56ef693..b13b7bb6c 100644
--- a/exporter/otlpexporter/config_test.go
+++ b/exporter/otlpexporter/config_test.go
@@ -52,6 +52,7 @@ func TestUnmarshalConfig(t *testing.T) {
 			},
 			QueueConfig: exporterhelper.QueueConfig{
 				Enabled:      true,
+				Sizer:        exporterhelper.RequestSizerTypeRequests,
 				NumConsumers: 2,
 				QueueSize:    10,
 			},
