diff --git a/pdata/pmetric/generated_metricslice.go b/pdata/pmetric/generated_metricslice.go
index 35fc14c7d..1bffcddb3 100644
--- a/pdata/pmetric/generated_metricslice.go
+++ b/pdata/pmetric/generated_metricslice.go
@@ -99,7 +99,7 @@ func (es MetricSlice) EnsureCapacity(newCap int) {
 // It returns the newly added Metric.
 func (es MetricSlice) AppendEmpty() Metric {
 	es.state.AssertMutable()
-	*es.orig = append(*es.orig, internal.NewOrigPtrMetric())
+	*es.orig = append(*es.orig, internal.NewOrigMetric())
 	return es.At(es.Len() - 1)
 }
 
