diff --git a/pdata/ptrace/generated_scopespansslice.go b/pdata/ptrace/generated_scopespansslice.go
index 0583df2c9..b0a85d611 100644
--- a/pdata/ptrace/generated_scopespansslice.go
+++ b/pdata/ptrace/generated_scopespansslice.go
@@ -148,7 +148,7 @@ func (es ScopeSpansSlice) RemoveIf(f func(ScopeSpans) bool) {
 // CopyTo copies all elements from the current slice overriding the destination.
 func (es ScopeSpansSlice) CopyTo(dest ScopeSpansSlice) {
 	dest.state.AssertMutable()
-	*dest.orig = copyOrigScopeSpansSlice(*dest.orig, *es.orig)
+	*dest.orig = internal.CopyOrigScopeSpansSlice(*dest.orig, *es.orig)
 }
 
 // Sort sorts the ScopeSpans elements within ScopeSpansSlice given the
@@ -180,32 +180,3 @@ func (ms ScopeSpansSlice) unmarshalJSONIter(iter *json.Iterator) {
 		return true
 	})
 }
-
-func copyOrigScopeSpansSlice(dest, src []*otlptrace.ScopeSpans) []*otlptrace.ScopeSpans {
-	var newDest []*otlptrace.ScopeSpans
-	if cap(dest) < len(src) {
-		newDest = make([]*otlptrace.ScopeSpans, len(src))
-		// Copy old pointers to re-use.
-		copy(newDest, dest)
-		// Add new pointers for missing elements from len(dest) to len(srt).
-		for i := len(dest); i < len(src); i++ {
-			newDest[i] = &otlptrace.ScopeSpans{}
-		}
-	} else {
-		newDest = dest[:len(src)]
-		// Cleanup the rest of the elements so GC can free the memory.
-		// This can happen when len(src) < len(dest) < cap(dest).
-		for i := len(src); i < len(dest); i++ {
-			dest[i] = nil
-		}
-		// Add new pointers for missing elements.
-		// This can happen when len(dest) < len(src) < cap(dest).
-		for i := len(dest); i < len(src); i++ {
-			newDest[i] = &otlptrace.ScopeSpans{}
-		}
-	}
-	for i := range src {
-		copyOrigScopeSpans(newDest[i], src[i])
-	}
-	return newDest
-}
