diff --git a/pdata/pcommon/common.go b/pdata/pcommon/common.go
index 07d122c79..22af8e4cd 100644
--- a/pdata/pcommon/common.go
+++ b/pdata/pcommon/common.go
@@ -681,7 +681,7 @@ func (m Map) Insert(k string, v Value) {
 
 // InsertNull adds a null Value to the map when the key does not exist.
 // No action is applied to the map where the key already exists.
-// Deprecated: [0.59.0] Use Insert(key, NewValueEmpty()) instead.
+// Deprecated: [0.59.0] Use Get and UpsertEmpty instead.
 func (m Map) InsertNull(k string) {
 	if _, existing := m.Get(k); !existing {
 		*m.getOrig() = append(*m.getOrig(), newAttributeKeyValueNull(k))
