diff --git a/exporter/exporterhelper/logs_batch.go b/exporter/exporterhelper/logs_batch.go
index 0d97239a3..4849c8eba 100644
--- a/exporter/exporterhelper/logs_batch.go
+++ b/exporter/exporterhelper/logs_batch.go
@@ -8,57 +8,78 @@ import (
 	"errors"
 
 	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/sizer"
 	"go.opentelemetry.io/collector/pdata/plog"
 )
 
+func sizerFromConfig(cfg exporterbatcher.SizeConfig) sizer.LogsSizer {
+	switch cfg.Sizer {
+	case exporterbatcher.SizerTypeItems:
+		return &sizer.LogsCountSizer{}
+	case exporterbatcher.SizerTypeBytes:
+		return &sizer.LogsBytesSizer{}
+	default:
+		return &sizer.LogsCountSizer{}
+	}
+}
+
 // MergeSplit splits and/or merges the provided logs request and the current request into one or more requests
 // conforming with the MaxSizeConfig.
 func (req *logsRequest) MergeSplit(_ context.Context, cfg exporterbatcher.SizeConfig, r2 Request) ([]Request, error) {
+	sizer := sizerFromConfig(cfg)
 	if r2 != nil {
 		req2, ok := r2.(*logsRequest)
 		if !ok {
 			return nil, errors.New("invalid input type")
 		}
-		req2.mergeTo(req)
+		req2.mergeTo(req, sizer)
 	}
 
 	// If no limit we can simply merge the new request into the current and return.
 	if cfg.MaxSize == 0 {
 		return []Request{req}, nil
 	}
-	return req.split(cfg)
+
+	return req.split(cfg.MaxSize, sizer), nil
 }
 
-func (req *logsRequest) mergeTo(dst *logsRequest) {
-	dst.setCachedItemsCount(dst.ItemsCount() + req.ItemsCount())
-	req.setCachedItemsCount(0)
+func (req *logsRequest) mergeTo(dst *logsRequest, sizer sizer.LogsSizer) {
+	if sizer != nil {
+		dst.setCachedSize(dst.Size(sizer) + req.Size(sizer))
+		req.setCachedSize(0)
+	}
 	req.ld.ResourceLogs().MoveAndAppendTo(dst.ld.ResourceLogs())
 }
 
-func (req *logsRequest) split(cfg exporterbatcher.SizeConfig) ([]Request, error) {
+func (req *logsRequest) split(maxSize int, sizer sizer.LogsSizer) []Request {
 	var res []Request
-	for req.ItemsCount() > cfg.MaxSize {
-		ld := extractLogs(req.ld, cfg.MaxSize)
-		size := ld.LogRecordCount()
-		req.setCachedItemsCount(req.ItemsCount() - size)
-		res = append(res, &logsRequest{ld: ld, pusher: req.pusher, cachedItemsCount: size})
+	for req.Size(sizer) > maxSize {
+		ld := extractLogs(req.ld, maxSize, sizer)
+		size := sizer.LogsSize(ld)
+		req.setCachedSize(req.Size(sizer) - size)
+		res = append(res, &logsRequest{ld: ld, pusher: req.pusher, cachedSize: size})
 	}
 	res = append(res, req)
-	return res, nil
+	return res
 }
 
 // extractLogs extracts logs from the input logs and returns a new logs with the specified number of log records.
-func extractLogs(srcLogs plog.Logs, count int) plog.Logs {
+func extractLogs(srcLogs plog.Logs, capacity int, sizer sizer.LogsSizer) plog.Logs {
+	capacityReached := false
 	destLogs := plog.NewLogs()
+	capacityLeft := capacity - sizer.LogsSize(destLogs)
 	srcLogs.ResourceLogs().RemoveIf(func(srcRL plog.ResourceLogs) bool {
-		if count == 0 {
+		if capacityReached {
 			return false
 		}
-		needToExtract := resourceLogsCount(srcRL) > count
+		needToExtract := sizer.ResourceLogsSize(srcRL) > capacityLeft
 		if needToExtract {
-			srcRL = extractResourceLogs(srcRL, count)
+			srcRL, capacityReached = extractResourceLogs(srcRL, capacityLeft, sizer)
+			if srcRL.ScopeLogs().Len() == 0 {
+				return false
+			}
 		}
-		count -= resourceLogsCount(srcRL)
+		capacityLeft -= sizer.DeltaSize(sizer.ResourceLogsSize(srcRL))
 		srcRL.MoveTo(destLogs.ResourceLogs().AppendEmpty())
 		return !needToExtract
 	})
@@ -66,46 +87,45 @@ func extractLogs(srcLogs plog.Logs, count int) plog.Logs {
 }
 
 // extractResourceLogs extracts resource logs and returns a new resource logs with the specified number of log records.
-func extractResourceLogs(srcRL plog.ResourceLogs, count int) plog.ResourceLogs {
+func extractResourceLogs(srcRL plog.ResourceLogs, capacity int, sizer sizer.LogsSizer) (plog.ResourceLogs, bool) {
+	capacityReached := false
 	destRL := plog.NewResourceLogs()
 	destRL.SetSchemaUrl(srcRL.SchemaUrl())
 	srcRL.Resource().CopyTo(destRL.Resource())
+	capacityLeft := capacity - sizer.ResourceLogsSize(destRL)
 	srcRL.ScopeLogs().RemoveIf(func(srcSL plog.ScopeLogs) bool {
-		if count == 0 {
+		if capacityReached {
 			return false
 		}
-		needToExtract := srcSL.LogRecords().Len() > count
+		needToExtract := sizer.ScopeLogsSize(srcSL) > capacityLeft
 		if needToExtract {
-			srcSL = extractScopeLogs(srcSL, count)
+			srcSL, capacityReached = extractScopeLogs(srcSL, capacityLeft, sizer)
+			if srcSL.LogRecords().Len() == 0 {
+				return false
+			}
 		}
-		count -= srcSL.LogRecords().Len()
+		capacityLeft -= sizer.DeltaSize(sizer.ScopeLogsSize(srcSL))
 		srcSL.MoveTo(destRL.ScopeLogs().AppendEmpty())
 		return !needToExtract
 	})
-	return destRL
+	return destRL, capacityReached
 }
 
 // extractScopeLogs extracts scope logs and returns a new scope logs with the specified number of log records.
-func extractScopeLogs(srcSL plog.ScopeLogs, count int) plog.ScopeLogs {
+func extractScopeLogs(srcSL plog.ScopeLogs, capacity int, sizer sizer.LogsSizer) (plog.ScopeLogs, bool) {
+	capacityReached := false
 	destSL := plog.NewScopeLogs()
 	destSL.SetSchemaUrl(srcSL.SchemaUrl())
 	srcSL.Scope().CopyTo(destSL.Scope())
+	capacityLeft := capacity - sizer.ScopeLogsSize(destSL)
 	srcSL.LogRecords().RemoveIf(func(srcLR plog.LogRecord) bool {
-		if count == 0 {
+		if capacityReached || sizer.LogRecordSize(srcLR) > capacityLeft {
+			capacityReached = true
 			return false
 		}
+		capacityLeft -= sizer.DeltaSize(sizer.LogRecordSize(srcLR))
 		srcLR.MoveTo(destSL.LogRecords().AppendEmpty())
-		count--
 		return true
 	})
-	return destSL
-}
-
-// resourceLogsCount calculates the total number of log records in the plog.ResourceLogs.
-func resourceLogsCount(rl plog.ResourceLogs) int {
-	count := 0
-	for k := 0; k < rl.ScopeLogs().Len(); k++ {
-		count += rl.ScopeLogs().At(k).LogRecords().Len()
-	}
-	return count
+	return destSL, capacityReached
 }
