diff --git a/pdata/ptrace/json.go b/pdata/ptrace/json.go
index 26fce0786..d3893221b 100644
--- a/pdata/ptrace/json.go
+++ b/pdata/ptrace/json.go
@@ -6,8 +6,6 @@ package ptrace // import "go.opentelemetry.io/collector/pdata/ptrace"
 import (
 	"slices"
 
-	jsoniter "github.com/json-iterator/go"
-
 	"go.opentelemetry.io/collector/pdata/internal"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
@@ -22,7 +20,7 @@ func (*JSONMarshaler) MarshalTraces(td Traces) ([]byte, error) {
 	dest := json.BorrowStream(nil)
 	defer json.ReturnStream(dest)
 	td.marshalJSONStream(dest)
-	return slices.Clone(dest.Buffer()), dest.Error
+	return slices.Clone(dest.Buffer()), dest.Error()
 }
 
 // JSONUnmarshaler unmarshals OTLP/JSON formatted-bytes to pdata.Traces.
@@ -30,24 +28,24 @@ type JSONUnmarshaler struct{}
 
 // UnmarshalTraces from OTLP/JSON format into pdata.Traces.
 func (*JSONUnmarshaler) UnmarshalTraces(buf []byte) (Traces, error) {
-	iter := jsoniter.ConfigFastest.BorrowIterator(buf)
-	defer jsoniter.ConfigFastest.ReturnIterator(iter)
+	iter := json.BorrowIterator(buf)
+	defer json.ReturnIterator(iter)
 	td := NewTraces()
 	td.unmarshalJSONIter(iter)
-	if iter.Error != nil {
-		return Traces{}, iter.Error
+	if iter.Error() != nil {
+		return Traces{}, iter.Error()
 	}
 	otlp.MigrateTraces(td.getOrig().ResourceSpans)
 	return td, nil
 }
 
-func (ms ResourceSpans) unmarshalJSONIter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+func (ms ResourceSpans) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "resource":
 			internal.UnmarshalJSONIterResource(internal.NewResource(&ms.orig.Resource, ms.state), iter)
 		case "scopeSpans", "scope_spans":
-			iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+			iter.ReadArrayCB(func(iter *json.Iterator) bool {
 				ms.ScopeSpans().AppendEmpty().unmarshalJSONIter(iter)
 				return true
 			})
@@ -60,13 +58,13 @@ func (ms ResourceSpans) unmarshalJSONIter(iter *jsoniter.Iterator) {
 	})
 }
 
-func (ms ScopeSpans) unmarshalJSONIter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+func (ms ScopeSpans) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "scope":
 			internal.UnmarshalJSONIterInstrumentationScope(internal.NewInstrumentationScope(&ms.orig.Scope, ms.state), iter)
 		case "spans":
-			iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+			iter.ReadArrayCB(func(iter *json.Iterator) bool {
 				ms.Spans().AppendEmpty().unmarshalJSONIter(iter)
 				return true
 			})
@@ -79,8 +77,8 @@ func (ms ScopeSpans) unmarshalJSONIter(iter *jsoniter.Iterator) {
 	})
 }
 
-func (ms Span) unmarshalJSONIter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+func (ms Span) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "traceId", "trace_id":
 			ms.orig.TraceId.UnmarshalJSONIter(iter)
@@ -91,33 +89,33 @@ func (ms Span) unmarshalJSONIter(iter *jsoniter.Iterator) {
 		case "parentSpanId", "parent_span_id":
 			ms.orig.ParentSpanId.UnmarshalJSONIter(iter)
 		case "flags":
-			ms.orig.Flags = json.ReadUint32(iter)
+			ms.orig.Flags = iter.ReadUint32()
 		case "name":
 			ms.orig.Name = iter.ReadString()
 		case "kind":
-			ms.orig.Kind = otlptrace.Span_SpanKind(json.ReadEnumValue(iter, otlptrace.Span_SpanKind_value))
+			ms.orig.Kind = otlptrace.Span_SpanKind(iter.ReadEnumValue(otlptrace.Span_SpanKind_value))
 		case "startTimeUnixNano", "start_time_unix_nano":
-			ms.orig.StartTimeUnixNano = json.ReadUint64(iter)
+			ms.orig.StartTimeUnixNano = iter.ReadUint64()
 		case "endTimeUnixNano", "end_time_unix_nano":
-			ms.orig.EndTimeUnixNano = json.ReadUint64(iter)
+			ms.orig.EndTimeUnixNano = iter.ReadUint64()
 		case "attributes":
 			internal.UnmarshalJSONIterMap(internal.NewMap(&ms.orig.Attributes, ms.state), iter)
 		case "droppedAttributesCount", "dropped_attributes_count":
-			ms.orig.DroppedAttributesCount = json.ReadUint32(iter)
+			ms.orig.DroppedAttributesCount = iter.ReadUint32()
 		case "events":
-			iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+			iter.ReadArrayCB(func(iter *json.Iterator) bool {
 				ms.Events().AppendEmpty().unmarshalJSONIter(iter)
 				return true
 			})
 		case "droppedEventsCount", "dropped_events_count":
-			ms.orig.DroppedEventsCount = json.ReadUint32(iter)
+			ms.orig.DroppedEventsCount = iter.ReadUint32()
 		case "links":
-			iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+			iter.ReadArrayCB(func(iter *json.Iterator) bool {
 				ms.Links().AppendEmpty().unmarshalJSONIter(iter)
 				return true
 			})
 		case "droppedLinksCount", "dropped_links_count":
-			ms.orig.DroppedLinksCount = json.ReadUint32(iter)
+			ms.orig.DroppedLinksCount = iter.ReadUint32()
 		case "status":
 			ms.Status().unmarshalJSONIter(iter)
 		default:
@@ -127,13 +125,13 @@ func (ms Span) unmarshalJSONIter(iter *jsoniter.Iterator) {
 	})
 }
 
-func (ms Status) unmarshalJSONIter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+func (ms Status) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "message":
 			ms.orig.Message = iter.ReadString()
 		case "code":
-			ms.orig.Code = otlptrace.Status_StatusCode(json.ReadEnumValue(iter, otlptrace.Status_StatusCode_value))
+			ms.orig.Code = otlptrace.Status_StatusCode(iter.ReadEnumValue(otlptrace.Status_StatusCode_value))
 		default:
 			iter.Skip()
 		}
@@ -141,8 +139,8 @@ func (ms Status) unmarshalJSONIter(iter *jsoniter.Iterator) {
 	})
 }
 
-func (ms SpanLink) unmarshalJSONIter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+func (ms SpanLink) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "traceId", "trace_id":
 			ms.orig.TraceId.UnmarshalJSONIter(iter)
@@ -153,9 +151,9 @@ func (ms SpanLink) unmarshalJSONIter(iter *jsoniter.Iterator) {
 		case "attributes":
 			internal.UnmarshalJSONIterMap(internal.NewMap(&ms.orig.Attributes, ms.state), iter)
 		case "droppedAttributesCount", "dropped_attributes_count":
-			ms.orig.DroppedAttributesCount = json.ReadUint32(iter)
+			ms.orig.DroppedAttributesCount = iter.ReadUint32()
 		case "flags":
-			ms.orig.Flags = json.ReadUint32(iter)
+			ms.orig.Flags = iter.ReadUint32()
 		default:
 			iter.Skip()
 		}
@@ -163,17 +161,17 @@ func (ms SpanLink) unmarshalJSONIter(iter *jsoniter.Iterator) {
 	})
 }
 
-func (ms SpanEvent) unmarshalJSONIter(iter *jsoniter.Iterator) {
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+func (ms SpanEvent) unmarshalJSONIter(iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "timeUnixNano", "time_unix_nano":
-			ms.orig.TimeUnixNano = json.ReadUint64(iter)
+			ms.orig.TimeUnixNano = iter.ReadUint64()
 		case "name":
 			ms.orig.Name = iter.ReadString()
 		case "attributes":
 			internal.UnmarshalJSONIterMap(internal.NewMap(&ms.orig.Attributes, ms.state), iter)
 		case "droppedAttributesCount", "dropped_attributes_count":
-			ms.orig.DroppedAttributesCount = json.ReadUint32(iter)
+			ms.orig.DroppedAttributesCount = iter.ReadUint32()
 		default:
 			iter.Skip()
 		}
