diff --git a/receiver/fluentforwardreceiver/factory.go b/receiver/fluentforwardreceiver/factory.go
index 5061ad641..80ee70113 100644
--- a/receiver/fluentforwardreceiver/factory.go
+++ b/receiver/fluentforwardreceiver/factory.go
@@ -50,7 +50,7 @@ func createLogsReceiver(
 	_ context.Context,
 	params component.ReceiverCreateParams,
 	cfg configmodels.Receiver,
-	consumer consumer.LogsConsumer,
+	consumer consumer.Logs,
 ) (component.LogsReceiver, error) {
 
 	rCfg := cfg.(*Config)
