diff --git a/service/pipelines/config_test.go b/service/pipelines/config_test.go
index eba5fa0cb..19ca0be6d 100644
--- a/service/pipelines/config_test.go
+++ b/service/pipelines/config_test.go
@@ -5,10 +5,8 @@ package pipelines
 
 import (
 	"errors"
-	"fmt"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/component"
@@ -36,7 +34,7 @@ func TestConfigValidate(t *testing.T) {
 				pipe.Processors = append(pipe.Processors, pipe.Processors...)
 				return cfg
 			},
-			expected: fmt.Errorf(`pipeline "traces": %w`, errors.New(`references processor "nop" multiple times`)),
+			expected: errors.New(`references processor "nop" multiple times`),
 		},
 		{
 			name: "missing-pipeline-receivers",
@@ -45,7 +43,7 @@ func TestConfigValidate(t *testing.T) {
 				cfg[pipeline.NewID(pipeline.SignalTraces)].Receivers = nil
 				return cfg
 			},
-			expected: fmt.Errorf(`pipeline "traces": %w`, errMissingServicePipelineReceivers),
+			expected: errMissingServicePipelineReceivers,
 		},
 		{
 			name: "missing-pipeline-exporters",
@@ -54,7 +52,7 @@ func TestConfigValidate(t *testing.T) {
 				cfg[pipeline.NewID(pipeline.SignalTraces)].Exporters = nil
 				return cfg
 			},
-			expected: fmt.Errorf(`pipeline "traces": %w`, errMissingServicePipelineExporters),
+			expected: errMissingServicePipelineExporters,
 		},
 		{
 			name: "missing-pipelines",
@@ -74,7 +72,7 @@ func TestConfigValidate(t *testing.T) {
 				}
 				return cfg
 			},
-			expected: errors.New(`pipeline "wrongtype": unknown signal "wrongtype"`),
+			expected: errors.New(`unknown signal "wrongtype"`),
 		},
 		{
 			name: "disabled-featuregate-profiles",
@@ -87,7 +85,7 @@ func TestConfigValidate(t *testing.T) {
 				}
 				return cfg
 			},
-			expected: errors.New(`pipeline "profiles": profiling signal support is at alpha level, gated under the "service.profilesSupport" feature gate`),
+			expected: errors.New(`profiling signal support is at alpha level, gated under the "service.profilesSupport" feature gate`),
 		},
 		{
 			name: "enabled-featuregate-profiles",
@@ -109,7 +107,11 @@ func TestConfigValidate(t *testing.T) {
 	for _, tt := range testCases {
 		t.Run(tt.name, func(t *testing.T) {
 			cfg := tt.cfgFn(t)
-			assert.Equal(t, tt.expected, cfg.Validate())
+			if tt.expected != nil {
+				require.ErrorContains(t, component.ValidateConfig(cfg), tt.expected.Error())
+			} else {
+				require.NoError(t, component.ValidateConfig(cfg))
+			}
 
 			// Clean up the profiles support gate, which may have been enabled in `cfgFn`.
 			require.NoError(t, featuregate.GlobalRegistry().Set(serviceProfileSupportGateID, false))
