diff --git a/service/collector.go b/service/collector.go
index 0c4687a49..85bf43202 100644
--- a/service/collector.go
+++ b/service/collector.go
@@ -183,7 +183,8 @@ LOOP:
 			col.logger.Info("Received shutdown request")
 			break LOOP
 		case <-ctx.Done():
-			col.logger.Error("Context done, terminating process", zap.Error(ctx.Err()))
+			col.logger.Info("Context done, terminating process", zap.Error(ctx.Err()))
+
 			// Call shutdown with background context as the passed in context has been canceled
 			return col.shutdown(context.Background())
 		}
