diff --git a/receiver/jaegerreceiver/config.go b/receiver/jaegerreceiver/config.go
index 406cc8d4a..d1920432a 100644
--- a/receiver/jaegerreceiver/config.go
+++ b/receiver/jaegerreceiver/config.go
@@ -18,7 +18,7 @@ import (
 	"go.opentelemetry.io/collector/config/configgrpc"
 	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/config/configmodels"
-	"go.opentelemetry.io/collector/config/configprotocol"
+	"go.opentelemetry.io/collector/config/confignet"
 )
 
 // The config field name to load the protocol map from
@@ -32,10 +32,10 @@ type RemoteSamplingConfig struct {
 }
 
 type Protocols struct {
-	GRPC          *configgrpc.GRPCServerSettings         `mapstructure:"grpc"`
-	ThriftHTTP    *confighttp.HTTPServerSettings         `mapstructure:"thrift_http"`
-	ThriftBinary  *configprotocol.ProtocolServerSettings `mapstructure:"thrift_binary"`
-	ThriftCompact *configprotocol.ProtocolServerSettings `mapstructure:"thrift_compact"`
+	GRPC          *configgrpc.GRPCServerSettings `mapstructure:"grpc"`
+	ThriftHTTP    *confighttp.HTTPServerSettings `mapstructure:"thrift_http"`
+	ThriftBinary  *confignet.TCPAddr             `mapstructure:"thrift_binary"`
+	ThriftCompact *confignet.TCPAddr             `mapstructure:"thrift_compact"`
 }
 
 // Config defines configuration for Jaeger receiver.
