diff --git a/service/internal/graph/host.go b/service/internal/graph/host.go
index c039d972b..692f6913c 100644
--- a/service/internal/graph/host.go
+++ b/service/internal/graph/host.go
@@ -11,7 +11,6 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componentstatus"
-	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/extension"
 	"go.opentelemetry.io/collector/featuregate"
@@ -35,7 +34,7 @@ type Host struct {
 	Receivers         builders.Receiver
 	Processors        *processor.Builder
 	Exporters         *exporter.Builder
-	Connectors        *connector.Builder
+	Connectors        builders.Connector
 	Extensions        *extension.Builder
 
 	ModuleInfo extension.ModuleInfo
