diff --git a/internal/fanoutconsumer/traces_test.go b/internal/fanoutconsumer/traces_test.go
index b17e674c4..212a81a69 100644
--- a/internal/fanoutconsumer/traces_test.go
+++ b/internal/fanoutconsumer/traces_test.go
@@ -9,6 +9,7 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumertest"
@@ -222,7 +223,7 @@ func TestTracesWhenErrors(t *testing.T) {
 	td := testdata.GenerateTraces(1)
 
 	for i := 0; i < 2; i++ {
-		assert.Error(t, tfc.ConsumeTraces(context.Background(), td))
+		require.Error(t, tfc.ConsumeTraces(context.Background(), td))
 	}
 
 	assert.Equal(t, td, p3.AllTraces()[0])
