diff --git a/internal/tools/tools.go b/internal/tools/tools.go
index 418c1ef77..854944169 100644
--- a/internal/tools/tools.go
+++ b/internal/tools/tools.go
@@ -29,7 +29,6 @@ import (
 	_ "github.com/jcchavezs/porto/cmd/porto"
 	_ "github.com/ory/go-acc"
 	_ "github.com/pavius/impi/cmd/impi"
-	_ "github.com/tcnksm/ghr"
 	_ "github.com/wadey/gocovmerge"
 	_ "go.opentelemetry.io/build-tools/checkdoc"
 	_ "go.opentelemetry.io/build-tools/chloggen"
