diff --git a/scraper/scraperhelper/obs_metrics.go b/scraper/scraperhelper/obs_metrics.go
index 8723215da..2c3105153 100644
--- a/scraper/scraperhelper/obs_metrics.go
+++ b/scraper/scraperhelper/obs_metrics.go
@@ -37,28 +37,28 @@ const (
 	formatKey = "format"
 )
 
-func newObsMetrics(delegate scraper.ScrapeMetricsFunc, receiverID component.ID, scraperID component.ID, telSettings component.TelemetrySettings) (scraper.ScrapeMetricsFunc, error) {
-	telemetryBuilder, errBuilder := metadata.NewTelemetryBuilder(telSettings)
+func wrapObsMetrics(sc scraper.Metrics, receiverID component.ID, scraperID component.ID, set component.TelemetrySettings) (scraper.Metrics, error) {
+	telemetryBuilder, errBuilder := metadata.NewTelemetryBuilder(set)
 	if errBuilder != nil {
 		return nil, errBuilder
 	}
 
-	tracer := metadata.Tracer(telSettings)
+	tracer := metadata.Tracer(set)
 	spanName := scraperKey + spanNameSep + scraperID.String() + spanNameSep + "ScrapeMetrics"
 	otelAttrs := metric.WithAttributeSet(attribute.NewSet(
 		attribute.String(receiverKey, receiverID.String()),
 		attribute.String(scraperKey, scraperID.String()),
 	))
 
-	return func(ctx context.Context) (pmetric.Metrics, error) {
+	scraperFuncs := func(ctx context.Context) (pmetric.Metrics, error) {
 		ctx, span := tracer.Start(ctx, spanName)
 		defer span.End()
 
-		md, err := delegate(ctx)
+		md, err := sc.ScrapeMetrics(ctx)
 		numScrapedMetrics := 0
 		numErroredMetrics := 0
 		if err != nil {
-			telSettings.Logger.Error("Error scraping metrics", zap.Error(err))
+			set.Logger.Error("Error scraping metrics", zap.Error(err))
 			var partialErr scrapererror.PartialScrapeError
 			if errors.As(err, &partialErr) {
 				numErroredMetrics = partialErr.Failed
@@ -85,5 +85,7 @@ func newObsMetrics(delegate scraper.ScrapeMetricsFunc, receiverID component.ID,
 		}
 
 		return md, err
-	}, nil
+	}
+
+	return scraper.NewMetrics(scraperFuncs, scraper.WithStart(sc.Start), scraper.WithShutdown(sc.Shutdown))
 }
