diff --git a/exporter/go.mod b/exporter/go.mod
index c837181d9..e353c8a8b 100644
--- a/exporter/go.mod
+++ b/exporter/go.mod
@@ -5,14 +5,14 @@ go 1.21
 require (
 	github.com/cenkalti/backoff/v4 v4.2.1
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector v0.95.0
-	go.opentelemetry.io/collector/component v0.95.0
-	go.opentelemetry.io/collector/config/configretry v0.95.0
-	go.opentelemetry.io/collector/config/configtelemetry v0.95.0
-	go.opentelemetry.io/collector/consumer v0.95.0
-	go.opentelemetry.io/collector/extension v0.95.0
-	go.opentelemetry.io/collector/pdata v1.2.0
-	go.opentelemetry.io/collector/receiver v0.95.0
+	go.opentelemetry.io/collector v0.96.0
+	go.opentelemetry.io/collector/component v0.96.0
+	go.opentelemetry.io/collector/config/configretry v0.96.0
+	go.opentelemetry.io/collector/config/configtelemetry v0.96.0
+	go.opentelemetry.io/collector/consumer v0.96.0
+	go.opentelemetry.io/collector/extension v0.96.0
+	go.opentelemetry.io/collector/pdata v1.3.0
+	go.opentelemetry.io/collector/receiver v0.96.0
 	go.opentelemetry.io/otel v1.24.0
 	go.opentelemetry.io/otel/metric v1.24.0
 	go.opentelemetry.io/otel/sdk v1.24.0
@@ -46,7 +46,7 @@ require (
 	github.com/prometheus/client_model v0.6.0 // indirect
 	github.com/prometheus/common v0.48.0 // indirect
 	github.com/prometheus/procfs v0.12.0 // indirect
-	go.opentelemetry.io/collector/confmap v0.95.0 // indirect
+	go.opentelemetry.io/collector/confmap v0.96.0 // indirect
 	go.opentelemetry.io/otel/exporters/prometheus v0.46.0 // indirect
 	go.opentelemetry.io/otel/sdk/metric v1.24.0 // indirect
 	golang.org/x/net v0.21.0 // indirect
