diff --git a/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go b/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go
index cee43688f..9d2ac6fed 100644
--- a/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go
+++ b/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"runtime"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -29,15 +30,12 @@ import (
 
 func newFakeRequestSettings() Settings[request.Request] {
 	return Settings[request.Request]{
-		Signal:    pipeline.SignalMetrics,
-		ID:        component.NewID(exportertest.NopType),
-		Telemetry: componenttest.NewNopTelemetrySettings(),
-		Encoding:  newFakeEncoding(&requesttest.FakeRequest{}),
-		Sizers: map[request.SizerType]request.Sizer[request.Request]{
-			request.SizerTypeRequests: request.RequestsSizer[request.Request]{},
-			request.SizerTypeItems:    request.NewItemsSizer(),
-			request.SizerTypeBytes:    newFakeBytesSizer(),
-		},
+		Signal:     pipeline.SignalMetrics,
+		ID:         component.NewID(exportertest.NopType),
+		Telemetry:  componenttest.NewNopTelemetrySettings(),
+		Encoding:   newFakeEncoding(&requesttest.FakeRequest{}),
+		ItemsSizer: request.NewItemsSizer(),
+		BytesSizer: requesttest.NewBytesSizer(),
 	}
 }
 
@@ -189,8 +187,11 @@ func TestQueueBatchPersistentEnabled_NoDataLossOnShutdown(t *testing.T) {
 	mockReq := &requesttest.FakeRequest{Items: 2}
 	qSet := newFakeRequestSettings()
 	qSet.Encoding = newFakeEncoding(mockReq)
+
+	consumed := &atomic.Bool{}
 	done := make(chan struct{})
 	qb, err := NewQueueBatch(qSet, cfg, func(context.Context, request.Request) error {
+		consumed.Store(true)
 		<-done
 		return experr.NewShutdownErr(errors.New("could not export data"))
 	})
@@ -207,7 +208,7 @@ func TestQueueBatchPersistentEnabled_NoDataLossOnShutdown(t *testing.T) {
 
 	// first wait for the item to be consumed from the queue
 	assert.Eventually(t, func() bool {
-		return qb.queue.Size() == 0
+		return consumed.Load()
 	}, 1*time.Second, 10*time.Millisecond)
 
 	// shuts down the exporter, unsent data should be preserved as in-flight data in the persistent queue.
