diff --git a/otelcol/collector_windows.go b/otelcol/collector_windows.go
index 142c811ee..3df08386b 100644
--- a/otelcol/collector_windows.go
+++ b/otelcol/collector_windows.go
@@ -83,7 +83,7 @@ func (s *windowsService) start(elog *eventlog.Log, colErrorChannel chan error) e
 	}
 
 	var err error
-	err = updateSettingsUsingFlags(&s.settings, s.flags, false)
+	err = updateSettingsUsingFlags(&s.settings, s.flags)
 	if err != nil {
 		return err
 	}
