diff --git a/otelcol/command.go b/otelcol/command.go
index cd0ee98e7..8516ec831 100644
--- a/otelcol/command.go
+++ b/otelcol/command.go
@@ -17,29 +17,15 @@ import (
 // Any URIs specified in CollectorSettings.ConfigProviderSettings.ResolverSettings.URIs
 // are considered defaults and will be overwritten by config flags passed as
 // command-line arguments to the executable.
-//
-// Deprecated: [v0.103.0] use NewCommandMustSetProvider instead
+// At least one Provider must be set.
 func NewCommand(set CollectorSettings) *cobra.Command {
-	return commandHelper(set, false)
-}
-
-// NewCommandMustSetProvider constructs a new cobra.Command using the given CollectorSettings.
-// Any URIs specified in CollectorSettings.ConfigProviderSettings.ResolverSettings.URIs
-// are considered defaults and will be overwritten by config flags passed as
-// command-line arguments to the executable.
-// At least one Provider must be supplied via CollectorSettings.ConfigProviderSettings.ResolverSettings.ProviderFactories.
-func NewCommandMustSetProvider(set CollectorSettings) *cobra.Command {
-	return commandHelper(set, true)
-}
-
-func commandHelper(set CollectorSettings, enforceProviders bool) *cobra.Command {
 	flagSet := flags(featuregate.GlobalRegistry())
 	rootCmd := &cobra.Command{
 		Use:          set.BuildInfo.Command,
 		Version:      set.BuildInfo.Version,
 		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, _ []string) error {
-			err := updateSettingsUsingFlags(&set, flagSet, enforceProviders)
+			err := updateSettingsUsingFlags(&set, flagSet)
 			if err != nil {
 				return err
 			}
@@ -52,13 +38,24 @@ func commandHelper(set CollectorSettings, enforceProviders bool) *cobra.Command
 		},
 	}
 	rootCmd.AddCommand(newComponentsCommand(set))
-	rootCmd.AddCommand(newValidateSubCommand(set, flagSet, enforceProviders))
+	rootCmd.AddCommand(newValidateSubCommand(set, flagSet))
 	rootCmd.Flags().AddGoFlagSet(flagSet)
 	return rootCmd
 }
 
+// NewCommandMustSetProvider constructs a new cobra.Command using the given CollectorSettings.
+// Any URIs specified in CollectorSettings.ConfigProviderSettings.ResolverSettings.URIs
+// are considered defaults and will be overwritten by config flags passed as
+// command-line arguments to the executable.
+// At least one Provider must be supplied via CollectorSettings.ConfigProviderSettings.ResolverSettings.ProviderFactories.
+//
+// Deprecated: [v0.104.0] use NewCommand instead
+func NewCommandMustSetProvider(set CollectorSettings) *cobra.Command {
+	return NewCommand(set)
+}
+
 // Puts command line flags from flags into the CollectorSettings, to be used during config resolution.
-func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet, enforceProviders bool) error {
+func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet) error {
 	resolverSet := &set.ConfigProviderSettings.ResolverSettings
 	configFlags := getConfigFlag(flags)
 
@@ -73,14 +70,8 @@ func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet, enfor
 		set.ConfigProviderSettings.ResolverSettings.DefaultScheme = "env"
 	}
 
-	// Provide a default set of providers and converters if none have been specified.
-	// TODO: Remove this after CollectorSettings.ConfigProvider is removed and instead
-	// do it in the builder.
-	if len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
-		if enforceProviders {
-			return errors.New("at least one Provider must be supplied")
-		}
-		set.ConfigProviderSettings = newDefaultConfigProviderSettings(resolverSet.URIs)
+	if len(resolverSet.ProviderFactories) == 0 {
+		return errors.New("at least one Provider must be supplied")
 	}
 	return nil
 }
