diff --git a/testbed/testbed/data_providers.go b/testbed/testbed/data_providers.go
index 54c170807..74964ed0c 100644
--- a/testbed/testbed/data_providers.go
+++ b/testbed/testbed/data_providers.go
@@ -19,6 +19,7 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"path/filepath"
 	"strconv"
 	"time"
 
@@ -327,7 +328,7 @@ type FileDataProvider struct {
 // NewFileDataProvider creates an instance of FileDataProvider which generates test data
 // loaded from a file.
 func NewFileDataProvider(filePath string, dataType configmodels.DataType) (*FileDataProvider, error) {
-	file, err := os.OpenFile(filePath, os.O_RDONLY, 0)
+	file, err := os.OpenFile(filepath.Clean(filePath), os.O_RDONLY, 0)
 	if err != nil {
 		return nil, err
 	}
