diff --git a/internal/goldendataset/pict_metrics_input_defs.go b/internal/goldendataset/pict_metrics_input_defs.go
index 9cf0e7f3a..fd57ec558 100644
--- a/internal/goldendataset/pict_metrics_input_defs.go
+++ b/internal/goldendataset/pict_metrics_input_defs.go
@@ -51,7 +51,7 @@ const (
 	LabelsMany PICTNumPtLabels = "ManyLabels"
 )
 
-// Enumerates the number of points on each metric.
+// PICTNumPtsPerMetric enum for the number of points on each metric.
 type PICTNumPtsPerMetric string
 
 const (
@@ -59,7 +59,7 @@ const (
 	NumPtsPerMetricMany PICTNumPtsPerMetric = "ManyPts"
 )
 
-// Enumerates the number of attributes on each resource.
+// PICTNumResourceAttrs enum for the number of attributes on each resource.
 type PICTNumResourceAttrs string
 
 const (
