diff --git a/service/collector_windows.go b/service/collector_windows.go
index e3f22c214..ca0302d23 100644
--- a/service/collector_windows.go
+++ b/service/collector_windows.go
@@ -22,6 +22,7 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"os"
 	"time"
 
 	"go.uber.org/zap"
@@ -56,11 +57,6 @@ func (s *windowsService) Execute(args []string, requests <-chan svc.ChangeReques
 		return false, 1501 // 1501: ERROR_EVENTLOG_CANT_START
 	}
 
-	// Parse all the flags manually.
-	if err = s.flags.Parse(args[1:]); err != nil {
-		return false, 1639 // 1639: ERROR_INVALID_COMMAND_LINE
-	}
-
 	colErrorChannel := make(chan error, 1)
 
 	changes <- svc.Status{State: svc.StartPending}
@@ -93,6 +89,11 @@ func (s *windowsService) Execute(args []string, requests <-chan svc.ChangeReques
 }
 
 func (s *windowsService) start(elog *eventlog.Log, colErrorChannel chan error) error {
+	// Parse all the flags manually.
+	if err := s.flags.Parse(os.Args[1:]); err != nil {
+		return err
+	}
+
 	if err := featuregate.GetRegistry().Apply(getFeatureGatesFlag(s.flags)); err != nil {
 		return err
 	}
