diff --git a/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl b/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl
index 52d8f5c98..26d8c8d47 100644
--- a/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl
+++ b/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl
@@ -77,7 +77,7 @@ func MarshalJSONStream{{ .structName }}(ms {{ .structName }}, dest *json.Stream)
 	dest.WriteArrayEnd()
 }
 
-// UnmarshalJSONStream{{ .structName }} unmarshals all properties from the current struct from the source iterator.
+// UnmarshalJSONIter{{ .structName }} unmarshals all properties from the current struct from the source iterator.
 func UnmarshalJSONIter{{ .structName }}(ms {{ .structName }}, iter *json.Iterator) {
 	iter.ReadArrayCB(func(iter *json.Iterator) bool {
 		*ms.orig = append(*ms.orig, {{ .emptyOriginElement }})
