diff --git a/go.mod b/go.mod
index 6dabf4484..d12c8e42d 100644
--- a/go.mod
+++ b/go.mod
@@ -19,7 +19,7 @@ require (
 	go.opentelemetry.io/collector/pdata v1.0.0-rcv0014
 	go.opentelemetry.io/collector/processor v0.85.0
 	go.opentelemetry.io/collector/receiver v0.85.0
-	go.opentelemetry.io/collector/service v0.0.0-00010101000000-000000000000
+	go.opentelemetry.io/collector/service v0.0.0-20230915215502-07938f20fcc7
 	go.opentelemetry.io/otel v1.18.0
 	go.opentelemetry.io/otel/exporters/prometheus v0.41.0
 	go.opentelemetry.io/otel/metric v1.18.0
