diff --git a/otelcol/otelcoltest/config_test.go b/otelcol/otelcoltest/config_test.go
index ba259ddc6..94d8c616e 100644
--- a/otelcol/otelcoltest/config_test.go
+++ b/otelcol/otelcoltest/config_test.go
@@ -11,6 +11,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pipeline"
 	"go.opentelemetry.io/collector/service/pipelines"
 )
 
@@ -48,14 +49,14 @@ func TestLoadConfig(t *testing.T) {
 	// Verify service.
 	require.Len(t, cfg.Service.Extensions, 1)
 	assert.Contains(t, cfg.Service.Extensions, component.MustNewID("nop"))
-	require.Len(t, cfg.Service.Pipelines, 1)
+	require.Len(t, cfg.Service.PipelinesWithPipelineID, 1)
 	assert.Equal(t,
 		&pipelines.PipelineConfig{
 			Receivers:  []component.ID{component.MustNewID("nop")},
 			Processors: []component.ID{component.MustNewID("nop")},
 			Exporters:  []component.ID{component.MustNewID("nop")},
 		},
-		cfg.Service.Pipelines[component.MustNewID("traces")],
+		cfg.Service.PipelinesWithPipelineID[pipeline.MustNewID("traces")],
 		"Did not load pipeline config correctly")
 }
 
