diff --git a/service/service_test.go b/service/service_test.go
index f9643e830..ebb55a8a2 100644
--- a/service/service_test.go
+++ b/service/service_test.go
@@ -75,7 +75,7 @@ func TestService_GetExtensions(t *testing.T) {
 	extMap := srv.GetExtensions()
 
 	assert.Len(t, extMap, 1)
-	assert.Contains(t, extMap, config.NewID("nop"))
+	assert.Contains(t, extMap, config.NewComponentID("nop"))
 }
 
 func TestService_GetExporters(t *testing.T) {
@@ -89,11 +89,11 @@ func TestService_GetExporters(t *testing.T) {
 	expMap := srv.GetExporters()
 	assert.Len(t, expMap, 3)
 	assert.Len(t, expMap[config.TracesDataType], 1)
-	assert.Contains(t, expMap[config.TracesDataType], config.NewID("nop"))
+	assert.Contains(t, expMap[config.TracesDataType], config.NewComponentID("nop"))
 	assert.Len(t, expMap[config.MetricsDataType], 1)
-	assert.Contains(t, expMap[config.MetricsDataType], config.NewID("nop"))
+	assert.Contains(t, expMap[config.MetricsDataType], config.NewComponentID("nop"))
 	assert.Len(t, expMap[config.LogsDataType], 1)
-	assert.Contains(t, expMap[config.LogsDataType], config.NewID("nop"))
+	assert.Contains(t, expMap[config.LogsDataType], config.NewComponentID("nop"))
 }
 
 func createExampleService(t *testing.T) *service {
