diff --git a/consumer/go.mod b/consumer/go.mod
index 207f91848..4abbc6e33 100644
--- a/consumer/go.mod
+++ b/consumer/go.mod
@@ -37,8 +37,6 @@ replace go.opentelemetry.io/collector/exporter => ../exporter
 
 replace go.opentelemetry.io/collector/extension => ../extension
 
-replace go.opentelemetry.io/collector/extension/auth => ../extension/auth
-
 replace go.opentelemetry.io/collector/extension/zpagesextension => ../extension/zpagesextension
 
 replace go.opentelemetry.io/collector/featuregate => ../featuregate
@@ -58,14 +56,6 @@ retract (
 
 replace go.opentelemetry.io/collector/connector => ../connector
 
-replace go.opentelemetry.io/collector/config/configcompression => ../config/configcompression
-
 replace go.opentelemetry.io/collector/config/confignet => ../config/confignet
 
-replace go.opentelemetry.io/collector/config/configopaque => ../config/configopaque
-
-replace go.opentelemetry.io/collector/config/configtls => ../config/configtls
-
 replace go.opentelemetry.io/collector/config/configtelemetry => ../config/configtelemetry
-
-replace go.opentelemetry.io/collector/config/configauth => ../config/configauth
