diff --git a/config/parser.go b/config/parser.go
index 24f08fcf9..5ee4b2aa6 100644
--- a/config/parser.go
+++ b/config/parser.go
@@ -12,7 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// package config implements the configuration Parser.
 package config
 
 import (
@@ -58,6 +57,7 @@ func ParserFromViper(v *viper.Viper) *Parser {
 	return &Parser{v: v}
 }
 
+// NewParserFromStringMap creates a parser from a map[string]interface{}.
 func NewParserFromStringMap(data map[string]interface{}) *Parser {
 	v := NewViper()
 	// Cannot return error because the subv is empty.
