diff --git a/internal/tools/go.mod b/internal/tools/go.mod
index 1ebcfd6fb..ceac8f969 100644
--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -6,7 +6,6 @@ require (
 	github.com/client9/misspell v0.3.4
 	github.com/golangci/golangci-lint v1.42.1
 	github.com/google/addlicense v1.0.0
-	github.com/jstemmer/go-junit-report v0.9.1
 	github.com/mjibson/esc v0.2.0
 	github.com/ory/go-acc v0.2.6
 	github.com/pavius/impi v0.0.3
