diff --git a/cmd/mdatagen/internal/templates/component_test.go.tmpl b/cmd/mdatagen/internal/templates/component_test.go.tmpl
index 99990cbbb..fc8a8b8d4 100644
--- a/cmd/mdatagen/internal/templates/component_test.go.tmpl
+++ b/cmd/mdatagen/internal/templates/component_test.go.tmpl
@@ -171,8 +171,8 @@ func TestComponentLifecycle(t *testing.T) {
 	factory := NewFactory()
 
 	tests := []struct{
-		name string
 		createFn func(ctx context.Context, set processor.Settings, cfg component.Config) (component.Component, error)
+		name string
 	}{
 {{ if supportsLogs }}
 		{
@@ -266,8 +266,8 @@ func TestComponentLifecycle(t *testing.T) {
 	factory := NewFactory()
 
 	tests := []struct{
-		name string
 		createFn func(ctx context.Context, set receiver.Settings, cfg component.Config) (component.Component, error)
+		name string
 	}{
 {{ if supportsLogs }}
 		{
@@ -335,8 +335,8 @@ func TestComponentLifecycle(t *testing.T) {
 	factory := NewFactory()
 
 	tests := []struct{
-		name string
 		createFn func(ctx context.Context, set scraper.Settings, cfg component.Config) (component.Component, error)
+		name string
 	}{
 {{ if supportsLogs }}
 		{
@@ -440,8 +440,8 @@ func TestComponentLifecycle(t *testing.T) {
 	factory := NewFactory()
 
 	tests := []struct{
-		name string
 		createFn func(ctx context.Context, set connector.Settings, cfg component.Config) (component.Component, error)
+		name string
 	}{
 {{ if supportsLogsToLogs }}
 		{
