diff --git a/service/telemetry/internal/otelinit/config.go b/service/telemetry/internal/otelinit/config.go
index a89838860..402090675 100644
--- a/service/telemetry/internal/otelinit/config.go
+++ b/service/telemetry/internal/otelinit/config.go
@@ -12,6 +12,7 @@ import (
 	"net/http"
 	"net/url"
 	"os"
+	"strconv"
 	"strings"
 	"sync"
 	"time"
@@ -142,10 +143,10 @@ func cardinalityFilter(filter attribute.Set) attribute.Filter {
 func initPrometheusExporter(prometheusConfig *config.Prometheus, asyncErrorChannel chan error, serverWG *sync.WaitGroup) (sdkmetric.Reader, *http.Server, error) {
 	promRegistry := prometheus.NewRegistry()
 	if prometheusConfig.Host == nil {
-		return nil, nil, fmt.Errorf("host must be specified")
+		return nil, nil, errors.New("host must be specified")
 	}
 	if prometheusConfig.Port == nil {
-		return nil, nil, fmt.Errorf("port must be specified")
+		return nil, nil, errors.New("port must be specified")
 	}
 
 	opts := []otelprom.Option{
@@ -163,7 +164,7 @@ func initPrometheusExporter(prometheusConfig *config.Prometheus, asyncErrorChann
 		return nil, nil, fmt.Errorf("error creating otel prometheus exporter: %w", err)
 	}
 
-	return exporter, InitPrometheusServer(promRegistry, net.JoinHostPort(*prometheusConfig.Host, fmt.Sprintf("%d", *prometheusConfig.Port)), asyncErrorChannel, serverWG), nil
+	return exporter, InitPrometheusServer(promRegistry, net.JoinHostPort(*prometheusConfig.Host, strconv.Itoa(*prometheusConfig.Port)), asyncErrorChannel, serverWG), nil
 }
 
 func initPullExporter(exporter config.MetricExporter, asyncErrorChannel chan error, serverWG *sync.WaitGroup) (sdkmetric.Reader, *http.Server, error) {
@@ -207,7 +208,7 @@ func initPeriodicExporter(ctx context.Context, exporter config.MetricExporter, o
 
 func normalizeEndpoint(endpoint string) string {
 	if !strings.HasPrefix(endpoint, "https://") && !strings.HasPrefix(endpoint, "http://") {
-		return fmt.Sprintf("http://%s", endpoint)
+		return "http://" + endpoint
 	}
 	return endpoint
 }
