diff --git a/vendor/github.com/openzipkin/zipkin-go/model/doc.go b/vendor/github.com/openzipkin/zipkin-go/model/doc.go
deleted file mode 100644
index 99c2ec39a..000000000
--- a/vendor/github.com/openzipkin/zipkin-go/model/doc.go
+++ /dev/null
@@ -1,9 +0,0 @@
-/*
-Package model contains the Zipkin V2 model which is used by the Zipkin Go
-tracer implementation.
-
-Third party instrumentation libraries can use the model and transport packages
-found in this Zipkin Go library to directly interface with the Zipkin Server or
-Zipkin Collectors without the need to use the tracer implementation itself.
-*/
-package model
