diff --git a/pdata/testdata/go.mod b/pdata/testdata/go.mod
index d79c0fee4..724afed99 100644
--- a/pdata/testdata/go.mod
+++ b/pdata/testdata/go.mod
@@ -3,8 +3,8 @@ module go.opentelemetry.io/collector/pdata/testdata
 go 1.23.0
 
 require (
-	go.opentelemetry.io/collector/pdata v1.28.0
-	go.opentelemetry.io/collector/pdata/pprofile v0.122.0
+	go.opentelemetry.io/collector/pdata v1.28.1
+	go.opentelemetry.io/collector/pdata/pprofile v0.122.1
 )
 
 require (
