diff --git a/internal/compression/grpc/gzip.go b/internal/compression/grpc/gzip.go
index bf3e9eefb..5b490f341 100644
--- a/internal/compression/grpc/gzip.go
+++ b/internal/compression/grpc/gzip.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
