diff --git a/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows_test.go b/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows_test.go
index 250e34ad9..3e41e3147 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_scraper_windows_test.go
@@ -24,57 +24,33 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal/windows/pdh"
+	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal/perfcounters"
 )
 
 func TestScrapeMetrics_Error(t *testing.T) {
 	type testCase struct {
-		name                                   string
-		diskReadBytesPerSecCounterReturnValue  interface{}
-		diskWriteBytesPerSecCounterReturnValue interface{}
-		diskReadsPerSecCounterReturnValue      interface{}
-		diskWritesPerSecCounterReturnValue     interface{}
-		avgDiskSecsPerReadCounterReturnValue   interface{}
-		avgDiskSecsPerWriteCounterReturnValue  interface{}
-		diskQueueLengthCounterReturnValue      interface{}
-		expectedErr                            string
+		name         string
+		scrapeErr    error
+		getObjectErr error
+		getValuesErr error
+		expectedErr  string
 	}
 
 	testCases := []testCase{
 		{
-			name:                                  "readBytesPerSecCounterError",
-			diskReadBytesPerSecCounterReturnValue: errors.New("err1"),
-			expectedErr:                           "err1",
+			name:        "scrapeError",
+			scrapeErr:   errors.New("err1"),
+			expectedErr: "err1",
 		},
 		{
-			name:                                   "writeBytesPerSecCounterError",
-			diskWriteBytesPerSecCounterReturnValue: errors.New("err1"),
-			expectedErr:                            "err1",
+			name:         "getObjectErr",
+			getObjectErr: errors.New("err1"),
+			expectedErr:  "err1",
 		},
 		{
-			name:                              "readsPerSecCounterError",
-			diskReadsPerSecCounterReturnValue: errors.New("err1"),
-			expectedErr:                       "err1",
-		},
-		{
-			name:                               "writesPerSecCounterError",
-			diskWritesPerSecCounterReturnValue: errors.New("err1"),
-			expectedErr:                        "err1",
-		},
-		{
-			name:                                 "avgSecsPerReadCounterError",
-			avgDiskSecsPerReadCounterReturnValue: errors.New("err1"),
-			expectedErr:                          "err1",
-		},
-		{
-			name:                                  "avgSecsPerReadWriteError",
-			avgDiskSecsPerWriteCounterReturnValue: errors.New("err1"),
-			expectedErr:                           "err1",
-		},
-		{
-			name:                              "avgDiskQueueLengthError",
-			diskQueueLengthCounterReturnValue: errors.New("err1"),
-			expectedErr:                       "err1",
+			name:         "getValuesErr",
+			getValuesErr: errors.New("err1"),
+			expectedErr:  "err1",
 		},
 	}
 
@@ -83,18 +59,12 @@ func TestScrapeMetrics_Error(t *testing.T) {
 			scraper, err := newDiskScraper(context.Background(), &Config{})
 			require.NoError(t, err, "Failed to create disk scraper: %v", err)
 
+			scraper.perfCounterScraper = perfcounters.NewMockPerfCounterScraperError(test.scrapeErr, test.getObjectErr, test.getValuesErr)
+
 			err = scraper.Initialize(context.Background())
 			require.NoError(t, err, "Failed to initialize disk scraper: %v", err)
 			defer func() { assert.NoError(t, scraper.Close(context.Background())) }()
 
-			scraper.diskReadBytesPerSecCounter = pdh.NewMockPerfCounter(test.diskReadBytesPerSecCounterReturnValue)
-			scraper.diskWriteBytesPerSecCounter = pdh.NewMockPerfCounter(test.diskWriteBytesPerSecCounterReturnValue)
-			scraper.diskReadsPerSecCounter = pdh.NewMockPerfCounter(test.diskReadsPerSecCounterReturnValue)
-			scraper.diskWritesPerSecCounter = pdh.NewMockPerfCounter(test.diskWritesPerSecCounterReturnValue)
-			scraper.avgDiskSecsPerReadCounter = pdh.NewMockPerfCounter(test.avgDiskSecsPerReadCounterReturnValue)
-			scraper.avgDiskSecsPerWriteCounter = pdh.NewMockPerfCounter(test.avgDiskSecsPerWriteCounterReturnValue)
-			scraper.diskQueueLengthCounter = pdh.NewMockPerfCounter(test.diskQueueLengthCounterReturnValue)
-
 			_, err = scraper.ScrapeMetrics(context.Background())
 			assert.EqualError(t, err, test.expectedErr)
 		})
