diff --git a/exporter/exporterhelper/xexporterhelper/go.mod b/exporter/exporterhelper/xexporterhelper/go.mod
index 460d9cf21..575d3b5ee 100644
--- a/exporter/exporterhelper/xexporterhelper/go.mod
+++ b/exporter/exporterhelper/xexporterhelper/go.mod
@@ -102,3 +102,5 @@ replace go.opentelemetry.io/collector/extension/extensiontest => ../../../extens
 replace go.opentelemetry.io/collector/featuregate => ../../../featuregate
 
 replace go.opentelemetry.io/collector/extension/xextension => ../../../extension/xextension
+
+replace go.opentelemetry.io/collector/confmap => ../../../confmap
