diff --git a/consumer/pdata/generated_trace_test.go b/consumer/pdata/generated_trace_test.go
index af39e1838..75cae88de 100644
--- a/consumer/pdata/generated_trace_test.go
+++ b/consumer/pdata/generated_trace_test.go
@@ -137,6 +137,24 @@ func TestResourceSpansSlice_MoveAndAppendTo(t *testing.T) {
 	}
 }
 
+func TestResourceSpansSlice_RemoveIf(t *testing.T) {
+	// Test RemoveIf on empty slice
+	emptySlice := NewResourceSpansSlice()
+	emptySlice.RemoveIf(func(el ResourceSpans) bool {
+		t.Fail()
+		return false
+	})
+
+	// Test RemoveIf
+	filtered := generateTestResourceSpansSlice()
+	pos := 0
+	filtered.RemoveIf(func(el ResourceSpans) bool {
+		pos++
+		return pos%3 == 0
+	})
+	assert.Equal(t, 5, filtered.Len())
+}
+
 func TestResourceSpans_CopyTo(t *testing.T) {
 	ms := NewResourceSpans()
 	generateTestResourceSpans().CopyTo(ms)
@@ -269,6 +287,24 @@ func TestInstrumentationLibrarySpansSlice_MoveAndAppendTo(t *testing.T) {
 	}
 }
 
+func TestInstrumentationLibrarySpansSlice_RemoveIf(t *testing.T) {
+	// Test RemoveIf on empty slice
+	emptySlice := NewInstrumentationLibrarySpansSlice()
+	emptySlice.RemoveIf(func(el InstrumentationLibrarySpans) bool {
+		t.Fail()
+		return false
+	})
+
+	// Test RemoveIf
+	filtered := generateTestInstrumentationLibrarySpansSlice()
+	pos := 0
+	filtered.RemoveIf(func(el InstrumentationLibrarySpans) bool {
+		pos++
+		return pos%3 == 0
+	})
+	assert.Equal(t, 5, filtered.Len())
+}
+
 func TestInstrumentationLibrarySpans_CopyTo(t *testing.T) {
 	ms := NewInstrumentationLibrarySpans()
 	generateTestInstrumentationLibrarySpans().CopyTo(ms)
@@ -401,6 +437,24 @@ func TestSpanSlice_MoveAndAppendTo(t *testing.T) {
 	}
 }
 
+func TestSpanSlice_RemoveIf(t *testing.T) {
+	// Test RemoveIf on empty slice
+	emptySlice := NewSpanSlice()
+	emptySlice.RemoveIf(func(el Span) bool {
+		t.Fail()
+		return false
+	})
+
+	// Test RemoveIf
+	filtered := generateTestSpanSlice()
+	pos := 0
+	filtered.RemoveIf(func(el Span) bool {
+		pos++
+		return pos%3 == 0
+	})
+	assert.Equal(t, 5, filtered.Len())
+}
+
 func TestSpan_CopyTo(t *testing.T) {
 	ms := NewSpan()
 	generateTestSpan().CopyTo(ms)
@@ -637,6 +691,24 @@ func TestSpanEventSlice_MoveAndAppendTo(t *testing.T) {
 	}
 }
 
+func TestSpanEventSlice_RemoveIf(t *testing.T) {
+	// Test RemoveIf on empty slice
+	emptySlice := NewSpanEventSlice()
+	emptySlice.RemoveIf(func(el SpanEvent) bool {
+		t.Fail()
+		return false
+	})
+
+	// Test RemoveIf
+	filtered := generateTestSpanEventSlice()
+	pos := 0
+	filtered.RemoveIf(func(el SpanEvent) bool {
+		pos++
+		return pos%3 == 0
+	})
+	assert.Equal(t, 5, filtered.Len())
+}
+
 func TestSpanEvent_CopyTo(t *testing.T) {
 	ms := NewSpanEvent()
 	generateTestSpanEvent().CopyTo(ms)
@@ -787,6 +859,24 @@ func TestSpanLinkSlice_MoveAndAppendTo(t *testing.T) {
 	}
 }
 
+func TestSpanLinkSlice_RemoveIf(t *testing.T) {
+	// Test RemoveIf on empty slice
+	emptySlice := NewSpanLinkSlice()
+	emptySlice.RemoveIf(func(el SpanLink) bool {
+		t.Fail()
+		return false
+	})
+
+	// Test RemoveIf
+	filtered := generateTestSpanLinkSlice()
+	pos := 0
+	filtered.RemoveIf(func(el SpanLink) bool {
+		pos++
+		return pos%3 == 0
+	})
+	assert.Equal(t, 5, filtered.Len())
+}
+
 func TestSpanLink_CopyTo(t *testing.T) {
 	ms := NewSpanLink()
 	generateTestSpanLink().CopyTo(ms)
