diff --git a/exporter/exporterhelper/internal/persistent_queue.go b/exporter/exporterhelper/internal/persistent_queue.go
index 97d5a1007..17bc2fe64 100644
--- a/exporter/exporterhelper/internal/persistent_queue.go
+++ b/exporter/exporterhelper/internal/persistent_queue.go
@@ -215,20 +215,27 @@ func (pq *persistentQueue[T]) putInternal(ctx context.Context, req T) error {
 	}
 
 	itemKey := getItemKey(pq.writeIndex)
-	pq.writeIndex++
+	newIndex := pq.writeIndex + 1
 
 	reqBuf, err := pq.marshaler(req)
 	if err != nil {
 		return err
 	}
-	err = pq.client.Batch(ctx,
-		storage.SetOperation(writeIndexKey, itemIndexToBytes(pq.writeIndex)),
-		storage.SetOperation(itemKey, reqBuf))
 
+	// Carry out a transaction where we both add the item and update the write index
+	ops := []storage.Operation{
+		storage.SetOperation(writeIndexKey, itemIndexToBytes(newIndex)),
+		storage.SetOperation(itemKey, reqBuf),
+	}
+	if storageErr := pq.client.Batch(ctx, ops...); storageErr != nil {
+		return storageErr
+	}
+
+	pq.writeIndex = newIndex
 	// Inform the loop that there's some data to process
 	pq.putChan <- struct{}{}
 
-	return err
+	return nil
 }
 
 // getNextItem pulls the next available item from the persistent storage along with a callback function that should be
