diff --git a/config/configopaque/opaque.go b/config/configopaque/opaque.go
index c79d053f6..b1712a212 100644
--- a/config/configopaque/opaque.go
+++ b/config/configopaque/opaque.go
@@ -4,7 +4,6 @@
 package configopaque // import "go.opentelemetry.io/collector/config/configopaque"
 
 import (
-	"encoding"
 	"fmt"
 )
 
@@ -14,31 +13,23 @@ type String string
 
 const maskedString = "[REDACTED]"
 
-var _ encoding.TextMarshaler = String("")
-
 // MarshalText marshals the string as `[REDACTED]`.
 func (s String) MarshalText() ([]byte, error) {
 	return []byte(maskedString), nil
 }
 
-var _ fmt.Stringer = String("")
-
 // String formats the string as `[REDACTED]`.
 // This is used for the %s and %q verbs.
 func (s String) String() string {
 	return maskedString
 }
 
-var _ fmt.GoStringer = String("")
-
 // GoString formats the string as `[REDACTED]`.
 // This is used for the %#v verb.
 func (s String) GoString() string {
 	return fmt.Sprintf("%#v", maskedString)
 }
 
-var _ encoding.BinaryMarshaler = String("")
-
 // MarshalBinary marshals the string `[REDACTED]` as []byte.
 func (s String) MarshalBinary() (text []byte, err error) {
 	return []byte(maskedString), nil
