diff --git a/testutil/metricstestutil/metricsutil.go b/testutil/metricstestutil/metricsutil.go
index 5ffed29ec..1d528ebdd 100644
--- a/testutil/metricstestutil/metricsutil.go
+++ b/testutil/metricstestutil/metricsutil.go
@@ -176,9 +176,9 @@ func SortedMetrics(metrics pdata.Metrics) pdata.Metrics {
 					for l := 0; l < m.DoubleGauge().DataPoints().Len(); l++ {
 						m.DoubleGauge().DataPoints().At(l).LabelsMap().Sort()
 					}
-				case pdata.MetricDataTypeDoubleSum:
-					for l := 0; l < m.DoubleSum().DataPoints().Len(); l++ {
-						m.DoubleSum().DataPoints().At(l).LabelsMap().Sort()
+				case pdata.MetricDataTypeSum:
+					for l := 0; l < m.Sum().DataPoints().Len(); l++ {
+						m.Sum().DataPoints().At(l).LabelsMap().Sort()
 					}
 				case pdata.MetricDataTypeIntHistogram:
 					for l := 0; l < m.IntHistogram().DataPoints().Len(); l++ {
