diff --git a/pdata/pprofile/location_test.go b/pdata/pprofile/location_test.go
index f9c401470..311e5b733 100644
--- a/pdata/pprofile/location_test.go
+++ b/pdata/pprofile/location_test.go
@@ -108,26 +108,26 @@ func TestLocationSwitchDictionary(t *testing.T) {
 			}(),
 			dst: func() ProfilesDictionary {
 				d := NewProfilesDictionary()
-				d.StringTable().Append("", "foo")
+				d.StringTable().Append("", "test")
 
 				d.MappingTable().AppendEmpty()
-				d.MappingTable().AppendEmpty()
+				m := d.MappingTable().AppendEmpty()
+				m.SetFilenameStrindex(1)
 				return d
 			}(),
 
 			wantLocation: func() Location {
 				l := NewLocation()
-				l.SetMappingIndex(2)
+				l.SetMappingIndex(1)
 				return l
 			}(),
 			wantDictionary: func() ProfilesDictionary {
 				d := NewProfilesDictionary()
-				d.StringTable().Append("", "foo", "test")
+				d.StringTable().Append("", "test")
 
-				d.MappingTable().AppendEmpty()
 				d.MappingTable().AppendEmpty()
 				m := d.MappingTable().AppendEmpty()
-				m.SetFilenameStrindex(2)
+				m.SetFilenameStrindex(1)
 				return d
 			}(),
 		},
@@ -194,26 +194,27 @@ func TestLocationSwitchDictionary(t *testing.T) {
 			}(),
 			dst: func() ProfilesDictionary {
 				d := NewProfilesDictionary()
-				d.StringTable().Append("", "foo")
+				d.StringTable().Append("", "test")
 
 				d.AttributeTable().AppendEmpty()
-				d.AttributeTable().AppendEmpty()
+				a := d.AttributeTable().AppendEmpty()
+				a.SetKeyStrindex(1)
+
 				return d
 			}(),
 
 			wantLocation: func() Location {
 				l := NewLocation()
-				l.AttributeIndices().Append(2)
+				l.AttributeIndices().Append(1)
 				return l
 			}(),
 			wantDictionary: func() ProfilesDictionary {
 				d := NewProfilesDictionary()
-				d.StringTable().Append("", "foo", "test")
+				d.StringTable().Append("", "test")
 
-				d.AttributeTable().AppendEmpty()
 				d.AttributeTable().AppendEmpty()
 				a := d.AttributeTable().AppendEmpty()
-				a.SetKeyStrindex(2)
+				a.SetKeyStrindex(1)
 				return d
 			}(),
 		},
@@ -280,26 +281,27 @@ func TestLocationSwitchDictionary(t *testing.T) {
 			}(),
 			dst: func() ProfilesDictionary {
 				d := NewProfilesDictionary()
-				d.StringTable().Append("", "foo")
+				d.StringTable().Append("", "test")
 
 				d.FunctionTable().AppendEmpty()
-				d.FunctionTable().AppendEmpty()
+				f := d.FunctionTable().AppendEmpty()
+				f.SetNameStrindex(1)
+
 				return d
 			}(),
 
 			wantLocation: func() Location {
 				l := NewLocation()
-				l.Lines().AppendEmpty().SetFunctionIndex(2)
+				l.Lines().AppendEmpty().SetFunctionIndex(1)
 				return l
 			}(),
 			wantDictionary: func() ProfilesDictionary {
 				d := NewProfilesDictionary()
-				d.StringTable().Append("", "foo", "test")
+				d.StringTable().Append("", "test")
 
-				d.FunctionTable().AppendEmpty()
 				d.FunctionTable().AppendEmpty()
 				f := d.FunctionTable().AppendEmpty()
-				f.SetNameStrindex(2)
+				f.SetNameStrindex(1)
 				return d
 			}(),
 		},
