diff --git a/pdata/internal/generated_wrapper_lineslice.go b/pdata/internal/generated_wrapper_lineslice.go
index e8aa796ca..e43c1abab 100644
--- a/pdata/internal/generated_wrapper_lineslice.go
+++ b/pdata/internal/generated_wrapper_lineslice.go
@@ -25,6 +25,7 @@ func CopyOrigLineSlice(dest, src []*otlpprofiles.Line) []*otlpprofiles.Line {
 		// Cleanup the rest of the elements so GC can free the memory.
 		// This can happen when len(src) < len(dest) < cap(dest).
 		for i := len(src); i < len(dest); i++ {
+			DeleteOrigLine(dest[i], true)
 			dest[i] = nil
 		}
 		// Add new pointers for missing elements.
