diff --git a/pdata/internal/generated_wrapper_resourceprofiles_test.go b/pdata/internal/generated_wrapper_resourceprofiles_test.go
index 39e56111a..df88884c4 100644
--- a/pdata/internal/generated_wrapper_resourceprofiles_test.go
+++ b/pdata/internal/generated_wrapper_resourceprofiles_test.go
@@ -14,6 +14,7 @@ import (
 	gootlpprofiles "go.opentelemetry.io/proto/slim/otlp/profiles/v1development"
 	"google.golang.org/protobuf/proto"
 
+	"go.opentelemetry.io/collector/featuregate"
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 )
@@ -39,20 +40,29 @@ func TestMarshalAndUnmarshalJSONOrigResourceProfilesUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalJSONOrigResourceProfiles(t *testing.T) {
 	for name, src := range genTestEncodingValuesResourceProfiles() {
-		t.Run(name, func(t *testing.T) {
-			stream := json.BorrowStream(nil)
-			defer json.ReturnStream(stream)
-			MarshalJSONOrigResourceProfiles(src, stream)
-			require.NoError(t, stream.Error())
-
-			iter := json.BorrowIterator(stream.Buffer())
-			defer json.ReturnIterator(iter)
-			dest := NewOrigResourceProfiles()
-			UnmarshalJSONOrigResourceProfiles(dest, iter)
-			require.NoError(t, iter.Error())
-
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				stream := json.BorrowStream(nil)
+				defer json.ReturnStream(stream)
+				MarshalJSONOrigResourceProfiles(src, stream)
+				require.NoError(t, stream.Error())
+
+				iter := json.BorrowIterator(stream.Buffer())
+				defer json.ReturnIterator(iter)
+				dest := NewOrigResourceProfiles()
+				UnmarshalJSONOrigResourceProfiles(dest, iter)
+				require.NoError(t, iter.Error())
+
+				assert.Equal(t, src, dest)
+				DeleteOrigResourceProfiles(dest, true)
+			})
+		}
 	}
 }
 
@@ -74,15 +84,25 @@ func TestMarshalAndUnmarshalProtoOrigResourceProfilesUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalProtoOrigResourceProfiles(t *testing.T) {
 	for name, src := range genTestEncodingValuesResourceProfiles() {
-		t.Run(name, func(t *testing.T) {
-			buf := make([]byte, SizeProtoOrigResourceProfiles(src))
-			gotSize := MarshalProtoOrigResourceProfiles(src, buf)
-			assert.Equal(t, len(buf), gotSize)
-
-			dest := NewOrigResourceProfiles()
-			require.NoError(t, UnmarshalProtoOrigResourceProfiles(dest, buf))
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				buf := make([]byte, SizeProtoOrigResourceProfiles(src))
+				gotSize := MarshalProtoOrigResourceProfiles(src, buf)
+				assert.Equal(t, len(buf), gotSize)
+
+				dest := NewOrigResourceProfiles()
+				require.NoError(t, UnmarshalProtoOrigResourceProfiles(dest, buf))
+
+				assert.Equal(t, src, dest)
+				DeleteOrigResourceProfiles(dest, true)
+			})
+		}
 	}
 }
 
