diff --git a/pdata/internal/generated_wrapper_summarydatapoint.go b/pdata/internal/generated_wrapper_summarydatapoint.go
index 9ab8835bf..6dc38eef2 100644
--- a/pdata/internal/generated_wrapper_summarydatapoint.go
+++ b/pdata/internal/generated_wrapper_summarydatapoint.go
@@ -10,6 +10,7 @@ import (
 	"encoding/binary"
 	"fmt"
 	"math"
+	"sync"
 
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
@@ -17,8 +18,40 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
+var protoPoolSummaryDataPoint = sync.Pool{
+	New: func() any {
+		return &otlpmetrics.SummaryDataPoint{}
+	},
+}
+
 func NewOrigSummaryDataPoint() *otlpmetrics.SummaryDataPoint {
-	return &otlpmetrics.SummaryDataPoint{}
+	if !UseProtoPooling.IsEnabled() {
+		return &otlpmetrics.SummaryDataPoint{}
+	}
+	return protoPoolSummaryDataPoint.Get().(*otlpmetrics.SummaryDataPoint)
+}
+
+func DeleteOrigSummaryDataPoint(orig *otlpmetrics.SummaryDataPoint, nullable bool) {
+	if orig == nil {
+		return
+	}
+
+	if !UseProtoPooling.IsEnabled() {
+		orig.Reset()
+		return
+	}
+
+	for i := range orig.Attributes {
+		DeleteOrigKeyValue(&orig.Attributes[i], false)
+	}
+	for i := range orig.QuantileValues {
+		DeleteOrigSummaryDataPoint_ValueAtQuantile(orig.QuantileValues[i], true)
+	}
+
+	orig.Reset()
+	if nullable {
+		protoPoolSummaryDataPoint.Put(orig)
+	}
 }
 
 func CopyOrigSummaryDataPoint(dest, src *otlpmetrics.SummaryDataPoint) {
