diff --git a/pdata/internal/generated_wrapper_span_link_test.go b/pdata/internal/generated_wrapper_span_link_test.go
index 0e05b8431..1d01f6558 100644
--- a/pdata/internal/generated_wrapper_span_link_test.go
+++ b/pdata/internal/generated_wrapper_span_link_test.go
@@ -14,6 +14,7 @@ import (
 	gootlptrace "go.opentelemetry.io/proto/slim/otlp/trace/v1"
 	"google.golang.org/protobuf/proto"
 
+	"go.opentelemetry.io/collector/featuregate"
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
@@ -40,20 +41,29 @@ func TestMarshalAndUnmarshalJSONOrigSpan_LinkUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalJSONOrigSpan_Link(t *testing.T) {
 	for name, src := range genTestEncodingValuesSpan_Link() {
-		t.Run(name, func(t *testing.T) {
-			stream := json.BorrowStream(nil)
-			defer json.ReturnStream(stream)
-			MarshalJSONOrigSpan_Link(src, stream)
-			require.NoError(t, stream.Error())
-
-			iter := json.BorrowIterator(stream.Buffer())
-			defer json.ReturnIterator(iter)
-			dest := NewOrigSpan_Link()
-			UnmarshalJSONOrigSpan_Link(dest, iter)
-			require.NoError(t, iter.Error())
-
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				stream := json.BorrowStream(nil)
+				defer json.ReturnStream(stream)
+				MarshalJSONOrigSpan_Link(src, stream)
+				require.NoError(t, stream.Error())
+
+				iter := json.BorrowIterator(stream.Buffer())
+				defer json.ReturnIterator(iter)
+				dest := NewOrigSpan_Link()
+				UnmarshalJSONOrigSpan_Link(dest, iter)
+				require.NoError(t, iter.Error())
+
+				assert.Equal(t, src, dest)
+				DeleteOrigSpan_Link(dest, true)
+			})
+		}
 	}
 }
 
@@ -75,15 +85,25 @@ func TestMarshalAndUnmarshalProtoOrigSpan_LinkUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalProtoOrigSpan_Link(t *testing.T) {
 	for name, src := range genTestEncodingValuesSpan_Link() {
-		t.Run(name, func(t *testing.T) {
-			buf := make([]byte, SizeProtoOrigSpan_Link(src))
-			gotSize := MarshalProtoOrigSpan_Link(src, buf)
-			assert.Equal(t, len(buf), gotSize)
-
-			dest := NewOrigSpan_Link()
-			require.NoError(t, UnmarshalProtoOrigSpan_Link(dest, buf))
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				buf := make([]byte, SizeProtoOrigSpan_Link(src))
+				gotSize := MarshalProtoOrigSpan_Link(src, buf)
+				assert.Equal(t, len(buf), gotSize)
+
+				dest := NewOrigSpan_Link()
+				require.NoError(t, UnmarshalProtoOrigSpan_Link(dest, buf))
+
+				assert.Equal(t, src, dest)
+				DeleteOrigSpan_Link(dest, true)
+			})
+		}
 	}
 }
 
