diff --git a/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go b/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go
index 116d9073b..ceaaa3c14 100644
--- a/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go
+++ b/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go
@@ -14,6 +14,7 @@ import (
 	gootlpcollectorlogs "go.opentelemetry.io/proto/slim/otlp/collector/logs/v1"
 	"google.golang.org/protobuf/proto"
 
+	"go.opentelemetry.io/collector/featuregate"
 	otlpcollectorlogs "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/logs/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 )
@@ -39,20 +40,29 @@ func TestMarshalAndUnmarshalJSONOrigExportLogsServiceResponseUnknown(t *testing.
 
 func TestMarshalAndUnmarshalJSONOrigExportLogsServiceResponse(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportLogsServiceResponse() {
-		t.Run(name, func(t *testing.T) {
-			stream := json.BorrowStream(nil)
-			defer json.ReturnStream(stream)
-			MarshalJSONOrigExportLogsServiceResponse(src, stream)
-			require.NoError(t, stream.Error())
-
-			iter := json.BorrowIterator(stream.Buffer())
-			defer json.ReturnIterator(iter)
-			dest := NewOrigExportLogsServiceResponse()
-			UnmarshalJSONOrigExportLogsServiceResponse(dest, iter)
-			require.NoError(t, iter.Error())
-
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				stream := json.BorrowStream(nil)
+				defer json.ReturnStream(stream)
+				MarshalJSONOrigExportLogsServiceResponse(src, stream)
+				require.NoError(t, stream.Error())
+
+				iter := json.BorrowIterator(stream.Buffer())
+				defer json.ReturnIterator(iter)
+				dest := NewOrigExportLogsServiceResponse()
+				UnmarshalJSONOrigExportLogsServiceResponse(dest, iter)
+				require.NoError(t, iter.Error())
+
+				assert.Equal(t, src, dest)
+				DeleteOrigExportLogsServiceResponse(dest, true)
+			})
+		}
 	}
 }
 
@@ -74,15 +84,25 @@ func TestMarshalAndUnmarshalProtoOrigExportLogsServiceResponseUnknown(t *testing
 
 func TestMarshalAndUnmarshalProtoOrigExportLogsServiceResponse(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportLogsServiceResponse() {
-		t.Run(name, func(t *testing.T) {
-			buf := make([]byte, SizeProtoOrigExportLogsServiceResponse(src))
-			gotSize := MarshalProtoOrigExportLogsServiceResponse(src, buf)
-			assert.Equal(t, len(buf), gotSize)
-
-			dest := NewOrigExportLogsServiceResponse()
-			require.NoError(t, UnmarshalProtoOrigExportLogsServiceResponse(dest, buf))
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				buf := make([]byte, SizeProtoOrigExportLogsServiceResponse(src))
+				gotSize := MarshalProtoOrigExportLogsServiceResponse(src, buf)
+				assert.Equal(t, len(buf), gotSize)
+
+				dest := NewOrigExportLogsServiceResponse()
+				require.NoError(t, UnmarshalProtoOrigExportLogsServiceResponse(dest, buf))
+
+				assert.Equal(t, src, dest)
+				DeleteOrigExportLogsServiceResponse(dest, true)
+			})
+		}
 	}
 }
 
