diff --git a/pdata/internal/generated_wrapper_keyvalue_test.go b/pdata/internal/generated_wrapper_keyvalue_test.go
index ce7798bba..05eed2970 100644
--- a/pdata/internal/generated_wrapper_keyvalue_test.go
+++ b/pdata/internal/generated_wrapper_keyvalue_test.go
@@ -14,6 +14,7 @@ import (
 	gootlpcommon "go.opentelemetry.io/proto/slim/otlp/common/v1"
 	"google.golang.org/protobuf/proto"
 
+	"go.opentelemetry.io/collector/featuregate"
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 )
@@ -39,20 +40,29 @@ func TestMarshalAndUnmarshalJSONOrigKeyValueUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalJSONOrigKeyValue(t *testing.T) {
 	for name, src := range genTestEncodingValuesKeyValue() {
-		t.Run(name, func(t *testing.T) {
-			stream := json.BorrowStream(nil)
-			defer json.ReturnStream(stream)
-			MarshalJSONOrigKeyValue(src, stream)
-			require.NoError(t, stream.Error())
-
-			iter := json.BorrowIterator(stream.Buffer())
-			defer json.ReturnIterator(iter)
-			dest := NewOrigKeyValue()
-			UnmarshalJSONOrigKeyValue(dest, iter)
-			require.NoError(t, iter.Error())
-
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				stream := json.BorrowStream(nil)
+				defer json.ReturnStream(stream)
+				MarshalJSONOrigKeyValue(src, stream)
+				require.NoError(t, stream.Error())
+
+				iter := json.BorrowIterator(stream.Buffer())
+				defer json.ReturnIterator(iter)
+				dest := NewOrigKeyValue()
+				UnmarshalJSONOrigKeyValue(dest, iter)
+				require.NoError(t, iter.Error())
+
+				assert.Equal(t, src, dest)
+				DeleteOrigKeyValue(dest, true)
+			})
+		}
 	}
 }
 
@@ -74,15 +84,25 @@ func TestMarshalAndUnmarshalProtoOrigKeyValueUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalProtoOrigKeyValue(t *testing.T) {
 	for name, src := range genTestEncodingValuesKeyValue() {
-		t.Run(name, func(t *testing.T) {
-			buf := make([]byte, SizeProtoOrigKeyValue(src))
-			gotSize := MarshalProtoOrigKeyValue(src, buf)
-			assert.Equal(t, len(buf), gotSize)
-
-			dest := NewOrigKeyValue()
-			require.NoError(t, UnmarshalProtoOrigKeyValue(dest, buf))
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				buf := make([]byte, SizeProtoOrigKeyValue(src))
+				gotSize := MarshalProtoOrigKeyValue(src, buf)
+				assert.Equal(t, len(buf), gotSize)
+
+				dest := NewOrigKeyValue()
+				require.NoError(t, UnmarshalProtoOrigKeyValue(dest, buf))
+
+				assert.Equal(t, src, dest)
+				DeleteOrigKeyValue(dest, true)
+			})
+		}
 	}
 }
 
