diff --git a/pdata/internal/generated_wrapper_line_test.go b/pdata/internal/generated_wrapper_line_test.go
index 8391bc6b5..de44dd90d 100644
--- a/pdata/internal/generated_wrapper_line_test.go
+++ b/pdata/internal/generated_wrapper_line_test.go
@@ -14,6 +14,7 @@ import (
 	gootlpprofiles "go.opentelemetry.io/proto/slim/otlp/profiles/v1development"
 	"google.golang.org/protobuf/proto"
 
+	"go.opentelemetry.io/collector/featuregate"
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 )
@@ -39,20 +40,29 @@ func TestMarshalAndUnmarshalJSONOrigLineUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalJSONOrigLine(t *testing.T) {
 	for name, src := range genTestEncodingValuesLine() {
-		t.Run(name, func(t *testing.T) {
-			stream := json.BorrowStream(nil)
-			defer json.ReturnStream(stream)
-			MarshalJSONOrigLine(src, stream)
-			require.NoError(t, stream.Error())
-
-			iter := json.BorrowIterator(stream.Buffer())
-			defer json.ReturnIterator(iter)
-			dest := NewOrigLine()
-			UnmarshalJSONOrigLine(dest, iter)
-			require.NoError(t, iter.Error())
-
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				stream := json.BorrowStream(nil)
+				defer json.ReturnStream(stream)
+				MarshalJSONOrigLine(src, stream)
+				require.NoError(t, stream.Error())
+
+				iter := json.BorrowIterator(stream.Buffer())
+				defer json.ReturnIterator(iter)
+				dest := NewOrigLine()
+				UnmarshalJSONOrigLine(dest, iter)
+				require.NoError(t, iter.Error())
+
+				assert.Equal(t, src, dest)
+				DeleteOrigLine(dest, true)
+			})
+		}
 	}
 }
 
@@ -74,15 +84,25 @@ func TestMarshalAndUnmarshalProtoOrigLineUnknown(t *testing.T) {
 
 func TestMarshalAndUnmarshalProtoOrigLine(t *testing.T) {
 	for name, src := range genTestEncodingValuesLine() {
-		t.Run(name, func(t *testing.T) {
-			buf := make([]byte, SizeProtoOrigLine(src))
-			gotSize := MarshalProtoOrigLine(src, buf)
-			assert.Equal(t, len(buf), gotSize)
-
-			dest := NewOrigLine()
-			require.NoError(t, UnmarshalProtoOrigLine(dest, buf))
-			assert.Equal(t, src, dest)
-		})
+		for _, pooling := range []bool{true, false} {
+			t.Run(name, func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				buf := make([]byte, SizeProtoOrigLine(src))
+				gotSize := MarshalProtoOrigLine(src, buf)
+				assert.Equal(t, len(buf), gotSize)
+
+				dest := NewOrigLine()
+				require.NoError(t, UnmarshalProtoOrigLine(dest, buf))
+
+				assert.Equal(t, src, dest)
+				DeleteOrigLine(dest, true)
+			})
+		}
 	}
 }
 
