diff --git a/pdata/internal/generated_wrapper_locationslice.go b/pdata/internal/generated_wrapper_locationslice.go
index cdb23edfe..83a10cc45 100644
--- a/pdata/internal/generated_wrapper_locationslice.go
+++ b/pdata/internal/generated_wrapper_locationslice.go
@@ -25,6 +25,7 @@ func CopyOrigLocationSlice(dest, src []*otlpprofiles.Location) []*otlpprofiles.L
 		// Cleanup the rest of the elements so GC can free the memory.
 		// This can happen when len(src) < len(dest) < cap(dest).
 		for i := len(src); i < len(dest); i++ {
+			DeleteOrigLocation(dest[i], true)
 			dest[i] = nil
 		}
 		// Add new pointers for missing elements.
