diff --git a/pdata/internal/generated_wrapper_histogram.go b/pdata/internal/generated_wrapper_histogram.go
index 7ac8b7a03..e5e33b884 100644
--- a/pdata/internal/generated_wrapper_histogram.go
+++ b/pdata/internal/generated_wrapper_histogram.go
@@ -8,14 +8,44 @@ package internal
 
 import (
 	"fmt"
+	"sync"
 
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
+var protoPoolHistogram = sync.Pool{
+	New: func() any {
+		return &otlpmetrics.Histogram{}
+	},
+}
+
 func NewOrigHistogram() *otlpmetrics.Histogram {
-	return &otlpmetrics.Histogram{}
+	if !UseProtoPooling.IsEnabled() {
+		return &otlpmetrics.Histogram{}
+	}
+	return protoPoolHistogram.Get().(*otlpmetrics.Histogram)
+}
+
+func DeleteOrigHistogram(orig *otlpmetrics.Histogram, nullable bool) {
+	if orig == nil {
+		return
+	}
+
+	if !UseProtoPooling.IsEnabled() {
+		orig.Reset()
+		return
+	}
+
+	for i := range orig.DataPoints {
+		DeleteOrigHistogramDataPoint(orig.DataPoints[i], true)
+	}
+
+	orig.Reset()
+	if nullable {
+		protoPoolHistogram.Put(orig)
+	}
 }
 
 func CopyOrigHistogram(dest, src *otlpmetrics.Histogram) {
