diff --git a/pdata/internal/generated_wrapper_entityref.go b/pdata/internal/generated_wrapper_entityref.go
index 033dd4cc1..30e63520b 100644
--- a/pdata/internal/generated_wrapper_entityref.go
+++ b/pdata/internal/generated_wrapper_entityref.go
@@ -8,6 +8,7 @@ package internal
 
 import (
 	"fmt"
+	"sync"
 
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
@@ -31,8 +32,33 @@ func NewEntityRef(orig *otlpcommon.EntityRef, state *State) EntityRef {
 	return EntityRef{orig: orig, state: state}
 }
 
+var protoPoolEntityRef = sync.Pool{
+	New: func() any {
+		return &otlpcommon.EntityRef{}
+	},
+}
+
 func NewOrigEntityRef() *otlpcommon.EntityRef {
-	return &otlpcommon.EntityRef{}
+	if !UseProtoPooling.IsEnabled() {
+		return &otlpcommon.EntityRef{}
+	}
+	return protoPoolEntityRef.Get().(*otlpcommon.EntityRef)
+}
+
+func DeleteOrigEntityRef(orig *otlpcommon.EntityRef, nullable bool) {
+	if orig == nil {
+		return
+	}
+
+	if !UseProtoPooling.IsEnabled() {
+		orig.Reset()
+		return
+	}
+
+	orig.Reset()
+	if nullable {
+		protoPoolEntityRef.Put(orig)
+	}
 }
 
 func CopyOrigEntityRef(dest, src *otlpcommon.EntityRef) {
