diff --git a/component/componenttest/go.mod b/component/componenttest/go.mod
index 1df44ffe3..ca2040cd6 100644
--- a/component/componenttest/go.mod
+++ b/component/componenttest/go.mod
@@ -26,7 +26,7 @@ require (
 	go.opentelemetry.io/auto/sdk v1.1.0 // indirect
 	go.opentelemetry.io/collector/featuregate v1.33.0 // indirect
 	go.opentelemetry.io/collector/internal/telemetry v0.127.0 // indirect
-	go.opentelemetry.io/contrib/bridges/otelzap v0.10.0 // indirect
+	go.opentelemetry.io/contrib/bridges/otelzap v0.11.0 // indirect
 	go.opentelemetry.io/otel v1.36.0 // indirect
 	go.opentelemetry.io/otel/log v0.12.2 // indirect
 	golang.org/x/net v0.39.0 // indirect
