diff --git a/exporter/exporterhelper/traces.go b/exporter/exporterhelper/traces.go
index 778c0d639..7c85ac6c5 100644
--- a/exporter/exporterhelper/traces.go
+++ b/exporter/exporterhelper/traces.go
@@ -87,7 +87,7 @@ func NewTracesExporter(
 }
 
 // RequestFromTracesFunc converts ptrace.Traces into a user-defined Request.
-// This API is at the early stage of development and may change without backward compatibility
+// Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
 type RequestFromTracesFunc func(context.Context, ptrace.Traces) (Request, error)
 
@@ -99,7 +99,7 @@ func requestFromTraces(pusher consumer.ConsumeTracesFunc) RequestFromTracesFunc
 }
 
 // NewTracesRequestExporter creates a new traces exporter based on a custom TracesConverter and RequestSender.
-// This API is at the early stage of development and may change without backward compatibility
+// Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
 func NewTracesRequestExporter(
 	_ context.Context,
