diff --git a/service/collector_windows.go b/service/collector_windows.go
index 3ab2bb6dc..7c7dc2308 100644
--- a/service/collector_windows.go
+++ b/service/collector_windows.go
@@ -30,6 +30,7 @@ import (
 	"golang.org/x/sys/windows/svc/eventlog"
 
 	"go.opentelemetry.io/collector/config/configmapprovider"
+	"go.opentelemetry.io/collector/config/configunmarshaler"
 )
 
 type WindowsService struct {
@@ -135,8 +136,8 @@ func openEventLog(serviceName string) (*eventlog.Log, error) {
 }
 
 func newWithWindowsEventLogCore(set CollectorSettings, elog *eventlog.Log) (*Collector, error) {
-	if set.ConfigMapProvider == nil {
-		set.ConfigMapProvider = configmapprovider.NewDefault(getConfigFlag(), getSetFlag())
+	if set.ConfigProvider == nil {
+		set.ConfigProvider = NewConfigProvider(configmapprovider.NewDefault(getConfigFlag(), getSetFlag()), configunmarshaler.NewDefault())
 	}
 	set.LoggingOptions = append(
 		set.LoggingOptions,
