diff --git a/service/settings.go b/service/settings.go
index 59a16de4f..9a8f6ff8b 100644
--- a/service/settings.go
+++ b/service/settings.go
@@ -20,8 +20,6 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
-	"go.opentelemetry.io/collector/config/configmapprovider"
-	"go.opentelemetry.io/collector/config/configunmarshaler"
 )
 
 // svcSettings holds configuration for building a new service.
@@ -59,16 +57,9 @@ type CollectorSettings struct {
 	// and manually handle the signals to shutdown the collector.
 	DisableGracefulShutdown bool
 
-	// ConfigMapProvider provides the configuration's config.Map.
-	// If it is not provided a default provider is used. The default provider loads the configuration
-	// from a config file define by the --config command line flag and overrides component's configuration
-	// properties supplied via --set command line flag.
-	// If the provider is configmapprovider.WatchableRetrieved, collector may reload the configuration upon error.
-	ConfigMapProvider configmapprovider.Provider
-
-	// ConfigUnmarshaler unmarshalls the configuration's Parser into the service configuration.
-	// If it is not provided a default unmarshaler is used.
-	ConfigUnmarshaler configunmarshaler.ConfigUnmarshaler
+	// ConfigProvider provides the service configuration.
+	// If the provider watches for configuration change, collector may reload the new configuration upon changes.
+	ConfigProvider ConfigProvider
 
 	// LoggingOptions provides a way to change behavior of zap logging.
 	LoggingOptions []zap.Option
