diff --git a/pdata/internal/generated_wrapper_gauge.go b/pdata/internal/generated_wrapper_gauge.go
index dccf195c2..182ff55ec 100644
--- a/pdata/internal/generated_wrapper_gauge.go
+++ b/pdata/internal/generated_wrapper_gauge.go
@@ -49,6 +49,10 @@ func DeleteOrigGauge(orig *otlpmetrics.Gauge, nullable bool) {
 }
 
 func CopyOrigGauge(dest, src *otlpmetrics.Gauge) {
+	// If copying to same object, just return.
+	if src == dest {
+		return
+	}
 	dest.DataPoints = CopyOrigNumberDataPointSlice(dest.DataPoints, src.DataPoints)
 }
 
