diff --git a/pdata/internal/generated_wrapper_span_link.go b/pdata/internal/generated_wrapper_span_link.go
index 71dd1d012..32201980a 100644
--- a/pdata/internal/generated_wrapper_span_link.go
+++ b/pdata/internal/generated_wrapper_span_link.go
@@ -54,6 +54,10 @@ func DeleteOrigSpan_Link(orig *otlptrace.Span_Link, nullable bool) {
 }
 
 func CopyOrigSpan_Link(dest, src *otlptrace.Span_Link) {
+	// If copying to same object, just return.
+	if src == dest {
+		return
+	}
 	dest.TraceId = src.TraceId
 	dest.SpanId = src.SpanId
 	CopyOrigTraceState(&dest.TraceState, &src.TraceState)
