diff --git a/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go b/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go
index ceaaa3c14..f317266ed 100644
--- a/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go
+++ b/pdata/internal/generated_wrapper_exportlogsserviceresponse_test.go
@@ -7,6 +7,7 @@
 package internal
 
 import (
+	"strconv"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -20,13 +21,23 @@ import (
 )
 
 func TestCopyOrigExportLogsServiceResponse(t *testing.T) {
-	src := NewOrigExportLogsServiceResponse()
-	dest := NewOrigExportLogsServiceResponse()
-	CopyOrigExportLogsServiceResponse(dest, src)
-	assert.Equal(t, NewOrigExportLogsServiceResponse(), dest)
-	*src = *GenTestOrigExportLogsServiceResponse()
-	CopyOrigExportLogsServiceResponse(dest, src)
-	assert.Equal(t, src, dest)
+	for name, src := range genTestEncodingValuesExportLogsServiceResponse() {
+		for _, pooling := range []bool{true, false} {
+			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
+				prevPooling := UseProtoPooling.IsEnabled()
+				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
+				defer func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), prevPooling))
+				}()
+
+				dest := NewOrigExportLogsServiceResponse()
+				CopyOrigExportLogsServiceResponse(dest, src)
+				assert.Equal(t, src, dest)
+				CopyOrigExportLogsServiceResponse(dest, dest)
+				assert.Equal(t, src, dest)
+			})
+		}
+	}
 }
 
 func TestMarshalAndUnmarshalJSONOrigExportLogsServiceResponseUnknown(t *testing.T) {
@@ -41,7 +52,7 @@ func TestMarshalAndUnmarshalJSONOrigExportLogsServiceResponseUnknown(t *testing.
 func TestMarshalAndUnmarshalJSONOrigExportLogsServiceResponse(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportLogsServiceResponse() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name, func(t *testing.T) {
+			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
@@ -85,7 +96,7 @@ func TestMarshalAndUnmarshalProtoOrigExportLogsServiceResponseUnknown(t *testing
 func TestMarshalAndUnmarshalProtoOrigExportLogsServiceResponse(t *testing.T) {
 	for name, src := range genTestEncodingValuesExportLogsServiceResponse() {
 		for _, pooling := range []bool{true, false} {
-			t.Run(name, func(t *testing.T) {
+			t.Run(name+"pooling_"+strconv.FormatBool(pooling), func(t *testing.T) {
 				prevPooling := UseProtoPooling.IsEnabled()
 				require.NoError(t, featuregate.GlobalRegistry().Set(UseProtoPooling.ID(), pooling))
 				defer func() {
