diff --git a/internal/tools/go.mod b/internal/tools/go.mod
index 42724b4bc..977c9a190 100644
--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -200,7 +200,7 @@ require (
 	golang.org/x/crypto v0.3.0 // indirect
 	golang.org/x/exp/typeparams v0.0.0-20221208152030-732eee02a75a // indirect
 	golang.org/x/mod v0.7.0 // indirect
-	golang.org/x/net v0.6.0 // indirect
+	golang.org/x/net v0.7.0 // indirect
 	golang.org/x/sync v0.1.0 // indirect
 	golang.org/x/sys v0.5.0 // indirect
 	golang.org/x/text v0.7.0 // indirect
