diff --git a/service/builder/receivers_builder.go b/service/builder/receivers_builder.go
index 9afeed00a..8db737746 100644
--- a/service/builder/receivers_builder.go
+++ b/service/builder/receivers_builder.go
@@ -96,7 +96,7 @@ func NewReceiversBuilder(
 	builtPipelines BuiltPipelines,
 	factories map[configmodels.Type]component.ReceiverFactoryBase,
 ) *ReceiversBuilder {
-	return &ReceiversBuilder{logger.With(zap.String(kindLogKey, kindLogReceiver)), config, builtPipelines, factories}
+	return &ReceiversBuilder{logger.With(zap.String(kindLogKey, kindLogsReceiver)), config, builtPipelines, factories}
 }
 
 // BuildProcessors receivers from config.
@@ -193,7 +193,7 @@ func (rb *ReceiversBuilder) attachReceiverToPipelines(
 
 	case configmodels.LogsDataType:
 		junction := buildFanoutLogConsumer(builtPipelines)
-		createdReceiver, err = createLogReceiver(context.Background(), factory, logger, config, junction)
+		createdReceiver, err = createLogsReceiver(context.Background(), factory, logger, config, junction)
 
 	default:
 		err = configerror.ErrDataTypeIsNotSupported
@@ -325,13 +325,13 @@ func buildFanoutMetricConsumer(pipelines []*builtPipeline) consumer.MetricsConsu
 	return processor.CreateMetricsFanOutConnector(pipelineConsumers)
 }
 
-func buildFanoutLogConsumer(pipelines []*builtPipeline) consumer.LogConsumer {
+func buildFanoutLogConsumer(pipelines []*builtPipeline) consumer.LogsConsumer {
 	// Optimize for the case when there is only one processor, no need to create junction point.
 	if len(pipelines) == 1 {
 		return pipelines[0].firstLC
 	}
 
-	var pipelineConsumers []consumer.LogConsumer
+	var pipelineConsumers []consumer.LogsConsumer
 	anyPipelineMutatesData := false
 	for _, pipeline := range pipelines {
 		pipelineConsumers = append(pipelineConsumers, pipeline.firstLC)
@@ -426,19 +426,19 @@ func createMetricsReceiver(
 	return factoryOld.CreateMetricsReceiver(context.Background(), logger, cfg, metricsConverter)
 }
 
-// createLogReceiver creates a log receiver using given factory and next consumer.
-func createLogReceiver(
+// createLogsReceiver creates a log receiver using given factory and next consumer.
+func createLogsReceiver(
 	ctx context.Context,
 	factoryBase component.ReceiverFactoryBase,
 	logger *zap.Logger,
 	cfg configmodels.Receiver,
-	nextConsumer consumer.LogConsumer,
-) (component.LogReceiver, error) {
-	factory, ok := factoryBase.(component.LogReceiverFactory)
+	nextConsumer consumer.LogsConsumer,
+) (component.LogsReceiver, error) {
+	factory, ok := factoryBase.(component.LogsReceiverFactory)
 	if !ok {
 		return nil, fmt.Errorf("receiver %q does support data type %q",
 			cfg.Name(), configmodels.LogsDataType)
 	}
 	creationParams := component.ReceiverCreateParams{Logger: logger}
-	return factory.CreateLogReceiver(ctx, creationParams, cfg, nextConsumer)
+	return factory.CreateLogsReceiver(ctx, creationParams, cfg, nextConsumer)
 }
