diff --git a/cmd/mdatagen/internal/metadata_test.go b/cmd/mdatagen/internal/metadata_test.go
index 952e6963c..41b90ac1e 100644
--- a/cmd/mdatagen/internal/metadata_test.go
+++ b/cmd/mdatagen/internal/metadata_test.go
@@ -77,6 +77,10 @@ func TestValidate(t *testing.T) {
 			name:    "testdata/no_metric_description.yaml",
 			wantErr: "metric \"default.metric\": missing metric description",
 		},
+		{
+			name:    "testdata/events/no_description.yaml",
+			wantErr: "event \"default.event\": missing event description",
+		},
 		{
 			name:    "testdata/no_metric_unit.yaml",
 			wantErr: "metric \"default.metric\": missing metric unit",
@@ -99,6 +103,10 @@ func TestValidate(t *testing.T) {
 			name:    "testdata/unknown_metric_attribute.yaml",
 			wantErr: "metric \"system.cpu.time\" refers to undefined attributes: [missing]",
 		},
+		{
+			name:    "testdata/events/unknown_attribute.yaml",
+			wantErr: "event \"system.event\" refers to undefined attributes: [missing]",
+		},
 		{
 			name:    "testdata/unused_attribute.yaml",
 			wantErr: "unused attributes: [unused_attr]",
