diff --git a/internal/otlptext/metrics.go b/internal/otlptext/metrics.go
index a107de27e..353679737 100644
--- a/internal/otlptext/metrics.go
+++ b/internal/otlptext/metrics.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package otlptext
+package otlptext // import "go.opentelemetry.io/collector/internal/otlptext"
 
 import (
 	"go.opentelemetry.io/collector/model/pdata"
