diff --git a/model/pdata/metrics.go b/model/pdata/metrics.go
index 4daeef0ce..f19a3d74d 100644
--- a/model/pdata/metrics.go
+++ b/model/pdata/metrics.go
@@ -111,8 +111,8 @@ func (md Metrics) DataPointCount() (dataPointCount int) {
 					dataPointCount += m.DoubleGauge().DataPoints().Len()
 				case MetricDataTypeIntSum:
 					dataPointCount += m.IntSum().DataPoints().Len()
-				case MetricDataTypeDoubleSum:
-					dataPointCount += m.DoubleSum().DataPoints().Len()
+				case MetricDataTypeSum:
+					dataPointCount += m.Sum().DataPoints().Len()
 				case MetricDataTypeIntHistogram:
 					dataPointCount += m.IntHistogram().DataPoints().Len()
 				case MetricDataTypeHistogram:
@@ -134,7 +134,7 @@ const (
 	MetricDataTypeIntGauge
 	MetricDataTypeDoubleGauge
 	MetricDataTypeIntSum
-	MetricDataTypeDoubleSum
+	MetricDataTypeSum
 	MetricDataTypeIntHistogram
 	MetricDataTypeHistogram
 	MetricDataTypeSummary
@@ -151,8 +151,8 @@ func (mdt MetricDataType) String() string {
 		return "DoubleGauge"
 	case MetricDataTypeIntSum:
 		return "IntSum"
-	case MetricDataTypeDoubleSum:
-		return "DoubleSum"
+	case MetricDataTypeSum:
+		return "Sum"
 	case MetricDataTypeIntHistogram:
 		return "IntHistogram"
 	case MetricDataTypeHistogram:
@@ -174,7 +174,7 @@ func (ms Metric) DataType() MetricDataType {
 	case *otlpmetrics.Metric_IntSum:
 		return MetricDataTypeIntSum
 	case *otlpmetrics.Metric_Sum:
-		return MetricDataTypeDoubleSum
+		return MetricDataTypeSum
 	case *otlpmetrics.Metric_IntHistogram:
 		return MetricDataTypeIntHistogram
 	case *otlpmetrics.Metric_Histogram:
@@ -195,7 +195,7 @@ func (ms Metric) SetDataType(ty MetricDataType) {
 		ms.orig.Data = &otlpmetrics.Metric_Gauge{Gauge: &otlpmetrics.Gauge{}}
 	case MetricDataTypeIntSum:
 		ms.orig.Data = &otlpmetrics.Metric_IntSum{IntSum: &otlpmetrics.IntSum{}}
-	case MetricDataTypeDoubleSum:
+	case MetricDataTypeSum:
 		ms.orig.Data = &otlpmetrics.Metric_Sum{Sum: &otlpmetrics.Sum{}}
 	case MetricDataTypeIntHistogram:
 		ms.orig.Data = &otlpmetrics.Metric_IntHistogram{IntHistogram: &otlpmetrics.IntHistogram{}}
@@ -227,11 +227,11 @@ func (ms Metric) IntSum() IntSum {
 	return newIntSum(ms.orig.Data.(*otlpmetrics.Metric_IntSum).IntSum)
 }
 
-// DoubleSum returns the data as DoubleSum.
-// Calling this function when DataType() != MetricDataTypeDoubleSum will cause a panic.
+// Sum returns the data as Sum.
+// Calling this function when DataType() != MetricDataTypeSum will cause a panic.
 // Calling this function on zero-initialized Metric will cause a panic.
-func (ms Metric) DoubleSum() DoubleSum {
-	return newDoubleSum(ms.orig.Data.(*otlpmetrics.Metric_Sum).Sum)
+func (ms Metric) Sum() Sum {
+	return newSum(ms.orig.Data.(*otlpmetrics.Metric_Sum).Sum)
 }
 
 // IntHistogram returns the data as IntHistogram.
@@ -271,7 +271,7 @@ func copyData(src, dest *otlpmetrics.Metric) {
 		dest.Data = data
 	case *otlpmetrics.Metric_Sum:
 		data := &otlpmetrics.Metric_Sum{Sum: &otlpmetrics.Sum{}}
-		newDoubleSum(srcData.Sum).CopyTo(newDoubleSum(data.Sum))
+		newSum(srcData.Sum).CopyTo(newSum(data.Sum))
 		dest.Data = data
 	case *otlpmetrics.Metric_IntHistogram:
 		data := &otlpmetrics.Metric_IntHistogram{IntHistogram: &otlpmetrics.IntHistogram{}}
