diff --git a/exporter/exporterparser/prometheus.go b/exporter/exporterparser/prometheus.go
index c86b1baf6..6ba1004be 100644
--- a/exporter/exporterparser/prometheus.go
+++ b/exporter/exporterparser/prometheus.go
@@ -23,6 +23,7 @@ import (
 
 	"github.com/census-instrumentation/opencensus-service/data"
 	"github.com/census-instrumentation/opencensus-service/exporter"
+	"github.com/spf13/viper"
 
 	// TODO: once this repository has been transferred to the
 	// official census-ecosystem location, update this import path.
@@ -33,34 +34,32 @@ import (
 
 type prometheusConfig struct {
 	// Namespace if set, exports metrics under the provided value.
-	Namespace string `yaml:"namespace"`
+	Namespace string `mapstructure:"namespace"`
 
 	// ConstLabels are values that are applied for every exported metric.
-	ConstLabels prometheus_golang.Labels `yaml:"const_labels"`
+	ConstLabels prometheus_golang.Labels `mapstructure:"const_labels"`
 
 	// The address on which the Prometheus scrape handler will be run on.
-	Address string `yaml:"address"`
+	Address string `mapstructure:"address"`
 }
 
 var errBlankPrometheusAddress = errors.New("expecting a non-blank address to run the Prometheus metrics handler")
 
-// PrometheusExportersFromYAML parses the yaml bytes and returns exporter.MetricsExporters
+// PrometheusExportersFromViper unmarshals the viper and returns exporter.MetricsExporters
 // targeting Prometheus according to the configuration settings.
 // It allows HTTP clients to scrape it on endpoint path "/metrics".
-func PrometheusExportersFromYAML(config []byte) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
+func PrometheusExportersFromViper(v *viper.Viper) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
 	var cfg struct {
-		Exporters *struct {
-			Prometheus *prometheusConfig `yaml:"prometheus"`
-		} `yaml:"exporters"`
+		Prometheus *prometheusConfig `mapstructure:"prometheus"`
 	}
-	if err := yamlUnmarshal(config, &cfg); err != nil {
+	if err := v.Unmarshal(&cfg); err != nil {
 		return nil, nil, nil, err
 	}
-	if cfg.Exporters == nil || cfg.Exporters.Prometheus == nil {
+	if cfg.Prometheus == nil {
 		return nil, nil, nil, nil
 	}
 
-	pcfg := cfg.Exporters.Prometheus
+	pcfg := cfg.Prometheus
 	addr := strings.TrimSpace(pcfg.Address)
 	if addr == "" {
 		err = errBlankPrometheusAddress
