diff --git a/exporter/exporterparser/stackdriver.go b/exporter/exporterparser/stackdriver.go
index 3f6240eaf..c77bba766 100644
--- a/exporter/exporterparser/stackdriver.go
+++ b/exporter/exporterparser/stackdriver.go
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	"contrib.go.opencensus.io/exporter/stackdriver"
+	"github.com/spf13/viper"
 	"go.opencensus.io/trace"
 
 	"github.com/census-instrumentation/opencensus-service/data"
@@ -28,10 +29,10 @@ import (
 )
 
 type stackdriverConfig struct {
-	ProjectID     string `yaml:"project,omitempty"`
-	EnableTracing bool   `yaml:"enable_tracing,omitempty"`
-	EnableMetrics bool   `yaml:"enable_metrics,omitempty"`
-	MetricPrefix  string `yaml:"metric_prefix,omitempty"`
+	ProjectID     string `mapstructure:"project,omitempty"`
+	EnableTracing bool   `mapstructure:"enable_tracing,omitempty"`
+	EnableMetrics bool   `mapstructure:"enable_metrics,omitempty"`
+	MetricPrefix  string `mapstructure:"metric_prefix,omitempty"`
 }
 
 type stackdriverExporter struct {
@@ -40,21 +41,16 @@ type stackdriverExporter struct {
 
 var _ exporter.TraceExporter = (*stackdriverExporter)(nil)
 
-// StackdriverTraceExportersFromYAML parses the yaml bytes and returns an exporter.TraceExporter targeting
+// StackdriverTraceExportersFromViper unmarshals the viper and returns an exporter.TraceExporter targeting
 // Stackdriver according to the configuration settings.
-func StackdriverTraceExportersFromYAML(config []byte) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
+func StackdriverTraceExportersFromViper(v *viper.Viper) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
 	var cfg struct {
-		Exporters *struct {
-			Stackdriver *stackdriverConfig `yaml:"stackdriver"`
-		} `yaml:"exporters"`
+		Stackdriver *stackdriverConfig `mapstructure:"stackdriver"`
 	}
-	if err := yamlUnmarshal(config, &cfg); err != nil {
+	if err := v.Unmarshal(&cfg); err != nil {
 		return nil, nil, nil, err
 	}
-	if cfg.Exporters == nil {
-		return nil, nil, nil, nil
-	}
-	sc := cfg.Exporters.Stackdriver
+	sc := cfg.Stackdriver
 	if sc == nil {
 		return nil, nil, nil, nil
 	}
