diff --git a/pdata/pmetric/generated_summarydatapointslice_test.go b/pdata/pmetric/generated_summarydatapointslice_test.go
index e48edb1dc..ef6509a7a 100644
--- a/pdata/pmetric/generated_summarydatapointslice_test.go
+++ b/pdata/pmetric/generated_summarydatapointslice_test.go
@@ -51,16 +51,25 @@ func TestSummaryDataPointSliceReadOnly(t *testing.T) {
 
 func TestSummaryDataPointSlice_CopyTo(t *testing.T) {
 	dest := NewSummaryDataPointSlice()
-	// Test CopyTo to empty
+	// Test CopyTo empty
 	NewSummaryDataPointSlice().CopyTo(dest)
 	assert.Equal(t, NewSummaryDataPointSlice(), dest)
 
-	// Test CopyTo larger slice
-	generateTestSummaryDataPointSlice().CopyTo(dest)
+	// Test CopyTo larger slice and EnsureCapacity
+	src := generateTestSummaryDataPointSlice()
+	src.CopyTo(dest)
 	assert.Equal(t, generateTestSummaryDataPointSlice(), dest)
 
 	// Test CopyTo same size slice
-	generateTestSummaryDataPointSlice().CopyTo(dest)
+	src.CopyTo(dest)
+	assert.Equal(t, generateTestSummaryDataPointSlice(), dest)
+}
+
+func TestSummaryDataPointSlice_CopyToAndEnsureCapacity(t *testing.T) {
+	dest := NewSummaryDataPointSlice()
+	src := generateTestSummaryDataPointSlice()
+	dest.EnsureCapacity(src.Len())
+	src.CopyTo(dest)
 	assert.Equal(t, generateTestSummaryDataPointSlice(), dest)
 }
 
