diff --git a/internal/otlptext/metrics.go b/internal/otlptext/metrics.go
index 4e3d5e69b..25a72ed1a 100644
--- a/internal/otlptext/metrics.go
+++ b/internal/otlptext/metrics.go
@@ -14,19 +14,17 @@
 
 package otlptext // import "go.opentelemetry.io/collector/internal/otlptext"
 
-import (
-	"go.opentelemetry.io/collector/model/pdata"
-)
+import "go.opentelemetry.io/collector/pdata/pmetric"
 
-// NewTextMetricsMarshaler returns a serializer.MetricsMarshaler to encode to OTLP text bytes.
-func NewTextMetricsMarshaler() pdata.MetricsMarshaler {
+// NewTextMetricsMarshaler returns a serializer.Marshaler to encode to OTLP text bytes.
+func NewTextMetricsMarshaler() pmetric.Marshaler {
 	return textMetricsMarshaler{}
 }
 
 type textMetricsMarshaler struct{}
 
-// MarshalMetrics pdata.Metrics to OTLP text.
-func (textMetricsMarshaler) MarshalMetrics(md pdata.Metrics) ([]byte, error) {
+// MarshalMetrics pmetric.Metrics to OTLP text.
+func (textMetricsMarshaler) MarshalMetrics(md pmetric.Metrics) ([]byte, error) {
 	buf := dataBuffer{}
 	rms := md.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
