diff --git a/pdata/pprofile/generated_locationslice.go b/pdata/pprofile/generated_locationslice.go
index 0aa908bbe..970041835 100644
--- a/pdata/pprofile/generated_locationslice.go
+++ b/pdata/pprofile/generated_locationslice.go
@@ -99,7 +99,7 @@ func (es LocationSlice) EnsureCapacity(newCap int) {
 // It returns the newly added Location.
 func (es LocationSlice) AppendEmpty() Location {
 	es.state.AssertMutable()
-	*es.orig = append(*es.orig, internal.NewOrigPtrLocation())
+	*es.orig = append(*es.orig, internal.NewOrigLocation())
 	return es.At(es.Len() - 1)
 }
 
