diff --git a/pdata/pprofile/generated_function.go b/pdata/pprofile/generated_function.go
index 61edff712..1d497d242 100644
--- a/pdata/pprofile/generated_function.go
+++ b/pdata/pprofile/generated_function.go
@@ -32,7 +32,7 @@ func newFunction(orig *otlpprofiles.Function, state *internal.State) Function {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewFunction() Function {
-	return newFunction(internal.NewOrigPtrFunction(), internal.NewState())
+	return newFunction(internal.NewOrigFunction(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
