diff --git a/config/configcheck/configcheck.go b/config/configcheck/configcheck.go
index 3a11e3ea7..04f97e4dd 100644
--- a/config/configcheck/configcheck.go
+++ b/config/configcheck/configcheck.go
@@ -24,7 +24,7 @@ import (
 	"regexp"
 	"strings"
 
-	"github.com/open-telemetry/opentelemetry-collector/component/componenterr"
+	"github.com/open-telemetry/opentelemetry-collector/component/componenterror"
 	"github.com/open-telemetry/opentelemetry-collector/config"
 )
 
@@ -56,7 +56,7 @@ func ValidateConfigFromFactories(factories config.Factories) error {
 		}
 	}
 
-	return componenterr.CombineErrors(errs)
+	return componenterror.CombineErrors(errs)
 }
 
 // ValidateConfig enforces that given configuration object is following the patterns
@@ -109,7 +109,7 @@ func validateConfigDataType(t reflect.Type) error {
 		// reflect.UnsafePointer.
 	}
 
-	if err := componenterr.CombineErrors(errs); err != nil {
+	if err := componenterror.CombineErrors(errs); err != nil {
 		return fmt.Errorf(
 			"type %q from package %q has invalid config settings: %v",
 			t.Name(),
