diff --git a/extension/auth/client_test.go b/extension/auth/client_test.go
index ea643b2cc..ec71e0115 100644
--- a/extension/auth/client_test.go
+++ b/extension/auth/client_test.go
@@ -45,7 +45,7 @@ func TestClientDefaultValues(t *testing.T) {
 
 func TestWithClientStart(t *testing.T) {
 	called := false
-	e := NewClient(WithClientStart(func(c context.Context, h component.Host) error {
+	e := NewClient(WithClientStart(func(_ context.Context, _ component.Host) error {
 		called = true
 		return nil
 	}))
@@ -60,7 +60,7 @@ func TestWithClientStart(t *testing.T) {
 
 func TestWithClientShutdown(t *testing.T) {
 	called := false
-	e := NewClient(WithClientShutdown(func(c context.Context) error {
+	e := NewClient(WithClientShutdown(func(_ context.Context) error {
 		called = true
 		return nil
 	}))
