diff --git a/translator/internaldata/metrics_to_oc.go b/translator/internaldata/metrics_to_oc.go
index e8798643f..78b33d784 100644
--- a/translator/internaldata/metrics_to_oc.go
+++ b/translator/internaldata/metrics_to_oc.go
@@ -18,7 +18,7 @@ import (
 	"sort"
 
 	ocmetrics "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
-	"github.com/golang/protobuf/ptypes/wrappers"
+	"google.golang.org/protobuf/types/known/wrapperspb"
 
 	"go.opentelemetry.io/collector/consumer/consumerdata"
 	"go.opentelemetry.io/collector/consumer/pdata"
@@ -440,14 +440,14 @@ func labelValuesToOC(labels pdata.StringMap, labelKeys *labelKeys) []*ocmetrics.
 	return labelValues
 }
 
-func int64Value(val uint64) *wrappers.Int64Value {
-	return &wrappers.Int64Value{
+func int64Value(val uint64) *wrapperspb.Int64Value {
+	return &wrapperspb.Int64Value{
 		Value: int64(val),
 	}
 }
 
-func doubleValue(val float64) *wrappers.DoubleValue {
-	return &wrappers.DoubleValue{
+func doubleValue(val float64) *wrapperspb.DoubleValue {
+	return &wrapperspb.DoubleValue{
 		Value: val,
 	}
 }
