diff --git a/pdata/ptrace/internal/ptracejson/json.go b/pdata/ptrace/internal/ptracejson/json.go
index 1989862cf..67b91d2b5 100644
--- a/pdata/ptrace/internal/ptracejson/json.go
+++ b/pdata/ptrace/internal/ptracejson/json.go
@@ -17,6 +17,7 @@ package ptracejson // import "go.opentelemetry.io/collector/pdata/ptrace/interna
 import (
 	"fmt"
 
+	"github.com/gogo/protobuf/jsonpb"
 	jsoniter "github.com/json-iterator/go"
 
 	otlpcollectortrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1"
@@ -25,6 +26,13 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal/otlp"
 )
 
+var JSONMarshaler = &jsonpb.Marshaler{
+	// https://github.com/open-telemetry/opentelemetry-specification/pull/2758
+	EnumsAsInts: true,
+	// https://github.com/open-telemetry/opentelemetry-specification/pull/2829
+	OrigName: false,
+}
+
 func UnmarshalTraceData(buf []byte, dest *otlptrace.TracesData) error {
 	iter := jsoniter.ConfigFastest.BorrowIterator(buf)
 	defer jsoniter.ConfigFastest.ReturnIterator(iter)
