diff --git a/receiver/otlpreceiver/otlphttp.go b/receiver/otlpreceiver/otlphttp.go
index dc5628b1a..d733e533c 100644
--- a/receiver/otlpreceiver/otlphttp.go
+++ b/receiver/otlpreceiver/otlphttp.go
@@ -15,7 +15,7 @@
 package otlpreceiver // import "go.opentelemetry.io/collector/receiver/otlpreceiver"
 
 import (
-	"io/ioutil"
+	"io"
 	"net/http"
 
 	spb "google.golang.org/genproto/googleapis/rpc/status"
@@ -111,7 +111,7 @@ func handleLogs(resp http.ResponseWriter, req *http.Request, logsReceiver *logs.
 }
 
 func readAndCloseBody(resp http.ResponseWriter, req *http.Request, encoder encoder) ([]byte, bool) {
-	body, err := ioutil.ReadAll(req.Body)
+	body, err := io.ReadAll(req.Body)
 	if err != nil {
 		writeError(resp, encoder, err, http.StatusBadRequest)
 		return nil, false
