diff --git a/translator/internaldata/traces_to_oc_test.go b/translator/internaldata/traces_to_oc_test.go
index ebf98fd1d..e089f778d 100644
--- a/translator/internaldata/traces_to_oc_test.go
+++ b/translator/internaldata/traces_to_oc_test.go
@@ -332,19 +332,6 @@ func TestInternalToOC(t *testing.T) {
 			},
 		},
 
-		{
-			name: "one-empty-one-nil-resource-spans",
-			td:   testdata.GenerateTraceDataOneEmptyOneNilResourceSpans(),
-			oc: []consumerdata.TraceData{
-				{
-					Node:         nil,
-					Resource:     nil,
-					Spans:        []*octrace.Span(nil),
-					SourceFormat: sourceFormat,
-				},
-			},
-		},
-
 		{
 			name: "no-libraries",
 			td:   testdata.GenerateTraceDataNoLibraries(),
@@ -371,19 +358,6 @@ func TestInternalToOC(t *testing.T) {
 			},
 		},
 
-		{
-			name: "one-empty-one-nil-instrumentation-library",
-			td:   testdata.GenerateTraceDataOneEmptyOneNilInstrumentationLibrary(),
-			oc: []consumerdata.TraceData{
-				{
-					Node:         ocNode,
-					Resource:     ocResource1,
-					Spans:        []*octrace.Span{},
-					SourceFormat: sourceFormat,
-				},
-			},
-		},
-
 		{
 			name: "one-span-no-resource",
 			td:   testdata.GenerateTraceDataOneSpanNoResource(),
@@ -410,19 +384,6 @@ func TestInternalToOC(t *testing.T) {
 			},
 		},
 
-		{
-			name: "one-span-one-nil",
-			td:   testdata.GenerateTraceDataOneSpanOneNil(),
-			oc: []consumerdata.TraceData{
-				{
-					Node:         ocNode,
-					Resource:     ocResource1,
-					Spans:        []*octrace.Span{ocSpan1},
-					SourceFormat: sourceFormat,
-				},
-			},
-		},
-
 		{
 			name: "two-spans-same-resource",
 			td:   testdata.GenerateTraceDataTwoSpansSameResource(),
@@ -456,8 +417,6 @@ func TestInternalToOC(t *testing.T) {
 		},
 	}
 
-	assert.EqualValues(t, testdata.NumTraceTests, len(tests))
-
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			assert.EqualValues(t, test.oc, TraceDataToOC(test.td))
