diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index c4e4a8ce2..8b023d5d0 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -93,3 +93,5 @@ replace go.opentelemetry.io/collector/component => ../../component
 replace go.opentelemetry.io/collector/semconv => ../../semconv
 
 replace go.opentelemetry.io/collector/consumer => ../../consumer
+
+replace go.opentelemetry.io/collector/service => ../../service
