diff --git a/config/configgrpc/go.mod b/config/configgrpc/go.mod
index 03136585b..b268a2dc3 100644
--- a/config/configgrpc/go.mod
+++ b/config/configgrpc/go.mod
@@ -59,7 +59,7 @@ require (
 	go.opentelemetry.io/collector/confmap v0.90.0 // indirect
 	go.opentelemetry.io/collector/extension v0.90.0 // indirect
 	go.opentelemetry.io/collector/featuregate v1.0.0 // indirect
-	go.opentelemetry.io/otel/exporters/prometheus v0.44.0 // indirect
+	go.opentelemetry.io/otel/exporters/prometheus v0.44.1-0.20231201153405-6027c1ae76f2 // indirect
 	go.opentelemetry.io/otel/metric v1.21.0 // indirect
 	go.opentelemetry.io/otel/sdk v1.21.0 // indirect
 	go.opentelemetry.io/otel/sdk/metric v1.21.0 // indirect
