diff --git a/pdata/plog/generated_scopelogsslice_test.go b/pdata/plog/generated_scopelogsslice_test.go
index 698d75530..4c250e84f 100644
--- a/pdata/plog/generated_scopelogsslice_test.go
+++ b/pdata/plog/generated_scopelogsslice_test.go
@@ -51,16 +51,25 @@ func TestScopeLogsSliceReadOnly(t *testing.T) {
 
 func TestScopeLogsSlice_CopyTo(t *testing.T) {
 	dest := NewScopeLogsSlice()
-	// Test CopyTo to empty
+	// Test CopyTo empty
 	NewScopeLogsSlice().CopyTo(dest)
 	assert.Equal(t, NewScopeLogsSlice(), dest)
 
-	// Test CopyTo larger slice
-	generateTestScopeLogsSlice().CopyTo(dest)
+	// Test CopyTo larger slice and EnsureCapacity
+	src := generateTestScopeLogsSlice()
+	src.CopyTo(dest)
 	assert.Equal(t, generateTestScopeLogsSlice(), dest)
 
 	// Test CopyTo same size slice
-	generateTestScopeLogsSlice().CopyTo(dest)
+	src.CopyTo(dest)
+	assert.Equal(t, generateTestScopeLogsSlice(), dest)
+}
+
+func TestScopeLogsSlice_CopyToAndEnsureCapacity(t *testing.T) {
+	dest := NewScopeLogsSlice()
+	src := generateTestScopeLogsSlice()
+	dest.EnsureCapacity(src.Len())
+	src.CopyTo(dest)
 	assert.Equal(t, generateTestScopeLogsSlice(), dest)
 }
 
