diff --git a/pdata/plog/generated_logrecordslice.go b/pdata/plog/generated_logrecordslice.go
index 63e04e141..8c09601b3 100644
--- a/pdata/plog/generated_logrecordslice.go
+++ b/pdata/plog/generated_logrecordslice.go
@@ -180,16 +180,30 @@ func (ms LogRecordSlice) unmarshalJSONIter(iter *json.Iterator) {
 }
 
 func copyOrigLogRecordSlice(dest, src []*otlplogs.LogRecord) []*otlplogs.LogRecord {
+	var newDest []*otlplogs.LogRecord
 	if cap(dest) < len(src) {
-		dest = make([]*otlplogs.LogRecord, len(src))
-		data := make([]otlplogs.LogRecord, len(src))
-		for i := range src {
-			dest[i] = &data[i]
+		newDest = make([]*otlplogs.LogRecord, len(src))
+		// Copy old pointers to re-use.
+		copy(newDest, dest)
+		// Add new pointers for missing elements from len(dest) to len(srt).
+		for i := len(dest); i < len(src); i++ {
+			newDest[i] = &otlplogs.LogRecord{}
+		}
+	} else {
+		newDest = dest[:len(src)]
+		// Cleanup the rest of the elements so GC can free the memory.
+		// This can happen when len(src) < len(dest) < cap(dest).
+		for i := len(src); i < len(dest); i++ {
+			dest[i] = nil
+		}
+		// Add new pointers for missing elements.
+		// This can happen when len(dest) < len(src) < cap(dest).
+		for i := len(dest); i < len(src); i++ {
+			newDest[i] = &otlplogs.LogRecord{}
 		}
 	}
-	dest = dest[:len(src)]
 	for i := range src {
-		copyOrigLogRecord(dest[i], src[i])
+		copyOrigLogRecord(newDest[i], src[i])
 	}
-	return dest
+	return newDest
 }
