diff --git a/receiver/prometheusreceiver/internal/metricfamily.go b/receiver/prometheusreceiver/internal/metricfamily.go
index eca80c3fd..e8f200a46 100644
--- a/receiver/prometheusreceiver/internal/metricfamily.go
+++ b/receiver/prometheusreceiver/internal/metricfamily.go
@@ -15,6 +15,7 @@
 package internal
 
 import (
+	"fmt"
 	"sort"
 	"strings"
 
@@ -22,6 +23,7 @@ import (
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/prometheus/prometheus/pkg/textparse"
 	"github.com/prometheus/prometheus/scrape"
+	"go.uber.org/zap"
 	"google.golang.org/protobuf/types/known/timestamppb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 )
@@ -46,7 +48,7 @@ type metricFamily struct {
 	groups            map[string]*metricGroup
 }
 
-func newMetricFamily(metricName string, mc MetadataCache) MetricFamily {
+func newMetricFamily(metricName string, mc MetadataCache, logger *zap.Logger) MetricFamily {
 	familyName := normalizeMetricName(metricName)
 
 	// lookup metadata based on familyName
@@ -62,11 +64,17 @@ func newMetricFamily(metricName string, mc MetadataCache) MetricFamily {
 			metadata.Metric = familyName
 			metadata.Type = textparse.MetricTypeUnknown
 		}
+	} else if !ok && isInternalMetric(metricName) {
+		metadata = defineInternalMetric(metricName, metadata, logger)
+	}
+	ocaMetricType := convToOCAMetricType(metadata.Type)
+	if ocaMetricType == metricspb.MetricDescriptor_UNSPECIFIED {
+		logger.Debug(fmt.Sprintf("Invalid metric : %s %+v", metricName, metadata))
 	}
 
 	return &metricFamily{
 		name:              familyName,
-		mtype:             convToOCAMetricType(metadata.Type),
+		mtype:             ocaMetricType,
 		mc:                mc,
 		droppedTimeseries: 0,
 		labelKeys:         make(map[string]bool),
@@ -77,6 +85,35 @@ func newMetricFamily(metricName string, mc MetadataCache) MetricFamily {
 	}
 }
 
+// Define manually the metadata of prometheus scrapper internal metrics
+func defineInternalMetric(metricName string, metadata scrape.MetricMetadata, logger *zap.Logger) scrape.MetricMetadata {
+	if metadata.Metric != "" && metadata.Type != "" && metadata.Help != "" {
+		logger.Debug("Internal metric seems already fully defined")
+		return metadata
+	}
+	metadata.Metric = metricName
+
+	switch metricName {
+	case scrapeUpMetricName:
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The scraping was successful"
+	case "scrape_duration_seconds":
+		metadata.Unit = "seconds"
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "Duration of the scrape"
+	case "scrape_samples_scraped":
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The number of samples the target exposed"
+	case "scrape_series_added":
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The approximate number of new series in this scrape"
+	case "scrape_samples_post_metric_relabeling":
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The number of samples remaining after metric relabeling was applied"
+	}
+	return metadata
+}
+
 func (mf *metricFamily) IsSameFamily(metricName string) bool {
 	// trim known suffix if necessary
 	familyName := normalizeMetricName(metricName)
