diff --git a/receiver/receiverprofiles/receiver_test.go b/receiver/receiverprofiles/receiver_test.go
index 2ba69c6ac..eb0945767 100644
--- a/receiver/receiverprofiles/receiver_test.go
+++ b/receiver/receiverprofiles/receiver_test.go
@@ -18,7 +18,7 @@ import (
 func TestNewFactoryWithProfiles(t *testing.T) {
 	var testType = component.MustNewType("test")
 	defaultCfg := struct{}{}
-	factory := receiver.NewFactory(
+	factory := NewFactory(
 		testType,
 		func() component.Config { return &defaultCfg },
 		WithProfiles(createProfiles, component.StabilityLevelAlpha),
