diff --git a/translator/internaldata/resource_to_oc.go b/translator/internaldata/resource_to_oc.go
index 565a94d4e..6c99f2b62 100644
--- a/translator/internaldata/resource_to_oc.go
+++ b/translator/internaldata/resource_to_oc.go
@@ -111,9 +111,6 @@ func internalResourceToOC(resource pdata.Resource) (*occommon.Node, *ocresource.
 			}
 			getProcessIdentifier(ocNode).Pid = uint32(pid)
 		case conventions.AttributeTelemetrySDKVersion:
-			if ocNode.LibraryInfo == nil {
-				ocNode.LibraryInfo = &occommon.LibraryInfo{}
-			}
 			getLibraryInfo(ocNode).CoreLibraryVersion = val
 		case conventions.OCAttributeExporterVersion:
 			getLibraryInfo(ocNode).ExporterVersion = val
