diff --git a/pdata/pmetric/exemplar_value_type_test.go b/pdata/pmetric/exemplar_value_type_test.go
index 9e7c89bd9..db83dfd26 100644
--- a/pdata/pmetric/exemplar_value_type_test.go
+++ b/pdata/pmetric/exemplar_value_type_test.go
@@ -13,5 +13,5 @@ func TestExemplarValueTypeString(t *testing.T) {
 	assert.Equal(t, "Empty", ExemplarValueTypeEmpty.String())
 	assert.Equal(t, "Int", ExemplarValueTypeInt.String())
 	assert.Equal(t, "Double", ExemplarValueTypeDouble.String())
-	assert.Equal(t, "", (ExemplarValueTypeDouble + 1).String())
+	assert.Empty(t, (ExemplarValueTypeDouble + 1).String())
 }
