diff --git a/pdata/internal/generated_wrapper_uint64slice.go b/pdata/internal/generated_wrapper_uint64slice.go
index 7eda6ac7b..c20fd9d0e 100644
--- a/pdata/internal/generated_wrapper_uint64slice.go
+++ b/pdata/internal/generated_wrapper_uint64slice.go
@@ -24,14 +24,10 @@ func NewUInt64SliceWrapper(orig *[]uint64, state *State) UInt64SliceWrapper {
 }
 
 func GenTestUInt64SliceWrapper() UInt64SliceWrapper {
-	orig := GenTestUint64Slice()
+	orig := []uint64{1, 2, 3}
 	return NewUInt64SliceWrapper(&orig, NewState())
 }
 
-func CopyUint64Slice(dst, src []uint64) []uint64 {
-	return append(dst[:0], src...)
-}
-
 func GenTestUint64Slice() []uint64 {
 	return []uint64{1, 2, 3}
 }
