diff --git a/pdata/internal/wrapper_traces.go b/pdata/internal/wrapper_traces.go
index 5e2b0db39..758a1cb27 100644
--- a/pdata/internal/wrapper_traces.go
+++ b/pdata/internal/wrapper_traces.go
@@ -3,22 +3,17 @@
 
 package internal // import "go.opentelemetry.io/collector/pdata/internal"
 
-import (
-	otlpcollectortrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1"
-	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
-)
-
 // TracesToProto internal helper to convert Traces to protobuf representation.
-func TracesToProto(l TracesWrapper) otlptrace.TracesData {
-	return otlptrace.TracesData{
+func TracesToProto(l TracesWrapper) TracesData {
+	return TracesData{
 		ResourceSpans: l.orig.ResourceSpans,
 	}
 }
 
 // TracesFromProto internal helper to convert protobuf representation to Traces.
 // This function set exclusive state assuming that it's called only once per Traces.
-func TracesFromProto(orig otlptrace.TracesData) TracesWrapper {
-	return NewTracesWrapper(&otlpcollectortrace.ExportTraceServiceRequest{
+func TracesFromProto(orig TracesData) TracesWrapper {
+	return NewTracesWrapper(&ExportTraceServiceRequest{
 		ResourceSpans: orig.ResourceSpans,
 	}, NewState())
 }
