diff --git a/component/receiver.go b/component/receiver.go
index 7e36d8f25..bcdf2a642 100644
--- a/component/receiver.go
+++ b/component/receiver.go
@@ -60,6 +60,9 @@ type ReceiverCreateParams struct {
 	// Logger that the factory can use during creation and can pass to the created
 	// component to be used later as well.
 	Logger *zap.Logger
+
+	// ApplicationStartInfo can be used by components for informational purposes
+	ApplicationStartInfo ApplicationStartInfo
 }
 
 // ReceiverFactory can create TraceReceiver and MetricsReceiver. This is the
