diff --git a/internal/collector/sampling/numeric_tag_filter.go b/internal/collector/sampling/numeric_tag_filter.go
index 3e3a08806..160903e50 100644
--- a/internal/collector/sampling/numeric_tag_filter.go
+++ b/internal/collector/sampling/numeric_tag_filter.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
