diff --git a/internal/collector/sampling/policy.go b/internal/collector/sampling/policy.go
index 0c2aa1d79..4bd362545 100644
--- a/internal/collector/sampling/policy.go
+++ b/internal/collector/sampling/policy.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -20,7 +20,7 @@ import (
 
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 
-	"github.com/census-instrumentation/opencensus-service/data"
+	"github.com/open-telemetry/opentelemetry-service/data"
 )
 
 // TraceData stores the sampling related trace data.
