diff --git a/confmap/internal/e2e/go.mod b/confmap/internal/e2e/go.mod
index 4d8c6918e..c0e995368 100644
--- a/confmap/internal/e2e/go.mod
+++ b/confmap/internal/e2e/go.mod
@@ -4,10 +4,10 @@ go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/config/configopaque v1.13.0
+	go.opentelemetry.io/collector/config/configopaque v1.14.0
 	go.opentelemetry.io/collector/confmap v1.14.1
-	go.opentelemetry.io/collector/confmap/provider/envprovider v0.107.0
-	go.opentelemetry.io/collector/confmap/provider/fileprovider v0.107.0
+	go.opentelemetry.io/collector/confmap/provider/envprovider v0.108.0
+	go.opentelemetry.io/collector/confmap/provider/fileprovider v0.108.0
 )
 
 require (
