diff --git a/service/extensions/extensions.go b/service/extensions/extensions.go
index 140d8b3b6..fcefcb33f 100644
--- a/service/extensions/extensions.go
+++ b/service/extensions/extensions.go
@@ -209,7 +209,7 @@ func New(ctx context.Context, set Settings, cfg Config, options ...Option) (*Ext
 	}
 
 	for _, extID := range cfg {
-		instanceID := componentstatus.NewInstanceIDWithPipelineIDs(extID, component.KindExtension)
+		instanceID := componentstatus.NewInstanceID(extID, component.KindExtension)
 		extSet := extension.Settings{
 			ID:                extID,
 			TelemetrySettings: set.Telemetry,
