diff --git a/connector/connectorprofiles/go.mod b/connector/connectorprofiles/go.mod
index 02fa46135..1858bafa5 100644
--- a/connector/connectorprofiles/go.mod
+++ b/connector/connectorprofiles/go.mod
@@ -6,14 +6,14 @@ require (
 	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v0.115.0
 	go.opentelemetry.io/collector/connector v0.115.0
-	go.opentelemetry.io/collector/connector/xconnector v0.115.0
+	go.opentelemetry.io/collector/connector/xconnector v0.0.0-20241214150434-e9bc4bde924e
 	go.opentelemetry.io/collector/consumer v1.21.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.115.0
-	go.opentelemetry.io/collector/consumer/xconsumer v0.115.1-0.20241213185000-4593ba7de234
+	go.opentelemetry.io/collector/consumer/xconsumer v0.0.0-20241214150434-e9bc4bde924e
 	go.opentelemetry.io/collector/pdata/pprofile v0.115.0
 	go.opentelemetry.io/collector/pdata/testdata v0.115.0
 	go.opentelemetry.io/collector/pipeline v0.115.0
-	go.opentelemetry.io/collector/pipeline/xpipeline v0.115.0
+	go.opentelemetry.io/collector/pipeline/xpipeline v0.0.0-20241214150434-e9bc4bde924e
 )
 
 require (
