diff --git a/processor/filterprocessor/filter_processor_test.go b/processor/filterprocessor/filter_processor_test.go
index 2f5fb4190..15d83e605 100644
--- a/processor/filterprocessor/filter_processor_test.go
+++ b/processor/filterprocessor/filter_processor_test.go
@@ -306,10 +306,7 @@ func TestFilterMetricProcessor(t *testing.T) {
 			// next stores the results of the filter metric processor
 			next := new(consumertest.MetricsSink)
 			cfg := &Config{
-				ProcessorSettings: config.ProcessorSettings{
-					TypeVal: typeStr,
-					NameVal: typeStr,
-				},
+				ProcessorSettings: config.NewProcessorSettings(typeStr),
 				Metrics: MetricFilters{
 					Include: test.inc,
 					Exclude: test.exc,
