diff --git a/translator/trace/oc_to_otlp_test.go b/translator/trace/oc_to_otlp_test.go
index cef642c41..4d6662ba9 100644
--- a/translator/trace/oc_to_otlp_test.go
+++ b/translator/trace/oc_to_otlp_test.go
@@ -92,16 +92,16 @@ func TestOcNodeResourceToOtlp(t *testing.T) {
 	otlpResource = ocNodeResourceToOtlp(node, resource)
 
 	expectedAttrs := map[string]string{
-		conventions.AttributeHostHostname:    "host1",
-		ocAttributeProcessID:                 "123",
-		ocAttributeProcessStartTime:          "2020-02-11T20:26:00Z",
-		conventions.AttributeLibraryLanguage: "CPP",
-		ocAttributeExporterVersion:           "v1.2.0",
-		conventions.AttributeLibraryVersion:  "v2.0.1",
-		conventions.AttributeServiceName:     "svcA",
-		"node-attr":                          "val1",
-		ocAttributeResourceType:              "good-resource",
-		"resource-attr":                      "val2",
+		conventions.AttributeHostHostname:       "host1",
+		conventions.OCAttributeProcessID:        "123",
+		conventions.OCAttributeProcessStartTime: "2020-02-11T20:26:00Z",
+		conventions.AttributeLibraryLanguage:    "CPP",
+		conventions.OCAttributeExporterVersion:  "v1.2.0",
+		conventions.AttributeLibraryVersion:     "v2.0.1",
+		conventions.AttributeServiceName:        "svcA",
+		"node-attr":                             "val1",
+		conventions.OCAttributeResourceType:     "good-resource",
+		"resource-attr":                         "val2",
 	}
 
 	assert.EqualValues(t, len(expectedAttrs), len(otlpResource.Attributes))
@@ -118,7 +118,7 @@ func TestOcNodeResourceToOtlp(t *testing.T) {
 			node.Attributes[k] = "this will be overridden 1"
 		}
 	}
-	resource.Labels[ocAttributeResourceType] = "this will be overridden 2"
+	resource.Labels[conventions.OCAttributeResourceType] = "this will be overridden 2"
 
 	// Convert again.
 	otlpResource = ocNodeResourceToOtlp(node, resource)
