diff --git a/exporter/exporter.go b/exporter/exporter.go
index 14013dca0..ec26b1084 100644
--- a/exporter/exporter.go
+++ b/exporter/exporter.go
@@ -83,6 +83,8 @@ type Factory interface {
 
 	// LogsExporterStability gets the stability level of the LogsExporter.
 	LogsExporterStability() component.StabilityLevel
+
+	unexportedFactoryFunc()
 }
 
 // FactoryOption apply changes to Factory.
@@ -134,7 +136,6 @@ func (f CreateLogsFunc) CreateLogsExporter(ctx context.Context, set CreateSettin
 }
 
 type factory struct {
-	component.Factory
 	cfgType component.Type
 	component.CreateDefaultConfigFunc
 	CreateTracesFunc
@@ -149,12 +150,7 @@ func (f *factory) Type() component.Type {
 	return f.cfgType
 }
 
-// CreateDefaultConfig creates the default configuration for the Component.
-//
-// TODO: Remove this when we remove the private func from component.Factory and add it to every specialized Factory.
-func (f *factory) CreateDefaultConfig() component.Config {
-	return f.CreateDefaultConfigFunc()
-}
+func (f *factory) unexportedFactoryFunc() {}
 
 func (f *factory) TracesExporterStability() component.StabilityLevel {
 	return f.tracesStabilityLevel
