diff --git a/testbed/testbed/test_bed.go b/testbed/testbed/test_bed.go
index 4a89cd202..12040f513 100644
--- a/testbed/testbed/test_bed.go
+++ b/testbed/testbed/test_bed.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
