diff --git a/internal/collector/processor/nodebatcher/config_test.go b/internal/collector/processor/nodebatcher/config_test.go
index 6ae9639f5..2d9a64888 100644
--- a/internal/collector/processor/nodebatcher/config_test.go
+++ b/internal/collector/processor/nodebatcher/config_test.go
@@ -23,7 +23,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/open-telemetry/opentelemetry-service/configv2"
-	"github.com/open-telemetry/opentelemetry-service/models"
+	"github.com/open-telemetry/opentelemetry-service/configv2/configmodels"
 	"github.com/open-telemetry/opentelemetry-service/processor"
 )
 
@@ -50,7 +50,7 @@ func TestLoadConfig(t *testing.T) {
 
 	assert.Equal(t, p1,
 		&ConfigV2{
-			ProcessorSettings: models.ProcessorSettings{
+			ProcessorSettings: configmodels.ProcessorSettings{
 				TypeVal: "batch",
 				NameVal: "batch/2",
 			},
