diff --git a/pdata/pprofile/go.mod b/pdata/pprofile/go.mod
index 1eef997d4..9103b86a1 100644
--- a/pdata/pprofile/go.mod
+++ b/pdata/pprofile/go.mod
@@ -8,7 +8,7 @@ require (
 	go.opentelemetry.io/proto/slim/otlp/collector/profiles/v1development v0.0.1
 	go.opentelemetry.io/proto/slim/otlp/profiles/v1development v0.0.1
 	go.uber.org/goleak v1.3.0
-	google.golang.org/grpc v1.74.2
+	google.golang.org/grpc v1.75.0
 	google.golang.org/protobuf v1.36.7
 )
 
@@ -23,10 +23,10 @@ require (
 	go.opentelemetry.io/collector/featuregate v1.38.0 // indirect
 	go.opentelemetry.io/proto/slim/otlp v1.7.1 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
-	golang.org/x/net v0.40.0 // indirect
+	golang.org/x/net v0.41.0 // indirect
 	golang.org/x/sys v0.33.0 // indirect
-	golang.org/x/text v0.25.0 // indirect
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20250528174236-200df99c418a // indirect
+	golang.org/x/text v0.26.0 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20250707201910-8d1bb00bc6a7 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
