diff --git a/testbed/testbed/receivers.go b/testbed/testbed/receivers.go
index 136ae16fd..a28ef723c 100644
--- a/testbed/testbed/receivers.go
+++ b/testbed/testbed/receivers.go
@@ -44,7 +44,7 @@ import (
 // from Collector and the corresponding entity in the Collector that sends this data is
 // an exporter.
 type DataReceiver interface {
-	Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer) error
+	Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer, lc consumer.LogsConsumer) error
 	Stop() error
 
 	// Generate a config string to place in exporter part of collector config
@@ -99,7 +99,7 @@ func NewOCDataReceiver(port int) *OCDataReceiver {
 	return &OCDataReceiver{DataReceiverBase: DataReceiverBase{Port: port}}
 }
 
-func (or *OCDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer) error {
+func (or *OCDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer, _ consumer.LogsConsumer) error {
 	factory := opencensusreceiver.NewFactory()
 	cfg := factory.CreateDefaultConfig().(*opencensusreceiver.Config)
 	cfg.SetName(or.ProtocolName())
@@ -146,13 +146,15 @@ type JaegerDataReceiver struct {
 	receiver component.TraceReceiver
 }
 
+var _ DataReceiver = (*JaegerDataReceiver)(nil)
+
 const DefaultJaegerPort = 14250
 
 func NewJaegerDataReceiver(port int) *JaegerDataReceiver {
 	return &JaegerDataReceiver{DataReceiverBase: DataReceiverBase{Port: port}}
 }
 
-func (jr *JaegerDataReceiver) Start(tc consumer.TraceConsumer, _ consumer.MetricsConsumer) error {
+func (jr *JaegerDataReceiver) Start(tc consumer.TraceConsumer, _ consumer.MetricsConsumer, _ consumer.LogsConsumer) error {
 	factory := jaegerreceiver.NewFactory()
 	cfg := factory.CreateDefaultConfig().(*jaegerreceiver.Config)
 	cfg.SetName(jr.ProtocolName())
@@ -190,6 +192,7 @@ type OTLPDataReceiver struct {
 	DataReceiverBase
 	traceReceiver   component.TraceReceiver
 	metricsReceiver component.MetricsReceiver
+	logReceiver     component.LogsReceiver
 }
 
 // Ensure OTLPDataReceiver implements DataReceiver.
@@ -203,7 +206,7 @@ func NewOTLPDataReceiver(port int) *OTLPDataReceiver {
 	return &OTLPDataReceiver{DataReceiverBase: DataReceiverBase{Port: port}}
 }
 
-func (or *OTLPDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer) error {
+func (or *OTLPDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer, lc consumer.LogsConsumer) error {
 	factory := otlpreceiver.NewFactory()
 	cfg := factory.CreateDefaultConfig().(*otlpreceiver.Config)
 	cfg.SetName(or.ProtocolName())
@@ -217,10 +220,17 @@ func (or *OTLPDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.Metrics
 	if or.metricsReceiver, err = factory.CreateMetricsReceiver(context.Background(), params, cfg, mc); err != nil {
 		return err
 	}
+	if or.logReceiver, err = factory.CreateLogsReceiver(context.Background(), params, cfg, lc); err != nil {
+		return err
+	}
+
 	if err = or.traceReceiver.Start(context.Background(), or); err != nil {
 		return err
 	}
-	return or.metricsReceiver.Start(context.Background(), or)
+	if err = or.metricsReceiver.Start(context.Background(), or); err != nil {
+		return err
+	}
+	return or.logReceiver.Start(context.Background(), or)
 }
 
 func (or *OTLPDataReceiver) Stop() error {
@@ -251,13 +261,15 @@ type ZipkinDataReceiver struct {
 	receiver component.TraceReceiver
 }
 
+var _ DataReceiver = (*ZipkinDataReceiver)(nil)
+
 const DefaultZipkinAddressPort = 9411
 
 func NewZipkinDataReceiver(port int) *ZipkinDataReceiver {
 	return &ZipkinDataReceiver{DataReceiverBase: DataReceiverBase{Port: port}}
 }
 
-func (zr *ZipkinDataReceiver) Start(tc consumer.TraceConsumer, _ consumer.MetricsConsumer) error {
+func (zr *ZipkinDataReceiver) Start(tc consumer.TraceConsumer, _ consumer.MetricsConsumer, _ consumer.LogsConsumer) error {
 	factory := zipkinreceiver.NewFactory()
 	cfg := factory.CreateDefaultConfig().(*zipkinreceiver.Config)
 	cfg.SetName(zr.ProtocolName())
@@ -303,7 +315,7 @@ func NewPrometheusDataReceiver(port int) *PrometheusDataReceiver {
 	return &PrometheusDataReceiver{DataReceiverBase: DataReceiverBase{Port: port}}
 }
 
-func (dr *PrometheusDataReceiver) Start(_ consumer.TraceConsumer, mc consumer.MetricsConsumer) error {
+func (dr *PrometheusDataReceiver) Start(_ consumer.TraceConsumer, mc consumer.MetricsConsumer, _ consumer.LogsConsumer) error {
 	factory := prometheusreceiver.NewFactory()
 	cfg := factory.CreateDefaultConfig().(*prometheusreceiver.Config)
 	addr := fmt.Sprintf("0.0.0.0:%d", dr.Port)
