diff --git a/component/telemetry.go b/component/telemetry.go
index d8b6b134c..f4d022316 100644
--- a/component/telemetry.go
+++ b/component/telemetry.go
@@ -37,13 +37,13 @@ type TelemetrySettings struct {
 
 	// ReportComponentStatus allows a component to report runtime changes in status. The service
 	// will automatically report status for a component during startup and shutdown. Components can
-	// use this method to report status after start and before shutdown. ReportComponentStatus
-	// will only return errors if the API used incorrectly. The two scenarios where an error will
-	// be returned are:
-	//
-	//   - An illegal state transition
-	//   - Calling this method before component startup
-	//
-	// If the API is being used properly, these errors are safe to ignore.
+	// use this method to report status after start and before shutdown.
+	// Deprecated: [v0.92.0] This function will be removed in a future release.
+	// Use ReportStatus instead.
 	ReportComponentStatus func(*StatusEvent) error
+
+	// ReportStatus allows a component to report runtime changes in status. The service
+	// will automatically report status for a component during startup and shutdown. Components can
+	// use this method to report status after start and before shutdown.
+	ReportStatus func(*StatusEvent)
 }
