diff --git a/pdata/internal/generated_wrapper_valuetype_test.go b/pdata/internal/generated_wrapper_valuetype_test.go
index 7f90a262e..268fbd8ee 100644
--- a/pdata/internal/generated_wrapper_valuetype_test.go
+++ b/pdata/internal/generated_wrapper_valuetype_test.go
@@ -46,3 +46,26 @@ func TestMarshalAndUnmarshalJSONOrigValueType(t *testing.T) {
 
 	assert.Equal(t, src, dest)
 }
+
+func TestMarshalAndUnmarshalProtoOrigValueType(t *testing.T) {
+	src := &otlpprofiles.ValueType{}
+	FillOrigTestValueType(src)
+	buf, err := MarshalProtoOrigValueType(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigValueType(src))
+
+	dest := &otlpprofiles.ValueType{}
+	require.NoError(t, UnmarshalProtoOrigValueType(dest, buf))
+	assert.Equal(t, src, dest)
+}
+
+func TestMarshalAndUnmarshalProtoOrigEmptyValueType(t *testing.T) {
+	src := &otlpprofiles.ValueType{}
+	buf, err := MarshalProtoOrigValueType(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigValueType(src))
+
+	dest := &otlpprofiles.ValueType{}
+	require.NoError(t, UnmarshalProtoOrigValueType(dest, buf))
+	assert.Equal(t, src, dest)
+}
