diff --git a/pdata/internal/generated_wrapper_histogram_test.go b/pdata/internal/generated_wrapper_histogram_test.go
index b1c3f13d0..03d9eacaa 100644
--- a/pdata/internal/generated_wrapper_histogram_test.go
+++ b/pdata/internal/generated_wrapper_histogram_test.go
@@ -46,3 +46,26 @@ func TestMarshalAndUnmarshalJSONOrigHistogram(t *testing.T) {
 
 	assert.Equal(t, src, dest)
 }
+
+func TestMarshalAndUnmarshalProtoOrigHistogram(t *testing.T) {
+	src := &otlpmetrics.Histogram{}
+	FillOrigTestHistogram(src)
+	buf, err := MarshalProtoOrigHistogram(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigHistogram(src))
+
+	dest := &otlpmetrics.Histogram{}
+	require.NoError(t, UnmarshalProtoOrigHistogram(dest, buf))
+	assert.Equal(t, src, dest)
+}
+
+func TestMarshalAndUnmarshalProtoOrigEmptyHistogram(t *testing.T) {
+	src := &otlpmetrics.Histogram{}
+	buf, err := MarshalProtoOrigHistogram(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigHistogram(src))
+
+	dest := &otlpmetrics.Histogram{}
+	require.NoError(t, UnmarshalProtoOrigHistogram(dest, buf))
+	assert.Equal(t, src, dest)
+}
