diff --git a/pdata/internal/generated_wrapper_summary.go b/pdata/internal/generated_wrapper_summary.go
index 736a609d3..a2503c5bf 100644
--- a/pdata/internal/generated_wrapper_summary.go
+++ b/pdata/internal/generated_wrapper_summary.go
@@ -9,6 +9,7 @@ package internal
 import (
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
 func CopyOrigSummary(dest, src *otlpmetrics.Summary) {
@@ -41,3 +42,22 @@ func UnmarshalJSONOrigSummary(orig *otlpmetrics.Summary, iter *json.Iterator) {
 		return true
 	})
 }
+
+func SizeProtoOrigSummary(orig *otlpmetrics.Summary) int {
+	var n int
+	var l int
+	_ = l
+	for i := range orig.DataPoints {
+		l = SizeProtoOrigSummaryDataPoint(orig.DataPoints[i])
+		n += 1 + proto.Sov(uint64(l)) + l
+	}
+	return n
+}
+
+func MarshalProtoOrigSummary(orig *otlpmetrics.Summary) ([]byte, error) {
+	return orig.Marshal()
+}
+
+func UnmarshalProtoOrigSummary(orig *otlpmetrics.Summary, buf []byte) error {
+	return orig.Unmarshal(buf)
+}
