diff --git a/pdata/internal/generated_wrapper_exemplar_test.go b/pdata/internal/generated_wrapper_exemplar_test.go
index eb14fee7d..604533f9b 100644
--- a/pdata/internal/generated_wrapper_exemplar_test.go
+++ b/pdata/internal/generated_wrapper_exemplar_test.go
@@ -46,3 +46,26 @@ func TestMarshalAndUnmarshalJSONOrigExemplar(t *testing.T) {
 
 	assert.Equal(t, src, dest)
 }
+
+func TestMarshalAndUnmarshalProtoOrigExemplar(t *testing.T) {
+	src := &otlpmetrics.Exemplar{}
+	FillOrigTestExemplar(src)
+	buf, err := MarshalProtoOrigExemplar(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigExemplar(src))
+
+	dest := &otlpmetrics.Exemplar{}
+	require.NoError(t, UnmarshalProtoOrigExemplar(dest, buf))
+	assert.Equal(t, src, dest)
+}
+
+func TestMarshalAndUnmarshalProtoOrigEmptyExemplar(t *testing.T) {
+	src := &otlpmetrics.Exemplar{}
+	buf, err := MarshalProtoOrigExemplar(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigExemplar(src))
+
+	dest := &otlpmetrics.Exemplar{}
+	require.NoError(t, UnmarshalProtoOrigExemplar(dest, buf))
+	assert.Equal(t, src, dest)
+}
