diff --git a/pdata/internal/generated_wrapper_exponentialhistogram_test.go b/pdata/internal/generated_wrapper_exponentialhistogram_test.go
index 1cab51499..c5b9f57b2 100644
--- a/pdata/internal/generated_wrapper_exponentialhistogram_test.go
+++ b/pdata/internal/generated_wrapper_exponentialhistogram_test.go
@@ -46,3 +46,26 @@ func TestMarshalAndUnmarshalJSONOrigExponentialHistogram(t *testing.T) {
 
 	assert.Equal(t, src, dest)
 }
+
+func TestMarshalAndUnmarshalProtoOrigExponentialHistogram(t *testing.T) {
+	src := &otlpmetrics.ExponentialHistogram{}
+	FillOrigTestExponentialHistogram(src)
+	buf, err := MarshalProtoOrigExponentialHistogram(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigExponentialHistogram(src))
+
+	dest := &otlpmetrics.ExponentialHistogram{}
+	require.NoError(t, UnmarshalProtoOrigExponentialHistogram(dest, buf))
+	assert.Equal(t, src, dest)
+}
+
+func TestMarshalAndUnmarshalProtoOrigEmptyExponentialHistogram(t *testing.T) {
+	src := &otlpmetrics.ExponentialHistogram{}
+	buf, err := MarshalProtoOrigExponentialHistogram(src)
+	require.NoError(t, err)
+	assert.Equal(t, len(buf), SizeProtoOrigExponentialHistogram(src))
+
+	dest := &otlpmetrics.ExponentialHistogram{}
+	require.NoError(t, UnmarshalProtoOrigExponentialHistogram(dest, buf))
+	assert.Equal(t, src, dest)
+}
