diff --git a/pdata/pmetric/generated_histogramdatapointslice_test.go b/pdata/pmetric/generated_histogramdatapointslice_test.go
index 0c522902d..93b476a79 100644
--- a/pdata/pmetric/generated_histogramdatapointslice_test.go
+++ b/pdata/pmetric/generated_histogramdatapointslice_test.go
@@ -28,9 +28,9 @@ func TestHistogramDataPointSlice(t *testing.T) {
 	emptyVal := NewHistogramDataPoint()
 	testVal := generateTestHistogramDataPoint()
 	for i := 0; i < 7; i++ {
-		el := es.AppendEmpty()
+		es.AppendEmpty()
 		assert.Equal(t, emptyVal, es.At(i))
-		fillTestHistogramDataPoint(el)
+		internal.FillOrigTestHistogramDataPoint((*es.orig)[i])
 		assert.Equal(t, testVal, es.At(i))
 	}
 	assert.Equal(t, 7, es.Len())
@@ -51,26 +51,9 @@ func TestHistogramDataPointSliceReadOnly(t *testing.T) {
 
 func TestHistogramDataPointSlice_CopyTo(t *testing.T) {
 	dest := NewHistogramDataPointSlice()
-	// Test CopyTo empty
-	NewHistogramDataPointSlice().CopyTo(dest)
-	assert.Equal(t, NewHistogramDataPointSlice(), dest)
-
-	// Test CopyTo larger slice
 	src := generateTestHistogramDataPointSlice()
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestHistogramDataPointSlice(), dest)
-
-	// Test CopyTo same size slice
-	src.CopyTo(dest)
-	assert.Equal(t, generateTestHistogramDataPointSlice(), dest)
-
-	// Test CopyTo smaller size slice
-	NewHistogramDataPointSlice().CopyTo(dest)
-	assert.Equal(t, 0, dest.Len())
-
-	// Test CopyTo larger slice with enough capacity
-	src.CopyTo(dest)
-	assert.Equal(t, generateTestHistogramDataPointSlice(), dest)
 }
 
 func TestHistogramDataPointSlice_EnsureCapacity(t *testing.T) {
@@ -194,15 +177,7 @@ func TestHistogramDataPointSlice_Sort(t *testing.T) {
 }
 
 func generateTestHistogramDataPointSlice() HistogramDataPointSlice {
-	es := NewHistogramDataPointSlice()
-	fillTestHistogramDataPointSlice(es)
-	return es
-}
-
-func fillTestHistogramDataPointSlice(es HistogramDataPointSlice) {
-	*es.orig = make([]*otlpmetrics.HistogramDataPoint, 7)
-	for i := 0; i < 7; i++ {
-		(*es.orig)[i] = &otlpmetrics.HistogramDataPoint{}
-		fillTestHistogramDataPoint(newHistogramDataPoint((*es.orig)[i], es.state))
-	}
+	ms := NewHistogramDataPointSlice()
+	*ms.orig = internal.GenerateOrigTestHistogramDataPointSlice()
+	return ms
 }
