diff --git a/pdata/pmetric/metrics_test.go b/pdata/pmetric/metrics_test.go
index e23947ec5..063589b39 100644
--- a/pdata/pmetric/metrics_test.go
+++ b/pdata/pmetric/metrics_test.go
@@ -13,6 +13,7 @@ import (
 	goproto "google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/emptypb"
 
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlpcollectormetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1"
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
@@ -31,11 +32,10 @@ func TestResourceMetricsWireCompatibility(t *testing.T) {
 	// this repository are wire compatible.
 
 	// Generate ResourceMetrics as pdata struct.
-	metrics := NewMetrics()
-	fillTestResourceMetricsSlice(metrics.ResourceMetrics())
+	md := generateTestMetrics()
 
 	// Marshal its underlying ProtoBuf to wire.
-	wire1, err := gogoproto.Marshal(metrics.getOrig())
+	wire1, err := gogoproto.Marshal(md.getOrig())
 	require.NoError(t, err)
 	assert.NotNil(t, wire1)
 
@@ -56,7 +56,7 @@ func TestResourceMetricsWireCompatibility(t *testing.T) {
 
 	// Now compare that the original and final ProtoBuf messages are the same.
 	// This proves that goproto and gogoproto marshaling/unmarshaling are wire compatible.
-	assert.True(t, assert.Equal(t, metrics.getOrig(), &gogoprotoRM))
+	assert.True(t, assert.Equal(t, md.getOrig(), &gogoprotoRM))
 }
 
 func TestMetricCount(t *testing.T) {
@@ -151,19 +151,19 @@ func TestDataPointCountWithEmpty(t *testing.T) {
 }
 
 func TestDataPointCountWithNilDataPoints(t *testing.T) {
-	metrics := NewMetrics()
-	ilm := metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
+	md := NewMetrics()
+	ilm := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
 	ilm.Metrics().AppendEmpty().SetEmptyGauge()
 	ilm.Metrics().AppendEmpty().SetEmptySum()
 	ilm.Metrics().AppendEmpty().SetEmptyHistogram()
 	ilm.Metrics().AppendEmpty().SetEmptyExponentialHistogram()
 	ilm.Metrics().AppendEmpty().SetEmptySummary()
-	assert.Equal(t, 0, metrics.DataPointCount())
+	assert.Equal(t, 0, md.DataPointCount())
 }
 
 func TestHistogramWithNilSum(t *testing.T) {
-	metrics := NewMetrics()
-	ilm := metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
+	md := NewMetrics()
+	ilm := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
 	histo := ilm.Metrics().AppendEmpty()
 	histogramDataPoints := histo.SetEmptyHistogram().DataPoints()
 	histogramDataPoints.AppendEmpty()
@@ -173,8 +173,8 @@ func TestHistogramWithNilSum(t *testing.T) {
 }
 
 func TestHistogramWithValidSum(t *testing.T) {
-	metrics := NewMetrics()
-	ilm := metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
+	md := NewMetrics()
+	ilm := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
 	histo := ilm.Metrics().AppendEmpty()
 	histogramDataPoints := histo.SetEmptyHistogram().DataPoints()
 	histogramDataPoints.AppendEmpty()
@@ -630,11 +630,10 @@ func TestOtlpToFromInternalExponentialHistogramMutating(t *testing.T) {
 }
 
 func TestMetricsCopyTo(t *testing.T) {
-	metrics := NewMetrics()
-	fillTestResourceMetricsSlice(metrics.ResourceMetrics())
+	md := generateTestMetrics()
 	metricsCopy := NewMetrics()
-	metrics.CopyTo(metricsCopy)
-	assert.Equal(t, metrics, metricsCopy)
+	md.CopyTo(metricsCopy)
+	assert.Equal(t, md, metricsCopy)
 }
 
 func TestReadOnlyMetricsInvalidUsage(t *testing.T) {
@@ -946,8 +945,7 @@ func generateMetricsEmptyDataPoints() Metrics {
 }
 
 func BenchmarkMetricsUsage(b *testing.B) {
-	metrics := NewMetrics()
-	fillTestResourceMetricsSlice(metrics.ResourceMetrics())
+	md := generateTestMetrics()
 
 	ts := pcommon.NewTimestampFromTime(time.Now())
 
@@ -955,8 +953,8 @@ func BenchmarkMetricsUsage(b *testing.B) {
 	b.ResetTimer()
 
 	for bb := 0; bb < b.N; bb++ {
-		for i := 0; i < metrics.ResourceMetrics().Len(); i++ {
-			rm := metrics.ResourceMetrics().At(i)
+		for i := 0; i < md.ResourceMetrics().Len(); i++ {
+			rm := md.ResourceMetrics().At(i)
 			res := rm.Resource()
 			res.Attributes().PutStr("foo", "bar")
 			v, ok := res.Attributes().Get("foo")
@@ -1000,8 +998,7 @@ func BenchmarkMetricsUsage(b *testing.B) {
 }
 
 func BenchmarkMetricsMarshalJSON(b *testing.B) {
-	md := NewMetrics()
-	fillTestResourceMetricsSlice(md.ResourceMetrics())
+	md := generateTestMetrics()
 	encoder := &JSONMarshaler{}
 
 	b.ReportAllocs()
@@ -1012,3 +1009,9 @@ func BenchmarkMetricsMarshalJSON(b *testing.B) {
 		require.NotNil(b, jsonBuf)
 	}
 }
+
+func generateTestMetrics() Metrics {
+	md := NewMetrics()
+	md.getOrig().ResourceMetrics = internal.GenerateOrigTestResourceMetricsSlice()
+	return md
+}
