diff --git a/internal/cmd/pdatagen/internal/pdata/templates/slice.go.tmpl b/internal/cmd/pdatagen/internal/pdata/templates/slice.go.tmpl
index 75d85147b..ce32c110b 100644
--- a/internal/cmd/pdatagen/internal/pdata/templates/slice.go.tmpl
+++ b/internal/cmd/pdatagen/internal/pdata/templates/slice.go.tmpl
@@ -23,12 +23,12 @@ import (
 type {{ .structName }} internal.{{ .structName }}Wrapper
 {{- else }}
 type {{ .structName }} struct {
-	orig *[]{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }}
+	orig *[]{{ if .elementNullable }}*{{ end }}internal.{{ .elementOriginName }}
 	state *internal.State
 }
 {{- end }}
 
-func new{{ .structName }}(orig *[]{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }}, state *internal.State) {{ .structName }} {
+func new{{ .structName }}(orig *[]{{ if .elementNullable }}*{{ end }}internal.{{ .elementOriginName }}, state *internal.State) {{ .structName }} {
 	{{- if .hasWrapper }}
 	return {{ .structName }}(internal.New{{ .structName }}Wrapper(orig, state))
 	{{- else }}
@@ -39,7 +39,7 @@ func new{{ .structName }}(orig *[]{{ if .elementNullable }}*{{ end }}{{ .element
 // New{{ .structName }} creates a {{ .structName }}Wrapper with 0 elements.
 // Can use "EnsureCapacity" to initialize with a given capacity.
 func New{{ .structName }}() {{ .structName }} {
-	orig := []{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }}(nil)
+	orig := []{{ if .elementNullable }}*{{ end }}internal.{{ .elementOriginName }}(nil)
 	return new{{ .structName }}(&orig, internal.NewState())
 }
 
@@ -94,7 +94,7 @@ func (es {{ .structName }}) EnsureCapacity(newCap int) {
 		return
 	}
 
-	newOrig := make([]{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }}, len(*es.{{ .origAccessor }}), newCap)
+	newOrig := make([]{{ if .elementNullable }}*{{ end }}internal.{{ .elementOriginName }}, len(*es.{{ .origAccessor }}), newCap)
 	copy(newOrig, *es.{{ .origAccessor }})
 	*es.{{ .origAccessor }} = newOrig
 }
@@ -103,7 +103,7 @@ func (es {{ .structName }}) EnsureCapacity(newCap int) {
 // It returns the newly added {{ .elementName }}.
 func (es {{ .structName }}) AppendEmpty() {{ .elementName }} {
 	es.{{ .stateAccessor }}.AssertMutable()
-	*es.{{ .origAccessor }} = append(*es.{{ .origAccessor }}, {{- if .elementNullable }}internal.New{{ .elementOriginName }}(){{ else }}{{ .elementOriginFullName }}{}{{ end }})
+	*es.{{ .origAccessor }} = append(*es.{{ .origAccessor }}, {{- if .elementNullable }}internal.New{{ .elementOriginName }}(){{ else }}internal.{{ .elementOriginName }}{}{{ end }})
 	return es.At(es.Len() - 1)
 }
 
@@ -163,7 +163,7 @@ func (es {{ .structName }}) CopyTo(dest {{ .structName }}) {
 	if es.{{ .origAccessor }} == dest.{{ .origAccessor }} {
 		return
 	}
-	*dest.{{ .origAccessor }} = internal.Copy{{ .elementOriginName }}Slice(*dest.{{ .origAccessor }}, *es.{{ .origAccessor }})
+	*dest.{{ .origAccessor }} = internal.Copy{{ .elementOriginName }}{{ if .elementNullable }}Ptr{{ end }}Slice(*dest.{{ .origAccessor }}, *es.{{ .origAccessor }})
 }
 
 {{ if .elementNullable -}}
@@ -177,7 +177,7 @@ func (es {{ .structName }}) Sort(less func(a, b {{ .elementName }}) bool) {
 {{- end }}
 
 {{ if .hasWrapper -}}
-func (ms {{ .structName }}) getOrig() *[]{{ if .elementNullable }}*{{ end }}{{ .elementOriginFullName }} {
+func (ms {{ .structName }}) getOrig() *[]{{ if .elementNullable }}*{{ end }}internal.{{ .elementOriginName }} {
 	return internal.Get{{ .structName }}Orig(internal.{{ .structName }}Wrapper(ms))
 }
 
