diff --git a/pdata/internal/generated_wrapper_sample.go b/pdata/internal/generated_wrapper_sample.go
index 03a17cebe..512dc3197 100644
--- a/pdata/internal/generated_wrapper_sample.go
+++ b/pdata/internal/generated_wrapper_sample.go
@@ -40,13 +40,15 @@ func CopyOrigSample(dest, src *otlpprofiles.Sample) {
 	dest.TimestampsUnixNano = CopyOrigUint64Slice(dest.TimestampsUnixNano, src.TimestampsUnixNano)
 }
 
-func FillOrigTestSample(orig *otlpprofiles.Sample) {
+func GenTestOrigSample() *otlpprofiles.Sample {
+	orig := NewOrigPtrSample()
 	orig.LocationsStartIndex = int32(13)
 	orig.LocationsLength = int32(13)
 	orig.Value = GenerateOrigTestInt64Slice()
 	orig.AttributeIndices = GenerateOrigTestInt32Slice()
 	orig.LinkIndex_ = &otlpprofiles.Sample_LinkIndex{LinkIndex: int32(13)}
 	orig.TimestampsUnixNano = GenerateOrigTestUint64Slice()
+	return orig
 }
 
 // MarshalJSONOrig marshals all properties from the current struct to the destination stream.
