diff --git a/pdata/internal/generated_wrapper_instrumentationscope_test.go b/pdata/internal/generated_wrapper_instrumentationscope_test.go
index a4e6958dc..fb00c8026 100644
--- a/pdata/internal/generated_wrapper_instrumentationscope_test.go
+++ b/pdata/internal/generated_wrapper_instrumentationscope_test.go
@@ -23,7 +23,7 @@ func TestCopyOrigInstrumentationScope(t *testing.T) {
 	dest := NewOrigPtrInstrumentationScope()
 	CopyOrigInstrumentationScope(dest, src)
 	assert.Equal(t, NewOrigPtrInstrumentationScope(), dest)
-	FillOrigTestInstrumentationScope(src)
+	*src = *GenTestOrigInstrumentationScope()
 	CopyOrigInstrumentationScope(dest, src)
 	assert.Equal(t, src, dest)
 }
@@ -38,7 +38,7 @@ func TestMarshalAndUnmarshalJSONOrigInstrumentationScopeUnknown(t *testing.T) {
 }
 
 func TestMarshalAndUnmarshalJSONOrigInstrumentationScope(t *testing.T) {
-	for name, src := range getEncodingTestValuesInstrumentationScope() {
+	for name, src := range genTestEncodingValuesInstrumentationScope() {
 		t.Run(name, func(t *testing.T) {
 			stream := json.BorrowStream(nil)
 			defer json.ReturnStream(stream)
@@ -56,6 +56,15 @@ func TestMarshalAndUnmarshalJSONOrigInstrumentationScope(t *testing.T) {
 	}
 }
 
+func TestMarshalAndUnmarshalProtoOrigInstrumentationScopeFailing(t *testing.T) {
+	for name, buf := range genTestFailingUnmarshalProtoValuesInstrumentationScope() {
+		t.Run(name, func(t *testing.T) {
+			dest := NewOrigPtrInstrumentationScope()
+			require.Error(t, UnmarshalProtoOrigInstrumentationScope(dest, buf))
+		})
+	}
+}
+
 func TestMarshalAndUnmarshalProtoOrigInstrumentationScopeUnknown(t *testing.T) {
 	dest := NewOrigPtrInstrumentationScope()
 	// message Test { required int64 field = 1313; } encoding { "field": "1234" }
@@ -64,7 +73,7 @@ func TestMarshalAndUnmarshalProtoOrigInstrumentationScopeUnknown(t *testing.T) {
 }
 
 func TestMarshalAndUnmarshalProtoOrigInstrumentationScope(t *testing.T) {
-	for name, src := range getEncodingTestValuesInstrumentationScope() {
+	for name, src := range genTestEncodingValuesInstrumentationScope() {
 		t.Run(name, func(t *testing.T) {
 			buf := make([]byte, SizeProtoOrigInstrumentationScope(src))
 			gotSize := MarshalProtoOrigInstrumentationScope(src, buf)
@@ -78,7 +87,7 @@ func TestMarshalAndUnmarshalProtoOrigInstrumentationScope(t *testing.T) {
 }
 
 func TestMarshalAndUnmarshalProtoViaProtobufInstrumentationScope(t *testing.T) {
-	for name, src := range getEncodingTestValuesInstrumentationScope() {
+	for name, src := range genTestEncodingValuesInstrumentationScope() {
 		t.Run(name, func(t *testing.T) {
 			buf := make([]byte, SizeProtoOrigInstrumentationScope(src))
 			gotSize := MarshalProtoOrigInstrumentationScope(src, buf)
@@ -97,13 +106,26 @@ func TestMarshalAndUnmarshalProtoViaProtobufInstrumentationScope(t *testing.T) {
 	}
 }
 
-func getEncodingTestValuesInstrumentationScope() map[string]*otlpcommon.InstrumentationScope {
+func genTestFailingUnmarshalProtoValuesInstrumentationScope() map[string][]byte {
+	return map[string][]byte{
+		"invalid_field":                          {0x02},
+		"Name/wrong_wire_type":                   {0xc},
+		"Name/missing_value":                     {0xa},
+		"Version/wrong_wire_type":                {0x14},
+		"Version/missing_value":                  {0x12},
+		"Attributes/wrong_wire_type":             {0x1c},
+		"Attributes/missing_value":               {0x1a},
+		"DroppedAttributesCount/wrong_wire_type": {0x24},
+		"DroppedAttributesCount/missing_value":   {0x20},
+	}
+}
+
+func genTestEncodingValuesInstrumentationScope() map[string]*otlpcommon.InstrumentationScope {
 	return map[string]*otlpcommon.InstrumentationScope{
-		"empty": NewOrigPtrInstrumentationScope(),
-		"fill_test": func() *otlpcommon.InstrumentationScope {
-			src := NewOrigPtrInstrumentationScope()
-			FillOrigTestInstrumentationScope(src)
-			return src
-		}(),
+		"empty":                       NewOrigPtrInstrumentationScope(),
+		"Name/test":                   {Name: "test_name"},
+		"Version/test":                {Version: "test_version"},
+		"Attributes/default_and_test": {Attributes: []otlpcommon.KeyValue{{}, *GenTestOrigKeyValue()}},
+		"DroppedAttributesCount/test": {DroppedAttributesCount: uint32(13)},
 	}
 }
