diff --git a/internal/data/common_test.go b/internal/data/common_test.go
index c6adb0a32..1b4a7bc05 100644
--- a/internal/data/common_test.go
+++ b/internal/data/common_test.go
@@ -112,28 +112,29 @@ func TestAttributeKeyValue(t *testing.T) {
 }
 
 func TestNilStringMap(t *testing.T) {
-	val, exist := NewStringMap(nil).Get("test_key")
+	val, exist := NewStringMap().Get("test_key")
 	assert.EqualValues(t, false, exist)
 	assert.EqualValues(t, StringKeyValue{nil}, val)
-	insertMap := NewStringMap(nil)
-	insertMap.Insert("key", "value")
-	assert.EqualValues(t, NewStringMap(map[string]string{"key": "value"}), insertMap)
-	updateMap := NewStringMap(nil)
-	updateMap.Update("key", "value")
-	assert.EqualValues(t, NewStringMap(nil), updateMap)
-	upsertMap := NewStringMap(nil)
-	upsertMap.Upsert("key", "value")
-	assert.EqualValues(t, NewStringMap(map[string]string{"key": "value"}), upsertMap)
-	deleteMap := NewStringMap(nil)
-	assert.EqualValues(t, false, deleteMap.Delete("key"))
-	assert.EqualValues(t, NewStringMap(nil), deleteMap)
-	assert.EqualValues(t, 0, NewStringMap(nil).Len())
-	assert.EqualValues(t, NewStringMap(nil), NewStringMap(nil).Sort())
+	insertMap := NewStringMap()
+	insertMap.Insert("k", "v")
+	assert.EqualValues(t, generateTestStringMap(), insertMap)
+	updateMap := NewStringMap()
+	updateMap.Update("k", "v")
+	assert.EqualValues(t, NewStringMap(), updateMap)
+	upsertMap := NewStringMap()
+	upsertMap.Upsert("k", "v")
+	assert.EqualValues(t, generateTestStringMap(), upsertMap)
+	deleteMap := NewStringMap()
+	assert.EqualValues(t, false, deleteMap.Delete("k"))
+	assert.EqualValues(t, NewStringMap(), deleteMap)
+	assert.EqualValues(t, 0, NewStringMap().Len())
+	assert.EqualValues(t, NewStringMap(), NewStringMap().Sort())
 }
 
 func TestStringMap(t *testing.T) {
-	origMap := map[string]string{"k0": "v0", "k1": "v1", "k2": "v2"}
-	sm := NewStringMap(origMap)
+	origRawMap := map[string]string{"k0": "v0", "k1": "v1", "k2": "v2"}
+	origMap := NewStringMap().InitFromMap(origRawMap)
+	sm := NewStringMap().InitFromMap(origRawMap)
 	assert.EqualValues(t, 3, sm.Len())
 
 	val, exist := sm.Get("k2")
@@ -145,53 +146,53 @@ func TestStringMap(t *testing.T) {
 	assert.EqualValues(t, StringKeyValue{nil}, val)
 
 	sm.Insert("k1", "v1")
-	assert.EqualValues(t, NewStringMap(origMap).Sort(), sm.Sort())
+	assert.EqualValues(t, origMap.Sort(), sm.Sort())
 	sm.Insert("k3", "v3")
 	assert.EqualValues(t, 4, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
 	assert.EqualValues(t, true, sm.Delete("k3"))
 	assert.EqualValues(t, 3, sm.Len())
-	assert.EqualValues(t, NewStringMap(origMap).Sort(), sm.Sort())
+	assert.EqualValues(t, origMap.Sort(), sm.Sort())
 
 	sm.Update("k3", "v3")
 	assert.EqualValues(t, 3, sm.Len())
-	assert.EqualValues(t, NewStringMap(origMap).Sort(), sm.Sort())
+	assert.EqualValues(t, origMap.Sort(), sm.Sort())
 	sm.Update("k2", "v3")
 	assert.EqualValues(t, 3, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v3"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v3"}).Sort(), sm.Sort())
 	sm.Update("k2", "v2")
 	assert.EqualValues(t, 3, sm.Len())
-	assert.EqualValues(t, NewStringMap(origMap).Sort(), sm.Sort())
+	assert.EqualValues(t, origMap.Sort(), sm.Sort())
 
 	sm.Upsert("k3", "v3")
 	assert.EqualValues(t, 4, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
 	sm.Upsert("k1", "v5")
 	assert.EqualValues(t, 4, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k0": "v0", "k1": "v5", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k0": "v0", "k1": "v5", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
 	sm.Upsert("k1", "v1")
 	assert.EqualValues(t, 4, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2", "k3": "v3"}).Sort(), sm.Sort())
 	assert.EqualValues(t, true, sm.Delete("k3"))
 	assert.EqualValues(t, 3, sm.Len())
-	assert.EqualValues(t, NewStringMap(origMap).Sort(), sm.Sort())
+	assert.EqualValues(t, origMap.Sort(), sm.Sort())
 
 	assert.EqualValues(t, false, sm.Delete("k3"))
 	assert.EqualValues(t, 3, sm.Len())
-	assert.EqualValues(t, NewStringMap(origMap).Sort(), sm.Sort())
+	assert.EqualValues(t, origMap.Sort(), sm.Sort())
 
 	assert.EqualValues(t, true, sm.Delete("k0"))
 	assert.EqualValues(t, 2, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k1": "v1", "k2": "v2"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k1": "v1", "k2": "v2"}).Sort(), sm.Sort())
 	assert.EqualValues(t, true, sm.Delete("k2"))
 	assert.EqualValues(t, 1, sm.Len())
-	assert.EqualValues(t, NewStringMap(map[string]string{"k1": "v1"}).Sort(), sm.Sort())
+	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"k1": "v1"}).Sort(), sm.Sort())
 	assert.EqualValues(t, true, sm.Delete("k1"))
 	assert.EqualValues(t, 0, sm.Len())
 }
 
 func TestStringMapIteration(t *testing.T) {
-	sm := NewStringMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2"})
+	sm := NewStringMap().InitFromMap(map[string]string{"k0": "v0", "k1": "v1", "k2": "v2"})
 	assert.EqualValues(t, 3, sm.Len())
 	sm.Sort()
 	for i := 0; i < sm.Len(); i++ {
@@ -202,13 +203,25 @@ func TestStringMapIteration(t *testing.T) {
 }
 
 func generateTestStringMap() StringMap {
-	return NewStringMap(map[string]string{
+	sm := NewStringMap()
+	fillTestStringMap(sm)
+	return sm
+}
+
+func fillTestStringMap(dest StringMap) {
+	dest.InitFromMap(map[string]string{
 		"k": "v",
 	})
 }
 
 func generateTestAttributeMap() AttributeMap {
-	return NewAttributeMap(map[string]AttributeValue{
+	am := NewAttributeMap()
+	fillTestAttributeMap(am)
+	return am
+}
+
+func fillTestAttributeMap(dest AttributeMap) {
+	dest.InitFromMap(map[string]AttributeValue{
 		"k": NewAttributeValueString("v"),
 	})
 }
