diff --git a/exporter/exportertest/nop_exporter.go b/exporter/exportertest/nop_exporter.go
index 7894271d4..6d6435ac4 100644
--- a/exporter/exportertest/nop_exporter.go
+++ b/exporter/exportertest/nop_exporter.go
@@ -17,7 +17,7 @@ import (
 
 var nopType = component.MustNewType("nop")
 
-// NewNopSettings returns a new nop settings for Create*Exporter functions.
+// NewNopSettings returns a new nop settings for Create* functions.
 func NewNopSettings() exporter.Settings {
 	return exporter.Settings{
 		ID:                component.NewIDWithName(nopType, uuid.NewString()),
@@ -31,37 +31,37 @@ func NewNopFactory() exporter.Factory {
 	return exporterprofiles.NewFactory(
 		nopType,
 		func() component.Config { return &nopConfig{} },
-		exporterprofiles.WithTraces(createTracesExporter, component.StabilityLevelStable),
-		exporterprofiles.WithMetrics(createMetricsExporter, component.StabilityLevelStable),
-		exporterprofiles.WithLogs(createLogsExporter, component.StabilityLevelStable),
-		exporterprofiles.WithProfiles(createProfilesExporter, component.StabilityLevelAlpha),
+		exporterprofiles.WithTraces(createTraces, component.StabilityLevelStable),
+		exporterprofiles.WithMetrics(createMetrics, component.StabilityLevelStable),
+		exporterprofiles.WithLogs(createLogs, component.StabilityLevelStable),
+		exporterprofiles.WithProfiles(createProfiles, component.StabilityLevelAlpha),
 	)
 }
 
-func createTracesExporter(context.Context, exporter.Settings, component.Config) (exporter.Traces, error) {
+func createTraces(context.Context, exporter.Settings, component.Config) (exporter.Traces, error) {
 	return nopInstance, nil
 }
 
-func createMetricsExporter(context.Context, exporter.Settings, component.Config) (exporter.Metrics, error) {
+func createMetrics(context.Context, exporter.Settings, component.Config) (exporter.Metrics, error) {
 	return nopInstance, nil
 }
 
-func createLogsExporter(context.Context, exporter.Settings, component.Config) (exporter.Logs, error) {
+func createLogs(context.Context, exporter.Settings, component.Config) (exporter.Logs, error) {
 	return nopInstance, nil
 }
 
-func createProfilesExporter(context.Context, exporter.Settings, component.Config) (exporterprofiles.Profiles, error) {
+func createProfiles(context.Context, exporter.Settings, component.Config) (exporterprofiles.Profiles, error) {
 	return nopInstance, nil
 }
 
 type nopConfig struct{}
 
-var nopInstance = &nopExporter{
+var nopInstance = &nop{
 	Consumer: consumertest.NewNop(),
 }
 
-// nopExporter stores consumed traces, metrics, logs and profiles for testing purposes.
-type nopExporter struct {
+// nop stores consumed traces, metrics, logs and profiles for testing purposes.
+type nop struct {
 	component.StartFunc
 	component.ShutdownFunc
 	consumertest.Consumer
