diff --git a/pdata/internal/generated_wrapper_uint64slice.go b/pdata/internal/generated_wrapper_uint64slice.go
index 2f17e07d5..954987d85 100644
--- a/pdata/internal/generated_wrapper_uint64slice.go
+++ b/pdata/internal/generated_wrapper_uint64slice.go
@@ -6,10 +6,6 @@
 
 package internal
 
-import (
-	"go.opentelemetry.io/collector/pdata/internal/json"
-)
-
 type UInt64Slice struct {
 	orig  *[]uint64
 	state *State
@@ -39,13 +35,3 @@ func CopyOrigUint64Slice(dst, src []uint64) []uint64 {
 func GenerateOrigTestUint64Slice() []uint64 {
 	return []uint64{1, 2, 3}
 }
-
-// UnmarshalJSONOrigUint64Slice unmarshals all properties from the current struct from the source iterator.
-func UnmarshalJSONOrigUint64Slice(iter *json.Iterator) []uint64 {
-	var orig []uint64
-	iter.ReadArrayCB(func(iter *json.Iterator) bool {
-		orig = append(orig, iter.ReadUint64())
-		return true
-	})
-	return orig
-}
