diff --git a/pdata/internal/generated_wrapper_exportlogsservicerequest.go b/pdata/internal/generated_wrapper_exportlogsservicerequest.go
index 36ea036aa..d160321ec 100644
--- a/pdata/internal/generated_wrapper_exportlogsservicerequest.go
+++ b/pdata/internal/generated_wrapper_exportlogsservicerequest.go
@@ -63,15 +63,18 @@ func MarshalJSONOrigExportLogsServiceRequest(orig *otlpcollectorlogs.ExportLogsS
 
 // UnmarshalJSONOrigLogs unmarshals all properties from the current struct from the source iterator.
 func UnmarshalJSONOrigExportLogsServiceRequest(orig *otlpcollectorlogs.ExportLogsServiceRequest, iter *json.Iterator) {
-	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
+	for f := iter.ReadObject(); f != ""; f = iter.ReadObject() {
 		switch f {
 		case "resourceLogs", "resource_logs":
-			orig.ResourceLogs = UnmarshalJSONOrigResourceLogsSlice(iter)
+			for iter.ReadArray() {
+				orig.ResourceLogs = append(orig.ResourceLogs, NewOrigResourceLogs())
+				UnmarshalJSONOrigResourceLogs(orig.ResourceLogs[len(orig.ResourceLogs)-1], iter)
+			}
+
 		default:
 			iter.Skip()
 		}
-		return true
-	})
+	}
 }
 
 func SizeProtoOrigExportLogsServiceRequest(orig *otlpcollectorlogs.ExportLogsServiceRequest) int {
