diff --git a/internal/processor/filterspan/filterspan.go b/internal/processor/filterspan/filterspan.go
index c7a5e34df..1e347bfba 100644
--- a/internal/processor/filterspan/filterspan.go
+++ b/internal/processor/filterspan/filterspan.go
@@ -19,17 +19,11 @@ import (
 	"fmt"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/processor/filterconfig"
 	"go.opentelemetry.io/collector/internal/processor/filterhelper"
 	"go.opentelemetry.io/collector/internal/processor/filterset"
 )
 
-var (
-	// TODO Add processor type invoking the NewMatcher in error text.
-	errAtLeastOneMatchFieldNeeded = errors.New(
-		`error creating processor. At least one ` +
-			`of "services", "span_names" or "attributes" field must be specified"`)
-)
-
 // TODO: Modify Matcher to invoke both the include and exclude properties so
 // calling processors will always have the same logic.
 // Matcher is an interface that allows matching a span against a configuration
@@ -60,13 +54,13 @@ type attributeMatcher struct {
 }
 
 // NewMatcher creates a span Matcher that matches based on the given MatchProperties.
-func NewMatcher(mp *MatchProperties) (Matcher, error) {
+func NewMatcher(mp *filterconfig.MatchProperties) (Matcher, error) {
 	if mp == nil {
 		return nil, nil
 	}
 
-	if len(mp.Services) == 0 && len(mp.SpanNames) == 0 && len(mp.Attributes) == 0 {
-		return nil, errAtLeastOneMatchFieldNeeded
+	if err := mp.ValidateForSpans(); err != nil {
+		return nil, err
 	}
 
 	var err error
@@ -102,12 +96,12 @@ func NewMatcher(mp *MatchProperties) (Matcher, error) {
 	}, nil
 }
 
-func newAttributesMatcher(mp *MatchProperties) (attributesMatcher, error) {
+func newAttributesMatcher(mp *filterconfig.MatchProperties) (attributesMatcher, error) {
 	// attribute matching is only supported with strict matching
 	if mp.Config.MatchType != filterset.Strict {
 		return nil, fmt.Errorf(
 			"%s=%s is not supported for %q",
-			filterset.MatchTypeFieldName, filterset.Regexp, AttributesFieldName,
+			filterset.MatchTypeFieldName, filterset.Regexp, filterconfig.AttributesFieldName,
 		)
 	}
 
