diff --git a/otelcol/collector.go b/otelcol/collector.go
index 6226d0768..f300dd502 100644
--- a/otelcol/collector.go
+++ b/otelcol/collector.go
@@ -26,7 +26,6 @@ import (
 	"go.opentelemetry.io/collector/extension"
 	"go.opentelemetry.io/collector/otelcol/internal/grpclog"
 	"go.opentelemetry.io/collector/processor"
-	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/service"
 )
 
@@ -187,11 +186,14 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	col.service, err = service.New(ctx, service.Settings{
 		BuildInfo:     col.set.BuildInfo,
 		CollectorConf: conf,
-		Receivers:     receiver.NewBuilder(cfg.Receivers, factories.Receivers),
-		Processors:    processor.NewBuilder(cfg.Processors, factories.Processors),
-		Exporters:     exporter.NewBuilder(cfg.Exporters, factories.Exporters),
-		Connectors:    connector.NewBuilder(cfg.Connectors, factories.Connectors),
-		Extensions:    extension.NewBuilder(cfg.Extensions, factories.Extensions),
+
+		ReceiversConfigs:   cfg.Receivers,
+		ReceiversFactories: factories.Receivers,
+
+		Processors: processor.NewBuilder(cfg.Processors, factories.Processors),
+		Exporters:  exporter.NewBuilder(cfg.Exporters, factories.Exporters),
+		Connectors: connector.NewBuilder(cfg.Connectors, factories.Connectors),
+		Extensions: extension.NewBuilder(cfg.Extensions, factories.Extensions),
 		ModuleInfo: extension.ModuleInfo{
 			Receiver:  factories.ReceiverModules,
 			Processor: factories.ProcessorModules,
