diff --git a/extension/healthcheckextension/config.go b/extension/healthcheckextension/config.go
index 1137eb874..f6fb1e931 100644
--- a/extension/healthcheckextension/config.go
+++ b/extension/healthcheckextension/config.go
@@ -22,7 +22,7 @@ import (
 // Config has the configuration for the extension enabling the health check
 // extension, used to report the health status of the service.
 type Config struct {
-	*config.ExtensionSettings `mapstructure:"-"`
+	config.ExtensionSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
 	// Port is the port used to publish the health check status.
 	// The default value is 13133.
