diff --git a/service/service.go b/service/service.go
index 477ed2266..386dddd20 100644
--- a/service/service.go
+++ b/service/service.go
@@ -139,6 +139,11 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 }
 
 // Start starts the extensions and pipelines. If Start fails Shutdown should be called to ensure a clean state.
+// Start does the following steps in order:
+// 1. Start all extensions.
+// 2. Notify extensions about Collector configuration
+// 3. Start all pipelines.
+// 4. Notify extensions that the pipeline is ready.
 func (srv *Service) Start(ctx context.Context) error {
 	srv.telemetrySettings.Logger.Info("Starting "+srv.buildInfo.Command+"...",
 		zap.String("Version", srv.buildInfo.Version),
@@ -170,6 +175,11 @@ func (srv *Service) Start(ctx context.Context) error {
 	return nil
 }
 
+// Shutdown the service. Shutdown will do the following steps in order:
+// 1. Notify extensions that the pipeline is shutting down.
+// 2. Shutdown all pipelines.
+// 3. Shutdown all extensions.
+// 4. Shutdown telemetry.
 func (srv *Service) Shutdown(ctx context.Context) error {
 	// Accumulate errors and proceed with shutting down remaining components.
 	var errs error
