diff --git a/exporter/exportertest/nop_exporter_test.go b/exporter/exportertest/nop_exporter_test.go
index ed7a36bc0..6ae95efc0 100644
--- a/exporter/exportertest/nop_exporter_test.go
+++ b/exporter/exportertest/nop_exporter_test.go
@@ -20,7 +20,7 @@ import (
 func TestNewNopFactory(t *testing.T) {
 	factory := NewNopFactory()
 	require.NotNil(t, factory)
-	assert.Equal(t, component.Type("nop"), factory.Type())
+	assert.Equal(t, component.MustNewType("nop"), factory.Type())
 	cfg := factory.CreateDefaultConfig()
 	assert.Equal(t, &nopConfig{}, cfg)
 
@@ -50,7 +50,7 @@ func TestNewNopBuilder(t *testing.T) {
 	factory := NewNopFactory()
 	cfg := factory.CreateDefaultConfig()
 	set := NewNopCreateSettings()
-	set.ID = component.NewID(typeStr)
+	set.ID = component.NewID(nopType)
 
 	traces, err := factory.CreateTracesExporter(context.Background(), set, cfg)
 	require.NoError(t, err)
