diff --git a/exporter/otlpexporter/go.mod b/exporter/otlpexporter/go.mod
index 481d05f58..c42d17551 100644
--- a/exporter/otlpexporter/go.mod
+++ b/exporter/otlpexporter/go.mod
@@ -10,6 +10,7 @@ require (
 	go.opentelemetry.io/collector/config/configcompression v0.91.0
 	go.opentelemetry.io/collector/config/configgrpc v0.91.0
 	go.opentelemetry.io/collector/config/configopaque v0.91.0
+	go.opentelemetry.io/collector/config/configretry v0.91.0
 	go.opentelemetry.io/collector/config/configtls v0.91.0
 	go.opentelemetry.io/collector/confmap v0.91.0
 	go.opentelemetry.io/collector/consumer v0.91.0
@@ -119,3 +120,5 @@ replace go.opentelemetry.io/collector/config/confighttp => ../../config/confight
 replace go.opentelemetry.io/collector/receiver/otlpreceiver => ../../receiver/otlpreceiver
 
 replace go.opentelemetry.io/collector/semconv => ../../semconv
+
+replace go.opentelemetry.io/collector/config/configretry => ../../config/configretry
