diff --git a/receiver/prometheusreceiver/config_test.go b/receiver/prometheusreceiver/config_test.go
index cf24cddb8..862ac1eac 100644
--- a/receiver/prometheusreceiver/config_test.go
+++ b/receiver/prometheusreceiver/config_test.go
@@ -22,9 +22,9 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/open-telemetry/opentelemetry-service/internal/configmodels"
-	"github.com/open-telemetry/opentelemetry-service/internal/configv2"
-	"github.com/open-telemetry/opentelemetry-service/internal/factories"
+	"github.com/open-telemetry/opentelemetry-service/pkg/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/pkg/configv2"
+	"github.com/open-telemetry/opentelemetry-service/pkg/factories"
 )
 
 var _ = configv2.RegisterTestFactories()
