diff --git a/component/componentstatus/go.mod b/component/componentstatus/go.mod
index d94436778..eef21bac2 100644
--- a/component/componentstatus/go.mod
+++ b/component/componentstatus/go.mod
@@ -32,3 +32,5 @@ replace go.opentelemetry.io/collector/pdata => ../../pdata
 replace go.opentelemetry.io/collector/pipeline => ../../pipeline
 
 replace go.opentelemetry.io/collector/featuregate => ../../featuregate
+
+replace go.opentelemetry.io/collector/internal/testutil => ../../internal/testutil
