diff --git a/extension/memorylimiterextension/go.mod b/extension/memorylimiterextension/go.mod
index 0ceb098a0..6d38d5f3b 100644
--- a/extension/memorylimiterextension/go.mod
+++ b/extension/memorylimiterextension/go.mod
@@ -3,7 +3,7 @@ module go.opentelemetry.io/collector/extension/memorylimiterextension
 go 1.22.0
 
 require (
-	github.com/stretchr/testify v1.9.0
+	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v0.114.0
 	go.opentelemetry.io/collector/component/componenttest v0.114.0
 	go.opentelemetry.io/collector/confmap v1.20.0
