diff --git a/go.mod b/go.mod
index ac692cf81..40cbd8517 100644
--- a/go.mod
+++ b/go.mod
@@ -12,8 +12,8 @@ require (
 	github.com/prometheus/client_golang v1.14.0
 	github.com/prometheus/client_model v0.3.0
 	github.com/prometheus/common v0.39.0
-	github.com/rs/cors v1.8.2
-	github.com/shirou/gopsutil/v3 v3.22.10
+	github.com/rs/cors v1.8.3
+	github.com/shirou/gopsutil/v3 v3.22.12
 	github.com/spf13/cobra v1.6.1
 	github.com/stretchr/testify v1.8.1
 	go.opencensus.io v0.24.0
