diff --git a/pdata/internal/generated_wrapper_entityref_test.go b/pdata/internal/generated_wrapper_entityref_test.go
index 92d643da7..56acc0689 100644
--- a/pdata/internal/generated_wrapper_entityref_test.go
+++ b/pdata/internal/generated_wrapper_entityref_test.go
@@ -50,9 +50,9 @@ func TestMarshalAndUnmarshalJSONOrigEntityRef(t *testing.T) {
 func TestMarshalAndUnmarshalProtoOrigEntityRef(t *testing.T) {
 	src := &otlpcommon.EntityRef{}
 	FillOrigTestEntityRef(src)
-	buf, err := MarshalProtoOrigEntityRef(src)
-	require.NoError(t, err)
-	assert.Equal(t, len(buf), SizeProtoOrigEntityRef(src))
+	buf := make([]byte, SizeProtoOrigEntityRef(src))
+	gotSize := MarshalProtoOrigEntityRef(src, buf)
+	assert.Equal(t, len(buf), gotSize)
 
 	dest := &otlpcommon.EntityRef{}
 	require.NoError(t, UnmarshalProtoOrigEntityRef(dest, buf))
@@ -61,9 +61,9 @@ func TestMarshalAndUnmarshalProtoOrigEntityRef(t *testing.T) {
 
 func TestMarshalAndUnmarshalProtoOrigEmptyEntityRef(t *testing.T) {
 	src := &otlpcommon.EntityRef{}
-	buf, err := MarshalProtoOrigEntityRef(src)
-	require.NoError(t, err)
-	assert.Equal(t, len(buf), SizeProtoOrigEntityRef(src))
+	buf := make([]byte, SizeProtoOrigEntityRef(src))
+	gotSize := MarshalProtoOrigEntityRef(src, buf)
+	assert.Equal(t, len(buf), gotSize)
 
 	dest := &otlpcommon.EntityRef{}
 	require.NoError(t, UnmarshalProtoOrigEntityRef(dest, buf))
