diff --git a/cmd/pdatagen/internal/base_fields.go b/cmd/pdatagen/internal/base_fields.go
index 66c0ccde9..b208731a7 100644
--- a/cmd/pdatagen/internal/base_fields.go
+++ b/cmd/pdatagen/internal/base_fields.go
@@ -53,6 +53,18 @@ func (ms ${structName}) Set${fieldName}(v ${returnType}) {
 	(*ms.orig).${originFieldName} = v
 }`
 
+const accessorsOneofPrimitiveTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) ${fieldName}() ${returnType} {
+	return (*ms.orig).GetAs${fieldType}()
+}
+
+// Set${fieldName} replaces the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) Set${fieldName}(v ${returnType}) {
+	(*ms.orig).${originFieldName} = &${originFullName}_As${fieldType}{
+		As${fieldType}: v,
+	}
+}`
+
 const accessorsPrimitiveAsDoubleTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
 func (ms ${structName}) ${fieldName}() ${returnType} {
 	return (*ms.orig).GetAsDouble()
@@ -462,3 +474,101 @@ func (pf *primitiveAsDoubleField) generateCopyToValue(sb *strings.Builder) {
 }
 
 var _ baseField = (*primitiveField)(nil)
+
+type oneOfPrimitiveValue struct {
+	name            string
+	defaultVal      string
+	testVal         string
+	returnType      string
+	originFieldName string
+	originFullName  string
+	fieldType       string
+}
+
+func (opv *oneOfPrimitiveValue) generateAccessors(ms baseStruct, sb *strings.Builder) {
+	sb.WriteString(os.Expand(accessorsOneofPrimitiveTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.getName()
+		case "fieldName":
+			return opv.name
+		case "lowerFieldName":
+			return strings.ToLower(opv.name)
+		case "returnType":
+			return opv.returnType
+		case "originFieldName":
+			return opv.originFieldName
+		case "originFullName":
+			return opv.originFullName
+		case "fieldType":
+			return opv.fieldType
+		default:
+			panic(name)
+		}
+	}))
+	sb.WriteString("\n")
+}
+
+func (opv *oneOfPrimitiveValue) generateAccessorsTest(ms baseStruct, sb *strings.Builder) {
+	sb.WriteString(os.Expand(accessorsPrimitiveTestTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.getName()
+		case "defaultVal":
+			return opv.defaultVal
+		case "fieldName":
+			return opv.name
+		case "testValue":
+			return opv.testVal
+		default:
+			panic(name)
+		}
+	}))
+	sb.WriteString("\n")
+}
+
+func (opv *oneOfPrimitiveValue) generateSetWithTestValue(sb *strings.Builder) {
+	sb.WriteString("\t tv.Set" + opv.name + "(" + opv.testVal + ")\n")
+}
+
+func (opv *oneOfPrimitiveValue) generateCopyToValue(sb *strings.Builder) {
+	sb.WriteString("\tcase NumberDataPointType" + opv.fieldType + ":\n")
+	sb.WriteString("\t dest.Set" + opv.name + "(ms." + opv.name + "())\n")
+}
+
+var _ baseField = (*oneOfPrimitiveValue)(nil)
+
+type numberField struct {
+	fields []*oneOfPrimitiveValue
+}
+
+func (nf *numberField) generateAccessors(ms baseStruct, sb *strings.Builder) {
+	for _, field := range nf.fields {
+		field.generateAccessors(ms, sb)
+	}
+}
+
+func (nf *numberField) generateAccessorsTest(ms baseStruct, sb *strings.Builder) {
+	for _, field := range nf.fields {
+		field.generateAccessorsTest(ms, sb)
+	}
+}
+
+func (nf *numberField) generateSetWithTestValue(sb *strings.Builder) {
+	for _, field := range nf.fields {
+		field.generateSetWithTestValue(sb)
+		// TODO: this test should be generated for all number values,
+		//       for now, it's ok to only set one value
+		return
+	}
+}
+
+func (nf *numberField) generateCopyToValue(sb *strings.Builder) {
+	sb.WriteString("\tswitch ms.Type() {\n")
+	for _, field := range nf.fields {
+		field.generateCopyToValue(sb)
+	}
+	sb.WriteString("\t}\n")
+}
+
+var _ baseField = (*numberField)(nil)
