diff --git a/service/service.go b/service/service.go
index 3a681cad3..ea76a1449 100644
--- a/service/service.go
+++ b/service/service.go
@@ -21,27 +21,31 @@ import (
 	"go.opentelemetry.io/otel/metric/nonrecording"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.uber.org/multierr"
+	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/service/internal"
 	"go.opentelemetry.io/collector/service/internal/extensions"
 	"go.opentelemetry.io/collector/service/internal/pipelines"
+	"go.opentelemetry.io/collector/service/internal/telemetry"
 	"go.opentelemetry.io/collector/service/internal/telemetrylogs"
 )
 
 // service represents the implementation of a component.Host.
 type service struct {
-	buildInfo component.BuildInfo
-	config    *Config
-	telemetry component.TelemetrySettings
-	host      *serviceHost
+	buildInfo            component.BuildInfo
+	config               *Config
+	telemetrySettings    component.TelemetrySettings
+	host                 *serviceHost
+	telemetryInitializer *telemetryInitializer
 }
 
 func newService(set *settings) (*service, error) {
 	srv := &service{
 		buildInfo: set.BuildInfo,
 		config:    set.Config,
-		telemetry: component.TelemetrySettings{
+		telemetrySettings: component.TelemetrySettings{
+			Logger: zap.NewNop(),
 			TracerProvider: sdktrace.NewTracerProvider(
 				// needed for supporting the zpages extension
 				sdktrace.WithSampler(internal.AlwaysRecord()),
@@ -54,26 +58,37 @@ func newService(set *settings) (*service, error) {
 			buildInfo:         set.BuildInfo,
 			asyncErrorChannel: set.AsyncErrorChannel,
 		},
+		telemetryInitializer: set.telemetry,
 	}
 
 	var err error
-	if srv.telemetry.Logger, err = telemetrylogs.NewLogger(set.Config.Service.Telemetry.Logs, set.LoggingOptions); err != nil {
+	if srv.telemetrySettings.Logger, err = telemetrylogs.NewLogger(set.Config.Service.Telemetry.Logs, set.LoggingOptions); err != nil {
 		return nil, fmt.Errorf("failed to get logger: %w", err)
 	}
 
+	if err = srv.telemetryInitializer.init(set.BuildInfo, srv.telemetrySettings.Logger, set.Config.Service.Telemetry, set.AsyncErrorChannel); err != nil {
+		return nil, fmt.Errorf("failed to initialize telemetry: %w", err)
+	}
+	srv.telemetrySettings.MeterProvider = srv.telemetryInitializer.mp
+
 	extensionsSettings := extensions.Settings{
-		Telemetry:         srv.telemetry,
+		Telemetry:         srv.telemetrySettings,
 		BuildInfo:         srv.buildInfo,
 		Configs:           srv.config.Extensions,
 		Factories:         srv.host.factories.Extensions,
 		ServiceExtensions: srv.config.Service.Extensions,
 	}
 	if srv.host.extensions, err = extensions.Build(context.Background(), extensionsSettings); err != nil {
-		return nil, fmt.Errorf("cannot build extensions: %w", err)
+		return nil, fmt.Errorf("failed build extensions: %w", err)
+	}
+
+	if srv.host.pipelines, err = pipelines.Build(context.Background(), srv.telemetrySettings, srv.buildInfo, srv.config, srv.host.factories); err != nil {
+		return nil, fmt.Errorf("failed build pipelines: %w", err)
 	}
 
-	if srv.host.pipelines, err = pipelines.Build(context.Background(), srv.telemetry, srv.buildInfo, srv.config, srv.host.factories); err != nil {
-		return nil, fmt.Errorf("cannot build pipelines: %w", err)
+	// The process telemetry initialization requires the ballast size, which is available after the extensions are initialized.
+	if err = telemetry.RegisterProcessMetrics(srv.telemetryInitializer.ocRegistry, getBallastSize(srv.host)); err != nil {
+		return nil, fmt.Errorf("failed to register process metrics: %w", err)
 	}
 
 	return srv, nil
