diff --git a/service/telemetry.go b/service/telemetry.go
index 402dba4cb..a25199d60 100644
--- a/service/telemetry.go
+++ b/service/telemetry.go
@@ -16,7 +16,6 @@ package service // import "go.opentelemetry.io/collector/service"
 
 import (
 	"errors"
-	"fmt"
 	"net/http"
 	"strings"
 	"sync"
@@ -24,10 +23,12 @@ import (
 
 	"contrib.go.opencensus.io/exporter/prometheus"
 	"github.com/google/uuid"
-	"go.opencensus.io/metric"
+	ocmetric "go.opencensus.io/metric"
 	"go.opencensus.io/metric/metricproducer"
 	"go.opencensus.io/stats/view"
 	otelprometheus "go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/nonrecording"
 	"go.opentelemetry.io/otel/sdk/metric/aggregator/histogram"
 	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
 	"go.opentelemetry.io/otel/sdk/metric/export/aggregation"
@@ -35,147 +36,134 @@ import (
 	selector "go.opentelemetry.io/otel/sdk/metric/selector/simple"
 	"go.uber.org/zap"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configtelemetry"
 	"go.opentelemetry.io/collector/internal/obsreportconfig"
 	"go.opentelemetry.io/collector/processor/batchprocessor"
 	semconv "go.opentelemetry.io/collector/semconv/v1.5.0"
 	"go.opentelemetry.io/collector/service/featuregate"
-	"go.opentelemetry.io/collector/service/internal/telemetry"
 )
 
-// collectorTelemetry is collector's own telemetry.
-var collectorTelemetry collectorTelemetryExporter = newColTelemetry(featuregate.GetRegistry())
+// collectorTelemetry is collector's own telemetrySettings.
+var collectorTelemetry = newColTelemetry(featuregate.GetRegistry())
 
 const (
 	zapKeyTelemetryAddress = "address"
 	zapKeyTelemetryLevel   = "level"
 
 	// useOtelForInternalMetricsfeatureGateID is the feature gate ID that controls whether the collector uses open
-	// telemetry for internal metrics.
+	// telemetrySettings for internal metrics.
 	useOtelForInternalMetricsfeatureGateID = "telemetry.useOtelForInternalMetrics"
 )
 
-type collectorTelemetryExporter interface {
-	init(svc *service) error
-	shutdown() error
-}
-
-type colTelemetry struct {
+type telemetryInitializer struct {
 	registry *featuregate.Registry
 	views    []*view.View
 
-	ocRegistry *metric.Registry
+	ocRegistry *ocmetric.Registry
+
+	mp metric.MeterProvider
 
 	server     *http.Server
 	doInitOnce sync.Once
 }
 
-func newColTelemetry(registry *featuregate.Registry) *colTelemetry {
+func newColTelemetry(registry *featuregate.Registry) *telemetryInitializer {
 	registry.MustRegister(featuregate.Gate{
 		ID:          useOtelForInternalMetricsfeatureGateID,
-		Description: "controls whether the collector to uses open telemetry for internal metrics",
+		Description: "controls whether the collector to uses OpenTelemetry for internal metrics",
 		Enabled:     false,
 	})
-	return &colTelemetry{registry: registry}
+	return &telemetryInitializer{
+		registry: registry,
+		mp:       nonrecording.NewNoopMeterProvider(),
+	}
 }
 
-func (tel *colTelemetry) init(svc *service) error {
+func (tel *telemetryInitializer) init(buildInfo component.BuildInfo, logger *zap.Logger, cfg ConfigServiceTelemetry, asyncErrorChannel chan error) error {
 	var err error
 	tel.doInitOnce.Do(
 		func() {
-			err = tel.initOnce(svc)
+			err = tel.initOnce(buildInfo, logger, cfg, asyncErrorChannel)
 		},
 	)
-	if err != nil {
-		return fmt.Errorf("failed to initialize telemetry: %w", err)
-	}
-	return nil
+	return err
 }
 
-func (tel *colTelemetry) initOnce(svc *service) error {
-	telemetryConf := svc.config.Telemetry
-
-	if telemetryConf.Metrics.Level == configtelemetry.LevelNone || telemetryConf.Metrics.Address == "" {
-		svc.telemetry.Logger.Info(
+func (tel *telemetryInitializer) initOnce(buildInfo component.BuildInfo, logger *zap.Logger, cfg ConfigServiceTelemetry, asyncErrorChannel chan error) error {
+	if cfg.Metrics.Level == configtelemetry.LevelNone || cfg.Metrics.Address == "" {
+		logger.Info(
 			"Skipping telemetry setup.",
-			zap.String(zapKeyTelemetryAddress, telemetryConf.Metrics.Address),
-			zap.String(zapKeyTelemetryLevel, telemetryConf.Metrics.Level.String()),
+			zap.String(zapKeyTelemetryAddress, cfg.Metrics.Address),
+			zap.String(zapKeyTelemetryLevel, cfg.Metrics.Level.String()),
 		)
 		return nil
 	}
 
-	svc.telemetry.Logger.Info("Setting up own telemetry...")
+	logger.Info("Setting up own telemetry...")
 
 	// Construct telemetry attributes from resource attributes.
 	telAttrs := map[string]string{}
-	for k, v := range telemetryConf.Resource {
+	for k, v := range cfg.Resource {
 		// nil value indicates that the attribute should not be included in the telemetry.
 		if v != nil {
 			telAttrs[k] = *v
 		}
 	}
 
-	if _, ok := telemetryConf.Resource[semconv.AttributeServiceInstanceID]; !ok {
+	if _, ok := cfg.Resource[semconv.AttributeServiceInstanceID]; !ok {
 		// AttributeServiceInstanceID is not specified in the config. Auto-generate one.
 		instanceUUID, _ := uuid.NewRandom()
 		instanceID := instanceUUID.String()
 		telAttrs[semconv.AttributeServiceInstanceID] = instanceID
 	}
 
-	if _, ok := telemetryConf.Resource[semconv.AttributeServiceVersion]; !ok {
+	if _, ok := cfg.Resource[semconv.AttributeServiceVersion]; !ok {
 		// AttributeServiceVersion is not specified in the config. Use the actual
 		// build version.
-		telAttrs[semconv.AttributeServiceVersion] = svc.buildInfo.Version
+		telAttrs[semconv.AttributeServiceVersion] = buildInfo.Version
 	}
 
 	var pe http.Handler
+	var err error
 	if tel.registry.IsEnabled(useOtelForInternalMetricsfeatureGateID) {
-		otelHandler, err := tel.initOpenTelemetry(svc)
-		if err != nil {
-			return err
-		}
-		pe = otelHandler
+		pe, err = tel.initOpenTelemetry()
 	} else {
-		ocHandler, err := tel.initOpenCensus(svc, telAttrs)
-		if err != nil {
-			return err
-		}
-		pe = ocHandler
+		pe, err = tel.initOpenCensus(cfg, telAttrs)
+	}
+	if err != nil {
+		return err
 	}
 
-	svc.telemetry.Logger.Info(
+	logger.Info(
 		"Serving Prometheus metrics",
-		zap.String(zapKeyTelemetryAddress, telemetryConf.Metrics.Address),
-		zap.String(zapKeyTelemetryLevel, telemetryConf.Metrics.Level.String()),
+		zap.String(zapKeyTelemetryAddress, cfg.Metrics.Address),
+		zap.String(zapKeyTelemetryLevel, cfg.Metrics.Level.String()),
 	)
 
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", pe)
 
 	tel.server = &http.Server{
-		Addr:    telemetryConf.Metrics.Address,
+		Addr:    cfg.Metrics.Address,
 		Handler: mux,
 	}
 
 	go func() {
 		if serveErr := tel.server.ListenAndServe(); serveErr != nil && !errors.Is(serveErr, http.ErrServerClosed) {
-			svc.host.asyncErrorChannel <- serveErr
+			asyncErrorChannel <- serveErr
 		}
 	}()
 
 	return nil
 }
 
-func (tel *colTelemetry) initOpenCensus(svc *service, telAttrs map[string]string) (http.Handler, error) {
-	tel.ocRegistry = metric.NewRegistry()
+func (tel *telemetryInitializer) initOpenCensus(cfg ConfigServiceTelemetry, telAttrs map[string]string) (http.Handler, error) {
+	tel.ocRegistry = ocmetric.NewRegistry()
 	metricproducer.GlobalManager().AddProducer(tel.ocRegistry)
 
-	if err := telemetry.RegisterProcessMetrics(tel.ocRegistry, getBallastSize(svc.host)); err != nil {
-		return nil, err
-	}
-
 	var views []*view.View
-	obsMetrics := obsreportconfig.Configure(svc.config.Telemetry.Metrics.Level)
+	obsMetrics := obsreportconfig.Configure(cfg.Metrics.Level)
 	views = append(views, batchprocessor.MetricViews()...)
 	views = append(views, obsMetrics.Views...)
 
@@ -204,7 +192,9 @@ func (tel *colTelemetry) initOpenCensus(svc *service, telAttrs map[string]string
 	return pe, nil
 }
 
-func (tel *colTelemetry) initOpenTelemetry(svc *service) (http.Handler, error) {
+func (tel *telemetryInitializer) initOpenTelemetry() (http.Handler, error) {
+	// Initialize the ocRegistry, still used by the process metrics.
+	tel.ocRegistry = ocmetric.NewRegistry()
 	config := otelprometheus.Config{}
 	c := controller.New(
 		processor.NewFactory(
@@ -221,11 +211,11 @@ func (tel *colTelemetry) initOpenTelemetry(svc *service) (http.Handler, error) {
 		return nil, err
 	}
 
-	svc.telemetry.MeterProvider = pe.MeterProvider()
+	tel.mp = pe.MeterProvider()
 	return pe, err
 }
 
-func (tel *colTelemetry) shutdown() error {
+func (tel *telemetryInitializer) shutdown() error {
 	metricproducer.GlobalManager().DeleteProducer(tel.ocRegistry)
 
 	view.Unregister(tel.views...)
