diff --git a/translator/conventions/opentelemetry.go b/translator/conventions/opentelemetry.go
index 298b0b285..e2b55ed59 100644
--- a/translator/conventions/opentelemetry.go
+++ b/translator/conventions/opentelemetry.go
@@ -15,18 +15,22 @@
 package conventions
 
 // OpenTelemetry Semantic Convention values for Resource attribute names.
-// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/README.md
 const (
 	AttributeServiceName          = "service.name"
 	AttributeServiceNamespace     = "service.namespace"
 	AttributeServiceInstance      = "service.instance.id"
 	AttributeServiceVersion       = "service.version"
-	AttributeLibraryName          = "library.name"
-	AttributeLibraryLanguage      = "library.language"
-	AttributeLibraryVersion       = "library.version"
+	AttributeTelemetrySDKName     = "telemetry.sdk.name"
+	AttributeTelemetrySDKLanguage = "telemetry.sdk.language"
+	AttributeTelemetrySDKVersion  = "telemetry.sdk.version"
 	AttributeContainerName        = "container.name"
 	AttributeContainerImage       = "container.image.name"
 	AttributeContainerTag         = "container.image.tag"
+	AttributeFaasName             = "faas.name"
+	AttributeFaasID               = "faas.id"
+	AttributeFaasVersion          = "faas.version"
+	AttributeFaasInstance         = "faas.instance"
 	AttributeK8sCluster           = "k8s.cluster.name"
 	AttributeK8sNamespace         = "k8s.namespace.name"
 	AttributeK8sPod               = "k8s.pod.name"
@@ -42,13 +46,10 @@ const (
 	AttributeCloudAccount         = "cloud.account.id"
 	AttributeCloudRegion          = "cloud.region"
 	AttributeCloudZone            = "cloud.zone"
-	AttributeTelemetrySDKName     = "telemetry.sdk.name"
-	AttributeTelemetrySDKLanguage = "telemetry.sdk.language"
-	AttributeTelemetrySDKVersion  = "telemetry.sdk.version"
 )
 
 // OpenTelemetry Semantic Convention values for general Span attribute names.
-// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-span-general.md
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/span-general.md
 const (
 	AttributeComponent    = "component"
 	AttributeNetTransport = "net.transport"
@@ -71,7 +72,7 @@ const (
 )
 
 // OpenTelemetry Semantic Convention attribute names for HTTP related attributes
-// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md
 const (
 	AttributeHTTPMethod     = "http.method"
 	AttributeHTTPURL        = "http.url"
@@ -90,7 +91,7 @@ const (
 )
 
 // OpenTelemetry Semantic Convention attribute names for database related attributes
-// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-database.md
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md
 const (
 	AttributeDBType      = "db.type"
 	AttributeDBInstance  = "db.instance"
@@ -100,7 +101,7 @@ const (
 )
 
 // OpenTelemetry Semantic Convention attribute names for gRPC related attributes
-// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md
 const (
 	AttributeRPCService              = "rpc.service"
 	EventTypeMessage                 = "message"
@@ -111,3 +112,38 @@ const (
 	AttributeMessageCompressedSize   = "message.compressed_size"
 	AttributeMessageUncompressedSize = "message.uncompressed_size"
 )
+
+// OpenTelemetry Semantic Convention attribute names for FaaS related attributes
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/faas.md
+const (
+	AttributeFaaSTrigger            = "faas.trigger"
+	AttributeFaaSExecution          = "faas.execution"
+	AttributeFaaSDocumentCollection = "faas.document.collection"
+	AttributeFaaSDocumentOperation  = "faas.document.operation"
+	AttributeFaaSDocumentTime       = "faas.document.time"
+	AttributeFaaSDocumentName       = "faas.document.name"
+	AttributeFaaSTime               = "faas.time"
+	AttributeFaaSCron               = "faas.cron"
+	FaaSTriggerDataSource           = "datasource"
+	FaaSTriggerHTTP                 = "http"
+	FaaSTriggerPubSub               = "pubsub"
+	FaaSTriggerTimer                = "timer"
+	FaaSTriggerOther                = "other"
+)
+
+// OpenTelemetry Semantic Convention attribute names for messaging system related attributes
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+const (
+	AttributeMessagingSystem                = "messaging.system"
+	AttributeMessagingDestination           = "messaging.destination"
+	AttributeMessagingDestinationKind       = "messaging.destination_kind"
+	AttributeMessagingTempDestination       = "messaging.temp_destination"
+	AttributeMessagingProtocol              = "messaging.protocol"
+	AttributeMessagingProtocolVersion       = "messaging.protocol_version"
+	AttributeMessagingURL                   = "messaging.url"
+	AttributeMessagingMessageID             = "messaging.message_id"
+	AttributeMessagingConversationID        = "messaging.conversation_id"
+	AttributeMessagingPayloadSize           = "messaging.message_payload_size_bytes"
+	AttributeMessagingPayloadCompressedSize = "messaging.message_payload_compressed_size_bytes"
+	AttributeMessagingOperation             = "messaging.operation"
+)
