diff --git a/connector/xconnector/go.mod b/connector/xconnector/go.mod
index 6dc9bccd6..94945f4bc 100644
--- a/connector/xconnector/go.mod
+++ b/connector/xconnector/go.mod
@@ -29,7 +29,7 @@ require (
 	go.opentelemetry.io/otel/metric v1.38.0 // indirect
 	go.opentelemetry.io/otel/trace v1.38.0 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
-	go.uber.org/zap v1.27.0 // indirect
+	go.uber.org/zap v1.27.1 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
