diff --git a/config/receiver.go b/config/receiver.go
index 51ab39632..e647e5eba 100644
--- a/config/receiver.go
+++ b/config/receiver.go
@@ -44,8 +44,3 @@ func (rs *ReceiverSettings) ID() component.ID {
 func (rs *ReceiverSettings) SetIDName(idName string) {
 	rs.id = component.NewIDWithName(rs.id.Type(), idName)
 }
-
-// Deprecated: [v0.65.0] Not needed anymore since the Validate() will be moved from Config interface to the optional ConfigValidator interface.
-func (rs *ReceiverSettings) Validate() error {
-	return nil
-}
