diff --git a/extension/ballastextension/factory.go b/extension/ballastextension/factory.go
index 112e7df32..e19dbee50 100644
--- a/extension/ballastextension/factory.go
+++ b/extension/ballastextension/factory.go
@@ -32,7 +32,7 @@ var memHandler = iruntime.TotalMemory
 
 // NewFactory creates a factory for FluentBit extension.
 func NewFactory() component.ExtensionFactory {
-	return component.NewExtensionFactoryWithStabilityLevel(typeStr, createDefaultConfig, createExtension, component.StabilityLevelBeta)
+	return component.NewExtensionFactory(typeStr, createDefaultConfig, createExtension, component.StabilityLevelBeta)
 }
 
 func createDefaultConfig() config.Extension {
