diff --git a/exporter/zipkinexporter/factory.go b/exporter/zipkinexporter/factory.go
index 65d015a5f..872cf8a8d 100644
--- a/exporter/zipkinexporter/factory.go
+++ b/exporter/zipkinexporter/factory.go
@@ -50,6 +50,8 @@ func createDefaultConfig() configmodels.Exporter {
 			TypeVal: typeStr,
 			NameVal: typeStr,
 		},
+		RetrySettings: exporterhelper.CreateDefaultRetrySettings(),
+		QueueSettings: exporterhelper.CreateDefaultQueueSettings(),
 		HTTPClientSettings: confighttp.HTTPClientSettings{
 			Timeout: defaultTimeout,
 			// We almost read 0 bytes, so no need to tune ReadBufferSize.
@@ -72,5 +74,15 @@ func createTraceExporter(
 		return nil, errors.New("exporter config requires a non-empty 'endpoint'")
 	}
 
-	return newTraceExporter(zc)
+	ze, err := createZipkinExporter(zc)
+	if err != nil {
+		return nil, err
+	}
+	return exporterhelper.NewTraceExporter(
+		zc,
+		ze.pushTraceData,
+		// explicitly disable since we rely on http.Client timeout logic.
+		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
+		exporterhelper.WithQueue(zc.QueueSettings),
+		exporterhelper.WithRetry(zc.RetrySettings))
 }
