diff --git a/cmd/pdatagen/internal/metrics_structs.go b/cmd/pdatagen/internal/metrics_structs.go
index 4def50ef2..4c4615cd0 100644
--- a/cmd/pdatagen/internal/metrics_structs.go
+++ b/cmd/pdatagen/internal/metrics_structs.go
@@ -340,7 +340,7 @@ var quantileValues = &messageValueStruct{
 	},
 }
 
-var intExemplarSlice = &sliceOfPtrs{
+var intExemplarSlice = &sliceOfValues{
 	structName: "IntExemplarSlice",
 	element:    intExemplar,
 }
@@ -363,7 +363,7 @@ var intExemplar = &messageValueStruct{
 	},
 }
 
-var doubleExemplarSlice = &sliceOfPtrs{
+var doubleExemplarSlice = &sliceOfValues{
 	structName: "DoubleExemplarSlice",
 	element:    doubleExemplar,
 }
