diff --git a/pdata/internal/generated_wrapper_scopemetrics_test.go b/pdata/internal/generated_wrapper_scopemetrics_test.go
index b3973ad05..c2d0496e0 100644
--- a/pdata/internal/generated_wrapper_scopemetrics_test.go
+++ b/pdata/internal/generated_wrapper_scopemetrics_test.go
@@ -10,8 +10,10 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestCopyOrigScopeMetrics(t *testing.T) {
@@ -23,3 +25,24 @@ func TestCopyOrigScopeMetrics(t *testing.T) {
 	CopyOrigScopeMetrics(dest, src)
 	assert.Equal(t, src, dest)
 }
+
+func TestMarshalAndUnmarshalJSONOrigScopeMetrics(t *testing.T) {
+	src := &otlpmetrics.ScopeMetrics{}
+	FillOrigTestScopeMetrics(src)
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	MarshalJSONOrigScopeMetrics(src, stream)
+	require.NoError(t, stream.Error())
+
+	// Append an unknown field at the start to ensure unknown fields are skipped
+	// and the unmarshal logic continues.
+	buf := stream.Buffer()
+	assert.EqualValues(t, '{', buf[0])
+	iter := json.BorrowIterator(append([]byte(`{"unknown": "string",`), buf[1:]...))
+	defer json.ReturnIterator(iter)
+	dest := &otlpmetrics.ScopeMetrics{}
+	UnmarshalJSONOrigScopeMetrics(dest, iter)
+	require.NoError(t, iter.Error())
+
+	assert.Equal(t, src, dest)
+}
