diff --git a/pdata/internal/generated_wrapper_profile_test.go b/pdata/internal/generated_wrapper_profile_test.go
index 20271e95b..21c6d2db5 100644
--- a/pdata/internal/generated_wrapper_profile_test.go
+++ b/pdata/internal/generated_wrapper_profile_test.go
@@ -10,8 +10,10 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestCopyOrigProfile(t *testing.T) {
@@ -23,3 +25,24 @@ func TestCopyOrigProfile(t *testing.T) {
 	CopyOrigProfile(dest, src)
 	assert.Equal(t, src, dest)
 }
+
+func TestMarshalAndUnmarshalJSONOrigProfile(t *testing.T) {
+	src := &otlpprofiles.Profile{}
+	FillOrigTestProfile(src)
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	MarshalJSONOrigProfile(src, stream)
+	require.NoError(t, stream.Error())
+
+	// Append an unknown field at the start to ensure unknown fields are skipped
+	// and the unmarshal logic continues.
+	buf := stream.Buffer()
+	assert.EqualValues(t, '{', buf[0])
+	iter := json.BorrowIterator(append([]byte(`{"unknown": "string",`), buf[1:]...))
+	defer json.ReturnIterator(iter)
+	dest := &otlpprofiles.Profile{}
+	UnmarshalJSONOrigProfile(dest, iter)
+	require.NoError(t, iter.Error())
+
+	assert.Equal(t, src, dest)
+}
