diff --git a/pdata/internal/generated_wrapper_span_linkslice.go b/pdata/internal/generated_wrapper_span_linkslice.go
index c43f6e1d5..ec2009fb3 100644
--- a/pdata/internal/generated_wrapper_span_linkslice.go
+++ b/pdata/internal/generated_wrapper_span_linkslice.go
@@ -8,6 +8,7 @@ package internal
 
 import (
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func CopyOrigSpan_LinkSlice(dest, src []*otlptrace.Span_Link) []*otlptrace.Span_Link {
@@ -47,3 +48,27 @@ func GenerateOrigTestSpan_LinkSlice() []*otlptrace.Span_Link {
 	}
 	return orig
 }
+
+// MarshalJSONOrigSpan_LinkSlice marshals all properties from the current struct to the destination stream.
+func MarshalJSONOrigSpan_LinkSlice(orig []*otlptrace.Span_Link, dest *json.Stream) {
+	dest.WriteArrayStart()
+	if len(orig) > 0 {
+		MarshalJSONOrigSpan_Link(orig[0], dest)
+	}
+	for i := 1; i < len(orig); i++ {
+		dest.WriteMore()
+		MarshalJSONOrigSpan_Link(orig[i], dest)
+	}
+	dest.WriteArrayEnd()
+}
+
+// UnmarshalJSONOrigSpan_LinkSlice unmarshals all properties from the current struct from the source iterator.
+func UnmarshalJSONOrigSpan_LinkSlice(iter *json.Iterator) []*otlptrace.Span_Link {
+	var orig []*otlptrace.Span_Link
+	iter.ReadArrayCB(func(iter *json.Iterator) bool {
+		orig = append(orig, &otlptrace.Span_Link{})
+		UnmarshalJSONOrigSpan_Link(orig[len(orig)-1], iter)
+		return true
+	})
+	return orig
+}
