diff --git a/pdata/internal/generated_wrapper_lineslice_test.go b/pdata/internal/generated_wrapper_lineslice_test.go
index 1ab7259a3..556adae5f 100644
--- a/pdata/internal/generated_wrapper_lineslice_test.go
+++ b/pdata/internal/generated_wrapper_lineslice_test.go
@@ -10,8 +10,10 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestCopyOrigLineSlice(t *testing.T) {
@@ -38,3 +40,19 @@ func TestCopyOrigLineSlice(t *testing.T) {
 	dest = CopyOrigLineSlice(dest, src)
 	assert.Equal(t, GenerateOrigTestLineSlice(), dest)
 }
+
+func TestMarshalAndUnmarshalJSONOrigLineSlice(t *testing.T) {
+	src := GenerateOrigTestLineSlice()
+
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	MarshalJSONOrigLineSlice(src, stream)
+	require.NoError(t, stream.Error())
+
+	iter := json.BorrowIterator(stream.Buffer())
+	defer json.ReturnIterator(iter)
+	dest := UnmarshalJSONOrigLineSlice(iter)
+	require.NoError(t, iter.Error())
+
+	assert.Equal(t, src, dest)
+}
