diff --git a/pdata/internal/generated_wrapper_entityrefslice_test.go b/pdata/internal/generated_wrapper_entityrefslice_test.go
index 9b3943063..7f9d97894 100644
--- a/pdata/internal/generated_wrapper_entityrefslice_test.go
+++ b/pdata/internal/generated_wrapper_entityrefslice_test.go
@@ -10,8 +10,10 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestCopyOrigEntityRefSlice(t *testing.T) {
@@ -38,3 +40,19 @@ func TestCopyOrigEntityRefSlice(t *testing.T) {
 	dest = CopyOrigEntityRefSlice(dest, src)
 	assert.Equal(t, GenerateOrigTestEntityRefSlice(), dest)
 }
+
+func TestMarshalAndUnmarshalJSONOrigEntityRefSlice(t *testing.T) {
+	src := GenerateOrigTestEntityRefSlice()
+
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	MarshalJSONOrigEntityRefSlice(src, stream)
+	require.NoError(t, stream.Error())
+
+	iter := json.BorrowIterator(stream.Buffer())
+	defer json.ReturnIterator(iter)
+	dest := UnmarshalJSONOrigEntityRefSlice(iter)
+	require.NoError(t, iter.Error())
+
+	assert.Equal(t, src, dest)
+}
