diff --git a/pdata/internal/generated_wrapper_resourcemetricsslice_test.go b/pdata/internal/generated_wrapper_resourcemetricsslice_test.go
index e1626a0fc..757433056 100644
--- a/pdata/internal/generated_wrapper_resourcemetricsslice_test.go
+++ b/pdata/internal/generated_wrapper_resourcemetricsslice_test.go
@@ -10,8 +10,10 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func TestCopyOrigResourceMetricsSlice(t *testing.T) {
@@ -38,3 +40,19 @@ func TestCopyOrigResourceMetricsSlice(t *testing.T) {
 	dest = CopyOrigResourceMetricsSlice(dest, src)
 	assert.Equal(t, GenerateOrigTestResourceMetricsSlice(), dest)
 }
+
+func TestMarshalAndUnmarshalJSONOrigResourceMetricsSlice(t *testing.T) {
+	src := GenerateOrigTestResourceMetricsSlice()
+
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	MarshalJSONOrigResourceMetricsSlice(src, stream)
+	require.NoError(t, stream.Error())
+
+	iter := json.BorrowIterator(stream.Buffer())
+	defer json.ReturnIterator(iter)
+	dest := UnmarshalJSONOrigResourceMetricsSlice(iter)
+	require.NoError(t, iter.Error())
+
+	assert.Equal(t, src, dest)
+}
