diff --git a/pdata/internal/generated_wrapper_entityref.go b/pdata/internal/generated_wrapper_entityref.go
index 3222be133..0f5f2d349 100644
--- a/pdata/internal/generated_wrapper_entityref.go
+++ b/pdata/internal/generated_wrapper_entityref.go
@@ -35,57 +35,57 @@ func GenerateTestEntityRef() EntityRef {
 	return NewEntityRef(&orig, &state)
 }
 
-// MarshalJSONStream marshals all properties from the current struct to the destination stream.
-func MarshalJSONStreamEntityRef(ms EntityRef, dest *json.Stream) {
+func CopyOrigEntityRef(dest, src *otlpcommon.EntityRef) {
+	dest.SchemaUrl = src.SchemaUrl
+	dest.Type = src.Type
+	dest.IdKeys = CopyOrigStringSlice(dest.IdKeys, src.IdKeys)
+	dest.DescriptionKeys = CopyOrigStringSlice(dest.DescriptionKeys, src.DescriptionKeys)
+}
+
+func FillOrigTestEntityRef(orig *otlpcommon.EntityRef) {
+	orig.SchemaUrl = "test_schemaurl"
+	orig.Type = "test_type"
+	orig.IdKeys = GenerateOrigTestStringSlice()
+	orig.DescriptionKeys = GenerateOrigTestStringSlice()
+}
+
+// MarshalJSONOrig marshals all properties from the current struct to the destination stream.
+func MarshalJSONOrigEntityRef(orig *otlpcommon.EntityRef, dest *json.Stream) {
 	dest.WriteObjectStart()
-	if ms.orig.SchemaUrl != "" {
+	if orig.SchemaUrl != "" {
 		dest.WriteObjectField("schemaUrl")
-		dest.WriteString(ms.orig.SchemaUrl)
+		dest.WriteString(orig.SchemaUrl)
 	}
-	if ms.orig.Type != "" {
+	if orig.Type != "" {
 		dest.WriteObjectField("type")
-		dest.WriteString(ms.orig.Type)
+		dest.WriteString(orig.Type)
 	}
-	if len(ms.orig.IdKeys) > 0 {
+	if len(orig.IdKeys) > 0 {
 		dest.WriteObjectField("idKeys")
-		MarshalJSONStreamStringSlice(NewStringSlice(&ms.orig.IdKeys, ms.state), dest)
+		MarshalJSONOrigStringSlice(orig.IdKeys, dest)
 	}
-	if len(ms.orig.DescriptionKeys) > 0 {
+	if len(orig.DescriptionKeys) > 0 {
 		dest.WriteObjectField("descriptionKeys")
-		MarshalJSONStreamStringSlice(NewStringSlice(&ms.orig.DescriptionKeys, ms.state), dest)
+		MarshalJSONOrigStringSlice(orig.DescriptionKeys, dest)
 	}
 	dest.WriteObjectEnd()
 }
 
-// UnmarshalJSONIterEntityRef unmarshals all properties from the current struct from the source iterator.
-func UnmarshalJSONIterEntityRef(ms EntityRef, iter *json.Iterator) {
+// UnmarshalJSONOrigEntityRef unmarshals all properties from the current struct from the source iterator.
+func UnmarshalJSONOrigEntityRef(orig *otlpcommon.EntityRef, iter *json.Iterator) {
 	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "schemaUrl", "schema_url":
-			ms.orig.SchemaUrl = iter.ReadString()
+			orig.SchemaUrl = iter.ReadString()
 		case "type":
-			ms.orig.Type = iter.ReadString()
+			orig.Type = iter.ReadString()
 		case "idKeys", "id_keys":
-			UnmarshalJSONIterStringSlice(NewStringSlice(&ms.orig.IdKeys, ms.state), iter)
+			orig.IdKeys = UnmarshalJSONOrigStringSlice(iter)
 		case "descriptionKeys", "description_keys":
-			UnmarshalJSONIterStringSlice(NewStringSlice(&ms.orig.DescriptionKeys, ms.state), iter)
+			orig.DescriptionKeys = UnmarshalJSONOrigStringSlice(iter)
 		default:
 			iter.Skip()
 		}
 		return true
 	})
 }
-
-func CopyOrigEntityRef(dest, src *otlpcommon.EntityRef) {
-	dest.SchemaUrl = src.SchemaUrl
-	dest.Type = src.Type
-	dest.IdKeys = CopyOrigStringSlice(dest.IdKeys, src.IdKeys)
-	dest.DescriptionKeys = CopyOrigStringSlice(dest.DescriptionKeys, src.DescriptionKeys)
-}
-
-func FillOrigTestEntityRef(orig *otlpcommon.EntityRef) {
-	orig.SchemaUrl = "test_schemaurl"
-	orig.Type = "test_type"
-	orig.IdKeys = GenerateOrigTestStringSlice()
-	orig.DescriptionKeys = GenerateOrigTestStringSlice()
-}
