diff --git a/pdata/internal/generated_wrapper_span_link.go b/pdata/internal/generated_wrapper_span_link.go
index 529c07126..c6eed2e11 100644
--- a/pdata/internal/generated_wrapper_span_link.go
+++ b/pdata/internal/generated_wrapper_span_link.go
@@ -9,6 +9,7 @@ package internal
 import (
 	"go.opentelemetry.io/collector/pdata/internal/data"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func CopyOrigSpan_Link(dest, src *otlptrace.Span_Link) {
@@ -28,3 +29,56 @@ func FillOrigTestSpan_Link(orig *otlptrace.Span_Link) {
 	orig.Attributes = GenerateOrigTestKeyValueSlice()
 	orig.DroppedAttributesCount = uint32(13)
 }
+
+// MarshalJSONOrig marshals all properties from the current struct to the destination stream.
+func MarshalJSONOrigSpan_Link(orig *otlptrace.Span_Link, dest *json.Stream) {
+	dest.WriteObjectStart()
+	if orig.TraceId != data.TraceID([16]byte{}) {
+		dest.WriteObjectField("traceId")
+		orig.TraceId.MarshalJSONStream(dest)
+	}
+	if orig.SpanId != data.SpanID([8]byte{}) {
+		dest.WriteObjectField("spanId")
+		orig.SpanId.MarshalJSONStream(dest)
+	}
+	if orig.TraceState != "" {
+		dest.WriteObjectField("traceState")
+		MarshalJSONOrigTraceState(&orig.TraceState, dest)
+	}
+	if orig.Flags != uint32(0) {
+		dest.WriteObjectField("flags")
+		dest.WriteUint32(orig.Flags)
+	}
+	if len(orig.Attributes) > 0 {
+		dest.WriteObjectField("attributes")
+		MarshalJSONOrigKeyValueSlice(orig.Attributes, dest)
+	}
+	if orig.DroppedAttributesCount != uint32(0) {
+		dest.WriteObjectField("droppedAttributesCount")
+		dest.WriteUint32(orig.DroppedAttributesCount)
+	}
+	dest.WriteObjectEnd()
+}
+
+// UnmarshalJSONOrigSpanLink unmarshals all properties from the current struct from the source iterator.
+func UnmarshalJSONOrigSpan_Link(orig *otlptrace.Span_Link, iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
+		switch f {
+		case "traceId", "trace_id":
+			orig.TraceId.UnmarshalJSONIter(iter)
+		case "spanId", "span_id":
+			orig.SpanId.UnmarshalJSONIter(iter)
+		case "traceState", "trace_state":
+			UnmarshalJSONOrigTraceState(&orig.TraceState, iter)
+		case "flags":
+			orig.Flags = iter.ReadUint32()
+		case "attributes":
+			orig.Attributes = UnmarshalJSONOrigKeyValueSlice(iter)
+		case "droppedAttributesCount", "dropped_attributes_count":
+			orig.DroppedAttributesCount = iter.ReadUint32()
+		default:
+			iter.Skip()
+		}
+		return true
+	})
+}
