diff --git a/pdata/internal/generated_wrapper_summary.go b/pdata/internal/generated_wrapper_summary.go
index 32986c60c..736a609d3 100644
--- a/pdata/internal/generated_wrapper_summary.go
+++ b/pdata/internal/generated_wrapper_summary.go
@@ -8,6 +8,7 @@ package internal
 
 import (
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 func CopyOrigSummary(dest, src *otlpmetrics.Summary) {
@@ -17,3 +18,26 @@ func CopyOrigSummary(dest, src *otlpmetrics.Summary) {
 func FillOrigTestSummary(orig *otlpmetrics.Summary) {
 	orig.DataPoints = GenerateOrigTestSummaryDataPointSlice()
 }
+
+// MarshalJSONOrig marshals all properties from the current struct to the destination stream.
+func MarshalJSONOrigSummary(orig *otlpmetrics.Summary, dest *json.Stream) {
+	dest.WriteObjectStart()
+	if len(orig.DataPoints) > 0 {
+		dest.WriteObjectField("dataPoints")
+		MarshalJSONOrigSummaryDataPointSlice(orig.DataPoints, dest)
+	}
+	dest.WriteObjectEnd()
+}
+
+// UnmarshalJSONOrigSummary unmarshals all properties from the current struct from the source iterator.
+func UnmarshalJSONOrigSummary(orig *otlpmetrics.Summary, iter *json.Iterator) {
+	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
+		switch f {
+		case "dataPoints", "data_points":
+			orig.DataPoints = UnmarshalJSONOrigSummaryDataPointSlice(iter)
+		default:
+			iter.Skip()
+		}
+		return true
+	})
+}
