diff --git a/receiver/otlpreceiver/factory.go b/receiver/otlpreceiver/factory.go
index 164455722..24cb581e6 100644
--- a/receiver/otlpreceiver/factory.go
+++ b/receiver/otlpreceiver/factory.go
@@ -67,7 +67,7 @@ func createDefaultConfig() config.Receiver {
 
 // CreateTracesReceiver creates a  trace receiver based on provided config.
 func createTracesReceiver(
-	ctx context.Context,
+	_ context.Context,
 	set component.ReceiverCreateSettings,
 	cfg config.Receiver,
 	nextConsumer consumer.Traces,
@@ -76,7 +76,7 @@ func createTracesReceiver(
 		return newOtlpReceiver(cfg.(*Config), set.Logger)
 	})
 
-	if err := r.Unwrap().(*otlpReceiver).registerTraceConsumer(ctx, nextConsumer); err != nil {
+	if err := r.Unwrap().(*otlpReceiver).registerTraceConsumer(nextConsumer); err != nil {
 		return nil, err
 	}
 	return r, nil
@@ -84,7 +84,7 @@ func createTracesReceiver(
 
 // CreateMetricsReceiver creates a metrics receiver based on provided config.
 func createMetricsReceiver(
-	ctx context.Context,
+	_ context.Context,
 	set component.ReceiverCreateSettings,
 	cfg config.Receiver,
 	consumer consumer.Metrics,
@@ -93,7 +93,7 @@ func createMetricsReceiver(
 		return newOtlpReceiver(cfg.(*Config), set.Logger)
 	})
 
-	if err := r.Unwrap().(*otlpReceiver).registerMetricsConsumer(ctx, consumer); err != nil {
+	if err := r.Unwrap().(*otlpReceiver).registerMetricsConsumer(consumer); err != nil {
 		return nil, err
 	}
 	return r, nil
@@ -101,7 +101,7 @@ func createMetricsReceiver(
 
 // CreateLogReceiver creates a log receiver based on provided config.
 func createLogReceiver(
-	ctx context.Context,
+	_ context.Context,
 	set component.ReceiverCreateSettings,
 	cfg config.Receiver,
 	consumer consumer.Logs,
@@ -110,7 +110,7 @@ func createLogReceiver(
 		return newOtlpReceiver(cfg.(*Config), set.Logger)
 	})
 
-	if err := r.Unwrap().(*otlpReceiver).registerLogsConsumer(ctx, consumer); err != nil {
+	if err := r.Unwrap().(*otlpReceiver).registerLogsConsumer(consumer); err != nil {
 		return nil, err
 	}
 	return r, nil
