diff --git a/receiver/opencensusreceiver/factory.go b/receiver/opencensusreceiver/factory.go
index 7116d4f70..484e37dcb 100644
--- a/receiver/opencensusreceiver/factory.go
+++ b/receiver/opencensusreceiver/factory.go
@@ -60,7 +60,7 @@ func createTraceReceiver(
 	_ component.ReceiverCreateParams,
 	cfg configmodels.Receiver,
 	nextConsumer consumer.TracesConsumer,
-) (component.TraceReceiver, error) {
+) (component.TracesReceiver, error) {
 	r, err := createReceiver(cfg)
 	if err != nil {
 		return nil, err
@@ -116,6 +116,6 @@ func createReceiver(cfg configmodels.Receiver) (*ocReceiver, error) {
 
 // This is the map of already created OpenCensus receivers for particular configurations.
 // We maintain this map because the Factory is asked trace and metric receivers separately
-// when it gets CreateTraceReceiver() and CreateMetricsReceiver() but they must not
+// when it gets CreateTracesReceiver() and CreateMetricsReceiver() but they must not
 // create separate objects, they must use one ocReceiver object per configuration.
 var receivers = map[*Config]*ocReceiver{}
