diff --git a/service/go.mod b/service/go.mod
index b73d39b16..12d153207 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -14,16 +14,16 @@ require (
 	go.opentelemetry.io/collector/component/componentstatus v0.107.0
 	go.opentelemetry.io/collector/config/confighttp v0.107.0
 	go.opentelemetry.io/collector/config/configtelemetry v0.107.0
-	go.opentelemetry.io/collector/confmap v0.107.0
+	go.opentelemetry.io/collector/confmap v1.14.0
 	go.opentelemetry.io/collector/connector v0.107.0
 	go.opentelemetry.io/collector/consumer v0.107.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.107.0
 	go.opentelemetry.io/collector/exporter v0.107.0
 	go.opentelemetry.io/collector/extension v0.107.0
 	go.opentelemetry.io/collector/extension/zpagesextension v0.107.0
-	go.opentelemetry.io/collector/featuregate v1.13.0
+	go.opentelemetry.io/collector/featuregate v1.14.0
 	go.opentelemetry.io/collector/internal/globalgates v0.107.0
-	go.opentelemetry.io/collector/pdata v1.13.0
+	go.opentelemetry.io/collector/pdata v1.14.0
 	go.opentelemetry.io/collector/pdata/testdata v0.107.0
 	go.opentelemetry.io/collector/processor v0.107.0
 	go.opentelemetry.io/collector/receiver v0.107.0
@@ -79,12 +79,12 @@ require (
 	github.com/tklauser/go-sysconf v0.3.12 // indirect
 	github.com/tklauser/numcpus v0.6.1 // indirect
 	github.com/yusufpapurcu/wmi v1.2.4 // indirect
-	go.opentelemetry.io/collector/client v1.13.0 // indirect
+	go.opentelemetry.io/collector/client v1.14.0 // indirect
 	go.opentelemetry.io/collector/component/componentprofiles v0.107.0 // indirect
 	go.opentelemetry.io/collector/config/configauth v0.107.0 // indirect
-	go.opentelemetry.io/collector/config/configcompression v1.13.0 // indirect
-	go.opentelemetry.io/collector/config/configopaque v1.13.0 // indirect
-	go.opentelemetry.io/collector/config/configtls v1.13.0 // indirect
+	go.opentelemetry.io/collector/config/configcompression v1.14.0 // indirect
+	go.opentelemetry.io/collector/config/configopaque v1.14.0 // indirect
+	go.opentelemetry.io/collector/config/configtls v1.14.0 // indirect
 	go.opentelemetry.io/collector/config/internal v0.107.0 // indirect
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.107.0 // indirect
 	go.opentelemetry.io/collector/extension/auth v0.107.0 // indirect
