diff --git a/cmd/mdatagen/templates/component_telemetry_test.go.tmpl b/cmd/mdatagen/templates/component_telemetry_test.go.tmpl
index bf4b3ba2a..58c0ab7d6 100644
--- a/cmd/mdatagen/templates/component_telemetry_test.go.tmpl
+++ b/cmd/mdatagen/templates/component_telemetry_test.go.tmpl
@@ -7,6 +7,9 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	{{- if or isConnector isExporter isExtension isProcessor isReceiver }}
+	"go.opentelemetry.io/otel/metric"
+	{{- end }}
 	"go.opentelemetry.io/otel/sdk/metric/metricdata"
 	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
 	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
