diff --git a/pdata/pprofile/generated_resourceprofilesslice.go b/pdata/pprofile/generated_resourceprofilesslice.go
index a592c54ef..c78115dbf 100644
--- a/pdata/pprofile/generated_resourceprofilesslice.go
+++ b/pdata/pprofile/generated_resourceprofilesslice.go
@@ -145,22 +145,7 @@ func (es ResourceProfilesSlice) RemoveIf(f func(ResourceProfiles) bool) {
 // CopyTo copies all elements from the current slice overriding the destination.
 func (es ResourceProfilesSlice) CopyTo(dest ResourceProfilesSlice) {
 	dest.state.AssertMutable()
-	srcLen := es.Len()
-	destCap := cap(*dest.orig)
-	if srcLen <= destCap {
-		(*dest.orig) = (*dest.orig)[:srcLen:destCap]
-		for i := range *es.orig {
-			newResourceProfiles((*es.orig)[i], es.state).CopyTo(newResourceProfiles((*dest.orig)[i], dest.state))
-		}
-		return
-	}
-	origs := make([]otlpprofiles.ResourceProfiles, srcLen)
-	wrappers := make([]*otlpprofiles.ResourceProfiles, srcLen)
-	for i := range *es.orig {
-		wrappers[i] = &origs[i]
-		newResourceProfiles((*es.orig)[i], es.state).CopyTo(newResourceProfiles(wrappers[i], dest.state))
-	}
-	*dest.orig = wrappers
+	*dest.orig = copyOrigResourceProfilesSlice(*dest.orig, *es.orig)
 }
 
 // Sort sorts the ResourceProfiles elements within ResourceProfilesSlice given the
@@ -170,3 +155,18 @@ func (es ResourceProfilesSlice) Sort(less func(a, b ResourceProfiles) bool) {
 	es.state.AssertMutable()
 	sort.SliceStable(*es.orig, func(i, j int) bool { return less(es.At(i), es.At(j)) })
 }
+
+func copyOrigResourceProfilesSlice(dest, src []*otlpprofiles.ResourceProfiles) []*otlpprofiles.ResourceProfiles {
+	if cap(dest) < len(src) {
+		dest = make([]*otlpprofiles.ResourceProfiles, len(src))
+		data := make([]otlpprofiles.ResourceProfiles, len(src))
+		for i := range src {
+			dest[i] = &data[i]
+		}
+	}
+	dest = dest[:len(src)]
+	for i := range src {
+		copyOrigResourceProfiles(dest[i], src[i])
+	}
+	return dest
+}
