diff --git a/pdata/pmetric/metrics_test.go b/pdata/pmetric/metrics_test.go
index ad7977fb5..6e4c5ff2c 100644
--- a/pdata/pmetric/metrics_test.go
+++ b/pdata/pmetric/metrics_test.go
@@ -9,6 +9,7 @@ import (
 
 	gogoproto "github.com/gogo/protobuf/proto"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	goproto "google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/emptypb"
 
@@ -35,23 +36,23 @@ func TestResourceMetricsWireCompatibility(t *testing.T) {
 
 	// Marshal its underlying ProtoBuf to wire.
 	wire1, err := gogoproto.Marshal(metrics.getOrig())
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.NotNil(t, wire1)
 
 	// Unmarshal from the wire to OTLP Protobuf in goproto's representation.
 	var goprotoMessage emptypb.Empty
 	err = goproto.Unmarshal(wire1, &goprotoMessage)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	// Marshal to the wire again.
 	wire2, err := goproto.Marshal(&goprotoMessage)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.NotNil(t, wire2)
 
 	// Unmarshal from the wire into gogoproto's representation.
 	var gogoprotoRM otlpcollectormetrics.ExportMetricsServiceRequest
 	err = gogoproto.Unmarshal(wire2, &gogoprotoRM)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	// Now compare that the original and final ProtoBuf messages are the same.
 	// This proves that goproto and gogoproto marshaling/unmarshaling are wire compatible.
