diff --git a/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go b/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go
index 5be6838b3..b402b991f 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go
@@ -26,7 +26,6 @@ import (
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/internal/processor/filterset"
-	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
 	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal/metadata"
 )
 
@@ -38,7 +37,7 @@ const (
 // scraper for Network Metrics
 type scraper struct {
 	config    *Config
-	startTime pdata.TimestampUnixNano
+	startTime pdata.Timestamp
 	includeFS filterset.FilterSet
 	excludeFS filterset.FilterSet
 
@@ -77,7 +76,7 @@ func (s *scraper) start(context.Context, component.Host) error {
 		return err
 	}
 
-	s.startTime = pdata.TimestampUnixNano(bootTime * 1e9)
+	s.startTime = pdata.Timestamp(bootTime * 1e9)
 	return nil
 }
 
@@ -99,8 +98,8 @@ func (s *scraper) scrape(_ context.Context) (pdata.MetricSlice, error) {
 	return metrics, errors.Combine()
 }
 
-func (s *scraper) scrapeAndAppendNetworkCounterMetrics(metrics pdata.MetricSlice, startTime pdata.TimestampUnixNano) error {
-	now := internal.TimeToUnixNano(time.Now())
+func (s *scraper) scrapeAndAppendNetworkCounterMetrics(metrics pdata.MetricSlice, startTime pdata.Timestamp) error {
+	now := pdata.TimestampFromTime(time.Now())
 
 	// get total stats only
 	ioCounters, err := s.ioCounters( /*perNetworkInterfaceController=*/ true)
@@ -123,7 +122,7 @@ func (s *scraper) scrapeAndAppendNetworkCounterMetrics(metrics pdata.MetricSlice
 	return nil
 }
 
-func initializeNetworkPacketsMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.TimestampUnixNano, ioCountersSlice []net.IOCountersStat) {
+func initializeNetworkPacketsMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.Timestamp, ioCountersSlice []net.IOCountersStat) {
 	metricIntf.Init(metric)
 
 	idps := metric.IntSum().DataPoints()
@@ -134,7 +133,7 @@ func initializeNetworkPacketsMetric(metric pdata.Metric, metricIntf metadata.Met
 	}
 }
 
-func initializeNetworkDroppedPacketsMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.TimestampUnixNano, ioCountersSlice []net.IOCountersStat) {
+func initializeNetworkDroppedPacketsMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.Timestamp, ioCountersSlice []net.IOCountersStat) {
 	metricIntf.Init(metric)
 
 	idps := metric.IntSum().DataPoints()
@@ -145,7 +144,7 @@ func initializeNetworkDroppedPacketsMetric(metric pdata.Metric, metricIntf metad
 	}
 }
 
-func initializeNetworkErrorsMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.TimestampUnixNano, ioCountersSlice []net.IOCountersStat) {
+func initializeNetworkErrorsMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.Timestamp, ioCountersSlice []net.IOCountersStat) {
 	metricIntf.Init(metric)
 
 	idps := metric.IntSum().DataPoints()
@@ -156,7 +155,7 @@ func initializeNetworkErrorsMetric(metric pdata.Metric, metricIntf metadata.Metr
 	}
 }
 
-func initializeNetworkIOMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.TimestampUnixNano, ioCountersSlice []net.IOCountersStat) {
+func initializeNetworkIOMetric(metric pdata.Metric, metricIntf metadata.MetricIntf, startTime, now pdata.Timestamp, ioCountersSlice []net.IOCountersStat) {
 	metricIntf.Init(metric)
 
 	idps := metric.IntSum().DataPoints()
@@ -167,7 +166,7 @@ func initializeNetworkIOMetric(metric pdata.Metric, metricIntf metadata.MetricIn
 	}
 }
 
-func initializeNetworkDataPoint(dataPoint pdata.IntDataPoint, startTime, now pdata.TimestampUnixNano, deviceLabel, directionLabel string, value int64) {
+func initializeNetworkDataPoint(dataPoint pdata.IntDataPoint, startTime, now pdata.Timestamp, deviceLabel, directionLabel string, value int64) {
 	labelsMap := dataPoint.LabelsMap()
 	labelsMap.Insert(metadata.Labels.NetworkDevice, deviceLabel)
 	labelsMap.Insert(metadata.Labels.NetworkDirection, directionLabel)
@@ -177,7 +176,7 @@ func initializeNetworkDataPoint(dataPoint pdata.IntDataPoint, startTime, now pda
 }
 
 func (s *scraper) scrapeAndAppendNetworkConnectionsMetric(metrics pdata.MetricSlice) error {
-	now := internal.TimeToUnixNano(time.Now())
+	now := pdata.TimestampFromTime(time.Now())
 
 	connections, err := s.connections("tcp")
 	if err != nil {
@@ -204,7 +203,7 @@ func getTCPConnectionStatusCounts(connections []net.ConnectionStat) map[string]i
 	return tcpStatuses
 }
 
-func initializeNetworkConnectionsMetric(metric pdata.Metric, now pdata.TimestampUnixNano, connectionStateCounts map[string]int64) {
+func initializeNetworkConnectionsMetric(metric pdata.Metric, now pdata.Timestamp, connectionStateCounts map[string]int64) {
 	metadata.Metrics.SystemNetworkConnections.Init(metric)
 
 	idps := metric.IntSum().DataPoints()
@@ -217,7 +216,7 @@ func initializeNetworkConnectionsMetric(metric pdata.Metric, now pdata.Timestamp
 	}
 }
 
-func initializeNetworkConnectionsDataPoint(dataPoint pdata.IntDataPoint, now pdata.TimestampUnixNano, protocolLabel, stateLabel string, value int64) {
+func initializeNetworkConnectionsDataPoint(dataPoint pdata.IntDataPoint, now pdata.Timestamp, protocolLabel, stateLabel string, value int64) {
 	labelsMap := dataPoint.LabelsMap()
 	labelsMap.Insert(metadata.Labels.NetworkProtocol, protocolLabel)
 	labelsMap.Insert(metadata.Labels.NetworkState, stateLabel)
