diff --git a/extension/ballastextension/go.mod b/extension/ballastextension/go.mod
index 15a7d2d51..054f7cd7c 100644
--- a/extension/ballastextension/go.mod
+++ b/extension/ballastextension/go.mod
@@ -87,3 +87,7 @@ replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/con
 replace go.opentelemetry.io/collector/pdata/testdata => ../../pdata/testdata
 
 replace go.opentelemetry.io/collector/pdata/pprofile => ../../pdata/pprofile
+
+replace go.opentelemetry.io/collector/consumer/consumerprofiles => ../../consumer/consumerprofiles
+
+replace go.opentelemetry.io/collector/consumer/consumertest => ../../consumer/consumertest
