diff --git a/connector/go.mod b/connector/go.mod
index d77fae82c..8efd22f6c 100644
--- a/connector/go.mod
+++ b/connector/go.mod
@@ -11,7 +11,7 @@ require (
 	go.opentelemetry.io/collector/consumer v0.107.0
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.107.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.107.0
-	go.opentelemetry.io/collector/pdata v1.13.0
+	go.opentelemetry.io/collector/pdata v1.14.0
 	go.opentelemetry.io/collector/pdata/testdata v0.107.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/multierr v1.11.0
