diff --git a/service/defaultcomponents/defaults_test.go b/service/defaultcomponents/defaults_test.go
index 38bfc5929..ff9ffbd87 100644
--- a/service/defaultcomponents/defaults_test.go
+++ b/service/defaultcomponents/defaults_test.go
@@ -47,7 +47,6 @@ import (
 	"go.opentelemetry.io/collector/receiver/opencensusreceiver"
 	"go.opentelemetry.io/collector/receiver/otlpreceiver"
 	"go.opentelemetry.io/collector/receiver/prometheusreceiver"
-	"go.opentelemetry.io/collector/receiver/vmmetricsreceiver"
 	"go.opentelemetry.io/collector/receiver/zipkinreceiver"
 )
 
@@ -63,7 +62,6 @@ func TestDefaultComponents(t *testing.T) {
 		"prometheus":  &prometheusreceiver.Factory{},
 		"opencensus":  &opencensusreceiver.Factory{},
 		"otlp":        &otlpreceiver.Factory{},
-		"vmmetrics":   &vmmetricsreceiver.Factory{},
 		"hostmetrics": hostmetricsreceiver.NewFactory(),
 	}
 	expectedProcessors := map[configmodels.Type]component.ProcessorFactoryBase{
