diff --git a/service/internal/refconsumer/profiles_test.go b/service/internal/refconsumer/profiles_test.go
index 5d6b0d7fd..f74293589 100644
--- a/service/internal/refconsumer/profiles_test.go
+++ b/service/internal/refconsumer/profiles_test.go
@@ -11,16 +11,16 @@ import (
 
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/featuregate"
-	"go.opentelemetry.io/collector/internal/telemetry"
 	"go.opentelemetry.io/collector/pdata/testdata"
 	"go.opentelemetry.io/collector/pdata/xpdata/pref"
+	"go.opentelemetry.io/collector/service/internal/metadata"
 )
 
 func TestProfilesNopWhenGateDisabled(t *testing.T) {
 	initial := pref.UseProtoPooling.IsEnabled()
 	require.NoError(t, featuregate.GlobalRegistry().Set(pref.UseProtoPooling.ID(), false))
 	t.Cleanup(func() {
-		require.NoError(t, featuregate.GlobalRegistry().Set(telemetry.NewPipelineTelemetryGate.ID(), initial))
+		require.NoError(t, featuregate.GlobalRegistry().Set(metadata.TelemetryNewPipelineTelemetryFeatureGate.ID(), initial))
 	})
 
 	refCons := NewProfiles(consumertest.NewNop())
@@ -34,7 +34,7 @@ func TestProfiles(t *testing.T) {
 	initial := pref.UseProtoPooling.IsEnabled()
 	require.NoError(t, featuregate.GlobalRegistry().Set(pref.UseProtoPooling.ID(), true))
 	t.Cleanup(func() {
-		require.NoError(t, featuregate.GlobalRegistry().Set(telemetry.NewPipelineTelemetryGate.ID(), initial))
+		require.NoError(t, featuregate.GlobalRegistry().Set(metadata.TelemetryNewPipelineTelemetryFeatureGate.ID(), initial))
 	})
 
 	refCons := NewProfiles(consumertest.NewNop())
