diff --git a/observability/observability_test.go b/observability/observability_test.go
index fd766b392..3bf368472 100644
--- a/observability/observability_test.go
+++ b/observability/observability_test.go
@@ -30,7 +30,8 @@ const (
 )
 
 func TestTracePieplineRecordedMetrics(t *testing.T) {
-	defer observabilitytest.SetupRecordedMetricsTest()()
+	doneFn := observabilitytest.SetupRecordedMetricsTest()
+	defer doneFn()
 
 	receiverCtx := observability.ContextWithReceiverName(context.Background(), receiverName)
 	observability.RecordTraceReceiverMetrics(receiverCtx, 17, 13)
