diff --git a/consumer/pdata/generated_trace.go b/consumer/pdata/generated_trace.go
index 7a2a1e9e8..5e5256ab3 100644
--- a/consumer/pdata/generated_trace.go
+++ b/consumer/pdata/generated_trace.go
@@ -523,23 +523,23 @@ func (ms Span) SetKind(v SpanKind) {
 	(*ms.orig).Kind = otlptrace.Span_SpanKind(v)
 }
 
-// StartTime returns the starttime associated with this Span.
-func (ms Span) StartTime() Timestamp {
+// StartTimestamp returns the starttimestamp associated with this Span.
+func (ms Span) StartTimestamp() Timestamp {
 	return Timestamp((*ms.orig).StartTimeUnixNano)
 }
 
-// SetStartTime replaces the starttime associated with this Span.
-func (ms Span) SetStartTime(v Timestamp) {
+// SetStartTimestamp replaces the starttimestamp associated with this Span.
+func (ms Span) SetStartTimestamp(v Timestamp) {
 	(*ms.orig).StartTimeUnixNano = uint64(v)
 }
 
-// EndTime returns the endtime associated with this Span.
-func (ms Span) EndTime() Timestamp {
+// EndTimestamp returns the endtimestamp associated with this Span.
+func (ms Span) EndTimestamp() Timestamp {
 	return Timestamp((*ms.orig).EndTimeUnixNano)
 }
 
-// SetEndTime replaces the endtime associated with this Span.
-func (ms Span) SetEndTime(v Timestamp) {
+// SetEndTimestamp replaces the endtimestamp associated with this Span.
+func (ms Span) SetEndTimestamp(v Timestamp) {
 	(*ms.orig).EndTimeUnixNano = uint64(v)
 }
 
@@ -601,8 +601,8 @@ func (ms Span) CopyTo(dest Span) {
 	dest.SetParentSpanID(ms.ParentSpanID())
 	dest.SetName(ms.Name())
 	dest.SetKind(ms.Kind())
-	dest.SetStartTime(ms.StartTime())
-	dest.SetEndTime(ms.EndTime())
+	dest.SetStartTimestamp(ms.StartTimestamp())
+	dest.SetEndTimestamp(ms.EndTimestamp())
 	ms.Attributes().CopyTo(dest.Attributes())
 	dest.SetDroppedAttributesCount(ms.DroppedAttributesCount())
 	ms.Events().CopyTo(dest.Events())
