diff --git a/consumer/pdata/metric_test.go b/consumer/pdata/metric_test.go
index 6511f7d80..cd958ac1a 100644
--- a/consumer/pdata/metric_test.go
+++ b/consumer/pdata/metric_test.go
@@ -342,12 +342,12 @@ func TestOtlpToInternalReadOnly(t *testing.T) {
 	int64DataPoints := metricInt.IntGauge().DataPoints()
 	assert.EqualValues(t, 2, int64DataPoints.Len())
 	// First point
-	assert.EqualValues(t, startTime, int64DataPoints.At(0).StartTime())
+	assert.EqualValues(t, startTime, int64DataPoints.At(0).StartTimestamp())
 	assert.EqualValues(t, endTime, int64DataPoints.At(0).Timestamp())
 	assert.EqualValues(t, 123, int64DataPoints.At(0).Value())
 	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key0": "value0"}), int64DataPoints.At(0).LabelsMap())
 	// Second point
-	assert.EqualValues(t, startTime, int64DataPoints.At(1).StartTime())
+	assert.EqualValues(t, startTime, int64DataPoints.At(1).StartTimestamp())
 	assert.EqualValues(t, endTime, int64DataPoints.At(1).Timestamp())
 	assert.EqualValues(t, 456, int64DataPoints.At(1).Value())
 	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key1": "value1"}), int64DataPoints.At(1).LabelsMap())
@@ -363,12 +363,12 @@ func TestOtlpToInternalReadOnly(t *testing.T) {
 	doubleDataPoints := dsd.DataPoints()
 	assert.EqualValues(t, 2, doubleDataPoints.Len())
 	// First point
-	assert.EqualValues(t, startTime, doubleDataPoints.At(0).StartTime())
+	assert.EqualValues(t, startTime, doubleDataPoints.At(0).StartTimestamp())
 	assert.EqualValues(t, endTime, doubleDataPoints.At(0).Timestamp())
 	assert.EqualValues(t, 123.1, doubleDataPoints.At(0).Value())
 	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key0": "value0"}), doubleDataPoints.At(0).LabelsMap())
 	// Second point
-	assert.EqualValues(t, startTime, doubleDataPoints.At(1).StartTime())
+	assert.EqualValues(t, startTime, doubleDataPoints.At(1).StartTimestamp())
 	assert.EqualValues(t, endTime, doubleDataPoints.At(1).Timestamp())
 	assert.EqualValues(t, 456.1, doubleDataPoints.At(1).Value())
 	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key1": "value1"}), doubleDataPoints.At(1).LabelsMap())
@@ -384,13 +384,13 @@ func TestOtlpToInternalReadOnly(t *testing.T) {
 	histogramDataPoints := dhd.DataPoints()
 	assert.EqualValues(t, 2, histogramDataPoints.Len())
 	// First point
-	assert.EqualValues(t, startTime, histogramDataPoints.At(0).StartTime())
+	assert.EqualValues(t, startTime, histogramDataPoints.At(0).StartTimestamp())
 	assert.EqualValues(t, endTime, histogramDataPoints.At(0).Timestamp())
 	assert.EqualValues(t, []float64{1, 2}, histogramDataPoints.At(0).ExplicitBounds())
 	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key0": "value0"}), histogramDataPoints.At(0).LabelsMap())
 	assert.EqualValues(t, []uint64{10, 15, 1}, histogramDataPoints.At(0).BucketCounts())
 	// Second point
-	assert.EqualValues(t, startTime, histogramDataPoints.At(1).StartTime())
+	assert.EqualValues(t, startTime, histogramDataPoints.At(1).StartTimestamp())
 	assert.EqualValues(t, endTime, histogramDataPoints.At(1).Timestamp())
 	assert.EqualValues(t, []float64{1}, histogramDataPoints.At(1).ExplicitBounds())
 	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key1": "value1"}), histogramDataPoints.At(1).LabelsMap())
@@ -458,8 +458,8 @@ func TestOtlpToFromInternalIntGaugeMutating(t *testing.T) {
 	igd.DataPoints().Resize(1)
 	assert.EqualValues(t, 1, igd.DataPoints().Len())
 	int64DataPoints := igd.DataPoints()
-	int64DataPoints.At(0).SetStartTime(Timestamp(startTime + 1))
-	assert.EqualValues(t, startTime+1, int64DataPoints.At(0).StartTime())
+	int64DataPoints.At(0).SetStartTimestamp(Timestamp(startTime + 1))
+	assert.EqualValues(t, startTime+1, int64DataPoints.At(0).StartTimestamp())
 	int64DataPoints.At(0).SetTimestamp(Timestamp(endTime + 1))
 	assert.EqualValues(t, endTime+1, int64DataPoints.At(0).Timestamp())
 	int64DataPoints.At(0).SetValue(124)
@@ -538,8 +538,8 @@ func TestOtlpToFromInternalDoubleSumMutating(t *testing.T) {
 	dsd.DataPoints().Resize(1)
 	assert.EqualValues(t, 1, dsd.DataPoints().Len())
 	doubleDataPoints := dsd.DataPoints()
-	doubleDataPoints.At(0).SetStartTime(Timestamp(startTime + 1))
-	assert.EqualValues(t, startTime+1, doubleDataPoints.At(0).StartTime())
+	doubleDataPoints.At(0).SetStartTimestamp(Timestamp(startTime + 1))
+	assert.EqualValues(t, startTime+1, doubleDataPoints.At(0).StartTimestamp())
 	doubleDataPoints.At(0).SetTimestamp(Timestamp(endTime + 1))
 	assert.EqualValues(t, endTime+1, doubleDataPoints.At(0).Timestamp())
 	doubleDataPoints.At(0).SetValue(124.1)
@@ -619,8 +619,8 @@ func TestOtlpToFromInternalHistogramMutating(t *testing.T) {
 	dhd.DataPoints().Resize(1)
 	assert.EqualValues(t, 1, dhd.DataPoints().Len())
 	histogramDataPoints := dhd.DataPoints()
-	histogramDataPoints.At(0).SetStartTime(Timestamp(startTime + 1))
-	assert.EqualValues(t, startTime+1, histogramDataPoints.At(0).StartTime())
+	histogramDataPoints.At(0).SetStartTimestamp(Timestamp(startTime + 1))
+	assert.EqualValues(t, startTime+1, histogramDataPoints.At(0).StartTimestamp())
 	histogramDataPoints.At(0).SetTimestamp(Timestamp(endTime + 1))
 	assert.EqualValues(t, endTime+1, histogramDataPoints.At(0).Timestamp())
 	histogramDataPoints.At(0).LabelsMap().Delete("key0")
