diff --git a/internal/tools/tools.go b/internal/tools/tools.go
index 06f143c37..1abe8027c 100644
--- a/internal/tools/tools.go
+++ b/internal/tools/tools.go
@@ -11,6 +11,7 @@ package tools // import "go.opentelemetry.io/collector/internal/tools"
 // This ensures that all systems use the same version of tools in addition to regular dependencies.
 
 import (
+	_ "github.com/Antonboom/testifylint"
 	_ "github.com/a8m/envsubst/cmd/envsubst"
 	_ "github.com/client9/misspell/cmd/misspell"
 	_ "github.com/golangci/golangci-lint/cmd/golangci-lint"
