diff --git a/cmd/mdatagen/internal/metric.go b/cmd/mdatagen/internal/metric.go
index c1bcf9bf1..0b9a26bf2 100644
--- a/cmd/mdatagen/internal/metric.go
+++ b/cmd/mdatagen/internal/metric.go
@@ -12,6 +12,7 @@ import (
 	"golang.org/x/text/cases"
 	"golang.org/x/text/language"
 
+	"go.opentelemetry.io/collector/cmd/mdatagen/internal/helpers"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/pdata/pmetric"
@@ -22,11 +23,11 @@ var reNonAlnum = regexp.MustCompile(`[^a-z0-9_]+`)
 type MetricName string
 
 func (mn MetricName) Render() (string, error) {
-	return FormatIdentifier(string(mn), true)
+	return helpers.FormatIdentifier(string(mn), true)
 }
 
 func (mn MetricName) RenderUnexported() (string, error) {
-	return FormatIdentifier(string(mn), false)
+	return helpers.FormatIdentifier(string(mn), false)
 }
 
 type Metric struct {
