diff --git a/pdata/xpdata/request/traces_request.go b/pdata/xpdata/request/traces_request.go
index ba06cffa3..68fc6d1e1 100644
--- a/pdata/xpdata/request/traces_request.go
+++ b/pdata/xpdata/request/traces_request.go
@@ -7,8 +7,6 @@ import (
 	"context"
 	"fmt"
 
-	"google.golang.org/protobuf/proto"
-
 	"go.opentelemetry.io/collector/pdata/internal"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	reqint "go.opentelemetry.io/collector/pdata/xpdata/request/internal"
@@ -23,7 +21,7 @@ func MarshalTraces(ctx context.Context, ld ptrace.Traces) ([]byte, error) {
 		TracesData:     &otlpTraces,
 		RequestContext: &rc,
 	}
-	return proto.Marshal(&tr)
+	return tr.Marshal()
 }
 
 // UnmarshalTraces unmarshals a byte slice into ptrace.Traces and the context.
@@ -32,7 +30,7 @@ func UnmarshalTraces(buf []byte) (context.Context, ptrace.Traces, error) {
 		return context.Background(), ptrace.Traces{}, ErrInvalidFormat
 	}
 	tr := reqint.TracesRequest{}
-	if err := proto.Unmarshal(buf, &tr); err != nil {
+	if err := tr.Unmarshal(buf); err != nil {
 		return context.Background(), ptrace.Traces{}, fmt.Errorf("failed to unmarshal traces request: %w", err)
 	}
 	return decodeContext(tr.RequestContext), ptrace.Traces(internal.TracesFromProto(*tr.TracesData)), nil
