diff --git a/component/extension.go b/component/extension.go
index 4615a2139..0c4a33244 100644
--- a/component/extension.go
+++ b/component/extension.go
@@ -36,13 +36,13 @@ type Extension interface {
 type PipelineWatcher interface {
 	// Ready notifies the Extension that all pipelines were built and the
 	// receivers were started, i.e.: the service is ready to receive data
-	// (notice that it may already have received data when this method is called).
+	// (note that it may already have received data when this method is called).
 	Ready() error
 
 	// NotReady notifies the Extension that all receivers are about to be stopped,
 	// i.e.: pipeline receivers will not accept new data.
 	// This is sent before receivers are stopped, so the Extension can take any
-	// appropriate action before that happens.
+	// appropriate actions before that happens.
 	NotReady() error
 }
 
@@ -65,7 +65,7 @@ type ExtensionFactory interface {
 	// configuration and should not cause side-effects that prevent the creation
 	// of multiple instances of the Extension.
 	// The object returned by this method needs to pass the checks implemented by
-	// 'configcheck.ValidateConfig'. It is recommended to have such check in the
+	// 'configcheck.ValidateConfig'. It is recommended to have these checks in the
 	// tests of any implementation of the Factory interface.
 	CreateDefaultConfig() config.Extension
 
