diff --git a/exporter/exporterhelper/internal/requesttest/sink.go b/exporter/exporterhelper/internal/requesttest/sink.go
index ef4a1a7b7..5ff2d2d28 100644
--- a/exporter/exporterhelper/internal/requesttest/sink.go
+++ b/exporter/exporterhelper/internal/requesttest/sink.go
@@ -18,6 +18,7 @@ func NewSink() *Sink {
 type Sink struct {
 	requestsCount int
 	itemsCount    int
+	bytesCount    int
 	mu            sync.Mutex
 	exportErr     error
 }
@@ -48,6 +49,7 @@ func (s *Sink) Export(ctx context.Context, req request.Request) error {
 	}
 	s.requestsCount++
 	s.itemsCount += r.Items
+	s.bytesCount += r.Bytes
 	return nil
 }
 
@@ -68,3 +70,9 @@ func (s *Sink) ItemsCount() int {
 	defer s.mu.Unlock()
 	return s.itemsCount
 }
+
+func (s *Sink) BytesCount() int {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.bytesCount
+}
