diff --git a/internal/data/trace.go b/internal/data/trace.go
index a7c2f6df6..4485c7ab3 100644
--- a/internal/data/trace.go
+++ b/internal/data/trace.go
@@ -15,6 +15,7 @@
 package data
 
 import (
+	"github.com/golang/protobuf/proto"
 	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
 )
 
@@ -33,8 +34,8 @@ func TraceDataFromOtlp(orig []*otlptrace.ResourceSpans) TraceData {
 }
 
 // TraceDataToOtlp converts the internal TraceData to the OTLP.
-func TraceDataToOtlp(md TraceData) []*otlptrace.ResourceSpans {
-	return *md.orig
+func TraceDataToOtlp(td TraceData) []*otlptrace.ResourceSpans {
+	return *td.orig
 }
 
 // NewTraceData creates a new TraceData.
@@ -43,6 +44,17 @@ func NewTraceData() TraceData {
 	return TraceData{&orig}
 }
 
+// Clone returns a copy of TraceData.
+func (td TraceData) Clone() TraceData {
+	otlp := TraceDataToOtlp(td)
+	resourceSpansClones := make([]*otlptrace.ResourceSpans, 0, len(otlp))
+	for _, resourceSpans := range otlp {
+		resourceSpansClones = append(resourceSpansClones,
+			proto.Clone(resourceSpans).(*otlptrace.ResourceSpans))
+	}
+	return TraceDataFromOtlp(resourceSpansClones)
+}
+
 // SpanCount calculates the total number of spans.
 func (td TraceData) SpanCount() int {
 	spanCount := 0
