diff --git a/consumer/pdata/doc.go b/consumer/pdata/doc.go
index 4b4ca0b04..be88c2a78 100644
--- a/consumer/pdata/doc.go
+++ b/consumer/pdata/doc.go
@@ -22,7 +22,7 @@
 // protocol. Note that the underlying data structure is kept private so that in the
 // future we are free to make changes to it to make more optimal.
 //
-// Most of internal data structures must be created via New* functions. Zero-initialized
+// Most of the internal data structures must be created via New* functions. Zero-initialized
 // structures in most cases are not valid (read comments for each struct to know if it
 // is the case). This is a slight deviation from idiomatic Go to avoid unnecessary
 // pointer checks in dozens of functions which assume the invariant that "orig" member
