diff --git a/internal/iruntime/mem_info.go b/internal/iruntime/mem_info.go
index fc36fccf7..59e532065 100644
--- a/internal/iruntime/mem_info.go
+++ b/internal/iruntime/mem_info.go
@@ -4,7 +4,7 @@
 package iruntime // import "go.opentelemetry.io/collector/internal/iruntime"
 
 import (
-	"github.com/shirou/gopsutil/v3/mem"
+	"github.com/shirou/gopsutil/v4/mem"
 )
 
 // readMemInfo returns the total memory
