diff --git a/exporter/fileexporter/config.go b/exporter/fileexporter/config.go
index 60e0f3dce..9ae876279 100644
--- a/exporter/fileexporter/config.go
+++ b/exporter/fileexporter/config.go
@@ -15,7 +15,7 @@
 package fileexporter
 
 import (
-	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/configmodels"
 )
 
 // Config defines configuration for file exporter.
