diff --git a/processor/queuedprocessor/config_test.go b/processor/queuedprocessor/config_test.go
index 0e8523187..e1e38ef6d 100644
--- a/processor/queuedprocessor/config_test.go
+++ b/processor/queuedprocessor/config_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
