diff --git a/translator/internaldata/resource_to_oc.go b/translator/internaldata/resource_to_oc.go
index 91d1f9394..aa5ac5b7b 100644
--- a/translator/internaldata/resource_to_oc.go
+++ b/translator/internaldata/resource_to_oc.go
@@ -23,8 +23,8 @@ import (
 	"go.opencensus.io/resource/resourcekeys"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
-	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/internal/occonventions"
+	"go.opentelemetry.io/collector/model/pdata"
 	"go.opentelemetry.io/collector/translator/conventions"
 	tracetranslator "go.opentelemetry.io/collector/translator/trace"
 )
