diff --git a/exporter/exporterhelper/metrics.go b/exporter/exporterhelper/metrics.go
index 251668231..6ca7a2a6f 100644
--- a/exporter/exporterhelper/metrics.go
+++ b/exporter/exporterhelper/metrics.go
@@ -32,14 +32,11 @@ var (
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
 func NewMetricsQueueBatchSettings() QueueBatchSettings {
 	return QueueBatchSettings{
-		Encoding: metricsEncoding{},
-		Sizers: map[RequestSizerType]request.Sizer[Request]{
-			RequestSizerTypeRequests: NewRequestsSizer(),
-			RequestSizerTypeItems:    request.NewItemsSizer(),
-			RequestSizerTypeBytes: request.BaseSizer{
-				SizeofFunc: func(req request.Request) int64 {
-					return int64(metricsMarshaler.MetricsSize(req.(*metricsRequest).md))
-				},
+		Encoding:   metricsEncoding{},
+		ItemsSizer: request.NewItemsSizer(),
+		BytesSizer: request.BaseSizer{
+			SizeofFunc: func(req request.Request) int64 {
+				return int64(metricsMarshaler.MetricsSize(req.(*metricsRequest).md))
 			},
 		},
 	}
