diff --git a/component/component.go b/component/component.go
index 40267a416..7e8c1426d 100644
--- a/component/component.go
+++ b/component/component.go
@@ -7,16 +7,6 @@ package component // import "go.opentelemetry.io/collector/component"
 
 import (
 	"context"
-
-	"go.opentelemetry.io/collector/pipeline"
-)
-
-var (
-	// ErrDataTypeIsNotSupported can be returned by receiver, exporter, processor or connector factory funcs that create the
-	// Component if the particular telemetry data type is not supported by the receiver, exporter, processor or connector factory.
-	//
-	// Deprecated: [v0.110.0] Use pipeline.ErrSignalNotSupported instead
-	ErrDataTypeIsNotSupported = pipeline.ErrSignalNotSupported
 )
 
 // Component is either a receiver, exporter, processor, connector, or an extension.
