diff --git a/processor/processorhelper/logs_test.go b/processor/processorhelper/logs_test.go
index 82dbfd666..8ca9f2436 100644
--- a/processor/processorhelper/logs_test.go
+++ b/processor/processorhelper/logs_test.go
@@ -93,13 +93,11 @@ func TestLogsConcurrency(t *testing.T) {
 
 	var wg sync.WaitGroup
 	for range 10 {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
+		wg.Go(func() {
 			for range 10000 {
 				assert.NoError(t, lp.ConsumeLogs(context.Background(), incomingLogs))
 			}
-		}()
+		})
 	}
 	wg.Wait()
 	assert.NoError(t, lp.Shutdown(context.Background()))
