diff --git a/pdata/pprofile/generated_resourceprofilesslice_test.go b/pdata/pprofile/generated_resourceprofilesslice_test.go
index 7ff599194..2ae723d1d 100644
--- a/pdata/pprofile/generated_resourceprofilesslice_test.go
+++ b/pdata/pprofile/generated_resourceprofilesslice_test.go
@@ -141,6 +141,14 @@ func TestResourceProfilesSlice_RemoveIf(t *testing.T) {
 	assert.Equal(t, 5, filtered.Len())
 }
 
+func TestResourceProfilesSlice_RemoveIfAll(t *testing.T) {
+	got := generateTestResourceProfilesSlice()
+	got.RemoveIf(func(el ResourceProfiles) bool {
+		return true
+	})
+	assert.Equal(t, 0, got.Len())
+}
+
 func TestResourceProfilesSliceAll(t *testing.T) {
 	ms := generateTestResourceProfilesSlice()
 	assert.NotEmpty(t, ms.Len())
