diff --git a/pdata/internal/generated_wrapper_resource.go b/pdata/internal/generated_wrapper_resource.go
index 115862738..0f13655b7 100644
--- a/pdata/internal/generated_wrapper_resource.go
+++ b/pdata/internal/generated_wrapper_resource.go
@@ -34,8 +34,7 @@ func NewResource(orig *otlpresource.Resource, state *State) Resource {
 func GenerateTestResource() Resource {
 	orig := otlpresource.Resource{}
 	FillOrigTestResource(&orig)
-	state := StateMutable
-	return NewResource(&orig, &state)
+	return NewResource(&orig, NewState())
 }
 
 func NewOrigResource() otlpresource.Resource {
