diff --git a/processor/processortest/nop_processor_test.go b/processor/processortest/nop_processor_test.go
index 44f581bd2..77e9131ed 100644
--- a/processor/processortest/nop_processor_test.go
+++ b/processor/processortest/nop_processor_test.go
@@ -22,7 +22,7 @@ import (
 func TestNewNopFactory(t *testing.T) {
 	factory := NewNopFactory()
 	require.NotNil(t, factory)
-	assert.Equal(t, component.Type("nop"), factory.Type())
+	assert.Equal(t, component.MustNewType("nop"), factory.Type())
 	cfg := factory.CreateDefaultConfig()
 	assert.Equal(t, &nopConfig{}, cfg)
 
@@ -55,7 +55,7 @@ func TestNewNopBuilder(t *testing.T) {
 	factory := NewNopFactory()
 	cfg := factory.CreateDefaultConfig()
 	set := NewNopCreateSettings()
-	set.ID = component.NewID(typeStr)
+	set.ID = component.NewID(nopType)
 
 	traces, err := factory.CreateTracesProcessor(context.Background(), set, cfg, consumertest.NewNop())
 	require.NoError(t, err)
