diff --git a/receiver/otlpreceiver/internal/metrics/otlp.go b/receiver/otlpreceiver/internal/metrics/otlp.go
index 514910958..9e5abc257 100644
--- a/receiver/otlpreceiver/internal/metrics/otlp.go
+++ b/receiver/otlpreceiver/internal/metrics/otlp.go
@@ -48,16 +48,16 @@ func New(id config.ComponentID, nextConsumer consumer.Metrics, set component.Rec
 }
 
 // Export implements the service Export metrics func.
-func (r *Receiver) Export(ctx context.Context, req pmetricotlp.Request) (pmetricotlp.Response, error) {
+func (r *Receiver) Export(ctx context.Context, req pmetricotlp.ExportRequest) (pmetricotlp.ExportResponse, error) {
 	md := req.Metrics()
 	dataPointCount := md.DataPointCount()
 	if dataPointCount == 0 {
-		return pmetricotlp.NewResponse(), nil
+		return pmetricotlp.NewExportResponse(), nil
 	}
 
 	ctx = r.obsrecv.StartMetricsOp(ctx)
 	err := r.nextConsumer.ConsumeMetrics(ctx, md)
 	r.obsrecv.EndMetricsOp(ctx, dataFormatProtobuf, dataPointCount, err)
 
-	return pmetricotlp.NewResponse(), err
+	return pmetricotlp.NewExportResponse(), err
 }
