diff --git a/internal/collector/zipkin/scribe/receiver.go b/internal/collector/zipkin/scribe/receiver.go
index a0105a410..b3fa00c93 100644
--- a/internal/collector/zipkin/scribe/receiver.go
+++ b/internal/collector/zipkin/scribe/receiver.go
@@ -30,18 +30,18 @@ import (
 )
 
 // Start starts the Zipkin Scribe receiver endpoint.
-func Start(logger *zap.Logger, v *viper.Viper, traceConsumer consumer.TraceConsumer) (receiver.TraceReceiver, error) {
+func Start(logger *zap.Logger, v *viper.Viper, traceConsumer consumer.TraceConsumer, asyncErrorChan chan<- error) (receiver.TraceReceiver, error) {
 	rOpts, err := builder.NewDefaultZipkinScribeReceiverCfg().InitFromViper(v)
 	if err != nil {
 		return nil, err
 	}
 
-	sr, err := zipkinscribereceiver.NewReceiver(rOpts.Address, rOpts.Port, rOpts.Category)
+	sr, err := zipkinscribereceiver.NewReceiver(rOpts.Address, rOpts.Port, rOpts.Category, traceConsumer)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to create the Zipkin Scribe receiver: %v", err)
 	}
 
-	if err := sr.StartTraceReception(context.Background(), traceConsumer); err != nil {
+	if err := sr.StartTraceReception(context.Background(), asyncErrorChan); err != nil {
 		return nil, fmt.Errorf("Cannot start Zipkin Scribe receiver %+v: %v", rOpts, err)
 	}
 
