diff --git a/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl b/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
index 97816a9a2..a3dd9c12c 100644
--- a/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
+++ b/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
@@ -21,7 +21,7 @@ import (
 )
 
 type Telemetry struct {
-	componenttest.Telemetry
+	*componenttest.Telemetry
 }
 
 func SetupTelemetry(opts ...componenttest.TelemetryOption) Telemetry {
@@ -50,9 +50,18 @@ func (tt *Telemetry) AssertMetrics(t *testing.T, expected []metricdata.Metrics,
 	require.Equal(t, len(expected), lenMetrics(md))
 }
 
+{{ if or isConnector isExporter isExtension isProcessor isReceiver isScraper }}
+func NewSettings(tt *componenttest.Telemetry) {{ .Status.Class }}.Settings {
+set := {{ .Status.Class }}test.NewNopSettings()
+set.ID = component.NewID(component.MustNewType("{{ .Type }}"))
+set.TelemetrySettings = tt.NewTelemetrySettings()
+return set
+}
+{{- end }}
+
 {{ range $name, $metric := .Telemetry.Metrics }}
 
-func AssertEqual{{ $name.Render }}(t *testing.T, tt componenttest.Telemetry, dps []metricdata.{{- if eq $metric.Data.Type "Histogram" }} {{$metric.Data.Type}} {{- end }}DataPoint[{{ $metric.Data.BasicType }}], opts ...metricdatatest.Option) {
+func AssertEqual{{ $name.Render }}(t *testing.T, tt *componenttest.Telemetry, dps []metricdata.{{- if eq $metric.Data.Type "Histogram" }} {{$metric.Data.Type}} {{- end }}DataPoint[{{ $metric.Data.BasicType }}], opts ...metricdatatest.Option) {
 	want := metricdata.Metrics{
 		Name:        "otelcol_{{ $name }}",
 		Description: "{{ $metric.Description }}{{ $metric.Stability }}",
