diff --git a/consumer/pdata/common.go b/consumer/pdata/common.go
index a1b45ae10..079228758 100644
--- a/consumer/pdata/common.go
+++ b/consumer/pdata/common.go
@@ -280,6 +280,10 @@ func (a AttributeValue) CopyTo(dest AttributeValue) {
 
 // Equal checks for equality, it returns true if the objects are equal otherwise false.
 func (a AttributeValue) Equal(av AttributeValue) bool {
+	if a.orig == av.orig {
+		return true
+	}
+
 	if a.orig.Value == nil || av.orig.Value == nil {
 		return a.orig.Value == av.orig.Value
 	}
@@ -293,8 +297,30 @@ func (a AttributeValue) Equal(av AttributeValue) bool {
 		return v.IntValue == av.orig.GetIntValue()
 	case *otlpcommon.AnyValue_DoubleValue:
 		return v.DoubleValue == av.orig.GetDoubleValue()
+	case *otlpcommon.AnyValue_ArrayValue:
+		vv := v.ArrayValue.GetValues()
+		avv := av.orig.GetArrayValue().GetValues()
+		if len(vv) != len(avv) {
+			return false
+		}
+
+		for i, val := range avv {
+			val := val
+			av := newAttributeValue(&vv[i])
+
+			// According to the specification, array values must be scalar.
+			if avType := av.Type(); avType == AttributeValueARRAY || avType == AttributeValueMAP {
+				return false
+			}
+
+			if !av.Equal(newAttributeValue(&val)) {
+				return false
+			}
+		}
+		return true
 	}
-	// TODO: handle MAP and ARRAY data types.
+
+	// TODO: handle MAP data type
 	return false
 }
 
