diff --git a/consumer/consumererror/consumererrorprofiles/go.mod b/consumer/consumererror/consumererrorprofiles/go.mod
index 517a94d17..68c35440d 100644
--- a/consumer/consumererror/consumererrorprofiles/go.mod
+++ b/consumer/consumererror/consumererrorprofiles/go.mod
@@ -4,7 +4,7 @@ go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/consumer v0.109.0
+	go.opentelemetry.io/collector/consumer v0.111.0
 	go.opentelemetry.io/collector/pdata/pprofile v0.111.0
 	go.opentelemetry.io/collector/pdata/testdata v0.111.0
 )
