diff --git a/extension/healthcheckextension/config.go b/extension/healthcheckextension/config.go
index 53bc96981..0937c30e0 100644
--- a/extension/healthcheckextension/config.go
+++ b/extension/healthcheckextension/config.go
@@ -31,6 +31,6 @@ type Config struct {
 
 	// TCPAddr represents a tcp endpoint address that is to publish the health
 	// check status.
-	// The default endpoint is "localhost:13133".
+	// The default endpoint is "0.0.0.0:13133".
 	TCPAddr confignet.TCPAddr `mapstructure:",squash"`
 }
