diff --git a/pdata/internal/generated_wrapper_int64slice.go b/pdata/internal/generated_wrapper_int64slice.go
index ec4df4f49..f8bf7aca1 100644
--- a/pdata/internal/generated_wrapper_int64slice.go
+++ b/pdata/internal/generated_wrapper_int64slice.go
@@ -6,32 +6,32 @@
 
 package internal
 
-type Int64Slice struct {
+type Int64SliceWrapper struct {
 	orig  *[]int64
 	state *State
 }
 
-func GetOrigInt64Slice(ms Int64Slice) *[]int64 {
+func GetInt64SliceOrig(ms Int64SliceWrapper) *[]int64 {
 	return ms.orig
 }
 
-func GetInt64SliceState(ms Int64Slice) *State {
+func GetInt64SliceState(ms Int64SliceWrapper) *State {
 	return ms.state
 }
 
-func NewInt64Slice(orig *[]int64, state *State) Int64Slice {
-	return Int64Slice{orig: orig, state: state}
+func NewInt64SliceWrapper(orig *[]int64, state *State) Int64SliceWrapper {
+	return Int64SliceWrapper{orig: orig, state: state}
 }
 
-func GenerateTestInt64Slice() Int64Slice {
-	orig := GenerateOrigTestInt64Slice()
-	return NewInt64Slice(&orig, NewState())
+func GenTestInt64SliceWrapper() Int64SliceWrapper {
+	orig := GenTestInt64Slice()
+	return NewInt64SliceWrapper(&orig, NewState())
 }
 
-func CopyOrigInt64Slice(dst, src []int64) []int64 {
+func CopyInt64Slice(dst, src []int64) []int64 {
 	return append(dst[:0], src...)
 }
 
-func GenerateOrigTestInt64Slice() []int64 {
+func GenTestInt64Slice() []int64 {
 	return []int64{1, 2, 3}
 }
