diff --git a/pdata/ptrace/generated_spaneventslice_test.go b/pdata/ptrace/generated_spaneventslice_test.go
index 965a9e6a4..874085bd1 100644
--- a/pdata/ptrace/generated_spaneventslice_test.go
+++ b/pdata/ptrace/generated_spaneventslice_test.go
@@ -28,9 +28,9 @@ func TestSpanEventSlice(t *testing.T) {
 	emptyVal := NewSpanEvent()
 	testVal := generateTestSpanEvent()
 	for i := 0; i < 7; i++ {
-		el := es.AppendEmpty()
+		es.AppendEmpty()
 		assert.Equal(t, emptyVal, es.At(i))
-		fillTestSpanEvent(el)
+		internal.FillOrigTestSpan_Event((*es.orig)[i])
 		assert.Equal(t, testVal, es.At(i))
 	}
 	assert.Equal(t, 7, es.Len())
@@ -51,26 +51,9 @@ func TestSpanEventSliceReadOnly(t *testing.T) {
 
 func TestSpanEventSlice_CopyTo(t *testing.T) {
 	dest := NewSpanEventSlice()
-	// Test CopyTo empty
-	NewSpanEventSlice().CopyTo(dest)
-	assert.Equal(t, NewSpanEventSlice(), dest)
-
-	// Test CopyTo larger slice
 	src := generateTestSpanEventSlice()
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestSpanEventSlice(), dest)
-
-	// Test CopyTo same size slice
-	src.CopyTo(dest)
-	assert.Equal(t, generateTestSpanEventSlice(), dest)
-
-	// Test CopyTo smaller size slice
-	NewSpanEventSlice().CopyTo(dest)
-	assert.Equal(t, 0, dest.Len())
-
-	// Test CopyTo larger slice with enough capacity
-	src.CopyTo(dest)
-	assert.Equal(t, generateTestSpanEventSlice(), dest)
 }
 
 func TestSpanEventSlice_EnsureCapacity(t *testing.T) {
@@ -194,15 +177,7 @@ func TestSpanEventSlice_Sort(t *testing.T) {
 }
 
 func generateTestSpanEventSlice() SpanEventSlice {
-	es := NewSpanEventSlice()
-	fillTestSpanEventSlice(es)
-	return es
-}
-
-func fillTestSpanEventSlice(es SpanEventSlice) {
-	*es.orig = make([]*otlptrace.Span_Event, 7)
-	for i := 0; i < 7; i++ {
-		(*es.orig)[i] = &otlptrace.Span_Event{}
-		fillTestSpanEvent(newSpanEvent((*es.orig)[i], es.state))
-	}
+	ms := NewSpanEventSlice()
+	*ms.orig = internal.GenerateOrigTestSpan_EventSlice()
+	return ms
 }
