diff --git a/otelcol/unmarshaler_test.go b/otelcol/unmarshaler_test.go
index 9fdb08324..8a82da874 100644
--- a/otelcol/unmarshaler_test.go
+++ b/otelcol/unmarshaler_test.go
@@ -67,8 +67,7 @@ func TestUnmarshalUnknownTopLevel(t *testing.T) {
 		"unknown_section": nil,
 	})
 	_, err = unmarshal(conf, factories)
-	require.Error(t, err)
-	assert.Contains(t, err.Error(), "'' has invalid keys: unknown_section")
+	assert.ErrorContains(t, err, "'' has invalid keys: unknown_section")
 }
 
 func TestPipelineConfigUnmarshalError(t *testing.T) {
@@ -136,8 +135,7 @@ func TestPipelineConfigUnmarshalError(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			pips := new(pipelines.ConfigWithPipelineID)
 			err := tt.conf.Unmarshal(&pips)
-			require.Error(t, err)
-			assert.Contains(t, err.Error(), tt.expectError)
+			assert.ErrorContains(t, err, tt.expectError)
 		})
 	}
 }
