diff --git a/service/internal/builder/exporters_builder.go b/service/internal/builder/exporters_builder.go
index 149e6725e..0ddd28684 100644
--- a/service/internal/builder/exporters_builder.go
+++ b/service/internal/builder/exporters_builder.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package builder
+package builder // import "go.opentelemetry.io/collector/service/internal/builder"
 
 import (
 	"context"
