diff --git a/obsreport/obsreport_receiver.go b/obsreport/obsreport_receiver.go
index 059e5ef2b..1a0a73b0f 100644
--- a/obsreport/obsreport_receiver.go
+++ b/obsreport/obsreport_receiver.go
@@ -27,83 +27,39 @@ import (
 	"go.opentelemetry.io/collector/internal/obsreportconfig/obsmetrics"
 )
 
-// startReceiveOptions has the options related to starting a receive operation.
-type startReceiveOptions struct {
-	// LongLivedCtx when true indicates that the context passed in the call
-	// outlives the individual receive operation. See WithLongLivedCtx() for
-	// more information.
-	LongLivedCtx bool
-}
-
-// StartReceiveOption function applues changes to startReceiveOptions.
-type StartReceiveOption func(*startReceiveOptions)
-
-// WithLongLivedCtx indicates that the context passed in the call outlives the
-// receive operation at hand. Typically the long lived context is associated
-// to a connection, eg.: a gRPC stream or a TCP connection, for which many
-// batches of data are received in individual operations without a corresponding
-// new context per operation.
-//
-// Example:
-//
-//    func (r *receiver) ClientConnect(ctx context.Context, rcvChan <-chan pdata.Traces) {
-//        longLivedCtx := obsreport.ReceiverContext(ctx, r.config.Name(), r.transport, "")
-//        for {
-//            // Since the context outlives the individual receive operations call obsreport using
-//            // WithLongLivedCtx().
-//            ctx := obsreport.StartTracesOp(
-//                longLivedCtx,
-//                r.config.Name(),
-//                r.transport,
-//                obsreport.WithLongLivedCtx())
-//
-//            td, ok := <-rcvChan
-//            var err error
-//            if ok {
-//                err = r.nextConsumer.ConsumeTraces(ctx, td)
-//            }
-//            obsreport.EndTracesOp(
-//                ctx,
-//                r.format,
-//                len(td.Spans),
-//                err)
-//            if !ok {
-//                break
-//            }
-//        }
-//    }
-//
-func WithLongLivedCtx() StartReceiveOption {
-	return func(opts *startReceiveOptions) {
-		opts.LongLivedCtx = true
-	}
-}
-
 // Receiver is a helper to add obersvability to a component.Receiver.
 type Receiver struct {
-	receiverID config.ComponentID
-	transport  string
+	receiverID   config.ComponentID
+	transport    string
+	longLivedCtx bool
 }
 
 // ReceiverSettings are settings for creating an Receiver.
 type ReceiverSettings struct {
 	ReceiverID config.ComponentID
 	Transport  string
+	// LongLivedCtx when true indicates that the context passed in the call
+	// outlives the individual receive operation.
+	// Typically the long lived context is associated to a connection,
+	// eg.: a gRPC stream, for which many batches of data are received in individual
+	// operations without a corresponding new context per operation.
+	LongLivedCtx bool
 }
 
 // NewReceiver creates a new Receiver.
 func NewReceiver(cfg ReceiverSettings) *Receiver {
 	return &Receiver{
-		receiverID: cfg.ReceiverID,
-		transport:  cfg.Transport,
+		receiverID:   cfg.ReceiverID,
+		transport:    cfg.Transport,
+		longLivedCtx: cfg.LongLivedCtx,
 	}
 }
 
 // StartTracesOp is called when a request is received from a client.
 // The returned context should be used in other calls to the obsreport functions
 // dealing with the same receive operation.
-func (rec *Receiver) StartTracesOp(operationCtx context.Context, opt ...StartReceiveOption) context.Context {
-	return rec.startOp(operationCtx, obsmetrics.ReceiveTraceDataOperationSuffix, opt...)
+func (rec *Receiver) StartTracesOp(operationCtx context.Context) context.Context {
+	return rec.startOp(operationCtx, obsmetrics.ReceiveTraceDataOperationSuffix)
 }
 
 // EndTracesOp completes the receive operation that was started with
@@ -120,8 +76,8 @@ func (rec *Receiver) EndTracesOp(
 // StartLogsOp is called when a request is received from a client.
 // The returned context should be used in other calls to the obsreport functions
 // dealing with the same receive operation.
-func (rec *Receiver) StartLogsOp(operationCtx context.Context, opt ...StartReceiveOption) context.Context {
-	return rec.startOp(operationCtx, obsmetrics.ReceiverLogsOperationSuffix, opt...)
+func (rec *Receiver) StartLogsOp(operationCtx context.Context) context.Context {
+	return rec.startOp(operationCtx, obsmetrics.ReceiverLogsOperationSuffix)
 }
 
 // EndLogsOp completes the receive operation that was started with
@@ -138,8 +94,8 @@ func (rec *Receiver) EndLogsOp(
 // StartMetricsOp is called when a request is received from a client.
 // The returned context should be used in other calls to the obsreport functions
 // dealing with the same receive operation.
-func (rec *Receiver) StartMetricsOp(operationCtx context.Context, opt ...StartReceiveOption) context.Context {
-	return rec.startOp(operationCtx, obsmetrics.ReceiverMetricsOperationSuffix, opt...)
+func (rec *Receiver) StartMetricsOp(operationCtx context.Context) context.Context {
+	return rec.startOp(operationCtx, obsmetrics.ReceiverMetricsOperationSuffix)
 }
 
 // EndMetricsOp completes the receive operation that was started with
@@ -171,20 +127,11 @@ func ReceiverContext(
 
 // startOp creates the span used to trace the operation. Returning
 // the updated context with the created span.
-func (rec *Receiver) startOp(
-	receiverCtx context.Context,
-	operationSuffix string,
-	opt ...StartReceiveOption,
-) context.Context {
-	var opts startReceiveOptions
-	for _, o := range opt {
-		o(&opts)
-	}
-
+func (rec *Receiver) startOp(receiverCtx context.Context, operationSuffix string) context.Context {
 	var ctx context.Context
 	var span *trace.Span
 	spanName := obsmetrics.ReceiverPrefix + rec.receiverID.String() + operationSuffix
-	if !opts.LongLivedCtx {
+	if !rec.longLivedCtx {
 		ctx, span = trace.StartSpan(receiverCtx, spanName)
 	} else {
 		// Since the receiverCtx is long lived do not use it to start the span.
