diff --git a/vendor/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext/app_types.go b/vendor/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext/app_types.go
new file mode 100644
index 000000000..7fb18e586
--- /dev/null
+++ b/vendor/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext/app_types.go
@@ -0,0 +1,19 @@
+package ext // import "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+
+const (
+	// AppTypeWeb specifies the Web span type and can be used as a tag value
+	// for a span's SpanType tag.
+	AppTypeWeb = "web"
+
+	// AppTypeDB specifies the DB span type and can be used as a tag value
+	// for a span's SpanType tag.
+	AppTypeDB = "db"
+
+	// AppTypeCache specifies the Cache span type and can be used as a tag value
+	// for a span's SpanType tag.
+	AppTypeCache = "cache"
+
+	// AppTypeRPC specifies the RPC span type and can be used as a tag value
+	// for a span's SpanType tag.
+	AppTypeRPC = "rpc"
+)
