diff --git a/pdata/xpdata/request/internal/request.proto b/pdata/xpdata/request/internal/request.proto
index 39f9c6154..347a00b89 100644
--- a/pdata/xpdata/request/internal/request.proto
+++ b/pdata/xpdata/request/internal/request.proto
@@ -24,6 +24,9 @@ message SpanContext {
 // RequestContext represents metadata associated with a telemetry export request.
 message RequestContext {
   SpanContext span_context = 1;
+
+  // ClientMetadata contains additional metadata about the client making the request.
+  repeated opentelemetry.proto.common.v1.KeyValue client_metadata = 2 [ (gogoproto.nullable) = false ];
 }
 
 // The following messages are wrappers around standard OpenTelemetry data types.
