diff --git a/exporter/exporterhelper/queued_retry_experimental.go b/exporter/exporterhelper/queued_retry_experimental.go
index 8e91cf34a..1949f23b2 100644
--- a/exporter/exporterhelper/queued_retry_experimental.go
+++ b/exporter/exporterhelper/queued_retry_experimental.go
@@ -200,7 +200,7 @@ func (qrs *queuedRetrySender) start(ctx context.Context, host component.Host) er
 
 	// Start reporting queue length metric
 	if qrs.cfg.Enabled {
-		err := queueSizeGauge.UpsertEntry(func() int64 {
+		err := globalInstruments.queueSize.UpsertEntry(func() int64 {
 			return int64(qrs.queue.Size())
 		}, metricdata.NewLabelValue(qrs.fullName()))
 		if err != nil {
@@ -215,7 +215,7 @@ func (qrs *queuedRetrySender) start(ctx context.Context, host component.Host) er
 func (qrs *queuedRetrySender) shutdown() {
 	// Cleanup queue metrics reporting
 	if qrs.cfg.Enabled {
-		_ = queueSizeGauge.UpsertEntry(func() int64 {
+		_ = globalInstruments.queueSize.UpsertEntry(func() int64 {
 			return int64(0)
 		}, metricdata.NewLabelValue(qrs.fullName()))
 	}
