diff --git a/exporter/exporterhelper/traces.go b/exporter/exporterhelper/traces.go
index cb69719c8..5b6618496 100644
--- a/exporter/exporterhelper/traces.go
+++ b/exporter/exporterhelper/traces.go
@@ -114,7 +114,7 @@ func NewTracesExporter(
 		req := newTracesRequest(ctx, td, pusher)
 		err := be.sender.send(req)
 		if errors.Is(err, errSendingQueueIsFull) {
-			be.obsrep.recordTracesEnqueueFailure(req.context(), req.count())
+			be.obsrep.recordTracesEnqueueFailure(req.context(), int64(req.count()))
 		}
 		return err
 	}, bs.consumerOptions...)
