diff --git a/exporter/otlpexporter/otlp_test.go b/exporter/otlpexporter/otlp_test.go
index 5ff5f5360..eff256936 100644
--- a/exporter/otlpexporter/otlp_test.go
+++ b/exporter/otlpexporter/otlp_test.go
@@ -172,12 +172,12 @@ func TestSendData(t *testing.T) {
 				SpanId:    spanID,
 				Name:      &octrace.TruncatableString{Value: "operationB"},
 				Kind:      octrace.Span_SERVER,
-				StartTime: internal.UnixnanoToTimestamp(data.TimestampUnixNano(unixnanos)),
-				EndTime:   internal.UnixnanoToTimestamp(data.TimestampUnixNano(unixnanos)),
+				StartTime: internal.UnixNanoToTimestamp(data.TimestampUnixNano(unixnanos)),
+				EndTime:   internal.UnixNanoToTimestamp(data.TimestampUnixNano(unixnanos)),
 				TimeEvents: &octrace.Span_TimeEvents{
 					TimeEvent: []*octrace.Span_TimeEvent{
 						{
-							Time: internal.UnixnanoToTimestamp(data.TimestampUnixNano(unixnanos)),
+							Time: internal.UnixNanoToTimestamp(data.TimestampUnixNano(unixnanos)),
 							Value: &octrace.Span_TimeEvent_Annotation_{
 								Annotation: &octrace.Span_TimeEvent_Annotation{
 									Description: &octrace.TruncatableString{Value: "event1"},
@@ -245,11 +245,11 @@ func TestSendData(t *testing.T) {
 								SpanId:            spanID,
 								Name:              "operationB",
 								Kind:              otlptrace.Span_SERVER,
-								StartTimeUnixnano: unixnanos,
-								EndTimeUnixnano:   unixnanos,
+								StartTimeUnixNano: unixnanos,
+								EndTimeUnixNano:   unixnanos,
 								Events: []*otlptrace.Span_Event{
 									{
-										TimeUnixnano: unixnanos,
+										TimeUnixNano: unixnanos,
 										Name:         "event1",
 										Attributes: []*otlpcommon.AttributeKeyValue{
 											{
