diff --git a/exporter/otlpexporter/go.mod b/exporter/otlpexporter/go.mod
index 25ccd8171..206d99e13 100644
--- a/exporter/otlpexporter/go.mod
+++ b/exporter/otlpexporter/go.mod
@@ -4,7 +4,7 @@ go 1.23.0
 
 require (
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector v0.119.0
+	go.opentelemetry.io/collector v0.119.1-0.20250217085205-36e046737251
 	go.opentelemetry.io/collector/component v0.119.0
 	go.opentelemetry.io/collector/component/componenttest v0.119.0
 	go.opentelemetry.io/collector/config/configauth v0.119.0
