diff --git a/cmd/mdatagen/go.mod b/cmd/mdatagen/go.mod
index ff8812914..2eed28f19 100644
--- a/cmd/mdatagen/go.mod
+++ b/cmd/mdatagen/go.mod
@@ -184,8 +184,6 @@ replace go.opentelemetry.io/collector/extension/extensionauth => ../../extension
 
 replace go.opentelemetry.io/collector/service/hostcapabilities => ../../service/hostcapabilities
 
-replace go.opentelemetry.io/collector/confmap/provider/yamlprovider => ../../confmap/provider/yamlprovider
-
 replace go.opentelemetry.io/collector/config/configtls => ../../config/configtls
 
 replace go.opentelemetry.io/collector/config/configcompression => ../../config/configcompression
