diff --git a/translator/internaldata/oc_testdata_test.go b/translator/internaldata/oc_testdata_test.go
index 1fffff247..3fdbe8bcf 100644
--- a/translator/internaldata/oc_testdata_test.go
+++ b/translator/internaldata/oc_testdata_test.go
@@ -499,7 +499,7 @@ func generateResourceWithOcNodeAndResource() pdata.Resource {
 	resource.Attributes().InitFromMap(map[string]pdata.AttributeValue{
 		conventions.OCAttributeProcessStartTime:   pdata.NewAttributeValueString("2020-02-11T20:26:00Z"),
 		conventions.AttributeHostName:             pdata.NewAttributeValueString("host1"),
-		conventions.OCAttributeProcessID:          pdata.NewAttributeValueInt(123),
+		conventions.AttributeProcessID:            pdata.NewAttributeValueInt(123),
 		conventions.AttributeTelemetrySDKVersion:  pdata.NewAttributeValueString("v2.0.1"),
 		conventions.OCAttributeExporterVersion:    pdata.NewAttributeValueString("v1.2.0"),
 		conventions.AttributeTelemetrySDKLanguage: pdata.NewAttributeValueString("cpp"),
