diff --git a/cmd/builder/internal/builder/main_test.go b/cmd/builder/internal/builder/main_test.go
index c4b510036..b50857f73 100644
--- a/cmd/builder/internal/builder/main_test.go
+++ b/cmd/builder/internal/builder/main_test.go
@@ -363,6 +363,7 @@ func TestReplaceStatementsAreComplete(t *testing.T) {
 	// Configure all components that we want to use elsewhere in these tests.
 	// This ensures the resulting go.mod file has maximum coverage of modules
 	// that exist in the Core repository.
+	usedNames := make(map[string]int)
 	cfg.Exporters, err = parseModules([]Module{
 		{
 			GoMod: "go.opentelemetry.io/collector/exporter/debugexporter v1.9999.9999",
@@ -376,7 +377,7 @@ func TestReplaceStatementsAreComplete(t *testing.T) {
 		{
 			GoMod: "go.opentelemetry.io/collector/exporter/otlphttpexporter v1.9999.9999",
 		},
-	})
+	}, usedNames)
 	require.NoError(t, err)
 	cfg.Receivers, err = parseModules([]Module{
 		{
@@ -385,13 +386,13 @@ func TestReplaceStatementsAreComplete(t *testing.T) {
 		{
 			GoMod: "go.opentelemetry.io/collector/receiver/otlpreceiver v1.9999.9999",
 		},
-	})
+	}, usedNames)
 	require.NoError(t, err)
 	cfg.Extensions, err = parseModules([]Module{
 		{
 			GoMod: "go.opentelemetry.io/collector/extension/zpagesextension v1.9999.9999",
 		},
-	})
+	}, usedNames)
 	require.NoError(t, err)
 	cfg.Processors, err = parseModules([]Module{
 		{
@@ -400,7 +401,7 @@ func TestReplaceStatementsAreComplete(t *testing.T) {
 		{
 			GoMod: "go.opentelemetry.io/collector/processor/memorylimiterprocessor v1.9999.9999",
 		},
-	})
+	}, usedNames)
 	require.NoError(t, err)
 
 	require.NoError(t, cfg.Validate())
