diff --git a/receiver/receivertest/go.mod b/receiver/receivertest/go.mod
index bf72a835b..1963ba39e 100644
--- a/receiver/receivertest/go.mod
+++ b/receiver/receivertest/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/component v0.113.0
-	go.opentelemetry.io/collector/component/componenttest v0.113.0
+	go.opentelemetry.io/collector/component/componenttest v0.0.0-20241112045949-9c4c007a12b4
 	go.opentelemetry.io/collector/consumer v0.113.0
 	go.opentelemetry.io/collector/consumer/consumererror v0.113.0
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.113.0
