diff --git a/config/configunmarshaler/defaultunmarshaler.go b/config/configunmarshaler/defaultunmarshaler.go
index 967208b69..719af4445 100644
--- a/config/configunmarshaler/defaultunmarshaler.go
+++ b/config/configunmarshaler/defaultunmarshaler.go
@@ -199,7 +199,6 @@ func unmarshalService(srvRaw map[string]interface{}) (config.Service, error) {
 	}
 
 	for id, pipeline := range srv.Pipelines {
-		pipeline.Name = id.String()
 		pipeline.InputType = config.DataType(id.Type())
 		switch pipeline.InputType {
 		case config.TracesDataType:
