diff --git a/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_config_test.go b/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_config_test.go
index cc18cc857..9eae774d6 100644
--- a/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_config_test.go
+++ b/cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_config_test.go
@@ -32,11 +32,7 @@ func TestMetricsBuilderConfig(t *testing.T) {
 						AggregationStrategy: AggregationStrategySum,
 						EnabledAttributes:   []string{"string_attr", "state", "enum_attr", "slice_attr", "map_attr"},
 					},
-					DefaultMetricToBeRemoved: MetricConfig{
-						Enabled:             true,
-						AggregationStrategy: AggregationStrategySum,
-						EnabledAttributes:   []string{},
-					},
+					DefaultMetricToBeRemoved: MetricConfig{Enabled: true},
 					MetricInputType: MetricConfig{
 						Enabled:             true,
 						AggregationStrategy: AggregationStrategySum,
@@ -79,11 +75,7 @@ func TestMetricsBuilderConfig(t *testing.T) {
 						AggregationStrategy: AggregationStrategySum,
 						EnabledAttributes:   []string{"string_attr", "state", "enum_attr", "slice_attr", "map_attr"},
 					},
-					DefaultMetricToBeRemoved: MetricConfig{
-						Enabled:             false,
-						AggregationStrategy: AggregationStrategySum,
-						EnabledAttributes:   []string{},
-					},
+					DefaultMetricToBeRemoved: MetricConfig{Enabled: false},
 					MetricInputType: MetricConfig{
 						Enabled:             false,
 						AggregationStrategy: AggregationStrategySum,
