diff --git a/semconv/semconv_test.go b/semconv/semconv_test.go
index fcee8460f..2d528bd23 100644
--- a/semconv/semconv_test.go
+++ b/semconv/semconv_test.go
@@ -10,6 +10,7 @@ import (
 
 	version "github.com/hashicorp/go-version"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestAllSemConvFilesAreCrated(t *testing.T) {
@@ -17,13 +18,13 @@ func TestAllSemConvFilesAreCrated(t *testing.T) {
 	var expectedFiles = []string{"generated_resource.go", "generated_trace.go", "schema.go", "nonstandard.go"}
 
 	files, err := os.ReadDir(".")
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	constraints, err := version.NewConstraint("> v1.16.0")
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	attrgroupconstraints, err := version.NewConstraint("> v1.22.0")
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	for _, f := range files {
 		if !f.IsDir() {
@@ -31,7 +32,7 @@ func TestAllSemConvFilesAreCrated(t *testing.T) {
 		}
 
 		ver, err := version.NewVersion(f.Name())
-		assert.NoError(t, err)
+		require.NoError(t, err)
 
 		expected := make([]string, len(expectedFiles))
 		copy(expected, expectedFiles)
