diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index 514c867b9..9b9a3f195 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -33,6 +33,7 @@ consumer/consumerprofiles                      @open-telemetry/collector-approve
 consumer/xconsumer                             @open-telemetry/collector-approvers @mx-psi @dmathieu
 connector/connectorprofiles                    @open-telemetry/collector-approvers @mx-psi @dmathieu
 exporter/exporterhelper/exporterhelperprofiles @open-telemetry/collector-approvers @mx-psi @dmathieu
+exporter/exporterhelper/xexporterhelper        @open-telemetry/collector-approvers @mx-psi @dmathieu
 exporter/exporterprofiles                      @open-telemetry/collector-approvers @mx-psi @dmathieu
 exporter/xexporter                             @open-telemetry/collector-approvers @mx-psi @dmathieu
 processor/processorprofiles                    @open-telemetry/collector-approvers @mx-psi @dmathieu
