diff --git a/exporter/internal/otlptext/logs_test.go b/exporter/internal/otlptext/logs_test.go
index a1d15676d..e8e237094 100644
--- a/exporter/internal/otlptext/logs_test.go
+++ b/exporter/internal/otlptext/logs_test.go
@@ -65,7 +65,7 @@ func TestLogsText(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := NewTextLogsMarshaler().MarshalLogs(tt.in)
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			out, err := os.ReadFile(filepath.Join("testdata", "logs", tt.out))
 			require.NoError(t, err)
 			expected := strings.ReplaceAll(string(out), "\r", "")
