diff --git a/service/internal/fanoutconsumer/metrics.go b/service/internal/fanoutconsumer/metrics.go
index 9a9d221d1..371643252 100644
--- a/service/internal/fanoutconsumer/metrics.go
+++ b/service/internal/fanoutconsumer/metrics.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package fanoutconsumer
+package fanoutconsumer // import "go.opentelemetry.io/collector/service/internal/fanoutconsumer"
 
 import (
 	"context"
