diff --git a/exporter/exporterhelper/common_test.go b/exporter/exporterhelper/common_test.go
index c719ed6b0..0937bf348 100644
--- a/exporter/exporterhelper/common_test.go
+++ b/exporter/exporterhelper/common_test.go
@@ -26,13 +26,14 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
 var (
 	defaultID       = component.NewID("test")
-	defaultSettings = func() component.ExporterCreateSettings {
+	defaultSettings = func() exporter.CreateSettings {
 		set := componenttest.NewNopExporterCreateSettings()
 		set.ID = defaultID
 		return set
