diff --git a/exporter/exporterhelper/logs.go b/exporter/exporterhelper/logs.go
index 7cb6287a7..60770aab4 100644
--- a/exporter/exporterhelper/logs.go
+++ b/exporter/exporterhelper/logs.go
@@ -21,6 +21,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
 	"go.opentelemetry.io/collector/pdata/plog"
 )
@@ -77,14 +78,14 @@ type logsExporter struct {
 	consumer.Logs
 }
 
-// NewLogsExporter creates a component.LogsExporter that records observability metrics and wraps every request with a Span.
+// NewLogsExporter creates a exporter.Logs that records observability metrics and wraps every request with a Span.
 func NewLogsExporter(
 	_ context.Context,
-	set component.ExporterCreateSettings,
+	set exporter.CreateSettings,
 	cfg component.Config,
 	pusher consumer.ConsumeLogsFunc,
 	options ...Option,
-) (component.LogsExporter, error) {
+) (exporter.Logs, error) {
 	if cfg == nil {
 		return nil, errNilConfig
 	}
