diff --git a/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl b/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
index a3dd9c12c..a5c538af4 100644
--- a/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
+++ b/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
@@ -20,23 +20,31 @@ import (
 	{{- end }}
 )
 
+// Deprecated: [v0.119.0] Use componenttest.Telemetry
 type Telemetry struct {
 	*componenttest.Telemetry
 }
 
+// Deprecated: [v0.119.0] Use componenttest.NewTelemetry
 func SetupTelemetry(opts ...componenttest.TelemetryOption) Telemetry {
 	return Telemetry{ Telemetry: componenttest.NewTelemetry(opts...) }
 }
 
 {{ if or isConnector isExporter isExtension isProcessor isReceiver isScraper }}
+// Deprecated: [v0.119.0] Use {{ .Package }}test.NewSettings
 func (tt *Telemetry) NewSettings() {{ .Status.Class }}.Settings {
-	set := {{ .Status.Class }}test.NewNopSettings()
-	set.ID = component.NewID(component.MustNewType("{{ .Type }}"))
-	set.TelemetrySettings = tt.NewTelemetrySettings()
-	return set
+	return NewSettings(tt.Telemetry)
+}
+
+func NewSettings(tt *componenttest.Telemetry) {{ .Status.Class }}.Settings {
+set := {{ .Status.Class }}test.NewNopSettings()
+set.ID = component.NewID(component.MustNewType("{{ .Type }}"))
+set.TelemetrySettings = tt.NewTelemetrySettings()
+return set
 }
 {{- end }}
 
+// Deprecated: [v0.119.0] Use {{ .Package }}test.AssertEqual*
 func (tt *Telemetry) AssertMetrics(t *testing.T, expected []metricdata.Metrics, opts ...metricdatatest.Option) {
 	var md metricdata.ResourceMetrics
 	require.NoError(t, tt.Reader.Collect(context.Background(), &md))
@@ -50,15 +58,6 @@ func (tt *Telemetry) AssertMetrics(t *testing.T, expected []metricdata.Metrics,
 	require.Equal(t, len(expected), lenMetrics(md))
 }
 
-{{ if or isConnector isExporter isExtension isProcessor isReceiver isScraper }}
-func NewSettings(tt *componenttest.Telemetry) {{ .Status.Class }}.Settings {
-set := {{ .Status.Class }}test.NewNopSettings()
-set.ID = component.NewID(component.MustNewType("{{ .Type }}"))
-set.TelemetrySettings = tt.NewTelemetrySettings()
-return set
-}
-{{- end }}
-
 {{ range $name, $metric := .Telemetry.Metrics }}
 
 func AssertEqual{{ $name.Render }}(t *testing.T, tt *componenttest.Telemetry, dps []metricdata.{{- if eq $metric.Data.Type "Histogram" }} {{$metric.Data.Type}} {{- end }}DataPoint[{{ $metric.Data.BasicType }}], opts ...metricdatatest.Option) {
