diff --git a/extension/zpagesextension/generated_component_test.go b/extension/zpagesextension/generated_component_test.go
index fba4d348c..85ab73915 100644
--- a/extension/zpagesextension/generated_component_test.go
+++ b/extension/zpagesextension/generated_component_test.go
@@ -31,18 +31,18 @@ func TestComponentLifecycle(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, sub.Unmarshal(&cfg))
 	t.Run("shutdown", func(t *testing.T) {
-		e, err := factory.CreateExtension(context.Background(), extensiontest.NewNopSettings(), cfg)
+		e, err := factory.Create(context.Background(), extensiontest.NewNopSettings(), cfg)
 		require.NoError(t, err)
 		err = e.Shutdown(context.Background())
 		require.NoError(t, err)
 	})
 	t.Run("lifecycle", func(t *testing.T) {
-		firstExt, err := factory.CreateExtension(context.Background(), extensiontest.NewNopSettings(), cfg)
+		firstExt, err := factory.Create(context.Background(), extensiontest.NewNopSettings(), cfg)
 		require.NoError(t, err)
 		require.NoError(t, firstExt.Start(context.Background(), componenttest.NewNopHost()))
 		require.NoError(t, firstExt.Shutdown(context.Background()))
 
-		secondExt, err := factory.CreateExtension(context.Background(), extensiontest.NewNopSettings(), cfg)
+		secondExt, err := factory.Create(context.Background(), extensiontest.NewNopSettings(), cfg)
 		require.NoError(t, err)
 		require.NoError(t, secondExt.Start(context.Background(), componenttest.NewNopHost()))
 		require.NoError(t, secondExt.Shutdown(context.Background()))
