diff --git a/cmd/schemagen/schemagen/cli_all.go b/cmd/schemagen/schemagen/cli_all.go
index 5acb4b8db..00cbc0a1b 100644
--- a/cmd/schemagen/schemagen/cli_all.go
+++ b/cmd/schemagen/schemagen/cli_all.go
@@ -16,7 +16,7 @@ package schemagen
 
 import (
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config"
 )
 
 // createAllSchemaFiles creates config yaml schema files for all registered components
@@ -27,8 +27,8 @@ func createAllSchemaFiles(components component.Factories, env env) {
 	}
 }
 
-func getAllConfigs(components component.Factories) []configmodels.NamedEntity {
-	var cfgs []configmodels.NamedEntity
+func getAllConfigs(components component.Factories) []config.NamedEntity {
+	var cfgs []config.NamedEntity
 	for _, f := range components.Receivers {
 		cfgs = append(cfgs, f.CreateDefaultConfig())
 	}
