diff --git a/processor/factory_test.go b/processor/factory_test.go
index e5b5be1a7..6332e056f 100644
--- a/processor/factory_test.go
+++ b/processor/factory_test.go
@@ -17,6 +17,7 @@ package processor
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-service/config/configerror"
@@ -25,11 +26,12 @@ import (
 )
 
 type TestFactory struct {
+	name string
 }
 
 // Type gets the type of the Processor config created by this factory.
 func (f *TestFactory) Type() string {
-	return "exampleoption"
+	return f.name
 }
 
 // CreateDefaultConfig creates the default configuration for the Processor.
@@ -55,30 +57,41 @@ func (f *TestFactory) CreateMetricsProcessor(
 	return nil, configerror.ErrDataTypeIsNotSupported
 }
 
-func TestRegisterProcessorFactory(t *testing.T) {
-	f := TestFactory{}
-	err := RegisterFactory(&f)
-	if err != nil {
-		t.Fatalf("cannot register factory")
+func TestFactoriesBuilder(t *testing.T) {
+	type testCase struct {
+		in  []Factory
+		out map[string]Factory
+		err bool
 	}
 
-	if &f != GetFactory(f.Type()) {
-		t.Fatalf("cannot find factory")
+	testCases := []testCase{
+		{
+			in: []Factory{
+				&TestFactory{"p1"},
+				&TestFactory{"p2"},
+			},
+			out: map[string]Factory{
+				"p1": &TestFactory{"p1"},
+				"p2": &TestFactory{"p2"},
+			},
+			err: false,
+		},
+		{
+			in: []Factory{
+				&TestFactory{"p1"},
+				&TestFactory{"p1"},
+			},
+			err: true,
+		},
 	}
 
-	// Verify that attempt to register a factory with duplicate name panics
-	paniced := false
-	func() {
-		defer func() {
-			if r := recover(); r != nil {
-				paniced = true
-			}
-		}()
-
-		err = RegisterFactory(&f)
-	}()
-
-	if !paniced {
-		t.Fatalf("must panic on double registration")
+	for _, c := range testCases {
+		out, err := Build(c.in...)
+		if c.err {
+			assert.NotNil(t, err)
+			continue
+		}
+		assert.Nil(t, err)
+		assert.Equal(t, c.out, out)
 	}
 }
