diff --git a/receiver/receivertest/go.mod b/receiver/receivertest/go.mod
index 342e26f45..9fa28935d 100644
--- a/receiver/receivertest/go.mod
+++ b/receiver/receivertest/go.mod
@@ -41,7 +41,7 @@ require (
 	golang.org/x/text v0.21.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20241202173237-19429a94021a // indirect
 	google.golang.org/grpc v1.70.0 // indirect
-	google.golang.org/protobuf v1.36.4 // indirect
+	google.golang.org/protobuf v1.36.5 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
