diff --git a/service/defaultcomponents/defaults_test.go b/service/defaultcomponents/defaults_test.go
index 264572cbf..598e60ff7 100644
--- a/service/defaultcomponents/defaults_test.go
+++ b/service/defaultcomponents/defaults_test.go
@@ -17,7 +17,6 @@
 package defaultcomponents
 
 import (
-	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -38,6 +37,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/processor/batchprocessor"
 	"github.com/open-telemetry/opentelemetry-collector/processor/memorylimiter"
 	"github.com/open-telemetry/opentelemetry-collector/processor/queuedprocessor"
+	"github.com/open-telemetry/opentelemetry-collector/processor/resourceprocessor"
 	"github.com/open-telemetry/opentelemetry-collector/processor/samplingprocessor/probabilisticsamplerprocessor"
 	"github.com/open-telemetry/opentelemetry-collector/processor/samplingprocessor/tailsamplingprocessor"
 	"github.com/open-telemetry/opentelemetry-collector/processor/spanprocessor"
@@ -67,6 +67,7 @@ func TestDefaultComponents(t *testing.T) {
 	}
 	expectedProcessors := map[configmodels.Type]component.ProcessorFactoryBase{
 		"attributes":            &attributesprocessor.Factory{},
+		"resource":              &resourceprocessor.Factory{},
 		"queued_retry":          &queuedprocessor.Factory{},
 		"batch":                 &batchprocessor.Factory{},
 		"memory_limiter":        &memorylimiter.Factory{},
@@ -85,7 +86,6 @@ func TestDefaultComponents(t *testing.T) {
 	}
 
 	factories, err := Components()
-	fmt.Println(err)
 	assert.Nil(t, err)
 	assert.Equal(t, expectedExtensions, factories.Extensions)
 	assert.Equal(t, expectedReceivers, factories.Receivers)
