diff --git a/service/telemetry/factory.go b/service/telemetry/factory.go
index 310574743..0cff573e2 100644
--- a/service/telemetry/factory.go
+++ b/service/telemetry/factory.go
@@ -7,7 +7,7 @@ import (
 	"context"
 	"time"
 
-	config "go.opentelemetry.io/contrib/config/v0.3.0"
+	config "go.opentelemetry.io/contrib/otelconf/v0.3.0"
 	"go.opentelemetry.io/otel/log"
 	"go.opentelemetry.io/otel/metric"
 	"go.opentelemetry.io/otel/trace"
