diff --git a/cmd/mdatagen/internal/command.go b/cmd/mdatagen/internal/command.go
index a094bc248..3e2d45fd2 100644
--- a/cmd/mdatagen/internal/command.go
+++ b/cmd/mdatagen/internal/command.go
@@ -339,13 +339,10 @@ func getTemplateFuncMap(md Metadata) template.FuncMap {
 		"casesTitle":  cases.Title(language.English).String,
 		"toLowerCase": strings.ToLower,
 		"toCamelCase": func(s string) string {
-			caser := cases.Title(language.English).String
-			parts := strings.Split(s, "_")
-			var result strings.Builder
-			for _, part := range parts {
-				fmt.Fprintf(&result, "%s", caser(part))
-			}
-			return result.String()
+			return joinCamelCase(strings.Split(s, "_"), true)
+		},
+		"toLowerCamelCase": func(s string) string {
+			return joinCamelCase(strings.Split(s, "_"), false)
 		},
 		"inc":       func(i int) int { return i + 1 },
 		"distroURL": distroURL,
@@ -570,3 +567,16 @@ func generateConfigGoStruct(md Metadata, outputDir string) error {
 	fns := cfggen.WithCfgFns(getTemplateFuncMap(md), rootPkg, md.PackageName)
 	return generateFileWithFns(tmplFile, dstFile, md, packageName, fns)
 }
+
+func joinCamelCase(parts []string, exported bool) string {
+	caser := cases.Title(language.English).String
+	var result strings.Builder
+	for i, part := range parts {
+		if i == 0 && !exported {
+			fmt.Fprintf(&result, "%s", strings.ToLower(part))
+		} else {
+			fmt.Fprintf(&result, "%s", caser(part))
+		}
+	}
+	return result.String()
+}
