diff --git a/exporter/exporterhelper/common_test.go b/exporter/exporterhelper/common_test.go
index c3cd97b9e..3df557eb3 100644
--- a/exporter/exporterhelper/common_test.go
+++ b/exporter/exporterhelper/common_test.go
@@ -31,15 +31,10 @@ import (
 var (
 	okStatus = trace.Status{Code: trace.StatusCodeOK}
 
-	defaultExporterName = "test"
-	defaultExporterCfg  = &config.ExporterSettings{
-		TypeVal: "test",
-		NameVal: defaultExporterName,
-	}
-
+	defaultExporterCfg  = config.NewExporterSettings(config.NewID(typeStr))
 	exporterTag, _      = tag.NewKey("exporter")
 	defaultExporterTags = []tag.Tag{
-		{Key: exporterTag, Value: defaultExporterName},
+		{Key: exporterTag, Value: "test"},
 	}
 )
 
@@ -49,7 +44,7 @@ func TestErrorToStatus(t *testing.T) {
 }
 
 func TestBaseExporter(t *testing.T) {
-	be := newBaseExporter(defaultExporterCfg, zap.NewNop())
+	be := newBaseExporter(&defaultExporterCfg, zap.NewNop())
 	require.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))
 	require.NoError(t, be.Shutdown(context.Background()))
 }
@@ -57,7 +52,7 @@ func TestBaseExporter(t *testing.T) {
 func TestBaseExporterWithOptions(t *testing.T) {
 	want := errors.New("my error")
 	be := newBaseExporter(
-		defaultExporterCfg,
+		&defaultExporterCfg,
 		zap.NewNop(),
 		WithStart(func(ctx context.Context, host component.Host) error { return want }),
 		WithShutdown(func(ctx context.Context) error { return want }),
