diff --git a/pdata/internal/json/enum.go b/pdata/internal/json/enum.go
index 4fbe19343..02dd2b776 100644
--- a/pdata/internal/json/enum.go
+++ b/pdata/internal/json/enum.go
@@ -15,7 +15,7 @@ func ReadEnumValue(iter *jsoniter.Iterator, valueMap map[string]int32) int32 {
 		return iter.ReadInt32()
 	case jsoniter.StringValue:
 		val, ok := valueMap[iter.ReadString()]
-		// Same behavior with official protbuf JSON decoder,
+		// Same behavior with official protobuf JSON decoder,
 		// see https://github.com/open-telemetry/opentelemetry-proto-go/pull/81
 		if !ok {
 			iter.ReportError("ReadEnumValue", "unknown string value")
