diff --git a/receiver/hostmetricsreceiver/hostmetrics_receiver.go b/receiver/hostmetricsreceiver/hostmetrics_receiver.go
index dd5bdd832..e93369559 100644
--- a/receiver/hostmetricsreceiver/hostmetrics_receiver.go
+++ b/receiver/hostmetricsreceiver/hostmetrics_receiver.go
@@ -19,6 +19,7 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/pkg/errors"
 	"go.opencensus.io/trace"
 	"go.uber.org/zap"
 
@@ -32,8 +33,11 @@ import (
 
 // receiver is the type that scrapes various host metrics.
 type receiver struct {
-	config   *Config
-	scrapers []internal.Scraper
+	config *Config
+
+	hostMetricScrapers     []internal.Scraper
+	resourceMetricScrapers []internal.ResourceScraper
+
 	consumer consumer.MetricsConsumer
 	done     chan struct{}
 }
@@ -43,33 +47,72 @@ func newHostMetricsReceiver(
 	ctx context.Context,
 	logger *zap.Logger,
 	config *Config,
-	factories map[string]internal.Factory,
+	factories map[string]internal.ScraperFactory,
+	resourceFactories map[string]internal.ResourceScraperFactory,
 	consumer consumer.MetricsConsumer,
 ) (*receiver, error) {
 
-	scrapers := make([]internal.Scraper, 0)
+	hostMetricScrapers := make([]internal.Scraper, 0)
+	resourceMetricScrapers := make([]internal.ResourceScraper, 0)
+
 	for key, cfg := range config.Scrapers {
-		factory := factories[key]
-		if factory == nil {
-			return nil, fmt.Errorf("host metrics scraper factory not found for key: %s", key)
+		hostMetricsScraper, ok, err := createHostMetricsScraper(ctx, logger, key, cfg, factories)
+		if err != nil {
+			errors.Wrapf(err, "failed to create scraper for key %q", key)
 		}
 
-		scraper, err := factory.CreateMetricsScraper(ctx, logger, cfg)
+		if ok {
+			hostMetricScrapers = append(hostMetricScrapers, hostMetricsScraper)
+			continue
+		}
+
+		resourceMetricsScraper, ok, err := createResourceMetricsScraper(ctx, logger, key, cfg, resourceFactories)
 		if err != nil {
-			return nil, fmt.Errorf("cannot create scraper: %s", err.Error())
+			errors.Wrapf(err, "failed to create resource scraper for key %q", key)
 		}
-		scrapers = append(scrapers, scraper)
+
+		if ok {
+			resourceMetricScrapers = append(resourceMetricScrapers, resourceMetricsScraper)
+			continue
+		}
+
+		return nil, fmt.Errorf("host metrics scraper factory not found for key: %q", key)
 	}
 
 	hmr := &receiver{
-		config:   config,
-		scrapers: scrapers,
-		consumer: consumer,
+		config:                 config,
+		hostMetricScrapers:     hostMetricScrapers,
+		resourceMetricScrapers: resourceMetricScrapers,
+		consumer:               consumer,
 	}
 
 	return hmr, nil
 }
 
+func createHostMetricsScraper(ctx context.Context, logger *zap.Logger, key string, cfg internal.Config, factories map[string]internal.ScraperFactory) (scraper internal.Scraper, ok bool, err error) {
+	factory := factories[key]
+	if factory == nil {
+		ok = false
+		return
+	}
+
+	ok = true
+	scraper, err = factory.CreateMetricsScraper(ctx, logger, cfg)
+	return
+}
+
+func createResourceMetricsScraper(ctx context.Context, logger *zap.Logger, key string, cfg internal.Config, factories map[string]internal.ResourceScraperFactory) (scraper internal.ResourceScraper, ok bool, err error) {
+	factory := factories[key]
+	if factory == nil {
+		ok = false
+		return
+	}
+
+	ok = true
+	scraper, err = factory.CreateMetricsScraper(ctx, logger, cfg)
+	return
+}
+
 // Start initializes the underlying scrapers and begins scraping
 // host metrics based on the OS platform.
 func (hmr *receiver) Start(ctx context.Context, host component.Host) error {
@@ -90,7 +133,7 @@ func (hmr *receiver) Shutdown(ctx context.Context) error {
 }
 
 func (hmr *receiver) initializeScrapers(ctx context.Context, host component.Host) {
-	for _, scraper := range hmr.scrapers {
+	for _, scraper := range hmr.allScrapers() {
 		err := scraper.Initialize(ctx)
 		if err != nil {
 			host.ReportFatalError(err)
@@ -119,11 +162,36 @@ func (hmr *receiver) scrapeMetrics(ctx context.Context) {
 	ctx, span := trace.StartSpan(ctx, "hostmetricsreceiver.ScrapeMetrics")
 	defer span.End()
 
+	var errors []error
 	metricData := data.NewMetricData()
+
+	if err := hmr.scrapeAndAppendHostMetrics(ctx, metricData); err != nil {
+		errors = append(errors, err)
+	}
+
+	if err := hmr.scrapeAndAppendResourceMetrics(ctx, metricData); err != nil {
+		errors = append(errors, err)
+	}
+
+	if len(errors) > 0 {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Error(s) when scraping metrics: %v", componenterror.CombineErrors(errors))})
+	}
+
+	if err := hmr.consumer.ConsumeMetrics(ctx, pdatautil.MetricsFromInternalMetrics(metricData)); err != nil {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Unable to process metrics: %v", err)})
+		return
+	}
+}
+
+func (hmr *receiver) scrapeAndAppendHostMetrics(ctx context.Context, metricData data.MetricData) error {
+	if len(hmr.hostMetricScrapers) == 0 {
+		return nil
+	}
+
 	metrics := internal.InitializeMetricSlice(metricData)
 
 	var errors []error
-	for _, scraper := range hmr.scrapers {
+	for _, scraper := range hmr.hostMetricScrapers {
 		scraperMetrics, err := scraper.ScrapeMetrics(ctx)
 		if err != nil {
 			errors = append(errors, err)
@@ -133,21 +201,39 @@ func (hmr *receiver) scrapeMetrics(ctx context.Context) {
 	}
 
 	if len(errors) > 0 {
-		span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Error(s) when scraping metrics: %v", componenterror.CombineErrors(errors))})
+		return componenterror.CombineErrors(errors)
 	}
 
-	if metrics.Len() > 0 {
-		err := hmr.consumer.ConsumeMetrics(ctx, pdatautil.MetricsFromInternalMetrics(metricData))
+	return nil
+}
+
+func (hmr *receiver) scrapeAndAppendResourceMetrics(ctx context.Context, metricData data.MetricData) error {
+	if len(hmr.resourceMetricScrapers) == 0 {
+		return nil
+	}
+
+	rm := metricData.ResourceMetrics()
+
+	var errors []error
+	for _, scraper := range hmr.resourceMetricScrapers {
+		scraperResourceMetrics, err := scraper.ScrapeMetrics(ctx)
 		if err != nil {
-			span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Unable to process metrics: %v", err)})
-			return
+			errors = append(errors, err)
 		}
+
+		scraperResourceMetrics.MoveAndAppendTo(rm)
+	}
+
+	if len(errors) > 0 {
+		return componenterror.CombineErrors(errors)
 	}
+
+	return nil
 }
 
 func (hmr *receiver) closeScrapers(ctx context.Context) error {
 	var errs []error
-	for _, scraper := range hmr.scrapers {
+	for _, scraper := range hmr.allScrapers() {
 		err := scraper.Close(ctx)
 		if err != nil {
 			errs = append(errs, err)
@@ -160,3 +246,15 @@ func (hmr *receiver) closeScrapers(ctx context.Context) error {
 
 	return nil
 }
+
+func (hmr *receiver) allScrapers() []internal.BaseScraper {
+	allScrapers := make([]internal.BaseScraper, len(hmr.hostMetricScrapers)+len(hmr.resourceMetricScrapers))
+	for i, hostMetricScraper := range hmr.hostMetricScrapers {
+		allScrapers[i] = hostMetricScraper
+	}
+	startIdx := len(hmr.hostMetricScrapers)
+	for i, resourceMetricScraper := range hmr.resourceMetricScrapers {
+		allScrapers[startIdx+i] = resourceMetricScraper
+	}
+	return allScrapers
+}
