diff --git a/service/telemetry_test.go b/service/telemetry_test.go
index a4144e278..e22c7c88f 100644
--- a/service/telemetry_test.go
+++ b/service/telemetry_test.go
@@ -23,6 +23,7 @@ import (
 	"go.opentelemetry.io/collector/internal/testutil"
 	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
 	"go.opentelemetry.io/collector/service/internal/proctelemetry"
+	"go.opentelemetry.io/collector/service/internal/servicetelemetry"
 	"go.opentelemetry.io/collector/service/telemetry"
 )
 
@@ -272,7 +273,7 @@ func TestTelemetryInit(t *testing.T) {
 			}
 			otelRes := buildResource(buildInfo, *tc.cfg)
 			res := pdataFromSdk(otelRes)
-			settings := component.TelemetrySettings{
+			settings := servicetelemetry.TelemetrySettings{
 				Logger:   zap.NewNop(),
 				Resource: res,
 			}
