diff --git a/service/internal/fanoutconsumer/cloningconsumer.go b/service/internal/fanoutconsumer/cloningconsumer.go
index 14709c57a..ad266c4dc 100644
--- a/service/internal/fanoutconsumer/cloningconsumer.go
+++ b/service/internal/fanoutconsumer/cloningconsumer.go
@@ -19,7 +19,7 @@ import (
 
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumererror"
-	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/model/pdata"
 )
 
 // NewMetricsCloning wraps multiple metrics consumers in a single one and clones the data
