diff --git a/processor/go.mod b/processor/go.mod
index 0ed3bec05..de9d00ca6 100644
--- a/processor/go.mod
+++ b/processor/go.mod
@@ -38,7 +38,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/prometheus/client_golang v1.19.1 // indirect
 	github.com/prometheus/client_model v0.6.1 // indirect
-	github.com/prometheus/common v0.53.0 // indirect
+	github.com/prometheus/common v0.54.0 // indirect
 	github.com/prometheus/procfs v0.15.0 // indirect
 	go.opentelemetry.io/collector/confmap v0.102.0 // indirect
 	go.opentelemetry.io/otel/exporters/prometheus v0.49.0 // indirect
