diff --git a/cmd/mdatagen/metricdata.go b/cmd/mdatagen/metricdata.go
index f4515136a..5bf77985c 100644
--- a/cmd/mdatagen/metricdata.go
+++ b/cmd/mdatagen/metricdata.go
@@ -125,7 +125,7 @@ func (d *gauge) Unmarshal(parser *confmap.Conf) error {
 	if err := d.MetricValueType.Unmarshal(parser); err != nil {
 		return err
 	}
-	return parser.Unmarshal(d)
+	return parser.Unmarshal(d, confmap.WithIgnoreUnused())
 }
 
 func (d gauge) Type() string {
@@ -155,7 +155,7 @@ func (d *sum) Unmarshal(parser *confmap.Conf) error {
 	if err := d.MetricValueType.Unmarshal(parser); err != nil {
 		return err
 	}
-	return parser.Unmarshal(d)
+	return parser.Unmarshal(d, confmap.WithIgnoreUnused())
 }
 
 // TODO: Currently, this func will not be called because of https://github.com/open-telemetry/opentelemetry-collector/issues/6671. Uncomment function and
