diff --git a/confmap/provider.go b/confmap/provider.go
index 161e24739..3338d72bd 100644
--- a/confmap/provider.go
+++ b/confmap/provider.go
@@ -9,6 +9,8 @@ import (
 
 	"go.uber.org/zap"
 	"gopkg.in/yaml.v3"
+
+	"go.opentelemetry.io/collector/internal/globalgates"
 )
 
 // ProviderSettings are the settings to initialize a Provider.
@@ -140,7 +142,11 @@ func NewRetrievedFromYAML(yamlBytes []byte, opts ...RetrievedOption) (*Retrieved
 
 	switch v := rawConf.(type) {
 	case string:
-		opts = append(opts, withStringRepresentation(v))
+		val := v
+		if globalgates.StrictlyTypedInputGate.IsEnabled() {
+			val = string(yamlBytes)
+		}
+		return NewRetrieved(val, append(opts, withStringRepresentation(val))...)
 	case int, int32, int64, float32, float64, bool, map[string]any:
 		opts = append(opts, withStringRepresentation(string(yamlBytes)))
 	}
