diff --git a/config/configretry/backoff.go b/config/configretry/backoff.go
index f87260084..605556ba9 100644
--- a/config/configretry/backoff.go
+++ b/config/configretry/backoff.go
@@ -59,7 +59,13 @@ func (bs *BackOffConfig) Validate() error {
 		return errors.New("'max_interval' must be non-negative")
 	}
 	if bs.MaxElapsedTime < 0 {
-		return errors.New("'max_elapsed' time must be non-negative")
+		return errors.New("'max_elapsed_time' must be non-negative")
+	}
+	if bs.MaxElapsedTime < bs.InitialInterval {
+		return errors.New("'max_elapsed_time' must not be less than 'initial_interval'")
+	}
+	if bs.MaxElapsedTime < bs.MaxInterval {
+		return errors.New("'max_elapsed_time' must not be less than 'max_interval'")
 	}
 	return nil
 }
