diff --git a/config/configtest/configtest_test.go b/config/configtest/configtest_test.go
index 9a70f588d..9829b8548 100644
--- a/config/configtest/configtest_test.go
+++ b/config/configtest/configtest_test.go
@@ -1,4 +1,4 @@
-// Copyright 2020 OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
