diff --git a/exporter/debugexporter/internal/otlptext/traces.go b/exporter/debugexporter/internal/otlptext/traces.go
index 90d5c400a..e3d9acfb3 100644
--- a/exporter/debugexporter/internal/otlptext/traces.go
+++ b/exporter/debugexporter/internal/otlptext/traces.go
@@ -39,7 +39,7 @@ func (textTracesMarshaler) MarshalTraces(td ptrace.Traces) ([]byte, error) {
 				buf.logAttr("ID", span.SpanID())
 				buf.logAttr("Name", span.Name())
 				buf.logAttr("Kind", span.Kind().String())
-				if ts := span.TraceState().AsRaw(); len(ts) != 0 {
+				if ts := span.TraceState().AsRaw(); ts != "" {
 					buf.logAttr("TraceState", ts)
 				}
 				buf.logAttr("Start time", span.StartTimestamp().String())
