diff --git a/config/configgrpc/go.mod b/config/configgrpc/go.mod
index 92a9ae735..88ca54516 100644
--- a/config/configgrpc/go.mod
+++ b/config/configgrpc/go.mod
@@ -15,7 +15,7 @@ require (
 	go.opentelemetry.io/collector/config/internal v0.96.0
 	go.opentelemetry.io/collector/extension/auth v0.96.0
 	go.opentelemetry.io/collector/pdata v1.3.0
-	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.47.0
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.49.0
 	go.opentelemetry.io/otel v1.24.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
@@ -23,7 +23,6 @@ require (
 )
 
 require (
-	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.2.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
