diff --git a/obsreport/obsreporttest/obsreporttest_test.go b/obsreport/obsreporttest/obsreporttest_test.go
index 64d8ad57e..22879049d 100644
--- a/obsreport/obsreporttest/obsreporttest_test.go
+++ b/obsreport/obsreporttest/obsreporttest_test.go
@@ -27,7 +27,6 @@ import (
 
 const (
 	exporter  = "fakeExporter"
-	processor = "fakeProcessor"
 	receiver  = "fakeReicever"
 	transport = "fakeTransport"
 	format    = "fakeFormat"
