diff --git a/config/configgrpc/configgrpc.go b/config/configgrpc/configgrpc.go
index 04b1662e7..33b6b5270 100644
--- a/config/configgrpc/configgrpc.go
+++ b/config/configgrpc/configgrpc.go
@@ -112,10 +112,6 @@ type ClientConfig struct {
 	Middlewares []configmiddleware.Config `mapstructure:"middlewares,omitempty"`
 }
 
-func ptr[T any](v T) *T {
-	return &v
-}
-
 // NewDefaultClientConfig returns a new instance of ClientConfig with default values.
 func NewDefaultClientConfig() ClientConfig {
 	return ClientConfig{
@@ -127,8 +123,8 @@ func NewDefaultClientConfig() ClientConfig {
 
 // KeepaliveServerConfig is the configuration for keepalive.
 type KeepaliveServerConfig struct {
-	ServerParameters  *KeepaliveServerParameters  `mapstructure:"server_parameters,omitempty"`
-	EnforcementPolicy *KeepaliveEnforcementPolicy `mapstructure:"enforcement_policy,omitempty"`
+	ServerParameters  configoptional.Optional[KeepaliveServerParameters]  `mapstructure:"server_parameters,omitempty"`
+	EnforcementPolicy configoptional.Optional[KeepaliveEnforcementPolicy] `mapstructure:"enforcement_policy,omitempty"`
 	// prevent unkeyed literal initialization
 	_ struct{}
 }
@@ -136,8 +132,8 @@ type KeepaliveServerConfig struct {
 // NewDefaultKeepaliveServerConfig returns a new instance of KeepaliveServerConfig with default values.
 func NewDefaultKeepaliveServerConfig() KeepaliveServerConfig {
 	return KeepaliveServerConfig{
-		ServerParameters:  ptr(NewDefaultKeepaliveServerParameters()),
-		EnforcementPolicy: ptr(NewDefaultKeepaliveEnforcementPolicy()),
+		ServerParameters:  configoptional.Some(NewDefaultKeepaliveServerParameters()),
+		EnforcementPolicy: configoptional.Some(NewDefaultKeepaliveEnforcementPolicy()),
 	}
 }
 
@@ -487,8 +483,8 @@ func (gss *ServerConfig) getGrpcServerOptions(
 	// https://github.com/grpc/grpc-go/blob/120728e1f775e40a2a764341939b78d666b08260/internal/transport/http2_server.go#L184-L200
 	if gss.Keepalive.HasValue() {
 		keepaliveConfig := gss.Keepalive.Get()
-		if keepaliveConfig.ServerParameters != nil {
-			svrParams := keepaliveConfig.ServerParameters
+		if keepaliveConfig.ServerParameters.HasValue() {
+			svrParams := keepaliveConfig.ServerParameters.Get()
 			opts = append(opts, grpc.KeepaliveParams(keepalive.ServerParameters{
 				MaxConnectionIdle:     svrParams.MaxConnectionIdle,
 				MaxConnectionAge:      svrParams.MaxConnectionAge,
@@ -501,8 +497,8 @@ func (gss *ServerConfig) getGrpcServerOptions(
 		// to apply them over zero/nil values before passing these as grpc.ServerOptions.
 		// The following shows the server code for applying default grpc.ServerOptions.
 		// https://github.com/grpc/grpc-go/blob/120728e1f775e40a2a764341939b78d666b08260/internal/transport/http2_server.go#L202-L205
-		if keepaliveConfig.EnforcementPolicy != nil {
-			enfPol := keepaliveConfig.EnforcementPolicy
+		if keepaliveConfig.EnforcementPolicy.HasValue() {
+			enfPol := keepaliveConfig.EnforcementPolicy.Get()
 			opts = append(opts, grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
 				MinTime:             enfPol.MinTime,
 				PermitWithoutStream: enfPol.PermitWithoutStream,
