diff --git a/internal/e2e/go.mod b/internal/e2e/go.mod
index cc9ae37a9..1deb3e718 100644
--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -23,7 +23,7 @@ require (
 	go.opentelemetry.io/collector/extension v0.109.0
 	go.opentelemetry.io/collector/pdata v1.15.0
 	go.opentelemetry.io/collector/pdata/testdata v0.109.0
-	go.opentelemetry.io/collector/pipeline v0.0.0-20240923143859-77bb849aa07e
+	go.opentelemetry.io/collector/pipeline v0.0.0-20240923154032-388e56cdb156
 	go.opentelemetry.io/collector/receiver v0.109.0
 	go.opentelemetry.io/collector/receiver/otlpreceiver v0.109.0
 	go.opentelemetry.io/collector/service v0.109.0
@@ -85,7 +85,7 @@ require (
 	go.opentelemetry.io/collector/extension/extensioncapabilities v0.109.0 // indirect
 	go.opentelemetry.io/collector/featuregate v1.15.0 // indirect
 	go.opentelemetry.io/collector/internal/globalgates v0.109.0 // indirect
-	go.opentelemetry.io/collector/internal/globalsignal v0.0.0-20240923143859-77bb849aa07e // indirect
+	go.opentelemetry.io/collector/internal/globalsignal v0.0.0-20240923154032-388e56cdb156 // indirect
 	go.opentelemetry.io/collector/pdata/pprofile v0.109.0 // indirect
 	go.opentelemetry.io/collector/processor v0.109.0 // indirect
 	go.opentelemetry.io/collector/processor/processorprofiles v0.109.0 // indirect
