diff --git a/exporter/debugexporter/internal/otlptext/databuffer.go b/exporter/debugexporter/internal/otlptext/databuffer.go
index 9f52ff2e6..56bfbd852 100644
--- a/exporter/debugexporter/internal/otlptext/databuffer.go
+++ b/exporter/debugexporter/internal/otlptext/databuffer.go
@@ -348,7 +348,7 @@ func (b *dataBuffer) logProfileSamples(ss pprofile.SampleSlice, dic pprofile.Pro
 
 		if lai := sample.AttributeIndices().Len(); lai > 0 {
 			b.logEntry("        Attributes:")
-			for j := 0; j < lai; j++ {
+			for j := range lai {
 				attr := dic.AttributeTable().At(int(sample.AttributeIndices().At(j)))
 				b.logEntry("             -> %s: %s", dic.StringTable().At(int(attr.KeyStrindex())), attr.Value().AsRaw())
 			}
@@ -385,7 +385,7 @@ func (b *dataBuffer) logProfileLocations(ls pprofile.LocationSlice) {
 		b.logEntry("    Mapping index: %d", location.MappingIndex())
 		b.logEntry("    Address: %d", location.Address())
 		if ll := location.Line().Len(); ll > 0 {
-			for j := 0; j < ll; j++ {
+			for j := range ll {
 				b.logEntry("    Line #%d", j)
 				line := location.Line().At(j)
 				b.logEntry("        Function index: %d", line.FunctionIndex())
