diff --git a/config/configauth/configauth.go b/config/configauth/configauth.go
index dc275f705..ccee4216e 100644
--- a/config/configauth/configauth.go
+++ b/config/configauth/configauth.go
@@ -34,15 +34,7 @@ func NewDefaultAuthentication() *Authentication {
 
 // GetServerAuthenticator attempts to select the appropriate auth.Server from the list of extensions,
 // based on the requested extension name. If an authenticator is not found, an error is returned.
-//
-// Deprecated: [v0.103.0] Use GetServerAuthenticatorContext instead.
-func (a Authentication) GetServerAuthenticator(extensions map[component.ID]component.Component) (auth.Server, error) {
-	return a.GetServerAuthenticatorContext(context.Background(), extensions)
-}
-
-// GetServerAuthenticatorContext attempts to select the appropriate auth.Server from the list of extensions,
-// based on the requested extension name. If an authenticator is not found, an error is returned.
-func (a Authentication) GetServerAuthenticatorContext(_ context.Context, extensions map[component.ID]component.Component) (auth.Server, error) {
+func (a Authentication) GetServerAuthenticator(_ context.Context, extensions map[component.ID]component.Component) (auth.Server, error) {
 	if ext, found := extensions[a.AuthenticatorID]; found {
 		if server, ok := ext.(auth.Server); ok {
 			return server, nil
@@ -53,15 +45,18 @@ func (a Authentication) GetServerAuthenticatorContext(_ context.Context, extensi
 	return nil, fmt.Errorf("failed to resolve authenticator %q: %w", a.AuthenticatorID, errAuthenticatorNotFound)
 }
 
-// Deprecated: [v0.103.0] Use GetClientAuthenticatorContext instead.
-func (a Authentication) GetClientAuthenticator(extensions map[component.ID]component.Component) (auth.Client, error) {
-	return a.GetClientAuthenticatorContext(context.Background(), extensions)
+// GetServerAuthenticatorContext attempts to select the appropriate auth.Server from the list of extensions,
+// based on the requested extension name. If an authenticator is not found, an error is returned.
+//
+// Deprecated: [v0.105.0] Use GetServerAuthenticator instead.
+func (a Authentication) GetServerAuthenticatorContext(ctx context.Context, extensions map[component.ID]component.Component) (auth.Server, error) {
+	return a.GetServerAuthenticator(ctx, extensions)
 }
 
-// GetClientAuthenticatorContext attempts to select the appropriate auth.Client from the list of extensions,
+// GetClientAuthenticator attempts to select the appropriate auth.Client from the list of extensions,
 // based on the component id of the extension. If an authenticator is not found, an error is returned.
 // This should be only used by HTTP clients.
-func (a Authentication) GetClientAuthenticatorContext(_ context.Context, extensions map[component.ID]component.Component) (auth.Client, error) {
+func (a Authentication) GetClientAuthenticator(_ context.Context, extensions map[component.ID]component.Component) (auth.Client, error) {
 	if ext, found := extensions[a.AuthenticatorID]; found {
 		if client, ok := ext.(auth.Client); ok {
 			return client, nil
@@ -70,3 +65,12 @@ func (a Authentication) GetClientAuthenticatorContext(_ context.Context, extensi
 	}
 	return nil, fmt.Errorf("failed to resolve authenticator %q: %w", a.AuthenticatorID, errAuthenticatorNotFound)
 }
+
+// GetClientAuthenticatorContext attempts to select the appropriate auth.Client from the list of extensions,
+// based on the component id of the extension. If an authenticator is not found, an error is returned.
+// This should be only used by HTTP clients.
+//
+// Deprecated: [v0.105.0] Use GetClientAuthenticatorContext instead.
+func (a Authentication) GetClientAuthenticatorContext(ctx context.Context, extensions map[component.ID]component.Component) (auth.Client, error) {
+	return a.GetClientAuthenticator(ctx, extensions)
+}
