diff --git a/pdata/ptrace/generated_status.go b/pdata/ptrace/generated_status.go
index 3f637090a..2040674d6 100644
--- a/pdata/ptrace/generated_status.go
+++ b/pdata/ptrace/generated_status.go
@@ -75,6 +75,10 @@ func (ms Status) SetMessage(v string) {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Status) CopyTo(dest Status) {
 	dest.state.AssertMutable()
-	dest.SetCode(ms.Code())
-	dest.SetMessage(ms.Message())
+	copyOrigStatus(dest.orig, ms.orig)
+}
+
+func copyOrigStatus(dest, src *otlptrace.Status) {
+	dest.Code = src.Code
+	dest.Message = src.Message
 }
