diff --git a/exporter/exporterhelper/internal/queuebatch/queue_batch.go b/exporter/exporterhelper/internal/queuebatch/queue_batch.go
index ce25b1911..f0e5b7300 100644
--- a/exporter/exporterhelper/internal/queuebatch/queue_batch.go
+++ b/exporter/exporterhelper/internal/queuebatch/queue_batch.go
@@ -62,31 +62,23 @@ func newQueueBatch(
 		return nil, fmt.Errorf("queue_batch: unsupported sizer %q", cfg.Sizer)
 	}
 
-	var b Batcher[request.Request]
+	bSet := batcherSettings[request.Request]{
+		sizerType:   cfg.Sizer,
+		sizer:       sizer,
+		partitioner: set.Partitioner,
+		next:        next,
+		maxWorkers:  cfg.NumConsumers,
+	}
+	if oldBatcher {
+		// If a user configures the old batcher, we only can support "items" sizer.
+		bSet.sizerType = request.SizerTypeItems
+		bSet.sizer = request.NewItemsSizer()
+	}
+	b := NewBatcher(cfg.Batch, bSet)
 	if cfg.Batch != nil {
-		if oldBatcher {
-			// If user configures the old batcher we only can support "items" sizer.
-			b = newMultiBatcher(*cfg.Batch, batcherSettings[request.Request]{
-				sizerType:   request.SizerTypeItems,
-				sizer:       request.NewItemsSizer(),
-				partitioner: set.Partitioner,
-				next:        next,
-				maxWorkers:  cfg.NumConsumers,
-			})
-		} else {
-			b = newMultiBatcher(*cfg.Batch, batcherSettings[request.Request]{
-				sizerType:   cfg.Sizer,
-				sizer:       sizer,
-				partitioner: set.Partitioner,
-				next:        next,
-				maxWorkers:  cfg.NumConsumers,
-			})
-		}
-		// Keep the number of queue consumers to 1 if batching is enabled until we support sharding as described in
-		// https://github.com/open-telemetry/opentelemetry-collector/issues/12473
+		// If batching is enabled, keep the number of queue consumers to 1 if batching is enabled until we support
+		// sharding as described in https://github.com/open-telemetry/opentelemetry-collector/issues/12473
 		cfg.NumConsumers = 1
-	} else {
-		b = newDisabledBatcher[request.Request](next)
 	}
 
 	var q Queue[request.Request]
