diff --git a/translator/trace/jaeger/jaegerproto_to_protospan.go b/translator/trace/jaeger/jaegerproto_to_protospan.go
index 37ca3e967..c8aa352ac 100644
--- a/translator/trace/jaeger/jaegerproto_to_protospan.go
+++ b/translator/trace/jaeger/jaegerproto_to_protospan.go
@@ -25,9 +25,9 @@ import (
 	model "github.com/jaegertracing/jaeger/model"
 	"github.com/jaegertracing/jaeger/thrift-gen/jaeger"
 
-	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
-	"github.com/open-telemetry/opentelemetry-service/internal"
-	tracetranslator "github.com/open-telemetry/opentelemetry-service/translator/trace"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
 )
 
 // ProtoBatchToOCProto converts a single Jaeger Proto batch of spans to a OC proto batch.
