diff --git a/config/configopaque/go.mod b/config/configopaque/go.mod
index 8ae5ac646..f900f35dd 100644
--- a/config/configopaque/go.mod
+++ b/config/configopaque/go.mod
@@ -17,7 +17,7 @@ require (
 	github.com/hashicorp/go-version v1.8.0 // indirect
 	github.com/knadh/koanf/maps v0.1.2 // indirect
 	github.com/knadh/koanf/providers/confmap v1.0.0 // indirect
-	github.com/knadh/koanf/v2 v2.3.0 // indirect
+	github.com/knadh/koanf/v2 v2.3.2 // indirect
 	github.com/mitchellh/copystructure v1.2.0 // indirect
 	github.com/mitchellh/reflectwalk v1.0.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
