diff --git a/exporter/otlpexporter/config_test.go b/exporter/otlpexporter/config_test.go
index f19d87961..2705bdd6a 100644
--- a/exporter/otlpexporter/config_test.go
+++ b/exporter/otlpexporter/config_test.go
@@ -89,6 +89,38 @@ func TestUnmarshalConfig(t *testing.T) {
 		}, cfg)
 }
 
+func TestUnmarshalDefaultBatchConfig(t *testing.T) {
+	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "default-batch.yaml"))
+	require.NoError(t, err)
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig()
+	require.NoError(t, cm.Unmarshal(&cfg))
+	require.NoError(t, xconfmap.Validate(&cfg))
+	assert.Equal(t,
+		&Config{
+			TimeoutConfig: exporterhelper.TimeoutConfig{
+				Timeout: 10 * time.Second,
+			},
+			RetryConfig: configretry.NewDefaultBackOffConfig(),
+			QueueConfig: exporterhelper.QueueBatchConfig{
+				Enabled:      true,
+				Sizer:        exporterhelper.RequestSizerTypeRequests,
+				QueueSize:    1000,
+				NumConsumers: 10,
+				Batch: configoptional.Some(exporterhelper.BatchConfig{
+					FlushTimeout: 200 * time.Millisecond,
+					Sizer:        exporterhelper.RequestSizerTypeItems,
+					MinSize:      8192,
+				}),
+			},
+			ClientConfig: configgrpc.ClientConfig{
+				Endpoint:        "1.2.3.4:1234",
+				Compression:     "gzip",
+				WriteBufferSize: 512 * 1024,
+			},
+		}, cfg)
+}
+
 func TestUnmarshalInvalidConfig(t *testing.T) {
 	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "invalid_configs.yaml"))
 	require.NoError(t, err)
