diff --git a/cmd/occollector/app/builder/sampling_builder.go b/cmd/occollector/app/builder/sampling_builder.go
index df33de9a1..fe3234592 100644
--- a/cmd/occollector/app/builder/sampling_builder.go
+++ b/cmd/occollector/app/builder/sampling_builder.go
@@ -33,6 +33,9 @@ const (
 	// NoSampling mode is the default and means that all data arriving at the collector
 	// is passed ahead.
 	NoSampling Mode = "no-sampling"
+	// HeadSampling is the mode in which trace data is sampled at ingestion, without seeing
+	// the whole trace data.
+	HeadSampling Mode = "head"
 	// TailSampling is the mode in which trace data is temporarily retained until an evaluation
 	// if the trace should be sampled is performed.
 	TailSampling Mode = "tail"
