diff --git a/confmap/provider/httpsprovider/go.mod b/confmap/provider/httpsprovider/go.mod
index 5bb4b5ae1..04a910490 100644
--- a/confmap/provider/httpsprovider/go.mod
+++ b/confmap/provider/httpsprovider/go.mod
@@ -3,7 +3,7 @@ module go.opentelemetry.io/collector/confmap/provider/httpsprovider
 go 1.21
 
 require (
-	github.com/stretchr/testify v1.8.4
+	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/confmap v0.96.0
 	go.uber.org/goleak v1.3.0
 )
