diff --git a/exporter/zipkinexporter/factory.go b/exporter/zipkinexporter/factory.go
index 12cbf9642..368f0fecb 100644
--- a/exporter/zipkinexporter/factory.go
+++ b/exporter/zipkinexporter/factory.go
@@ -41,7 +41,7 @@ func NewFactory() component.ExporterFactory {
 	return exporterhelper.NewFactory(
 		typeStr,
 		createDefaultConfig,
-		exporterhelper.WithTraces(createTraceExporter))
+		exporterhelper.WithTraces(createTracesExporter))
 }
 
 func createDefaultConfig() config.Exporter {
@@ -62,7 +62,7 @@ func createDefaultConfig() config.Exporter {
 	}
 }
 
-func createTraceExporter(
+func createTracesExporter(
 	_ context.Context,
 	params component.ExporterCreateParams,
 	cfg config.Exporter,
@@ -78,7 +78,7 @@ func createTraceExporter(
 	if err != nil {
 		return nil, err
 	}
-	return exporterhelper.NewTraceExporter(
+	return exporterhelper.NewTracesExporter(
 		zc,
 		params.Logger,
 		ze.pushTraceData,
