diff --git a/cmd/mdatagen/main_test.go b/cmd/mdatagen/main_test.go
index 62c2159fe..f13693dab 100644
--- a/cmd/mdatagen/main_test.go
+++ b/cmd/mdatagen/main_test.go
@@ -21,6 +21,7 @@ func TestRunContents(t *testing.T) {
 	tests := []struct {
 		yml                    string
 		wantMetricsGenerated   bool
+		wantMetricsContext     bool
 		wantConfigGenerated    bool
 		wantTelemetryGenerated bool
 		wantStatusGenerated    bool
@@ -98,6 +99,12 @@ func TestRunContents(t *testing.T) {
 			yml:     "invalid_telemetry_missing_value_type_for_histogram.yaml",
 			wantErr: true,
 		},
+		{
+			yml:                  "async_metric.yaml",
+			wantMetricsGenerated: true,
+			wantConfigGenerated:  true,
+			wantStatusGenerated:  true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.yml, func(t *testing.T) {
@@ -120,10 +127,18 @@ foo
 			}
 			require.NoError(t, err)
 
+			var contents []byte
 			if tt.wantMetricsGenerated {
 				require.FileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_metrics.go"))
 				require.FileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_metrics_test.go"))
 				require.FileExists(t, filepath.Join(tmpdir, "documentation.md"))
+				contents, err = os.ReadFile(filepath.Join(tmpdir, "internal/metadata/generated_metrics.go")) // nolint: gosec
+				require.NoError(t, err)
+				if tt.wantMetricsContext {
+					require.Contains(t, string(contents), "\"context\"")
+				} else {
+					require.NotContains(t, string(contents), "\"context\"")
+				}
 			} else {
 				require.NoFileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_metrics.go"))
 				require.NoFileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_metrics_test.go"))
@@ -141,6 +156,13 @@ foo
 				require.FileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_telemetry.go"))
 				require.FileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_telemetry_test.go"))
 				require.FileExists(t, filepath.Join(tmpdir, "documentation.md"))
+				contents, err = os.ReadFile(filepath.Join(tmpdir, "internal/metadata/generated_telemetry.go")) // nolint: gosec
+				require.NoError(t, err)
+				if tt.wantMetricsContext {
+					require.Contains(t, string(contents), "\"context\"")
+				} else {
+					require.NotContains(t, string(contents), "\"context\"")
+				}
 			} else {
 				require.NoFileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_telemetry.go"))
 				require.NoFileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_telemetry_test.go"))
@@ -150,7 +172,6 @@ foo
 				require.NoFileExists(t, filepath.Join(tmpdir, "documentation.md"))
 			}
 
-			var contents []byte
 			if tt.wantStatusGenerated {
 				require.FileExists(t, filepath.Join(tmpdir, "internal/metadata/generated_status.go"))
 				contents, err = os.ReadFile(filepath.Join(tmpdir, "README.md")) // nolint: gosec
