diff --git a/exporter/opencensusexporter/factory_test.go b/exporter/opencensusexporter/factory_test.go
index 6452368bd..e534267f0 100644
--- a/exporter/opencensusexporter/factory_test.go
+++ b/exporter/opencensusexporter/factory_test.go
@@ -24,12 +24,9 @@ import (
 	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config/configcheck"
 	"go.opentelemetry.io/collector/config/configgrpc"
 	"go.opentelemetry.io/collector/config/configtls"
-	"go.opentelemetry.io/collector/exporter/exportertest"
-	"go.opentelemetry.io/collector/receiver/opencensusreceiver"
 	"go.opentelemetry.io/collector/testutil"
 )
 
@@ -51,25 +48,7 @@ func TestCreateMetricsExporter(t *testing.T) {
 }
 
 func TestCreateTraceExporter(t *testing.T) {
-	// This test is about creating the exporter and stopping it. However, the
-	// exporter keeps trying to update its connection state in the background
-	// so unless there is a receiver enabled the stop call can return different
-	// results. Standing up a receiver to ensure that stop don't report errors.
-	rcvFactory := opencensusreceiver.NewFactory()
-	require.NotNil(t, rcvFactory)
-	rcvCfg := rcvFactory.CreateDefaultConfig().(*opencensusreceiver.Config)
-	rcvCfg.NetAddr.Endpoint = testutil.GetAvailableLocalAddress(t)
-
-	rcv, err := rcvFactory.CreateTraceReceiver(
-		context.Background(),
-		component.ReceiverCreateParams{Logger: zap.NewNop()},
-		rcvCfg,
-		new(exportertest.SinkTraceExporter))
-	require.NotNil(t, rcv)
-	require.Nil(t, err)
-	require.Nil(t, rcv.Start(context.Background(), componenttest.NewNopHost()))
-	defer rcv.Shutdown(context.Background())
-
+	endpoint := testutil.GetAvailableLocalAddress(t)
 	tests := []struct {
 		name     string
 		config   Config
@@ -81,69 +60,69 @@ func TestCreateTraceExporter(t *testing.T) {
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
 					Endpoint: "",
 				},
+				NumWorkers: 3,
 			},
 			mustFail: true,
 		},
 		{
-			name: "UseSecure",
+			name: "ZeroNumWorkers",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
+					Endpoint: endpoint,
 					TLSSetting: configtls.TLSClientSetting{
 						Insecure: false,
 					},
 				},
+				NumWorkers: 0,
 			},
+			mustFail: true,
 		},
 		{
-			name: "ReconnectionDelay",
+			name: "UseSecure",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
+					Endpoint: endpoint,
+					TLSSetting: configtls.TLSClientSetting{
+						Insecure: false,
+					},
 				},
-				ReconnectionDelay: 5 * time.Second,
+				NumWorkers: 3,
 			},
 		},
 		{
 			name: "Keepalive",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
+					Endpoint: endpoint,
 					Keepalive: &configgrpc.KeepaliveClientConfig{
 						Time:                30 * time.Second,
 						Timeout:             25 * time.Second,
 						PermitWithoutStream: true,
 					},
 				},
+				NumWorkers: 3,
 			},
 		},
 		{
 			name: "Compression",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint:    rcvCfg.NetAddr.Endpoint,
+					Endpoint:    endpoint,
 					Compression: configgrpc.CompressionGzip,
 				},
+				NumWorkers: 3,
 			},
 		},
 		{
 			name: "Headers",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
+					Endpoint: endpoint,
 					Headers: map[string]string{
 						"hdr1": "val1",
 						"hdr2": "val2",
 					},
 				},
-			},
-		},
-		{
-			name: "NumConsumers",
-			config: Config{
-				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
-				},
 				NumWorkers: 3,
 			},
 		},
@@ -151,9 +130,10 @@ func TestCreateTraceExporter(t *testing.T) {
 			name: "CompressionError",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint:    rcvCfg.NetAddr.Endpoint,
+					Endpoint:    endpoint,
 					Compression: "unknown compression",
 				},
+				NumWorkers: 3,
 			},
 			mustFail: true,
 		},
@@ -161,26 +141,28 @@ func TestCreateTraceExporter(t *testing.T) {
 			name: "CaCert",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
+					Endpoint: endpoint,
 					TLSSetting: configtls.TLSClientSetting{
 						TLSSetting: configtls.TLSSetting{
 							CAFile: "testdata/test_cert.pem",
 						},
 					},
 				},
+				NumWorkers: 3,
 			},
 		},
 		{
 			name: "CertPemFileError",
 			config: Config{
 				GRPCClientSettings: configgrpc.GRPCClientSettings{
-					Endpoint: rcvCfg.NetAddr.Endpoint,
+					Endpoint: endpoint,
 					TLSSetting: configtls.TLSClientSetting{
 						TLSSetting: configtls.TLSSetting{
 							CAFile: "nosuchfile",
 						},
 					},
 				},
+				NumWorkers: 3,
 			},
 			mustFail: true,
 		},
@@ -189,21 +171,22 @@ func TestCreateTraceExporter(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			factory := &Factory{}
-			consumer, err := factory.CreateTraceExporter(zap.NewNop(), &tt.config)
-
-			if tt.mustFail {
-				assert.NotNil(t, err)
-			} else {
-				assert.NoError(t, err)
-				assert.NotNil(t, consumer)
 
-				err = consumer.Shutdown(context.Background())
-				if err != nil {
-					// Since the endpoint of opencensus exporter doesn't actually exist,
-					// exporter may already stop because it cannot connect.
-					assert.Equal(t, err.Error(), "rpc error: code = Canceled desc = grpc: the client connection is closing")
-				}
-			}
+			tReceiver, tErr := factory.CreateTraceExporter(zap.NewNop(), &tt.config)
+			checkErrorsAndShutdown(t, tReceiver, tErr, tt.mustFail)
+			mReceiver, mErr := factory.CreateMetricsExporter(zap.NewNop(), &tt.config)
+			checkErrorsAndShutdown(t, mReceiver, mErr, tt.mustFail)
 		})
 	}
 }
+
+func checkErrorsAndShutdown(t *testing.T, receiver component.Receiver, err error, mustFail bool) {
+	if mustFail {
+		assert.NotNil(t, err)
+	} else {
+		assert.NoError(t, err)
+		assert.NotNil(t, receiver)
+
+		require.NoError(t, receiver.Shutdown(context.Background()))
+	}
+}
