diff --git a/service/telemetry/otelconftelemetry/logger.go b/service/telemetry/otelconftelemetry/logger.go
index fe6231a6f..f4b80c7fb 100644
--- a/service/telemetry/otelconftelemetry/logger.go
+++ b/service/telemetry/otelconftelemetry/logger.go
@@ -63,7 +63,7 @@ func createLogger(
 	// add them to the logger using With, because that would apply to all logs,
 	// even ones exported through the core that wraps the LoggerProvider,
 	// meaning that the attributes would be exported twice.
-	if res != nil && len(res.Attributes()) > 0 {
+	if !cfg.Logs.DisableZapResource && res != nil && len(res.Attributes()) > 0 {
 		logger = logger.WithOptions(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
 			var fields []zap.Field
 			for _, attr := range res.Attributes() {
