diff --git a/pdata/xpdata/request/traces_request.go b/pdata/xpdata/request/traces_request.go
index 3c323b464..c427ccfc7 100644
--- a/pdata/xpdata/request/traces_request.go
+++ b/pdata/xpdata/request/traces_request.go
@@ -9,19 +9,18 @@ import (
 
 	"go.opentelemetry.io/collector/pdata/internal"
 	"go.opentelemetry.io/collector/pdata/ptrace"
-	reqint "go.opentelemetry.io/collector/pdata/xpdata/request/internal"
 )
 
 // MarshalTraces marshals ptrace.Traces along with the context into a byte slice.
 func MarshalTraces(ctx context.Context, ld ptrace.Traces) ([]byte, error) {
-	otlpTraces := internal.TracesToProto(internal.TracesWrapper(ld))
-	rc := encodeContext(ctx)
-	tr := reqint.TracesRequest{
+	tr := internal.TracesRequest{
 		FormatVersion:  requestFormatVersion,
-		TracesData:     &otlpTraces,
-		RequestContext: &rc,
+		TracesData:     internal.TracesToProto(internal.TracesWrapper(ld)),
+		RequestContext: encodeContext(ctx),
 	}
-	return tr.Marshal()
+	buf := make([]byte, tr.SizeProto())
+	tr.MarshalProto(buf)
+	return buf, nil
 }
 
 // UnmarshalTraces unmarshals a byte slice into ptrace.Traces and the context.
@@ -30,9 +29,9 @@ func UnmarshalTraces(buf []byte) (context.Context, ptrace.Traces, error) {
 	if !isRequestPayloadV1(buf) {
 		return ctx, ptrace.Traces{}, ErrInvalidFormat
 	}
-	tr := reqint.TracesRequest{}
-	if err := tr.Unmarshal(buf); err != nil {
+	tr := internal.TracesRequest{}
+	if err := tr.UnmarshalProto(buf); err != nil {
 		return ctx, ptrace.Traces{}, fmt.Errorf("failed to unmarshal traces request: %w", err)
 	}
-	return decodeContext(ctx, tr.RequestContext), ptrace.Traces(internal.TracesFromProto(*tr.TracesData)), nil
+	return decodeContext(ctx, tr.RequestContext), ptrace.Traces(internal.TracesFromProto(tr.TracesData)), nil
 }
