diff --git a/config/confighttp/confighttp_test.go b/config/confighttp/confighttp_test.go
index 3b2b47cc4..fa1442aff 100644
--- a/config/confighttp/confighttp_test.go
+++ b/config/confighttp/confighttp_test.go
@@ -34,19 +34,11 @@ import (
 	"go.opentelemetry.io/collector/extension/extensionauth/extensionauthtest"
 )
 
-type customRoundTripper struct{}
-
-var _ http.RoundTripper = (*customRoundTripper)(nil)
-
-func (c *customRoundTripper) RoundTrip(*http.Request) (*http.Response, error) {
-	return nil, nil
-}
-
-func mustNewServerAuth(t *testing.T, opts ...extensionauth.ServerOption) extensionauth.Server {
+func build[T any, O any](t *testing.T, builder func(...O) (T, error), opts ...O) T {
 	t.Helper()
-	srv, err := extensionauth.NewServer(opts...)
+	obj, err := builder(opts...)
 	require.NoError(t, err)
-	return srv
+	return obj
 }
 
 var (
@@ -61,7 +53,7 @@ var (
 func TestAllHTTPClientSettings(t *testing.T) {
 	host := &mockHost{
 		ext: map[component.ID]component.Component{
-			testAuthID: &extensionauthtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+			testAuthID: build(t, extensionauth.NewClient),
 		},
 	}
 
@@ -187,7 +179,7 @@ func TestAllHTTPClientSettings(t *testing.T) {
 func TestPartialHTTPClientSettings(t *testing.T) {
 	host := &mockHost{
 		ext: map[component.ID]component.Component{
-			testAuthID: &extensionauthtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+			testAuthID: build(t, extensionauth.NewClient),
 		},
 	}
 
@@ -339,7 +331,20 @@ func TestHTTPClientSettingsError(t *testing.T) {
 	}
 }
 
+var _ http.RoundTripper = &customRoundTripper{}
+
+type customRoundTripper struct{}
+
+func (c *customRoundTripper) RoundTrip(*http.Request) (*http.Response, error) {
+	return nil, nil
+}
+
 func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
+	customRoundTripperOpt := extensionauth.WithClientRoundTripper(
+		func(http.RoundTripper) (http.RoundTripper, error) {
+			return &customRoundTripper{}, nil
+		})
+
 	tests := []struct {
 		name      string
 		shouldErr bool
@@ -355,9 +360,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 			shouldErr: false,
 			host: &mockHost{
 				ext: map[component.ID]component.Component{
-					mockID: &extensionauthtest.MockClient{
-						ResultRoundTripper: &customRoundTripper{},
-					},
+					mockID: build(t, extensionauth.NewClient),
 				},
 			},
 		},
@@ -370,7 +373,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 			shouldErr: true,
 			host: &mockHost{
 				ext: map[component.ID]component.Component{
-					mockID: &extensionauthtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+					mockID: build(t, extensionauth.NewClient),
 				},
 			},
 		},
@@ -392,7 +395,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 			shouldErr: false,
 			host: &mockHost{
 				ext: map[component.ID]component.Component{
-					mockID: &extensionauthtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+					mockID: build(t, extensionauth.NewClient, customRoundTripperOpt),
 				},
 			},
 		},
@@ -406,7 +409,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 			shouldErr: false,
 			host: &mockHost{
 				ext: map[component.ID]component.Component{
-					mockID: &extensionauthtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+					mockID: build(t, extensionauth.NewClient, customRoundTripperOpt),
 				},
 			},
 		},
@@ -420,7 +423,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 			shouldErr: false,
 			host: &mockHost{
 				ext: map[component.ID]component.Component{
-					mockID: &extensionauthtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+					mockID: build(t, extensionauth.NewClient, customRoundTripperOpt),
 				},
 			},
 		},
@@ -433,9 +436,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 			shouldErr: true,
 			host: &mockHost{
 				ext: map[component.ID]component.Component{
-					mockID: &extensionauthtest.MockClient{
-						ResultRoundTripper: &customRoundTripper{}, MustError: true,
-					},
+					mockID: build(t, extensionauthtest.NewErrorClient),
 				},
 			},
 		},
@@ -832,7 +833,7 @@ func TestHttpCorsWithSettings(t *testing.T) {
 
 	host := &mockHost{
 		ext: map[component.ID]component.Component{
-			mockID: mustNewServerAuth(t,
+			mockID: build(t, extensionauth.NewServer,
 				extensionauth.WithServerAuthenticate(func(ctx context.Context, _ map[string][]string) (context.Context, error) {
 					return ctx, errors.New("Settings failed")
 				}),
@@ -1144,7 +1145,7 @@ func TestServerAuth(t *testing.T) {
 
 	host := &mockHost{
 		ext: map[component.ID]component.Component{
-			mockID: mustNewServerAuth(t,
+			mockID: build(t, extensionauth.NewServer,
 				extensionauth.WithServerAuthenticate(func(ctx context.Context, _ map[string][]string) (context.Context, error) {
 					authCalled = true
 					return ctx, nil
@@ -1195,7 +1196,7 @@ func TestFailedServerAuth(t *testing.T) {
 	}
 	host := &mockHost{
 		ext: map[component.ID]component.Component{
-			mockID: mustNewServerAuth(t,
+			mockID: build(t, extensionauth.NewServer,
 				extensionauth.WithServerAuthenticate(func(ctx context.Context, _ map[string][]string) (context.Context, error) {
 					return ctx, errors.New("Settings failed")
 				}),
@@ -1376,7 +1377,7 @@ func TestAuthWithQueryParams(t *testing.T) {
 
 	host := &mockHost{
 		ext: map[component.ID]component.Component{
-			mockID: mustNewServerAuth(t,
+			mockID: build(t, extensionauth.NewServer,
 				extensionauth.WithServerAuthenticate(func(ctx context.Context, sources map[string][]string) (context.Context, error) {
 					require.Len(t, sources, 1)
 					assert.Equal(t, "1", sources["auth"][0])
