diff --git a/internal/collector/processor/nodebatcher/factory.go b/internal/collector/processor/nodebatcher/factory.go
index 78f930b73..cf8776f2c 100644
--- a/internal/collector/processor/nodebatcher/factory.go
+++ b/internal/collector/processor/nodebatcher/factory.go
@@ -23,24 +23,24 @@ import (
 	"github.com/open-telemetry/opentelemetry-service/processor"
 )
 
-var _ = processor.RegisterProcessorFactory(&processorFactory{})
+var _ = processor.RegisterFactory(&factory{})
 
 const (
 	// The value of "type" key in configuration.
 	typeStr = "batch"
 )
 
-// processorFactory is the factory for batch processor.
-type processorFactory struct {
+// factory is the factory for batch processor.
+type factory struct {
 }
 
 // Type gets the type of the config created by this factory.
-func (f *processorFactory) Type() string {
+func (f *factory) Type() string {
 	return typeStr
 }
 
 // CreateDefaultConfig creates the default configuration for processor.
-func (f *processorFactory) CreateDefaultConfig() models.Processor {
+func (f *factory) CreateDefaultConfig() models.Processor {
 	removeAfterTicks := int(defaultRemoveAfterCycles)
 	sendBatchSize := int(defaultSendBatchSize)
 	tickTime := defaultTickTime
@@ -60,7 +60,7 @@ func (f *processorFactory) CreateDefaultConfig() models.Processor {
 }
 
 // CreateTraceProcessor creates a trace processor based on this config.
-func (f *processorFactory) CreateTraceProcessor(
+func (f *factory) CreateTraceProcessor(
 	logger *zap.Logger,
 	nextConsumer consumer.TraceConsumer,
 	c models.Processor,
@@ -96,7 +96,7 @@ func (f *processorFactory) CreateTraceProcessor(
 }
 
 // CreateMetricsProcessor creates a metrics processor based on this config.
-func (f *processorFactory) CreateMetricsProcessor(
+func (f *factory) CreateMetricsProcessor(
 	logger *zap.Logger,
 	nextConsumer consumer.MetricsConsumer,
 	cfg models.Processor,
