diff --git a/processor/attributesprocessor/factory.go b/processor/attributesprocessor/factory.go
index 9405eab16..ef3c605a8 100644
--- a/processor/attributesprocessor/factory.go
+++ b/processor/attributesprocessor/factory.go
@@ -31,7 +31,7 @@ const (
 	typeStr = "attributes"
 )
 
-var processorCapabilities = component.ProcessorCapabilities{MutatesConsumedData: true}
+var processorCapabilities = consumer.Capabilities{MutatesData: true}
 
 // NewFactory returns a new factory for the Attributes processor.
 func NewFactory() component.ProcessorFactory {
