diff --git a/pdata/pmetric/json_test.go b/pdata/pmetric/json_test.go
index 8d1d5ac74..3030fb2fb 100644
--- a/pdata/pmetric/json_test.go
+++ b/pdata/pmetric/json_test.go
@@ -531,9 +531,7 @@ func TestExemplarInvalidTraceID(t *testing.T) {
 	iter := jsoniter.ConfigFastest.BorrowIterator([]byte(jsonStr))
 	defer jsoniter.ConfigFastest.ReturnIterator(iter)
 	NewExemplar().unmarshalJsoniter(iter)
-	if assert.Error(t, iter.Error) {
-		assert.Contains(t, iter.Error.Error(), "parse trace_id")
-	}
+	assert.ErrorContains(t, iter.Error, "parse trace_id")
 }
 
 func TestExemplarInvalidSpanID(t *testing.T) {
@@ -541,9 +539,7 @@ func TestExemplarInvalidSpanID(t *testing.T) {
 	iter := jsoniter.ConfigFastest.BorrowIterator([]byte(jsonStr))
 	defer jsoniter.ConfigFastest.ReturnIterator(iter)
 	NewExemplar().unmarshalJsoniter(iter)
-	if assert.Error(t, iter.Error) {
-		assert.Contains(t, iter.Error.Error(), "parse span_id")
-	}
+	assert.ErrorContains(t, iter.Error, "parse span_id")
 }
 
 func TestExemplar(t *testing.T) {
