diff --git a/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go b/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go
index cf5012f40..614b93dfe 100644
--- a/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go
+++ b/processor/processorhelper/internal/metadatatest/generated_telemetrytest.go
@@ -16,7 +16,7 @@ func AssertEqualProcessorIncomingItems(t *testing.T, tt *componenttest.Telemetry
 	want := metricdata.Metrics{
 		Name:        "otelcol_processor_incoming_items",
 		Description: "Number of items passed to the processor. [Alpha]",
-		Unit:        "{items}",
+		Unit:        "{item}",
 		Data: metricdata.Sum[int64]{
 			Temporality: metricdata.CumulativeTemporality,
 			IsMonotonic: true,
@@ -47,7 +47,7 @@ func AssertEqualProcessorOutgoingItems(t *testing.T, tt *componenttest.Telemetry
 	want := metricdata.Metrics{
 		Name:        "otelcol_processor_outgoing_items",
 		Description: "Number of items emitted from the processor. [Alpha]",
-		Unit:        "{items}",
+		Unit:        "{item}",
 		Data: metricdata.Sum[int64]{
 			Temporality: metricdata.CumulativeTemporality,
 			IsMonotonic: true,
