diff --git a/internal/otlptext/logs_test.go b/internal/otlptext/logs_test.go
index cafa2f1c8..69c29fd63 100644
--- a/internal/otlptext/logs_test.go
+++ b/internal/otlptext/logs_test.go
@@ -39,7 +39,8 @@ func TestLogs(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			logs := Logs(tt.args.ld)
+			logs, err := NewTextLogsMarshaler().Marshal(tt.args.ld)
+			assert.NoError(t, err)
 			if !tt.empty {
 				assert.NotEmpty(t, logs)
 			}
