diff --git a/exporter/exporterhelper/internal/hosttest/hosttest_test.go b/exporter/exporterhelper/internal/hosttest/hosttest_test.go
index e3bdd44b6..b78d5ebf4 100644
--- a/exporter/exporterhelper/internal/hosttest/hosttest_test.go
+++ b/exporter/exporterhelper/internal/hosttest/hosttest_test.go
@@ -23,6 +23,6 @@ func TestMockHost(t *testing.T) {
 	ext := map[component.ID]component.Component{
 		component.MustNewID("test"): &nopExtension{},
 	}
-	host := &MockHost{Ext: ext}
-	assert.Equal(t, ext, host.Ext)
+	host := NewHost(ext)
+	assert.Equal(t, ext, host.GetExtensions())
 }
