diff --git a/translator/trace/jaeger/protospan_to_jaegerthrift.go b/translator/trace/jaeger/protospan_to_jaegerthrift.go
index d901a0608..1b881ceb0 100644
--- a/translator/trace/jaeger/protospan_to_jaegerthrift.go
+++ b/translator/trace/jaeger/protospan_to_jaegerthrift.go
@@ -202,6 +202,8 @@ func ocSpansToJaegerSpans(ocSpans []*tracepb.Span) ([]*jaeger.Span, error) {
 			}
 		}
 
+		jSpan.Tags = appendJaegerThriftTagFromOCStatus(jSpan.Tags, ocSpan.Status)
+
 		jSpans = append(jSpans, jSpan)
 	}
 
@@ -248,6 +250,35 @@ func ocLinksToJaegerReferences(ocSpanLinks *tracepb.Span_Links) ([]*jaeger.SpanR
 	return jRefs, nil
 }
 
+func appendJaegerThriftTagFromOCStatus(jTags []*jaeger.Tag, ocStatus *tracepb.Status) []*jaeger.Tag {
+	if ocStatus == nil {
+		return jTags
+	}
+
+	for _, jt := range jTags {
+		if jt.Key == tracetranslator.TagStatusCode || jt.Key == tracetranslator.TagStatusMsg {
+			return jTags
+		}
+	}
+
+	code := int64(ocStatus.Code)
+	jTags = append(jTags, &jaeger.Tag{
+		Key:   tracetranslator.TagStatusCode,
+		VLong: &code,
+		VType: jaeger.TagType_LONG,
+	})
+
+	if ocStatus.Message != "" {
+		jTags = append(jTags, &jaeger.Tag{
+			Key:   tracetranslator.TagStatusMsg,
+			VStr:  &ocStatus.Message,
+			VType: jaeger.TagType_STRING,
+		})
+	}
+
+	return jTags
+}
+
 func appendJaegerTagFromOCSpanKind(jTags []*jaeger.Tag, ocSpanKind tracepb.Span_SpanKind) []*jaeger.Tag {
 	// TODO: (@pjanotti): Replace any OpenTracing literals by importing github.com/opentracing/opentracing-go/ext?
 	var tagValue string
