diff --git a/exporter/exporterhelper/internal/bounded_memory_queue_test.go b/exporter/exporterhelper/internal/bounded_memory_queue_test.go
index b9b689d70..176484ec0 100644
--- a/exporter/exporterhelper/internal/bounded_memory_queue_test.go
+++ b/exporter/exporterhelper/internal/bounded_memory_queue_test.go
@@ -8,9 +8,7 @@ package internal
 import (
 	"context"
 	"fmt"
-	"reflect"
-	"sync"
-	"sync/atomic"
+	"strconv"
 	"testing"
 	"time"
 
@@ -26,55 +24,45 @@ import (
 func TestBoundedQueue(t *testing.T) {
 	q := NewBoundedMemoryQueue[string](1)
 
-	waitCh := make(chan struct{})
-
-	consumerState := newConsumerState(t)
-
-	consumers := NewQueueConsumers(q, 1, func(_ context.Context, item string) error {
-		consumerState.record(item)
-		<-waitCh
-		return nil
-	})
-	assert.NoError(t, consumers.Start(context.Background(), componenttest.NewNopHost()))
-
 	assert.NoError(t, q.Offer(context.Background(), "a"))
 
-	// at this point "a" may or may not have been received by the consumer go-routine
-	// so let's make sure it has been
-	consumerState.waitToConsumeOnce()
-
-	// at this point the item must have been read off the queue, but the consumer is blocked
+	numConsumed := 0
+	assert.True(t, q.Consume(func(ctx context.Context, item string) error {
+		assert.Equal(t, "a", item)
+		numConsumed++
+		return nil
+	}))
+	assert.Equal(t, 1, numConsumed)
 	assert.Equal(t, 0, q.Size())
-	consumerState.assertConsumed(map[string]bool{
-		"a": true,
-	})
 
 	// produce two more items. The first one should be accepted, but not consumed.
 	assert.NoError(t, q.Offer(context.Background(), "b"))
 	assert.Equal(t, 1, q.Size())
+
 	// the second should be rejected since the queue is full
 	assert.ErrorIs(t, q.Offer(context.Background(), "c"), ErrQueueIsFull)
 	assert.Equal(t, 1, q.Size())
 
-	close(waitCh) // unblock consumer
-
-	consumerState.assertConsumed(map[string]bool{
-		"a": true,
-		"b": true,
-	})
-
-	// now that consumers are unblocked, we can add more items
-	expected := map[string]bool{
-		"a": true,
-		"b": true,
-	}
-	for _, item := range []string{"d", "e", "f"} {
-		assert.NoError(t, q.Offer(context.Background(), item))
-		expected[item] = true
-		consumerState.assertConsumed(expected)
+	assert.True(t, q.Consume(func(ctx context.Context, item string) error {
+		assert.Equal(t, "b", item)
+		numConsumed++
+		return nil
+	}))
+	assert.Equal(t, 2, numConsumed)
+
+	for _, toAddItem := range []string{"d", "e", "f"} {
+		assert.NoError(t, q.Offer(context.Background(), toAddItem))
+		assert.True(t, q.Consume(func(ctx context.Context, item string) error {
+			assert.Equal(t, toAddItem, item)
+			numConsumed++
+			return nil
+		}))
 	}
-
-	assert.NoError(t, consumers.Shutdown(context.Background()))
+	assert.Equal(t, 5, numConsumed)
+	assert.NoError(t, q.Shutdown(context.Background()))
+	assert.False(t, q.Consume(func(ctx context.Context, item string) error {
+		panic(item)
+	}))
 }
 
 // In this test we run a queue with many items and a slow consumer.
@@ -86,48 +74,27 @@ func TestBoundedQueue(t *testing.T) {
 func TestShutdownWhileNotEmpty(t *testing.T) {
 	q := NewBoundedMemoryQueue[string](1000)
 
-	consumerState := newConsumerState(t)
-
-	waitChan := make(chan struct{})
-	consumers := NewQueueConsumers(q, 5, func(_ context.Context, item string) error {
-		<-waitChan
-		consumerState.record(item)
-		return nil
-	})
-	assert.NoError(t, consumers.Start(context.Background(), componenttest.NewNopHost()))
+	assert.NoError(t, q.Start(context.Background(), componenttest.NewNopHost()))
+	for i := 0; i < 10; i++ {
+		assert.NoError(t, q.Offer(context.Background(), strconv.FormatInt(int64(i), 10)))
+	}
+	assert.NoError(t, q.Shutdown(context.Background()))
 
-	assert.NoError(t, q.Offer(context.Background(), "a"))
-	assert.NoError(t, q.Offer(context.Background(), "b"))
-	assert.NoError(t, q.Offer(context.Background(), "c"))
-	assert.NoError(t, q.Offer(context.Background(), "d"))
-	assert.NoError(t, q.Offer(context.Background(), "e"))
-	assert.NoError(t, q.Offer(context.Background(), "f"))
-	assert.NoError(t, q.Offer(context.Background(), "g"))
-	assert.NoError(t, q.Offer(context.Background(), "h"))
-	assert.NoError(t, q.Offer(context.Background(), "i"))
-	assert.NoError(t, q.Offer(context.Background(), "j"))
-
-	go func() {
-		assert.NoError(t, consumers.Shutdown(context.Background()))
-	}()
-
-	// wait a bit to ensure shutdown is called and unblock the consumers.
-	time.Sleep(100 * time.Millisecond)
-	close(waitChan)
-
-	consumerState.assertConsumed(map[string]bool{
-		"a": true,
-		"b": true,
-		"c": true,
-		"d": true,
-		"e": true,
-		"f": true,
-		"g": true,
-		"h": true,
-		"i": true,
-		"j": true,
-	})
+	assert.Equal(t, 10, q.Size())
+	numConsumed := 0
+	for i := 0; i < 10; i++ {
+		assert.True(t, q.Consume(func(ctx context.Context, item string) error {
+			assert.Equal(t, strconv.FormatInt(int64(i), 10), item)
+			numConsumed++
+			return nil
+		}))
+	}
+	assert.Equal(t, 10, numConsumed)
 	assert.Equal(t, 0, q.Size())
+
+	assert.False(t, q.Consume(func(ctx context.Context, item string) error {
+		panic(item)
+	}))
 }
 
 func Benchmark_QueueUsage_10000_1_50000(b *testing.B) {
@@ -188,51 +155,6 @@ func queueUsage(b *testing.B, capacity int, numConsumers int, numberOfItems int)
 	}
 }
 
-type consumerState struct {
-	sync.Mutex
-	t            *testing.T
-	consumed     map[string]bool
-	consumedOnce *atomic.Bool
-}
-
-func newConsumerState(t *testing.T) *consumerState {
-	return &consumerState{
-		t:            t,
-		consumed:     make(map[string]bool),
-		consumedOnce: &atomic.Bool{},
-	}
-}
-
-func (s *consumerState) record(val string) {
-	s.Lock()
-	defer s.Unlock()
-	s.consumed[val] = true
-	s.consumedOnce.Store(true)
-}
-
-func (s *consumerState) snapshot() map[string]bool {
-	s.Lock()
-	defer s.Unlock()
-	out := make(map[string]bool)
-	for k, v := range s.consumed {
-		out[k] = v
-	}
-	return out
-}
-
-func (s *consumerState) waitToConsumeOnce() {
-	require.Eventually(s.t, s.consumedOnce.Load, 2*time.Second, 10*time.Millisecond, "expected to consumer once")
-}
-
-func (s *consumerState) assertConsumed(expected map[string]bool) {
-	for i := 0; i < 1000; i++ {
-		if snapshot := s.snapshot(); !reflect.DeepEqual(snapshot, expected) {
-			time.Sleep(time.Millisecond)
-		}
-	}
-	assert.Equal(s.t, expected, s.snapshot())
-}
-
 func TestZeroSizeNoConsumers(t *testing.T) {
 	q := NewBoundedMemoryQueue[string](0)
 
