diff --git a/otelcol/flags_test.go b/otelcol/flags_test.go
index 3a8e3bd40..0e8bc0340 100644
--- a/otelcol/flags_test.go
+++ b/otelcol/flags_test.go
@@ -19,6 +19,8 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/featuregate"
 )
 
 func TestSetFlag(t *testing.T) {
@@ -67,7 +69,7 @@ func TestSetFlag(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			flgs := flags()
+			flgs := flags(featuregate.NewRegistry())
 			err := flgs.Parse(tt.args)
 			if tt.expectedErr != "" {
 				assert.EqualError(t, err, tt.expectedErr)
