diff --git a/cmd/occollector/app/collector/healthcheck.go b/cmd/occollector/app/collector/healthcheck.go
index 3c2b81a04..5b77fde0b 100644
--- a/cmd/occollector/app/collector/healthcheck.go
+++ b/cmd/occollector/app/collector/healthcheck.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
