diff --git a/internal/telemetry/componentattribute/attribute.go b/internal/telemetry/componentattribute/attribute.go
new file mode 100644
index 000000000..78532bb89
--- /dev/null
+++ b/internal/telemetry/componentattribute/attribute.go
@@ -0,0 +1,12 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package componentattribute // import "go.opentelemetry.io/collector/internal/telemetry/componentattribute"
+
+const (
+	ComponentKindKey = "otelcol.component.kind"
+	ComponentIDKey   = "otelcol.component.id"
+	PipelineIDKey    = "otelcol.pipeline.id"
+	SignalKey        = "otelcol.signal"
+	SignalOutputKey  = "otelcol.signal.output"
+)
