diff --git a/pdata/pprofile/generated_location.go b/pdata/pprofile/generated_location.go
index 4be12b087..81b70836f 100644
--- a/pdata/pprofile/generated_location.go
+++ b/pdata/pprofile/generated_location.go
@@ -9,7 +9,6 @@ package pprofile
 import (
 	"go.opentelemetry.io/collector/pdata/internal"
 	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
-	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
@@ -111,50 +110,3 @@ func (ms Location) CopyTo(dest Location) {
 	dest.state.AssertMutable()
 	internal.CopyOrigLocation(dest.orig, ms.orig)
 }
-
-// marshalJSONStream marshals all properties from the current struct to the destination stream.
-func (ms Location) marshalJSONStream(dest *json.Stream) {
-	dest.WriteObjectStart()
-	if ms.HasMappingIndex() {
-		dest.WriteObjectField("mappingIndex")
-		dest.WriteInt32(ms.MappingIndex())
-	}
-	if ms.orig.Address != uint64(0) {
-		dest.WriteObjectField("address")
-		dest.WriteUint64(ms.orig.Address)
-	}
-	if len(ms.orig.Line) > 0 {
-		dest.WriteObjectField("line")
-		ms.Line().marshalJSONStream(dest)
-	}
-	if ms.orig.IsFolded != false {
-		dest.WriteObjectField("isFolded")
-		dest.WriteBool(ms.orig.IsFolded)
-	}
-	if len(ms.orig.AttributeIndices) > 0 {
-		dest.WriteObjectField("attributeIndices")
-		internal.MarshalJSONStreamInt32Slice(internal.NewInt32Slice(&ms.orig.AttributeIndices, ms.state), dest)
-	}
-	dest.WriteObjectEnd()
-}
-
-// unmarshalJSONIter unmarshals all properties from the current struct from the source iterator.
-func (ms Location) unmarshalJSONIter(iter *json.Iterator) {
-	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
-		switch f {
-		case "mappingIndex", "mapping_index":
-			ms.orig.MappingIndex_ = &otlpprofiles.Location_MappingIndex{MappingIndex: iter.ReadInt32()}
-		case "address":
-			ms.orig.Address = iter.ReadUint64()
-		case "line":
-			ms.Line().unmarshalJSONIter(iter)
-		case "isFolded", "is_folded":
-			ms.orig.IsFolded = iter.ReadBool()
-		case "attributeIndices", "attribute_indices":
-			internal.UnmarshalJSONIterInt32Slice(internal.NewInt32Slice(&ms.orig.AttributeIndices, ms.state), iter)
-		default:
-			iter.Skip()
-		}
-		return true
-	})
-}
