diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index cd22f4043..f6c2e1451 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -5,7 +5,7 @@ go 1.24.0
 require (
 	github.com/foxboron/go-tpm-keyfiles v0.0.0-20250903184740-5d135037bd4d
 	github.com/fsnotify/fsnotify v1.9.0
-	github.com/google/go-tpm v0.9.6
+	github.com/google/go-tpm v0.9.7
 	github.com/stretchr/testify v1.11.1
 	go.opentelemetry.io/collector/config/configopaque v1.45.0
 	go.opentelemetry.io/collector/confmap/xconfmap v0.139.0
