diff --git a/proto_patch.sed b/proto_patch.sed
index fb5160a15..6e396adeb 100644
--- a/proto_patch.sed
+++ b/proto_patch.sed
@@ -54,6 +54,9 @@ s+\(.*\)PartialSuccess partial_success = \(.*\);+\1PartialSuccess partial_succes
 s+opentelemetry.proto.profiles.v1development.Profile \(.*\);+opentelemetry.proto.profiles.v1development.Profile \1\
   [ (gogoproto.nullable) = false ];+g
 
+s+ProfilesDictionary \(.*\);+ProfilesDictionary \1\
+	[ (gogoproto.nullable) = false ];+g
+
 s+bytes profile_id = \(.*\);+bytes profile_id = \1\
 	[\
 	// Use custom ProfileId data type for this field.\
