diff --git a/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl b/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl
index 26d8c8d47..8e7da6122 100644
--- a/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl
+++ b/internal/cmd/pdatagen/internal/templates/slice_internal.go.tmpl
@@ -32,20 +32,42 @@ func New{{ .structName }}(orig *[]{{ .originElementType }}, state *State) {{ .st
 }
 
 func CopyOrig{{ .structName }}(dest, src []{{ .originElementType }}) []{{ .originElementType }} {
+	var newDest []{{ .originElementType }}
 	if cap(dest) < len(src) {
-		dest = make([]{{ .originElementType }}, len(src))
 		{{- if eq .type "sliceOfPtrs" }}
-		data := make([]{{ .originName }}, len(src))
-		for i := range src {
-			dest[i] = &data[i]
+		newDest = make([]{{ .originElementType }}, len(src))
+		// Copy old pointers to re-use.
+		copy(newDest, dest)
+		// Add new pointers for missing elements from len(dest) to len(srt).
+		for i := len(dest); i < len(src); i++ {
+			newDest[i] = &{{ .originName }}{}
+		}
+		{{- else }}
+		newDest = make([]{{ .originElementType }}, len(src))
+		{{- end }}
+	} else {
+		newDest = dest[:len(src)]
+		// Cleanup the rest of the elements so GC can free the memory.
+		// This can happen when len(src) < len(dest) < cap(dest).
+		{{- if eq .type "sliceOfPtrs" }}
+		for i := len(src); i < len(dest); i++ {
+		dest[i] = nil
+		}
+		// Add new pointers for missing elements.
+		// This can happen when len(dest) < len(src) < cap(dest).
+		for i := len(dest); i < len(src); i++ {
+		newDest[i] = &{{ .originName }}{}
+		}
+		{{- else }}
+		for i := len(src); i < len(dest); i++ {
+		dest[i] = {{ .emptyOriginElement }}
 		}
 		{{- end }}
 	}
-	dest = dest[:len(src)]
 	for i := range src {
-		CopyOrig{{ .elementName }}({{ .originElementPtr }}dest[i], {{ .originElementPtr }}src[i])
+		CopyOrig{{ .elementName }}({{ .originElementPtr }}newDest[i], {{ .originElementPtr }}src[i])
 	}
-	return dest
+	return newDest
 }
 
 func GenerateTest{{ .structName }}() {{ .structName }} {
