diff --git a/processor/filterprocessor/filter_processor_test.go b/processor/filterprocessor/filter_processor_test.go
index f9542b00e..fca58e499 100644
--- a/processor/filterprocessor/filter_processor_test.go
+++ b/processor/filterprocessor/filter_processor_test.go
@@ -29,7 +29,6 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
-	"go.opentelemetry.io/collector/consumer/consumerdata"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/internal/goldendataset"
@@ -333,9 +332,9 @@ func TestFilterMetricProcessor(t *testing.T) {
 			ctx := context.Background()
 			assert.NoError(t, fmp.Start(ctx, nil))
 
-			mds := make([]consumerdata.MetricsData, len(test.inMN))
+			mds := make([]internaldata.MetricsData, len(test.inMN))
 			for i, metrics := range test.inMN {
-				mds[i] = consumerdata.MetricsData{
+				mds[i] = internaldata.MetricsData{
 					Metrics: metrics,
 				}
 			}
