diff --git a/extension/ballastextension/go.mod b/extension/ballastextension/go.mod
index a6035aa3f..3056be7cd 100644
--- a/extension/ballastextension/go.mod
+++ b/extension/ballastextension/go.mod
@@ -4,7 +4,7 @@ module go.opentelemetry.io/collector/extension/ballastextension
 go 1.21
 
 require (
-	github.com/stretchr/testify v1.8.4
+	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector v0.96.0
 	go.opentelemetry.io/collector/component v0.96.0
 	go.opentelemetry.io/collector/confmap v0.96.0
