diff --git a/consumer/consumererror/go.mod b/consumer/consumererror/go.mod
index 52a8a9b5e..3f8d09d07 100644
--- a/consumer/consumererror/go.mod
+++ b/consumer/consumererror/go.mod
@@ -3,7 +3,7 @@ module go.opentelemetry.io/collector/consumer/consumererror
 go 1.24
 
 require (
-	github.com/stretchr/testify v1.11.0
+	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/pdata v1.39.0
 	go.opentelemetry.io/collector/pdata/pprofile v0.133.0
 	go.opentelemetry.io/collector/pdata/testdata v0.133.0
