diff --git a/processor/batchprocessor/batch_processor.go b/processor/batchprocessor/batch_processor.go
index 3f1d69cd7..0e7d65a04 100644
--- a/processor/batchprocessor/batch_processor.go
+++ b/processor/batchprocessor/batch_processor.go
@@ -47,7 +47,7 @@ type batchProcessor struct {
 	sendBatchSize    int
 	sendBatchMaxSize int
 
-	newItem chan interface{}
+	newItem chan any
 	batch   batch
 
 	shutdownC  chan struct{}
@@ -64,7 +64,7 @@ type batch interface {
 	itemCount() int
 
 	// add item to the current batch
-	add(item interface{})
+	add(item any)
 }
 
 var _ consumer.Traces = (*batchProcessor)(nil)
@@ -85,7 +85,7 @@ func newBatchProcessor(set processor.CreateSettings, cfg *Config, batch batch, u
 		sendBatchSize:    int(cfg.SendBatchSize),
 		sendBatchMaxSize: int(cfg.SendBatchMaxSize),
 		timeout:          cfg.Timeout,
-		newItem:          make(chan interface{}, runtime.NumCPU()),
+		newItem:          make(chan any, runtime.NumCPU()),
 		batch:            batch,
 		shutdownC:        make(chan struct{}, 1),
 	}, nil
@@ -147,7 +147,7 @@ func (bp *batchProcessor) startProcessingCycle() {
 	}
 }
 
-func (bp *batchProcessor) processItem(item interface{}) {
+func (bp *batchProcessor) processItem(item any) {
 	bp.batch.add(item)
 	sent := false
 	for bp.batch.itemCount() >= bp.sendBatchSize {
@@ -226,7 +226,7 @@ func newBatchTraces(nextConsumer consumer.Traces) *batchTraces {
 }
 
 // add updates current batchTraces by adding new TraceData object
-func (bt *batchTraces) add(item interface{}) {
+func (bt *batchTraces) add(item any) {
 	td := item.(ptrace.Traces)
 	newSpanCount := td.SpanCount()
 	if newSpanCount == 0 {
@@ -296,7 +296,7 @@ func (bm *batchMetrics) itemCount() int {
 	return bm.dataPointCount
 }
 
-func (bm *batchMetrics) add(item interface{}) {
+func (bm *batchMetrics) add(item any) {
 	md := item.(pmetric.Metrics)
 
 	newDataPointCount := md.DataPointCount()
@@ -342,7 +342,7 @@ func (bl *batchLogs) itemCount() int {
 	return bl.logCount
 }
 
-func (bl *batchLogs) add(item interface{}) {
+func (bl *batchLogs) add(item any) {
 	ld := item.(plog.Logs)
 
 	newLogsCount := ld.LogRecordCount()
