diff --git a/pdata/pmetric/pmetricotlp/response.go b/pdata/pmetric/pmetricotlp/response.go
index 0ecc69303..f7754504b 100644
--- a/pdata/pmetric/pmetricotlp/response.go
+++ b/pdata/pmetric/pmetricotlp/response.go
@@ -42,7 +42,7 @@ func (ms ExportResponse) MarshalJSON() ([]byte, error) {
 	defer json.ReturnStream(dest)
 	dest.WriteObjectStart()
 	dest.WriteObjectField("partialSuccess")
-	ms.PartialSuccess().marshalJSONStream(dest)
+	internal.MarshalJSONOrigExportMetricsPartialSuccess(&ms.orig.PartialSuccess, dest)
 	dest.WriteObjectEnd()
 	return slices.Clone(dest.Buffer()), dest.Error()
 }
@@ -64,7 +64,7 @@ func (ms ExportResponse) unmarshalJSONIter(iter *json.Iterator) {
 	iter.ReadObjectCB(func(iter *json.Iterator, f string) bool {
 		switch f {
 		case "partial_success", "partialSuccess":
-			ms.PartialSuccess().unmarshalJSONIter(iter)
+			internal.UnmarshalJSONOrigExportMetricsPartialSuccess(&ms.orig.PartialSuccess, iter)
 		default:
 			iter.Skip()
 		}
