diff --git a/internal/testdata/metric.go b/internal/testdata/metric.go
index 4860f62e6..38d9ee79a 100644
--- a/internal/testdata/metric.go
+++ b/internal/testdata/metric.go
@@ -74,17 +74,6 @@ func GenerateMetricsOneMetric() pdata.Metrics {
 	return md
 }
 
-func GenerateMetricsOneSumMetric() pdata.Metrics {
-	// TODO: this is currently used in place of GenerateMetricsOneMetric
-	// because the OTLP receiver converts IntSum to Sum types and causes a
-	// failure in TestSendMetrics. Once Sum supports setting either Int
-	// or Double, this function can be removed.
-	md := GenerateMetricsOneEmptyInstrumentationLibrary()
-	rm0ils0 := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0)
-	initSumMetric(rm0ils0.Metrics().AppendEmpty())
-	return md
-}
-
 func GenerateMetricsTwoMetrics() pdata.Metrics {
 	md := GenerateMetricsOneEmptyInstrumentationLibrary()
 	rm0ils0 := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0)
@@ -93,18 +82,6 @@ func GenerateMetricsTwoMetrics() pdata.Metrics {
 	return md
 }
 
-func GenerateMetricsTwoSumMetrics() pdata.Metrics {
-	// TODO: this is currently used in place of GenerateMetricsTwoMetrics
-	// because the OTLP receiver converts IntSum to Sum types and causes a
-	// failure in TestSendMetrics. Once Sum supports setting either Int
-	// or Double, this function can be removed.
-	md := GenerateMetricsOneEmptyInstrumentationLibrary()
-	rm0ils0 := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0)
-	initSumMetric(rm0ils0.Metrics().AppendEmpty())
-	initSumMetric(rm0ils0.Metrics().AppendEmpty())
-	return md
-}
-
 func GenerateMetricsOneCounterOneSummaryMetrics() pdata.Metrics {
 	md := GenerateMetricsOneEmptyInstrumentationLibrary()
 	rm0ils0 := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0)
@@ -182,22 +159,6 @@ func GeneratMetricsAllTypesWithSampleDatapoints() pdata.Metrics {
 	return md
 }
 
-func initSumMetric(im pdata.Metric) {
-	initMetric(im, TestSumIntMetricName, pdata.MetricDataTypeSum)
-
-	idps := im.Sum().DataPoints()
-	idp0 := idps.AppendEmpty()
-	initMetricLabels1(idp0.LabelsMap())
-	idp0.SetStartTimestamp(TestMetricStartTimestamp)
-	idp0.SetTimestamp(TestMetricTimestamp)
-	idp0.SetDoubleVal(123)
-	idp1 := idps.AppendEmpty()
-	initMetricLabels2(idp1.LabelsMap())
-	idp1.SetStartTimestamp(TestMetricStartTimestamp)
-	idp1.SetTimestamp(TestMetricTimestamp)
-	idp1.SetDoubleVal(456)
-}
-
 func initGaugeIntMetric(im pdata.Metric) {
 	initMetric(im, TestGaugeIntMetricName, pdata.MetricDataTypeGauge)
 
