diff --git a/internal/otlptext/logs.go b/internal/otlptext/logs.go
index ccbdc278d..1676de15d 100644
--- a/internal/otlptext/logs.go
+++ b/internal/otlptext/logs.go
@@ -18,15 +18,15 @@ import (
 	"go.opentelemetry.io/collector/model/pdata"
 )
 
-// NewTextLogsMarshaler returns a serializer.LogsMarshaler to encode to OTLP json bytes.
+// NewTextLogsMarshaler returns a serializer.LogsMarshaler to encode to OTLP text bytes.
 func NewTextLogsMarshaler() pdata.LogsMarshaler {
-	return logsMarshaler{}
+	return textLogsMarshaler{}
 }
 
-type logsMarshaler struct{}
+type textLogsMarshaler struct{}
 
 // MarshalLogs pdata.Logs to OTLP text.
-func (logsMarshaler) MarshalLogs(ld pdata.Logs) ([]byte, error) {
+func (textLogsMarshaler) MarshalLogs(ld pdata.Logs) ([]byte, error) {
 	buf := dataBuffer{}
 	rls := ld.ResourceLogs()
 	for i := 0; i < rls.Len(); i++ {
