diff --git a/pdata/ptrace/generated_span.go b/pdata/ptrace/generated_span.go
index 0b7001510..416237a4d 100644
--- a/pdata/ptrace/generated_span.go
+++ b/pdata/ptrace/generated_span.go
@@ -10,6 +10,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal"
 	"go.opentelemetry.io/collector/pdata/internal/data"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
@@ -204,6 +205,74 @@ func (ms Span) CopyTo(dest Span) {
 	copyOrigSpan(dest.orig, ms.orig)
 }
 
+// marshalJSONStream marshals all properties from the current struct to the destination stream.
+func (ms Span) marshalJSONStream(dest *json.Stream) {
+	dest.WriteObjectStart()
+	if ms.orig.TraceId != data.TraceID([16]byte{}) {
+		dest.WriteObjectField("traceId")
+		ms.orig.TraceId.MarshalJSONStream(dest)
+	}
+	if ms.orig.SpanId != data.SpanID([8]byte{}) {
+		dest.WriteObjectField("spanId")
+		ms.orig.SpanId.MarshalJSONStream(dest)
+	}
+	if ms.orig.TraceState != "" {
+		dest.WriteObjectField("traceState")
+		internal.MarshalJSONStreamTraceState(internal.NewTraceState(&ms.orig.TraceState, ms.state), dest)
+	}
+	if ms.orig.ParentSpanId != data.SpanID([8]byte{}) {
+		dest.WriteObjectField("parentSpanId")
+		ms.orig.ParentSpanId.MarshalJSONStream(dest)
+	}
+	if ms.orig.Name != "" {
+		dest.WriteObjectField("name")
+		dest.WriteString(ms.orig.Name)
+	}
+	if ms.orig.Flags != uint32(0) {
+		dest.WriteObjectField("flags")
+		dest.WriteUint32(ms.orig.Flags)
+	}
+	if ms.orig.Kind != otlptrace.Span_SpanKind(0) {
+		dest.WriteObjectField("kind")
+		ms.Kind().marshalJSONStream(dest)
+	}
+	if ms.orig.StartTimeUnixNano != 0 {
+		dest.WriteObjectField("startTimeUnixNano")
+		dest.WriteUint64(ms.orig.StartTimeUnixNano)
+	}
+	if ms.orig.EndTimeUnixNano != 0 {
+		dest.WriteObjectField("endTimeUnixNano")
+		dest.WriteUint64(ms.orig.EndTimeUnixNano)
+	}
+	if len(ms.orig.Attributes) > 0 {
+		dest.WriteObjectField("attributes")
+		internal.MarshalJSONStreamMap(internal.NewMap(&ms.orig.Attributes, ms.state), dest)
+	}
+	if ms.orig.DroppedAttributesCount != uint32(0) {
+		dest.WriteObjectField("droppedAttributesCount")
+		dest.WriteUint32(ms.orig.DroppedAttributesCount)
+	}
+	if len(ms.orig.Events) > 0 {
+		dest.WriteObjectField("events")
+		ms.Events().marshalJSONStream(dest)
+	}
+	if ms.orig.DroppedEventsCount != uint32(0) {
+		dest.WriteObjectField("droppedEventsCount")
+		dest.WriteUint32(ms.orig.DroppedEventsCount)
+	}
+	if len(ms.orig.Links) > 0 {
+		dest.WriteObjectField("links")
+		ms.Links().marshalJSONStream(dest)
+	}
+	if ms.orig.DroppedLinksCount != uint32(0) {
+		dest.WriteObjectField("droppedLinksCount")
+		dest.WriteUint32(ms.orig.DroppedLinksCount)
+	}
+	dest.WriteObjectField("status")
+	ms.Status().marshalJSONStream(dest)
+	dest.WriteObjectEnd()
+}
+
 func copyOrigSpan(dest, src *otlptrace.Span) {
 	dest.TraceId = src.TraceId
 	dest.SpanId = src.SpanId
