diff --git a/pdata/ptrace/generated_spanevent_test.go b/pdata/ptrace/generated_spanevent_test.go
index 707dfec8e..4bab2abf1 100644
--- a/pdata/ptrace/generated_spanevent_test.go
+++ b/pdata/ptrace/generated_spanevent_test.go
@@ -13,6 +13,7 @@ import (
 
 	"go.opentelemetry.io/collector/pdata/internal"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
@@ -41,6 +42,20 @@ func TestSpanEvent_CopyTo(t *testing.T) {
 	assert.Panics(t, func() { ms.CopyTo(newSpanEvent(&otlptrace.Span_Event{}, &sharedState)) })
 }
 
+func TestSpanEvent_MarshalAndUnmarshal(t *testing.T) {
+	stream := json.BorrowStream(nil)
+	defer json.ReturnStream(stream)
+	src := generateTestSpanEvent()
+	src.marshalJSONStream(stream)
+
+	iter := json.BorrowIterator(stream.Buffer())
+	defer json.ReturnIterator(iter)
+	dest := NewSpanEvent()
+	dest.unmarshalJsoniter(iter)
+
+	assert.Equal(t, src, dest)
+}
+
 func TestSpanEvent_Timestamp(t *testing.T) {
 	ms := NewSpanEvent()
 	assert.Equal(t, pcommon.Timestamp(0), ms.Timestamp())
