diff --git a/pdata/ptrace/generated_status.go b/pdata/ptrace/generated_status.go
index e39f91c95..d21d9015f 100644
--- a/pdata/ptrace/generated_status.go
+++ b/pdata/ptrace/generated_status.go
@@ -9,6 +9,7 @@ package ptrace
 import (
 	"go.opentelemetry.io/collector/pdata/internal"
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 // Status is an optional final status for this span. Semantically, when Status was not
@@ -78,6 +79,20 @@ func (ms Status) CopyTo(dest Status) {
 	copyOrigStatus(dest.orig, ms.orig)
 }
 
+// marshalJSONStream marshals all properties from the current struct to the destination stream.
+func (ms Status) marshalJSONStream(dest *json.Stream) {
+	dest.WriteObjectStart()
+	if ms.orig.Code != 0 {
+		dest.WriteObjectField("code")
+		ms.Code().marshalJSONStream(dest)
+	}
+	if ms.orig.Message != "" {
+		dest.WriteObjectField("message")
+		dest.WriteString(ms.orig.Message)
+	}
+	dest.WriteObjectEnd()
+}
+
 func copyOrigStatus(dest, src *otlptrace.Status) {
 	dest.Code = src.Code
 	dest.Message = src.Message
