diff --git a/receiver/jaegerreceiver/trace_receiver_test.go b/receiver/jaegerreceiver/trace_receiver_test.go
index 407a8d70c..b1f18687b 100644
--- a/receiver/jaegerreceiver/trace_receiver_test.go
+++ b/receiver/jaegerreceiver/trace_receiver_test.go
@@ -309,8 +309,8 @@ func expectedTraceData(t1, t2, t3 time.Time) pdata.Traces {
 	span0.SetParentSpanID(parentSpanID)
 	span0.SetTraceID(traceID)
 	span0.SetName("DBSearch")
-	span0.SetStartTime(pdata.TimestampUnixNano(uint64(t1.UnixNano())))
-	span0.SetEndTime(pdata.TimestampUnixNano(uint64(t2.UnixNano())))
+	span0.SetStartTime(pdata.TimestampFromTime(t1))
+	span0.SetEndTime(pdata.TimestampFromTime(t2))
 	span0.Status().SetCode(pdata.StatusCodeError)
 	span0.Status().SetMessage("Stale indices")
 
@@ -318,8 +318,8 @@ func expectedTraceData(t1, t2, t3 time.Time) pdata.Traces {
 	span1.SetSpanID(parentSpanID)
 	span1.SetTraceID(traceID)
 	span1.SetName("ProxyFetch")
-	span1.SetStartTime(pdata.TimestampUnixNano(uint64(t2.UnixNano())))
-	span1.SetEndTime(pdata.TimestampUnixNano(uint64(t3.UnixNano())))
+	span1.SetStartTime(pdata.TimestampFromTime(t2))
+	span1.SetEndTime(pdata.TimestampFromTime(t3))
 	span1.Status().SetCode(pdata.StatusCodeError)
 	span1.Status().SetMessage("Frontend crash")
 
