diff --git a/receiver/receiverhelper/generated_component_telemetry_test.go b/receiver/receiverhelper/generated_component_telemetry_test.go
index 41d3c0bf8..e858cbac4 100644
--- a/receiver/receiverhelper/generated_component_telemetry_test.go
+++ b/receiver/receiverhelper/generated_component_telemetry_test.go
@@ -21,8 +21,8 @@ type componentTestTelemetry struct {
 	meterProvider *sdkmetric.MeterProvider
 }
 
-func (tt *componentTestTelemetry) NewCreateSettings() receiver.CreateSettings {
-	settings := receivertest.NewNopCreateSettings()
+func (tt *componentTestTelemetry) NewSettings() receiver.Settings {
+	settings := receivertest.NewNopSettings()
 	settings.MeterProvider = tt.meterProvider
 	settings.ID = component.NewID(component.MustNewType("receiverhelper"))
 
