diff --git a/processor/batchprocessor/go.mod b/processor/batchprocessor/go.mod
index 4973b0267..174edf493 100644
--- a/processor/batchprocessor/go.mod
+++ b/processor/batchprocessor/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/stretchr/testify v1.8.1
 	go.opencensus.io v0.24.0
 	go.opentelemetry.io/collector v0.64.1
-	go.opentelemetry.io/collector/component v0.64.1
+	go.opentelemetry.io/collector/component v0.0.0-20221115212826-19db2f15c85c
 	go.opentelemetry.io/collector/pdata v0.64.1
 	go.uber.org/zap v1.23.0
 )
