diff --git a/consumer/consumer.go b/consumer/consumer.go
index acf3baa3c..2accdb79d 100644
--- a/consumer/consumer.go
+++ b/consumer/consumer.go
@@ -22,40 +22,40 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/internal/data"
 )
 
-// BaseMetricsConsumer defines a common interface for MetricsConsumer and MetricsConsumerV2.
+// BaseMetricsConsumer defines a common interface for MetricsConsumerOld and MetricsConsumer.
 type BaseMetricsConsumer interface{}
 
-// MetricsConsumer is an interface that receives consumerdata.MetricsData, process it as needed, and
+// MetricsConsumerOld is an interface that receives consumerdata.MetricsData, process it as needed, and
 // sends it to the next processing node if any or to the destination.
 //
 // ConsumeMetricsData receives consumerdata.MetricsData for processing by the MetricsConsumer.
-type MetricsConsumer interface {
+type MetricsConsumerOld interface {
 	BaseMetricsConsumer
 	ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error
 }
 
-// MetricsConsumerV2 is the new metrics consumer interface that receives data.MetricData, processes it
+// MetricsConsumer is the new metrics consumer interface that receives data.MetricData, processes it
 // as needed, and sends it to the next processing node if any or to the destination.
-type MetricsConsumerV2 interface {
+type MetricsConsumer interface {
 	BaseMetricsConsumer
 	ConsumeMetrics(ctx context.Context, md data.MetricData) error
 }
 
-// BaseTraceConsumer defines a common interface for TraceConsumer and TraceConsumerV2.
+// BaseTraceConsumer defines a common interface for TraceConsumerOld and TraceConsumer.
 type BaseTraceConsumer interface{}
 
-// TraceConsumer is an interface that receives consumerdata.TraceData, process it as needed, and
+// TraceConsumerOld is an interface that receives consumerdata.TraceData, process it as needed, and
 // sends it to the next processing node if any or to the destination.
 //
 // ConsumeTraceData receives consumerdata.TraceData for processing by the TraceConsumer.
-type TraceConsumer interface {
+type TraceConsumerOld interface {
 	BaseTraceConsumer
 	ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error
 }
 
-// TraceConsumerV2 is an interface that receives data.TraceData, processes it
+// TraceConsumer is an interface that receives data.TraceData, processes it
 // as needed, and sends it to the next processing node if any or to the destination.
-type TraceConsumerV2 interface {
+type TraceConsumer interface {
 	BaseTraceConsumer
 	// ConsumeTrace receives data.TraceData for processing.
 	ConsumeTrace(ctx context.Context, td data.TraceData) error
