diff --git a/service/collector_windows_test.go b/service/collector_windows_test.go
index 05e070c87..e79d547d8 100644
--- a/service/collector_windows_test.go
+++ b/service/collector_windows_test.go
@@ -30,7 +30,7 @@ import (
 )
 
 func TestWindowsService_Execute(t *testing.T) {
-	os.Args = []string{"otelcol", "--config", "testdata/otelcol-config-minimal.yaml"}
+	os.Args = []string{"otelcol", "--config", "testdata/otelcol-config.yaml"}
 
 	factories, err := defaultcomponents.Components()
 	require.NoError(t, err)
