diff --git a/service/collector_test.go b/service/collector_test.go
index 5410abce0..9aa7dced7 100644
--- a/service/collector_test.go
+++ b/service/collector_test.go
@@ -145,7 +145,7 @@ func TestCollector_ReportError(t *testing.T) {
 	require.NoError(t, err)
 
 	cmd := NewCommand(col)
-	cmd.SetArgs([]string{"--config=testdata/otelcol-config-minimal.yaml"})
+	cmd.SetArgs([]string{"--config=testdata/otelcol-config.yaml"})
 
 	colDone := make(chan struct{})
 	go func() {
