diff --git a/exporter/exporterhelper/traces_test.go b/exporter/exporterhelper/traces_test.go
index 4b5950c22..b0e339c6a 100644
--- a/exporter/exporterhelper/traces_test.go
+++ b/exporter/exporterhelper/traces_test.go
@@ -184,11 +184,10 @@ func TestTraces_WithPersistentQueue(t *testing.T) {
 }
 
 func TestTraces_WithRecordMetrics(t *testing.T) {
-	tt, err := componenttest.SetupTelemetry(fakeTracesName)
-	require.NoError(t, err)
+	tt := componenttest.NewTelemetry()
 	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
 
-	te, err := NewTraces(context.Background(), exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()}, &fakeTracesConfig, newTraceDataPusher(nil))
+	te, err := NewTraces(context.Background(), exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.NewTelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()}, &fakeTracesConfig, newTraceDataPusher(nil))
 	require.NoError(t, err)
 	require.NotNil(t, te)
 
@@ -196,11 +195,10 @@ func TestTraces_WithRecordMetrics(t *testing.T) {
 }
 
 func TestTraces_pLogModifiedDownStream_WithRecordMetrics(t *testing.T) {
-	tt, err := componenttest.SetupTelemetry(fakeTracesName)
-	require.NoError(t, err)
+	tt := componenttest.NewTelemetry()
 	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
 
-	te, err := NewTraces(context.Background(), exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()}, &fakeTracesConfig, newTraceDataPusherModifiedDownstream(nil), WithCapabilities(consumer.Capabilities{MutatesData: true}))
+	te, err := NewTraces(context.Background(), exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.NewTelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()}, &fakeTracesConfig, newTraceDataPusherModifiedDownstream(nil), WithCapabilities(consumer.Capabilities{MutatesData: true}))
 	assert.NotNil(t, te)
 	require.NoError(t, err)
 	td := testdata.GenerateTraces(2)
@@ -208,7 +206,7 @@ func TestTraces_pLogModifiedDownStream_WithRecordMetrics(t *testing.T) {
 	require.NoError(t, te.ConsumeTraces(context.Background(), td))
 	assert.Equal(t, 0, td.SpanCount())
 
-	metadatatest.AssertEqualExporterSentSpans(t, tt.Telemetry,
+	metadatatest.AssertEqualExporterSentSpans(t, tt,
 		[]metricdata.DataPoint[int64]{
 			{
 				Attributes: attribute.NewSet(
@@ -219,12 +217,11 @@ func TestTraces_pLogModifiedDownStream_WithRecordMetrics(t *testing.T) {
 }
 
 func TestTracesRequest_WithRecordMetrics(t *testing.T) {
-	tt, err := componenttest.SetupTelemetry(fakeTracesName)
-	require.NoError(t, err)
+	tt := componenttest.NewTelemetry()
 	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
 
 	te, err := NewTracesRequest(context.Background(),
-		exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
+		exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.NewTelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
 		requesttest.RequestFromTracesFunc(nil))
 	require.NoError(t, err)
 	require.NotNil(t, te)
@@ -234,11 +231,10 @@ func TestTracesRequest_WithRecordMetrics(t *testing.T) {
 
 func TestTraces_WithRecordMetrics_ReturnError(t *testing.T) {
 	want := errors.New("my_error")
-	tt, err := componenttest.SetupTelemetry(fakeTracesName)
-	require.NoError(t, err)
+	tt := componenttest.NewTelemetry()
 	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
 
-	te, err := NewTraces(context.Background(), exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()}, &fakeTracesConfig, newTraceDataPusher(want))
+	te, err := NewTraces(context.Background(), exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.NewTelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()}, &fakeTracesConfig, newTraceDataPusher(want))
 	require.NoError(t, err)
 	require.NotNil(t, te)
 
@@ -247,12 +243,11 @@ func TestTraces_WithRecordMetrics_ReturnError(t *testing.T) {
 
 func TestTracesRequest_WithRecordMetrics_RequestSenderError(t *testing.T) {
 	want := errors.New("export_error")
-	tt, err := componenttest.SetupTelemetry(fakeTracesName)
-	require.NoError(t, err)
+	tt := componenttest.NewTelemetry()
 	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
 
 	te, err := NewTracesRequest(context.Background(),
-		exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.TelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
+		exporter.Settings{ID: fakeTracesName, TelemetrySettings: tt.NewTelemetrySettings(), BuildInfo: component.NewDefaultBuildInfo()},
 		requesttest.RequestFromTracesFunc(want))
 	require.NoError(t, err)
 	require.NotNil(t, te)
@@ -383,7 +378,7 @@ func newTraceDataPusherModifiedDownstream(retError error) consumer.ConsumeTraces
 	}
 }
 
-func checkRecordedMetricsForTraces(t *testing.T, tt componenttest.TestTelemetry, id component.ID, te exporter.Traces, wantError error) {
+func checkRecordedMetricsForTraces(t *testing.T, tt *componenttest.Telemetry, id component.ID, te exporter.Traces, wantError error) {
 	td := testdata.GenerateTraces(2)
 	const numBatches = 7
 	for i := 0; i < numBatches; i++ {
@@ -392,7 +387,7 @@ func checkRecordedMetricsForTraces(t *testing.T, tt componenttest.TestTelemetry,
 
 	// TODO: When the new metrics correctly count partial dropped fix this.
 	if wantError != nil {
-		metadatatest.AssertEqualExporterSendFailedSpans(t, tt.Telemetry,
+		metadatatest.AssertEqualExporterSendFailedSpans(t, tt,
 			[]metricdata.DataPoint[int64]{
 				{
 					Attributes: attribute.NewSet(
@@ -401,7 +396,7 @@ func checkRecordedMetricsForTraces(t *testing.T, tt componenttest.TestTelemetry,
 				},
 			}, metricdatatest.IgnoreTimestamp(), metricdatatest.IgnoreExemplars())
 	} else {
-		metadatatest.AssertEqualExporterSentSpans(t, tt.Telemetry,
+		metadatatest.AssertEqualExporterSentSpans(t, tt,
 			[]metricdata.DataPoint[int64]{
 				{
 					Attributes: attribute.NewSet(
