diff --git a/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_windows_test.go b/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_windows_test.go
index 34acdea35..565d9c56b 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_windows_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_windows_test.go
@@ -24,10 +24,11 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal/perfcounters"
 )
 
-func TestScrapeMetrics_Errors(t *testing.T) {
+func TestScrape_Errors(t *testing.T) {
 	type testCase struct {
 		name              string
 		pageSize          uint64
@@ -36,6 +37,7 @@ func TestScrapeMetrics_Errors(t *testing.T) {
 		getObjectErr      error
 		getValuesErr      error
 		expectedErr       string
+		expectedErrCount  int
 		expectedUsedValue int64
 		expectedFreeValue int64
 	}
@@ -57,27 +59,32 @@ func TestScrapeMetrics_Errors(t *testing.T) {
 			name:             "pageFileError",
 			getPageFileStats: func() ([]*pageFileData, error) { return nil, errors.New("err1") },
 			expectedErr:      "err1",
+			expectedErrCount: swapUsageMetricsLen,
 		},
 		{
-			name:        "scrapeError",
-			scrapeErr:   errors.New("err1"),
-			expectedErr: "err1",
+			name:             "scrapeError",
+			scrapeErr:        errors.New("err1"),
+			expectedErr:      "err1",
+			expectedErrCount: pagingMetricsLen,
 		},
 		{
-			name:         "getObjectErr",
-			getObjectErr: errors.New("err1"),
-			expectedErr:  "err1",
+			name:             "getObjectErr",
+			getObjectErr:     errors.New("err1"),
+			expectedErr:      "err1",
+			expectedErrCount: pagingMetricsLen,
 		},
 		{
-			name:         "getValuesErr",
-			getValuesErr: errors.New("err1"),
-			expectedErr:  "err1",
+			name:             "getValuesErr",
+			getValuesErr:     errors.New("err1"),
+			expectedErr:      "err1",
+			expectedErrCount: pagingMetricsLen,
 		},
 		{
 			name:             "multipleErrors",
 			getPageFileStats: func() ([]*pageFileData, error) { return nil, errors.New("err1") },
 			getObjectErr:     errors.New("err2"),
 			expectedErr:      "[err1; err2]",
+			expectedErrCount: swapUsageMetricsLen + pagingMetricsLen,
 		},
 	}
 
@@ -97,11 +104,17 @@ func TestScrapeMetrics_Errors(t *testing.T) {
 
 			err := scraper.Initialize(context.Background())
 			require.NoError(t, err, "Failed to initialize swap scraper: %v", err)
-			defer func() { assert.NoError(t, scraper.Close(context.Background())) }()
 
-			metrics, err := scraper.ScrapeMetrics(context.Background())
+			metrics, err := scraper.Scrape(context.Background())
 			if test.expectedErr != "" {
 				assert.EqualError(t, err, test.expectedErr)
+
+				isPartial := consumererror.IsPartialScrapeError(err)
+				assert.True(t, isPartial)
+				if isPartial {
+					assert.Equal(t, test.expectedErrCount, err.(consumererror.PartialScrapeError).Failed)
+				}
+
 				return
 			}
 
