diff --git a/extension/ballastextension/go.mod b/extension/ballastextension/go.mod
index 26c929916..d6671acca 100644
--- a/extension/ballastextension/go.mod
+++ b/extension/ballastextension/go.mod
@@ -1,7 +1,7 @@
 // Deprecated: Use the GOMEMLIMIT environment variable instead.
 module go.opentelemetry.io/collector/extension/ballastextension
 
-go 1.21.0
+go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.9.0
