diff --git a/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_linux.go b/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_linux.go
index 5158eff9b..26dec4e23 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_linux.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper_linux.go
@@ -32,9 +32,7 @@ func appendUnixSystemSpecificProcessesMetrics(metrics pdata.MetricSlice, startIn
 
 func initializeProcessesCreatedMetric(metric pdata.Metric, now pdata.Timestamp, misc *load.MiscStat) {
 	metadata.Metrics.SystemProcessesCreated.Init(metric)
-
-	ddps := metric.IntSum().DataPoints()
-	ddps.Resize(1)
-	ddps.At(0).SetTimestamp(now)
-	ddps.At(0).SetValue(int64(misc.ProcsCreated))
+	ddp := metric.IntSum().DataPoints().AppendEmpty()
+	ddp.SetTimestamp(now)
+	ddp.SetValue(int64(misc.ProcsCreated))
 }
