diff --git a/config/configoptional/package_test.go b/config/configoptional/package_test.go
deleted file mode 100644
index c64bec583..000000000
--- a/config/configoptional/package_test.go
+++ /dev/null
@@ -1,14 +0,0 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
-
-package configoptional
-
-import (
-	"testing"
-
-	"go.uber.org/goleak"
-)
-
-func TestMain(m *testing.M) {
-	goleak.VerifyTestMain(m)
-}
