diff --git a/client/go.mod b/client/go.mod
index 284fd4d7d..b07b427c4 100644
--- a/client/go.mod
+++ b/client/go.mod
@@ -4,8 +4,8 @@ go 1.23.0
 
 require (
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/consumer v1.36.1
-	go.opentelemetry.io/collector/pdata v1.36.1
+	go.opentelemetry.io/collector/consumer v1.37.0
+	go.opentelemetry.io/collector/pdata v1.37.0
 	go.uber.org/goleak v1.3.0
 )
 
