diff --git a/pdata/internal/wrapper_slice.go b/pdata/internal/wrapper_slice.go
index 94317dc01..a1f2a9780 100644
--- a/pdata/internal/wrapper_slice.go
+++ b/pdata/internal/wrapper_slice.go
@@ -4,8 +4,6 @@
 package internal // import "go.opentelemetry.io/collector/pdata/internal"
 
 import (
-	jsoniter "github.com/json-iterator/go"
-
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 )
@@ -67,8 +65,8 @@ func MarshalJSONStreamSlice(ms Slice, dest *json.Stream) {
 	dest.WriteArrayEnd()
 }
 
-func UnmarshalJSONIterSlice(ms Slice, iter *jsoniter.Iterator) {
-	iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+func UnmarshalJSONIterSlice(ms Slice, iter *json.Iterator) {
+	iter.ReadArrayCB(func(iter *json.Iterator) bool {
 		*ms.orig = append(*ms.orig, otlpcommon.AnyValue{})
 		UnmarshalJSONIterValue(NewValue(&(*ms.orig)[len(*ms.orig)-1], ms.state), iter)
 		return true
