diff --git a/featuregate/gates.go b/featuregate/gates.go
index 0817a5b31..c31eeb770 100644
--- a/featuregate/gates.go
+++ b/featuregate/gates.go
@@ -19,12 +19,106 @@ import (
 	"sync"
 )
 
-// Gate represents an individual feature that may be enabled or disabled based
+// Stage represents the Gate's lifecycle and what is the expected state of it.
+type Stage int8
+
+const (
+	// Alpha is used when creating a new feature and the Gate must be explicitly enabled
+	// by the operator.
+	//
+	// The Gate will be disabled by default.
+	Alpha Stage = iota
+	// Beta is used when the feature flag is well tested and is enabled by default,
+	// but can be disabled by a Gate.
+	//
+	// The Gate will be enabled by default.
+	Beta
+	// Stable is used when feature is permanently enabled and can not be disabled by a Gate.
+	// This value is used to provide feedback to the user that the gate will be removed in the next version.
+	//
+	// The Gate will be enabled by default and will return an error if modified.
+	Stable
+)
+
+func (s Stage) String() string {
+	switch s {
+	case Alpha:
+		return "Alpha"
+	case Beta:
+		return "Beta"
+	case Stable:
+		return "Stable"
+	}
+	return "unknown"
+}
+
+// Gate is an immutable object that is owned by the `Registry`
+// to represents an individual feature that may be enabled or disabled based
 // on the lifecycle state of the feature and CLI flags specified by the user.
 type Gate struct {
-	ID          string
+	// Deprecated: [v0.64.0] Use GetID() instead to read,
+	//				use `Registry.RegisterID` to set value.
+	ID string
+	// Deprecated: [v0.64.0] use GetDescription to read,
+	// 				use `WithRegisterDescription` to set using `Registry.RegisterID`.
 	Description string
-	Enabled     bool
+	// Deprecated: [v0.64.0] use `IsEnabled(id)` to read,
+	//              use `Registry.Apply` to set.
+	Enabled        bool
+	stage          Stage
+	referenceURL   string
+	removalVersion string
+}
+
+// RegistryOption allows for configuration additional information
+// about a gate that can be exposed throughout the application
+type RegistryOption func(g *Gate)
+
+// WithRegisterDescription adds the description to the provided `Gate“.
+func WithRegisterDescription(description string) RegistryOption {
+	return func(g *Gate) {
+		g.Description = description
+	}
+}
+
+// WithRegisterReferenceURL adds an URL that has
+// all the contextual information about the `Gate`.
+func WithRegisterReferenceURL(url string) RegistryOption {
+	return func(g *Gate) {
+		g.referenceURL = url
+	}
+}
+
+// WithRegisterRemovalVersion is used when the `Gate` is considered `Stable`,
+// to inform users that referencing the gate is no longer needed.
+func WithRegisterRemovalVersion(version string) RegistryOption {
+	return func(g *Gate) {
+		g.removalVersion = version
+	}
+}
+
+func (g *Gate) GetID() string {
+	return g.ID
+}
+
+func (g *Gate) IsEnabled() bool {
+	return g.Enabled
+}
+
+func (g *Gate) GetDescription() string {
+	return g.Description
+}
+
+func (g *Gate) GetStage() Stage {
+	return g.stage
+}
+
+func (g *Gate) GetReferenceURL() string {
+	return g.referenceURL
+}
+
+func (g *Gate) GetRemovalVersion() string {
+	return g.removalVersion
 }
 
 var reg = NewRegistry()
@@ -54,6 +148,9 @@ func (r *Registry) Apply(cfg map[string]bool) error {
 		if !ok {
 			return fmt.Errorf("feature gate %s is unregistered", id)
 		}
+		if g.stage == Stable {
+			return fmt.Errorf("feature gate %s is stable, can not be modified", id)
+		}
 		g.Enabled = val
 		r.gates[g.ID] = g
 	}
@@ -69,6 +166,7 @@ func (r *Registry) IsEnabled(id string) bool {
 }
 
 // MustRegister like Register but panics if a Gate with the same ID is already registered.
+// Deprecated: [v0.64.0] Use MustRegisterID instead.
 func (r *Registry) MustRegister(g Gate) {
 	if err := r.Register(g); err != nil {
 		panic(err)
@@ -76,6 +174,7 @@ func (r *Registry) MustRegister(g Gate) {
 }
 
 // Register registers a Gate. May only be called in an init() function.
+// Deprecated: [v0.64.0] Use RegisterID instead.
 func (r *Registry) Register(g Gate) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
@@ -86,6 +185,41 @@ func (r *Registry) Register(g Gate) error {
 	return nil
 }
 
+// MustRegisterID like RegisterID but panics if an invalid ID or gate options are provided.
+func (r *Registry) MustRegisterID(id string, stage Stage, opts ...RegistryOption) {
+	if err := r.RegisterID(id, stage, opts...); err != nil {
+		panic(err)
+	}
+}
+
+func (r *Registry) RegisterID(id string, stage Stage, opts ...RegistryOption) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if _, ok := r.gates[id]; ok {
+		return fmt.Errorf("attempted to add pre-existing gate %q", id)
+	}
+	g := Gate{
+		ID:    id,
+		stage: stage,
+	}
+	for _, opt := range opts {
+		opt(&g)
+	}
+	switch g.stage {
+	case Alpha:
+		g.Enabled = false
+	case Beta, Stable:
+		g.Enabled = true
+	default:
+		return fmt.Errorf("unknown stage value %q for gate %q", stage, id)
+	}
+	if g.stage == Stable && g.removalVersion == "" {
+		return fmt.Errorf("no removal version set for stable gate %q", id)
+	}
+	r.gates[id] = g
+	return nil
+}
+
 // List returns a slice of copies of all registered Gates.
 func (r *Registry) List() []Gate {
 	r.mu.RLock()
