diff --git a/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_others.go b/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_others.go
index e9eaba4da..1de7e3afb 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_others.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_scraper_others.go
@@ -23,6 +23,7 @@ import (
 	"github.com/shirou/gopsutil/host"
 	"github.com/shirou/gopsutil/mem"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
@@ -50,8 +51,7 @@ func newSwapScraper(_ context.Context, cfg *Config) *scraper {
 	return &scraper{config: cfg, bootTime: host.BootTime, virtualMemory: mem.VirtualMemory, swapMemory: mem.SwapMemory}
 }
 
-// Initialize
-func (s *scraper) Initialize(_ context.Context) error {
+func (s *scraper) start(context.Context, component.Host) error {
 	bootTime, err := s.bootTime()
 	if err != nil {
 		return err
@@ -61,8 +61,7 @@ func (s *scraper) Initialize(_ context.Context) error {
 	return nil
 }
 
-// Scrape
-func (s *scraper) Scrape(_ context.Context) (pdata.MetricSlice, error) {
+func (s *scraper) scrape(_ context.Context) (pdata.MetricSlice, error) {
 	metrics := pdata.NewMetricSlice()
 
 	var errors []error
