diff --git a/internal/otlptext/traces_test.go b/internal/otlptext/traces_test.go
index 3591f9483..e91daa73a 100644
--- a/internal/otlptext/traces_test.go
+++ b/internal/otlptext/traces_test.go
@@ -37,7 +37,7 @@ func TestTraces(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			traces, err := NewTextTracesMarshaler().Marshal(tt.args.td)
+			traces, err := NewTextTracesMarshaler().MarshalTraces(tt.args.td)
 			assert.NoError(t, err)
 			if !tt.empty {
 				assert.NotEmpty(t, traces)
