diff --git a/exporter/prometheusremotewriteexporter/exporter.go b/exporter/prometheusremotewriteexporter/exporter.go
index be41159d5..e309e0eb7 100644
--- a/exporter/prometheusremotewriteexporter/exporter.go
+++ b/exporter/prometheusremotewriteexporter/exporter.go
@@ -131,7 +131,7 @@ func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
 					switch metric.DataType() {
 					case pdata.MetricDataTypeGauge:
 						dataPoints := metric.Gauge().DataPoints()
-						if err := prwe.addDoubleDataPointSlice(dataPoints, tsMap, resource, metric); err != nil {
+						if err := prwe.addNumberDataPointSlice(dataPoints, tsMap, resource, metric); err != nil {
 							dropped++
 							errs = append(errs, err)
 						}
@@ -143,7 +143,7 @@ func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
 						}
 					case pdata.MetricDataTypeSum:
 						dataPoints := metric.Sum().DataPoints()
-						if err := prwe.addDoubleDataPointSlice(dataPoints, tsMap, resource, metric); err != nil {
+						if err := prwe.addNumberDataPointSlice(dataPoints, tsMap, resource, metric); err != nil {
 							dropped++
 							errs = append(errs, err)
 						}
@@ -230,12 +230,12 @@ func (prwe *PRWExporter) addIntDataPointSlice(dataPoints pdata.IntDataPointSlice
 	return nil
 }
 
-func (prwe *PRWExporter) addDoubleDataPointSlice(dataPoints pdata.DoubleDataPointSlice, tsMap map[string]*prompb.TimeSeries, resource pdata.Resource, metric pdata.Metric) error {
+func (prwe *PRWExporter) addNumberDataPointSlice(dataPoints pdata.NumberDataPointSlice, tsMap map[string]*prompb.TimeSeries, resource pdata.Resource, metric pdata.Metric) error {
 	if dataPoints.Len() == 0 {
 		return consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name()))
 	}
 	for x := 0; x < dataPoints.Len(); x++ {
-		addSingleDoubleDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+		addSingleNumberDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
 	}
 	return nil
 }
