diff --git a/processor/samplingprocessor/tailsamplingprocessor/factory_test.go b/processor/samplingprocessor/tailsamplingprocessor/factory_test.go
index 15e5a7043..c097bb8d5 100644
--- a/processor/samplingprocessor/tailsamplingprocessor/factory_test.go
+++ b/processor/samplingprocessor/tailsamplingprocessor/factory_test.go
@@ -15,25 +15,25 @@
 package tailsamplingprocessor
 
 import (
+	"context"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"go.uber.org/zap"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configcheck"
 	"go.opentelemetry.io/collector/exporter/exportertest"
 )
 
 func TestCreateDefaultConfig(t *testing.T) {
-	factory := &Factory{}
-
-	cfg := factory.CreateDefaultConfig()
+	cfg := createDefaultConfig()
 	assert.NotNil(t, cfg, "failed to create default config")
 	assert.NoError(t, configcheck.ValidateConfig(cfg))
 }
 
 func TestCreateProcessor(t *testing.T) {
-	factory := &Factory{}
+	factory := NewFactory()
 
 	cfg := factory.CreateDefaultConfig().(*Config)
 	// Manually set required fields
@@ -45,11 +45,8 @@ func TestCreateProcessor(t *testing.T) {
 		},
 	}
 
-	tp, err := factory.CreateTraceProcessor(zap.NewNop(), exportertest.NewNopTraceExporterOld(), cfg)
+	params := component.ProcessorCreateParams{Logger: zap.NewNop()}
+	tp, err := factory.CreateTraceProcessor(context.Background(), params, exportertest.NewNopTraceExporter(), cfg)
 	assert.NotNil(t, tp)
 	assert.NoError(t, err, "cannot create trace processor")
-
-	mp, err := factory.CreateMetricsProcessor(zap.NewNop(), nil, cfg)
-	assert.Nil(t, mp)
-	assert.Error(t, err, "should not be able to create metric processor")
 }
