diff --git a/service/internal/configunmarshaler/defaultunmarshaler_test.go b/service/internal/configunmarshaler/defaultunmarshaler_test.go
index be849d7bb..5c4983ca5 100644
--- a/service/internal/configunmarshaler/defaultunmarshaler_test.go
+++ b/service/internal/configunmarshaler/defaultunmarshaler_test.go
@@ -107,44 +107,43 @@ func TestDecodeConfig_Invalid(t *testing.T) {
 		{name: "invalid-receiver-type", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-processor-type", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-exporter-type", expected: errUnmarshalTopLevelStructure},
-		{name: "invalid-pipeline-type", expected: errUnmarshalService},
+		{name: "invalid-pipeline-type", expected: errUnmarshalTopLevelStructure},
 
 		{name: "invalid-extension-name-after-slash", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-receiver-name-after-slash", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-processor-name-after-slash", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-exporter-name-after-slash", expected: errUnmarshalTopLevelStructure},
-		{name: "invalid-pipeline-name-after-slash", expected: errUnmarshalService},
+		{name: "invalid-pipeline-name-after-slash", expected: errUnmarshalTopLevelStructure},
 
 		{name: "unknown-extension-type", expected: errUnmarshalExtension, expectedMessage: "extensions"},
 		{name: "unknown-receiver-type", expected: errUnmarshalReceiver, expectedMessage: "receivers"},
 		{name: "unknown-processor-type", expected: errUnmarshalProcessor, expectedMessage: "processors"},
 		{name: "unknown-exporter-type", expected: errUnmarshalExporter, expectedMessage: "exporters"},
-		{name: "unknown-pipeline-type", expected: errUnmarshalService, expectedMessage: "pipelines"},
 
 		{name: "duplicate-extension", expected: errUnmarshalTopLevelStructure, expectedMessage: "duplicate name"},
 		{name: "duplicate-receiver", expected: errUnmarshalTopLevelStructure, expectedMessage: "duplicate name"},
 		{name: "duplicate-processor", expected: errUnmarshalTopLevelStructure, expectedMessage: "duplicate name"},
 		{name: "duplicate-exporter", expected: errUnmarshalTopLevelStructure, expectedMessage: "duplicate name"},
-		{name: "duplicate-pipeline", expected: errUnmarshalService, expectedMessage: "duplicate name"},
+		{name: "duplicate-pipeline", expected: errUnmarshalTopLevelStructure, expectedMessage: "duplicate name"},
 
 		{name: "invalid-top-level-section", expected: errUnmarshalTopLevelStructure, expectedMessage: "top level"},
 		{name: "invalid-extension-section", expected: errUnmarshalExtension, expectedMessage: "extensions"},
 		{name: "invalid-receiver-section", expected: errUnmarshalReceiver, expectedMessage: "receivers"},
 		{name: "invalid-processor-section", expected: errUnmarshalProcessor, expectedMessage: "processors"},
 		{name: "invalid-exporter-section", expected: errUnmarshalExporter, expectedMessage: "exporters"},
-		{name: "invalid-service-section", expected: errUnmarshalService},
-		{name: "invalid-service-extensions-section", expected: errUnmarshalService},
-		{name: "invalid-pipeline-section", expected: errUnmarshalService, expectedMessage: "pipelines"},
-		{name: "invalid-sequence-value", expected: errUnmarshalService, expectedMessage: "pipelines"},
+		{name: "invalid-service-section", expected: errUnmarshalTopLevelStructure},
+		{name: "invalid-service-extensions-section", expected: errUnmarshalTopLevelStructure},
+		{name: "invalid-pipeline-section", expected: errUnmarshalTopLevelStructure, expectedMessage: "pipelines"},
+		{name: "invalid-sequence-value", expected: errUnmarshalTopLevelStructure, expectedMessage: "pipelines"},
 
 		{name: "invalid-extension-sub-config", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-receiver-sub-config", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-processor-sub-config", expected: errUnmarshalTopLevelStructure},
 		{name: "invalid-exporter-sub-config", expected: errUnmarshalTopLevelStructure},
-		{name: "invalid-pipeline-sub-config", expected: errUnmarshalService},
+		{name: "invalid-pipeline-sub-config", expected: errUnmarshalTopLevelStructure},
 
-		{name: "invalid-logs-level", expected: errUnmarshalService},
-		{name: "invalid-metrics-level", expected: errUnmarshalService},
+		{name: "invalid-logs-level", expected: errUnmarshalTopLevelStructure},
+		{name: "invalid-metrics-level", expected: errUnmarshalTopLevelStructure},
 	}
 
 	factories, err := componenttest.NopFactories()
