diff --git a/consumer/pdata/generated_trace.go b/consumer/pdata/generated_trace.go
index 48daa5691..41dc04e86 100644
--- a/consumer/pdata/generated_trace.go
+++ b/consumer/pdata/generated_trace.go
@@ -18,7 +18,7 @@
 package pdata
 
 import (
-	otlpcommon "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/common/v1"
+	"go.opentelemetry.io/collector/internal/data"
 	otlptrace "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/trace/v1"
 )
 
@@ -547,7 +547,7 @@ func (ms Span) TraceID() TraceID {
 //
 // Important: This causes a runtime error if IsNil() returns "true".
 func (ms Span) SetTraceID(v TraceID) {
-	(*ms.orig).TraceId = otlpcommon.TraceID(v)
+	(*ms.orig).TraceId = data.TraceID(v)
 }
 
 // SpanID returns the spanid associated with this Span.
@@ -561,7 +561,7 @@ func (ms Span) SpanID() SpanID {
 //
 // Important: This causes a runtime error if IsNil() returns "true".
 func (ms Span) SetSpanID(v SpanID) {
-	(*ms.orig).SpanId = otlpcommon.SpanID(v)
+	(*ms.orig).SpanId = data.SpanID(v)
 }
 
 // TraceState returns the tracestate associated with this Span.
@@ -589,7 +589,7 @@ func (ms Span) ParentSpanID() SpanID {
 //
 // Important: This causes a runtime error if IsNil() returns "true".
 func (ms Span) SetParentSpanID(v SpanID) {
-	(*ms.orig).ParentSpanId = otlpcommon.SpanID(v)
+	(*ms.orig).ParentSpanId = data.SpanID(v)
 }
 
 // Name returns the name associated with this Span.
@@ -1128,7 +1128,7 @@ func (ms SpanLink) TraceID() TraceID {
 //
 // Important: This causes a runtime error if IsNil() returns "true".
 func (ms SpanLink) SetTraceID(v TraceID) {
-	(*ms.orig).TraceId = otlpcommon.TraceID(v)
+	(*ms.orig).TraceId = data.TraceID(v)
 }
 
 // SpanID returns the spanid associated with this SpanLink.
@@ -1142,7 +1142,7 @@ func (ms SpanLink) SpanID() SpanID {
 //
 // Important: This causes a runtime error if IsNil() returns "true".
 func (ms SpanLink) SetSpanID(v SpanID) {
-	(*ms.orig).SpanId = otlpcommon.SpanID(v)
+	(*ms.orig).SpanId = data.SpanID(v)
 }
 
 // TraceState returns the tracestate associated with this SpanLink.
