diff --git a/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go b/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go
index 728272ab1..5f51a4bfe 100644
--- a/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go
+++ b/exporter/exporterhelper/internal/queuebatch/queue_batch_test.go
@@ -95,17 +95,15 @@ func TestQueueBatchDoNotPreserveCancellation(t *testing.T) {
 }
 
 func TestQueueBatchHappyPath(t *testing.T) {
-	cfg := Config{
-		Enabled:      true,
-		QueueSize:    10,
-		NumConsumers: 1,
-	}
+	cfg := newTestConfig()
+	cfg.BlockOnOverflow = false
+	cfg.QueueSize = 56
 	sink := requesttest.NewSink()
 	qb, err := NewQueueBatch(newFakeRequestSettings(), cfg, sink.Export)
 	require.NoError(t, err)
 
 	for i := 0; i < 10; i++ {
-		require.NoError(t, qb.Send(context.Background(), &requesttest.FakeRequest{Items: i}))
+		require.NoError(t, qb.Send(context.Background(), &requesttest.FakeRequest{Items: i + 1}))
 	}
 
 	// expect queue to be full
@@ -113,7 +111,9 @@ func TestQueueBatchHappyPath(t *testing.T) {
 
 	require.NoError(t, qb.Start(context.Background(), componenttest.NewNopHost()))
 	assert.Eventually(t, func() bool {
-		return sink.RequestsCount() == 10 && sink.ItemsCount() == 45
+		// Because batching is used, cannot guarantee that will be 1 batch or multiple because of the flush interval.
+		// Check only for total items count.
+		return sink.ItemsCount() == 55
 	}, 1*time.Second, 10*time.Millisecond)
 	require.NoError(t, qb.Shutdown(context.Background()))
 }
