diff --git a/confmap/provider/httpprovider/go.mod b/confmap/provider/httpprovider/go.mod
index 5292c6436..cae8da3f9 100644
--- a/confmap/provider/httpprovider/go.mod
+++ b/confmap/provider/httpprovider/go.mod
@@ -4,7 +4,7 @@ go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/confmap v1.16.0
+	go.opentelemetry.io/collector/confmap v1.17.0
 	go.uber.org/goleak v1.3.0
 )
 
