diff --git a/pdata/internal/generated_proto_tcpaddr.go b/pdata/internal/generated_proto_tcpaddr.go
index d2590bdbb..fb903efab 100644
--- a/pdata/internal/generated_proto_tcpaddr.go
+++ b/pdata/internal/generated_proto_tcpaddr.go
@@ -65,9 +65,7 @@ func CopyTCPAddr(dest, src *TCPAddr) *TCPAddr {
 		dest = NewTCPAddr()
 	}
 	dest.IP = src.IP
-
 	dest.Port = src.Port
-
 	dest.Zone = src.Zone
 
 	return dest
@@ -164,13 +162,15 @@ func (orig *TCPAddr) SizeProto() int {
 	var n int
 	var l int
 	_ = l
+
 	l = len(orig.IP)
 	if l > 0 {
 		n += 1 + proto.Sov(uint64(l)) + l
 	}
-	if orig.Port != 0 {
+	if orig.Port != int64(0) {
 		n += 1 + proto.Sov(uint64(orig.Port))
 	}
+
 	l = len(orig.Zone)
 	if l > 0 {
 		n += 1 + proto.Sov(uint64(l)) + l
@@ -190,7 +190,7 @@ func (orig *TCPAddr) MarshalProto(buf []byte) int {
 		pos--
 		buf[pos] = 0xa
 	}
-	if orig.Port != 0 {
+	if orig.Port != int64(0) {
 		pos = proto.EncodeVarint(buf, pos, uint64(orig.Port))
 		pos--
 		buf[pos] = 0x10
@@ -245,7 +245,6 @@ func (orig *TCPAddr) UnmarshalProto(buf []byte) error {
 			if err != nil {
 				return err
 			}
-
 			orig.Port = int64(num)
 
 		case 3:
