diff --git a/pdata/internal/generated_proto_resourcespans.go b/pdata/internal/generated_proto_resourcespans.go
index bd70dedb8..f008db670 100644
--- a/pdata/internal/generated_proto_resourcespans.go
+++ b/pdata/internal/generated_proto_resourcespans.go
@@ -79,7 +79,6 @@ func CopyResourceSpans(dest, src *ResourceSpans) *ResourceSpans {
 	dest.ScopeSpans = CopyScopeSpansPtrSlice(dest.ScopeSpans, src.ScopeSpans)
 
 	dest.SchemaUrl = src.SchemaUrl
-
 	dest.DeprecatedScopeSpans = CopyScopeSpansPtrSlice(dest.DeprecatedScopeSpans, src.DeprecatedScopeSpans)
 
 	return dest
@@ -206,6 +205,7 @@ func (orig *ResourceSpans) SizeProto() int {
 		l = orig.ScopeSpans[i].SizeProto()
 		n += 1 + proto.Sov(uint64(l)) + l
 	}
+
 	l = len(orig.SchemaUrl)
 	if l > 0 {
 		n += 1 + proto.Sov(uint64(l)) + l
