diff --git a/pdata/internal/generated_proto_ipaddr.go b/pdata/internal/generated_proto_ipaddr.go
index 2cd6029a6..37a6f62ed 100644
--- a/pdata/internal/generated_proto_ipaddr.go
+++ b/pdata/internal/generated_proto_ipaddr.go
@@ -64,7 +64,6 @@ func CopyIPAddr(dest, src *IPAddr) *IPAddr {
 		dest = NewIPAddr()
 	}
 	dest.IP = src.IP
-
 	dest.Zone = src.Zone
 
 	return dest
@@ -155,10 +154,12 @@ func (orig *IPAddr) SizeProto() int {
 	var n int
 	var l int
 	_ = l
+
 	l = len(orig.IP)
 	if l > 0 {
 		n += 1 + proto.Sov(uint64(l)) + l
 	}
+
 	l = len(orig.Zone)
 	if l > 0 {
 		n += 1 + proto.Sov(uint64(l)) + l
