diff --git a/pdata/internal/generated_proto_exporttracepartialsuccess.go b/pdata/internal/generated_proto_exporttracepartialsuccess.go
index df8025572..c971f7cd7 100644
--- a/pdata/internal/generated_proto_exporttracepartialsuccess.go
+++ b/pdata/internal/generated_proto_exporttracepartialsuccess.go
@@ -65,7 +65,6 @@ func CopyExportTracePartialSuccess(dest, src *ExportTracePartialSuccess) *Export
 		dest = NewExportTracePartialSuccess()
 	}
 	dest.RejectedSpans = src.RejectedSpans
-
 	dest.ErrorMessage = src.ErrorMessage
 
 	return dest
@@ -155,9 +154,10 @@ func (orig *ExportTracePartialSuccess) SizeProto() int {
 	var n int
 	var l int
 	_ = l
-	if orig.RejectedSpans != 0 {
+	if orig.RejectedSpans != int64(0) {
 		n += 1 + proto.Sov(uint64(orig.RejectedSpans))
 	}
+
 	l = len(orig.ErrorMessage)
 	if l > 0 {
 		n += 1 + proto.Sov(uint64(l)) + l
@@ -169,7 +169,7 @@ func (orig *ExportTracePartialSuccess) MarshalProto(buf []byte) int {
 	pos := len(buf)
 	var l int
 	_ = l
-	if orig.RejectedSpans != 0 {
+	if orig.RejectedSpans != int64(0) {
 		pos = proto.EncodeVarint(buf, pos, uint64(orig.RejectedSpans))
 		pos--
 		buf[pos] = 0x8
@@ -209,7 +209,6 @@ func (orig *ExportTracePartialSuccess) UnmarshalProto(buf []byte) error {
 			if err != nil {
 				return err
 			}
-
 			orig.RejectedSpans = int64(num)
 
 		case 2:
