diff --git a/pdata/plog/generated_logrecord.go b/pdata/plog/generated_logrecord.go
index 14583d349..f93ca7866 100644
--- a/pdata/plog/generated_logrecord.go
+++ b/pdata/plog/generated_logrecord.go
@@ -10,6 +10,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/internal"
 	"go.opentelemetry.io/collector/pdata/internal/data"
 	otlplogs "go.opentelemetry.io/collector/pdata/internal/data/protogen/logs/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
@@ -166,6 +167,54 @@ func (ms LogRecord) CopyTo(dest LogRecord) {
 	copyOrigLogRecord(dest.orig, ms.orig)
 }
 
+// marshalJSONStream marshals all properties from the current struct to the destination stream.
+func (ms LogRecord) marshalJSONStream(dest *json.Stream) {
+	dest.WriteObjectStart()
+	if ms.orig.ObservedTimeUnixNano != 0 {
+		dest.WriteObjectField("observedTimeUnixNano")
+		dest.WriteUint64(ms.orig.ObservedTimeUnixNano)
+	}
+	if ms.orig.TimeUnixNano != 0 {
+		dest.WriteObjectField("timeUnixNano")
+		dest.WriteUint64(ms.orig.TimeUnixNano)
+	}
+	if ms.orig.TraceId != data.TraceID([16]byte{}) {
+		dest.WriteObjectField("traceId")
+		ms.orig.TraceId.MarshalJSONStream(dest)
+	}
+	if ms.orig.SpanId != data.SpanID([8]byte{}) {
+		dest.WriteObjectField("spanId")
+		ms.orig.SpanId.MarshalJSONStream(dest)
+	}
+	if ms.orig.Flags != 0 {
+		dest.WriteObjectField("flags")
+		dest.WriteUint32(ms.orig.Flags)
+	}
+	if ms.orig.EventName != "" {
+		dest.WriteObjectField("eventName")
+		dest.WriteString(ms.orig.EventName)
+	}
+	if ms.orig.SeverityText != "" {
+		dest.WriteObjectField("severityText")
+		dest.WriteString(ms.orig.SeverityText)
+	}
+	if ms.orig.SeverityNumber != otlplogs.SeverityNumber(0) {
+		dest.WriteObjectField("severityNumber")
+		ms.SeverityNumber().marshalJSONStream(dest)
+	}
+	dest.WriteObjectField("body")
+	internal.MarshalJSONStreamValue(internal.NewValue(&ms.orig.Body, ms.state), dest)
+	if len(ms.orig.Attributes) > 0 {
+		dest.WriteObjectField("attributes")
+		internal.MarshalJSONStreamMap(internal.NewMap(&ms.orig.Attributes, ms.state), dest)
+	}
+	if ms.orig.DroppedAttributesCount != uint32(0) {
+		dest.WriteObjectField("droppedAttributesCount")
+		dest.WriteUint32(ms.orig.DroppedAttributesCount)
+	}
+	dest.WriteObjectEnd()
+}
+
 func copyOrigLogRecord(dest, src *otlplogs.LogRecord) {
 	dest.ObservedTimeUnixNano = src.ObservedTimeUnixNano
 	dest.TimeUnixNano = src.TimeUnixNano
