diff --git a/service/telemetry_test.go b/service/telemetry_test.go
index c95a42ab9..2377a0770 100644
--- a/service/telemetry_test.go
+++ b/service/telemetry_test.go
@@ -16,7 +16,6 @@ import (
 	"go.opencensus.io/stats/view"
 	"go.opentelemetry.io/contrib/config"
 	"go.opentelemetry.io/otel/metric"
-	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configtelemetry"
@@ -241,11 +240,10 @@ func TestTelemetryInit(t *testing.T) {
 			}
 			set := meterProviderSettings{
 				res:               resource.New(component.NewDefaultBuildInfo(), tc.cfg.Resource),
-				logger:            zap.NewNop(),
 				cfg:               tc.cfg.Metrics,
 				asyncErrorChannel: make(chan error),
 			}
-			mp, err := newMeterProvider(set, tc.disableHighCard, tc.extendedConfig)
+			mp, err := newMeterProvider(set, tc.disableHighCard)
 			require.NoError(t, err)
 			defer func() {
 				if prov, ok := mp.(interface{ Shutdown(context.Context) error }); ok {
