diff --git a/pdata/pprofile/links.go b/pdata/pprofile/links.go
index b548a2f60..13a9192a0 100644
--- a/pdata/pprofile/links.go
+++ b/pdata/pprofile/links.go
@@ -5,43 +5,27 @@ package pprofile // import "go.opentelemetry.io/collector/pdata/pprofile"
 
 import (
 	"errors"
-	"fmt"
 	"math"
 )
 
 var errTooManyLinkTableEntries = errors.New("too many entries in LinkTable")
 
-// SetLink updates a LinkTable and a Sample's LinkIndex to
-// add or update a link.
-func SetLink(table LinkSlice, record Sample, li Link) error {
-	idx := int(record.LinkIndex())
-	if idx > 0 {
-		if idx >= table.Len() {
-			return fmt.Errorf("index value %d out of range for LinkIndex", idx)
-		}
-		mapAt := table.At(idx)
-		if mapAt.Equal(li) {
-			// Link already exists, nothing to do.
-			return nil
-		}
-	}
-
+// SetLink updates a LinkTable, adding or providing a value and returns its
+// index.
+func SetLink(table LinkSlice, li Link) (int32, error) {
 	for j, l := range table.All() {
 		if l.Equal(li) {
 			if j > math.MaxInt32 {
-				return errTooManyLinkTableEntries
+				return 0, errTooManyLinkTableEntries
 			}
-			// Add the index of the existing link to the indices.
-			record.SetLinkIndex(int32(j)) //nolint:gosec // G115 overflow checked
-			return nil
+			return int32(j), nil //nolint:gosec // G115 overflow checked
 		}
 	}
 
 	if table.Len() >= math.MaxInt32 {
-		return errTooManyLinkTableEntries
+		return 0, errTooManyLinkTableEntries
 	}
 
 	li.CopyTo(table.AppendEmpty())
-	record.SetLinkIndex(int32(table.Len() - 1)) //nolint:gosec // G115 overflow checked
-	return nil
+	return int32(table.Len() - 1), nil //nolint:gosec // G115 overflow checked
 }
