diff --git a/exporter/opencensusexporter/opencensus.go b/exporter/opencensusexporter/opencensus.go
index b61e8a686..ac6387b36 100644
--- a/exporter/opencensusexporter/opencensus.go
+++ b/exporter/opencensusexporter/opencensus.go
@@ -26,7 +26,6 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 
-	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/translator/internaldata"
 )
@@ -58,7 +57,7 @@ type ocExporter struct {
 	metadata       metadata.MD
 }
 
-func newOcExporter(_ context.Context, cfg *Config) (*ocExporter, error) {
+func newOcExporter(ctx context.Context, cfg *Config) (*ocExporter, error) {
 	if cfg.Endpoint == "" {
 		return nil, errors.New("OpenCensus exporter cfg requires an Endpoint")
 	}
@@ -67,46 +66,22 @@ func newOcExporter(_ context.Context, cfg *Config) (*ocExporter, error) {
 		return nil, errors.New("OpenCensus exporter cfg requires at least one worker")
 	}
 
-	oce := &ocExporter{
-		cfg:      cfg,
-		metadata: metadata.New(cfg.GRPCClientSettings.Headers),
-	}
-	return oce, nil
-}
-
-// start creates the gRPC client Connection
-func (oce *ocExporter) start(ctx context.Context, host component.Host) error {
-	dialOpts, err := oce.cfg.GRPCClientSettings.ToDialOptions(host.GetExtensions())
+	dialOpts, err := cfg.GRPCClientSettings.ToDialOptions()
 	if err != nil {
-		return err
-	}
-	var clientConn *grpc.ClientConn
-	if clientConn, err = grpc.DialContext(ctx, oce.cfg.GRPCClientSettings.Endpoint, dialOpts...); err != nil {
-		return err
+		return nil, err
 	}
 
-	oce.grpcClientConn = clientConn
-
-	if oce.tracesClients != nil {
-		oce.traceSvcClient = agenttracepb.NewTraceServiceClient(oce.grpcClientConn)
-		// Try to create rpc clients now.
-		for i := 0; i < oce.cfg.NumWorkers; i++ {
-			// Populate the channel with NumWorkers nil RPCs to keep the number of workers
-			// constant in the channel.
-			oce.tracesClients <- nil
-		}
+	var clientConn *grpc.ClientConn
+	if clientConn, err = grpc.DialContext(ctx, cfg.GRPCClientSettings.Endpoint, dialOpts...); err != nil {
+		return nil, err
 	}
 
-	if oce.metricsClients != nil {
-		oce.metricsSvcClient = agentmetricspb.NewMetricsServiceClient(oce.grpcClientConn)
-		// Try to create rpc clients now.
-		for i := 0; i < oce.cfg.NumWorkers; i++ {
-			// Populate the channel with NumWorkers nil RPCs to keep the number of workers
-			// constant in the channel.
-			oce.metricsClients <- nil
-		}
+	oce := &ocExporter{
+		cfg:            cfg,
+		grpcClientConn: clientConn,
+		metadata:       metadata.New(cfg.GRPCClientSettings.Headers),
 	}
-	return nil
+	return oce, nil
 }
 
 func (oce *ocExporter) shutdown(context.Context) error {
@@ -134,7 +109,14 @@ func newTracesExporter(ctx context.Context, cfg *Config) (*ocExporter, error) {
 	if err != nil {
 		return nil, err
 	}
-	oce.tracesClients = make(chan *tracesClientWithCancel, oce.cfg.NumWorkers)
+	oce.traceSvcClient = agenttracepb.NewTraceServiceClient(oce.grpcClientConn)
+	oce.tracesClients = make(chan *tracesClientWithCancel, cfg.NumWorkers)
+	// Try to create rpc clients now.
+	for i := 0; i < cfg.NumWorkers; i++ {
+		// Populate the channel with NumWorkers nil RPCs to keep the number of workers
+		// constant in the channel.
+		oce.tracesClients <- nil
+	}
 	return oce, nil
 }
 
@@ -143,7 +125,14 @@ func newMetricsExporter(ctx context.Context, cfg *Config) (*ocExporter, error) {
 	if err != nil {
 		return nil, err
 	}
-	oce.metricsClients = make(chan *metricsClientWithCancel, oce.cfg.NumWorkers)
+	oce.metricsSvcClient = agentmetricspb.NewMetricsServiceClient(oce.grpcClientConn)
+	oce.metricsClients = make(chan *metricsClientWithCancel, cfg.NumWorkers)
+	// Try to create rpc clients now.
+	for i := 0; i < cfg.NumWorkers; i++ {
+		// Populate the channel with NumWorkers nil RPCs to keep the number of workers
+		// constant in the channel.
+		oce.metricsClients <- nil
+	}
 	return oce, nil
 }
 
