diff --git a/service/internal/graph/graph_test.go b/service/internal/graph/graph_test.go
index 6795f27d4..b4dcf4f90 100644
--- a/service/internal/graph/graph_test.go
+++ b/service/internal/graph/graph_test.go
@@ -1836,8 +1836,8 @@ func TestGraphBuildErrors(t *testing.T) {
 	}
 }
 
-// // This includes all tests from the previous implmentation, plus a new one
-// // relevant only to the new graph-based implementation.
+// This includes all tests from the previous implmentation, plus a new one
+// relevant only to the new graph-based implementation.
 func TestGraphFailToStartAndShutdown(t *testing.T) {
 	errReceiverFactory := newErrReceiverFactory()
 	errProcessorFactory := newErrProcessorFactory()
