diff --git a/pdata/ptrace/generated_scopespansslice_test.go b/pdata/ptrace/generated_scopespansslice_test.go
index 18bbdd88a..c34ec4c96 100644
--- a/pdata/ptrace/generated_scopespansslice_test.go
+++ b/pdata/ptrace/generated_scopespansslice_test.go
@@ -19,8 +19,7 @@ import (
 func TestScopeSpansSlice(t *testing.T) {
 	es := NewScopeSpansSlice()
 	assert.Equal(t, 0, es.Len())
-	state := internal.StateMutable
-	es = newScopeSpansSlice(&[]*otlptrace.ScopeSpans{}, &state)
+	es = newScopeSpansSlice(&[]*otlptrace.ScopeSpans{}, internal.NewState())
 	assert.Equal(t, 0, es.Len())
 
 	emptyVal := NewScopeSpans()
@@ -35,8 +34,9 @@ func TestScopeSpansSlice(t *testing.T) {
 }
 
 func TestScopeSpansSliceReadOnly(t *testing.T) {
-	sharedState := internal.StateReadOnly
-	es := newScopeSpansSlice(&[]*otlptrace.ScopeSpans{}, &sharedState)
+	sharedState := internal.NewState()
+	sharedState.MarkReadOnly()
+	es := newScopeSpansSlice(&[]*otlptrace.ScopeSpans{}, sharedState)
 	assert.Equal(t, 0, es.Len())
 	assert.Panics(t, func() { es.AppendEmpty() })
 	assert.Panics(t, func() { es.EnsureCapacity(2) })
