diff --git a/pdata/ptrace/generated_spanevent.go b/pdata/ptrace/generated_spanevent.go
index 9b423df71..b9fed0384 100644
--- a/pdata/ptrace/generated_spanevent.go
+++ b/pdata/ptrace/generated_spanevent.go
@@ -34,8 +34,7 @@ func newSpanEvent(orig *otlptrace.Span_Event, state *internal.State) SpanEvent {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewSpanEvent() SpanEvent {
-	state := internal.StateMutable
-	return newSpanEvent(&otlptrace.Span_Event{}, &state)
+	return newSpanEvent(&otlptrace.Span_Event{}, internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
