diff --git a/pdata/ptrace/generated_span.go b/pdata/ptrace/generated_span.go
index 2a471186d..a14a7e6ac 100644
--- a/pdata/ptrace/generated_span.go
+++ b/pdata/ptrace/generated_span.go
@@ -35,8 +35,7 @@ func newSpan(orig *otlptrace.Span, state *internal.State) Span {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewSpan() Span {
-	state := internal.StateMutable
-	return newSpan(&otlptrace.Span{}, &state)
+	return newSpan(&otlptrace.Span{}, internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
