diff --git a/service/go.mod b/service/go.mod
index 16db8d101..ae03378ec 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -21,7 +21,7 @@ require (
 	go.opentelemetry.io/collector/extension v0.104.0
 	go.opentelemetry.io/collector/extension/zpagesextension v0.104.0
 	go.opentelemetry.io/collector/featuregate v1.11.0
-	go.opentelemetry.io/collector/internal/globalgates v0.0.0-20240705161705-b127da089038
+	go.opentelemetry.io/collector/internal/globalgates v0.0.0-20240715203212-534768cf7a80
 	go.opentelemetry.io/collector/pdata v1.11.0
 	go.opentelemetry.io/collector/pdata/testdata v0.104.0
 	go.opentelemetry.io/collector/processor v0.104.0
