diff --git a/obsreport/obsreporttest/obsreporttest.go b/obsreport/obsreporttest/obsreporttest.go
index 1b4171dae..6ec5a1e97 100644
--- a/obsreport/obsreporttest/obsreporttest.go
+++ b/obsreport/obsreporttest/obsreporttest.go
@@ -46,46 +46,46 @@ var (
 	processorTag, _ = tag.NewKey("processor")
 )
 
-type TestTelemetrySettings struct {
+type TestTelemetry struct {
 	component.TelemetrySettings
 	SpanRecorder *tracetest.SpanRecorder
 	views        []*view.View
 }
 
 // ToExporterCreateSettings returns ExporterCreateSettings with configured TelemetrySettings
-func (tts *TestTelemetrySettings) ToExporterCreateSettings() component.ExporterCreateSettings {
+func (tts *TestTelemetry) ToExporterCreateSettings() component.ExporterCreateSettings {
 	exporterSettings := componenttest.NewNopExporterCreateSettings()
 	exporterSettings.TelemetrySettings = tts.TelemetrySettings
 	return exporterSettings
 }
 
 // ToProcessorCreateSettings returns ProcessorCreateSettings with configured TelemetrySettings
-func (tts *TestTelemetrySettings) ToProcessorCreateSettings() component.ProcessorCreateSettings {
+func (tts *TestTelemetry) ToProcessorCreateSettings() component.ProcessorCreateSettings {
 	processorSettings := componenttest.NewNopProcessorCreateSettings()
 	processorSettings.TelemetrySettings = tts.TelemetrySettings
 	return processorSettings
 }
 
 // ToReceiverCreateSettings returns ReceiverCreateSettings with configured TelemetrySettings
-func (tts *TestTelemetrySettings) ToReceiverCreateSettings() component.ReceiverCreateSettings {
+func (tts *TestTelemetry) ToReceiverCreateSettings() component.ReceiverCreateSettings {
 	receiverSettings := componenttest.NewNopReceiverCreateSettings()
 	receiverSettings.TelemetrySettings = tts.TelemetrySettings
 	return receiverSettings
 }
 
 // Shutdown unregisters any views and shuts down the SpanRecorder
-func (tts *TestTelemetrySettings) Shutdown(ctx context.Context) error {
+func (tts *TestTelemetry) Shutdown(ctx context.Context) error {
 	view.Unregister(tts.views...)
 	return tts.SpanRecorder.Shutdown(ctx)
 }
 
-// SetupRecordedMetricsTest does setup the testing environment to check the metrics recorded by receivers, producers or exporters.
-// The caller should defer a call to Shutdown the returned TestTelemetrySettings.
-func SetupRecordedMetricsTest() (TestTelemetrySettings, error) {
+// SetupTelemetry does setup the testing environment to check the metrics recorded by receivers, producers or exporters.
+// The caller should defer a call to Shutdown the returned TestTelemetry.
+func SetupTelemetry() (TestTelemetry, error) {
 	sr := new(tracetest.SpanRecorder)
 	tp := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(sr))
 
-	settings := TestTelemetrySettings{
+	settings := TestTelemetry{
 		TelemetrySettings: componenttest.NewNopTelemetrySettings(),
 		SpanRecorder:      sr,
 	}
@@ -101,7 +101,7 @@ func SetupRecordedMetricsTest() (TestTelemetrySettings, error) {
 }
 
 // CheckExporterTraces checks that for the current exported values for trace exporter metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckExporterTraces(exporter config.ComponentID, acceptedSpans, droppedSpans int64) error {
 	exporterTags := tagsForExporterView(exporter)
 	if err := checkValueForView(exporterTags, acceptedSpans, "exporter/sent_spans"); err != nil {
@@ -111,7 +111,7 @@ func CheckExporterTraces(exporter config.ComponentID, acceptedSpans, droppedSpan
 }
 
 // CheckExporterMetrics checks that for the current exported values for metrics exporter metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckExporterMetrics(exporter config.ComponentID, acceptedMetricsPoints, droppedMetricsPoints int64) error {
 	exporterTags := tagsForExporterView(exporter)
 	if err := checkValueForView(exporterTags, acceptedMetricsPoints, "exporter/sent_metric_points"); err != nil {
@@ -121,7 +121,7 @@ func CheckExporterMetrics(exporter config.ComponentID, acceptedMetricsPoints, dr
 }
 
 // CheckExporterLogs checks that for the current exported values for logs exporter metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckExporterLogs(exporter config.ComponentID, acceptedLogRecords, droppedLogRecords int64) error {
 	exporterTags := tagsForExporterView(exporter)
 	if err := checkValueForView(exporterTags, acceptedLogRecords, "exporter/sent_log_records"); err != nil {
@@ -131,7 +131,7 @@ func CheckExporterLogs(exporter config.ComponentID, acceptedLogRecords, droppedL
 }
 
 // CheckProcessorTraces checks that for the current exported values for trace exporter metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckProcessorTraces(processor config.ComponentID, acceptedSpans, refusedSpans, droppedSpans int64) error {
 	processorTags := tagsForProcessorView(processor)
 	if err := checkValueForView(processorTags, acceptedSpans, "processor/accepted_spans"); err != nil {
@@ -144,7 +144,7 @@ func CheckProcessorTraces(processor config.ComponentID, acceptedSpans, refusedSp
 }
 
 // CheckProcessorMetrics checks that for the current exported values for metrics exporter metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckProcessorMetrics(processor config.ComponentID, acceptedMetricPoints, refusedMetricPoints, droppedMetricPoints int64) error {
 	processorTags := tagsForProcessorView(processor)
 	if err := checkValueForView(processorTags, acceptedMetricPoints, "processor/accepted_metric_points"); err != nil {
@@ -157,7 +157,7 @@ func CheckProcessorMetrics(processor config.ComponentID, acceptedMetricPoints, r
 }
 
 // CheckProcessorLogs checks that for the current exported values for logs exporter metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckProcessorLogs(processor config.ComponentID, acceptedLogRecords, refusedLogRecords, droppedLogRecords int64) error {
 	processorTags := tagsForProcessorView(processor)
 	if err := checkValueForView(processorTags, acceptedLogRecords, "processor/accepted_log_records"); err != nil {
@@ -170,7 +170,7 @@ func CheckProcessorLogs(processor config.ComponentID, acceptedLogRecords, refuse
 }
 
 // CheckReceiverTraces checks that for the current exported values for trace receiver metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckReceiverTraces(receiver config.ComponentID, protocol string, acceptedSpans, droppedSpans int64) error {
 	receiverTags := tagsForReceiverView(receiver, protocol)
 	if err := checkValueForView(receiverTags, acceptedSpans, "receiver/accepted_spans"); err != nil {
@@ -180,7 +180,7 @@ func CheckReceiverTraces(receiver config.ComponentID, protocol string, acceptedS
 }
 
 // CheckReceiverLogs checks that for the current exported values for logs receiver metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckReceiverLogs(receiver config.ComponentID, protocol string, acceptedLogRecords, droppedLogRecords int64) error {
 	receiverTags := tagsForReceiverView(receiver, protocol)
 	if err := checkValueForView(receiverTags, acceptedLogRecords, "receiver/accepted_log_records"); err != nil {
@@ -190,7 +190,7 @@ func CheckReceiverLogs(receiver config.ComponentID, protocol string, acceptedLog
 }
 
 // CheckReceiverMetrics checks that for the current exported values for metrics receiver metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckReceiverMetrics(receiver config.ComponentID, protocol string, acceptedMetricPoints, droppedMetricPoints int64) error {
 	receiverTags := tagsForReceiverView(receiver, protocol)
 	if err := checkValueForView(receiverTags, acceptedMetricPoints, "receiver/accepted_metric_points"); err != nil {
@@ -200,7 +200,7 @@ func CheckReceiverMetrics(receiver config.ComponentID, protocol string, accepted
 }
 
 // CheckScraperMetrics checks that for the current exported values for metrics scraper metrics match given values.
-// When this function is called it is required to also call SetupRecordedMetricsTest as first thing.
+// When this function is called it is required to also call SetupTelemetry as first thing.
 func CheckScraperMetrics(receiver config.ComponentID, scraper config.ComponentID, scrapedMetricPoints, erroredMetricPoints int64) error {
 	scraperTags := tagsForScraperView(receiver, scraper)
 	if err := checkValueForView(scraperTags, scrapedMetricPoints, "scraper/scraped_metric_points"); err != nil {
