diff --git a/cmd/mdatagen/templates/status.go.tmpl b/cmd/mdatagen/templates/status.go.tmpl
index 850058b9c..d35cf76ff 100644
--- a/cmd/mdatagen/templates/status.go.tmpl
+++ b/cmd/mdatagen/templates/status.go.tmpl
@@ -10,6 +10,7 @@ import (
 
 var (
 	Type = component.MustNewType("{{ .Type }}")
+	scopeName = "{{ .ScopeName }}"
 )
 
 const (
@@ -21,9 +22,9 @@ const (
 )
 
 func Meter(settings component.TelemetrySettings) metric.Meter {
-	return settings.MeterProvider.Meter("{{ .ScopeName }}")
+	return settings.MeterProvider.Meter(scopeName)
 }
 
 func Tracer(settings component.TelemetrySettings) trace.Tracer {
-	return settings.TracerProvider.Tracer("{{ .ScopeName }}")
+	return settings.TracerProvider.Tracer(scopeName)
 }
