diff --git a/extension/ballastextension/go.mod b/extension/ballastextension/go.mod
index 0439b86c9..7238bf114 100644
--- a/extension/ballastextension/go.mod
+++ b/extension/ballastextension/go.mod
@@ -6,6 +6,7 @@ require (
 	github.com/stretchr/testify v1.8.1
 	go.opentelemetry.io/collector v0.66.0
 	go.opentelemetry.io/collector/component v0.66.0
+	go.opentelemetry.io/collector/confmap v0.66.0
 	go.uber.org/zap v1.23.0
 )
 
@@ -47,6 +48,8 @@ replace go.opentelemetry.io/collector => ../../
 
 replace go.opentelemetry.io/collector/component => ../../component
 
+replace go.opentelemetry.io/collector/confmap => ../../confmap
+
 replace go.opentelemetry.io/collector/featuregate => ../../featuregate
 
 replace go.opentelemetry.io/collector/pdata => ../../pdata
