diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index e3590fe84..7fb77bee1 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -30,6 +30,7 @@ extension/experimental/storage  @open-telemetry/collector-approvers @swiatekm
 # Profiling-related modules
 pdata/pprofile                                  @open-telemetry/collector-approvers @mx-psi @dmathieu
 consumer/consumerprofiles                       @open-telemetry/collector-approvers @mx-psi @dmathieu
+consumer/xconsumer                             @open-telemetry/collector-approvers @mx-psi @dmathieu
 connector/connectorprofiles                     @open-telemetry/collector-approvers @mx-psi @dmathieu
 exporter/exporterhelper/exporterhelperprofiles  @open-telemetry/collector-approvers @mx-psi @dmathieu
 exporter/exporterprofiles                       @open-telemetry/collector-approvers @mx-psi @dmathieu
