diff --git a/go.mod b/go.mod
index 7af9f1862..0509b8236 100644
--- a/go.mod
+++ b/go.mod
@@ -20,7 +20,6 @@ require (
 	github.com/spf13/cobra v1.6.1
 	github.com/stretchr/testify v1.8.1
 	go.opencensus.io v0.23.0
-	go.opentelemetry.io/collector/extension/ballastextension v0.0.0-20221102221454-bfc45c16c979
 	go.opentelemetry.io/collector/extension/zpagesextension v0.0.0-20221102221454-bfc45c16c979
 	go.opentelemetry.io/collector/pdata v0.63.1
 	go.opentelemetry.io/collector/processor/batchprocessor v0.0.0-00010101000000-000000000000
@@ -84,8 +83,6 @@ replace go.opentelemetry.io/collector/semconv => ./semconv
 
 replace go.opentelemetry.io/collector/pdata => ./pdata
 
-replace go.opentelemetry.io/collector/extension/ballastextension => ./extension/ballastextension
-
 replace go.opentelemetry.io/collector/extension/zpagesextension => ./extension/zpagesextension
 
 replace go.opentelemetry.io/collector/processor/batchprocessor => ./processor/batchprocessor
