diff --git a/processor/nodebatcherprocessor/node_batcher.go b/processor/nodebatcherprocessor/node_batcher.go
index 68abc76d1..e05ae0e69 100644
--- a/processor/nodebatcherprocessor/node_batcher.go
+++ b/processor/nodebatcherprocessor/node_batcher.go
@@ -76,7 +76,7 @@ type batcher struct {
 var _ consumer.TraceConsumer = (*batcher)(nil)
 
 // NewBatcher creates a new batcher that batches spans by node and resource
-func NewBatcher(name string, logger *zap.Logger, sender consumer.TraceConsumer, opts ...Option) consumer.TraceConsumer {
+func NewBatcher(name string, logger *zap.Logger, sender consumer.TraceConsumer, opts ...Option) processor.TraceProcessor {
 	// Init with defaults
 	b := &batcher{
 		name:   name,
@@ -109,6 +109,10 @@ func (b *batcher) ConsumeTraceData(ctx context.Context, td consumerdata.TraceDat
 	return nil
 }
 
+func (b *batcher) GetCapabilities() processor.Capabilities {
+	return processor.Capabilities{MutatesConsumedData: false}
+}
+
 func (b *batcher) genBucketID(node *commonpb.Node, resource *resourcepb.Resource, spanFormat string) string {
 	h := sha256.New()
 	if node != nil {
