diff --git a/config/config.go b/config/config.go
index 8acf0bd17..27332e523 100644
--- a/config/config.go
+++ b/config/config.go
@@ -191,7 +191,7 @@ func loadReceivers(v *viper.Viper, factories map[string]receiver.Factory) (confi
 	enabledReceiver := false
 
 	// Prepare resulting map
-	receivers := make(configmodels.Receivers, 0)
+	receivers := make(configmodels.Receivers)
 
 	// Iterate over input map and create a config for each.
 	for key := range keyMap {
@@ -271,7 +271,7 @@ func loadExporters(v *viper.Viper, factories map[string]exporter.Factory) (confi
 	keyMap := v.GetStringMap(exportersKeyName)
 
 	// Prepare resulting map
-	exporters := make(configmodels.Exporters, 0)
+	exporters := make(configmodels.Exporters)
 
 	// Iterate over exporters and create a config for each.
 	for key := range keyMap {
@@ -328,7 +328,7 @@ func loadProcessors(v *viper.Viper, factories map[string]processor.Factory) (con
 	keyMap := v.GetStringMap(processorsKeyName)
 
 	// Prepare resulting map.
-	processors := make(configmodels.Processors, 0)
+	processors := make(configmodels.Processors)
 
 	// Iterate over processors and create a config for each.
 	for key := range keyMap {
@@ -385,7 +385,7 @@ func loadPipelines(v *viper.Viper) (configmodels.Pipelines, error) {
 	keyMap := v.GetStringMap(pipelinesKeyName)
 
 	// Prepare resulting map.
-	pipelines := make(configmodels.Pipelines, 0)
+	pipelines := make(configmodels.Pipelines)
 
 	// Iterate over input map and create a config for each.
 	for key := range keyMap {
