diff --git a/exporter/exporterhelper/internal/bounded_memory_queue_test.go b/exporter/exporterhelper/internal/bounded_memory_queue_test.go
index a91b8bdc8..e5bbedf0f 100644
--- a/exporter/exporterhelper/internal/bounded_memory_queue_test.go
+++ b/exporter/exporterhelper/internal/bounded_memory_queue_test.go
@@ -19,12 +19,12 @@ package internal
 import (
 	"reflect"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.uber.org/atomic"
 )
 
 type stringRequest struct {
@@ -159,7 +159,7 @@ func newConsumerState(t *testing.T) *consumerState {
 	return &consumerState{
 		t:            t,
 		consumed:     make(map[string]bool),
-		consumedOnce: atomic.NewBool(false),
+		consumedOnce: &atomic.Bool{},
 	}
 }
 
