diff --git a/processor/processortest/nop_processor.go b/processor/processortest/nop_processor.go
index c227789f6..e72fa5797 100644
--- a/processor/processortest/nop_processor.go
+++ b/processor/processortest/nop_processor.go
@@ -17,6 +17,9 @@ package processortest
 import (
 	"context"
 
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
 	"github.com/open-telemetry/opentelemetry-collector/consumer"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
 	"github.com/open-telemetry/opentelemetry-collector/processor"
@@ -51,3 +54,44 @@ func NewNopTraceProcessor(nextTraceProcessor consumer.TraceConsumer) consumer.Tr
 func NewNopMetricsProcessor(nextMetricsProcessor consumer.MetricsConsumer) consumer.MetricsConsumer {
 	return &nopProcessor{nextMetricsProcessor: nextMetricsProcessor}
 }
+
+// NopProcessorFactory allows the creation of the no operation processor via
+// config, so it can be used in tests that cannot create it directly.
+type NopProcessorFactory struct{}
+
+var _ processor.Factory = (*NopProcessorFactory)(nil)
+
+// Type gets the type of the Processor created by this factory.
+func (npf *NopProcessorFactory) Type() string {
+	return "nop"
+}
+
+// CreateDefaultConfig creates the default configuration for the Processor.
+func (npf *NopProcessorFactory) CreateDefaultConfig() configmodels.Processor {
+	return &configmodels.ProcessorSettings{
+		TypeVal: npf.Type(),
+		NameVal: npf.Type(),
+	}
+}
+
+// CreateTraceProcessor creates a trace processor based on this config.
+// If the processor type does not support tracing or if the config is not valid
+// error will be returned instead.
+func (npf *NopProcessorFactory) CreateTraceProcessor(
+	logger *zap.Logger,
+	nextConsumer consumer.TraceConsumer,
+	cfg configmodels.Processor,
+) (processor.TraceProcessor, error) {
+	return &nopProcessor{nextTraceProcessor: nextConsumer}, nil
+}
+
+// CreateMetricsProcessor creates a metrics processor based on this config.
+// If the processor type does not support metrics or if the config is not valid
+// error will be returned instead.
+func (npf *NopProcessorFactory) CreateMetricsProcessor(
+	logger *zap.Logger,
+	nextConsumer consumer.MetricsConsumer,
+	cfg configmodels.Processor,
+) (processor.MetricsProcessor, error) {
+	return &nopProcessor{nextMetricsProcessor: nextConsumer}, nil
+}
