diff --git a/exporter/loggingexporter/go.mod b/exporter/loggingexporter/go.mod
index 5c646ebad..5b2647933 100644
--- a/exporter/loggingexporter/go.mod
+++ b/exporter/loggingexporter/go.mod
@@ -5,10 +5,10 @@ go 1.21.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/component v0.105.0
-	go.opentelemetry.io/collector/config/configtelemetry v0.105.0
-	go.opentelemetry.io/collector/confmap v0.105.0
-	go.opentelemetry.io/collector/exporter v0.105.0
+	go.opentelemetry.io/collector/component v0.106.0
+	go.opentelemetry.io/collector/config/configtelemetry v0.106.0
+	go.opentelemetry.io/collector/confmap v0.106.0
+	go.opentelemetry.io/collector/exporter v0.106.0
 	go.opentelemetry.io/collector/pdata v1.12.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
@@ -39,16 +39,16 @@ require (
 	github.com/prometheus/client_model v0.6.1 // indirect
 	github.com/prometheus/common v0.55.0 // indirect
 	github.com/prometheus/procfs v0.15.1 // indirect
-	go.opentelemetry.io/collector v0.105.0 // indirect
+	go.opentelemetry.io/collector v0.106.0 // indirect
 	go.opentelemetry.io/collector/config/configretry v1.12.0 // indirect
-	go.opentelemetry.io/collector/consumer v0.105.0 // indirect
-	go.opentelemetry.io/collector/consumer/consumerprofiles v0.105.0 // indirect
-	go.opentelemetry.io/collector/consumer/consumertest v0.0.0-20240726175034-c3a11297650a // indirect
-	go.opentelemetry.io/collector/extension v0.105.0 // indirect
+	go.opentelemetry.io/collector/consumer v0.106.0 // indirect
+	go.opentelemetry.io/collector/consumer/consumerprofiles v0.106.0 // indirect
+	go.opentelemetry.io/collector/consumer/consumertest v0.106.0 // indirect
+	go.opentelemetry.io/collector/extension v0.106.0 // indirect
 	go.opentelemetry.io/collector/featuregate v1.12.0 // indirect
-	go.opentelemetry.io/collector/internal/globalgates v0.105.0 // indirect
-	go.opentelemetry.io/collector/pdata/pprofile v0.105.0 // indirect
-	go.opentelemetry.io/collector/receiver v0.105.0 // indirect
+	go.opentelemetry.io/collector/internal/globalgates v0.106.0 // indirect
+	go.opentelemetry.io/collector/pdata/pprofile v0.106.0 // indirect
+	go.opentelemetry.io/collector/receiver v0.106.0 // indirect
 	go.opentelemetry.io/otel v1.28.0 // indirect
 	go.opentelemetry.io/otel/exporters/prometheus v0.50.0 // indirect
 	go.opentelemetry.io/otel/metric v1.28.0 // indirect
