diff --git a/cmd/mdatagen/internal/command.go b/cmd/mdatagen/internal/command.go
index 49278ad4e..8758c8a6d 100644
--- a/cmd/mdatagen/internal/command.go
+++ b/cmd/mdatagen/internal/command.go
@@ -361,6 +361,16 @@ func templatize(tmplFile string, md Metadata) *template.Template {
 			}).ParseFS(TemplateFS, strings.ReplaceAll(tmplFile, "\\", "/")))
 }
 
+func executeTemplate(tmplFile string, md Metadata, goPackage string) ([]byte, error) {
+	tmpl := templatize(tmplFile, md)
+	buf := bytes.Buffer{}
+
+	if err := tmpl.Execute(&buf, TemplateContext{Metadata: md, Package: goPackage}); err != nil {
+		return []byte{}, fmt.Errorf("failed executing template: %w", err)
+	}
+	return buf.Bytes(), nil
+}
+
 func inlineReplace(tmplFile string, outputFile string, md Metadata, start string, end string) error {
 	var readmeContents []byte
 	var err error
@@ -373,20 +383,16 @@ func inlineReplace(tmplFile string, outputFile string, md Metadata, start string
 		return nil
 	}
 
-	tmpl := templatize(tmplFile, md)
-	buf := bytes.Buffer{}
-
 	if md.GithubProject == "" {
 		md.GithubProject = "open-telemetry/opentelemetry-collector-contrib"
 	}
 
-	if err := tmpl.Execute(&buf, TemplateContext{Metadata: md, Package: "metadata"}); err != nil {
-		return fmt.Errorf("failed executing template: %w", err)
+	buf, err := executeTemplate(tmplFile, md, "metadata")
+	if err != nil {
+		return err
 	}
 
-	result := buf.String()
-
-	s := re.ReplaceAllString(string(readmeContents), result)
+	s := re.ReplaceAllString(string(readmeContents), string(buf))
 	if err := os.WriteFile(outputFile, []byte(s), 0600); err != nil {
 		return fmt.Errorf("failed writing %q: %w", outputFile, err)
 	}
@@ -395,21 +401,17 @@ func inlineReplace(tmplFile string, outputFile string, md Metadata, start string
 }
 
 func generateFile(tmplFile string, outputFile string, md Metadata, goPackage string) error {
-	tmpl := templatize(tmplFile, md)
-	buf := bytes.Buffer{}
-
-	if err := tmpl.Execute(&buf, TemplateContext{Metadata: md, Package: goPackage}); err != nil {
-		return fmt.Errorf("failed executing template: %w", err)
-	}
-
 	if err := os.Remove(outputFile); err != nil && !errors.Is(err, os.ErrNotExist) {
 		return fmt.Errorf("unable to remove generated file %q: %w", outputFile, err)
 	}
 
-	result := buf.Bytes()
+	result, err := executeTemplate(tmplFile, md, goPackage)
+	if err != nil {
+		return err
+	}
 	var formatErr error
 	if strings.HasSuffix(outputFile, ".go") {
-		if formatted, err := format.Source(buf.Bytes()); err == nil {
+		if formatted, err := format.Source(result); err == nil {
 			result = formatted
 		} else {
 			formatErr = fmt.Errorf("failed formatting %s:%w", outputFile, err)
