diff --git a/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go b/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go
index 6adb56b3c..8bac6aec5 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go
@@ -107,11 +107,7 @@ func (s *scraper) ScrapeMetrics(_ context.Context) (pdata.MetricSlice, error) {
 		appendSystemSpecificMetrics(metrics, 1, usages)
 	}
 
-	if len(errors) > 0 {
-		return metrics, componenterror.CombineErrors(errors)
-	}
-
-	return metrics, nil
+	return metrics, componenterror.CombineErrors(errors)
 }
 
 func initializeFileSystemUsageMetric(metric pdata.Metric, deviceUsages []*deviceUsage) {
