diff --git a/receiver/otlpreceiver/config.go b/receiver/otlpreceiver/config.go
index 622952d77..c89f1d559 100644
--- a/receiver/otlpreceiver/config.go
+++ b/receiver/otlpreceiver/config.go
@@ -4,6 +4,7 @@
 package otlpreceiver // import "go.opentelemetry.io/collector/receiver/otlpreceiver"
 
 import (
+	"encoding"
 	"errors"
 	"fmt"
 	"net/url"
@@ -21,17 +22,35 @@ const (
 	protoHTTP = "protocols::http"
 )
 
+type SanitizedURLPath string
+
+var _ encoding.TextUnmarshaler = (*SanitizedURLPath)(nil)
+
+func (s *SanitizedURLPath) UnmarshalText(text []byte) error {
+	u, err := url.Parse(string(text))
+	if err != nil {
+		return fmt.Errorf("invalid HTTP URL path set for signal: %w", err)
+	}
+
+	if !path.IsAbs(u.Path) {
+		u.Path = "/" + u.Path
+	}
+
+	*s = SanitizedURLPath(u.Path)
+	return nil
+}
+
 type HTTPConfig struct {
 	ServerConfig confighttp.ServerConfig `mapstructure:",squash"`
 
 	// The URL path to receive traces on. If omitted "/v1/traces" will be used.
-	TracesURLPath string `mapstructure:"traces_url_path,omitempty"`
+	TracesURLPath SanitizedURLPath `mapstructure:"traces_url_path,omitempty"`
 
 	// The URL path to receive metrics on. If omitted "/v1/metrics" will be used.
-	MetricsURLPath string `mapstructure:"metrics_url_path,omitempty"`
+	MetricsURLPath SanitizedURLPath `mapstructure:"metrics_url_path,omitempty"`
 
 	// The URL path to receive logs on. If omitted "/v1/logs" will be used.
-	LogsURLPath string `mapstructure:"logs_url_path,omitempty"`
+	LogsURLPath SanitizedURLPath `mapstructure:"logs_url_path,omitempty"`
 
 	// prevent unkeyed literal initialization
 	_ struct{}
@@ -78,32 +97,7 @@ func (cfg *Config) Unmarshal(conf *confmap.Conf) error {
 
 	if !conf.IsSet(protoHTTP) {
 		cfg.HTTP = nil
-	} else {
-		var err error
-
-		if cfg.HTTP.TracesURLPath, err = sanitizeURLPath(cfg.HTTP.TracesURLPath); err != nil {
-			return err
-		}
-		if cfg.HTTP.MetricsURLPath, err = sanitizeURLPath(cfg.HTTP.MetricsURLPath); err != nil {
-			return err
-		}
-		if cfg.HTTP.LogsURLPath, err = sanitizeURLPath(cfg.HTTP.LogsURLPath); err != nil {
-			return err
-		}
 	}
 
 	return nil
 }
-
-// Verify signal URL path sanity
-func sanitizeURLPath(urlPath string) (string, error) {
-	u, err := url.Parse(urlPath)
-	if err != nil {
-		return "", fmt.Errorf("invalid HTTP URL path set for signal: %w", err)
-	}
-
-	if !path.IsAbs(u.Path) {
-		u.Path = "/" + u.Path
-	}
-	return u.Path, nil
-}
