diff --git a/consumer/pdata/generated_trace.go b/consumer/pdata/generated_trace.go
index 1bfb543eb..8841ad90f 100644
--- a/consumer/pdata/generated_trace.go
+++ b/consumer/pdata/generated_trace.go
@@ -140,8 +140,8 @@ func (es ResourceSpansSlice) Resize(newLen int) {
 // given ResourceSpans at that new position.  The original ResourceSpans
 // could still be referenced so do not reuse it after passing it to this
 // method.
-func (es ResourceSpansSlice) Append(e *ResourceSpans) {
-	(*es.orig) = append((*es.orig), *e.orig)
+func (es ResourceSpansSlice) Append(e ResourceSpans) {
+	*es.orig = append(*es.orig, *e.orig)
 }
 
 // InstrumentationLibrarySpans is a collection of spans from a LibraryInstrumentation.
@@ -331,8 +331,8 @@ func (es InstrumentationLibrarySpansSlice) Resize(newLen int) {
 // given InstrumentationLibrarySpans at that new position.  The original InstrumentationLibrarySpans
 // could still be referenced so do not reuse it after passing it to this
 // method.
-func (es InstrumentationLibrarySpansSlice) Append(e *InstrumentationLibrarySpans) {
-	(*es.orig) = append((*es.orig), *e.orig)
+func (es InstrumentationLibrarySpansSlice) Append(e InstrumentationLibrarySpans) {
+	*es.orig = append(*es.orig, *e.orig)
 }
 
 // InstrumentationLibrarySpans is a collection of spans from a LibraryInstrumentation.
@@ -522,8 +522,8 @@ func (es SpanSlice) Resize(newLen int) {
 // given Span at that new position.  The original Span
 // could still be referenced so do not reuse it after passing it to this
 // method.
-func (es SpanSlice) Append(e *Span) {
-	(*es.orig) = append((*es.orig), *e.orig)
+func (es SpanSlice) Append(e Span) {
+	*es.orig = append(*es.orig, *e.orig)
 }
 
 // Span represents a single operation within a trace.
@@ -895,8 +895,8 @@ func (es SpanEventSlice) Resize(newLen int) {
 // given SpanEvent at that new position.  The original SpanEvent
 // could still be referenced so do not reuse it after passing it to this
 // method.
-func (es SpanEventSlice) Append(e *SpanEvent) {
-	(*es.orig) = append((*es.orig), *e.orig)
+func (es SpanEventSlice) Append(e SpanEvent) {
+	*es.orig = append(*es.orig, *e.orig)
 }
 
 // SpanEvent is a time-stamped annotation of the span, consisting of user-supplied
@@ -1121,8 +1121,8 @@ func (es SpanLinkSlice) Resize(newLen int) {
 // given SpanLink at that new position.  The original SpanLink
 // could still be referenced so do not reuse it after passing it to this
 // method.
-func (es SpanLinkSlice) Append(e *SpanLink) {
-	(*es.orig) = append((*es.orig), *e.orig)
+func (es SpanLinkSlice) Append(e SpanLink) {
+	*es.orig = append(*es.orig, *e.orig)
 }
 
 // SpanLink is a pointer from the current span to another span in the same trace or in a
