diff --git a/pdata/internal/generated_proto_logrecord.go b/pdata/internal/generated_proto_logrecord.go
index 6d5057b57..c3e449246 100644
--- a/pdata/internal/generated_proto_logrecord.go
+++ b/pdata/internal/generated_proto_logrecord.go
@@ -12,6 +12,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -40,7 +41,7 @@ var (
 )
 
 func NewLogRecord() *LogRecord {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &LogRecord{}
 	}
 	return protoPoolLogRecord.Get().(*LogRecord)
@@ -51,7 +52,7 @@ func DeleteLogRecord(orig *LogRecord, nullable bool) {
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
