diff --git a/pdata/internal/generated_proto_profilesdata.go b/pdata/internal/generated_proto_profilesdata.go
index 45a876f6d..491a81955 100644
--- a/pdata/internal/generated_proto_profilesdata.go
+++ b/pdata/internal/generated_proto_profilesdata.go
@@ -11,6 +11,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -31,7 +32,7 @@ var (
 )
 
 func NewProfilesData() *ProfilesData {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &ProfilesData{}
 	}
 	return protoPoolProfilesData.Get().(*ProfilesData)
@@ -42,7 +43,7 @@ func DeleteProfilesData(orig *ProfilesData, nullable bool) {
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
