diff --git a/pdata/internal/generated_proto_logsrequest.go b/pdata/internal/generated_proto_logsrequest.go
index 94b915df0..4e18c3050 100644
--- a/pdata/internal/generated_proto_logsrequest.go
+++ b/pdata/internal/generated_proto_logsrequest.go
@@ -12,6 +12,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -30,7 +31,7 @@ var (
 )
 
 func NewLogsRequest() *LogsRequest {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &LogsRequest{}
 	}
 	return protoPoolLogsRequest.Get().(*LogsRequest)
@@ -41,7 +42,7 @@ func DeleteLogsRequest(orig *LogsRequest, nullable bool) {
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
