diff --git a/pdata/internal/generated_proto_summarydatapoint.go b/pdata/internal/generated_proto_summarydatapoint.go
index ccd99fa18..5402264ee 100644
--- a/pdata/internal/generated_proto_summarydatapoint.go
+++ b/pdata/internal/generated_proto_summarydatapoint.go
@@ -13,6 +13,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -36,7 +37,7 @@ var (
 )
 
 func NewSummaryDataPoint() *SummaryDataPoint {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &SummaryDataPoint{}
 	}
 	return protoPoolSummaryDataPoint.Get().(*SummaryDataPoint)
@@ -47,7 +48,7 @@ func DeleteSummaryDataPoint(orig *SummaryDataPoint, nullable bool) {
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
