diff --git a/pdata/internal/generated_proto_exportlogsserviceresponse.go b/pdata/internal/generated_proto_exportlogsserviceresponse.go
index 5570fe93d..b2271520f 100644
--- a/pdata/internal/generated_proto_exportlogsserviceresponse.go
+++ b/pdata/internal/generated_proto_exportlogsserviceresponse.go
@@ -11,6 +11,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -28,7 +29,7 @@ var (
 )
 
 func NewExportLogsServiceResponse() *ExportLogsServiceResponse {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &ExportLogsServiceResponse{}
 	}
 	return protoPoolExportLogsServiceResponse.Get().(*ExportLogsServiceResponse)
@@ -39,7 +40,7 @@ func DeleteExportLogsServiceResponse(orig *ExportLogsServiceResponse, nullable b
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
