diff --git a/pdata/internal/generated_proto_exportmetricsserviceresponse.go b/pdata/internal/generated_proto_exportmetricsserviceresponse.go
index ae78a10f9..222d5825f 100644
--- a/pdata/internal/generated_proto_exportmetricsserviceresponse.go
+++ b/pdata/internal/generated_proto_exportmetricsserviceresponse.go
@@ -11,6 +11,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -28,7 +29,7 @@ var (
 )
 
 func NewExportMetricsServiceResponse() *ExportMetricsServiceResponse {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &ExportMetricsServiceResponse{}
 	}
 	return protoPoolExportMetricsServiceResponse.Get().(*ExportMetricsServiceResponse)
@@ -39,7 +40,7 @@ func DeleteExportMetricsServiceResponse(orig *ExportMetricsServiceResponse, null
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
