diff --git a/pdata/internal/generated_proto_tracesrequest.go b/pdata/internal/generated_proto_tracesrequest.go
index b8bb39687..a54dedaef 100644
--- a/pdata/internal/generated_proto_tracesrequest.go
+++ b/pdata/internal/generated_proto_tracesrequest.go
@@ -12,6 +12,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -30,7 +31,7 @@ var (
 )
 
 func NewTracesRequest() *TracesRequest {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &TracesRequest{}
 	}
 	return protoPoolTracesRequest.Get().(*TracesRequest)
@@ -41,7 +42,7 @@ func DeleteTracesRequest(orig *TracesRequest, nullable bool) {
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
