diff --git a/pdata/internal/generated_proto_link.go b/pdata/internal/generated_proto_link.go
index 6df8aa77f..9d62117e8 100644
--- a/pdata/internal/generated_proto_link.go
+++ b/pdata/internal/generated_proto_link.go
@@ -11,6 +11,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -29,7 +30,7 @@ var (
 )
 
 func NewLink() *Link {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &Link{}
 	}
 	return protoPoolLink.Get().(*Link)
@@ -40,7 +41,7 @@ func DeleteLink(orig *Link, nullable bool) {
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
