diff --git a/pdata/internal/generated_proto_exportlogsservicerequest.go b/pdata/internal/generated_proto_exportlogsservicerequest.go
index 7fb48b36a..cdfb591d7 100644
--- a/pdata/internal/generated_proto_exportlogsservicerequest.go
+++ b/pdata/internal/generated_proto_exportlogsservicerequest.go
@@ -11,6 +11,7 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/pdata/internal/json"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
@@ -29,7 +30,7 @@ var (
 )
 
 func NewExportLogsServiceRequest() *ExportLogsServiceRequest {
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		return &ExportLogsServiceRequest{}
 	}
 	return protoPoolExportLogsServiceRequest.Get().(*ExportLogsServiceRequest)
@@ -40,7 +41,7 @@ func DeleteExportLogsServiceRequest(orig *ExportLogsServiceRequest, nullable boo
 		return
 	}
 
-	if !UseProtoPooling.IsEnabled() {
+	if !metadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 		orig.Reset()
 		return
 	}
