diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index d55518ef2..7bab5d072 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -5,7 +5,7 @@ go 1.22.0
 require (
 	github.com/golang/snappy v0.0.4
 	github.com/klauspost/compress v1.17.11
-	github.com/pierrec/lz4/v4 v4.1.21
+	github.com/pierrec/lz4/v4 v4.1.22
 	github.com/rs/cors v1.11.1
 	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/client v1.22.0
