diff --git a/pdata/pprofile/generated_valuetypeslice.go b/pdata/pprofile/generated_valuetypeslice.go
index 4ea8e814e..99c910e66 100644
--- a/pdata/pprofile/generated_valuetypeslice.go
+++ b/pdata/pprofile/generated_valuetypeslice.go
@@ -34,8 +34,7 @@ func newValueTypeSlice(orig *[]*otlpprofiles.ValueType, state *internal.State) V
 // Can use "EnsureCapacity" to initialize with a given capacity.
 func NewValueTypeSlice() ValueTypeSlice {
 	orig := []*otlpprofiles.ValueType(nil)
-	state := internal.StateMutable
-	return newValueTypeSlice(&orig, &state)
+	return newValueTypeSlice(&orig, internal.NewState())
 }
 
 // Len returns the number of elements in the slice.
