diff --git a/pdata/pprofile/generated_profilesslice.go b/pdata/pprofile/generated_profilesslice.go
index 75d014112..0d6ff9eaa 100644
--- a/pdata/pprofile/generated_profilesslice.go
+++ b/pdata/pprofile/generated_profilesslice.go
@@ -34,8 +34,7 @@ func newProfilesSlice(orig *[]*otlpprofiles.Profile, state *internal.State) Prof
 // Can use "EnsureCapacity" to initialize with a given capacity.
 func NewProfilesSlice() ProfilesSlice {
 	orig := []*otlpprofiles.Profile(nil)
-	state := internal.StateMutable
-	return newProfilesSlice(&orig, &state)
+	return newProfilesSlice(&orig, internal.NewState())
 }
 
 // Len returns the number of elements in the slice.
