diff --git a/pdata/pprofile/generated_attributeunit.go b/pdata/pprofile/generated_attributeunit.go
index 641dd2ae2..ebc02d5c5 100644
--- a/pdata/pprofile/generated_attributeunit.go
+++ b/pdata/pprofile/generated_attributeunit.go
@@ -32,8 +32,7 @@ func newAttributeUnit(orig *otlpprofiles.AttributeUnit, state *internal.State) A
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewAttributeUnit() AttributeUnit {
-	state := internal.StateMutable
-	return newAttributeUnit(&otlpprofiles.AttributeUnit{}, &state)
+	return newAttributeUnit(&otlpprofiles.AttributeUnit{}, internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
