diff --git a/pdata/pprofile/generated_resourceprofiles.go b/pdata/pprofile/generated_resourceprofiles.go
index f6b4fb259..7b58f4043 100644
--- a/pdata/pprofile/generated_resourceprofiles.go
+++ b/pdata/pprofile/generated_resourceprofiles.go
@@ -33,8 +33,7 @@ func newResourceProfiles(orig *otlpprofiles.ResourceProfiles, state *internal.St
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewResourceProfiles() ResourceProfiles {
-	state := internal.StateMutable
-	return newResourceProfiles(&otlpprofiles.ResourceProfiles{}, &state)
+	return newResourceProfiles(&otlpprofiles.ResourceProfiles{}, internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
