diff --git a/obsreport/obsreporttest/obsreporttest.go b/obsreport/obsreporttest/obsreporttest.go
index 8146756c1..ef04537e6 100644
--- a/obsreport/obsreporttest/obsreporttest.go
+++ b/obsreport/obsreporttest/obsreporttest.go
@@ -30,6 +30,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/internal/obsreportconfig"
 )
 
@@ -59,7 +60,7 @@ type TestTelemetry struct {
 }
 
 // ToExporterCreateSettings returns ExporterCreateSettings with configured TelemetrySettings
-func (tts *TestTelemetry) ToExporterCreateSettings() component.ExporterCreateSettings {
+func (tts *TestTelemetry) ToExporterCreateSettings() exporter.CreateSettings {
 	set := componenttest.NewNopExporterCreateSettings()
 	set.TelemetrySettings = tts.TelemetrySettings
 	set.ID = tts.id
