diff --git a/exporter/exporterhelper/metrics_test.go b/exporter/exporterhelper/metrics_test.go
index 4f42edcf6..dc559d9f9 100644
--- a/exporter/exporterhelper/metrics_test.go
+++ b/exporter/exporterhelper/metrics_test.go
@@ -337,7 +337,7 @@ func newPushMetricsDataModifiedDownstream(retError error) consumer.ConsumeMetric
 func checkRecordedMetricsForMetrics(t *testing.T, tt *componenttest.Telemetry, id component.ID, me exporter.Metrics, wantError error) {
 	md := testdata.GenerateMetrics(2)
 	const numBatches = 7
-	for i := 0; i < numBatches; i++ {
+	for range numBatches {
 		require.Equal(t, wantError, me.ConsumeMetrics(context.Background(), md))
 	}
 
@@ -368,7 +368,7 @@ func generateMetricsTraffic(t *testing.T, tracer trace.Tracer, me exporter.Metri
 	md := testdata.GenerateMetrics(1)
 	ctx, span := tracer.Start(context.Background(), fakeMetricsParentSpanName)
 	defer span.End()
-	for i := 0; i < numRequests; i++ {
+	for range numRequests {
 		require.Equal(t, wantError, me.ConsumeMetrics(ctx, md))
 	}
 }
