diff --git a/pdata/pcommon/value.go b/pdata/pcommon/value.go
index 73a95bcf2..5c7972ced 100644
--- a/pdata/pcommon/value.go
+++ b/pdata/pcommon/value.go
@@ -452,6 +452,33 @@ func (v Value) AsRaw() any {
 	return fmt.Sprintf("<Unknown OpenTelemetry value type %q>", v.Type())
 }
 
+func (v Value) Equal(c Value) bool {
+	if v.Type() != c.Type() {
+		return false
+	}
+
+	switch v.Type() {
+	case ValueTypeEmpty:
+		return true
+	case ValueTypeStr:
+		return v.Str() == c.Str()
+	case ValueTypeBool:
+		return v.Bool() == c.Bool()
+	case ValueTypeDouble:
+		return v.Double() == c.Double()
+	case ValueTypeInt:
+		return v.Int() == c.Int()
+	case ValueTypeBytes:
+		return v.Bytes().Equal(c.Bytes())
+	case ValueTypeMap:
+		return v.Map().Equal(c.Map())
+	case ValueTypeSlice:
+		return v.Slice().Equal(c.Slice())
+	}
+
+	return false
+}
+
 func newKeyValueString(k string, v string) otlpcommon.KeyValue {
 	orig := otlpcommon.KeyValue{Key: k}
 	state := internal.StateMutable
