diff --git a/consumer/consumerdata/consumerdata.go b/consumer/consumerdata/consumerdata.go
index 0e7dc649b..6859dc1e9 100644
--- a/consumer/consumerdata/consumerdata.go
+++ b/consumer/consumerdata/consumerdata.go
@@ -20,6 +20,7 @@ import (
 	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
 )
 
 // MetricsData is a struct that groups proto metrics with a unique node and a resource.
@@ -36,3 +37,18 @@ type TraceData struct {
 	Spans        []*tracepb.Span
 	SourceFormat string
 }
+
+// OTLPTraceData is a struct that groups proto spans with a resource. This is the
+// newer version of TraceData, using OTLP-based representation.
+type OTLPTraceData struct {
+	ResourceSpanList []*otlptrace.ResourceSpans
+}
+
+// SpanCount calculates the total number of spans.
+func (td OTLPTraceData) SpanCount() int {
+	spanCount := 0
+	for _, rsl := range td.ResourceSpanList {
+		spanCount += len(rsl.Spans)
+	}
+	return spanCount
+}
