diff --git a/internal/data/trace_test.go b/internal/data/trace_test.go
index f09d1608f..e950ad58b 100644
--- a/internal/data/trace_test.go
+++ b/internal/data/trace_test.go
@@ -98,21 +98,25 @@ func TestNewSpanLinkSlice(t *testing.T) {
 }
 
 func TestAttrs(t *testing.T) {
-	attrs := AttributesMap{"attr1": NewAttributeValueString("abc")}
+	attrs := NewAttributeMap(AttributesMap{"attr1": NewAttributeValueString("abc")})
 
 	span := NewSpan()
 	assert.EqualValues(t, 0, span.DroppedAttributesCount())
-	span.SetAttributes(NewAttributes(attrs, 123))
-	assert.EqualValues(t, 123, span.DroppedAttributesCount())
+	span.SetAttributes(attrs)
 	assert.EqualValues(t, attrs, span.Attributes())
+	span.SetDroppedAttributesCount(123)
+	assert.EqualValues(t, 123, span.DroppedAttributesCount())
 
-	event := NewSpanEvent(0, "", NewAttributes(attrs, 234))
-	assert.EqualValues(t, 234, event.DroppedAttributesCount())
+	event := NewSpanEvent()
+	event.SetAttributes(attrs)
 	assert.EqualValues(t, attrs, event.Attributes())
+	event.SetDroppedAttributesCount(234)
+	assert.EqualValues(t, 234, event.DroppedAttributesCount())
 
 	link := NewSpanLink()
 	assert.EqualValues(t, 0, link.DroppedAttributesCount())
-	link.SetAttributes(NewAttributes(attrs, 456))
-	assert.EqualValues(t, 456, link.DroppedAttributesCount())
+	link.SetAttributes(attrs)
 	assert.EqualValues(t, attrs, link.Attributes())
+	link.SetDroppedAttributesCount(456)
+	assert.EqualValues(t, 456, link.DroppedAttributesCount())
 }
