diff --git a/exporter/debugexporter/go.mod b/exporter/debugexporter/go.mod
index 1b9e82f85..747eb68b2 100644
--- a/exporter/debugexporter/go.mod
+++ b/exporter/debugexporter/go.mod
@@ -65,8 +65,6 @@ replace go.opentelemetry.io/collector/receiver => ../../receiver
 
 replace go.opentelemetry.io/collector/extension => ../../extension
 
-replace go.opentelemetry.io/collector/connector => ../../connector
-
 replace go.opentelemetry.io/collector/processor => ../../processor
 
 replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/configtelemetry
