diff --git a/pdata/internal/generated_wrapper_uint64slice.go b/pdata/internal/generated_wrapper_uint64slice.go
index 7aa29d1db..c04027dcc 100644
--- a/pdata/internal/generated_wrapper_uint64slice.go
+++ b/pdata/internal/generated_wrapper_uint64slice.go
@@ -6,6 +6,10 @@
 
 package internal
 
+import (
+	"go.opentelemetry.io/collector/pdata/internal/json"
+)
+
 type UInt64Slice struct {
 	orig  *[]uint64
 	state *State
@@ -39,3 +43,16 @@ func GenerateTestUInt64Slice() UInt64Slice {
 	FillTestUInt64Slice(ms)
 	return ms
 }
+
+// MarshalJSONStream marshals all properties from the current struct to the destination stream.
+func MarshalJSONStreamUInt64Slice(ms UInt64Slice, dest *json.Stream) {
+	dest.WriteArrayStart()
+	if len(*ms.orig) > 0 {
+		dest.WriteUint64((*ms.orig)[0])
+	}
+	for i := 1; i < len((*ms.orig)); i++ {
+		dest.WriteMore()
+		dest.WriteUint64((*ms.orig)[i])
+	}
+	dest.WriteArrayEnd()
+}
