diff --git a/pdata/internal/generated_wrapper_entityref.go b/pdata/internal/generated_wrapper_entityref.go
index e330c2157..5ccf61837 100644
--- a/pdata/internal/generated_wrapper_entityref.go
+++ b/pdata/internal/generated_wrapper_entityref.go
@@ -8,6 +8,7 @@ package internal
 
 import (
 	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 type EntityRef struct {
@@ -48,3 +49,25 @@ func FillTestEntityRef(tv EntityRef) {
 	FillTestStringSlice(NewStringSlice(&tv.orig.IdKeys, tv.state))
 	FillTestStringSlice(NewStringSlice(&tv.orig.DescriptionKeys, tv.state))
 }
+
+// MarshalJSONStream marshals all properties from the current struct to the destination stream.
+func MarshalJSONStreamEntityRef(ms EntityRef, dest *json.Stream) {
+	dest.WriteObjectStart()
+	if ms.orig.SchemaUrl != "" {
+		dest.WriteObjectField("schemaUrl")
+		dest.WriteString(ms.orig.SchemaUrl)
+	}
+	if ms.orig.Type != "" {
+		dest.WriteObjectField("type")
+		dest.WriteString(ms.orig.Type)
+	}
+	if len(ms.orig.IdKeys) > 0 {
+		dest.WriteObjectField("idKeys")
+		MarshalJSONStreamStringSlice(NewStringSlice(&ms.orig.IdKeys, ms.state), dest)
+	}
+	if len(ms.orig.DescriptionKeys) > 0 {
+		dest.WriteObjectField("descriptionKeys")
+		MarshalJSONStreamStringSlice(NewStringSlice(&ms.orig.DescriptionKeys, ms.state), dest)
+	}
+	dest.WriteObjectEnd()
+}
