diff --git a/component/component.go b/component/component.go
index 5548e1db3..210becf4b 100644
--- a/component/component.go
+++ b/component/component.go
@@ -18,7 +18,6 @@ import (
 	"context"
 
 	"go.opentelemetry.io/collector/config"
-	"go.opentelemetry.io/collector/internal/internalinterface"
 )
 
 // Component is either a receiver, exporter, processor, or an extension.
@@ -99,7 +98,18 @@ const (
 // This interface cannot be directly implemented. Implementations must
 // use the factory helpers for the appropriate component type.
 type Factory interface {
-	internalinterface.InternalInterface
 	// Type gets the type of the component created by this factory.
 	Type() config.Type
+
+	unexportedFactoryFunc()
+}
+
+type baseFactory struct {
+	cfgType config.Type
+}
+
+func (baseFactory) unexportedFactoryFunc() {}
+
+func (bf baseFactory) Type() config.Type {
+	return bf.cfgType
 }
