diff --git a/processor/processorhelper/xprocessorhelper/profiles_test.go b/processor/processorhelper/xprocessorhelper/profiles_test.go
index c4cc53b8a..d30a018d4 100644
--- a/processor/processorhelper/xprocessorhelper/profiles_test.go
+++ b/processor/processorhelper/xprocessorhelper/profiles_test.go
@@ -89,13 +89,11 @@ func TestProfilesConcurrency(t *testing.T) {
 
 	var wg sync.WaitGroup
 	for range 10 {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
+		wg.Go(func() {
 			for range 10000 {
 				assert.NoError(t, pp.ConsumeProfiles(context.Background(), incomingProfiles))
 			}
-		}()
+		})
 	}
 	wg.Wait()
 	assert.NoError(t, pp.Shutdown(context.Background()))
