diff --git a/service/telemetry/otelconftelemetry/sdk.go b/service/telemetry/otelconftelemetry/sdk.go
index 7b4456ab3..fc332a75f 100644
--- a/service/telemetry/otelconftelemetry/sdk.go
+++ b/service/telemetry/otelconftelemetry/sdk.go
@@ -8,7 +8,7 @@ import (
 
 	config "go.opentelemetry.io/contrib/otelconf/v0.3.0"
 	sdkresource "go.opentelemetry.io/otel/sdk/resource"
-	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.37.0"
 )
 
 func newSDK(ctx context.Context, res *sdkresource.Resource, conf config.OpenTelemetryConfiguration) (config.SDK, error) {
