diff --git a/exporter/opencensusexporter/config_test.go b/exporter/opencensusexporter/config_test.go
index 62d688d71..b1046c649 100644
--- a/exporter/opencensusexporter/config_test.go
+++ b/exporter/opencensusexporter/config_test.go
@@ -29,7 +29,7 @@ import (
 var _ = configv2.RegisterTestFactories()
 
 func TestLoadConfig(t *testing.T) {
-	factory := exporter.GetExporterFactory(typeStr)
+	factory := exporter.GetFactory(typeStr)
 
 	config, err := configv2.LoadConfigFile(t, path.Join(".", "testdata", "config.yaml"))
 
