diff --git a/pdata/internal/generated_wrapper_exporttraceservicerequest_test.go b/pdata/internal/generated_wrapper_exporttraceservicerequest_test.go
index d6e08c6f5..ae1d12815 100644
--- a/pdata/internal/generated_wrapper_exporttraceservicerequest_test.go
+++ b/pdata/internal/generated_wrapper_exporttraceservicerequest_test.go
@@ -19,10 +19,10 @@ import (
 )
 
 func TestCopyOrigExportTraceServiceRequest(t *testing.T) {
-	src := &otlpcollectortrace.ExportTraceServiceRequest{}
-	dest := &otlpcollectortrace.ExportTraceServiceRequest{}
+	src := NewOrigPtrExportTraceServiceRequest()
+	dest := NewOrigPtrExportTraceServiceRequest()
 	CopyOrigExportTraceServiceRequest(dest, src)
-	assert.Equal(t, &otlpcollectortrace.ExportTraceServiceRequest{}, dest)
+	assert.Equal(t, NewOrigPtrExportTraceServiceRequest(), dest)
 	FillOrigTestExportTraceServiceRequest(src)
 	CopyOrigExportTraceServiceRequest(dest, src)
 	assert.Equal(t, src, dest)
@@ -31,10 +31,10 @@ func TestCopyOrigExportTraceServiceRequest(t *testing.T) {
 func TestMarshalAndUnmarshalJSONOrigExportTraceServiceRequestUnknown(t *testing.T) {
 	iter := json.BorrowIterator([]byte(`{"unknown": "string"}`))
 	defer json.ReturnIterator(iter)
-	dest := &otlpcollectortrace.ExportTraceServiceRequest{}
+	dest := NewOrigPtrExportTraceServiceRequest()
 	UnmarshalJSONOrigExportTraceServiceRequest(dest, iter)
 	require.NoError(t, iter.Error())
-	assert.Equal(t, &otlpcollectortrace.ExportTraceServiceRequest{}, dest)
+	assert.Equal(t, NewOrigPtrExportTraceServiceRequest(), dest)
 }
 
 func TestMarshalAndUnmarshalJSONOrigExportTraceServiceRequest(t *testing.T) {
@@ -47,7 +47,7 @@ func TestMarshalAndUnmarshalJSONOrigExportTraceServiceRequest(t *testing.T) {
 
 			iter := json.BorrowIterator(stream.Buffer())
 			defer json.ReturnIterator(iter)
-			dest := &otlpcollectortrace.ExportTraceServiceRequest{}
+			dest := NewOrigPtrExportTraceServiceRequest()
 			UnmarshalJSONOrigExportTraceServiceRequest(dest, iter)
 			require.NoError(t, iter.Error())
 
@@ -57,10 +57,10 @@ func TestMarshalAndUnmarshalJSONOrigExportTraceServiceRequest(t *testing.T) {
 }
 
 func TestMarshalAndUnmarshalProtoOrigExportTraceServiceRequestUnknown(t *testing.T) {
-	dest := &otlpcollectortrace.ExportTraceServiceRequest{}
+	dest := NewOrigPtrExportTraceServiceRequest()
 	// message Test { required int64 field = 1313; } encoding { "field": "1234" }
 	require.NoError(t, UnmarshalProtoOrigExportTraceServiceRequest(dest, []byte{0x88, 0x52, 0xD2, 0x09}))
-	assert.Equal(t, &otlpcollectortrace.ExportTraceServiceRequest{}, dest)
+	assert.Equal(t, NewOrigPtrExportTraceServiceRequest(), dest)
 }
 
 func TestMarshalAndUnmarshalProtoOrigExportTraceServiceRequest(t *testing.T) {
@@ -70,7 +70,7 @@ func TestMarshalAndUnmarshalProtoOrigExportTraceServiceRequest(t *testing.T) {
 			gotSize := MarshalProtoOrigExportTraceServiceRequest(src, buf)
 			assert.Equal(t, len(buf), gotSize)
 
-			dest := &otlpcollectortrace.ExportTraceServiceRequest{}
+			dest := NewOrigPtrExportTraceServiceRequest()
 			require.NoError(t, UnmarshalProtoOrigExportTraceServiceRequest(dest, buf))
 			assert.Equal(t, src, dest)
 		})
@@ -90,7 +90,7 @@ func TestMarshalAndUnmarshalProtoViaProtobufExportTraceServiceRequest(t *testing
 			goBuf, err := proto.Marshal(goDest)
 			require.NoError(t, err)
 
-			dest := &otlpcollectortrace.ExportTraceServiceRequest{}
+			dest := NewOrigPtrExportTraceServiceRequest()
 			require.NoError(t, UnmarshalProtoOrigExportTraceServiceRequest(dest, goBuf))
 			assert.Equal(t, src, dest)
 		})
@@ -99,9 +99,9 @@ func TestMarshalAndUnmarshalProtoViaProtobufExportTraceServiceRequest(t *testing
 
 func getEncodingTestValuesExportTraceServiceRequest() map[string]*otlpcollectortrace.ExportTraceServiceRequest {
 	return map[string]*otlpcollectortrace.ExportTraceServiceRequest{
-		"empty": {},
+		"empty": NewOrigPtrExportTraceServiceRequest(),
 		"fill_test": func() *otlpcollectortrace.ExportTraceServiceRequest {
-			src := &otlpcollectortrace.ExportTraceServiceRequest{}
+			src := NewOrigPtrExportTraceServiceRequest()
 			FillOrigTestExportTraceServiceRequest(src)
 			return src
 		}(),
