diff --git a/config/processor.go b/config/processor.go
index 1713b3347..f9c20fd86 100644
--- a/config/processor.go
+++ b/config/processor.go
@@ -27,7 +27,7 @@ type Processors map[ComponentID]Processor
 
 // ProcessorSettings defines common settings for a processor configuration.
 // Specific processors can embed this struct and extend it with more fields if needed.
-// When embedded in the processor config it must be with `mapstructure:"-"` tag.
+// When embedded in the processor config it must be with `mapstructure:",squash"` tag.
 type ProcessorSettings struct {
 	id ComponentID `mapstructure:"-"`
 }
