diff --git a/exporter/exporterhelper/internal/persistent_storage_batch_test.go b/exporter/exporterhelper/internal/persistent_storage_batch_test.go
index 9a7aef221..496a767b6 100644
--- a/exporter/exporterhelper/internal/persistent_storage_batch_test.go
+++ b/exporter/exporterhelper/internal/persistent_storage_batch_test.go
@@ -12,9 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build enable_unstable
-// +build enable_unstable
-
 package internal
 
 import (
