diff --git a/internal/goldendataset/metric_gen.go b/internal/goldendataset/metric_gen.go
index f642ce69b..26e6ef2ce 100644
--- a/internal/goldendataset/metric_gen.go
+++ b/internal/goldendataset/metric_gen.go
@@ -160,7 +160,7 @@ func populateIntPoints(cfg MetricCfg, pts pdata.IntDataPointSlice) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
+		pt.SetStartTimestamp(pdata.Timestamp(cfg.StartTime))
 		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
 		pt.SetValue(int64(cfg.PtVal + i))
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -171,7 +171,7 @@ func populateDoublePoints(cfg MetricCfg, pts pdata.DoubleDataPointSlice) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
+		pt.SetStartTimestamp(pdata.Timestamp(cfg.StartTime))
 		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
 		pt.SetValue(float64(cfg.PtVal + i))
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -183,7 +183,7 @@ func populateDoubleHistogram(cfg MetricCfg, dh pdata.Histogram) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
+		pt.SetStartTimestamp(pdata.Timestamp(cfg.StartTime))
 		ts := getTimestamp(cfg.StartTime, cfg.StepSize, i)
 		pt.SetTimestamp(ts)
 		populatePtLabels(cfg, pt.LabelsMap())
@@ -220,7 +220,7 @@ func populateIntHistogram(cfg MetricCfg, dh pdata.IntHistogram) {
 	pts.Resize(cfg.NumPtsPerMetric)
 	for i := 0; i < cfg.NumPtsPerMetric; i++ {
 		pt := pts.At(i)
-		pt.SetStartTime(pdata.Timestamp(cfg.StartTime))
+		pt.SetStartTimestamp(pdata.Timestamp(cfg.StartTime))
 		ts := getTimestamp(cfg.StartTime, cfg.StepSize, i)
 		pt.SetTimestamp(ts)
 		populatePtLabels(cfg, pt.LabelsMap())
