diff --git a/internal/testdata/trace.go b/internal/testdata/trace.go
index 589570d88..e7b3eca34 100644
--- a/internal/testdata/trace.go
+++ b/internal/testdata/trace.go
@@ -54,7 +54,7 @@ func fillSpanOne(span ptrace.Span) {
 	ev0 := evs.AppendEmpty()
 	ev0.SetTimestamp(spanEventTimestamp)
 	ev0.SetName("event-with-attr")
-	initSpanEventAttributes(ev0.Attributes())
+	ev0.Attributes().PutStr("span-event-attr", "span-event-attr-val")
 	ev0.SetDroppedAttributesCount(2)
 	ev1 := evs.AppendEmpty()
 	ev1.SetTimestamp(spanEventTimestamp)
@@ -71,7 +71,7 @@ func fillSpanTwo(span ptrace.Span) {
 	span.SetStartTimestamp(spanStartTimestamp)
 	span.SetEndTimestamp(spanEndTimestamp)
 	link0 := span.Links().AppendEmpty()
-	initSpanLinkAttributes(link0.Attributes())
+	link0.Attributes().PutStr("span-link-attr", "span-link-attr-val")
 	link0.SetDroppedAttributesCount(4)
 	link1 := span.Links().AppendEmpty()
 	link1.SetDroppedAttributesCount(4)
