diff --git a/exporter/exporterhelper/internal/bounded_memory_queue_test.go b/exporter/exporterhelper/internal/bounded_memory_queue_test.go
index 4c8d08714..ef2cc4aa8 100644
--- a/exporter/exporterhelper/internal/bounded_memory_queue_test.go
+++ b/exporter/exporterhelper/internal/bounded_memory_queue_test.go
@@ -233,17 +233,6 @@ func (s *consumerState) assertConsumed(expected map[string]bool) {
 	assert.Equal(s.t, expected, s.snapshot())
 }
 
-func TestZeroSizeWithConsumers(t *testing.T) {
-	q := NewBoundedMemoryQueue[string](0)
-
-	consumers := NewQueueConsumers(q, 1, func(context.Context, string) bool { return true })
-	assert.NoError(t, consumers.Start(context.Background(), componenttest.NewNopHost()))
-
-	assert.NoError(t, q.Offer(context.Background(), "a")) // in process
-
-	assert.NoError(t, consumers.Shutdown(context.Background()))
-}
-
 func TestZeroSizeNoConsumers(t *testing.T) {
 	q := NewBoundedMemoryQueue[string](0)
 
