diff --git a/service/internal/testcomponents/example_router_test.go b/service/internal/testcomponents/example_router_test.go
index 2f44e0490..f868b9954 100644
--- a/service/internal/testcomponents/example_router_test.go
+++ b/service/internal/testcomponents/example_router_test.go
@@ -48,7 +48,7 @@ func TestTracesRouter(t *testing.T) {
 
 	cfg := ExampleRouterConfig{Traces: &LeftRightConfig{Left: leftID, Right: rightID}}
 	tr, err := ExampleRouterFactory.CreateTracesToTraces(
-		context.Background(), connectortest.NewNopCreateSettings(), cfg, router)
+		context.Background(), connectortest.NewNopSettings(), cfg, router)
 	assert.NoError(t, err)
 	assert.False(t, tr.Capabilities().MutatesData)
 
@@ -87,7 +87,7 @@ func TestMetricsRouter(t *testing.T) {
 
 	cfg := ExampleRouterConfig{Metrics: &LeftRightConfig{Left: leftID, Right: rightID}}
 	mr, err := ExampleRouterFactory.CreateMetricsToMetrics(
-		context.Background(), connectortest.NewNopCreateSettings(), cfg, router)
+		context.Background(), connectortest.NewNopSettings(), cfg, router)
 	assert.NoError(t, err)
 	assert.False(t, mr.Capabilities().MutatesData)
 
@@ -126,7 +126,7 @@ func TestLogsRouter(t *testing.T) {
 
 	cfg := ExampleRouterConfig{Logs: &LeftRightConfig{Left: leftID, Right: rightID}}
 	lr, err := ExampleRouterFactory.CreateLogsToLogs(
-		context.Background(), connectortest.NewNopCreateSettings(), cfg, router)
+		context.Background(), connectortest.NewNopSettings(), cfg, router)
 	assert.NoError(t, err)
 	assert.False(t, lr.Capabilities().MutatesData)
 
