diff --git a/exporter/exporterhelper/internal/queuebatch/partition_batcher_test.go b/exporter/exporterhelper/internal/queuebatch/partition_batcher_test.go
index 404228f04..fe2951c83 100644
--- a/exporter/exporterhelper/internal/queuebatch/partition_batcher_test.go
+++ b/exporter/exporterhelper/internal/queuebatch/partition_batcher_test.go
@@ -47,13 +47,13 @@ func TestPartitionBatcher_NoSplit_MinThresholdZero_TimeoutDisabled(t *testing.T)
 		{
 			name:       "bytes/one_worker",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 1,
 		},
 		{
 			name:       "bytes/three_workers",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 3,
 		},
 	}
@@ -113,13 +113,13 @@ func TestPartitionBatcher_NoSplit_TimeoutDisabled(t *testing.T) {
 		{
 			name:       "bytes/one_worker",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 1,
 		},
 		{
 			name:       "bytes/three_workers",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 3,
 		},
 	}
@@ -194,13 +194,13 @@ func TestPartitionBatcher_NoSplit_WithTimeout(t *testing.T) {
 		{
 			name:       "bytes/one_worker",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 1,
 		},
 		{
 			name:       "bytes/three_workers",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 3,
 		},
 	}
@@ -265,13 +265,13 @@ func TestPartitionBatcher_Split_TimeoutDisabled(t *testing.T) {
 		{
 			name:       "bytes/one_worker",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 1,
 		},
 		{
 			name:       "bytes/three_workers",
 			sizerType:  request.SizerTypeBytes,
-			sizer:      request.NewItemsSizer(),
+			sizer:      request.NewBytesSizer(),
 			maxWorkers: 3,
 		},
 	}
