diff --git a/pdata/plog/generated_resourcelogs.go b/pdata/plog/generated_resourcelogs.go
index 6d3a0c7fd..bf2c91632 100644
--- a/pdata/plog/generated_resourcelogs.go
+++ b/pdata/plog/generated_resourcelogs.go
@@ -74,7 +74,11 @@ func (ms ResourceLogs) ScopeLogs() ScopeLogsSlice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms ResourceLogs) CopyTo(dest ResourceLogs) {
 	dest.state.AssertMutable()
-	ms.Resource().CopyTo(dest.Resource())
-	dest.SetSchemaUrl(ms.SchemaUrl())
-	ms.ScopeLogs().CopyTo(dest.ScopeLogs())
+	copyOrigResourceLogs(dest.orig, ms.orig)
+}
+
+func copyOrigResourceLogs(dest, src *otlplogs.ResourceLogs) {
+	internal.CopyOrigResource(&dest.Resource, &src.Resource)
+	dest.SchemaUrl = src.SchemaUrl
+	dest.ScopeLogs = copyOrigScopeLogsSlice(dest.ScopeLogs, src.ScopeLogs)
 }
