diff --git a/component/receiver.go b/component/receiver.go
index 44b57070b..54d77f2ac 100644
--- a/component/receiver.go
+++ b/component/receiver.go
@@ -30,7 +30,7 @@ type Receiver interface {
 
 // A TracesReceiver is an "arbitrary data"-to-"internal format" converter.
 // Its purpose is to translate data from the wild into internal trace format.
-// TracesReceiver feeds a consumer.TracesConsumer with data.
+// TracesReceiver feeds a consumer.Traces with data.
 //
 // For example it could be Zipkin data source which translates Zipkin spans into pdata.Traces.
 type TracesReceiver interface {
@@ -39,7 +39,7 @@ type TracesReceiver interface {
 
 // A MetricsReceiver is an "arbitrary data"-to-"internal format" converter.
 // Its purpose is to translate data from the wild into internal metrics format.
-// MetricsReceiver feeds a consumer.MetricsConsumer with data.
+// MetricsReceiver feeds a consumer.Metrics with data.
 //
 // For example it could be Prometheus data source which translates Prometheus metrics into pdata.Metrics.
 type MetricsReceiver interface {
@@ -48,7 +48,7 @@ type MetricsReceiver interface {
 
 // A LogsReceiver is a "log data"-to-"internal format" converter.
 // Its purpose is to translate data from the wild into internal data format.
-// LogsReceiver feeds a consumer.LogsConsumer with data.
+// LogsReceiver feeds a consumer.Logs with data.
 type LogsReceiver interface {
 	Receiver
 }
@@ -81,17 +81,17 @@ type ReceiverFactory interface {
 	// If the receiver type does not support tracing or if the config is not valid
 	// error will be returned instead.
 	CreateTracesReceiver(ctx context.Context, params ReceiverCreateParams,
-		cfg configmodels.Receiver, nextConsumer consumer.TracesConsumer) (TracesReceiver, error)
+		cfg configmodels.Receiver, nextConsumer consumer.Traces) (TracesReceiver, error)
 
 	// CreateMetricsReceiver creates a metrics receiver based on this config.
 	// If the receiver type does not support metrics or if the config is not valid
 	// error will be returned instead.
 	CreateMetricsReceiver(ctx context.Context, params ReceiverCreateParams,
-		cfg configmodels.Receiver, nextConsumer consumer.MetricsConsumer) (MetricsReceiver, error)
+		cfg configmodels.Receiver, nextConsumer consumer.Metrics) (MetricsReceiver, error)
 
 	// CreateLogsReceiver creates a log receiver based on this config.
 	// If the receiver type does not support the data type or if the config is not valid
 	// error will be returned instead.
 	CreateLogsReceiver(ctx context.Context, params ReceiverCreateParams,
-		cfg configmodels.Receiver, nextConsumer consumer.LogsConsumer) (LogsReceiver, error)
+		cfg configmodels.Receiver, nextConsumer consumer.Logs) (LogsReceiver, error)
 }
