diff --git a/translator/trace/jaeger/jaegerproto_to_traces_test.go b/translator/trace/jaeger/jaegerproto_to_traces_test.go
index 586f064f5..2dc73fa50 100644
--- a/translator/trace/jaeger/jaegerproto_to_traces_test.go
+++ b/translator/trace/jaeger/jaegerproto_to_traces_test.go
@@ -204,18 +204,34 @@ func TestProtoBatchToInternalTraces(t *testing.T) {
 			td: generateTraceDataResourceOnly(),
 		},
 
+		{
+			name: "no-resource-attrs",
+			jb: model.Batch{
+				Process: &model.Process{
+					ServiceName: tracetranslator.ResourceNoAttrs,
+				},
+			},
+			td: generateTraceDataResourceOnlyWithNoAttrs(),
+		},
+
 		{
 			name: "one-span-no-resources",
 			jb: model.Batch{
+				Process: &model.Process{
+					ServiceName: tracetranslator.ResourceNotSet,
+				},
 				Spans: []*model.Span{
-					generateProtoSpan(),
+					generateProtoSpanWithTraceState(),
 				},
 			},
-			td: generateTraceDataOneSpanNoResource(),
+			td: generateTraceDataOneSpanNoResourceWithTraceState(),
 		},
 		{
 			name: "two-spans-child-parent",
 			jb: model.Batch{
+				Process: &model.Process{
+					ServiceName: tracetranslator.ResourceNotSet,
+				},
 				Spans: []*model.Span{
 					generateProtoSpan(),
 					generateProtoChildSpan(),
@@ -227,6 +243,9 @@ func TestProtoBatchToInternalTraces(t *testing.T) {
 		{
 			name: "two-spans-with-follower",
 			jb: model.Batch{
+				Process: &model.Process{
+					ServiceName: tracetranslator.ResourceNotSet,
+				},
 				Spans: []*model.Span{
 					generateProtoSpan(),
 					generateProtoFollowerSpan(),
@@ -246,6 +265,8 @@ func TestProtoBatchToInternalTraces(t *testing.T) {
 
 func TestSetInternalSpanStatus(t *testing.T) {
 
+	nilStatus := pdata.NewSpanStatus()
+
 	okStatus := pdata.NewSpanStatus()
 	okStatus.InitEmpty()
 	okStatus.SetCode(pdata.StatusCode(otlptrace.Status_Ok))
@@ -281,7 +302,7 @@ func TestSetInternalSpanStatus(t *testing.T) {
 		{
 			name:             "No tags set -> OK status",
 			attrs:            pdata.NewAttributeMap().InitFromMap(map[string]pdata.AttributeValue{}),
-			status:           okStatus,
+			status:           nilStatus,
 			attrsModifiedLen: 0,
 		},
 		{
@@ -390,6 +411,44 @@ func TestProtoBatchesToInternalTraces(t *testing.T) {
 	assert.EqualValues(t, expected, got)
 }
 
+func TestJSpanKindToInternal(t *testing.T) {
+	tests := []struct {
+		jSpanKind    string
+		otlpSpanKind pdata.SpanKind
+	}{
+		{
+			jSpanKind:    "client",
+			otlpSpanKind: pdata.SpanKindCLIENT,
+		},
+		{
+			jSpanKind:    "server",
+			otlpSpanKind: pdata.SpanKindSERVER,
+		},
+		{
+			jSpanKind:    "producer",
+			otlpSpanKind: pdata.SpanKindPRODUCER,
+		},
+		{
+			jSpanKind:    "consumer",
+			otlpSpanKind: pdata.SpanKindCONSUMER,
+		},
+		{
+			jSpanKind:    "internal",
+			otlpSpanKind: pdata.SpanKindINTERNAL,
+		},
+		{
+			jSpanKind:    "all-others",
+			otlpSpanKind: pdata.SpanKindUNSPECIFIED,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.jSpanKind, func(t *testing.T) {
+			assert.Equal(t, test.otlpSpanKind, jSpanKindToInternal(test.jSpanKind))
+		})
+	}
+}
+
 func generateTraceDataResourceOnly() pdata.Traces {
 	td := testdata.GenerateTraceDataOneEmptyResourceSpans()
 	rs := td.ResourceSpans().At(0).Resource()
@@ -399,6 +458,13 @@ func generateTraceDataResourceOnly() pdata.Traces {
 	return td
 }
 
+func generateTraceDataResourceOnlyWithNoAttrs() pdata.Traces {
+	td := testdata.GenerateTraceDataOneEmptyResourceSpans()
+	rs := td.ResourceSpans().At(0).Resource()
+	rs.InitEmpty()
+	return td
+}
+
 func generateProtoProcess() *model.Process {
 	return &model.Process{
 		ServiceName: "service-1",
@@ -425,9 +491,7 @@ func generateTraceDataOneSpanNoResource() pdata.Traces {
 	span.SetKind(pdata.SpanKindCLIENT)
 	span.Events().At(0).SetTimestamp(testSpanEventTimestamp)
 	span.Events().At(0).SetDroppedAttributesCount(0)
-	span.Events().At(0).Attributes().InitFromMap(map[string]pdata.AttributeValue{
-		"message": pdata.NewAttributeValueString("event-with-attr"),
-	})
+	span.Events().At(0).SetName("event-with-attr")
 	span.Events().At(1).SetTimestamp(testSpanEventTimestamp)
 	span.Events().At(1).SetDroppedAttributesCount(0)
 	span.Events().At(1).SetName("")
@@ -435,6 +499,13 @@ func generateTraceDataOneSpanNoResource() pdata.Traces {
 	return td
 }
 
+func generateTraceDataOneSpanNoResourceWithTraceState() pdata.Traces {
+	td := generateTraceDataOneSpanNoResource()
+	span := td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0)
+	span.SetTraceState("lasterror=f39cd56cc44274fd5abd07ef1164246d10ce2955")
+	return td
+}
+
 func generateProtoSpan() *model.Span {
 	return &model.Span{
 		TraceID: model.NewTraceID(
@@ -450,10 +521,77 @@ func generateProtoSpan() *model.Span {
 				Timestamp: testSpanEventTime,
 				Fields: []model.KeyValue{
 					{
-						Key:   "message",
+						Key:   tracetranslator.TagMessage,
+						VType: model.ValueType_STRING,
+						VStr:  "event-with-attr",
+					},
+					{
+						Key:   "span-event-attr",
+						VType: model.ValueType_STRING,
+						VStr:  "span-event-attr-val",
+					},
+				},
+			},
+			{
+				Timestamp: testSpanEventTime,
+				Fields: []model.KeyValue{
+					{
+						Key:    "attr-int",
+						VType:  model.ValueType_INT64,
+						VInt64: 123,
+					},
+				},
+			},
+		},
+		Tags: []model.KeyValue{
+			{
+				Key:   tracetranslator.TagSpanKind,
+				VType: model.ValueType_STRING,
+				VStr:  string(tracetranslator.OpenTracingSpanKindClient),
+			},
+			{
+				Key:    tracetranslator.TagStatusCode,
+				VType:  model.ValueType_INT64,
+				VInt64: tracetranslator.OCCancelled,
+			},
+			{
+				Key:   tracetranslator.TagError,
+				VBool: true,
+				VType: model.ValueType_BOOL,
+			},
+			{
+				Key:   tracetranslator.TagStatusMsg,
+				VType: model.ValueType_STRING,
+				VStr:  "status-cancelled",
+			},
+		},
+	}
+}
+
+func generateProtoSpanWithTraceState() *model.Span {
+	return &model.Span{
+		TraceID: model.NewTraceID(
+			binary.BigEndian.Uint64([]byte{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8}),
+			binary.BigEndian.Uint64([]byte{0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80}),
+		),
+		SpanID:        model.NewSpanID(binary.BigEndian.Uint64([]byte{0xAF, 0xAE, 0xAD, 0xAC, 0xAB, 0xAA, 0xA9, 0xA8})),
+		OperationName: "operationA",
+		StartTime:     testSpanStartTime,
+		Duration:      testSpanEndTime.Sub(testSpanStartTime),
+		Logs: []model.Log{
+			{
+				Timestamp: testSpanEventTime,
+				Fields: []model.KeyValue{
+					{
+						Key:   tracetranslator.TagMessage,
 						VType: model.ValueType_STRING,
 						VStr:  "event-with-attr",
 					},
+					{
+						Key:   "span-event-attr",
+						VType: model.ValueType_STRING,
+						VStr:  "span-event-attr-val",
+					},
 				},
 			},
 			{
@@ -488,6 +626,11 @@ func generateProtoSpan() *model.Span {
 				VType: model.ValueType_STRING,
 				VStr:  "status-cancelled",
 			},
+			{
+				Key:   tracetranslator.TagW3CTraceState,
+				VType: model.ValueType_STRING,
+				VStr:  "lasterror=f39cd56cc44274fd5abd07ef1164246d10ce2955",
+			},
 		},
 	}
 }
