diff --git a/testbed/correctness/metrics/metrics_correctness_test.go b/testbed/correctness/metrics/metrics_correctness_test.go
index c98397ccc..8729e5fd8 100644
--- a/testbed/correctness/metrics/metrics_correctness_test.go
+++ b/testbed/correctness/metrics/metrics_correctness_test.go
@@ -92,7 +92,7 @@ func testWithMetricsGoldenDataset(
 
 func getTestMetrics(t *testing.T) []pdata.Metrics {
 	const file = "../../../internal/goldendataset/testdata/generated_pict_pairs_metrics.txt"
-	mds, err := goldendataset.GenerateMetricDatas(file)
+	mds, err := goldendataset.GenerateMetrics(file)
 	require.NoError(t, err)
 	return mds
 }
