diff --git a/consumer/xconsumer/go.mod b/consumer/xconsumer/go.mod
index 0304bd881..11970ac6b 100644
--- a/consumer/xconsumer/go.mod
+++ b/consumer/xconsumer/go.mod
@@ -4,8 +4,8 @@ go 1.25.0
 
 require (
 	github.com/stretchr/testify v1.11.1
-	go.opentelemetry.io/collector/consumer v1.51.0
-	go.opentelemetry.io/collector/pdata/pprofile v0.146.0
+	go.opentelemetry.io/collector/consumer v1.52.0
+	go.opentelemetry.io/collector/pdata/pprofile v0.146.1
 )
 
 require (
@@ -15,8 +15,8 @@ require (
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.3-0.20250322232337-35a7c28c31ee // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
-	go.opentelemetry.io/collector/featuregate v1.51.0 // indirect
-	go.opentelemetry.io/collector/pdata v1.51.0 // indirect
+	go.opentelemetry.io/collector/featuregate v1.52.0 // indirect
+	go.opentelemetry.io/collector/pdata v1.52.0 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
