diff --git a/pdata/pmetric/generated_summary.go b/pdata/pmetric/generated_summary.go
index e247537dc..b53c65b1b 100644
--- a/pdata/pmetric/generated_summary.go
+++ b/pdata/pmetric/generated_summary.go
@@ -32,7 +32,7 @@ func newSummary(orig *otlpmetrics.Summary, state *internal.State) Summary {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewSummary() Summary {
-	return newSummary(internal.NewOrigSummary(), internal.NewState())
+	return newSummary(internal.NewSummary(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
@@ -44,7 +44,7 @@ func (ms Summary) MoveTo(dest Summary) {
 	if ms.orig == dest.orig {
 		return
 	}
-	internal.DeleteOrigSummary(dest.orig, false)
+	internal.DeleteSummary(dest.orig, false)
 	*dest.orig, *ms.orig = *ms.orig, *dest.orig
 }
 
@@ -56,5 +56,5 @@ func (ms Summary) DataPoints() SummaryDataPointSlice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Summary) CopyTo(dest Summary) {
 	dest.state.AssertMutable()
-	internal.CopyOrigSummary(dest.orig, ms.orig)
+	internal.CopySummary(dest.orig, ms.orig)
 }
