diff --git a/pdata/pmetric/generated_metric_test.go b/pdata/pmetric/generated_metric_test.go
index edfb46e92..61c8de869 100644
--- a/pdata/pmetric/generated_metric_test.go
+++ b/pdata/pmetric/generated_metric_test.go
@@ -26,8 +26,8 @@ func TestMetric_MoveTo(t *testing.T) {
 	assert.Equal(t, generateTestMetric(), dest)
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { ms.MoveTo(newMetric(internal.NewOrigMetric(), sharedState)) })
-	assert.Panics(t, func() { newMetric(internal.NewOrigMetric(), sharedState).MoveTo(dest) })
+	assert.Panics(t, func() { ms.MoveTo(newMetric(internal.NewMetric(), sharedState)) })
+	assert.Panics(t, func() { newMetric(internal.NewMetric(), sharedState).MoveTo(dest) })
 }
 
 func TestMetric_CopyTo(t *testing.T) {
@@ -40,7 +40,7 @@ func TestMetric_CopyTo(t *testing.T) {
 	assert.Equal(t, orig, ms)
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { ms.CopyTo(newMetric(internal.NewOrigMetric(), sharedState)) })
+	assert.Panics(t, func() { ms.CopyTo(newMetric(internal.NewMetric(), sharedState)) })
 }
 
 func TestMetric_Name(t *testing.T) {
@@ -82,7 +82,7 @@ func TestMetric_Gauge(t *testing.T) {
 	ms := NewMetric()
 	ms.SetEmptyGauge()
 	assert.Equal(t, NewGauge(), ms.Gauge())
-	ms.orig.GetData().(*otlpmetrics.Metric_Gauge).Gauge = internal.GenTestOrigGauge()
+	ms.orig.GetData().(*otlpmetrics.Metric_Gauge).Gauge = internal.GenTestGauge()
 	assert.Equal(t, MetricTypeGauge, ms.Type())
 	assert.Equal(t, generateTestGauge(), ms.Gauge())
 	sharedState := internal.NewState()
@@ -94,7 +94,7 @@ func TestMetric_Sum(t *testing.T) {
 	ms := NewMetric()
 	ms.SetEmptySum()
 	assert.Equal(t, NewSum(), ms.Sum())
-	ms.orig.GetData().(*otlpmetrics.Metric_Sum).Sum = internal.GenTestOrigSum()
+	ms.orig.GetData().(*otlpmetrics.Metric_Sum).Sum = internal.GenTestSum()
 	assert.Equal(t, MetricTypeSum, ms.Type())
 	assert.Equal(t, generateTestSum(), ms.Sum())
 	sharedState := internal.NewState()
@@ -106,7 +106,7 @@ func TestMetric_Histogram(t *testing.T) {
 	ms := NewMetric()
 	ms.SetEmptyHistogram()
 	assert.Equal(t, NewHistogram(), ms.Histogram())
-	ms.orig.GetData().(*otlpmetrics.Metric_Histogram).Histogram = internal.GenTestOrigHistogram()
+	ms.orig.GetData().(*otlpmetrics.Metric_Histogram).Histogram = internal.GenTestHistogram()
 	assert.Equal(t, MetricTypeHistogram, ms.Type())
 	assert.Equal(t, generateTestHistogram(), ms.Histogram())
 	sharedState := internal.NewState()
@@ -118,7 +118,7 @@ func TestMetric_ExponentialHistogram(t *testing.T) {
 	ms := NewMetric()
 	ms.SetEmptyExponentialHistogram()
 	assert.Equal(t, NewExponentialHistogram(), ms.ExponentialHistogram())
-	ms.orig.GetData().(*otlpmetrics.Metric_ExponentialHistogram).ExponentialHistogram = internal.GenTestOrigExponentialHistogram()
+	ms.orig.GetData().(*otlpmetrics.Metric_ExponentialHistogram).ExponentialHistogram = internal.GenTestExponentialHistogram()
 	assert.Equal(t, MetricTypeExponentialHistogram, ms.Type())
 	assert.Equal(t, generateTestExponentialHistogram(), ms.ExponentialHistogram())
 	sharedState := internal.NewState()
@@ -130,7 +130,7 @@ func TestMetric_Summary(t *testing.T) {
 	ms := NewMetric()
 	ms.SetEmptySummary()
 	assert.Equal(t, NewSummary(), ms.Summary())
-	ms.orig.GetData().(*otlpmetrics.Metric_Summary).Summary = internal.GenTestOrigSummary()
+	ms.orig.GetData().(*otlpmetrics.Metric_Summary).Summary = internal.GenTestSummary()
 	assert.Equal(t, MetricTypeSummary, ms.Type())
 	assert.Equal(t, generateTestSummary(), ms.Summary())
 	sharedState := internal.NewState()
@@ -141,11 +141,11 @@ func TestMetric_Summary(t *testing.T) {
 func TestMetric_Metadata(t *testing.T) {
 	ms := NewMetric()
 	assert.Equal(t, pcommon.NewMap(), ms.Metadata())
-	ms.orig.Metadata = internal.GenerateOrigTestKeyValueSlice()
-	assert.Equal(t, pcommon.Map(internal.GenerateTestMap()), ms.Metadata())
+	ms.orig.Metadata = internal.GenTestKeyValueSlice()
+	assert.Equal(t, pcommon.Map(internal.GenTestMapWrapper()), ms.Metadata())
 }
 
 func generateTestMetric() Metric {
-	ms := newMetric(internal.GenTestOrigMetric(), internal.NewState())
+	ms := newMetric(internal.GenTestMetric(), internal.NewState())
 	return ms
 }
