diff --git a/internal/tools/go.mod b/internal/tools/go.mod
index 3151a4858..b996beead 100644
--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -12,6 +12,7 @@ require (
 	github.com/jcchavezs/porto v0.7.0
 	github.com/pavius/impi v0.0.3
 	github.com/rhysd/actionlint v1.7.7
+	go.opentelemetry.io/build-tools/checkapi v0.23.0
 	go.opentelemetry.io/build-tools/checkfile v0.23.0
 	go.opentelemetry.io/build-tools/chloggen v0.23.0
 	go.opentelemetry.io/build-tools/crosslink v0.23.0
