diff --git a/service/telemetry/telemetrytest/go.mod b/service/telemetry/telemetrytest/go.mod
index c94af06cf..6593028ee 100644
--- a/service/telemetry/telemetrytest/go.mod
+++ b/service/telemetry/telemetrytest/go.mod
@@ -57,7 +57,7 @@ require (
 	go.yaml.in/yaml/v2 v2.4.3 // indirect
 	go.yaml.in/yaml/v3 v3.0.4 // indirect
 	golang.org/x/net v0.51.0 // indirect
-	golang.org/x/sys v0.41.0 // indirect
+	golang.org/x/sys v0.42.0 // indirect
 	golang.org/x/text v0.34.0 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20260226221140-a57be14db171 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20260226221140-a57be14db171 // indirect
