diff --git a/pdata/internal/generated_wrapper_stringslice.go b/pdata/internal/generated_wrapper_stringslice.go
index 2e5c7d510..856562a72 100644
--- a/pdata/internal/generated_wrapper_stringslice.go
+++ b/pdata/internal/generated_wrapper_stringslice.go
@@ -27,11 +27,6 @@ func NewStringSlice(orig *[]string, state *State) StringSlice {
 	return StringSlice{orig: orig, state: state}
 }
 
-func CopyOrigStringSlice(dst, src []string) []string {
-	dst = dst[:0]
-	return append(dst, src...)
-}
-
 func FillTestStringSlice(ms StringSlice) {
 	*ms.orig = []string{"a", "b", "c"}
 }
@@ -64,3 +59,7 @@ func UnmarshalJSONIterStringSlice(ms StringSlice, iter *json.Iterator) {
 		return true
 	})
 }
+
+func CopyOrigStringSlice(dst, src []string) []string {
+	return append(dst[:0], src...)
+}
