diff --git a/exporter/exporterhelper/metricshelper_test.go b/exporter/exporterhelper/metricshelper_test.go
index 3f98497b3..f286444d5 100644
--- a/exporter/exporterhelper/metricshelper_test.go
+++ b/exporter/exporterhelper/metricshelper_test.go
@@ -20,24 +20,23 @@ import (
 
 	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"go.opencensus.io/trace"
 
 	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
 	"github.com/open-telemetry/opentelemetry-service/exporter"
 )
 
-// TODO https://github.com/open-telemetry/opentelemetry-service/issues/266
-// Migrate tests to use testify/assert instead of t.Fatal pattern.
 func TestMetricsExporter_InvalidName(t *testing.T) {
-	if _, err := NewMetricsExporter("", newPushMetricsData(0, nil)); err != errEmptyExporterName {
-		t.Fatalf("NewMetricsExporter returns: Want %v Got %v", errEmptyExporterName, err)
-	}
+	me, err := NewMetricsExporter("", newPushMetricsData(0, nil))
+	require.Nil(t, me)
+	require.Equal(t, errEmptyExporterName, err)
 }
 
 func TestMetricsExporter_NilPushMetricsData(t *testing.T) {
-	if _, err := NewMetricsExporter(fakeExporterName, nil); err != errNilPushMetricsData {
-		t.Fatalf("NewMetricsExporter returns: Want %v Got %v", errNilPushMetricsData, err)
-	}
+	me, err := NewMetricsExporter(fakeExporterName, nil)
+	require.Nil(t, me)
+	require.Equal(t, errNilPushMetricsData, err)
 }
 
 func TestMetricsExporter_Default(t *testing.T) {
@@ -57,36 +56,30 @@ func TestMetricsExporter_Default_ReturnError(t *testing.T) {
 	td := consumerdata.MetricsData{}
 	want := errors.New("my_error")
 	te, err := NewMetricsExporter(fakeExporterName, newPushMetricsData(0, want))
-	if err != nil {
-		t.Fatalf("NewMetricsExporter returns: Want nil Got %v", err)
-	}
-	if err := te.ConsumeMetricsData(context.Background(), td); err != want {
-		t.Fatalf("ConsumeMetricsData returns: Want %v Got %v", want, err)
-	}
+	require.Nil(t, err)
+	require.NotNil(t, te)
+	require.Equal(t, want, te.ConsumeMetricsData(context.Background(), td))
 }
 
 func TestMetricsExporter_WithSpan(t *testing.T) {
 	te, err := NewMetricsExporter(fakeExporterName, newPushMetricsData(0, nil), WithSpanName(fakeSpanName))
-	if err != nil {
-		t.Fatalf("NewMetricsExporter returns: Want nil Got %v", err)
-	}
+	require.Nil(t, err)
+	require.NotNil(t, te)
 	checkWrapSpanForMetricsExporter(t, te, nil, 0)
 }
 
 func TestMetricsExporter_WithSpan_NonZeroDropped(t *testing.T) {
 	te, err := NewMetricsExporter(fakeExporterName, newPushMetricsData(1, nil), WithSpanName(fakeSpanName))
-	if err != nil {
-		t.Fatalf("NewMetricsExporter returns: Want nil Got %v", err)
-	}
+	require.Nil(t, err)
+	require.NotNil(t, te)
 	checkWrapSpanForMetricsExporter(t, te, nil, 1)
 }
 
 func TestMetricsExporter_WithSpan_ReturnError(t *testing.T) {
 	want := errors.New("my_error")
 	te, err := NewMetricsExporter(fakeExporterName, newPushMetricsData(0, want), WithSpanName(fakeSpanName))
-	if err != nil {
-		t.Fatalf("NewMetricsExporter returns: Want nil Got %v", err)
-	}
+	require.Nil(t, err)
+	require.NotNil(t, te)
 	checkWrapSpanForMetricsExporter(t, te, want, 0)
 }
 
@@ -124,9 +117,7 @@ func generateMetricsTraffic(t *testing.T, te exporter.MetricsExporter, numReques
 	ctx, span := trace.StartSpan(context.Background(), fakeParentSpanName, trace.WithSampler(trace.AlwaysSample()))
 	defer span.End()
 	for i := 0; i < numRequests; i++ {
-		if err := te.ConsumeMetricsData(ctx, td); err != wantError {
-			t.Fatalf("Want %v Got %v", wantError, err)
-		}
+		require.Equal(t, wantError, te.ConsumeMetricsData(ctx, td))
 	}
 }
 
@@ -142,32 +133,18 @@ func checkWrapSpanForMetricsExporter(t *testing.T, te exporter.MetricsExporter,
 	ocSpansSaver.mu.Lock()
 	defer ocSpansSaver.mu.Unlock()
 
-	if len(ocSpansSaver.spanData) == 0 {
-		t.Fatal("No exported span data.")
-	}
+	require.NotEqual(t, 0, len(ocSpansSaver.spanData), "No exported span data")
 
 	gotSpanData := ocSpansSaver.spanData[:]
-	if g, w := len(gotSpanData), numRequests+1; g != w {
-		t.Fatalf("Spandata count: Want %d Got %d", w, g)
-	}
+	require.Equal(t, numRequests+1, len(gotSpanData))
 
 	parentSpan := gotSpanData[numRequests]
-	if g, w := parentSpan.Name, fakeParentSpanName; g != w {
-		t.Fatalf("Parent span name: Want %s Got %s\nSpanData %v", w, g, parentSpan)
-	}
+	require.Equalf(t, fakeParentSpanName, parentSpan.Name, "SpanData %v", parentSpan)
 
 	for _, sd := range gotSpanData[:numRequests] {
-		if g, w := sd.ParentSpanID, parentSpan.SpanContext.SpanID; g != w {
-			t.Fatalf("Exporter span not a child: Want %d Got %d\nSpanData %v", w, g, sd)
-		}
-		if g, w := sd.Status, errToStatus(wantError); g != w {
-			t.Fatalf("Status: Want %v Got %v\nSpanData %v", w, g, sd)
-		}
-		if g, w := sd.Attributes[numReceivedMetricsAttribute], int64(1); g != w {
-			t.Fatalf("Number of received spans attribute: Want %d Got %d\nSpanData %v", w, g, sd)
-		}
-		if g, w := sd.Attributes[numDroppedMetricsAttribute], int64(droppedSpans); g != w {
-			t.Fatalf("Number of dropped spans attribute: Want %d Got %d\nSpanData %v", w, g, sd)
-		}
+		require.Equalf(t, parentSpan.SpanContext.SpanID, sd.ParentSpanID, "Exporter span not a child\nSpanData %v", sd)
+		require.Equalf(t, errToStatus(wantError), sd.Status, "SpanData %v", sd)
+		require.Equalf(t, int64(1), sd.Attributes[numReceivedMetricsAttribute], "SpanData %v", sd)
+		require.Equalf(t, int64(droppedSpans), sd.Attributes[numDroppedMetricsAttribute], "SpanData %v", sd)
 	}
 }
