diff --git a/service/pipelines/config.go b/service/pipelines/config.go
index b30afa009..154d848a1 100644
--- a/service/pipelines/config.go
+++ b/service/pipelines/config.go
@@ -8,9 +8,9 @@ import (
 	"fmt"
 
 	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/pipeline"
 	"go.opentelemetry.io/collector/pipeline/xpipeline"
+	"go.opentelemetry.io/collector/service/internal/metadata"
 )
 
 var (
@@ -18,19 +18,7 @@ var (
 	errMissingServicePipelineReceivers = errors.New("must have at least one receiver")
 	errMissingServicePipelineExporters = errors.New("must have at least one exporter")
 
-	serviceProfileSupportGateID = "service.profilesSupport"
-	serviceProfileSupportGate   = featuregate.GlobalRegistry().MustRegister(
-		serviceProfileSupportGateID,
-		featuregate.StageAlpha,
-		featuregate.WithRegisterFromVersion("v0.112.0"),
-		featuregate.WithRegisterDescription("Controls whether profiles support can be enabled"),
-	)
-	AllowNoPipelines = featuregate.GlobalRegistry().MustRegister(
-		"service.AllowNoPipelines",
-		featuregate.StageAlpha,
-		featuregate.WithRegisterFromVersion("v0.122.0"),
-		featuregate.WithRegisterDescription("Allow starting the Collector without starting any pipelines."),
-	)
+	AllowNoPipelines = metadata.ServiceAllowNoPipelinesFeatureGate
 )
 
 // Config defines the configurable settings for service telemetry.
@@ -38,11 +26,11 @@ type Config map[pipeline.ID]*PipelineConfig
 
 func (cfg Config) Validate() error {
 	// Must have at least one pipeline unless explicitly disabled.
-	if !AllowNoPipelines.IsEnabled() && len(cfg) == 0 {
+	if !metadata.ServiceAllowNoPipelinesFeatureGate.IsEnabled() && len(cfg) == 0 {
 		return errMissingServicePipelines
 	}
 
-	if !serviceProfileSupportGate.IsEnabled() {
+	if !metadata.ServiceProfilesSupportFeatureGate.IsEnabled() {
 		// Check that all pipelines have at least one receiver and one exporter, and they reference
 		// only configured components.
 		for pipelineID := range cfg {
@@ -50,7 +38,7 @@ func (cfg Config) Validate() error {
 				return fmt.Errorf(
 					"pipeline %q: profiling signal support is at alpha level, gated under the %q feature gate",
 					pipelineID.String(),
-					serviceProfileSupportGateID,
+					metadata.ServiceProfilesSupportFeatureGate.ID(),
 				)
 			}
 		}
