diff --git a/model/otlp/pb_marshaler.go b/model/otlp/pb_marshaler.go
index e5a8279b6..1d0b1702e 100644
--- a/model/otlp/pb_marshaler.go
+++ b/model/otlp/pb_marshaler.go
@@ -15,56 +15,19 @@
 package otlp // import "go.opentelemetry.io/collector/model/otlp"
 
 import (
-	ipdata "go.opentelemetry.io/collector/model/internal"
-	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
 // NewProtobufTracesMarshaler returns a pdata.TracesMarshaler. Marshals to OTLP binary protobuf bytes.
-func NewProtobufTracesMarshaler() pdata.TracesMarshaler {
-	return newPbMarshaler()
-}
+// Deprecated: [v0.49.0] Use ptrace.NewProtoMarshaler instead.
+var NewProtobufTracesMarshaler = ptrace.NewProtoMarshaler
 
 // NewProtobufMetricsMarshaler returns a pdata.MetricsMarshaler. Marshals to OTLP binary protobuf bytes.
-func NewProtobufMetricsMarshaler() pdata.MetricsMarshaler {
-	return newPbMarshaler()
-}
+// Deprecated: [v0.49.0] Use pmetric.NewProtoMarshaler instead.
+var NewProtobufMetricsMarshaler = pmetric.NewProtoMarshaler
 
 // NewProtobufLogsMarshaler returns a pdata.LogsMarshaler. Marshals to OTLP binary protobuf bytes.
-func NewProtobufLogsMarshaler() pdata.LogsMarshaler {
-	return newPbMarshaler()
-}
-
-// TODO(#3842): Figure out how we want to represent/return *Sizers.
-type pbMarshaler struct{}
-
-func newPbMarshaler() *pbMarshaler {
-	return &pbMarshaler{}
-}
-
-var _ pdata.TracesSizer = (*pbMarshaler)(nil)
-var _ pdata.MetricsSizer = (*pbMarshaler)(nil)
-var _ pdata.LogsSizer = (*pbMarshaler)(nil)
-
-func (e *pbMarshaler) MarshalLogs(ld pdata.Logs) ([]byte, error) {
-	return ipdata.LogsToOtlp(ld).Marshal()
-}
-
-func (e *pbMarshaler) MarshalMetrics(md pdata.Metrics) ([]byte, error) {
-	return ipdata.MetricsToOtlp(md).Marshal()
-}
-
-func (e *pbMarshaler) MarshalTraces(td pdata.Traces) ([]byte, error) {
-	return ipdata.TracesToOtlp(td).Marshal()
-}
-
-func (e *pbMarshaler) TracesSize(td pdata.Traces) int {
-	return ipdata.TracesToOtlp(td).Size()
-}
-
-func (e *pbMarshaler) MetricsSize(md pdata.Metrics) int {
-	return ipdata.MetricsToOtlp(md).Size()
-}
-
-func (e *pbMarshaler) LogsSize(ld pdata.Logs) int {
-	return ipdata.LogsToOtlp(ld).Size()
-}
+// Deprecated: [v0.49.0] Use plog.NewProtoMarshaler instead.
+var NewProtobufLogsMarshaler = plog.NewProtoMarshaler
