diff --git a/model/otlp/pb_unmarshaler.go b/model/otlp/pb_unmarshaler.go
index 5a5fc1b89..d8fc4b7a4 100644
--- a/model/otlp/pb_unmarshaler.go
+++ b/model/otlp/pb_unmarshaler.go
@@ -15,48 +15,19 @@
 package otlp // import "go.opentelemetry.io/collector/model/otlp"
 
 import (
-	ipdata "go.opentelemetry.io/collector/model/internal"
-	otlplogs "go.opentelemetry.io/collector/model/internal/data/protogen/logs/v1"
-	otlpmetrics "go.opentelemetry.io/collector/model/internal/data/protogen/metrics/v1"
-	otlptrace "go.opentelemetry.io/collector/model/internal/data/protogen/trace/v1"
-	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
-type pbUnmarshaler struct{}
-
 // NewProtobufTracesUnmarshaler returns a model.TracesUnmarshaler. Unmarshals from OTLP binary protobuf bytes.
-func NewProtobufTracesUnmarshaler() pdata.TracesUnmarshaler {
-	return newPbUnmarshaler()
-}
+// Deprecated: [v0.49.0] Use ptrace.NewProtoUnmarshaler instead.
+var NewProtobufTracesUnmarshaler = ptrace.NewProtoUnmarshaler
 
 // NewProtobufMetricsUnmarshaler returns a model.MetricsUnmarshaler. Unmarshals from OTLP binary protobuf bytes.
-func NewProtobufMetricsUnmarshaler() pdata.MetricsUnmarshaler {
-	return newPbUnmarshaler()
-}
+// Deprecated: [v0.49.0] Use pmetric.NewProtoUnmarshaler instead.
+var NewProtobufMetricsUnmarshaler = pmetric.NewProtoUnmarshaler
 
 // NewProtobufLogsUnmarshaler returns a model.LogsUnmarshaler. Unmarshals from OTLP binary protobuf bytes.
-func NewProtobufLogsUnmarshaler() pdata.LogsUnmarshaler {
-	return newPbUnmarshaler()
-}
-
-func newPbUnmarshaler() *pbUnmarshaler {
-	return &pbUnmarshaler{}
-}
-
-func (d *pbUnmarshaler) UnmarshalLogs(buf []byte) (pdata.Logs, error) {
-	ld := &otlplogs.LogsData{}
-	err := ld.Unmarshal(buf)
-	return ipdata.LogsFromOtlp(ld), err
-}
-
-func (d *pbUnmarshaler) UnmarshalMetrics(buf []byte) (pdata.Metrics, error) {
-	md := &otlpmetrics.MetricsData{}
-	err := md.Unmarshal(buf)
-	return ipdata.MetricsFromOtlp(md), err
-}
-
-func (d *pbUnmarshaler) UnmarshalTraces(buf []byte) (pdata.Traces, error) {
-	td := &otlptrace.TracesData{}
-	err := td.Unmarshal(buf)
-	return ipdata.TracesFromOtlp(td), err
-}
+// Deprecated: [v0.49.0] Use plog.NewProtoUnmarshaler instead.
+var NewProtobufLogsUnmarshaler = plog.NewProtoUnmarshaler
