diff --git a/client/go.mod b/client/go.mod
index 2ea83d2fd..fc6443c02 100644
--- a/client/go.mod
+++ b/client/go.mod
@@ -4,8 +4,8 @@ go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/consumer v1.23.0
-	go.opentelemetry.io/collector/pdata v1.23.0
+	go.opentelemetry.io/collector/consumer v1.24.0
+	go.opentelemetry.io/collector/pdata v1.24.0
 	go.uber.org/goleak v1.3.0
 )
 
