diff --git a/pdata/pprofile/generated_location_test.go b/pdata/pprofile/generated_location_test.go
index 3427774a9..eedce7e69 100644
--- a/pdata/pprofile/generated_location_test.go
+++ b/pdata/pprofile/generated_location_test.go
@@ -26,8 +26,8 @@ func TestLocation_MoveTo(t *testing.T) {
 	assert.Equal(t, generateTestLocation(), dest)
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { ms.MoveTo(newLocation(&otlpprofiles.Location{}, sharedState)) })
-	assert.Panics(t, func() { newLocation(&otlpprofiles.Location{}, sharedState).MoveTo(dest) })
+	assert.Panics(t, func() { ms.MoveTo(newLocation(internal.NewOrigPtrLocation(), sharedState)) })
+	assert.Panics(t, func() { newLocation(internal.NewOrigPtrLocation(), sharedState).MoveTo(dest) })
 }
 
 func TestLocation_CopyTo(t *testing.T) {
@@ -40,7 +40,7 @@ func TestLocation_CopyTo(t *testing.T) {
 	assert.Equal(t, orig, ms)
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { ms.CopyTo(newLocation(&otlpprofiles.Location{}, sharedState)) })
+	assert.Panics(t, func() { ms.CopyTo(newLocation(internal.NewOrigPtrLocation(), sharedState)) })
 }
 
 func TestLocation_MappingIndex(t *testing.T) {
