diff --git a/translator/internaldata/traces_to_oc.go b/translator/internaldata/traces_to_oc.go
index 98c3f0728..518792f74 100644
--- a/translator/internaldata/traces_to_oc.go
+++ b/translator/internaldata/traces_to_oc.go
@@ -386,10 +386,6 @@ func spanIDToOC(sid pdata.SpanID) []byte {
 }
 
 func statusToOC(status pdata.SpanStatus) (*octrace.Status, *octrace.AttributeValue) {
-	if status.IsNil() {
-		return nil, nil
-	}
-
 	var attr *octrace.AttributeValue
 	var oc int32
 	switch status.Code() {
