diff --git a/otelcol/otelcoltest/nop_factories.go b/otelcol/otelcoltest/nop_factories.go
index 930db9fc8..326430f8e 100644
--- a/otelcol/otelcoltest/nop_factories.go
+++ b/otelcol/otelcoltest/nop_factories.go
@@ -4,43 +4,24 @@
 package otelcoltest // import "go.opentelemetry.io/collector/otelcol/otelcoltest"
 
 import (
-	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/connector/connectortest"
-	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/exportertest"
-	"go.opentelemetry.io/collector/extension"
 	"go.opentelemetry.io/collector/extension/extensiontest"
 	"go.opentelemetry.io/collector/otelcol"
-	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/processor/processortest"
-	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/receiver/receivertest"
 )
 
 // NopFactories returns a otelcol.Factories with all nop factories.
 func NopFactories() (otelcol.Factories, error) {
 	var factories otelcol.Factories
-	var err error
 
-	if factories.Extensions, err = extension.MakeFactoryMap(extensiontest.NewNopFactory()); err != nil {
-		return otelcol.Factories{}, err
-	}
+	// MakeFactoryMap can never return an error with a single Factory
+	factories.Extensions, _ = otelcol.MakeFactoryMap(extensiontest.NewNopFactory())
+	factories.Receivers, _ = otelcol.MakeFactoryMap(receivertest.NewNopFactory())
+	factories.Exporters, _ = otelcol.MakeFactoryMap(exportertest.NewNopFactory())
+	factories.Processors, _ = otelcol.MakeFactoryMap(processortest.NewNopFactory())
+	factories.Connectors, _ = otelcol.MakeFactoryMap(connectortest.NewNopFactory())
 
-	if factories.Receivers, err = receiver.MakeFactoryMap(receivertest.NewNopFactory()); err != nil {
-		return otelcol.Factories{}, err
-	}
-
-	if factories.Exporters, err = exporter.MakeFactoryMap(exportertest.NewNopFactory()); err != nil {
-		return otelcol.Factories{}, err
-	}
-
-	if factories.Processors, err = processor.MakeFactoryMap(processortest.NewNopFactory()); err != nil {
-		return otelcol.Factories{}, err
-	}
-
-	if factories.Connectors, err = connector.MakeFactoryMap(connectortest.NewNopFactory()); err != nil {
-		return otelcol.Factories{}, err
-	}
-
-	return factories, err
+	return factories, nil
 }
