diff --git a/processor/batchprocessor/go.mod b/processor/batchprocessor/go.mod
index e69d61047..9d6f36a77 100644
--- a/processor/batchprocessor/go.mod
+++ b/processor/batchprocessor/go.mod
@@ -5,13 +5,13 @@ go 1.22.0
 require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector v0.107.0
-	go.opentelemetry.io/collector/client v1.13.0
+	go.opentelemetry.io/collector/client v1.14.0
 	go.opentelemetry.io/collector/component v0.107.0
 	go.opentelemetry.io/collector/config/configtelemetry v0.107.0
-	go.opentelemetry.io/collector/confmap v0.107.0
+	go.opentelemetry.io/collector/confmap v1.14.0
 	go.opentelemetry.io/collector/consumer v0.107.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.107.0
-	go.opentelemetry.io/collector/pdata v1.13.0
+	go.opentelemetry.io/collector/pdata v1.14.0
 	go.opentelemetry.io/collector/pdata/testdata v0.107.0
 	go.opentelemetry.io/collector/processor v0.107.0
 	go.opentelemetry.io/otel v1.28.0
