diff --git a/exporter/exporterparser/datadog.go b/exporter/exporterparser/datadog.go
index b8340726a..40a5d783d 100644
--- a/exporter/exporterparser/datadog.go
+++ b/exporter/exporterparser/datadog.go
@@ -46,6 +46,8 @@ type datadogExporter struct {
 	exporter *datadog.Exporter
 }
 
+// DatadogTraceExportersFromYAML parses the yaml bytes and returns an exporter.TraceExporter targeting
+// Datadog according to the configuration settings.
 func DatadogTraceExportersFromYAML(config []byte) (tes []exporter.TraceExporter, doneFns []func() error, err error) {
 	var cfg struct {
 		Exporters *struct {
