diff --git a/pdata/internal/cmd/pdatagen/internal/packages.go b/pdata/internal/cmd/pdatagen/internal/packages.go
index 47708bc97..8b0e2bb47 100644
--- a/pdata/internal/cmd/pdatagen/internal/packages.go
+++ b/pdata/internal/cmd/pdatagen/internal/packages.go
@@ -45,7 +45,7 @@ type PackageInfo struct {
 // GenerateFiles generates files with the configured data structures for this Package.
 func (p *Package) GenerateFiles() error {
 	for _, s := range p.structs {
-		path := filepath.Join("pdata", p.info.path, "generated_"+strings.ToLower(s.getName())+".go")
+		path := filepath.Join(p.info.path, "generated_"+strings.ToLower(s.getName())+".go")
 		if err := os.WriteFile(path, s.generate(p.info), 0600); err != nil {
 			return err
 		}
@@ -56,7 +56,7 @@ func (p *Package) GenerateFiles() error {
 // GenerateTestFiles generates files with tests for the configured data structures for this Package.
 func (p *Package) GenerateTestFiles() error {
 	for _, s := range p.structs {
-		path := filepath.Join("pdata", p.info.path, "generated_"+strings.ToLower(s.getName())+"_test.go")
+		path := filepath.Join(p.info.path, "generated_"+strings.ToLower(s.getName())+"_test.go")
 		if err := os.WriteFile(path, s.generateTests(p.info), 0600); err != nil {
 			return err
 		}
@@ -71,7 +71,7 @@ func (p *Package) GenerateInternalFiles() error {
 	}
 
 	for _, s := range p.structs {
-		path := filepath.Join("pdata", "internal", "generated_wrapper_"+strings.ToLower(s.getName())+".go")
+		path := filepath.Join("internal", "generated_wrapper_"+strings.ToLower(s.getName())+".go")
 		if err := os.WriteFile(path, s.generateInternal(p.info), 0600); err != nil {
 			return err
 		}
