diff --git a/receiver/scraperhelper/internal/metadata/generated_telemetry.go b/receiver/scraperhelper/internal/metadata/generated_telemetry.go
index 6c76bccd5..4aa060e9c 100644
--- a/receiver/scraperhelper/internal/metadata/generated_telemetry.go
+++ b/receiver/scraperhelper/internal/metadata/generated_telemetry.go
@@ -33,6 +33,9 @@ type telemetryBuilderOption func(*TelemetryBuilder)
 // for a component
 func NewTelemetryBuilder(settings component.TelemetrySettings, options ...telemetryBuilderOption) (*TelemetryBuilder, error) {
 	builder := TelemetryBuilder{}
+	for _, op := range options {
+		op(&builder)
+	}
 	var err, errs error
 	meter := Meter(settings)
 	builder.ScraperErroredMetricPoints, err = meter.Int64Counter(
