diff --git a/model/pdata/generated_common.go b/model/pdata/generated_common.go
index a726420cc..4bbef6f83 100644
--- a/model/pdata/generated_common.go
+++ b/model/pdata/generated_common.go
@@ -158,15 +158,6 @@ func (es AnyValueArray) Resize(newLen int) {
 	}
 }
 
-// Append will increase the length of the AnyValueArray by one and set the
-// given AttributeValue at that new position.  The original AttributeValue
-// could still be referenced so do not reuse it after passing it to this
-// method.
-// Deprecated: Use AppendEmpty.
-func (es AnyValueArray) Append(e AttributeValue) {
-	*es.orig = append(*es.orig, *e.orig)
-}
-
 // AppendEmpty will append to the end of the slice an empty AttributeValue.
 // It returns the newly added AttributeValue.
 func (es AnyValueArray) AppendEmpty() AttributeValue {
