diff --git a/exporter/exportertest/sink_exporter_test.go b/exporter/exportertest/sink_exporter_test.go
index 127c2f749..6348d74d3 100644
--- a/exporter/exportertest/sink_exporter_test.go
+++ b/exporter/exportertest/sink_exporter_test.go
@@ -25,6 +25,7 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/consumer/pdatautil"
 	"go.opentelemetry.io/collector/internal/data/testdata"
+	"go.opentelemetry.io/collector/internal/dataold/testdataold"
 )
 
 func TestSinkTraceExporter(t *testing.T) {
@@ -58,7 +59,7 @@ func TestSinkTraceExporter_Error(t *testing.T) {
 func TestSinkMetricsExporter(t *testing.T) {
 	sink := new(SinkMetricsExporter)
 	require.NoError(t, sink.Start(context.Background(), componenttest.NewNopHost()))
-	md := testdata.GenerateMetricDataOneMetric()
+	md := testdataold.GenerateMetricDataOneMetric()
 	want := make([]pdata.Metrics, 0, 7)
 	for i := 0; i < 7; i++ {
 		require.NoError(t, sink.ConsumeMetrics(context.Background(), pdatautil.MetricsFromInternalMetrics(md)))
@@ -76,7 +77,7 @@ func TestSinkMetricsExporter_Error(t *testing.T) {
 	sink := new(SinkMetricsExporter)
 	require.NoError(t, sink.Start(context.Background(), componenttest.NewNopHost()))
 	sink.SetConsumeMetricsError(errors.New("my error"))
-	md := testdata.GenerateMetricDataOneMetric()
+	md := testdataold.GenerateMetricDataOneMetric()
 	require.Error(t, sink.ConsumeMetrics(context.Background(), pdatautil.MetricsFromInternalMetrics(md)))
 	assert.Len(t, sink.AllMetrics(), 0)
 	assert.Equal(t, 0, sink.MetricsCount())
