diff --git a/processor/xprocessor/go.mod b/processor/xprocessor/go.mod
index 778aa1fd0..3c92d9c93 100644
--- a/processor/xprocessor/go.mod
+++ b/processor/xprocessor/go.mod
@@ -18,7 +18,6 @@ require (
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
-	go.opentelemetry.io/collector/config/configtelemetry v0.119.0 // indirect
 	go.opentelemetry.io/collector/consumer v1.25.0 // indirect
 	go.opentelemetry.io/collector/pdata v1.25.0 // indirect
 	go.opentelemetry.io/collector/pdata/pprofile v0.119.0 // indirect
@@ -38,8 +37,6 @@ require (
 
 replace go.opentelemetry.io/collector/processor => ../
 
-replace go.opentelemetry.io/collector/config/configtelemetry => ../../config/configtelemetry
-
 replace go.opentelemetry.io/collector/pdata/pprofile => ../../pdata/pprofile
 
 replace go.opentelemetry.io/collector/consumer => ../../consumer
