diff --git a/consumer/go.mod b/consumer/go.mod
index 1435ded0b..5142fc7b5 100644
--- a/consumer/go.mod
+++ b/consumer/go.mod
@@ -4,8 +4,8 @@ go 1.21
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/pdata v1.5.0
-	go.opentelemetry.io/collector/pdata/testdata v0.98.0
+	go.opentelemetry.io/collector/pdata v1.6.0
+	go.opentelemetry.io/collector/pdata/testdata v0.99.0
 	go.uber.org/goleak v1.3.0
 )
 
