diff --git a/internal/processor/filterspan/config.go b/internal/processor/filterspan/config.go
index 90b640297..a3d6cde10 100644
--- a/internal/processor/filterspan/config.go
+++ b/internal/processor/filterspan/config.go
@@ -1,4 +1,4 @@
-// Copyright 2020, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
