diff --git a/service/config_test.go b/service/config_test.go
index 2dd0246e7..413887de7 100644
--- a/service/config_test.go
+++ b/service/config_test.go
@@ -9,6 +9,7 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/contrib/config"
 	"go.uber.org/zap/zapcore"
 
 	"go.opentelemetry.io/collector/component"
@@ -67,7 +68,7 @@ func TestConfigValidate(t *testing.T) {
 			cfgFn: func() *Config {
 				cfg := generateConfig()
 				cfg.Telemetry.Metrics.Level = configtelemetry.LevelBasic
-				cfg.Telemetry.Metrics.Address = ""
+				cfg.Telemetry.Metrics.Readers = nil
 				return cfg
 			},
 			expected: nil,
@@ -96,8 +97,13 @@ func generateConfig() *Config {
 				InitialFields:     map[string]any{"fieldKey": "filed-value"},
 			},
 			Metrics: telemetry.MetricsConfig{
-				Level:   configtelemetry.LevelNormal,
-				Address: ":8080",
+				Level: configtelemetry.LevelNormal,
+				Readers: []config.MetricReader{{
+					Pull: &config.PullMetricReader{Exporter: config.MetricExporter{Prometheus: &config.Prometheus{
+						Host: newPtr("localhost"),
+						Port: newPtr(8080),
+					}}}},
+				},
 			},
 		},
 		Extensions: extensions.Config{component.MustNewID("nop")},
