diff --git a/connector/go.mod b/connector/go.mod
index 911d40905..f2b20ff3e 100644
--- a/connector/go.mod
+++ b/connector/go.mod
@@ -64,3 +64,5 @@ replace go.opentelemetry.io/collector/receiver => ../receiver
 replace go.opentelemetry.io/collector/semconv => ../semconv
 
 replace go.opentelemetry.io/collector/exporter => ../exporter
+
+replace go.opentelemetry.io/collector/config/configcompression => ../config/configcompression
