diff --git a/exporter/exporterhelper/queue_batch.go b/exporter/exporterhelper/queue_batch.go
index 7e0527c55..f4e3a4390 100644
--- a/exporter/exporterhelper/queue_batch.go
+++ b/exporter/exporterhelper/queue_batch.go
@@ -4,6 +4,8 @@
 package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
 
 import (
+	"context"
+
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/queue"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/queuebatch"
@@ -38,11 +40,11 @@ type BatchConfig = queuebatch.BatchConfig
 // QueueBatchEncoding defines the encoding to be used if persistent queue is configured.
 // Duplicate definition with queuebatch.Encoding since aliasing generics is not supported by default.
 type QueueBatchEncoding[T any] interface {
-	// Marshal is a function that can marshal a request into bytes.
-	Marshal(T) ([]byte, error)
+	// Marshal is a function that can marshal a request and its context into bytes.
+	Marshal(context.Context, T) ([]byte, error)
 
-	// Unmarshal is a function that can unmarshal bytes into a request.
-	Unmarshal([]byte) (T, error)
+	// Unmarshal is a function that can unmarshal bytes into a request and its context.
+	Unmarshal([]byte) (context.Context, T, error)
 }
 
 var ErrQueueIsFull = queue.ErrQueueIsFull
