diff --git a/otelcol/command_components.go b/otelcol/command_components.go
index faa9adafb..92ffeef40 100644
--- a/otelcol/command_components.go
+++ b/otelcol/command_components.go
@@ -11,6 +11,7 @@ import (
 	"go.yaml.in/yaml/v3"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/extension"
@@ -118,18 +119,12 @@ func newComponentsCommand(set CollectorSettings) *cobra.Command {
 			}
 			components.BuildInfo = set.BuildInfo
 
-			for providerScheme, providerModuleModule := range set.ProviderModules {
-				components.Providers = append(components.Providers, componentWithoutStability{
-					Scheme: providerScheme,
-					Module: providerModuleModule,
-				})
-			}
-
-			for _, converterModule := range set.ConverterModules {
-				components.Converters = append(components.Converters, componentWithoutStability{
-					Module: converterModule,
-				})
-			}
+			components.Providers = append(components.Providers, sortProvidersByScheme(
+				set.ProviderModules,
+				set.ConfigProviderSettings.ResolverSettings.ProviderFactories,
+				set.ConfigProviderSettings.ResolverSettings.ProviderSettings,
+			)...)
+			components.Converters = append(components.Converters, sortConverterModules(set.ConverterModules)...)
 
 			yamlData, err := yaml.Marshal(components)
 			if err != nil {
@@ -164,8 +159,44 @@ func sortFactoriesByType[T component.Factory](factories map[component.Type]T) []
 	return sortedFactories
 }
 
-func isComponentAlias[T component.Factory](component T) bool {
-	if al, ok := any(component).(componentalias.TypeAliasHolder); ok {
+func sortProvidersByScheme(providerModules map[string]string, provFactories []confmap.ProviderFactory, set confmap.ProviderSettings) []componentWithoutStability {
+	schemes := make([]string, 0, len(provFactories))
+	for _, f := range provFactories {
+		provF := f.Create(set)
+		scheme := provF.Scheme()
+		if !isComponentAlias(f) {
+			schemes = append(schemes, scheme)
+		}
+	}
+
+	sort.Strings(schemes)
+
+	providerComponents := make([]componentWithoutStability, 0, len(providerModules))
+	for _, scheme := range schemes {
+		providerComponents = append(providerComponents, componentWithoutStability{
+			Scheme: scheme,
+			Module: providerModules[scheme],
+		})
+	}
+	return providerComponents
+}
+
+func sortConverterModules(modules []string) []componentWithoutStability {
+	sortedModulesCopy := make([]string, len(modules))
+	copy(sortedModulesCopy, modules)
+	sort.Strings(sortedModulesCopy)
+
+	sortedModules := make([]componentWithoutStability, 0, len(modules))
+	for _, mod := range sortedModulesCopy {
+		sortedModules = append(sortedModules, componentWithoutStability{
+			Module: mod,
+		})
+	}
+	return sortedModules
+}
+
+func isComponentAlias(component any) bool {
+	if al, ok := component.(componentalias.TypeAliasHolder); ok {
 		return al.DeprecatedAlias().String() != ""
 	}
 	return false
