diff --git a/config/configretry/backoff_test.go b/config/configretry/backoff_test.go
index ad9151d69..a0adb3cea 100644
--- a/config/configretry/backoff_test.go
+++ b/config/configretry/backoff_test.go
@@ -67,11 +67,17 @@ func TestInvalidMaxElapsedTime(t *testing.T) {
 	cfg.MaxElapsedTime = -1
 	assert.Error(t, cfg.Validate())
 	cfg.MaxElapsedTime = 60
+	// MaxElapsedTime is 60, InitialInterval is 5s, so it should be invalid
 	assert.Error(t, cfg.Validate())
 	cfg.InitialInterval = 0
+	// MaxElapsedTime is 60, MaxInterval is 30s, so it should be invalid
 	assert.Error(t, cfg.Validate())
 	cfg.MaxInterval = 0
 	assert.NoError(t, cfg.Validate())
+	cfg.InitialInterval = 50
+	// MaxElapsedTime is 0, so it should be valid
+	cfg.MaxElapsedTime = 0
+	assert.NoError(t, cfg.Validate())
 }
 
 func TestDisabledWithInvalidValues(t *testing.T) {
