diff --git a/translator/trace/zipkin/protospan_to_zipkinv1_test.go b/translator/trace/zipkin/protospan_to_zipkinv1_test.go
index 944462576..6af0d4489 100644
--- a/translator/trace/zipkin/protospan_to_zipkinv1_test.go
+++ b/translator/trace/zipkin/protospan_to_zipkinv1_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019 OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
