diff --git a/pdata/pmetric/generated_histogram.go b/pdata/pmetric/generated_histogram.go
index 8ca62e3be..1e111c1e3 100644
--- a/pdata/pmetric/generated_histogram.go
+++ b/pdata/pmetric/generated_histogram.go
@@ -69,7 +69,7 @@ func (ms Histogram) DataPoints() HistogramDataPointSlice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Histogram) CopyTo(dest Histogram) {
 	dest.state.AssertMutable()
-	copyOrigHistogram(dest.orig, ms.orig)
+	internal.CopyOrigHistogram(dest.orig, ms.orig)
 }
 
 // marshalJSONStream marshals all properties from the current struct to the destination stream.
@@ -100,8 +100,3 @@ func (ms Histogram) unmarshalJSONIter(iter *json.Iterator) {
 		return true
 	})
 }
-
-func copyOrigHistogram(dest, src *otlpmetrics.Histogram) {
-	dest.AggregationTemporality = src.AggregationTemporality
-	dest.DataPoints = copyOrigHistogramDataPointSlice(dest.DataPoints, src.DataPoints)
-}
