diff --git a/pdata/pmetric/generated_summary.go b/pdata/pmetric/generated_summary.go
index b19ddeddd..dccaaef62 100644
--- a/pdata/pmetric/generated_summary.go
+++ b/pdata/pmetric/generated_summary.go
@@ -58,7 +58,7 @@ func (ms Summary) DataPoints() SummaryDataPointSlice {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Summary) CopyTo(dest Summary) {
 	dest.state.AssertMutable()
-	copyOrigSummary(dest.orig, ms.orig)
+	internal.CopyOrigSummary(dest.orig, ms.orig)
 }
 
 // marshalJSONStream marshals all properties from the current struct to the destination stream.
@@ -83,7 +83,3 @@ func (ms Summary) unmarshalJSONIter(iter *json.Iterator) {
 		return true
 	})
 }
-
-func copyOrigSummary(dest, src *otlpmetrics.Summary) {
-	dest.DataPoints = copyOrigSummaryDataPointSlice(dest.DataPoints, src.DataPoints)
-}
