diff --git a/service/internal/graph/util_test.go b/service/internal/graph/util_test.go
index e6a643055..c296dd854 100644
--- a/service/internal/graph/util_test.go
+++ b/service/internal/graph/util_test.go
@@ -8,6 +8,9 @@ import (
 	"errors"
 	"hash/fnv"
 	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/connector"
@@ -17,13 +20,14 @@ import (
 	"go.opentelemetry.io/collector/consumer/xconsumer"
 	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/xexporter"
+	"go.opentelemetry.io/collector/featuregate"
+	"go.opentelemetry.io/collector/internal/telemetry"
 	"go.opentelemetry.io/collector/pipeline"
 	"go.opentelemetry.io/collector/pipeline/xpipeline"
 	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/processor/xprocessor"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/receiver/xreceiver"
-	"go.opentelemetry.io/collector/service/internal/testcomponents"
 	"go.opentelemetry.io/collector/service/pipelines"
 )
 
@@ -154,29 +158,6 @@ func expectedInstances(m pipelines.Config, pID pipeline.ID) (int, int) {
 	return r, e
 }
 
-// connector needs to be unwrapped to access component as ExampleConnector
-func unwrapExampleConnector(c *connectorNode) *testcomponents.ExampleConnector {
-	switch ct := c.Component.(type) {
-	case componentTraces: // consumes traces, emits traces
-		return ct.Component.(*testcomponents.ExampleConnector)
-	case connector.Traces: // consumes traces, emits something else
-		return ct.(*testcomponents.ExampleConnector)
-	case componentMetrics: // consumes metrics, emits metrics
-		return ct.Component.(*testcomponents.ExampleConnector)
-	case connector.Metrics: // consumes metrics, emits something else
-		return ct.(*testcomponents.ExampleConnector)
-	case componentLogs: // consumes logs, emits logs
-		return ct.Component.(*testcomponents.ExampleConnector)
-	case connector.Logs: // consumes logs, emits something else
-		return ct.(*testcomponents.ExampleConnector)
-	case componentProfiles: // consumes profiles, emits profiles
-		return ct.Component.(*testcomponents.ExampleConnector)
-	case xconnector.Profiles: // consumes profiles, emits something else
-		return ct.(*testcomponents.ExampleConnector)
-	}
-	return nil
-}
-
 func newBadReceiverFactory() receiver.Factory {
 	return receiver.NewFactory(component.MustNewType("bf"), func() component.Config {
 		return &struct{}{}
@@ -328,3 +309,11 @@ func (e errComponent) Start(context.Context, component.Host) error {
 func (e errComponent) Shutdown(context.Context) error {
 	return errors.New("my error")
 }
+
+func setObsConsumerGateForTest(t *testing.T, enabled bool) {
+	initial := telemetry.NewPipelineTelemetryGate.IsEnabled()
+	require.NoError(t, featuregate.GlobalRegistry().Set(telemetry.NewPipelineTelemetryGate.ID(), enabled))
+	t.Cleanup(func() {
+		require.NoError(t, featuregate.GlobalRegistry().Set(telemetry.NewPipelineTelemetryGate.ID(), initial))
+	})
+}
