diff --git a/config/configopaque/doc.go b/config/configopaque/doc.go
index 7153ca1eb..a7d043e84 100644
--- a/config/configopaque/doc.go
+++ b/config/configopaque/doc.go
@@ -5,6 +5,6 @@
 // Use configopaque.String on the type of sensitive fields, to mask the
 // opaque string as `[REDACTED]`.
 //
-// This ensure that no sensitive information is leaked when printing the
+// This ensures that no sensitive information is leaked when printing the
 // full Collector configurations.
 package configopaque // import "go.opentelemetry.io/collector/config/configopaque"
