diff --git a/cmd/mdatagen/internal/templates/documentation.md.tmpl b/cmd/mdatagen/internal/templates/documentation.md.tmpl
index d8b882544..461ab21da 100644
--- a/cmd/mdatagen/internal/templates/documentation.md.tmpl
+++ b/cmd/mdatagen/internal/templates/documentation.md.tmpl
@@ -233,7 +233,7 @@ The following entities are defined for this component:
 
 {{- if $entity.Identity }}
 
-**Identity Attributes:**
+**Identifying Attributes:**
 {{- range $entity.Identity }}
 - `{{ .Ref }}`
 {{- end }}
@@ -241,7 +241,7 @@ The following entities are defined for this component:
 
 {{- if $entity.Description }}
 
-**Description Attributes:**
+**Descriptive Attributes:**
 {{- range $entity.Description }}
 - `{{ .Ref }}`
 {{- end }}
