diff --git a/extension/auth/go.mod b/extension/auth/go.mod
index 2b1864eae..1ad709c1d 100644
--- a/extension/auth/go.mod
+++ b/extension/auth/go.mod
@@ -5,6 +5,7 @@ go 1.22.0
 require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/component v0.113.0
+	go.opentelemetry.io/collector/component/componenttest v0.113.0
 	go.opentelemetry.io/collector/extension v0.113.0
 	go.uber.org/goleak v1.3.0
 	google.golang.org/grpc v1.67.1
@@ -36,6 +37,8 @@ require (
 
 replace go.opentelemetry.io/collector/component => ../../component
 
+replace go.opentelemetry.io/collector/component/componenttest => ../../component/componenttest
+
 replace go.opentelemetry.io/collector/extension => ../
 
 replace go.opentelemetry.io/collector/pdata => ../../pdata
