diff --git a/cmd/builder/internal/builder/config.go b/cmd/builder/internal/builder/config.go
index 344076234..ca17b6446 100644
--- a/cmd/builder/internal/builder/config.go
+++ b/cmd/builder/internal/builder/config.go
@@ -16,7 +16,7 @@ import (
 	"go.uber.org/zap"
 )
 
-const defaultOtelColVersion = "0.89.0"
+const defaultOtelColVersion = "0.90.0"
 
 // ErrInvalidGoMod indicates an invalid gomod
 var ErrInvalidGoMod = errors.New("invalid gomod specification for module")
