diff --git a/exporter/debugexporter/internal/normal/profiles.go b/exporter/debugexporter/internal/normal/profiles.go
index 09502776a..2fee8b65d 100644
--- a/exporter/debugexporter/internal/normal/profiles.go
+++ b/exporter/debugexporter/internal/normal/profiles.go
@@ -42,7 +42,7 @@ func (normalProfilesMarshaler) MarshalProfiles(pd pprofile.Profiles) ([]byte, er
 				buffer.WriteString(profile.ProfileID().String())
 
 				buffer.WriteString(" samples=")
-				buffer.WriteString(strconv.Itoa(profile.Sample().Len()))
+				buffer.WriteString(strconv.Itoa(profile.Samples().Len()))
 
 				if profile.AttributeIndices().Len() > 0 {
 					attrs := []string{}
