diff --git a/pdata/pprofile/mapping.go b/pdata/pprofile/mapping.go
index 7989f3c14..2b857fb23 100644
--- a/pdata/pprofile/mapping.go
+++ b/pdata/pprofile/mapping.go
@@ -18,7 +18,7 @@ func (ms Mapping) Equal(val Mapping) bool {
 // dictionary to another.
 func (ms Mapping) switchDictionary(src, dst ProfilesDictionary) error {
 	if ms.FilenameStrindex() > 0 {
-		if src.StringTable().Len() < int(ms.FilenameStrindex()) {
+		if src.StringTable().Len() <= int(ms.FilenameStrindex()) {
 			return fmt.Errorf("invalid filename index %d", ms.FilenameStrindex())
 		}
 
@@ -30,7 +30,7 @@ func (ms Mapping) switchDictionary(src, dst ProfilesDictionary) error {
 	}
 
 	for i, v := range ms.AttributeIndices().All() {
-		if src.AttributeTable().Len() < int(v) {
+		if src.AttributeTable().Len() <= int(v) {
 			return fmt.Errorf("invalid attribute index %d", v)
 		}
 
