diff --git a/pdata/pprofile/stack.go b/pdata/pprofile/stack.go
index 87fd216ef..8d4f5be1d 100644
--- a/pdata/pprofile/stack.go
+++ b/pdata/pprofile/stack.go
@@ -26,7 +26,7 @@ func (ms Stack) Equal(val Stack) bool {
 // dictionary to another.
 func (ms Stack) switchDictionary(src, dst ProfilesDictionary) error {
 	for i, v := range ms.LocationIndices().All() {
-		if src.LocationTable().Len() < int(v) {
+		if src.LocationTable().Len() <= int(v) {
 			return fmt.Errorf("invalid location index %d", v)
 		}
 
