diff --git a/model/pdata/logs_alias.go b/model/pdata/logs_alias.go
index 9007003ba..ac6df12d4 100644
--- a/model/pdata/logs_alias.go
+++ b/model/pdata/logs_alias.go
@@ -119,15 +119,3 @@ const (
 	// Deprecated: [v0.49.0] Use plog.SeverityNumberFATAL4 instead.
 	SeverityNumberFATAL4 = plog.SeverityNumberFATAL4
 )
-
-// Deprecated: [v0.48.0] Use ScopeLogsSlice instead.
-type InstrumentationLibraryLogsSlice = plog.ScopeLogsSlice
-
-// Deprecated: [v0.48.0] Use NewScopeLogsSlice instead.
-var NewInstrumentationLibraryLogsSlice = plog.NewScopeLogsSlice
-
-// Deprecated: [v0.48.0] Use ScopeLogs instead.
-type InstrumentationLibraryLogs = plog.ScopeLogs
-
-// Deprecated: [v0.48.0] Use NewScopeLogs instead.
-var NewInstrumentationLibraryLogs = plog.NewScopeLogs
