diff --git a/pdata/ptrace/generated_scopespans.go b/pdata/ptrace/generated_scopespans.go
index 5ea42e985..c21edcc10 100644
--- a/pdata/ptrace/generated_scopespans.go
+++ b/pdata/ptrace/generated_scopespans.go
@@ -33,7 +33,7 @@ func newScopeSpans(orig *otlptrace.ScopeSpans, state *internal.State) ScopeSpans
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewScopeSpans() ScopeSpans {
-	return newScopeSpans(internal.NewOrigScopeSpans(), internal.NewState())
+	return newScopeSpans(internal.NewScopeSpans(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
@@ -45,13 +45,13 @@ func (ms ScopeSpans) MoveTo(dest ScopeSpans) {
 	if ms.orig == dest.orig {
 		return
 	}
-	internal.DeleteOrigScopeSpans(dest.orig, false)
+	internal.DeleteScopeSpans(dest.orig, false)
 	*dest.orig, *ms.orig = *ms.orig, *dest.orig
 }
 
 // Scope returns the scope associated with this ScopeSpans.
 func (ms ScopeSpans) Scope() pcommon.InstrumentationScope {
-	return pcommon.InstrumentationScope(internal.NewInstrumentationScope(&ms.orig.Scope, ms.state))
+	return pcommon.InstrumentationScope(internal.NewInstrumentationScopeWrapper(&ms.orig.Scope, ms.state))
 }
 
 // Spans returns the Spans associated with this ScopeSpans.
@@ -73,5 +73,5 @@ func (ms ScopeSpans) SetSchemaUrl(v string) {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms ScopeSpans) CopyTo(dest ScopeSpans) {
 	dest.state.AssertMutable()
-	internal.CopyOrigScopeSpans(dest.orig, ms.orig)
+	internal.CopyScopeSpans(dest.orig, ms.orig)
 }
