diff --git a/confmap/internal/mapstructure/encoder.go b/confmap/internal/mapstructure/encoder.go
index 638ad1ef7..ffc0bdc29 100644
--- a/confmap/internal/mapstructure/encoder.go
+++ b/confmap/internal/mapstructure/encoder.go
@@ -141,9 +141,6 @@ func (e *Encoder) encodeSlice(value reflect.Value) (any, error) {
 			Kind:   value.Kind(),
 		}
 	}
-	if value.IsNil() {
-		return []any(nil), nil
-	}
 	result := make([]any, value.Len())
 	for i := 0; i < value.Len(); i++ {
 		var err error
