diff --git a/receiver/hostmetricsreceiver/internal/scraper/cpuscraper/config.go b/receiver/hostmetricsreceiver/internal/scraper/cpuscraper/config.go
index a8b5cdafe..36062be78 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/cpuscraper/config.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/cpuscraper/config.go
@@ -19,10 +19,4 @@ import "go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
 // Config relating to CPU Metric Scraper.
 type Config struct {
 	internal.ConfigSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
-
-	// If `true`, stats will be generated for the system as a whole _as well
-	// as_ for each individual CPU/core in the system and will be distinguished
-	// by the `cpu` dimension.  If `false`, stats will only be generated for
-	// the system as a whole that will not include a `cpu` dimension.
-	ReportPerCPU bool `mapstructure:"report_per_cpu"`
 }
