diff --git a/exporter/prometheusexporter/factory.go b/exporter/prometheusexporter/factory.go
index 9bb85bc67..1eb7efd55 100644
--- a/exporter/prometheusexporter/factory.go
+++ b/exporter/prometheusexporter/factory.go
@@ -27,24 +27,24 @@ import (
 	"github.com/orijtech/prometheus-go-metrics-exporter"
 )
 
-var _ = exporter.RegisterExporterFactory(&exporterFactory{})
+var _ = exporter.RegisterFactory(&factory{})
 
 const (
 	// The value of "type" key in configuration.
 	typeStr = "prometheus"
 )
 
-// exporterFactory is the factory for Prometheus exporter.
-type exporterFactory struct {
+// factory is the factory for Prometheus exporter.
+type factory struct {
 }
 
 // Type gets the type of the Exporter config created by this factory.
-func (f *exporterFactory) Type() string {
+func (f *factory) Type() string {
 	return typeStr
 }
 
 // CreateDefaultConfig creates the default configuration for exporter.
-func (f *exporterFactory) CreateDefaultConfig() models.Exporter {
+func (f *factory) CreateDefaultConfig() models.Exporter {
 	return &ConfigV2{
 		ExporterSettings: models.ExporterSettings{
 			TypeVal: typeStr,
@@ -55,12 +55,12 @@ func (f *exporterFactory) CreateDefaultConfig() models.Exporter {
 }
 
 // CreateTraceExporter creates a trace exporter based on this config.
-func (f *exporterFactory) CreateTraceExporter(logger *zap.Logger, config models.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+func (f *factory) CreateTraceExporter(logger *zap.Logger, config models.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
 	return nil, nil, models.ErrDataTypeIsNotSupported
 }
 
 // CreateMetricsExporter creates a metrics exporter based on this config.
-func (f *exporterFactory) CreateMetricsExporter(logger *zap.Logger, cfg models.Exporter) (consumer.MetricsConsumer, exporter.StopFunc, error) {
+func (f *factory) CreateMetricsExporter(logger *zap.Logger, cfg models.Exporter) (consumer.MetricsConsumer, exporter.StopFunc, error) {
 	pcfg := cfg.(*ConfigV2)
 
 	addr := strings.TrimSpace(pcfg.Endpoint)
