diff --git a/component/component.go b/component/component.go
index 2f2da6f09..40267a416 100644
--- a/component/component.go
+++ b/component/component.go
@@ -7,13 +7,16 @@ package component // import "go.opentelemetry.io/collector/component"
 
 import (
 	"context"
-	"errors"
+
+	"go.opentelemetry.io/collector/pipeline"
 )
 
 var (
 	// ErrDataTypeIsNotSupported can be returned by receiver, exporter, processor or connector factory funcs that create the
 	// Component if the particular telemetry data type is not supported by the receiver, exporter, processor or connector factory.
-	ErrDataTypeIsNotSupported = errors.New("telemetry type is not supported")
+	//
+	// Deprecated: [v0.110.0] Use pipeline.ErrSignalNotSupported instead
+	ErrDataTypeIsNotSupported = pipeline.ErrSignalNotSupported
 )
 
 // Component is either a receiver, exporter, processor, connector, or an extension.
