diff --git a/pdata/xpdata/pref/gate.go b/pdata/xpdata/pref/gate.go
index ad2deec9e..2a53acd6e 100644
--- a/pdata/xpdata/pref/gate.go
+++ b/pdata/xpdata/pref/gate.go
@@ -4,17 +4,8 @@
 package pref // import "go.opentelemetry.io/collector/pdata/xpdata/pref"
 
 import (
-	"go.opentelemetry.io/collector/featuregate"
-	"go.opentelemetry.io/collector/pdata/internal"
+	"go.opentelemetry.io/collector/pdata/internal/metadata"
 )
 
 // UseProtoPooling temporary expose public to allow testing.
-var UseProtoPooling = internal.UseProtoPooling
-
-var EnableRefCounting = featuregate.GlobalRegistry().MustRegister(
-	"pdata.enableRefCounting",
-	featuregate.StageBeta,
-	featuregate.WithRegisterDescription("When enabled, enables using ref counting to know when pdata memory can be freed up. This featuregate is here only to protect if unexpected bugs happens because of ref counting logic."),
-	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector/issues/13631"),
-	featuregate.WithRegisterFromVersion("v0.133.0"),
-)
+var UseProtoPooling = metadata.PdataUseProtoPoolingFeatureGate
