diff --git a/pdata/xpdata/pref/metrics.go b/pdata/xpdata/pref/metrics.go
index 02610780d..8d4e56f1a 100644
--- a/pdata/xpdata/pref/metrics.go
+++ b/pdata/xpdata/pref/metrics.go
@@ -7,7 +7,9 @@ import (
 	"reflect"
 
 	"go.opentelemetry.io/collector/pdata/internal"
+	pmetadata "go.opentelemetry.io/collector/pdata/internal/metadata"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/xpdata/internal/metadata"
 )
 
 // MarkPipelineOwnedMetrics marks the pmetric.Metrics data as owned by the pipeline, returns true if the data were
@@ -17,18 +19,18 @@ func MarkPipelineOwnedMetrics(md pmetric.Metrics) bool {
 }
 
 func RefMetrics(md pmetric.Metrics) {
-	if EnableRefCounting.IsEnabled() {
+	if metadata.PdataEnableRefCountingFeatureGate.IsEnabled() {
 		internal.GetMetricsState(internal.MetricsWrapper(md)).Ref()
 	}
 }
 
 func UnrefMetrics(md pmetric.Metrics) {
-	if EnableRefCounting.IsEnabled() {
+	if metadata.PdataEnableRefCountingFeatureGate.IsEnabled() {
 		if !internal.GetMetricsState(internal.MetricsWrapper(md)).Unref() {
 			return
 		}
 		// Don't call DeleteExportLogsServiceRequest without the gate because we reset the data and that may still cause issues.
-		if internal.UseProtoPooling.IsEnabled() {
+		if pmetadata.PdataUseProtoPoolingFeatureGate.IsEnabled() {
 			internal.DeleteExportMetricsServiceRequest(internal.GetMetricsOrig(internal.MetricsWrapper(md)), true)
 		}
 	}
