diff --git a/pdata/pprofile/generated_scopeprofilesslice.go b/pdata/pprofile/generated_scopeprofilesslice.go
index 766ee3831..3f2b452ce 100644
--- a/pdata/pprofile/generated_scopeprofilesslice.go
+++ b/pdata/pprofile/generated_scopeprofilesslice.go
@@ -10,7 +10,7 @@ import (
 	"sort"
 
 	"go.opentelemetry.io/collector/pdata/internal"
-	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1experimental"
+	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
 )
 
 // ScopeProfilesSlice logically represents a slice of ScopeProfiles.
