diff --git a/internal/processor/filterexpr/matcher.go b/internal/processor/filterexpr/matcher.go
index 54a39278c..d3ae381ac 100644
--- a/internal/processor/filterexpr/matcher.go
+++ b/internal/processor/filterexpr/matcher.go
@@ -44,12 +44,8 @@ func NewMatcher(expression string) (*Matcher, error) {
 func (m *Matcher) MatchMetric(metric pdata.Metric) (bool, error) {
 	metricName := metric.Name()
 	switch metric.DataType() {
-	case pdata.MetricDataTypeIntGauge:
-		return m.matchIntGauge(metricName, metric.IntGauge())
 	case pdata.MetricDataTypeGauge:
 		return m.matchGauge(metricName, metric.Gauge())
-	case pdata.MetricDataTypeIntSum:
-		return m.matchIntSum(metricName, metric.IntSum())
 	case pdata.MetricDataTypeSum:
 		return m.matchSum(metricName, metric.Sum())
 	case pdata.MetricDataTypeHistogram:
@@ -59,20 +55,6 @@ func (m *Matcher) MatchMetric(metric pdata.Metric) (bool, error) {
 	}
 }
 
-func (m *Matcher) matchIntGauge(metricName string, gauge pdata.IntGauge) (bool, error) {
-	pts := gauge.DataPoints()
-	for i := 0; i < pts.Len(); i++ {
-		matched, err := m.matchEnv(metricName, pts.At(i).LabelsMap())
-		if err != nil {
-			return false, err
-		}
-		if matched {
-			return true, nil
-		}
-	}
-	return false, nil
-}
-
 func (m *Matcher) matchGauge(metricName string, gauge pdata.Gauge) (bool, error) {
 	pts := gauge.DataPoints()
 	for i := 0; i < pts.Len(); i++ {
@@ -101,20 +83,6 @@ func (m *Matcher) matchSum(metricName string, sum pdata.Sum) (bool, error) {
 	return false, nil
 }
 
-func (m *Matcher) matchIntSum(metricName string, sum pdata.IntSum) (bool, error) {
-	pts := sum.DataPoints()
-	for i := 0; i < pts.Len(); i++ {
-		matched, err := m.matchEnv(metricName, pts.At(i).LabelsMap())
-		if err != nil {
-			return false, err
-		}
-		if matched {
-			return true, nil
-		}
-	}
-	return false, nil
-}
-
 func (m *Matcher) matchDoubleHistogram(metricName string, histogram pdata.Histogram) (bool, error) {
 	pts := histogram.DataPoints()
 	for i := 0; i < pts.Len(); i++ {
