diff --git a/exporter/debugexporter/internal/otlptext/profiles_test.go b/exporter/debugexporter/internal/otlptext/profiles_test.go
index 0a44252b3..4334eeaaf 100644
--- a/exporter/debugexporter/internal/otlptext/profiles_test.go
+++ b/exporter/debugexporter/internal/otlptext/profiles_test.go
@@ -53,15 +53,9 @@ func extendProfiles(profiles pprofile.Profiles) pprofile.Profiles {
 		switch i % 2 {
 		case 0:
 			profile := sc.Profiles().At(i)
-			profile.Profile().LocationIndices().FromRaw([]int64{1})
-			label := profile.Profile().Sample().At(0).Label().AppendEmpty()
-			label.SetKey(1)
-			label.SetStr(2)
-			label.SetNum(3)
-			label.SetNumUnit(4)
+			profile.LocationIndices().FromRaw([]int32{1})
 
-			location := profile.Profile().Location().AppendEmpty()
-			location.SetID(2)
+			location := profile.LocationTable().AppendEmpty()
 			location.SetMappingIndex(3)
 			location.SetAddress(4)
 			line := location.Line().AppendEmpty()
@@ -69,50 +63,43 @@ func extendProfiles(profiles pprofile.Profiles) pprofile.Profiles {
 			line.SetLine(2)
 			line.SetColumn(3)
 			location.SetIsFolded(true)
-			location.SetTypeIndex(5)
-			location.Attributes().FromRaw([]uint64{6, 7})
+			location.AttributeIndices().FromRaw([]int32{6, 7})
 
-			_ = profile.Profile().AttributeTable().FromRaw(map[string]any{
-				"value": map[string]any{
-					"intValue": "42",
-				},
-			})
+			at := profile.AttributeTable()
+			a := at.AppendEmpty()
+			a.SetKey("intValue")
+			a.Value().SetInt(42)
 
-			attributeUnits := profile.Profile().AttributeUnits().AppendEmpty()
-			attributeUnits.SetAttributeKey(1)
-			attributeUnits.SetUnit(5)
+			attributeUnits := profile.AttributeUnits().AppendEmpty()
+			attributeUnits.SetAttributeKeyStrindex(1)
+			attributeUnits.SetUnitStrindex(5)
 
-			profile.Profile().StringTable().Append("foobar")
+			profile.StringTable().Append("foobar")
 		case 1:
 			profile := sc.Profiles().At(i)
-			profile.Profile().SetDropFrames(1)
-			profile.Profile().SetKeepFrames(2)
 
-			mapping := profile.Profile().Mapping().AppendEmpty()
-			mapping.SetID(1)
+			mapping := profile.MappingTable().AppendEmpty()
 			mapping.SetMemoryStart(2)
 			mapping.SetMemoryLimit(3)
 			mapping.SetFileOffset(4)
-			mapping.SetFilename(5)
-			mapping.SetBuildID(6)
-			mapping.Attributes().FromRaw([]uint64{7, 8})
+			mapping.SetFilenameStrindex(5)
+			mapping.AttributeIndices().FromRaw([]int32{7, 8})
 			mapping.SetHasFunctions(true)
 			mapping.SetHasFilenames(true)
 			mapping.SetHasLineNumbers(true)
 			mapping.SetHasInlineFrames(true)
 
-			function := profile.Profile().Function().AppendEmpty()
-			function.SetID(1)
-			function.SetName(2)
-			function.SetSystemName(3)
-			function.SetFilename(4)
+			function := profile.FunctionTable().AppendEmpty()
+			function.SetNameStrindex(2)
+			function.SetSystemNameStrindex(3)
+			function.SetFilenameStrindex(4)
 			function.SetStartLine(5)
 
-			linkTable := profile.Profile().LinkTable().AppendEmpty()
+			linkTable := profile.LinkTable().AppendEmpty()
 			linkTable.SetTraceID([16]byte{0x03, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10})
 			linkTable.SetSpanID([8]byte{0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18})
 
-			profile.Profile().Comment().FromRaw([]int64{1, 2})
+			profile.CommentStrindices().FromRaw([]int32{1, 2})
 		}
 	}
 	return profiles
