diff --git a/receiver/otlpreceiver/config.go b/receiver/otlpreceiver/config.go
index 00f5d8796..8e512f5a2 100644
--- a/receiver/otlpreceiver/config.go
+++ b/receiver/otlpreceiver/config.go
@@ -22,7 +22,7 @@ const (
 )
 
 type HTTPConfig struct {
-	*confighttp.HTTPServerConfig `mapstructure:",squash"`
+	*confighttp.ServerConfig `mapstructure:",squash"`
 
 	// The URL path to receive traces on. If omitted "/v1/traces" will be used.
 	TracesURLPath string `mapstructure:"traces_url_path,omitempty"`
