diff --git a/pdata/pcommon/spanid_test.go b/pdata/pcommon/spanid_test.go
index d4c4c9316..72b838220 100644
--- a/pdata/pcommon/spanid_test.go
+++ b/pdata/pcommon/spanid_test.go
@@ -21,7 +21,7 @@ import (
 )
 
 func TestSpanID(t *testing.T) {
-	sid := InvalidSpanID()
+	sid := EmptySpanID
 	assert.EqualValues(t, [8]byte{}, sid.Bytes())
 	assert.True(t, sid.IsEmpty())
 	assert.Equal(t, "", sid.HexString())
