diff --git a/processor/memorylimiter/factory.go b/processor/memorylimiter/factory.go
index 78a98bd73..ce1d316b9 100644
--- a/processor/memorylimiter/factory.go
+++ b/processor/memorylimiter/factory.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
