diff --git a/processor/batchprocessor/splitmetrics_test.go b/processor/batchprocessor/splitmetrics_test.go
index fd40ec8f9..06238174c 100644
--- a/processor/batchprocessor/splitmetrics_test.go
+++ b/processor/batchprocessor/splitmetrics_test.go
@@ -192,7 +192,7 @@ func TestSplitMetricsAllTypes(t *testing.T) {
 	assert.Equal(t, "test-metric-int-0-1", gaugeDouble.Name())
 	assert.Equal(t, 1, sumInt.Sum().DataPoints().Len())
 	assert.Equal(t, pmetric.AggregationTemporalityCumulative, sumInt.Sum().AggregationTemporality())
-	assert.Equal(t, true, sumInt.Sum().IsMonotonic())
+	assert.True(t, sumInt.Sum().IsMonotonic())
 	assert.Equal(t, "test-metric-int-0-2", sumInt.Name())
 
 	split = splitMetrics(splitSize, md)
@@ -202,11 +202,11 @@ func TestSplitMetricsAllTypes(t *testing.T) {
 	sumDouble := split.ResourceMetrics().At(0).ScopeMetrics().At(0).Metrics().At(1)
 	assert.Equal(t, 1, sumInt.Sum().DataPoints().Len())
 	assert.Equal(t, pmetric.AggregationTemporalityCumulative, sumInt.Sum().AggregationTemporality())
-	assert.Equal(t, true, sumInt.Sum().IsMonotonic())
+	assert.True(t, sumInt.Sum().IsMonotonic())
 	assert.Equal(t, "test-metric-int-0-2", sumInt.Name())
 	assert.Equal(t, 1, sumDouble.Sum().DataPoints().Len())
 	assert.Equal(t, pmetric.AggregationTemporalityCumulative, sumDouble.Sum().AggregationTemporality())
-	assert.Equal(t, true, sumDouble.Sum().IsMonotonic())
+	assert.True(t, sumDouble.Sum().IsMonotonic())
 	assert.Equal(t, "test-metric-int-0-3", sumDouble.Name())
 
 	split = splitMetrics(splitSize, md)
@@ -216,7 +216,7 @@ func TestSplitMetricsAllTypes(t *testing.T) {
 	histogram := split.ResourceMetrics().At(0).ScopeMetrics().At(0).Metrics().At(1)
 	assert.Equal(t, 1, sumDouble.Sum().DataPoints().Len())
 	assert.Equal(t, pmetric.AggregationTemporalityCumulative, sumDouble.Sum().AggregationTemporality())
-	assert.Equal(t, true, sumDouble.Sum().IsMonotonic())
+	assert.True(t, sumDouble.Sum().IsMonotonic())
 	assert.Equal(t, "test-metric-int-0-3", sumDouble.Name())
 	assert.Equal(t, 1, histogram.Histogram().DataPoints().Len())
 	assert.Equal(t, pmetric.AggregationTemporalityCumulative, histogram.Histogram().AggregationTemporality())
