diff --git a/config/configunmarshaler/defaultunmarshaler.go b/config/configunmarshaler/defaultunmarshaler.go
index 3d45e505a..5df58acd8 100644
--- a/config/configunmarshaler/defaultunmarshaler.go
+++ b/config/configunmarshaler/defaultunmarshaler.go
@@ -19,6 +19,8 @@ import (
 	"os"
 	"reflect"
 
+	"go.uber.org/zap/zapcore"
+
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configparser"
@@ -33,6 +35,7 @@ const (
 	_ configErrorCode = iota
 
 	errInvalidTypeAndNameKey
+	errInvalidLogsLevel
 	errUnknownType
 	errDuplicateName
 	errUnmarshalTopLevelStructureError
@@ -77,10 +80,21 @@ type configSettings struct {
 }
 
 type serviceSettings struct {
+	Telemetry  serviceTelemetrySettings    `mapstructure:"telemetry"`
 	Extensions []string                    `mapstructure:"extensions"`
 	Pipelines  map[string]pipelineSettings `mapstructure:"pipelines"`
 }
 
+type serviceTelemetrySettings struct {
+	Logs serviceTelemetryLogsSettings `mapstructure:"logs"`
+}
+
+type serviceTelemetryLogsSettings struct {
+	Level       string `mapstructure:"level"`
+	Development bool   `mapstructure:"development"`
+	Encoding    string `mapstructure:"encoding"`
+}
+
 type pipelineSettings struct {
 	Receivers  []string `mapstructure:"receivers"`
 	Processors []string `mapstructure:"processors"`
@@ -103,7 +117,19 @@ func (*defaultUnmarshaler) Unmarshal(v *configparser.ConfigMap, factories compon
 	// Unmarshal the config.
 
 	// Struct to validate top level sections.
-	var rawCfg configSettings
+	rawCfg := configSettings{
+		// Setup default telemetry values as in service/logger.go.
+		// TODO: Add a component.ServiceFactory to allow this to be defined by the Service.
+		Service: serviceSettings{
+			Telemetry: serviceTelemetrySettings{
+				Logs: serviceTelemetryLogsSettings{
+					Level:       "INFO",
+					Development: false,
+					Encoding:    "console",
+				},
+			},
+		},
+	}
 	if err := v.UnmarshalExact(&rawCfg); err != nil {
 		return nil, &configError{
 			code: errUnmarshalTopLevelStructureError,
@@ -221,6 +247,21 @@ func unmarshalExtensions(exts map[string]map[string]interface{}, factories map[c
 
 func unmarshalService(rawService serviceSettings) (config.Service, error) {
 	var ret config.Service
+
+	var lvl zapcore.Level
+	if err := lvl.UnmarshalText([]byte(rawService.Telemetry.Logs.Level)); err != nil {
+		return ret, &configError{
+			msg:  fmt.Sprintf(`service telemetry logs invalid level: %q, valid values are "DEBUG", "INFO", "WARN", "ERROR", "DPANIC", "PANIC", "FATAL"`, rawService.Telemetry.Logs.Level),
+			code: errInvalidLogsLevel,
+		}
+	}
+
+	ret.Telemetry.Logs = config.ServiceTelemetryLogs{
+		Level:       lvl,
+		Development: rawService.Telemetry.Logs.Development,
+		Encoding:    rawService.Telemetry.Logs.Encoding,
+	}
+
 	ret.Extensions = make([]config.ComponentID, 0, len(rawService.Extensions))
 	for _, extIDStr := range rawService.Extensions {
 		id, err := config.NewIDFromString(extIDStr)
