diff --git a/receiver/zipkinreceiver/factory.go b/receiver/zipkinreceiver/factory.go
index ca916c508..bb31a570e 100644
--- a/receiver/zipkinreceiver/factory.go
+++ b/receiver/zipkinreceiver/factory.go
@@ -39,7 +39,7 @@ type Factory struct {
 }
 
 // Type gets the type of the Receiver config created by this factory.
-func (f *Factory) Type() string {
+func (f *Factory) Type() configmodels.Type {
 	return typeStr
 }
 
