diff --git a/exporter/debugexporter/internal/otlptext/profiles_test.go b/exporter/debugexporter/internal/otlptext/profiles_test.go
index 74284931a..fc2d4bf3b 100644
--- a/exporter/debugexporter/internal/otlptext/profiles_test.go
+++ b/exporter/debugexporter/internal/otlptext/profiles_test.go
@@ -60,14 +60,14 @@ func extendProfiles(profiles pprofile.Profiles) pprofile.Profiles {
 	line.SetFunctionIndex(1)
 	line.SetLine(2)
 	line.SetColumn(3)
-	location.SetIsFolded(true)
 	location.AttributeIndices().FromRaw([]int32{6, 7})
+	dic.StringTable().Append("intValue")
 	at := dic.AttributeTable()
 	a := at.AppendEmpty()
-	a.SetKey("intValue")
+	a.SetKeyStrindex(1)
 	a.Value().SetInt(42)
-	attributeUnits := dic.AttributeUnits().AppendEmpty()
-	attributeUnits.SetAttributeKeyStrindex(1)
+	attributeUnits := dic.AttributeTable().AppendEmpty()
+	attributeUnits.SetKeyStrindex(2)
 	attributeUnits.SetUnitStrindex(5)
 	dic.StringTable().Append("foobar")
 	mapping := dic.MappingTable().AppendEmpty()
@@ -76,10 +76,6 @@ func extendProfiles(profiles pprofile.Profiles) pprofile.Profiles {
 	mapping.SetFileOffset(4)
 	mapping.SetFilenameStrindex(5)
 	mapping.AttributeIndices().FromRaw([]int32{7, 8})
-	mapping.SetHasFunctions(true)
-	mapping.SetHasFilenames(true)
-	mapping.SetHasLineNumbers(true)
-	mapping.SetHasInlineFrames(true)
 	function := dic.FunctionTable().AppendEmpty()
 	function.SetNameStrindex(2)
 	function.SetSystemNameStrindex(3)
@@ -96,7 +92,7 @@ func extendProfiles(profiles pprofile.Profiles) pprofile.Profiles {
 		switch i % 2 {
 		case 0:
 			profile := sc.Profiles().At(i)
-			profile.LocationIndices().FromRaw([]int32{1})
+			profile.AttributeIndices().FromRaw([]int32{1})
 		case 1:
 			profile := sc.Profiles().At(i)
 			profile.CommentStrindices().FromRaw([]int32{1, 2})
@@ -116,16 +112,14 @@ func generateProfilesWithEntityRefs() pprofile.Profiles {
 	profile := sp.Profiles().AppendEmpty()
 
 	sample := profile.Sample().AppendEmpty()
-	sample.SetLocationsStartIndex(0)
-	sample.SetLocationsLength(1)
-	sample.Value().FromRaw([]int64{100})
+	sample.Values().Append(100)
 
 	dic := pd.Dictionary()
 	dic.StringTable().Append("")
 	dic.StringTable().Append("cpu")
 	dic.StringTable().Append("nanoseconds")
 
-	sampleType := profile.SampleType().AppendEmpty()
+	sampleType := profile.SampleType()
 	sampleType.SetTypeStrindex(1)
 	sampleType.SetUnitStrindex(2)
 
