diff --git a/processor/batchprocessor/batch_processor_test.go b/processor/batchprocessor/batch_processor_test.go
index 4d201f806..9329e8af0 100644
--- a/processor/batchprocessor/batch_processor_test.go
+++ b/processor/batchprocessor/batch_processor_test.go
@@ -159,7 +159,7 @@ func TestBatchProcessorSentBySize(t *testing.T) {
 		}
 	}
 
-	viewData, err := view.RetrieveData(statBatchSendSize.Name())
+	viewData, err := view.RetrieveData("processor/batch/" + statBatchSendSize.Name())
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(viewData))
 	distData := viewData[0].Data.(*view.DistributionData)
@@ -168,7 +168,7 @@ func TestBatchProcessorSentBySize(t *testing.T) {
 	assert.Equal(t, sendBatchSize, int(distData.Min))
 	assert.Equal(t, sendBatchSize, int(distData.Max))
 
-	viewData, err = view.RetrieveData(statBatchSendSizeBytes.Name())
+	viewData, err = view.RetrieveData("processor/batch/" + statBatchSendSizeBytes.Name())
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(viewData))
 	distData = viewData[0].Data.(*view.DistributionData)
@@ -341,7 +341,7 @@ func TestBatchMetricProcessor_BatchSize(t *testing.T) {
 		}
 	}
 
-	viewData, err := view.RetrieveData(statBatchSendSize.Name())
+	viewData, err := view.RetrieveData("processor/batch/" + statBatchSendSize.Name())
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(viewData))
 	distData := viewData[0].Data.(*view.DistributionData)
@@ -350,7 +350,7 @@ func TestBatchMetricProcessor_BatchSize(t *testing.T) {
 	assert.Equal(t, cfg.SendBatchSize, uint32(distData.Min))
 	assert.Equal(t, cfg.SendBatchSize, uint32(distData.Max))
 
-	viewData, err = view.RetrieveData(statBatchSendSizeBytes.Name())
+	viewData, err = view.RetrieveData("processor/batch/" + statBatchSendSizeBytes.Name())
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(viewData))
 	distData = viewData[0].Data.(*view.DistributionData)
@@ -597,7 +597,7 @@ func TestBatchLogProcessor_BatchSize(t *testing.T) {
 		}
 	}
 
-	viewData, err := view.RetrieveData(statBatchSendSize.Name())
+	viewData, err := view.RetrieveData("processor/batch/" + statBatchSendSize.Name())
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(viewData))
 	distData := viewData[0].Data.(*view.DistributionData)
@@ -606,7 +606,7 @@ func TestBatchLogProcessor_BatchSize(t *testing.T) {
 	assert.Equal(t, cfg.SendBatchSize, uint32(distData.Min))
 	assert.Equal(t, cfg.SendBatchSize, uint32(distData.Max))
 
-	viewData, err = view.RetrieveData(statBatchSendSizeBytes.Name())
+	viewData, err = view.RetrieveData("processor/batch/" + statBatchSendSizeBytes.Name())
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(viewData))
 	distData = viewData[0].Data.(*view.DistributionData)
