diff --git a/component/componenttest/nop_telemetry.go b/component/componenttest/nop_telemetry.go
index 637801cee..171e9daa4 100644
--- a/component/componenttest/nop_telemetry.go
+++ b/component/componenttest/nop_telemetry.go
@@ -21,8 +21,7 @@ func NewNopTelemetrySettings() component.TelemetrySettings {
 		MeterProvider:  noopmetric.NewMeterProvider(),
 		MetricsLevel:   configtelemetry.LevelNone,
 		Resource:       pcommon.NewResource(),
-		ReportComponentStatus: func(*component.StatusEvent) error {
-			return nil
+		ReportStatus: func(*component.StatusEvent) {
 		},
 	}
 }
