diff --git a/exporter/exporterhelper/internal/request/request.go b/exporter/exporterhelper/internal/request/request.go
index 425bea84d..c0a44082d 100644
--- a/exporter/exporterhelper/internal/request/request.go
+++ b/exporter/exporterhelper/internal/request/request.go
@@ -25,6 +25,8 @@ type Request interface {
 	// Experimental: This API is at the early stage of development and may change without backward compatibility
 	// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
 	MergeSplit(context.Context, int, SizerType, Request) ([]Request, error)
+	// BytesSize returns the size of the request in bytes.
+	BytesSize() int
 }
 
 // ErrorHandler is an optional interface that can be implemented by Request to provide a way handle partial
