diff --git a/confmap/provider/httpprovider/go.mod b/confmap/provider/httpprovider/go.mod
index 47f56d571..a4802ecbc 100644
--- a/confmap/provider/httpprovider/go.mod
+++ b/confmap/provider/httpprovider/go.mod
@@ -4,7 +4,7 @@ go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.10.0
-	go.opentelemetry.io/collector/confmap v1.23.0
+	go.opentelemetry.io/collector/confmap v1.24.0
 	go.uber.org/goleak v1.3.0
 )
 
