diff --git a/exporter/prometheusexporter/prometheus_test.go b/exporter/prometheusexporter/prometheus_test.go
index e68081a07..e530bbcbe 100644
--- a/exporter/prometheusexporter/prometheus_test.go
+++ b/exporter/prometheusexporter/prometheus_test.go
@@ -320,8 +320,8 @@ func metricBuilder(delta int64, prefix string) pdata.Metrics {
 	d1.SetIsMonotonic(true)
 	d1.SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
 	dp1 := d1.DataPoints().AppendEmpty()
-	dp1.SetStartTimestamp(pdata.TimestampFromTime(time.Unix(1543160298+delta, 100000090)))
-	dp1.SetTimestamp(pdata.TimestampFromTime(time.Unix(1543160298+delta, 100000997)))
+	dp1.SetStartTimestamp(pdata.NewTimestampFromTime(time.Unix(1543160298+delta, 100000090)))
+	dp1.SetTimestamp(pdata.NewTimestampFromTime(time.Unix(1543160298+delta, 100000997)))
 	dp1.Attributes().UpsertString("os", "windows")
 	dp1.Attributes().UpsertString("arch", "x86")
 	dp1.SetIntVal(99 + delta)
@@ -335,8 +335,8 @@ func metricBuilder(delta int64, prefix string) pdata.Metrics {
 	d2.SetIsMonotonic(true)
 	d2.SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
 	dp2 := d2.DataPoints().AppendEmpty()
-	dp2.SetStartTimestamp(pdata.TimestampFromTime(time.Unix(1543160298, 100000090)))
-	dp2.SetTimestamp(pdata.TimestampFromTime(time.Unix(1543160298, 100000997)))
+	dp2.SetStartTimestamp(pdata.NewTimestampFromTime(time.Unix(1543160298, 100000090)))
+	dp2.SetTimestamp(pdata.NewTimestampFromTime(time.Unix(1543160298, 100000997)))
 	dp2.Attributes().UpsertString("os", "linux")
 	dp2.Attributes().UpsertString("arch", "x86")
 	dp2.SetIntVal(100 + delta)
