diff --git a/pipeline/pipelineprofiles/config.go b/pipeline/pipelineprofiles/config.go
index 6802cf15c..a81887086 100644
--- a/pipeline/pipelineprofiles/config.go
+++ b/pipeline/pipelineprofiles/config.go
@@ -1,10 +1,12 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+// Deprecated: [0.116.0] Use go.opentelemetry.io/collector/pipeline/xpipeline instead.
 package pipelineprofiles // import "go.opentelemetry.io/collector/pipeline/pipelineprofiles"
 
 import (
 	"go.opentelemetry.io/collector/pipeline/internal/globalsignal"
 )
 
+// Deprecated: [0.116.0] Use xpipeline.SignalProfiles instead.
 var SignalProfiles = globalsignal.MustNewSignal("profiles")
