diff --git a/go.mod b/go.mod
index 2e246080c..355626754 100644
--- a/go.mod
+++ b/go.mod
@@ -19,7 +19,7 @@ require (
 	go.opentelemetry.io/collector/pdata v1.0.0-rcv0018
 	go.opentelemetry.io/collector/processor v0.89.0
 	go.opentelemetry.io/collector/receiver v0.89.0
-	go.opentelemetry.io/contrib/config v0.1.0
+	go.opentelemetry.io/contrib/config v0.1.1
 	go.opentelemetry.io/otel v1.21.0
 	go.opentelemetry.io/otel/exporters/prometheus v0.44.0
 	go.opentelemetry.io/otel/sdk v1.21.0
