diff --git a/pdata/pcommon/generated_stringslice.go b/pdata/pcommon/generated_stringslice.go
index b16564570..57680fd18 100644
--- a/pdata/pcommon/generated_stringslice.go
+++ b/pdata/pcommon/generated_stringslice.go
@@ -106,6 +106,10 @@ func (ms StringSlice) Append(elms ...string) {
 func (ms StringSlice) MoveTo(dest StringSlice) {
 	ms.getState().AssertMutable()
 	dest.getState().AssertMutable()
+	// If they point to the same data, they are the same, nothing to do.
+	if ms.getOrig() == dest.getOrig() {
+		return
+	}
 	*dest.getOrig() = *ms.getOrig()
 	*ms.getOrig() = nil
 }
