diff --git a/pdata/internal/generated_wrapper_keyvalueslice.go b/pdata/internal/generated_wrapper_keyvalueslice.go
index 0fb24b36d..12ec71be5 100644
--- a/pdata/internal/generated_wrapper_keyvalueslice.go
+++ b/pdata/internal/generated_wrapper_keyvalueslice.go
@@ -20,7 +20,7 @@ func CopyOrigKeyValueSlice(dest, src []otlpcommon.KeyValue) []otlpcommon.KeyValu
 		// Cleanup the rest of the elements so GC can free the memory.
 		// This can happen when len(src) < len(dest) < cap(dest).
 		for i := len(src); i < len(dest); i++ {
-			dest[i] = otlpcommon.KeyValue{}
+			dest[i].Reset()
 		}
 	}
 	for i := range src {
@@ -31,13 +31,8 @@ func CopyOrigKeyValueSlice(dest, src []otlpcommon.KeyValue) []otlpcommon.KeyValu
 
 func GenerateOrigTestKeyValueSlice() []otlpcommon.KeyValue {
 	orig := make([]otlpcommon.KeyValue, 5)
-	orig[0] = otlpcommon.KeyValue{}
-	orig[1] = otlpcommon.KeyValue{}
 	FillOrigTestKeyValue(&orig[1])
-	orig[2] = otlpcommon.KeyValue{}
-	orig[3] = otlpcommon.KeyValue{}
-	FillOrigTestKeyValue(&orig[1])
-	orig[4] = otlpcommon.KeyValue{}
+	FillOrigTestKeyValue(&orig[3])
 	return orig
 }
 
