diff --git a/service/internal/testcomponents/example_connector_test.go b/service/internal/testcomponents/example_connector_test.go
index 1e5dfd518..2877dc00b 100644
--- a/service/internal/testcomponents/example_connector_test.go
+++ b/service/internal/testcomponents/example_connector_test.go
@@ -26,11 +26,11 @@ import (
 func TestExampleConnector(t *testing.T) {
 	conn := &ExampleConnector{}
 	host := componenttest.NewNopHost()
-	assert.False(t, conn.Started)
+	assert.False(t, conn.Started())
 	assert.NoError(t, conn.Start(context.Background(), host))
-	assert.True(t, conn.Started)
+	assert.True(t, conn.Started())
 
-	assert.False(t, conn.Stopped)
+	assert.False(t, conn.Stopped())
 	assert.NoError(t, conn.Shutdown(context.Background()))
-	assert.True(t, conn.Stopped)
+	assert.True(t, conn.Stopped())
 }
