diff --git a/service/internal/fanoutconsumer/logs.go b/service/internal/fanoutconsumer/logs.go
index 8673bacb2..e72c19109 100644
--- a/service/internal/fanoutconsumer/logs.go
+++ b/service/internal/fanoutconsumer/logs.go
@@ -27,8 +27,8 @@ import (
 
 // NewLogs wraps multiple log consumers in a single one.
 // It fanouts the incoming data to all the consumers, and does smart routing:
-//  * Clones only to the consumer that needs to mutate the data.
-//  * If all consumers needs to mutate the data one will get the original data.
+//   - Clones only to the consumer that needs to mutate the data.
+//   - If all consumers needs to mutate the data one will get the original data.
 func NewLogs(lcs []consumer.Logs) consumer.Logs {
 	if len(lcs) == 1 {
 		// Don't wrap if no need to do it.
