diff --git a/otelcol/command.go b/otelcol/command.go
index c56a90791..207aa1088 100644
--- a/otelcol/command.go
+++ b/otelcol/command.go
@@ -30,7 +30,7 @@ func NewCommand(set CollectorSettings) *cobra.Command {
 		Version:      set.BuildInfo.Version,
 		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := featuregate.GetRegistry().Apply(getFeatureGatesFlag(flagSet)); err != nil {
+			if err := featuregate.GlobalRegistry().Apply(getFeatureGatesFlag(flagSet)); err != nil {
 				return err
 			}
 			if set.ConfigProvider == nil {
