diff --git a/exporter/exporterhelper/logs.go b/exporter/exporterhelper/logs.go
index 3837d9a90..ace067082 100644
--- a/exporter/exporterhelper/logs.go
+++ b/exporter/exporterhelper/logs.go
@@ -23,20 +23,19 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
-	"go.opentelemetry.io/collector/model/otlp"
-	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/plog"
 )
 
-var logsMarshaler = otlp.NewProtobufLogsMarshaler()
-var logsUnmarshaler = otlp.NewProtobufLogsUnmarshaler()
+var logsMarshaler = plog.NewProtoMarshaler()
+var logsUnmarshaler = plog.NewProtoUnmarshaler()
 
 type logsRequest struct {
 	baseRequest
-	ld     pdata.Logs
+	ld     plog.Logs
 	pusher consumer.ConsumeLogsFunc
 }
 
-func newLogsRequest(ctx context.Context, ld pdata.Logs, pusher consumer.ConsumeLogsFunc) request {
+func newLogsRequest(ctx context.Context, ld plog.Logs, pusher consumer.ConsumeLogsFunc) request {
 	return &logsRequest{
 		baseRequest: baseRequest{ctx: ctx},
 		ld:          ld,
@@ -107,7 +106,7 @@ func NewLogsExporter(
 		}
 	})
 
-	lc, err := consumer.NewLogs(func(ctx context.Context, ld pdata.Logs) error {
+	lc, err := consumer.NewLogs(func(ctx context.Context, ld plog.Logs) error {
 		req := newLogsRequest(ctx, ld, pusher)
 		err := be.sender.send(req)
 		if errors.Is(err, errSendingQueueIsFull) {
