diff --git a/semconv/semconv_test.go b/semconv/semconv_test.go
index 2d528bd23..bb36e335e 100644
--- a/semconv/semconv_test.go
+++ b/semconv/semconv_test.go
@@ -15,7 +15,7 @@ import (
 
 func TestAllSemConvFilesAreCrated(t *testing.T) {
 	// Files that have to be present in each semconv package
-	var expectedFiles = []string{"generated_resource.go", "generated_trace.go", "schema.go", "nonstandard.go"}
+	expectedFiles := []string{"generated_resource.go", "generated_trace.go", "schema.go", "nonstandard.go"}
 
 	files, err := os.ReadDir(".")
 	require.NoError(t, err)
