diff --git a/processor/addattributesprocessor/addattributesprocessor.go b/processor/addattributesprocessor/addattributesprocessor.go
index 83a86d818..dab11cda9 100644
--- a/processor/addattributesprocessor/addattributesprocessor.go
+++ b/processor/addattributesprocessor/addattributesprocessor.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -21,9 +21,9 @@ import (
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 	"github.com/spf13/cast"
 
-	"github.com/census-instrumentation/opencensus-service/consumer"
-	"github.com/census-instrumentation/opencensus-service/data"
-	"github.com/census-instrumentation/opencensus-service/processor"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/data"
+	"github.com/open-telemetry/opentelemetry-service/processor"
 )
 
 var (
