diff --git a/service/internal/testcomponents/example_receiver.go b/service/internal/testcomponents/example_receiver.go
index 9ed3a24ae..87d65f039 100644
--- a/service/internal/testcomponents/example_receiver.go
+++ b/service/internal/testcomponents/example_receiver.go
@@ -10,19 +10,19 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/consumer/xconsumer"
 	"go.opentelemetry.io/collector/receiver"
-	"go.opentelemetry.io/collector/receiver/receiverprofiles"
+	"go.opentelemetry.io/collector/receiver/xreceiver"
 )
 
 var receiverType = component.MustNewType("examplereceiver")
 
 // ExampleReceiverFactory is factory for ExampleReceiver.
-var ExampleReceiverFactory = receiverprofiles.NewFactory(
+var ExampleReceiverFactory = xreceiver.NewFactory(
 	receiverType,
 	createReceiverDefaultConfig,
-	receiverprofiles.WithTraces(createTraces, component.StabilityLevelDevelopment),
-	receiverprofiles.WithMetrics(createMetrics, component.StabilityLevelDevelopment),
-	receiverprofiles.WithLogs(createLogs, component.StabilityLevelDevelopment),
-	receiverprofiles.WithProfiles(createProfiles, component.StabilityLevelDevelopment),
+	xreceiver.WithTraces(createTraces, component.StabilityLevelDevelopment),
+	xreceiver.WithMetrics(createMetrics, component.StabilityLevelDevelopment),
+	xreceiver.WithLogs(createLogs, component.StabilityLevelDevelopment),
+	xreceiver.WithProfiles(createProfiles, component.StabilityLevelDevelopment),
 )
 
 func createReceiverDefaultConfig() component.Config {
@@ -71,7 +71,7 @@ func createProfiles(
 	_ receiver.Settings,
 	cfg component.Config,
 	nextConsumer xconsumer.Profiles,
-) (receiverprofiles.Profiles, error) {
+) (xreceiver.Profiles, error) {
 	tr := createReceiver(cfg)
 	tr.ConsumeProfilesFunc = nextConsumer.ConsumeProfiles
 	return tr, nil
