diff --git a/cmd/pdatagen/internal/files.go b/cmd/pdatagen/internal/files.go
index 4971e1187..81485b060 100644
--- a/cmd/pdatagen/internal/files.go
+++ b/cmd/pdatagen/internal/files.go
@@ -53,7 +53,7 @@ type File struct {
 	structs []baseStruct
 }
 
-// GenerateFile generates the file string.
+// GenerateFile generates the configured data structures for this File.
 func (f *File) GenerateFile() string {
 	var sb strings.Builder
 
@@ -75,6 +75,7 @@ func (f *File) GenerateFile() string {
 	return sb.String()
 }
 
+// GenerateTestFile generates tests for the configured data structures for this File.
 func (f *File) GenerateTestFile() string {
 	var sb strings.Builder
 
