diff --git a/exporter/go.mod b/exporter/go.mod
index 62dbfb8ff..1361f18e8 100644
--- a/exporter/go.mod
+++ b/exporter/go.mod
@@ -93,3 +93,5 @@ replace go.opentelemetry.io/collector/connector => ../connector
 replace go.opentelemetry.io/collector/config/configcompression => ../config/configcompression
 
 replace go.opentelemetry.io/collector/config/confignet => ../config/confignet
+
+replace go.opentelemetry.io/collector/config/configopaque => ../config/configopaque
