diff --git a/config/configtls/go.mod b/config/configtls/go.mod
index 6113e3f2f..a20d4e213 100644
--- a/config/configtls/go.mod
+++ b/config/configtls/go.mod
@@ -5,7 +5,7 @@ go 1.20
 require (
 	github.com/fsnotify/fsnotify v1.7.0
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector/config/configopaque v0.90.0
+	go.opentelemetry.io/collector/config/configopaque v0.90.1
 )
 
 require (
