diff --git a/cmd/mdatagen/internal/metadata.go b/cmd/mdatagen/internal/metadata.go
index 4a8ae82b3..f177de89f 100644
--- a/cmd/mdatagen/internal/metadata.go
+++ b/cmd/mdatagen/internal/metadata.go
@@ -26,6 +26,8 @@ type Metadata struct {
 	Parent string `mapstructure:"parent"`
 	// Status information for the component.
 	Status *Status `mapstructure:"status"`
+	// Spatial Re-aggregation featuregate.
+	ReaggregationEnabled bool `mapstructure:"reaggregation_enabled"`
 	// The name of the package that will be generated.
 	GeneratedPackageName string `mapstructure:"generated_package_name"`
 	// Telemetry information for the component.
@@ -412,6 +414,17 @@ func (a Attribute) IsConditional() bool {
 	return a.RequirementLevel == AttributeRequirementLevelConditionallyRequired
 }
 
+// IsRequired returns true if the attribute is required.
+func (a Attribute) IsRequired() bool {
+	return a.RequirementLevel == AttributeRequirementLevelRequired
+}
+
+// IsNotOptIn returns true if the attribute is any requirement_level above
+// opt_in
+func (a Attribute) IsNotOptIn() bool {
+	return a.RequirementLevel != AttributeRequirementLevelOptIn
+}
+
 // UnmarshalText implements the encoding.TextUnmarshaler interface.
 func (rl *AttributeRequirementLevel) UnmarshalText(text []byte) error {
 	switch string(text) {
@@ -476,6 +489,31 @@ func (a Attribute) TestValue() string {
 	return ""
 }
 
+func (a Attribute) TestValueTwo() string {
+	if a.Enum != nil {
+		return fmt.Sprintf(`%q`, a.Enum[1])
+	}
+	switch a.Type.ValueType {
+	case pcommon.ValueTypeEmpty:
+		return ""
+	case pcommon.ValueTypeStr:
+		return fmt.Sprintf(`"%s-val-2"`, a.FullName)
+	case pcommon.ValueTypeInt:
+		return strconv.Itoa(len(a.FullName) + 1)
+	case pcommon.ValueTypeDouble:
+		return fmt.Sprintf("%f", 1.1+float64(len(a.FullName)))
+	case pcommon.ValueTypeBool:
+		return strconv.FormatBool(len(a.FullName)%2 == 1)
+	case pcommon.ValueTypeMap:
+		return fmt.Sprintf(`map[string]any{"key3": "%s-val3", "key4": "%s-val4"}`, a.FullName, a.FullName)
+	case pcommon.ValueTypeSlice:
+		return fmt.Sprintf(`[]any{"%s-item3", "%s-item4"}`, a.FullName, a.FullName)
+	case pcommon.ValueTypeBytes:
+		return fmt.Sprintf(`[]byte("%s-val-2")`, a.FullName)
+	}
+	return ""
+}
+
 type Signal struct {
 	// Enabled defines whether the signal is enabled by default.
 	Enabled bool `mapstructure:"enabled"`
