diff --git a/pdata/pprofile/generated_attributetableslice_test.go b/pdata/pprofile/generated_attributetableslice_test.go
index 9a1ba1ea8..d1660b01c 100644
--- a/pdata/pprofile/generated_attributetableslice_test.go
+++ b/pdata/pprofile/generated_attributetableslice_test.go
@@ -54,7 +54,7 @@ func TestAttributeTableSlice_CopyTo(t *testing.T) {
 	NewAttributeTableSlice().CopyTo(dest)
 	assert.Equal(t, NewAttributeTableSlice(), dest)
 
-	// Test CopyTo larger slice and EnsureCapacity
+	// Test CopyTo larger slice
 	src := generateTestAttributeTableSlice()
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestAttributeTableSlice(), dest)
@@ -62,12 +62,12 @@ func TestAttributeTableSlice_CopyTo(t *testing.T) {
 	// Test CopyTo same size slice
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestAttributeTableSlice(), dest)
-}
 
-func TestAttributeTableSlice_CopyToAndEnsureCapacity(t *testing.T) {
-	dest := NewAttributeTableSlice()
-	src := generateTestAttributeTableSlice()
-	dest.EnsureCapacity(src.Len())
+	// Test CopyTo smaller size slice
+	NewAttributeTableSlice().CopyTo(dest)
+	assert.Equal(t, 0, dest.Len())
+
+	// Test CopyTo larger slice with enough capacity
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestAttributeTableSlice(), dest)
 }
