diff --git a/pdata/ptrace/json_test.go b/pdata/ptrace/json_test.go
index e16406f04..77b620f89 100644
--- a/pdata/ptrace/json_test.go
+++ b/pdata/ptrace/json_test.go
@@ -159,23 +159,6 @@ func BenchmarkJSONUnmarshal(b *testing.B) {
 	})
 }
 
-func TestReadInt64(t *testing.T) {
-	var data = `{"intAsNumber":1,"intAsString":"1"}`
-	iter := jsoniter.ConfigFastest.BorrowIterator([]byte(data))
-	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
-		switch f {
-		case "intAsNumber":
-			v := readInt64(iter)
-			assert.Equal(t, int64(1), v)
-		case "intAsString":
-			v := readInt64(iter)
-			assert.Equal(t, int64(1), v)
-		}
-		return true
-	})
-	assert.NoError(t, iter.Error)
-}
-
 func TestReadTraceDataUnknownField(t *testing.T) {
 	jsonStr := `{"extra":""}`
 	iter := jsoniter.ConfigFastest.BorrowIterator([]byte(jsonStr))
