diff --git a/exporter/exporter.go b/exporter/exporter.go
index 7267c45ad..22f817f39 100644
--- a/exporter/exporter.go
+++ b/exporter/exporter.go
@@ -15,6 +15,9 @@
 package exporter // import "go.opentelemetry.io/collector/exporter"
 
 import (
+	"context"
+	"fmt"
+
 	"go.opentelemetry.io/collector/component"
 )
 
@@ -37,30 +40,133 @@ type CreateSettings = component.ExporterCreateSettings //nolint:staticcheck
 type Factory = component.ExporterFactory //nolint:staticcheck
 
 // FactoryOption apply changes to Factory.
-type FactoryOption = component.ExporterFactoryOption //nolint:staticcheck
+type FactoryOption interface {
+	// applyExporterFactoryOption applies the option.
+	applyExporterFactoryOption(o *factory)
+}
+
+var _ FactoryOption = (*factoryOptionFunc)(nil)
+
+// factoryOptionFunc is an ExporterFactoryOption created through a function.
+type factoryOptionFunc func(*factory)
+
+func (f factoryOptionFunc) applyExporterFactoryOption(o *factory) {
+	f(o)
+}
 
 // CreateTracesFunc is the equivalent of Factory.CreateTraces.
-type CreateTracesFunc = component.CreateTracesExporterFunc //nolint:staticcheck
+type CreateTracesFunc func(context.Context, CreateSettings, component.Config) (Traces, error)
+
+// CreateTracesExporter implements ExporterFactory.CreateTracesExporter().
+func (f CreateTracesFunc) CreateTracesExporter(ctx context.Context, set CreateSettings, cfg component.Config) (Traces, error) {
+	if f == nil {
+		return nil, component.ErrDataTypeIsNotSupported
+	}
+	return f(ctx, set, cfg)
+}
 
 // CreateMetricsFunc is the equivalent of Factory.CreateMetrics.
-type CreateMetricsFunc = component.CreateMetricsExporterFunc //nolint:staticcheck
+type CreateMetricsFunc func(context.Context, CreateSettings, component.Config) (Metrics, error)
+
+// CreateMetricsExporter implements ExporterFactory.CreateMetricsExporter().
+func (f CreateMetricsFunc) CreateMetricsExporter(ctx context.Context, set CreateSettings, cfg component.Config) (Metrics, error) {
+	if f == nil {
+		return nil, component.ErrDataTypeIsNotSupported
+	}
+	return f(ctx, set, cfg)
+}
 
 // CreateLogsFunc is the equivalent of Factory.CreateLogs.
-type CreateLogsFunc = component.CreateLogsExporterFunc //nolint:staticcheck
+type CreateLogsFunc func(context.Context, CreateSettings, component.Config) (Logs, error)
+
+// CreateLogsExporter implements Factory.CreateLogsExporter().
+func (f CreateLogsFunc) CreateLogsExporter(ctx context.Context, set CreateSettings, cfg component.Config) (Logs, error) {
+	if f == nil {
+		return nil, component.ErrDataTypeIsNotSupported
+	}
+	return f(ctx, set, cfg)
+}
+
+type factory struct {
+	component.Factory
+	cfgType component.Type
+	component.CreateDefaultConfigFunc
+	CreateTracesFunc
+	tracesStabilityLevel component.StabilityLevel
+	CreateMetricsFunc
+	metricsStabilityLevel component.StabilityLevel
+	CreateLogsFunc
+	logsStabilityLevel component.StabilityLevel
+}
+
+func (f *factory) Type() component.Type {
+	return f.cfgType
+}
+
+// CreateDefaultConfig creates the default configuration for the Component.
+//
+// TODO: Remove this when we remove the private func from component.Factory and add it to every specialized Factory.
+func (f *factory) CreateDefaultConfig() component.Config {
+	return f.CreateDefaultConfigFunc()
+}
+
+func (f *factory) TracesExporterStability() component.StabilityLevel {
+	return f.tracesStabilityLevel
+}
+
+func (f *factory) MetricsExporterStability() component.StabilityLevel {
+	return f.metricsStabilityLevel
+}
+
+func (f *factory) LogsExporterStability() component.StabilityLevel {
+	return f.logsStabilityLevel
+}
 
 // WithTraces overrides the default "error not supported" implementation for CreateTracesExporter and the default "undefined" stability level.
-var WithTraces = component.WithTracesExporter //nolint:staticcheck
+func WithTraces(createTraces CreateTracesFunc, sl component.StabilityLevel) FactoryOption {
+	return factoryOptionFunc(func(o *factory) {
+		o.tracesStabilityLevel = sl
+		o.CreateTracesFunc = createTraces
+	})
+}
 
 // WithMetrics overrides the default "error not supported" implementation for CreateMetricsExporter and the default "undefined" stability level.
-var WithMetrics = component.WithMetricsExporter //nolint:staticcheck
+func WithMetrics(createMetrics CreateMetricsFunc, sl component.StabilityLevel) FactoryOption {
+	return factoryOptionFunc(func(o *factory) {
+		o.metricsStabilityLevel = sl
+		o.CreateMetricsFunc = createMetrics
+	})
+}
 
 // WithLogs overrides the default "error not supported" implementation for CreateLogsExporter and the default "undefined" stability level.
-var WithLogs = component.WithLogsExporter //nolint:staticcheck
+func WithLogs(createLogs CreateLogsFunc, sl component.StabilityLevel) FactoryOption {
+	return factoryOptionFunc(func(o *factory) {
+		o.logsStabilityLevel = sl
+		o.CreateLogsFunc = createLogs
+	})
+}
 
 // NewFactory returns a Factory.
-var NewFactory = component.NewExporterFactory //nolint:staticcheck
-
-// MakeFactoryMap takes a list of exporter factories and returns a map
-// with factory type as keys. It returns a non-nil error when more than one factories
-// have the same type.
-var MakeFactoryMap = component.MakeExporterFactoryMap //nolint:staticcheck
+func NewFactory(cfgType component.Type, createDefaultConfig component.CreateDefaultConfigFunc, options ...FactoryOption) Factory {
+	f := &factory{
+		cfgType:                 cfgType,
+		CreateDefaultConfigFunc: createDefaultConfig,
+	}
+	for _, opt := range options {
+		opt.applyExporterFactoryOption(f)
+	}
+	return f
+}
+
+// MakeFactoryMap takes a list of factories and returns a map with Factory type as keys.
+// It returns a non-nil error when there are factories with duplicate type.
+func MakeFactoryMap(factories ...Factory) (map[component.Type]Factory, error) {
+	fMap := map[component.Type]Factory{}
+	for _, f := range factories {
+		if _, ok := fMap[f.Type()]; ok {
+			return fMap, fmt.Errorf("duplicate exporter factory %q", f.Type())
+		}
+		fMap[f.Type()] = f
+	}
+	return fMap, nil
+}
