diff --git a/confmap/provider/envprovider/go.mod b/confmap/provider/envprovider/go.mod
index e3c320813..fcf8c09ee 100644
--- a/confmap/provider/envprovider/go.mod
+++ b/confmap/provider/envprovider/go.mod
@@ -4,7 +4,7 @@ go 1.22.0
 
 require (
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/collector/confmap v1.19.0
+	go.opentelemetry.io/collector/confmap v1.20.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
 )
