diff --git a/config/configgrpc/configgrpc.go b/config/configgrpc/configgrpc.go
index 12a425271..4ed941f8c 100644
--- a/config/configgrpc/configgrpc.go
+++ b/config/configgrpc/configgrpc.go
@@ -116,14 +116,22 @@ type GRPCServerSettings struct {
 
 	// Configures the protocol to use TLS.
 	// The default value is nil, which will cause the protocol to not use TLS.
-	TLSCredentials *configtls.TLSServerSetting `mapstructure:"tls_credentials, omitempty"`
+	TLSCredentials *configtls.TLSServerSetting `mapstructure:"tls_credentials,omitempty"`
 
 	// MaxRecvMsgSizeMiB sets the maximum size (in MiB) of messages accepted by the server.
-	MaxRecvMsgSizeMiB uint64 `mapstructure:"max_recv_msg_size_mib,omitempty"`
+	MaxRecvMsgSizeMiB uint64 `mapstructure:"max_recv_msg_size_mib"`
 
 	// MaxConcurrentStreams sets the limit on the number of concurrent streams to each ServerTransport.
 	// It has effect only for streaming RPCs.
-	MaxConcurrentStreams uint32 `mapstructure:"max_concurrent_streams,omitempty"`
+	MaxConcurrentStreams uint32 `mapstructure:"max_concurrent_streams"`
+
+	// The WriteBufferSize for client gRPC. See grpc.ReadBufferSize
+	// (https://godoc.org/google.golang.org/grpc#ReadBufferSize).
+	ReadBufferSize int `mapstructure:"read_buffer_size"`
+
+	// The WriteBufferSize for client gRPC. See grpc.WriteBufferSize
+	// (https://godoc.org/google.golang.org/grpc#WriteBufferSize).
+	WriteBufferSize int `mapstructure:"write_buffer_size"`
 
 	// Keepalive anchor for all the settings related to keepalive.
 	Keepalive *KeepaliveServerConfig `mapstructure:"keepalive,omitempty"`
@@ -191,6 +199,14 @@ func (gss *GRPCServerSettings) ToServerOption() ([]grpc.ServerOption, error) {
 		opts = append(opts, grpc.MaxConcurrentStreams(gss.MaxConcurrentStreams))
 	}
 
+	if gss.ReadBufferSize > 0 {
+		opts = append(opts, grpc.ReadBufferSize(gss.ReadBufferSize))
+	}
+
+	if gss.WriteBufferSize > 0 {
+		opts = append(opts, grpc.WriteBufferSize(gss.WriteBufferSize))
+	}
+
 	// The default values referenced in the GRPC docs are set within the server, so this code doesn't need
 	// to apply them over zero/nil values before passing these as grpc.ServerOptions.
 	// The following shows the server code for applying default grpc.ServerOptions.
