diff --git a/translator/internaldata/oc_testdata_test.go b/translator/internaldata/oc_testdata_test.go
index 2f528ac25..9575cfd6e 100644
--- a/translator/internaldata/oc_testdata_test.go
+++ b/translator/internaldata/oc_testdata_test.go
@@ -492,16 +492,16 @@ func generateResourceWithOcNodeAndResource() pdata.Resource {
 	resource := pdata.NewResource()
 	resource.InitEmpty()
 	resource.Attributes().InitFromMap(map[string]pdata.AttributeValue{
-		conventions.OCAttributeProcessStartTime: pdata.NewAttributeValueString("2020-02-11T20:26:00Z"),
-		conventions.AttributeHostHostname:       pdata.NewAttributeValueString("host1"),
-		conventions.OCAttributeProcessID:        pdata.NewAttributeValueInt(123),
-		conventions.AttributeLibraryVersion:     pdata.NewAttributeValueString("v2.0.1"),
-		conventions.OCAttributeExporterVersion:  pdata.NewAttributeValueString("v1.2.0"),
-		conventions.AttributeLibraryLanguage:    pdata.NewAttributeValueString("CPP"),
-		conventions.OCAttributeResourceType:     pdata.NewAttributeValueString("good-resource"),
-		"node-str-attr":                         pdata.NewAttributeValueString("node-str-attr-val"),
-		"resource-str-attr":                     pdata.NewAttributeValueString("resource-str-attr-val"),
-		"resource-int-attr":                     pdata.NewAttributeValueInt(123),
+		conventions.OCAttributeProcessStartTime:   pdata.NewAttributeValueString("2020-02-11T20:26:00Z"),
+		conventions.AttributeHostHostname:         pdata.NewAttributeValueString("host1"),
+		conventions.OCAttributeProcessID:          pdata.NewAttributeValueInt(123),
+		conventions.AttributeTelemetrySDKVersion:  pdata.NewAttributeValueString("v2.0.1"),
+		conventions.OCAttributeExporterVersion:    pdata.NewAttributeValueString("v1.2.0"),
+		conventions.AttributeTelemetrySDKLanguage: pdata.NewAttributeValueString("CPP"),
+		conventions.OCAttributeResourceType:       pdata.NewAttributeValueString("good-resource"),
+		"node-str-attr":                           pdata.NewAttributeValueString("node-str-attr-val"),
+		"resource-str-attr":                       pdata.NewAttributeValueString("resource-str-attr-val"),
+		"resource-int-attr":                       pdata.NewAttributeValueInt(123),
 	})
 	return resource
 }
