diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 2a9c38a19..2c465b549 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -39,7 +39,7 @@ require (
 	github.com/hashicorp/go-version v1.7.0 // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
-	github.com/prometheus/client_golang v1.20.2 // indirect
+	github.com/prometheus/client_golang v1.20.3 // indirect
 	github.com/prometheus/client_model v0.6.1 // indirect
 	github.com/prometheus/common v0.59.1 // indirect
 	github.com/prometheus/procfs v0.15.1 // indirect
