diff --git a/pdata/pmetric/generated_metrics_test.go b/pdata/pmetric/generated_metrics_test.go
index 3e4a67f8f..8a27d5d2a 100644
--- a/pdata/pmetric/generated_metrics_test.go
+++ b/pdata/pmetric/generated_metrics_test.go
@@ -1058,7 +1058,10 @@ func TestHistogramDataPoint_Sum(t *testing.T) {
 	ms := NewHistogramDataPoint()
 	assert.Equal(t, float64(0.0), ms.Sum())
 	ms.SetSum(float64(17.13))
+	assert.True(t, ms.HasSum())
 	assert.Equal(t, float64(17.13), ms.Sum())
+	ms.RemoveSum()
+	assert.False(t, ms.HasSum())
 }
 
 func TestHistogramDataPoint_BucketCounts(t *testing.T) {
@@ -1094,14 +1097,20 @@ func TestHistogramDataPoint_Min(t *testing.T) {
 	ms := NewHistogramDataPoint()
 	assert.Equal(t, float64(0.0), ms.Min())
 	ms.SetMin(float64(9.23))
+	assert.True(t, ms.HasMin())
 	assert.Equal(t, float64(9.23), ms.Min())
+	ms.RemoveMin()
+	assert.False(t, ms.HasMin())
 }
 
 func TestHistogramDataPoint_Max(t *testing.T) {
 	ms := NewHistogramDataPoint()
 	assert.Equal(t, float64(0.0), ms.Max())
 	ms.SetMax(float64(182.55))
+	assert.True(t, ms.HasMax())
 	assert.Equal(t, float64(182.55), ms.Max())
+	ms.RemoveMax()
+	assert.False(t, ms.HasMax())
 }
 
 func TestExponentialHistogramDataPointSlice(t *testing.T) {
@@ -1266,7 +1275,10 @@ func TestExponentialHistogramDataPoint_Sum(t *testing.T) {
 	ms := NewExponentialHistogramDataPoint()
 	assert.Equal(t, float64(0.0), ms.Sum())
 	ms.SetSum(float64(17.13))
+	assert.True(t, ms.HasSum())
 	assert.Equal(t, float64(17.13), ms.Sum())
+	ms.RemoveSum()
+	assert.False(t, ms.HasSum())
 }
 
 func TestExponentialHistogramDataPoint_Scale(t *testing.T) {
@@ -1314,14 +1326,20 @@ func TestExponentialHistogramDataPoint_Min(t *testing.T) {
 	ms := NewExponentialHistogramDataPoint()
 	assert.Equal(t, float64(0.0), ms.Min())
 	ms.SetMin(float64(9.23))
+	assert.True(t, ms.HasMin())
 	assert.Equal(t, float64(9.23), ms.Min())
+	ms.RemoveMin()
+	assert.False(t, ms.HasMin())
 }
 
 func TestExponentialHistogramDataPoint_Max(t *testing.T) {
 	ms := NewExponentialHistogramDataPoint()
 	assert.Equal(t, float64(0.0), ms.Max())
 	ms.SetMax(float64(182.55))
+	assert.True(t, ms.HasMax())
 	assert.Equal(t, float64(182.55), ms.Max())
+	ms.RemoveMax()
+	assert.False(t, ms.HasMax())
 }
 
 func TestExponentialHistogramDataPointBuckets_MoveTo(t *testing.T) {
