diff --git a/consumer/consumertest/sink.go b/consumer/consumertest/sink.go
index 954403fae..06ae77fdb 100644
--- a/consumer/consumertest/sink.go
+++ b/consumer/consumertest/sink.go
@@ -25,6 +25,7 @@ import (
 // TracesSink is a consumer.Traces that acts like a sink that
 // stores all traces and allows querying them for testing.
 type TracesSink struct {
+	nonMutatingConsumer
 	mu         sync.Mutex
 	traces     []pdata.Traces
 	spansCount int
@@ -72,6 +73,7 @@ func (ste *TracesSink) Reset() {
 // MetricsSink is a consumer.Metrics that acts like a sink that
 // stores all metrics and allows querying them for testing.
 type MetricsSink struct {
+	nonMutatingConsumer
 	mu           sync.Mutex
 	metrics      []pdata.Metrics
 	metricsCount int
@@ -119,6 +121,7 @@ func (sme *MetricsSink) Reset() {
 // LogsSink is a consumer.Logs that acts like a sink that
 // stores all logs and allows querying them for testing.
 type LogsSink struct {
+	nonMutatingConsumer
 	mu              sync.Mutex
 	logs            []pdata.Logs
 	logRecordsCount int
