diff --git a/receiver/zipkinreceiver/config.go b/receiver/zipkinreceiver/config.go
index e38afd1fd..dfa70803d 100644
--- a/receiver/zipkinreceiver/config.go
+++ b/receiver/zipkinreceiver/config.go
@@ -14,7 +14,7 @@
 
 package zipkinreceiver
 
-import "github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+import "go.opentelemetry.io/collector/config/configmodels"
 
 // Config defines configuration for Zipkin receiver.
 type Config struct {
