diff --git a/confmap/provider/httpprovider/provider.go b/confmap/provider/httpprovider/provider.go
index 966faa510..4c762adc2 100644
--- a/confmap/provider/httpprovider/provider.go
+++ b/confmap/provider/httpprovider/provider.go
@@ -16,13 +16,3 @@ import (
 func NewWithSettings(set confmap.ProviderSettings) confmap.Provider {
 	return configurablehttpprovider.New(configurablehttpprovider.HTTPScheme, set)
 }
-
-// New returns a new confmap.Provider that reads the configuration from a http server.
-//
-// This Provider supports "http" scheme.
-//
-// One example for HTTP URI is: http://localhost:3333/getConfig
-// Deprecated: [v0.94.0] Use NewWithSettings instead.
-func New() confmap.Provider {
-	return NewWithSettings(confmap.ProviderSettings{})
-}
