diff --git a/pdata/pcommon/generated_resource.go b/pdata/pcommon/generated_resource.go
index 66cd5156d..f734d213e 100644
--- a/pdata/pcommon/generated_resource.go
+++ b/pdata/pcommon/generated_resource.go
@@ -18,10 +18,10 @@ import (
 //
 // Must use NewResource function to create new instances.
 // Important: zero-initialized instance is not valid for use.
-type Resource internal.Resource
+type Resource internal.ResourceWrapper
 
 func newResource(orig *otlpresource.Resource, state *internal.State) Resource {
-	return Resource(internal.NewResource(orig, state))
+	return Resource(internal.NewResourceWrapper(orig, state))
 }
 
 // NewResource creates a new empty Resource.
@@ -29,7 +29,7 @@ func newResource(orig *otlpresource.Resource, state *internal.State) Resource {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewResource() Resource {
-	return newResource(internal.NewOrigResource(), internal.NewState())
+	return newResource(internal.NewResource(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
@@ -41,13 +41,13 @@ func (ms Resource) MoveTo(dest Resource) {
 	if ms.getOrig() == dest.getOrig() {
 		return
 	}
-	internal.DeleteOrigResource(dest.getOrig(), false)
+	internal.DeleteResource(dest.getOrig(), false)
 	*dest.getOrig(), *ms.getOrig() = *ms.getOrig(), *dest.getOrig()
 }
 
 // Attributes returns the Attributes associated with this Resource.
 func (ms Resource) Attributes() Map {
-	return Map(internal.NewMap(&ms.getOrig().Attributes, ms.getState()))
+	return Map(internal.NewMapWrapper(&ms.getOrig().Attributes, ms.getState()))
 }
 
 // DroppedAttributesCount returns the droppedattributescount associated with this Resource.
@@ -64,13 +64,13 @@ func (ms Resource) SetDroppedAttributesCount(v uint32) {
 // CopyTo copies all properties from the current struct overriding the destination.
 func (ms Resource) CopyTo(dest Resource) {
 	dest.getState().AssertMutable()
-	internal.CopyOrigResource(dest.getOrig(), ms.getOrig())
+	internal.CopyResource(dest.getOrig(), ms.getOrig())
 }
 
 func (ms Resource) getOrig() *otlpresource.Resource {
-	return internal.GetOrigResource(internal.Resource(ms))
+	return internal.GetResourceOrig(internal.ResourceWrapper(ms))
 }
 
 func (ms Resource) getState() *internal.State {
-	return internal.GetResourceState(internal.Resource(ms))
+	return internal.GetResourceState(internal.ResourceWrapper(ms))
 }
