diff --git a/pdata/pcommon/generated_resource_test.go b/pdata/pcommon/generated_resource_test.go
index 7754bc46c..ac0309363 100644
--- a/pdata/pcommon/generated_resource_test.go
+++ b/pdata/pcommon/generated_resource_test.go
@@ -25,8 +25,8 @@ func TestResource_MoveTo(t *testing.T) {
 	assert.Equal(t, generateTestResource(), dest)
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { ms.MoveTo(newResource(internal.NewOrigResource(), sharedState)) })
-	assert.Panics(t, func() { newResource(internal.NewOrigResource(), sharedState).MoveTo(dest) })
+	assert.Panics(t, func() { ms.MoveTo(newResource(internal.NewResource(), sharedState)) })
+	assert.Panics(t, func() { newResource(internal.NewResource(), sharedState).MoveTo(dest) })
 }
 
 func TestResource_CopyTo(t *testing.T) {
@@ -39,14 +39,14 @@ func TestResource_CopyTo(t *testing.T) {
 	assert.Equal(t, orig, ms)
 	sharedState := internal.NewState()
 	sharedState.MarkReadOnly()
-	assert.Panics(t, func() { ms.CopyTo(newResource(internal.NewOrigResource(), sharedState)) })
+	assert.Panics(t, func() { ms.CopyTo(newResource(internal.NewResource(), sharedState)) })
 }
 
 func TestResource_Attributes(t *testing.T) {
 	ms := NewResource()
 	assert.Equal(t, NewMap(), ms.Attributes())
-	ms.getOrig().Attributes = internal.GenerateOrigTestKeyValueSlice()
-	assert.Equal(t, Map(internal.GenerateTestMap()), ms.Attributes())
+	ms.getOrig().Attributes = internal.GenTestKeyValueSlice()
+	assert.Equal(t, Map(internal.GenTestMapWrapper()), ms.Attributes())
 }
 
 func TestResource_DroppedAttributesCount(t *testing.T) {
@@ -60,6 +60,6 @@ func TestResource_DroppedAttributesCount(t *testing.T) {
 }
 
 func generateTestResource() Resource {
-	ms := newResource(internal.GenTestOrigResource(), internal.NewState())
+	ms := newResource(internal.GenTestResource(), internal.NewState())
 	return ms
 }
