diff --git a/receiver/zipkinreceiver/proto_parse_test.go b/receiver/zipkinreceiver/proto_parse_test.go
index 74205a08b..31c1906af 100644
--- a/receiver/zipkinreceiver/proto_parse_test.go
+++ b/receiver/zipkinreceiver/proto_parse_test.go
@@ -117,8 +117,8 @@ func TestConvertSpansToTraceSpans_protobuf(t *testing.T) {
 	span0.SetSpanID(pdata.NewSpanID([8]byte{0xF7, 0xF6, 0xF5, 0xF4, 0xF3, 0xF2, 0xF1, 0xF0}))
 	span0.SetParentSpanID(pdata.NewSpanID([8]byte{0xF7, 0xF6, 0xF5, 0xF4, 0xF3, 0xF2, 0xF1, 0xF0}))
 	span0.SetName("ProtoSpan1")
-	span0.SetStartTimestamp(pdata.TimestampFromTime(now))
-	span0.SetEndTimestamp(pdata.TimestampFromTime(now.Add(12 * time.Second)))
+	span0.SetStartTimestamp(pdata.NewTimestampFromTime(now))
+	span0.SetEndTimestamp(pdata.NewTimestampFromTime(now.Add(12 * time.Second)))
 	span0.Attributes().UpsertString(conventions.AttributeNetHostIP, "192.168.0.1")
 	span0.Attributes().UpsertInt(conventions.AttributeNetHostPort, 8009)
 	span0.Attributes().UpsertString(conventions.AttributeNetPeerName, "memcached")
@@ -133,8 +133,8 @@ func TestConvertSpansToTraceSpans_protobuf(t *testing.T) {
 	span1.SetSpanID(pdata.NewSpanID([8]byte{0x67, 0x66, 0x65, 0x64, 0x63, 0x62, 0x61, 0x60}))
 	span1.SetParentSpanID(pdata.NewSpanID([8]byte{0x17, 0x16, 0x15, 0x14, 0x13, 0x12, 0x11, 0x10}))
 	span1.SetName("CacheWarmUp")
-	span1.SetStartTimestamp(pdata.TimestampFromTime(now.Add(-10 * time.Hour)))
-	span1.SetEndTimestamp(pdata.TimestampFromTime(now.Add(-10 * time.Hour).Add(7 * time.Second)))
+	span1.SetStartTimestamp(pdata.NewTimestampFromTime(now.Add(-10 * time.Hour)))
+	span1.SetEndTimestamp(pdata.NewTimestampFromTime(now.Add(-10 * time.Hour).Add(7 * time.Second)))
 	span1.Attributes().UpsertString(conventions.AttributeNetHostIP, "10.0.0.13")
 	span1.Attributes().UpsertInt(conventions.AttributeNetHostPort, 8009)
 	span1.Attributes().UpsertString(conventions.AttributeNetPeerName, "redis")
@@ -143,9 +143,9 @@ func TestConvertSpansToTraceSpans_protobuf(t *testing.T) {
 	span1.Attributes().UpsertString(tracetranslator.TagSpanKind, string(tracetranslator.OpenTracingSpanKindProducer))
 	span1.Events().EnsureCapacity(2)
 	span1.Events().AppendEmpty().SetName("DB reset")
-	span1.Events().At(0).SetTimestamp(pdata.TimestampFromTime(now.Add(-10 * time.Hour)))
+	span1.Events().At(0).SetTimestamp(pdata.NewTimestampFromTime(now.Add(-10 * time.Hour)))
 	span1.Events().AppendEmpty().SetName("GC Cycle 39")
-	span1.Events().At(1).SetTimestamp(pdata.TimestampFromTime(now.Add(-10 * time.Hour)))
+	span1.Events().At(1).SetTimestamp(pdata.NewTimestampFromTime(now.Add(-10 * time.Hour)))
 
 	assert.Equal(t, want.SpanCount(), reqs.SpanCount())
 	assert.Equal(t, want.ResourceSpans().Len(), reqs.ResourceSpans().Len())
