diff --git a/receiver/kafkareceiver/kafka_receiver_test.go b/receiver/kafkareceiver/kafka_receiver_test.go
index 8f946904c..a1238c841 100644
--- a/receiver/kafkareceiver/kafka_receiver_test.go
+++ b/receiver/kafkareceiver/kafka_receiver_test.go
@@ -37,13 +37,24 @@ import (
 
 func TestNewReceiver_version_err(t *testing.T) {
 	c := Config{
+		Encoding:        defaultEncoding,
 		ProtocolVersion: "none",
 	}
-	r, err := newReceiver(c, component.ReceiverCreateParams{}, exportertest.NewNopTraceExporter())
+	r, err := newReceiver(c, component.ReceiverCreateParams{}, defaultUnmarshallers(), exportertest.NewNopTraceExporter())
 	assert.Error(t, err)
 	assert.Nil(t, r)
 }
 
+func TestNewReceiver_encoding_err(t *testing.T) {
+	c := Config{
+		Encoding: "foo",
+	}
+	r, err := newReceiver(c, component.ReceiverCreateParams{}, defaultUnmarshallers(), exportertest.NewNopTraceExporter())
+	require.Error(t, err)
+	assert.Nil(t, r)
+	assert.EqualError(t, err, errUnrecognizedEncoding.Error())
+}
+
 func TestReceiverStart(t *testing.T) {
 	testClient := testConsumerGroup{once: &sync.Once{}}
 	c := kafkaConsumer{
@@ -100,7 +111,7 @@ func TestConsumerGroupHandler(t *testing.T) {
 	defer view.Unregister(views...)
 
 	c := consumerGroupHandler{
-		unmarshaller: &protoUnmarshaller{},
+		unmarshaller: &otlpProtoUnmarshaller{},
 		logger:       zap.NewNop(),
 		ready:        make(chan bool),
 		nextConsumer: exportertest.NewNopTraceExporter(),
@@ -144,7 +155,7 @@ func TestConsumerGroupHandler(t *testing.T) {
 
 func TestConsumerGroupHandler_error_unmarshall(t *testing.T) {
 	c := consumerGroupHandler{
-		unmarshaller: &protoUnmarshaller{},
+		unmarshaller: &otlpProtoUnmarshaller{},
 		logger:       zap.NewNop(),
 		ready:        make(chan bool),
 		nextConsumer: exportertest.NewNopTraceExporter(),
@@ -170,7 +181,7 @@ func TestConsumerGroupHandler_error_nextConsumer(t *testing.T) {
 	consumerError := fmt.Errorf("failed to consumer")
 	nextConsumer.SetConsumeTraceError(consumerError)
 	c := consumerGroupHandler{
-		unmarshaller: &protoUnmarshaller{},
+		unmarshaller: &otlpProtoUnmarshaller{},
 		logger:       zap.NewNop(),
 		ready:        make(chan bool),
 		nextConsumer: nextConsumer,
