diff --git a/service/parserprovider/inmemory.go b/service/parserprovider/inmemory.go
index 9d4fcc62c..910f923fc 100644
--- a/service/parserprovider/inmemory.go
+++ b/service/parserprovider/inmemory.go
@@ -31,7 +31,7 @@ func NewInMemory(buf io.Reader) ParserProvider {
 }
 
 func (inp *inMemoryProvider) Get(context.Context) (*configparser.ConfigMap, error) {
-	return configparser.NewParserFromBuffer(inp.buf)
+	return configparser.NewConfigMapFromBuffer(inp.buf)
 }
 
 func (inp *inMemoryProvider) Close(context.Context) error {
