diff --git a/pdata/pcommon/slice_test.go b/pdata/pcommon/slice_test.go
index 95cccc98f..28c6441cb 100644
--- a/pdata/pcommon/slice_test.go
+++ b/pdata/pcommon/slice_test.go
@@ -69,6 +69,14 @@ func TestSlice_CopyTo(t *testing.T) {
 	assert.Equal(t, Slice(internal.GenerateTestSlice()), dest)
 }
 
+func TestSlice_CopyToAndEnsureCapacity(t *testing.T) {
+	dest := NewSlice()
+	src := Slice(internal.GenerateTestSlice())
+	dest.EnsureCapacity(src.Len())
+	src.CopyTo(dest)
+	assert.Equal(t, Slice(internal.GenerateTestSlice()), dest)
+}
+
 func TestSlice_EnsureCapacity(t *testing.T) {
 	es := Slice(internal.GenerateTestSlice())
 	// Test ensure smaller capacity.
