diff --git a/pdata/pmetric/generated_resourcemetricsslice_test.go b/pdata/pmetric/generated_resourcemetricsslice_test.go
index 8d0000126..34dcc8e84 100644
--- a/pdata/pmetric/generated_resourcemetricsslice_test.go
+++ b/pdata/pmetric/generated_resourcemetricsslice_test.go
@@ -55,7 +55,7 @@ func TestResourceMetricsSlice_CopyTo(t *testing.T) {
 	NewResourceMetricsSlice().CopyTo(dest)
 	assert.Equal(t, NewResourceMetricsSlice(), dest)
 
-	// Test CopyTo larger slice and EnsureCapacity
+	// Test CopyTo larger slice
 	src := generateTestResourceMetricsSlice()
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestResourceMetricsSlice(), dest)
@@ -63,12 +63,12 @@ func TestResourceMetricsSlice_CopyTo(t *testing.T) {
 	// Test CopyTo same size slice
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestResourceMetricsSlice(), dest)
-}
 
-func TestResourceMetricsSlice_CopyToAndEnsureCapacity(t *testing.T) {
-	dest := NewResourceMetricsSlice()
-	src := generateTestResourceMetricsSlice()
-	dest.EnsureCapacity(src.Len())
+	// Test CopyTo smaller size slice
+	NewResourceMetricsSlice().CopyTo(dest)
+	assert.Equal(t, 0, dest.Len())
+
+	// Test CopyTo larger slice with enough capacity
 	src.CopyTo(dest)
 	assert.Equal(t, generateTestResourceMetricsSlice(), dest)
 }
