diff --git a/service/config_provider_test.go b/service/config_provider_test.go
index 93cc5a2ca..500771161 100644
--- a/service/config_provider_test.go
+++ b/service/config_provider_test.go
@@ -28,9 +28,9 @@ import (
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configtest"
-	"go.opentelemetry.io/collector/config/configunmarshaler"
 	"go.opentelemetry.io/collector/config/experimental/configsource"
 	"go.opentelemetry.io/collector/config/mapprovider/filemapprovider"
+	"go.opentelemetry.io/collector/internal/configunmarshaler"
 )
 
 type mockProvider struct {
