diff --git a/processor/queuedprocessor/queued_processor.go b/processor/queuedprocessor/queued_processor.go
index 44dcec0e9..42bd2380c 100644
--- a/processor/queuedprocessor/queued_processor.go
+++ b/processor/queuedprocessor/queued_processor.go
@@ -135,6 +135,12 @@ func (sp *queuedSpanProcessor) GetCapabilities() processor.Capabilities {
 	return processor.Capabilities{MutatesConsumedData: false}
 }
 
+// Shutdown is invoked during service shutdown.
+func (sp *queuedSpanProcessor) Shutdown() error {
+	// TODO: flush the queue.
+	return nil
+}
+
 func (sp *queuedSpanProcessor) processItemFromQueue(item *queueItem) {
 	startTime := time.Now()
 	err := sp.sender.ConsumeTraceData(item.ctx, item.td)
