diff --git a/confmap/confmaptest/configtest_test.go b/confmap/confmaptest/configtest_test.go
index ae1e592de..8175ffaa1 100644
--- a/confmap/confmaptest/configtest_test.go
+++ b/confmap/confmaptest/configtest_test.go
@@ -15,20 +15,56 @@
 package confmaptest
 
 import (
+	"context"
 	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/confmap"
 )
 
-func TestLoadConfigMap_FileNotFound(t *testing.T) {
+func TestLoadConfFileNotFound(t *testing.T) {
 	_, err := LoadConf("file/not/found")
 	assert.Error(t, err)
 }
 
-func TestLoadConfigMap(t *testing.T) {
+func TestLoadConfInvalidYAML(t *testing.T) {
+	_, err := LoadConf(filepath.Join("testdata", "invalid.yaml"))
+	require.Error(t, err)
+}
+
+func TestLoadConf(t *testing.T) {
 	cfg, err := LoadConf(filepath.Join("testdata", "simple.yaml"))
 	require.NoError(t, err)
 	assert.Equal(t, map[string]interface{}{"floating": 3.14}, cfg.ToStringMap())
 }
+
+func TestValidateProviderScheme(t *testing.T) {
+	assert.NoError(t, ValidateProviderScheme(&schemeProvider{scheme: "file"}))
+	assert.NoError(t, ValidateProviderScheme(&schemeProvider{scheme: "s3"}))
+	assert.NoError(t, ValidateProviderScheme(&schemeProvider{scheme: "a.l-l+"}))
+	// Too short.
+	assert.Error(t, ValidateProviderScheme(&schemeProvider{scheme: "a"}))
+	// Invalid first character.
+	assert.Error(t, ValidateProviderScheme(&schemeProvider{scheme: "3s"}))
+	// Invalid underscore character.
+	assert.Error(t, ValidateProviderScheme(&schemeProvider{scheme: "all_"}))
+}
+
+type schemeProvider struct {
+	scheme string
+}
+
+func (s schemeProvider) Retrieve(context.Context, string, confmap.WatcherFunc) (confmap.Retrieved, error) {
+	return confmap.Retrieved{}, nil
+}
+
+func (s schemeProvider) Scheme() string {
+	return s.scheme
+}
+
+func (s schemeProvider) Shutdown(ctx context.Context) error {
+	return nil
+}
