diff --git a/config/internal/warning.go b/config/internal/warning.go
index f9e32fc1c..26eca49a7 100644
--- a/config/internal/warning.go
+++ b/config/internal/warning.go
@@ -10,7 +10,7 @@ import (
 
 	"go.uber.org/zap"
 
-	"go.opentelemetry.io/collector/internal/localhostgate"
+	"go.opentelemetry.io/collector/internal/globalgates"
 )
 
 func shouldWarn(endpoint string) bool {
@@ -38,14 +38,14 @@ func shouldWarn(endpoint string) bool {
 
 // WarnOnUnspecifiedHost emits a warning if an endpoint has an unspecified host.
 func WarnOnUnspecifiedHost(logger *zap.Logger, endpoint string) {
-	if !localhostgate.UseLocalHostAsDefaultHostfeatureGate.IsEnabled() && shouldWarn(endpoint) {
+	if !globalgates.UseLocalHostAsDefaultHostfeatureGate.IsEnabled() && shouldWarn(endpoint) {
 		logger.Warn(
 			"Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks. Enable the feature gate to change the default and remove this warning.",
 			zap.String(
 				"documentation",
 				"https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks",
 			),
-			zap.String("feature gate ID", localhostgate.UseLocalHostAsDefaultHostID),
+			zap.String("feature gate ID", globalgates.UseLocalHostAsDefaultHostID),
 		)
 	}
 }
