diff --git a/obsreport/obsreport_processor.go b/obsreport/obsreport_processor.go
index f4a385811..2725e83d7 100644
--- a/obsreport/obsreport_processor.go
+++ b/obsreport/obsreport_processor.go
@@ -117,23 +117,23 @@ func ProcessorMetricViews(configType string, legacyViews []*view.View) []*view.V
 var gProcessorObsReport = &ProcessorObsReport{level: configtelemetry.LevelNone}
 
 type ProcessorObsReport struct {
-	level configtelemetry.Level
+	level    configtelemetry.Level
+	mutators []tag.Mutator
 }
 
-// WrapContext adds the keys used when recording observability metrics to
-// the given context returning the newly created context. This context should
-// be used in related calls to the obsreport functions so metrics are properly
-// recorded.
-func (*ProcessorObsReport) WrapContext(ctx context.Context, processor string) context.Context {
-	ctx, _ = tag.New(ctx, tag.Upsert(tagKeyProcessor, processor, tag.WithTTL(tag.TTLNoPropagation)))
-	return ctx
+func NewProcessorObsReport(level configtelemetry.Level, processorName string) *ProcessorObsReport {
+	return &ProcessorObsReport{
+		level:    level,
+		mutators: []tag.Mutator{tag.Upsert(tagKeyProcessor, processorName, tag.WithTTL(tag.TTLNoPropagation))},
+	}
 }
 
 // TracesAccepted reports that the trace data was accepted.
 func (por *ProcessorObsReport) TracesAccepted(ctx context.Context, numSpans int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedSpans.M(int64(numSpans)),
 			mProcessorRefusedSpans.M(0),
 			mProcessorDroppedSpans.M(0),
@@ -144,8 +144,9 @@ func (por *ProcessorObsReport) TracesAccepted(ctx context.Context, numSpans int)
 // TracesRefused reports that the trace data was refused.
 func (por *ProcessorObsReport) TracesRefused(ctx context.Context, numSpans int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedSpans.M(0),
 			mProcessorRefusedSpans.M(int64(numSpans)),
 			mProcessorDroppedSpans.M(0),
@@ -156,8 +157,9 @@ func (por *ProcessorObsReport) TracesRefused(ctx context.Context, numSpans int)
 // TracesDropped reports that the trace data was dropped.
 func (por *ProcessorObsReport) TracesDropped(ctx context.Context, numSpans int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedSpans.M(0),
 			mProcessorRefusedSpans.M(0),
 			mProcessorDroppedSpans.M(int64(numSpans)),
@@ -168,8 +170,9 @@ func (por *ProcessorObsReport) TracesDropped(ctx context.Context, numSpans int)
 // MetricsAccepted reports that the metrics were accepted.
 func (por *ProcessorObsReport) MetricsAccepted(ctx context.Context, numPoints int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedMetricPoints.M(int64(numPoints)),
 			mProcessorRefusedMetricPoints.M(0),
 			mProcessorDroppedMetricPoints.M(0),
@@ -180,8 +183,9 @@ func (por *ProcessorObsReport) MetricsAccepted(ctx context.Context, numPoints in
 // MetricsRefused reports that the metrics were refused.
 func (por *ProcessorObsReport) MetricsRefused(ctx context.Context, numPoints int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedMetricPoints.M(0),
 			mProcessorRefusedMetricPoints.M(int64(numPoints)),
 			mProcessorDroppedMetricPoints.M(0),
@@ -192,8 +196,9 @@ func (por *ProcessorObsReport) MetricsRefused(ctx context.Context, numPoints int
 // MetricsDropped reports that the metrics were dropped.
 func (por *ProcessorObsReport) MetricsDropped(ctx context.Context, numPoints int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedMetricPoints.M(0),
 			mProcessorRefusedMetricPoints.M(0),
 			mProcessorDroppedMetricPoints.M(int64(numPoints)),
@@ -204,8 +209,9 @@ func (por *ProcessorObsReport) MetricsDropped(ctx context.Context, numPoints int
 // LogsAccepted reports that the logs were accepted.
 func (por *ProcessorObsReport) LogsAccepted(ctx context.Context, numRecords int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedLogRecords.M(int64(numRecords)),
 			mProcessorRefusedLogRecords.M(0),
 			mProcessorDroppedLogRecords.M(0),
@@ -216,8 +222,9 @@ func (por *ProcessorObsReport) LogsAccepted(ctx context.Context, numRecords int)
 // LogsRefused reports that the logs were refused.
 func (por *ProcessorObsReport) LogsRefused(ctx context.Context, numRecords int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedLogRecords.M(0),
 			mProcessorRefusedLogRecords.M(int64(numRecords)),
 			mProcessorDroppedMetricPoints.M(0),
@@ -228,64 +235,12 @@ func (por *ProcessorObsReport) LogsRefused(ctx context.Context, numRecords int)
 // LogsDropped reports that the logs were dropped.
 func (por *ProcessorObsReport) LogsDropped(ctx context.Context, numRecords int) {
 	if por.level != configtelemetry.LevelNone {
-		stats.Record(
+		stats.RecordWithTags(
 			ctx,
+			por.mutators,
 			mProcessorAcceptedLogRecords.M(0),
 			mProcessorRefusedLogRecords.M(0),
 			mProcessorDroppedLogRecords.M(int64(numRecords)),
 		)
 	}
 }
-
-// ProcessorContext adds the keys used when recording observability metrics to
-// the given context returning the newly created context. This context should
-// be used in related calls to the obsreport functions so metrics are properly
-// recorded.
-func ProcessorContext(ctx context.Context, processor string) context.Context {
-	return gProcessorObsReport.WrapContext(ctx, processor)
-}
-
-// ProcessorTraceDataAccepted reports that the trace data was accepted.
-func ProcessorTraceDataAccepted(ctx context.Context, numSpans int) {
-	gProcessorObsReport.TracesAccepted(ctx, numSpans)
-}
-
-// ProcessorTraceDataRefused reports that the trace data was refused.
-func ProcessorTraceDataRefused(ctx context.Context, numSpans int) {
-	gProcessorObsReport.TracesRefused(ctx, numSpans)
-}
-
-// ProcessorTraceDataDropped reports that the trace data was dropped.
-func ProcessorTraceDataDropped(ctx context.Context, numSpans int) {
-	gProcessorObsReport.TracesDropped(ctx, numSpans)
-}
-
-// ProcessorMetricsDataAccepted reports that the metrics were accepted.
-func ProcessorMetricsDataAccepted(ctx context.Context, numPoints int) {
-	gProcessorObsReport.MetricsAccepted(ctx, numPoints)
-}
-
-// ProcessorMetricsDataRefused reports that the metrics were refused.
-func ProcessorMetricsDataRefused(ctx context.Context, numPoints int) {
-	gProcessorObsReport.MetricsRefused(ctx, numPoints)
-}
-
-// ProcessorMetricsDataDropped reports that the metrics were dropped.
-func ProcessorMetricsDataDropped(ctx context.Context, numPoints int) {
-	gProcessorObsReport.MetricsDropped(ctx, numPoints)
-}
-
-// ProcessorLogRecordsAccepted reports that the metrics were accepted.
-func ProcessorLogRecordsAccepted(ctx context.Context, numRecords int) {
-	gProcessorObsReport.LogsAccepted(ctx, numRecords)
-}
-
-// ProcessorLogRecordsRefused reports that the metrics were refused.
-func ProcessorLogRecordsRefused(ctx context.Context, numRecords int) {
-	gProcessorObsReport.LogsRefused(ctx, numRecords)
-}
-
-// ProcessorLogRecordsDropped reports that the metrics were dropped.
-func ProcessorLogRecordsDropped(ctx context.Context, numRecords int) {
-	gProcessorObsReport.LogsDropped(ctx, numRecords)
-}
