diff --git a/consumer/go.mod b/consumer/go.mod
index 0acb7760d..ee08c43f8 100644
--- a/consumer/go.mod
+++ b/consumer/go.mod
@@ -34,23 +34,13 @@ replace go.opentelemetry.io/collector/component => ../component
 
 replace go.opentelemetry.io/collector/confmap => ../confmap
 
-replace go.opentelemetry.io/collector/exporter => ../exporter
-
-replace go.opentelemetry.io/collector/extension => ../extension
-
 replace go.opentelemetry.io/collector/featuregate => ../featuregate
 
 replace go.opentelemetry.io/collector/pdata => ../pdata
 
-replace go.opentelemetry.io/collector/processor => ../processor
-
-replace go.opentelemetry.io/collector/receiver => ../receiver
-
 retract (
 	v0.76.0 // Depends on retracted pdata v1.0.0-rc10 module, use v0.76.1
 	v0.69.0 // Release failed, use v0.69.1
 )
 
 replace go.opentelemetry.io/collector/config/configtelemetry => ../config/configtelemetry
-
-replace go.opentelemetry.io/collector/config/configretry => ../config/configretry
