diff --git a/testbed/tests/trace_test.go b/testbed/tests/trace_test.go
index c45550b20..15a4873ca 100644
--- a/testbed/tests/trace_test.go
+++ b/testbed/tests/trace_test.go
@@ -261,7 +261,7 @@ func verifySingleSpan(
 	startCounter := tc.MockBackend.DataItemsReceived()
 
 	// Send one span.
-	td := pdata.NewTraceData()
+	td := pdata.NewTraces()
 	td.ResourceSpans().Resize(1)
 	td.ResourceSpans().At(0).Resource().InitEmpty()
 	td.ResourceSpans().At(0).Resource().Attributes().InitFromMap(map[string]pdata.AttributeValue{
