diff --git a/translator/trace/zipkin/traces_to_zipkinv2_test.go b/translator/trace/zipkin/traces_to_zipkinv2_test.go
index 5003f131c..7a23c6efb 100644
--- a/translator/trace/zipkin/traces_to_zipkinv2_test.go
+++ b/translator/trace/zipkin/traces_to_zipkinv2_test.go
@@ -77,6 +77,12 @@ func TestInternalTracesToZipkinSpans(t *testing.T) {
 			zs:   make([]*zipkinmodel.SpanModel, 0),
 			err:  errors.New("TraceID is nil"),
 		},
+		{
+			name: "oneSpanOneNil",
+			td:   generateTraceOneSpanOneNilWithTraceID(),
+			zs:   []*zipkinmodel.SpanModel{zipkinOneSpan()},
+			err:  nil,
+		},
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
@@ -111,3 +117,49 @@ func TestInternalTracesToZipkinSpansAndBack(t *testing.T) {
 		assert.Equal(t, td.SpanCount(), tdFromZS.SpanCount())
 	}
 }
+
+func generateTraceOneSpanOneNilWithTraceID() pdata.Traces {
+	td := testdata.GenerateTraceDataOneSpanOneNil()
+	span := td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0)
+	span.SetTraceID(pdata.NewTraceID([]byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+		0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10}))
+	span.SetSpanID(pdata.NewSpanID([]byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08}))
+	return td
+}
+
+func zipkinOneSpan() *zipkinmodel.SpanModel {
+	trueBool := true
+	return &zipkinmodel.SpanModel{
+		SpanContext: zipkinmodel.SpanContext{
+			TraceID:  zipkinmodel.TraceID{High: 72623859790382856, Low: 651345242494996240},
+			ID:       72623859790382856,
+			ParentID: nil,
+			Debug:    false,
+			Sampled:  &trueBool,
+			Err:      errors.New("status-cancelled"),
+		},
+		LocalEndpoint: &zipkinmodel.Endpoint{
+			ServiceName: "OTLPResourceNoServiceName",
+		},
+		RemoteEndpoint: nil,
+		Annotations: []zipkinmodel.Annotation{
+			{
+				Timestamp: testdata.TestSpanEventTime,
+				Value:     "event-with-attr|{\"span-event-attr\":\"span-event-attr-val\"}|2",
+			},
+			{
+				Timestamp: testdata.TestSpanEventTime,
+				Value:     "event|{}|2",
+			},
+		},
+		Tags: map[string]string{
+			"resource-attr":  "resource-attr-val-1",
+			"status.code":    "STATUS_CODE_CANCELLED",
+			"status.message": "status-cancelled",
+		},
+		Name:      "operationA",
+		Timestamp: testdata.TestSpanStartTime,
+		Duration:  1000000468,
+		Shared:    false,
+	}
+}
