diff --git a/exporter/exportertest/go.mod b/exporter/exportertest/go.mod
index c04e53b0c..6b642de47 100644
--- a/exporter/exportertest/go.mod
+++ b/exporter/exportertest/go.mod
@@ -19,7 +19,7 @@ require (
 	go.opentelemetry.io/collector/pipeline v1.51.0
 	go.opentelemetry.io/collector/receiver v1.51.0
 	go.opentelemetry.io/collector/receiver/receivertest v0.145.0
-	google.golang.org/grpc v1.78.0
+	google.golang.org/grpc v1.79.1
 )
 
 require (
