diff --git a/cmd/otelsvc/main.go b/cmd/otelsvc/main.go
index 25844d14a..5a17292c0 100644
--- a/cmd/otelsvc/main.go
+++ b/cmd/otelsvc/main.go
@@ -19,12 +19,21 @@ package main
 import (
 	"log"
 
-	_ "github.com/open-telemetry/opentelemetry-service/receiver/vmmetricsreceiver"
+	"github.com/open-telemetry/opentelemetry-service/defaults"
 	"github.com/open-telemetry/opentelemetry-service/service"
 )
 
 func main() {
-	if err := service.App.StartUnified(); err != nil {
-		log.Fatalf("Failed to run the service: %v", err)
+	handleErr := func(err error) {
+		if err != nil {
+			log.Fatalf("Failed to run the service: %v", err)
+		}
 	}
+
+	receivers, processors, exporters, err := defaults.Components()
+	handleErr(err)
+
+	svc := service.New(receivers, processors, exporters)
+	err = svc.StartUnified()
+	handleErr(err)
 }
