diff --git a/service/service.go b/service/service.go
index d28bba03a..c3a2e6b4e 100644
--- a/service/service.go
+++ b/service/service.go
@@ -87,6 +87,7 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 		useOtel = *set.useOtel
 	}
 	disableHighCard := obsreportconfig.DisableHighCardinalityMetricsfeatureGate.IsEnabled()
+	extendedConfig := obsreportconfig.UseOtelWithSDKConfigurationForInternalTelemetryFeatureGate.IsEnabled()
 	srv := &Service{
 		buildInfo: set.BuildInfo,
 		host: &serviceHost{
@@ -98,7 +99,7 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 			buildInfo:         set.BuildInfo,
 			asyncErrorChannel: set.AsyncErrorChannel,
 		},
-		telemetryInitializer: newColTelemetry(useOtel, disableHighCard),
+		telemetryInitializer: newColTelemetry(useOtel, disableHighCard, extendedConfig),
 	}
 	var err error
 	srv.telemetry, err = telemetry.New(ctx, telemetry.Settings{ZapOptions: set.LoggingOptions}, cfg.Telemetry)
