diff --git a/pdata/pmetric/generated_exemplarslice.go b/pdata/pmetric/generated_exemplarslice.go
index b88f3eb30..d6fb809cf 100644
--- a/pdata/pmetric/generated_exemplarslice.go
+++ b/pdata/pmetric/generated_exemplarslice.go
@@ -10,7 +10,6 @@ import (
 	"iter"
 
 	"go.opentelemetry.io/collector/pdata/internal"
-	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
 )
 
 // ExemplarSlice logically represents a slice of Exemplar.
@@ -21,18 +20,18 @@ import (
 // Must use NewExemplarSlice function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type ExemplarSlice struct {
-	orig  *[]otlpmetrics.Exemplar
+	orig  *[]internal.Exemplar
 	state *internal.State
 }
 
-func newExemplarSlice(orig *[]otlpmetrics.Exemplar, state *internal.State) ExemplarSlice {
+func newExemplarSlice(orig *[]internal.Exemplar, state *internal.State) ExemplarSlice {
 	return ExemplarSlice{orig: orig, state: state}
 }
 
 // NewExemplarSlice creates a ExemplarSliceWrapper with 0 elements.
 // Can use "EnsureCapacity" to initialize with a given capacity.
 func NewExemplarSlice() ExemplarSlice {
-	orig := []otlpmetrics.Exemplar(nil)
+	orig := []internal.Exemplar(nil)
 	return newExemplarSlice(&orig, internal.NewState())
 }
 
@@ -89,7 +88,7 @@ func (es ExemplarSlice) EnsureCapacity(newCap int) {
 		return
 	}
 
-	newOrig := make([]otlpmetrics.Exemplar, len(*es.orig), newCap)
+	newOrig := make([]internal.Exemplar, len(*es.orig), newCap)
 	copy(newOrig, *es.orig)
 	*es.orig = newOrig
 }
@@ -98,7 +97,7 @@ func (es ExemplarSlice) EnsureCapacity(newCap int) {
 // It returns the newly added Exemplar.
 func (es ExemplarSlice) AppendEmpty() Exemplar {
 	es.state.AssertMutable()
-	*es.orig = append(*es.orig, otlpmetrics.Exemplar{})
+	*es.orig = append(*es.orig, internal.Exemplar{})
 	return es.At(es.Len() - 1)
 }
 
