diff --git a/pdata/internal/generated_plog.go b/pdata/internal/generated_plog.go
index 601d47cb6..2fd319d65 100644
--- a/pdata/internal/generated_plog.go
+++ b/pdata/internal/generated_plog.go
@@ -649,20 +649,6 @@ func (ms LogRecord) SetSeverityNumber(v SeverityNumber) {
 	(*ms.orig).SeverityNumber = otlplogs.SeverityNumber(v)
 }
 
-// Name returns the name associated with this LogRecord.
-//
-// Deprecated: [v0.50.0] it was removed from the data model.
-func (ms LogRecord) Name() string {
-	return (*ms.orig).Name
-}
-
-// SetName replaces the name associated with this LogRecord.
-//
-// Deprecated: [v0.50.0] it was removed from the data model.
-func (ms LogRecord) SetName(v string) {
-	(*ms.orig).Name = v
-}
-
 // Body returns the body associated with this LogRecord.
 func (ms LogRecord) Body() Value {
 	return newValue(&(*ms.orig).Body)
@@ -692,7 +678,6 @@ func (ms LogRecord) CopyTo(dest LogRecord) {
 	dest.SetFlags(ms.Flags())
 	dest.SetSeverityText(ms.SeverityText())
 	dest.SetSeverityNumber(ms.SeverityNumber())
-	dest.SetName(ms.Name())
 	ms.Body().CopyTo(dest.Body())
 	ms.Attributes().CopyTo(dest.Attributes())
 	dest.SetDroppedAttributesCount(ms.DroppedAttributesCount())
