diff --git a/pdata/internal/generated_wrapper_span_event_test.go b/pdata/internal/generated_wrapper_span_event_test.go
index a6d333563..13594b5bf 100644
--- a/pdata/internal/generated_wrapper_span_event_test.go
+++ b/pdata/internal/generated_wrapper_span_event_test.go
@@ -20,10 +20,10 @@ import (
 )
 
 func TestCopyOrigSpan_Event(t *testing.T) {
-	src := NewOrigPtrSpan_Event()
-	dest := NewOrigPtrSpan_Event()
+	src := NewOrigSpan_Event()
+	dest := NewOrigSpan_Event()
 	CopyOrigSpan_Event(dest, src)
-	assert.Equal(t, NewOrigPtrSpan_Event(), dest)
+	assert.Equal(t, NewOrigSpan_Event(), dest)
 	*src = *GenTestOrigSpan_Event()
 	CopyOrigSpan_Event(dest, src)
 	assert.Equal(t, src, dest)
@@ -32,10 +32,10 @@ func TestCopyOrigSpan_Event(t *testing.T) {
 func TestMarshalAndUnmarshalJSONOrigSpan_EventUnknown(t *testing.T) {
 	iter := json.BorrowIterator([]byte(`{"unknown": "string"}`))
 	defer json.ReturnIterator(iter)
-	dest := NewOrigPtrSpan_Event()
+	dest := NewOrigSpan_Event()
 	UnmarshalJSONOrigSpan_Event(dest, iter)
 	require.NoError(t, iter.Error())
-	assert.Equal(t, NewOrigPtrSpan_Event(), dest)
+	assert.Equal(t, NewOrigSpan_Event(), dest)
 }
 
 func TestMarshalAndUnmarshalJSONOrigSpan_Event(t *testing.T) {
@@ -48,7 +48,7 @@ func TestMarshalAndUnmarshalJSONOrigSpan_Event(t *testing.T) {
 
 			iter := json.BorrowIterator(stream.Buffer())
 			defer json.ReturnIterator(iter)
-			dest := NewOrigPtrSpan_Event()
+			dest := NewOrigSpan_Event()
 			UnmarshalJSONOrigSpan_Event(dest, iter)
 			require.NoError(t, iter.Error())
 
@@ -60,17 +60,17 @@ func TestMarshalAndUnmarshalJSONOrigSpan_Event(t *testing.T) {
 func TestMarshalAndUnmarshalProtoOrigSpan_EventFailing(t *testing.T) {
 	for name, buf := range genTestFailingUnmarshalProtoValuesSpan_Event() {
 		t.Run(name, func(t *testing.T) {
-			dest := NewOrigPtrSpan_Event()
+			dest := NewOrigSpan_Event()
 			require.Error(t, UnmarshalProtoOrigSpan_Event(dest, buf))
 		})
 	}
 }
 
 func TestMarshalAndUnmarshalProtoOrigSpan_EventUnknown(t *testing.T) {
-	dest := NewOrigPtrSpan_Event()
+	dest := NewOrigSpan_Event()
 	// message Test { required int64 field = 1313; } encoding { "field": "1234" }
 	require.NoError(t, UnmarshalProtoOrigSpan_Event(dest, []byte{0x88, 0x52, 0xD2, 0x09}))
-	assert.Equal(t, NewOrigPtrSpan_Event(), dest)
+	assert.Equal(t, NewOrigSpan_Event(), dest)
 }
 
 func TestMarshalAndUnmarshalProtoOrigSpan_Event(t *testing.T) {
@@ -80,7 +80,7 @@ func TestMarshalAndUnmarshalProtoOrigSpan_Event(t *testing.T) {
 			gotSize := MarshalProtoOrigSpan_Event(src, buf)
 			assert.Equal(t, len(buf), gotSize)
 
-			dest := NewOrigPtrSpan_Event()
+			dest := NewOrigSpan_Event()
 			require.NoError(t, UnmarshalProtoOrigSpan_Event(dest, buf))
 			assert.Equal(t, src, dest)
 		})
@@ -100,7 +100,7 @@ func TestMarshalAndUnmarshalProtoViaProtobufSpan_Event(t *testing.T) {
 			goBuf, err := proto.Marshal(goDest)
 			require.NoError(t, err)
 
-			dest := NewOrigPtrSpan_Event()
+			dest := NewOrigSpan_Event()
 			require.NoError(t, UnmarshalProtoOrigSpan_Event(dest, goBuf))
 			assert.Equal(t, src, dest)
 		})
@@ -123,7 +123,7 @@ func genTestFailingUnmarshalProtoValuesSpan_Event() map[string][]byte {
 
 func genTestEncodingValuesSpan_Event() map[string]*otlptrace.Span_Event {
 	return map[string]*otlptrace.Span_Event{
-		"empty":                       NewOrigPtrSpan_Event(),
+		"empty":                       NewOrigSpan_Event(),
 		"TimeUnixNano/test":           {TimeUnixNano: uint64(13)},
 		"Name/test":                   {Name: "test_name"},
 		"Attributes/default_and_test": {Attributes: []otlpcommon.KeyValue{{}, *GenTestOrigKeyValue()}},
