diff --git a/pdata/internal/generated_wrapper_sampleslice.go b/pdata/internal/generated_wrapper_sampleslice.go
index 3ee1f2189..c897417db 100644
--- a/pdata/internal/generated_wrapper_sampleslice.go
+++ b/pdata/internal/generated_wrapper_sampleslice.go
@@ -19,7 +19,7 @@ func CopyOrigSampleSlice(dest, src []*otlpprofiles.Sample) []*otlpprofiles.Sampl
 		copy(newDest, dest)
 		// Add new pointers for missing elements from len(dest) to len(srt).
 		for i := len(dest); i < len(src); i++ {
-			newDest[i] = NewOrigPtrSample()
+			newDest[i] = NewOrigSample()
 		}
 	} else {
 		newDest = dest[:len(src)]
@@ -31,7 +31,7 @@ func CopyOrigSampleSlice(dest, src []*otlpprofiles.Sample) []*otlpprofiles.Sampl
 		// Add new pointers for missing elements.
 		// This can happen when len(dest) < len(src) < cap(dest).
 		for i := len(dest); i < len(src); i++ {
-			newDest[i] = NewOrigPtrSample()
+			newDest[i] = NewOrigSample()
 		}
 	}
 	for i := range src {
@@ -42,11 +42,11 @@ func CopyOrigSampleSlice(dest, src []*otlpprofiles.Sample) []*otlpprofiles.Sampl
 
 func GenerateOrigTestSampleSlice() []*otlpprofiles.Sample {
 	orig := make([]*otlpprofiles.Sample, 5)
-	orig[0] = NewOrigPtrSample()
+	orig[0] = NewOrigSample()
 	orig[1] = GenTestOrigSample()
-	orig[2] = NewOrigPtrSample()
+	orig[2] = NewOrigSample()
 	orig[3] = GenTestOrigSample()
-	orig[4] = NewOrigPtrSample()
+	orig[4] = NewOrigSample()
 	return orig
 }
 
@@ -54,7 +54,7 @@ func GenerateOrigTestSampleSlice() []*otlpprofiles.Sample {
 func UnmarshalJSONOrigSampleSlice(iter *json.Iterator) []*otlpprofiles.Sample {
 	var orig []*otlpprofiles.Sample
 	iter.ReadArrayCB(func(iter *json.Iterator) bool {
-		orig = append(orig, NewOrigPtrSample())
+		orig = append(orig, NewOrigSample())
 		UnmarshalJSONOrigSample(orig[len(orig)-1], iter)
 		return true
 	})
