diff --git a/pdata/internal/generated_wrapper_numberdatapoint.go b/pdata/internal/generated_wrapper_numberdatapoint.go
index 72ad5f81f..4e2412511 100644
--- a/pdata/internal/generated_wrapper_numberdatapoint.go
+++ b/pdata/internal/generated_wrapper_numberdatapoint.go
@@ -11,16 +11,13 @@ import (
 	"fmt"
 	"math"
 
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
 	otlpmetrics "go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 	"go.opentelemetry.io/collector/pdata/internal/proto"
 )
 
-func NewOrigNumberDataPoint() otlpmetrics.NumberDataPoint {
-	return otlpmetrics.NumberDataPoint{}
-}
-
-func NewOrigPtrNumberDataPoint() *otlpmetrics.NumberDataPoint {
+func NewOrigNumberDataPoint() *otlpmetrics.NumberDataPoint {
 	return &otlpmetrics.NumberDataPoint{}
 }
 
@@ -39,7 +36,7 @@ func CopyOrigNumberDataPoint(dest, src *otlpmetrics.NumberDataPoint) {
 }
 
 func GenTestOrigNumberDataPoint() *otlpmetrics.NumberDataPoint {
-	orig := NewOrigPtrNumberDataPoint()
+	orig := NewOrigNumberDataPoint()
 	orig.Attributes = GenerateOrigTestKeyValueSlice()
 	orig.StartTimeUnixNano = 1234567890
 	orig.TimeUnixNano = 1234567890
@@ -232,7 +229,7 @@ func UnmarshalProtoOrigNumberDataPoint(orig *otlpmetrics.NumberDataPoint, buf []
 				return err
 			}
 			startPos := pos - length
-			orig.Attributes = append(orig.Attributes, NewOrigKeyValue())
+			orig.Attributes = append(orig.Attributes, otlpcommon.KeyValue{})
 			err = UnmarshalProtoOrigKeyValue(&orig.Attributes[len(orig.Attributes)-1], buf[startPos:pos])
 			if err != nil {
 				return err
@@ -298,7 +295,7 @@ func UnmarshalProtoOrigNumberDataPoint(orig *otlpmetrics.NumberDataPoint, buf []
 				return err
 			}
 			startPos := pos - length
-			orig.Exemplars = append(orig.Exemplars, NewOrigExemplar())
+			orig.Exemplars = append(orig.Exemplars, otlpmetrics.Exemplar{})
 			err = UnmarshalProtoOrigExemplar(&orig.Exemplars[len(orig.Exemplars)-1], buf[startPos:pos])
 			if err != nil {
 				return err
