diff --git a/pdata/internal/generated_wrapper_resourcelogsslice.go b/pdata/internal/generated_wrapper_resourcelogsslice.go
index 57d7951ef..07a45c650 100644
--- a/pdata/internal/generated_wrapper_resourcelogsslice.go
+++ b/pdata/internal/generated_wrapper_resourcelogsslice.go
@@ -19,7 +19,7 @@ func CopyOrigResourceLogsSlice(dest, src []*otlplogs.ResourceLogs) []*otlplogs.R
 		copy(newDest, dest)
 		// Add new pointers for missing elements from len(dest) to len(srt).
 		for i := len(dest); i < len(src); i++ {
-			newDest[i] = NewOrigPtrResourceLogs()
+			newDest[i] = NewOrigResourceLogs()
 		}
 	} else {
 		newDest = dest[:len(src)]
@@ -31,7 +31,7 @@ func CopyOrigResourceLogsSlice(dest, src []*otlplogs.ResourceLogs) []*otlplogs.R
 		// Add new pointers for missing elements.
 		// This can happen when len(dest) < len(src) < cap(dest).
 		for i := len(dest); i < len(src); i++ {
-			newDest[i] = NewOrigPtrResourceLogs()
+			newDest[i] = NewOrigResourceLogs()
 		}
 	}
 	for i := range src {
@@ -42,11 +42,11 @@ func CopyOrigResourceLogsSlice(dest, src []*otlplogs.ResourceLogs) []*otlplogs.R
 
 func GenerateOrigTestResourceLogsSlice() []*otlplogs.ResourceLogs {
 	orig := make([]*otlplogs.ResourceLogs, 5)
-	orig[0] = NewOrigPtrResourceLogs()
+	orig[0] = NewOrigResourceLogs()
 	orig[1] = GenTestOrigResourceLogs()
-	orig[2] = NewOrigPtrResourceLogs()
+	orig[2] = NewOrigResourceLogs()
 	orig[3] = GenTestOrigResourceLogs()
-	orig[4] = NewOrigPtrResourceLogs()
+	orig[4] = NewOrigResourceLogs()
 	return orig
 }
 
@@ -54,7 +54,7 @@ func GenerateOrigTestResourceLogsSlice() []*otlplogs.ResourceLogs {
 func UnmarshalJSONOrigResourceLogsSlice(iter *json.Iterator) []*otlplogs.ResourceLogs {
 	var orig []*otlplogs.ResourceLogs
 	iter.ReadArrayCB(func(iter *json.Iterator) bool {
-		orig = append(orig, NewOrigPtrResourceLogs())
+		orig = append(orig, NewOrigResourceLogs())
 		UnmarshalJSONOrigResourceLogs(orig[len(orig)-1], iter)
 		return true
 	})
