diff --git a/pdata/ptrace/traces.go b/pdata/ptrace/traces.go
index 757bd67f9..9c2baa76f 100644
--- a/pdata/ptrace/traces.go
+++ b/pdata/ptrace/traces.go
@@ -36,14 +36,6 @@ func NewTraces() Traces {
 	return newTraces(&otlpcollectortrace.ExportTraceServiceRequest{})
 }
 
-// MoveTo moves the Traces instance overriding the destination and
-// resetting the current instance to its zero value.
-// Deprecated: [1.0.0-rc5] The method can be replaced with a plain assignment.
-func (ms Traces) MoveTo(dest Traces) {
-	*dest.getOrig() = *ms.getOrig()
-	*ms.getOrig() = otlpcollectortrace.ExportTraceServiceRequest{}
-}
-
 // CopyTo copies the Traces instance overriding the destination.
 func (ms Traces) CopyTo(dest Traces) {
 	ms.ResourceSpans().CopyTo(dest.ResourceSpans())
