diff --git a/pdata/pprofile/generated_location.go b/pdata/pprofile/generated_location.go
index f8820577f..81af42688 100644
--- a/pdata/pprofile/generated_location.go
+++ b/pdata/pprofile/generated_location.go
@@ -112,7 +112,9 @@ func (ms Location) CopyTo(dest Location) {
 }
 
 func copyOrigLocation(dest, src *otlpprofiles.Location) {
-	if src.MappingIndex_ != nil {
+	if src.MappingIndex_ == nil {
+		dest.MappingIndex_ = nil
+	} else {
 		dest.MappingIndex_ = &otlpprofiles.Location_MappingIndex{MappingIndex: src.GetMappingIndex()}
 	}
 	dest.Address = src.Address
