diff --git a/internal/telemetry/componentattribute/attribute.go b/internal/telemetry/componentattribute/attribute.go
index 78532bb89..a246af4da 100644
--- a/internal/telemetry/componentattribute/attribute.go
+++ b/internal/telemetry/componentattribute/attribute.go
@@ -3,6 +3,12 @@
 
 package componentattribute // import "go.opentelemetry.io/collector/internal/telemetry/componentattribute"
 
+import (
+	"slices"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
 const (
 	ComponentKindKey = "otelcol.component.kind"
 	ComponentIDKey   = "otelcol.component.id"
@@ -10,3 +16,10 @@ const (
 	SignalKey        = "otelcol.signal"
 	SignalOutputKey  = "otelcol.signal.output"
 )
+
+func RemoveAttributes(attrs attribute.Set, fields ...string) attribute.Set {
+	attrs, _ = attribute.NewSetWithFiltered(attrs.ToSlice(), func(kv attribute.KeyValue) bool {
+		return !slices.Contains(fields, string(kv.Key))
+	})
+	return attrs
+}
