diff --git a/config/connector.go b/config/connector.go
index 3ec7f73f8..5637b9b59 100644
--- a/config/connector.go
+++ b/config/connector.go
@@ -32,13 +32,3 @@ func NewConnectorSettings(id component.ID) ConnectorSettings {
 }
 
 var _ component.Config = (*ConnectorSettings)(nil)
-
-// ID returns the connector ComponentID.
-func (cs *ConnectorSettings) ID() component.ID {
-	return cs.id
-}
-
-// SetIDName sets the connector name.
-func (cs *ConnectorSettings) SetIDName(idName string) {
-	cs.id = component.NewIDWithName(cs.id.Type(), idName)
-}
