diff --git a/cmd/mdatagen/templates/component_telemetry_test.go.tmpl b/cmd/mdatagen/templates/component_telemetry_test.go.tmpl
index 03a702acd..3c3f06140 100644
--- a/cmd/mdatagen/templates/component_telemetry_test.go.tmpl
+++ b/cmd/mdatagen/templates/component_telemetry_test.go.tmpl
@@ -21,7 +21,7 @@ type componentTestTelemetry struct {
 	meterProvider *sdkmetric.MeterProvider
 }
 
-{{- if isReceiver }}
+{{- if (or isExporter isReceiver) }}
 func (tt *componentTestTelemetry) NewSettings() {{ .Status.Class }}.Settings {
 	settings := {{ .Status.Class }}test.NewNopSettings()
 	settings.MeterProvider = tt.meterProvider
