diff --git a/pdata/pprofile/generated_valuetype.go b/pdata/pprofile/generated_valuetype.go
index da66dc05d..8b05aa2fe 100644
--- a/pdata/pprofile/generated_valuetype.go
+++ b/pdata/pprofile/generated_valuetype.go
@@ -8,7 +8,6 @@ package pprofile
 
 import (
 	"go.opentelemetry.io/collector/pdata/internal"
-	otlpprofiles "go.opentelemetry.io/collector/pdata/internal/data/protogen/profiles/v1development"
 )
 
 // ValueType describes the type and units of a value, with an optional aggregation temporality.
@@ -19,11 +18,11 @@ import (
 // Must use NewValueType function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type ValueType struct {
-	orig  *otlpprofiles.ValueType
+	orig  *internal.ValueType
 	state *internal.State
 }
 
-func newValueType(orig *otlpprofiles.ValueType, state *internal.State) ValueType {
+func newValueType(orig *internal.ValueType, state *internal.State) ValueType {
 	return ValueType{orig: orig, state: state}
 }
 
@@ -78,7 +77,7 @@ func (ms ValueType) AggregationTemporality() AggregationTemporality {
 // SetAggregationTemporality replaces the aggregationtemporality associated with this ValueType.
 func (ms ValueType) SetAggregationTemporality(v AggregationTemporality) {
 	ms.state.AssertMutable()
-	ms.orig.AggregationTemporality = otlpprofiles.AggregationTemporality(v)
+	ms.orig.AggregationTemporality = internal.AggregationTemporality(v)
 }
 
 // CopyTo copies all properties from the current struct overriding the destination.
