diff --git a/pdata/plog/logs_test.go b/pdata/plog/logs_test.go
index b8909e462..f633b820f 100644
--- a/pdata/plog/logs_test.go
+++ b/pdata/plog/logs_test.go
@@ -47,14 +47,14 @@ func TestLogRecordCountWithEmpty(t *testing.T) {
 	assert.Zero(t, NewLogs().LogRecordCount())
 	assert.Zero(t, newLogs(&otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{{}},
-	}).LogRecordCount())
+	}, new(internal.State)).LogRecordCount())
 	assert.Zero(t, newLogs(&otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
 				ScopeLogs: []*otlplogs.ScopeLogs{{}},
 			},
 		},
-	}).LogRecordCount())
+	}, new(internal.State)).LogRecordCount())
 	assert.Equal(t, 1, newLogs(&otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
@@ -65,14 +65,7 @@ func TestLogRecordCountWithEmpty(t *testing.T) {
 				},
 			},
 		},
-	}).LogRecordCount())
-}
-
-func TestToFromLogOtlp(t *testing.T) {
-	otlp := &otlpcollectorlog.ExportLogsServiceRequest{}
-	logs := newLogs(otlp)
-	assert.Equal(t, NewLogs(), logs)
-	assert.Equal(t, otlp, logs.getOrig())
+	}, new(internal.State)).LogRecordCount())
 }
 
 func TestResourceLogsWireCompatibility(t *testing.T) {
@@ -108,13 +101,6 @@ func TestResourceLogsWireCompatibility(t *testing.T) {
 	assert.Equal(t, ms.getOrig(), &gogoprotoRS2)
 }
 
-func TestLogsCopyTo(t *testing.T) {
-	ld := generateTestLogs()
-	logsCopy := NewLogs()
-	ld.CopyTo(logsCopy)
-	assert.Equal(t, ld, logsCopy)
-}
-
 func TestReadOnlyLogsInvalidUsage(t *testing.T) {
 	ld := NewLogs()
 	assert.False(t, ld.IsReadOnly())
@@ -183,9 +169,3 @@ func BenchmarkLogsMarshalJSON(b *testing.B) {
 		require.NotNil(b, jsonBuf)
 	}
 }
-
-func generateTestLogs() Logs {
-	ld := NewLogs()
-	ld.getOrig().ResourceLogs = internal.GenerateOrigTestResourceLogsSlice()
-	return ld
-}
