diff --git a/otelcol/unmarshaler_test.go b/otelcol/unmarshaler_test.go
index fc579e8f6..e410c0972 100644
--- a/otelcol/unmarshaler_test.go
+++ b/otelcol/unmarshaler_test.go
@@ -39,7 +39,7 @@ func TestUnmarshalEmptyAllSections(t *testing.T) {
 	factories, err := nopFactories()
 	assert.NoError(t, err)
 
-	conf := confmap.NewFromStringMap(map[string]interface{}{
+	conf := confmap.NewFromStringMap(map[string]any{
 		"receivers":  nil,
 		"processors": nil,
 		"exporters":  nil,
@@ -71,7 +71,7 @@ func TestUnmarshalUnknownTopLevel(t *testing.T) {
 	factories, err := nopFactories()
 	assert.NoError(t, err)
 
-	conf := confmap.NewFromStringMap(map[string]interface{}{
+	conf := confmap.NewFromStringMap(map[string]any{
 		"unknown_section": nil,
 	})
 	_, err = unmarshal(conf, factories)
@@ -89,7 +89,7 @@ func TestPipelineConfigUnmarshalError(t *testing.T) {
 	}{
 		{
 			name: "duplicate-pipeline",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
 				"traces/ pipe": nil,
 				"traces /pipe": nil,
 			}),
@@ -97,15 +97,15 @@ func TestPipelineConfigUnmarshalError(t *testing.T) {
 		},
 		{
 			name: "invalid-pipeline-name-after-slash",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
 				"metrics/": nil,
 			}),
 			expectError: "in \"metrics/\" id: the part after / should not be empty",
 		},
 		{
 			name: "invalid-pipeline-section",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
-				"traces": map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
+				"traces": map[string]any{
 					"unknown_section": nil,
 				},
 			}),
@@ -113,24 +113,24 @@ func TestPipelineConfigUnmarshalError(t *testing.T) {
 		},
 		{
 			name: "invalid-pipeline-sub-config",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
 				"traces": "string",
 			}),
 			expectError: "'[traces]' expected a map, got 'string'",
 		},
 		{
 			name: "invalid-pipeline-type",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
 				"/metrics": nil,
 			}),
 			expectError: "in \"/metrics\" id: the part before / should not be empty",
 		},
 		{
 			name: "invalid-sequence-value",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
-				"traces": map[string]interface{}{
-					"receivers": map[string]interface{}{
-						"nop": map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
+				"traces": map[string]any{
+					"receivers": map[string]any{
+						"nop": map[string]any{
 							"some": "config",
 						},
 					},
@@ -160,9 +160,9 @@ func TestServiceUnmarshalError(t *testing.T) {
 	}{
 		{
 			name: "invalid-logs-level",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
-				"telemetry": map[string]interface{}{
-					"logs": map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
+				"telemetry": map[string]any{
+					"logs": map[string]any{
 						"level": "UNKNOWN",
 					},
 				},
@@ -171,9 +171,9 @@ func TestServiceUnmarshalError(t *testing.T) {
 		},
 		{
 			name: "invalid-metrics-level",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
-				"telemetry": map[string]interface{}{
-					"metrics": map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
+				"telemetry": map[string]any{
+					"metrics": map[string]any{
 						"level": "unknown",
 					},
 				},
@@ -182,10 +182,10 @@ func TestServiceUnmarshalError(t *testing.T) {
 		},
 		{
 			name: "invalid-service-extensions-section",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
-				"extensions": []interface{}{
-					map[string]interface{}{
-						"nop": map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
+				"extensions": []any{
+					map[string]any{
+						"nop": map[string]any{
 							"some": "config",
 						},
 					},
@@ -195,14 +195,14 @@ func TestServiceUnmarshalError(t *testing.T) {
 		},
 		{
 			name: "invalid-service-section",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
 				"unknown_section": "string",
 			}),
 			expectError: "'' has invalid keys: unknown_section",
 		},
 		{
 			name: "invalid-pipelines-config",
-			conf: confmap.NewFromStringMap(map[string]interface{}{
+			conf: confmap.NewFromStringMap(map[string]any{
 				"pipelines": "string",
 			}),
 			expectError: "'pipelines' expected a map, got 'string'",
