diff --git a/pdata/xpdata/entity/generated_entityref.go b/pdata/xpdata/entity/generated_entityref.go
index 424114be5..96ff7d516 100644
--- a/pdata/xpdata/entity/generated_entityref.go
+++ b/pdata/xpdata/entity/generated_entityref.go
@@ -28,7 +28,7 @@ func newEntityRef(orig *otlpcommon.EntityRef, state *internal.State) EntityRef {
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewEntityRef() EntityRef {
-	return newEntityRef(internal.NewOrigPtrEntityRef(), internal.NewState())
+	return newEntityRef(internal.NewOrigEntityRef(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
