diff --git a/internal/testdata/log.go b/internal/testdata/log.go
index 623eabf51..e9fb22900 100644
--- a/internal/testdata/log.go
+++ b/internal/testdata/log.go
@@ -29,13 +29,13 @@ var (
 	TestLogTimestamp = pdata.TimestampFromTime(TestLogTime)
 )
 
-func GenerateLogDataOneEmptyResourceLogs() pdata.Logs {
+func GenerateLogsOneEmptyResourceLogs() pdata.Logs {
 	ld := pdata.NewLogs()
 	ld.ResourceLogs().AppendEmpty()
 	return ld
 }
 
-func generateLogOtlpOneEmptyResourceLogs() *otlpcollectorlog.ExportLogsServiceRequest {
+func generateLogsOtlpOneEmptyResourceLogs() *otlpcollectorlog.ExportLogsServiceRequest {
 	return &otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{},
@@ -43,8 +43,8 @@ func generateLogOtlpOneEmptyResourceLogs() *otlpcollectorlog.ExportLogsServiceRe
 	}
 }
 
-func GenerateLogDataNoLogRecords() pdata.Logs {
-	ld := GenerateLogDataOneEmptyResourceLogs()
+func GenerateLogsNoLogRecords() pdata.Logs {
+	ld := GenerateLogsOneEmptyResourceLogs()
 	initResource1(ld.ResourceLogs().At(0).Resource())
 	return ld
 }
@@ -59,14 +59,14 @@ func generateLogOtlpNoLogRecords() *otlpcollectorlog.ExportLogsServiceRequest {
 	}
 }
 
-func GenerateLogDataOneEmptyLogs() pdata.Logs {
-	ld := GenerateLogDataNoLogRecords()
+func GenerateLogsOneEmptyLogRecord() pdata.Logs {
+	ld := GenerateLogsNoLogRecords()
 	rs0 := ld.ResourceLogs().At(0)
 	rs0.InstrumentationLibraryLogs().AppendEmpty().Logs().AppendEmpty()
 	return ld
 }
 
-func generateLogOtlpOneEmptyLogs() *otlpcollectorlog.ExportLogsServiceRequest {
+func generateLogsOtlpOneEmptyLogRecord() *otlpcollectorlog.ExportLogsServiceRequest {
 	return &otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
@@ -83,14 +83,14 @@ func generateLogOtlpOneEmptyLogs() *otlpcollectorlog.ExportLogsServiceRequest {
 	}
 }
 
-func GenerateLogDataOneLogNoResource() pdata.Logs {
-	ld := GenerateLogDataOneEmptyResourceLogs()
+func GenerateLogsOneLogRecordNoResource() pdata.Logs {
+	ld := GenerateLogsOneEmptyResourceLogs()
 	rs0 := ld.ResourceLogs().At(0)
 	fillLogOne(rs0.InstrumentationLibraryLogs().AppendEmpty().Logs().AppendEmpty())
 	return ld
 }
 
-func generateLogOtlpOneLogNoResource() *otlpcollectorlog.ExportLogsServiceRequest {
+func generateLogsOtlpOneLogRecordNoResource() *otlpcollectorlog.ExportLogsServiceRequest {
 	return &otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
@@ -106,13 +106,13 @@ func generateLogOtlpOneLogNoResource() *otlpcollectorlog.ExportLogsServiceReques
 	}
 }
 
-func GenerateLogDataOneLog() pdata.Logs {
-	ld := GenerateLogDataOneEmptyLogs()
+func GenerateLogsOneLogRecord() pdata.Logs {
+	ld := GenerateLogsOneEmptyLogRecord()
 	fillLogOne(ld.ResourceLogs().At(0).InstrumentationLibraryLogs().At(0).Logs().At(0))
 	return ld
 }
 
-func generateLogOtlpOneLog() *otlpcollectorlog.ExportLogsServiceRequest {
+func generateLogsOtlpOneLogRecord() *otlpcollectorlog.ExportLogsServiceRequest {
 	return &otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
@@ -129,16 +129,15 @@ func generateLogOtlpOneLog() *otlpcollectorlog.ExportLogsServiceRequest {
 	}
 }
 
-func GenerateLogDataTwoLogsSameResource() pdata.Logs {
-	ld := GenerateLogDataOneEmptyLogs()
+func GenerateLogsTwoLogRecordsSameResource() pdata.Logs {
+	ld := GenerateLogsOneEmptyLogRecord()
 	logs := ld.ResourceLogs().At(0).InstrumentationLibraryLogs().At(0).Logs()
 	fillLogOne(logs.At(0))
 	fillLogTwo(logs.AppendEmpty())
 	return ld
 }
 
-// generateLogOtlpSameResourceTwologs returns the OTLP representation of the GenerateLogOtlpSameResourceTwologs.
-func generateLogOtlpSameResourceTwoLogs() *otlpcollectorlog.ExportLogsServiceRequest {
+func generateLogsOtlpTwoLogRecordsSameResource() *otlpcollectorlog.ExportLogsServiceRequest {
 	return &otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
@@ -156,7 +155,7 @@ func generateLogOtlpSameResourceTwoLogs() *otlpcollectorlog.ExportLogsServiceReq
 	}
 }
 
-func GenerateLogDataTwoLogsSameResourceOneDifferent() pdata.Logs {
+func GenerateLogsTwoLogRecordsSameResourceOneDifferent() pdata.Logs {
 	ld := pdata.NewLogs()
 	rl0 := ld.ResourceLogs().AppendEmpty()
 	initResource1(rl0.Resource())
@@ -169,7 +168,7 @@ func GenerateLogDataTwoLogsSameResourceOneDifferent() pdata.Logs {
 	return ld
 }
 
-func generateLogOtlpTwoLogsSameResourceOneDifferent() *otlpcollectorlog.ExportLogsServiceRequest {
+func generateLogsOtlpTwoLogRecordsSameResourceOneDifferent() *otlpcollectorlog.ExportLogsServiceRequest {
 	return &otlpcollectorlog.ExportLogsServiceRequest{
 		ResourceLogs: []*otlplogs.ResourceLogs{
 			{
@@ -292,8 +291,8 @@ func generateOtlpLogThree() *otlplogs.LogRecord {
 	}
 }
 
-func GenerateLogDataManyLogsSameResource(count int) pdata.Logs {
-	ld := GenerateLogDataOneEmptyLogs()
+func GenerateLogsManyLogRecordsSameResource(count int) pdata.Logs {
+	ld := GenerateLogsOneEmptyLogRecord()
 	logs := ld.ResourceLogs().At(0).InstrumentationLibraryLogs().At(0).Logs()
 	logs.Resize(count)
 	for i := 0; i < count; i++ {
