diff --git a/internal/testdata/log_test.go b/internal/testdata/log_test.go
index 8da2f0165..c8f6fea53 100644
--- a/internal/testdata/log_test.go
+++ b/internal/testdata/log_test.go
@@ -34,38 +34,38 @@ func generateAllLogTestCases() []logTestCase {
 	return []logTestCase{
 		{
 			name: "one-empty-resource-logs",
-			ld:   GenerateLogDataOneEmptyResourceLogs(),
-			otlp: generateLogOtlpOneEmptyResourceLogs(),
+			ld:   GenerateLogsOneEmptyResourceLogs(),
+			otlp: generateLogsOtlpOneEmptyResourceLogs(),
 		},
 		{
 			name: "no-log-records",
-			ld:   GenerateLogDataNoLogRecords(),
+			ld:   GenerateLogsNoLogRecords(),
 			otlp: generateLogOtlpNoLogRecords(),
 		},
 		{
 			name: "one-empty-log-record",
-			ld:   GenerateLogDataOneEmptyLogs(),
-			otlp: generateLogOtlpOneEmptyLogs(),
+			ld:   GenerateLogsOneEmptyLogRecord(),
+			otlp: generateLogsOtlpOneEmptyLogRecord(),
 		},
 		{
 			name: "one-log-record-no-resource",
-			ld:   GenerateLogDataOneLogNoResource(),
-			otlp: generateLogOtlpOneLogNoResource(),
+			ld:   GenerateLogsOneLogRecordNoResource(),
+			otlp: generateLogsOtlpOneLogRecordNoResource(),
 		},
 		{
 			name: "one-log-record",
-			ld:   GenerateLogDataOneLog(),
-			otlp: generateLogOtlpOneLog(),
+			ld:   GenerateLogsOneLogRecord(),
+			otlp: generateLogsOtlpOneLogRecord(),
 		},
 		{
 			name: "two-records-same-resource",
-			ld:   GenerateLogDataTwoLogsSameResource(),
-			otlp: generateLogOtlpSameResourceTwoLogs(),
+			ld:   GenerateLogsTwoLogRecordsSameResource(),
+			otlp: generateLogsOtlpTwoLogRecordsSameResource(),
 		},
 		{
 			name: "two-records-same-resource-one-different",
-			ld:   GenerateLogDataTwoLogsSameResourceOneDifferent(),
-			otlp: generateLogOtlpTwoLogsSameResourceOneDifferent(),
+			ld:   GenerateLogsTwoLogRecordsSameResourceOneDifferent(),
+			otlp: generateLogsOtlpTwoLogRecordsSameResourceOneDifferent(),
 		},
 	}
 }
