diff --git a/service/internal/configunmarshaler/receivers.go b/service/internal/configunmarshaler/receivers.go
index f5bff7262..2e5e75b64 100644
--- a/service/internal/configunmarshaler/receivers.go
+++ b/service/internal/configunmarshaler/receivers.go
@@ -54,7 +54,6 @@ func (r *Receivers) Unmarshal(conf *confmap.Conf) error {
 
 		// Create the default config for this receiver.
 		receiverCfg := factory.CreateDefaultConfig()
-		receiverCfg.SetIDName(id.Name()) //nolint:staticcheck
 
 		// Now that the default config struct is created we can Unmarshal into it,
 		// and it will apply user-defined config on top of the default.
