diff --git a/service/internal/fanoutconsumer/metrics.go b/service/internal/fanoutconsumer/metrics.go
index 637e73162..9ea5c9c92 100644
--- a/service/internal/fanoutconsumer/metrics.go
+++ b/service/internal/fanoutconsumer/metrics.go
@@ -69,7 +69,9 @@ func (msc *metricsConsumer) ConsumeMetrics(ctx context.Context, md pmetric.Metri
 	// the incoming data to a mutating consumer is used that may change the incoming data before
 	// cloning.
 	for _, mc := range msc.clone {
-		errs = multierr.Append(errs, mc.ConsumeMetrics(ctx, md.Clone()))
+		clonedMetrics := pmetric.NewMetrics()
+		md.CopyTo(clonedMetrics)
+		errs = multierr.Append(errs, mc.ConsumeMetrics(ctx, clonedMetrics))
 	}
 	for _, mc := range msc.pass {
 		errs = multierr.Append(errs, mc.ConsumeMetrics(ctx, md))
