diff --git a/extension/xextension/go.mod b/extension/xextension/go.mod
index f1e681d52..a6baa5b3b 100644
--- a/extension/xextension/go.mod
+++ b/extension/xextension/go.mod
@@ -5,9 +5,9 @@ go 1.25.0
 require (
 	github.com/stretchr/testify v1.11.1
 	go.opentelemetry.io/collector/component v1.51.0
-	go.opentelemetry.io/collector/component/componenttest v0.145.0
+	go.opentelemetry.io/collector/component/componenttest v0.146.0
 	go.opentelemetry.io/collector/extension v1.51.0
-	go.opentelemetry.io/collector/internal/componentalias v0.145.0
+	go.opentelemetry.io/collector/internal/componentalias v0.146.0
 )
 
 require (
