diff --git a/cmd/builder/go.mod b/cmd/builder/go.mod
index f46a87486..e4f46b8e6 100644
--- a/cmd/builder/go.mod
+++ b/cmd/builder/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/knadh/koanf/v2 v2.2.2
 	github.com/spf13/cobra v1.9.1
 	github.com/spf13/pflag v1.0.7
-	github.com/stretchr/testify v1.10.0
+	github.com/stretchr/testify v1.11.0
 	go.uber.org/goleak v1.3.0
 	go.uber.org/multierr v1.11.0
 	go.uber.org/zap v1.27.0
