diff --git a/pdata/ptrace/pb.go b/pdata/ptrace/pb.go
index acc4440d7..e16414590 100644
--- a/pdata/ptrace/pb.go
+++ b/pdata/ptrace/pb.go
@@ -19,12 +19,12 @@ import (
 	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
 )
 
-// NewProtoMarshaler returns a Marshaler. Marshals to OTLP binary protobuf bytes.
-func NewProtoMarshaler() Marshaler {
+// NewProtoMarshaler returns a MarshalerSizer.
+// Marshals to OTLP binary protobuf bytes and calculates the size of the marshaled Traces.
+func NewProtoMarshaler() MarshalerSizer {
 	return newPbMarshaler()
 }
 
-// TODO(#3842): Figure out how we want to represent/return *Sizers.
 type pbMarshaler struct{}
 
 func newPbMarshaler() *pbMarshaler {
