diff --git a/internal/cmd/pdatagen/internal/templates/primitive_slice_test.go.tmpl b/internal/cmd/pdatagen/internal/templates/primitive_slice_test.go.tmpl
index 34369bab1..9c555df70 100644
--- a/internal/cmd/pdatagen/internal/templates/primitive_slice_test.go.tmpl
+++ b/internal/cmd/pdatagen/internal/templates/primitive_slice_test.go.tmpl
@@ -164,22 +164,6 @@ func Test{{ .structName }}Equal(t *testing.T) {
 	assert.True(t, ms.Equal(ms2))
 }
 
-func Test{{ .structName }}_MarshalAndUnmarshalJSON(t *testing.T) {
-	stream := json.BorrowStream(nil)
-	defer json.ReturnStream(stream)
-	src := New{{ .structName }}()
-	*src.getOrig() = internal.GenerateOrigTest{{ .elementOriginName }}Slice()
-	internal.MarshalJSONStream{{ .structName }}(internal.{{ .structName }}(src), stream)
-	require.NoError(t, stream.Error())
-
-	iter := json.BorrowIterator(stream.Buffer())
-	defer json.ReturnIterator(iter)
-	dest := New{{ .structName }}()
-	internal.UnmarshalJSONIter{{ .structName }}(internal.{{ .structName }}(dest), iter)
-
-	assert.Equal(t, src, dest)
-}
-
 func Benchmark{{ .structName }}Equal(b *testing.B) {
 	ms := New{{ .structName }}()
 	ms.Append({{ .testOrigVal }})
