diff --git a/processor/processortest/go.mod b/processor/processortest/go.mod
index dbdea5d52..3074ea5ca 100644
--- a/processor/processortest/go.mod
+++ b/processor/processortest/go.mod
@@ -7,7 +7,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector/component v0.113.0
 	go.opentelemetry.io/collector/component/componentstatus v0.113.0
-	go.opentelemetry.io/collector/component/componenttest v0.113.0
+	go.opentelemetry.io/collector/component/componenttest v0.0.0-20241112045949-9c4c007a12b4
 	go.opentelemetry.io/collector/consumer v0.113.0
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.113.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.113.0
