diff --git a/confmap/resolver.go b/confmap/resolver.go
index f2a6f595e..ee8058a56 100644
--- a/confmap/resolver.go
+++ b/confmap/resolver.go
@@ -183,13 +183,8 @@ func (mr *Resolver) Resolve(ctx context.Context) (*Conf, error) {
 		if err != nil {
 			return nil, err
 		}
-		if enableMergeAppendOption.IsEnabled() {
-			// only use MergeAppend when enableMergeAppendOption featuregate is enabled.
-			err = retMap.mergeAppend(retCfgMap)
-		} else {
-			err = retMap.Merge(retCfgMap)
-		}
-		if err != nil {
+
+		if err := retMap.Merge(retCfgMap); err != nil {
 			return nil, err
 		}
 	}
