diff --git a/Makefile.Common b/Makefile.Common
index 44fcdeb56..9a10afbb7 100644
--- a/Makefile.Common
+++ b/Makefile.Common
@@ -1,5 +1,7 @@
 # ALL_PKGS is the list of all packages where ALL_SRC files reside.
 ALL_PKGS := $(sort $(shell go list ./...))
+# COVER_PKGS is the list of packages to include in the coverage
+COVER_PKGS := $(shell go list ./... | tr "\n" ",")
 
 GOTEST_OPT?= -race -timeout 120s
 GOCMD?= go
@@ -19,8 +21,6 @@ CHECKDOC    := $(TOOLS_BIN_DIR)/checkdoc
 CHLOG       := $(TOOLS_BIN_DIR)/chloggen
 CROSSLINK   := $(TOOLS_BIN_DIR)/crosslink
 ENVSUBST    := $(TOOLS_BIN_DIR)/envsubst
-GO_ACC      := $(TOOLS_BIN_DIR)/go-acc
-GOCOVMERGE  := $(TOOLS_BIN_DIR)/gocovmerge
 GOIMPORTS   := $(TOOLS_BIN_DIR)/goimports
 GOVULNCHECK := $(TOOLS_BIN_DIR)/govulncheck
 LINT        := $(TOOLS_BIN_DIR)/golangci-lint
@@ -50,7 +50,8 @@ test:
 
 .PHONY: test-with-cover
 test-with-cover: $(GO_ACC)
-	$(GO_ACC) --output=coverage.out ./...
+	mkdir -p $(PWD)/coverage/unit
+	$(GOTEST) $(GOTEST_OPT) -cover ./... -covermode=atomic -coverpkg $(COVER_PKGS) -args -test.gocoverdir="$(PWD)/coverage/unit"
 
 .PHONY: benchmark
 benchmark:
