diff --git a/cmd/occollector/app/builder/pipelines_builder_test.go b/cmd/occollector/app/builder/pipelines_builder_test.go
index fef7c08e5..3b2c551e7 100644
--- a/cmd/occollector/app/builder/pipelines_builder_test.go
+++ b/cmd/occollector/app/builder/pipelines_builder_test.go
@@ -32,7 +32,7 @@ import (
 )
 
 // Ensure attributes processor is registered.
-var _ = addattributesprocessor.ConfigV2{}
+var _ = addattributesprocessor.Config{}
 
 // Register test factories used in the pipelines_builder.yaml test config.
 var _ = config.RegisterTestFactories()
