diff --git a/pdata/internal/cmd/pdatagen/internal/pprofile_package.go b/pdata/internal/cmd/pdatagen/internal/pprofile_package.go
index aca1593a0..5e7a7b3df 100644
--- a/pdata/internal/cmd/pdatagen/internal/pprofile_package.go
+++ b/pdata/internal/cmd/pdatagen/internal/pprofile_package.go
@@ -27,6 +27,7 @@ var pprofile = &Package{
 	structs: []baseStruct{
 		resourceProfilesSlice,
 		resourceProfiles,
+		profilesDictionary,
 		scopeProfilesSlice,
 		scopeProfiles,
 		profilesSlice,
@@ -71,6 +72,42 @@ var resourceProfiles = &messageValueStruct{
 	},
 }
 
+var profilesDictionary = &messageValueStruct{
+	structName:     "ProfilesDictionary",
+	description:    "// ProfilesDictionary is the reference table containing all data shared by profiles across the message being sent.",
+	originFullName: "otlpprofiles.ProfilesDictionary",
+	fields: []baseField{
+		&sliceField{
+			fieldName:   "MappingTable",
+			returnSlice: mappingSlice,
+		},
+		&sliceField{
+			fieldName:   "LocationTable",
+			returnSlice: locationSlice,
+		},
+		&sliceField{
+			fieldName:   "FunctionTable",
+			returnSlice: functionSlice,
+		},
+		&sliceField{
+			fieldName:   "LinkTable",
+			returnSlice: linkSlice,
+		},
+		&sliceField{
+			fieldName:   "StringTable",
+			returnSlice: stringSlice,
+		},
+		&sliceField{
+			fieldName:   "AttributeTable",
+			returnSlice: attributeTableSlice,
+		},
+		&sliceField{
+			fieldName:   "AttributeUnits",
+			returnSlice: attributeUnitSlice,
+		},
+	},
+}
+
 var scopeProfilesSlice = &sliceOfPtrs{
 	structName: "ScopeProfilesSlice",
 	element:    scopeProfiles,
@@ -108,38 +145,10 @@ var profile = &messageValueStruct{
 			fieldName:   "Sample",
 			returnSlice: sampleSlice,
 		},
-		&sliceField{
-			fieldName:   "MappingTable",
-			returnSlice: mappingSlice,
-		},
-		&sliceField{
-			fieldName:   "LocationTable",
-			returnSlice: locationSlice,
-		},
 		&sliceField{
 			fieldName:   "LocationIndices",
 			returnSlice: int32Slice,
 		},
-		&sliceField{
-			fieldName:   "FunctionTable",
-			returnSlice: functionSlice,
-		},
-		&sliceField{
-			fieldName:   "AttributeTable",
-			returnSlice: attributeTableSlice,
-		},
-		&sliceField{
-			fieldName:   "AttributeUnits",
-			returnSlice: attributeUnitSlice,
-		},
-		&sliceField{
-			fieldName:   "LinkTable",
-			returnSlice: linkSlice,
-		},
-		&sliceField{
-			fieldName:   "StringTable",
-			returnSlice: stringSlice,
-		},
 		&primitiveTypedField{
 			fieldName:       "Time",
 			originFieldName: "TimeNanos",
@@ -188,7 +197,7 @@ var profile = &messageValueStruct{
 			returnSlice: int32Slice,
 		},
 		&primitiveField{
-			fieldName:  "DefaultSampleTypeStrindex",
+			fieldName:  "DefaultSampleTypeIndex",
 			returnType: "int32",
 			defaultVal: "int32(0)",
 			testVal:    "int32(1)",
