diff --git a/internal/cmd/pdatagen/internal/slice_field.go b/internal/cmd/pdatagen/internal/slice_field.go
index 2bae8f5f5..9ff41de8e 100644
--- a/internal/cmd/pdatagen/internal/slice_field.go
+++ b/internal/cmd/pdatagen/internal/slice_field.go
@@ -7,18 +7,12 @@ import (
 	"text/template"
 )
 
-const sliceAccessorTemplate = `// {{ .fieldName }} returns the {{ .originFieldName }} associated with this {{ .structName }}.
+const sliceAccessorTemplate = `// {{ .fieldName }} returns the {{ .fieldName }} associated with this {{ .structName }}.
 func (ms {{ .structName }}) {{ .fieldName }}() {{ .packageName }}{{ .returnType }} {
 	{{- if .isCommon }}
-	return {{ .packageName }}{{ .returnType }}(internal.New{{ .returnType }}(&ms.{{ .origAccessor }}.{{ .originFieldName }}
-	{{- if .isBaseStructCommon -}}
-	, internal.Get{{ .structName }}State(internal.{{ .structName }}(ms))
-	{{- else -}}
-	, ms.state
-	{{- end -}}
-	))
+	return {{ .packageName }}{{ .returnType }}(internal.New{{ .returnType }}(&ms.{{ .origAccessor }}.{{ .originFieldName }}, ms.{{ .stateAccessor }}))
 	{{- else }}
-	return new{{ .returnType }}(&ms.{{ .origAccessor }}.{{ .originFieldName }}, ms.state)
+	return new{{ .returnType }}(&ms.{{ .origAccessor }}.{{ .originFieldName }}, ms.{{ .stateAccessor }})
 	{{- end }}
 }`
 
@@ -36,11 +30,10 @@ const sliceAccessorsTestTemplate = `func Test{{ .structName }}_{{ .fieldName }}(
 
 const sliceSetTestTemplate = `{{ if .isCommon -}}
 	{{ if not .isBaseStructCommon }}internal.{{ end }}FillTest{{ .returnType }}(
-	{{- if not .isBaseStructCommon }}internal.{{ end }}New
+	{{- if not .isBaseStructCommon }}internal.{{ end }}New{{ .returnType }}(&tv.orig.{{ .originFieldName }}, tv.state))
 	{{- else -}}
-	fillTest{{ .returnType }}(new
-	{{-	end -}}
-	{{ .returnType }}(&tv.orig.{{ .originFieldName }}, tv.state))`
+	fillTest{{ .returnType }}(tv.{{ .fieldName }}())
+	{{-	end }}`
 
 const sliceCopyOrigTemplate = `dest.{{ .originFieldName }} = 
 {{- if .isCommon }}{{ if not .isBaseStructCommon }}internal.{{ end }}CopyOrig{{ else }}copyOrig{{ end }}
@@ -65,10 +58,9 @@ const sliceUnmarshalJSONTemplate = `case "{{ lowerFirst .originFieldName }}"{{ i
 	{{- end }}`
 
 type SliceField struct {
-	fieldName       string
-	originFieldName string
-	returnSlice     baseSlice
-	hideAccessors   bool
+	fieldName     string
+	returnSlice   baseSlice
+	hideAccessors bool
 }
 
 func (sf *SliceField) GenerateAccessors(ms *messageStruct) string {
@@ -109,8 +101,9 @@ func (sf *SliceField) GenerateUnmarshalJSON(ms *messageStruct) string {
 
 func (sf *SliceField) templateFields(ms *messageStruct) map[string]any {
 	return map[string]any{
-		"structName": ms.getName(),
-		"fieldName":  sf.fieldName,
+		"structName":      ms.getName(),
+		"fieldName":       sf.fieldName,
+		"originFieldName": sf.fieldName,
 		"packageName": func() string {
 			if sf.returnSlice.getPackageName() != ms.packageName {
 				return sf.returnSlice.getPackageName() + "."
@@ -122,12 +115,6 @@ func (sf *SliceField) templateFields(ms *messageStruct) map[string]any {
 		"stateAccessor":      stateAccessor(ms.packageName),
 		"isCommon":           usedByOtherDataTypes(sf.returnSlice.getPackageName()),
 		"isBaseStructCommon": usedByOtherDataTypes(ms.packageName),
-		"originFieldName": func() string {
-			if sf.originFieldName == "" {
-				return sf.fieldName
-			}
-			return sf.originFieldName
-		}(),
 	}
 }
 
