diff --git a/receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go b/receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go
index f91bc8f04..b86df6fa0 100644
--- a/receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go
@@ -73,9 +73,7 @@ func (s *scraper) scrape(_ context.Context) (pdata.MetricSlice, error) {
 
 func initializeLoadMetric(metric pdata.Metric, metricDescriptor metadata.MetricIntf, now pdata.Timestamp, value float64) {
 	metricDescriptor.Init(metric)
-	idps := metric.DoubleGauge().DataPoints()
-	idps.Resize(1)
-	dp := idps.At(0)
+	dp := metric.DoubleGauge().DataPoints().AppendEmpty()
 	dp.SetTimestamp(now)
 	dp.SetValue(value)
 }
