diff --git a/otelcol/go.mod b/otelcol/go.mod
index d820379ef..f84b3a9af 100644
--- a/otelcol/go.mod
+++ b/otelcol/go.mod
@@ -68,6 +68,7 @@ require (
 	go.opentelemetry.io/collector/consumer v0.108.1 // indirect
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.108.1 // indirect
 	go.opentelemetry.io/collector/consumer/consumertest v0.108.1 // indirect
+	go.opentelemetry.io/collector/exporter/exporterprofiles v0.108.1 // indirect
 	go.opentelemetry.io/collector/internal/globalgates v0.108.1 // indirect
 	go.opentelemetry.io/collector/pdata v1.14.1 // indirect
 	go.opentelemetry.io/collector/pdata/pprofile v0.108.1 // indirect
@@ -170,3 +171,5 @@ replace go.opentelemetry.io/collector/receiver/receiverprofiles => ../receiver/r
 replace go.opentelemetry.io/collector/processor/processorprofiles => ../processor/processorprofiles
 
 replace go.opentelemetry.io/collector/connector/connectorprofiles => ../connector/connectorprofiles
+
+replace go.opentelemetry.io/collector/exporter/exporterprofiles => ../exporter/exporterprofiles
