diff --git a/confmap/provider/yamlprovider/provider_test.go b/confmap/provider/yamlprovider/provider_test.go
index 8959bb0de..c56ba8e60 100644
--- a/confmap/provider/yamlprovider/provider_test.go
+++ b/confmap/provider/yamlprovider/provider_test.go
@@ -21,7 +21,7 @@ func TestValidateProviderScheme(t *testing.T) {
 func TestEmpty(t *testing.T) {
 	sp := createProvider()
 	_, err := sp.Retrieve(context.Background(), "", nil)
-	assert.Error(t, err)
+	require.Error(t, err)
 	assert.NoError(t, sp.Shutdown(context.Background()))
 }
 
@@ -39,9 +39,9 @@ func TestInvalidYAML(t *testing.T) {
 func TestOneValue(t *testing.T) {
 	sp := createProvider()
 	ret, err := sp.Retrieve(context.Background(), "yaml:processors::batch::timeout: 2s", nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	retMap, err := ret.AsConf()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.Equal(t, map[string]any{
 		"processors": map[string]any{
 			"batch": map[string]any{
@@ -55,9 +55,9 @@ func TestOneValue(t *testing.T) {
 func TestNamedComponent(t *testing.T) {
 	sp := createProvider()
 	ret, err := sp.Retrieve(context.Background(), "yaml:processors::batch/foo::timeout: 3s", nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	retMap, err := ret.AsConf()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.Equal(t, map[string]any{
 		"processors": map[string]any{
 			"batch/foo": map[string]any{
@@ -71,9 +71,9 @@ func TestNamedComponent(t *testing.T) {
 func TestMapEntry(t *testing.T) {
 	sp := createProvider()
 	ret, err := sp.Retrieve(context.Background(), "yaml:processors: {batch/foo::timeout: 3s, batch::timeout: 2s}", nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	retMap, err := ret.AsConf()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.Equal(t, map[string]any{
 		"processors": map[string]any{
 			"batch/foo": map[string]any{
@@ -90,9 +90,9 @@ func TestMapEntry(t *testing.T) {
 func TestArrayEntry(t *testing.T) {
 	sp := createProvider()
 	ret, err := sp.Retrieve(context.Background(), "yaml:service::extensions: [zpages, zpages/foo]", nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	retMap, err := ret.AsConf()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.Equal(t, map[string]any{
 		"service": map[string]any{
 			"extensions": []any{
@@ -107,9 +107,9 @@ func TestArrayEntry(t *testing.T) {
 func TestNewLine(t *testing.T) {
 	sp := createProvider()
 	ret, err := sp.Retrieve(context.Background(), "yaml:processors::batch/foo::timeout: 3s\nprocessors::batch::timeout: 2s", nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	retMap, err := ret.AsConf()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.Equal(t, map[string]any{
 		"processors": map[string]any{
 			"batch/foo": map[string]any{
@@ -126,9 +126,9 @@ func TestNewLine(t *testing.T) {
 func TestDotSeparator(t *testing.T) {
 	sp := createProvider()
 	ret, err := sp.Retrieve(context.Background(), "yaml:processors.batch.timeout: 4s", nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	retMap, err := ret.AsConf()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	assert.Equal(t, map[string]any{"processors.batch.timeout": "4s"}, retMap.ToStringMap())
 	assert.NoError(t, sp.Shutdown(context.Background()))
 }
