diff --git a/pdata/internal/generated_proto_span.go b/pdata/internal/generated_proto_span.go
index 94f5adbd8..4df8e982e 100644
--- a/pdata/internal/generated_proto_span.go
+++ b/pdata/internal/generated_proto_span.go
@@ -18,22 +18,22 @@ import (
 // Span represents a single operation within a trace.
 // See Span definition in OTLP: https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto
 type Span struct {
-	TraceId                TraceID
-	SpanId                 SpanID
 	TraceState             string
-	ParentSpanId           SpanID
-	Flags                  uint32
 	Name                   string
-	Kind                   SpanKind
+	Attributes             []KeyValue
+	Events                 []*SpanEvent
+	Links                  []*SpanLink
+	Status                 Status
 	StartTimeUnixNano      uint64
 	EndTimeUnixNano        uint64
-	Attributes             []KeyValue
+	Flags                  uint32
+	Kind                   SpanKind
 	DroppedAttributesCount uint32
-	Events                 []*SpanEvent
 	DroppedEventsCount     uint32
-	Links                  []*SpanLink
 	DroppedLinksCount      uint32
-	Status                 Status
+	TraceId                TraceID
+	SpanId                 SpanID
+	ParentSpanId           SpanID
 }
 
 var (
