diff --git a/pdata/internal/generated_proto_mapping.go b/pdata/internal/generated_proto_mapping.go
index e1772794e..8a91a0762 100644
--- a/pdata/internal/generated_proto_mapping.go
+++ b/pdata/internal/generated_proto_mapping.go
@@ -16,11 +16,11 @@ import (
 
 // Mapping describes the mapping of a binary in memory, including its address range, file offset, and metadata like build ID
 type Mapping struct {
+	AttributeIndices []int32
 	MemoryStart      uint64
 	MemoryLimit      uint64
 	FileOffset       uint64
 	FilenameStrindex int32
-	AttributeIndices []int32
 }
 
 var (
