diff --git a/component/telemetry.go b/component/telemetry.go
index 1f4e011e7..d8b6b134c 100644
--- a/component/telemetry.go
+++ b/component/telemetry.go
@@ -45,5 +45,5 @@ type TelemetrySettings struct {
 	//   - Calling this method before component startup
 	//
 	// If the API is being used properly, these errors are safe to ignore.
-	ReportComponentStatus StatusFunc
+	ReportComponentStatus func(*StatusEvent) error
 }
