diff --git a/config/confighttp/go.mod b/config/confighttp/go.mod
index 541aa6f3e..8556c2ee9 100644
--- a/config/confighttp/go.mod
+++ b/config/confighttp/go.mod
@@ -5,7 +5,7 @@ go 1.21.0
 require (
 	github.com/golang/snappy v0.0.4
 	github.com/klauspost/compress v1.17.9
-	github.com/rs/cors v1.10.1
+	github.com/rs/cors v1.11.0
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/collector v0.104.0
 	go.opentelemetry.io/collector/component v0.104.0
