diff --git a/receiver/xreceiver/go.mod b/receiver/xreceiver/go.mod
index b2441ddf6..23d916b2c 100644
--- a/receiver/xreceiver/go.mod
+++ b/receiver/xreceiver/go.mod
@@ -3,7 +3,7 @@ module go.opentelemetry.io/collector/receiver/xreceiver
 go 1.24
 
 require (
-	github.com/stretchr/testify v1.11.0
+	github.com/stretchr/testify v1.10.0
 	go.opentelemetry.io/collector/component v1.39.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.133.0
 	go.opentelemetry.io/collector/consumer/xconsumer v0.133.0
