diff --git a/cmd/otelcorecol/go.mod b/cmd/otelcorecol/go.mod
index 8ac6011ec..3c3cc55bc 100644
--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -6,23 +6,23 @@ go 1.20
 
 require (
 	github.com/stretchr/testify v1.8.4
-	go.opentelemetry.io/collector/component v0.88.0
-	go.opentelemetry.io/collector/connector v0.88.0
-	go.opentelemetry.io/collector/connector/forwardconnector v0.88.0
-	go.opentelemetry.io/collector/exporter v0.88.0
-	go.opentelemetry.io/collector/exporter/debugexporter v0.88.0
-	go.opentelemetry.io/collector/exporter/loggingexporter v0.88.0
-	go.opentelemetry.io/collector/exporter/otlpexporter v0.88.0
-	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.88.0
-	go.opentelemetry.io/collector/extension v0.88.0
-	go.opentelemetry.io/collector/extension/ballastextension v0.88.0
-	go.opentelemetry.io/collector/extension/zpagesextension v0.88.0
-	go.opentelemetry.io/collector/otelcol v0.88.0
-	go.opentelemetry.io/collector/processor v0.88.0
-	go.opentelemetry.io/collector/processor/batchprocessor v0.88.0
-	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.88.0
-	go.opentelemetry.io/collector/receiver v0.88.0
-	go.opentelemetry.io/collector/receiver/otlpreceiver v0.88.0
+	go.opentelemetry.io/collector/component v0.89.0
+	go.opentelemetry.io/collector/connector v0.89.0
+	go.opentelemetry.io/collector/connector/forwardconnector v0.89.0
+	go.opentelemetry.io/collector/exporter v0.89.0
+	go.opentelemetry.io/collector/exporter/debugexporter v0.89.0
+	go.opentelemetry.io/collector/exporter/loggingexporter v0.89.0
+	go.opentelemetry.io/collector/exporter/otlpexporter v0.89.0
+	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.89.0
+	go.opentelemetry.io/collector/extension v0.89.0
+	go.opentelemetry.io/collector/extension/ballastextension v0.89.0
+	go.opentelemetry.io/collector/extension/zpagesextension v0.89.0
+	go.opentelemetry.io/collector/otelcol v0.89.0
+	go.opentelemetry.io/collector/processor v0.89.0
+	go.opentelemetry.io/collector/processor/batchprocessor v0.89.0
+	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.89.0
+	go.opentelemetry.io/collector/receiver v0.89.0
+	go.opentelemetry.io/collector/receiver/otlpreceiver v0.89.0
 	golang.org/x/sys v0.14.0
 )
 
@@ -77,23 +77,23 @@ require (
 	github.com/tklauser/numcpus v0.6.1 // indirect
 	github.com/yusufpapurcu/wmi v1.2.3 // indirect
 	go.opencensus.io v0.24.0 // indirect
-	go.opentelemetry.io/collector v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/configauth v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/configcompression v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/configgrpc v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/confighttp v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/confignet v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/configopaque v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/configtelemetry v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/configtls v0.88.0 // indirect
-	go.opentelemetry.io/collector/config/internal v0.88.0 // indirect
-	go.opentelemetry.io/collector/confmap v0.88.0 // indirect
-	go.opentelemetry.io/collector/consumer v0.88.0 // indirect
-	go.opentelemetry.io/collector/extension/auth v0.88.0 // indirect
-	go.opentelemetry.io/collector/featuregate v1.0.0-rcv0017 // indirect
-	go.opentelemetry.io/collector/pdata v1.0.0-rcv0017 // indirect
-	go.opentelemetry.io/collector/semconv v0.88.0 // indirect
-	go.opentelemetry.io/collector/service v0.88.0 // indirect
+	go.opentelemetry.io/collector v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/configauth v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/configcompression v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/configgrpc v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/confighttp v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/confignet v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/configopaque v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/configtelemetry v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/configtls v0.89.0 // indirect
+	go.opentelemetry.io/collector/config/internal v0.89.0 // indirect
+	go.opentelemetry.io/collector/confmap v0.89.0 // indirect
+	go.opentelemetry.io/collector/consumer v0.89.0 // indirect
+	go.opentelemetry.io/collector/extension/auth v0.89.0 // indirect
+	go.opentelemetry.io/collector/featuregate v1.0.0-rcv0018 // indirect
+	go.opentelemetry.io/collector/pdata v1.0.0-rcv0018 // indirect
+	go.opentelemetry.io/collector/semconv v0.89.0 // indirect
+	go.opentelemetry.io/collector/service v0.89.0 // indirect
 	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.46.0 // indirect
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.45.0 // indirect
 	go.opentelemetry.io/contrib/propagators/b3 v1.20.0 // indirect
