diff --git a/connector/connectorprofiles/go.mod b/connector/connectorprofiles/go.mod
index be353d2b8..e98d1fd94 100644
--- a/connector/connectorprofiles/go.mod
+++ b/connector/connectorprofiles/go.mod
@@ -7,7 +7,7 @@ require (
 	go.opentelemetry.io/collector v0.111.0
 	go.opentelemetry.io/collector/component v0.111.0
 	go.opentelemetry.io/collector/connector v0.111.0
-	go.opentelemetry.io/collector/consumer v0.111.1-0.20241022132505-11767fed9dfe
+	go.opentelemetry.io/collector/consumer v0.111.1-0.20241022144230-02ceda1496be
 	go.opentelemetry.io/collector/consumer/consumerprofiles v0.111.0
 	go.opentelemetry.io/collector/consumer/consumertest v0.111.0
 	go.opentelemetry.io/collector/pdata/pprofile v0.111.0
