diff --git a/cmd/mdatagen/internal/loader.go b/cmd/mdatagen/internal/loader.go
index 028964605..bc6d29a53 100644
--- a/cmd/mdatagen/internal/loader.go
+++ b/cmd/mdatagen/internal/loader.go
@@ -5,6 +5,8 @@ package internal // import "go.opentelemetry.io/collector/cmd/mdatagen/internal"
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"os/exec"
 	"path/filepath"
 	"strings"
@@ -42,7 +44,7 @@ func LoadMetadata(filePath string) (Metadata, error) {
 		return md, err
 	}
 	if md.ScopeName == "" {
-		md.ScopeName, err = packageName()
+		md.ScopeName, err = packageName(filepath.Dir(filePath))
 		if err != nil {
 			return md, err
 		}
@@ -80,11 +82,17 @@ func shortFolderName(filePath string) string {
 	return parentFolder
 }
 
-func packageName() (string, error) {
+func packageName(filePath string) (string, error) {
 	cmd := exec.Command("go", "list", "-f", "{{.ImportPath}}")
+	cmd.Dir = filePath
 	output, err := cmd.Output()
 	if err != nil {
-		return "", err
+		var ee *exec.ExitError
+		if errors.As(err, &ee) {
+			return "", fmt.Errorf("unable to determine package name: %v failed: (stderr) %v", cmd.Args, string(ee.Stderr))
+		}
+
+		return "", fmt.Errorf("unable to determine package name: %v failed: %v %w", cmd.Args, string(output), err)
 	}
 	return strings.TrimSpace(string(output)), nil
 }
