diff --git a/receiver/opencensusreceiver/opencensus_test.go b/receiver/opencensusreceiver/opencensus_test.go
index e3060ecdc..491db1370 100644
--- a/receiver/opencensusreceiver/opencensus_test.go
+++ b/receiver/opencensusreceiver/opencensus_test.go
@@ -44,6 +44,9 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configtls"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/internal/internalconsumertest"
 	"go.opentelemetry.io/collector/obsreport/obsreporttest"
@@ -613,3 +616,29 @@ func TestOCReceiverMetrics_HandleNextConsumerResponse(t *testing.T) {
 		}
 	}
 }
+
+func TestInvalidTLSCredentials(t *testing.T) {
+	cfg := Config{
+		ReceiverSettings: config.ReceiverSettings{
+			NameVal: "IncorrectTLS",
+		},
+		GRPCServerSettings: configgrpc.GRPCServerSettings{
+			TLSSetting: &configtls.TLSServerSetting{
+				TLSSetting: configtls.TLSSetting{
+					CertFile: "willfail",
+				},
+			},
+		},
+	}
+	opt := cfg.buildOptions()
+	assert.NotNil(t, opt)
+
+	addr := testutil.GetAvailableLocalAddress(t)
+	ocr, err := newOpenCensusReceiver("invalidtls", "tcp", addr, nil, nil, opt...)
+	assert.NoError(t, err)
+	assert.NotNil(t, ocr)
+
+	srv, err := ocr.grpcServer(componenttest.NewNopHost())
+	assert.EqualError(t, err, `failed to load TLS config: for auth via TLS, either both certificate and key must be supplied, or neither`)
+	assert.Nil(t, srv)
+}
