diff --git a/service/internal/graph/graph_test.go b/service/internal/graph/graph_test.go
index fcc2ea6cf..8b98d020f 100644
--- a/service/internal/graph/graph_test.go
+++ b/service/internal/graph/graph_test.go
@@ -16,6 +16,7 @@ import (
 	"gonum.org/v1/gonum/graph/simple"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/connector/connectortest"
@@ -145,16 +146,16 @@ func TestGraphStartStop(t *testing.T) {
 
 			pg := &Graph{componentGraph: simple.NewDirectedGraph()}
 			pg.telemetry = componenttest.NewNopTelemetrySettings()
-			pg.instanceIDs = make(map[int64]*component.InstanceID)
+			pg.instanceIDs = make(map[int64]*componentstatus.InstanceID)
 
 			for _, edge := range tt.edges {
 				f, t := &testNode{id: edge[0]}, &testNode{id: edge[1]}
-				pg.instanceIDs[f.ID()] = &component.InstanceID{}
-				pg.instanceIDs[t.ID()] = &component.InstanceID{}
+				pg.instanceIDs[f.ID()] = &componentstatus.InstanceID{}
+				pg.instanceIDs[t.ID()] = &componentstatus.InstanceID{}
 				pg.componentGraph.SetEdge(simple.Edge{F: f, T: t})
 			}
 
-			require.NoError(t, pg.StartAll(ctx, componenttest.NewNopHost(), statustest.NewNopStatusReporter()))
+			require.NoError(t, pg.StartAll(ctx, &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}))
 			for _, edge := range tt.edges {
 				assert.Greater(t, ctx.order[edge[0]], ctx.order[edge[1]])
 			}
@@ -176,7 +177,7 @@ func TestGraphStartStopCycle(t *testing.T) {
 	c1 := &testNode{id: component.MustNewIDWithName("c", "1")}
 	e1 := &testNode{id: component.MustNewIDWithName("e", "1")}
 
-	pg.instanceIDs = map[int64]*component.InstanceID{
+	pg.instanceIDs = map[int64]*componentstatus.InstanceID{
 		r1.ID(): {},
 		p1.ID(): {},
 		c1.ID(): {},
@@ -188,7 +189,7 @@ func TestGraphStartStopCycle(t *testing.T) {
 	pg.componentGraph.SetEdge(simple.Edge{F: c1, T: e1})
 	pg.componentGraph.SetEdge(simple.Edge{F: c1, T: p1}) // loop back
 
-	err := pg.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter())
+	err := pg.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})})
 	assert.Error(t, err)
 	assert.Contains(t, err.Error(), `topo: no topological ordering: cyclic components`)
 
@@ -208,7 +209,7 @@ func TestGraphStartStopComponentError(t *testing.T) {
 		id:          component.MustNewIDWithName("e", "1"),
 		shutdownErr: errors.New("bar"),
 	}
-	pg.instanceIDs = map[int64]*component.InstanceID{
+	pg.instanceIDs = map[int64]*componentstatus.InstanceID{
 		r1.ID(): {},
 		e1.ID(): {},
 	}
@@ -216,7 +217,7 @@ func TestGraphStartStopComponentError(t *testing.T) {
 		F: r1,
 		T: e1,
 	})
-	assert.EqualError(t, pg.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter()), "foo")
+	assert.EqualError(t, pg.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}), "foo")
 	assert.EqualError(t, pg.ShutdownAll(context.Background(), statustest.NewNopStatusReporter()), "bar")
 }
 
@@ -224,7 +225,7 @@ func TestConnectorPipelinesGraph(t *testing.T) {
 	tests := []struct {
 		name                string
 		pipelineConfigs     pipelines.Config
-		expectedPerExporter int // requires symmetry in pipelines
+		expectedPerExporter int // requires symmetry in Pipelines
 	}{
 		{
 			name: "pipelines_simple.yaml",
@@ -804,7 +805,7 @@ func TestConnectorPipelinesGraph(t *testing.T) {
 
 			assert.Equal(t, len(test.pipelineConfigs), len(pg.pipelines))
 
-			assert.NoError(t, pg.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter()))
+			assert.NoError(t, pg.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}))
 
 			mutatingPipelines := make(map[component.ID]bool, len(test.pipelineConfigs))
 
@@ -881,7 +882,7 @@ func TestConnectorPipelinesGraph(t *testing.T) {
 				}
 			}
 
-			// Check that connectors are correctly inheriting mutability from downstream pipelines
+			// Check that Connectors are correctly inheriting mutability from downstream Pipelines
 			for expPipelineID, expPipeline := range pg.pipelines {
 				for _, exp := range expPipeline.exporters {
 					expConn, ok := exp.(*connectorNode)
@@ -891,7 +892,7 @@ func TestConnectorPipelinesGraph(t *testing.T) {
 					if expConn.getConsumer().Capabilities().MutatesData {
 						continue
 					}
-					// find all the pipelines of the same type where this connector is a receiver
+					// find all the Pipelines of the same type where this connector is a receiver
 					var inheritMutatesData bool
 					for recPipelineID, recPipeline := range pg.pipelines {
 						if recPipelineID == expPipelineID || recPipelineID.Type() != expPipelineID.Type() {
@@ -909,9 +910,9 @@ func TestConnectorPipelinesGraph(t *testing.T) {
 				}
 			}
 
-			// Push data into the pipelines. The list of receivers is retrieved directly from the overall
-			// component graph because we do not want to duplicate signal inputs to receivers that are
-			// shared between pipelines. The `allReceivers` function also excludes connectors, which we do
+			// Push data into the Pipelines. The list of Receivers is retrieved directly from the overall
+			// component graph because we do not want to duplicate signal inputs to Receivers that are
+			// shared between Pipelines. The `allReceivers` function also excludes Connectors, which we do
 			// not want to directly inject with signals.
 			allReceivers := pg.getReceivers()
 			for _, c := range allReceivers[component.DataTypeTraces] {
@@ -979,8 +980,8 @@ func TestConnectorPipelinesGraph(t *testing.T) {
 				}
 			}
 
-			// Get the list of exporters directly from the overall component graph. Like receivers,
-			// exclude connectors and validate each exporter once regardless of sharing between pipelines.
+			// Get the list of Exporters directly from the overall component graph. Like Receivers,
+			// exclude Connectors and validate each exporter once regardless of sharing between Pipelines.
 			allExporters := pg.GetExporters()
 			for _, e := range allExporters[component.DataTypeTraces] {
 				tracesExporter := e.(*testcomponents.ExampleExporter)
@@ -1142,7 +1143,7 @@ func TestConnectorRouter(t *testing.T) {
 
 	assert.Equal(t, len(set.PipelineConfigs), len(pg.pipelines))
 
-	// Get a handle for the traces receiver and both exporters
+	// Get a handle for the traces receiver and both Exporters
 	tracesReceiver := allReceivers[component.DataTypeTraces][rcvrID].(*testcomponents.ExampleReceiver)
 	tracesRight := allExporters[component.DataTypeTraces][expRightID].(*testcomponents.ExampleExporter)
 	tracesLeft := allExporters[component.DataTypeTraces][expLeftID].(*testcomponents.ExampleExporter)
@@ -1164,7 +1165,7 @@ func TestConnectorRouter(t *testing.T) {
 	assert.Equal(t, 3, len(tracesRight.Traces))
 	assert.Equal(t, 2, len(tracesLeft.Traces))
 
-	// Get a handle for the metrics receiver and both exporters
+	// Get a handle for the metrics receiver and both Exporters
 	metricsReceiver := allReceivers[component.DataTypeMetrics][rcvrID].(*testcomponents.ExampleReceiver)
 	metricsRight := allExporters[component.DataTypeMetrics][expRightID].(*testcomponents.ExampleExporter)
 	metricsLeft := allExporters[component.DataTypeMetrics][expLeftID].(*testcomponents.ExampleExporter)
@@ -1186,7 +1187,7 @@ func TestConnectorRouter(t *testing.T) {
 	assert.Equal(t, 3, len(metricsRight.Metrics))
 	assert.Equal(t, 2, len(metricsLeft.Metrics))
 
-	// Get a handle for the logs receiver and both exporters
+	// Get a handle for the logs receiver and both Exporters
 	logsReceiver := allReceivers[component.DataTypeLogs][rcvrID].(*testcomponents.ExampleReceiver)
 	logsRight := allExporters[component.DataTypeLogs][expRightID].(*testcomponents.ExampleExporter)
 	logsLeft := allExporters[component.DataTypeLogs][expLeftID].(*testcomponents.ExampleExporter)
@@ -2193,7 +2194,7 @@ func TestGraphFailToStartAndShutdown(t *testing.T) {
 			}
 			pipelines, err := Build(context.Background(), set)
 			assert.NoError(t, err)
-			assert.Error(t, pipelines.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter()))
+			assert.Error(t, pipelines.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}))
 			assert.Error(t, pipelines.ShutdownAll(context.Background(), statustest.NewNopStatusReporter()))
 		})
 
@@ -2207,7 +2208,7 @@ func TestGraphFailToStartAndShutdown(t *testing.T) {
 			}
 			pipelines, err := Build(context.Background(), set)
 			assert.NoError(t, err)
-			assert.Error(t, pipelines.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter()))
+			assert.Error(t, pipelines.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}))
 			assert.Error(t, pipelines.ShutdownAll(context.Background(), statustest.NewNopStatusReporter()))
 		})
 
@@ -2221,7 +2222,7 @@ func TestGraphFailToStartAndShutdown(t *testing.T) {
 			}
 			pipelines, err := Build(context.Background(), set)
 			assert.NoError(t, err)
-			assert.Error(t, pipelines.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter()))
+			assert.Error(t, pipelines.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}))
 			assert.Error(t, pipelines.ShutdownAll(context.Background(), statustest.NewNopStatusReporter()))
 		})
 
@@ -2241,7 +2242,7 @@ func TestGraphFailToStartAndShutdown(t *testing.T) {
 				}
 				pipelines, err := Build(context.Background(), set)
 				assert.NoError(t, err)
-				assert.Error(t, pipelines.StartAll(context.Background(), componenttest.NewNopHost(), statustest.NewNopStatusReporter()))
+				assert.Error(t, pipelines.StartAll(context.Background(), &Host{Reporter: status.NewReporter(func(*componentstatus.InstanceID, *componentstatus.Event) {}, func(error) {})}))
 				assert.Error(t, pipelines.ShutdownAll(context.Background(), statustest.NewNopStatusReporter()))
 			})
 		}
@@ -2258,7 +2259,7 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 	eStErr := &testNode{id: component.MustNewIDWithName("e_st_err", "1"), startErr: assert.AnError}
 	eSdErr := &testNode{id: component.MustNewIDWithName("e_sd_err", "1"), shutdownErr: assert.AnError}
 
-	instanceIDs := map[*testNode]*component.InstanceID{
+	instanceIDs := map[*testNode]*componentstatus.InstanceID{
 		rNoErr: {ID: rNoErr.id},
 		rStErr: {ID: rStErr.id},
 		rSdErr: {ID: rSdErr.id},
@@ -2268,7 +2269,7 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 	}
 
 	// compare two maps of status events ignoring timestamp
-	assertEqualStatuses := func(t *testing.T, evMap1, evMap2 map[*component.InstanceID][]*component.StatusEvent) {
+	assertEqualStatuses := func(t *testing.T, evMap1, evMap2 map[*componentstatus.InstanceID][]*componentstatus.Event) {
 		assert.Equal(t, len(evMap1), len(evMap2))
 		for id, evts1 := range evMap1 {
 			evts2 := evMap2[id]
@@ -2286,35 +2287,35 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 	for _, tc := range []struct {
 		name             string
 		edge             [2]*testNode
-		expectedStatuses map[*component.InstanceID][]*component.StatusEvent
+		expectedStatuses map[*componentstatus.InstanceID][]*componentstatus.Event
 		startupErr       error
 		shutdownErr      error
 	}{
 		{
 			name: "successful startup/shutdown",
 			edge: [2]*testNode{rNoErr, eNoErr},
-			expectedStatuses: map[*component.InstanceID][]*component.StatusEvent{
+			expectedStatuses: map[*componentstatus.InstanceID][]*componentstatus.Event{
 				instanceIDs[rNoErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewStatusEvent(component.StatusStopped),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewEvent(componentstatus.StatusStopped),
 				},
 				instanceIDs[eNoErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewStatusEvent(component.StatusStopped),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewEvent(componentstatus.StatusStopped),
 				},
 			},
 		},
 		{
 			name: "early startup error",
 			edge: [2]*testNode{rNoErr, eStErr},
-			expectedStatuses: map[*component.InstanceID][]*component.StatusEvent{
+			expectedStatuses: map[*componentstatus.InstanceID][]*componentstatus.Event{
 				instanceIDs[eStErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewPermanentErrorEvent(assert.AnError),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewPermanentErrorEvent(assert.AnError),
 				},
 			},
 			startupErr: assert.AnError,
@@ -2322,16 +2323,16 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 		{
 			name: "late startup error",
 			edge: [2]*testNode{rStErr, eNoErr},
-			expectedStatuses: map[*component.InstanceID][]*component.StatusEvent{
+			expectedStatuses: map[*componentstatus.InstanceID][]*componentstatus.Event{
 				instanceIDs[rStErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewPermanentErrorEvent(assert.AnError),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewPermanentErrorEvent(assert.AnError),
 				},
 				instanceIDs[eNoErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewStatusEvent(component.StatusStopped),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewEvent(componentstatus.StatusStopped),
 				},
 			},
 			startupErr: assert.AnError,
@@ -2339,18 +2340,18 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 		{
 			name: "early shutdown error",
 			edge: [2]*testNode{rSdErr, eNoErr},
-			expectedStatuses: map[*component.InstanceID][]*component.StatusEvent{
+			expectedStatuses: map[*componentstatus.InstanceID][]*componentstatus.Event{
 				instanceIDs[rSdErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewPermanentErrorEvent(assert.AnError),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewPermanentErrorEvent(assert.AnError),
 				},
 				instanceIDs[eNoErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewStatusEvent(component.StatusStopped),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewEvent(componentstatus.StatusStopped),
 				},
 			},
 			shutdownErr: assert.AnError,
@@ -2358,18 +2359,18 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 		{
 			name: "late shutdown error",
 			edge: [2]*testNode{rNoErr, eSdErr},
-			expectedStatuses: map[*component.InstanceID][]*component.StatusEvent{
+			expectedStatuses: map[*componentstatus.InstanceID][]*componentstatus.Event{
 				instanceIDs[rNoErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewStatusEvent(component.StatusStopped),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewEvent(componentstatus.StatusStopped),
 				},
 				instanceIDs[eSdErr]: {
-					component.NewStatusEvent(component.StatusStarting),
-					component.NewStatusEvent(component.StatusOK),
-					component.NewStatusEvent(component.StatusStopping),
-					component.NewPermanentErrorEvent(assert.AnError),
+					componentstatus.NewEvent(componentstatus.StatusStarting),
+					componentstatus.NewEvent(componentstatus.StatusOK),
+					componentstatus.NewEvent(componentstatus.StatusStopping),
+					componentstatus.NewPermanentErrorEvent(assert.AnError),
 				},
 			},
 			shutdownErr: assert.AnError,
@@ -2379,8 +2380,8 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 			pg := &Graph{componentGraph: simple.NewDirectedGraph()}
 			pg.telemetry = componenttest.NewNopTelemetrySettings()
 
-			actualStatuses := make(map[*component.InstanceID][]*component.StatusEvent)
-			rep := status.NewReporter(func(id *component.InstanceID, ev *component.StatusEvent) {
+			actualStatuses := make(map[*componentstatus.InstanceID][]*componentstatus.Event)
+			rep := status.NewReporter(func(id *componentstatus.InstanceID, ev *componentstatus.Event) {
 				actualStatuses[id] = append(actualStatuses[id], ev)
 			}, func(error) {
 			})
@@ -2388,13 +2389,13 @@ func TestStatusReportedOnStartupShutdown(t *testing.T) {
 			rep.Ready()
 
 			e0, e1 := tc.edge[0], tc.edge[1]
-			pg.instanceIDs = map[int64]*component.InstanceID{
+			pg.instanceIDs = map[int64]*componentstatus.InstanceID{
 				e0.ID(): instanceIDs[e0],
 				e1.ID(): instanceIDs[e1],
 			}
 			pg.componentGraph.SetEdge(simple.Edge{F: e0, T: e1})
 
-			assert.Equal(t, tc.startupErr, pg.StartAll(context.Background(), componenttest.NewNopHost(), rep))
+			assert.Equal(t, tc.startupErr, pg.StartAll(context.Background(), &Host{Reporter: rep}))
 			assert.Equal(t, tc.shutdownErr, pg.ShutdownAll(context.Background(), rep))
 			assertEqualStatuses(t, tc.expectedStatuses, actualStatuses)
 		})
@@ -2424,7 +2425,7 @@ func (g *Graph) getReceivers() map[component.DataType]map[component.ID]component
 //
 // Expect one instance of each receiver and exporter, unless it is a connector.
 //
-// For connectors:
+// For Connectors:
 // - Let E equal the number of pipeline types in which the connector is used as an exporter.
 // - Let R equal the number of pipeline types in which the connector is used as a receiver.
 //
