diff --git a/cmd/occollector/app/collector/telemetry.go b/cmd/occollector/app/collector/telemetry.go
index d6ca87f1c..60e8dca87 100644
--- a/cmd/occollector/app/collector/telemetry.go
+++ b/cmd/occollector/app/collector/telemetry.go
@@ -29,7 +29,7 @@ import (
 	"github.com/census-instrumentation/opencensus-service/internal/collector/processor"
 	"github.com/census-instrumentation/opencensus-service/internal/collector/processor/nodebatcher"
 	"github.com/census-instrumentation/opencensus-service/internal/collector/processor/queued"
-	"github.com/census-instrumentation/opencensus-service/internal/collector/processor/tail_sampling"
+	"github.com/census-instrumentation/opencensus-service/internal/collector/processor/tailsampling"
 	"github.com/census-instrumentation/opencensus-service/internal/collector/telemetry"
 )
 
