diff --git a/exporter/exporterhelper/metrics_batch.go b/exporter/exporterhelper/metrics_batch.go
index 390517c69..8ded9fa36 100644
--- a/exporter/exporterhelper/metrics_batch.go
+++ b/exporter/exporterhelper/metrics_batch.go
@@ -13,7 +13,7 @@ import (
 
 // MergeSplit splits and/or merges the provided metrics request and the current request into one or more requests
 // conforming with the MaxSizeConfig.
-func (req *metricsRequest) MergeSplit(_ context.Context, cfg exporterbatcher.MaxSizeConfig, r2 Request) ([]Request, error) {
+func (req *metricsRequest) MergeSplit(_ context.Context, cfg exporterbatcher.SizeConfig, r2 Request) ([]Request, error) {
 	if r2 != nil {
 		req2, ok := r2.(*metricsRequest)
 		if !ok {
@@ -23,7 +23,7 @@ func (req *metricsRequest) MergeSplit(_ context.Context, cfg exporterbatcher.Max
 	}
 
 	// If no limit we can simply merge the new request into the current and return.
-	if cfg.MaxSizeItems == 0 {
+	if cfg.MaxSize == 0 {
 		return []Request{req}, nil
 	}
 	return req.split(cfg)
@@ -35,10 +35,10 @@ func (req *metricsRequest) mergeTo(dst *metricsRequest) {
 	req.md.ResourceMetrics().MoveAndAppendTo(dst.md.ResourceMetrics())
 }
 
-func (req *metricsRequest) split(cfg exporterbatcher.MaxSizeConfig) ([]Request, error) {
+func (req *metricsRequest) split(cfg exporterbatcher.SizeConfig) ([]Request, error) {
 	var res []Request
-	for req.ItemsCount() > cfg.MaxSizeItems {
-		md := extractMetrics(req.md, cfg.MaxSizeItems)
+	for req.ItemsCount() > cfg.MaxSize {
+		md := extractMetrics(req.md, cfg.MaxSize)
 		size := md.DataPointCount()
 		req.setCachedItemsCount(req.ItemsCount() - size)
 		res = append(res, &metricsRequest{md: md, pusher: req.pusher, cachedItemsCount: size})
