diff --git a/exporter/internal/queue/default_batcher_test.go b/exporter/internal/queue/default_batcher_test.go
index e93746f83..3f2c6253e 100644
--- a/exporter/internal/queue/default_batcher_test.go
+++ b/exporter/internal/queue/default_batcher_test.go
@@ -15,7 +15,10 @@ import (
 
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+	"go.opentelemetry.io/collector/exporter/exporterqueue"
+	"go.opentelemetry.io/collector/exporter/exportertest"
 	"go.opentelemetry.io/collector/exporter/internal"
+	"go.opentelemetry.io/collector/pipeline"
 )
 
 func TestDefaultBatcher_NoSplit_MinThresholdZero_TimeoutDisabled(t *testing.T) {
@@ -45,11 +48,13 @@ func TestDefaultBatcher_NoSplit_MinThresholdZero_TimeoutDisabled(t *testing.T) {
 				MinSizeItems: 0,
 			}
 
-			q := NewBoundedMemoryQueue[internal.Request](
-				MemoryQueueSettings[internal.Request]{
-					Sizer:    &RequestSizer[internal.Request]{},
-					Capacity: 10,
-				})
+			q := exporterqueue.NewMemoryQueueFactory[internal.Request]()(
+				context.Background(),
+				exporterqueue.Settings{
+					Signal:           pipeline.SignalTraces,
+					ExporterSettings: exportertest.NewNopSettings(),
+				},
+				exporterqueue.NewDefaultConfig())
 
 			ba, err := NewBatcher(cfg, q,
 				func(ctx context.Context, req internal.Request) error { return req.Export(ctx) },
@@ -105,11 +110,13 @@ func TestDefaultBatcher_NoSplit_TimeoutDisabled(t *testing.T) {
 				MinSizeItems: 10,
 			}
 
-			q := NewBoundedMemoryQueue[internal.Request](
-				MemoryQueueSettings[internal.Request]{
-					Sizer:    &RequestSizer[internal.Request]{},
-					Capacity: 10,
-				})
+			q := exporterqueue.NewMemoryQueueFactory[internal.Request]()(
+				context.Background(),
+				exporterqueue.Settings{
+					Signal:           pipeline.SignalTraces,
+					ExporterSettings: exportertest.NewNopSettings(),
+				},
+				exporterqueue.NewDefaultConfig())
 
 			ba, err := NewBatcher(cfg, q,
 				func(ctx context.Context, req internal.Request) error { return req.Export(ctx) },
@@ -175,11 +182,13 @@ func TestDefaultBatcher_NoSplit_WithTimeout(t *testing.T) {
 				MinSizeItems: 100,
 			}
 
-			q := NewBoundedMemoryQueue[internal.Request](
-				MemoryQueueSettings[internal.Request]{
-					Sizer:    &RequestSizer[internal.Request]{},
-					Capacity: 10,
-				})
+			q := exporterqueue.NewMemoryQueueFactory[internal.Request]()(
+				context.Background(),
+				exporterqueue.Settings{
+					Signal:           pipeline.SignalTraces,
+					ExporterSettings: exportertest.NewNopSettings(),
+				},
+				exporterqueue.NewDefaultConfig())
 
 			ba, err := NewBatcher(cfg, q,
 				func(ctx context.Context, req internal.Request) error { return req.Export(ctx) },
@@ -245,11 +254,13 @@ func TestDefaultBatcher_Split_TimeoutDisabled(t *testing.T) {
 				MaxSizeItems: 100,
 			}
 
-			q := NewBoundedMemoryQueue[internal.Request](
-				MemoryQueueSettings[internal.Request]{
-					Sizer:    &RequestSizer[internal.Request]{},
-					Capacity: 10,
-				})
+			q := exporterqueue.NewMemoryQueueFactory[internal.Request]()(
+				context.Background(),
+				exporterqueue.Settings{
+					Signal:           pipeline.SignalTraces,
+					ExporterSettings: exportertest.NewNopSettings(),
+				},
+				exporterqueue.NewDefaultConfig())
 
 			ba, err := NewBatcher(cfg, q,
 				func(ctx context.Context, req internal.Request) error { return req.Export(ctx) },
@@ -292,11 +303,13 @@ func TestDefaultBatcher_Shutdown(t *testing.T) {
 	batchCfg.MinSizeItems = 10
 	batchCfg.FlushTimeout = 100 * time.Second
 
-	q := NewBoundedMemoryQueue[internal.Request](
-		MemoryQueueSettings[internal.Request]{
-			Sizer:    &RequestSizer[internal.Request]{},
-			Capacity: 10,
-		})
+	q := exporterqueue.NewMemoryQueueFactory[internal.Request]()(
+		context.Background(),
+		exporterqueue.Settings{
+			Signal:           pipeline.SignalTraces,
+			ExporterSettings: exportertest.NewNopSettings(),
+		},
+		exporterqueue.NewDefaultConfig())
 
 	ba, err := NewBatcher(batchCfg, q,
 		func(ctx context.Context, req internal.Request) error { return req.Export(ctx) },
