diff --git a/consumer/pdata/metric_test.go b/consumer/pdata/metric_test.go
index bc3d6bb5e..3d6f7dcd6 100644
--- a/consumer/pdata/metric_test.go
+++ b/consumer/pdata/metric_test.go
@@ -332,6 +332,45 @@ func TestMetricAndDataPointCountWithNil(t *testing.T) {
 
 }
 
+func TestMetricAndDataPointCountWithNilDataPoints(t *testing.T) {
+	metrics := NewMetrics()
+	rm := NewResourceMetrics()
+	rm.InitEmpty()
+	metrics.ResourceMetrics().Append(rm)
+	ilm := NewInstrumentationLibraryMetrics()
+	ilm.InitEmpty()
+	rm.InstrumentationLibraryMetrics().Append(ilm)
+	intGauge := NewMetric()
+	intGauge.InitEmpty()
+	ilm.Metrics().Append(intGauge)
+	intGauge.SetDataType(MetricDataTypeIntGauge)
+	doubleGauge := NewMetric()
+	doubleGauge.InitEmpty()
+	ilm.Metrics().Append(doubleGauge)
+	doubleGauge.SetDataType(MetricDataTypeDoubleGauge)
+	intHistogram := NewMetric()
+	intHistogram.InitEmpty()
+	ilm.Metrics().Append(intHistogram)
+	intHistogram.SetDataType(MetricDataTypeIntHistogram)
+	doubleHistogram := NewMetric()
+	doubleHistogram.InitEmpty()
+	ilm.Metrics().Append(doubleHistogram)
+	doubleHistogram.SetDataType(MetricDataTypeDoubleHistogram)
+	intSum := NewMetric()
+	intSum.InitEmpty()
+	ilm.Metrics().Append(intSum)
+	intSum.SetDataType(MetricDataTypeIntSum)
+	doubleSum := NewMetric()
+	doubleSum.InitEmpty()
+	ilm.Metrics().Append(doubleSum)
+	doubleSum.SetDataType(MetricDataTypeDoubleSum)
+
+	ms, dps := metrics.MetricAndDataPointCount()
+
+	assert.EqualValues(t, 6, ms)
+	assert.EqualValues(t, 0, dps)
+}
+
 func TestOtlpToInternalReadOnly(t *testing.T) {
 	metricData := MetricsFromOtlp([]*otlpmetrics.ResourceMetrics{
 		{
