diff --git a/pdata/ptrace/generated_scopespans.go b/pdata/ptrace/generated_scopespans.go
index 03188d2ed..51d492581 100644
--- a/pdata/ptrace/generated_scopespans.go
+++ b/pdata/ptrace/generated_scopespans.go
@@ -33,7 +33,7 @@ func newScopeSpans(orig *otlptrace.ScopeSpans, state *internal.State) ScopeSpans
 // This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
 // OR directly access the member if this is embedded in another struct.
 func NewScopeSpans() ScopeSpans {
-	return newScopeSpans(internal.NewOrigPtrScopeSpans(), internal.NewState())
+	return newScopeSpans(internal.NewOrigScopeSpans(), internal.NewState())
 }
 
 // MoveTo moves all properties from the current struct overriding the destination and
