diff --git a/otelcol/unmarshaler.go b/otelcol/unmarshaler.go
index ed97d975e..d38e140b0 100644
--- a/otelcol/unmarshaler.go
+++ b/otelcol/unmarshaler.go
@@ -27,11 +27,9 @@ type configSettings struct {
 // unmarshal the configSettings from a confmap.Conf.
 // After the config is unmarshalled, `Validate()` must be called to validate.
 func unmarshal(v *confmap.Conf, factories Factories) (*configSettings, error) {
-	// TODO remove these params once SDK and resource creation are encapsulated
-	// within the otelconftelemetry factory. They are not used when creating
-	// the default config.
+	// TODO: inject the telemetry factory through factories, once available.
 	// See https://github.com/open-telemetry/opentelemetry-collector/issues/4970
-	telFactory := otelconftelemetry.NewFactory(nil, nil)
+	telFactory := otelconftelemetry.NewFactory()
 	defaultTelConfig := *telFactory.CreateDefaultConfig().(*otelconftelemetry.Config)
 
 	// Unmarshal top level sections and validate.
