diff --git a/receiver/zipkinreceiver/zipkinscribereceiver/scribe_receiver.go b/receiver/zipkinreceiver/zipkinscribereceiver/scribe_receiver.go
index bfdd0ffc9..17bd51dea 100644
--- a/receiver/zipkinreceiver/zipkinscribereceiver/scribe_receiver.go
+++ b/receiver/zipkinreceiver/zipkinscribereceiver/scribe_receiver.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -26,10 +26,10 @@ import (
 	"github.com/jaegertracing/jaeger/thrift-gen/zipkincore"
 	"github.com/omnition/scribe-go/if/scribe/gen-go/scribe"
 
-	"github.com/census-instrumentation/opencensus-service/consumer"
-	"github.com/census-instrumentation/opencensus-service/observability"
-	"github.com/census-instrumentation/opencensus-service/receiver"
-	zipkintranslator "github.com/census-instrumentation/opencensus-service/translator/trace/zipkin"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/observability"
+	"github.com/open-telemetry/opentelemetry-service/receiver"
+	zipkintranslator "github.com/open-telemetry/opentelemetry-service/translator/trace/zipkin"
 )
 
 var (
