diff --git a/featuregate/registry.go b/featuregate/registry.go
index b2dc2e19c..69486d623 100644
--- a/featuregate/registry.go
+++ b/featuregate/registry.go
@@ -200,7 +200,7 @@ func (r *Registry) Set(id string, enabled bool) error {
 // VisitAll visits all the gates in lexicographical order, calling fn for each.
 func (r *Registry) VisitAll(fn func(*Gate)) {
 	var gates []*Gate
-	r.gates.Range(func(key, value any) bool {
+	r.gates.Range(func(_, value any) bool {
 		gates = append(gates, value.(*Gate))
 		return true
 	})
