diff --git a/otelcol/config.go b/otelcol/config.go
index 2bfd863e2..fa2a624f2 100644
--- a/otelcol/config.go
+++ b/otelcol/config.go
@@ -9,6 +9,7 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/service"
+	"go.opentelemetry.io/collector/service/pipelines"
 )
 
 var (
@@ -50,13 +51,13 @@ func (cfg *Config) Validate() error {
 
 	// Currently, there is no default receiver enabled.
 	// The configuration must specify at least one receiver to be valid.
-	if len(cfg.Receivers) == 0 {
+	if !pipelines.AllowNoPipelines.IsEnabled() && len(cfg.Receivers) == 0 {
 		return errMissingReceivers
 	}
 
 	// Currently, there is no default exporter enabled.
 	// The configuration must specify at least one exporter to be valid.
-	if len(cfg.Exporters) == 0 {
+	if !pipelines.AllowNoPipelines.IsEnabled() && len(cfg.Exporters) == 0 {
 		return errMissingExporters
 	}
 
