diff --git a/service/telemetry/logger.go b/service/telemetry/logger.go
index 09aa82a92..fa8bccd82 100644
--- a/service/telemetry/logger.go
+++ b/service/telemetry/logger.go
@@ -4,13 +4,11 @@
 package telemetry // import "go.opentelemetry.io/collector/service/telemetry"
 
 import (
-	"go.opentelemetry.io/contrib/bridges/otelzap"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/log"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 
-	"go.opentelemetry.io/collector/internal/telemetry"
 	"go.opentelemetry.io/collector/internal/telemetry/componentattribute"
 )
 
@@ -43,54 +41,29 @@ func newLogger(set Settings, cfg Config) (*zap.Logger, log.LoggerProvider, error
 
 	var lp log.LoggerProvider
 
-	if telemetry.NewPipelineTelemetryGate.IsEnabled() {
-		logger = logger.WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
-			core = componentattribute.NewConsoleCoreWithAttributes(core, attribute.NewSet())
+	logger = logger.WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+		core = componentattribute.NewConsoleCoreWithAttributes(core, attribute.NewSet())
 
-			if len(cfg.Logs.Processors) > 0 && set.SDK != nil {
-				lp = set.SDK.LoggerProvider()
-
-				core = componentattribute.NewOTelTeeCoreWithAttributes(
-					core,
-					lp,
-					"go.opentelemetry.io/collector/service/telemetry",
-					cfg.Logs.Level,
-					attribute.NewSet(),
-				)
-			}
-
-			if cfg.Logs.Sampling != nil && cfg.Logs.Sampling.Enabled {
-				core = componentattribute.NewWrapperCoreWithAttributes(core, func(c zapcore.Core) zapcore.Core {
-					return newSampledCore(c, cfg.Logs.Sampling)
-				})
-			}
-
-			return core
-		}))
-	} else {
 		if len(cfg.Logs.Processors) > 0 && set.SDK != nil {
 			lp = set.SDK.LoggerProvider()
 
-			logger = logger.WithOptions(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
-				core, err := zapcore.NewIncreaseLevelCore(zapcore.NewTee(
-					c,
-					otelzap.NewCore("go.opentelemetry.io/collector/service/telemetry",
-						otelzap.WithLoggerProvider(lp),
-					),
-				), zap.NewAtomicLevelAt(cfg.Logs.Level))
-				if err != nil {
-					panic(err)
-				}
-				return core
-			}))
+			core = componentattribute.NewOTelTeeCoreWithAttributes(
+				core,
+				lp,
+				"go.opentelemetry.io/collector/service/telemetry",
+				cfg.Logs.Level,
+				attribute.NewSet(),
+			)
 		}
 
 		if cfg.Logs.Sampling != nil && cfg.Logs.Sampling.Enabled {
-			logger = logger.WithOptions(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
+			core = componentattribute.NewWrapperCoreWithAttributes(core, func(c zapcore.Core) zapcore.Core {
 				return newSampledCore(c, cfg.Logs.Sampling)
-			}))
+			})
 		}
-	}
+
+		return core
+	}))
 
 	return logger, lp, nil
 }
