diff --git a/cmd/builder/internal/builder/config.go b/cmd/builder/internal/builder/config.go
index 5c3c533e4..90b5eacd0 100644
--- a/cmd/builder/internal/builder/config.go
+++ b/cmd/builder/internal/builder/config.go
@@ -26,7 +26,7 @@ import (
 	"go.uber.org/zap"
 )
 
-const defaultOtelColVersion = "0.64.0"
+const defaultOtelColVersion = "0.64.1"
 
 // ErrInvalidGoMod indicates an invalid gomod
 var ErrInvalidGoMod = errors.New("invalid gomod specification for module")
