diff --git a/processor/cloningfanoutconnector.go b/processor/cloningfanoutconnector.go
index 68f060575..afed82a53 100644
--- a/processor/cloningfanoutconnector.go
+++ b/processor/cloningfanoutconnector.go
@@ -28,6 +28,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/consumer/converter"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/pdatautil"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
 )
 
 // This file contains implementations of cloning Trace/Metrics connectors
@@ -215,6 +216,39 @@ func (tfc traceCloningFanOutConnector) ConsumeTraces(ctx context.Context, td pda
 	return componenterror.CombineErrors(errs)
 }
 
+// NewLogCloningFanOutConnector wraps multiple trace consumers in a single one.
+func NewLogCloningFanOutConnector(lcs []consumer.LogConsumer) consumer.LogConsumer {
+	return LogCloningFanOutConnector(lcs)
+}
+
+type LogCloningFanOutConnector []consumer.LogConsumer
+
+var _ consumer.LogConsumer = (*LogCloningFanOutConnector)(nil)
+
+// ConsumeLogs exports the span data to all  consumers wrapped by the current one.
+func (lfc LogCloningFanOutConnector) ConsumeLogs(ctx context.Context, ld data.Logs) error {
+	var errs []error
+
+	// Fan out to first len-1 consumers.
+	for i := 0; i < len(lfc)-1; i++ {
+		// Create a clone of data. We need to clone because consumers may modify the data.
+		clone := ld.Clone()
+		if err := lfc[i].ConsumeLogs(ctx, clone); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(lfc) > 0 {
+		// Give the original data to the last consumer.
+		lastTc := lfc[len(lfc)-1]
+		if err := lastTc.ConsumeLogs(ctx, ld); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return componenterror.CombineErrors(errs)
+}
+
 func cloneTraceDataOld(td consumerdata.TraceData) consumerdata.TraceData {
 	clone := consumerdata.TraceData{
 		SourceFormat: td.SourceFormat,
