diff --git a/config/config.go b/config/config.go
index fa314e721..49f8bd9c3 100644
--- a/config/config.go
+++ b/config/config.go
@@ -781,14 +781,8 @@ func validatePipelineProcessors(
 	pipeline *configmodels.Pipeline,
 	logger *zap.Logger,
 ) error {
-	if pipeline.InputType == configmodels.TracesDataType {
-		// Traces pipeline must have at least one processor.
-		if len(pipeline.Processors) == 0 {
-			return &configError{
-				code: errPipelineMustHaveProcessors,
-				msg:  fmt.Sprintf("pipeline %q must have at least one processor", pipeline.Name),
-			}
-		}
+	if len(pipeline.Processors) == 0 {
+		return nil
 	}
 
 	// Validate pipeline processor name references
