diff --git a/internal/factories/factories_test.go b/internal/factories/factories_test.go
index 1c68b3911..8b73a5bb5 100644
--- a/internal/factories/factories_test.go
+++ b/internal/factories/factories_test.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenCensus Authors
+// Copyright 2019, OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -18,12 +18,12 @@ import (
 	"context"
 	"testing"
 
-	"github.com/census-instrumentation/opencensus-service/processor"
+	"github.com/open-telemetry/opentelemetry-service/processor"
 
-	"github.com/census-instrumentation/opencensus-service/consumer"
-	"github.com/census-instrumentation/opencensus-service/receiver"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/receiver"
 
-	"github.com/census-instrumentation/opencensus-service/internal/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/internal/configmodels"
 )
 
 type ExampleReceiverFactory struct {
