diff --git a/service/internal/builder/extensions_builder.go b/service/internal/builder/extensions_builder.go
index cf610ceee..096677495 100644
--- a/service/internal/builder/extensions_builder.go
+++ b/service/internal/builder/extensions_builder.go
@@ -132,11 +132,13 @@ func BuildExtensions(
 		}
 
 		set := component.ExtensionCreateSettings{
-			Logger: logger.With(
-				zap.String(zapKindKey, zapKindExtension),
-				zap.String(zapNameKey, extID.String())),
-			TracerProvider: tracerProvider,
-			BuildInfo:      buildInfo,
+			TelemetryCreateSettings: component.TelemetryCreateSettings{
+				Logger: logger.With(
+					zap.String(zapKindKey, zapKindExtension),
+					zap.String(zapNameKey, extID.String())),
+				TracerProvider: tracerProvider,
+			},
+			BuildInfo: buildInfo,
 		}
 		ext, err := buildExtension(context.Background(), factory, set, extCfg)
 		if err != nil {
