diff --git a/config/confighttp/confighttp.go b/config/confighttp/confighttp.go
index 66c417fc3..dd14a6ca5 100644
--- a/config/confighttp/confighttp.go
+++ b/config/confighttp/confighttp.go
@@ -271,6 +271,15 @@ func (hss *HTTPServerSettings) ToServer(host component.Host, settings component.
 		handler = maxRequestBodySizeInterceptor(handler, hss.MaxRequestBodySize)
 	}
 
+	if hss.Auth != nil {
+		authenticator, err := hss.Auth.GetServerAuthenticator(host.GetExtensions())
+		if err != nil {
+			return nil, err
+		}
+
+		handler = authInterceptor(handler, authenticator.Authenticate)
+	}
+
 	if hss.CORS != nil && len(hss.CORS.AllowedOrigins) > 0 {
 		co := cors.Options{
 			AllowedOrigins:   hss.CORS.AllowedOrigins,
@@ -282,15 +291,6 @@ func (hss *HTTPServerSettings) ToServer(host component.Host, settings component.
 	}
 	// TODO: emit a warning when non-empty CorsHeaders and empty CorsOrigins.
 
-	if hss.Auth != nil {
-		authenticator, err := hss.Auth.GetServerAuthenticator(host.GetExtensions())
-		if err != nil {
-			return nil, err
-		}
-
-		handler = authInterceptor(handler, authenticator.Authenticate)
-	}
-
 	// Enable OpenTelemetry observability plugin.
 	// TODO: Consider to use component ID string as prefix for all the operations.
 	handler = otelhttp.NewHandler(
