diff --git a/exporter/otlpexporter/config_test.go b/exporter/otlpexporter/config_test.go
index a8a6ec98a..c6e1751c2 100644
--- a/exporter/otlpexporter/config_test.go
+++ b/exporter/otlpexporter/config_test.go
@@ -29,7 +29,6 @@ import (
 	"go.opentelemetry.io/collector/config/configtest"
 	"go.opentelemetry.io/collector/config/configtls"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
-	"go.opentelemetry.io/collector/extension/bearertokenauthextension"
 )
 
 func TestLoadConfig(t *testing.T) {
@@ -39,10 +38,6 @@ func TestLoadConfig(t *testing.T) {
 	factory := NewFactory()
 	factories.Exporters[typeStr] = factory
 
-	authFactory := bearertokenauthextension.NewFactory()
-	authType := authFactory.Type()
-	factories.Extensions[authType] = authFactory
-
 	cfg, err := configtest.LoadConfigAndValidate(path.Join(".", "testdata", "config.yaml"), factories)
 
 	require.NoError(t, err)
@@ -90,7 +85,7 @@ func TestLoadConfig(t *testing.T) {
 				},
 				WriteBufferSize: 512 * 1024,
 				BalancerName:    "round_robin",
-				Auth:            &configauth.Authentication{AuthenticatorName: string(authType)},
+				Auth:            &configauth.Authentication{AuthenticatorName: "nop"},
 			},
 		})
 }
