diff --git a/service/go.mod b/service/go.mod
index 8544fc263..9dbe6596b 100644
--- a/service/go.mod
+++ b/service/go.mod
@@ -78,6 +78,7 @@ require (
 	github.com/tklauser/numcpus v0.6.1 // indirect
 	github.com/yusufpapurcu/wmi v1.2.4 // indirect
 	go.opentelemetry.io/collector/client v0.106.1 // indirect
+	go.opentelemetry.io/collector/component/componentprofiles v0.0.0-00010101000000-000000000000 // indirect
 	go.opentelemetry.io/collector/config/configauth v0.106.1 // indirect
 	go.opentelemetry.io/collector/config/configcompression v1.12.0 // indirect
 	go.opentelemetry.io/collector/config/configopaque v1.12.0 // indirect
@@ -161,4 +162,6 @@ replace go.opentelemetry.io/collector/consumer/consumerprofiles => ../consumer/c
 
 replace go.opentelemetry.io/collector/consumer/consumertest => ../consumer/consumertest
 
+replace go.opentelemetry.io/collector/component/componentprofiles => ../component/componentprofiles
+
 replace go.opentelemetry.io/collector/client => ../client
