diff --git a/confmap/expand_test.go b/confmap/expand_test.go
index 9cca35c70..25f544968 100644
--- a/confmap/expand_test.go
+++ b/confmap/expand_test.go
@@ -577,3 +577,46 @@ func TestResolverDefaultProviderExpand(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, map[string]any{"foo": "localhost"}, cfgMap.ToStringMap())
 }
+
+func Test_EscapedEnvVars(t *testing.T) {
+	const mapValue2 = "some map value"
+
+	expectedMap := map[string]any{
+		"test_map": map[string]any{
+			"recv.1":  "$MAP_VALUE_1",
+			"recv.2":  "$$MAP_VALUE_2",
+			"recv.3":  "$$MAP_VALUE_3",
+			"recv.4":  "$" + mapValue2,
+			"recv.5":  "some${MAP_VALUE_4}text",
+			"recv.6":  "${ONE}${TWO}",
+			"recv.7":  "text$",
+			"recv.8":  "$",
+			"recv.9":  "${1}${env:2}",
+			"recv.10": "some${env:MAP_VALUE_4}text",
+			"recv.11": "${env:" + mapValue2 + "}",
+			"recv.12": "${env:${MAP_VALUE_2}}",
+			"recv.13": "env:MAP_VALUE_2}${MAP_VALUE_2}{",
+			"recv.14": "${env:MAP_VALUE_2${MAP_VALUE_2}",
+			"recv.15": "$" + mapValue2,
+		}}
+
+	fileProvider := newFakeProvider("file", func(_ context.Context, uri string, _ WatcherFunc) (*Retrieved, error) {
+		return NewRetrieved(newConfFromFile(t, uri[5:]))
+	})
+	envProvider := newFakeProvider("env", func(_ context.Context, uri string, _ WatcherFunc) (*Retrieved, error) {
+		if uri == "env:MAP_VALUE_2" {
+			return NewRetrieved(mapValue2)
+		}
+		return nil, errors.New("should not be expanding any other env vars")
+	})
+
+	resolver, err := NewResolver(ResolverSettings{URIs: []string{filepath.Join("testdata", "expand-escaped-env.yaml")}, ProviderFactories: []ProviderFactory{fileProvider, envProvider}, ConverterFactories: nil, DefaultScheme: "env"})
+	require.NoError(t, err)
+
+	// Test that expanded configs are the same with the simple config with no env vars.
+	cfgMap, err := resolver.Resolve(context.Background())
+	require.NoError(t, err)
+	m := cfgMap.ToStringMap()
+	assert.Equal(t, expectedMap, m)
+
+}
