diff --git a/exporter/otlpexporter/go.mod b/exporter/otlpexporter/go.mod
index f7a4cd8ae..59cec3d2c 100644
--- a/exporter/otlpexporter/go.mod
+++ b/exporter/otlpexporter/go.mod
@@ -152,8 +152,6 @@ replace go.opentelemetry.io/collector/exporter/exportertest => ../exportertest
 
 replace go.opentelemetry.io/collector/extension/extensiontest => ../../extension/extensiontest
 
-replace go.opentelemetry.io/collector/extension/extensionauth/extensionauthtest => ../../extension/extensionauth/extensionauthtest
-
 replace go.opentelemetry.io/collector/featuregate => ../../featuregate
 
 replace go.opentelemetry.io/collector/extension/xextension => ../../extension/xextension
