diff --git a/service/telemetry/otelconftelemetry/logger_test.go b/service/telemetry/otelconftelemetry/logger_test.go
index b921b1245..92ba18e42 100644
--- a/service/telemetry/otelconftelemetry/logger_test.go
+++ b/service/telemetry/otelconftelemetry/logger_test.go
@@ -17,7 +17,9 @@ import (
 	config "go.opentelemetry.io/contrib/otelconf/v0.3.0"
 	"go.opentelemetry.io/otel/log"
 	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
+	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
@@ -31,7 +33,7 @@ const (
 	testValue     = "test-value"
 )
 
-func TestNewLogger(t *testing.T) {
+func TestCreateLogger(t *testing.T) {
 	tests := []struct {
 		name         string
 		wantCoreType any
@@ -43,6 +45,29 @@ func TestNewLogger(t *testing.T) {
 			cfg:     Config{},
 			wantErr: errors.New("no encoder name specified"),
 		},
+		{
+			name: "log config with invalid processors",
+			cfg: Config{
+				Logs: LogsConfig{
+					Level:             zapcore.DebugLevel,
+					Development:       true,
+					Encoding:          "console",
+					DisableCaller:     true,
+					DisableStacktrace: true,
+					InitialFields:     map[string]any{"fieldKey": "filed-value"},
+					Processors: []config.LogRecordProcessor{
+						{
+							Batch: &config.BatchLogRecordProcessor{
+								Exporter: config.LogRecordExporter{
+									OTLP: &config.OTLP{}, // missing required fields
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: errors.New("no valid log exporter"),
+		},
 		{
 			name: "log config with no processors",
 			cfg: Config{
@@ -103,20 +128,21 @@ func TestNewLogger(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			buildInfo := component.BuildInfo{}
-			factory := NewFactory()
 
-			providers, err := factory.CreateProviders(context.Background(), telemetry.Settings{BuildInfo: buildInfo}, &tt.cfg)
+			_, provider, err := createLogger(context.Background(), telemetry.Settings{
+				BuildInfo: buildInfo,
+			}, &tt.cfg)
 			if tt.wantErr != nil {
 				require.ErrorContains(t, err, tt.wantErr.Error())
 			} else {
 				require.NoError(t, err)
-				require.NoError(t, providers.Shutdown(context.Background()))
+				require.NoError(t, provider.Shutdown(context.Background()))
 			}
 		})
 	}
 }
 
-func TestNewLoggerWithResource(t *testing.T) {
+func TestCreateLoggerWithResource(t *testing.T) {
 	tests := []struct {
 		name           string
 		buildInfo      component.BuildInfo
@@ -195,8 +221,15 @@ func TestNewLoggerWithResource(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			set := telemetry.Settings{BuildInfo: tt.buildInfo}
-			cfg := Config{
+			core, observedLogs := observer.New(zapcore.DebugLevel)
+			set := telemetry.Settings{
+				BuildInfo: tt.buildInfo,
+				ZapOptions: []zap.Option{
+					// Redirect logs to the observer core
+					zap.WrapCore(func(zapcore.Core) zapcore.Core { return core }),
+				},
+			}
+			cfg := &Config{
 				Logs: LogsConfig{
 					Level:    zapcore.InfoLevel,
 					Encoding: "json",
@@ -204,9 +237,13 @@ func TestNewLoggerWithResource(t *testing.T) {
 				Resource: tt.resourceConfig,
 			}
 
-			tel, observedLogs := newTelemetryProviders(t, set, &cfg)
-			mylogger := tel.Logger()
-			mylogger.Info("Test log message")
+			logger, loggerProvider, err := createLogger(t.Context(), set, cfg)
+			require.NoError(t, err)
+			defer func() {
+				assert.NoError(t, loggerProvider.Shutdown(t.Context()))
+			}()
+
+			logger.Info("Test log message")
 			require.Len(t, observedLogs.All(), 1)
 
 			entry := observedLogs.All()[0]
@@ -266,6 +303,9 @@ func TestLoggerProvider(t *testing.T) {
 
 		receivedLogs++
 	})
+	defer func() {
+		assert.NoError(t, lp.Shutdown(t.Context()))
+	}()
 
 	// Generate some logs to send to the backend
 	logger := lp.Logger("name")
@@ -279,7 +319,7 @@ func TestLoggerProvider(t *testing.T) {
 	require.Equal(t, totalLogs, receivedLogs)
 }
 
-func newOTLPLoggerProvider(t *testing.T, level zapcore.Level, handler http.HandlerFunc) log.LoggerProvider {
+func newOTLPLoggerProvider(t *testing.T, level zapcore.Level, handler http.HandlerFunc) telemetry.LoggerProvider {
 	srv := createBackend("/v1/logs", handler)
 	t.Cleanup(srv.Close)
 
@@ -295,7 +335,7 @@ func newOTLPLoggerProvider(t *testing.T, level zapcore.Level, handler http.Handl
 		},
 	}}
 
-	cfg := Config{
+	cfg := &Config{
 		Logs: LogsConfig{
 			Level:      level,
 			Encoding:   "json",
@@ -308,10 +348,9 @@ func newOTLPLoggerProvider(t *testing.T, level zapcore.Level, handler http.Handl
 		},
 	}
 
-	tel, _ := newTelemetryProviders(t, telemetry.Settings{}, &cfg)
-	lp := tel.LoggerProvider()
-	require.NotNil(t, lp)
-	return lp
+	_, loggerProvider, err := createLogger(t.Context(), telemetry.Settings{}, cfg)
+	require.NoError(t, err)
+	return loggerProvider
 }
 
 func createBackend(endpoint string, handler func(http.ResponseWriter, *http.Request)) *httptest.Server {
