diff --git a/receiver/opencensusreceiver/config.go b/receiver/opencensusreceiver/config.go
index 8f01d2825..9d97eb3ec 100644
--- a/receiver/opencensusreceiver/config.go
+++ b/receiver/opencensusreceiver/config.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
