diff --git a/internal/goldendataset/resource_generator.go b/internal/goldendataset/resource_generator.go
index 39e84b840..ae379e114 100644
--- a/internal/goldendataset/resource_generator.go
+++ b/internal/goldendataset/resource_generator.go
@@ -91,8 +91,8 @@ func appendCloudK8sAttributes(attrMap pdata.AttributeMap) {
 	attrMap.UpsertString(conventions.AttributeK8SNamespaceName, "monitoring")
 	attrMap.UpsertString(conventions.AttributeK8SDeploymentName, "otel-collector")
 	attrMap.UpsertString(conventions.AttributeK8SDeploymentUID, "4D614B27-EDAF-409B-B631-6963D8F6FCD4")
-	attrMap.UpsertString(conventions.AttributeK8SReplicasetName, "otel-collector-2983fd34")
-	attrMap.UpsertString(conventions.AttributeK8SReplicasetUID, "EC7D59EF-D5B6-48B7-881E-DA6B7DD539B6")
+	attrMap.UpsertString(conventions.AttributeK8SReplicaSetName, "otel-collector-2983fd34")
+	attrMap.UpsertString(conventions.AttributeK8SReplicaSetUID, "EC7D59EF-D5B6-48B7-881E-DA6B7DD539B6")
 	attrMap.UpsertString(conventions.AttributeK8SPodName, "otel-collector-6484db5844-c6f9m")
 	attrMap.UpsertString(conventions.AttributeK8SPodUID, "FDFD941E-2A7A-4945-B601-88DD486161A4")
 	attrMap.UpsertString(conventions.AttributeHostID, "ec2e3fdaffa294348bdf355156b94cda")
