diff --git a/cmd/mdatagen/internal/loader_test.go b/cmd/mdatagen/internal/loader_test.go
index 4ea52e111..7ab082a8f 100644
--- a/cmd/mdatagen/internal/loader_test.go
+++ b/cmd/mdatagen/internal/loader_test.go
@@ -15,6 +15,10 @@ import (
 	"go.opentelemetry.io/collector/pdata/pmetric"
 )
 
+func boolPtr(b bool) *bool {
+	return &b
+}
+
 func TestTwoPackagesInDirectory(t *testing.T) {
 	contents, err := os.ReadFile("testdata/twopackages.yaml")
 	require.NoError(t, err)
@@ -70,7 +74,7 @@ func TestLoadMetadata(t *testing.T) {
 				ResourceAttributes: map[AttributeName]Attribute{
 					"string.resource.attr": {
 						Description: "Resource attribute with any string value.",
-						Enabled:     true,
+						EnabledPtr:  boolPtr(true),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeStr,
 						},
@@ -78,7 +82,7 @@ func TestLoadMetadata(t *testing.T) {
 					},
 					"string.enum.resource.attr": {
 						Description: "Resource attribute with a known set of string values.",
-						Enabled:     true,
+						EnabledPtr:  boolPtr(true),
 						Enum:        []string{"one", "two"},
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeStr,
@@ -87,7 +91,7 @@ func TestLoadMetadata(t *testing.T) {
 					},
 					"optional.resource.attr": {
 						Description: "Explicitly disabled ResourceAttribute.",
-						Enabled:     false,
+						EnabledPtr:  boolPtr(false),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeStr,
 						},
@@ -95,7 +99,7 @@ func TestLoadMetadata(t *testing.T) {
 					},
 					"slice.resource.attr": {
 						Description: "Resource attribute with a slice value.",
-						Enabled:     true,
+						EnabledPtr:  boolPtr(true),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeSlice,
 						},
@@ -103,7 +107,7 @@ func TestLoadMetadata(t *testing.T) {
 					},
 					"map.resource.attr": {
 						Description: "Resource attribute with a map value.",
-						Enabled:     true,
+						EnabledPtr:  boolPtr(true),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeMap,
 						},
@@ -114,7 +118,7 @@ func TestLoadMetadata(t *testing.T) {
 						Warnings: Warnings{
 							IfEnabledNotSet: "This resource_attribute will be disabled by default soon.",
 						},
-						Enabled: true,
+						EnabledPtr: boolPtr(true),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeStr,
 						},
@@ -125,7 +129,7 @@ func TestLoadMetadata(t *testing.T) {
 						Warnings: Warnings{
 							IfConfigured: "This resource_attribute is deprecated and will be removed soon.",
 						},
-						Enabled: false,
+						EnabledPtr: boolPtr(false),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeStr,
 						},
@@ -136,7 +140,7 @@ func TestLoadMetadata(t *testing.T) {
 						Warnings: Warnings{
 							IfEnabled: "This resource_attribute is deprecated and will be removed soon.",
 						},
-						Enabled: true,
+						EnabledPtr: boolPtr(true),
 						Type: ValueType{
 							ValueType: pcommon.ValueTypeStr,
 						},
