diff --git a/cmd/mdatagen/go.mod b/cmd/mdatagen/go.mod
index d7106c58b..ccc858ef1 100644
--- a/cmd/mdatagen/go.mod
+++ b/cmd/mdatagen/go.mod
@@ -114,7 +114,3 @@ replace go.opentelemetry.io/collector/component/componentstatus => ../../compone
 replace go.opentelemetry.io/collector/processor => ../../processor
 
 replace go.opentelemetry.io/collector/consumer/consumererror => ../../consumer/consumererror
-
-replace go.opentelemetry.io/collector/scraper => ../../scraper
-
-replace go.opentelemetry.io/collector/scraper/scraperhelper => ../../scraper/scraperhelper
