diff --git a/confmap/confmap.go b/confmap/confmap.go
index 3fac0c146..ec6669b21 100644
--- a/confmap/confmap.go
+++ b/confmap/confmap.go
@@ -171,6 +171,15 @@ func (l *Conf) Merge(in *Conf) error {
 	return l.k.Merge(in.k)
 }
 
+// mergeAppend merges the input given configuration into the existing config.
+// Note that the given map may be modified.
+// Additionally, mergeAppend performs deduplication when merging lists.
+// For example, if listA = [extension1, extension2] and listB = [extension1, extension3],
+// the resulting list will be [extension1, extension2, extension3].
+func (l *Conf) mergeAppend(in *Conf) error {
+	return l.k.Load(confmap.Provider(in.ToStringMap(), ""), nil, koanf.WithMergeFunc(mergeAppend))
+}
+
 // Sub returns new Conf instance representing a sub-config of this instance.
 // It returns an error is the sub-config is not a map[string]any (use Get()), and an empty Map if none exists.
 func (l *Conf) Sub(key string) (*Conf, error) {
