diff --git a/internal/data/testdata/trace.go b/internal/data/testdata/trace.go
index a8c5f2b34..fa6d9df6d 100644
--- a/internal/data/testdata/trace.go
+++ b/internal/data/testdata/trace.go
@@ -311,7 +311,7 @@ func generateOtlpSpanOne() *otlptrace.Span {
 		},
 		DroppedEventsCount: 1,
 		Status: &otlptrace.Status{
-			Code:    otlptrace.Status_Cancelled,
+			Code:    otlptrace.Status_STATUS_CODE_CANCELLED,
 			Message: "status-cancelled",
 		},
 	}
