diff --git a/processor/samplingprocessor/tailsamplingprocessor/sampling/rate_limiting.go b/processor/samplingprocessor/tailsamplingprocessor/sampling/rate_limiting.go
index 68887712f..8cd619a48 100644
--- a/processor/samplingprocessor/tailsamplingprocessor/sampling/rate_limiting.go
+++ b/processor/samplingprocessor/tailsamplingprocessor/sampling/rate_limiting.go
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright The OpenTelemetry Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
